diff -rupN ndiswrapper-1.59/driver/ndis.c ndiswrapper-1.59-patched/driver/ndis.c
--- ndiswrapper-1.59/driver/ndis.c	2013-11-28 20:42:36.000000000 +0100
+++ ndiswrapper-1.59-patched/driver/ndis.c	2015-03-12 16:21:18.249996689 +0100
@@ -438,7 +438,7 @@ static int read_setting(struct nt_list *
 	struct wrap_device_setting *setting;
 	mutex_lock(&loader_mutex);
 	nt_list_for_each_entry(setting, setting_list, list) {
-		if (strnicmp(keyname, setting->name, length) == 0) {
+		if (strncasecmp(keyname, setting->name, length) == 0) {
 			TRACE2("setting %s='%s'", keyname, setting->value);
 			mutex_unlock(&loader_mutex);
 			*param = ndis_encode_setting(setting, type);
@@ -502,7 +502,7 @@ wstdcall void WIN_FUNC(NdisWriteConfigur
 
 	mutex_lock(&loader_mutex);
 	nt_list_for_each_entry(setting, &nmb->wnd->wd->settings, list) {
-		if (strnicmp(keyname, setting->name, ansi.length) == 0) {
+		if (strncasecmp(keyname, setting->name, ansi.length) == 0) {
 			mutex_unlock(&loader_mutex);
 			if (ndis_decode_setting(setting, param))
 				*status = NDIS_STATUS_FAILURE;

