diff -Naur xine-ui-r3058/misc/Makefile.in xine-ui-r3058a/misc/Makefile.in
--- xine-ui-r3058/misc/Makefile.in	2010-10-03 18:08:26.000000000 +0200
+++ xine-ui-r3058a/misc/Makefile.in	2010-10-04 15:49:22.074629965 +0200
@@ -310,7 +310,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = desktops visuals
+SUBDIRS = visuals
 
 # splash-foo/manifest contains names of extra source files, distributed only.
 MANIFEST := $(shell test ! -f splash-@SOURCE_SPLASH@/manifest || cat splash-@SOURCE_SPLASH@/manifest)
