diff -Naur xine-ui-cvs+20090315/src/xitk/Imlib-light/load.c xine-ui-cvs+20090315a/src/xitk/Imlib-light/load.c
--- xine-ui-cvs+20090315/src/xitk/Imlib-light/load.c	2009-01-01 15:21:19.000000000 +0100
+++ xine-ui-cvs+20090315a/src/xitk/Imlib-light/load.c	2010-07-12 17:12:58.393996094 +0200
@@ -198,7 +198,7 @@
     return 0;
   fread(buf, 1, 8, f);
   rewind(f);
-  return (int)png_check_sig(buf, 8);
+  return (int)(png_sig_cmp(buf, 0, 8) == 0);
 }
 
 ImlibImage * Imlib_load_image(ImlibData * id, char *file) {
