diff -Naur xine-ui-r3058/configure xine-ui-r3058a/configure
--- xine-ui-r3058/configure	2010-10-03 18:08:27.000000000 +0200
+++ xine-ui-r3058a/configure	2010-11-21 15:17:57.314166074 +0100
@@ -13404,7 +13404,7 @@
 SYSCONF_DIR=`makeexpand "$sysconfdir"`
 
 
-INCLUDES='-I$(top_srcdir) -I$(top_builddir) -I$(top_srcdir)/src -I$(top_builddir)/src -I$(top_srcdir)/src/common -I$(top_builddir)/src/common -I$(top_srcdir)/src/xitk/xine-toolkit -I$(top_builddir)/src/xitk/xine-toolkit -I$(prefix)/include $(INTLDIR)'
+INCLUDES='-I$(top_srcdir) -I$(top_builddir) -I$(top_srcdir)/src -I$(top_builddir)/src -I$(top_srcdir)/src/common -I$(top_builddir)/src/common -I$(top_srcdir)/src/xitk/xine-toolkit -I$(top_builddir)/src/xitk/xine-toolkit $(INTLDIR)'
 
 
 
