From 312f8999caaf02dfcda2d51fcd6f26fbb80389d7 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sun, 5 Nov 2017 13:04:39 +0100
Subject: [PATCH 10/13] windowing/gbm: register DRM PRIME decoder and renderer

---
 xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp b/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
index cdd99da9d5..e7ef372187 100644
--- a/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
+++ b/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
@@ -24,6 +24,9 @@
 #include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGLES.h"
 #endif
 
+#include "cores/VideoPlayer/DVDCodecs/Video/DRMPRIME.h"
+#include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.h"
+
 #include "cores/RetroPlayer/process/RPProcessInfo.h"
 #include "cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.h"
 #include "cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.h"
@@ -52,6 +55,9 @@ bool CWinSystemGbmGLESContext::InitWindowSystem()
     return false;
   }
 
+  CRendererDRMPRIME::Register();
+  DRMPRIME::CDecoder::Register();
+
 #if defined (HAVE_LIBVA)
   VADisplay vaDpy = static_cast<VADisplay>(CWinSystemGbm::GetVaDisplay());
   bool general, hevc;
-- 
2.11.0

