--- a/configure	2014-11-08 16:54:58.827651509 +0100
+++ b/configure	2014-11-08 16:54:53.127651678 +0100
@@ -3702,16 +3702,17 @@
             subarch=$(echo $cpu | sed 's/[^a-z0-9]//g')
         ;;
         *)
-            cpuflags="-mcpu=$cpu"
             case $cpu in
-                cortex-a*)                               subarch=armv7a  ;;
-                cortex-r*)                               subarch=armv7r  ;;
-                cortex-m*)                 enable thumb; subarch=armv7m  ;;
+                cortex-a*)                               subarch=armv7-a ;;
+                cortex-r*)                               subarch=armv7-r ;;
+                cortex-m*)                 enable thumb; subarch=armv7-m ;;
                 arm11*)                                  subarch=armv6   ;;
                 arm[79]*e*|arm9[24]6*|arm96*|arm102[26]) subarch=armv5te ;;
-                armv4*|arm7*|arm9[24]*)                  subarch=armv4   ;;
+                arm4*|arm7*|arm9[24]*)                   subarch=armv4   ;;
                 *)                             subarch=$(probe_arm_arch) ;;
             esac
+            cpuflags="-march=$subarch -mtune=$cpu"
+            subarch=$(echo $subarch | sed 's/-//g')
         ;;
     esac
