--- xf86-video-nvidia-legacy-173.14.39/./usr/src/nv/nv-linux.h.org	2014-04-01 07:01:36.348238027 +0200
+++ xf86-video-nvidia-legacy-173.14.39/./usr/src/nv/nv-linux.h	2014-04-01 07:01:40.448238098 +0200
@@ -208,6 +208,11 @@
 #if defined(CONFIG_ACPI) 
 #include <acpi/acpi.h>
 #include <acpi/acpi_drivers.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
+#include <acpi/acpi_bus.h>
+#endif
+
 #if defined(NV_ACPI_DEVICE_OPS_HAS_MATCH) || defined(ACPI_VIDEO_HID)
 #define NV_LINUX_ACPI_EVENTS_SUPPORTED 1
 #endif
