--- play/play.cpp	2013-09-05 12:03:09.000000000 +0200
+++ play/play.cpp.new	2015-02-11 17:11:25.702068936 +0100
@@ -1352,7 +1352,7 @@
     }
 
     LOCK_PIXMAPS;
-    while ((pm = RenderPixmaps())) {
+    while (cPixmapMemory *pm = dynamic_cast<cPixmapMemory *>(RenderPixmaps())) {
 	int x;
 	int y;
 	int w;
@@ -1367,7 +1367,7 @@
 	    pm->Data());
 	OsdDrawARGB(x, y, w, h, pm->Data());
 
-	delete pm;
+	DestroyPixmap(pm);
     }
     cMyOsd::Dirty = 0;
 }

