diff -Naur qt-mobility-opensource-src-1.1.0-beta2.orig//configure qt-mobility-opensource-src-1.1.0-beta2/configure
--- qt-mobility-opensource-src-1.1.0-beta2.orig//configure	2010-11-07 16:29:11.752427999 +0100
+++ qt-mobility-opensource-src-1.1.0-beta2/configure	2010-11-07 16:29:32.012428009 +0100
@@ -532,8 +537,9 @@
     "$MAKE" clean >> "$CONFIG_LOG"
     printf "."
     "$MAKE" >> "$CONFIG_LOG" 2>&1
+    RES=$?
     printf ". "
-    if [ -e ./$2 ]; then
+    if [ $RES -eq 0 ]; then
         echo "OK"
         echo "$2_enabled = yes" >> "$CONFIG_IN"
     else
