Index: btfm/kfmapp/makefile
===================================================================
--- btfm.orig/kfmapp/makefile	2010-12-06 19:11:05.000000000 +0100
+++ btfm/kfmapp/makefile	2010-12-06 19:11:44.000000000 +0100
@@ -10,7 +10,7 @@
 all: kfmapp
 
 kfmapp: kfmapp.c
-	$(CC) $^ -o $@ $(LIB_DIR) -lpthread -lm -lasound
+	$(CC) $(CFLAGS) $^ -o $@ $(LIB_DIR) -lpthread -lm -lasound
 
 .PHONY: clean
 clean:
Index: btfm/uim/makefile
===================================================================
--- btfm.orig/uim/makefile	2010-12-06 19:11:04.000000000 +0100
+++ btfm/uim/makefile	2010-12-06 19:12:00.000000000 +0100
@@ -8,7 +8,7 @@
 all: uim
 
 uim: uim.c
-	$(CC) $^ -o $@
+	$(CC) $(CFLAGS) $^ -o $@
 
 .PHONY: clean
 clean:
