--- mono-3.8.0/mcs/build/profiles/basic.make	2014-08-28 17:22:10.000000000 +0100
+++ mono-3.8.0.new/mcs/build/profiles/basic.make	2014-09-10 14:51:23.689662419 +0100
@@ -12,7 +12,7 @@
 PROFILE_RUNTIME = $(with_mono_path_monolite) $(RUNTIME)
 BOOTSTRAP_MCS = $(PROFILE_RUNTIME) $(RUNTIME_FLAGS) $(MONOLITE_MCS) -sdk:2
 else
-PROFILE_RUNTIME = $(EXTERNAL_RUNTIME)
+PROFILE_RUNTIME = $(with_mono_path) $(EXTERNAL_RUNTIME)
 BOOTSTRAP_MCS = $(EXTERNAL_MCS)
 endif
 
