--- vdr-mcli-188/device.c	2011-08-22 14:38:29.000000000 +0200
+++ vdr-mcli-188/device.c.org	2013-05-03 17:12:48.288671147 +0200
@@ -474,7 +474,7 @@
 	} else {
 		m_disabletimeout = TEMP_DISABLE_TIMEOUT_DEFAULT;
 	}
-	bool cam_force=!EITScanner.UsesDevice(this);
+	bool cam_force=true;
 	if(cam_force && !CheckCAM(Channel, true)) {
 #ifdef DEBUG_TUNE
 		printf("No CAM on %d available even after tried to steal one\n", CardIndex () + 1);
