--- vdr-mcli-188/Makefile.org	2013-12-18 15:44:09.907336770 +0100
+++ vdr-mcli-188/Makefile	2013-12-18 15:44:27.971336231 +0100
@@ -62,13 +62,13 @@
 
 ### Includes and Defines (add further entries here):
 
-ifdef RBMINI
+#ifdef RBMINI
   XML_INC := -I/usr/arm-linux-gnueabi/include/libxml2
   XML_LIB := -lxml2
-else
-  XML_INC := `xml2-config --cflags`
-  XML_LIB := `xml2-config --libs`
-endif
+#else
+#  XML_INC := `xml2-config --cflags`
+#  XML_LIB := `xml2-config --libs`
+#endif
 
 
 ifdef MCLI_SHARED
