diff -ruN sleeptimer-0.8-git20110722/i18n.c sleeptimer-0.8git-20110722/i18n.c
--- sleeptimer-0.8-git20110722/i18n.c	2011-07-24 19:09:44.000000000 +0200
+++ sleeptimer-0.8git-20110722/i18n.c	2012-03-26 19:08:49.755045896 +0200
@@ -9,8 +9,7 @@
 #include "i18n.h"
 
 #if VDRVERSNUM < 10507
-
-const tI18nPhrase Phrases[] = {
+//const tI18nPhrase Phrases[] = {
   { "Not active",// English
     "Nicht aktiv", //Deutsch
     "",// Slovenski
diff -ruN sleeptimer-0.8-git20110722/i18n.h sleeptimer-0.8git-20110722/i18n.h
--- sleeptimer-0.8-git20110722/i18n.h	2011-07-24 19:09:44.000000000 +0200
+++ sleeptimer-0.8git-20110722/i18n.h	2012-03-26 19:09:10.286047528 +0200
@@ -11,6 +11,6 @@
 
 #include <vdr/i18n.h>
 
-extern const tI18nPhrase Phrases[];
+//extern const tI18nPhrase Phrases[];
 
 #endif //_I18N__H
diff -ruN sleeptimer-0.8-git20110722/Makefile sleeptimer-0.8git-20110722/Makefile
--- sleeptimer-0.8-git20110722/Makefile	2012-03-26 19:23:16.819114782 +0200
+++ sleeptimer-0.8git-20110722/Makefile	2012-03-26 19:23:01.338113552 +0200
@@ -20,7 +20,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Werror=overloaded-virtual
 
 ### The directory environment:
 
@@ -50,7 +50,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o i18n.o
+OBJS = $(PLUGIN).o
 
 ### Implicit rules:
 
