--- libnsbmp-0.0.3/build/makefiles/Makefile.tools.org	2013-03-26 12:40:00.250592303 +0100
+++ libnsbmp-0.0.3/build/makefiles/Makefile.tools	2013-03-26 12:40:11.218592490 +0100
@@ -225,7 +225,7 @@
     PKGCONFIG ?=
   else
     # Cross compiling for Windows -- assumes mingw toolchain
-    MINGW_INSTALL_ENV ?= /usr/local/mingw
+    MINGW_INSTALL_ENV ?= /usr/mingw
 
     CC__ := i586-mingw32msvc-gcc
     CXX__ := i586-mingw32msvc-g++
@@ -275,7 +275,7 @@
 endif
 
 # Default prefix
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 
 ###############################################################################
 # Tool defaults
