Index: gstreamer0.10-ffmpeg-0.10.10.2/ext/ffmpeg/gstffmpeg.c
===================================================================
--- gstreamer0.10-ffmpeg-0.10.10.2.orig/ext/ffmpeg/gstffmpeg.c	2010-05-08 13:47:43.000000000 +0200
+++ gstreamer0.10-ffmpeg-0.10.10.2/ext/ffmpeg/gstffmpeg.c	2010-06-27 11:36:08.937172000 +0200
@@ -138,6 +138,14 @@
 
   av_register_all ();
 
+  gst_plugin_add_dependency_simple (plugin, NULL,
+      "/usr/lib:/usr/local/lib",
+      "libavcodec.so.52,"
+      "libavformat.so.52,"
+      "libpostproc.so.51,"
+      "libswscale.so.0",
+      GST_PLUGIN_DEPENDENCY_FLAG_NONE);
+
   gst_ffmpegenc_register (plugin);
   gst_ffmpegdec_register (plugin);
   gst_ffmpegdemux_register (plugin);
