From fadeffdd10f1836c23cdd97d8a93011d913065a3 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Mon, 10 Jun 2019 19:06:15 +0200
Subject: [PATCH] Avoid C99 syntax

---
 edid-decode.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/edid-decode.c b/edid-decode.c
index 3507941..608520a 100644
--- a/edid-decode.c
+++ b/edid-decode.c
@@ -1477,9 +1477,10 @@ static void cta_y420vdb(const unsigned char *x, unsigned int length)
 static void cta_y420cmdb(const unsigned char *x, unsigned int length)
 {
     printf("    Video modes supporting YCbCr 4:2:0 in addition to other color formats:\n");
-    int index = 0;
-	for (unsigned int i = 0; i < length; i++) {
-        for (int bit = 0x01; bit <= 0x80; bit <<= 1) {
+    int bit, index = 0;
+    unsigned int i;
+    for (i = 0; i < length; i++) {
+        for (bit = 0x01; bit <= 0x80; bit <<= 1) {
             if (x[i] & bit && index < victable_size) {
                 unsigned char vic = victable[index];
                 printf("      VIC %3d %s\n", vic, edid_cta_modes1[vic].name);
@@ -1850,7 +1851,8 @@ static void cta_hf_block(const unsigned char *x, unsigned int length)
 static void cta_vs_data_block(const unsigned char* x, unsigned int length) {
     if (length > 0) {
         printf("    Bytes ");
-        for (unsigned int i = 0; i < length; i++)
+        unsigned int i;
+        for (i = 0; i < length; i++)
             printf("%02x", x[i]);
         printf("\n");
     }
-- 
2.7.4

