--- dos2unix-6.0.3/Makefile.org	2013-11-29 23:44:04.536852195 +0100
+++ dos2unix-6.0.3/Makefile	2013-11-29 23:44:25.680851779 +0100
@@ -36,10 +36,10 @@
 include version.mk
 
 
-CC		= gcc
-CPP		= cpp
-CPP_FLAGS_POD	= ALL
-STRIP		= strip
+#CC		= gcc
+#CPP		= cpp
+#CPP_FLAGS_POD	= ALL
+#STRIP		= strip
 
 PACKAGE		= dos2unix
 UNIX2DOS	= unix2dos
--- dos2unix-6.0.3/Makefile.org	2013-12-10 15:51:15.829322327 +0100
+++ dos2unix-6.0.3/Makefile	2013-12-10 15:51:24.001322465 +0100
@@ -47,7 +47,7 @@
 UNIX2MAC	= unix2mac
 
 # Native Language Support (NLS)
-ENABLE_NLS	= 1
+#ENABLE_NLS	= 1
 # Large File Support (LFS)
 LFS             = 1
 DEBUG = 0
