diff -Naur linux-2.6.32-tegra-10.8.2/init/main.c linux-2.6.32-tegra-10.8.2a//init/main.c
--- linux-2.6.32-tegra-10.8.2/init/main.c	2010-09-04 10:07:59.000000000 +0200
+++ linux-2.6.32-tegra-10.8.2a//init/main.c	2010-09-04 10:51:49.493153760 +0200
@@ -811,8 +811,13 @@
 	system_state = SYSTEM_RUNNING;
 	numa_default_policy();
 
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0) {
+		sys_mknod("/dev/console", S_IFCHR|0600, (TTYAUX_MAJOR<<8)|1);
+		if (sys_open("/dev/console", O_RDWR, 0) < 0)
+			printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+		sys_unlink("/dev/console");
+	}
+
 
 	(void) sys_dup(0);
 	(void) sys_dup(0);
