diff -Naur linux-2.6.32-tegra-10.8.2/include/linux/Kbuild linux-2.6.32-tegra-10.8.2a/include/linux/Kbuild
--- linux-2.6.32-tegra-10.8.2/include/linux/Kbuild	2010-09-04 10:07:59.000000000 +0200
+++ linux-2.6.32-tegra-10.8.2a/include/linux/Kbuild	2010-09-04 13:50:15.506156893 +0200
@@ -239,6 +239,8 @@
 unifdef-y += if_phonet.h
 unifdef-y += if_pppol2tp.h
 unifdef-y += if_pppox.h
+unifdef-y += if_pppolac.h
+unifdef-y += if_pppopns.h
 unifdef-y += if_tr.h
 unifdef-y += if_tunnel.h
 unifdef-y += if_vlan.h
