--- snowball-init-1.4.1-0+b064be21de/Makefile.orig	2012-05-16 10:27:35.036721382 +0300
+++ snowball-init-1.4.1-0+b064be21de/Makefile	2012-05-16 10:28:40.749050139 +0300
@@ -3,9 +3,6 @@
 
 install:
 	# install the traditional-style init script to init.d/
-	install -d ${DESTDIR}/etc/init.d
-	install -m 0775 snowball ${DESTDIR}/etc/init.d/snowball
+	install -d ${DESTDIR}/usr/local/sbin
+	install -m 0775 snowball ${DESTDIR}/usr/local/sbin/snowball
 	
-	# install the upstart-style task to init/
-	install -d ${DESTDIR}/etc/init
-	install -m 0664 snowball.conf ${DESTDIR}/etc/init/snowball.conf
