--- a/Makefile.am	2017-08-02 14:29:47.660465453 +0200
+++ b/Makefile.am	2017-08-02 14:29:47.672465398 +0200
@@ -34,6 +34,11 @@
 	cp ./extra/wav/bird-calls.wav $(INSTALL_PREFIX)/inadyn-mt/extra/wav
 
 	cp ./extra/servers_additional.cfg $(INSTALL_PREFIX)/inadyn-mt/extra
+else
+
+	mkdir $(INSTALL_PREFIX)/inadyn-mt/extra
+
+	cp ./extra/servers_additional.cfg $(INSTALL_PREFIX)/inadyn-mt/extra
 endif
 
 uninstall-hook:
