Description: Fixes problems with text2skin skin enigma.
Author: Andreas Brugger <brougs78@gmx.net>
Origin: http://vdrportal.de/board/thread.php?postid=343665#post343665

Index: vdr/osdbase.c
===================================================================
--- vdr.orig/osdbase.c	2011-12-10 22:22:04.000000000 +0100
+++ vdr/osdbase.c	2011-12-11 01:48:58.000000000 +0100
@@ -297,6 +297,7 @@
 
 void cOsdMenu::CursorUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -335,6 +336,7 @@
 
 void cOsdMenu::CursorDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -375,6 +377,7 @@
 
 void cOsdMenu::PageUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current -= displayMenuItems;
@@ -409,6 +412,7 @@
 
 void cOsdMenu::PageDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current += displayMenuItems;
