Description: Use ncursesw header file
Author: Tobias Grimm <tg@e-tobi.net>

Index: vdr/PLUGINS/src/skincurses/skincurses.c
===================================================================
--- vdr.orig/PLUGINS/src/skincurses/skincurses.c	2011-12-10 22:22:04.000000000 +0100
+++ vdr/PLUGINS/src/skincurses/skincurses.c	2011-12-11 01:49:11.000000000 +0100
@@ -6,7 +6,7 @@
  * $Id: skincurses.c 2.7 2011/08/21 11:04:38 kls Exp $
  */
 
-#include <ncurses.h>
+#include <ncursesw/ncurses.h>
 #include <vdr/osd.h>
 #include <vdr/plugin.h>
 #include <vdr/skins.h>
