Description: Patch for VDR
Author: Bexbier <Bexbier@vdr-portal> 
Origin: http://www.vdr-portal.de/board16-video-disk-recorder/board8-vdr-grundlagen/p964596-bug-pause-100-cpu-auslastung/#post964596

diff -urNad vdr-1.7.22-orig/menu.c vdr-1.7.22-hdtimeshift-wait/menu.c
--- vdr-1.7.22-orig/menu.c	2011-12-22 15:38:36.000000000 +0100
+++ vdr-1.7.22-hdtimeshift-wait/menu.c	2011-12-22 15:39:51.008120051 +0100
@@ -4883,11 +4883,11 @@
   Skins.Message(mtStatus, tr("Pausing live video..."));
   cReplayControl::SetRecording(NULL, NULL); // make sure the new cRecordControl will set cReplayControl::LastReplayed()
   if (Start(NULL, true)) {
-     cCondWait::SleepMs(2000); // allow recorded file to fill up enough to start replaying
+     cCondWait::SleepMs(5000); // allow recorded file to fill up enough to start replaying
      cReplayControl *rc = new cReplayControl;
      cControl::Launch(rc);
      cControl::Attach();
-     cCondWait::SleepMs(1000); // allow device to replay some frames, so we have a picture
+     cCondWait::SleepMs(2000); // allow device to replay some frames, so we have a picture
      Skins.Message(mtStatus, NULL);
      rc->ProcessKey(kPause); // pause, allowing replay mode display
      return true;
