--- vdr-1.7.22.orig/remux.c
+++ vdr-1.7.22/remux.c
@@ -790,6 +790,11 @@ const uchar *cTsToPes::GetPes(int &Lengt
         uchar *p = data + offset - 6;
         if (p != data) {
            p -= 3;
+           if (p < data) {
+               esyslog("ERROR: pointer before data offset=%d", offset);
+              Reset();
+              return NULL;
+              }
            memmove(p, data, 4);
            }
         int l = min(length - offset, MAXPESLENGTH);
