Description: This patch is needed for the yaepg plugin.
Author: bball950@yahoo.com
Origin: http://www.hoochvdr.info/yaepg/vdr-1.3.19-core-yaepg.diff

Index: vdr/PLUGINS/src/dvbsddevice/dvbsdffosd.c
===================================================================
--- vdr.orig/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2011-12-10 21:50:59.000000000 +0100
+++ vdr/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2011-12-11 02:29:56.000000000 +0100
@@ -77,6 +77,10 @@
             Cmd(OSD_SetWindow, 0, i + 1);
             Cmd(OSD_Close);
             }
+        if (vidWin.bpp != 0) {
+           Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+           Cmd(OSD_Close);
+           }
         shown = false;
         }
      }
@@ -111,6 +115,10 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_Close);
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_Close);
+        }
      shown = false;
      }
   return cOsd::SetAreas(Areas, NumAreas);
@@ -194,6 +202,11 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_MoveWindow, 0, Left() + Bitmap->X0(), Top() + Bitmap->Y0());
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_OpenRaw, vidWin.bpp, vidWin.x1, vidWin.y1,
+            vidWin.x2, vidWin.y2, (void *)0);
+        }
      shown = true;
      }
 }
Index: vdr/device.h
===================================================================
--- vdr.orig/device.h	2011-12-11 02:29:53.000000000 +0100
+++ vdr/device.h	2011-12-11 02:29:56.000000000 +0100
@@ -293,12 +293,12 @@
          ///< Direction (only the sign of Direction is evaluated, positive values
          ///< switch to higher channel numbers).
 private:
-  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
-         ///< Sets the device to the given channel (general setup).
 protected:
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
          ///< Sets the device to the given channel (actual physical setup).
 public:
+  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
+         ///< Sets the device to the given channel (general setup).
   static int CurrentChannel(void) { return primaryDevice ? currentChannel : 0; }
          ///< Returns the number of the current channel on the primary device.
   static void SetCurrentChannel(const cChannel *Channel) { currentChannel = Channel ? Channel->Number() : 0; }
Index: vdr/osd.c
===================================================================
--- vdr.orig/osd.c	2011-12-10 21:50:59.000000000 +0100
+++ vdr/osd.c	2011-12-11 02:29:56.000000000 +0100
@@ -1622,6 +1622,7 @@
          }
       }
   Osds.Append(this);
+  vidWin.bpp = 0;
 }
 
 cOsd::~cOsd()
Index: vdr/osd.h
===================================================================
--- vdr.orig/osd.h	2011-12-10 21:59:26.000000000 +0100
+++ vdr/osd.h	2011-12-11 02:29:56.000000000 +0100
@@ -921,6 +921,7 @@
        ///<        MyOsdDrawPixmap(Left() + pm->ViewPort().X(), Top() + pm->ViewPort().Y(), pm->Data(), w, h, h * d);
        ///<        delete pm;
        ///<        }
+  tArea vidWin;
   };
 
 #define MAXOSDIMAGES 64
