diff --git a/displaychannel.c b/displaychannel.c
index 0d8208a..23890ae 100644
--- a/displaychannel.c
+++ b/displaychannel.c
@@ -806,9 +806,9 @@ void cNopacityDisplayChannel::Action(void) {
             pixmapSignalMeter->SetAlpha(Alpha);
             pixmapSignalLabel->SetAlpha(Alpha);
         }
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/displaymenu.c b/displaymenu.c
index 6add8b8..c77d9b4 100644
--- a/displaymenu.c
+++ b/displaymenu.c
@@ -765,9 +765,9 @@ void cNopacityDisplayMenu::Action(void) {
         }
         for (cNopacityTimer *t = timers.First(); Running() && t; t = timers.Next(t))
             t->SetAlpha(Alpha);
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/displaymessage.c b/displaymessage.c
index cc831a3..474f137 100644
--- a/displaymessage.c
+++ b/displaymessage.c
@@ -70,9 +70,9 @@ void cNopacityDisplayMessage::Action(void) {
         int Alpha = t * ALPHA_OPAQUE;
         pixmapBackground->SetAlpha(Alpha);
         pixmap->SetAlpha(Alpha);
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/displayreplay.c b/displayreplay.c
index 0f3c845..fe9570b 100644
--- a/displayreplay.c
+++ b/displayreplay.c
@@ -371,9 +371,9 @@ void cNopacityDisplayReplay::Action(void) {
         pixmapPlay->SetAlpha(Alpha);
         pixmapFwd->SetAlpha(Alpha);
         pixmapFwdSpeed->SetAlpha(Alpha);
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/displaytracks.c b/displaytracks.c
index bbcc8ce..34ac1a1 100644
--- a/displaytracks.c
+++ b/displaytracks.c
@@ -197,9 +197,9 @@ void cNopacityDisplayTracks::Action(void) {
         for (cNopacityMenuItem *item = menuItems.First(); Running() && item; item = menuItems.Next(item)) {
             item->SetAlpha(Alpha);
         }
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/displayvolume.c b/displayvolume.c
index 5bbca51..48b3a6e 100644
--- a/displayvolume.c
+++ b/displayvolume.c
@@ -141,9 +141,9 @@ void cNopacityDisplayVolume::Action(void) {
         pixmapBackgroundBottom->SetAlpha(Alpha);
         pixmapProgressBar->SetAlpha(Alpha);
         pixmapLabel->SetAlpha(Alpha);
+        cPixmap::Unlock();
         if (Running())
             osd->Flush();
-        cPixmap::Unlock();
         int Delta = cTimeMs::Now() - Now;
         if (Running() && (Delta < FrameTime))
             cCondWait::SleepMs(FrameTime - Delta);
diff --git a/textwindow.c b/textwindow.c
index 6e55ef9..b83110b 100644
--- a/textwindow.c
+++ b/textwindow.c
@@ -115,9 +115,9 @@ void cNopacityTextWindow::Action(void) {
             int Alpha = t * ALPHA_OPAQUE;
             pixmapBackground->SetAlpha(Alpha);
             pixmap->SetAlpha(Alpha);
+            cPixmap::Unlock();
             if (Running())
                 osd->Flush();
-            cPixmap::Unlock();
             int Delta = cTimeMs::Now() - Now;
             if (Running() && (Delta < FadeFrameTime))
                 cCondWait::SleepMs(FadeFrameTime - Delta);
