--- a/configure.in	2013-07-21 14:39:43.000000000 +0200
+++ b/configure.in	2013-07-21 14:43:16.000000000 +0200
@@ -671,8 +671,8 @@
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_RASPBERRY_PI")
      AC_DEFINE(HAS_EGLGLES, [1], [Define if supporting EGL based GLES Framebuffer])
      USE_OMXLIB=1; AC_DEFINE([HAVE_OMXLIB],[1],["Define to 1 if OMX libs is enabled"])
-     CFLAGS="$CFLAGS"
-     CXXFLAGS="$CXXFLAGS"
+     CFLAGS="$CFLAGS -Wno-psabi"
+     CXXFLAGS="$CXXFLAGS -Wno-psabi"
      ffmpeg_target_os=linux
      ;;
 esac
@@ -761,8 +761,8 @@
   ARCH="arm"
   use_arch="arm"
 elif test "$use_arch" = "arm"; then
-  CFLAGS="$CFLAGS -mno-apcs-stack-check"
-  CXXFLAGS="$CXXFLAGS -mno-apcs-stack-check"
+  CFLAGS="$CFLAGS -mno-apcs-stack-check -Wno-psabi"
+  CXXFLAGS="$CXXFLAGS -mno-apcs-stack-check -Wno-psabi"
   FFMPEG_EXTRACFLAGS=""
   if test "$use_tegra" = "yes"; then
     # Compile for ARMv7a architecture, need to test gcc for vfpv3-d16 support 
@@ -770,8 +770,8 @@
     CFLAGS="-mfpu=vfpv3-d16"
     AC_COMPILE_IFELSE(
       [AC_LANG_SOURCE([int foo;])],
-      [ CFLAGS="$SAVE_CFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb-interwork"
-        CXXFLAGS="$CXXFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb-interwork"
+      [ CFLAGS="$SAVE_CFLAGS -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb-interwork"
+        CXXFLAGS="$CXXFLAGS -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb-interwork"
         FFMPEG_EXTRACFLAGS="$FFMPEG_EXTRACFLAGS -mtune=cortex-a9 -mfpu=vfpv3-d16"
         use_cpu=cortex-a9],
       [ CFLAGS="$SAVE_CFLAGS -Wa,-march=armv6 -mtune=cortex-a8 -mthumb-interwork"
