--- a/configure.in	2013-07-21 14:43:16.000000000 +0200
+++ b/configure.in	2013-07-21 16:54:15.000000000 +0200
@@ -628,7 +628,9 @@
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_POWERPC64")
      ;;
   arm*-*-linux-gnu*)
-     use_texturepacker=no
+     if test "x$use_texturepacker" = "xauto" ; then
+       use_texturepacker=no
+     fi
      ARCH="arm"
      use_arch="arm"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
