diff --git a/Makefile.in b/Makefile.in
index 9d850e4..8d5ec93 100755
--- a/Makefile.in
+++ b/Makefile.in
@@ -18,8 +18,12 @@ DVDPLAYER_ARCHIVES=xbmc/cores/dvdplayer/DVDPlayer.a \
                    xbmc/cores/dvdplayer/DVDDemuxers/DVDDemuxers.a \
                    xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreams.a \
                    xbmc/cores/dvdplayer/DVDSubtitles/DVDSubtitles.a
+		   
+		   
+GSTPLAYER_ARCHIVES=xbmc/cores/gstplayer/GstPlayer.a
 
 DIRECTORY_ARCHIVES=$(DVDPLAYER_ARCHIVES) \
+                   $(GSTPLAYER_ARCHIVES) \
                    lib/SlingboxLib/SlingboxLib.a \
                    lib/libRTV/librtv.a \
                    lib/libUPnP/libupnp.a \
diff --git a/configure.in b/configure.in
index 3b1a1f3..10eab9b 100755
--- a/configure.in
+++ b/configure.in
@@ -1900,6 +1900,7 @@ OUTPUT_FILES="Makefile \
     xbmc/cdrip/Makefile \
     xbmc/cores/Makefile \
     xbmc/cores/VideoRenderers/Makefile \
+    xbmc/cores/gstplayer/Makefile \
     xbmc/cores/dvdplayer/Makefile \
     lib/Makefile \
     lib/libdvd/Makefile \

