diff --git a/xbmc/cores/gstplayer/GstPlayer.cpp b/xbmc/cores/gstplayer/GstPlayer.cpp
index 201730d..40385c4 100644
--- a/xbmc/cores/gstplayer/GstPlayer.cpp
+++ b/xbmc/cores/gstplayer/GstPlayer.cpp
@@ -244,7 +244,7 @@ int CGstPlayer::OutputPicture(GstBuffer * gstbuffer)
 
   GstCaps * caps = GST_BUFFER_CAPS(gstbuffer);
 
-  if (caps){
+  if (caps) {
     gint width, height;
     guint32 format;
     GstStructure * structure = gst_caps_get_structure (caps, 0);
@@ -261,6 +261,9 @@ int CGstPlayer::OutputPicture(GstBuffer * gstbuffer)
     pPicture->iDisplayWidth = pPicture->iWidth = width;
     pPicture->iDisplayHeight = pPicture->iHeight = height;
 
+    pPicture->iDisplayX = 0;
+    pPicture->iDisplayY = 0;
+
     if (format==GST_STR_FOURCC("NV12")){
       pPicture->data[0] = GST_BUFFER_DATA(gstbuffer);
       pPicture->data[1] = GST_BUFFER_DATA(gstbuffer)+m_output.width*m_output.height;
