diff -Naur xbmc-pvr-20120428-gite714180/configure.in xbmc-pvr-20120428-gite714180a/configure.in
--- xbmc-pvr-20120428-gite714180/configure.in	2012-05-26 19:14:03.898018585 -0700
+++ xbmc-pvr-20120428-gite714180a/configure.in	2012-05-26 19:19:30.137534896 -0700
@@ -1244,6 +1244,11 @@
   USE_EXTERNAL_FFMPEG=1
   AC_DEFINE([USE_EXTERNAL_FFMPEG], [1], [Whether to use external FFmpeg libraries.])
 
+  if test "x$cross_compiling" = "xyes"; then
+    if test "x$use_vdpau" = "xyes"; then
+      AC_MSG_NOTICE("Skipping VDPAU test as we are cross compiling")
+    fi
+  else
   # Disable vdpau support if external libavcodec doesn't have it
   AC_RUN_IFELSE(
     AC_LANG_PROGRAM([[#include <libavcodec/avcodec.h>]],
@@ -1256,6 +1261,7 @@
       use_vdpau=no
       AC_MSG_RESULT($ffmpeg_vdpau_not_supported)
     fi])
+  fi
 
   # Other headers to include if available.
   AC_CHECK_HEADERS([libavutil/mathematics.h],,)
