diff --git a/xbmc/cores/gstplayer/GstPlayer.cpp b/xbmc/cores/gstplayer/GstPlayer.cpp
index 059a8d3..f13d2a6 100644
--- a/xbmc/cores/gstplayer/GstPlayer.cpp
+++ b/xbmc/cores/gstplayer/GstPlayer.cpp
@@ -365,7 +365,7 @@ static void gst_xbmc_src_uri_handler_init (gpointer g_iface,
   int ret;
   GstBuffer *buf;
   int rest_bytes = 0;
-  int timeout = 10;
+  int timeout = 256;
 
   if (G_UNLIKELY (src->read_position != offset)) {
     off_t res;
