diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
index 498c63b..76373f1 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
@@ -122,6 +122,11 @@ void CLinuxRendererGLES::UnRefBuf(int index)
   {
     buf.eglImageHandle->UnRef();
     buf.eglImageHandle = NULL;
+    YV12Image* im     = &buf.image;
+    im->cropX = 0;
+    im->cropY = 0;
+    im->cropWidth = 0;
+    im->cropHeight = 0;
   }
 }
 
@@ -1653,6 +1658,7 @@ bool CLinuxRendererGLES::CreateYV12Texture(int index)
   /* since we also want the field textures, pitch must be texture aligned */
   YV12Image &im     = m_buffers[index].image;
   YUVFIELDS &fields = m_buffers[index].fields;
+  m_textureTarget = GL_TEXTURE_2D;
 
   DeleteYV12Texture(index);
 

