diff --git a/configure.in b/configure.in
index 722e30b..402bf29 100644
--- a/configure.in
+++ b/configure.in
@@ -1865,6 +1865,7 @@ if test "x$use_gstreamer" != "xno"; then
       INCLUDES="$INCLUDES $GSTREAMER_CFLAGS $GSTREAMER_BASE_CFLAGS $GSTREAMER_PLUGINS_BASE_CFLAGS"
       LIBS="$LIBS $GSTREAMER_LIBS $GSTREAMER_BASE_LIBS $GSTREAMER_PLUGINS_BASE_LIBS"
       USE_GSTREAMER=1
+      XB_ADD_PLAYER([GSTPLAYER], [gstplayer])
       AC_DEFINE([HAVE_LIBGSTREAMER], [1], [Define to 1 if you have the 'GStreamer' library.])
     else
       if test "x$use_gstreamer" = "xyes"; then
diff --git a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
index 111e282..b76c990 100644
--- a/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoRenderers/LinuxRendererGLES.cpp
@@ -683,7 +688,7 @@ void CLinuxRendererGLES::LoadShaders(int field)
       // Try GLSL shaders if supported and user requested auto or GLSL.
       if (glCreateProgram)
       {
-   	    if (CONF_FLAGS_FORMAT_MASK(m_iFlags) == CONF_FLAGS_FORMAT_EGLIMG)
+   	    if (m_format == RENDER_FMT_EGLIMG)
         {
           m_pYUVShader = new EGLImageExternalShader();
           CLog::Log(LOGNOTICE, "GL: Selecting eglImage shader");
diff --git a/xbmc/cores/VideoRenderers/RenderManager.cpp b/xbmc/cores/VideoRenderers/RenderManager.cpp
index cf45e72..950ee8c 100644
--- a/xbmc/cores/VideoRenderers/RenderManager.cpp
+++ b/xbmc/cores/VideoRenderers/RenderManager.cpp
@@ -610,7 +610,7 @@ void CXBMCRenderManager::FlipPage(volatile bool& bStop, double timestamp /* = 0L
   if (timestamp != 0.0)
   {
     /* wait untill render thread have flipped buffers */
-    timeout = m_presenttime + 1.0;
+    double timeout = m_presenttime + 1.0;
     while(m_presentstep == PRESENT_FLIP && !bStop)
     {
       if(!m_presentevent.WaitMSec(100) && GetPresentTime() > timeout && !bStop)
diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodec.h b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodec.h
index 270085c..a2c39af 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodec.h
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodec.h
@@ -21,6 +21,7 @@
  */
 
 #include "system.h"
+#include "system_gl.h"
 #include "threads/SingleLock.h"
 
 #include <vector>


