--- a/lib/libmpo/CMakeLists.txt	2017-09-08 10:41:40.134870458 +0200
+++ b/lib/libmpoCMakeLists.txt	2017-09-08 10:41:40.146876456 +0200
@@ -46,9 +46,9 @@
 add_library(mpo_obj OBJECT ${source_files})
 #add_library(mpo SHARED $<TARGET_OBJECTS:mpo_obj>)
 add_library(mpo_static STATIC $<TARGET_OBJECTS:mpo_obj>)
-target_link_libraries(mpo jpeg)
+#target_link_libraries(mpo jpeg)
 set_target_properties(mpo_obj PROPERTIES POSITION_INDEPENDENT_CODE ON)
-set_target_properties(mpo PROPERTIES POSITION_INDEPENDENT_CODE ON)
+#set_target_properties(mpo PROPERTIES POSITION_INDEPENDENT_CODE ON)
 set_target_properties(mpo_static PROPERTIES POSITION_INDEPENDENT_CODE ON)
 add_subdirectory(examples)
 
