diff -ruN vdr_1.7.22_ext//Makefile vdr_1.7.22_ext_mld//Makefile
--- vdr_1.7.22_ext//Makefile	2011-12-06 23:43:44.000000000 +0100
+++ vdr_1.7.22_ext_mld//Makefile	2011-12-06 23:56:18.000000000 +0100
@@ -180,7 +180,7 @@
 	@failed="";\
 	noapiv="";\
 	for i in `ls $(PLUGINDIR)/src | grep -v '[^a-z0-9]'`; do\
-	    echo "Plugin $$i:";\
+	    true echo "Plugin $$i:";\
 	    if ! grep -q "\$$(LIBDIR)/.*\$$(APIVERSION)" "$(PLUGINDIR)/src/$$i/Makefile" ; then\
 	       echo "ERROR: plugin $$i doesn't honor APIVERSION - not compiled!";\
 	       noapiv="$$noapiv $$i";\
diff -ruN vdr_1.7.22_ext//menu.c vdr_1.7.22_ext_mld//menu.c
--- vdr_1.7.22_ext//menu.c	2011-12-06 23:41:39.000000000 +0100
+++ vdr_1.7.22_ext_mld//menu.c	2011-12-06 23:56:18.000000000 +0100
@@ -3821,7 +3821,7 @@
 {
   Clear();
   char buffer[64];
-  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("Setup"), VDRVERSION);
+  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("VDR Setup"), VDRVERSION);
   SetTitle(buffer);
   SetHasHotkeys();
   Add(new cOsdItem(hk(tr("OSD")),           osUser1));
@@ -3900,9 +3900,9 @@
 {
 #ifdef USE_SETUP
   // Load Menu Configuration
-  cString menuXML = cString::sprintf("%s/setup/vdr-menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
+  cString menuXML = cString::sprintf("%s/../menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
   if (access(menuXML, 04) == -1)
-     menuXML = cString::sprintf("%s/setup/vdr-menu.xml", cPlugin::ConfigDirectory());
+     menuXML = cString::sprintf("%s/../menu.xml", cPlugin::ConfigDirectory());
   subMenu.LoadXml(menuXML);
   nrDynamicMenuEntries = 0;
 #endif /* SETUP */
@@ -4052,7 +4052,7 @@
          else if (strcmp(item, "Recordings") == 0)
             Add(new cOsdItem(hk(tr("Recordings")), osRecordings));
          else if (strcmp(item, "Setup") == 0) {
-            cString itemSetup = cString::sprintf("%s%s", tr("Setup"), *subMenu.GetMenuSuffix());
+            cString itemSetup = cString::sprintf("%s%s", tr("VDR Setup"), *subMenu.GetMenuSuffix());
             Add(new cOsdItem(hk(itemSetup), osSetup));
             }
          else if (strcmp(item, "Commands") == 0 && Commands.Count() > 0) {
@@ -4103,9 +4103,9 @@
   // More basic menu items:
 
 #ifdef USE_PINPLUGIN
-  if (!cStatus::MsgMenuItemProtected("Setup", true)) Add(new cOsdItem(hk(tr("Setup")), osSetup));
+  if (!cStatus::MsgMenuItemProtected("Setup", true)) Add(new cOsdItem(hk(tr("VDR Setup")), osSetup));
 #else
-  Add(new cOsdItem(hk(tr("Setup")),      osSetup));
+  Add(new cOsdItem(hk(tr("VDR Setup")),      osSetup));
 #endif /* PINPLUGIN */
   if (Commands.Count())
 #ifdef USE_PINPLUGIN
diff -ruN vdr_1.7.22_ext//menuitems.c vdr_1.7.22_ext_mld//menuitems.c
--- vdr_1.7.22_ext//menuitems.c	2011-12-06 23:41:39.000000000 +0100
+++ vdr_1.7.22_ext_mld//menuitems.c	2011-12-06 23:56:18.000000000 +0100
@@ -1098,7 +1098,7 @@
 
 void cMenuSetupPage::SetSection(const char *Section)
 {
-  SetTitle(cString::sprintf("%s - %s", tr("Setup"), Section));
+  SetTitle(cString::sprintf("%s - %s", tr("VDR Setup"), Section));
 }
 
 eOSState cMenuSetupPage::ProcessKey(eKeys Key)
diff -ruN vdr_1.7.22_ext//submenu.c vdr_1.7.22_ext_mld//submenu.c
--- vdr_1.7.22_ext//submenu.c	2011-12-06 23:41:39.000000000 +0100
+++ vdr_1.7.22_ext_mld//submenu.c	2011-12-06 23:56:18.000000000 +0100
@@ -446,10 +446,6 @@
   if ((ok = xmlDoc.LoadFile())) {
      if ((root = xmlDoc.FirstChildElement("menus")) != NULL) {
         cString tmp = root->Attribute("suffix");
-#ifdef USE_WAREAGLEICON
-        if      (strcmp(tmp, "ICON_FOLDER") == 0)      tmp = cString::sprintf(" %s", IsLangUtf8() ? ICON_FOLDER_UTF8 : ICON_FOLDER);
-        else if (strcmp(tmp, "ICON_MOVE_FOLDER") == 0) tmp = cString::sprintf(" %s", IsLangUtf8() ? ICON_MOVE_FOLDER_UTF8 : ICON_MOVE_FOLDER);
-#endif /* WAREAGLEICON */
         if (*tmp)
            _menuSuffix = tmp;
         else
