diff -ruN vdr-1.7.22//svdrp.c vdr_1.7.22_ext_mld//svdrp.c
--- vdr-1.7.22//svdrp.c	2011-12-07 00:05:15.000000000 +0100
+++ vdr_1.7.22_ext_mld//svdrp.c	2011-12-07 00:16:10.000000000 +0100
@@ -399,6 +399,7 @@
   length = BUFSIZ;
   cmdLine = MALLOC(char, length);
   lastActivity = 0;
+  executing = false;
   isyslog("SVDRP listening on port %d", Port);
 }
 
@@ -1697,7 +1698,8 @@
         }
      if (NewConnection)
         lastActivity = time(NULL);
-     while (file.Ready(false)) {
+     if (!executing)
+        while (file.Ready(false)) {
            unsigned char c;
            int r = safe_read(file, &c, 1);
            if (r > 0) {
@@ -1708,7 +1710,9 @@
                  // make sure the string is terminated:
                  cmdLine[numChars] = 0;
                  // showtime!
+                 executing = true;
                  Execute(cmdLine);
+                 executing = false;
                  numChars = 0;
                  if (length > BUFSIZ) {
                     free(cmdLine); // let's not tie up too much memory
diff -ruN vdr-1.7.22//svdrp.h vdr_1.7.22_ext_mld//svdrp.h
--- vdr-1.7.22//svdrp.h	2011-12-07 00:05:15.000000000 +0100
+++ vdr_1.7.22_ext_mld//svdrp.h	2011-12-07 00:16:10.000000000 +0100
@@ -50,6 +50,7 @@
   char *cmdLine;
   time_t lastActivity;
   static char *grabImageDir;
+  bool executing;
   void Close(bool SendReply = false, bool Timeout = false);
   bool Send(const char *s, int length = -1);
   void Reply(int Code, const char *fmt, ...) __attribute__ ((format (printf, 3, 4)));
