diff -ruN vdr-1.7.22.f//config.c vdr-1.7.22.g//config.c
--- vdr-1.7.22.f//config.c	2011-12-07 00:54:03.000000000 +0100
+++ vdr-1.7.22.g//config.c	2011-12-07 01:02:47.000000000 +0100
@@ -491,6 +491,8 @@
   InitialVolume = -1;
   ChannelsWrap = 0;
   EmergencyExit = 1;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
 #ifdef USE_LIVEBUFFER
   LiveBufferSize = 15;
   LiveBufferMaxFileSize = 100;
@@ -728,6 +730,8 @@
   else if (!strcasecmp(Name, "DeviceBondings"))      DeviceBondings     = Value;
   else if (!strcasecmp(Name, "ChannelsWrap"))        ChannelsWrap       = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
 #ifdef USE_LIRCSETTINGS
   else if (!strcasecmp(Name, "LircRepeatDelay"))     LircRepeatDelay    = atoi(Value);
   else if (!strcasecmp(Name, "LircRepeatFreq"))      LircRepeatFreq     = atoi(Value);
@@ -887,6 +891,8 @@
   Store("DeviceBondings",     DeviceBondings);
   Store("ChannelsWrap",       ChannelsWrap);
   Store("EmergencyExit",      EmergencyExit);
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
 #ifdef USE_LIVEBUFFER
   Store("LiveBufferSize",     LiveBufferSize);
 #endif  /* LIVEBUFFER */
diff -ruN vdr-1.7.22.f//config.h vdr-1.7.22.g//config.h
--- vdr-1.7.22.f//config.h	2011-12-07 00:54:03.000000000 +0100
+++ vdr-1.7.22.g//config.h	2011-12-07 01:00:11.000000000 +0100
@@ -359,6 +359,8 @@
   int InitialVolume;
   int ChannelsWrap;
   int EmergencyExit;
+  int JumpSeconds;
+  int JumpSecondsSlow;
 #ifdef USE_LIVEBUFFER
   int LiveBufferSize;
   int LiveBufferMaxFileSize;
diff -ruN vdr-1.7.22.f//menu.c vdr-1.7.22.g//menu.c
--- vdr-1.7.22.f//menu.c	2011-12-07 00:54:03.000000000 +0100
+++ vdr-1.7.22.g//menu.c	2011-12-07 01:00:11.000000000 +0100
@@ -3698,6 +3698,8 @@
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Play&Jump"), &data.PlayJump));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause at last mark"), &data.PauseLastMark));
 #endif /* JUMPPLAY */
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds Slow"), &data.JumpSecondsSlow));
 }
 
 void cMenuSetupReplay::Store(void)
@@ -6009,9 +6011,13 @@
     case kFastFwd:
     case kRight:   Forward(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+    case kGreen:   SkipSeconds(-Setup.JumpSeconds); break;
+    case k1|k_Repeat:
+    case k1:       SkipSeconds(-Setup.JumpSecondsSlow); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
+    case k3|k_Repeat:
+    case k3:       SkipSeconds(Setup.JumpSecondsSlow); break;
 #ifdef USE_LIVEBUFFER
     case kRed:     if(cRecordControls::GetLiveIndex(fileName)) {
 
