diff -Naur libpciaccess-0.10.orig/configure libpciaccess-0.10/configure
--- libpciaccess-0.10.orig/configure	2008-04-12 15:06:28.000000000 +0200
+++ libpciaccess-0.10/configure	2008-04-12 15:06:49.000000000 +0200
@@ -19455,7 +19455,7 @@
 if test "${ac_cv_file__usr_include_asm_mtrr_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
