diff -Naur ffmpeg-r21465.orig/configure ffmpeg-r21465/configure
--- ffmpeg-r21465.orig/configure	2010-01-26 22:11:16.877989507 +0100
+++ ffmpeg-r21465/configure	2010-01-26 22:11:51.428064673 +0100
@@ -2508,7 +2508,7 @@
     die "ERROR: No version of libdc1394 found "
 fi
 
-SDL_CONFIG="${cross_prefix}sdl-config"
+SDL_CONFIG="sdl-config"
 if "${SDL_CONFIG}" --version > /dev/null 2>&1; then
     sdl_cflags=$("${SDL_CONFIG}" --cflags)
     sdl_libs=$("${SDL_CONFIG}" --libs)
