--- ffmpeg-0.11/configure.orig	2012-05-25 21:02:15.000000000 +0300
+++ ffmpeg-0.11/configure	2012-06-03 23:55:31.870186439 +0300
@@ -3261,7 +3261,7 @@
     die "ERROR: No version of libdc1394 found "
 fi
 
-SDL_CONFIG="${cross_prefix}sdl-config"
+SDL_CONFIG="sdl-config"
 if check_pkg_config sdl SDL_events.h SDL_PollEvent; then
     check_cpp_condition SDL.h "(SDL_MAJOR_VERSION<<16 | SDL_MINOR_VERSION<<8 | SDL_PATCHLEVEL) >= 0x010201" $sdl_cflags &&
     enable sdl &&
