diff -Naur a/configure b/configure
--- a/configure	2011-10-02 22:04:36.000000000 +0200
+++ b/configure	2011-10-27 08:03:21.784188968 +0200
@@ -2957,7 +2957,7 @@
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
 
-SDL_CONFIG="${cross_prefix}sdl-config"
+SDL_CONFIG="sdl-config"
 if check_pkg_config sdl SDL_version.h SDL_Linked_Version; then
     check_cpp_condition SDL.h "(SDL_MAJOR_VERSION<<16 | SDL_MINOR_VERSION<<8 | SDL_PATCHLEVEL) >= 0x010201" $sdl_cflags &&
     enable sdl &&
