diff -Naur ffmpeg-r24750.orig/libavcodec/vp8.c ffmpeg-r24750/libavcodec/vp8.c
--- ffmpeg-r24750.orig/libavcodec/vp8.c	2010-08-09 19:57:06.953534030 +0200
+++ ffmpeg-r24750/libavcodec/vp8.c	2010-08-09 20:37:59.762291901 +0200
@@ -541,7 +541,10 @@
     VP8Macroblock *mb_edge[3] = { mb + 2 /* top */,
                                   mb - 1 /* left */,
                                   mb + 1 /* top-left */ };
-    enum { EDGE_TOP, EDGE_LEFT, EDGE_TOPLEFT };
+#define EDGE_TOP     0
+#define EDGE_LEFT    1
+#define EDGE_TOPLEFT 2
+
     VP56mv near_mv[4]  = {{ 0 }};
     enum { CNT_ZERO, CNT_NEAREST, CNT_NEAR, CNT_SPLITMV };
     int idx = CNT_ZERO;
