diff -Naur u-boot-2010.12/include/configs/igep0020.h u-boot-2010.12a/include/configs/igep0020.h
--- u-boot-2010.12/include/configs/igep0020.h	2010-12-22 20:22:14.000000000 +0100
+++ u-boot-2010.12a/include/configs/igep0020.h	2011-02-13 09:46:22.107499541 +0100
@@ -131,7 +131,7 @@
 
 /* Environment information */
 #define CONFIG_BOOTCOMMAND \
-	"mmc init 0 ; fatload mmc 0 0x80000000 setup.ini ; source \0"
+	"mmc rescan 0; fatload mmc 0:1 0x80000000 boot.ini ; source \0"
 
 #define CONFIG_BOOTDELAY		3
 
diff -Naur u-boot-2010.12/include/configs/igep0030.h u-boot-2010.12a/include/configs/igep0030.h
--- u-boot-2010.12/include/configs/igep0030.h	2010-12-22 20:22:14.000000000 +0100
+++ u-boot-2010.12a/include/configs/igep0030.h	2011-02-13 09:46:22.107499541 +0100
@@ -129,7 +129,7 @@
 
 /* Environment information */
 #define CONFIG_BOOTCOMMAND \
-	"mmc init 0 ; fatload mmc 0 0x80000000 setup.ini ; source \0"
+	"mmc rescan 0; fatload mmc 0:1 0x80000000 boot.ini ; source \0"
 
 #define CONFIG_BOOTDELAY		3
 
