diff -Naur u-boot-2011.03-rc1/include/configs/igep0020.h u-boot-2011.03-rc1a/include/configs/igep0020.h
--- u-boot-2011.03-rc1/include/configs/igep0020.h	2011-02-19 10:52:36.505883111 +0100
+++ u-boot-2011.03-rc1a/include/configs/igep0020.h	2011-02-19 10:53:24.445896445 +0100
@@ -203,7 +203,7 @@
 
 #define ONENAND_ENV_OFFSET		0x260000 /* environment starts here */
 
-#define CONFIG_ENV_IS_IN_ONENAND	1
+#define CONFIG_ENV_IS_NOWHERE
 #define CONFIG_ENV_SIZE			(512 << 10) /* Total Size Environment */
 #define CONFIG_ENV_ADDR			ONENAND_ENV_OFFSET
 
diff -Naur u-boot-2011.03-rc1/include/configs/igep0030.h u-boot-2011.03-rc1a/include/configs/igep0030.h
--- u-boot-2011.03-rc1/include/configs/igep0030.h	2011-02-19 10:52:36.509846059 +0100
+++ u-boot-2011.03-rc1a/include/configs/igep0030.h	2011-02-19 10:53:42.237859336 +0100
@@ -201,7 +201,7 @@
 
 #define ONENAND_ENV_OFFSET		0x260000 /* environment starts here */
 
-#define CONFIG_ENV_IS_IN_ONENAND	1
+#define CONFIG_ENV_IS_NOWHERE
 #define CONFIG_ENV_SIZE			(512 << 10) /* Total Size Environment */
 #define CONFIG_ENV_ADDR			ONENAND_ENV_OFFSET
 
