--- vdr-skinclassic-0.0.3/Makefile.org	2012-04-11 15:25:41.104406543 +0200
+++ vdr-skinclassic-0.0.3/Makefile	2012-04-11 15:26:10.404406557 +0200
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -O2 -fPIC -Wall -Werror=overloaded-virtual
 
 ### The directory environment:
 
@@ -50,7 +50,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o i18n.o
+OBJS = $(PLUGIN).o
 
 ### Implicit rules:
 
