diff -Naur a/src/drivers/drivers.mak b/src/drivers/drivers.mak
--- a/src/drivers/drivers.mak	2014-06-30 06:42:53.000000000 +0100
+++ b/src/drivers/drivers.mak	2014-06-30 06:55:12.034918567 +0100
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 else
   ifdef CONFIG_LIBNL_TINY
     DRV_LIBS += -lnl-tiny
diff -Naur a/src/drivers/drivers.mk b/src/drivers/drivers.mk
--- a/src/drivers/drivers.mk	2014-06-30 06:42:53.000000000 +0100
+++ b/src/drivers/drivers.mk	2014-06-30 06:55:13.125783428 +0100
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 else
   ifdef CONFIG_LIBNL_TINY
     DRV_LIBS += -lnl-tiny
