diff -Naur vdr-vcd-0.9/Makefile vdr-vcd-0.9a/Makefile
--- vdr-vcd-0.9/Makefile	2008-01-16 16:29:39.000000000 +0100
+++ vdr-vcd-0.9a/Makefile	2008-09-14 14:47:27.000000000 +0200
@@ -47,7 +47,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES = -I$(VDRINC) -I$(DVBDIR)
+INCLUDES = -I$(VDRDIR)/include -I$(DVBDIR) -I.
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -D_GNU_SOURCE
 
