--- vdr-plugin-vcd-0.9.orig/debian/watch
+++ vdr-plugin-vcd-0.9/debian/watch
@@ -0,0 +1,2 @@
+version=2
+http://www.heiligenmann.de/vdr/download/vdr-vcd-(.*)\.tgz
--- vdr-plugin-vcd-0.9.orig/debian/compat
+++ vdr-plugin-vcd-0.9/debian/compat
@@ -0,0 +1 @@
+5
--- vdr-plugin-vcd-0.9.orig/debian/install
+++ vdr-plugin-vcd-0.9/debian/install
@@ -0,0 +1 @@
+libvdr-*.so.* usr/lib/vdr/plugins
--- vdr-plugin-vcd-0.9.orig/debian/changelog
+++ vdr-plugin-vcd-0.9/debian/changelog
@@ -0,0 +1,618 @@
+vdr-plugin-vcd (0.9-36yavdr18~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Sun, 25 Mar 2012 22:31:13 +0200
+
+vdr-plugin-vcd (0.9-36yavdr17~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Thu, 15 Mar 2012 10:23:15 +0100
+
+vdr-plugin-vcd (0.9-36yavdr16~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Tue, 13 Mar 2012 19:22:17 +0100
+
+vdr-plugin-vcd (0.9-36yavdr15~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Thu, 23 Feb 2012 09:06:26 +0100
+
+vdr-plugin-vcd (0.9-36yavdr14~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Mon, 13 Feb 2012 20:13:43 +0100
+
+vdr-plugin-vcd (0.9-36yavdr14~oneiric) oneiric; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Thu, 29 Dec 2011 11:17:24 +0100
+
+vdr-plugin-vcd (0.9-36yavdr13~oneiric) oneiric; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Sun, 25 Dec 2011 11:43:39 +0100
+
+vdr-plugin-vcd (0.9-36yavdr12~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Sat, 24 Dec 2011 11:32:40 +0100
+
+vdr-plugin-vcd (0.9-36yavdr11~precise) precise; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Mon, 17 Oct 2011 15:53:32 +0200
+
+vdr-plugin-vcd (0.9-36yavdr10~natty) natty; urgency=medium
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Mon, 05 Sep 2011 18:28:53 +0200
+
+vdr-plugin-vcd (0.9-36yavdr9~natty) natty; urgency=medium
+
+  * rebuild
+
+ -- Marco Scholl <develop@marco-scholl.de>  Thu, 25 Aug 2011 18:04:00 +0200
+
+vdr-plugin-vcd (0.9-36yavdr8~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Thu, 18 Aug 2011 20:11:15 +0200
+
+vdr-plugin-vcd (0.9-36yavdr7~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Wed, 15 Jun 2011 13:53:57 +0200
+
+vdr-plugin-vcd (0.9-36yavdr6~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Tue, 14 Jun 2011 20:50:16 +0200
+
+vdr-plugin-vcd (0.9-36yavdr5~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Wed, 25 May 2011 14:01:04 +0200
+
+vdr-plugin-vcd (0.9-36yavdr4~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Sun, 22 May 2011 23:02:19 +0200
+
+vdr-plugin-vcd (0.9-36yavdr3~natty) natty; urgency=low
+
+  * rebuild
+
+ -- Holger Schvestka <hotzenplotz5@gmx.de>  Sun, 03 Apr 2011 21:32:55 +0200
+
+vdr-plugin-vcd (0.9-9) unstable; urgency=low
+
+  * Release for vdrdevel 1.7.6
+  * Added ${misc:Depends}
+  * Bumped standards version to 3.8.1
+  * Updated debian/copyright
+  * Changed section to "video"
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Wed, 29 Apr 2009 23:11:11 +0200
+
+vdr-plugin-vcd (0.9-8) unstable; urgency=low
+
+  * Added 90_vcd-0.9-1.7.3.dpatch
+  * Updated debian/copyright
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Sun, 19 Apr 2009 19:42:25 +0200
+
+vdr-plugin-vcd (0.9-7) unstable; urgency=low
+
+  [ Tobias Grimm ]
+  * Dropped patchlevel control field
+  * Build-Depend on vdr-dev (>=1.6.0-5)
+  * Bumped Standards-Version to 3.8.0
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Sat, 19 Jul 2008 18:35:11 +0200
+
+vdr-plugin-vcd (0.9-6) unstable; urgency=low
+
+  * Upload to unstable
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Mon, 09 Jun 2008 00:14:00 +0200
+
+vdr-plugin-vcd (0.9-5) experimental; urgency=low
+
+  * Increased package version to force rebuild for vdr 1.6.0-1ctvdr7
+
+ -- Tobias Grimm <tg@e-tobi.net>  Mon, 12 May 2008 13:17:38 +0200
+
+vdr-plugin-vcd (0.9-4) experimental; urgency=low
+
+  [ Thomas Schmidt ]
+  * Build-Depend on vdr-dev (>=1.6.0)
+  * Updated years in debian/copyright
+
+  [ Tobias Grimm ]
+  * Using COMPAT=5 now
+
+ -- Tobias Grimm <tg@e-tobi.net>  Fri, 04 Apr 2008 14:34:34 +0200
+
+vdr-plugin-vcd (0.9-3) experimental; urgency=low
+
+  * Force rebuild for vdr 1.5.15
+
+ -- Tobias Grimm <tg@e-tobi.net>  Mon, 18 Feb 2008 21:09:23 +0100
+
+vdr-plugin-vcd (0.9-2) experimental; urgency=low
+
+  * Force update for new VDR release
+  * Added 01_Makefile-fPIC-fix.dpatch
+
+ -- Tobias Grimm <tg@e-tobi.net>  Sun, 17 Feb 2008 15:19:17 +0100
+
+vdr-plugin-vcd (0.9-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Tobias Grimm <tg@e-tobi.net>  Tue, 22 Jan 2008 01:02:39 +0100
+
+vdr-plugin-vcd (0.8-2) unstable; urgency=low
+
+   * Switched Build-System to cdbs, Build-Depend on cdbs
+   * Bumped Standards-Version to 3.7.3
+   * Added Homepage field to debian/control
+   * Renamed XS-Vcs-* fields to Vcs-* in debian/control 
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Fri, 28 Dec 2007 21:59:01 +0100
+
+vdr-plugin-vcd (0.8-1) unstable; urgency=low
+
+  * New upstream release
+  * Changed debian/watch to match the changed upstream URL (closes: #450314)
+  * Removed because of upstream fixes:
+    - 01_Makefile-fPIC-fix.dpatch
+    - 02_vdr-1.3.38-fix.dpatch
+    - 90_APIVERSION.dpatch
+    - 91_vcd-0.7-1.5.0.dpatch
+  * Updated debian/copyright
+
+ -- Tobias Grimm <tg@e-tobi.net>  Wed, 12 Dec 2007 20:05:56 +0100
+
+vdr-plugin-vcd (0.7-38) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.12
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue, 20 Nov 2007 23:46:34 +0100
+
+vdr-plugin-vcd (0.7-37) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.11
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue,  6 Nov 2007 23:34:25 +0100
+
+vdr-plugin-vcd (0.7-36) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.10
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue, 16 Oct 2007 23:51:05 +0200
+
+vdr-plugin-vcd (0.7-35) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.9
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue, 28 Aug 2007 01:01:16 +0200
+
+vdr-plugin-vcd (0.7-34) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.8
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Thu, 23 Aug 2007 01:09:14 +0200
+
+vdr-plugin-vcd (0.7-33) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.6
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue, 14 Aug 2007 01:46:27 +0200
+
+vdr-plugin-vcd (0.7-32) unstable; urgency=low
+
+  * Release for vdrdevel 1.5.5
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Wed, 27 Jun 2007 23:02:49 +0200
+
+vdr-plugin-vcd (0.7-31) unstable; urgency=low
+
+  * Added XS-Vcs-Svn and XS-Vcs-Browser fields to debian/control
+  * Removed version from Build-Dependancy to dpatch - even Sarge has the
+    required version
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Mon, 28 May 2007 20:27:25 +0200
+
+vdr-plugin-vcd (0.7-30) experimental; urgency=low
+
+  * Release for vdrdevel 1.5.2
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Sat, 28 Apr 2007 00:00:48 +0200
+
+vdr-plugin-vcd (0.7-29) experimental; urgency=low
+
+  * Release for vdrdevel 1.5.1
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Tue, 27 Feb 2007 19:59:25 +0100
+
+vdr-plugin-vcd (0.7-28) experimental; urgency=low
+
+  * Added 91_vcd-0.7-1.5.0.dpatch
+  * Added myself to Uploaders
+
+ -- Thomas GÃŒnther <tom@toms-cafe.de>  Sun, 21 Jan 2007 16:48:07 +0100
+
+vdr-plugin-vcd (0.7-27) experimental; urgency=low
+
+  [ Thomas GÃŒnther ]
+  * Replaced VDRdevel adaptions in debian/rules with make-special-vdr
+  * Adapted call of dependencies.sh and patchlevel.sh to the new location
+    in /usr/share/vdr-dev/
+  * Install HISTORY as changelog
+  * Using debian/compat now
+  * Shortened debian/copyright
+
+  [ Tobias Grimm ]
+  * Build-Depend on vdr-dev (>=1.4.5-1)
+
+ -- Tobias Grimm <tg@e-tobi.net>  Sun, 14 Jan 2007 20:24:05 +0100
+
+vdr-plugin-vcd (0.7-26) unstable; urgency=low
+
+  * Build-Depend on vdr-dev (>=1.4.4-1)
+
+ -- Tobias Grimm <tg@e-tobi.net>  Sun,  5 Nov 2006 13:51:08 +0100
+
+vdr-plugin-vcd (0.7-25) unstable; urgency=low
+
+  * Build-Depend on vdr-dev (>=1.4.3-1)
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Wed, 27 Sep 2006 08:30:11 +0200
+
+vdr-plugin-vcd (0.7-24) unstable; urgency=low
+
+  * Bumped Standards-Version to 3.7.2
+  * Added note about the Debian Maintainers to debian/copyright
+  * Added Tobias Grimm to Uploaders
+  * Build-Depend on vdr-dev (>=1.4.2-1)
+
+ -- Thomas Schmidt <tschmidt@debian.org>  Mon, 28 Aug 2006 20:59:15 +0200
+
+vdr-plugin-vcd (0.7-23) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.4.1-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 18 Jun 2006 15:55:04 +0200
+
+vdr-plugin-vcd (0.7-22) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.4.0-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 30 Apr 2006 21:34:34 +0200
+
+vdr-plugin-vcd (0.7-21) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.3.48-1)
+  * Tobias Grimm <tg@e-tobi.net>
+    - Added 90_APIVERSION.dpatch
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Mon, 24 Apr 2006 20:17:02 +0200
+
+vdr-plugin-vcd (0.7-20) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.3.46-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Thu, 13 Apr 2006 00:28:59 +0200
+
+vdr-plugin-vcd (0.7-19) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.3.45-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Mon, 27 Mar 2006 23:17:59 +0200
+
+vdr-plugin-vcd (0.7-18) unstable; urgency=low
+
+  * Tobias Grimm <tg@e-tobi.net>
+    - Depend/Build-Depend on vdr (>=1.3.41-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Tue, 31 Jan 2006 20:03:22 +0100
+
+vdr-plugin-vcd (0.7-17) unstable; urgency=low
+
+  * Tobias Grimm <tg@e-tobi.net>
+    - Build-Depend on vdr-dev (>=1.3.40-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Thu, 26 Jan 2006 19:45:10 +0100
+
+vdr-plugin-vcd (0.7-16) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr-dev (>=1.3.39-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 22 Jan 2006 12:52:11 +0100
+
+vdr-plugin-vcd (0.7-15) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.38-1)
+    - Added 02_vdr-1.3.38-fix.dpatch
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Tue, 10 Jan 2006 18:43:43 +0100
+
+vdr-plugin-vcd (0.7-14) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.37-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Mon, 28 Nov 2005 22:59:09 +0100
+
+vdr-plugin-vcd (0.7-13) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.35-1)
+    - Corrected FSF address in debian/copyright
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Wed,  2 Nov 2005 17:16:56 +0100
+
+vdr-plugin-vcd (0.7-12) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.33-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun,  2 Oct 2005 12:25:58 +0200
+
+vdr-plugin-vcd (0.7-11) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.32-1)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 11 Sep 2005 22:04:54 +0200
+
+vdr-plugin-vcd (0.7-10) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Build-Depend on vdr (>=1.3.31-1)
+    - Use dependencies.sh from vdr-dev to set vdr dependencies and
+      conflicts to the right vdr version when building the package
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Fri,  2 Sep 2005 18:24:09 +0200
+
+vdr-plugin-vcd (0.7-9) unstable; urgency=low
+
+  * Tobias Grimm <tg@e-tobi.net>
+    - Depend/Build-Depend on vdr (>=1.3.30-1)
+    - Conflict with vdr (>=1.3.30.99)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Tue, 23 Aug 2005 19:19:09 +0200
+
+vdr-plugin-vcd (0.7-8) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.29-1)
+    - Conflict with vdr (>=1.3.29.99)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Mon, 15 Aug 2005 19:52:27 +0200
+
+vdr-plugin-vcd (0.7-7) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.28-1)
+    - Conflict with vdr (>=1.3.29)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Wed, 10 Aug 2005 21:22:01 +0200
+
+vdr-plugin-vcd (0.7-6) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Bumped Standards-Version to 3.6.2
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Wed, 29 Jun 2005 18:43:32 +0200
+
+vdr-plugin-vcd (0.7-5) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.27-1)
+    - Conflict with vdr (>=1.3.28)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 19 Jun 2005 18:12:36 +0200
+
+vdr-plugin-vcd (0.7-4) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.26-1)
+    - Conflict with vdr (>=1.3.27)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 12 Jun 2005 17:16:05 +0200
+
+vdr-plugin-vcd (0.7-3) experimental; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.25-1)
+    - Conflict with vdr (>=1.3.26)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Tue, 31 May 2005 15:00:11 +0200
+
+vdr-plugin-vcd (0.7-2) experimental; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Depend/Build-Depend on vdr (>=1.3.24-1)
+    - Conflict with vdr (>=1.3.25)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Mon,  9 May 2005 17:56:31 +0200
+
+vdr-plugin-vcd (0.7-1) experimental; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - New upstream release
+    - Depend/Build-Depend on vdr (>=1.3.23-1)
+    - Conflict with vdr (>=1.3.24)
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sun, 10 Apr 2005 14:04:16 +0200
+
+vdr-plugin-vcd (0.0.6c-2) unstable; urgency=low
+
+  * Thomas Schmidt <tschmidt@debian.org>
+    - Use patchlevel.sh subst in debian/rules not the old code,
+      therefore Build-Depend on vdr (>=1.2.6-7)
+    - Build-depend on dpatch (>= 2.0.9)
+    - Converted existing dpatch-files to the new short format
+    - Changed my email-address to the debian one
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sat, 26 Mar 2005 17:53:07 +0100
+
+vdr-plugin-vcd (0.0.6c-1) unstable; urgency=low
+
+  * Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>
+    - First upload to the official archive (closes: #273312)
+    - New upstream release
+    - Added watch-file
+    - Added patchlevel-mechanism, to be compatible with c't-vdr		
+    - Use dpatch, so build-dep on dpatch
+    - Add 01_Makefile-fPIC-fix.dpatch
+    - Use dh_install
+    - Changed maintainer to Debian VDR and DVB Packaging Team 
+      <pkg-vdr-dvb-devel@lists.alioth.debian.org>
+    - Changed architecture to any
+
+ -- Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>  Sat, 25 Sep 2004 11:55:01 +0200
+
+vdr-plugin-vcd (0.0.6b-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sat, 22 Nov 2003 13:53:07 +0100
+
+vdr-plugin-vcd (0.0.6-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Mon, 17 Nov 2003 23:47:51 +0100
+
+vdr-plugin-vcd (0.0.5b-1) unstable; urgency=low
+
+  * New upstream release
+  * Rebuild for vdr 1.2.6
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun, 16 Nov 2003 15:14:52 +0100
+
+vdr-plugin-vcd (0.0.5-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun, 12 Oct 2003 12:58:13 +0200
+
+vdr-plugin-vcd (0.0.4i-6) unstable; urgency=low
+
+  * Rebuild for vdr 1.2.5
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Wed, 17 Sep 2003 20:33:38 +0200
+
+vdr-plugin-vcd (0.0.4i-5) unstable; urgency=low
+
+  * changed the plugin-dir to /usr/lib/vdr/plugins, to be FHS compliant
+    Thanks to Henning Glawe <glaweh@physik.fu-berlin.de> for pointing me 
+    to this one
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun, 14 Sep 2003 16:35:01 +0200
+
+vdr-plugin-vcd (0.0.4i-4) unstable; urgency=low
+
+  * Changed debian/rules and debian/control (added debian/control.template),
+    so that the correct dependancy to vdr is filled out automatically on the
+    version of vdr-dev, you have installed
+  * Bump Standards-Version to 3.6.1
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Wed, 10 Sep 2003 10:36:47 +0200
+
+vdr-plugin-vcd (0.0.4i-3) unstable; urgency=low
+
+  * Rebuild for vdr 1.2.2
+  * Bump Standards-Version to 3.6.0
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Wed, 13 Aug 2003 10:58:55 +0200
+
+vdr-plugin-vcd (0.0.4i-2) unstable; urgency=low
+
+  * Rebuild for vdr 1.2.1
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Mon,  9 Jun 2003 19:10:25 +0200
+
+vdr-plugin-vcd (0.0.4i-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun,  1 Jun 2003 23:36:18 +0200
+
+vdr-plugin-vcd (0.0.4h-2) unstable; urgency=low
+
+  * Rebuild for vdr 1.2.0
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun,  1 Jun 2003 22:01:58 +0200
+
+vdr-plugin-vcd (0.0.4h-1) unstable; urgency=low
+
+  * New upstream release
+  * Rebuild for vdr 1.1.29
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Wed, 30 Apr 2003 18:04:49 +0200
+
+vdr-plugin-vcd (0.0.4g-1) unstable; urgency=low
+
+  * New upstream release
+  * Rebuild for vdr 1.1.28
+  * Bump Standars-Version to 3.5.9
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Tue, 22 Apr 2003 13:34:25 +0200
+
+vdr-plugin-vcd (0.0.4f-3) unstable; urgency=low
+
+  * rebuild for vdr 1.1.27
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun, 13 Apr 2003 15:50:09 +0200
+
+vdr-plugin-vcd (0.0.4f-2) unstable; urgency=low
+
+  * rebuild for vdr 1.1.25 
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Wed,  5 Mar 2003 20:15:52 +0100
+
+vdr-plugin-vcd (0.0.4f-1) unstable; urgency=low
+
+  * New upstream release - ok, now thats enough for today ;-)
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun,  2 Feb 2003 16:42:26 +0100
+
+vdr-plugin-vcd (0.0.4e-1) unstable; urgency=low
+
+  * New upstream release
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun,  2 Feb 2003 16:03:27 +0100
+
+vdr-plugin-vcd (0.0.4b-1) unstable; urgency=low
+
+  * Initial Release.
+
+ -- Thomas Schmidt <thomas.schmidt@in.stud.tu-ilmenau.de>  Sun,  2 Feb 2003 15:21:57 +0100
+
--- vdr-plugin-vcd-0.9.orig/debian/copyright
+++ vdr-plugin-vcd-0.9/debian/copyright
@@ -0,0 +1,39 @@
+Upstream Homepage:
+    http://www.heiligenmann.de/vdr/
+
+Upstream Author:
+    Thomas Heiligenmann <thomas@heiligenmann.de>
+    Andreas Schultz <aschultz@warp10.net>
+
+Debian Maintainers:
+    Tobias Grimm <etobi@debian.org>
+    Thomas Schmidt <tschmidt@debian.org>
+    Thomas GÃŒnther <tom@toms-cafe.de>
+
+Copyright:
+    (C) 2002 - 2008 Thomas Heiligenmann
+
+Copyright (Debian packaging):
+    (C) 2002 - 2009 Thomas Schmidt, Thomas GÃŒnther, Tobias Grimm
+
+License:
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+    The complete text of the GNU General Public License can be found
+    in /usr/share/common-licenses/GPL-2 on most Debian systems.
+
+License (Debian packaging):
+    The Debian packaging is licensed under the GPL, version 2 or any
+    later version, see /usr/share/common-licenses/GPL-2.
--- vdr-plugin-vcd-0.9.orig/debian/control
+++ vdr-plugin-vcd-0.9/debian/control
@@ -0,0 +1,17 @@
+Source: vdr-plugin-vcd
+Section: video
+Priority: extra
+Maintainer: Debian VDR Team <pkg-vdr-dvb-devel@lists.alioth.debian.org>
+Uploaders: Thomas Schmidt <tschmidt@debian.org>, Tobias Grimm <tg@e-tobi.net>, Thomas GÃŒnther <tom@toms-cafe.de>
+Build-Depends: debhelper (>= 5), cdbs, dpatch, vdr-dev (>= 1.6.0-5)
+Standards-Version: 3.8.1
+Vcs-Svn: svn://svn.debian.org/pkg-vdr-dvb/vdr/vdr-plugin-vcd/trunk/
+Vcs-Browser: http://svn.debian.org/wsvn/pkg-vdr-dvb/vdr/vdr-plugin-vcd/trunk/
+Homepage: http://www.heiligenmann.de/vdr/
+
+Package: vdr-plugin-vcd
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}, ${vdr:Depends}
+Description: VDR Plugin for playing (S)VCD's
+ This plugin for the Linux Video Disc Recorder VDR allows you to play your
+ (S)VCD's within VDR.
--- vdr-plugin-vcd-0.9.orig/debian/rules
+++ vdr-plugin-vcd-0.9/debian/rules
@@ -0,0 +1,19 @@
+#!/usr/bin/make -f
+
+include /usr/share/cdbs/1/rules/debhelper.mk
+include /usr/share/cdbs/1/rules/dpatch.mk
+
+DEB_INSTALL_CHANGELOGS_ALL = HISTORY
+
+MAKE_OPTIONS = DVBDIR=/usr VDRDIR=/usr/include/vdr LIBDIR=.
+
+common-build-arch::
+	$(MAKE) all $(MAKE_OPTIONS)
+
+clean::
+	$(MAKE) -o .dependencies clean $(MAKE_OPTIONS)
+	rm -f libvdr-*.so.*
+
+common-binary-predeb-arch::
+	sh /usr/share/vdr-dev/dependencies.sh
+
--- vdr-plugin-vcd-0.9.orig/debian/docs
+++ vdr-plugin-vcd-0.9/debian/docs
@@ -0,0 +1 @@
+README
--- vdr-plugin-vcd-0.9.orig/debian/patches/01_Makefile-fPIC-fix.dpatch
+++ vdr-plugin-vcd-0.9/debian/patches/01_Makefile-fPIC-fix.dpatch
@@ -0,0 +1,19 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 01_Makefile-fPIC-fix.dpatch by Tobias Grimm <tg@e-tobi.net>
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: No description.
+
+@DPATCH@
+diff -urNad vdr-plugin-vcd-0.9~/Makefile vdr-plugin-vcd-0.9/Makefile
+--- vdr-plugin-vcd-0.9~/Makefile	2008-01-16 16:29:39.000000000 +0100
++++ vdr-plugin-vcd-0.9/Makefile	2008-02-17 15:18:06.000000000 +0100
+@@ -16,7 +16,7 @@
+ ### The C++ compiler and options:
+ 
+ CXX      ?= g++
+-CXXFLAGS ?= -O2 -Wall -Woverloaded-virtual
++CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
+ 
+ ### The directory environment:
+ 
--- vdr-plugin-vcd-0.9.orig/debian/patches/90_vcd-0.9-1.7.3.dpatch
+++ vdr-plugin-vcd-0.9/debian/patches/90_vcd-0.9-1.7.3.dpatch
@@ -0,0 +1,23 @@
+#! /bin/sh /usr/share/dpatch/dpatch-run
+## 90_vcd-0.9-1.7.3.dpatch by Thomas Günther <tom@toms-cafe.de>
+## http://toms-cafe.de/vdr/download/vcd-0.9-1.7.3.diff
+##
+## All lines beginning with `## DP:' are a description of the patch.
+## DP: Changes for VDR >= 1.7.3.
+
+@DPATCH@
+--- vcd-0.9/player.h
++++ vcd-0.9/player.h
+@@ -42,6 +42,12 @@
+ #define VIDEO_STREAM_SC     0xE0
+ #define SC_MASK             0xE0
+ 
++// from remux.h (VDR <= 1.7.2)
++#define I_FRAME    1
++
++// from recording.h (VDR <= 1.7.2)
++#define FRAMESPERSEC 25
++
+ class cVcdPlayer;
+ 
+ class cVcdPlayerControl : public cControl {
--- vdr-plugin-vcd-0.9.orig/debian/patches/00list
+++ vdr-plugin-vcd-0.9/debian/patches/00list
@@ -0,0 +1,2 @@
+01_Makefile-fPIC-fix
+90_vcd-0.9-1.7.3
