--- a/CMakeLists.txt	2018-10-24 17:34:19.606695916 +0200
+++ b/CMakeLists.txt	2018-10-24 17:34:12.774915212 +0200
@@ -10,7 +10,7 @@
 find_package(p8-platform REQUIRED)
 find_package(kodiplatform REQUIRED)
 find_package(Threads REQUIRED)
-find_package(Iconv REQUIRED)
+#find_package(Iconv REQUIRED)
 
 include_directories(${PROJECT_SOURCE_DIR}/lib/libsacd
                     ${PROJECT_SOURCE_DIR}/lib/libsacd/common
