--- a/CMakeLists.txt    2020-02-29 20:21:40.709322821 +0100
+++ b/CMakeLists.txt    2020-02-29 20:21:43.713302589 +0100
@@ -4,7 +4,6 @@
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR})

 find_package(Kodi REQUIRED)
-find_package(Iconv REQUIRED)
 find_package(Threads REQUIRED)
 if(WIN32)
   find_package(PThreads4W REQUIRED)
@@ -29,7 +28,7 @@
                  src/Helpers.h)

 if(NOT WIN32)
-  set(DEPLIBS sacd ${CMAKE_THREAD_LIBS_INIT} ${ICONV_LIBRARY})
+  set(DEPLIBS sacd ${CMAKE_THREAD_LIBS_INIT})
 else()
   set(DEPLIBS sacd ${CMAKE_THREAD_LIBS_INIT} ${PTHREADS4W_LIBRARIES} ${ICONV_LIBRARY} ws2_32)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4996")
