--- ./sound/soc/ux500/ux500_pcm.h.orig	2012-04-18 14:48:33.000000000 +0300
+++ ./sound/soc/ux500/ux500_pcm.h	2012-06-14 12:22:32.371031995 +0300
@@ -25,7 +25,10 @@
 #define UX500_PLATFORM_MAX_CHANNELS 8
 
 #define UX500_PLATFORM_PERIODS_BYTES_MIN	128
-#define UX500_PLATFORM_PERIODS_BYTES_MAX	(64 * PAGE_SIZE)
+/* Our DMA supports maximum of 131070 bytes segment size
+ * So, we can support period_size of max 32767 frames
+ */
+#define UX500_PLATFORM_PERIODS_BYTES_MAX       131070
 #define UX500_PLATFORM_PERIODS_MIN		2
 #define UX500_PLATFORM_PERIODS_MAX		48
 #define UX500_PLATFORM_BUFFER_BYTES_MAX		(2048 * PAGE_SIZE)
