--- a/receiver.h	2012-04-03 23:22:25.000000000 +0200
+++ b/receiver.h.mod	2012-04-03 23:19:50.000000000 +0200
@@ -14,7 +14,7 @@
 
 #define MAXRECEIVEPIDS  64 // the maximum number of PIDs per receiver
 
-//#define LEGACY_CRECEIVER // Code enclosed with this macro is deprecated and may be removed in a future version
+#define LEGACY_CRECEIVER // Code enclosed with this macro is deprecated and may be removed in a future version
 
 class cReceiver {
   friend class cDevice;
