diff -Naur xf86-input-evdev-2.5.0.orig/src/emuMB.c xf86-input-evdev-2.5.0/src/emuMB.c
--- xf86-input-evdev-2.5.0.orig/src/emuMB.c	2010-08-23 19:34:57.850510623 +0200
+++ xf86-input-evdev-2.5.0/src/emuMB.c	2010-08-23 19:36:08.280510875 +0200
@@ -270,7 +270,7 @@
     EvdevPtr     pEvdev = (EvdevPtr)pInfo->private;
     int ms;
 
-    if (pEvdev->emulateMB.pending)
+    if (pEvdev && pEvdev->emulateMB.pending)
     {
         ms = pEvdev->emulateMB.expires - GetTimeInMillis();
         if (ms <= 0)
@@ -286,7 +286,7 @@
     EvdevPtr        pEvdev= (EvdevPtr) pInfo->private;
     int             ms;
 
-    if (pEvdev->emulateMB.pending)
+    if (pEvdev && pEvdev->emulateMB.pending)
     {
         ms = pEvdev->emulateMB.expires - GetTimeInMillis ();
         if (ms <= 0)
