--- a/Makefile	2014-11-29 16:31:18.710503807 +0100
+++ b/Makefile	2014-11-29 16:31:18.694503808 +0100
@@ -56,10 +56,10 @@
 	$(CC) $(CFLAGS) $(LIBUSB_CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^) $(LIBS) $(LIBUSB_LIBS)
 
 nand-part: nand-part-main.c nand-part.c nand-part-a10.h nand-part-a20.h
-	$(CC) $(CFLAGS) -c -o nand-part-main.o nand-part-main.c
-	$(CC) $(CFLAGS) -c -o nand-part-a10.o nand-part.c -D A10
-	$(CC) $(CFLAGS) -c -o nand-part-a20.o nand-part.c -D A20
-	$(CC) $(LDFLAGS) -o $@ nand-part-main.o nand-part-a10.o nand-part-a20.o $(LIBS)
+	$(CROSS_COMPILE)gcc $(CFLAGS) -c -o nand-part-main.o nand-part-main.c
+	$(CROSS_COMPILE)gcc $(CFLAGS) -c -o nand-part-a10.o nand-part.c -D A10
+	$(CROSS_COMPILE)gcc $(CFLAGS) -c -o nand-part-a20.o nand-part.c -D A20
+	$(CROSS_COMPILE)gcc $(LDFLAGS) -o $@ nand-part-main.o nand-part-a10.o nand-part-a20.o
 
 %: %.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^) $(LIBS)
