diff -Naur libnih-1.0.0/nih/logging.c libnih-1.0.0a/nih/logging.c
--- libnih-1.0.0/nih/logging.c	2009-11-28 17:41:01.000000000 +0100
+++ libnih-1.0.0a/nih/logging.c	2010-01-24 18:23:04.000000000 +0100
@@ -39,13 +39,6 @@
 
 
 /**
- * __abort_msg:
- *
- * A glibc variable that keeps the assertion message in the core dump.
- **/
-extern char *__abort_msg;
-
-/**
  * logger:
  *
  * Function used to output log messages.
@@ -110,21 +103,6 @@
 }
 
 
-/**
- * nih_log_abort_message:
- * @message: message to be logged.
- *
- * Save @message in the glibc __abort_msg variable so it can be retrieved
- * by debuggers if we should crash at this point.
- **/
-static void
-nih_log_abort_message (const char *message)
-{
-	if (__abort_msg)
-		nih_discard (__abort_msg);
-
-	__abort_msg = NIH_MUST (nih_strdup (NULL, message));
-}
 
 /**
  * nih_log_message:
@@ -160,9 +138,6 @@
 	message = NIH_MUST (nih_vsprintf (NULL, format, args));
 	va_end (args);
 
-	if (priority >= NIH_LOG_FATAL)
-		nih_log_abort_message (message);
-
 	/* Output the message */
 	ret = logger (priority, message);
 
