--- a/xineDevice.c
+++ b/xineDevice.c
@@ -300,19 +300,14 @@
 //#endif
   }
 
-  void cXineDevice::TrickSpeed(int Speed)
-  {
-    TrickSpeed(Speed, false);
-  }
-
-  void cXineDevice::TrickSpeed(int Speed, bool IBP)
+  void cXineDevice::TrickSpeed(int Speed, bool Forward)
   {
     f = false;
     ts = Speed;
 
     xfprintf(stderr, "TrickSpeed: %d\n", Speed);
     m_xineLib.execFuncTrickSpeedMode(lastCmdWasClear);
-    m_xineLib.execFuncSetSpeed(100.0 / Speed * (IBP ? 12 : 1));
+    m_xineLib.execFuncSetSpeed(100.0 / Speed);
     m_xineLib.execFuncWait();
     m_xineLib.freeze(false);
     m_xineLib.pause(false);
--- a/xineDevice.h
+++ b/xineDevice.h
@@ -50,8 +50,7 @@
     virtual bool CanReplay(void) const;
     virtual bool SetPlayMode(ePlayMode PlayMode);
     virtual bool HasIBPTrickSpeed(void);
-    virtual void TrickSpeed(int Speed, bool IBP);
-    virtual void TrickSpeed(int Speed);
+    virtual void TrickSpeed(int Speed, bool Forward);
     virtual void Clear(void);
     virtual void Play(void);
     virtual void Freeze(void);
--- a/xineOsd.c
+++ b/xineOsd.c
@@ -490,7 +490,7 @@
 }
       }
 
-      while (cPixmapMemory *pm = RenderPixmaps())
+      while (cPixmapMemory *pm = dynamic_cast<cPixmapMemory*>(RenderPixmaps()))
       {
 if (!head)
 {
@@ -510,7 +510,11 @@
 );
 */
         m_pRawOsd->Copy(pm, pm->DrawPort().Shifted(-pm->DrawPort().Point()), pm->ViewPort().Point());
-        delete pm; 
+#if APIVERSNUM >= 20110
+        DestroyPixmap(pm);
+#else
+        delete pm;
+#endif
       }
     }
     else if (m_pRawOsd)
