--- vdr-xine-0.9.4/xineLib.c.org	2013-10-23 16:51:39.582154388 +0200
+++ vdr-xine-0.9.4/xineLib.c	2013-10-22 23:25:36.629077016 +0200
@@ -4164,7 +4164,7 @@
     cMutexLock ioLock(&m_ioMutex);
 
     enum funcs grab_func;
-#if XINE_VERSION_CODE >= 10201
+/*#if XINE_VERSION_CODE >= 10201
     if (m_xine_version >= 905)
     {
       data_grab_image_v2_t data;
@@ -4180,6 +4180,7 @@
     }
     else
 #endif
+*/
     {
       data_grab_image_t data;
       data.header.func = func_grab_image;
