diff -Naur graphlcd-base-0.1.5/tools/convpic/Makefile graphlcd-base-0.1.5a/tools/convpic/Makefile
--- graphlcd-base-0.1.5/tools/convpic/Makefile	2007-02-25 18:41:30.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/convpic/Makefile	2008-10-04 15:26:48.000000000 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-0.1.5/tools/crtfont/Makefile graphlcd-base-0.1.5a/tools/crtfont/Makefile
--- graphlcd-base-0.1.5/tools/crtfont/Makefile	2007-02-25 18:41:30.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/crtfont/Makefile	2008-10-04 15:26:08.000000000 +0200
@@ -35,7 +35,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-0.1.5/tools/genfont/Makefile graphlcd-base-0.1.5a/tools/genfont/Makefile
--- graphlcd-base-0.1.5/tools/genfont/Makefile	2007-02-25 18:41:30.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/genfont/Makefile	2008-10-04 15:26:23.000000000 +0200
@@ -38,7 +38,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 	
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-0.1.5/tools/lcdtestpattern/Makefile graphlcd-base-0.1.5a/tools/lcdtestpattern/Makefile
--- graphlcd-base-0.1.5/tools/lcdtestpattern/Makefile	2007-02-25 18:41:31.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/lcdtestpattern/Makefile	2008-10-04 15:26:34.000000000 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-0.1.5/tools/showpic/Makefile graphlcd-base-0.1.5a/tools/showpic/Makefile
--- graphlcd-base-0.1.5/tools/showpic/Makefile	2007-02-25 18:41:30.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/showpic/Makefile	2008-10-04 15:26:57.000000000 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
diff -Naur graphlcd-base-0.1.5/tools/showtext/Makefile graphlcd-base-0.1.5a/tools/showtext/Makefile
--- graphlcd-base-0.1.5/tools/showtext/Makefile	2007-02-25 18:41:30.000000000 +0100
+++ graphlcd-base-0.1.5a/tools/showtext/Makefile	2008-10-04 15:27:09.000000000 +0200
@@ -36,7 +36,7 @@
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root -s $(PRGNAME) $(BINDIR)
+	install -m 755 -s $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
