--- a/configure	2015-01-04 10:44:46.311507547 +0100
+++ b/configure	2015-01-04 10:44:46.315507546 +0100
@@ -16708,7 +16708,7 @@
   fi

   # if we are cross-compiling, tell ffmpeg so
-  target_os=`echo $host_os | sed 's/-gnu//'`
+  target_os=`echo $host_os | cut -f1 -d-`
   if test "x$cross_compiling" = xyes; then
     embffmpeg_configure_args="$embffmpeg_configure_args --enable-cross-compile \
         --target-os=$target_os --arch=$host_cpu --cross-prefix=$host_alias-"
--- a/configure.ac	2015-01-04 10:44:46.311507547 +0100
+++ b/configure.ac	2015-01-04 10:44:46.315507546 +0100
@@ -326,7 +326,7 @@
   fi

   # if we are cross-compiling, tell ffmpeg so
-  target_os=`echo $host_os | sed 's/-gnu//'`
+  target_os=`echo $host_os | cut -f1 -d-`
   if test "x$cross_compiling" = xyes; then
     embffmpeg_configure_args="$embffmpeg_configure_args --enable-cross-compile \
         --target-os=$target_os --arch=$host_cpu --cross-prefix=$host_alias-"
