diff -uNr vdr-xineliboutput-1.1.0+rab8cf79.orig/Makefile vdr-xineliboutput-1.1.0+rab8cf79/Makefile
--- vdr-xineliboutput-1.1.0+rab8cf79.orig/Makefile	2013-10-02 16:05:42.000000000 +0800
+++ vdr-xineliboutput-1.1.0+rab8cf79/Makefile	2013-10-02 18:41:15.549440823 +0800
@@ -340,6 +340,7 @@
 install-lib: $(TARGETS_VDR)
 ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
 	@echo Installing $^
+	@mkdir -p  $(DESTDIR)$(LIBDIR)
 	install -D $^ $(DESTDIR)$(LIBDIR)/
 endif
 
