diff -uNr vdr-xineliboutput-1.1.0+rdd73be3.orig/Makefile vdr-xineliboutput-1.1.0+rdd73be3/Makefile
--- vdr-xineliboutput-1.1.0+rdd73be3.orig/Makefile	2014-07-14 23:33:00.084138870 +0800
+++ vdr-xineliboutput-1.1.0+rdd73be3/Makefile	2014-07-14 23:54:58.134674724 +0800
@@ -358,7 +358,7 @@
 	$(INSTALL) $@ $(LIBDIR)/
 endif
 $(VDRSXFE): $(OBJS_SXFE)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_SXFE) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_CEC) $(LIBS_PTHREAD) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_SXFE) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_CEC) $(LIBS_PTHREAD) -lstdc++ -o $@
 
 #
 # vdr-fbfe
@@ -370,7 +370,7 @@
 	$(INSTALL) $@ $(LIBDIR)/
 endif
 $(VDRFBFE): $(OBJS_FBFE)
-	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_FBFE) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_CEC) $(LIBS_PTHREAD) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_FBFE) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_CEC) $(LIBS_PTHREAD) -lstdc++ -o $@
 
 #
 # xine plugins
