--- vdr-xineliboutput-2.0.1/Makefile.orig	2014-07-01 21:17:14.000000000 +0200
+++ vdr-xineliboutput-2.0.1/Makefile	2014-10-13 15:29:43.728607335 +0200
@@ -27,8 +27,8 @@
 # Override configuration here or in ../../../Make.config
 #
 
-#NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal.mpv
-#STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
+NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal_sifbox_720x576.mpg
+STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
 XINELIBOUTPUT_CONFIGURE_OPTS =
 
 
