diff -uNr vdr-xineliboutput-cvs20130707.orig/configure vdr-xineliboutput-cvs20130707/configure
--- vdr-xineliboutput-cvs20130707.orig/configure	2012-08-30 14:44:20.000000000 +0800
+++ vdr-xineliboutput-cvs20130707/configure	2013-08-07 22:09:22.462923276 +0800
@@ -376,7 +376,7 @@
     test_library X11  xshape       "X11/extensions/shape.h"    "-lXext"      "XShapeQueryExtension(0,0,0)"
     test_library X11  xdpms        "X11/extensions/dpms.h"     "-lXext"      "DPMSDisable(0)"
     test_library X11  xinerama     "X11/extensions/Xinerama.h" "-lXinerama"  "XineramaQueryScreens(0,0)"
-    test_library X11  opengl       "GL/glx.h"                  "-lGL -lGLU"  "glXQueryVersion(0,0,0)"
+    test_library X11  gl           "GL/glx.h"                  "-lGL -lGLU -lglut -lstdc++"  "glXQueryVersion(0,0,0)"
     test_library none vdpau        "vdpau/vdpau_x11.h"         "-lvdpau"     "vdp_device_create_x11(0,0,0,0)"
     test_library X11  dbus-glib-1  \
       "dbus/dbus-glib.h" \
