--- extrecmenu/mymenurecordings.c.orig	2009-03-22 14:59:47.000000000 +0100
+++ extrecmenu/mymenurecordings.c	2009-03-22 20:39:10.000000000 +0100
@@ -352,13 +352,13 @@
 {
    // set tabs
    if(mysetup.ShowRecDate&&mysetup.ShowRecTime&&!mysetup.ShowRecLength) // recording date and time are shown, recording length not
-     SetCols(2,8,6);
+     SetCols(2,8,5);
    else if(mysetup.ShowRecDate&&mysetup.ShowRecTime) // all details are shown
-     SetCols(2,8,6,4);
+     SetCols(2,8,5,4);
    else if(mysetup.ShowRecDate&&!mysetup.ShowRecTime) // recording time is not shown
      SetCols(2,8,4);
    else if(!mysetup.ShowRecDate&&mysetup.ShowRecTime&&mysetup.ShowRecLength) // recording date is not shown
-     SetCols(2,6,4);
+     SetCols(2,5,4);
    else // recording date and time are not shown; even if recording length should be not shown we must set two tabs because of the details of the directories
    {
      if(mysetup.ShowNewRecs)
