--- extrecmenu/tools.c.orig	2009-03-22 16:06:50.000000000 +0100
+++ extrecmenu/tools.c	2009-03-22 17:39:51.000000000 +0100
@@ -207,13 +207,8 @@
     }
     t++;
   }
-  if(mysetup.DescendSorting)
-  {
-    if(SortByName)
-      *s1=1;
-    else
-      *(s2+1)=255;
-  }
+  if(mysetup.DescendSorting && !SortByName)
+    *(s2+1)=255;
   else
     *s1=255;
  
@@ -231,7 +226,7 @@
   char *s2=StripEpisodeName(strdup(item->filename+strlen(VideoDirectory)));
 
   int compare;
-  if(mysetup.DescendSorting)
+  if(mysetup.DescendSorting && !SortByName)
     compare=strcasecmp(s2,s1);
   else
     compare=strcasecmp(s1,s2);
