diff -ruN extrecmenu-1.1-plain/mymenurecordings.c extrecmenu-1.1-graphtft/mymenurecordings.c
--- extrecmenu-1.1-plain/mymenurecordings.c	2007-10-19 18:36:10.000000000 +0200
+++ extrecmenu-1.1-graphtft/mymenurecordings.c	2008-03-17 18:54:43.000000000 +0100
@@ -37,6 +37,7 @@
     myMenuRecordingInfo(const cRecording *Recording,bool WithButtons = false);
     virtual void Display(void);
     virtual eOSState ProcessKey(eKeys Key);
+    virtual const char* MenuKind() { return "MenuExtRecording"; }
 };
 
 myMenuRecordingInfo::myMenuRecordingInfo(const cRecording *Recording, bool WithButtons):cOsdMenu(tr("Recording info"))
@@ -51,6 +52,8 @@
 {
   cOsdMenu::Display();
 
+  cStatus::MsgOsdSetRecording(recording);
+
   if(mysetup.UseVDRsRecInfoMenu)
   {
     DisplayMenu()->SetRecording(recording);
@@ -93,6 +96,7 @@
     text << tr("Lifetime") << ": " << recording->lifetime << "\n";
 
     DisplayMenu()->SetText(text.str().c_str(),false);
+    cStatus::MsgOsdTextItem(text.str().c_str());
   }
 }
 
diff -ruN extrecmenu-1.1-plain/mymenurecordings.h extrecmenu-1.1-graphtft/mymenurecordings.h
--- extrecmenu-1.1-plain/mymenurecordings.h	2007-09-29 22:25:35.000000000 +0200
+++ extrecmenu-1.1-graphtft/mymenurecordings.h	2008-03-17 18:55:45.000000000 +0100
@@ -54,6 +54,7 @@
   ~myMenuRecordings();
   void Set(bool Refresh=false,char *current=NULL);
   virtual eOSState ProcessKey(eKeys Key);
+  virtual const char* MenuKind() { return "MenuExtRecordings"; }
 };
 
 // --- myMenuRenameRecording --------------------------------------------------
