diff -Naur gtk-2.20.0/Makefile.in gtk-2.20.0a/Makefile.in
--- gtk-2.20.0/Makefile.in	2010-03-23 04:33:02.000000000 +0100
+++ gtk-2.20.0a/Makefile.in	2010-09-16 11:46:48.581675827 +0200
@@ -480,7 +480,7 @@
 	|| { echo "Gtk+Tests:ERROR: Failed to start Xvfb environment for X11 target tests."; exit 1; } \
 	&& DISPLAY=:$$XID && export DISPLAY
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules tests perf contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
