--- netsurf-2.9/Makefile.org	2013-03-26 15:52:43.454790794 +0100
+++ netsurf-2.9/Makefile	2013-03-26 15:53:16.006791352 +0100
@@ -3,7 +3,8 @@
 # NetSurf Source makefile for libraries and browser
 
 export TARGET ?= gtk
-export PKG_CONFIG_PATH = $(TMP_PREFIX)/lib/pkgconfig
+#export PKG_CONFIG_PATH = $(TMP_PREFIX)/lib/pkgconfig
+export PKG_CONFIG_PATH = ${SYSROOT_PREFIX}/usr/local/lib/pkgconfig
 TMP_PREFIX := $(CURDIR)/prefix-$(TARGET)
 
 #NetSurf
