--- systemd-206/rules/60-cdrom_id.rules	2013-07-22 00:43:28.119183329 +0200
+++ systemd-206/rules/60-cdrom_id.rules-new	2013-10-25 08:31:54.669373445 +0200
@@ -13,7 +13,7 @@
 
 # import device and media properties and lock tray to
 # enable the receiving of media eject button events
-IMPORT{program}="cdrom_id --lock-media $devnode"
+IMPORT{program}="cdrom_id $devnode"
 
 KERNEL=="sr0", SYMLINK+="cdrom", OPTIONS+="link_priority=-100"
 
