--- a/src/basic/missing.h	2015-09-19 13:02:28.880381363 +0200
+++ b/src/basic/missing.h	2015-09-19 13:02:28.896381363 +0200
@@ -126,6 +126,10 @@
 #define SOL_NETLINK 270
 #endif

+#ifndef NTF_SELF
+#define NTF_SELF        0x02
+#endif
+
 #if !HAVE_DECL_PIVOT_ROOT
 static inline int pivot_root(const char *new_root, const char *put_old) {
         return syscall(SYS_pivot_root, new_root, put_old);
@@ -1026,6 +1026,10 @@
 #define RENAME_NOREPLACE (1 << 0)
 #endif

+#ifndef __NR_kcmp
+#define __NR_kcmp 272
+#endif
+
 #if !HAVE_DECL_KCMP
 static inline int kcmp(pid_t pid1, pid_t pid2, int type, unsigned long idx1, unsigned long idx2) {
         return syscall(__NR_kcmp, pid1, pid2, type, idx1, idx2);
