diff -Naur fftw2-2.1.5/fftw/Makefile.in fftw2-2.1.5_patched/fftw/Makefile.in
--- fftw2-2.1.5/fftw/Makefile.in	2003-03-24 08:09:57.000000000 +0100
+++ fftw2-2.1.5_patched/fftw/Makefile.in	2012-02-22 20:55:17.907141021 +0100
@@ -652,7 +652,7 @@
 
 @FFTW_PREFIX1@fftw.h: fftw.h
 	rm -f $@
-	sed 's/\"fftw/\"@FFTW_PREFIX@fftw/g;s/\"rfftw/\"@FFTW_PREFIX@rfftw/g' $(srcdir)/fftw.h > $@
+	sed 's/\"fftw/\"@FFTW_PREFIX@fftw/g;s/\"rfftw/\"@FFTW_PREFIX@rfftw/g' fftw.h > $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Naur fftw2-2.1.5/fftw/Makefile.am fftw2-2.1.5_patched/fftw/Makefile.am
--- fftw2-2.1.5/fftw/Makefile.am	2003-03-24 08:05:32.000000000 +0100
+++ fftw2-2.1.5_patched/fftw/Makefile.am	2012-02-22 21:00:09.607141040 +0100
@@ -24,7 +24,7 @@
 
 @FFTW_PREFIX1@fftw.h: fftw.h
 	rm -f $@
-	sed 's/\"fftw/\"@FFTW_PREFIX@fftw/g;s/\"rfftw/\"@FFTW_PREFIX@rfftw/g' $(srcdir)/fftw.h > $@
+	sed 's/\"fftw/\"@FFTW_PREFIX@fftw/g;s/\"rfftw/\"@FFTW_PREFIX@rfftw/g' fftw.h > $@
 
 CLEANFILES = @FFTW_PREFIX1@fftw.h
 
