--- a/cursor.bdf	2010-01-24 17:45:11.000000000 +0100
+++ b/cursor.bdf	2010-01-24 17:44:49.000000000 +0100
@@ -13,45 +13,45 @@
 ENCODING -1 0
 SWIDTH 548 0
 DWIDTH 17 0
-BBX 14 14 -6 -8
+BBX 8 14 0 -14
 BITMAP
-e01d
-f03c
-f87c
-7cf8
-3ff0
-1fe0
-0fc0
-0fc0
-1fe0
-3ff0
-7cf8
-f87c
-f03c
-e01c
+80
+c0
+e0
+f0
+f8
+fc
+fe
+ff
+f8
+d8
+8c
+0c
+06
+06
 ENDCHAR
 STARTCHAR X_cursor_mask
 ENCODING -1 1
 SWIDTH 548 0
 DWIDTH 17 0
-BBX 16 16 -7 -9
+BBX 10 16 -1 -15
 BITMAP
-f00f
-f81f
-fc3f
-fe7f
-7ffe
-3ffc
-1ff8
-0ff0
-0ff0
-1ff8
-3ffc
-7ffe
-fe7f
-fc3f
-f81f
-f00f
+c003
+e000
+f000
+f800
+fc00
+fe00
+ff00
+ff80
+ffc0
+ffc0
+fe00
+ef00
+cf00
+0780
+0780
+0300
 ENDCHAR
 STARTCHAR arrow
 ENCODING -1 2
