--- a/CMakeLists.txt	2016-03-10 00:16:52.000000000 +0100
+++ b/CMakeLists.txt	2016-03-18 09:02:22.407423835 +0100
@@ -33,8 +33,7 @@
 endif()

 set(asplib_INCLUDE_DIRS ${PROJECT_SOURCE_DIR}
-                        ${PROJECT_SOURCE_DIR}/Biquads
-                        ${CMAKE_INSTALL_PREFIX}/include/asplib)
+                        ${PROJECT_SOURCE_DIR}/Biquads)

 include_directories(${asplib_INCLUDE_DIRS})
