diff -Naur enca-1.13/configure enca-1.13a/configure
--- enca-1.13/configure	2010-02-09 11:23:03.000000000 +0100
+++ enca-1.13a/configure	2010-09-14 17:14:55.000000000 +0200
@@ -12036,15 +12036,6 @@
 fi
 
 
-if test "$prefix" = "NONE"; then
-  LDFLAGS="$LDFLAGS -L$ac_default_prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$ac_default_prefix/include"
-else
-  LDFLAGS="$LDFLAGS -L$prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$prefix/include"
-fi
-
-
 case "$target" in
 NONE) yeti_libm_target="$host" ;;
 *) yeti_libm_target="$target" ;;
diff -Naur enca-1.13/Makefile.in enca-1.13a/Makefile.in
--- enca-1.13/Makefile.in	2010-02-09 11:23:03.000000000 +0100
+++ enca-1.13a/Makefile.in	2010-09-14 17:15:39.000000000 +0200
@@ -99,7 +99,7 @@
 	distdir dist dist-all distcheck
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = tools script lib src devel-docs test data
+DIST_SUBDIRS = script lib src devel-docs data
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
@@ -264,10 +264,10 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-@MAINTAINER_MODE_FALSE@SUBDIRS = tools script lib src devel-docs test
+@MAINTAINER_MODE_FALSE@SUBDIRS = script lib src devel-docs
 
 # @(#) $Id: Makefile.am,v 1.35 2005/11/24 10:21:32 yeti Exp $
-@MAINTAINER_MODE_TRUE@SUBDIRS = tools data script lib src devel-docs test
+@MAINTAINER_MODE_TRUE@SUBDIRS = data script lib src devel-docs
 man_MANS = man/enca.1
 DISTCHECK_CONFIGURE_FLAGS = --enable-maintainer-mode
 ACLOCAL_AMFLAGS = -I m4
