--- vdr-epgsync-1.0.0/setup.c.orig	2013-06-14 10:00:08.576027415 +0200
+++ vdr-epgsync-1.0.0/setup.c	2013-06-14 10:00:42.956028005 +0200
@@ -88,7 +88,7 @@
 	channelTypeTexts[ctIptv] = "IP";
 
 	Add(new cMenuEditBoolItem(tr("Hide mainmenu entry"), &setupTmp.hideMainMenuEntry));
-	Add(new cMenuEditStrItem(tr("Server IP"), setupTmp.serverIp, 15, ".1234567890"));
+	Add(new cMenuEditStrItem(tr("Server IP"), setupTmp.serverIp, 16, ".1234567890"));
 	Add(new cMenuEditIntItem(tr("Server port"), &setupTmp.serverPort, 0, 65535, tr("from svdrpservice")));
 	Add(new cMenuEditIntItem(tr("Connection attempts"), &setupTmp.connectAttempts, 0, 10));
 	Add(new cMenuEditBoolItem(tr("Update \"now\" and \"next\" first"), &setupTmp.nowNext));
