diff -urNad /usr/src/cvs/vdrdevel-plugin-solitaire/Makefile vdrdevel-plugin-solitaire/Makefile
--- /usr/src/cvs/vdrdevel-plugin-solitaire/Makefile	Fri Aug 27 18:07:06 2004
+++ vdrdevel-plugin-solitaire/Makefile	Sat Sep 18 18:30:11 2004
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
