Index: solitaire-0.0.2.orig/tools/list.h
===================================================================
--- solitaire-0.0.2.orig.orig/tools/list.h	2012-04-04 20:40:26.313840218 +0200
+++ solitaire-0.0.2.orig/tools/list.h	2012-04-04 20:40:33.693840222 +0200
@@ -523,7 +523,7 @@
 	return neu;
 }
 
-typedef class cTBList<cTBString> cStringList;
+//typedef class cTBList<cTBString> cStringList;
 typedef class cTBList<void*> cPtrList;
 
 // Special Overrides for String lists
