Index: vdr-plugin-solitaire-0.0.2/Makefile
===================================================================
--- vdr-plugin-solitaire-0.0.2.orig/Makefile	2012-04-04 20:24:25.273839638 +0200
+++ vdr-plugin-solitaire-0.0.2/Makefile	2012-04-04 20:25:07.603839664 +0200
@@ -46,7 +46,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o cards.o game.o osd.o cursor.o i18n.o \
+OBJS = $(PLUGIN).o cards.o game.o osd.o cursor.o \
 	\
 	tools/collection.o tools/list.o tools/shared.o tools/string.o tools/tools.o 
 
Index: vdr-plugin-solitaire-0.0.2/i18n.c
===================================================================
--- vdr-plugin-solitaire-0.0.2.orig/i18n.c	2012-04-04 20:23:43.283839612 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,148 +0,0 @@
-// originally from freecell by Sascha Volkenandt (sascha@akv-soft.de)
-
-#include "i18n.h"
-
-const tI18nPhrase Phrases[] = {
-	{	"Solitaire",							// English
-		"Solitaire",							// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Pasianssi",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"The well-known card game",// English
-		"Das bekannte Kartenspiel",// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Korttipeliklassikko",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"You won!",							// English
-		"Sie haben gewonnen!",	// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Voitit!",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"Select stack",					// English
-		"Stapel wählen",				// Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Valitse pino",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"Press OK to start a new game",// English
-		"OK beginnt ein neues Spiel",// Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Käynnistä uusi peli painamalla 'OK'",// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"This is Solitaire, a card-game for the Video Disk Recorder.\n\n"
-		"Up/Down: Jump cells/stacks\n"
-		"Left/Right: Move hand/cards\n"
-		"OK: Grab card or stack\n"
-		"Up/Down: Select/Deselect cards\n"
-		"OK: Confirm stack selection\n"
-		"Blue: Start new game\n\n"
-		"Press OK to resume playing", // English
-		"Dies ist Solitaire, ein Kartenspiel für den Video Disk Recorder.\n\n"
-		"Rauf/Runter: Zellen/Stapel springen\n"
-		"Links/Rechts: Hand/Stapel bewegen\n"
-		"OK: Karte oder Stapel nehmen\n"
-		"Rauf/Runter: Karten An-/Abwählen\n"
-		"OK: Stapelwahl bestätigen\n"
-		"Blau: Neues Spiel starten\n\n"
-		"OK drücken um fortzusetzen", // Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Pasianssi - korttipeliklassikko VDR:lle.\n\n" 
-		"'Ylös/Alas': Siirry kenttien/pinojen välillä\n" 
-		"'Vasen/Oikea': Siirrä kursoria/korttia\n" 
-		"'OK': Valitse kortti tai pino\n" 
-		"'Ylös/Alas': Valitse kortti\n" 
-		"'OK': Hyväksy pinon valinta\n" 
-		"'Sininen': Käynnistä uusi peli\n\n" 
-		"Paina 'OK' jatkaaksesi peliä",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{ NULL }
-};
Index: vdr-plugin-solitaire-0.0.2/i18n.h
===================================================================
--- vdr-plugin-solitaire-0.0.2.orig/i18n.h	2012-04-04 20:23:46.133839614 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,9 +0,0 @@
-#ifndef VDR_SOLITAIRE_I18N_H
-#define VDR_SOLITAIRE_I18N_H
-
-#include <vdr/i18n.h>
-
-extern const tI18nPhrase Phrases[];
-
-#endif // VDR_SOLITAIRE_I18N_H
-
Index: vdr-plugin-solitaire-0.0.2/solitaire.c
===================================================================
--- vdr-plugin-solitaire-0.0.2.orig/solitaire.c	2012-04-04 20:24:13.343839630 +0200
+++ vdr-plugin-solitaire-0.0.2/solitaire.c	2012-04-04 20:25:35.013839680 +0200
@@ -9,7 +9,6 @@
 
 #include <vdr/plugin.h>
 #include "game.h"
-#include "i18n.h"
 
 static const char *VERSION        = "0.0.2";
 static const char *DESCRIPTION    = "The well-known card game";
@@ -47,7 +46,6 @@
 
 bool cPluginSolitaire::Start(void)
 {
-	RegisterI18n(Phrases);
 	return true;
 }
 
