--- python-pycrypto-2.6.1/setup.py	2013-10-14 23:38:10.000000000 +0200
+++ python-pycrypto-2.6.1/setup.py-new	2015-03-25 18:21:53.362042846 +0100
@@ -370,7 +370,7 @@
       'ext_modules': plat_ext + [
             # _fastmath (uses GNU mp library)
             Extension("Crypto.PublicKey._fastmath",
-                      include_dirs=['src/','/usr/include/'],
+                      include_dirs=['src/'],
                       libraries=['gmp'],
                       sources=["src/_fastmath.c"]),
 
