mathinline is required for __CPU_HAS_FSQRT macro

diff -Nur glibc-2.5.orig/sysdeps/powerpc/fpu/e_sqrt.c glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c
--- glibc-2.5.orig/sysdeps/powerpc/fpu/e_sqrt.c	Sat Sep 30 11:33:31 2006
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrt.c	Sat Sep 30 11:48:38 2006
@@ -24,6 +24,7 @@
 
 #include <sysdep.h>
 #include <ldsodefs.h>
+#include <bits/mathinline.h>
 
 static const double almost_half = 0.5000000000000001;	/* 0.5 + 2^-53 */
 static const ieee_float_shape_type a_nan = {.word = 0x7fc00000 };
diff -Nur glibc-2.5.orig/sysdeps/powerpc/fpu/e_sqrtf.c glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c
--- glibc-2.5.orig/sysdeps/powerpc/fpu/e_sqrtf.c	Sat Sep 30 11:33:34 2006
+++ glibc-2.5/sysdeps/powerpc/fpu/e_sqrtf.c	Sat Sep 30 11:48:44 2006
@@ -24,6 +24,7 @@
 
 #include <sysdep.h>
 #include <ldsodefs.h>
+#include <bits/mathinline.h>
 
 static const float almost_half = 0.50000006;	/* 0.5 + 2^-24 */
 static const ieee_float_shape_type a_nan = {.word = 0x7fc00000 };
