diff -Naur glibc-2.7.orig/configure glibc-2.7/configure
--- glibc-2.7.orig/configure	2008-08-05 14:59:47.000000000 +0200
+++ glibc-2.7/configure	2008-08-05 14:59:59.000000000 +0200
@@ -5063,7 +5064,8 @@
 # thing on a system that doesn't need fixincludes.  (Not presently a problem.)
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  ccfixedheaders=`$CC -print-file-name=include-fixed`
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccfixedheaders \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
     cxxversion=`$CXX -dumpversion 2>&5` &&
