diff -Naur mex-0.1.6/mex/Makefile.in mex-0.1.6a/mex/Makefile.in
--- mex-0.1.6/mex/Makefile.in	2011-08-31 18:32:56.000000000 +0200
+++ mex-0.1.6a/mex/Makefile.in	2011-09-20 16:26:19.205579867 +0200
@@ -798,7 +798,6 @@
 libmex_@MEX_API_VERSION@_la_CPPFLAGS = \
 	$(common_includes)		\
 	-DG_LOG_DOMAIN=\"Mex\"		\
-	-DG_DISABLE_DEPRECATED		\
 	-DG_DISABLE_SINGLE_INCLUDES	\
 	-DG_COGL_DISABLE_DEPRECATED	\
 	-DG_CLUTTER_DISABLE_DEPRECATED	\
