diff -Naur a/base/src/distribution/core.py b/base/src/distribution/core.py
--- a/base/src/distribution/core.py	2011-08-30 07:53:42.000000000 +0200
+++ b/base/src/distribution/core.py	2011-08-31 07:23:38.497049109 +0200
@@ -32,7 +32,7 @@
 import tempfile
 import time
 import platform
-import distutils.core
+import setuptools
 import distutils.cmd
 import distutils.sysconfig
 import distutils.dist
@@ -350,7 +350,7 @@
             include_dirs.extend(lib.include_dirs)
             libraries.extend(lib.libraries)
 
-        ext = distutils.core.Extension(self.output, self.files,
+        ext = setuptools.Extension(self.output, self.files,
                                        library_dirs=library_dirs,
                                        include_dirs=include_dirs,
                                        libraries=libraries,
@@ -494,9 +494,8 @@
 
     # Use setuptools if --egg was passed.  We don't use setuptools by default (yet?)
     # because it changes certain behaviours (like install --prefix=[...]).
-    if '--egg' in sys.argv:
-        sys.argv.remove('--egg')
-        import setuptools
+    
+    import setuptools
 
     # Add --egg option
     opt = ('egg', None, "Install with setuptools (use eggs)")
@@ -673,7 +672,7 @@
         build_py.opts_2to3 = kwargs['opts_2to3']
         del kwargs['opts_2to3']
 
-    distutils.core.setup(**kwargs)
+    setuptools.setup(**kwargs)
 
     # Run cleanup on extensions (for example to delete config.h)
     for ext in kwargs['ext_modules']:
