diff -Naur vdr-vcd-0.9/menucontrol.c vdr-vcd-0.9a/menucontrol.c
--- vdr-vcd-0.9/menucontrol.c	2008-01-16 13:43:39.000000000 +0100
+++ vdr-vcd-0.9a/menucontrol.c	2009-04-17 11:01:44.000000000 +0200
@@ -284,8 +284,8 @@
 {
 #define STAY_SECONDS_OFF_END 10
   int Seconds = (timeSearchTime >> 24) * 36000 + ((timeSearchTime & 0x00FF0000) >> 16) * 3600 + ((timeSearchTime & 0x0000FF00) >> 8) * 600 + (timeSearchTime & 0x000000FF) * 60;
-  int Current = (lastCurrent / FRAMESPERSEC);
-  int Total = (lastTotal / FRAMESPERSEC);
+  int Current = (lastCurrent / DEFAULTFRAMESPERSECOND);
+  int Total = (lastTotal / DEFAULTFRAMESPERSECOND);
   switch (Key) {
     case k0 ... k9:
          if (timeSearchPos < 4) {
diff -Naur vdr-vcd-0.9/player.c vdr-vcd-0.9a/player.c
--- vdr-vcd-0.9/player.c	2008-01-16 13:26:38.000000000 +0100
+++ vdr-vcd-0.9a/player.c	2009-04-17 11:01:44.000000000 +0200
@@ -31,6 +31,11 @@
 #define MAX_SPEEDS    3 // the offset of the maximum speed from normal speed in either direction
 #define SPEED_MULT   12 // the speed multiplier
 
+// Picture types:
+#define NO_PICTURE 0
+#define I_FRAME    1
+#define P_FRAME    2
+#define B_FRAME    3
 
 // --- cVcdPlayer ------------------------------------------------------------
 
