--- vdr-plugin-extrecmenu-1.2+git20100901.orig/mymenurecordings.c
+++ vdr-plugin-extrecmenu-1.2+git20100901/mymenurecordings.c
@@ -463,7 +463,7 @@ myMenuRecordings::myMenuRecordings(const
   if(wasdvd&&!cControl::Control())
   {
     char *cmd=NULL;
-    if(-1!=asprintf(&cmd,"dvdarchive.sh umount \"%s\"",*strescape(myReplayControl::LastReplayed(),"'\\\"$")))
+    if(-1!=asprintf(&cmd,"/usr/share/vdr-plugin-extrecmenu/dvdarchive.sh umount \"%s\"",*strescape(myReplayControl::LastReplayed(),"'\\\"$")))
     {
       isyslog("[extrecmenu] calling %s to unmount dvd",cmd);
       int result=SystemExec(cmd);
@@ -913,7 +913,7 @@ eOSState myMenuRecordings::Play()
                 {
                   strcpy(path,recording->FileName());
                   name=strrchr(path,'/')+1;
-                  if(-1!=asprintf(&msg,"dvdarchive.sh mount \"%s\" '%s'",*strescape(path,"'"),*strescape(name,"'\\\"$")))
+                  if(-1!=asprintf(&msg,"/usr/share/vdr-plugin-extrecmenu/dvdarchive.sh mount \"%s\" '%s'",*strescape(path,"'"),*strescape(name,"'\\\"$")))
                   {
                     isyslog("[extrecmenu] calling %s to mount dvd",msg);
                     int result=SystemExec(msg);

