diff -r 29b7b5f231c8 device-tmpl.c
--- a/device-tmpl.c	Sat Jul 21 20:51:33 2012 +0200
+++ b/device-tmpl.c	Thu Jan 30 01:10:26 2014 +0400
@@ -41,7 +41,6 @@
 #endif //!SASC
 protected:
 #ifndef SASC
-  virtual bool Ready(void);
   virtual bool SetPid(cPidHandle *Handle, int Type, bool On);
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
   virtual bool OpenDvr(void);
@@ -164,12 +163,6 @@
   return cam || hwciadapter;
 }
 
-bool SCDEVICE::Ready(void)
-{
-  return (cam         ? cam->Ready():true) &&
-         (hwciadapter ? hwciadapter->Ready():true);
-}
-
 bool SCDEVICE::SetPid(cPidHandle *Handle, int Type, bool On)
 {
   if(cam) cam->SetPid(Type,Handle->pid,On);

