diff -Naur gcc-4.6.1.orig//libstdc++-v3/src/Makefile.am gcc-4.6.1/libstdc++-v3/src/Makefile.am
--- gcc-4.6.1.orig//libstdc++-v3/src/Makefile.am	2011-07-08 19:10:47.748474171 +0200
+++ gcc-4.6.1/libstdc++-v3/src/Makefile.am	2011-07-08 19:11:24.568474193 +0200
@@ -444,6 +444,11 @@
 CXXLINK = $(LIBTOOL) --tag CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link \
 	  $(CXX) $(OPT_LDFLAGS) $(SECTION_LDFLAGS) $(AM_CXXFLAGS) $(LTLDFLAGS) -o $@
 
+install-exec-local:
+ifeq ($(enable_shared),yes)
+	$(AR) cru libstdc++_pic.a .libs/*.o $(top_builddir)/libsupc++/*.o
+	$(INSTALL_DATA) libstdc++_pic.a $(DESTDIR)$(toolexeclibdir)
+endif
 
 # Added bits to build debug library.
 if GLIBCXX_BUILD_DEBUG
diff -Naur gcc-4.6.1.orig//libstdc++-v3/src/Makefile.in gcc-4.6.1/libstdc++-v3/src/Makefile.in
--- gcc-4.6.1.orig//libstdc++-v3/src/Makefile.in	2011-07-08 19:10:47.748474171 +0200
+++ gcc-4.6.1/libstdc++-v3/src/Makefile.in	2011-07-08 19:13:09.768474251 +0200
@@ -725,7 +725,7 @@
 
 install-dvi-am:
 
-install-exec-am: install-toolexeclibLTLIBRARIES
+install-exec-am: install-toolexeclibLTLIBRARIES install-exec-local
 
 install-html: install-html-am
 
@@ -775,6 +775,7 @@
 	info-am install install-am install-data install-data-am \
 	install-data-local install-dvi install-dvi-am install-exec \
 	install-exec-am install-html install-html-am install-info \
+	install-exec-local \
 	install-info-am install-man install-pdf install-pdf-am \
 	install-ps install-ps-am install-strip \
 	install-toolexeclibLTLIBRARIES installcheck installcheck-am \
@@ -1021,6 +1022,12 @@
 	(cd ${debugdir} && $(MAKE) \
 	toolexeclibdir=$(glibcxx_toolexeclibdir)/debug install)
 
++install-exec-local:
+ifeq ($(enable_shared),yes)
+	$(AR) cru libstdc++_pic.a *.o $(top_builddir)/libsupc++/*.o
+	$(INSTALL_DATA) libstdc++_pic.a $(DESTDIR)$(toolexeclibdir)
+endif
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
