--- a/lib/libsquish/Makefile.in	2017-01-14 12:51:01.142900968 +0100
+++ b/lib/libsquish/Makefile.in	2017-01-14 12:51:01.154900878 +0100
@@ -14,7 +14,7 @@
 CXXFLAGS           += -I.
 CXXFLAGS_FOR_BUILD += -I.
 LIB                 = libsquish.a
-NATIVE_LIB          = libsquish-native.so
+NATIVE_LIB          = libsquish-native.a
 CLEAN_FILES        += $(NATIVE_LIB)
 
 ifeq ($(findstring Darwin,$(shell uname -s)),Darwin)
@@ -28,7 +28,8 @@
 ifeq ($(findstring Darwin,$(shell uname -s)),Darwin)
 	$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) $(SRCS) -dynamiclib -install_name `pwd`/$(NATIVE_LIB) -o $@
 else
-	$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) $(SRCS) -shared -fPIC -Wl,-soname,`pwd`/$(NATIVE_LIB) -o $@
+	$(foreach f,$(SRCS),$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) -c $(f) -I. -fPIC -o $(patsubst %.cpp,%-native.o,$(f));)
+	ar rcs $(NATIVE_LIB) $(patsubst %.cpp,%-native.o,$(SRCS))
 endif
 
 include ../../Makefile.include
--- a/tools/TexturePacker/Makefile.in	2017-01-14 20:19:56.883765508 +0100
+++ b/tools/TexturePacker/Makefile.in	2017-01-14 20:19:56.895765428 +0100
@@ -31,7 +31,7 @@
 
 $(TARGET): $(SRCS) @abs_top_srcdir@/xbmc/guilib/XBTF.h
 # TexturePacker run native on build system, build it with native tools
-	make -C @abs_top_srcdir@/lib/libsquish/ libsquish-native.so
-	$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) $(DEFINES) $(NATIVE_ARCH) $(SRCS) $(LDFLAGS_FOR_BUILD) -o $(TARGET)
+	make -C @abs_top_srcdir@/lib/libsquish/ libsquish-native.a
+	$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) $(DEFINES) $(NATIVE_ARCH) $(SRCS) $(LDFLAGS_FOR_BUILD) -static-libstdc++ -o $(TARGET)
 
 include @abs_top_srcdir@/Makefile.include
