--- a/xbmc/music/MusicThumbLoader.cpp	2014-09-01 10:21:25.000000000 +0200
+++ b/xbmc/music/MusicThumbLoader.cpp	2016-02-18 07:51:49.151908795 +0100
@@ -152,8 +152,11 @@
 
   if (!pItem->HasArt("thumb"))
   {
+    EmbeddedArt ea;
+
     // Look for embedded art
-    if (pItem->HasMusicInfoTag() && !pItem->GetMusicInfoTag()->GetCoverArtInfo().empty())
+    if (pItem->HasMusicInfoTag() && 
+        (!pItem->GetMusicInfoTag()->GetCoverArtInfo().empty() || GetEmbeddedThumb(pItem->GetPath(), ea)))
     {
       // The item has got embedded art but user thumbs overrule, so check for those first
       if (!FillThumb(*pItem, false)) // Check for user thumbs but ignore folder thumbs
@@ -161,6 +164,7 @@
         // No user thumb, use embedded art
         CStdString thumb = CTextureUtils::GetWrappedImageURL(pItem->GetPath(), "music");
         pItem->SetArt("thumb", thumb);
+        SetCachedImage(*pItem, "thumb", thumb);
       }
     }
     else
--- a/xbmc/music/tags/MusicInfoTag.h	2016-02-18 07:51:49.155908795 +0100
+++ b/xbmc/music/tags/MusicInfoTag.h	2016-02-18 07:52:34.519906772 +0100
@@ -49,7 +49,7 @@
   class EmbeddedArtInfo : public IArchivable
   {
   public:
-    EmbeddedArtInfo() {};
+    EmbeddedArtInfo() { size = 0; };
     EmbeddedArtInfo(size_t size, const std::string &mime);
     void set(size_t size, const std::string &mime);
     void clear();
