--- a/lib/libsidplay2/Makefile.in	2019-11-18 23:25:12.288173005 +0100
+++ b/lib/libsidplay2/Makefile.in	2019-11-18 23:25:12.296172945 +0100
@@ -5,7 +5,7 @@
          -Ibuilders/resid-builder/include/sidplay/builders \
          -Ilibsidplay/include/sidplay -Iresid -Ilibsidplay/include -fPIC
 CFLAGS += $(SIDFLAGS)
-CXXFLAGS += $(SIDFLAGS)
+CXXFLAGS += $(SIDFLAGS) -Wno-narrowing
 ifeq ($(ARCH), powerpc-osx)
 	BUNDLE1_O=-lbundle1.o -L/usr/lib/gcc/powerpc-apple-darwin8/4.0.1/ -lgcc
 endif
