diff -Naur crystalhd-4.4+6cd0dc968-old/driver/linux/crystalhd_lnx.c crystalhd-4.4+6cd0dc968-new/driver/linux/crystalhd_lnx.c
--- crystalhd-4.4+6cd0dc968-old/driver/linux/crystalhd_lnx.c	2016-07-24 18:25:39.000000000 +0200
+++ crystalhd-4.4+6cd0dc968-new/driver/linux/crystalhd_lnx.c	2016-10-22 16:08:21.299565669 +0200
@@ -785,11 +785,19 @@
 }
 
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 24)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0)
+static const struct pci_device_id chd_dec_pci_id_table[] = {
+	{ PCI_VDEVICE(BROADCOM, 0x1612), 8 },
+	{ PCI_VDEVICE(BROADCOM, 0x1615), 8 },
+	{ 0, },
+};
+#else
 static DEFINE_PCI_DEVICE_TABLE(chd_dec_pci_id_table) = {
 	{ PCI_VDEVICE(BROADCOM, 0x1612), 8 },
 	{ PCI_VDEVICE(BROADCOM, 0x1615), 8 },
 	{ 0, },
 };
+#endif
 #else
 static struct pci_device_id chd_dec_pci_id_table[] = {
 /*	vendor, device, subvendor, subdevice, class, classmask, driver_data */
