diff -Naur crystalhd-3.9.0-git+20101203.orig//driver/linux/crystalhd_flea_ddr.c crystalhd-3.9.0-git+20101203/driver/linux/crystalhd_flea_ddr.c
--- crystalhd-3.9.0-git+20101203.orig//driver/linux/crystalhd_flea_ddr.c	2011-10-30 20:07:35.195760010 +0100
+++ crystalhd-3.9.0-git+20101203/driver/linux/crystalhd_flea_ddr.c	2011-10-30 20:08:01.375760634 +0100
@@ -24,6 +24,8 @@
  * along with this driver.  If not, see <http://www.gnu.org/licenses/>.
  **********************************************************************/
 
+#include <linux/delay.h>
+
 #include "crystalhd_hw.h"
 #include "crystalhd_flea_ddr.h"
 
