--- vdr-vompserver-0.3.3-3/Makefile.org	2012-12-27 22:47:32.984580352 +0100
+++ vdr-vompserver-0.3.3-3/Makefile	2012-12-27 22:48:01.384580835 +0100
@@ -32,7 +32,7 @@
 
 APIVERSNUM = $(shell grep 'define APIVERSNUM ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
 ifeq ($(shell test $(APIVERSNUM) -ge 10713; echo $$?),0) # thanks streamdev
-include $(VDRDIR)/Make.global
+-include $(VDRDIR)/Make.global
 endif
 
 ### Allow user defined options to overwrite defaults:
