diff -Nur -x '*.orig' -x '*~' gstreamer0.10-ffmpeg/ext/ffmpeg/gstffmpegdec.c gstreamer0.10-ffmpeg.new/ext/ffmpeg/gstffmpegdec.c
--- gstreamer0.10-ffmpeg/ext/ffmpeg/gstffmpegdec.c	2010-06-05 19:37:26.000000000 +0200
+++ gstreamer0.10-ffmpeg.new/ext/ffmpeg/gstffmpegdec.c	2010-07-06 11:45:50.000000000 +0200
@@ -2945,7 +2945,7 @@
       case CODEC_ID_RV30:
       case CODEC_ID_RV40:
       case CODEC_ID_COOK:
-        rank = GST_RANK_PRIMARY;
+        rank = GST_RANK_SECONDARY;
         break;
         /* DVVIDEO: we have a good dv decoder, fast on both ppc as well as x86.
          * They say libdv's quality is better though. leave as secondary.
