Index: gstreamer0.10-ffmpeg-0.10.10.2/ltmain.sh
===================================================================
--- gstreamer0.10-ffmpeg-0.10.10.2.orig/ltmain.sh	2010-06-27 10:52:20.000000000 +0200
+++ gstreamer0.10-ffmpeg-0.10.10.2/ltmain.sh	2010-06-27 11:36:09.861174141 +0200
@@ -5041,6 +5041,11 @@
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -5394,6 +5399,15 @@
 	lib=
 	found=no
 	case $deplib in
+	-Wl,--as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
