From 3643b38e70f5ea30b7037d8b3e7aa6df0195c861 Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Fri, 11 Nov 2011 16:41:13 +0100
Subject: [PATCH] tests/run: Fix crash if first hard disk is not SMART capable

---
 tests/run |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/tests/run b/tests/run
index 2fae2c8..fca6503 100755
--- a/tests/run
+++ b/tests/run
@@ -1223,6 +1223,7 @@ class Smart(UDisksTestCase):
                 self.assertEqual(e._dbus_error_name, 'org.freedesktop.UDisks.Error.Failed')
         else:
             sys.stderr.write('[N/A] ')
+            info = self.get_info(devname=hd)
             self.assertEqual(info['ATA SMART'], 'not available')
             self.assertEqual(props.Get(I_D, 'DriveAtaSmartTimeCollected'), 0)
             self.assertEqual(props.Get(I_D, 'DriveAtaSmartBlob'), [])
-- 
1.7.7.1

