--- udisks-1.0.4/data/80-udisks.rules	2012-09-07 13:26:42.825296761 +0200
+++ udisks-1.0.4/data/80-udisks.rules-new	2012-09-08 12:58:11.009508110 +0200
@@ -71,7 +71,7 @@
 #
 
 # skip rules for inappropriate block devices
-KERNEL=="fd*|mtd*|nbd*|gnbd*|btibm*", GOTO="probe_parttable_end"
+KERNEL=="fd*|mtd*|nbd*|gnbd*|btibm*|loop*", GOTO="probe_parttable_end"
 
 # never access non-cdrom removable ide devices, the drivers are causing event loops on open()
 KERNEL=="hd*[!0-9]", ATTR{removable}=="1", SUBSYSTEMS=="ide", ATTRS{media}=="disk|floppy", GOTO="probe_parttable_end"
