diff -urN vdr-1.7.40.orig/Makefile vdr-1.7.40/Makefile
--- vdr-1.7.40.orig/Makefile	2013-03-10 21:03:05.264167563 +0100
+++ vdr-1.7.40/Makefile	2013-03-10 21:03:36.380168097 +0100
@@ -29,11 +29,11 @@
 PLUGINDIR ?= $(CWD)/PLUGINS
 
 DESTDIR   ?=
-VIDEODIR  ?= /srv/vdr/video
-CONFDIR   ?= /var/lib/vdr
+VIDEODIR  ?= /data/tv/tv0
+CONFDIR   ?= /etc/vdr
 CACHEDIR  ?= /var/cache/vdr
 
-PREFIX    ?= /usr/local
+PREFIX    ?= /usr
 BINDIR    ?= $(PREFIX)/bin
 INCDIR    ?= $(PREFIX)/include
 LIBDIR    ?= $(PREFIX)/lib/vdr
diff -urN vdr-1.7.40.orig/menu.c vdr-1.7.40/menu.c
--- vdr-1.7.40.orig/menu.c	2013-03-10 21:03:05.252167561 +0100
+++ vdr-1.7.40/menu.c	2013-03-10 21:03:36.380168097 +0100
@@ -3479,7 +3479,7 @@
 {
   Clear();
   char buffer[64];
-  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("Setup"), VDRVERSION);
+  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("VDR Setup"), VDRVERSION);
   SetTitle(buffer);
   SetHasHotkeys();
   Add(new cOsdItem(hk(tr("OSD")),           osUser1));
diff -urN vdr-1.7.40.orig/menuitems.c vdr-1.7.40/menuitems.c
--- vdr-1.7.40.orig/menuitems.c	2013-03-10 21:03:05.268167562 +0100
+++ vdr-1.7.40/menuitems.c	2013-03-10 21:03:36.380168097 +0100
@@ -1132,7 +1132,7 @@
 
 void cMenuSetupPage::SetSection(const char *Section)
 {
-  SetTitle(cString::sprintf("%s - %s", tr("Setup"), Section));
+  SetTitle(cString::sprintf("%s - %s", tr("VDR Setup"), Section));
 }
 
 eOSState cMenuSetupPage::ProcessKey(eKeys Key)
