diff -urN vdr-1.7.40.orig/config.c vdr-1.7.40/config.c
--- vdr-1.7.40.orig/config.c	2013-03-10 21:05:21.944169909 +0100
+++ vdr-1.7.40/config.c	2013-03-10 21:05:46.848170336 +0100
@@ -392,6 +392,7 @@
   DiSEqC = 0;
   SetSystemTime = 0;
   TimeSource = 0;
+  ChannelBlocker = 0;
   TimeTransponder = 0;
   StandardCompliance = STANDARD_DVB;
   MarginStart = 2;
@@ -599,6 +600,7 @@
   else if (!strcasecmp(Name, "SetSystemTime"))       SetSystemTime      = atoi(Value);
   else if (!strcasecmp(Name, "TimeSource"))          TimeSource         = cSource::FromString(Value);
   else if (!strcasecmp(Name, "TimeTransponder"))     TimeTransponder    = atoi(Value);
+  else if (!strcasecmp(Name, "ChannelBlocker"))      ChannelBlocker     = atoi(Value);
   else if (!strcasecmp(Name, "StandardCompliance"))  StandardCompliance = atoi(Value);
   else if (!strcasecmp(Name, "MarginStart"))         MarginStart        = atoi(Value);
   else if (!strcasecmp(Name, "MarginStop"))          MarginStop         = atoi(Value);
@@ -709,6 +711,7 @@
   Store("SetSystemTime",      SetSystemTime);
   Store("TimeSource",         cSource::ToString(TimeSource));
   Store("TimeTransponder",    TimeTransponder);
+  Store("ChannelBlocker",     ChannelBlocker);
   Store("StandardCompliance", StandardCompliance);
   Store("MarginStart",        MarginStart);
   Store("MarginStop",         MarginStop);
diff -urN vdr-1.7.40.orig/config.h vdr-1.7.40/config.h
--- vdr-1.7.40.orig/config.h	2013-03-10 21:05:21.948169909 +0100
+++ vdr-1.7.40/config.h	2013-03-10 21:05:46.848170336 +0100
@@ -272,6 +272,7 @@
   int SetSystemTime;
   int TimeSource;
   int TimeTransponder;
+  int ChannelBlocker;
   int StandardCompliance;
   int MarginStart, MarginStop;
   int AudioLanguages[I18N_MAX_LANGUAGES + 1];
diff -urN vdr-1.7.40.orig/dvbdevice.c vdr-1.7.40/dvbdevice.c
--- vdr-1.7.40.orig/dvbdevice.c	2013-03-10 21:05:21.944169909 +0100
+++ vdr-1.7.40/dvbdevice.c	2013-03-10 21:05:46.852170335 +0100
@@ -1424,6 +1424,10 @@
 
 bool cDvbDevice::ProvidesSource(int Source) const
 {
+  if (Setup.ChannelBlocker == 1) {
+      if (IsPrimaryDevice()) return false;
+      //isyslog("ChannelBlocker aktive on primary Interface");
+  }
   int type = Source & cSource::st_Mask;
   return type == cSource::stNone
       || type == cSource::stAtsc  && ProvidesDeliverySystem(SYS_ATSC)
diff -urN vdr-1.7.40.orig/menu.c vdr-1.7.40/menu.c
--- vdr-1.7.40.orig/menu.c	2013-03-10 21:05:21.944169909 +0100
+++ vdr-1.7.40/menu.c	2013-03-10 21:05:46.856170336 +0100
@@ -3162,6 +3162,8 @@
          }
      }
 
+  Add(new cMenuEditBoolItem(tr("Setup.DVB$Channel Blocker"), &data.ChannelBlocker));
+
   SetCurrent(Get(current));
   Display();
 }
diff -urN vdr-1.7.40.orig/po/de_DE.po vdr-1.7.40/po/de_DE.po
--- vdr-1.7.40.orig/po/de_DE.po	2013-03-10 21:05:21.952169909 +0100
+++ vdr-1.7.40/po/de_DE.po	2013-03-10 21:05:46.856170336 +0100
@@ -893,6 +893,9 @@
 msgid "Setup.EPG$Preferred language"
 msgstr "Bevorzugte Sprache"
 
+msgid "Setup.DVB$Channel Blocker"
+msgstr "Tuner des primären DVB blocken"
+
 msgid "pan&scan"
 msgstr "pan&scan"
 
