--- vdr-1.7.38/Makefile.org	2013-02-18 15:41:51.335382299 +0100
+++ vdr-1.7.38/Makefile	2013-02-18 15:42:13.787381628 +0100
@@ -151,6 +151,7 @@
 	@echo "resdir=$(RESDIR)" >> $@
 	@echo "libdir=$(LIBDIR)" >> $@
 	@echo "locdir=$(LOCDIR)" >> $@
+	@echo "incdir=$(INCDIR)" >> $@
 	@echo "plgcfg=$(PLGCFG)" >> $@
 	@echo "apiversion=$(APIVERSION)" >> $@
 	@echo "cflags=$(CFLAGS) $(CDEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
