diff -Naur vlc-1.1.1.orig/src/interface/intf_eject.c vlc-1.1.1/src/interface/intf_eject.c
--- vlc-1.1.1.orig/src/interface/intf_eject.c	2010-07-23 22:51:01.981229112 +0200
+++ vlc-1.1.1/src/interface/intf_eject.c	2010-07-23 22:51:23.621229272 +0200
@@ -65,6 +65,8 @@
 #       include <linux/ucdrom.h>
 #   endif
 
+#   define u8 uint8_t
+
 #   include <scsi/scsi.h>
 #   include <scsi/sg.h>
 #   include <scsi/scsi_ioctl.h>
