diff -Naur vlc-1.1.1.orig/modules/video_output/omapfb.c vlc-1.1.1/modules/video_output/omapfb.c
--- vlc-1.1.1.orig/modules/video_output/omapfb.c	2010-07-23 22:51:02.461230046 +0200
+++ vlc-1.1.1/modules/video_output/omapfb.c	2010-07-23 22:56:30.151228994 +0200
@@ -397,7 +397,7 @@
         var_SetBool( p_vout, "fullscreen", p_vout->b_fullscreen );
 
         if( p_vout->p_sys->owner_window )
-            vout_window_SetFullscreen( p_vout->p_sys->owner_window,
+            vout_window_SetFullScreen( p_vout->p_sys->owner_window,
                                        p_vout->b_fullscreen );
         p_vout->i_changes &= ~VOUT_FULLSCREEN_CHANGE;
     }
