diff -Naur vlc-1.1.1.orig/modules/access/dvdnav.c vlc-1.1.1/modules/access/dvdnav.c
--- vlc-1.1.1.orig/modules/access/dvdnav.c	2010-07-23 22:51:02.441230643 +0200
+++ vlc-1.1.1/modules/access/dvdnav.c	2010-07-24 11:48:36.971230343 +0200
@@ -752,6 +752,7 @@
 
 #if defined(HAVE_DVDNAV_GET_VIDEO_RESOLUTION)
         uint32_t i_width, i_height;
+        extern int dvdnav_get_video_resolution(dvdnav_t *self, uint32_t *width, uint32_t *height);
         if( dvdnav_get_video_resolution( p_sys->dvdnav,
                                          &i_width, &i_height ) )
             i_width = i_height = 0;
