--- vdr-1.7.38/Makefile.org	2013-02-23 14:08:50.481229900 +0100
+++ vdr-1.7.38/Makefile	2013-02-23 14:11:55.897233085 +0100
@@ -10,11 +10,11 @@
 
 # Compiler flags:
 
-CC       ?= gcc
-CFLAGS   ?= -g -O3 -Wall
+#CC       ?= gcc
+#CFLAGS   ?= -g -O3 -Wall
 
-CXX      ?= g++
-CXXFLAGS ?= -g -O3 -Wall -Werror=overloaded-virtual -Wno-parentheses
+#CXX      ?= g++
+#CXXFLAGS ?= -g -O3 -Wall -Werror=overloaded-virtual -Wno-parentheses
 
 CDEFINES  = -D_GNU_SOURCE
 CDEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
@@ -113,6 +113,7 @@
 DEFINES += -DRESDIR=\"$(RESDIR)\"
 DEFINES += -DPLUGINDIR=\"$(LIBDIR)\"
 DEFINES += -DLOCDIR=\"$(LOCDIR)\"
+DEFINES += -DINCDIR=\"$(INCDIR)\"
 
 # The version numbers of VDR and the plugin API (taken from VDR's "config.h"):
 
