diff -ur jemalloc-3.5.1/configure.ac /home/jnettlet/tmp/jemalloc-3.5.1/configure.ac
--- jemalloc-3.5.1/configure.ac	2014-02-26 01:49:15.000000000 +0100
+++ /home/jnettlet/tmp/jemalloc-3.5.1/configure.ac	2014-03-10 02:51:10.339223241 +0100
@@ -1022,7 +1022,8 @@
     return 0;
 ]])],
                              [je_cv_static_page_shift=`cat conftest.out`],
-                             [je_cv_static_page_shift=undefined]))
+                             [je_cv_static_page_shift=undefined], 
+                             [je_cv_static_page_shift=12]))
 
 if test "x$je_cv_static_page_shift" != "xundefined"; then
    AC_DEFINE_UNQUOTED([STATIC_PAGE_SHIFT], [$je_cv_static_page_shift])
Only in /home/jnettlet/tmp/jemalloc-3.5.1/doc: html.xsl
Only in /home/jnettlet/tmp/jemalloc-3.5.1/doc: jemalloc.xml
Only in /home/jnettlet/tmp/jemalloc-3.5.1/doc: manpages.xsl
diff -ur jemalloc-3.5.1/include/jemalloc/internal/atomic.h /home/jnettlet/tmp/jemalloc-3.5.1/include/jemalloc/internal/atomic.h
--- jemalloc-3.5.1/include/jemalloc/internal/atomic.h	2014-02-26 01:49:15.000000000 +0100
+++ /home/jnettlet/tmp/jemalloc-3.5.1/include/jemalloc/internal/atomic.h	2014-03-10 02:51:10.189230373 +0100
@@ -145,7 +145,7 @@
 
 /******************************************************************************/
 /* 32-bit operations. */
-#ifdef __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4
+#if (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) || defined(__arm__))
 JEMALLOC_INLINE uint32_t
 atomic_add_uint32(uint32_t *p, uint32_t x)
 {
