diff -ruN vdr-plugin-dbus2vdr-12/shutdown.c vdr-plugin-dbus2vdr-12.edit/shutdown.c
--- vdr-plugin-dbus2vdr-12/shutdown.c	2013-06-01 09:51:24.000000000 +0200
+++ vdr-plugin-dbus2vdr-12.edit/shutdown.c	2013-10-19 13:49:53.931094714 +0200
@@ -39,7 +39,11 @@
        return;
        }
 
+#if APIVERSNUM > 20101
+    if (RecordingsHandler.Active()) {
+#else
     if (cCutter::Active()) {
+#endif
        SendReply(Invocation, 902, "cutter is active", 0, "");
        return;
        }
