diff -ruN vdr-1.7.21.org//livebuffer.c vdr-1.7.21/livebuffer.c
--- vdr-1.7.21.org//livebuffer.c	2011-10-05 14:59:58.000000000 +0200
+++ vdr-1.7.21/livebuffer.c	2011-10-05 15:08:11.508902180 +0200
@@ -278,10 +278,10 @@
 
 bool cLiveRecorder::Cleanup() {
 	if(FileName()) 
-                if(-1 == system(cString::sprintf("ls -l %s/* | grep -- '->' | sed -e's/.*-> //' | xargs rm -rf", FileName()))) // for symlink video.xx
+                if(-1 == system(cString::sprintf("ls -l %s/* 2>/dev/null | grep -- '->' | sed -e's/.*-> //' | xargs rm -rf", FileName()))) // for symlink video.xx
                         return false;
         else 
-		if(-1 == system(cString::sprintf("rm -rf %s/*", FileName())))
+		if(-1 == system(cString::sprintf("rm -rf %s/* 2>/dev/null", FileName())))
 			return false;
 	return true;
 }; // cLiveRecorder::Cleanup
