diff -ruN vdr-1.7.22.d//menu.c vdr-1.7.22.e/menu.c
--- vdr-1.7.22.d//menu.c	2011-12-07 00:31:53.000000000 +0100
+++ vdr-1.7.22.e/menu.c	2011-12-07 00:47:27.000000000 +0100
@@ -5329,11 +5329,11 @@
   Skins.Message(mtStatus, tr("Pausing live video..."));
   cReplayControl::SetRecording(NULL, NULL); // make sure the new cRecordControl will set cReplayControl::LastReplayed()
   if (Start(NULL, true)) {
-     cCondWait::SleepMs(2000); // allow recorded file to fill up enough to start replaying
+     cCondWait::SleepMs(5000); // allow recorded file to fill up enough to start replaying
      cReplayControl *rc = new cReplayControl;
      cControl::Launch(rc);
      cControl::Attach();
-     cCondWait::SleepMs(1000); // allow device to replay some frames, so we have a picture
+     cCondWait::SleepMs(2000); // allow device to replay some frames, so we have a picture
      Skins.Message(mtStatus, NULL);
      rc->ProcessKey(kPause); // pause, allowing replay mode display
      return true;
