diff -Nru vdr-1.7.15-vanilla/pat.c vdr-1.7.15-disable_ca_updates/pat.c
--- vdr-1.7.15-vanilla/pat.c	2010-06-06 19:11:15.000000000 +0300
+++ vdr-1.7.15-disable_ca_updates/pat.c	2010-06-06 20:42:47.000000000 +0300
@@ -490,6 +490,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, Atypes, ALangs, Dpids, Dtypes, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
diff -Nru vdr-1.7.15-vanilla/eitscan.c vdr-1.7.15-disable_eitscan/eitscan.c
--- vdr-1.7.15-vanilla/eitscan.c	2010-06-06 19:11:14.000000000 +0300
+++ vdr-1.7.15-disable_eitscan/eitscan.c	2010-06-06 20:44:07.000000000 +0300
@@ -146,7 +146,7 @@
                if (Device) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (!Device->Receiving()) {
