diff -Naur lcd4linux-0.10.1-RC1.orig/cfg.c lcd4linux-0.10.1-RC1/cfg.c
--- lcd4linux-0.10.1-RC1.orig/cfg.c	2007-03-23 11:16:58.000000000 +0100
+++ lcd4linux-0.10.1-RC1/cfg.c	2007-03-23 11:18:07.000000000 +0100
@@ -453,6 +453,7 @@
 	return 0;
 
     error = 0;
+#if 0
     if (!S_ISREG(stbuf.st_mode)) {
 	error("security error: '%s' is not a regular file", file);
 	error = -1;
@@ -465,6 +466,7 @@
 	error("security error: group or other have access to '%s'", file);
 	error = -1;
     }
+#endif
     return error;
 }
 
