diff -Naur lcd4linux-0.10.0.orig/cfg.c lcd4linux-0.10.0/cfg.c
--- lcd4linux-0.10.0.orig/cfg.c	2006-04-10 01:33:40.000000000 +0200
+++ lcd4linux-0.10.0/cfg.c	2006-04-10 01:34:26.000000000 +0200
@@ -645,6 +645,7 @@
 	return 0;
 
     error = 0;
+#if 0
     if (!S_ISREG(stbuf.st_mode)) {
 	error("security error: '%s' is not a regular file", file);
 	error = -1;
@@ -657,6 +658,7 @@
 	error("security error: group or other have access to '%s'", file);
 	error = -1;
     }
+#endif
     return error;
 }
 
