--- lcd4linux-0.10.0.orig/drv_generic_text.c	2005-05-08 06:32:44.000000000 +0200
+++ lcd4linux-0.10.0/drv_generic_text.c	2006-03-29 01:47:52.000000000 +0200
@@ -364,16 +364,16 @@
     int i;
     int flag = 0;
 
-    char *line1[] = { "* LCD4Linux " VERSION " *",
-	"LCD4Linux " VERSION,
-	"* LCD4Linux *",
-	"LCD4Linux",
-	"L4Linux",
+    char *line1[] = { "GeeXboX OMC Welcome",
+	"GeeXboX Welcome",
+	"GeeXboX OMC",
+	"GeeXboX",
+	"GB",
 	NULL
     };
 
-    char *line2[] = { "http://lcd4linux.bulix.org",
-	"lcd4linux.bulix.org",
+    char *line2[] = { "http://www.geexbox.org",
+	"www.geexbox.org",
 	NULL
     };
 
