* sent upstreaÃm

--- lcd4linux-0.10.1-RC1.orig/widget_bar.c	2007-04-28 00:57:56.000000000 +0200
+++ lcd4linux-0.10.1-RC1/widget_bar.c	2007-04-28 00:57:58.000000000 +0200
@@ -84,7 +84,7 @@
     /* maximum: if expression is empty, do auto-scaling */
     if (property_valid(&Bar->expr_max)) {
 	property_eval(&Bar->expr_max);
-	max = P2N(&Bar->expr_min);
+	max = P2N(&Bar->expr_max);
     } else {
 	max = Bar->max;
 	if (val1 > max)
