--- vdr-2.0.1/osd.c.org	2013-05-03 16:06:45.804789236 +0200
+++ vdr-2.0.1/osd.c	2013-05-03 16:07:05.724788643 +0200
@@ -1645,6 +1645,7 @@
          }
       }
   Osds.Append(this);
+  vidWin.bpp = 0;
 }
 
 cOsd::~cOsd()
--- vdr-2.0.1/osd.h.org	2013-05-03 16:06:51.304789075 +0200
+++ vdr-2.0.1/osd.h	2013-05-03 16:07:28.364787965 +0200
@@ -926,6 +926,7 @@
        ///<        MyOsdDrawPixmap(Left() + pm->ViewPort().X(), Top() + pm->ViewPort().Y(), pm->Data(), w, h, h * d);
        ///<        delete pm;
        ///<        }
+       tArea vidWin;
        static bool pinValid;
   };
 
