diff -urN vdr-2.0.1/plugin.h vdr-2.0.1.MCLI/plugin.h
--- vdr-2.0.1/plugin.h	2012-09-01 15:08:54.000000000 +0200
+++ vdr-2.0.1.MCLI/plugin.h	2013-05-03 14:30:41.268961035 +0200
@@ -48,6 +48,8 @@
   virtual const char *MainMenuEntry(void);
   virtual cOsdObject *MainMenuAction(void);
 
+  virtual cOsdObject *AltMenuAction(void) { return NULL; };
+
   virtual cMenuSetupPage *SetupMenu(void);
   virtual bool SetupParse(const char *Name, const char *Value);
   void SetupStore(const char *Name, const char *Value = NULL);
diff -urN vdr-2.0.1/vdr.c vdr-2.0.1.MCLI/vdr.c
--- vdr-2.0.1/vdr.c	2013-03-15 11:44:54.000000000 +0100
+++ vdr-2.0.1.MCLI/vdr.c	2013-05-03 14:29:11.308963713 +0200
@@ -1005,6 +1005,19 @@
            Recordings.Update();
            DeletedRecordings.Update();
            }
+        cPlugin *mcliPlugin = cPluginManager::GetPlugin("mcli");
+        if (mcliPlugin) {
+           if (!ShutdownHandler.countdown) { // if kPower has been pressed, cMenuShutdown takes precedence over other menus
+              cOsdObject *MyMenu = mcliPlugin->AltMenuAction();
+              if (MyMenu) { // is there any cam-menu waiting?
+                 DELETE_MENU;
+                 if (cControl::Control())
+                    cControl::Control()->Hide();
+                    Menu = MyMenu;
+                    Menu->Show();
+                 }
+              }
+           }
         // CAM control:
         if (!Menu && !cOsd::IsOpen())
            Menu = CamControl();
