--- a/render.c.orig	2011-11-06 22:23:42.000000000 +0100
+++ b/render.c	2011-11-07 22:38:11.000000000 +0100
@@ -334,7 +334,7 @@
 		const cFont *defFont = cFont::GetFont(fontOsd);
 		const char *dummy = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789 ";
 		//if (defFont != Object->Font())
-		mTabScale = 1.08 * (float)Object->Font()->Width(dummy) / (float)defFont->Width(dummy);
+		mTabScale = 1.25 * (float)Object->Font()->Width(dummy) / (float)defFont->Width(dummy);
 		mTabScaleSet = true;
 	}
 
@@ -404,19 +404,19 @@
 			txPoint prog_pos = Pos;
 			txSize prog_size = Size;
 
-			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + 4),
+			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + 8),
 			              txSize(prog_size.w, 2), Object->Fg());
-			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + 4),
-			              txSize(2, prog_size.h - 8), Object->Fg());
-			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + prog_size.h - 6),
+			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + 8),
+			              txSize(2, prog_size.h), Object->Fg());
+			DrawRectangle(txPoint(prog_pos.x, prog_pos.y + prog_size.h + 6),
 			              txSize(prog_size.w, 2), Object->Fg());
-			DrawRectangle(txPoint(prog_pos.x + prog_size.w - 2, prog_pos.y + 4),
-			              txSize(2, prog_size.h - 8), Object->Fg());
+			DrawRectangle(txPoint(prog_pos.x + prog_size.w - 2, prog_pos.y + 8),
+			              txSize(2, prog_size.h), Object->Fg());
 
 			prog_pos.x += 4;
-			prog_pos.y += 8;
+			prog_pos.y += 12;
 			prog_size.w -= 8;
-			prog_size.h -= 16;
+			prog_size.h -= 8;
 			DrawProgressbar(prog_pos, prog_size, current, total, Object->Bg(),
 			                Object->Fg(), NULL, NULL, NULL, NULL);
 		}
