--- vdr-plugin-text2skin-1.3+git20100419.orig/common.c
+++ vdr-plugin-text2skin-1.3+git20100419/common.c
@@ -146,7 +146,7 @@ bool GetFrontendHasSignal(void)
 std::string AddExtInfoToDescription(const char *Title, const char *ShortText, const char *Description, const char *Aux, bool StripAux)
 {
 	// max. width so lines don't get wrapped
-	#define MAX_CHARS 50
+	#define MAX_CHARS 100
 
 	// prepare the description
 	std::stringstream desc("");
@@ -180,9 +180,13 @@ std::string AddExtInfoToDescription(cons
 				list->First(); r && i < 5; r = list->Next(r)) {
 					i++;
 					std::stringstream buf;
+					cChannel *channel = Channels.GetByChannelID(r->event->ChannelID(), true, true);
+					if (channel)
+					buf << " - ";
+					buf << channel->ShortName(true);
 					buf << " - ";
 					buf << *DayDateTime(r->event->StartTime());
-					buf << ": " << r->event->Title();
+					buf << " - " << r->event->Title();
 					if (!isempty(r->event->ShortText())) buf << "~" << r->event->ShortText();
 					desc << FitToWidth(buf, MAX_CHARS) << "\n";
 				}
--- vdr-plugin-text2skin-1.3+git20100419.orig/Makefile
+++ vdr-plugin-text2skin-1.3+git20100419/Makefile
@@ -1,6 +1,7 @@
 #DEBUG=1
 #BENCH=1
 STRIP=strip
+DEVELOPMENT_FEATURES=1
 
 # BEWARE that you can not use GraphTFT together with
 # Text2Skin if you use Imlib2! (That's why I actually implemented ImageMagick)
