diff -Naur libsndfile-1.0.21/Makefile.in libsndfile-1.0.21a/Makefile.in
--- libsndfile-1.0.21/Makefile.in	2009-12-13 07:46:40.000000000 +0100
+++ libsndfile-1.0.21a/Makefile.in	2010-09-26 09:58:56.464544415 +0200
@@ -314,7 +314,7 @@
 top_srcdir = @top_srcdir@
 DISTCHECK_CONFIGURE_FLAGS = --enable-gcc-werror
 @BUILD_OCTAVE_MOD_TRUE@octave_dir = Octave
-SUBDIRS = M4 man doc Win32 src $(octave_dir) examples regtest tests programs
+SUBDIRS = M4 man doc Win32 src
 DIST_SUBDIRS = M4 man doc Win32 src Octave examples regtest tests programs
 EXTRA_DIST = libsndfile.spec.in sndfile.pc.in
 pkgconfigdir = $(libdir)/pkgconfig
