--- a/configure.ac	2018-04-27 11:37:27.740638256 +0200
+++ b/configure.ac	2018-04-27 11:37:45.336638044 +0200
@@ -1506,7 +1506,7 @@
 if test x"$grub_mkfont_excuse" = x ; then
   # Check for freetype libraries.
   freetype_cflags=`$FREETYPE --cflags`
-  freetype_libs=`$FREETYPE --libs`
+  freetype_libs=`$FREETYPE --static --libs`
   SAVED_CPPFLAGS="$CPPFLAGS"
   SAVED_LIBS="$LIBS"
   CPPFLAGS="$CPPFLAGS $freetype_cflags"
@@ -1567,7 +1567,7 @@
 if test x"$grub_build_mkfont_excuse" = x ; then
   # Check for freetype libraries.
   build_freetype_cflags=`$BUILD_FREETYPE --cflags`
-  build_freetype_libs=`$BUILD_FREETYPE --libs`
+  build_freetype_libs=`$BUILD_FREETYPE --static --libs`
   SAVED_CPPFLAGS_2="$CPPFLAGS"
   SAVED_LIBS="$LIBS"
   CPPFLAGS="$CPPFLAGS $build_freetype_cflags"
