diff -Nur pciutils-2.1.11.orig/lib/Makefile pciutils-2.1.11/lib/Makefile
--- pciutils-2.1.11.orig/lib/Makefile	Tue Apr 12 11:20:36 2005
+++ pciutils-2.1.11/lib/Makefile	Tue Apr 12 11:22:21 2005
@@ -4,6 +4,9 @@
 
 include config.mk
 
+AR?=ar
+RANLIB?=ranlib
+
 OBJS=access.o generic.o dump.o names.o filter.o
 INCL=internal.h pci.h config.h
 
@@ -47,8 +50,8 @@
 
 $(PCILIB): $(OBJS)
 	rm -f $@
-	ar rcs $@ $^
-	ranlib $@
+	$(AR) rcs $@ $^
+	$(RANLIB) $@
 
 access.o: access.c $(INCL)
 i386-ports.o: i386-ports.c $(INCL)
