diff -Nur pciutils-2.2.0.orig/lib/Makefile pciutils-2.2.0/lib/Makefile
--- pciutils-2.2.0.orig/lib/Makefile	Fri Aug 13 23:29:06 2004
+++ pciutils-2.2.0/lib/Makefile	Thu Nov  3 18:03:49 2005
@@ -3,7 +3,7 @@
 
 include config.mk
 
-OBJS=access.o generic.o dump.o names.o filter.o
+OBJS=access.o generic.o names.o filter.o
 INCL=internal.h pci.h config.h header.h sysdep.h types.h
 
 PCILIB=libpci.a
diff -Nur pciutils-2.2.0.orig/lib/configure pciutils-2.2.0/lib/configure
--- pciutils-2.2.0.orig/lib/configure	Wed Sep 21 14:42:03 2005
+++ pciutils-2.2.0/lib/configure	Thu Nov  3 18:03:41 2005
@@ -38,21 +38,13 @@
 case $sys in
 	linux*)
 		case $rel in
-			2.[1-9]*|[3-9]*)	echo_n " sysfs proc"
+			2.[1-9]*|[3-9]*)	echo_n " sysfs"
 						echo >>$c '#define PCI_HAVE_PM_LINUX_SYSFS'
-						echo >>$c '#define PCI_HAVE_PM_LINUX_PROC'
 						echo >>$c '#define PCI_HAVE_LINUX_BYTEORDER_H'
-						echo >>$c '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
 						echo >>$c '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
 						ok=1
 						;;
 		esac
-		case $cpu in
-				i386)		echo_n " i386-ports"
-						echo >>$c '#define PCI_HAVE_PM_INTEL_CONF'
-						ok=1
-						;;
-		esac
 		echo >>$c '#define PCI_HAVE_64BIT_ADDRESS'
 		;;
 	sunos)
@@ -96,8 +88,7 @@
 		;;
 esac
 
-echo >>$c '#define PCI_HAVE_PM_DUMP'
-echo " dump"
+echo ""
 if [ -z "$ok" ] ; then
 	echo "WARNING: No real configuration access method is available."
 fi
