diff -Naur laptop-detect-0.12.1.orig/laptop-detect.sh laptop-detect-0.12.1/laptop-detect.sh
--- laptop-detect-0.12.1.orig/laptop-detect.sh	2007-05-02 19:21:31.000000000 +0200
+++ laptop-detect-0.12.1/laptop-detect.sh	2007-05-02 19:21:47.000000000 +0200
@@ -25,7 +25,7 @@
 
 # check for any ACPI batteries
 if [ -d /proc/acpi/battery ]; then
-        results=`find /proc/acpi/battery/ -mindepth 1 -type d`
+        results=`find /proc/acpi/battery/ -type d`
         if [ ! -z "$results" ]; then
         #we're almost certainly a laptop
 #                echo "We're a laptop" >&2;
