diff -uNr vdr-sc-1.0+29b7b5f231c8.orig/device.c vdr-sc-1.0+29b7b5f231c8/device.c
--- vdr-sc-1.0+29b7b5f231c8.orig/device.c	2014-01-09 18:32:04.138504122 +0800
+++ vdr-sc-1.0+29b7b5f231c8/device.c	2014-01-09 18:34:33.990508008 +0800
@@ -411,7 +411,11 @@
 
 void cScDevices::SetForceBudget(int n)
 {
+#if APIVERSNUM >= 20103
+   if(n>=0 && n<MAXDEVICES) budget|=(1<<n);
+#else
    if(n>=0 && n<MAXDVBDEVICES) budget|=(1<<n);
+#endif
 }
 
 bool cScDevices::ForceBudget(int n)
