--- vdr-xineliboutput-1.1.0/Makefile.orig	2013-11-09 09:04:16.972721851 +0100
+++ vdr-xineliboutput-1.1.0/Makefile	2013-11-09 09:04:11.896721765 +0100
@@ -29,6 +29,8 @@
 
 #NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal.mpv
 #STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
+NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal_sifbox_720x576.mpg
+STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
 XINELIBOUTPUT_CONFIGURE_OPTS =
 
 
@@ -43,20 +45,20 @@
 ### The C++ compiler and options:
 ###
 
-CXX      ?= g++
-CC       ?= gcc 
+#CXX      ?= g++
+#CC       ?= gcc 
 OPTFLAGS ?= 
 
-ifeq ($(ARCH_APPLE_DARWIN), yes)
-    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
-    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
-    LDFLAGS_SO ?= -fvisibility=hidden
-else
-    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g
-    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g
-    LDFLAGS_SO ?= -fvisibility=hidden
-    LDFLAGS_SO += -shared
-endif
+#ifeq ($(ARCH_APPLE_DARWIN), yes)
+#    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
+#    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
+#    LDFLAGS_SO ?= -fvisibility=hidden
+#else
+#    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g
+#    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g
+#    LDFLAGS_SO ?= -fvisibility=hidden
+#    LDFLAGS_SO += -shared
+#endif
 
 ###
 ### The directory environment:
@@ -68,7 +70,7 @@
 LOCDIR = $(call PKGCFG,locdir)
 PLGCFG = $(call PKGCFG,plgcfg)
 VIDEODIR = $(call PKGCFG,videodir)
-
+DESTDIR ?= ./.install/
 TMPDIR  ?= /tmp
 BINDIR  ?= /usr/bin
 
