--- vdr-xineliboutput-2.0.0/Makefile.org	2013-11-09 11:06:52.376848112 +0100
+++ vdr-xineliboutput-2.0.0/Makefile	2013-11-09 11:08:55.976850233 +0100
@@ -4,7 +4,7 @@
 # See the main source file 'xineliboutput.c' for copyright information and
 # how to reach the author.
 #
-# $Id: Makefile,v 1.116 2013-10-28 15:31:45 rofafor Exp $
+# $Id: Makefile,v 1.110 2012-07-29 08:55:00 phintuka Exp $
 #
 
 # The official name of this plugin.
@@ -15,7 +15,6 @@
 
 _default: all
 
-# Keep VDR Makefile happy - it requires $(LIBDIR)/.$(APIVERSION) somewhere in this file ...
 
 # check for Apple Darwin
 ARCH_APPLE_DARWIN = no
@@ -27,8 +26,6 @@
 # Override configuration here or in ../../../Make.config
 #
 
-#NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal.mpv
-#STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
 NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal_sifbox_720x576.mpg
 STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
 XINELIBOUTPUT_CONFIGURE_OPTS =
@@ -45,26 +42,31 @@
 ### The C++ compiler and options:
 ###
 
-#CXX      ?= g++
-#CC       ?= gcc 
+CXX      ?= g++
+CC       ?= gcc 
 OPTFLAGS ?= 
 
-#ifeq ($(ARCH_APPLE_DARWIN), yes)
-#    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
-#    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
-#    LDFLAGS_SO ?= -fvisibility=hidden
-#else
-#    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g
-#    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g
-#    LDFLAGS_SO ?= -fvisibility=hidden
-#    LDFLAGS_SO += -shared
-#endif
+ifeq ($(ARCH_APPLE_DARWIN), yes)
+    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
+    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g -fno-common -bundle -flat_namespace -undefined suppress
+    LDFLAGS_SO ?= -fvisibility=hidden
+else
+    CXXFLAGS   ?= -O3 -pipe -Wall -Woverloaded-virtual -fPIC -g
+    CFLAGS     ?= -O3 -pipe -Wall -fPIC -g
+    LDFLAGS_SO ?= -fvisibility=hidden
+    LDFLAGS_SO += -shared
+endif
 
 ###
 ### The directory environment:
 ###
 
-# Use package data if installed...otherwise assume we're under the VDR source directory:
+VDRDIR  ?= ../../..
+LIBDIR  ?= ../../lib
+TMPDIR  ?= /tmp
+BINDIR  ?= /usr/bin
+DESTDIR ?= ./.install/
+
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
 LIBDIR = $(call PKGCFG,libdir)
 LOCDIR = $(call PKGCFG,locdir)
@@ -76,10 +78,14 @@
 
 INSTALL ?= install
 
+VDRINCDIR ?= $(VDRDIR)/include
+
 ###
 ### Allow user defined options to overwrite defaults:
 ###
 
+-include $(VDRDIR)/Make.global
+-include $(VDRDIR)/Make.config
 -include Make.config
 
 
@@ -87,28 +93,38 @@
 ### check for VDR
 ###
 
-APIVERSION = $(call PKGCFG,apiversion)
+ifeq ($(ARCH_APPLE_DARWIN), yes)
+    VDRVERSION = $(shell sed -ne '/define VDRVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
+    APIVERSION = $(shell sed -ne '/define APIVERSION/s/^.*"\(.*\)".*$$/\1/p' $(VDRDIR)/config.h)
+else
+    VDRVERSION = $(shell sed -ne '/define VDRVERSION/ { s/^.*"\(.*\)".*$$/\1/; p }' $(VDRDIR)/config.h)
+    APIVERSION = $(shell sed -ne '/define APIVERSION/ { s/^.*"\(.*\)".*$$/\1/; p }' $(VDRDIR)/config.h)
+endif
 
 VDR_TREE = no
-ifeq ($(strip $(APIVERSION)),)
+ifeq ($(strip $(VDRVERSION)),)
     $(warning ********************************************************)
     $(warning VDR not detected ! VDR plugins will not be compiled.    )
     $(warning ********************************************************)
     CONFIGURE_OPTS += --disable-vdr
 else
-    export CFLAGS   = $(call PKGCFG,cflags)
-    export CXXFLAGS = $(call PKGCFG,cxxflags)
+    ifeq ($(strip $(APIVERSION)),)
+        $(warning VDR APIVERSION missing, using VDRVERSION $(VDRVERSION) )
+        APIVERSION = $(VDRVERSION)
+    endif
+    CONFIGURE_OPTS += --add-cflags=-I$(VDRDIR)
 
-    ifeq ($(VDRDIR),)
-        $(warning Building outside VDR source tree)
-    else
+    ifeq ($(VDRDIR), ../../..)
         $(warning Building inside VDR source tree)
         VDR_TREE = yes
+    else
+        $(warning ********************************************************)
+        $(warning VDR source tree not detected !                          )
+        $(warning VDR plugins will not be installed.                      )
+        $(warning ********************************************************)
     endif
 endif
 
-### Allow user defined options to overwrite defaults:
--include $(PLGCFG)
 
 ###
 ### run configure script
@@ -172,6 +188,8 @@
 ### Includes and Defines (add further entries here):
 ###
 
+INCLUDES  += -I$(VDRINCDIR)
+
 ifeq ($(ARCH_APPLE_DARWIN), yes)
     INCLUDES  += -I/sw/include
     LIBDIRS   += -L/sw/lib
@@ -250,7 +268,7 @@
 $(DEPFILE): Makefile config.mak
 	@rm -f $@
 	@for i in $(OBJS:%.o=%.c) $(OBJS_SXFE:%.o=%.c) $(OBJS_FBFE:%.o=%.c) $(OBJS_XINE:%.o=%.c) ; do \
-	  $(MAKEDEP) $(CXXFLAGS) $(DEFINES) $(INCLUDES) -MT "`dirname $$i`/`basename $$i .c`.o" $$i >>$@ ; \
+	  $(MAKEDEP) $(DEFINES) $(INCLUDES) -MT "`dirname $$i`/`basename $$i .c`.o" $$i >>$@ ; \
 	done
 
 -include $(DEPFILE)
@@ -278,32 +296,32 @@
 	$(CC) $(CFLAGS) -c $(DEFINES) $(INCLUDES) $(CFLAGS_X11) $(CFLAGS_AVUTIL) $(OPTFLAGS) -o $@ $<
 
 ### Internationalization (I18N):
+ifeq ($(HAVE_I18N), yes)
+ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
 PODIR     = po
+LOCALEDIR ?= $(DESTDIR)$(VDRDIR)/locale
 I18Npo    = $(wildcard $(PODIR)/*.po)
-I18Nmo    = $(addsuffix .mo, $(foreach file, $(I18Npo), $(basename $(file))))
-I18Nmsgs  = $(addprefix $(DESTDIR)$(LOCDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
+I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/$(PLUGIN).pot
 
 %.mo: %.po
 	msgfmt -c -o $@ $<
 
 $(I18Npot): $(wildcard *.c)
-	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --package-name=vdr-$(PLUGIN) --package-version=$(VERSION) --msgid-bugs-address='<phintuka@users.sourceforge.net>' -o $@ $^
+	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --msgid-bugs-address='<phintuka@users.sourceforge.net>' -o $@ $^
 
 %.po: $(I18Npot)
-	msgmerge -U --no-wrap --no-location --backup=none -q -N $@ $<
+	msgmerge -U --no-wrap --no-location --backup=none -q $@ $<
 	@touch $@
 
-$(I18Nmsgs): $(DESTDIR)$(LOCDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
-ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
-	@echo Installing $^
-	install -D -m644 $< $@
+$(I18Nmsgs): $(LOCALEDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
+	@mkdir -p $(dir $@)
+	cp $< $@
+endif
 endif
 
 .PHONY: i18n
-i18n: $(I18Nmo) $(I18Npot)
-
-install-i18n: $(I18Nmsgs)
+i18n: $(I18Nmsgs)
 
 ###
 ### targets
@@ -335,18 +353,12 @@
 #
 
 $(VDRPLUGIN): $(OBJS) $(OBJS_MPG)
-	$(CXX) $(CXXFLAGS) $(LDFLAGS_SO) $(LDFLAGS) -shared $(OBJS) $(OBJS_MPG) $(LIBS) $(LIBS_VDR) -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS_SO) $(LDFLAGS) $(OBJS) $(OBJS_MPG) $(LIBS) $(LIBS_VDR) -o $@
 ifeq ($(VDR_TREE), yes)
-	$(INSTALL) $@ $(LIBDIR)/
-endif
-
-install-lib: $(TARGETS_VDR)
-ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
-	@echo Installing $^
-	install -D $^ $(DESTDIR)$(LIBDIR)/
+	@-rm -rf $(LIBDIR)/$@
+	@cp $@ $(LIBDIR)/$@
 endif
-
-install: install-lib install-i18n
+# Keep VDR Makefile happy - it requires $(LIBDIR)/.$(APIVERSION) somewhere in this file ...
 
 #
 # vdr-sxfe
@@ -355,7 +367,8 @@
 $(VDRPLUGIN_SXFE): $(OBJS_SXFE_SO)
 	$(CC) $(CFLAGS) $(LDFLAGS_SO) $(LDFLAGS) $(OBJS_SXFE_SO) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) -o $@
 ifeq ($(VDR_TREE), yes)
-	$(INSTALL) $@ $(LIBDIR)/
+	@-rm -rf $(LIBDIR)/$(VDRPLUGIN_SXFE)
+	@cp $@ $(LIBDIR)/$(VDRPLUGIN_SXFE)
 endif
 $(VDRSXFE): $(OBJS_SXFE)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_SXFE) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_PTHREAD) -o $@
@@ -367,7 +380,8 @@
 $(VDRPLUGIN_FBFE): $(OBJS_FBFE_SO)
 	$(CC) $(CFLAGS) $(LDFLAGS_SO) $(LDFLAGS) $(OBJS_FBFE_SO) $(LIBS_XINE) $(LIBS_JPEG) -o $@
 ifeq ($(VDR_TREE), yes)
-	$(INSTALL) $@ $(LIBDIR)/
+	@-rm -rf $(LIBDIR)/$(VDRPLUGIN_FBFE)
+	@cp $@ $(LIBDIR)/$(VDRPLUGIN_FBFE)
 endif
 $(VDRFBFE): $(OBJS_FBFE)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS_FBFE) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_PTHREAD) -o $@
@@ -391,34 +405,58 @@
 
 install: all
 ifeq ($(XINELIBOUTPUT_XINEPLUGIN), yes)
-	@mkdir -p $(DESTDIR)$(XINEPLUGINDIR)/post
-	@echo Installing $(DESTDIR)$(XINEPLUGINDIR)/$(XINEINPUTVDR)
-	@-rm -rf $(DESTDIR)$(XINEPLUGINDIR)/$(XINEINPUTVDR)
-	@$(INSTALL) -m 0644 $(XINEINPUTVDR) $(DESTDIR)$(XINEPLUGINDIR)/$(XINEINPUTVDR)
-	@echo Installing $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
-	@-rm -rf $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
-	@$(INSTALL) -m 0644 $(XINEPOSTAUTOCROP) $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
-	@echo Installing $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
-	@-rm -rf $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
-	@$(INSTALL) -m 0644 $(XINEPOSTSWSCALE) $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
-	@echo Installing $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
-	@-rm -rf $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
-	@$(INSTALL) -m 0644 $(XINEPOSTAUDIOCHANNEL) $(DESTDIR)$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
+	@mkdir -p $(DESTDIR)/$(XINEPLUGINDIR)/post
+	@echo Installing $(DESTDIR)/$(XINEPLUGINDIR)/$(XINEINPUTVDR)
+	@-rm -rf $(DESTDIR)/$(XINEPLUGINDIR)/$(XINEINPUTVDR)
+	@$(INSTALL) -m 0644 $(XINEINPUTVDR) $(DESTDIR)/$(XINEPLUGINDIR)/$(XINEINPUTVDR)
+	@echo Installing $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
+	@-rm -rf $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
+	@$(INSTALL) -m 0644 $(XINEPOSTAUTOCROP) $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUTOCROP)
+	@echo Installing $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
+	@-rm -rf $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
+	@$(INSTALL) -m 0644 $(XINEPOSTSWSCALE) $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTSWSCALE)
+	@echo Installing $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
+	@-rm -rf $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
+	@$(INSTALL) -m 0644 $(XINEPOSTAUDIOCHANNEL) $(DESTDIR)/$(XINEPLUGINDIR)/post/$(XINEPOSTAUDIOCHANNEL)
+endif
+ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
+  ifeq ($(VDR_TREE), no)
+	@echo Installing $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN)
+	@mkdir -p $(DESTDIR)$(LIBDIR)
+	@-rm -rf $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN)
+	@$(INSTALL) -D -m 0755 $(VDRPLUGIN) $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN)
+  endif
 endif
 ifeq ($(XINELIBOUTPUT_FB), yes)
-	@echo Installing $(DESTDIR)$(BINDIR)/vdr-fbfe
-	@mkdir -p $(DESTDIR)$(BINDIR)
-	@-rm -rf $(DESTDIR)$(BINDIR)/vdr-fbfe
-	@$(INSTALL) -m 0755 vdr-fbfe $(DESTDIR)$(BINDIR)/vdr-fbfe
+	@echo Installing $(DESTDIR)/$(BINDIR)/vdr-fbfe
+	@mkdir -p $(DESTDIR)/$(BINDIR)
+	@-rm -rf $(DESTDIR)/$(BINDIR)/vdr-fbfe
+	@$(INSTALL) -m 0755 vdr-fbfe $(DESTDIR)/$(BINDIR)/vdr-fbfe
+  ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
+    ifeq ($(VDR_TREE), no)
+	@echo Installing $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_FBFE)
+	@mkdir -p $(DESTDIR)$(LIBDIR)
+	@-rm -rf $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_FBFE)
+	@$(INSTALL) -m 0755 $(VDRPLUGIN_FBFE) $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_FBFE)
+    endif
+  endif
 endif
 ifeq ($(XINELIBOUTPUT_X11), yes)
-	@echo Installing $(DESTDIR)$(BINDIR)/vdr-sxfe
-	@mkdir -p $(DESTDIR)$(BINDIR)
-	@-rm -rf $(DESTDIR)$(BINDIR)/vdr-sxfe
-	@$(INSTALL) -m 0755 vdr-sxfe $(DESTDIR)$(BINDIR)/vdr-sxfe
+	@echo Installing $(DESTDIR)/$(BINDIR)/vdr-sxfe
+	@mkdir -p $(DESTDIR)/$(BINDIR)
+	@-rm -rf $(DESTDIR)/$(BINDIR)/vdr-sxfe
+	@$(INSTALL) -m 0755 vdr-sxfe $(DESTDIR)/$(BINDIR)/vdr-sxfe
+  ifeq ($(XINELIBOUTPUT_VDRPLUGIN), yes)
+    ifeq ($(VDR_TREE), no)
+	@echo Installing $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_SXFE)
+	@mkdir -p $(DESTDIR)$(LIBDIR)
+	@-rm -rf $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_SXFE)
+	@$(INSTALL) -m 0755 $(VDRPLUGIN_SXFE) $(DESTDIR)$(LIBDIR)/$(VDRPLUGIN_SXFE)
+    endif
+  endif
 endif
 
-dist: $(I18Npo) clean
+dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
 	@mkdir $(TMPDIR)/$(ARCHIVE)
 	@cp -a * $(TMPDIR)/$(ARCHIVE)
@@ -433,4 +471,6 @@
 		xine/*.flc $(VDR_FBFE) $(VDR_SXFE) mpg2c black_720x576.c \
 		nosignal_720x576.c vdrlogo_720x576.c vdr-sxfe vdr-fbfe \
 		features.h config.mak configure.log
+ifeq ($(HAVE_I18N), yes)
 	@-rm -f $(PODIR)/*.mo $(PODIR)/*.pot
+endif
