--- a/src/sna/sna_driver.c	2015-09-29 15:00:33.279063283 +0200
+++ b/src/sna/sna_driver.c	2015-09-29 15:00:33.295062939 +0200
@@ -59,6 +59,7 @@
 
 #include <sys/ioctl.h>
 #include <sys/fcntl.h>
+#include <sys/stat.h>
 #include <sys/poll.h>
 #include "i915_drm.h"
 
--- a/src/uxa/intel_driver.c	2015-09-29 15:11:52.383051759 +0200
+++ b/src/uxa/intel_driver.c	2015-09-29 15:11:52.411051665 +0200
@@ -44,6 +44,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <errno.h>
+#include <sys/stat.h>
 
 #include "xorg-server.h"
 #include "xf86.h"