diff -Naur xf86-video-vmware-10.16.8.orig/src/vmware.c xf86-video-vmware-10.16.8/src/vmware.c
--- xf86-video-vmware-10.16.8.orig/src/vmware.c	2009-10-03 19:50:22.000000000 +0200
+++ xf86-video-vmware-10.16.8/src/vmware.c	2009-10-03 19:51:15.000000000 +0200
@@ -640,8 +640,6 @@
         return FALSE;
     }
 
-    xf86LoaderReqSymLists(vgahwSymbols, NULL);
-
     if (!vgaHWGetHWRec(pScrn)) {
         return FALSE;
     }
@@ -959,7 +957,6 @@
         VMWAREFreeRec(pScrn);
         return FALSE;
     }
-    xf86LoaderReqSymLists(fbSymbols, shadowfbSymbols, NULL);
 
     /* Need ramdac for hwcursor */
     if (pVMWARE->hwCursor) {
@@ -967,7 +964,6 @@
             VMWAREFreeRec(pScrn);
             return FALSE;
         }
-        xf86LoaderReqSymLists(ramdacSymbols, NULL);
     }
 
     return TRUE;
@@ -2051,9 +2047,6 @@
         setupDone = TRUE;
         xf86AddDriver(&VMWARE, module, VMWARE_DRIVER_FUNC);
 
-        LoaderRefSymLists(vgahwSymbols, fbSymbols, ramdacSymbols,
-                          shadowfbSymbols, NULL);
-
         return (pointer)1;
     }
     if (errmaj) {
