diff -Naur a/udev/rules.d/98-eventlircd.rules.disabled.in b/udev/rules.d/98-eventlircd.rules.disabled.in
--- a/udev/rules.d/98-eventlircd.rules.disabled.in	2011-10-23 10:13:04.000000000 +0200
+++ b/udev/rules.d/98-eventlircd.rules.disabled.in	2013-11-11 12:37:08.998439128 +0100
@@ -54,7 +54,7 @@
 # Ask eventlircd to handle input event devices created by ir-core (kernel).
 #-------------------------------------------------------------------------------
 SUBSYSTEMS=="rc", \
-  ENV{eventlircd_enable}="true"
+  ENV{eventlircd_enable}="true", \
   ENV{eventlircd_evmap}="ircore.evmap"
 
 #-------------------------------------------------------------------------------
@@ -69,7 +69,7 @@
 ENV{ID_USB_INTERFACES}=="", IMPORT{program}="usb_id --export %p"
 
 ENV{ID_VENDOR_ID}=="0419", ENV{ID_MODEL_ID}=="0001", \
-  ENV{eventlircd_enable}="true",
+  ENV{eventlircd_enable}="true", \
   ENV{eventlircd_evmap}="03_$env{ID_VENDOR_ID}_$env{ID_MODEL_ID}.evmap"
 
 ENV{ID_VENDOR_ID}=="05a4", ENV{ID_MODEL_ID}=="9881", \
