diff -Naur eventlircd-r44/udev/rules.d/98-eventlircd.rules.disabled.in eventlircd-r44-new/udev/rules.d/98-eventlircd.rules.disabled.in
--- eventlircd-r44/udev/rules.d/98-eventlircd.rules.disabled.in	2011-10-23 10:13:04.000000000 +0200
+++ eventlircd-r44-new/udev/rules.d/98-eventlircd.rules.disabled.in	2013-11-11 12:37:08.998439128 +0100
@@ -58,6 +58,14 @@
   ENV{eventlircd_evmap}="ircore.evmap"
 
 #-------------------------------------------------------------------------------
+# Ask eventlircd to handle snvs_power_key
+#-------------------------------------------------------------------------------
+
+ATTRS{name}=="snvs_power_key", \
+  ENV{eventlircd_enable}="true"
+  ENV{eventlircd_evmap}="ircore.evmap"
+
+#-------------------------------------------------------------------------------
 # Ask eventlircd to handle USB HID devices that show up as event devices and are
 # known to be remote controls. For simplicity, the event map file names have the
 # format <BUSTYPE>_<VENDOR>_<PRODUCT>.evmap.
