--- a/udev/rules.d/98-eventlircd.rules.disabled.in	2014-06-14 07:43:22.388073799 +0200
+++ b/udev/rules.d/98-eventlircd.rules.disabled.in	2014-06-14 07:46:18.700068005 +0200
@@ -95,6 +95,10 @@
 ENV{ID_VENDOR_ID}=="147a", ENV{ID_MODEL_ID}=="e02d", \
   ENV{eventlircd_enable}="true", \
   ENV{eventlircd_evmap}="03_$env{ID_VENDOR_ID}_$env{ID_MODEL_ID}.evmap"
+  
+ENV{ID_VENDOR_ID}=="6253", ENV{ID_MODEL_ID}=="0100", \
+  ENV{eventlircd_enable}="true", \
+  ENV{eventlircd_evmap}="03_$env{ID_VENDOR_ID}_$env{ID_MODEL_ID}.evmap"
 
 # Enable wake-on-usb for the USB remotes.
 ENV{eventlircd_enable}=="true", RUN+="wakeup_enable"
