--- iptraf-3.0.0/src/install.sh.old	2002-12-26 04:07:18.000000000 +0100
+++ iptraf-3.0.0/src/install.sh	2013-11-11 16:17:09.000133916 +0100
@@ -14,11 +14,15 @@
 fi
 
 INSTALL=/usr/bin/install
-TARGET=$1
-WORKDIR=$2
-LOGDIR=$3
-LOCKDIR=$4
-
+TARGET=./.install$1
+WORKDIR=./.install$2
+LOGDIR=./.install$3
+LOCKDIR=./.install$4
+
+mkdir -p ./.install$1 || true
+mkdir -p ./.install$2 || true
+mkdir -p ./.install$3 || true
+mkdir -p ./.install$4 || true
 echo
 echo "*** Installing executable programs and preparing work directories"
 echo
@@ -74,35 +78,5 @@
 fi
 ####################################################################
 
-cat version > $WORKDIR/version
-
-echo 
-echo
-
-echo "======================================================================"
-echo
-echo "Please read the RELEASE-NOTES file for important new information about"
-echo "this version. You can view this file now (will require the 'less'"
-echo "program in /usr/bin.  Press Q to quit when done)."
-echo
-echo -n "Would you like to view the RELEASE-NOTES file now (Y/N)? "; read YESNO
-
-if [ "$YESNO" = "y" -o "$YESNO" = "Y" ]; then
-    less ../RELEASE-NOTES
-fi
-
-clear
-echo
-echo "====================================================================="
-echo
-echo "Thank you for installing IPTraf.  You can now start IPTraf by issuing"
-echo "the command"
-echo
-echo "    $TARGET/iptraf"
-echo
-echo "at your shell prompt.  You can also add $TARGET to your PATH environment"
-echo "variable to avoid having to type the pathname when invoking the program."
-echo
-
 exit 0
 
