--- iptraf-3.0.0/src/Makefile.mod	2013-11-11 16:22:18.848139233 +0100
+++ iptraf-3.0.0/src/Makefile	2013-11-11 16:22:37.028139546 +0100
@@ -68,12 +68,12 @@
 # installation target directory.  The iptraf and rvnamed programs get
 # stored here.  iptraf also exec's rvnamed from this directory.
 
-TARGET		= /usr/local/bin
+TARGET		= /usr/bin
 
 # The IPTraf working directory; if you change this.  Starting with this
 # version, this definition overrides dirs.h.
 
-WORKDIR		= /var/local/iptraf
+WORKDIR		= /var/iptraf
 
 # The IPTraf log file directory.  IPTraf log files are placed here.
 # This definition overrides dirs.h
