--- iptraf-3.0.0/src/hostmon.c.org      2014-10-19 11:06:30.076416029 +0200
+++ iptraf-3.0.0/src/hostmon.c  2014-10-19 11:07:59.544417541 +0200
@@ -31,7 +31,7 @@
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <net/if_arp.h>
 #include <stdlib.h>
 #include <time.h>
--- iptraf-3.0.0/src/othptab.c.org      2014-10-19 11:06:37.720416158 +0200
+++ iptraf-3.0.0/src/othptab.c  2014-10-19 11:08:31.088418074 +0200
@@ -18,7 +18,7 @@

 #include <asm/types.h>
 #include <linux/if_ether.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <linux/if_fddi.h>
 #include <winops.h>
 #include "arphdr.h"
--- iptraf-3.0.0/src/packet.c.org       2014-10-19 11:06:44.788416278 +0200
+++ iptraf-3.0.0/src/packet.c   2014-10-19 11:08:48.596418370 +0200
@@ -36,7 +36,7 @@
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <linux/isdn.h>
 #include <linux/sockios.h>
 #include <msgboxes.h>
--- iptraf-3.0.0/src/tcptable.h.org     2014-10-19 11:06:53.320416422 +0200
+++ iptraf-3.0.0/src/tcptable.h 2014-10-19 11:09:07.884418696 +0200
@@ -23,7 +23,7 @@
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
--- iptraf-3.0.0/src/tr.c.org   2014-10-19 11:06:58.556416510 +0200
+++ iptraf-3.0.0/src/tr.c       2014-10-19 11:09:23.000418951 +0200
@@ -7,7 +7,7 @@
  */

 #include <asm/types.h>
-#include <linux/if_tr.h>
+#include <netinet/if_tr.h>
 #include <netinet/in.h>

 unsigned int get_tr_ip_offset(unsigned char *pkt)
