diff -Naur a/src/osd/sdl/sdl.mak b/src/osd/sdl/sdl.mak
--- a/src/osd/sdl/sdl.mak	2011-06-26 08:13:06.000000000 +0200
+++ b/src/osd/sdl/sdl.mak	2011-06-26 08:28:12.276979839 +0200
@@ -286,9 +286,9 @@
 	$(SDLOBJ)/watchdog.o
 
 # Add SDL1.3 support
-ifdef SDL_INSTALL_ROOT
-OSDOBJS += $(SDLOBJ)/draw13.o
-endif
+# ifdef SDL_INSTALL_ROOT
+# OSDOBJS += $(SDLOBJ)/draw13.o
+# endif
 
 # add an ARCH define
 DEFS += "-DSDLMAME_ARCH=$(ARCHOPTS)" -DSYNC_IMPLEMENTATION=$(SYNC_IMPLEMENTATION)
