Index: vdr-plugin-osdpip-0.1.1/receiver.c
===================================================================
--- vdr-plugin-osdpip-0.1.1.orig/receiver.c	2012-03-28 22:17:25.433910311 +0200
+++ vdr-plugin-osdpip-0.1.1/receiver.c	2012-03-28 22:20:13.513910412 +0200
@@ -35,7 +35,9 @@
 
 cOsdPipReceiver::cOsdPipReceiver(const cChannel *Channel,
     cRingBufferFrame *ESBuffer):
-#if (APIVERSNUM < 10500)
+#if VDRVERSNUM >= 10727
+    cReceiver(Channel, 0),
+#elif (APIVERSNUM < 10500)
     cReceiver(Channel->Ca(), 0, Channel->Vpid()),
 #else
     cReceiver(Channel->GetChannelID(), 0, Channel->Vpid()),
