diff -ruN osdpip-org//Makefile osdpip-mod//Makefile
--- osdpip-org//Makefile	2012-02-12 13:41:01.000000000 +0100
+++ osdpip-mod//Makefile	2012-04-02 21:14:18.000000000 +0200
@@ -48,7 +48,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+CXXFLAGS ?= -g -O2 -Wall -Werror=overloaded-virtual -Wno-parentheses
 
 ### The directory environment:
 
@@ -107,7 +107,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o osd_info.o remux.o osd.o receiver.o setup.o i18n.o pes.o quantize.o decoder.o
+OBJS = $(PLUGIN).o osd_info.o remux.o osd.o receiver.o setup.o pes.o quantize.o decoder.o
 
 ### The main target:
 
diff -ruN osdpip-org//osd_info.c osdpip-mod//osd_info.c
--- osdpip-org//osd_info.c	2012-02-12 13:41:01.000000000 +0100
+++ osdpip-mod//osd_info.c	2012-04-02 21:16:44.717626252 +0200
@@ -21,7 +21,6 @@
 
 #include "osd_info.h"
 #include "setup.h"
-#include "i18n.h"
 
 #include <vdr/device.h>
 #include <vdr/thread.h>
diff -ruN osdpip-org//osdpip.c osdpip-mod//osdpip.c
--- osdpip-org//osdpip.c	2012-02-12 13:41:01.000000000 +0100
+++ osdpip-mod//osdpip.c	2012-04-02 21:14:40.000000000 +0200
@@ -23,9 +23,6 @@
 #include "decoder.h"
 #include "osd.h"
 #include "setup.h"
-#if (APIVERSNUM < 10507)
-#include "i18n.h"
-#endif
 
 #include <vdr/plugin.h>
 
@@ -83,9 +80,6 @@
 
 bool cPluginOsdpip::Start(void)
 {
-#if (APIVERSNUM < 10507)
-    RegisterI18n(Phrases);
-#endif
     return true;
 }
 
diff -ruN osdpip-org//setup.c osdpip-mod//setup.c
--- osdpip-org//setup.c	2012-02-12 13:41:01.000000000 +0100
+++ osdpip-mod//setup.c	2012-04-02 21:20:28.818737529 +0200
@@ -21,7 +21,6 @@
 #include <vdr/config.h>
 
 #include "setup.h"
-#include "i18n.h"
 
 const int kColorDepths = 5;
 const int kSizes = 11;
