diff -Naur qt-mobility-opensource-src-1.1.0-beta2.orig//configure qt-mobility-opensource-src-1.1.0-beta2/configure
--- qt-mobility-opensource-src-1.1.0-beta2.orig//configure	2010-11-07 16:29:11.752427999 +0100
+++ qt-mobility-opensource-src-1.1.0-beta2/configure	2010-11-07 16:29:32.012428009 +0100
@@ -161,6 +161,7 @@
     echo "                    example: -sdk /Developer/SDKs/MacOSX10.6.sdk"
     echo "-qmake-exec <name>  Sets custom binary name for qmake binary"
     echo "                    (default: qmake)"
+    echo "-qmake-specs <name> Sets custom specs name for qmake binary"
     echo
     
     rm -f "$CONFIG_IN"
@@ -245,6 +246,10 @@
             QMAKE_EXEC="$2"
             shift
             ;;
+        -qmake-specs)
+            QMKSPEC="-spec $2"
+            shift
+            ;;
         -modules)
             MOBILITY_MODULES_UNPARSED=$2
             #reset default selection
