diff -Naur qt-mobility-opensource-src-1.2.0.orig//configure qt-mobility-opensource-src-1.2.0/configure
--- qt-mobility-opensource-src-1.2.0.orig//configure	2011-05-18 20:21:20.607063999 +0200
+++ qt-mobility-opensource-src-1.2.0/configure	2011-05-18 20:22:29.597064000 +0200
@@ -164,6 +164,7 @@
     echo "                    (default is empty)"
     echo "-qmake-exec <name>  Sets custom binary name for qmake binary"
     echo "                    (default: qmake)"
+    echo "-qmake-specs <name> Sets custom specs name for qmake binary"
     echo "-no-nfc-symbian     Disables the NFC Sybmian backend."
 #    echo "-test-sim           Use simulator backend for testing systeminfo"
 #    echo "-staticconfig <name>"
@@ -258,6 +259,10 @@
             QMAKE_EXEC="$2"
             shift
             ;;
+        -qmake-specs)
+            QMKSPEC="-spec $2"
+            shift
+            ;;
         -modules)
             MOBILITY_MODULES_UNPARSED=$2
             ORGANIZER_REQUESTED=
