diff -Naur qt-mobility-opensource-src-1.1.0-beta2.orig//plugins/declarative/messaging/messaging.pro qt-mobility-opensource-src-1.1.0-beta2/plugins/declarative/messaging/messaging.pro
--- qt-mobility-opensource-src-1.1.0-beta2.orig//plugins/declarative/messaging/messaging.pro	2010-11-07 16:29:11.532427998 +0100
+++ qt-mobility-opensource-src-1.1.0-beta2/plugins/declarative/messaging/messaging.pro	2010-11-07 16:46:16.592427999 +0100
@@ -37,7 +37,7 @@
         QMAKE_LFLAGS += -F$$(QMF_LIBDIR)
         LIBS += -framework qtopiamail
     } else {
-        LIBS += -L$$(QMF_LIBDIR) -l$$qtLibraryTarget(qtopiamail)
+        LIBS += -L$$(QMF_LIBDIR)
     }
 }
 
