diff -ruN 1.7.41_24/menu.c 1.7.41_26/menu.c
--- 1.7.41_24/menu.c	2013-03-18 22:41:54.000000000 +0100
+++ 1.7.41_26/menu.c	2013-03-18 22:46:57.000000000 +0100
@@ -3544,7 +3544,7 @@
 {
   Clear();
   char buffer[64];
-  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("Setup"), VDRVERSION);
+  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("VDR Setup"), VDRVERSION);
   SetTitle(buffer);
   SetHasHotkeys();
   Add(new cOsdItem(hk(tr("OSD")),           osUser1));
@@ -3622,9 +3622,9 @@
 :cOsdMenu("")
 {
   // Load Menu Configuration
-  cString menuXML = cString::sprintf("%s/setup/vdr-menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
+  cString menuXML = cString::sprintf("%s/../menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
   if (access(menuXML, 04) == -1)
-     menuXML = cString::sprintf("%s/setup/vdr-menu.xml", cPlugin::ConfigDirectory());
+     menuXML = cString::sprintf("%s/../menu.xml", cPlugin::ConfigDirectory());
   subMenu.LoadXml(menuXML);
   nrDynamicMenuEntries = 0;
   SetMenuCategory(mcMain);
@@ -3709,7 +3709,7 @@
          else if (strcmp(item, "Recordings") == 0)
             Add(new cOsdItem(hk(tr("Recordings")), osRecordings));
          else if (strcmp(item, "Setup") == 0) {
-            cString itemSetup = cString::sprintf("%s%s", tr("Setup"), *subMenu.GetMenuSuffix());
+            cString itemSetup = cString::sprintf("%s%s", tr("VDR Setup"), *subMenu.GetMenuSuffix());
             Add(new cOsdItem(hk(itemSetup), osSetup));
             }
          else if (strcmp(item, "Commands") == 0 && Commands.Count() > 0) {
diff -ruN 1.7.41_24/menuitems.c 1.7.41_26/menuitems.c
--- 1.7.41_24/menuitems.c	2013-02-15 15:20:29.000000000 +0100
+++ 1.7.41_26/menuitems.c	2013-03-18 22:44:00.000000000 +0100
@@ -1132,7 +1132,7 @@
 
 void cMenuSetupPage::SetSection(const char *Section)
 {
-  SetTitle(cString::sprintf("%s - %s", tr("Setup"), Section));
+  SetTitle(cString::sprintf("%s - %s", tr("VDR Setup"), Section));
 }
 
 eOSState cMenuSetupPage::ProcessKey(eKeys Key)
