--- vdr-smarttvweb-0.98/Makefile.old	2013-09-17 17:43:07.905659899 +0200
+++ vdr-smarttvweb-0.98/Makefile	2013-09-17 17:41:14.821657958 +0200
@@ -16,22 +16,24 @@
 
 
 CXX      ?= g++
-#ifdef DEBUG
+ifdef DEBUG
 CXXFLAGS ?= -g -O0 -fPIC -Wall -Woverloaded-virtual #-Werror
-#else
-#CXXFLAGS ?= -fPIC -Wall -Woverloaded-virtual #-Werror
+else
+CXXFLAGS ?= -fPIC -Wall -Woverloaded-virtual #-Werror
 #CXXFLAGS ?= -O2 -fPIC -Wall -Woverloaded-virtual #-Werror
-#endif
+endif
 
 ### The directory environment:
 
-VDRDIR = ../../..
-LIBDIR = ../../lib
+#VDRDIR = ../../..
+VDRDIR = /usr/include/vdr
+LIBDIR = .
+#LIBDIR = ../../lib
 TMPDIR = /tmp
 
 ### Allow user defined options to overwrite defaults:
 
--include $(VDRDIR)/Make.config
+#-include $(VDRDIR)/Make.config
 
 ### read standlone settings if there
 -include .standalone
@@ -55,7 +57,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o smarttvfactory.o httpresource.o httpclient.o mngurls.o log.o url.o stvw_cfg.o responsebase.o responsefile.o responsevdrdir.o responsememblk.o
+OBJS = $(PLUGIN).o smarttvfactory.o httpresource.o log.o url.o stvw_cfg.o
 
 OBJS2 = 
 
