--- vdr-plugin-yaepghd-0.0.3-ce-epgimages.orig/yaepghd.c
+++ vdr-plugin-yaepghd-0.0.3-ce-epgimages/yaepghd.c
@@ -2967,7 +2967,7 @@ cYaepghd::Show(void)
                          BG_IMAGE->Bpp());
    osd->SetAreas(&mainWin, 1);
 
-#ifdef YAEPGHDVERSNUM
+
    /* Set up the video window parameters */
    if (VID_WIN_GEOM.w != 0 && VID_WIN_GEOM.h != 0) {
       osd->vidWin.x1 = VID_WIN_GEOM.x;
@@ -2976,7 +2976,7 @@ cYaepghd::Show(void)
       osd->vidWin.y2 = VID_WIN_GEOM.y + VID_WIN_GEOM.h;
       osd->vidWin.bpp = 12;
    }
-#endif
+
 
 #ifdef YAEPGHD_REEL_EHD
    reelVidWin->Open(VID_WIN_GEOM);

