--- a/yaepghd.c
+++ b/yaepghd.c
@@ -229,7 +229,7 @@
 	const cEvent	*event;
 //out
 	cTimer		*timer;
-	int		timerMatch;
+	eTimerMatch		timerMatch;
 	int		timerType;
 	bool		isRemote;
 };
@@ -1720,7 +1720,7 @@
    const char* v=NULL;
    const char* r=NULL;
    
-   int timerMatch=tmNone; 
+   eTimerMatch timerMatch=tmNone; 
    cTimer* ti;
 
    if (iRemoteTimer && pRemoteTimers && event) {  
@@ -3019,7 +3019,7 @@
 cYaepghd::AddDelTimer(void)
  {
       const cEvent *event=gridEvents->Event();
-	  int timerMatch;
+	  eTimerMatch timerMatch;
 	  cTimer *ti;
 	  ti=Timers.GetMatch(event, &timerMatch);
 	  if (timerMatch==tmFull)
@@ -3245,7 +3245,7 @@
                state = osEnd;
 		 }
          else{
-		    int timerMatch;
+		    eTimerMatch timerMatch;
 			Timers.GetMatch(event, &timerMatch);
 		    if (!(timerMatch==tmFull)){
 				
@@ -3275,7 +3275,7 @@
 		    }
  	        else {
 			   AddDelTimer();  // delete timer
-			   int timerMatch;
+			   eTimerMatch timerMatch;
 			   Timers.GetMatch(event, &timerMatch);
                if (timerMatch==tmNone){	
 			      messageBox = new cYaepgMsg();
@@ -3291,7 +3291,7 @@
       case kRed:
          if (event && event->EventID()!=0){
             if (iRecDlgRed) {
-               int timerMatch;
+               eTimerMatch timerMatch;
 			   Timers.GetMatch(event, &timerMatch);
                if (!(timerMatch==tmFull)){				
 	              if (iRemoteTimer && pRemoteTimers) {
@@ -3320,7 +3320,7 @@
 		       }
  	           else {
 			      AddDelTimer();  // delete timer
-			      int timerMatch;
+			      eTimerMatch timerMatch;
 			      Timers.GetMatch(event, &timerMatch);
                   if (timerMatch==tmNone){	
 			         messageBox = new cYaepgMsg();
