--- gdb-6.1.1.orig/bfd/config.bfd	Sat Mar 13 14:37:09 2004
+++ gdb-6.1.1/bfd/config.bfd	Tue Dec  7 17:49:50 2004
@@ -488,6 +488,10 @@
     targ_selvecs="i386linux_vec bfd_efi_app_ia32_vec"
     targ64_selvecs=bfd_elf64_x86_64_vec
     ;;
+  i[3-7]86-*-linux-uclibc*)
+    targ_defvec=bfd_elf32_i386_vec
+    targ_selvecs="i386linux_vec bfd_efi_app_ia32_vec"
+    ;;
 #ifdef BFD64
   x86_64-*-freebsd* | x86_64-*-kfreebsd*-gnu)
     targ_defvec=bfd_elf64_x86_64_vec
