diff -Naur fbi-1.31.orig/fbi.c fbi-1.31/fbi.c
--- fbi-1.31.orig/fbi.c	2004-04-03 00:20:19.000000000 +0200
+++ fbi-1.31/fbi.c	2004-04-03 00:55:02.000000000 +0200
@@ -942,7 +942,9 @@
 	    redraw = 1;
 	    left += steps;
 
-	} else if (0 == strcmp(key, "\x1b[5~")) {
+	} else if (0 == strcmp(key, "\x1b[5~") ||
+                   0 == strcmp(key, "b")       ||
+                   0 == strcmp(key, "B")) {
 	    return KEY_PGUP;
 	} else if (0 == strcmp(key, "\x1b[6~") ||
 		   0 == strcmp(key, "n")       ||
diff -Naur fbi-1.31.orig/lirc.c fbi-1.31/lirc.c
--- fbi-1.31.orig/lirc.c	2004-04-03 00:20:19.000000000 +0200
+++ fbi-1.31/lirc.c	2004-04-03 00:20:31.000000000 +0200
@@ -22,7 +22,7 @@
 	    fprintf(stderr,"lirc: no infrared remote support available\n");
 	return -1;
     }
-    if (0 != lirc_readconfig(NULL,&config,NULL)) {
+    if (0 != lirc_readconfig("/etc/lircrc",&config,NULL)) {
 	config = NULL;
     }
     if (debug)
