diff -Naur fbi-1.31.orig/fbtools.c fbi-1.31/fbtools.c
--- fbi-1.31.orig/fbtools.c	2004-12-30 01:56:26.767298760 +0100
+++ fbi-1.31/fbtools.c	2004-12-30 01:56:36.956749728 +0100
@@ -265,6 +265,8 @@
 	fprintf(stderr,"access %s: %s\n",vtname,strerror(errno));
 	exit(1);
     }
+
+    /* do not fork!
     switch (fork()) {
     case 0:
 	break;
@@ -274,6 +276,8 @@
     default:
 	exit(0);
     }
+    */
+
     close(tty);
     close(0);
     close(1);
