--- fbi-1.31.orig/mk/Variables.mk	Wed Nov 17 14:54:13 2004
+++ fbi-1.31/mk/Variables.mk	Wed Nov 17 14:54:29 2004
@@ -33,12 +33,6 @@
 CFLAGS	+= -Wall -Wmissing-prototypes -Wstrict-prototypes \
 	   -Wpointer-arith -Wunused
 
-# add /usr/local to the search path if something is in there ...
-ifneq ($(wildcard /usr/local/include/*.h),)
-  CFLAGS  += -I/usr/local/include
-  LDFLAGS += -L/usr/local/$(LIB)
-endif
-
 # fixup include path for $(srcdir) != "."
 ifneq ($(srcdir),.)
   CFLAGS  += -I. -I$(srcdir)
