diff -Nur fbi-1.31.orig/fbi.c fbi-1.31/fbi.c
--- fbi-1.31.orig/fbi.c	2005-12-18 17:25:44.000000000 +0800
+++ fbi-1.31/fbi.c	2005-12-19 00:58:33.000000000 +0800
@@ -923,10 +923,12 @@
 		return KEY_SPACE;
 	    }
 
-	} else if (0 == strcmp(key, "\x1b[A") && img->i.height > fb_var.yres) {
+	} else if (( 0 == strcmp(key, "\x1b[A") || 0 == strcmp(key, "u"))
+			&& img->i.height > fb_var.yres) {
 	    redraw = 1;
 	    top -= steps;
-	} else if (0 == strcmp(key, "\x1b[B") && img->i.height > fb_var.yres) {
+	} else if (( 0 == strcmp(key, "\x1b[B") || 0 == strcmp(key, "d"))
+			&& img->i.height > fb_var.yres) {
 	    redraw = 1;
 	    top += steps;
 	} else if (0 == strcmp(key, "\x1b[1~") && img->i.height > fb_var.yres) {
@@ -935,10 +937,12 @@
 	} else if (0 == strcmp(key, "\x1b[4~")) {
 	    redraw = 1;
 	    top = img->i.height - fb_var.yres;
-	} else if (0 == strcmp(key, "\x1b[D") && img->i.width > fb_var.xres) {
+	} else if ((0 == strcmp(key, "\x1b[D") || 0 == strcmp(key, "<"))
+			&& img->i.width > fb_var.xres) {
 	    redraw = 1;
 	    left -= steps;
-	} else if (0 == strcmp(key, "\x1b[C") && img->i.width > fb_var.xres) {
+	} else if ((0 == strcmp(key, "\x1b[C") || 0 == strcmp(key, ">"))
+			&& img->i.width > fb_var.xres) {
 	    redraw = 1;
 	    left += steps;
 
