diff -Naur fbi-1.31a/fbtools.c fbi-1.31/fbtools.c
--- fbi-1.31a/fbtools.c	2009-06-10 16:20:35.000000000 +0200
+++ fbi-1.31/fbtools.c	2009-06-10 16:22:47.000000000 +0200
@@ -21,8 +21,6 @@
 #include <linux/vt.h>
 #include <linux/fb.h>
 
-#include <asm/page.h>
-
 #include "fbtools.h"
 
 /* -------------------------------------------------------------------- */
@@ -428,7 +426,7 @@
 	goto err;
     }
 #endif
-    fb_mem_offset = (unsigned long)(fb_fix.smem_start) & (~PAGE_MASK);
+    fb_mem_offset = (unsigned long)(fb_fix.smem_start) & ~(sysconf(_SC_PAGE_SIZE)-1);
     fb_mem = mmap(NULL,fb_fix.smem_len+fb_mem_offset,
 		  PROT_READ|PROT_WRITE,MAP_SHARED,fb,0);
     if (-1L == (long)fb_mem) {
