diff -Naur fbi-1.31a/fbi.c fbi-1.31/fbi.c
--- fbi-1.31a/fbi.c	2009-06-14 15:00:44.000000000 +0200
+++ fbi-1.31/fbi.c	2009-06-14 15:01:01.000000000 +0200
@@ -407,7 +407,8 @@
 	    continue;
 	}
 	title[tag] = exif_tag_get_title(tags[tag]);
-	value[tag] = strdup(exif_entry_get_value(ee));
+	value[tag] = malloc(128);
+	exif_entry_get_value(ee, value[tag], 128);
 	len = strlen(title[tag]);
 	if (l1 < len)
 	    l1 = len;
