--- irmplircd-0.1/Makefile.org	2013-01-06 11:17:33.000000000 +0100
+++ irmplircd-0.1/Makefile	2013-04-21 18:57:36.505833093 +0200
@@ -4,7 +4,7 @@
 
 CC ?= gcc
 CFLAGS ?= -Wall -g -O2 -pipe -Ic_hashmap #-DDEBUG
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 INSTALL ?= install
 STRIP ?= strip
 SBINDIR ?= $(PREFIX)/sbin
@@ -35,11 +35,12 @@
 install: install-sbin install-man
 
 install-sbin: $(SBIN_IRMPLIRCD) $(SBIN_IRMPEXEC)
-	mkdir -p $(SBINDIR)
+	mkdir -p ./.install/$(SBINDIR)
+	mkdir -p ./.install/$(BINDIR)
 	$(STRIP) $(SBIN_IRMPLIRCD)
 	$(STRIP) $(SBIN_IRMPEXEC)
-	$(INSTALL) $(SBIN_IRMPLIRCD) $(SBINDIR)/
-	$(INSTALL) $(SBIN_IRMPEXEC) $(BINDIR)/
+	$(INSTALL) $(SBIN_IRMPLIRCD) ./.install/$(SBINDIR)/
+	$(INSTALL) $(SBIN_IRMPEXEC) ./.install/$(BINDIR)/
 
 install-man: $(MAN1) $(MAN5) $(MAN8)
 	mkdir -p $(DESTDIR)$(MANDIR)/man8/
