--- a/ld/ldfile.c	2015-11-13 09:27:42.000000000 +0100
+++ b/ld/ldfile.c	2016-11-15 19:09:31.658371254 +0100
@@ -102,6 +102,17 @@ ldfile_add_library_path (const char *nam
   if (!cmdline && config.only_cmd_line_lib_dirs)
     return;
 
+  /* skip those directories when linking */
+  if ((!strncmp (name, "/lib", 4)) ||
+      (!strncmp (name, "/usr/lib", 8)) ||
+      (!strncmp (name, "/usr/local/lib", 14)) ||
+      (!strncmp (name, "/usr/X11R6/lib", 14)))
+  {
+    einfo (_("%P: warning: library search path \"%s\" is unsafe for "
+             "cross-compilation, ignore it\n"), name);
+    return;
+  }
+
   new_dirs = (search_dirs_type *) xmalloc (sizeof (search_dirs_type));
   new_dirs->next = NULL;
   new_dirs->cmdline = cmdline;
