--- riff-loadercomm-0.4.2+rcf4825b61e/lcmodule/Makefile.orig	2012-04-14 11:15:38.110295675 +0300
+++ riff-loadercomm-0.4.2+rcf4825b61e/lcmodule/Makefile	2012-04-14 11:19:46.039521497 +0300
@@ -98,10 +98,10 @@
 endif
 
 $(LIB_x32): $(LIBOBJ_x32)
-	$(CC) -shared -m32 -Wl,-soname,$(LIB_x32) -o $(LIB_x32) -lc $(addprefix $(BUILDFOLDER)/$(LIB_x32_OBJ_DIR)/, $(^F))
+	$(CC) -shared -m32 -Wl,-soname,$(LIB_x32) -o $(LIB_x32) -lc -lrt -ldl $(addprefix $(BUILDFOLDER)/$(LIB_x32_OBJ_DIR)/, $(^F))
 
 $(LIB_x64): $(LIBOBJ_x64)
-	$(CC) -shared -Wl,-soname,$(LIB_x64) -o $(LIB_x64) -lc $(addprefix $(BUILDFOLDER)/$(LIB_x64_OBJ_DIR)/, $(^F))
+	$(CC) -shared -Wl,-soname,$(LIB_x64) -o $(LIB_x64) -lc -lrt -ldl $(addprefix $(BUILDFOLDER)/$(LIB_x64_OBJ_DIR)/, $(^F))
 
 $(BUILDFOLDER)/$(LIB_x32_OBJ_DIR)/%.o: ${TRANSPORT_PATH}/source/%.c $(AUTOGEN_FILES)
 	@mkdir -p $(dir $@)
