diff -Naur connman-0.61/plugins/ntpd.c connman-0.61a/plugins/ntpd.c
--- connman-0.61/plugins/ntpd.c	2010-07-25 06:50:07.000000000 +0200
+++ connman-0.61a/plugins/ntpd.c	2010-09-24 11:41:25.591375366 +0200
@@ -120,23 +120,6 @@
 	connman_task_destroy(ntpdate->task);
 }
 
-static void ntpdate_add_peer(struct ntpdate_task *ntpdate, char *peer)
-{
-	FILE *conf_file;
-
-	DBG("%s", peer);
-
-	conf_file = fdopen(ntpdate->conf_fd, "a+");
-	if (conf_file == NULL) {
-		connman_error("fdopen failed");
-		return;
-	}
-
-	fprintf(conf_file, "server %s iburst\n", peer);
-
-	fclose(conf_file);
-}
-
 static int ntpdate(void)
 {
 	int err;
@@ -158,36 +141,20 @@
 		goto error_task;
 	}
 
-	connman_task_add_argument(ntpdate->task, "-g", NULL);
 	connman_task_add_argument(ntpdate->task, "-q", NULL);
 
-	/* The servers are added through a temp configuration file */
-	ntpdate->conf_fd = g_file_open_tmp("connman.ntp.conf_XXXXXX",
-						&ntpdate->conf_path, &g_err);
-	if  (ntpdate->conf_fd == -1) {
-		err = g_err->code;
-		g_free(g_err);
-		goto error_open;
-	}
-
-	connman_task_add_argument(ntpdate->task, "-c", ntpdate->conf_path);
-
-	DBG("conf path %s", ntpdate->conf_path);
-
 	for (list = pending_peers; list; list = list->next) {
 		peer = list->data;
 
-		ntpdate_add_peer(ntpdate, peer->server);
+		connman_task_add_argument(ntpdate->task, "-p", peer->server);
 	}
 
 	for (list = peers; list; list = list->next) {
 		peer = list->data;
 
-		ntpdate_add_peer(ntpdate, peer->server);
+		connman_task_add_argument(ntpdate->task, "-p", peer->server);
 	}
 
-	close(ntpdate->conf_fd);
-
 	return connman_task_run(ntpdate->task, ntpdate_died, ntpdate,
 						NULL, NULL, NULL);
 error_open:
