diff -Naur connman-0.61/plugins/meego.c connman-0.61a/plugins/meego.c
--- connman-0.61/plugins/meego.c	2010-06-17 02:49:12.000000000 +0200
+++ connman-0.61a/plugins/meego.c	2010-09-24 11:52:22.655654297 +0200
@@ -27,7 +27,7 @@
 #include <connman/timeserver.h>
 #include <connman/plugin.h>
 
-#define MEEGO_NTP_SERVER "ntp.meego.com"
+#define MEEGO_NTP_SERVER "0.geexbox.pool.ntp.org"
 
 static int meego_init(void)
 {
