--- vtuner-2.1+hgb6fa0d2b133b/build/arm/Makefile	2012-02-14 21:22:36.000000000 +0100
+++ vtuner-2.1+hgb6fa0d2b133b/build/arm/Makefile-new	2014-06-09 10:35:47.597015898 +0200
@@ -8,8 +8,8 @@
 DRIVER=vtuner-dvb-3
 STRIP=$(STRIP-$(ARCH))
 
-all: ../../dist/$(ARCH)/vtunerd.$(ARCH) 
-	$(STRIP) ../../dist/$(ARCH)/vtunerd.$(ARCH) 
+all: ../../dist/$(ARCH)/vtunerd.$(ARCH) ../../dist/$(ARCH)/vtunerc.$(ARCH)
+	$(STRIP) ../../dist/$(ARCH)/vtunerd.$(ARCH) ../../dist/$(ARCH)/vtunerc.$(ARCH)
 	
 -include ../../Make.rules
 	
