--- a/qwtconfig.pri	2017-11-23 08:03:25.256705551 +0100
+++ b/qwtconfig.pri	2017-11-23 08:03:25.288700480 +0100
@@ -8,7 +8,7 @@
 VERSION      = $${VER_MAJ}.$${VER_MIN}.$${VER_PAT}
 
 unix {
-    INSTALLBASE    = /usr/local/qwt-$$VERSION
+    INSTALLBASE    = /usr
 }
 
 win32 {
@@ -16,8 +16,8 @@
 }
 
 target.path    = $$INSTALLBASE/lib
-headers.path   = $$INSTALLBASE/include
-doc.path       = $$INSTALLBASE/doc
+headers.path   = $$INSTALLBASE/include/qwt-$$VERSION
+doc.path       = $$INSTALLBASE/share/qwt-$$VERSION
 
 ######################################################################
 # qmake internal options
--- a/designer/designer.pro	2017-11-23 08:09:42.040978449 +0100
+++ b/designer/designer.pro	2017-11-23 08:09:42.068974010 +0100
@@ -126,7 +126,7 @@
         RESOURCES += \
             qwt_designer_plugin.qrc
 
-        target.path = $$[QT_INSTALL_PLUGINS]/designer
+        target.path = $$INSTALLBASE/lib/qt4/plugins/designer
         INSTALLS += target
     }
 }
