--- a/qwtconfig.pri	2017-11-23 08:03:25.256705551 +0100
+++ b/qwtconfig.pri	2017-11-23 08:03:25.288700480 +0100
@@ -98,7 +98,7 @@
 # QwtSVGItem.
 ######################################################################
 
-#CONFIG     += QwtSVGItem
+CONFIG     += QwtSVGItem
 
 ######################################################################
 # You can use the MathML renderer of the Qt solutions package to 
