--- startupfiles/startfiles.cfg	2012-05-24 14:55:42.000000000 +0300
+++ startupfiles.new/startfiles.cfg	2012-03-06 21:53:31.000000000 +0200
@@ -1,9 +1,9 @@
 # section     relative file name        align    address    load address
 # The address is the exact offset into the image.
 # Partition 2 (VFAT) starts at 0x400000
-ISSW          /boot/boot_image_issw.bin       -1       0          -1
-X-LOADER      /boot/boot_image_x-loader.bin   -1       0          -1
-MEM_INIT      /boot/mem_init.bin               0       0x080000    0
-PWR_MGT       /boot/power_management.bin       0       0x090000    0
-UBOOT_ENV     /boot/u-boot-env.bin             0       0x0F8000    0
-NORMAL        /usr/lib/u-boot/snowball/u-boot.bin                 0       0x100000    0
+ISSW          boot_image_issw.bin       -1       0          -1
+X-LOADER      boot_image_x-loader.bin   -1       0          -1
+MEM_INIT      mem_init.bin               0       0x080000    0
+PWR_MGT       power_management.bin       0       0x090000    0
+UBOOT_ENV     u-boot-env.bin             0       0x0F8000    0
+NORMAL        u-boot.bin                 0       0x100000    0
