diff -Naur binutils-2.21.51.0.9.orig//gas/config/tc-arm.c binutils-2.21.51.0.9/gas/config/tc-arm.c
--- binutils-2.21.51.0.9.orig//gas/config/tc-arm.c	2011-05-15 16:39:28.717064007 +0200
+++ binutils-2.21.51.0.9/gas/config/tc-arm.c	2011-05-15 16:40:54.077064009 +0200
@@ -5356,6 +5356,11 @@
   bfd_boolean is_apsr = FALSE;
   bfd_boolean m_profile = ARM_CPU_HAS_FEATURE (selected_cpu, arm_ext_m);
 
+  /* march=all, will match the m_profile case too, so things like 'mrsr2,cpsr'
+   * will now fail. Workaround by saying it's not a m profile */
+  if (selected_cpu.core == arm_arch_any.core)
+         m_profile = FALSE;
+
   /* CPSR's and SPSR's can now be lowercase.  This is just a convenience
      feature for ease of use and backwards compatibility.  */
   p = *str;
