diff -Naur binutils-2.20.51.0.8/ld/ldfile.c binutils-2.20.51.0.8a/ld/ldfile.c
--- binutils-2.20.51.0.8/ld/ldfile.c	2010-12-20 11:48:12.428579902 +0100
+++ binutils-2.20.51.0.8a/ld/ldfile.c	2010-12-20 11:48:35.844627726 +0100
@@ -98,6 +98,45 @@
 void
 ldfile_add_library_path (const char *name, bfd_boolean cmdline)
 {
+
+  if ( strstr(name, "/lib" ) == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /lib in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/lib32") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /lib32 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/lib64") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /lib64 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/lib") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/lib in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/lib32") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/lib32 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/lib64") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/lib64 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/local/lib") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/local/lib in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/local/lib32") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/local/lib32 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/usr/local/lib64") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /usr/local/lib64 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/opt/lib") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /opt/lib in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/opt/lib32") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /opt/lib32 in LIBRARYPATH: %s\n", name);
+    abort();
+  } else if ( strstr(name, "/opt/lib64") == name ) {
+    fprintf (stderr, "CROSS COMPILE Badness: /opt/lib64 in LIBRARYPATH: %s\n", name);
+    abort();
+  }
+
   search_dirs_type *new_dirs;
 
   if (!cmdline && config.only_cmd_line_lib_dirs)
