diff -Naur BeautifulSoup-3.0.8.1-old/setup.py BeautifulSoup-3.0.8.1-new/setup.py
--- BeautifulSoup-3.0.8.1-old/setup.py	2010-04-09 12:00:55.000000000 -0700
+++ BeautifulSoup-3.0.8.1-new/setup.py	2010-06-27 16:45:17.000000000 -0700
@@ -1,4 +1,4 @@
-from distutils.core import setup
+from setuptools import setup
 import unittest
 import warnings
 warnings.filterwarnings("ignore", "Unknown distribution option")
