--- libva-xvba-0.8.0/src/utils_glx.h.orig	2012-10-30 18:37:25.235652004 +0200
+++ libva-xvba-0.8.0/src/utils_glx.h	2012-10-30 18:37:47.715530214 +0200
@@ -44,6 +44,8 @@
 #define GL_PROGRAM_UNDER_NATIVE_LIMITS GL_PROGRAM_UNDER_NATIVE_LIMITS_ARB
 #endif
 
+typedef void (APIENTRYP PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+
 const char *
 gl_get_error_string(GLenum error)
     attribute_hidden;
