--- Python-2.7.1/setup.py.orig	2012-03-25 23:23:44.439637479 +0300
+++ Python-2.7.1/setup.py	2012-03-25 23:24:10.971637468 +0300
@@ -782,7 +782,7 @@
                             depends = ['md5.h']) )
 
         min_sha2_openssl_ver = 0x00908000
-        if COMPILED_WITH_PYDEBUG or openssl_ver < min_sha2_openssl_ver:
+        if COMPILED_WITH_PYDEBUG or openssl_ver < min_sha2_openssl_ver or True:
             # OpenSSL doesn't do these until 0.9.8 so we'll bring our own hash
             exts.append( Extension('_sha256', ['sha256module.c']) )
             exts.append( Extension('_sha512', ['sha512module.c']) )
