--- a/configure	2011-06-11 18:46:28.000000000 +0300
+++ b/configure	2012-04-03 14:16:35.507402002 +0300
@@ -604,6 +604,9 @@
 #endif"
 
 ac_subst_vars='LTLIBOBJS
+CROSS_COMPILING
+PGEN_FOR_BUILD
+PYTHON_FOR_BUILD
 SRCDIRS
 THREADHEADERS
 UNICODE_OBJS
@@ -619,12 +622,21 @@
 MACHDEP_OBJS
 DYNLOADFILE
 DLINCLDIR
+LIB2TO3
+TEST_MODULES
+PYDOC
+CURSES
+TK
+BSDDB
+SQLITE3
 THREADOBJ
 LDLAST
 USE_THREAD_MODULE
 SIGNAL_OBJS
 USE_SIGNAL_MODULE
 LIBFFI_INCLUDEDIR
+EXPAT
+DISABLED_EXTENSIONS
 PKG_CONFIG
 SHLIBS
 CFLAGSFORSHARED
@@ -749,7 +761,7 @@
 with_pydebug
 enable_toolbox_glue
 with_libs
-with_system_expat
+with_expat
 with_system_ffi
 with_dbmliborder
 with_signal_module
@@ -757,6 +769,21 @@
 with_threads
 with_thread
 with_pth
+enable_sqlite3
+enable_codecs_cjk
+enable_nis
+enable_ssl
+enable_bz2
+enable_zlib
+enable_dbm
+enable_gdbm
+enable_bsddb
+enable_unicodedata
+enable_tk
+enable_curses
+enable_pydoc
+enable_test_modules
+enable_lib2to3
 enable_ipv6
 with_doc_strings
 with_tsc
@@ -777,7 +804,9 @@
 LDFLAGS
 LIBS
 CPPFLAGS
-CPP'
+CPP
+PYTHON_FOR_BUILD
+PGEN_FOR_BUILD'
 
 
 # Initialize some variables set by options.
@@ -1396,6 +1425,21 @@
   --enable-shared         disable/enable building shared python library
   --enable-profiling      enable C-level code profiling
   --enable-toolbox-glue   disable/enable MacOSX glue code for extensions
+  --disable-sqlite3       disable sqlite3
+  --disable-codecs-cjk    disable CJK codecs
+  --disable-nis           disable NIS
+  --disable-ssl           disable SSL
+  --disable-bz2           disable BZIP2
+  --disable-zlib          disable ZLIB
+  --disable-dbm           disable DBM
+  --disable-gdbm          disable GDBM
+  --disable-bsddb         disable BerkeyleyDB
+  --disable-unicodedata   disable unicodedata
+  --disable-tk            disable tk
+  --disable-curses        disable curses
+  --disable-pydoc         disable pydoc
+  --disable-test-modules  disable test modules
+  --disable-lib2to3       disable lib2to3
   --enable-ipv6           Enable ipv6 (with ipv4) support
   --disable-ipv6          Disable ipv6 support
   --enable-big-digits[=BITS]
@@ -1419,8 +1463,8 @@
   --with-suffix=.exe      set executable suffix
   --with-pydebug          build with Py_DEBUG defined
   --with-libs='lib1 ...'  link against additional libs
-  --with-system-expat     build pyexpat module using an installed expat
-                          library
+  --with-expat            select which expat version to use: system, builtin,
+                          none
   --with-system-ffi       build _ctypes module using an installed ffi library
   --with-dbmliborder=db1:db2:...
                           order to check db backends for dbm. Valid value is a
@@ -1451,6 +1495,10 @@
   CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir> if
               you have headers in a nonstandard directory <include dir>
   CPP         C preprocessor
+  PYTHON_FOR_BUILD
+              build system Python
+  PGEN_FOR_BUILD
+              build system Python pgen
 
 Use these variables to override the choices made by `configure' or to help
 it to find libraries and programs with nonstandard names/locations.
@@ -8351,20 +8399,30 @@
 fi
 
 
+
+
 # Check for use of the system expat library
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for --with-system-expat" >&5
-$as_echo_n "checking for --with-system-expat... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for --with-expat" >&5
+$as_echo_n "checking for --with-expat... " >&6; }
 
-# Check whether --with-system_expat was given.
-if test "${with_system_expat+set}" = set; then :
-  withval=$with_system_expat;
+# Check whether --with-expat was given.
+if test "${with_expat+set}" = set; then :
+  withval=$with_expat;
 else
-  with_system_expat="no"
+  with_expat="builtin"
 fi
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_system_expat" >&5
-$as_echo "$with_system_expat" >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $with_expat" >&5
+$as_echo "$with_expat" >&6; }
+
+if test "$with_expat" != "none"; then
+   EXPAT=yes
+else
+   DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} pyexpat"
+   EXPAT=no
+fi
+
 
 # Check for use of the system libffi library
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for --with-system-ffi" >&5
@@ -9067,6 +9125,152 @@
 fi
 
 
+# Check whether --enable-sqlite3 was given.
+if test "${enable_sqlite3+set}" = set; then :
+  enableval=$enable_sqlite3;  SQLITE3="${enableval}"
+else
+   SQLITE3=yes
+fi
+
+
+if test "$SQLITE3" = "no" ; then
+   DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _sqlite3"
+fi
+
+# Check whether --enable-codecs-cjk was given.
+if test "${enable_codecs_cjk+set}" = set; then :
+  enableval=$enable_codecs_cjk;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _codecs_kr _codecs_jp _codecs_cn _codecs_tw _codecs_hk"
+  	  fi
+fi
+
+
+# Check whether --enable-nis was given.
+if test "${enable_nis+set}" = set; then :
+  enableval=$enable_nis;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} nis"
+  	  fi
+fi
+
+
+# Check whether --enable-ssl was given.
+if test "${enable_ssl+set}" = set; then :
+  enableval=$enable_ssl;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ssl"
+  	  fi
+fi
+
+
+# Check whether --enable-bz2 was given.
+if test "${enable_bz2+set}" = set; then :
+  enableval=$enable_bz2;  if test "$enableval" = "no"; then
+            DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} bz2"
+         fi
+fi
+
+
+# Check whether --enable-zlib was given.
+if test "${enable_zlib+set}" = set; then :
+  enableval=$enable_zlib;  if test "$enableval" = "no"; then
+            DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} zlib"
+         fi
+fi
+
+
+# Check whether --enable-dbm was given.
+if test "${enable_dbm+set}" = set; then :
+  enableval=$enable_dbm;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} dbm"
+  	  fi
+fi
+
+
+# Check whether --enable-gdbm was given.
+if test "${enable_gdbm+set}" = set; then :
+  enableval=$enable_gdbm;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} gdbm"
+  	  fi
+fi
+
+
+
+# Check whether --enable-bsddb was given.
+if test "${enable_bsddb+set}" = set; then :
+  enableval=$enable_bsddb;  if test "$enableval" = "no"; then
+    	     BSDDB=no
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} bsddb"
+  	  else
+	     BSDDB=yes
+  	  fi
+else
+   BSDDB=yes
+fi
+
+
+# Check whether --enable-unicodedata was given.
+if test "${enable_unicodedata+set}" = set; then :
+  enableval=$enable_unicodedata;  if test "$enableval" = "no"; then
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} unicodedata"
+  	  fi
+fi
+
+
+
+# Check whether --enable-tk was given.
+if test "${enable_tk+set}" = set; then :
+  enableval=$enable_tk;  TK="${enableval}"
+else
+   TK=yes
+fi
+
+
+if test "$TK" = "no"; then
+   DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _tkinter"
+fi
+
+
+# Check whether --enable-curses was given.
+if test "${enable_curses+set}" = set; then :
+  enableval=$enable_curses;  CURSES="${enableval}"
+else
+   CURSES=yes
+fi
+
+
+if test "$CURSES" = "no"; then
+   DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _curses _curses_panel"
+fi
+
+
+
+# Check whether --enable-pydoc was given.
+if test "${enable_pydoc+set}" = set; then :
+  enableval=$enable_pydoc;  PYDOC="${enableval}"
+else
+   PYDOC=yes
+fi
+
+
+
+
+# Check whether --enable-test-modules was given.
+if test "${enable_test_modules+set}" = set; then :
+  enableval=$enable_test_modules;  TEST_MODULES="${enableval}"
+else
+   TEST_MODULES=yes
+fi
+
+
+
+
+# Check whether --enable-lib2to3 was given.
+if test "${enable_lib2to3+set}" = set; then :
+  enableval=$enable_lib2to3;  LIB2TO3="${enableval}"
+else
+   LIB2TO3=yes
+fi
+
+
 # Check for enable-ipv6
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking if --enable-ipv6 is specified" >&5
@@ -13849,6 +14053,30 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: done" >&5
 $as_echo "done" >&6; }
 
+if test "$cross_compiling" = "yes"; then
+   { $as_echo "$as_me:${as_lineno-$LINENO}: checking python for build" >&5
+$as_echo_n "checking python for build... " >&6; }
+   PYTHON_FOR_BUILD="${PYTHON_FOR_BUILD}"
+   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_FOR_BUILD" >&5
+$as_echo "$PYTHON_FOR_BUILD" >&6; }
+   { $as_echo "$as_me:${as_lineno-$LINENO}: checking pgen for build" >&5
+$as_echo_n "checking pgen for build... " >&6; }
+   PGEN_FOR_BUILD="${PGEN_FOR_BUILD}"
+   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PGEN_FOR_BUILD" >&5
+$as_echo "$PGEN_FOR_BUILD" >&6; }
+else
+   PYTHON_FOR_BUILD='$(BUILDPYTHON)'
+   PGEN_FOR_BUILD='$(BUILDPGEN)'
+fi
+
+
+
+
+
+
+CROSS_COMPILING=$cross_compiling
+
+
 # generate output files
 ac_config_files="$ac_config_files Makefile.pre Modules/Setup.config Misc/python.pc"
 
@@ -13937,6 +14165,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: not updating unwritable cache $cache_file" >&5
 $as_echo "$as_me: not updating unwritable cache $cache_file" >&6;}
   fi
+U=
 fi
 rm -f confcache
 
@@ -14494,6 +14723,7 @@
     $ac_shift
     case $ac_optarg in
     *\'*) ac_optarg=`$as_echo "$ac_optarg" | sed "s/'/'\\\\\\\\''/g"` ;;
+    '') as_fn_error $? "missing file argument" ;;
     esac
     as_fn_append CONFIG_HEADERS " '$ac_optarg'"
     ac_need_defaults=false;;
