diff --git a/src/lib/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp b/src/lib/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp
index 27a4f56..936f31c 100644
--- a/src/lib/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp
+++ b/src/lib/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp
@@ -217,7 +217,7 @@ uint16_t CTDA995xCECAdapterCommunication::GetFirmwareVersion(void)
 
   m_dev->Ioctl(CEC_IOCTL_GET_SW_VERSION, &vers);
   
-  return (vers.majorVersionNr * 100) + vers.minorVersionNr;
+  return vers.majorVersionNr;
 }
 
 
