--- vdr-avahi4vdr-10.0/Makefile.old	2013-06-09 07:31:56.340054753 +0200
+++ vdr-avahi4vdr-10.0/Makefile	2013-06-09 07:32:08.128054954 +0200
@@ -29,7 +29,7 @@
 ### The compiler options:
 
 export CFLAGS   = $(call PKGCFG,cflags)
-export CXXFLAGS = $(call PKGCFG,cxxflags) $(shell pkg-config --cflags dbus-1) $(shell pkg-config --cflags avahi-client) $(shell pkg-config --cflags uuid)
+export CXXFLAGS = $(call PKGCFG,cxxflags) $(shell pkg-config --cflags dbus-1) $(shell pkg-config --cflags avahi-client) $(shell pkg-config --cflags uuid) -fPIC
 export LDADD += $(shell pkg-config --libs dbus-1) $(shell pkg-config --libs avahi-client) $(shell pkg-config --libs uuid)
 
 ### The version number of VDR's plugin API:
