From dfe48bed08ba4c1bbacf0941a804dccea273731a Mon Sep 17 00:00:00 2001
From: Peter Meerwald <p.meerwald@bct-electronic.com>
Date: Mon, 16 Jan 2012 23:48:38 +0100
Subject: [PATCH] build: workaround for --enable-backend issues

---
 configure.ac |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 2163e07..919aa2b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -115,28 +115,36 @@ case "${host_os}" in
 esac
 
 AC_ARG_ENABLE(backend,
-  AC_HELP_STRING([--enable-backend],[sse,mmx,neon,arm,all (default all)]),
+  AC_HELP_STRING([--enable-backend],[sse,mmx,neon,arm,altivec,c64x,all (default all)]),
     [], [enable_backend=all])
 case "${enable_backend}" in
   sse)
+    ENABLE_BACKEND_MMX=yes
     ENABLE_BACKEND_SSE=yes
     AC_DEFINE(ENABLE_BACKEND_SSE, 1, [Enable SSE backend])
+    AC_DEFINE(ENABLE_BACKEND_MMX, 1, [Enable MMX backend])
     ;;
   mmx)
     ENABLE_BACKEND_MMX=yes
+    ENABLE_BACKEND_SSE=yes
     AC_DEFINE(ENABLE_BACKEND_MMX, 1, [Enable MMX backend])
+    AC_DEFINE(ENABLE_BACKEND_SSE, 1, [Enable SSE backend])
     ;;
   altivec)
     ENABLE_BACKEND_ALTIVEC=yes
     AC_DEFINE(ENABLE_BACKEND_ALTIVEC, 1, [Enable Altivec backend])
     ;;
   neon)
+    ENABLE_BACKEND_ARM=yes
+    AC_DEFINE(ENABLE_BACKEND_ARM, 1, [Enable Arm backend])
     ENABLE_BACKEND_NEON=yes
     AC_DEFINE(ENABLE_BACKEND_NEON, 1, [Enable NEON backend])
     ;;
   arm)
     ENABLE_BACKEND_ARM=yes
     AC_DEFINE(ENABLE_BACKEND_ARM, 1, [Enable Arm backend])
+    ENABLE_BACKEND_NEON=yes
+    AC_DEFINE(ENABLE_BACKEND_NEON, 1, [Enable NEON backend])
     ;;
   c64x)
     ENABLE_BACKEND_C64X=yes
-- 
1.7.4.1

