--- cloog-ppl-0.15.11/configure.orig	2012-07-04 21:54:00.030293031 +0000
+++ cloog-ppl-0.15.11/configure	2012-07-04 21:55:17.994679649 +0000
@@ -11210,32 +11210,6 @@
 fi
 
 
-
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for version 0.10 (or later revision) of PPL" >&5
-$as_echo_n "checking for version 0.10 (or later revision) of PPL... " >&6; }
-	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include "ppl_c.h"
-int
-main ()
-{
-
-	#if PPL_VERSION_MAJOR != 0 || PPL_VERSION_MINOR < 10
-	choke me
-	#endif
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-else
-  as_fn_error $? "Can't find correct version of PPL." "$LINENO" 5
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-
 	LIBS="$LIBS -lppl_c -lppl -lgmpxx"
 
 $as_echo "#define CLOOG_PPL_BACKEND 1" >>confdefs.h
