--- a/glib/gnulib/vasnprintf.c	2013-08-04 12:17:08.834087337 +0200
+++ b/glib/gnulib/vasnprintf.c	2013-08-04 12:17:08.850087336 +0200
@@ -344,6 +344,20 @@
 	    result[length] = '%';
 	    length += 1;
 	  }
+	else if (dp->conversion == 'm')
+	  {
+	    char *m;
+	    size_t n;
+
+	    if (!(dp->arg_index < 0))
+	      abort ();
+	    
+	    m = strerror(errno);
+	    n = strlen(m);
+	    ENSURE_ALLOCATION (length + n + 1);
+	    memcpy(&result[length], m, n);
+	    length += n;
+	  }
 	else
 	  {
 	    if (!(dp->arg_index >= 0))
--- a/glib/gnulib/printf-parse.c	2013-08-04 12:08:33.830102554 +0200
+++ b/glib/gnulib/printf-parse.c	2013-08-04 12:08:33.846102552 +0200
@@ -449,6 +449,7 @@
 		  else
 		    type = TYPE_COUNT_INT_POINTER;
 		  break;
+		case 'm':
 		case '%':
 		  type = TYPE_NONE;
 		  break;
