diff -Naur cdparanoia-III-alpha9.8.orig/interface/utils.h cdparanoia-III-alpha9.8/interface/utils.h
--- cdparanoia-III-alpha9.8.orig/interface/utils.h	2000-04-20 00:41:04.000000000 +0200
+++ cdparanoia-III-alpha9.8/interface/utils.h	2005-01-14 18:25:16.000000000 +0100
@@ -112,6 +112,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      break;
     }
   }
 }
@@ -127,6 +128,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      break;
     }
   }
 }
@@ -169,6 +171,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      break;
     }
   }
   if(malloced)free(buffer);
@@ -205,6 +208,7 @@
       break;
     case CDDA_MESSAGE_FORGETIT:
     default:
+      break;
     }
   }
   if(malloced)free(buffer);
