debian bug #168683

diff -Nur cdparanoia-III-alpha9.8.orig/interface/scan_devices.c cdparanoia-III-alpha9.8/interface/scan_devices.c
--- cdparanoia-III-alpha9.8.orig/interface/scan_devices.c	Mon Mar 26 07:44:01 2001
+++ cdparanoia-III-alpha9.8/interface/scan_devices.c	Tue Jul 26 12:55:30 2005
@@ -617,7 +617,8 @@
 
   /* It would seem some TOSHIBA CDROMs gets things wrong */
  
-  if (!strncmp (p + 8, "TOSHIBA", 7) &&
+  if (p &&
+      !strncmp (p + 8, "TOSHIBA", 7) &&
       !strncmp (p + 16, "CD-ROM", 6) &&
       p[0] == TYPE_DISK) {
     p[0] = TYPE_ROM;
