diff -Naur cdparanoia-III-alpha9.8.orig/paranoia/paranoia.c cdparanoia-III-alpha9.8/paranoia/paranoia.c
--- cdparanoia-III-alpha9.8.orig/paranoia/paranoia.c	2007-10-06 17:54:20.000000000 +0200
+++ cdparanoia-III-alpha9.8/paranoia/paranoia.c	2007-10-06 17:55:35.000000000 +0200
@@ -46,6 +46,10 @@
 
   **************************************************************/
 
+#ifndef INTX_MAX
+#define INT_MAX               ((int)(~0U>>1))
+#endif
+
 #include <stdlib.h>
 #include <unistd.h>
 #include <stdio.h>
