diff -Naur xf86-video-openchrome-0.2.904.orig/libxvmc/viaXvMCPriv.h xf86-video-openchrome-0.2.904/libxvmc/viaXvMCPriv.h
--- xf86-video-openchrome-0.2.904.orig/libxvmc/viaXvMCPriv.h	2010-02-16 22:38:31.993258711 +0100
+++ xf86-video-openchrome-0.2.904/libxvmc/viaXvMCPriv.h	2010-02-16 22:40:10.550757606 +0100
@@ -25,6 +25,7 @@
 #ifndef _VIAXVMCPRIV_H
 #define _VIAXVMCPRIV_H 1
 
+#include <inttypes.h>
 #include <X11/Xlibint.h>
 #include <X11/extensions/XvMC.h>
 #include <X11/extensions/XvMClib.h>
