diff -Naur kaa-base-0.6.0/src/distribution/core.py kaa-base-0.6.0a/src/distribution/core.py
--- kaa-base-0.6.0/src/distribution/core.py	2009-05-25 20:19:31.000000000 +0200
+++ kaa-base-0.6.0a/src/distribution/core.py	2011-02-20 17:07:40.015256694 +0100
@@ -30,7 +30,7 @@
 import sys
 import tempfile
 import time
-import distutils.core
+import setuptools
 import distutils.cmd
 import distutils.sysconfig
 
@@ -344,7 +344,7 @@
 
     def convert(self):
         """
-        Convert Extension into a distutils.core.Extension.
+        Convert Extension into a setuptools.Extension.
         """
         library_dirs = self.library_dirs[:]
         include_dirs = self.include_dirs[:]
@@ -354,7 +354,7 @@
             include_dirs.extend(lib.include_dirs)
             libraries.extend(lib.libraries)
 
-        ext = distutils.core.Extension(self.output, self.files,
+        ext = setuptools.Extension(self.output, self.files,
                                        library_dirs=library_dirs,
                                        include_dirs=include_dirs,
                                        libraries=libraries,
@@ -598,7 +598,7 @@
         del kwargs['rpminfo']
 
     # run the distutils.setup function
-    result = distutils.core.setup(**kwargs)
+    result = setuptools.setup(**kwargs)
     # Run cleanup on extensions (for example to delete config.h)
     for ext in kwargs['ext_modules']:
         ext._kaa_ext.clean()
