--- vdr-smarttvweb-1.00/Makefile.org	2014-11-21 17:02:47.440034782 +0100
+++ vdr-smarttvweb-1.00/Makefile	2014-11-21 17:03:05.704035096 +0100
@@ -112,7 +112,6 @@
 	cp widget.conf $(DESTDIR)$(CONFDIR)/plugins/smarttvweb; \
 	cp -r web/ $(DESTDIR)$(CONFDIR)/plugins/smarttvweb; \
 	cp smarttvweb.conf $(DESTDIR)$(CONFDIR)/plugins/smarttvweb; \
-	chown -R vdr:vdr $(DESTDIR)$(CONFDIR)/plugins/smarttvweb; \
 	fi
 	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
 
