diff -Naur live/liveMedia/include/H263plusVideoStreamFramer.hh live-ok/liveMedia/include/H263plusVideoStreamFramer.hh
--- live/liveMedia/include/H263plusVideoStreamFramer.hh	2006-01-27 10:14:43.000000000 +0100
+++ live-ok/liveMedia/include/H263plusVideoStreamFramer.hh	2008-06-21 14:56:28.000000000 +0200
@@ -41,7 +41,7 @@
   virtual ~H263plusVideoStreamFramer();
 
 
-// private:
+public:
   static void continueReadProcessing(void* clientData,
 				     unsigned char* ptr, unsigned size,
 				     struct timeval presentationTime);
