diff -urN redis-2.8.7.orig/src/Makefile redis-2.8.7/src/Makefile
--- redis-2.8.7.orig/src/Makefile	2014-03-09 15:28:15.866117401 +0100
+++ redis-2.8.7/src/Makefile	2014-03-09 15:31:53.959156851 +0100
@@ -83,9 +83,8 @@
 endif
 
 ifeq ($(MALLOC),jemalloc)
-	DEPENDENCY_TARGETS+= jemalloc
-	FINAL_CFLAGS+= -DUSE_JEMALLOC -I../deps/jemalloc/include
-	FINAL_LIBS+= ../deps/jemalloc/lib/libjemalloc.a -ldl
+	FINAL_CFLAGS+= -DUSE_JEMALLOC -DJEMALLOC_NO_DEMANGLE 
+	FINAL_LIBS+= -ljemalloc -ldl
 endif
 
 REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS)
