--- graphtft-0.3.4/display.c.orig	2011-10-18 10:55:14.000000000 +0200
+++ graphtft-0.3.4/display.c	2012-03-11 11:52:32.545415663 +0100
@@ -931,12 +931,15 @@
 // Osd Channel Switch
 //***************************************************************************
 
-void cGraphTFTDisplay::ChannelSwitch(const cDevice* Device, int ChannelNumber) 
-{
+#if VDRVERSNUM >= 10726
+void cGraphTFTDisplay::ChannelSwitch(const cDevice* Device, int ChannelNumber, bool LiveView) {
+#else
+void cGraphTFTDisplay::ChannelSwitch(const cDevice* Device, int ChannelNumber) {
+    bool LiveView = Device && Device->IsPrimaryDevice() && !EITScanner.UsesDevice(Device);
+#endif
    tell(5, "ChannelSwitch on %p: %d", Device, ChannelNumber);
 
-   if (Device->IsPrimaryDevice() 
-       && !EITScanner.UsesDevice(Device) 
+   if (LiveView 
        && _channel != ChannelNumber 
        && cDevice::CurrentChannel() != _channel) 
    {
--- graphtft-0.3.4/display.h.orig	2011-10-18 10:55:14.000000000 +0200
+++ graphtft-0.3.4/display.h	2012-03-11 11:57:28.705392921 +0100
@@ -348,7 +348,11 @@
 
       // status interface
 
+#if VDRVERSNUM >= 10726
+      virtual void ChannelSwitch(const cDevice* Device, int ChannelNumber, bool LiveView);
+#else
       virtual void ChannelSwitch(const cDevice* Device, int ChannelNumber);
+#endif
       virtual void OsdSetEvent(const cEvent* event);
       virtual void OsdSetRecording(const cRecording* recording);
       virtual void OsdChannel(const char* Text);
