--- a/src/libFLAC/include/private/cpu.h	2017-04-12 08:09:49.160175707 +0200
+++ b/src/libFLAC/include/private/cpu.h	2017-04-12 08:09:49.176175507 +0200
@@ -74,37 +74,30 @@
 #elif defined __GNUC__
   #if (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 9)) /* since GCC 4.9 -msse.. compiler options aren't necessary */
     #define FLAC__SSE_TARGET(x) __attribute__ ((__target__ (x)))
+  #else /* for GCC older than 4.9 */
+    #define FLAC__SSE_TARGET(x)
+  #endif /* GCC version */
+  #ifdef __SSE__
     #define FLAC__SSE_SUPPORTED 1
+  #endif
+  #ifdef __SSE2__
     #define FLAC__SSE2_SUPPORTED 1
+  #endif
+  #ifdef __SSSE3__
     #define FLAC__SSSE3_SUPPORTED 1
+  #endif
+  #ifdef __SSE4_1__
     #define FLAC__SSE4_1_SUPPORTED 1
+  #endif
+  #ifdef __AVX__
     #define FLAC__AVX_SUPPORTED 1
+  #endif
+  #ifdef __AVX2__
     #define FLAC__AVX2_SUPPORTED 1
+  #endif
+  #ifdef __FMA__
     #define FLAC__FMA_SUPPORTED 1
-  #else /* for GCC older than 4.9 */
-    #define FLAC__SSE_TARGET(x)
-    #ifdef __SSE__
-      #define FLAC__SSE_SUPPORTED 1
-    #endif
-    #ifdef __SSE2__
-      #define FLAC__SSE2_SUPPORTED 1
-    #endif
-    #ifdef __SSSE3__
-      #define FLAC__SSSE3_SUPPORTED 1
-    #endif
-    #ifdef __SSE4_1__
-      #define FLAC__SSE4_1_SUPPORTED 1
-    #endif
-    #ifdef __AVX__
-      #define FLAC__AVX_SUPPORTED 1
-    #endif
-    #ifdef __AVX2__
-      #define FLAC__AVX2_SUPPORTED 1
-    #endif
-    #ifdef __FMA__
-      #define FLAC__FMA_SUPPORTED 1
-    #endif
-  #endif /* GCC version */
+  #endif
 #endif /* compiler version */
 #endif /* intrinsics support */
 
