diff -Nur autoplay-0.0.5.orig/autoplay.c autoplay-0.0.5/autoplay.c
--- autoplay-0.0.5.orig/autoplay.c	Sat Jun 26 20:43:24 2004
+++ autoplay-0.0.5/autoplay.c	Fri Dec 10 15:01:54 2004
@@ -164,6 +164,15 @@
 }
 
 int
+open_device(const char *dev)
+{
+  int fd = open(dev, O_RDONLY | O_NONBLOCK);
+  if (fd >= 0)
+    ioctl(fd, CDROM_LOCKDOOR, 0);
+  return fd;
+}
+
+int
 main (int argc, char **argv)
 {
   cd_drive drive, *drives, *ptr;
@@ -211,27 +220,29 @@
 
   for (ptr=drives, drive=*ptr; drive; ptr++, drive=*ptr)
     {
-      drive->fd = open(drive->dev, O_RDONLY | O_NONBLOCK);
-      if (drive->fd < 0)
+      if ((drive->fd = open_device(drive->dev)) < 0)
         return 3;
     }
 
   while (1)
     {
       usleep(1000000);
-
       for (ptr=drives, drive=*ptr; drive; ptr++, drive=*ptr)
         {
           if (drive->status == CDS_NO_INFO)
             continue;
 
+          if (drive->fd < 0 && (drive->fd = open_device(drive->dev)) < 0)
+            continue;
+
           status = ioctl (drive->fd, CDROM_DRIVE_STATUS, CDSL_CURRENT);
           switch (status)
             {
             case CDS_NO_INFO:
               /* Drive do not support status request :-( */
               drive->status = status;
-              close (drive->fd);
+              close(drive->fd);
+              drive->fd = -1;
               break;
 
             case CDS_DISC_OK:
@@ -249,6 +260,9 @@
                       case CDS_DATA_1:
                       case CDS_DATA_2:
                         /* it's a data CD */
+                        close(drive->fd);
+                        drive->fd = -1;
+
                         sprintf (filename, "%s/video_ts", drive->mnt);
                         if (!stat (filename, &st) && S_ISDIR (st.st_mode))
                           {
