diff -Nur autoplay-0.0.5.orig/autoplay.c autoplay-0.0.5/autoplay.c
--- autoplay-0.0.5.orig/autoplay.c	Mon Dec 20 11:53:44 2004
+++ autoplay-0.0.5/autoplay.c	Mon Dec 20 11:52:54 2004
@@ -172,6 +172,15 @@
   return fd;
 }
 
+void
+unlock_supermount(cd_drive drive)
+{
+  char buf[256];
+  sprintf(buf, "umount \"%s\" && mount -t supermount -o ro,dev=%s,fs=auto,tray_lock=onwrite none \"%s\"",
+      drive->mnt, drive->dev, drive->mnt);
+  system(buf);
+}
+
 int
 main (int argc, char **argv)
 {
@@ -266,24 +275,28 @@
                         sprintf (filename, "%s/video_ts", drive->mnt);
                         if (!stat (filename, &st) && S_ISDIR (st.st_mode))
                           {
+                            unlock_supermount(drive);
                             printf(play_dvd_cmd);
                             break;
                           }
                         sprintf (filename, "%s/VIDEO_TS", drive->mnt);
                         if (!stat (filename, &st) && S_ISDIR (st.st_mode))
                           {
+                            unlock_supermount(drive);
                             printf(play_dvd_cmd);
                             break;
                           }
                         sprintf (filename, "%s/vcd", drive->mnt);
                         if (!stat (filename, &st) && S_ISDIR (st.st_mode))
                           {
+                            unlock_supermount(drive);
                             printf("play_vcd\nset_menu null\n");
                             break;
                           }
                         sprintf (filename, "%s/svcd", drive->mnt);
                         if (!stat (filename, &st) && S_ISDIR (st.st_mode))
                           {
+                            unlock_supermount(drive);
                             printf("play_vcd\nset_menu null\n");
                             break;
                           }
@@ -316,7 +329,10 @@
                                   {
                                     exts=xcd_exts;
                                     if (build_playlist (drive->mnt, -1) >= 1)
+                                    {
+                                      unlock_supermount(drive);
                                       printf ("play_vcd\nset_menu null\n");
+                                    }
                                   }
                               }
                           }
