diff -Naur autoplay.orig/autoplay.c autoplay/autoplay.c
--- autoplay.orig/autoplay.c	2005-01-12 21:34:23.000000000 +0100
+++ autoplay/autoplay.c	2005-01-12 21:34:53.000000000 +0100
@@ -174,7 +174,7 @@
   struct stat st;
   int i, m, n, x, status;
 
-  if (argc < 3)
+  if (argc < 4)
     return 1;
 
   m = 0;
@@ -240,6 +240,8 @@
                 {
                   /* a media as been inserted */
                   drive->status = status;
+                  if (atoi (argv[3]) > 1 && atoi (argv[3]) < 100)
+                    ioctl (drive->fd, CDROM_SELECT_SPEED, atoi (argv[3]));
                   status = ioctl (drive->fd, CDROM_DISC_STATUS, CDSL_CURRENT);
                   switch (status)
                     {
