--- glcdprocdriver-0.0.6/Makefile.org	2013-03-17 18:51:23.189626742 +0100
+++ glcdprocdriver-0.0.6/Makefile	2013-03-17 18:51:56.597627316 +0100
@@ -2,12 +2,13 @@
 # Makefile for the glcdprocdriver graphlcd-base wrapper "meta-driver"
 #
 
--include Make.config
+include Make.config
 
 
 PROJECT = glcdprocdriver
 VERSION = 0.0.6
 ARCHIVE = $(PROJECT)-$(VERSION)
+LIBDIR = $(DESTDIR)/usr/lib
 PACKAGE = $(ARCHIVE)
 TMPDIR = /tmp
 
