--- a/Makefile	2015-01-10 23:31:18.751993045 +0100
+++ b/Makefile	2015-01-10 23:31:18.763993041 +0100
@@ -797,7 +797,7 @@
 	$(Q)$(MAKE) $(build)=$@
 
 define filechk_kernel.release
-	echo "$(KERNELVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))"
+	echo "$(KERNELVERSION)"
 endef
 
 # Store (new) KERNELRELEASE string in include/config/kernel.release
@@ -1332,7 +1332,7 @@
 	$(PERL) $(src)/scripts/checkstack.pl $(CHECKSTACK_ARCH)
 
 kernelrelease:
-	@echo "$(KERNELVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))"
+	@echo "$(KERNELVERSION)"
 
 kernelversion:
 	@echo $(KERNELVERSION)
