diff -Naur gst-openmax-git20091114/configure gst-openmax-git20091114a/configure
--- gst-openmax-git20091114/configure	2010-06-02 18:43:27.000000000 +0200
+++ gst-openmax-git20091114a/configure	2010-06-02 18:53:30.988659314 +0200
@@ -10967,7 +10967,7 @@
 $as_echo_n "checking to see if compiler understands -Werror... " >&6; }
 
   save_CFLAGS="$CFLAGS"
-  CFLAGS="$CFLAGS -Werror"
+  CFLAGS="$CFLAGS"
 
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -10989,7 +10989,7 @@
   CFLAGS="$save_CFLAGS"
 
   if test "X$flag_ok" = Xyes ; then
-    ERROR_CFLAGS="$ERROR_CFLAGS -Werror"
+    ERROR_CFLAGS="$ERROR_CFLAGS"
     true
   else
 
