diff -Naur elfutils-0.152/libelf/Makefile.am elfutils-0.152_patched/libelf/Makefile.am
--- elfutils-0.152/libelf/Makefile.am	2010-06-29 06:09:31.000000000 +0200
+++ elfutils-0.152_patched/libelf/Makefile.am	2011-11-18 11:15:08.102744543 +0100
@@ -93,7 +93,7 @@
 libelf_pic_a_SOURCES =
 am_libelf_pic_a_OBJECTS = $(libelf_a_SOURCES:.c=.os)
 
-libelf_so_LDLIBS =
+libelf_so_LDLIBS = -lintl
 if USE_LOCKS
 libelf_so_LDLIBS += -lpthread
 endif
--- elfutils-0.152/libdw/Makefile.am	2010-09-10 15:18:31.000000000 +0200
+++ elfutils-0.152_patched/libdw/Makefile.am	2011-11-18 11:23:57.992744271 +0100
@@ -98,6 +98,8 @@
 libdw_pic_a_SOURCES =
 am_libdw_pic_a_OBJECTS = $(libdw_a_SOURCES:.c=.os)
 
+libdw_so_LDLIBS = -lintl
+
 libdw_so_SOURCES =
 libdw.so: $(srcdir)/libdw.map libdw_pic.a \
 	  ../libdwfl/libdwfl_pic.a ../libebl/libebl.a \
@@ -108,7 +110,7 @@
 		-Wl,--enable-new-dtags,-rpath,$(pkglibdir) \
 		-Wl,--version-script,$<,--no-undefined \
 		-Wl,--whole-archive $(filter-out $<,$^) -Wl,--no-whole-archive\
-		-ldl $(zip_LIBS)
+		-ldl $(zip_LIBS) $(libdw_so_LDLIBS)
 	if readelf -d $@ | fgrep -q TEXTREL; then exit 1; fi
 	ln -fs $@ $@.$(VERSION)
 
--- elfutils-0.152/libcpu/Makefile.am	2010-09-10 15:18:31.000000000 +0200
+++ elfutils-0.152_patched/libcpu/Makefile.am	2011-11-18 11:28:18.162744138 +0100
@@ -61,7 +61,7 @@
 i386_parse.o: i386_parse.c i386.mnemonics
 i386_parse_CFLAGS = -DNMNES="`wc -l < i386.mnemonics`"
 i386_lex.o: i386_parse.h
-i386_gendis_LDADD = $(libeu) -lm $(libmudflap)
+i386_gendis_LDADD = $(libeu) -lm $(libmudflap) -lintl
 
 i386_parse.h: i386_parse.c ;
 
--- elfutils-0.152/libasm/Makefile.am	2010-06-29 06:09:31.000000000 +0200
+++ elfutils-0.152_patched/libasm/Makefile.am	2011-11-18 11:48:22.932743522 +0100
@@ -55,7 +55,7 @@
 libasm_pic_a_SOURCES =
 am_libasm_pic_a_OBJECTS = $(libasm_a_SOURCES:.c=.os)
 
-libasm_so_LDLIBS =
+libasm_so_LDLIBS = -lintl
 if USE_LOCKS
 libasm_so_LDLIBS += -lpthread
 endif
