--- elfutils-0.152/libcpu/Makefile.am	2011-11-18 11:28:52.242744121 +0100
+++ _elfutils-0.152/libcpu/Makefile.am	2011-11-18 11:43:50.572743661 +0100
@@ -47,7 +47,7 @@
 	m4 -D$* -DDISASSEMBLER $< > $@
 
 %_dis.h: %_defs i386_gendis
-	./i386_gendis $< > $@
+#	./i386_gendis $< > $@
 
 %.mnemonics: %_defs
 	sed '1,/^%%/d;/^#/d;/^[[:space:]]*$$/d;s/[^:]*:\([^[:space:]]*\).*/MNE(\1)/;s/{[^}]*}//g;/INVALID/d' \
