--- a/src/Makefile.am      2012-02-11 20:29:10.649076826 +0100
+++ b/src/Makefile.am
      2012-02-11 20:22:16.369087340 +0100
@@ -83,11 +83,11 @@
 if BUILD_STATIC
 libasm = ../libasm/libasm.a
 libdw = ../libdw/libdw.a $(zip_LIBS) $(libelf) $(libebl) -ldl
-libelf = ../libelf/libelf.a
+libelf = ../libelf/libelf.a -lintl
 else
 libasm = ../libasm/libasm.so
 libdw = ../libdw/libdw.so
-libelf = ../libelf/libelf.so
+libelf = ../libelf/libelf.so -lintl
 endif
 libebl = ../libebl/libebl.a
 libeu = ../lib/libeu.a

