diff -Naur mame-0.137/src/osd/sdl/sdl.mak mame-0.137a/src/osd/sdl/sdl.mak
--- mame-0.137/src/osd/sdl/sdl.mak	2010-02-14 03:47:30.000000000 +0100
+++ mame-0.137a/src/osd/sdl/sdl.mak	2010-03-13 09:40:07.658850282 +0100
@@ -452,12 +452,6 @@
 DEFS += -DSDLMAME_X11
 LIBS += -lX11 -lXinerama
 
-# the new debugger relies on GTK+ in addition to the base SDLMAME needs
-# Non-X11 builds can not use the debugger
-CCOMFLAGS += `pkg-config --cflags gtk+-2.0` `pkg-config --cflags gconf-2.0`
-LIBS += `pkg-config --libs gtk+-2.0` `pkg-config --libs gconf-2.0`
-#CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
-
 # some systems still put important things in a different prefix
 LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
 # make sure we can find X headers
