diff -Naur mame-0.137/src/osd/sdl/sdl.mak mame-0.137a/src/osd/sdl/sdl.mak
--- mame-0.137/src/osd/sdl/sdl.mak	2010-03-13 09:41:21.735145903 +0100
+++ mame-0.137a/src/osd/sdl/sdl.mak	2010-03-13 09:40:57.130889286 +0100
@@ -451,11 +451,6 @@
 # Default libs
 DEFS += -DSDLMAME_X11
 LIBS += -lX11 -lXinerama
-
-# some systems still put important things in a different prefix
-LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
-# make sure we can find X headers
-CCOMFLAGS += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
 endif # NO_X11
 
 #-------------------------------------------------
