diff -Naur a/src/osd/sdl/sdl.mak b/src/osd/sdl/sdl.mak
--- a/src/osd/sdl/sdl.mak	2011-06-26 08:09:42.000000000 +0200
+++ b/src/osd/sdl/sdl.mak	2011-06-26 08:13:06.575651470 +0200
@@ -490,9 +490,9 @@
 #CCOMFLAGS += -DGTK_DISABLE_DEPRECATED
 
 # some systems still put important things in a different prefix
-LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
+# LIBS += -L/usr/X11/lib -L/usr/X11R6/lib -L/usr/openwin/lib
 # make sure we can find X headers
-INCPATH += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
+# INCPATH += -I/usr/X11/include -I/usr/X11R6/include -I/usr/openwin/include
 endif # NO_X11
 
 #-------------------------------------------------
