diff -Naur mame-0.137/src/osd/sdl/sdl.mak mame-0.137a/src/osd/sdl/sdl.mak
--- mame-0.137/src/osd/sdl/sdl.mak	2010-07-10 09:28:48.416886907 +0200
+++ mame-0.137a/src/osd/sdl/sdl.mak	2010-07-10 09:29:47.076973961 +0200
@@ -274,11 +274,6 @@
 	$(SDLOBJ)/window.o \
 	$(SDLOBJ)/output.o
 
-# Add SDL1.3 support
-ifdef SDL_INSTALL_ROOT
-OSDOBJS += $(SDLOBJ)/draw13.o
-endif
-
 # add an ARCH define
 DEFS += "-DSDLMAME_ARCH=$(ARCHOPTS)" -DSYNC_IMPLEMENTATION=$(SYNC_IMPLEMENTATION)
 
