From 4f16832c764978ba0d1a74af17876d6b94d15acb Mon Sep 17 00:00:00 2001
From: Frederic Plourde <frederic.plourde@collabora.co.uk>
Date: Mon, 25 Jun 2012 14:33:55 +0200
Subject: [PATCH] PVR support for 2D buffers

---
 src/omap_exa.h |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/omap_exa.h b/src/omap_exa.h
index 4dc818d..efad95e 100644
--- a/src/omap_exa.h
+++ b/src/omap_exa.h
@@ -141,6 +141,13 @@ Bool OMAPPrepareAccess(PixmapPtr pPixmap, int index);
 void OMAPFinishAccess(PixmapPtr pPixmap, int index);
 Bool OMAPPixmapIsOffscreen(PixmapPtr pPixmap);
 
+static inline OMAPPixmapPrivPtr
+OMAPPixmapPriv(PixmapPtr pPixmap)
+{
+    OMAPPixmapPrivPtr priv = exaGetPixmapDriverPrivate(pPixmap);
+    return priv;
+}
+
 static inline struct omap_bo *
 OMAPPixmapBo(PixmapPtr pPixmap)
 {
-- 
1.7.9.5

