Index: xf86-video-omap/src/Makefile.am
===================================================================
--- xf86-video-omap.orig/src/Makefile.am	2012-03-27 12:44:10.000000000 +0000
+++ xf86-video-omap/src/Makefile.am	2012-03-27 12:49:12.000000000 +0000
@@ -49,3 +49,6 @@
          omap_xv.c \
          omap_dri2.c \
          omap_driver.c
+
+omap_drvincludedir = ${includedir}
+omap_drvinclude_HEADERS = omap_driver.h omap_exa.h omap_util.h compat-api.h
