From 182c0dd8fabf0ebe5f78a1c961ea5e44da5390ed Mon Sep 17 00:00:00 2001
From: Frederic Plourde <frederic.plourde@collabora.co.uk>
Date: Wed, 13 Jun 2012 16:10:25 -0400
Subject: [PATCH 1/8] dri2: turn vblank error into a WARNING

---
 src/omap_dri2.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/omap_dri2.c b/src/omap_dri2.c
index 1ab48e0..d4d0f16 100644
--- a/src/omap_dri2.c
+++ b/src/omap_dri2.c
@@ -313,7 +313,7 @@ OMAPDRI2GetMSC(DrawablePtr pDraw, CARD64 *ust, CARD64 *msc)
 	if (ret) {
 		static int limit = 5;
 		if (limit) {
-			ERROR_MSG("get vblank counter failed: %s", strerror(errno));
+			WARNING_MSG("get vblank counter failed: %s", strerror(errno));
 			limit--;
 		}
 		return FALSE;
-- 
1.7.9.5

