--- a/modules/demux/avformat/demux.c	2013-12-03 10:12:34.000000000 +0100
+++ b/modules/demux/avformat/demux.c	2014-10-12 22:07:54.044769366 +0200
@@ -106,7 +106,7 @@ int OpenDemux( vlc_object_t *p_this )
 {
     demux_t       *p_demux = (demux_t*)p_this;
     demux_sys_t   *p_sys;
-    AVProbeData   pd;
+    AVProbeData   pd = {0};
     AVInputFormat *fmt = NULL;
     unsigned int  i;
     int64_t       i_start_time = -1;
