diff -Nur gcc-3.4.5.orig/libstdc++-v3/config/locale/uclibc/c_locale.h gcc-3.4.5/libstdc++-v3/config/locale/uclibc/c_locale.h
--- gcc-3.4.5.orig/libstdc++-v3/config/locale/uclibc/c_locale.h	Fri Mar  3 16:38:52 2006
+++ gcc-3.4.5/libstdc++-v3/config/locale/uclibc/c_locale.h	Fri Mar  3 16:40:34 2006
@@ -100,7 +100,11 @@
 # endif
 #endif
 
+#ifdef _GLIBCXX_USE_C99
       const int __ret = std::snprintf(__out, __size, __fmt, __prec, __v);
+#else
+      const int __ret = std::sprintf(__out, __fmt, __prec, __v);
+#endif
 
 #ifdef __UCLIBC_HAS_XLOCALE__
       __gnu_cxx::__uselocale(__old);
