From 856f4de49ad3df487dd3b046c3de7061bb02e3ab Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Sun, 4 Apr 2010 06:47:55 -0500
Subject: [PATCH 13/21] v4l2sink: Add support for omap_vout driver

---
 sys/v4l2/gstv4l2sink.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/gstv4l2sink.c b/sys/v4l2/gstv4l2sink.c
index 20236b9..c0d45a0 100644
--- a/sys/v4l2/gstv4l2sink.c
+++ b/sys/v4l2/gstv4l2sink.c
@@ -739,6 +739,17 @@ gst_v4l2sink_buffer_alloc (GstBaseSink * bsink, guint64 offset, guint size,
         no_pending_streamon = TRUE;
       }
 
+      /* workaround for bug in omap_vout driver, when we ask for more
+       * than four buffers:
+       */
+      if (!strcmp ("omap_vout", driver)) {
+        if (v4l2sink->num_buffers > 4) {
+          v4l2sink->num_buffers = 4;
+          GST_DEBUG_OBJECT (v4l2sink,
+              "limiting to 4 buffers to work-around omap_vout driver bug");
+        }
+      }
+
       /* set_caps() might not be called yet.. so just to make sure: */
       if (!gst_v4l2sink_set_caps (bsink, caps)) {
         return GST_FLOW_ERROR;
-- 
1.7.1

