From bd559fa93cb7e1c57970f22beb38f4bd7cd6b510 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Tue, 28 Jun 2011 00:34:45 -0300
Subject: [PATCH 14/19] fixes to mux h264 stream utilizing ctts

---
 gst/isomp4/gstqtmux.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/gst/isomp4/gstqtmux.c b/gst/isomp4/gstqtmux.c
index 530c611..e94582c 100644
--- a/gst/isomp4/gstqtmux.c
+++ b/gst/isomp4/gstqtmux.c
@@ -2241,7 +2241,12 @@ again:
   } else {
     nsamples = 1;
     sample_size = GST_BUFFER_SIZE (last_buf);
-    if (pad->have_dts) {
+    /* note: by default offset_end will be 0, but qtdemux (and perhaps
+     * others) sets this to -1.  So treat either as invalid values.
+     */
+    if (pad->have_dts &&
+        (GST_BUFFER_OFFSET_END (last_buf) != -1) &&
+        (GST_BUFFER_OFFSET_END (last_buf) != 0)) {
       gint64 scaled_dts;
       pad->last_dts = GST_BUFFER_OFFSET_END (last_buf);
       if ((gint64) (pad->last_dts) < 0) {
@@ -3002,6 +3007,7 @@ gst_qt_mux_video_sink_set_caps (GstPad * pad, GstCaps * caps)
     }
 
     entry.fourcc = FOURCC_avc1;
+    qtpad->have_dts = TRUE;
     if (qtpad->avg_bitrate == 0) {
       gint avg_bitrate = 0;
       gst_structure_get_int (structure, "bitrate", &avg_bitrate);
-- 
1.7.1

