From 901510563490ea416c3b2b2edd650c03de9797cb Mon Sep 17 00:00:00 2001
From: Alejandro Gonzalez <agonzalez@dextratech.com>
Date: Thu, 16 Dec 2010 10:14:06 -0500
Subject: [PATCH 15/19] qtmux: Store 1st TS when detect we get 1st buffer

---
 gst/isomp4/gstqtmux.c |   26 ++++++++++----------------
 1 files changed, 10 insertions(+), 16 deletions(-)

diff --git a/gst/isomp4/gstqtmux.c b/gst/isomp4/gstqtmux.c
index e94582c..9e7ac79 100644
--- a/gst/isomp4/gstqtmux.c
+++ b/gst/isomp4/gstqtmux.c
@@ -2093,7 +2093,6 @@ again:
   }
 
   if (last_buf == NULL) {
-#ifndef GST_DISABLE_GST_DEBUG
     if (buf == NULL) {
       GST_DEBUG_OBJECT (qtmux, "Pad %s has no previous buffer stored and "
           "received NULL buffer, doing nothing",
@@ -2102,8 +2101,17 @@ again:
       GST_LOG_OBJECT (qtmux,
           "Pad %s has no previous buffer stored, storing now",
           GST_PAD_NAME (pad->collect.pad));
+      if (GST_CLOCK_TIME_IS_VALID (GST_BUFFER_TIMESTAMP (buf))) {
+        pad->first_ts = GST_BUFFER_TIMESTAMP (buf);
+      } else {
+        GST_DEBUG_OBJECT (qtmux, "First buffer for pad %s has no timestamp, "
+            "using 0 as first timestamp", GST_PAD_NAME (pad->collect.pad));
+        pad->first_ts = 0;
+      }
+      GST_DEBUG_OBJECT (qtmux, "Stored first timestamp for pad %s %"
+          GST_TIME_FORMAT, GST_PAD_NAME (pad->collect.pad),
+          GST_TIME_ARGS (pad->first_ts));
     }
-#endif
     pad->last_buf = buf;
     goto exit;
   } else
@@ -2319,20 +2327,6 @@ again:
     qtmux->longest_chunk = duration;
   }
 
-  /* if this is the first buffer, store the timestamp */
-  if (G_UNLIKELY (pad->first_ts == GST_CLOCK_TIME_NONE) && last_buf) {
-    if (GST_CLOCK_TIME_IS_VALID (GST_BUFFER_TIMESTAMP (last_buf))) {
-      pad->first_ts = GST_BUFFER_TIMESTAMP (last_buf);
-    } else {
-      GST_DEBUG_OBJECT (qtmux, "First buffer for pad %s has no timestamp, "
-          "using 0 as first timestamp", GST_PAD_NAME (pad->collect.pad));
-      pad->first_ts = 0;
-    }
-    GST_DEBUG_OBJECT (qtmux, "Stored first timestamp for pad %s %"
-        GST_TIME_FORMAT, GST_PAD_NAME (pad->collect.pad),
-        GST_TIME_ARGS (pad->first_ts));
-  }
-
   /* now we go and register this buffer/sample all over */
   /* note that a new chunk is started each time (not fancy but works) */
   if (qtmux->moov_recov_file) {
-- 
1.7.1

