From a960d1fdcd14d31ec119e4f3f3f47f1b698ea66d Mon Sep 17 00:00:00 2001
From: Olivier Naudan <o-naudan@ti.com>
Date: Fri, 8 Oct 2010 13:42:46 +0200
Subject: [PATCH 31/32] Set GSTPB_REQ to 0.10.30, as 0.10.30.1 is not well understood by apt-get

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9da105f..8a713d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,7 @@ AM_PROG_LIBTOOL
 
 dnl *** required versions of GStreamer stuff ***
 GST_REQ=0.10.30
-GSTPB_REQ=0.10.30.1
+GSTPB_REQ=0.10.30
 
 dnl *** autotools stuff ****
 
-- 
1.7.1

