From b141a4222b8f8c6dfa48166681eda5cce9fb7ca2 Mon Sep 17 00:00:00 2001
From: Alejandro Gonzalez <x0092071@ti.com>
Date: Wed, 8 Jun 2011 17:19:41 -0500
Subject: [PATCH] Comment PA_STREAM_ADJUST_LATENCY flag to avoid Underruns

---
 ext/pulse/pulsesink.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/ext/pulse/pulsesink.c b/ext/pulse/pulsesink.c
index a8d019c..3eb87f4 100644
--- a/ext/pulse/pulsesink.c
+++ b/ext/pulse/pulsesink.c
@@ -735,7 +735,11 @@ gst_pulseringbuffer_acquire (GstRingBuffer * buf, GstRingBufferSpec * spec)
   /* construct the flags */
   flags = PA_STREAM_INTERPOLATE_TIMING | PA_STREAM_AUTO_TIMING_UPDATE |
 #ifdef HAVE_PULSE_0_9_11
-      PA_STREAM_ADJUST_LATENCY |
+  //With this flag, every time stream is moved from sink
+  // PA tries to adjust latency in order to do not loose buffers
+  // but this cause Underruns and audio glitches after several 
+  // sink switches. 
+  /*  PA_STREAM_ADJUST_LATENCY |*/
 #endif
       PA_STREAM_START_CORKED;
 
-- 
1.7.1

