Index: gst-plugins-good0.10-0.10.23.2/ext/pulse/plugin.c
===================================================================
--- gst-plugins-good0.10-0.10.23.2.orig/ext/pulse/plugin.c	2010-03-23 01:19:28.000000000 +0100
+++ gst-plugins-good0.10-0.10.23.2/ext/pulse/plugin.c	2010-06-26 21:22:43.737204876 +0200
@@ -41,11 +41,11 @@
   bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 #endif
 
-  if (!gst_element_register (plugin, "pulsesink", GST_RANK_PRIMARY,
+  if (!gst_element_register (plugin, "pulsesink", GST_RANK_PRIMARY+10,
           GST_TYPE_PULSESINK))
     return FALSE;
 
-  if (!gst_element_register (plugin, "pulsesrc", GST_RANK_PRIMARY,
+  if (!gst_element_register (plugin, "pulsesrc", GST_RANK_PRIMARY+10,
           GST_TYPE_PULSESRC))
     return FALSE;
 
