Index: vdr-plugin-freecell-0.0.2/Makefile
===================================================================
--- vdr-plugin-freecell-0.0.2.orig/Makefile	2012-03-29 20:07:14.000000000 +0200
+++ vdr-plugin-freecell-0.0.2/Makefile	2012-03-29 20:07:14.000000000 +0200
@@ -46,7 +46,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o cards.o game.o osd.o cursor.o i18n.o \
+OBJS = $(PLUGIN).o cards.o game.o osd.o cursor.o \
 	\
 	tools/collection.o tools/list.o tools/shared.o tools/string.o tools/tools.o 
 
Index: vdr-plugin-freecell-0.0.2/freecell.c
===================================================================
--- vdr-plugin-freecell-0.0.2.orig/freecell.c	2012-03-29 20:06:35.353957756 +0200
+++ vdr-plugin-freecell-0.0.2/freecell.c	2012-03-29 20:41:59.783959041 +0200
@@ -8,7 +8,6 @@
 
 #include <vdr/plugin.h>
 #include "game.h"
-#include "i18n.h"
 
 static const char *VERSION        = "0.0.2";
 static const char *DESCRIPTION    = "The well-known card game";
@@ -46,7 +45,6 @@
 
 bool cPluginFreecell::Start(void)
 {
-	RegisterI18n(Phrases);
   return true;
 }
 
Index: vdr-plugin-freecell-0.0.2/game.c
===================================================================
--- vdr-plugin-freecell-0.0.2.orig/game.c	2012-03-29 20:06:35.433957756 +0200
+++ vdr-plugin-freecell-0.0.2/game.c	2012-03-29 20:07:14.000000000 +0200
@@ -2,7 +2,6 @@
 #include "cards.h"
 #include "osd.h"
 #include "cursor.h"
-#include "i18n.h"
 #include <math.h>
 
 cFreecellGame::cFreecellGame(void): cOsdObject(true) {
Index: vdr-plugin-freecell-0.0.2/i18n.c
===================================================================
--- vdr-plugin-freecell-0.0.2.orig/i18n.c	2012-03-29 20:06:35.393957756 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,146 +0,0 @@
-#include "i18n.h"
-
-const tI18nPhrase Phrases[] = {
-	{	"Freecell",							// English
-		"Freecell",							// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Vapaakenttä",					// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"The well-known card game",// English
-		"Das bekannte Kartenspiel",// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Korttipeliklassikko",	// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"You won!",							// English
-		"Sie haben gewonnen!",	// Deutsch
-		"",											// Slovenski
-		"",											// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Voitit!",							// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika / Greek
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"Select stack",					// English
-		"Stapel wählen",				// Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Valitse pino",					// Suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"Press OK to start a new game",// English
-		"OK beginnt ein neues Spiel",// Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Käynnistä uusi peli painamalla 'OK'",// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{	"This is Freecell, a card-game for the Video Disk Recorder.\n\n"
-		"Up/Down: Jump cells/stacks\n"
-		"Left/Right: Move hand/cards\n"
-		"OK: Grab card or stack\n"
-		"Up/Down: Select/Deselect cards\n"
-		"OK: Confirm stack selection\n"
-		"Blue: Start new game\n\n"
-		"Press OK to resume playing", // English
-		"Dies ist Freecell, ein Kartenspiel für den Video Disk Recorder.\n\n"
-		"Rauf/Runter: Zellen/Stapel springen\n"
-		"Links/Rechts: Hand/Stapel bewegen\n"
-		"OK: Karte oder Stapel nehmen\n"
-		"Rauf/Runter: Karten An-/Abwählen\n"
-		"OK: Stapelwahl bestätigen\n"
-		"Blau: Neues Spiel starten\n\n"
-		"OK drücken um fortzusetzen", // Deutsch
-		"",											// Slovenski
-		"", 										// Italiano
-		"",											// Nederlands
-		"",											// Português
-		"",											// Français
-		"",											// Norsk
-		"Vapaakenttä-pasianssi.\n\n" 
-		"'Ylös/Alas': Siirry kenttien/pinojen välillä\n" 
-		"'Vasen/Oikea': Siirrä kursoria/korttia\n" 
-		"'OK': Valitse kortti tai pino\n" 
-		"'Ylös/Alas': Valitse kortti\n" 
-		"'OK': Hyväksy pinon valinta\n" 
-		"'Sininen': Käynnistä uusi peli\n\n" 
-		"Paina 'OK' jatkaaksesi peliä",// suomi
-		"",											// Polski
-		"",											// Español
-		"",											// Ellinika
-		"",											// Svenska
-		"",											// Romaneste
-		"",											// Magyar
-		"",											// Catala
-#if VDRVERSNUM >= 10300
-		""                      // Russian
-#endif
-	},
-	{ NULL }
-};
Index: vdr-plugin-freecell-0.0.2/i18n.h
===================================================================
--- vdr-plugin-freecell-0.0.2.orig/i18n.h	2012-03-29 20:06:35.413957756 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,9 +0,0 @@
-#ifndef VDR_FREECELL_I18N_H
-#define VDR_FREECELL_I18N_H
-
-#include <vdr/i18n.h>
-
-extern const tI18nPhrase Phrases[];
-
-#endif // VDR_FREECELL_I18N_H
-
