--- gmp-5.0.1.orig/Makefile.in	2010-08-12 18:05:36.000000000 +0200
+++ gmp-5.0.1/Makefile.in	2010-08-12 18:05:56.000000000 +0200
@@ -434,7 +434,7 @@
 LIBMP_LT_CURRENT = 4
 LIBMP_LT_REVISION = 21
 LIBMP_LT_AGE = 1
-SUBDIRS = tests mpn mpz mpq mpf printf scanf cxx mpbsd demos tune doc
+SUBDIRS = mpn mpz mpq mpf printf scanf cxx mpbsd tune
 
 # The ansi2knr setups for the build programs are the same as the normal
 # automake ansi2knr rules, but using $(CC_FOR_BUILD) instead of $(CC).
--- gmp-5.0.1.orig/Makefile.am	2010-08-12 18:06:06.000000000 +0200
+++ gmp-5.0.1/Makefile.am	2010-08-12 18:06:25.000000000 +0200
@@ -92,7 +92,7 @@
 LIBMP_LT_AGE =      1
 
 
-SUBDIRS = tests mpn mpz mpq mpf printf scanf cxx mpbsd demos tune doc
+SUBDIRS = mpn mpz mpq mpf printf scanf cxx mpbsd tune
 
 EXTRA_DIST = configfsf.guess configfsf.sub .gdbinit INSTALL.autoconf
 
