--- a/extended_frontend.h	2017-01-07 10:06:17.000000000 +0100
+++ b/extended_frontend.h	2020-11-21 08:35:20.223203663 +0100
@@ -43,6 +43,70 @@
  *
  *****************************************************************************/

+#if DVB_API < 60
+
+#define FEC_2_5                    (FEC_9_10 + 1)
+
+#define QAM_4_NR                   (DQPSK + 1)
+
+#define TRANSMISSION_MODE_C1       (TRANSMISSION_MODE_32K + 1)
+#define TRANSMISSION_MODE_C3780    (TRANSMISSION_MODE_32K + 2)
+
+#define GUARD_INTERVAL_PN420       (GUARD_INTERVAL_19_256 + 1)
+#define GUARD_INTERVAL_PN595       (GUARD_INTERVAL_19_256 + 2)
+#define GUARD_INTERVAL_PN945       (GUARD_INTERVAL_19_256 + 3)
+
+enum fe_interleaving {
+    INTERLEAVING_NONE,
+    INTERLEAVING_AUTO,
+    INTERLEAVING_240,
+    INTERLEAVING_720,
+};
+
+#define SYS_DTMB SYS_DMBTH
+
+#define DTV_STREAM_ID DTV_ISDBS_TS_ID
+#define DTV_ISDBS_TS_ID_LEGACY DTV_ISDBS_TS_ID
+#define DTV_DVBT2_PLP_ID_LEGACY DTV_DVBT2_PLP_ID
+
+#endif
+
+
+#ifndef DTV_ATSCMH_FIC_VER
+#define DTV_ATSCMH_FIC_VER      45
+#define DTV_ATSCMH_PARADE_ID        46
+#define DTV_ATSCMH_NOG          47
+#define DTV_ATSCMH_TNOG         48
+#define DTV_ATSCMH_SGN          49
+#define DTV_ATSCMH_PRC          50
+#define DTV_ATSCMH_RS_FRAME_MODE    51
+#define DTV_ATSCMH_RS_FRAME_ENSEMBLE    52
+#define DTV_ATSCMH_RS_CODE_MODE_PRI 53
+#define DTV_ATSCMH_RS_CODE_MODE_SEC 54
+#define DTV_ATSCMH_SCCC_BLOCK_MODE  55
+#define DTV_ATSCMH_SCCC_CODE_MODE_A 56
+#define DTV_ATSCMH_SCCC_CODE_MODE_B 57
+#define DTV_ATSCMH_SCCC_CODE_MODE_C 58
+#define DTV_ATSCMH_SCCC_CODE_MODE_D 59
+#endif
+
+#ifndef DTV_INTERLEAVING
+#define DTV_INTERLEAVING            60
+#define DTV_LNA                 61
+#endif
+
+#ifndef DTV_STAT_SIGNAL_STRENGTH
+#define DTV_STAT_SIGNAL_STRENGTH    62
+#define DTV_STAT_CNR            63
+#define DTV_STAT_PRE_ERROR_BIT_COUNT    64
+#define DTV_STAT_PRE_TOTAL_BIT_COUNT    65
+#define DTV_STAT_POST_ERROR_BIT_COUNT   66
+#define DTV_STAT_POST_TOTAL_BIT_COUNT   67
+#define DTV_STAT_ERROR_BLOCK_COUNT  68
+#define DTV_STAT_TOTAL_BLOCK_COUNT  69
+#endif
+
+
 #ifndef fe_polarization // 300468 v181  6.2.13.2 Satellite delivery system descriptor
 typedef enum fe_polarization {
         POLARIZATION_HORIZONTAL,
