--- vdr-autostart-0.9.6/Makefile.kk	2013-06-13 11:26:22.247664412 +0200
+++ vdr-autostart-0.9.6/Makefile	2013-06-13 11:27:26.731665518 +0200
@@ -70,6 +70,7 @@
 # Options for DVDRead
 LIBS += $(shell pkg-config --libs dvdread)
 CXXFLAGS += $(shell pkg-config --cflags dvdread)
+CXXFLAGS += -fPIC
 
 ### The main target:
 
--- vdr-autostart-0.9.6/detector/Makefile.kk	2013-06-13 11:31:18.979669506 +0200
+++ vdr-autostart-0.9.6/detector/Makefile	2013-06-13 11:27:37.147665698 +0200
@@ -12,6 +12,7 @@
 # Options for DVDRead
 LIBS += $(shell pkg-config --libs dvdread)
 CXXFLAGS += $(shell pkg-config --cflags dvdread)
+CXXFLAGS += -fPIC
 
 OBJS = cdiotester.o configfileparser.o dbusdevkit.o \
 		filetester.o mediadetector.o mediatester.o \
--- vdr-autostart-0.9.6/Makefile.org	2013-06-13 11:59:24.507698439 +0200
+++ vdr-autostart-0.9.6/Makefile	2013-06-13 11:59:16.255698297 +0200
@@ -13,7 +13,9 @@
 PKGCFG = $(if $(VDRDIR),$(shell pkg-config --variable=$(1) $(VDRDIR)/vdr.pc),$(shell pkg-config --variable=$(1) vdr || pkg-config --variable=$(1) ../../../vdr.pc))
 CFGDIR = $(call PKGCFG,configdir)/plugins/$(PLUGIN)
 PLGCFG = $(call PKGCFG,plgcfg)
-
+LIBDIR = $(DESTDIR)$(call PKGCFG,libdir)
+LOCDIR = $(DESTDIR)$(call PKGCFG,locdir)
+PLGCFG = $(call PKGCFG,plgcfg)
 ifneq (exists, $(shell pkg-config dbus-1 && echo exists))
   $(warning ******************************************************************)
   $(warning 'dbus development' not detected! ')
@@ -95,7 +97,7 @@
 PODIR     = po
 I18Npo    = $(wildcard $(PODIR)/*.po)
 I18Nmo    = $(addsuffix .mo, $(foreach file, $(I18Npo), $(basename $(file))))
-I18Nmsgs  = $(addprefix $(DESTDIR)$(LOCDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
+I18Nmsgs  = $(addprefix $(LOCDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/$(PLUGIN).pot
 
 %.mo: %.po
@@ -108,7 +110,7 @@
 	msgmerge -U --no-wrap --no-location --backup=none -q -N $@ $<
 	@touch $@
 
-$(I18Nmsgs): $(DESTDIR)$(LOCDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
+$(I18Nmsgs): $(LOCDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
 	install -D -m644 $< $@
 
 .PHONY: i18n
@@ -126,7 +128,7 @@
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) detector.a $(LIBS) -o $@
 
 install-lib: $(SOFILE)
-	install -D $^ $(DESTDIR)$(LIBDIR)/$^.$(APIVERSION)
+	install -D $^ $(LIBDIR)/$^.$(APIVERSION)
 
 install: install-lib install-i18n
 
