diff -Naur showtime-r5753/support/linux.mk showtime-r5753a/support/linux.mk
--- showtime-r5753/support/linux.mk	2010-10-04 15:26:15.000000000 +0200
+++ showtime-r5753a/support/linux.mk	2010-10-09 17:44:14.641817063 +0200
@@ -5,12 +5,12 @@
 MAN=man/showtime.1
 
 install: ${PROG} ${MAN}
-	mkdir -p $(INSTBIN)
-	install -s ${PROG} $(INSTBIN)
+	mkdir -p ${DESTDIR}$(INSTBIN)
+	install -s ${PROG} ${DESTDIR}$(INSTBIN)
 
-	mkdir -p $(INSTMAN)
-	install ${MAN} $(INSTMAN)
+	mkdir -p ${DESTDIR}$(INSTMAN)
+	install ${MAN} ${DESTDIR}$(INSTMAN)
 
 uninstall:
-	rm -f $(INSTBIN)/${PROG}
-	rm -f $(INSTMAN)/${MAN}
+	rm -f ${DESTDIR}$(INSTBIN)/${PROG}
+	rm -f ${DESTDIR}$(INSTMAN)/${MAN}
