--- libeXosip2-3.5.0/configure.in	2011-01-04 13:06:06.000000000 +0100
+++ /home/cspiel/dev/openbricks/build/build.brettl3.eglibc/libeXosip2-3.5.0/configure.in	2012-02-24 11:09:58.840307975 +0100
@@ -44,7 +44,6 @@
 AC_CONFIG_AUX_DIR(scripts)
 AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_ACLOCAL_INCLUDE(scripts)

 AC_CANONICAL_HOST

@@ -117,7 +116,6 @@
     dnl FSM_LIB="-lpthread"
   else
     EXOSIP_FLAGS="$EXOSIP_FLAGS -DOSIP_MT"
-    ACX_PTHREAD()
   fi
 fi

