--- a/src/Makefile.am	2010-01-06 15:57:57.000000000 +0100
+++ b/src/Makefile.am	2012-02-24 11:48:31.048265735 +0100
@@ -9,7 +9,7 @@
 #	-Wstrict-prototypes
 #	-Werror
 #
-AM_CFLAGS = @EXOSIP_FLAGS@ @CFLAGS@  @PTHREAD_CFLAGS@
+AM_CFLAGS = @EXOSIP_FLAGS@ @CFLAGS@ `pkg-config --cflags pthread-stubs`
 
 libeXosip2_la_SOURCES =           \
 eXosip.c         eXconf.c         \
@@ -47,7 +47,7 @@
 sdp_offans.c
 endif
 
-libeXosip2_la_LDFLAGS = -version-info $(LIBEXOSIP_SO_VERSION) -no-undefined
-libeXosip2_la_LIBADD = @EXOSIP_LIB@ @PTHREAD_LIBS@ $(OSIP_LIBS)
+libeXosip2_la_LDFLAGS = -version-info $(LIBEXOSIP_SO_VERSION) `pkg-config --libs pthread-stubs` -no-undefined
+libeXosip2_la_LIBADD = @EXOSIP_LIB@ $(OSIP_LIBS)
 
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/include $(OSIP_CFLAGS)
