--- a/dump-vlc-m3u.c	2014-10-28 11:25:20.975450544 +0100
+++ b/dump-vlc-m3u.c	2014-10-28 11:25:20.991450544 +0100
@@ -78,7 +78,12 @@
 
 static const char * vlc_delsys (int guard_interval) {
         switch(guard_interval) {          
+#ifdef SYS_DVBC_ANNEX_AC
                 case SYS_DVBC_ANNEX_A:    return "dvb-c";
+                case SYS_DVBC_ANNEX_C:    return "dvb-c";
+#else
+                case SYS_DVBC_ANNEX_AC:   return "dvb-c";
+#endif
                 case SYS_DVBC_ANNEX_B:    return "dvb-c";
                 case SYS_DVBT        :    return "dvb-t";
                 case SYS_DVBS        :    return "dvb-s";
@@ -88,7 +93,6 @@
                 case SYS_ISDBC       :    return "isdb-t";
                 case SYS_ATSC        :    return "atsc";
                 case SYS_DVBT2       :    return "dvb-t2";
-                case SYS_DVBC_ANNEX_C:    return "dvb-c";
                 default:                  return "unknown";
                 }                        
 }
