--- vdr-dbus2vdr-1.0/Makefile	2012-03-22 00:14:20.000000000 +0100
+++ vdr-dbus2vdr-1.0/Makefile.ok	2012-03-27 16:13:01.579695225 +0200
@@ -20,14 +20,16 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -g -O3 -Wall -Woverloaded-virtual -Wno-parentheses
+#CXXFLAGS ?= -g -O3 -Wall -Woverloaded-virtual -Wno-parentheses
+CXXFLAGS ?= -lintl -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LDFLAGS = -lintl 
 CXXADD   += `pkg-config --cflags dbus-1` `libpng-config --cflags`
 LDADD    += `pkg-config --libs dbus-1` `libpng-config --ldflags`
 
 ### The directory environment:
 
-VDRDIR ?= ../../..
-LIBDIR ?= ../../lib
+#VDRDIR ?= ../../..
+#LIBDIR ?= ../../lib
 TMPDIR ?= /tmp
 
 ### Make sure that necessary options are included:
