
  Author: Goswin Brederlow
  Status: pending

diff -Nurd grub-0.95+cvs20040624/stage2/char_io.c grub-0.95+cvs20040624-mrvn/stage2/char_io.c
--- grub-0.95+cvs20040624/stage2/char_io.c	2004-05-23 18:45:44.000000000 +0200
+++ grub-0.95+cvs20040624-mrvn/stage2/char_io.c	2004-08-10 11:58:16.000000000 +0200
@@ -1175,10 +1175,10 @@
 #endif /* ! STAGE1_5 */
 
 int
-memcheck (int addr, int len)
+memcheck (unsigned long addr, unsigned long len)
 {
 #ifdef GRUB_UTIL
-  static int start_addr (void)
+  static unsigned long start_addr (void)
     {
       int ret;
 # if defined(HAVE_START_SYMBOL)
@@ -1189,7 +1189,7 @@
       return ret;
     }
 
-  static int end_addr (void)
+  static unsigned long end_addr (void)
     {
       int ret;
 # if defined(HAVE_END_SYMBOL)
diff -Nurd grub-0.95+cvs20040624/stage2/shared.h grub-0.95+cvs20040624-mrvn/stage2/shared.h
--- grub-0.95+cvs20040624/stage2/shared.h	2004-06-20 15:48:47.000000000 +0200
+++ grub-0.95+cvs20040624-mrvn/stage2/shared.h	2004-08-10 11:58:16.000000000 +0200
@@ -911,7 +911,7 @@
 int nul_terminate (char *str);
 int get_based_digit (int c, int base);
 int safe_parse_maxint (char **str_ptr, int *myint_ptr);
-int memcheck (int start, int len);
+int memcheck (unsigned long start, unsigned long len);
 void grub_putstr (const char *str);
 
 #ifndef NO_DECOMPRESSION
