--- grub2-r2404.orig/configure	2009-07-12 09:01:38.000000000 -0400
+++ grub2-r2404/configure	2009-07-26 17:11:32.000000000 -0400
@@ -9269,8 +9269,8 @@
 $as_echo "$as_me: error: freetype2 libraries are required to build \`grub-mkfont'" >&2;}
    { (exit 1); exit 1; }; }
   fi
-  freetype_cflags=`freetype-config --cflags`
-  freetype_libs=`freetype-config --libs`
+  freetype_cflags=`$FREETYPE --cflags`
+  freetype_libs=`$FREETYPE --libs`
 fi
 
 
