diff -Naur grub-1.96.orig/configure grub-1.96/configure
--- grub-1.96.orig/configure	2008-02-03 08:54:58.000000000 -0500
+++ grub-1.96/configure	2009-06-09 08:42:46.000000000 -0400
@@ -4575,7 +4575,7 @@
    { (exit 1); exit 1; }; } ;;
 esac
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -6788,7 +6788,7 @@
 if test "${grub_cv_i386_check_nested_functions+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
