diff -Naur grub-1.96.orig/include/grub/i386/linux.h grub-1.96/include/grub/i386/linux.h
--- grub-1.96.orig/include/grub/i386/linux.h	2007-07-21 14:22:34.000000000 -0400
+++ grub-1.96/include/grub/i386/linux.h	2009-06-13 13:05:49.000000000 -0400
@@ -40,8 +42,9 @@
 #define GRUB_LINUX_VID_MODE_ASK		0xFFFD
 
 #define GRUB_LINUX_CL_OFFSET		0x9000
-#define GRUB_LINUX_CL_END_OFFSET	0x90FF
-#define GRUB_LINUX_SETUP_MOVE_SIZE	0x9100
+#define GRUB_LINUX_CL_END_OFFSET \
+    (GRUB_LINUX_CL_OFFSET + COMMAND_LINE_SIZE)
+#define GRUB_LINUX_SETUP_MOVE_SIZE (GRUB_LINUX_CL_END_OFFSET + 1)
 #define GRUB_LINUX_CL_MAGIC		0xA33F
 
 #define GRUB_LINUX_EFI_SIGNATURE	\
diff -Naur grub-1.96.orig/kern/i386/loader.S grub-1.96/kern/i386/loader.S
--- grub-1.96.orig/kern/i386/loader.S	2008-01-06 02:15:07.000000000 -0500
+++ grub-1.96/kern/i386/loader.S	2009-06-13 13:07:57.000000000 -0400
@@ -83,7 +83,7 @@
 	/* copy the real mode code */
 	movl	EXT_C(grub_linux_tmp_addr), %esi
 	movl	%ebx, %edi
-	movl	$GRUB_LINUX_SETUP_MOVE_SIZE, %ecx
+	movl	$(GRUB_LINUX_CL_OFFSET + COMMAND_LINE_SIZE + 1), %ecx
 	cld
 	rep
 	movsb
