diff -Naur tntnet-1.6.2/framework/common/Makefile.am tntnet-1.6.2a/framework/common/Makefile.am
--- tntnet-1.6.2/framework/common/Makefile.am	2008-02-22 21:17:04.000000000 +0100
+++ tntnet-1.6.2a/framework/common/Makefile.am	2008-05-14 18:33:51.000000000 +0200
@@ -106,7 +106,7 @@
 	tnt/tcpjob.h
 
 libtntnet_la_CXXFLAGS = -DPKGLIBDIR=\"@libdir@/@PACKAGE@\" $(PTHREAD_CFLAGS)
-libtntnet_la_LDFLAGS = -version-info @sonumber@ -lz -lcxxtools $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
+libtntnet_la_LDFLAGS = -version-info @sonumber@ -lz -lcxxtools -ldl $(PTHREAD_LIBS) @SHARED_LIB_FLAG@
 
 if MAKE_GNUTLS
 libtntnet_la_SOURCES += \
diff -Naur tntnet-1.6.2/framework/common/Makefile.in tntnet-1.6.2a/framework/common/Makefile.in
--- tntnet-1.6.2/framework/common/Makefile.in	2008-04-20 22:53:16.000000000 +0200
+++ tntnet-1.6.2a/framework/common/Makefile.in	2008-05-14 18:33:42.000000000 +0200
@@ -339,7 +339,7 @@
 	tnt/tcpjob.h $(am__append_2) $(am__append_6)
 libtntnet_la_CXXFLAGS = -DPKGLIBDIR=\"@libdir@/@PACKAGE@\" \
 	$(PTHREAD_CFLAGS) $(am__append_4)
-libtntnet_la_LDFLAGS = -version-info @sonumber@ -lz -lcxxtools \
+libtntnet_la_LDFLAGS = -version-info @sonumber@ -lz -lcxxtools -ldl \
 	$(PTHREAD_LIBS) @SHARED_LIB_FLAG@ $(am__append_3) \
 	$(am__append_7)
 all: config.h
diff -Naur tntnet-1.6.2/tntnet-config.in tntnet-1.6.2a/tntnet-config.in
--- tntnet-1.6.2/tntnet-config.in	2008-03-19 22:14:58.000000000 +0100
+++ tntnet-1.6.2a/tntnet-config.in	2008-05-14 18:33:05.000000000 +0200
@@ -108,7 +108,7 @@
 ECPPC=env \${EXTRA_ENV} ecppc
 TNTNET=env \${EXTRA_ENV} tntnet
 CXXFLAGS+=-I$includedir -fPIC -O2
-LDFLAGS+=@SHARED_LIB_FLAG@ -L$libdir -ltntnet -lcxxtools
+LDFLAGS+=@SHARED_LIB_FLAG@ -L$libdir -ltntnet -lcxxtools -lz -ldl
 
 .ecpp.cpp:
 	\${ECPPC} \${ECPPFLAGS} \${ECPPFLAGS_CPP} \$<
@@ -180,7 +212,7 @@
         ;;
 
     --libs)
-        echo -L$libdir -ltntnet -lcxxtools
+        echo -L$libdir -ltntnet -lcxxtools -lz -ldl
         ;;
 
     --makefile=*)
