diff -Naur gmp-5.0.1/configure gmp-5.0.1a/configure
--- gmp-5.0.1/configure	2010-02-06 13:43:19.000000000 +0100
+++ gmp-5.0.1a/configure	2010-08-01 15:58:34.313007768 +0200
@@ -3142,17 +3142,6 @@
 
 
 
-# If --target is not used then $target_alias is empty, but if say
-# "./configure athlon-pc-freebsd3.5" is used, then all three of
-# $build_alias, $host_alias and $target_alias are set to
-# "athlon-pc-freebsd3.5".
-#
-if test -n "$target_alias" && test "$target_alias" != "$host_alias"; then
-  as_fn_error "--target is not appropriate for GMP
-Use --build=CPU-VENDOR-OS if you need to specify your CPU and/or system
-explicitly.  Use --host if cross-compiling (see \"Installing GMP\" in the
-manual for more on this)." "$LINENO" 5
-fi
 
 gmp_configm4="config.m4"
 gmp_tmpconfigm4=cnfm4.tmp
