--- a/i18n.h.org	2009-05-01 19:30:58.000000000 +0200
+++ a/i18n.h	2012-04-01 11:11:55.080297631 +0200
@@ -20,19 +20,7 @@
 	// Implemented as a Meyers-Singleton
 	class i18n
 	{
-	private:
-		static const tI18nPhrase m_phrases[];
-
-		int m_osdLanguage;
-
-	protected:
-		static i18n& get();
-
 	public:
-		i18n();
-
-		static const tI18nPhrase* get_phrases() { return m_phrases; }
-
 		static const char* translate( const char* text );
 	};
 
