diff -Naur libdvdnav-r1099/src/dvdnav.h libdvdnav-r1099a/src/dvdnav.h
--- libdvdnav-r1099/src/dvdnav.h	2008-06-12 22:46:29.000000000 +0200
+++ libdvdnav-r1099a/src/dvdnav.h	2008-06-22 18:23:04.000000000 +0200
@@ -290,7 +290,7 @@
  * The number of entries in *times is the result of the function.
  * On error *times is NULL and the output is 0
  */
-uint32_t dvdnav_describe_title_chapters(dvdnav_t *this, int32_t title, uint64_t **times, uint64_t *duration);
+uint32_t dvdnav_describe_title_chapters(dvdnav_t *self, int32_t title, uint64_t **times, uint64_t *duration);
 
 /*
  * Play the specified amount of parts of the specified title of
