--- pulseaudio-0.9.21/src/Makefile.in.orig	2009-11-23 05:19:00.000000000 +0100
+++ pulseaudio-0.9.21/src/Makefile.in	2010-05-10 13:45:30.000000000 +0200
@@ -2517,7 +2517,7 @@
 xdgautostartdir = $(sysconfdir)/xdg/autostart
 alsaprofilesetsdir = $(datadir)/pulseaudio/alsa-mixer/profile-sets
 alsapathsdir = $(datadir)/pulseaudio/alsa-mixer/paths
-udevrulesdir = /lib/udev/rules.d
+udevrulesdir = $(prefix)/lib/udev/rules.d
 dbuspolicydir = $(sysconfdir)/dbus-1/system.d
 
 ###################################
