diff -Naur pulseaudio-0.9.21/src/Makefile.in pulseaudio-0.9.21a/src/Makefile.in
--- pulseaudio-0.9.21/src/Makefile.in	2010-05-18 15:28:30.216664178 +0200
+++ pulseaudio-0.9.21a/src/Makefile.in	2010-05-18 15:28:20.977454253 +0200
@@ -2556,17 +2556,17 @@
 	$(LIBSNDFILE_CFLAGS) \
 	$(LIBSPEEX_CFLAGS) \
 	-DPA_BUILDDIR=\"$(abs_builddir)\" \
-	-DPA_DLSEARCHPATH=\"$(modlibexecdir)\" \
-	-DPA_DEFAULT_CONFIG_DIR=\"$(PA_DEFAULT_CONFIG_DIR)\" \
-	-DPA_BINARY=\"$(PA_BINARY)\" \
-	-DPA_SYSTEM_RUNTIME_PATH=\"$(PA_SYSTEM_RUNTIME_PATH)\" \
-	-DPA_SYSTEM_CONFIG_PATH=\"$(PA_SYSTEM_CONFIG_PATH)\" \
-	-DPA_SYSTEM_STATE_PATH=\"$(PA_SYSTEM_STATE_PATH)\" \
+	-DPA_DLSEARCHPATH=\"/usr/lib/pulse\" \
+	-DPA_DEFAULT_CONFIG_DIR=\"/etc/pulse\" \
+	-DPA_BINARY=\"/usr/bin/pulseaudio\" \
+	-DPA_SYSTEM_RUNTIME_PATH=\"/var/run/pulse\" \
+	-DPA_SYSTEM_CONFIG_PATH=\"/var/lib/pulse\" \
+	-DPA_SYSTEM_STATE_PATH=\"/var/lib/pulse\" \
 	-DAO_REQUIRE_CAS \
-	-DPULSE_LOCALEDIR=\"$(pulselocaledir)\" \
-	-DPA_MACHINE_ID=\"$(localstatedir)/lib/dbus/machine-id\" \
-        -DPA_ALSA_PATHS_DIR=\"$(alsapathsdir)\" \
-        -DPA_ALSA_PROFILE_SETS_DIR=\"$(alsaprofilesetsdir)\"
+	-DPULSE_LOCALEDIR=\"/usr/share/locale\" \
+	-DPA_MACHINE_ID=\"/var/lib/dbus/machine-id\" \
+        -DPA_ALSA_PATHS_DIR=\"/usr/share/pulseaudio/alsa-mixer/paths\" \
+        -DPA_ALSA_PROFILE_SETS_DIR=\"/usr/share/pulseaudio/alsa-mixer/profile-sets\"
 
 AM_LIBADD = $(PTHREAD_LIBS) $(INTLLIBS)
 AM_LDADD = $(PTHREAD_LIBS) $(INTLLIBS)
