--- pulseaudio-2.0/src/Makefile.in	2012-05-11 15:31:10.000000000 +0200
+++ pulseaudio-2.0/src/Makefile.in-new	2012-08-11 08:44:11.321526440 +0200
@@ -2974,11 +2974,6 @@
 @HAVE_DBUS_TRUE@dbuspolicy_DATA = \
 @HAVE_DBUS_TRUE@		daemon/pulseaudio-system.conf
 
-@HAVE_X11_TRUE@xdgautostart_in_files = \
-@HAVE_X11_TRUE@		daemon/pulseaudio.desktop.in \
-@HAVE_X11_TRUE@		daemon/pulseaudio-kde.desktop.in
-
-@HAVE_X11_TRUE@xdgautostart_DATA = $(xdgautostart_in_files:.desktop.in=.desktop)
 
 ###################################
 #          Includes               #
