--- a/libcedarv/linux-armhf/Makefile	2013-10-12 23:49:56.694757365 +0200
+++ b/libcedarv/linux-armhf/Makefile	2013-10-12 23:49:56.698757364 +0200
@@ -23,12 +23,16 @@
 OBJS = $(SRCS:%.c=%.o)
 
 LIB = libcedarv.a
+DLL = libcedarx.so
 
-all: $(LIB)
+all: $(DLL) $(LIB)
 
 $(LIB): $(OBJS)
 	$(AR) cr $(LIB) $(OBJS)
 
+$(DLL): $(OBJS)
+	$(CC) -o $(DLL) -shared -fPIC $(OBJS) -L./libvecore -lvecore
+
 clean:
 	rm -f $(LIB) $(OBJS)
 
