--- a/libcedarv/linux-armhf/adapter/libve_adapter.c	2013-06-16 13:04:17.887411813 +0200
+++ b/libcedarv/linux-armhf/adapter/libve_adapter.c	2013-06-16 13:04:17.903411816 +0200
@@ -121,7 +121,7 @@
 		return -1;
 
 	ioctl(cedarv_osal_ctx->fd, IOCTL_GET_ENV_INFO, (unsigned long)&cedarv_osal_ctx->env_info);
-	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (int)cedarv_osal_ctx->env_info.address_macc);
+	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (off_t)cedarv_osal_ctx->env_info.address_macc);
 
 	pthread_mutex_init(&cedarv_osal_mutex, NULL);
 
--- a/libcedarv/linux-armel/adapter/libve_adapter.c	2013-06-16 13:08:20.063415585 +0200
+++ b/libcedarv/linux-armel/adapter/libve_adapter.c	2013-06-16 13:08:20.075415585 +0200
@@ -123,7 +123,7 @@
 		return -1;
 
 	ioctl(cedarv_osal_ctx->fd, IOCTL_GET_ENV_INFO, (unsigned long)&cedarv_osal_ctx->env_info);
-	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (int)cedarv_osal_ctx->env_info.address_macc);
+	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (off_t)cedarv_osal_ctx->env_info.address_macc);
 
 	pthread_mutex_init(&cedarv_osal_mutex, NULL);
 
--- a/libcedarv/android/adapter/libve_adapter.c	2013-06-16 13:10:47.631417882 +0200
+++ b/libcedarv/android/adapter/libve_adapter.c	2013-06-16 13:10:47.647417882 +0200
@@ -120,7 +120,7 @@
 		return -1;
 
 	ioctl(cedarv_osal_ctx->fd, IOCTL_GET_ENV_INFO, (unsigned long)&cedarv_osal_ctx->env_info);
-	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (int)cedarv_osal_ctx->env_info.address_macc);
+	cedarv_osal_ctx->env_info.address_macc = (unsigned int)mmap(NULL, 2048, PROT_READ | PROT_WRITE, MAP_SHARED, cedarv_osal_ctx->fd, (off_t)cedarv_osal_ctx->env_info.address_macc);
 
 	pthread_mutex_init(&cedarv_osal_mutex, NULL);
 
