diff -r fa78c26853b1 src/hmac_sha256.c
--- a/src/hmac_sha256.c	Thu Nov 26 22:57:30 2009 +0100
+++ b/src/hmac_sha256.c	Sat Nov 28 10:10:18 2009 +0100
@@ -28,6 +28,13 @@
 vh_hmac_sha256_new (const char *key)
 {
   gcry_md_hd_t hd;
+  
+  if (!gcry_control (GCRYCTL_INITIALIZATION_FINISHED_P))
+  {
+    gcry_check_version (GCRYPT_VERSION);
+    gcry_control (GCRYCTL_DISABLE_SECMEM, 0);
+    gcry_control (GCRYCTL_INITIALIZATION_FINISHED, 0);
+  }
 
   gcry_md_open (&hd, GCRY_MD_SHA256, GCRY_MD_FLAG_HMAC);
   if (!hd)
