diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -887,7 +887,6 @@
 
 echolog "Checking for libavformat ..."
 check_lib libavformat/avformat.h av_metadata_conv -lavformat || die "Error, can't find libavformat !"
-add_pkgconfig_requires libavformat
 
 
 #################################################
@@ -903,7 +902,6 @@
 
 echolog "Checking for sqlite3 ..."
 check_lib sqlite3.h sqlite3_version -lsqlite3 || die "Error, can't find sqlite3 !"
-add_pkgconfig_requires sqlite3
 
 
 #################################################
@@ -1017,7 +1015,6 @@
   if [ "$?" = 0 ]; then
     echolog "(yes)"
     add_cflags -DUSE_XML
-    add_pkgconfig_requires libxml-2.0
   else
     echolog "(no)"
     grabber_forced $xml_dep_grabber
@@ -1049,7 +1046,6 @@
   check_lib libavutil/md5.h av_md5_sum -lavutil
   if [ "$?" = 0 ]; then
     echolog "(yes)"
-    add_pkgconfig_requires libavutil
   else
     echolog "(no)"
     grabber_forced $avutil_dep_grabber
@@ -1104,7 +1100,6 @@
   check_lib libexif/exif-data.h exif_entry_get_value -lexif
   if [ "$?" = 0 ]; then
     echolog "(yes)"
-    add_pkgconfig_requires libexif
   else
     echolog "(no)"
     grabber_forced $exif_dep_grabber
@@ -1137,7 +1132,6 @@
   if [ "$?" = 0 ]; then
     echolog "(yes)"
     add_cflags -DUSE_LAVC
-    add_pkgconfig_requires libavcodec
   else
     echolog "(no)"
     grabber_forced $avcodec_dep_grabber
@@ -1165,7 +1159,6 @@
     check_lib nfo.h nfo_init -lnfo
     [ "$?" != 0 ] && die "Error, libnfo available but the compilation failed !"
     echolog "(yes)"
-    add_pkgconfig_requires libnfo
   else
     echolog "(no)"
     grabber_forced $nfo_dep_grabber
@@ -1194,7 +1187,6 @@
   if [ "$?" = 0 ]; then
     echolog "(yes)"
     add_cflags -DUSE_GRABBER
-    add_pkgconfig_requires libcurl
   else
     echolog "(no)"
     if enabled grabber_all; then
