--- /home/ben/GeeXboX/openbricks-next-x32-enna/build/build.netbook.eglibc/elementary-r56610/data/themes/default.edc	2011-01-31 19:45:04.000000000 +0100
+++ a/data/themes/default.edc	2011-01-31 20:51:07.119173002 +0100
@@ -12502,7 +12502,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
@@ -12696,7 +12696,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
@@ -16375,7 +16375,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
@@ -17485,7 +17485,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
@@ -18036,7 +18036,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
@@ -18653,7 +18653,7 @@
                   size: 10;
                   min: 1 1;
 //                  min: 0 1;
-                  align: -1.0 0.5;
+                  align: 0.0 0.5;
                   text_class: "list_item";
                }
             }
