--- elementary-r56610/src/bin/Makefile.am.orig	2012-02-17 11:12:44.074256001 +0200
+++ elementary-r56610/src/bin/Makefile.am	2012-02-17 11:14:09.034256003 +0200
@@ -95,18 +95,18 @@
 test_bubble.c \
 test_store.c
 
-elementary_test_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@ @my_libs@
+elementary_test_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@ @ELEMENTARY_EFREET_LIBS@ @ELEMENTARY_LIBS@ @my_libs@
 elementary_test_LDFLAGS =
 
 elementary_config_SOURCES = \
 config.c
 
-elementary_config_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@
+elementary_config_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@ @ELEMENTARY_EFREET_LIBS@ @ELEMENTARY_LIBS@
 elementary_config_LDFLAGS =
 
 if BUILD_QUICKLAUNCH
 elementary_quicklaunch_SOURCES = quicklaunch.c
-elementary_quicklaunch_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@
+elementary_quicklaunch_LDADD = $(top_builddir)/src/lib/libelementary.la @ELEMENTARY_EWEATHER_LIBS@ @ELEMENTARY_LIBS@
 elementary_quicklaunch_LDFLAGS =
 
 if BUILD_RUN
