diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2011-06-14 11:21:10.000000000 -0500
+++ b/configure.ac	2013-07-12 14:28:23.011053315 -0500
@@ -26,7 +26,7 @@
 AC_CONFIG_SRCDIR([Makefile.am])
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE
-AM_CONFIG_HEADER([src/config.h])
+AC_CONFIG_HEADERS([src/config.h])
 
 LIBVA_SDS_VERSION_0_29=libva_sds_version_0_29
 LIBVA_SDS_VERSION_0_30=libva_sds_version_0_30
diff -ur a/src/utils_glx.h b/src/utils_glx.h
--- a/src/utils_glx.h	2011-06-14 11:21:10.000000000 -0500
+++ b/src/utils_glx.h	2013-07-12 14:28:23.011053315 -0500
@@ -25,6 +25,13 @@
 #include <GL/glext.h>
 #include <GL/glx.h>
 
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+   block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+   Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+
 #ifndef GL_FRAMEBUFFER_BINDING
 #define GL_FRAMEBUFFER_BINDING GL_FRAMEBUFFER_BINDING_EXT
 #endif
diff -ur a/src/xvba_dump.c b/src/xvba_dump.c
--- a/src/xvba_dump.c	2011-06-14 11:21:10.000000000 -0500
+++ b/src/xvba_dump.c	2013-07-12 14:28:36.025052831 -0500
@@ -150,8 +150,6 @@
         _(VAEncSequenceParameterBufferType);
         _(VAEncPictureParameterBufferType);
         _(VAEncSliceParameterBufferType);
-        _(VAEncH264VUIBufferType);
-        _(VAEncH264SEIBufferType);
 #endif
 #undef _
     }
diff -ur a/src/xvba_gate.h b/src/xvba_gate.h
--- a/src/xvba_gate.h	2011-06-14 11:21:10.000000000 -0500
+++ b/src/xvba_gate.h	2013-07-12 14:28:23.012053315 -0500
@@ -26,6 +26,13 @@
 
 #if USE_GLX
 #include <GL/glx.h>
+
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+   block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+   Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
 #endif
 
 #define XVBA_FAKE                       XVBA_FOURCC('F','A','K','E')
diff -ur a/src/xvba_video_glx.h b/src/xvba_video_glx.h
--- a/src/xvba_video_glx.h	2011-06-14 11:21:10.000000000 -0500
+++ b/src/xvba_video_glx.h	2013-07-12 14:28:23.012053315 -0500
@@ -25,6 +25,14 @@
 #include <va/va_backend.h>
 #include <GL/gl.h>
 #include <GL/glx.h>
+
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+   block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+   Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+
 #include <pthread.h>
 #include "object_heap.h"
 #include "xvba_video_x11.h"
