--- a/tools/kmap2qmap/kmap2qmap.pro	2015-10-22 09:42:04.766530937 +0200
+++ b/tools/kmap2qmap/kmap2qmap.pro	2015-10-22 09:42:04.778530937 +0200
@@ -10,3 +10,6 @@
 
 # Input
 SOURCES += main.cpp
+
+target.path=$$[QT_INSTALL_BINS]
+INSTALLS        += target
--- a/tools/kmap2qmap-host/kmap2qmap.pro	1970-01-01 01:00:00.000000000 +0100
+++ b/tools/kmap2qmap-host/kmap2qmap.pro	2015-10-22 11:12:03.426544797 +0200
@@ -0,0 +1,19 @@
+
+TEMPLATE = app
+DESTDIR = ../../bin-host
+
+include(../../src/tools/bootstrap/bootstrap.pri)
+
+DEFINES -= \
+    QT_NO_DATASTREAM \
+    QT_NO_CAST_TO_ASCII \
+    QT_NO_CAST_FROM_ASCII
+
+DEPENDPATH += $$QT_SOURCE_TREE/src/gui/embedded
+INCLUDEPATH += $$QT_SOURCE_TREE/src/gui/embedded
+
+# Input
+SOURCES += ../kmap2qmap/main.cpp
+
+target.path=$$[QT_INSTALL_BINS]
+INSTALLS        += target
