--- a/mkspecs/common/gcc-base.conf	2017-04-12 15:52:44.866987783 +0200
+++ b/mkspecs/common/gcc-base.conf	2017-04-12 15:52:44.866987783 +0200
@@ -42,7 +42,7 @@
 QMAKE_CFLAGS_YACC           += -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_HIDESYMS       += -fvisibility=hidden
 
-QMAKE_CXXFLAGS            += $$QMAKE_CFLAGS
+QMAKE_CXXFLAGS            += $$QMAKE_CFLAGS -std=gnu++98
 QMAKE_CXXFLAGS_DEPS       += $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON    += $$QMAKE_CFLAGS_WARN_ON
 QMAKE_CXXFLAGS_WARN_OFF   += $$QMAKE_CFLAGS_WARN_OFF
