--- a/src/xmlpatterns/api/qcoloroutput_p.h	2017-04-12 15:13:14.156791648 +0200
+++ b/src/xmlpatterns/api/qcoloroutput_p.h	2017-04-12 15:13:14.180791346 +0200
@@ -70,8 +70,8 @@
             ForegroundShift = 10,
             BackgroundShift = 20,
             SpecialShift    = 20,
-            ForegroundMask  = ((1 << ForegroundShift) - 1) << ForegroundShift,
-            BackgroundMask  = ((1 << BackgroundShift) - 1) << BackgroundShift
+            ForegroundMask  = 0x1f << ForegroundShift,
+            BackgroundMask  = 0x07 << BackgroundShift
         };
 
     public:
