From b4a3e260f5123d6fbd4670776232af5405a7c5a2 Mon Sep 17 00:00:00 2001
From: Matthieu Herrb <matthieu.herrb@laas.fr>
Date: Thu, 13 Jun 2013 21:42:36 +0000
Subject: Fix loading the driver without XAA

Signed-off-by: Matthieu Herrb <matthieu.herrb@laas.fr>
Reviewed-by: Thomas Klausner <wiz@NetBSD.org>
---
diff --git a/src/i740_dga.c b/src/i740_dga.c
index 532d614..c16cce3 100644
--- a/src/i740_dga.c
+++ b/src/i740_dga.c
@@ -41,10 +41,10 @@
 
 static Bool I740_OpenFramebuffer(ScrnInfoPtr, char **, unsigned char **, int *, int *, int *);
 static Bool I740_SetMode(ScrnInfoPtr, DGAModePtr);
-static void I740_Sync(ScrnInfoPtr);
 static int  I740_GetViewport(ScrnInfoPtr);
 static void I740_SetViewport(ScrnInfoPtr, int, int, int);
 #ifdef HAVE_XAA_H
+static void I740_Sync(ScrnInfoPtr);
 static void I740_FillRect(ScrnInfoPtr, int, int, int, int, unsigned long);
 static void I740_BlitRect(ScrnInfoPtr, int, int, int, int, int, int);
 #if 0
@@ -57,8 +57,8 @@ static DGAFunctionRec I740DGAFuncs = {
    I740_SetMode,
    I740_SetViewport,
    I740_GetViewport,
-   I740_Sync,
 #ifdef HAVE_XAA_H
+   I740_Sync,
    I740_FillRect,
    I740_BlitRect,
 #if 0
@@ -67,7 +67,7 @@ static DGAFunctionRec I740DGAFuncs = {
    NULL
 #endif
 #else
-   NULL, NULL, NULL
+   NULL, NULL, NULL, NULL
 #endif
 };
 
--
cgit v0.9.0.2-2-gbebe
