diff -Nur cpufreqd-2.0.0-pre2.orig/configure cpufreqd-2.0.0-pre2/configure
--- cpufreqd-2.0.0-pre2.orig/configure	Wed Oct 26 22:09:19 2005
+++ cpufreqd-2.0.0-pre2/configure	Fri Nov 18 13:04:54 2005
@@ -22701,7 +22701,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcpufreq  $LIBS"
+LIBS="-lcpufreq -lsysfs  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -22759,7 +22759,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_cpufreq_cpufreq_cpu_exists" >&5
 echo "${ECHO_T}$ac_cv_lib_cpufreq_cpufreq_cpu_exists" >&6
 if test $ac_cv_lib_cpufreq_cpufreq_cpu_exists = yes; then
-   CPUFREQD_LDFLAGS="${CPUFREQD_LDFLAGS} -lcpufreq"
+   CPUFREQD_LDFLAGS="${CPUFREQD_LDFLAGS} -lcpufreq -lsysfs"
 else
    { { echo "$as_me:$LINENO: error: You need libcpufreq from cpufrequtils to build cpufreqd" >&5
 echo "$as_me: error: You need libcpufreq from cpufrequtils to build cpufreqd" >&2;}
