* this patch also significantly reduce the binary size.

diff -Naur cpufreqd-2.0.0.orig/src/cpufreqd_log.h cpufreqd-2.0.0/src/cpufreqd_log.h
--- cpufreqd-2.0.0.orig/src/cpufreqd_log.h	2005-12-29 17:36:05.000000000 +0100
+++ cpufreqd-2.0.0/src/cpufreqd_log.h	2005-12-29 17:36:51.000000000 +0100
@@ -32,7 +32,6 @@
 
 extern void cpufreqd_log (const int prio, const char *fmt, ...);
 
-#define clog(__prio,__fmt,...) \
-	cpufreqd_log(__prio, "%-25s: "__fmt, __func__, ## __VA_ARGS__)
+#define clog(__prio,__fmt,...)  do { } while(0)
 
 #endif
