--- hplip-3.12.11/Makefile.am	2012-11-20 10:55:15.000000000 +0100
+++ hplip-3.12.11/Makefile.am-new	2013-01-10 18:52:30.845505072 +0100
@@ -552,24 +552,24 @@
 install-dist_hplip_LogDATA:
 if FULL_BUILD
 	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 	chmod 777 $(DESTDIR)$(hplip_Logdir)
 endif	#FULL_BUILD
 if HPLIP_BUILD
 	test -z "$(DESTDIR)$(hplip_Logdir)" || mkdir -p $(DESTDIR)$(hplip_Logdir)
-	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
+#	chgrp "lp" -R $(DESTDIR)$(hplip_Logdir)
 	chmod 777 $(DESTDIR)$(hplip_Logdir)
 endif	#FULL_BUILD
 
 install-dist_hplip_tmpDATA:
 if FULL_BUILD
 	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 	chmod 1777 $(DESTDIR)$(hplip_tmpdir)
 endif	#FULL_BUILD
 if HPLIP_BUILD
 	test -z "$(DESTDIR)$(hplip_tmpdir)" || mkdir -p $(DESTDIR)$(hplip_tmpdir)
-	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
+#	chgrp "lp" -R $(DESTDIR)$(hplip_tmpdir)
 	chmod 1777 $(DESTDIR)$(hplip_tmpdir)
 endif	#FULL_BUILD
 
