diff --git a/src/template/ADSPHelpers.cpp b/src/template/ADSPHelpers.cpp
index 5aa6a7d..f66e77f 100644
--- a/src/template/ADSPHelpers.cpp
+++ b/src/template/ADSPHelpers.cpp
@@ -24,7 +24,7 @@
 using namespace std;
 
 #if !defined(TARGET_WINDOWS)
-  inline float abs(float Val)
+  inline float iabs(float Val)
   {
     if(Val >= 0.0f)
     {
@@ -441,7 +441,7 @@ float CADSPHelpers::Convert_dB_TO_Value(float dB)
 
 float CADSPHelpers::Convert_Value_TO_dB(float Scale)
 {
-  return 20.0f*log10f(abs(Scale));
+  return 20.0f*log10f(iabs(Scale));
 }
 
 // reserved for future implementation
