diff -Naur xf86-video-radeonhd-0.0.1+git20070918.orig/configure xf86-video-radeonhd-0.0.1+git20070918/configure
--- xf86-video-radeonhd-0.0.1+git20070918.orig/configure	2007-10-06 19:06:07.000000000 +0200
+++ xf86-video-radeonhd-0.0.1+git20070918/configure	2007-10-06 19:06:18.000000000 +0200
@@ -21120,7 +21120,7 @@
 if test "${ac_cv_file_src_AtomBios_includes_atombios_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
@@ -21144,7 +21144,7 @@
 if test "${ac_cv_file_src_AtomBios_includes_Decoder_h+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
