--- a/src/mapi/glapi/glapi_x86-64.S
+++ b/src/mapi/glapi/glapi_x86-64.S
@@ -30885,18 +30885,6 @@ GL_PREFIX(EGLImageTargetTexture2DOES):
 	.globl GL_PREFIX(FramebufferTextureLayer) ; .set GL_PREFIX(FramebufferTextureLayer), GL_PREFIX(FramebufferTextureLayerEXT)
 	.globl GL_PREFIX(ProvokingVertex) ; .set GL_PREFIX(ProvokingVertex), GL_PREFIX(ProvokingVertexEXT)
 
-#if defined(GLX_USE_TLS) && defined(__linux__)
-	.section ".note.ABI-tag", "a"
-	.p2align 2
-	.long	1f - 0f   /* name length */
-	.long	3f - 2f   /* data length */
-	.long	1         /* note length */
-0:	.asciz "GNU"      /* vendor name */
-1:	.p2align 2
-2:	.long	0         /* note data: the ABI tag */
-	.long	2,4,20    /* Minimum kernel version w/TLS */
-3:	.p2align 2        /* pad out section */
-#endif /* GLX_USE_TLS */
 
 #if defined (__ELF__) && defined (__linux__)
 	.section .note.GNU-stack,"",%progbits
--- a/src/mapi/glapi/glapi_x86.S
+++ b/src/mapi/glapi/glapi_x86.S
@@ -1279,18 +1279,6 @@ GLNAME(gl_dispatch_functions_start):
 		ALIGNTEXT16
 GLNAME(gl_dispatch_functions_end):
 
-#if defined(GLX_USE_TLS) && defined(__linux__)
-	.section ".note.ABI-tag", "a"
-	.p2align 2
-	.long	1f - 0f   /* name length */
-	.long	3f - 2f   /* data length */
-	.long	1         /* note length */
-0:	.asciz "GNU"      /* vendor name */
-1:	.p2align 2
-2:	.long	0         /* note data: the ABI tag */
-	.long	2,4,20    /* Minimum kernel version w/TLS */
-3:	.p2align 2        /* pad out section */
-#endif /* GLX_USE_TLS */
 
 #if defined (__ELF__) && defined (__linux__)
 	.section .note.GNU-stack,"",%progbits
