diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/dri/intel/intel_screen.c Mesa-7.8.2/src/mesa/drivers/dri/intel/intel_screen.c
--- Mesa-7.8.2.orig/src/mesa/drivers/dri/intel/intel_screen.c	2010-08-01 00:03:10.097738521 +0200
+++ Mesa-7.8.2/src/mesa/drivers/dri/intel/intel_screen.c	2010-08-01 00:06:19.626487391 +0200
@@ -51,7 +51,7 @@
 PUBLIC const char __driConfigOptions[] =
    DRI_CONF_BEGIN
    DRI_CONF_SECTION_PERFORMANCE
-      DRI_CONF_VBLANK_MODE(DRI_CONF_VBLANK_ALWAYS_SYNC)
+      DRI_CONF_VBLANK_MODE(DRI_CONF_VBLANK_DEF_INTERVAL_0)
       /* Options correspond to DRI_CONF_BO_REUSE_DISABLED,
        * DRI_CONF_BO_REUSE_ALL
        */
