diff -Naur Mesa-7.10.orig//configs/autoconf.in Mesa-7.10/configs/autoconf.in
--- Mesa-7.10.orig//configs/autoconf.in	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/autoconf.in	2011-02-15 21:01:31.031171001 +0100
@@ -33,6 +33,8 @@
 LLVM_LIBS = @LLVM_LIBS@
 GLW_CFLAGS = @GLW_CFLAGS@
 GLUT_CFLAGS = @GLUT_CFLAGS@
+DRI_CFLAGS = @DRI_CFLAGS@
+DRI_CXXFLAGS = @DRI_CXXFLAGS@
 
 TALLOC_LIBS = @TALLOC_LIBS@
 TALLOC_CFLAGS = @TALLOC_CFLAGS@
@@ -103,7 +105,10 @@
 GALLIUM_DRIVERS = $(foreach DIR,$(GALLIUM_DRIVERS_DIRS),$(TOP)/src/gallium/drivers/$(DIR)/lib$(DIR).a)
 
 # Driver specific build vars
-DRI_DIRS = @DRI_DIRS@ 
+DRI_DIRS = @DRI_DIRS@
+DRICORE_GLSL_LIBS = @DRICORE_GLSL_LIBS@
+DRICORE_LIBS = @DRICORE_LIBS@
+DRICORE_LIB_DEPS = @DRICORE_LIB_DEPS@
 EGL_PLATFORMS = @EGL_PLATFORMS@
 EGL_CLIENT_APIS = @EGL_CLIENT_APIS@
 
@@ -131,6 +136,7 @@
 VG_LIB_DEPS = $(EXTRA_LIB_PATH) @VG_LIB_DEPS@
 
 # DRI dependencies
+MESA_MODULES = @MESA_MODULES@
 DRI_LIB_DEPS = $(EXTRA_LIB_PATH) @DRI_LIB_DEPS@
 LIBDRM_CFLAGS = @LIBDRM_CFLAGS@
 LIBDRM_LIB = @LIBDRM_LIBS@
diff -Naur Mesa-7.10.orig//configs/default Mesa-7.10/configs/default
--- Mesa-7.10.orig//configs/default	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/default	2011-02-15 21:01:31.031171001 +0100
@@ -85,6 +85,9 @@
 TALLOC_LIBS = `pkg-config --libs talloc`
 TALLOC_CFLAGS = `pkg-config --cflags talloc`
 
+DRI_CFLAGS = $(CFLAGS)
+DRI_CXXFLAGS = $(CXXFLAGS)
+
 # Optional assembly language optimization files for libGL
 MESA_ASM_SOURCES = 
 
diff -Naur Mesa-7.10.orig//configs/freebsd-dri Mesa-7.10/configs/freebsd-dri
--- Mesa-7.10.orig//configs/freebsd-dri	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/freebsd-dri	2011-02-15 21:01:31.031171001 +0100
@@ -30,9 +30,11 @@
 MESA_ASM_SOURCES = 
 
 # Library/program dependencies
+MESA_MODULES  = $(TOP)/src/mesa/libmesa.a
+
 LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
 LIBDRM_LIB = `pkg-config --libs libdrm`
-DRI_LIB_DEPS = -L/usr/local/lib -lm -pthread -lexpat $(LIBDRM_LIB)
+DRI_LIB_DEPS = $(MESA_MODULES) -L/usr/local/lib -lm -pthread -lexpat $(LIBDRM_LIB)
 GL_LIB_DEPS = -L/usr/local/lib -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 	-lm -pthread $(LIBDRM_LIB)
 
diff -Naur Mesa-7.10.orig//configs/linux-dri Mesa-7.10/configs/linux-dri
--- Mesa-7.10.orig//configs/linux-dri	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/linux-dri	2011-02-15 21:01:31.031171001 +0100
@@ -43,9 +43,11 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=-L/usr/X11R6/lib
 
+MESA_MODULES  = $(TOP)/src/mesa/libmesa.a
+
 LIBDRM_CFLAGS = $(shell pkg-config --cflags libdrm)
 LIBDRM_LIB = $(shell pkg-config --libs libdrm)
-DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl -ltalloc $(LIBDRM_LIB)
+DRI_LIB_DEPS  = $(MESA_MODULES) $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl -ltalloc $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 		-lm -lpthread -ldl $(LIBDRM_LIB)
 
diff -Naur Mesa-7.10.orig//configs/linux-dri-xcb Mesa-7.10/configs/linux-dri-xcb
--- Mesa-7.10.orig//configs/linux-dri-xcb	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/linux-dri-xcb	2011-02-15 21:01:31.031171001 +0100
@@ -41,9 +41,11 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=$(shell pkg-config --libs-only-L x11)
 
+MESA_MODULES  = $(TOP)/src/mesa/libmesa.a
+
 LIBDRM_CFLAGS = $(shell pkg-config --cflags libdrm)
 LIBDRM_LIB = $(shell pkg-config --libs libdrm)
-DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
+DRI_LIB_DEPS  = $(MESA_MODULES) $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lm -lpthread -ldl \
                 $(LIBDRM_LIB) $(shell pkg-config --libs xcb) $(shell pkg-config --libs x11-xcb) $(shell pkg-config --libs xcb-glx)
 
diff -Naur Mesa-7.10.orig//configs/linux-egl Mesa-7.10/configs/linux-egl
--- Mesa-7.10.orig//configs/linux-egl	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/linux-egl	2011-02-15 21:01:31.031171001 +0100
@@ -38,9 +38,11 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=-L/usr/X11R6/lib
 
+MESA_MODULES  = $(TOP)/src/mesa/libmesa.a
+
 LIBDRM_CFLAGS = $(shell pkg-config --cflags libdrm)
 LIBDRM_LIB = $(shell pkg-config --libs libdrm)
-DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
+DRI_LIB_DEPS  = $(MESA_MODULES) $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 		-lm -lpthread -ldl \
                 $(LIBDRM_LIB)
diff -Naur Mesa-7.10.orig//configs/linux-indirect Mesa-7.10/configs/linux-indirect
--- Mesa-7.10.orig//configs/linux-indirect	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configs/linux-indirect	2011-02-15 21:01:31.031171001 +0100
@@ -42,7 +42,8 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=-L/usr/X11R6/lib
 
-DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl
+MESA_MODULES  = $(TOP)/src/mesa/libmesa.a
+DRI_LIB_DEPS  = $(MESA_MODULES) $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lm -lpthread -ldl
 
 
diff -Naur Mesa-7.10.orig//configure Mesa-7.10/configure
--- Mesa-7.10.orig//configure	2011-02-15 21:01:24.841171002 +0100
+++ Mesa-7.10/configure	2011-02-15 21:01:35.421170999 +0100
@@ -662,6 +662,12 @@
 LIBDRM_RADEON_LIBS
 LIBDRM_RADEON_CFLAGS
 HAVE_XF86VIDMODE
+MESA_MODULES
+DRI_CFLAGS
+DRI_CXXFLAGS
+DRICORE_LIB_DEPS
+DRICORE_GLSL_LIBS
+DRICORE_LIBS
 GLESv2_PC_LIB_PRIV
 GLESv2_LIB_DEPS
 GLESv1_CM_PC_LIB_PRIV
@@ -823,6 +829,7 @@
 with_driver
 with_x
 enable_xcb
+enable_shared_dricore
 enable_glx_tls
 with_dri_driverdir
 with_dri_searchpath
@@ -1551,6 +1558,8 @@
   --enable-gles-overlay   build separate OpenGL ES only libraries [default=no]
   --enable-openvg         enable support for OpenVG API [default=no]
   --enable-xcb            use XCB for GLX [default=disabled]
+  --enable-shared-dricore link DRI modules with shared core DRI routines
+                          [default=disabled]
   --enable-glx-tls        enable TLS support in GLX [default=disabled]
   --disable-driglx-direct enable direct rendering in GLX and EGL for DRI
                           [default=enabled]
@@ -7488,6 +7497,36 @@
 
 
 
+# Check whether --enable-shared-dricore was given.
+if test "${enable_shared_dricore+set}" = set; then :
+  enableval=$enable_shared_dricore; enable_dricore="$enableval"
+else
+  enable_dricore=no
+fi
+
+if test "$mesa_driver" = dri ; then
+   if test "$enable_dricore" = yes ; then
+      DRICORE_GLSL_LIBS='$(TOP)/$(LIB_DIR)/libglsl.so'
+      DRICORE_LIBS='$(TOP)/$(LIB_DIR)/libdricore.so'
+      DRICORE_LIB_DEPS='-L$(TOP)/$(LIB_DIR) -Wl,-R$(DRI_DRIVER_INSTALL_DIR) -lglsl'
+      DRI_LIB_DEPS='-L$(TOP)/$(LIB_DIR) -Wl,-R$(DRI_DRIVER_INSTALL_DIR) -ldricore -lglsl'
+      DRI_CFLAGS='$(filter-out -fvisibility=hidden,$(CFLAGS)) -DUSE_DRICORE'
+      DRI_CXXFLAGS='$(filter-out -fvisibility=hidden,$(CXXFLAGS)) -DUSE_DRICORE'
+      MESA_MODULES='$(DRICORE_LIBS) $(DRICORE_GLSL_LIBS)'
+   else
+      DRI_CFLAGS='$(CFLAGS)'
+      DRI_CXXFLAGS='$(CXXFLAGS)'
+      DRI_LIB_DEPS='$(TOP)/src/mesa/libmesa.a'
+      MESA_MODULES='$(TOP)/src/mesa/libmesa.a'
+   fi
+fi
+
+
+
+
+
+
+
 
 
 
@@ -7776,8 +7815,8 @@
 
     fi
 
-    # put all the necessary libs together
-    DRI_LIB_DEPS="$SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS $TALLOC_LIBS"
+    # put all the necessary libs together, including possibly libdricore
+    DRI_LIB_DEPS="$DRI_LIB_DEPS $SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS $TALLOC_LIBS"
 fi
 
 
diff -Naur Mesa-7.10.orig//configure.ac Mesa-7.10/configure.ac
--- Mesa-7.10.orig//configure.ac	2011-02-15 21:01:24.831171002 +0100
+++ Mesa-7.10/configure.ac	2011-02-15 21:01:31.031171001 +0100
@@ -796,6 +796,34 @@
 AC_SUBST([GLESv2_PC_LIB_PRIV])
 
 
+AC_ARG_ENABLE([shared-dricore],
+    [AS_HELP_STRING([--enable-shared-dricore],
+        [link DRI modules with shared core DRI routines @<:@default=disabled@:>@])],
+    [enable_dricore="$enableval"],
+    [enable_dricore=no])
+if test "$mesa_driver" = dri ; then
+   if test "$enable_dricore" = yes ; then
+      DRICORE_GLSL_LIBS='$(TOP)/$(LIB_DIR)/libglsl.so'
+      DRICORE_LIBS='$(TOP)/$(LIB_DIR)/libdricore.so'
+      DRICORE_LIB_DEPS='-L$(TOP)/$(LIB_DIR) -Wl,-R$(DRI_DRIVER_INSTALL_DIR) -lglsl'
+      DRI_LIB_DEPS='-L$(TOP)/$(LIB_DIR) -Wl,-R$(DRI_DRIVER_INSTALL_DIR) -ldricore -lglsl'
+      DRI_CFLAGS='$(filter-out -fvisibility=hidden,$(CFLAGS)) -DUSE_DRICORE'
+      DRI_CXXFLAGS='$(filter-out -fvisibility=hidden,$(CXXFLAGS)) -DUSE_DRICORE'
+      MESA_MODULES='$(DRICORE_LIBS) $(DRICORE_GLSL_LIBS)'
+   else
+      DRI_CFLAGS='$(CFLAGS)'
+      DRI_CXXFLAGS='$(CXXFLAGS)'
+      DRI_LIB_DEPS='$(TOP)/src/mesa/libmesa.a'
+      MESA_MODULES='$(TOP)/src/mesa/libmesa.a'
+   fi
+fi
+AC_SUBST([DRICORE_LIBS])
+AC_SUBST([DRICORE_GLSL_LIBS])
+AC_SUBST([DRICORE_LIB_DEPS])
+AC_SUBST([DRI_CXXFLAGS])
+AC_SUBST([DRI_CFLAGS])
+AC_SUBST([MESA_MODULES])
+
 AC_SUBST([HAVE_XF86VIDMODE])
 
 PKG_CHECK_MODULES([LIBDRM_RADEON],
@@ -963,8 +991,8 @@
             [AC_MSG_ERROR([Expat required for DRI.])])
     fi
 
-    # put all the necessary libs together
-    DRI_LIB_DEPS="$SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS $TALLOC_LIBS"
+    # put all the necessary libs together, including possibly libdricore
+    DRI_LIB_DEPS="$DRI_LIB_DEPS $SELINUX_LIBS $LIBDRM_LIBS $EXPAT_LIB -lm -lpthread $DLOPEN_LIBS $TALLOC_LIBS"
 fi
 AC_SUBST([DRI_DIRS])
 AC_SUBST([EXPAT_INCLUDES])
diff -Naur Mesa-7.10.orig//src/glsl/Makefile Mesa-7.10/src/glsl/Makefile
--- Mesa-7.10.orig//src/glsl/Makefile	2011-02-15 21:01:24.751171002 +0100
+++ Mesa-7.10/src/glsl/Makefile	2011-02-15 21:01:31.031171001 +0100
@@ -112,6 +112,9 @@
 	$(C_SOURCES:.c=.o) \
 	$(CXX_SOURCES:.cpp=.o)
 
+DRICORE_OBJ_DIR = obj-visible
+OBJECTS_DRICORE = $(addprefix $(DRICORE_OBJ_DIR)/,$(OBJECTS))
+
 INCLUDES = \
 	$(TALLOC_CFLAGS) \
 	-I. \
@@ -128,7 +131,14 @@
 
 ##### TARGETS #####
 
-default: depend lib$(LIBNAME).a $(APPS)
+default: depend lib$(LIBNAME).a $(APPS) $(DRICORE_GLSL_LIBS)
+
+libglsl.so : $(OBJECTS_DRICORE) Makefile
+	$(MKLIB) -cplusplus -noprefix -o $@ $(OBJECTS_DRICORE)
+
+$(TOP)/$(LIB_DIR)/libglsl.so: libglsl.so
+	$(INSTALL) -d $(TOP)/$(LIB_DIR)
+	$(INSTALL) -m 755 libglsl.so $(TOP)/$(LIB_DIR)
 
 lib$(LIBNAME).a: $(OBJECTS) Makefile $(TOP)/src/glsl/Makefile.template
 	$(MKLIB) -cplusplus -o $(LIBNAME) -static $(OBJECTS)
@@ -162,6 +172,14 @@
 .c.o:
 	$(CC) -c $(INCLUDES) $(CFLAGS) $(DEFINES) $< -o $@
 
+$(DRICORE_OBJ_DIR)/%.o : %.cpp
+	@mkdir -p $(dir $@)
+	$(CXX) -c $(INCLUDES) $(DRI_CXXFLAGS) $(DEFINES) $< -o $@
+
+$(DRICORE_OBJ_DIR)/%.o : %.c
+	@mkdir -p $(dir $@)
+	$(CC) -c $(INCLUDES) $(DRI_CFLAGS) $(DEFINES) $< -o $@
+
 glsl_lexer.cpp: glsl_lexer.lpp
 	flex --nounistd -o$@  $<
 
diff -Naur Mesa-7.10.orig//src/glsl/Makefile.orig Mesa-7.10/src/glsl/Makefile.orig
--- Mesa-7.10.orig//src/glsl/Makefile.orig	1970-01-01 01:00:00.000000000 +0100
+++ Mesa-7.10/src/glsl/Makefile.orig	2011-01-04 18:44:17.000000000 +0100
@@ -0,0 +1,186 @@
+#src/glsl/pp/Makefile
+
+TOP = ../..
+
+include $(TOP)/configs/current
+
+LIBNAME = glsl
+
+LIBGLCPP_SOURCES = \
+	glcpp/glcpp-lex.c \
+	glcpp/glcpp-parse.c \
+	glcpp/pp.c
+
+GLCPP_SOURCES = \
+	$(LIBGLCPP_SOURCES) \
+	glcpp/glcpp.c
+
+C_SOURCES = \
+	strtod.c \
+	$(LIBGLCPP_SOURCES)
+
+CXX_SOURCES = \
+	ast_expr.cpp \
+	ast_function.cpp \
+	ast_to_hir.cpp \
+	ast_type.cpp \
+	builtin_function.cpp \
+	glsl_lexer.cpp \
+	glsl_parser.cpp \
+	glsl_parser_extras.cpp \
+	glsl_types.cpp \
+	glsl_symbol_table.cpp \
+	hir_field_selection.cpp \
+	ir_basic_block.cpp \
+	ir_clone.cpp \
+	ir_constant_expression.cpp \
+	ir.cpp \
+	ir_expression_flattening.cpp \
+	ir_function_can_inline.cpp \
+	ir_function.cpp \
+	ir_hierarchical_visitor.cpp \
+	ir_hv_accept.cpp \
+	ir_import_prototypes.cpp \
+	ir_print_visitor.cpp \
+	ir_reader.cpp \
+	ir_rvalue_visitor.cpp \
+	ir_set_program_inouts.cpp \
+	ir_validate.cpp \
+	ir_variable.cpp \
+	ir_variable_refcount.cpp \
+	linker.cpp \
+	link_functions.cpp \
+	loop_analysis.cpp \
+	loop_controls.cpp \
+	loop_unroll.cpp \
+	lower_discard.cpp \
+	lower_if_to_cond_assign.cpp \
+	lower_instructions.cpp \
+	lower_jumps.cpp \
+	lower_mat_op_to_vec.cpp \
+	lower_noise.cpp \
+	lower_texture_projection.cpp \
+	lower_variable_index_to_cond_assign.cpp \
+	lower_vec_index_to_cond_assign.cpp \
+	lower_vec_index_to_swizzle.cpp \
+	lower_vector.cpp \
+	opt_algebraic.cpp \
+	opt_constant_folding.cpp \
+	opt_constant_propagation.cpp \
+	opt_constant_variable.cpp \
+	opt_copy_propagation.cpp \
+	opt_dead_code.cpp \
+	opt_dead_code_local.cpp \
+	opt_dead_functions.cpp \
+	opt_discard_simplification.cpp \
+	opt_function_inlining.cpp \
+	opt_if_simplification.cpp \
+	opt_noop_swizzle.cpp \
+	opt_redundant_jumps.cpp \
+	opt_structure_splitting.cpp \
+	opt_swizzle_swizzle.cpp \
+	opt_tree_grafting.cpp \
+	s_expression.cpp
+
+LIBS = \
+	$(TOP)/src/glsl/libglsl.a \
+	$(TALLOC_LIBS)
+
+APPS = glsl_compiler glcpp/glcpp
+
+GLSL2_C_SOURCES = \
+	../mesa/program/hash_table.c \
+	../mesa/program/symbol_table.c
+GLSL2_CXX_SOURCES = \
+	main.cpp
+
+GLSL2_OBJECTS = \
+	$(GLSL2_C_SOURCES:.c=.o) \
+	$(GLSL2_CXX_SOURCES:.cpp=.o)
+
+### Basic defines ###
+
+DEFINES += \
+	$(LIBRARY_DEFINES) \
+	$(API_DEFINES)
+
+GLCPP_OBJECTS = \
+	$(GLCPP_SOURCES:.c=.o) \
+	../mesa/program/hash_table.o
+
+OBJECTS = \
+	$(C_SOURCES:.c=.o) \
+	$(CXX_SOURCES:.cpp=.o)
+
+INCLUDES = \
+	$(TALLOC_CFLAGS) \
+	-I. \
+	-I../mesa \
+	-I../mapi \
+	-I../../include \
+	$(LIBRARY_INCLUDES)
+
+ALL_SOURCES = \
+	$(C_SOURCES) \
+	$(CXX_SOURCES) \
+	$(GLSL2_CXX_SOURCES) \
+	$(GLSL2_C_SOURCES)
+
+##### TARGETS #####
+
+default: depend lib$(LIBNAME).a $(APPS)
+
+lib$(LIBNAME).a: $(OBJECTS) Makefile $(TOP)/src/glsl/Makefile.template
+	$(MKLIB) -cplusplus -o $(LIBNAME) -static $(OBJECTS)
+
+depend: $(ALL_SOURCES) Makefile
+	rm -f depend
+	touch depend
+	$(MKDEP) $(MKDEP_OPTIONS) $(INCLUDES) $(ALL_SOURCES) 2> /dev/null
+
+# Remove .o and backup files
+clean:
+	rm -f $(GLCPP_OBJECTS) $(GLSL2_OBJECTS) $(OBJECTS) lib$(LIBNAME).a depend depend.bak
+	-rm -f $(APPS)
+
+# Dummy target
+install:
+	@echo -n ""
+
+
+##### RULES #####
+
+glsl_compiler: $(GLSL2_OBJECTS) libglsl.a
+	$(APP_CXX) $(INCLUDES) $(CFLAGS) $(LDFLAGS) $(GLSL2_OBJECTS) $(LIBS) -o $@
+
+glcpp/glcpp: $(GLCPP_OBJECTS) libglsl.a
+	$(APP_CC) $(INCLUDES) $(CFLAGS) $(LDFLAGS) $(GLCPP_OBJECTS) $(LIBS) -o $@
+
+.cpp.o:
+	$(CXX) -c $(INCLUDES) $(CXXFLAGS) $(DEFINES) $< -o $@
+
+.c.o:
+	$(CC) -c $(INCLUDES) $(CFLAGS) $(DEFINES) $< -o $@
+
+glsl_lexer.cpp: glsl_lexer.lpp
+	flex --nounistd -o$@  $<
+
+glsl_parser.cpp: glsl_parser.ypp
+	bison -v -o "$@" -p "_mesa_glsl_" --defines=glsl_parser.h $<
+
+glcpp/glcpp-lex.c: glcpp/glcpp-lex.l
+	flex --nounistd -o$@  $<
+
+glcpp/glcpp-parse.c: glcpp/glcpp-parse.y
+	bison -v -o "$@" --defines=glcpp/glcpp-parse.h $<
+
+builtins: builtin_function.cpp builtins/profiles/* builtins/ir/* builtins/tools/generate_builtins.py builtins/tools/texture_builtins.py
+	@echo Bootstrapping the compiler...
+	cp builtins/tools/builtin_function.cpp .
+	make glsl_compiler
+	@echo Regenerating builtin_function.cpp...
+	$(PYTHON2) $(PYTHON_FLAGS) builtins/tools/generate_builtins.py > builtin_function.cpp
+	@echo Rebuilding the real compiler...
+	make glsl_compiler
+
+-include depend
diff -Naur Mesa-7.10.orig//src/mesa/drivers/dri/Makefile.template Mesa-7.10/src/mesa/drivers/dri/Makefile.template
--- Mesa-7.10.orig//src/mesa/drivers/dri/Makefile.template	2011-02-15 21:01:24.681171001 +0100
+++ Mesa-7.10/src/mesa/drivers/dri/Makefile.template	2011-02-15 21:01:31.031171001 +0100
@@ -1,7 +1,5 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
-
 COMMON_GALLIUM_SOURCES = \
         ../common/utils.c \
         ../common/vblank.c \
@@ -39,13 +37,13 @@
 ##### RULES #####
 
 .c.o:
-	$(CC) -c $(INCLUDES) $(CFLAGS) $(DRIVER_DEFINES) $< -o $@
+	$(CC) -c $(INCLUDES) $(DRI_CFLAGS) $(DRIVER_DEFINES) $< -o $@
 
 .cpp.o:
-	$(CC) -c $(INCLUDES) $(CXXFLAGS) $(DRIVER_DEFINES) $< -o $@
+	$(CC) -c $(INCLUDES) $(DRI_CXXFLAGS) $(DRIVER_DEFINES) $< -o $@
 
 .S.o:
-	$(CC) -c $(INCLUDES) $(CFLAGS) $(DRIVER_DEFINES) $< -o $@
+	$(CC) -c $(INCLUDES) $(DRI_CFLAGS) $(DRIVER_DEFINES) $< -o $@
 
 
 ##### TARGETS #####
@@ -57,10 +55,10 @@
 lib: symlinks subdirs depend
 	@$(MAKE) $(LIBNAME) $(TOP)/$(LIB_DIR)/$(LIBNAME)
 
-$(LIBNAME): $(OBJECTS) $(MESA_MODULES) $(EXTRA_MODULES) Makefile \
+$(LIBNAME): $(OBJECTS) $(EXTRA_MODULES) $(MESA_MODULES) Makefile \
 		$(TOP)/src/mesa/drivers/dri/Makefile.template $(TOP)/src/mesa/drivers/dri/common/dri_test.o
 	$(MKLIB) -o $@.tmp -noprefix -linker '$(CXX)' -ldflags '$(LDFLAGS)' \
-		$(OBJECTS) $(MESA_MODULES) $(EXTRA_MODULES) $(DRI_LIB_DEPS)
+		$(OBJECTS) $(EXTRA_MODULES) $(DRI_LIB_DEPS)
 	$(CXX) $(CFLAGS) -o $@.test $(TOP)/src/mesa/drivers/dri/common/dri_test.o $@.tmp $(DRI_LIB_DEPS)
 	@rm -f $@.test
 	mv -f $@.tmp $@
diff -Naur Mesa-7.10.orig//src/mesa/Makefile Mesa-7.10/src/mesa/Makefile
--- Mesa-7.10.orig//src/mesa/Makefile	2011-02-15 21:01:24.711171002 +0100
+++ Mesa-7.10/src/mesa/Makefile	2011-02-15 21:01:31.031171001 +0100
@@ -15,6 +15,7 @@
 MESA_OBJ_DIR := .
 ES1_OBJ_DIR := objs-es1
 ES2_OBJ_DIR := objs-es2
+DRICORE_OBJ_DIR := objs-dricore
 
 
 include sources.mak
@@ -23,6 +24,7 @@
 ES1_OBJECTS := $(addprefix $(ES1_OBJ_DIR)/, $(MESA_OBJECTS))
 ES2_OBJECTS := $(addprefix $(ES2_OBJ_DIR)/, $(MESA_OBJECTS))
 MESA_OBJECTS := $(addprefix $(MESA_OBJ_DIR)/, $(MESA_OBJECTS))
+DRICORE_OBJECTS := $(addprefix $(DRICORE_OBJ_DIR)/, $(MESA_OBJECTS))
 
 ES1_GALLIUM_OBJECTS := $(addprefix $(ES1_OBJ_DIR)/, $(MESA_GALLIUM_OBJECTS))
 ES2_GALLIUM_OBJECTS := $(addprefix $(ES2_OBJ_DIR)/, $(MESA_GALLIUM_OBJECTS))
@@ -32,6 +34,7 @@
 MESA_CPPFLAGS := $(API_DEFINES) $(DEFINES)
 ES1_CPPFLAGS := -DFEATURE_ES1=1 $(DEFINES)
 ES2_CPPFLAGS := -DFEATURE_ES2=1 $(DEFINES)
+DRICORE_CPPFLAGS = $(MESA_CPPFLAGS)
 
 # append include dirs
 MESA_CPPFLAGS += $(INCLUDE_DIRS) $(TALLOC_CFLAGS)
@@ -43,18 +46,24 @@
 CXXFLAGS := $(filter-out $(DEFINES), $(CXXFLAGS))
 
 # LLVM is needed for the state tracker
-MESA_CFLAGS := $(LLVM_CFLAGS)
-ES1_CFLAGS := $(LLVM_CFLAGS)
-ES2_CFLAGS := $(LLVM_CFLAGS)
+MESA_CFLAGS := $(LLVM_CFLAGS) $(CFLAGS)
+ES1_CFLAGS := $(LLVM_CFLAGS) $(CFLAGS)
+ES2_CFLAGS := $(LLVM_CFLAGS) $(CFLAGS)
+DRICORE_CFLAGS := $(LLVM_CFLAGS) $(DRI_CFLAGS)
+
+MESA_CXXFLAGS := $(LLVM_CFLAGS) $(CXXFLAGS)
+ES1_CXXFLAGS := $(LLVM_CFLAGS) $(CXXFLAGS)
+ES2_CXXFLAGS := $(LLVM_CFLAGS) $(CXXFLAGS)
+DRICORE_CXXFLAGS := $(LLVM_CFLAGS) $(DRI_CXXFLAGS)
 
 define mesa-cc-c
 	@mkdir -p $(dir $@)
-	$(CC) -c -o $@ $< $($(1)_CPPFLAGS) $($(1)_CFLAGS) $(CFLAGS)
+	$(CC) -c -o $@ $< $($(1)_CPPFLAGS) $($(1)_CFLAGS)
 endef
 
 define mesa-cxx-c
 	@mkdir -p $(dir $@)
-	$(CXX) -c -o $@ $< $($(1)_CPPFLAGS) $($(1)_CFLAGS) $(CXXFLAGS)
+	$(CXX) -c -o $@ $< $($(1)_CPPFLAGS) $($(1)_CFLAGS)
 endef
 
 $(MESA_OBJ_DIR)/%.o: %.c
@@ -84,11 +93,20 @@
 $(ES2_OBJ_DIR)/%.o: %.S
 	$(call mesa-cc-c,ES2)
 
+$(DRICORE_OBJ_DIR)/%.o: %.c
+	$(call mesa-cc-c,DRICORE)
+
+$(DRICORE_OBJ_DIR)/%.o: %.cpp
+	$(call mesa-cxx-c,DRICORE)
+
+$(DRICORE_OBJ_DIR)/%.o: %.S
+	$(call mesa-cc-c,DRICORE)
+
 
 # Default: build dependencies, then asm_subdirs, GLSL built-in lib,
 # then convenience libs (.a) and finally the device drivers:
 default: $(DEPENDS) asm_subdirs \
-	$(MESA_LIBS) $(ES1_LIBS) $(ES2_LIBS) driver_subdirs
+	$(MESA_LIBS) $(ES1_LIBS) $(ES2_LIBS) $(DRICORE_LIBS) driver_subdirs
 
 main/api_exec_es1.c: main/APIspec.xml main/es_generator.py main/APIspecutil.py main/APIspec.py
 	$(PYTHON2) $(PYTHON_FLAGS) main/es_generator.py -S main/APIspec.xml -V GLES1.1 > $@
@@ -109,6 +127,15 @@
 libes2.a: $(ES2_OBJECTS) $(GLSL_LIBS)
 	@$(MKLIB) -o es2 -static $(ES2_OBJECTS) $(GLSL_LIBS)
 
+# Shared dricore library for classic DRI drivers
+libdricore.so: $(DRICORE_OBJECTS) $(DRICORE_GLSL_LIBS)
+	@$(MKLIB) -o libdricore.so -noprefix -cplusplus \
+		$(DRICORE_LIB_DEPS) $(DRICORE_OBJECTS)
+
+$(TOP)/$(LIB_DIR)/libdricore.so: libdricore.so
+	@$(INSTALL) -d $(TOP)/$(LIB_DIR)
+	@$(INSTALL) -m 755 libdricore.so $(TOP)/$(LIB_DIR)
+
 # Make archive of subset of core mesa object files for gallium
 libmesagallium.a: $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
 	@ $(MKLIB) -o mesagallium -static $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
@@ -121,7 +148,7 @@
 
 ######################################################################
 # Device drivers
-driver_subdirs: $(MESA_LIBS)
+driver_subdirs: $(MESA_LIBS) $(DRICORE_LIBS)
 	@ (cd drivers && $(MAKE))
 
 
@@ -165,9 +192,12 @@
 new_install:
 	(cd drivers && $(MAKE) install)
 
+ifneq (,$(DRICORE_LIBS))
+DRICORE_INSTALL_TARGET = install-dricore
+endif
 
 # XXX replace this with new_install above someday
-install: default
+install: default $(DRICORE_INSTALL_TARGET)
 	@for driver in $(DRIVER_DIRS) ; do \
 	  case "$$driver" in \
 	    osmesa) if [ "$(DRIVER_DIRS)" = osmesa ]; then \
@@ -229,6 +259,12 @@
 	cd drivers/dri && $(MAKE) install
 
 
+install-dricore: default
+	$(INSTALL) -d $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
+	$(INSTALL) -m 755 $(DRICORE_LIBS) $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
+	$(INSTALL) -m 755 $(DRICORE_GLSL_LIBS) $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR$)
+
+
 
 # Emacs tags
 tags:
@@ -244,7 +280,12 @@
 	-rm -rf $(ES2_OBJ_DIR)
 	-rm -f depend.es2 depend.es2.bak
 
-clean: clean-es1 clean-es2
+clean-dricore:
+	-rm -f libdricore.so
+	-rm -f $(DRICORE_LIBS)
+	-rm -rf $(DRICORE_OBJ_DIR)
+
+clean: clean-es1 clean-es2 clean-dricore
 	-rm -f */*.o
 	-rm -f */*/*.o
 	-rm -f depend depend.bak libmesa.a libmesagallium.a
diff -Naur Mesa-7.10.orig//src/mesa/x86/read_rgba_span_x86.S Mesa-7.10/src/mesa/x86/read_rgba_span_x86.S
--- Mesa-7.10.orig//src/mesa/x86/read_rgba_span_x86.S	2011-02-15 21:01:24.641171001 +0100
+++ Mesa-7.10/src/mesa/x86/read_rgba_span_x86.S	2011-02-15 21:01:31.031171001 +0100
@@ -77,7 +77,9 @@
  */
 
 .globl _generic_read_RGBA_span_BGRA8888_REV_MMX
+#ifndef USE_DRICORE
 .hidden _generic_read_RGBA_span_BGRA8888_REV_MMX
+#endif
 	.type	_generic_read_RGBA_span_BGRA8888_REV_MMX, @function
 _generic_read_RGBA_span_BGRA8888_REV_MMX:
 	pushl	%ebx
@@ -172,7 +174,9 @@
  */
 
 .globl _generic_read_RGBA_span_BGRA8888_REV_SSE
+#ifndef USE_DRICORE
 .hidden _generic_read_RGBA_span_BGRA8888_REV_SSE
+#endif
 	.type	_generic_read_RGBA_span_BGRA8888_REV_SSE, @function
 _generic_read_RGBA_span_BGRA8888_REV_SSE:
 	pushl	%esi
@@ -335,7 +339,9 @@
 
 	.text
 .globl _generic_read_RGBA_span_BGRA8888_REV_SSE2
+#ifndef USE_DRICORE
 .hidden _generic_read_RGBA_span_BGRA8888_REV_SSE2
+#endif
 	.type	_generic_read_RGBA_span_BGRA8888_REV_SSE2, @function
 _generic_read_RGBA_span_BGRA8888_REV_SSE2:
 	pushl	%esi
@@ -494,7 +500,9 @@
 
 	.text
 	.globl	_generic_read_RGBA_span_RGB565_MMX
+#ifndef USE_DRICORE
         .hidden _generic_read_RGBA_span_RGB565_MMX
+#endif
 	.type	_generic_read_RGBA_span_RGB565_MMX, @function
 
 _generic_read_RGBA_span_RGB565_MMX:
