diff --git a/src/gallium/winsys/radeon/drm/radeon_r300.c b/src/gallium/winsys/radeon/drm/radeon_r300.c
index 420522f..2d1bd1e 100644
--- a/src/gallium/winsys/radeon/drm/radeon_r300.c
+++ b/src/gallium/winsys/radeon/drm/radeon_r300.c
@@ -119,7 +119,7 @@ static struct r300_winsys_buffer *radeon_r300_winsys_buffer_from_handle(struct r
 
     if (stride)
         *stride = whandle->stride;
-    if (size)
+    if (size && _buf)
         *size = _buf->base.size;
 
     return radeon_libdrm_winsys_buffer(_buf);
