diff -Naur Mesa-7.8.2.orig/bin/mklib Mesa-7.8.2/bin/mklib
--- Mesa-7.8.2.orig/bin/mklib	2010-08-01 00:03:10.278990646 +0200
+++ Mesa-7.8.2/bin/mklib	2010-08-01 00:08:19.886491443 +0200
@@ -313,9 +313,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
diff -Naur Mesa-7.8.2.orig/configs/linux-dri Mesa-7.8.2/configs/linux-dri
--- Mesa-7.8.2.orig/configs/linux-dri	2010-08-01 00:03:10.278990646 +0200
+++ Mesa-7.8.2/configs/linux-dri	2010-08-01 00:11:08.873417880 +0200
@@ -29,6 +29,8 @@
 
 CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
 	$(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
+HOST_CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
+	$(HOST_OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
 
 CXXFLAGS = -Wall $(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES)
 
diff -Naur Mesa-7.8.2.orig/src/glsl/apps/Makefile Mesa-7.8.2/src/glsl/apps/Makefile
--- Mesa-7.8.2.orig/src/glsl/apps/Makefile	2010-08-01 00:03:10.248990123 +0200
+++ Mesa-7.8.2/src/glsl/apps/Makefile	2010-08-01 00:20:04.437737537 +0200
@@ -26,10 +26,10 @@
 .SUFFIXES: .c
 
 .c:
-	$(APP_CC) $(INCLUDES) $(CFLAGS) $(LDFLAGS) $< $(LIBS) -o $@
+	$(HOST_CC) $(INCLUDES) $(HOST_CFLAGS) $(HOST_LDFLAGS) $< $(LIBS) -o $@
 
 .c.o:
-	$(APP_CC) -c $(INCLUDES) $(CFLAGS) $(DEFINES) $< -o $@
+	$(HOST_CC) -c $(INCLUDES) $(HOST_CFLAGS) $(DEFINES) $< -o $@
 
 
 ##### TARGETS #####
