diff -Naur mesa-7.1rc1-20080706.orig/bin/mklib mesa-7.1rc1-20080706/bin/mklib
--- mesa-7.1rc1-20080706.orig/bin/mklib	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/bin/mklib	2008-07-06 14:11:03.000000000 +0200
@@ -211,9 +211,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
diff -Naur mesa-7.1rc1-20080706.orig/configs/linux-dri mesa-7.1rc1-20080706/configs/linux-dri
--- mesa-7.1rc1-20080706.orig/configs/linux-dri	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/configs/linux-dri	2008-07-06 14:16:00.000000000 +0200
@@ -14,6 +14,7 @@
 #MKDEP_OPTIONS = -MF depend
 
 OPT_FLAGS  = -O -g
+HOST_OPT_FLAGS = $(OPT_FLAGS)
 PIC_FLAGS  = -fPIC
 
 # Add '-DGLX_USE_TLS' to ARCH_FLAGS to enable TLS support.
@@ -29,6 +30,8 @@
 
 CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
 	$(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
+HOST_CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
+	$(HOST_OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
 
 CXXFLAGS = -Wall $(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES)
 
diff -Naur mesa-7.1rc1-20080706.orig/src/glx/mini/Makefile mesa-7.1rc1-20080706/src/glx/mini/Makefile
--- mesa-7.1rc1-20080706.orig/src/glx/mini/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/glx/mini/Makefile	2008-07-06 14:13:32.000000000 +0200
@@ -66,9 +66,6 @@
 
 depend: $(C_SOURCES) $(ASM_SOURCES)
 	touch depend
-	$(MKDEP) $(MKDEP_OPTIONS) $(INCLUDES) $(C_SOURCES) $(ASM_SOURCES) \
-		> /dev/null 
-
 
 # Emacs tags
 tags:
@@ -84,5 +81,3 @@
 	-rm -f drmtest $(TOP)/$(LIB_DIR)/libGL.so*
 	-rm -f *.o *~
 	-rm -f depend depend.bak
-
-include depend
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/dri/Makefile.template mesa-7.1rc1-20080706/src/mesa/drivers/dri/Makefile.template
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/dri/Makefile.template	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/dri/Makefile.template	2008-07-06 14:13:28.000000000 +0200
@@ -78,9 +78,6 @@
 
 depend: $(C_SOURCES) $(ASM_SOURCES) $(SYMLINKS)
 	touch depend
-	$(MKDEP) $(MKDEP_OPTIONS) $(DRIVER_DEFINES) $(INCLUDES) $(C_SOURCES) \
-		$(ASM_SOURCES)
-
 
 # Emacs tags
 tags:
@@ -96,6 +93,3 @@
 install: $(LIBNAME)
 	$(INSTALL) -d $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
 	$(INSTALL) -m 755 $(LIBNAME) $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
-
-
--include depend
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/Makefile mesa-7.1rc1-20080706/src/mesa/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/Makefile	2008-07-06 14:13:52.000000000 +0200
@@ -61,9 +61,6 @@
 depend: $(ALL_SOURCES)
 	@ echo "running $(MKDEP)"
 	@ touch depend
-	@$(MKDEP) $(MKDEP_OPTIONS) $(DEFINES) $(INCLUDE_DIRS) $(ALL_SOURCES) \
-		> /dev/null 2>/dev/null
-
 
 ######################################################################
 # Installation rules
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/x86/Makefile mesa-7.1rc1-20080706/src/mesa/x86/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/x86/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/x86/Makefile	2008-07-06 14:15:00.000000000 +0200
@@ -21,7 +21,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(HOST_CC) $(INCLUDE_DIRS) $(HOST_CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
