diff -Naur mesa-git-20080113.orig/bin/mklib mesa-git-20080113/bin/mklib
--- mesa-git-20080113.orig/bin/mklib	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/bin/mklib	2008-01-14 19:34:50.000000000 +0100
@@ -199,9 +199,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
diff -Naur mesa-git-20080113.orig/src/glx/x11/Makefile mesa-git-20080113/src/glx/x11/Makefile
--- mesa-git-20080113.orig/src/glx/x11/Makefile	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/glx/x11/Makefile	2008-01-14 19:37:38.000000000 +0100
@@ -72,9 +72,6 @@
 
 depend: $(SOURCES) $(MESA_GLAPI_SOURCES) $(MESA_ASM_API) Makefile
 	touch depend
-	$(MKDEP) $(MKDEP_OPTIONS) $(INCLUDES) $(SOURCES) \
-		$(MESA_GLAPI_SOURCES) $(MESA_ASM_API) 
-
 
 # Emacs tags
 tags:
diff -Naur mesa-git-20080113.orig/src/mesa/drivers/dri/Makefile.template mesa-git-20080113/src/mesa/drivers/dri/Makefile.template
--- mesa-git-20080113.orig/src/mesa/drivers/dri/Makefile.template	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/mesa/drivers/dri/Makefile.template	2008-01-14 19:38:02.000000000 +0100
@@ -79,9 +79,6 @@
 
 depend: $(C_SOURCES) $(ASM_SOURCES) $(SYMLINKS)
 	touch depend
-	$(MKDEP) $(MKDEP_OPTIONS) $(DRIVER_DEFINES) $(INCLUDES) $(C_SOURCES) \
-		$(ASM_SOURCES) 2>&1 /dev/null
-
 
 # Emacs tags
 tags:
@@ -97,6 +94,3 @@
 install: $(LIBNAME)
 	$(INSTALL) -d $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
 	$(INSTALL) -m 755 $(LIBNAME) $(DESTDIR)$(DRI_DRIVER_INSTALL_DIR)
-
-
-include depend
diff -Naur mesa-git-20080113.orig/src/mesa/Makefile mesa-git-20080113/src/mesa/Makefile
--- mesa-git-20080113.orig/src/mesa/Makefile	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/mesa/Makefile	2008-01-14 19:39:08.000000000 +0100
@@ -146,9 +146,6 @@
 depend: $(ALL_SOURCES)
 	@ echo "running $(MKDEP)"
 	@ touch depend
-	@$(MKDEP) $(MKDEP_OPTIONS) $(DEFINES) $(INCLUDE_DIRS) $(ALL_SOURCES) \
-		> /dev/null 2>/dev/null
-
 
 subdirs:
 	@ if echo "$(ASM_FLAGS)" | grep -q USE_X86_ASM ; then \
diff -Naur mesa-git-20080113.orig/src/mesa/x86/Makefile mesa-git-20080113/src/mesa/x86/Makefile
--- mesa-git-20080113.orig/src/mesa/x86/Makefile	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/mesa/x86/Makefile	2008-01-14 19:39:28.000000000 +0100
@@ -21,7 +21,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(HOST_CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
