Index: mesa/src/mesa/drivers/dri/Makefile
===================================================================
--- mesa.orig/src/mesa/drivers/dri/Makefile	2011-02-18 21:02:23.777115341 +1100
+++ mesa/src/mesa/drivers/dri/Makefile	2011-02-18 21:03:12.557186884 +1100
@@ -26,6 +26,7 @@
 	-e 's,@INSTALL_INC_DIR@,$(INSTALL_INC_DIR),' \
 	-e 's,@VERSION@,$(MESA_MAJOR).$(MESA_MINOR).$(MESA_TINY),' \
 	-e 's,@DRI_DRIVER_DIR@,$(DRI_DRIVER_INSTALL_DIR),' \
+	-e 's,@DRI_DRIVER_SEARCH_DIR@,$(DRI_DRIVER_SEARCH_DIR),' \
 	-e 's,@DRI_PC_REQ_PRIV@,$(DRI_PC_REQ_PRIV),'
 
 dri.pc: dri.pc.in
Index: mesa/src/mesa/drivers/dri/dri.pc.in
===================================================================
--- mesa.orig/src/mesa/drivers/dri/dri.pc.in	2011-02-18 21:02:23.717115253 +1100
+++ mesa/src/mesa/drivers/dri/dri.pc.in	2011-02-18 21:02:45.337146969 +1100
@@ -3,6 +3,7 @@
 libdir=@INSTALL_LIB_DIR@
 includedir=@INSTALL_INC_DIR@
 dridriverdir=@DRI_DRIVER_DIR@
+drisearchdirs=@DRI_DRIVER_SEARCH_DIR@
 
 Name: dri
 Description: Direct Rendering Infrastructure
