diff -Naur Mesa-7.8.2.orig/src/gallium/winsys/drm/i965/xlib/Makefile Mesa-7.8.2/src/gallium/winsys/drm/i965/xlib/Makefile
--- Mesa-7.8.2.orig/src/gallium/winsys/drm/i965/xlib/Makefile	2010-08-01 00:03:10.188990678 +0200
+++ Mesa-7.8.2/src/gallium/winsys/drm/i965/xlib/Makefile	2010-08-01 00:44:00.626488530 +0200
@@ -38,8 +38,8 @@
 	$(TOP)/src/gallium/drivers/i965/libi965.a \
 	$(TOP)/src/gallium/drivers/trace/libtrace.a \
 	$(TOP)/src/gallium/state_trackers/glx/xlib/libxlib.a \
-	$(TOP)/src/mesa/libglapi.a \
-	$(TOP)/src/mesa/libmesagallium.a \
+	$(TOP)/src/mesa/libglapi.so \
+	$(TOP)/src/mesa/libmesagallium.so \
 	$(GALLIUM_AUXILIARIES) 
 
 #	$(TOP)/src/gallium/drivers/i965/lib/libi9xx.a \
diff -Naur Mesa-7.8.2.orig/src/gallium/winsys/drm/Makefile.template Mesa-7.8.2/src/gallium/winsys/drm/Makefile.template
--- Mesa-7.8.2.orig/src/gallium/winsys/drm/Makefile.template	2010-08-01 00:03:10.188990678 +0200
+++ Mesa-7.8.2/src/gallium/winsys/drm/Makefile.template	2010-08-01 00:44:59.801168069 +0200
@@ -1,7 +1,7 @@
 # -*-makefile-*-
 
 MESA_MODULES = \
-	$(TOP)/src/mesa/libmesagallium.a \
+	$(TOP)/src/mesa/libmesagallium.so \
 	$(GALLIUM_AUXILIARIES)
 
 COMMON_GALLIUM_SOURCES = \
diff -Naur Mesa-7.8.2.orig/src/gallium/winsys/xlib/Makefile Mesa-7.8.2/src/gallium/winsys/xlib/Makefile
--- Mesa-7.8.2.orig/src/gallium/winsys/xlib/Makefile	2010-08-01 00:03:10.188990678 +0200
+++ Mesa-7.8.2/src/gallium/winsys/xlib/Makefile	2010-08-01 00:44:19.867737569 +0200
@@ -42,8 +42,8 @@
 LIBS = \
 	$(GALLIUM_DRIVERS) \
 	$(TOP)/src/gallium/state_trackers/glx/xlib/libxlib.a \
-	$(TOP)/src/mesa/libglapi.a \
-	$(TOP)/src/mesa/libmesagallium.a \
+	$(TOP)/src/mesa/libglapi.so \
+	$(TOP)/src/mesa/libmesagallium.so \
 	$(GALLIUM_AUXILIARIES) \
 	$(CELL_SPU_LIB) \
 
diff -Naur Mesa-7.8.2.orig/src/glx/Makefile Mesa-7.8.2/src/glx/Makefile
--- Mesa-7.8.2.orig/src/glx/Makefile	2010-08-01 00:03:10.268990541 +0200
+++ Mesa-7.8.2/src/glx/Makefile	2010-08-01 00:44:35.457737434 +0200
@@ -39,7 +39,7 @@
 	  dri2_glx.c \
 	  dri2.c
 
-GLAPI_LIB = $(TOP)/src/mesa/libglapi.a
+GLAPI_LIB = $(TOP)/src/mesa/libglapi.so
 
 OBJECTS = $(SOURCES:.c=.o)
 
@@ -73,7 +73,7 @@
 		$(GL_LIB_DEPS) $(OBJECTS) $(GLAPI_LIB)
 
 $(GLAPI_LIB):
-	@$(MAKE) -C $(TOP)/src/mesa libglapi.a
+	@$(MAKE) -C $(TOP)/src/mesa libglapi.so
 
 depend: $(SOURCES) Makefile
 	rm -f depend
diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/dri/Makefile.template Mesa-7.8.2/src/mesa/drivers/dri/Makefile.template
--- Mesa-7.8.2.orig/src/mesa/drivers/dri/Makefile.template	2010-08-01 00:03:10.108990151 +0200
+++ Mesa-7.8.2/src/mesa/drivers/dri/Makefile.template	2010-08-01 00:31:34.722714187 +0200
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.so
 
 COMMON_GALLIUM_SOURCES = \
         ../common/utils.c \
diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/fbdev/Makefile Mesa-7.8.2/src/mesa/drivers/fbdev/Makefile
--- Mesa-7.8.2.orig/src/mesa/drivers/fbdev/Makefile	2010-08-01 00:03:10.057738271 +0200
+++ Mesa-7.8.2/src/mesa/drivers/fbdev/Makefile	2010-08-01 00:32:52.356487634 +0200
@@ -14,7 +14,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .c.o:
diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/glslcompiler/Makefile Mesa-7.8.2/src/mesa/drivers/glslcompiler/Makefile
--- Mesa-7.8.2.orig/src/mesa/drivers/glslcompiler/Makefile	2010-08-01 00:03:10.047737802 +0200
+++ Mesa-7.8.2/src/mesa/drivers/glslcompiler/Makefile	2010-08-01 00:42:18.410808562 +0200
@@ -10,8 +10,8 @@
 OBJECTS = \
 	glslcompiler.o \
 	../common/driverfuncs.o \
-	../../libmesa.a \
-	../../libglapi.a
+	../../libmesa.so \
+	../../libglapi.so
 
 INCLUDES = \
 	-I$(TOP)/include \
diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/osmesa/Makefile Mesa-7.8.2/src/mesa/drivers/osmesa/Makefile
--- Mesa-7.8.2.orig/src/mesa/drivers/osmesa/Makefile	2010-08-01 00:03:10.057738271 +0200
+++ Mesa-7.8.2/src/mesa/drivers/osmesa/Makefile	2010-08-01 00:34:13.956579457 +0200
@@ -20,8 +20,8 @@
 	-I$(TOP)/src/mesa/main
 
 CORE_MESA = \
-	$(TOP)/src/mesa/libmesa.a \
-	$(TOP)/src/mesa/libglapi.a \
+	$(TOP)/src/mesa/libmesa.so \
+	$(TOP)/src/mesa/libglapi.so \
 	$(TOP)/src/glsl/cl/libglslcl.a \
 	$(TOP)/src/glsl/pp/libglslpp.a
 
diff -Naur Mesa-7.8.2.orig/src/mesa/drivers/x11/Makefile Mesa-7.8.2/src/mesa/drivers/x11/Makefile
--- Mesa-7.8.2.orig/src/mesa/drivers/x11/Makefile	2010-08-01 00:03:10.047737802 +0200
+++ Mesa-7.8.2/src/mesa/drivers/x11/Makefile	2010-08-01 00:35:01.658987495 +0200
@@ -44,7 +44,7 @@
 	-I$(TOP)/src/mesa/main \
 	$(X11_INCLUDES)
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 
diff -Naur Mesa-7.8.2.orig/src/mesa/Makefile Mesa-7.8.2/src/mesa/Makefile
--- Mesa-7.8.2.orig/src/mesa/Makefile	2010-08-01 00:03:10.127719018 +0200
+++ Mesa-7.8.2/src/mesa/Makefile	2010-08-01 00:40:04.276488262 +0200
@@ -19,9 +19,9 @@
 
 
 # Default: build dependencies, then asm_subdirs, GLSL built-in lib,
-# then convenience libs (.a) and finally the device drivers:
-default: depend asm_subdirs glsl_builtin libmesa.a libmesagallium.a \
-	libglapi.a driver_subdirs
+# then convenience libs (.so) and finally the device drivers:
+default: depend asm_subdirs glsl_builtin libmesa.so libmesagallium.so \
+	libglapi.so driver_subdirs
 
 
 
@@ -29,20 +29,20 @@
 # Helper libraries used by many drivers:
 
 # Make archive of core mesa object files
-libmesa.a: $(MESA_OBJECTS) $(GLSL_LIBS)
-	@ $(MKLIB) -o mesa -static $(MESA_OBJECTS) $(GLSL_LIBS)
+libmesa.so: $(MESA_OBJECTS) $(GLSL_LIBS)
+	@ $(MKLIB) -o mesa -dlopen $(MESA_OBJECTS) $(GLSL_LIBS)
 
 # Make archive of subset of core mesa object files for gallium
-libmesagallium.a: $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
-	@ $(MKLIB) -o mesagallium -static $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
+libmesagallium.so: $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
+	@ $(MKLIB) -o mesagallium -dlopen $(MESA_GALLIUM_OBJECTS) $(GLSL_LIBS)
 
 # Make archive of gl* API dispatcher functions only
-libglapi.a: $(GLAPI_OBJECTS)
-	@ $(MKLIB) -o glapi -static $(GLAPI_OBJECTS)
+libglapi.so: $(GLAPI_OBJECTS)
+	@ $(MKLIB) -o glapi -dlopen $(GLAPI_OBJECTS)
 
 ######################################################################
 # Device drivers
-driver_subdirs: libmesa.a libglapi.a libmesagallium.a
+driver_subdirs: libmesa.so libglapi.so libmesagallium.so
 	@ (cd drivers && $(MAKE))
 
 
@@ -154,7 +154,7 @@
 clean:
 	-rm -f */*.o
 	-rm -f */*/*.o
-	-rm -f depend depend.bak libmesa.a libglapi.a libmesagallium.a
+	-rm -f depend depend.bak libmesa.so libglapi.so libmesagallium.so
 	-rm -f drivers/*/*.o
 	-rm -f *.pc
 	-rm -f shader/slang/library/*_gc.h
