diff -Naur mesa-git-20080113.orig/src/mesa/drivers/dri/Makefile.template mesa-git-20080113/src/mesa/drivers/dri/Makefile.template
--- mesa-git-20080113.orig/src/mesa/drivers/dri/Makefile.template	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/mesa/drivers/dri/Makefile.template	2008-01-19 16:00:29.000000000 +0100
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.so
 
 COMMON_SOURCES = \
         ../../common/driverfuncs.c \
@@ -70,7 +70,7 @@
 
 $(LIBNAME): $(OBJECTS) $(MESA_MODULES) $(WINOBJ) Makefile $(TOP)/src/mesa/drivers/dri/Makefile.template
 	$(TOP)/bin/mklib -ldflags '$(LDFLAGS)' -noprefix -o $@ \
-		$(OBJECTS) $(MESA_MODULES) $(WINOBJ) $(DRI_LIB_DEPS)
+		$(OBJECTS) $(WINOBJ) $(DRI_LIB_DEPS)
 
 
 $(TOP)/$(LIB_DIR)/$(LIBNAME): $(LIBNAME)
diff -Naur mesa-git-20080113.orig/src/mesa/Makefile mesa-git-20080113/src/mesa/Makefile
--- mesa-git-20080113.orig/src/mesa/Makefile	2008-01-14 19:34:26.000000000 +0100
+++ mesa-git-20080113/src/mesa/Makefile	2008-01-19 15:59:41.000000000 +0100
@@ -41,7 +41,7 @@
 ######################################################################
 # BeOS driver target
 
-beos: depend subdirs libmesa.a
+beos: depend subdirs libmesa.so
 	cd drivers/beos; $(MAKE)
 
 
@@ -49,13 +49,13 @@
 # Linux DRI drivers
 
 # Make archive of core object files
-libmesa.a: $(SOLO_OBJECTS)
-	@ $(TOP)/bin/mklib -o mesa -static $(SOLO_OBJECTS);
+libmesa.so: $(SOLO_OBJECTS)
+	@ $(TOP)/bin/mklib -o mesa -dlopen $(SOLO_OBJECTS);
 	@if [ "${CONFIG_NAME}" = "beos" ] ; then \
 		mimeset -f "$@" ; \
 	fi
 
-linux-solo: depend subdirs libmesa.a
+linux-solo: depend subdirs libmesa.so
 	cd drivers/dri ; $(MAKE)
 
 
@@ -190,7 +190,7 @@
 clean:
 	-rm -f */*.o
 	-rm -f */*/*.o
-	-rm -f depend depend.bak libmesa.a
+	-rm -f depend depend.bak libmesa.so
 	-rm -f drivers/*/*.o
 	(cd drivers/dri && $(MAKE) clean)
 	(cd x86 && $(MAKE) clean)
