diff -Naur mesa-7.0.3.orig/src/mesa/drivers/dri/Makefile.template mesa-7.0.3/src/mesa/drivers/dri/Makefile.template
--- mesa-7.0.3.orig/src/mesa/drivers/dri/Makefile.template	2008-04-12 15:30:40.000000000 +0200
+++ mesa-7.0.3/src/mesa/drivers/dri/Makefile.template	2008-04-12 15:32:08.000000000 +0200
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.so
 
 COMMON_SOURCES = \
         ../../common/driverfuncs.c \
@@ -74,7 +74,7 @@
 
 $(LIBNAME): $(OBJECTS) $(MESA_MODULES) $(WINOBJ) Makefile $(TOP)/src/mesa/drivers/dri/Makefile.template
 	$(TOP)/bin/mklib -ldflags '$(LDFLAGS)' -noprefix -o $@ \
-		$(OBJECTS) $(MESA_MODULES) $(WINOBJ) $(DRI_LIB_DEPS)
+		$(OBJECTS) $(WINOBJ) $(DRI_LIB_DEPS)
 
 
 $(TOP)/$(LIB_DIR)/$(LIBNAME): $(LIBNAME)
diff -Naur mesa-7.0.3.orig/src/mesa/Makefile mesa-7.0.3/src/mesa/Makefile
--- mesa-7.0.3.orig/src/mesa/Makefile	2008-04-12 15:39:26.000000000 +0200
+++ mesa-7.0.3/src/mesa/Makefile	2008-04-12 15:39:39.000000000 +0200
@@ -43,7 +43,7 @@
 ######################################################################
 # BeOS driver target
 
-beos: depend subdirs libmesa.a
+beos: depend subdirs libmesa.so
 	cd drivers/beos; $(MAKE)
 
 
@@ -51,13 +51,13 @@
 # Linux DRI drivers
 
 # Make archive of core object files
-libmesa.a: $(SOLO_OBJECTS)
-	@ $(TOP)/bin/mklib -o mesa -static $(SOLO_OBJECTS);
+libmesa.so: $(SOLO_OBJECTS)
+	@ $(TOP)/bin/mklib -o mesa -dlopen $(SOLO_OBJECTS);
 	@if [ "${CONFIG_NAME}" = "beos" ] ; then \
 		mimeset -f "$@" ; \
 	fi
 
-linux-solo: depend subdirs libmesa.a
+linux-solo: depend subdirs libmesa.so
 	cd drivers/dri ; $(MAKE)
 
 
@@ -191,7 +191,7 @@
 clean:
 	-rm -f */*.o
 	-rm -f */*/*.o
-	-rm -f depend depend.bak libmesa.a
+	-rm -f depend depend.bak libmesa.so
 	-rm -f drivers/*/*.o
 	(cd drivers/dri ; $(MAKE) clean)
 	(cd x86 ; $(MAKE) clean)
