diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/directfb/Makefile mesa-7.1rc1-20080706/src/mesa/drivers/directfb/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/directfb/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/directfb/Makefile	2008-07-06 14:24:31.000000000 +0200
@@ -25,7 +25,7 @@
 
 DIRECTFBGL_MESA = libidirectfbgl_mesa.so
 
-LIBS = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+LIBS = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .c.o:
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/dri/Makefile.template mesa-7.1rc1-20080706/src/mesa/drivers/dri/Makefile.template
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/dri/Makefile.template	2008-07-06 14:17:14.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/dri/Makefile.template	2008-07-06 14:18:41.000000000 +0200
@@ -1,6 +1,6 @@
 # -*-makefile-*-
 
-MESA_MODULES = $(TOP)/src/mesa/libmesa.a
+MESA_MODULES = $(TOP)/src/mesa/libmesa.so
 
 COMMON_SOURCES = \
         ../common/utils.c \
@@ -69,7 +69,7 @@
 
 $(LIBNAME): $(OBJECTS) $(MESA_MODULES) $(WINOBJ) Makefile $(TOP)/src/mesa/drivers/dri/Makefile.template
 	$(TOP)/bin/mklib -ldflags '$(LDFLAGS)' -noprefix -o $@ \
-		$(OBJECTS) $(MESA_MODULES) $(WINOBJ) $(DRI_LIB_DEPS)
+		$(OBJECTS) $(WINOBJ) $(DRI_LIB_DEPS)
 
 
 $(TOP)/$(LIB_DIR)/$(LIBNAME): $(LIBNAME)
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/fbdev/Makefile mesa-7.1rc1-20080706/src/mesa/drivers/fbdev/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/fbdev/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/fbdev/Makefile	2008-07-06 14:23:57.000000000 +0200
@@ -14,7 +14,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .c.o:
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/glslcompiler/Makefile mesa-7.1rc1-20080706/src/mesa/drivers/glslcompiler/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/glslcompiler/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/glslcompiler/Makefile	2008-07-06 14:25:20.000000000 +0200
@@ -13,7 +13,7 @@
 	../../glapi/glthread.o \
 	../../main/dispatch.o \
 	../common/driverfuncs.o \
-	../../libmesa.a
+	../../libmesa.so
 
 INCLUDES = \
 	-I$(TOP)/include \
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/osmesa/Makefile mesa-7.1rc1-20080706/src/mesa/drivers/osmesa/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/osmesa/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/osmesa/Makefile	2008-07-06 14:23:39.000000000 +0200
@@ -19,7 +19,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 .PHONY: osmesa8
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/drivers/x11/Makefile mesa-7.1rc1-20080706/src/mesa/drivers/x11/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/drivers/x11/Makefile	2008-07-06 14:09:54.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/drivers/x11/Makefile	2008-07-06 14:23:04.000000000 +0200
@@ -43,7 +43,7 @@
 	-I$(TOP)/src/mesa \
 	-I$(TOP)/src/mesa/main
 
-CORE_MESA = $(TOP)/src/mesa/libmesa.a $(TOP)/src/mesa/libglapi.a
+CORE_MESA = $(TOP)/src/mesa/libmesa.so $(TOP)/src/mesa/libglapi.so
 
 
 
diff -Naur mesa-7.1rc1-20080706.orig/src/mesa/Makefile mesa-7.1rc1-20080706/src/mesa/Makefile
--- mesa-7.1rc1-20080706.orig/src/mesa/Makefile	2008-07-06 14:17:14.000000000 +0200
+++ mesa-7.1rc1-20080706/src/mesa/Makefile	2008-07-06 14:21:54.000000000 +0200
@@ -20,8 +20,8 @@
 
 
 # Default: build dependencies, then asm_subdirs, then convenience
-# libs (.a) and finally the device drivers:
-default: depend asm_subdirs libmesa.a libglapi.a driver_subdirs
+# libs (.so) and finally the device drivers:
+default: depend asm_subdirs libmesa.so libglapi.so driver_subdirs
 
 
 
@@ -29,17 +29,17 @@
 # Helper libraries used by many drivers:
 
 # Make archive of core mesa object files
-libmesa.a: $(MESA_OBJECTS)
-	@ $(TOP)/bin/mklib -o mesa -static $(MESA_OBJECTS)
+libmesa.so: $(MESA_OBJECTS)
+	@ $(TOP)/bin/mklib -o mesa -dlopen $(MESA_OBJECTS)
 
 # Make archive of gl* API dispatcher functions only
-libglapi.a: $(GLAPI_OBJECTS)
-	@ $(TOP)/bin/mklib -o glapi -static $(GLAPI_OBJECTS)
+libglapi.so: $(GLAPI_OBJECTS)
+	@ $(TOP)/bin/mklib -o glapi -dlopen $(GLAPI_OBJECTS)
 
 
 ######################################################################
 # Device drivers
-driver_subdirs: libmesa.a libglapi.a
+driver_subdirs: libmesa.so libglapi.so
 	(cd drivers && $(MAKE))
 
 
@@ -123,7 +123,7 @@
 clean:
 	-rm -f */*.o
 	-rm -f */*/*.o
-	-rm -f depend depend.bak libmesa.a libglapi.a
+	-rm -f depend depend.bak libmesa.so libglapi.so
 	-rm -f drivers/*/*.o
 	-@cd drivers/dri && $(MAKE) clean
 	-@cd drivers/x11 && $(MAKE) clean
