diff a/configure.ac b/configure.ac
--- a/configure.ac	2012-01-30 19:02:00.000000000 +0100
+++ b/configure.ac	2012-01-30 19:02:30.189102590 +0100
@@ -1771,7 +1771,7 @@
     if test "x$LLVM_CONFIG" != xno; then
 	LLVM_VERSION=`$LLVM_CONFIG --version | sed 's/svn.*//g'`
 	LLVM_CFLAGS=`$LLVM_CONFIG --cppflags|sed -e 's/-DNDEBUG\>//g' -e 's/-pedantic//g'`
-	LLVM_LIBS="`$LLVM_CONFIG --libs`"
+	LLVM_LIBS="-lLLVM-$LLVM_VERSION"
 
 	LLVM_LDFLAGS=`$LLVM_CONFIG --ldflags`
 	DEFINES="$DEFINES -D__STDC_CONSTANT_MACROS"
