--- mesa-8.0.3/src/egl/wayland/wayland-drm/Makefile.orig	2012-07-25 14:19:44.466152883 +0300
+++ mesa-8.0.3/src/egl/wayland/wayland-drm/Makefile	2012-07-25 14:20:00.046068735 +0300
@@ -19,7 +19,7 @@
 prefix=$(shell $(PKG_CONFIG) --variable=prefix wayland-server)
 exec_prefx=$(shell $(PKG_CONFIG) --variable=exec_prefix wayland-server)
 wayland_protocoldir = $(PWD)/protocol
-wayland_scanner=$(exec_prefix)/bin/wayland-scanner
+wayland_scanner=wayland-scanner
 
 default: depend libwayland-drm.a $(GEN_SOURCES) $(GEN_HEADERS)
 
