--- mesa-8.0.2/bin/mklib.orig	2012-04-20 03:28:20.858442770 +0300
+++ mesa-8.0.2/bin/mklib	2012-04-20 03:30:10.946988676 +0300
@@ -334,9 +334,6 @@
 	    # environment.  If so, pass -m32 flag to linker.
 	    set ${OBJECTS}
 	    ABI32=`file $1 | grep 32-bit`
-	    if [ "${ABI32}" -a `uname -m` = "x86_64" ] ; then
-		OPTS="-m32 ${OPTS}"
-	    fi
 
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
@@ -391,9 +388,6 @@
 	    # environment.  If so, pass -m32 flag to linker.
 	    set ${OBJECTS}
 	    ABI32=`file $1 | grep 32-bit`
-	    if [ "${ABI32}" -a `uname -m` = "x86_64" ] ; then
-		OPTS="-m32 ${OPTS}"
-	    fi
             if [ "${ALTOPTS}" ] ; then
                 OPTS=${ALTOPTS}
             fi
@@ -464,9 +458,9 @@
 		set ${OBJECTS}
 		ABI32=`file $1 | grep 32-bit`
 		if [ "${ABI32}" ] ; then
-		    OPTS="-m32 -shared -Wl,-Bdynamic"
+		    OPTS="-shared -Wl,-Bdynamic"
 		else
-		    OPTS="-m64 -shared -Wl,-Bdynamic"
+		    OPTS="-shared -Wl,-Bdynamic"
 		fi
 	    fi
 
