--- vdr-epgsync-0.0.4/setup.c.orig	2013-03-04 18:28:39.225499203 +0100
+++ vdr-epgsync-0.0.4/setup.c	2013-03-04 18:28:47.233499340 +0100
@@ -90,7 +90,7 @@
 #endif
 
 	Add(new cMenuEditBoolItem(tr("Hide mainmenu entry"), &setupTmp.hideMainMenuEntry));
-	Add(new cMenuEditStrItem(tr("Server IP"), setupTmp.serverIp, 15, ".1234567890"));
+	Add(new cMenuEditStrItem(tr("Server IP"), setupTmp.serverIp, 16, ".1234567890"));
 	Add(new cMenuEditIntItem(tr("Server port"), &setupTmp.serverPort, 1, 65535));
 	Add(new cMenuEditIntItem(tr("Connection attempts"), &setupTmp.connectAttempts, 0, 10));
 	Add(new cMenuEditBoolItem(tr("Update \"now\" and \"next\" first"), &setupTmp.nowNext));
