--- cairo-1.12.14/boilerplate/cairo-boilerplate-vg.c.orig	2013-12-27 01:44:35.967928385 -0700
+++ cairo-1.12.14/boilerplate/cairo-boilerplate-vg.c	2013-12-27 01:46:40.235931464 -0700
@@ -215,7 +215,7 @@
 	EGL_ALPHA_SIZE, 8,
 	EGL_SURFACE_TYPE, EGL_PBUFFER_BIT,
 	EGL_RENDERABLE_TYPE, EGL_OPENVG_BIT,
-	None
+	EGL_NONE
     };
     int rgb_attribs[] = {
 	EGL_RED_SIZE, 8,
@@ -225,7 +225,7 @@
 	EGL_VG_ALPHA_FORMAT, EGL_VG_ALPHA_FORMAT_PRE_BIT,
 	EGL_SURFACE_TYPE, EGL_PBUFFER_BIT,
 	EGL_RENDERABLE_TYPE, EGL_OPENVG_BIT,
-	None
+	EGL_NONE
     };
     int dummy_attribs[] = {
 	EGL_WIDTH, 8, EGL_HEIGHT, 8,
