--- a/src/libcec/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp	2017-02-05 09:10:10.414054951 +0100
+++ b/src/libcec/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp	2017-02-05 09:10:10.458054775 +0100
@@ -268,7 +268,8 @@
       
       for (int la = CECDEVICE_TV; la < CECDEVICE_BROADCAST; la++)
       {
-        m_logicalAddresses.Set(cec_logical_address(la));  
+        if ((info.LogicalAddressMask >> la) & 1)
+          m_logicalAddresses.Set(cec_logical_address(la));  
       }
     }
 
