--- vdr-svdrpservice-1.0.0/setup.c.org	2013-06-14 10:06:32.748034009 +0200
+++ vdr-svdrpservice-1.0.0/setup.c	2013-06-14 10:06:45.996034237 +0200
@@ -54,7 +54,7 @@
 
 cSvdrpServiceMenuSetup::cSvdrpServiceMenuSetup() {
 	setupTmp = SvdrpServiceSetup;
-	Add(new cMenuEditStrItem(tr("Default server IP"), setupTmp.serverIp, 15, ".1234567890"));
+	Add(new cMenuEditStrItem(tr("Default server IP"), setupTmp.serverIp, 16, ".1234567890"));
 	Add(new cMenuEditIntItem(tr("Default server port"), &setupTmp.serverPort, 1, 65535));
 	if (cSvdrpServiceSetup::opt_serverIp)
 	{
