diff -Naur linux-linaro-imx5x-2.6.38-1000.10.orig//arch/arm/mach-mx5/Kconfig linux-linaro-imx5x-2.6.38-1000.10/arch/arm/mach-mx5/Kconfig
--- linux-linaro-imx5x-2.6.38-1000.10.orig//arch/arm/mach-mx5/Kconfig	2011-10-12 21:57:05.213491715 +0200
+++ linux-linaro-imx5x-2.6.38-1000.10/arch/arm/mach-mx5/Kconfig	2011-10-12 21:57:21.713573518 +0200
@@ -146,6 +146,7 @@
 	select IMX_HAVE_PLATFORM_IMX_UART
 	select IMX_HAVE_PLATFORM_SDHCI_ESDHC_IMX
 	select IMX_HAVE_PLATFORM_SPI_IMX
+        select IMX_HAVE_PLATFORM_IMX_SRTC
 	select MXC_ULPI if USB_ULPI
 
 config MACH_MX51_EFIKAMX
