From 3996d3e862f648f503b6c0031ac5771fa38385fd Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Tue, 7 Nov 2017 13:17:42 -0800
Subject: [PATCH 3/3] hack for kodi

---
 libavcodec/v4l2_m2m_dec.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libavcodec/v4l2_m2m_dec.c b/libavcodec/v4l2_m2m_dec.c
index 82397fc8a1..ec217b328d 100644
--- a/libavcodec/v4l2_m2m_dec.c
+++ b/libavcodec/v4l2_m2m_dec.c
@@ -165,6 +165,10 @@ static av_cold int v4l2_decode_init(AVCodecContext *avctx)
     V4L2Context *output = &s->output;
     int ret;
 
+    // HACK for Kodi
+    if ((ret = avctx->get_format(avctx, avctx->codec->pix_fmts)) < 0)
+        return ret;
+
     /* if these dimensions are invalid (ie, 0 or too small) an event will be raised
      * by the v4l2 driver; this event will trigger a full pipeline reconfig and
      * the proper values will be retrieved from the kernel driver.
-- 
2.11.0

