--- tvheadend-3.0+1c96e30d/Makefile	2012-08-11 22:18:56.000000000 +0200
+++ tvheadend-3.0+1c96e30d.patch/Makefile	2012-08-14 20:08:11.182616136 +0200
@@ -24,7 +24,7 @@
 
 PROG=${BUILDDIR}/tvheadend
 
-CFLAGS += -Wall -Werror -Wwrite-strings -Wno-deprecated-declarations
+CFLAGS += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS += -Wmissing-prototypes -fms-extensions
 LDFLAGS += -lrt -ldl
 
