--- xavs/common/osdep.h.ark	2010-11-19 14:03:55.933333119 +0100
+++ xavs/common/osdep.h	2010-11-19 14:10:17.463333120 +0100
@@ -191,13 +191,13 @@ xavs_pthread_create (xavs_pthread_t * t,
 #define endian_fix32(x) (x)
 #define endian_fix16(x) (x)
 #else
-#if defined(__GNUC__) && HAVE_MMX
+#if defined(__GNUC__) && defined(HAVE_MMX)
 static ALWAYS_INLINE uint32_t endian_fix32( uint32_t x )
 {
     asm("bswap %0":"+r"(x));
     return x;
 }
-#elif defined(__GNUC__) && HAVE_ARMV6
+#elif defined(__GNUC__) && defined(HAVE_ARMV6)
 static ALWAYS_INLINE uint32_t endian_fix32( uint32_t x )
 {
     asm("rev %0, %0":"+r"(x));
@@ -209,7 +209,7 @@ static ALWAYS_INLINE uint32_t endian_fix
     return (x<<24) + ((x<<8)&0xff0000) + ((x>>8)&0xff00) + (x>>24);
 }
 #endif
-#if defined(__GNUC__) && ARCH_X86_64
+#if defined(__GNUC__) && defined(ARCH_X86_64)
 static ALWAYS_INLINE uint64_t endian_fix64( uint64_t x )
 {
     asm("bswap %0":"+r"(x));
