Index: pkg/Makefile.am
===================================================================
--- pkg.orig/Makefile.am	2011-12-07 15:39:55.013001000 +0100
+++ pkg/Makefile.am	2011-12-07 15:40:08.007157000 +0100
@@ -43,7 +43,7 @@
 
 if ENABLE_TEST
 SUBDIRS            += test
-bin_PROGRAMS        = dri2test dri2videotest
+bin_PROGRAMS        = dri2test
 
 COMMON_LDFLAGS      = -no-undefined
 COMMON_LDADD        = src/libdri2.la @DRI2_LIBS@
@@ -51,8 +51,4 @@
 dri2test_SOURCES    =
 dri2test_LDFLAGS    = $(COMMON_LDFLAGS)
 dri2test_LDADD      = test/libdri2test.la $(COMMON_LDADD)
-
-dri2videotest_SOURCES =
-dri2videotest_LDFLAGS = $(COMMON_LDFLAGS)
-dri2videotest_LDADD   = test/libdri2videotest.la $(COMMON_LDADD)
 endif
Index: pkg/test/Makefile.am
===================================================================
--- pkg.orig/test/Makefile.am	2011-12-07 15:40:16.000559000 +0100
+++ pkg/test/Makefile.am	2011-12-07 15:40:31.009444000 +0100
@@ -1,4 +1,4 @@
-lib_LTLIBRARIES = libdri2test.la libdri2videotest.la
+lib_LTLIBRARIES = libdri2test.la
 
 COMMON_SOURCES = dri2util.c
 COMMON_LIBADD  = @DRI2_LIBS@ 
@@ -19,7 +19,3 @@
 libdri2test_la_SOURCES = dri2test.c $(COMMON_SOURCES)
 libdri2test_la_LIBADD = $(COMMON_LIBADD)
 libdri2test_la_CFLAGS = $(COMMON_CFLAGS)
-
-libdri2videotest_la_SOURCES = dri2videotest.c $(COMMON_SOURCES)
-libdri2videotest_la_LIBADD = $(COMMON_LIBADD)
-libdri2videotest_la_CFLAGS = $(COMMON_CFLAGS)
