--- a/glob/glob.c	2018-05-03 09:01:18.329907856 +0200
+++ b/glob/glob.c	2018-05-03 09:18:49.866661471 +0200
@@ -208,7 +208,7 @@
 #endif /* __GNU_LIBRARY__ || __DJGPP__ */
 
 
-#if !defined __alloca && !defined __GNU_LIBRARY__
+#if !defined __alloca /* && !defined __GNU_LIBRARY__ */
 
 # ifdef	__GNUC__
 #  undef alloca
@@ -239,6 +239,8 @@
 # ifndef S_ISDIR
 #  define S_ISDIR(mode) (((mode) & S_IFMT) == S_IFDIR)
 # endif
+#else
+#  define __stat(fname, buf) __xstat (_STAT_VER, fname, buf)
 #endif
 
 #ifdef _LIBC
