--- a/Makefile.real	2014-05-04 12:10:36.212418348 +0200
+++ b/Makefile.real	2014-05-04 12:09:28.000000000 +0200
@@ -87,15 +87,15 @@
 	@$(MAKE) -C $(KLIB_BUILD) M=$(BACKPORT_PWD)			\
 		INSTALL_MOD_DIR=$(KMODDIR) $(KMODPATH_ARG)		\
 		modules_install
-	@./scripts/blacklist.sh $(KLIB)/ $(KLIB)/$(KMODDIR)
-	@./scripts/compress_modules.sh $(KLIB)/$(KMODDIR)
-	@./scripts/check_depmod.sh
-	@/sbin/depmod -a
-	@./scripts/update-initramfs.sh $(KLIB)
-	@echo
-	@echo Your backported driver modules should be installed now.
-	@echo Reboot.
-	@echo
+#	@./scripts/blacklist.sh $(KLIB)/ $(KLIB)/$(KMODDIR)
+#	@./scripts/compress_modules.sh $(KLIB)/$(KMODDIR)
+#	@./scripts/check_depmod.sh
+#	@/sbin/depmod -a
+#	@./scripts/update-initramfs.sh $(KLIB)
+#	@echo
+#	@echo Your backported driver modules should be installed now.
+#	echo Reboot.
+#	echo
 
 .PHONY: modules_install
 modules_install: install
