diff -Naur dbus-glib-0.82.orig/dbus/Makefile.am dbus-glib-0.82/dbus/Makefile.am
--- dbus-glib-0.82.orig/dbus/Makefile.am	2010-07-09 21:43:32.435969341 +0200
+++ dbus-glib-0.82/dbus/Makefile.am	2010-07-09 21:44:20.068421064 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 INCLUDES = \
 	-I$(top_srcdir)						\
diff -Naur dbus-glib-0.82.orig/dbus/Makefile.in dbus-glib-0.82/dbus/Makefile.in
--- dbus-glib-0.82.orig/dbus/Makefile.in	2010-07-09 21:43:32.435969341 +0200
+++ dbus-glib-0.82/dbus/Makefile.in	2010-07-09 21:44:31.340700515 +0200
@@ -352,7 +352,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = . examples
+SUBDIRS = .
 INCLUDES = \
 	-I$(top_srcdir)						\
 	-I$(top_builddir)					\
diff -Naur dbus-glib-0.82.orig/Makefile.am dbus-glib-0.82/Makefile.am
--- dbus-glib-0.82.orig/Makefile.am	2010-07-09 21:43:32.445969652 +0200
+++ dbus-glib-0.82/Makefile.am	2010-07-09 21:44:02.425968799 +0200
@@ -1,8 +1,8 @@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC=dbus-glib-1.pc
 
-SUBDIRS=dbus tools test doc
-DIST_SUBDIRS=dbus tools test doc m4
+SUBDIRS=dbus test doc
+DIST_SUBDIRS=dbus test doc m4
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
diff -Naur dbus-glib-0.82.orig/Makefile.in dbus-glib-0.82/Makefile.in
--- dbus-glib-0.82.orig/Makefile.in	2010-07-09 21:43:32.435969341 +0200
+++ dbus-glib-0.82/Makefile.in	2010-07-09 21:44:10.854829136 +0200
@@ -306,8 +306,8 @@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
 GLIB_PC = dbus-glib-1.pc
-SUBDIRS = dbus tools test doc
-DIST_SUBDIRS = dbus tools test doc m4
+SUBDIRS = dbus test doc
+DIST_SUBDIRS = dbus test doc m4
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = $(GLIB_PC)
 DISTCLEANFILES = \
