diff -Naur busybox-1.18.4.orig//miscutils/rfkill.c busybox-1.18.4/miscutils/rfkill.c
--- busybox-1.18.4.orig//miscutils/rfkill.c	2011-05-05 23:47:19.077064000 +0200
+++ busybox-1.18.4/miscutils/rfkill.c	2011-05-05 23:47:53.327064000 +0200
@@ -9,6 +9,10 @@
 #include "libbb.h"
 #include <linux/rfkill.h>
 
+#ifndef RFKILL_EVENT_SIZE_V1
+#define RFKILL_EVENT_SIZE_V1    8
+#endif
+
 enum {
 	OPT_b = (1 << 0), /* must be = 1 */
 	OPT_u = (1 << 1),
