--- busybox-1.3.0/scripts/kconfig/confdata.c.orig	2006-12-17 23:43:40.000000000 +0100
+++ busybox-1.3.0/scripts/kconfig/confdata.c	2006-12-17 23:44:08.000000000 +0100
@@ -151,9 +151,6 @@
 			if (!sym) {
 				conf_warning("trying to assign nonexistent symbol %s", line + 9);
 				break;
-			} else if (!(sym->flags & SYMBOL_NEW)) {
-				conf_warning("trying to reassign symbol %s", sym->name);
-				break;
 			}
 			switch (sym->type) {
 			case S_BOOLEAN:
@@ -181,9 +178,6 @@
 			if (!sym) {
 				conf_warning("trying to assign nonexistent symbol %s", line + 7);
 				break;
-			} else if (!(sym->flags & SYMBOL_NEW)) {
-				conf_warning("trying to reassign symbol %s", sym->name);
-				break;
 			}
 			switch (sym->type) {
 			case S_TRISTATE:
