--- busybox-1.00.orig/Makefile	Fri Oct  8 09:45:08 2004
+++ busybox-1.00/Makefile	Tue Dec 14 00:14:10 2004
@@ -196,10 +196,8 @@
 .depend: scripts/mkdep
 	rm -f .depend .hdepend;
 	mkdir -p include/config;
-	scripts/mkdep -I include -- \
-	  `find $(top_srcdir) -name \*.c -print | sed -e "s,^./,,"` >> .depend;
-	scripts/mkdep -I include -- \
-	  `find $(top_srcdir) -name \*.h -print | sed -e "s,^./,,"` >> .hdepend;
+	find $(top_srcdir) -name \*.c -print | sed -e "s,^./,," | xargs scripts/mkdep -I include -- | cat >> .depend;
+	find $(top_srcdir) -name \*.h -print | sed -e "s,^./,," | xargs scripts/mkdep -I include -- | cat >> .hdepend;
 
 depend dep: include/config.h .depend
 
