diff -Naur busybox-1.4.0.orig/Config.in busybox-1.4.0/Config.in
--- busybox-1.4.0.orig/Config.in	2007-01-21 17:41:20.000000000 +0100
+++ busybox-1.4.0/Config.in	2007-01-21 17:59:30.000000000 +0100
@@ -469,6 +469,7 @@
 source init/Config.in
 source loginutils/Config.in
 source e2fsprogs/Config.in
+source e2fsprogs/old_e2fsprogs/Config.in
 source modutils/Config.in
 source util-linux/Config.in
 source miscutils/Config.in
