diff -Naur busybox-1.4.1.orig/e2fsprogs/Config.in busybox-1.4.1/e2fsprogs/Config.in
--- busybox-1.4.1.orig/e2fsprogs/Config.in	2007-02-25 17:39:42.000000000 +0100
+++ busybox-1.4.1/e2fsprogs/Config.in	2007-02-25 18:35:14.000000000 +0100
@@ -64,4 +64,6 @@
 ### 	  findfs will search the disks in the system looking for a filesystem
 ### 	  which has a label matching label or a UUID equal to uuid.
 
+source e2fsprogs/old_e2fsprogs/Config.in
+
 endmenu
diff -Naur busybox-1.4.1.orig/e2fsprogs/old_e2fsprogs/ext2fs/Kbuild busybox-1.4.1/e2fsprogs/old_e2fsprogs/ext2fs/Kbuild
--- busybox-1.4.1.orig/e2fsprogs/old_e2fsprogs/ext2fs/Kbuild	2007-02-25 17:39:42.000000000 +0100
+++ busybox-1.4.1/e2fsprogs/old_e2fsprogs/ext2fs/Kbuild	2007-02-25 18:28:27.000000000 +0100
@@ -20,4 +20,4 @@
                    dirhash.o version.o flushb.o unlink.o check_desc.o valid_blk.o \
                    ext_attr.o bmap.o dblist_dir.o ext2fs_inline.o swapfs.o
 
-CFLAGS += -include $(srctree)/e2fsprogs/e2fsbb.h
+CFLAGS += -include $(srctree)/e2fsprogs/old_e2fsprogs/e2fsbb.h
diff -Naur busybox-1.4.1.orig/e2fsprogs/old_e2fsprogs/Kbuild busybox-1.4.1/e2fsprogs/old_e2fsprogs/Kbuild
--- busybox-1.4.1.orig/e2fsprogs/old_e2fsprogs/Kbuild	2007-02-25 17:39:42.000000000 +0100
+++ busybox-1.4.1/e2fsprogs/old_e2fsprogs/Kbuild	2007-02-25 18:16:47.000000000 +0100
@@ -13,4 +13,4 @@
 lib-$(CONFIG_MKE2FS)     += mke2fs.o util.o
 lib-$(CONFIG_TUNE2FS)    += tune2fs.o util.o
 
-CFLAGS += -include $(srctree)/e2fsprogs/e2fsbb.h
+CFLAGS += -include $(srctree)/e2fsprogs/old_e2fsprogs/e2fsbb.h
diff -Naur busybox-1.4.1.orig/include/applets.h busybox-1.4.1/include/applets.h
--- busybox-1.4.1.orig/include/applets.h	2007-02-25 17:39:42.000000000 +0100
+++ busybox-1.4.1/include/applets.h	2007-02-25 18:23:17.000000000 +0100
@@ -192,7 +192,7 @@
 USE_MDEV(APPLET(mdev, _BB_DIR_SBIN, _BB_SUID_NEVER))
 USE_MESG(APPLET(mesg, _BB_DIR_USR_BIN, _BB_SUID_NEVER))
 USE_MKDIR(APPLET_NOFORK(mkdir, mkdir, _BB_DIR_BIN, _BB_SUID_NEVER, mkdir))
-//USE_MKE2FS(APPLET(mke2fs, _BB_DIR_SBIN, _BB_SUID_NEVER))
+USE_MKE2FS(APPLET(mke2fs, _BB_DIR_SBIN, _BB_SUID_NEVER))
 USE_MKFIFO(APPLET(mkfifo, _BB_DIR_USR_BIN, _BB_SUID_NEVER))
 //USE_MKE2FS(APPLET_NOUSAGE(mkfs.ext2, mke2fs, _BB_DIR_SBIN, _BB_SUID_NEVER))
 //USE_MKE2FS(APPLET_NOUSAGE(mkfs.ext3, mke2fs, _BB_DIR_SBIN, _BB_SUID_NEVER))
diff -Naur busybox-1.4.1.orig/Makefile busybox-1.4.1/Makefile
--- busybox-1.4.1.orig/Makefile	2007-02-25 17:39:42.000000000 +0100
+++ busybox-1.4.1/Makefile	2007-02-25 18:26:53.000000000 +0100
@@ -429,6 +429,11 @@
 		coreutils/libcoreutils/ \
 		debianutils/ \
 		e2fsprogs/ \
+		e2fsprogs/old_e2fsprogs/ \
+		e2fsprogs/old_e2fsprogs/blkid/ \
+		e2fsprogs/old_e2fsprogs/e2p/ \
+		e2fsprogs/old_e2fsprogs/ext2fs/ \
+		e2fsprogs/old_e2fsprogs/uuid/ \
 		editors/ \
 		findutils/ \
 		init/ \
