diff -Naur busybox-1.13.2/e2fsprogs/old_e2fsprogs/e2p/e2p.h busybox-1.13.2a/e2fsprogs/old_e2fsprogs/e2p/e2p.h
--- busybox-1.13.2/e2fsprogs/old_e2fsprogs/e2p/e2p.h	2009-01-31 17:59:40.000000000 +0100
+++ busybox-1.13.2a/e2fsprogs/old_e2fsprogs/e2p/e2p.h	2009-01-31 17:57:48.000000000 +0100
@@ -38,7 +38,7 @@
 void list_super2(struct ext2_super_block * s, FILE *f);
 #define list_super(s) list_super2(s, stdout)
 void print_fs_errors (FILE *f, unsigned short errors);
-void print_flags (FILE *f, unsigned long flags, unsigned options);
+//void print_flags (FILE *f, unsigned long flags, unsigned options);
 void print_fs_state (FILE *f, unsigned short state);
 int setflags (int fd, unsigned long flags);
 int setversion (int fd, unsigned long version);
diff -Naur busybox-1.13.2/e2fsprogs/old_e2fsprogs/e2p/pf.c busybox-1.13.2a/e2fsprogs/old_e2fsprogs/e2p/pf.c
--- busybox-1.13.2/e2fsprogs/old_e2fsprogs/e2p/pf.c	2009-01-31 17:59:32.000000000 +0100
+++ busybox-1.13.2a/e2fsprogs/old_e2fsprogs/e2p/pf.c	2009-01-31 17:58:22.000000000 +0100
@@ -48,6 +48,7 @@
 	{ 0, NULL, NULL }
 };
 
+#if 0
 void print_flags (FILE *f, unsigned long flags, unsigned options)
 {
 	int long_opt = (options & PFOPT_LONG);
@@ -72,3 +73,4 @@
 	if (long_opt && first)
 		fputs("---", f);
 }
+#endif
