diff -Naur busybox-1.11.1.orig/e2fsprogs/old_e2fsprogs/e2p/e2p.h busybox-1.11.1/e2fsprogs/old_e2fsprogs/e2p/e2p.h
--- busybox-1.11.1.orig/e2fsprogs/old_e2fsprogs/e2p/e2p.h	2008-08-05 16:07:52.000000000 +0200
+++ busybox-1.11.1/e2fsprogs/old_e2fsprogs/e2p/e2p.h	2008-08-05 16:10:19.000000000 +0200
@@ -38,7 +38,7 @@
 void list_super2(struct ext2_super_block * s, FILE *f);
 #define list_super(s) list_super2(s, stdout)
 void print_fs_errors (FILE * f, unsigned short errors);
-void print_flags (FILE * f, unsigned long flags, unsigned options);
+//void print_flags (FILE * f, unsigned long flags, unsigned options);
 void print_fs_state (FILE * f, unsigned short state);
 int setflags (int fd, unsigned long flags);
 int setversion (int fd, unsigned long version);
diff -Naur busybox-1.11.1.orig/e2fsprogs/old_e2fsprogs/e2p/pf.c busybox-1.11.1/e2fsprogs/old_e2fsprogs/e2p/pf.c
--- busybox-1.11.1.orig/e2fsprogs/old_e2fsprogs/e2p/pf.c	2008-08-05 16:07:52.000000000 +0200
+++ busybox-1.11.1/e2fsprogs/old_e2fsprogs/e2p/pf.c	2008-08-05 16:10:24.000000000 +0200
@@ -48,6 +48,7 @@
 	{ 0, NULL, NULL }
 };
 
+#if 0
 void print_flags (FILE * f, unsigned long flags, unsigned options)
 {
 	int long_opt = (options & PFOPT_LONG);
@@ -72,3 +73,4 @@
 	if (long_opt && first)
 		fputs("---", f);
 }
+#endif
