diff -Naur busybox.orig/archival/lzmacat.c busybox/archival/lzmacat.c
--- busybox.orig/archival/lzmacat.c	2005-07-08 21:57:47.719055664 +0200
+++ busybox/archival/lzmacat.c	2005-07-08 21:58:30.403566632 +0200
@@ -70,21 +70,11 @@
 	fprintf(stderr, "Can't read uncompressed size!\n");
   }
 	
-  if (outSize == 0xFFFFFFFF) {
-	fprintf(stderr, "Stream version is not supported!\n");
-	return 1;
-  }
-	
   if( !fread(&zero, 1, sizeof(zero), infp)) {
 	fprintf(stderr, "Cannot read hi-byte uncompressed size!\n");
 	return 1;
   }
 
-  if (zero != 0) {
-	fprintf(stderr, "File too big (bigger than 4 GB)!\n");
-	return 1;
-  }
-
   if( LzmaDecodeProperties( &state.Properties, properties, LZMA_PROPERTIES_SIZE ) != LZMA_RESULT_OK ) 
   {
 	fprintf(stderr, "Lzma properties error!\n");
