Index: busybox-1.1.1/miscutils/eject.c
===================================================================
--- busybox-1.1.1.orig/miscutils/eject.c	2006-03-22 22:16:24.000000000 +0100
+++ busybox-1.1.1/miscutils/eject.c	2006-03-24 01:45:45.000000000 +0100
@@ -29,18 +29,10 @@
 {
 	unsigned long flags;
 	char *device;
-	struct mntent *m;
 
 	flags = bb_getopt_ulflags(argc, argv, "t");
 	device = argv[optind] ? : DEFAULT_CDROM;
 
-	if ((m = find_mount_point(device, bb_path_mtab_file))) {
-		if (umount(m->mnt_dir)) {
-			bb_error_msg_and_die("Can't umount");
-		} else if (ENABLE_FEATURE_MTAB_SUPPORT) {
-			erase_mtab(m->mnt_fsname);
-		}
-	}
 	if (ioctl(bb_xopen(device, (O_RDONLY | O_NONBLOCK)),
 				(flags ? CDROMCLOSETRAY : CDROMEJECT))) {
 		bb_perror_msg_and_die("%s", device);
