diff -Nur busybox.orig/miscutils/eject.c busybox/miscutils/eject.c
--- busybox.orig/miscutils/eject.c	Sat Jun  4 09:20:16 2005
+++ busybox/miscutils/eject.c	Sat Jul  2 12:20:11 2005
@@ -40,19 +40,10 @@
 {
 	unsigned long flags;
 	char *device;
-	struct mntent *m;
 	
 	flags = bb_getopt_ulflags(argc, argv, "t");
 	device=argv[optind] ? : DEFAULT_CDROM;
 	
-	if((m = find_mount_point(device, bb_path_mtab_file))) {
-		if(umount(m->mnt_dir))
-			bb_error_msg_and_die("Can't umount");
-#ifdef CONFIG_FEATURE_MTAB_SUPPORT
-		else
-			erase_mtab(m->mnt_fsname);	
-#endif
-	}
 	if (ioctl(bb_xopen( device, 
 	                   (O_RDONLY | O_NONBLOCK)), 
 	          ( flags ? CDROMCLOSETRAY : CDROMEJECT)))
