diff -Naur busybox-1.2.1.orig/include/usage.h busybox-1.2.1/include/usage.h
--- busybox-1.2.1.orig/include/usage.h	2006-09-28 23:24:58.000000000 +0200
+++ busybox-1.2.1/include/usage.h	2006-09-28 23:25:30.000000000 +0200
@@ -631,11 +631,12 @@
 	"Erik\\nis\\ncool\n")
 
 #define eject_trivial_usage \
-	"[-t] [DEVICE]"
+	"[-t] [-T] [DEVICE]"
 #define eject_full_usage \
 	"Eject specified DEVICE (or default /dev/cdrom).\n\n" \
 	"Options:\n" \
-	"\t-t\tclose tray"
+	"\t-t\tclose tray" \
+	"\t-T\tsmart open and close tray"
 
 #define ed_trivial_usage ""
 #define ed_full_usage ""
diff -Naur busybox-1.2.1.orig/miscutils/eject.c busybox-1.2.1/miscutils/eject.c
--- busybox-1.2.1.orig/miscutils/eject.c	2006-09-28 23:24:58.000000000 +0200
+++ busybox-1.2.1/miscutils/eject.c	2006-09-28 23:26:39.000000000 +0200
@@ -22,19 +22,35 @@
 /* various defines swiped from linux/cdrom.h */
 #define CDROMCLOSETRAY            0x5319  /* pendant of CDROMEJECT  */
 #define CDROMEJECT                0x5309  /* Ejects the cdrom media */
+#define CDROM_DRIVE_STATUS        0x5326  /* Get tray position, etc. */
 #define DEFAULT_CDROM             "/dev/cdrom"
+/* drive status possibilities returned by CDROM_DRIVE_STATUS ioctl */
+#define CDS_TRAY_OPEN        2
+
+#define FLAG_CLOSE  1
+#define FLAG_SMART  2
 
 int eject_main(int argc, char **argv)
 {
 	unsigned long flags;
 	char *device;
+	int dev, cmd;
 
-	flags = bb_getopt_ulflags(argc, argv, "t");
+	flags = bb_getopt_ulflags(argc, argv, "tT");
 	device = argv[optind] ? : DEFAULT_CDROM;
 
-	if (ioctl(bb_xopen(device, (O_RDONLY | O_NONBLOCK)),
-				(flags ? CDROMCLOSETRAY : CDROMEJECT))) {
+	dev = bb_xopen(device, O_RDONLY|O_NONBLOCK);
+	if (flags & FLAG_CLOSE ||
+		(flags & FLAG_SMART && ioctl(dev,CDROM_DRIVE_STATUS) == CDS_TRAY_OPEN))
+		cmd = CDROMCLOSETRAY;
+	else
+		cmd = CDROMEJECT;
+	if (ioctl(dev, cmd)) {
 		bb_perror_msg_and_die("%s", device);
 	}
+
+	if (ENABLE_FEATURE_CLEAN_UP)
+		close(dev);
+
 	return (EXIT_SUCCESS);
 }
