diff -Naur busybox-1.5.0.orig/shell/ash.c busybox-1.5.0/shell/ash.c
--- busybox-1.5.0.orig/shell/ash.c	2007-03-23 23:24:43.000000000 +0100
+++ busybox-1.5.0/shell/ash.c	2007-03-23 23:31:22.000000000 +0100
@@ -3985,6 +3985,47 @@
 	return retval;
 }
 
+#include <sys/file.h>
+
+static int
+flockcmd(int argc, char **argv)
+{
+	static int fd = -1;
+	int operation = 0, c;
+
+	while ((c = nextopt("ensu")))
+		switch (c) {
+		case 'e':
+			operation |= LOCK_EX;
+			break;
+		case 'n':
+			operation |= LOCK_NB;
+			break;
+		case 's':
+			operation |= LOCK_SH;
+			break;
+		case 'u':
+			if (fd == -1 || flock(fd, LOCK_UN))
+				return 127;
+			close(fd);
+			fd = -1;
+			return 0;
+		}
+
+	c = 127;
+	if (fd == -1 && *argptr && (fd = open(*argptr, O_RDONLY | O_CREAT)) != -1)
+	{
+		if (!flock(fd, operation)) {
+			c = 0;
+		} else {
+			close(fd);
+			fd = -1;
+		}
+	}
+
+	return c;
+}
+
 static int
 waitcmd(int argc, char **argv)
 {
@@ -8170,6 +8211,7 @@
 #if JOBS
 	{ BUILTIN_REGULAR       "fg", fg_bgcmd },
 #endif
+	{ BUILTIN_NOSPEC        "flock", flockcmd },
 #if ENABLE_ASH_GETOPTS
 	{ BUILTIN_REGULAR       "getopts", getoptscmd },
 #endif
