--- busybox-1.16.0/util-linux/switch_root.c.orig	2010-01-24 19:59:39.000000000 -0500
+++ busybox-1.16.0/util-linux/switch_root.c	2010-01-28 15:57:18.000000000 -0500
@@ -85,8 +85,10 @@
 	// Additional sanity checks: we're about to rm -rf /, so be REALLY SURE
 	// we mean it. I could make this a CONFIG option, but I would get email
 	// from all the people who WILL destroy their filesystems.
-	if (stat("/init", &st) != 0 || !S_ISREG(st.st_mode)) {
-		bb_error_msg_and_die("/init is not a regular file");
+	if ((stat("/init", &st) != 0 || !S_ISREG(st.st_mode))
+		&& (stat("/linuxrc", &st) != 0 || !S_ISREG(st.st_mode)))
+	{
+		bb_error_msg_and_die("neither /init nor /linuxrc is found and a regular file");
 	}
 	statfs("/", &stfs); // this never fails
 	if ((unsigned)stfs.f_type != RAMFS_MAGIC
