diff -Naur busybox-1.11.1.orig/shell/ash.c busybox-1.11.1/shell/ash.c
--- busybox-1.11.1.orig/shell/ash.c	2008-08-05 16:12:04.000000000 +0200
+++ busybox-1.11.1/shell/ash.c	2008-08-05 16:12:37.000000000 +0200
@@ -3976,6 +3976,31 @@
 	}
 }
 
+static int
+jobscmd(int argc ATTRIBUTE_UNUSED, char **argv)
+{
+	int mode, m;
+
+	mode = 0;
+	while ((m = nextopt("lp"))) {
+		if (m == 'l')
+			mode = SHOW_PID;
+		else
+			mode = SHOW_PGID;
+	}
+
+	argv = argptr;
+	if (*argv) {
+		do
+			showjob(stdout, getjob(*argv,0), mode);
+		while (*++argv);
+	} else
+		showjobs(stdout, mode);
+
+	return 0;
+}
+#endif /* JOBS */
+
 #include <sys/file.h>
 
 static int
@@ -4018,31 +4043,6 @@
 }
 
 static int
-jobscmd(int argc ATTRIBUTE_UNUSED, char **argv)
-{
-	int mode, m;
-
-	mode = 0;
-	while ((m = nextopt("lp"))) {
-		if (m == 'l')
-			mode = SHOW_PID;
-		else
-			mode = SHOW_PGID;
-	}
-
-	argv = argptr;
-	if (*argv) {
-		do
-			showjob(stdout, getjob(*argv,0), mode);
-		while (*++argv);
-	} else
-		showjobs(stdout, mode);
-
-	return 0;
-}
-#endif /* JOBS */
-
-static int
 getstatus(struct job *job)
 {
 	int status;
