diff -Nur busybox.orig/util-linux/umount.c busybox/util-linux/umount.c
--- busybox.orig/util-linux/umount.c	Fri Jul 22 16:44:34 2005
+++ busybox/util-linux/umount.c	Fri Jul 22 16:46:55 2005
@@ -247,7 +247,8 @@
 
 extern int umount_main(int argc, char **argv)
 {
-	char path[PATH_MAX], result = 0;
+	int result = 0;
+	char path[PATH_MAX];
 
 	if (argc < 2) {
 		bb_show_usage();
@@ -305,8 +306,8 @@
 		if (realpath(*argv, path) != NULL)
 			if (do_umount(path))
 				continue;
+		result = errno ? errno : 127;
 		bb_perror_msg("%s", path);
-		result++;
 	} while (--argc > 0 && ++argv);
 	return result;
 }
