diff -Nur busybox.orig/util-linux/umount.c busybox/util-linux/umount.c
--- busybox.orig/util-linux/umount.c	Sat Aug 27 09:20:22 2005
+++ busybox/util-linux/umount.c	Sat Aug 27 13:31:49 2005
@@ -30,7 +30,7 @@
 	char path[2*PATH_MAX];
 	struct mntent me;
 	FILE *fp;
-	int status=EXIT_SUCCESS;
+	int status=0;
 	struct mtab_list {
 		char *dir;
 		char *device;
@@ -119,7 +119,7 @@
 			/* Yes, the ENABLE is redundant here, but the optimizer for ARM
 			 * can't do simple constant propogation in local variables... */
 			if(ENABLE_FEATURE_MTAB_SUPPORT && useMtab && m) erase_mtab(m->dir);
-			status = EXIT_FAILURE;
+			status = errno ? errno : 127;
 			bb_perror_msg("Couldn't umount %s\n", path);
 		}
 		// Find next matching mtab entry for -a or umount /dev
