Index: busybox-1.1.1/util-linux/umount.c
===================================================================
--- busybox-1.1.1.orig/util-linux/umount.c	2006-03-22 22:16:26.000000000 +0100
+++ busybox-1.1.1/util-linux/umount.c	2006-03-24 01:47:09.000000000 +0100
@@ -36,7 +36,7 @@
 	char path[2*PATH_MAX];
 	struct mntent me;
 	FILE *fp;
-	int status = EXIT_SUCCESS;
+	int status = 0;
 	unsigned long opt;
 	struct mtab_list {
 		char *dir;
@@ -128,7 +128,7 @@
 			 * can't do simple constant propagation in local variables... */
 			if(ENABLE_FEATURE_MTAB_SUPPORT && !(opt & OPT_NO_MTAB) && m)
 				erase_mtab(m->dir);
-			status = EXIT_FAILURE;
+			status = errno ? errno : 127;
 			bb_perror_msg("Couldn't umount %s", path);
 		}
 		// Find next matching mtab entry for -a or umount /dev
