diff -Nur busybox.orig/util-linux/umount.c busybox/util-linux/umount.c
--- busybox.orig/util-linux/umount.c	Tue Sep 27 10:21:15 2005
+++ busybox/util-linux/umount.c	Tue Sep 27 20:01:55 2005
@@ -37,7 +37,7 @@
 	char path[2*PATH_MAX];
 	struct mntent me;
 	FILE *fp;
-	int status=EXIT_SUCCESS;
+	int status=0;
 	unsigned long opt;
 	struct mtab_list {
 		char *dir;
@@ -123,7 +123,7 @@
 			 * can't do simple constant propagation in local variables... */
 			if(ENABLE_FEATURE_MTAB_SUPPORT && !(opt & OPT_NO_MTAB) && m)
 				erase_mtab(m->dir);
-			status = EXIT_FAILURE;
+			status = errno ? errno : 127;
 			bb_perror_msg("Couldn't umount %s\n", path);
 		}
 		// Find next matching mtab entry for -a or umount /dev
