diff -Nur busybox-1.2.2.orig/util-linux/umount.c busybox-1.2.2/util-linux/umount.c
--- busybox-1.2.2.orig/util-linux/umount.c	2006-10-24 22:21:17.000000000 +0200
+++ busybox-1.2.2/util-linux/umount.c	2006-11-10 14:57:27.000000000 +0200
@@ -30,7 +30,7 @@
 	char path[2*PATH_MAX];
 	struct mntent me;
 	FILE *fp;
-	int status = EXIT_SUCCESS;
+	int status = 0;
 	unsigned long opt;
 	struct mtab_list {
 		char *dir;
@@ -117,7 +117,7 @@
 		}
 
 		if (curstat) {
-			status = EXIT_FAILURE;
+			status = errno ? errno : 127;
 			bb_perror_msg("Couldn't umount %s", zapit);
 		} else {
 			/* De-allocate the loop device.  This ioctl should be ignored on
