diff -Nur busybox.orig/include/libbb.h busybox/include/libbb.h
--- busybox.orig/include/libbb.h	Thu Sep 15 10:20:35 2005
+++ busybox/include/libbb.h	Thu Sep 15 10:58:43 2005
@@ -431,7 +431,9 @@
 extern void setup_environment ( const char *shell, int loginshell, int changeenv, const struct passwd *pw );
 extern int correct_password ( const struct passwd *pw );
 extern char *pw_encrypt(const char *clear, const char *salt);
+#ifdef CONFIG_FEATURE_SHADOWPASSWDS
 extern struct spwd *pwd_to_spwd(const struct passwd *pw);
+#endif
 extern int obscure(const char *old, const char *newval, const struct passwd *pwdp);
 
 extern int bb_xopen(const char *pathname, int flags);
diff -Nur busybox.orig/libbb/pwd2spwd.c busybox/libbb/pwd2spwd.c
--- busybox.orig/libbb/pwd2spwd.c	Thu Sep 15 10:20:30 2005
+++ busybox/libbb/pwd2spwd.c	Thu Sep 15 10:58:32 2005
@@ -28,6 +28,8 @@
  * SUCH DAMAGE.
  */
 
+#ifdef CONFIG_FEATURE_SHADOWPASSWDS
+
 #include <time.h>
 #include <sys/types.h>
 #include "libbb.h"
@@ -72,3 +74,4 @@
 	return &sp;
 }
 
+#endif
