diff -Naur busybox-1.7.0.orig/include/usage.h busybox-1.7.0/include/usage.h
--- busybox-1.7.0.orig/include/usage.h	2007-08-25 00:02:19.000000000 +0200
+++ busybox-1.7.0/include/usage.h	2007-08-25 00:02:37.000000000 +0200
@@ -3876,7 +3876,8 @@
 #define wget_trivial_usage \
        "[-c|--continue] [-s|--spider] [-q|--quiet] [-O|--output-document file]\n" \
        "		[--header 'header: value'] [-Y|--proxy on/off] [-P DIR]\n" \
-       "		[-U|--user-agent agent] url"
+       "		[-U|--user-agent agent] url" \
+       "		[-T|--read-timeout value] [-t|--tries value] url"
 #define wget_full_usage \
        "Retrieve files via HTTP or FTP" \
        "\n\nOptions:\n" \
@@ -3886,7 +3887,9 @@
        "	-P	Set directory prefix to DIR\n" \
        "	-O	Save to filename ('-' for stdout)\n" \
        "	-U	Adjust 'User-Agent' field\n" \
-       "	-Y	Use proxy ('on' or 'off')"
+       "	-Y	Use proxy ('on' or 'off')" \
+       "	-T	delay before timeout and shutdown\n" \
+       "	-t	number of retry"
 
 #define which_trivial_usage \
        "[COMMAND ...]"
diff -Naur busybox-1.7.0.orig/networking/wget.c busybox-1.7.0/networking/wget.c
--- busybox-1.7.0.orig/networking/wget.c	2007-08-25 00:02:19.000000000 +0200
+++ busybox-1.7.0/networking/wget.c	2007-08-25 00:10:43.000000000 +0200
@@ -88,12 +88,30 @@
 }
 #endif
 
+static void timeout_reached(int ignore)
+{
+	if (content_len == 0) /* still haven't been able to talk with server */
+		exit(-1);
+}
+
+static void alarm_timeout(int iwait)
+{
+	struct itimerval itv;
+
+	itv.it_interval.tv_sec = 0;
+	itv.it_interval.tv_usec = 0;
+	itv.it_value.tv_sec = iwait;
+	itv.it_value.tv_usec = 0;
+	setitimer(ITIMER_REAL, &itv, NULL);
+}
+
 int wget_main(int argc, char **argv);
 int wget_main(int argc, char **argv)
 {
 	char buf[512];
 	struct host_info server, target;
 	len_and_sockaddr *lsa;
+	struct sigaction sa;
 	int n, status;
 	int port;
 	int try = 5;
@@ -114,6 +132,8 @@
 	bool use_proxy = 1;             /* Use proxies if env vars are set  */
 	const char *proxy_flag = "on";  /* Use proxies if env vars are set  */
 	const char *user_agent = "Wget";/* "User-Agent" header field        */
+	const char *timeout = "900";    /* Read Timeout duration (-T)       */
+	char *nr_tries = NULL;          /* Number of retries (-t)           */
 	static const char keywords[] ALIGN1 =
 		"content-length\0""transfer-encoding\0""chunked\0""location\0";
 	enum {
@@ -129,6 +149,8 @@
 		WGET_OPT_USER_AGENT = 0x40,
 		WGET_OPT_PASSIVE    = 0x80,
 		WGET_OPT_HEADER     = 0x100,
+		WGET_OPT_TIMEOUT    = 0x200,
+		WGET_OPT_RETRY      = 0x400,
 	};
 #if ENABLE_FEATURE_WGET_LONG_OPTIONS
 	static const char wget_longopts[] ALIGN1 =
@@ -139,6 +161,8 @@
 		"output-document\0"  Required_argument "O"
 		"directory-prefix\0" Required_argument "P"
 		"proxy\0"            Required_argument "Y"
+		"read-timeout\0"     Required_argument 'T'
+		"tries\0"            Required_argument 't'
 		"user-agent\0"       Required_argument "U"
 		"passive-ftp\0"      No_argument       "\xff"
 		"header\0"           Required_argument "\xfe"
@@ -147,9 +171,10 @@
 #endif
 	/* server.allocated = target.allocated = NULL; */
 	opt_complementary = "-1" USE_FEATURE_WGET_LONG_OPTIONS(":\xfe::");
-	opt = getopt32(argv, "csqO:P:Y:U:",
+	opt = getopt32(argv, "csqO:P:Y:T:t:U:",
 				&fname_out, &dir_prefix,
-				&proxy_flag, &user_agent
+				&proxy_flag, &user_agent,
+				&timeout, &nr_tries
 				USE_FEATURE_WGET_LONG_OPTIONS(, &headers_llist)
 				);
 	if (strcmp(proxy_flag, "off") == 0) {
@@ -173,6 +198,9 @@
 	}
 #endif
 
+	if (opt & WGET_OPT_RETRY)
+		try = atoi(nr_tries);
+
 	parse_url(argv[optind], &target);
 	server.host = target.host;
 	server.port = target.port;
@@ -240,6 +268,13 @@
 		/* We leak result of xmalloc_sockaddr2dotted */
 	}
 
+	/* starting timer */
+	sa.sa_handler = timeout_reached;
+	sigemptyset(&sa.sa_mask);
+	sa.sa_flags = SA_RESTART;
+	sigaction(SIGALRM, &sa, NULL);
+	alarm_timeout(atoi(timeout));
+
 	if (use_proxy || !target.is_ftp) {
 		/*
 		 *  HTTP session
