diff -Naur busybox-1.4.1.orig/networking/interface.c busybox-1.4.1/networking/interface.c
--- busybox-1.4.1.orig/networking/interface.c	2007-02-01 13:36:09.000000000 +0100
+++ busybox-1.4.1/networking/interface.c	2007-02-01 13:36:38.000000000 +0100
@@ -850,7 +850,7 @@
 	unsigned val;
 
 	sap->sa_family = ether_hwtype.type;
-	ptr = sap->sa_data;
+	ptr = (unsigned char *) sap->sa_data;
 
 	i = 0;
 	orig = bufp;
