diff -Naur omapfbplay-git20101201/Makefile omapfbplay-git20101201a/Makefile
--- omapfbplay-git20101201/Makefile	2010-12-14 16:09:13.000000000 +0100
+++ omapfbplay-git20101201a/Makefile	2010-12-15 10:00:37.912562800 +0100
@@ -50,6 +50,9 @@
 DRV  = magic-head.o $(DRV-y) magic-tail.o
 OBJ  = $(addprefix $(O),$(CORE) $(DRV))
 
+DESTDIR ?=
+PREFIX ?= /usr
+
 $(O)omapfbplay: $(OBJ)
 
 $(O)%.o: %.c
@@ -61,4 +64,8 @@
 clean:
 	rm -f $(O)*.o $(O)*.d $(O)omapfbplay
 
+install:
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	cp -P omapfbplay $(DESTDIR)$(PREFIX)/bin
+
 -include $(OBJ:.o=.d)
