diff -Naur a/setup.py b/setup.py
--- a/setup.py	2011-02-21 13:35:39.118273802 +0100
+++ b/setup.py	2011-02-21 13:40:16.074271081 +0100
@@ -58,20 +58,11 @@
 print 'checking for pygame', '...',
 sys.__stdout__.flush()
 
-try:
-    import pygame
+pygame = os.environ.get('PYTHONXCPREFIX') + '/include'
+if pygame:
     print 'ok'
-    print 'checking for pygame header files', '...',
-    inc = re.sub("/(lib|lib64)/", "/include/",
-                 pygame.__path__[0]).replace("site-packages/", "")
-    if not os.path.isdir(inc):
-        raise ImportError
-    print 'ok'
-    check_library('sdl', '1.2.5')
-    pygame = inc
-except ImportError:
+else:
     print 'not installed'
-    pygame = False
 
 # extention modules
 modules = []
