diff -Naur a/lib/gifalloc.c b/lib/gifalloc.c
--- a/lib/gifalloc.c	2015-07-13 06:05:46.000000000 +0200
+++ b/lib/gifalloc.c	2018-04-27 09:50:38.000000000 +0200
@@ -8,7 +8,7 @@
 #include <stdio.h>
 #include <string.h>
 
-#include "gif_lib.h"
+#include "gif_lib_private.h"
 
 #define MAX(x, y)    (((x) > (y)) ? (x) : (y))
 
diff -Naur a/lib/gif_lib.h b/lib/gif_lib.h
--- a/lib/gif_lib.h	2016-04-02 17:35:30.000000000 +0200
+++ b/lib/gif_lib.h	2018-04-27 09:49:24.000000000 +0200
@@ -244,9 +244,6 @@
                                      GifPixelType ColorTransIn2[]);
 extern int GifBitSize(int n);
 
-extern void *
-reallocarray(void *optr, size_t nmemb, size_t size);
-
 /******************************************************************************
  Support for the in-core structures allocation (slurp mode).              
 ******************************************************************************/
diff -Naur a/lib/gif_lib_private.h b/lib/gif_lib_private.h
--- a/lib/gif_lib_private.h	2014-05-16 12:46:53.000000000 +0200
+++ b/lib/gif_lib_private.h	2018-04-27 09:49:46.000000000 +0200
@@ -10,6 +10,9 @@
 #include "gif_lib.h"
 #include "gif_hash.h"
 
+extern void *
+reallocarray(void *optr, size_t nmemb, size_t size);
+
 #define EXTENSION_INTRODUCER      0x21
 #define DESCRIPTOR_INTRODUCER     0x2c
 #define TERMINATOR_INTRODUCER     0x3b
