--- vdr-channellist-0.0.4/scripts/channellists-update.sh.org	2012-04-11 18:52:36.204412901 +0200
+++ vdr-channellist-0.0.4/scripts/channellists-update.sh	2012-04-11 18:56:08.064413011 +0200
@@ -7,7 +7,7 @@
 RELOAD="off"
 [ "$1" = "reload" ] && RELOAD="on"
 
-CHANNELS_CONF_NAME="/var/lib/vdr/plugins/channellists/channels.conf.linowsat"
+CHANNELS_CONF_NAME="/etc/vdr/plugins/channellists/channels.conf.linowsat"
 
 [ -f /tmp/channels.conf ] && rm -f /tmp/channels.conf
 
@@ -16,7 +16,7 @@
 
 if [ ! -f /tmp/channels.conf ]; then
 #   echo "Update failed!"
-   svdrpsend.pl mesg "Update failed!"
+   svdrpsend mesg "Update failed!"
    exit 1
 fi
 
@@ -24,5 +24,5 @@
 [ "$RELOAD" = "on" ] && svdrpsend.pl plug channellists load "$CHANNELS_CONF_NAME"
 
 #echo "Update done!"
-svdrpsend.pl mesg "Update done!"
+svdrpsend mesg "Update done!"
 exit 0
