diff -Naur cryptopy-1.2.5/crypto/hash/md5Hash.py cryptopy-1.2.5a/crypto/hash/md5Hash.py
--- cryptopy-1.2.5/crypto/hash/md5Hash.py	2002-10-21 17:38:55.000000000 -0700
+++ cryptopy-1.2.5a/crypto/hash/md5Hash.py	2012-05-21 21:14:16.254897390 -0700
@@ -1,7 +1,7 @@
 """   md5Hash.py
       Wrapper for python md5 module to support crypo module standard interface
 """
-import md5
+import hashlib
 from crypto.hash.hash import Hash
 
 class MD5(Hash):
@@ -14,7 +14,7 @@
         self.reset()
 
     def reset(self):
-        self.pymd5 = md5.new()
+        self.pymd5 = hashlib.md5()
     def update(self,data):
         """ Update the md5 object with the string arg. Repeated calls are
             equivalent to a single call with the concatenation of all the
diff -Naur cryptopy-1.2.5/crypto/hash/sha1Hash.py cryptopy-1.2.5a/crypto/hash/sha1Hash.py
--- cryptopy-1.2.5/crypto/hash/sha1Hash.py	2003-02-21 21:36:37.000000000 -0800
+++ cryptopy-1.2.5a/crypto/hash/sha1Hash.py	2012-05-21 21:14:16.254897390 -0700
@@ -5,7 +5,7 @@
     Copyright © (c) 2002 by Paul A. Lambert
     Read LICENSE.txt for license information.
 """
-import sha
+import hashlib
 from crypto.hash.hash import Hash
 
 class SHA1(Hash):
@@ -18,7 +18,7 @@
         self.reset()
 
     def reset(self):
-        self.pysha1 = sha.new()
+        self.pysha1 = hashlib.sha1()
 
     def update(self,data):
         """ Update the sha object with the string arg. Repeated calls are
