diff -Naur eee-0.3.orig/asus_eee.c eee-0.3/asus_eee.c
--- eee-0.3.orig/asus_eee.c	2008-06-28 16:26:49.000000000 +0200
+++ eee-0.3/asus_eee.c	2008-06-28 16:27:27.000000000 +0200
@@ -473,7 +473,7 @@
     for (i = 0; eee_proc_files[i].name; i++) {
         remove_proc_entry(eee_proc_files[i].name, eee_proc_rootdir);
     }
-    remove_proc_entry("eee", &proc_root);
+    remove_proc_entry("eee", NULL);
 }
 
 #define WRITABLE(x) do { if(!writable) (x) &= 0x0577; } while(0)
@@ -482,7 +482,7 @@
     int i;
 
     /* Create the /proc/eee directory. */
-    eee_proc_rootdir = proc_mkdir("eee", &proc_root);
+    eee_proc_rootdir = proc_mkdir("eee", NULL);
     if (!eee_proc_rootdir) {
         printk(KERN_ERR "%s: Unable to create /proc/eee\n", EEE_NAME);
         return false;
