diff -uNr vdr-plugin-dvbapi-1.0.2.orig/device.cpp vdr-plugin-dvbapi-1.0.2/device.cpp
--- vdr-plugin-dvbapi-1.0.2.orig/device.cpp	2014-01-09 00:05:51.939985664 +0800
+++ vdr-plugin-dvbapi-1.0.2/device.cpp	2014-01-09 00:07:51.807988774 +0800
@@ -117,7 +117,11 @@
 
 void cScDevices::SetForceBudget(int n)
 {
+#if APIVERSNUM >= 20103
+  if (n >= 0 && n < MAXDEVICES)
+#else
   if (n >= 0 && n < MAXDVBDEVICES)
+#endif
     budget |= (1 << n);
 }
 
