diff -ruN vdr-1.7.38_remote/config.c vdr-1.7.38/config.c
--- vdr-1.7.38_remote/config.c	2013-02-25 02:24:32.000000000 +0100
+++ vdr-1.7.38/config.c	2013-02-25 02:28:15.000000000 +0100
@@ -394,6 +394,7 @@
   DiSEqC = 0;
   SetSystemTime = 0;
   TimeSource = 0;
+  ChannelBlocker = 0;
   TimeTransponder = 0;
   StandardCompliance = STANDARD_DVB;
   MarginStart = 2;
@@ -642,6 +643,7 @@
   else if (!strcasecmp(Name, "SetSystemTime"))       SetSystemTime      = atoi(Value);
   else if (!strcasecmp(Name, "TimeSource"))          TimeSource         = cSource::FromString(Value);
   else if (!strcasecmp(Name, "TimeTransponder"))     TimeTransponder    = atoi(Value);
+  else if (!strcasecmp(Name, "ChannelBlocker"))      ChannelBlocker     = atoi(Value);
   else if (!strcasecmp(Name, "StandardCompliance"))  StandardCompliance = atoi(Value);
   else if (!strcasecmp(Name, "MarginStart"))         MarginStart        = atoi(Value);
   else if (!strcasecmp(Name, "MarginStop"))          MarginStop         = atoi(Value);
@@ -811,6 +813,7 @@
   Store("SetSystemTime",      SetSystemTime);
   Store("TimeSource",         cSource::ToString(TimeSource));
   Store("TimeTransponder",    TimeTransponder);
+  Store("ChannelBlocker",     ChannelBlocker);
   Store("StandardCompliance", StandardCompliance);
   Store("MarginStart",        MarginStart);
   Store("MarginStop",         MarginStop);
diff -ruN vdr-1.7.38_remote/config.h vdr-1.7.38/config.h
--- vdr-1.7.38_remote/config.h	2013-02-25 02:24:32.000000000 +0100
+++ vdr-1.7.38/config.h	2013-02-25 02:28:15.000000000 +0100
@@ -300,6 +300,7 @@
   int SetSystemTime;
   int TimeSource;
   int TimeTransponder;
+  int ChannelBlocker;
   int StandardCompliance;
   int MarginStart, MarginStop;
   int AudioLanguages[I18N_MAX_LANGUAGES + 1];
diff -ruN vdr-1.7.38_remote/dvbdevice.c vdr-1.7.38/dvbdevice.c
--- vdr-1.7.38_remote/dvbdevice.c	2013-02-25 02:24:32.000000000 +0100
+++ vdr-1.7.38/dvbdevice.c	2013-02-25 02:28:15.000000000 +0100
@@ -1445,6 +1445,10 @@
 
 bool cDvbDevice::ProvidesSource(int Source) const
 {
+  if (Setup.ChannelBlocker == 1) {
+      if (IsPrimaryDevice()) return false;
+      //isyslog("ChannelBlocker aktive on primary Interface");
+  }
   int type = Source & cSource::st_Mask;
   return type == cSource::stNone
       || type == cSource::stAtsc  && ProvidesDeliverySystem(SYS_ATSC)
diff -ruN vdr-1.7.38_remote/menu.c vdr-1.7.38/menu.c
--- vdr-1.7.38_remote/menu.c	2013-02-25 02:24:32.000000000 +0100
+++ vdr-1.7.38/menu.c	2013-02-25 02:28:15.000000000 +0100
@@ -3310,6 +3310,8 @@
          }
      }
 
+  Add(new cMenuEditBoolItem(tr("Setup.DVB$Channel Blocker"), &data.ChannelBlocker));
+
   SetCurrent(Get(current));
   Display();
 }
diff -ruN vdr-1.7.38_remote/po/de_DE.po vdr-1.7.38/po/de_DE.po
--- vdr-1.7.38_remote/po/de_DE.po	2013-02-25 02:24:32.000000000 +0100
+++ vdr-1.7.38/po/de_DE.po	2013-02-25 02:28:15.000000000 +0100
@@ -943,6 +943,9 @@
 msgid "Setup.EPG$Disable running VPS event"
 msgstr "Erk. des lauf. VPS-Events abschalten"
 
+msgid "Setup.DVB$Channel Blocker"
+msgstr "Tuner des primären DVB blocken"
+
 msgid "pan&scan"
 msgstr "pan&scan"
 
