diff -ruN vdr-1.7.38_block/eitscan.c vdr-1.7.38/eitscan.c
--- vdr-1.7.38_block/eitscan.c	2012-04-07 16:39:28.000000000 +0200
+++ vdr-1.7.38/eitscan.c	2013-02-25 02:30:57.000000000 +0100
@@ -145,7 +145,7 @@
                if (Device && Device->ProvidesEIT()) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (Device->Priority() < 0) {
diff -ruN vdr-1.7.38_block/pat.c vdr-1.7.38/pat.c
--- vdr-1.7.38_block/pat.c	2013-02-25 02:28:15.000000000 +0100
+++ vdr-1.7.38/pat.c	2013-02-25 02:33:29.000000000 +0100
@@ -578,6 +578,7 @@
               }
            Channel->SetTeletextSubtitlePages(TeletextSubtitlePages, NumTPages);
 #endif // USE_TTXTSUBS
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
