diff -ruN vdr-1.7.38_extern/config.c vdr-1.7.38/config.c
--- vdr-1.7.38_extern/config.c	2013-02-25 02:37:29.000000000 +0100
+++ vdr-1.7.38/config.c	2013-02-25 02:43:14.000000000 +0100
@@ -490,6 +490,8 @@
 #endif /* JUMPPLAY */
   CurrentChannel = -1;
   CurrentVolume = MAXVOLUME;
+  VolumeSteps = 51;
+  VolumeLinearize = 0;
   CurrentDolby = 0;
   InitialChannel = "";
   DeviceBondings = "";
@@ -740,6 +742,8 @@
   else if (!strcasecmp(Name, "CurrentVolume"))       CurrentVolume      = atoi(Value);
   else if (!strcasecmp(Name, "CurrentDolby"))        CurrentDolby       = atoi(Value);
   else if (!strcasecmp(Name, "InitialChannel"))      InitialChannel     = Value;
+  else if (!strcasecmp(Name, "VolumeSteps"))         VolumeSteps        = atoi(Value);
+  else if (!strcasecmp(Name, "VolumeLinearize"))     VolumeLinearize    = atoi(Value);
   else if (!strcasecmp(Name, "InitialVolume"))       InitialVolume      = atoi(Value);
 #ifdef USE_VOLCTRL
   else if (!strcasecmp(Name, "LRVolumeControl"))     LRVolumeControl    = atoi(Value);
@@ -910,6 +914,8 @@
   Store("CurrentVolume",      CurrentVolume);
   Store("CurrentDolby",       CurrentDolby);
   Store("InitialChannel",     InitialChannel);
+  Store("VolumeSteps",        VolumeSteps);
+  Store("VolumeLinearize",    VolumeLinearize);
   Store("InitialVolume",      InitialVolume);
 #ifdef USE_VOLCTRL
   Store("LRVolumeControl",    LRVolumeControl);
diff -ruN vdr-1.7.38_extern/config.h vdr-1.7.38/config.h
--- vdr-1.7.38_extern/config.h	2013-02-25 02:37:29.000000000 +0100
+++ vdr-1.7.38/config.h	2013-02-25 02:39:52.000000000 +0100
@@ -381,6 +381,8 @@
   int PauseLastMark;
 #endif /* JUMPPLAY */
   int CurrentVolume;
+  int VolumeSteps;
+  int VolumeLinearize;
   int CurrentDolby;
   int InitialVolume;
 #ifdef USE_VOLCTRL
diff -ruN vdr-1.7.38_extern/device.c vdr-1.7.38/device.c
--- vdr-1.7.38_extern/device.c	2013-02-25 02:37:29.000000000 +0100
+++ vdr-1.7.38/device.c	2013-02-25 02:39:52.000000000 +0100
@@ -11,6 +11,7 @@
 #include <errno.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
+#include <math.h>
 #include "audio.h"
 #include "channels.h"
 #include "i18n.h"
@@ -986,8 +987,9 @@
 void cDevice::SetVolume(int Volume, bool Absolute)
 {
   int OldVolume = volume;
-  volume = constrain(Absolute ? Volume : volume + Volume, 0, MAXVOLUME);
-  SetVolumeDevice(volume);
+  float VolumeDelta = (float)MAXVOLUME/Setup.VolumeSteps;
+  volume = min(max((int)(floor((Absolute ? Volume : volume + Volume)/VolumeDelta+0.5)*VolumeDelta), 0), MAXVOLUME);
+  SetVolumeDevice((int)(pow((float)volume/MAXVOLUME, log((Setup.VolumeLinearize+20)/40.0)/log(0.5))*MAXVOLUME));
   Absolute |= mute;
   cStatus::MsgSetVolume(Absolute ? volume : volume - OldVolume, Absolute);
   if (volume > 0) {
diff -ruN vdr-1.7.38_extern/device.h vdr-1.7.38/device.h
--- vdr-1.7.38_extern/device.h	2013-02-25 02:37:29.000000000 +0100
+++ vdr-1.7.38/device.h	2013-02-25 02:39:52.000000000 +0100
@@ -33,7 +33,7 @@
 #define MAXPIDHANDLES      64 // the maximum number of different PIDs per device
 #define MAXRECEIVERS       16 // the maximum number of receivers per device
 #define MAXVOLUME         255
-#define VOLUMEDELTA         5 // used to increase/decrease the volume
+#define VOLUMEDELTA       (MAXVOLUME/Setup.VolumeSteps) // used to increase/decrease the volume
 #define MAXOCCUPIEDTIMEOUT 99 // max. time (in seconds) a device may be occupied
 
 enum eSetChannelResult { scrOk, scrNotAvailable, scrNoTransfer, scrFailed };
diff -ruN vdr-1.7.38_extern/po/de_DE.po vdr-1.7.38/po/de_DE.po
--- vdr-1.7.38_extern/po/de_DE.po	2013-02-25 02:37:29.000000000 +0100
+++ vdr-1.7.38/po/de_DE.po	2013-02-25 02:39:52.000000000 +0100
@@ -1225,6 +1225,12 @@
 msgid "Setup.Miscellaneous$SVDRP timeout (s)"
 msgstr "SVDRP trennen bei Inaktivität (s)"
 
+msgid "Setup.Miscellaneous$Volume steps"
+msgstr "Anzahl Lautstärke Schritte"
+
+msgid "Setup.Miscellaneous$Volume linearize"
+msgstr "Lautstärke Kurve korrigieren"
+
 msgid "Setup.Miscellaneous$Zap timeout (s)"
 msgstr "Mindestzeit für Kanalhistorie (s)"
 
