--- 1/Makefile.org	2013-02-26 23:57:29.264790381 +0100
+++ 2/Makefile	2013-02-26 23:56:57.000000000 +0100
@@ -167,8 +167,8 @@
 	@echo "locdir=$(LOCDIR)" >> $@
 	@echo "plgcfg=$(PLGCFG)" >> $@
 	@echo "apiversion=$(APIVERSION)" >> $@
-	@echo "cflags=$(CFLAGS) $(CDEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
-	@echo "cxxflags=$(CXXFLAGS) $(CDEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
+	@echo "cflags=$(CFLAGS) $(DEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
+	@echo "cxxflags=$(CXXFLAGS) $(DEFINES) $(CINCLUDES) $(HDRDIR)" >> $@
 	@echo "" >> $@
 	@echo "Name: VDR" >> $@
 	@echo "Description: Video Disk Recorder" >> $@
