diff -Naur qt-everywhere-opensource-src-4.7.0-beta1.orig/configure qt-everywhere-opensource-src-4.7.0-beta1/configure
--- qt-everywhere-opensource-src-4.7.0-beta1.orig/configure	2010-06-26 00:12:42.504688041 +0200
+++ qt-everywhere-opensource-src-4.7.0-beta1/configure	2010-06-26 00:24:43.614688899 +0200
@@ -954,7 +954,7 @@
         shift
         VAL=$1
         ;;
-    -prefix|-docdir|-headerdir|-plugindir|-importdir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey|-sdk|-arch|-host-arch|-mysql_config)
+    -prefix|-docdir|-headerdir|-plugindir|-importdir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey|-sdk|-arch|-host-arch|-mysql_config|-crossarch)
         VAR=`echo $1 | sed "s,^-\(.*\),\1,"`
         shift
         VAL="$1"
@@ -1443,6 +1443,9 @@
     xplatform)
         XPLATFORM="$VAL"
         ;;
+    crossarch)
+        CROSSARCH="$VAL"
+        ;;
     debug-and-release)
         if [ "$VAL" = "yes" ] || [ "$VAL" = "no" ]; then
             CFG_DEBUG_RELEASE="$VAL"
@@ -2918,6 +2921,8 @@
     CFG_ARCH=$CFG_HOST_ARCH
 fi
 
+CFG_ARCH="$CROSSARCH"
+
 if [ -d "$relpath/src/corelib/arch/$CFG_ARCH" ]; then
     if [ "$OPT_VERBOSE" = "yes" ]; then
         echo "    '$CFG_ARCH' is supported"
