diff -Naur qt-everywhere-opensource-src-4.7.3.orig//configure qt-everywhere-opensource-src-4.7.3/configure
--- qt-everywhere-opensource-src-4.7.3.orig//configure	2011-05-16 18:35:19.787064000 +0200
+++ qt-everywhere-opensource-src-4.7.3/configure	2011-05-16 18:36:13.357064008 +0200
@@ -4865,7 +4865,7 @@
 
 # detect neon support
 if ( [ "$CFG_ARCH" = "arm" ] || [ "$CFG_ARCH" = "armv6" ] ) && [ "${CFG_NEON}" = "auto" ]; then
-    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon"; then
+    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon -mfloat-abi=softfp"; then
 	CFG_NEON=yes
     else
 	CFG_NEON=no
diff -Naur qt-everywhere-opensource-src-4.7.3.orig//src/gui/gui.pro qt-everywhere-opensource-src-4.7.3/src/gui/gui.pro
--- qt-everywhere-opensource-src-4.7.3.orig//src/gui/gui.pro	2011-05-16 18:34:50.557064000 +0200
+++ qt-everywhere-opensource-src-4.7.3/src/gui/gui.pro	2011-05-16 18:36:58.137064002 +0200
@@ -69,7 +69,7 @@
 
     DRAWHELPER_NEON_ASM_FILES = $$NEON_ASM
 
-    neon_compiler.commands = $$QMAKE_CXX -c -mfpu=neon
+    neon_compiler.commands = $$QMAKE_CXX -c -mfpu=neon -mfloat-abi=softfp
     neon_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
     neon_compiler.dependency_type = TYPE_C
     neon_compiler.output = ${QMAKE_VAR_OBJECTS_DIR}${QMAKE_FILE_BASE}$${first(QMAKE_EXT_OBJ)}
