diff -Naur qt-everywhere-opensource-src-4.7.0-rc1.orig/configure qt-everywhere-opensource-src-4.7.0-rc1/configure
--- qt-everywhere-opensource-src-4.7.0-rc1.orig/configure	2010-09-06 19:51:47.137528931 +0200
+++ qt-everywhere-opensource-src-4.7.0-rc1/configure	2010-09-06 19:52:34.427530007 +0200
@@ -4685,7 +4685,7 @@
 
 # detect neon support
 if ( [ "$CFG_ARCH" = "arm" ] || [ "$CFG_ARCH" = "armv6" ] ) && [ "${CFG_NEON}" = "auto" ]; then
-    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon"; then
+    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon -mfloat-abi=softfp"; then
 	CFG_NEON=yes
     else
 	CFG_NEON=no
diff -Naur qt-everywhere-opensource-src-4.7.0-rc1.orig/src/corelib/corelib.pro qt-everywhere-opensource-src-4.7.0-rc1/src/corelib/corelib.pro
--- qt-everywhere-opensource-src-4.7.0-rc1.orig/src/corelib/corelib.pro	2010-09-06 19:50:32.399206584 +0200
+++ qt-everywhere-opensource-src-4.7.0-rc1/src/corelib/corelib.pro	2010-09-06 19:53:16.590090136 +0200
@@ -71,6 +71,6 @@
 }
 neon {
     DEFINES += QT_HAVE_NEON
-    QMAKE_CXXFLAGS *= -mfpu=neon
+    QMAKE_CXXFLAGS *= -mfpu=neon -mfloat-abi=softfp
 }
 
diff -Naur qt-everywhere-opensource-src-4.7.0-rc1.orig/src/gui/painting/painting.pri qt-everywhere-opensource-src-4.7.0-rc1/src/gui/painting/painting.pri
--- qt-everywhere-opensource-src-4.7.0-rc1.orig/src/gui/painting/painting.pri	2010-09-06 19:50:31.937981894 +0200
+++ qt-everywhere-opensource-src-4.7.0-rc1/src/gui/painting/painting.pri	2010-09-06 19:53:38.347529597 +0200
@@ -387,7 +387,7 @@
     DEFINES += QT_HAVE_NEON
     HEADERS += painting/qdrawhelper_neon_p.h
     SOURCES += painting/qdrawhelper_neon.cpp
-    QMAKE_CXXFLAGS *= -mfpu=neon
+    QMAKE_CXXFLAGS *= -mfpu=neon -mfloat-abi=softfp
 
     DRAWHELPER_NEON_ASM_FILES = ../3rdparty/pixman/pixman-arm-neon-asm.S painting/qdrawhelper_neon_asm.S
 
