diff -Naur qt-everywhere-opensource-src-4.7.1.orig//configure qt-everywhere-opensource-src-4.7.1/configure
--- qt-everywhere-opensource-src-4.7.1.orig//configure	2010-11-09 19:45:31.741802002 +0100
+++ qt-everywhere-opensource-src-4.7.1/configure	2010-11-09 19:46:19.461802001 +0100
@@ -4843,7 +4843,7 @@
 
 # detect neon support
 if ( [ "$CFG_ARCH" = "arm" ] || [ "$CFG_ARCH" = "armv6" ] ) && [ "${CFG_NEON}" = "auto" ]; then
-    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon"; then
+    if "$unixtests/compile.test" "$XQMAKESPEC" "$QMAKE_CONFIG" $OPT_VERBOSE "$relpath" "$outpath" config.tests/unix/neon "neon" $L_FLAGS $I_FLAGS $l_FLAGS "-mfpu=neon -mfloat-abi=softfp"; then
 	CFG_NEON=yes
     else
 	CFG_NEON=no
diff -Naur qt-everywhere-opensource-src-4.7.1.orig//src/corelib/corelib.pro qt-everywhere-opensource-src-4.7.1/src/corelib/corelib.pro
--- qt-everywhere-opensource-src-4.7.1.orig//src/corelib/corelib.pro	2010-11-09 19:44:14.781801999 +0100
+++ qt-everywhere-opensource-src-4.7.1/src/corelib/corelib.pro	2010-11-09 20:05:10.741801999 +0100
@@ -55,7 +55,7 @@
 
 neon {
     DEFINES += QT_HAVE_NEON
-    QMAKE_CXXFLAGS *= -mfpu=neon
+    QMAKE_CXXFLAGS *= -mfpu=neon -mfloat-abi=softfp
 }
 
 
diff -Naur qt-everywhere-opensource-src-4.7.1.orig//src/gui/gui.pro qt-everywhere-opensource-src-4.7.1/src/gui/gui.pro
--- qt-everywhere-opensource-src-4.7.1.orig//src/gui/gui.pro	2010-11-09 19:44:12.861801999 +0100
+++ qt-everywhere-opensource-src-4.7.1/src/gui/gui.pro	2010-11-09 20:19:09.791801998 +0100
@@ -79,7 +79,7 @@
 
 neon:*-g++* {
     DEFINES += QT_HAVE_NEON
-    QMAKE_CXXFLAGS *= -mfpu=neon
+    QMAKE_CXXFLAGS *= -mfpu=neon -mfloat-abi=softfp
     HEADERS += $$NEON_HEADERS
     SOURCES += $$NEON_SOURCES
 
