diff -Naur qt-4.7.1.orig//src/openvg/qpaintengine_vg.cpp qt-4.7.1/src/openvg/qpaintengine_vg.cpp
--- qt-4.7.1.orig//src/openvg/qpaintengine_vg.cpp	2010-11-11 17:26:25.311802000 +0100
+++ qt-4.7.1/src/openvg/qpaintengine_vg.cpp	2010-11-11 17:28:36.281801999 +0100
@@ -70,11 +70,11 @@
 #define QVG_NO_RENDER_TO_MASK 1
 #endif
 
-#if !defined(QVG_NO_DRAW_GLYPHS)
-
 // use the same rounding as in qrasterizer.cpp (6 bit fixed point)
 static const qreal aliasedCoordinateDelta = 0.5 - 0.015625;
 
+#if !defined(QVG_NO_DRAW_GLYPHS)
+
 Q_DECL_IMPORT extern int qt_defaultDpiX();
 Q_DECL_IMPORT extern int qt_defaultDpiY();
 
