diff -Naur dbus-1.5.0.orig//dbus/dbus-message.c dbus-1.5.0/dbus/dbus-message.c
--- dbus-1.5.0.orig//dbus/dbus-message.c	2011-08-06 12:31:50.624248071 +0200
+++ dbus-1.5.0/dbus/dbus-message.c	2011-08-06 12:32:49.264248103 +0200
@@ -3955,7 +3955,7 @@
   DBusValidationMode mode;
   dbus_uint32_t n_unix_fds = 0;
 
-  mode = DBUS_VALIDATION_MODE_DATA_IS_UNTRUSTED;
+  mode = DBUS_VALIDATION_MODE_WE_TRUST_THIS_DATA_ABSOLUTELY;
   
   oom = FALSE;
 
