From 2217548bcdc7b6a2beb55c1a8820ba184827f32b Mon Sep 17 00:00:00 2001
From: Alexander Huemer <a.huemer@bct-electronic.com>
Date: Fri, 20 Jul 2012 10:45:42 +0200
Subject: [PATCH] media-ctl: do not treat automake warnings as errors

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 6903451..8b669b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE([-Wall foreign])
 
 # Checks for programs.
 AC_PROG_CC
-- 
1.7.4.1

