diff -Naur vdr-dxr3-0.2.8/Makefile vdr-dxr3-0.2.8a/Makefile
--- vdr-dxr3-0.2.8/Makefile	2008-03-06 19:31:27.000000000 +0100
+++ vdr-dxr3-0.2.8a/Makefile	2008-08-04 11:01:48.000000000 +0200
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      = g++
-CXXFLAGS = -O2 -Wall -Woverloaded-virtual
+CXXFLAGS = -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
@@ -41,15 +41,15 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(FFMDIR) -I$(FFMDIR)/libavcodec -I$(EM8300)
-LIBS     = -L$(FFMDIR)/libavcodec -lavcodec
+INCLUDES += -I$(VDRDIR)/include -I$(FFMDIR)/include -I$(FFMDIR)/include/libavcodec -I$(EM8300)
+LIBS     = -L$(FFMDIR)/lib -lavcodec
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 DEFINES += -D_GNU_SOURCE
 
 # where is the microcode for the dxr3 located?  em8300 driver version 0.15.2
 # and later installs it by default to /lib/firmware/em8300.bin, and earlier
 # versions to /usr/share/misc/em8300.uc.
-DEFINES += -DMICROCODE=\"/lib/firmware/em8300.bin\"
+DEFINES += -DMICROCODE=\"/firmware/em8300.bin\"
 
 # use OSD scaler borrowed from the Xine plugin?
 # comment this out to use the traditional vdr-dxr3 scaler
