diff -Naur vdr-dxr3-0.2.8/Makefile vdr-dxr3-0.2.8-ok/Makefile
--- vdr-dxr3-0.2.8/Makefile	2008-03-06 19:31:27.000000000 +0100
+++ vdr-dxr3-0.2.8-ok/Makefile	2008-07-27 14:39:36.000000000 +0200
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      = g++
-CXXFLAGS = -O2 -Wall -Woverloaded-virtual
+CXXFLAGS = -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
@@ -41,8 +41,8 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(FFMDIR) -I$(FFMDIR)/libavcodec -I$(EM8300)
-LIBS     = -L$(FFMDIR)/libavcodec -lavcodec
+INCLUDES += -I$(VDRDIR)/include -I$(FFMDIR)/include -I$(FFMDIR)/include/libavcodec -I$(EM8300)
+LIBS     = -L$(FFMDIR)/lib -lavcodec
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 DEFINES += -D_GNU_SOURCE
 
