diff -Naur vdr-dxr3-0.2.9/dxr3interface.c vdr-dxr3-0.2.9a/dxr3interface.c
--- vdr-dxr3-0.2.9/dxr3interface.c	2008-12-29 22:17:39.000000000 +0100
+++ vdr-dxr3-0.2.9a/dxr3interface.c	2009-01-04 10:16:25.000000000 +0100
@@ -1046,6 +1046,17 @@
 	exit(1);
     }
 
+    // set brightness/contrast/saturation
+    m_bcs.brightness = cDxr3ConfigData::Instance().GetBrightness();
+    m_bcs.contrast = cDxr3ConfigData::Instance().GetContrast();
+    m_bcs.saturation = cDxr3ConfigData::Instance().GetSaturation();
+    dsyslog("dxr3: configure: brightness=%d,contrast=%d,saturation=%d",
+        m_bcs.brightness, m_bcs.contrast, m_bcs.saturation);
+    if (ioctl(m_fdControl, EM8300_IOCTL_SETBCS, &m_bcs) < 0)
+    {
+        esyslog("dxr3: unable to set brightness/contrast/saturation: %m");
+    }
+
     ConfigureDeviceAudio();
 }
 
