diff -Naur ImageMagick-6.7.5-3//Magick++/lib/Image.cpp ImageMagick-patched-6.7.5-3//Magick++/lib/Image.cpp
--- ImageMagick-6.7.5-3//Magick++/lib/Image.cpp	2012-02-01 03:12:28.000000000 +0100
+++ ImageMagick-patched-6.7.5-3//Magick++/lib/Image.cpp	2012-02-10 15:00:12.809476677 +0100
@@ -623,10 +623,10 @@
     kernel_info=AcquireKernelInfo("1");
   kernel_info->width=order_;
   kernel_info->height=order_;
-  kernel_info->values=(MagickRealType *) color_matrix_;
+  kernel_info->values=(double*)color_matrix_;
   MagickCore::Image* newImage =
     ColorMatrixImage( image(), kernel_info, &exceptionInfo );
-  kernel_info->values=(MagickRealType *) NULL;
+  kernel_info->values=NULL;
   kernel_info=DestroyKernelInfo(kernel_info);
   replaceImage( newImage );
   throwException( exceptionInfo );
