diff -ruN vdr-1.7.31/menu.c vdr-1.7.31-lbmod//menu.c
--- vdr-1.7.31/menu.c	2012-10-02 21:59:53.000000000 +0200
+++ vdr-1.7.31-lbmod//menu.c	2012-10-02 21:58:56.000000000 +0200
@@ -5980,7 +5980,11 @@
 void cReplayControl::TimeSearch(void)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+	if (cRecordControls::GetLiveIndex(fileName)) {
+       if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+          return;
+       }
+    }
 #endif /*USE_LIVEBUFFER*/
   timeSearchTime = timeSearchPos = 0;
   timeSearchHide = false;
@@ -6001,7 +6005,11 @@
 void cReplayControl::MarkToggle(void)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+  if (cRecordControls::GetLiveIndex(fileName)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) { 
+     	return;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
   int Current, Total;
   if (GetIndex(Current, Total, true)) {
@@ -6032,7 +6040,11 @@
 void cReplayControl::MarkJump(bool Forward)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+  if (cRecordControls::GetLiveIndex(fileName)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+        return;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
   if (marks.Count()) {
      int Current, Total;
@@ -6063,7 +6075,11 @@
 void cReplayControl::MarkMove(bool Forward)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+  if (cRecordControls::GetLiveIndex(fileName)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+        return;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
   int Current, Total;
   if (GetIndex(Current, Total)) {
@@ -6090,7 +6106,11 @@
 void cReplayControl::EditCut(void)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+  if (cRecordControls::GetLiveIndex(fileName)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+        return;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
   if (*fileName) {
      Hide();
@@ -6119,7 +6139,11 @@
 void cReplayControl::EditTest(void)
 {
 #ifdef USE_LIVEBUFFER
-  if(cRecordControls::GetLiveIndex(fileName)) return;
+  if (cRecordControls::GetLiveIndex(fileName)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+   	    return;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
   int Current, Total;
   if (GetIndex(Current, Total)) {
@@ -6188,8 +6212,11 @@
      }
   bool DoShowMode = true;
 #ifdef USE_LIVEBUFFER
-  if (cRecordControls::GetLiveIndex(fileName) && (Key >= k0) && (Key <= k9))
-      return osSwitchChannel;
+  if (cRecordControls::GetLiveIndex(fileName) && (Key >= k0) && (Key <= k9)) {
+     if (Interface->Confirm(tr("Interrompere il Timeshift?"))) {
+        return osSwitchChannel;
+     }
+  }
 #endif /*USE_LIVEBUFFER*/
 #ifdef USE_VOLCTRL
   if (Setup.LRVolumeControl &&
@@ -6212,20 +6239,10 @@
   switch (int(Key)) {
     // Positioning:
 #ifdef USE_LIVEBUFFER
-    case kUp:      if(cRecordControls::GetLiveIndex(fileName)) {
-                      cDevice::SwitchChannel(1);
-                      return osEnd;
-                   } // if
-                   // NO break
-    case kPlay:
-        Play(); break;
-    case kDown:    if(cRecordControls::GetLiveIndex(fileName)) {
-                      cDevice::SwitchChannel(-1);
-                      return osEnd;
-                   } // if
-                   // NO break
-    case kPause:   Pause();
-      break;
+    case kUp:      Play(); break;
+    case kPlay:    Play(); break;
+    case kDown:    Pause(); break;
+    case kPause:   Pause(); break;
 #else
     case kPlay:
     case kUp:      Play(); break;
