--- vdr-scraper2vdr-0.1.0/lib/Makefile.org	2014-10-26 16:12:41.083363925 +0100
+++ vdr-scraper2vdr-0.1.0/lib/Makefile	2014-10-26 16:12:50.147364081 +0100
@@ -1,7 +1,7 @@
 
 
 all: 
-	g++ -ggdb -DPLGDIR='"."' test.c common.c config.c db.c tabledef.c -lrt -lz -lmysqlclient -o t
+	$(CXX) -ggdb -DPLGDIR='"."' test.c common.c config.c db.c tabledef.c -lrt -lz -lmysqlclient -o t
 
 clean:
 	rm -f *.o *.a *~ core
