diff -Naur omapfbplay-git20101201/omapfbplay.c omapfbplay-git20101201_patched//omapfbplay.c
--- omapfbplay-git20101201/omapfbplay.c	2011-11-04 17:07:27.038454632 +0100
+++ omapfbplay-git20101201_patched//omapfbplay.c	2011-11-04 17:08:14.388454662 +0100
@@ -74,7 +74,7 @@
     int i;
 
     for (i = 0; i < afc->nb_streams; i++) {
-        if (afc->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO && !st)
+        if (afc->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO && !st)
             st = afc->streams[i];
         else
             afc->streams[i]->discard = AVDISCARD_ALL;
