diff -Naur libva-1.0.4/test/Makefile.in libva-1.0.4a/test/Makefile.in
--- libva-1.0.4/test/Makefile.in	2010-08-02 20:19:51.000000000 +0200
+++ libva-1.0.4a/test/Makefile.in	2010-09-15 09:27:01.000000000 +0200
@@ -269,7 +269,7 @@
 vainfo_LDADD = $(top_builddir)/va/$(libvacorelib) $(top_builddir)/va/$(libvabackendlib)
 vainfo_DEPENDENCIES = $(top_builddir)/va/$(libvacorelib) $(top_builddir)/va/$(libvabackendlib)
 vainfo_SOURCES = vainfo.c
-SUBDIRS = basic decode encode putsurface
+SUBDIRS = 
 all: all-recursive
 
 .SUFFIXES:
