--- a/configure.in	2013-08-17 10:22:31.692948390 +0200
+++ b/configure.in	2013-08-17 10:22:31.692948390 +0200
@@ -750,8 +750,9 @@
   fi
   USE_EXTERNAL_FFMPEG=1
 elif test "$target_platform" = "target_raspberry_pi"; then
-  ARCH="arm"
-  use_arch="arm"
+  CFLAGS="$CFLAGS -Wno-psabi -mno-apcs-stack-check"
+  CXXFLAGS="$CXXFLAGS -Wno-psabi -mno-apcs-stack-check"
+  FFMPEG_EXTRACFLAGS="$FFMPEG_EXTRACFLAGS"
 elif test "$use_arch" = "arm"; then
   CFLAGS="$CFLAGS -mno-apcs-stack-check"
   CXXFLAGS="$CXXFLAGS -mno-apcs-stack-check"
