diff --git a/lib/module.js b/lib/module.js
index 77a2a8d..2361b55 100644
--- a/lib/module.js
+++ b/lib/module.js
@@ -508,7 +508,7 @@ Module._initPaths = function() {
     var homeDir = process.env.HOME;
   }
 
-  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node')];
+  var paths = [path.resolve(process.execPath, '..', '..', 'lib', 'node_modules')];
 
   if (homeDir) {
     paths.unshift(path.resolve(homeDir, '.node_libraries'));

