diff -Naur ecore-1.0.0-alpha.orig/src/lib/ecore_con/ecore_con_info.c ecore-1.0.0-alpha/src/lib/ecore_con/ecore_con_info.c
--- ecore-1.0.0-alpha.orig/src/lib/ecore_con/ecore_con_info.c	2010-08-21 11:17:32.512281357 +0200
+++ ecore-1.0.0-alpha/src/lib/ecore_con/ecore_con_info.c	2010-08-21 11:18:12.626030527 +0200
@@ -283,7 +283,7 @@
         
         err = write(fd[1], "", 1);
         close(fd[1]);
-#ifdef __USE_ISOC99
+#if 0
         _Exit(0);
 #else
         _exit(0);
