diff -Naur ecore-r48546/src/lib/ecore_con/ecore_con_info.c ecore-r48546a/src/lib/ecore_con/ecore_con_info.c
--- ecore-r48546/src/lib/ecore_con/ecore_con_info.c	2010-04-05 10:46:24.000000000 +0200
+++ ecore-r48546a/src/lib/ecore_con/ecore_con_info.c	2010-05-02 16:17:23.909919310 +0200
@@ -274,7 +274,7 @@
 	  freeaddrinfo(result);
 	err = write(fd[1], "", 1);
 	close(fd[1]);
-# ifdef __USE_ISOC99
+# if 0
 	_Exit(0);
 # else
 	_exit(0);
