diff -Nur em8300-0.15.0.orig/em8300setup/Makefile.in em8300-0.15.0/em8300setup/Makefile.in
--- em8300-0.15.0.orig/em8300setup/Makefile.in	Wed Jun  1 02:21:14 2005
+++ em8300-0.15.0/em8300setup/Makefile.in	Mon Jul 18 22:38:12 2005
@@ -174,7 +174,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 AM_CFLAGS = @GLOBAL_CFLAGS@
-INCLUDES = -I@srcdir@/include -I/usr/local/include
+INCLUDES = -I@srcdir@/include
 man_MANS = em8300setup.1
 noinst_HEADERS = 
 em8300setup_SOURCES = em8300setup.c
