diff -Naur em8300-0.16.3.orig/modules/em8300_ioctl32.c em8300-0.16.3/modules/em8300_ioctl32.c
--- em8300-0.16.3.orig/modules/em8300_ioctl32.c	2007-10-06 20:30:21.000000000 +0200
+++ em8300-0.16.3/modules/em8300_ioctl32.c	2007-10-06 20:30:45.000000000 +0200
@@ -32,6 +32,8 @@
 #include <linux/pci.h>
 #include <asm/io.h>
 #include <asm/uaccess.h>
+#include <linux/ioctl.h>
+#include <linux/fs.h>
 
 #include <linux/i2c.h>
 #include <linux/i2c-algo-bit.h>
diff -Naur em8300-0.16.3.orig/modules/em8300_ioctl.c em8300-0.16.3/modules/em8300_ioctl.c
--- em8300-0.16.3.orig/modules/em8300_ioctl.c	2007-10-06 20:30:21.000000000 +0200
+++ em8300-0.16.3/modules/em8300_ioctl.c	2007-10-06 20:30:41.000000000 +0200
@@ -17,6 +17,7 @@
 #include <asm/io.h>
 #include <asm/uaccess.h>
 #include <linux/wait.h>
+#include <linux/ioctl.h>
 
 #include <linux/i2c.h>
 #include <linux/i2c-algo-bit.h>
