diff -Nur em8300-0.15.1.orig/modules/adv717x.c em8300-0.15.1/modules/adv717x.c
--- em8300-0.15.1.orig/modules/adv717x.c	Sat Apr  9 23:23:54 2005
+++ em8300-0.15.1/modules/adv717x.c	Fri Oct 28 12:18:19 2005
@@ -85,12 +85,6 @@
 MODULE_PARM(color_bars, "1-" __MODULE_STRING(EM8300_MAX) "i");
 MODULE_PARM_DESC(color_bars, "If you set this to 1 a set of color bars will be displayed on your screen (used for testing if the chip is working). Defaults to 0.");
 
-#define i2c_is_isa_client(clientptr) \
-		((clientptr)->adapter->algo->id == I2C_ALGO_ISA)
-#define i2c_is_isa_adapter(adapptr) \
-		((adapptr)->algo->id == I2C_ALGO_ISA)
-
-
 #define ADV7175_REG_MR0 0
 #define ADV7175_REG_MR1 1
 #define ADV7175_REG_TTXRQ_CTRL 0x24
@@ -454,10 +448,6 @@
 	struct i2c_client *new_client;
 	int mr0, mr1;
 	int err;
-
-	if (i2c_is_isa_adapter(adapter)) {
-		return 0;
-	}
 
 	if (!i2c_check_functionality(adapter, I2C_FUNC_SMBUS_READ_BYTE | I2C_FUNC_SMBUS_WRITE_BYTE_DATA)) {
 		return 0;
diff -Nur em8300-0.15.1.orig/modules/bt865.c em8300-0.15.1/modules/bt865.c
--- em8300-0.15.1.orig/modules/bt865.c	Sat Apr  9 23:23:54 2005
+++ em8300-0.15.1/modules/bt865.c	Fri Oct 28 12:18:51 2005
@@ -65,12 +65,6 @@
 MODULE_PARM(rgb_mode, "1-" __MODULE_STRING(EM8300_MAX) "i");
 MODULE_PARM_DESC(rgb_mode, "If you set this to 1, RGB output is enabled. You will need to hack the DXR3 hardware. Defaults to 0.");
 
-#define i2c_is_isa_client(clientptr) \
-		((clientptr)->adapter->algo->id == I2C_ALGO_ISA)
-#define i2c_is_isa_adapter(adapptr) \
-		((adapptr)->algo->id == I2C_ALGO_ISA)
-
-
 static int bt865_attach_adapter(struct i2c_adapter *adapter);
 int bt865_detach_client(struct i2c_client *client);
 int bt865_command(struct i2c_client *client, unsigned int cmd, void *arg);
@@ -870,10 +864,6 @@
 	struct bt865_data_s *data;
 	struct i2c_client *new_client;
 	int err, chk = 0;
-
-	if (i2c_is_isa_adapter(adapter)) {
-		return 0;
-	}
 
 	chk = i2c_check_functionality(adapter,
 			I2C_FUNC_SMBUS_READ_BYTE | I2C_FUNC_SMBUS_WRITE_BYTE_DATA);
