diff -Naur em8300-0.16.3/modules/em8300_main.c em8300-0.16.3.edit/modules/em8300_main.c
--- em8300-0.16.3/modules/em8300_main.c	2007-12-14 00:19:24.000000000 +0100
+++ em8300-0.16.3.edit/modules/em8300_main.c	2007-12-14 00:19:18.000000000 +0100
@@ -686,7 +686,7 @@
 	init_waitqueue_head(&em->vbi_wait);
 	init_waitqueue_head(&em->sp_ptsfifo_wait);
 
-	result = request_irq(dev->irq, em8300_irq, SA_SHIRQ | SA_INTERRUPT, "em8300", (void *) em);
+	result = request_irq(dev->irq, em8300_irq, IRQF_SHARED | IRQF_DISABLED, "em8300", (void *) em);
 
 	if (result == -EINVAL) {
 		printk(KERN_ERR "em8300: Bad irq number or handler\n");
