diff -Naur em8300-0.16.4.orig/modules/em8300_alsa.c em8300-0.16.4/modules/em8300_alsa.c
--- em8300-0.16.4.orig/modules/em8300_alsa.c	2007-03-16 23:02:31.000000000 +0100
+++ em8300-0.16.4/modules/em8300_alsa.c	2008-08-31 20:43:37.000000000 +0200
@@ -30,7 +30,7 @@
 #include <linux/pci.h>
 #include <linux/stringify.h>
 #include <linux/version.h>
-#include <asm/semaphore.h>
+#include <linux/semaphore.h>
 
 #include "em8300_reg.h"
 
diff -Naur em8300-0.16.4.orig/modules/em8300_fifo.h em8300-0.16.4/modules/em8300_fifo.h
--- em8300-0.16.4.orig/modules/em8300_fifo.h	2004-03-14 21:00:19.000000000 +0100
+++ em8300-0.16.4/modules/em8300_fifo.h	2008-08-31 20:43:49.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef EM8300_FIFO_H
 #define EM8300_FIFO_H
 
-#include <asm/semaphore.h>
+#include <linux/semaphore.h>
 
 #define FIFOTYPE_AUDIO 1
 #define FIFOTYPE_VIDEO 2
