diff --git a/arch/arm/boot/dts/imx6q-sbc-fx6m.dts b/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
index ddb550f..6874f89 100644
--- a/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
+++ b/arch/arm/boot/dts/imx6q-sbc-fx6m.dts
@@ -20,6 +20,10 @@
 / {
 	model = "CompuLab CM-FX6 on SBC-FX6m";
 	compatible = "compulab,cm-fx6", "compulab,sbc-fx6m", "fsl,imx6q";
+	
+	aliases {
+		mmc0 = &usdhc3;
+	};
 };
 
 &hdmi_core {
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
index 85836d7..fb389fb 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6.dtsi
@@ -72,6 +72,7 @@
 
 &usdhc3 {
 	wp-gpios = <&gpio7 0 0>;
+	cd-gpios = <&gpio7 1 0>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
index 5394364..dc6449d 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6m.dtsi
@@ -93,6 +93,7 @@
 };
 
 &usdhc3 {
+	non-removable;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi b/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
index 01f73ae..f98a87d 100644
--- a/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sb-fx6x.dtsi
@@ -103,8 +103,8 @@
 	pinctrl-0 = <&pinctrl_usdhc3>;
 	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
-	cd-gpios = <&gpio7 1 0>;
 	no-1-8-v;
+	keep-power-in-suspend;
 	vmmc-supply = <&reg_3p3v>;
 	status = "disabled";
 };
