diff --git a/sound/soc/imx/imx-hdmi-dma.c b/sound/soc/imx/imx-hdmi-dma.c
index 84b3503..7e5ebd0 100644
--- a/sound/soc/imx/imx-hdmi-dma.c
+++ b/sound/soc/imx/imx-hdmi-dma.c
@@ -750,11 +750,24 @@ static void hdmi_dma_set_thrsld_incrtype(int channels)
 	pr_debug("HDMI_AHB_DMA_THRSLD          0x%02x\n", hdmi_readb(HDMI_AHB_DMA_THRSLD));
 }
 
+static void hdmi_dma_set_hbr(int channels)
+{
+	u8 value = hdmi_readb(HDMI_AHB_DMA_CONF0) & (~HDMI_AHB_DMA_CONF0_HBR);
+
+	/* non audio AES and 8 channels means we have to enable HBR */
+	if ((channels == 8) &&
+	    (iec_header.B.linear_pcm == 1))
+		value |= HDMI_AHB_DMA_CONF0_HBR;
+
+	hdmi_writeb(value, HDMI_AHB_DMA_CONF0);
+}
+
 static void hdmi_dma_configure_dma(int channels)
 {
 	hdmi_dma_enable_hlock(1);
 	hdmi_dma_set_thrsld_incrtype(channels);
 	hdmi_dma_enable_channels(channels);
+	hdmi_dma_set_hbr(channels);
 }
 
 
