--- linux-linux-imx+3d350f4/arch/arm/plat-mxc/cpufreq.c	2013-10-23 15:02:26.000000000 +0200
+++ linux-linux-imx+3d350f4/arch/arm/plat-mxc/cpufreq.c-new	2013-10-28 21:23:39.729214143 +0100
@@ -81,7 +81,7 @@
 	}
 	if (gp_volt == 0)
 		return ret;
-	/*Set the voltage for the GP domain. */
+	/*Set the voltage for				printk(KERN_DEBUG the GP domain. */
 	if (freq > org_cpu_rate) {
 		/* Check if the bus freq needs to be increased first */
 		bus_freq_update(cpu_clk, true);
@@ -90,8 +90,6 @@
 			ret = regulator_set_voltage(soc_regulator, soc_volt,
 							soc_volt);
 			if (ret < 0) {
-				printk(KERN_DEBUG
-					"COULD NOT SET SOC VOLTAGE!!!!\n");
 				goto err1;
 			}
 		}
@@ -101,21 +99,17 @@
 			ret = regulator_set_voltage(pu_regulator, pu_volt,
 							pu_volt);
 			if (ret < 0) {
-				printk(KERN_DEBUG
-					"COULD NOT SET PU VOLTAGE!!!!\n");
 				goto err2;
 			}
 		}
 		ret = regulator_set_voltage(cpu_regulator, gp_volt,
 					    gp_volt);
 		if (ret < 0) {
-			printk(KERN_DEBUG "COULD NOT SET GP VOLTAGE!!!!\n");
 			goto err3;
 		}
 	}
 	ret = clk_set_rate(cpu_clk, freq);
 	if (ret != 0) {
-		printk(KERN_DEBUG "cannot set CPU clock rate\n");
 		goto err4;
 	}
 
@@ -123,15 +117,12 @@
 		ret = regulator_set_voltage(cpu_regulator, gp_volt,
 					    gp_volt);
 		if (ret < 0) {
-			printk(KERN_DEBUG "COULD NOT SET GP VOLTAGE!!!!\n");
 			goto err5;
 		}
 		if (!IS_ERR(soc_regulator)) {
 			ret = regulator_set_voltage(soc_regulator, soc_volt,
 							soc_volt);
 			if (ret < 0) {
-				printk(KERN_DEBUG
-					"COULD NOT SET SOC VOLTAGE BACK!!!!\n");
 				goto err6;
 			}
 		}
@@ -141,8 +132,6 @@
 			ret = regulator_set_voltage(pu_regulator, pu_volt,
 							pu_volt);
 			if (ret < 0) {
-				printk(KERN_DEBUG
-					"COULD NOT SET PU VOLTAGE!!!!\n");
 				goto err7;
 			}
 		}
@@ -156,21 +145,18 @@
 	ret = regulator_set_voltage(soc_regulator, org_soc_volt,
 							org_soc_volt);
 	if (ret < 0) {
-		printk(KERN_DEBUG "COULD NOT RESTORE SOC VOLTAGE!!!!\n");
 		goto err7;
 	}
 
 err6:
 	ret = regulator_set_voltage(cpu_regulator, org_gp_volt, org_gp_volt);
 	if (ret < 0) {
-		printk(KERN_DEBUG "COULD NOT RESTORE GP VOLTAGE!!!!\n");
 		goto err6;
 	}
 
 err5:
 	ret = clk_set_rate(cpu_clk, org_cpu_rate);
 	if (ret != 0) {
-		printk(KERN_DEBUG "cannot restore CPU clock rate\n");
 		goto err5;
 	}
 	return -1;
@@ -178,7 +164,6 @@
 err4:
 	ret = regulator_set_voltage(cpu_regulator, org_gp_volt, org_gp_volt);
 	if (ret < 0) {
-		printk(KERN_DEBUG "COULD NOT RESTORE GP VOLTAGE!!!!\n");
 		goto err4;
 	}
 
@@ -187,7 +172,6 @@
 		regulator_is_enabled(pu_regulator)) {
 		ret = regulator_set_voltage(pu_regulator, org_pu_volt, org_pu_volt);
 		if (ret < 0) {
-			printk(KERN_DEBUG "COULD NOT RESTORE PU VOLTAGE!!!!\n");
 			goto err3;
 		}
 	}
@@ -195,7 +179,6 @@
 	ret = regulator_set_voltage(soc_regulator, org_soc_volt,
 							org_soc_volt);
 	if (ret < 0) {
-		printk(KERN_DEBUG "COULD NOT RESTORE SOC VOLTAGE!!!!\n");
 		goto err2;
 	}
 err1:
