diff --git a/sound/soc/imx/imx-hdmi-dma.c b/sound/soc/imx/imx-hdmi-dma.c
index 7e5ebd0..db2d771 100644
--- a/sound/soc/imx/imx-hdmi-dma.c
+++ b/sound/soc/imx/imx-hdmi-dma.c
@@ -161,8 +161,8 @@ EXPORT_SYMBOL(iec_header);
  *    is necessary for 6 ch.
  */
 #define HDMI_DMA_PERIOD_BYTES		(6144)
-#define HDMI_DMA_BUF_SIZE		(64 * 1024)
-#define HDMI_PCM_BUF_SIZE		(64 * 1024)
+#define HDMI_DMA_BUF_SIZE		(16 * 64 * 1024)
+#define HDMI_PCM_BUF_SIZE		(16 * 64 * 1024)
 
 struct imx_hdmi_dma_runtime_data *hdmi_dma_priv;
 
@@ -1156,7 +1156,7 @@ static struct snd_pcm_hardware snd_imx_hardware = {
 	.period_bytes_min = HDMI_DMA_PERIOD_BYTES / 2,
 	.period_bytes_max = HDMI_DMA_PERIOD_BYTES / 2,
 	.periods_min = 8,
-	.periods_max = 8,
+	.periods_max = 16 * 8,
 	.fifo_size = 0,
 };
 
