diff --git a/drivers/rtc/rtc-em3027.c b/drivers/rtc/rtc-em3027.c
index 4f4930a..448a8c1 100644
--- a/drivers/rtc/rtc-em3027.c
+++ b/drivers/rtc/rtc-em3027.c
@@ -150,6 +150,7 @@ static struct i2c_driver em3027_driver = {
 	.driver = {
 		   .name = "rtc-em3027",
 		   .of_match_table = of_match_ptr(em3027_of_match),
+		   .owner	= THIS_MODULE,
 	},
 	.probe = &em3027_probe,
 	.id_table = em3027_id,
