--- at-3.1.10/Makefile.org	2012-05-03 19:14:38.053347107 +0200
+++ at-3.1.10/Makefile	2012-05-03 19:14:51.293347113 +0200
@@ -27,8 +27,8 @@
 LEXLIB		= 
 
 CC 		= /root/openbricks/build/build.i386.eglibc/toolchain/bin/i686-openbricks-linux-gnu-gcc
-CFLAGS 		= -I$(srcdir) -O2 -fexcess-precision=fast -march=i686 -mtune=generic -fgraphite-identity -ftree-loop-distribution -floop-interchange -floop-strip-mine -floop-block -Wall -pipe -g -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE  -I/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/local/include -I/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/include 
-LDFLAGS		=   -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/lib -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/lib -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/local/lib 
+CFLAGS 		= -I$(srcdir) -O2 -fexcess-precision=fast -march=i686 -mtune=generic -fgraphite-identity -ftree-loop-distribution -floop-interchange -floop-strip-mine -floop-block -Wall -pipe -g -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE  -I/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/local/include -I/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/include -lrt -lm 
+LDFLAGS		=   -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/lib -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/lib -L/root/openbricks/build/build.i386.eglibc/toolchain/i686-openbricks-linux-gnu/sysroot/usr/local/lib -lrt -lm 
 LFILE		= $(ATJOB_DIR)/.SEQ
 DEFS 		= -DHAVE_CONFIG_H -DVERSION=\"$(VERSION)\" \
 		-DETCDIR=\"$(etcdir)\" -DLOADAVG_MX=$(LOADAVG_MX) \
