diff --git a/ca_netlink.c b/ca_netlink.c
index c186e10..a1177b3 100644
--- a/ca_netlink.c
+++ b/ca_netlink.c
@@ -70,7 +70,7 @@ int reply_ca(struct sk_buff *skb_2, struct genl_info *info)
 	return 0;
 }
 
-int netlink_send_cw(unsigned char ca_num, ca_descr_t *ca_descr) {
+int netlink_send_cw(int ca_num, ca_descr_t *ca_descr) {
         struct sk_buff *skb;
 	void *msg_head;
 	int ret;
@@ -104,7 +104,7 @@ int netlink_send_cw(unsigned char ca_num, ca_descr_t *ca_descr) {
 	return 0;
 }
 
-int netlink_send_pid(unsigned char ca_num, ca_pid_t *ca_pid) {
+int netlink_send_pid(int ca_num, ca_pid_t *ca_pid) {
         struct sk_buff *skb;
 	void *msg_head;
 	int ret;
diff --git a/ca_netlink.h b/ca_netlink.h
index 709c665..95023b4 100644
--- a/ca_netlink.h
+++ b/ca_netlink.h
@@ -6,8 +6,8 @@
 #include <net/genetlink.h>
 
 int reply_ca(struct sk_buff *skb_2, struct genl_info *info);
-int netlink_send_cw(unsigned char ca_num, ca_descr_t *ca_descr);
-int netlink_send_pid(unsigned char ca_num, ca_pid_t *ca_pid);
+int netlink_send_cw(int ca_num, ca_descr_t *ca_descr);
+int netlink_send_pid(int ca_num, ca_pid_t *ca_pid);
 int register_netlink(void);
 void unregister_netlink(void);
 
