diff -uNr dvbsoftwareca-20131124.orig/ca_netlink.c dvbsoftwareca-20131124/ca_netlink.c
--- dvbsoftwareca-20131124.orig/ca_netlink.c	2013-11-24 17:06:18.576006528 +0800
+++ dvbsoftwareca-20131124/ca_netlink.c	2013-11-24 17:07:22.956004737 +0800
@@ -161,18 +161,18 @@
 		return ret;
 	}
 	// register functions (commands) of the new family
-	ret = genl_register_ops(&ca_family, &ask_ca_size_ops);
-	if (ret) {
-		genl_unregister_family(&ca_family);
-		printk("dvbsoftwareca: genl_register_family error\n");
-	}
+	//ret = genl_register_ops(&ca_family, &ask_ca_size_ops);
+	//if (ret) {
+	//	genl_unregister_family(&ca_family);
+	//	printk("dvbsoftwareca: genl_register_family error\n");
+	//}
 
 	return ret;
 }
 
 void unregister_netlink(void) {
         // unregister the functions
-	genl_unregister_ops(&ca_family, &ask_ca_size_ops);
+	//genl_unregister_ops(&ca_family, &ask_ca_size_ops);
         // unregister the family
 	genl_unregister_family(&ca_family);
 }

