--- a/board/solidrun/mx6_cubox-i/mx6_cubox-i_spl.c	2016-12-21 10:49:50.416079959 +0100
+++ b/board/solidrun/mx6_cubox-i/mx6_cubox-i_spl.c	2016-12-21 10:49:50.432087759 +0100
@@ -87,8 +87,8 @@
 
 static const char *build_dts_name(void)
 {
-	char *dt_prefix;
-	char *dt_suffix;
+	const char *dt_prefix, *dt_suffix;
+	static char dt_name[48];
 	int val1, val2, val3;
 
 	switch (spl_get_imx_type()){
@@ -118,11 +118,12 @@
 
 	/*
 	 * Machine selection -
-	 * Machine        val1, val2
-	 * -------------------------
-	 * HB rev 3.x     x     0
-	 * CBi            0     1
-	 * HB             1     1
+	 * Machine        val1, val2, val3
+	 * -------------------------------
+	 * HB2            x     x     0
+	 * HB rev 3.x     x     0     1
+	 * CBi            0     1     1
+	 * HB             1     1     1
 	 */
 
 	if (val3 == 0) {
@@ -135,7 +136,7 @@
 		dt_suffix = "-hummingboard.dtb";
 	}
 	
-	return strcat(dt_prefix, dt_suffix);
+	return strcat(strcpy(dt_name, dt_prefix), dt_suffix);
 }
 
 #include "asm/arch/mx6_ddr_regs.h"
