diff -Naur libomxil-bellagio-0.9.2.1.orig/configure libomxil-bellagio-0.9.2.1/configure
--- libomxil-bellagio-0.9.2.1.orig/configure	2010-06-17 22:00:20.567188165 +0200
+++ libomxil-bellagio-0.9.2.1/configure	2010-06-17 22:00:40.807188076 +0200
@@ -15066,8 +15066,6 @@
 esac
 
 
-$as_echo "#define malloc rpl_malloc" >>confdefs.h
-
 fi
 
 
@@ -15133,8 +15131,6 @@
 esac
 
 
-$as_echo "#define realloc rpl_realloc" >>confdefs.h
-
 fi
 
 
