diff -Naur libomxil-bellagio-0.9.2.1/configure libomxil-bellagio-0.9.2.1a/configure
--- libomxil-bellagio-0.9.2.1/configure	2010-08-01 16:43:28.713012297 +0200
+++ libomxil-bellagio-0.9.2.1a/configure	2010-08-01 16:45:03.480979842 +0200
@@ -3103,7 +3103,7 @@
 
 ac_config_headers="$ac_config_headers config.h"
 
-CFLAGS="${CFLAGS} -Wall -Werror"
+CFLAGS="${CFLAGS} -Wall"
 
 ################################################################################
 # Set the shared versioning info, according to section 6.3 of the libtool info #
