--- libbmm-0.2.0-0ubuntu1~ppa8/Makefile	2011-03-07 18:48:46.000000000 +0100
+++ libbmm-0.2.0-0ubuntu1~ppa8/Makefile-new	2012-07-22 17:15:46.903242599 +0200
@@ -2,11 +2,11 @@
 # Makefile for BMM user library
 #
 
-ifeq ($(MAKERULES),)
-include ../../../Rules.make
-else
+#ifeq ($(MAKERULES),)
+#include ../../../Rules.make
+#else
 include $(MAKERULES)
-endif
+#endif
 
 ifeq "$(CONFIG_GEN_DEBUG)" "y"
 BMM_DEF := -DBMM_DEBUG
