diff -Naur LVM2.2.02.54.orig/daemons/clvmd/Makefile.in LVM2.2.02.54/daemons/clvmd/Makefile.in
--- LVM2.2.02.54.orig/daemons/clvmd/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/daemons/clvmd/Makefile.in	2010-07-11 19:23:14.072658804 +0200
@@ -98,7 +98,7 @@
 .PHONY: install_clvmd
 
 install_clvmd: $(TARGETS)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) clvmd \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) clvmd \
 		$(usrsbindir)/clvmd
 
 install: $(INSTALL_TARGETS)
diff -Naur LVM2.2.02.54.orig/daemons/cmirrord/Makefile.in LVM2.2.02.54/daemons/cmirrord/Makefile.in
--- LVM2.2.02.54.orig/daemons/cmirrord/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/daemons/cmirrord/Makefile.in	2010-07-11 19:23:14.002659014 +0200
@@ -36,5 +36,5 @@
 		$(LVMLIBS) $(LMLIBS) $(LIBS)
 
 install: $(TARGETS)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) cmirrord \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) cmirrord \
 		$(usrsbindir)/cmirrord
diff -Naur LVM2.2.02.54.orig/daemons/dmeventd/Makefile.in LVM2.2.02.54/daemons/dmeventd/Makefile.in
--- LVM2.2.02.54.orig/daemons/dmeventd/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/daemons/dmeventd/Makefile.in	2010-07-11 19:23:14.152660073 +0200
@@ -81,23 +81,23 @@
 		$(usrlibdir)/pkgconfig/devmapper-event.pc
 
 install_lib_dynamic: libdevmapper-event.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/libdevmapper-event.$(LIB_SUFFIX).$(LIB_VERSION)
 	$(LN_S) -f libdevmapper-event.$(LIB_SUFFIX).$(LIB_VERSION) \
 		$(libdir)/libdevmapper-event.$(LIB_SUFFIX)
 
 install_lib_static: libdevmapper-event.a
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/libdevmapper-event.a.$(LIB_VERSION)
 	$(LN_S) -f libdevmapper-event.a.$(LIB_VERSION) $(libdir)/libdevmapper-event.a
 
 install_lib: $(INSTALL_LIB_TARGETS)
 
 install_dmeventd_dynamic: dmeventd
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< $(sbindir)/$<
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< $(sbindir)/$<
 
 install_dmeventd_static: dmeventd.static
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< $(sbindir)/$<
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< $(sbindir)/$<
 
 install_dmeventd: $(INSTALL_DMEVENTD_TARGETS)
 
diff -Naur LVM2.2.02.54.orig/daemons/dmeventd/plugins/mirror/Makefile.in LVM2.2.02.54/daemons/dmeventd/plugins/mirror/Makefile.in
--- LVM2.2.02.54.orig/daemons/dmeventd/plugins/mirror/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/daemons/dmeventd/plugins/mirror/Makefile.in	2010-07-11 19:23:14.172659081 +0200
@@ -35,7 +35,7 @@
 LIBS += -ldevmapper @LIB_PTHREAD@ @LVM2CMD_LIB@
 
 install_lvm2: libdevmapper-event-lvm2mirror.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/$<.$(LIB_VERSION)
 	$(LN_S) -f $<.$(LIB_VERSION) $(libdir)/$<
 
diff -Naur LVM2.2.02.54.orig/daemons/dmeventd/plugins/snapshot/Makefile.in LVM2.2.02.54/daemons/dmeventd/plugins/snapshot/Makefile.in
--- LVM2.2.02.54.orig/daemons/dmeventd/plugins/snapshot/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/daemons/dmeventd/plugins/snapshot/Makefile.in	2010-07-11 19:23:14.162658677 +0200
@@ -35,7 +35,7 @@
 LIBS += -ldevmapper @LIB_PTHREAD@ @LVM2CMD_LIB@
 
 install_lvm2: libdevmapper-event-lvm2snapshot.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/$<.$(LIB_VERSION)
 	$(LN_S) -f $<.$(LIB_VERSION) $(libdir)/$<
 
diff -Naur LVM2.2.02.54.orig/lib/format1/Makefile.in LVM2.2.02.54/lib/format1/Makefile.in
--- LVM2.2.02.54.orig/lib/format1/Makefile.in	2010-07-11 19:22:20.261409052 +0200
+++ LVM2.2.02.54/lib/format1/Makefile.in	2010-07-11 19:23:12.672658750 +0200
@@ -32,7 +32,7 @@
 include ../../make.tmpl
 
 install: liblvm2format1.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2format1.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2format1.so.$(LIB_VERSION) $(libdir)/liblvm2format1.so
 
diff -Naur LVM2.2.02.54.orig/lib/format_pool/Makefile.in LVM2.2.02.54/lib/format_pool/Makefile.in
--- LVM2.2.02.54.orig/lib/format_pool/Makefile.in	2010-07-11 19:22:20.271409152 +0200
+++ LVM2.2.02.54/lib/format_pool/Makefile.in	2010-07-11 19:23:12.962658919 +0200
@@ -29,7 +29,7 @@
 include ../../make.tmpl
 
 install: liblvm2formatpool.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2formatpool.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2formatpool.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2formatpool.so
diff -Naur LVM2.2.02.54.orig/lib/locking/Makefile.in LVM2.2.02.54/lib/locking/Makefile.in
--- LVM2.2.02.54.orig/lib/locking/Makefile.in	2010-07-11 19:22:20.271409152 +0200
+++ LVM2.2.02.54/lib/locking/Makefile.in	2010-07-11 19:23:13.012658714 +0200
@@ -25,7 +25,7 @@
 include ../../make.tmpl
 
 install install_cluster: liblvm2clusterlock.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2clusterlock.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2clusterlock.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2clusterlock.so
diff -Naur LVM2.2.02.54.orig/lib/mirror/Makefile.in LVM2.2.02.54/lib/mirror/Makefile.in
--- LVM2.2.02.54.orig/lib/mirror/Makefile.in	2010-07-11 19:22:20.261409052 +0200
+++ LVM2.2.02.54/lib/mirror/Makefile.in	2010-07-11 19:23:12.602658781 +0200
@@ -25,7 +25,7 @@
 include ../../make.tmpl
 
 install: liblvm2mirror.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2mirror.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2mirror.so.$(LIB_VERSION) $(libdir)/liblvm2mirror.so
 
diff -Naur LVM2.2.02.54.orig/lib/snapshot/Makefile.in LVM2.2.02.54/lib/snapshot/Makefile.in
--- LVM2.2.02.54.orig/lib/snapshot/Makefile.in	2010-07-11 19:22:20.261409052 +0200
+++ LVM2.2.02.54/lib/snapshot/Makefile.in	2010-07-11 19:23:12.482658780 +0200
@@ -25,7 +25,7 @@
 include ../../make.tmpl
 
 install: liblvm2snapshot.so
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2snapshot.so.$(LIB_VERSION)
 	$(LN_S) -f liblvm2snapshot.so.$(LIB_VERSION) \
 		$(libdir)/liblvm2snapshot.so
diff -Naur LVM2.2.02.54.orig/libdm/libdevmapper.h LVM2.2.02.54/libdm/libdevmapper.h
--- LVM2.2.02.54.orig/libdm/libdevmapper.h	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/libdm/libdevmapper.h	2010-07-11 19:23:13.632658847 +0200
@@ -618,7 +618,7 @@
 /* Returns number of set bits */
 static inline unsigned hweight32(uint32_t i)
 {
-	unsigned r = (i & 0x55555555) + ((i >> 1) & 0x55555555);
+	unsigned r = (i & 0x75577555) + ((i >> 1) & 0x75577555);
 
 	r =    (r & 0x33333333) + ((r >>  2) & 0x33333333);
 	r =    (r & 0x0F0F0F0F) + ((r >>  4) & 0x0F0F0F0F);
diff -Naur LVM2.2.02.54.orig/libdm/Makefile.in LVM2.2.02.54/libdm/Makefile.in
--- LVM2.2.02.54.orig/libdm/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/libdm/Makefile.in	2010-07-11 19:23:13.642658903 +0200
@@ -84,7 +84,7 @@
 	$(LN_S) -f libdevmapper.a.$(LIB_VERSION_DM) $(libdir)/libdevmapper.a
 
 install_ioctl: ioctl/libdevmapper.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/libdevmapper.$(LIB_SUFFIX).$(LIB_VERSION_DM)
 
 install_pkgconfig:
@@ -92,7 +92,7 @@
 		$(usrlibdir)/pkgconfig/devmapper.pc
 
 install_ioctl_static: ioctl/libdevmapper.a
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/libdevmapper.a.$(LIB_VERSION_DM)
 
 $(VERSIONED_SHLIB): %.$(LIB_SUFFIX).$(LIB_VERSION_DM): $(interface)/%.$(LIB_SUFFIX)
diff -Naur LVM2.2.02.54.orig/liblvm/Makefile.in LVM2.2.02.54/liblvm/Makefile.in
--- LVM2.2.02.54.orig/liblvm/Makefile.in	2010-07-11 19:22:20.271409152 +0200
+++ LVM2.2.02.54/liblvm/Makefile.in	2010-07-11 19:23:13.442660039 +0200
@@ -70,13 +70,13 @@
 	$(INSTALL) -D $(OWNER) $(GROUP) -m 444 $(srcdir)/lvm2app.h $(includedir)/lvm2app.h
 
 install_dynamic: $(LIB_SHARED)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/$(LIB_SHARED).$(LIB_VERSION_APP)
 	$(LN_S) -f $(LIB_SHARED).$(LIB_VERSION_APP) \
 		$(libdir)/$(LIB_SHARED)
 
 install_static: $(LIB_STATIC)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/$(LIB_STATIC).$(LIB_VERSION_APP)
 	$(LN_S) -f $(LIB_STATIC).$(LIB_VERSION_APP) $(libdir)/$(LIB_STATIC)
 
diff -Naur LVM2.2.02.54.orig/scripts/lvm2create_initrd/lvm2create_initrd LVM2.2.02.54/scripts/lvm2create_initrd/lvm2create_initrd
--- LVM2.2.02.54.orig/scripts/lvm2create_initrd/lvm2create_initrd	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/scripts/lvm2create_initrd/lvm2create_initrd	2010-07-11 19:23:13.902660008 +0200
@@ -207,7 +207,7 @@
 exec chroot . /bin/sh -c "umount /initrd; blockdev --flushbufs /dev/ram0 ; exec /sbin/init $*" < dev/console > dev/console 2>&1
 
 INIT
-   chmod 555 $TMPMNT/sbin/init
+   chmod 755 $TMPMNT/sbin/init
 }
 
 # create lvm.conf file from dumpconfig. Just use filter options
diff -Naur LVM2.2.02.54.orig/scripts/Makefile.in LVM2.2.02.54/scripts/Makefile.in
--- LVM2.2.02.54.orig/scripts/Makefile.in	2010-07-11 19:22:20.281409166 +0200
+++ LVM2.2.02.54/scripts/Makefile.in	2010-07-11 19:23:13.852658811 +0200
@@ -19,12 +19,12 @@
 include ../make.tmpl
 
 install:
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(srcdir)/lvm_dump.sh \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(srcdir)/lvm_dump.sh \
 		$(sbindir)/lvmdump
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(srcdir)/vgimportclone.sh \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(srcdir)/vgimportclone.sh \
 		$(sbindir)/vgimportclone
 ifeq ("@FSADM@", "yes")
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(srcdir)/fsadm.sh \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(srcdir)/fsadm.sh \
 		$(sbindir)/fsadm
 endif
 
diff -Naur LVM2.2.02.54.orig/tools/Makefile.in LVM2.2.02.54/tools/Makefile.in
--- LVM2.2.02.54.orig/tools/Makefile.in	2010-07-11 19:22:20.271409152 +0200
+++ LVM2.2.02.54/tools/Makefile.in	2010-07-11 19:23:13.222659036 +0200
@@ -165,21 +165,21 @@
 endif
 
 install_cmdlib_dynamic: liblvm2cmd.$(LIB_SUFFIX)
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2cmd.$(LIB_SUFFIX).$(LIB_VERSION)
 	$(LN_S) -f liblvm2cmd.$(LIB_SUFFIX).$(LIB_VERSION) $(libdir)/liblvm2cmd.$(LIB_SUFFIX)
 	$(INSTALL) -D $(OWNER) $(GROUP) -m 444 $(srcdir)/lvm2cmd.h \
 		$(includedir)/lvm2cmd.h
 
 install_cmdlib_static: liblvm2cmd-static.a
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(libdir)/liblvm2cmd.a.$(LIB_VERSION)
 	$(LN_S) -f liblvm2cmd.a.$(LIB_VERSION) $(libdir)/liblvm2cmd.a
 	$(INSTALL) -D $(OWNER) $(GROUP) -m 444 $(srcdir)/lvm2cmd.h \
 		$(includedir)/lvm2cmd.h
 
 install_tools_dynamic: lvm .commands
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) lvm \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) lvm \
 		$(sbindir)/lvm
 	@echo Creating symbolic links for individual commands in $(sbindir)
 	@( \
@@ -190,14 +190,14 @@
 	)
 
 install_tools_static: lvm.static
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< \
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< \
 		$(staticdir)/lvm.static
 
 install_dmsetup_dynamic: dmsetup
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< $(sbindir)/$<
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< $(sbindir)/$<
 
 install_dmsetup_static: dmsetup.static
-	$(INSTALL) -D $(OWNER) $(GROUP) -m 555 $(STRIP) $< $(sbindir)/$<
+	$(INSTALL) -D $(OWNER) $(GROUP) -m 755 $(STRIP) $< $(sbindir)/$<
 
 install_device-mapper: $(INSTALL_DMSETUP_TARGETS)
 
