--- LVM.orig/configure  2009-07-03 14:49:21.000000000 -0400
+++ LVM/configure	2009-07-03 14:51:07.000000000 -0400
@@ -2731,7 +2731,7 @@
 $as_echo_n "checking whether the C compiler works... " >&6; }
 # FIXME: These cross compiler hacks should be removed for Autoconf 3.0
 # If not cross compiling, check that we can run a simple program.
-if test "$cross_compiling" != yes; then
+if test "$cross_compiling" != dummy; then
   if { ac_try='./$ac_file'
   { (case "(($ac_try" in
   *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
@@ -4336,7 +4336,7 @@
 
 if test $ac_cv_header_stdc = yes; then
   # /bin/cc in Irix-4.0.5 gets non-ANSI ctype macros unless using -ansi.
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   :
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -6423,7 +6423,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -6497,7 +6497,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_working_alloca_h=yes
@@ -6581,7 +6581,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_func_alloca_works=yes
@@ -6718,7 +6718,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -6756,7 +6756,7 @@
 if test "${ac_cv_c_stack_direction+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_c_stack_direction=0
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -6838,7 +6838,7 @@
 if test "${ac_cv_func_closedir_void+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_closedir_void=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7127,7 +7127,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -7162,7 +7162,7 @@
 if test "${ac_cv_func_fork_works+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_fork_works=cross
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7246,7 +7246,7 @@
 if test "${ac_cv_func_vfork_works+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_vfork_works=cross
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7421,7 +7421,7 @@
   rm -f conftest.sym conftest.file
 echo >conftest.file
 if test "$as_ln_s" = "ln -s" && ln -s conftest.file conftest.sym; then
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_lstat_dereferences_slashed_symlink=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7511,7 +7511,7 @@
 if test "${ac_cv_func_lstat_empty_string_bug+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_lstat_empty_string_bug=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7734,7 +7734,7 @@
 if test "${ac_cv_func_malloc_0_nonnull+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_malloc_0_nonnull=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7827,7 +7827,7 @@
 if test "${ac_cv_func_memcmp_working+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_memcmp_working=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8134,7 +8134,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -8168,7 +8168,7 @@
 if test "${ac_cv_func_mmap_fixed_mapped+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_mmap_fixed_mapped=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8357,7 +8357,7 @@
 if test "${ac_cv_func_stat_empty_string_bug+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_stat_empty_string_bug=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8434,7 +8434,7 @@
 if test "${ac_cv_func_strtod+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_strtod=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8588,7 +8588,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_func_pow=yes
@@ -8654,7 +8654,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_m_pow=yes
@@ -8755,7 +8755,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -8849,7 +8849,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_func__doprnt=yes
@@ -9851,7 +9851,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_c_getline=yes
@@ -9927,7 +9927,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_c_canonicalize_file_name=yes
@@ -10014,7 +10014,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_search_tgetent=$ac_res
@@ -10120,7 +10120,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_dl_dlopen=yes
@@ -10227,7 +10227,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_sepol_sepol_check_context=yes
@@ -10308,7 +10308,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_selinux_is_selinux_enabled=yes
@@ -10404,7 +10404,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_search_pthread_mutex_lock=$ac_res
@@ -10492,7 +10492,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_rt_clock_gettime=yes
@@ -10733,7 +10733,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_lib_readline_readline=yes
@@ -10851,7 +10851,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -11390,7 +11390,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
@@ -11478,7 +11478,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   ac_cv_search_getmntent=$ac_res
@@ -11983,7 +11983,7 @@
 	 test -z "$ac_c_werror_flag" ||
 	 test ! -s conftest.err
        } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
+	 test "$cross_compiling" = dummy ||
 	 $as_test_x conftest$ac_exeext
        }; then
   eval "$as_ac_var=yes"
