diff -Naur LVM2.2.02.52.orig/configure LVM2.2.02.52/configure
--- LVM2.2.02.52.orig/configure	2009-09-15 00:57:46.000000000 +0200
+++ LVM2.2.02.52/configure	2009-09-15 22:47:01.000000000 +0200
@@ -4413,7 +4413,7 @@
 
 if test $ac_cv_header_stdc = yes; then
   # /bin/cc in Irix-4.0.5 gets non-ANSI ctype macros unless using -ansi.
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   :
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -6833,7 +6833,7 @@
 if test "${ac_cv_c_stack_direction+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_c_stack_direction=0
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -6915,7 +6915,7 @@
 if test "${ac_cv_func_closedir_void+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_closedir_void=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7239,7 +7239,7 @@
 if test "${ac_cv_func_fork_works+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_fork_works=cross
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7323,7 +7323,7 @@
 if test "${ac_cv_func_vfork_works+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_vfork_works=cross
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7498,7 +7498,7 @@
   rm -f conftest.sym conftest.file
 echo >conftest.file
 if test "$as_ln_s" = "ln -s" && ln -s conftest.file conftest.sym; then
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_lstat_dereferences_slashed_symlink=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7588,7 +7588,7 @@
 if test "${ac_cv_func_lstat_empty_string_bug+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_lstat_empty_string_bug=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7811,7 +7811,7 @@
 if test "${ac_cv_func_malloc_0_nonnull+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_malloc_0_nonnull=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -7904,7 +7904,7 @@
 if test "${ac_cv_func_memcmp_working+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_memcmp_working=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8245,7 +8245,7 @@
 if test "${ac_cv_func_mmap_fixed_mapped+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_mmap_fixed_mapped=no
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8434,7 +8434,7 @@
 if test "${ac_cv_func_stat_empty_string_bug+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_stat_empty_string_bug=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
@@ -8511,7 +8511,7 @@
 if test "${ac_cv_func_strtod+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   ac_cv_func_strtod=no
 else
   cat >conftest.$ac_ext <<_ACEOF
