--- a/artiste/CMakeLists.txt	2014-12-14 14:01:04.000000000 +0100
+++ b/artiste/CMakeLists.txt	2016-03-15 13:39:24.185865101 +0100
@@ -120,6 +120,7 @@
     vsxu_engine
     vsx_widget
     rt
+    Xrandr
   )
 
   if (VSXU_STATIC EQUAL 1)
--- a/player/CMakeLists.txt	2014-12-14 14:01:04.000000000 +0100
+++ b/player/CMakeLists.txt	2016-03-15 13:39:24.185865101 +0100
@@ -76,6 +76,7 @@
     vsxu_engine_graphics
     vsxu_engine
     rt
+    Xrandr
   )
 endif()
 
--- a/profiler/CMakeLists.txt	2016-03-15 13:45:22.892088003 +0100
+++ b/profiler/CMakeLists.txt	2016-03-15 13:45:22.916088008 +0100
@@ -88,6 +88,7 @@
     vsxu_engine_graphics
     vsx_widget
     rt
+    Xrandr
   )
 
 endif(UNIX)
--- a/server/CMakeLists.txt	2016-03-15 13:47:59.816084387 +0100
+++ b/server/CMakeLists.txt	2016-03-15 13:47:59.836084359 +0100
@@ -111,6 +111,7 @@
       ${PNG_LIBRARIES}
       ${JPEG_LIBRARIES}
       rt
+      Xrandr
     )
   endif (VSXU_OPENGL_ES_2_0 EQUAL 1)
 
--- a/tools/obj2vxm/CMakeLists.txt	2016-03-15 13:50:15.998547216 +0100
+++ b/tools/obj2vxm/CMakeLists.txt	2016-03-15 13:50:16.014555197 +0100
@@ -58,6 +58,7 @@
 if(UNIX)
   target_link_libraries(${module_id}
     rt
+    dl
   )
   install(TARGETS ${module_id} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 endif(UNIX)
