diff -ruN yaepghd/yaepghd.c vdr-plugin-yaepghd/yaepghd.c
--- yaepghd/yaepghd.c	2013-04-14 20:28:58.051220580 +0200
+++ vdr-plugin-yaepghd/yaepghd.c	2013-04-14 20:29:56.474282023 +0200
@@ -2565,7 +2565,7 @@
 
    /* Fill in initial values for start/end input */
    startInput.UpdateTime(event->StartTime() - (Setup.MarginStart * 60));
-   endInput.UpdateTime(event->StartTime() + _event->Duration() + (Setup.MarginStart * 60));
+   endInput.UpdateTime(event->StartTime() + _event->Duration() + (Setup.MarginStop * 60));
 
 }
 
