--- nvtv-0.4.7.orig/configure	Mon Dec  6 10:41:43 2004
+++ nvtv-0.4.7/configure	Mon Dec  6 10:42:00 2004
@@ -2778,28 +2778,6 @@
 
 
 
-CFLAGS=""
-
-echo "$as_me:$LINENO: checking for debugging info" >&5
-echo $ECHO_N "checking for debugging info... $ECHO_C" >&6
-
-# Check whether --enable-debug or --disable-debug was given.
-if test "${enable_debug+set}" = set; then
-  enableval="$enable_debug"
-
-else
-  enable_debug=yes
-fi;
-
-if test "$enable_debug" = yes; then
-  CFLAGS="$CFLAGS -g"
-  echo "$as_me:$LINENO: result: enabled" >&5
-echo "${ECHO_T}enabled" >&6
-else
-  CFLAGS="$CFLAGS -O2"
-  echo "$as_me:$LINENO: result: disabled (optimizing on)" >&5
-echo "${ECHO_T}disabled (optimizing on)" >&6
-fi
 
 
 ac_ext=c
