diff -Naur nvtv-0.4.7.orig/src/nvtv.c nvtv-0.4.7/src/nvtv.c
--- nvtv-0.4.7.orig/src/nvtv.c	2004-12-30 01:48:25.507461376 +0100
+++ nvtv-0.4.7/src/nvtv.c	2004-12-30 01:48:37.258674920 +0100
@@ -172,6 +172,10 @@
   {"ChromaBandwidth", min:   0, max:100, SET_FIELD(chroma_bandwidth)},
   {"Sharpness",       min:   0, max:100, SET_FIELD(sharpness)},
   {"CrossColor",      min:   0, max:100, SET_FIELD(cross_color)},
+  {"TVPositionX",     min:-100, max:100, SET_FIELD(tv_hoffset)},
+  {"TVPositionY",     min:-100, max:100, SET_FIELD(tv_voffset)},
+  {"MonitorPositionX",     min:-100, max:100, SET_FIELD(mon_hoffset)},
+  {"MonitorPositionY",     min:-100, max:100, SET_FIELD(mon_voffset)},
   {NULL}
 };
 
