* fix compilation on non-x86 arch

diff -Naur nvtv-0.4.7.orig/src/mmio.h nvtv-0.4.7/src/mmio.h
--- nvtv-0.4.7.orig/src/mmio.h	2005-03-23 01:19:42.000000000 +0100
+++ nvtv-0.4.7/src/mmio.h	2005-03-23 01:20:04.000000000 +0100
@@ -51,7 +51,7 @@
 /* from xfree common/compiler.h */
 
 #ifndef __NetBSD__
-#ifdef linux 
+#if defined(linux) && defined(__i386__)
 #include <sys/io.h>
 
 #define OUTB(p,v) outb(v,p)
