--- udev-146.orig/configure	2009-10-31 09:36:26.000000000 -0400
+++ udev-146/configure	2009-10-31 09:36:34.000000000 -0400
@@ -12354,223 +12354,6 @@
 fi
 
 
-
-	{ $as_echo "$as_me:$LINENO: checking for acl_init in -lacl" >&5
-$as_echo_n "checking for acl_init in -lacl... " >&6; }
-if test "${ac_cv_lib_acl_acl_init+set}" = set; then
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lacl  $LIBS"
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char acl_init ();
-int
-main ()
-{
-return acl_init ();
-  ;
-  return 0;
-}
-_ACEOF
-rm -f conftest.$ac_objext conftest$ac_exeext
-if { (ac_try="$ac_link"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval ac_try_echo="\"\$as_me:$LINENO: $ac_try_echo\""
-$as_echo "$ac_try_echo") >&5
-  (eval "$ac_link") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest$ac_exeext && {
-	 test "$cross_compiling" = yes ||
-	 $as_test_x conftest$ac_exeext
-       }; then
-  ac_cv_lib_acl_acl_init=yes
-else
-  $as_echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	ac_cv_lib_acl_acl_init=no
-fi
-
-rm -rf conftest.dSYM
-rm -f core conftest.err conftest.$ac_objext conftest_ipa8_conftest.oo \
-      conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:$LINENO: result: $ac_cv_lib_acl_acl_init" >&5
-$as_echo "$ac_cv_lib_acl_acl_init" >&6; }
-if test "x$ac_cv_lib_acl_acl_init" = x""yes; then
-  :
-else
-  { { $as_echo "$as_me:$LINENO: error: libacl not found" >&5
-$as_echo "$as_me: error: libacl not found" >&2;}
-   { (exit 1); exit 1; }; }
-fi
-
-	if test "${ac_cv_header_acl_libacl_h+set}" = set; then
-  { $as_echo "$as_me:$LINENO: checking for acl/libacl.h" >&5
-$as_echo_n "checking for acl/libacl.h... " >&6; }
-if test "${ac_cv_header_acl_libacl_h+set}" = set; then
-  $as_echo_n "(cached) " >&6
-fi
-{ $as_echo "$as_me:$LINENO: result: $ac_cv_header_acl_libacl_h" >&5
-$as_echo "$ac_cv_header_acl_libacl_h" >&6; }
-else
-  # Is the header compilable?
-{ $as_echo "$as_me:$LINENO: checking acl/libacl.h usability" >&5
-$as_echo_n "checking acl/libacl.h usability... " >&6; }
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-$ac_includes_default
-#include <acl/libacl.h>
-_ACEOF
-rm -f conftest.$ac_objext
-if { (ac_try="$ac_compile"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval ac_try_echo="\"\$as_me:$LINENO: $ac_try_echo\""
-$as_echo "$ac_try_echo") >&5
-  (eval "$ac_compile") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } && {
-	 test -z "$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       } && test -s conftest.$ac_objext; then
-  ac_header_compiler=yes
-else
-  $as_echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-	ac_header_compiler=no
-fi
-
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-{ $as_echo "$as_me:$LINENO: result: $ac_header_compiler" >&5
-$as_echo "$ac_header_compiler" >&6; }
-
-# Is the header present?
-{ $as_echo "$as_me:$LINENO: checking acl/libacl.h presence" >&5
-$as_echo_n "checking acl/libacl.h presence... " >&6; }
-cat >conftest.$ac_ext <<_ACEOF
-/* confdefs.h.  */
-_ACEOF
-cat confdefs.h >>conftest.$ac_ext
-cat >>conftest.$ac_ext <<_ACEOF
-/* end confdefs.h.  */
-#include <acl/libacl.h>
-_ACEOF
-if { (ac_try="$ac_cpp conftest.$ac_ext"
-case "(($ac_try" in
-  *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
-  *) ac_try_echo=$ac_try;;
-esac
-eval ac_try_echo="\"\$as_me:$LINENO: $ac_try_echo\""
-$as_echo "$ac_try_echo") >&5
-  (eval "$ac_cpp conftest.$ac_ext") 2>conftest.er1
-  ac_status=$?
-  grep -v '^ *+' conftest.er1 >conftest.err
-  rm -f conftest.er1
-  cat conftest.err >&5
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); } >/dev/null && {
-	 test -z "$ac_c_preproc_warn_flag$ac_c_werror_flag" ||
-	 test ! -s conftest.err
-       }; then
-  ac_header_preproc=yes
-else
-  $as_echo "$as_me: failed program was:" >&5
-sed 's/^/| /' conftest.$ac_ext >&5
-
-  ac_header_preproc=no
-fi
-
-rm -f conftest.err conftest.$ac_ext
-{ $as_echo "$as_me:$LINENO: result: $ac_header_preproc" >&5
-$as_echo "$ac_header_preproc" >&6; }
-
-# So?  What about this header?
-case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
-  yes:no: )
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: accepted by the compiler, rejected by the preprocessor!" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: proceeding with the compiler's result" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: proceeding with the compiler's result" >&2;}
-    ac_header_preproc=yes
-    ;;
-  no:yes:* )
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: present but cannot be compiled" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: present but cannot be compiled" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h:     check for missing prerequisite headers?" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h:     check for missing prerequisite headers?" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: see the Autoconf documentation" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: see the Autoconf documentation" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h:     section \"Present But Cannot Be Compiled\"" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h:     section \"Present But Cannot Be Compiled\"" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: proceeding with the preprocessor's result" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: proceeding with the preprocessor's result" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: acl/libacl.h: in the future, the compiler will take precedence" >&5
-$as_echo "$as_me: WARNING: acl/libacl.h: in the future, the compiler will take precedence" >&2;}
-    ( cat <<\_ASBOX
-## -------------------------------------------- ##
-## Report this to linux-hotplug@vger.kernel.org ##
-## -------------------------------------------- ##
-_ASBOX
-     ) | sed "s/^/$as_me: WARNING:     /" >&2
-    ;;
-esac
-{ $as_echo "$as_me:$LINENO: checking for acl/libacl.h" >&5
-$as_echo_n "checking for acl/libacl.h... " >&6; }
-if test "${ac_cv_header_acl_libacl_h+set}" = set; then
-  $as_echo_n "(cached) " >&6
-else
-  ac_cv_header_acl_libacl_h=$ac_header_preproc
-fi
-{ $as_echo "$as_me:$LINENO: result: $ac_cv_header_acl_libacl_h" >&5
-$as_echo "$ac_cv_header_acl_libacl_h" >&6; }
-
-fi
-if test "x$ac_cv_header_acl_libacl_h" = x""yes; then
-  :
-else
-  { { $as_echo "$as_me:$LINENO: error: libacl header not found" >&5
-$as_echo "$as_me: error: libacl header not found" >&2;}
-   { (exit 1); exit 1; }; }
-fi
-
-
-
-
 pkg_failed=no
 { $as_echo "$as_me:$LINENO: checking for LIBUSB" >&5
 $as_echo_n "checking for LIBUSB... " >&6; }
@@ -12683,42 +12466,42 @@
 
 
 
-pkg_failed=no
-{ $as_echo "$as_me:$LINENO: checking for USBUTILS" >&5
-$as_echo_n "checking for USBUTILS... " >&6; }
-
-if test -n "$USBUTILS_CFLAGS"; then
-    pkg_cv_USBUTILS_CFLAGS="$USBUTILS_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"usbutils >= 0.82\"") >&5
-  ($PKG_CONFIG --exists --print-errors "usbutils >= 0.82") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_USBUTILS_CFLAGS=`$PKG_CONFIG --cflags "usbutils >= 0.82" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$USBUTILS_LIBS"; then
-    pkg_cv_USBUTILS_LIBS="$USBUTILS_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"usbutils >= 0.82\"") >&5
-  ($PKG_CONFIG --exists --print-errors "usbutils >= 0.82") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_USBUTILS_LIBS=`$PKG_CONFIG --libs "usbutils >= 0.82" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
+#pkg_failed=no
+#{ $as_echo "$as_me:$LINENO: checking for USBUTILS" >&5
+#$as_echo_n "checking for USBUTILS... " >&6; }
+#
+#if test -n "$USBUTILS_CFLAGS"; then
+#    pkg_cv_USBUTILS_CFLAGS="$USBUTILS_CFLAGS"
+# elif test -n "$PKG_CONFIG"; then
+#    if test -n "$PKG_CONFIG" && \
+#    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"usbutils >= 0.82\"") >&5
+#  ($PKG_CONFIG --exists --print-errors "usbutils >= 0.82") 2>&5
+#  ac_status=$?
+#  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
+#  (exit $ac_status); }; then
+#  pkg_cv_USBUTILS_CFLAGS=`$PKG_CONFIG --cflags "usbutils >= 0.82" 2>/dev/null`
+#else
+#  pkg_failed=yes
+#fi
+# else
+#    pkg_failed=untried
+#fi
+#if test -n "$USBUTILS_LIBS"; then
+#    pkg_cv_USBUTILS_LIBS="$USBUTILS_LIBS"
+# elif test -n "$PKG_CONFIG"; then
+#    if test -n "$PKG_CONFIG" && \
+#    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"usbutils >= 0.82\"") >&5
+#  ($PKG_CONFIG --exists --print-errors "usbutils >= 0.82") 2>&5
+#  ac_status=$?
+#  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
+#  (exit $ac_status); }; then
+#  pkg_cv_USBUTILS_LIBS=`$PKG_CONFIG --libs "usbutils >= 0.82" 2>/dev/null`
+#else
+#  pkg_failed=yes
+#fi
+# else
+#    pkg_failed=untried
+#fi
 
 
 
