diff -uNr enigma2pc-plugins-20130618.orig/Extensions/OpenWebIf/controllers/base.py enigma2pc-plugins-20130618/Extensions/OpenWebIf/controllers/base.py
--- enigma2pc-plugins-20130618.orig/Extensions/OpenWebIf/controllers/base.py	2013-06-18 16:19:43.000000000 +0800
+++ enigma2pc-plugins-20130618/Extensions/OpenWebIf/controllers/base.py	2013-09-16 19:40:41.539785683 +0800
@@ -137,12 +137,12 @@
 		ret['configsections'] = getConfigsSections()['sections']
 		ret['zapstream'] = getZapStream()['zapstream']
 		ret['box'] = "dmm"
-		if open("/usr/local/e2/etc/stb/info/model",'r').read().strip() == "Gigablue":
+		if open("/etc/stb/info/model",'r').read().strip() == "Gigablue":
 			ret['box'] = "gigablue"
-		if fileExists("/usr/local/e2/etc/stb/info/vumodel"):
-			ret['box'] = open("/usr/local/e2/etc/stb/info/vumodel").read().strip()
-		elif fileExists("/usr/local/e2/etc/stb/info/boxtype"):
-			ret['box'] = open("/usr/local/e2/etc/stb/info/boxtype").read().strip()
+		if fileExists("/etc/stb/info/vumodel"):
+			ret['box'] = open("/etc/stb/info/vumodel").read().strip()
+		elif fileExists("/etc/stb/info/boxtype"):
+			ret['box'] = open("/etc/stb/info/boxtype").read().strip()
 			
 		if ret["box"] == "duo" or ret["box"] == "solo" or ret["box"] == "uno":
 			ret["remote"] = "vu_normal"
diff -uNr enigma2pc-plugins-20130618.orig/Extensions/OpenWebIf/controllers/models/info.py enigma2pc-plugins-20130618/Extensions/OpenWebIf/controllers/models/info.py
--- enigma2pc-plugins-20130618.orig/Extensions/OpenWebIf/controllers/models/info.py	2013-06-18 16:19:43.000000000 +0800
+++ enigma2pc-plugins-20130618/Extensions/OpenWebIf/controllers/models/info.py	2013-09-16 19:42:23.227781429 +0800
@@ -66,26 +66,26 @@
 	model = "unknown"
 	chipset = "unknown"
 	
-	if fileExists("/usr/local/e2/etc/stb/info/vumodel"):
+	if fileExists("/etc/stb/info/vumodel"):
 		brand = "Vuplus"
-		f = open("/usr/local/e2/etc/stb/info/vumodel",'r')
+		f = open("/etc/stb/info/vumodel",'r')
  		model = f.readline().strip()
  		f.close()
-	elif fileExists("/usr/local/e2/etc/stb/info/boxtype"):
+	elif fileExists("/etc/stb/info/boxtype"):
 		brand = "Clarke-Xtrend"
-		f = open("/usr/local/e2/etc/stb/info/boxtype",'r')
+		f = open("/etc/stb/info/boxtype",'r')
  		model = f.readline().strip()
  		f.close()
 	else:
-		f = open("/usr/local/e2/etc/stb/info/model",'r')
+		f = open("/etc/stb/info/model",'r')
  		model = f.readline().strip()
  		f.close()
 
 	info['brand'] = brand
 	info['model'] = model
 
-	if fileExists("/usr/local/e2/etc/stb/info/chipset"):
-		f = open("/usr/local/e2/etc/stb/info/chipset",'r')
+	if fileExists("/etc/stb/info/chipset"):
+		f = open("/etc/stb/info/chipset",'r')
  		chipset = f.readline().strip()
  		f.close()
 		
