diff -uNr enigma2pc-plugins-20130618.orig/Extensions/OscamStatus/OscamStatus.cfg enigma2pc-plugins-20130618/Extensions/OscamStatus/OscamStatus.cfg
--- enigma2pc-plugins-20130618.orig/Extensions/OscamStatus/OscamStatus.cfg	2013-06-18 16:19:43.000000000 +0800
+++ enigma2pc-plugins-20130618/Extensions/OscamStatus/OscamStatus.cfg	2013-09-16 20:00:13.971736649 +0800
@@ -1 +1 @@
-oscam oscam 127.0.0.1 8080 shara 0
+openpctv openpctv 127.0.0.1 8888 shara 0
