diff -uNr enigma2pc-plugins-20130618.orig/PLi/SoftcamSetup/camcontrol.py enigma2pc-plugins-20130618/PLi/SoftcamSetup/camcontrol.py
--- enigma2pc-plugins-20130618.orig/PLi/SoftcamSetup/camcontrol.py	2013-06-18 16:19:43.000000000 +0800
+++ enigma2pc-plugins-20130618/PLi/SoftcamSetup/camcontrol.py	2013-08-17 08:34:20.331638698 +0800
@@ -1,18 +1,18 @@
 import os
 
 class CamControl:
-	'''CAM convention is that a softlink named /etc/init.c/softcam.* points
+	'''CAM convention is that a softlink named /lib/systecmd/system/oscam.server points
 	to the start/stop script.'''
 	def __init__(self, name):
 		self.name = name
-		self.link = '/etc/init.d/' + name
+		self.link = '/bin/' + name
 		if not os.path.exists(self.link):
 			print "[CamControl] No softcam link?", self.link
 
 	def getList(self):
 		result = []
 		prefix = self.name + '.'
-		for f in os.listdir("/etc/init.d"):
+		for f in os.listdir("/bin"):
 			if f.startswith(prefix):
 				result.append(f[len(prefix):])
 		return result
@@ -35,7 +35,7 @@
 		if not which:
 			which = "None"
 		dst = self.name + '.' + which
-		if not os.path.exists('/etc/init.d/' + dst):
+		if not os.path.exists('/bin/' + dst):
 			print "[CamControl] init script does not exist:", dst
 			return 
 		try:
diff -uNr enigma2pc-plugins-20130618.orig/PLi/SoftcamSetup/Sc.py enigma2pc-plugins-20130618/PLi/SoftcamSetup/Sc.py
--- enigma2pc-plugins-20130618.orig/PLi/SoftcamSetup/Sc.py	2013-06-18 16:19:43.000000000 +0800
+++ enigma2pc-plugins-20130618/PLi/SoftcamSetup/Sc.py	2013-08-17 08:28:49.175652548 +0800
@@ -101,9 +101,9 @@
 	def doStop(self):
 		self.activityTimer.stop()
 		if "c" in self.what:
-			self.cardserver.command('stop')
+			self.cardserver.command('systemctl stop oscam')
 		if "s" in self.what:
-			self.softcam.command('stop')
+			self.softcam.command('systemctl stop oscam')
 		self.oldref = self.session.nav.getCurrentlyPlayingServiceReference()
 		self.session.nav.stopService()
 		# Delay a second to give 'em a chance to stop
@@ -116,10 +116,10 @@
 		del self.activityTimer 
 		if "c" in self.what:
                         self.cardserver.select(self.cardservers.value)
-			self.cardserver.command('start')
+			self.cardserver.command('systemctl start oscam')
 		if "s" in self.what:
 			self.softcam.select(self.softcams.value)
-			self.softcam.command('start')
+			self.softcam.command('systemctl start oscam')
 		self.mbox.close()
 		self.close()
 		self.session.nav.playService(self.oldref)
