--- vdr-graphtftng-0.4.8/graphtft-fe/Makefile.org	2014-02-22 19:59:41.858336581 +0100
+++ vdr-graphtftng-0.4.8/graphtft-fe/Makefile	2014-02-22 19:59:49.814336344 +0100
@@ -4,7 +4,8 @@
 CXXFLAGS  += -Wno-deprecated-declarations
 LFLAGS    = -Wl,--no-undefined
 LIBS      = -lpthread -ljpeg -lX11
-LIBS      += $(shell imlib2-config --libs)
+#LIBS      += $(shell imlib2-config --libs)
+LIBS      += -lImlib2
 AR        = ar
 
 TARGET = graphtft-fe
