diff -ruN 1.7.41_10/osdbase.c 1.7.41_11/osdbase.c
--- 1.7.41_10/osdbase.c	2013-03-18 21:26:24.000000000 +0100
+++ 1.7.41_11/osdbase.c	2013-03-18 21:31:10.000000000 +0100
@@ -326,6 +326,7 @@
 
 void cOsdMenu::CursorUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -364,6 +365,7 @@
 
 void cOsdMenu::CursorDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -404,6 +406,7 @@
 
 void cOsdMenu::PageUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current -= displayMenuItems;
@@ -438,6 +441,7 @@
 
 void cOsdMenu::PageDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current += displayMenuItems;
