diff -ruN 1.7.41_30/eitscan.c 1.7.41_31/eitscan.c
--- 1.7.41_30/eitscan.c	2012-04-07 16:39:28.000000000 +0200
+++ 1.7.41_31/eitscan.c	2013-03-18 23:07:58.000000000 +0100
@@ -145,7 +145,7 @@
                if (Device && Device->ProvidesEIT()) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (Device->Priority() < 0) {
diff -ruN 1.7.41_30/pat.c 1.7.41_31/pat.c
--- 1.7.41_30/pat.c	2013-03-18 23:06:14.000000000 +0100
+++ 1.7.41_31/pat.c	2013-03-18 23:11:54.923243225 +0100
@@ -569,6 +569,7 @@
                  TeletextSubtitlePages[NumTPages++] = tTeletextSubtitlePage(manualPageNumber);
               }
            Channel->SetTeletextSubtitlePages(TeletextSubtitlePages, NumTPages);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
