diff -Naur newt-0.52.14//Makefile.in _newt-0.52.14//Makefile.in
--- newt-0.52.14/Makefile.in	2011-11-11 12:00:15.000000000 +0100
+++ newt-0.52.14_patched/Makefile.in	2011-11-18 14:28:06.055357145 +0100
@@ -5,7 +5,7 @@
 CPP = @CPP@
 CFLAGS = @CFLAGS@
 LDFLAGS = @LDFLAGS@
-CPPFLAGS = -D_GNU_SOURCE -I/usr/include/slang @CPPFLAGS@ 
+CPPFLAGS = -D_GNU_SOURCE  @CPPFLAGS@
 GNU_LD = @GNU_LD@
 
 VERSION = @VERSION@
