diff -Naur inputlirc-r11.orig/inputlircd.c inputlirc-r11/inputlircd.c
--- inputlirc-r11.orig/inputlircd.c	2007-05-14 16:46:21.000000000 +0200
+++ inputlirc-r11/inputlircd.c	2007-05-14 16:47:03.000000000 +0200
@@ -35,8 +35,9 @@
 #include <signal.h>
 #include <syslog.h>
 #include <pwd.h>
+#include <asm/types.h>
+#include <linux/input.h>
 
-#include </usr/include/linux/input.h>
 #include "names.h"
 
 typedef struct evdev {
diff -Naur inputlirc-r11.orig/Makefile inputlirc-r11/Makefile
--- inputlirc-r11.orig/Makefile	2007-05-14 16:46:21.000000000 +0200
+++ inputlirc-r11/Makefile	2007-05-14 16:47:35.000000000 +0200
@@ -27,10 +27,10 @@
 
 all: $(SBIN)
 
-names.h: /usr/include/linux/input.h gennames
+names.h: ${SYSROOT_PREFIX}/usr/include/linux/input.h gennames
 	./gennames $< > $@
 
-inputlircd: inputlircd.c /usr/include/linux/input.h names.h
+inputlircd: inputlircd.c ${SYSROOT_PREFIX}/usr/include/linux/input.h names.h
 	$(CC) $(CFLAGS) -o $@ $<
 
 install: install-sbin install-man
