diff -Naur gst-bcmdec-0.10.0.39-git+20101203.orig//src/decif.h gst-bcmdec-0.10.0.39-git+20101203/src/decif.h
--- gst-bcmdec-0.10.0.39-git+20101203.orig//src/decif.h	2010-12-04 15:01:15.122122002 +0100
+++ gst-bcmdec-0.10.0.39-git+20101203/src/decif.h	2010-12-04 15:02:17.542122001 +0100
@@ -27,6 +27,7 @@
 #define __BCMDECIF_H__
 
 #include "bc_dts_defs.h"
+#include "bc_dts_types.h"
 #include "libcrystalhd_if.h"
 
 #define PROC_TIMEOUT 3000
diff -Naur gst-bcmdec-0.10.0.39-git+20101203.orig//src/parse.c gst-bcmdec-0.10.0.39-git+20101203/src/parse.c
--- gst-bcmdec-0.10.0.39-git+20101203.orig//src/parse.c	2010-12-04 15:01:15.122122002 +0100
+++ gst-bcmdec-0.10.0.39-git+20101203/src/parse.c	2010-12-04 15:02:25.882122005 +0100
@@ -25,6 +25,7 @@
 #include <glib.h>
 #include <gst/gst.h>
 #include "bc_dts_defs.h"
+#include "bc_dts_types.h"
 #include "parse.h"
 
 void parse_init(Parse *parse)
