--- cwiid-r201/python/setup.py	2009-09-28 15:18:06.000000000 +0200
+++ cwiid-r201/python/setup.py-new	2012-11-08 11:34:34.251659003 +0100
@@ -1,4 +1,4 @@
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 
 setup(name='cwiid',
 	version='0.6.00',
--- cwiid-r201/wminput/Makefile.in	2012-11-08 11:33:55.321601904 +0100
+++ cwiid-r201/wminput/Makefile.in-new	2012-11-08 11:34:25.101645583 +0100
@@ -22,7 +22,7 @@
 
 ifdef PYTHON
 	SOURCES += py_plugin.c
-	CFLAGS += -I/usr/include/python@PYTHON_VERSION@
+	CFLAGS += -I$(LIB_P)/include/python@PYTHON_VERSION@
 	LDLIBS += -lpython@PYTHON_VERSION@
 endif
 
--- cwiid-r201/python/Makefile.in	2009-09-28 15:18:06.000000000 +0200
+++ cwiid-r201/python/Makefile.in-new	2012-11-08 11:34:14.721630359 +0100
@@ -7,7 +7,7 @@
 endif
 
 all:
-	$(PYTHON) setup.py build_ext $(DEBUGFLAGS) -I@top_builddir@/libcwiid -L@top_builddir@/libcwiid -lcwiid
+	CFLAGS="$(TARGET_CFLAGS)" $(PYTHON) setup.py build --cross-compile $(DEBUGFLAGS)
 
 install:
 	$(PYTHON) setup.py install --install-lib=${libdir}/python@PYTHON_VERSION@/site-packages $(SET_ROOT_DIR)
