--- a/lib/timidity/timidity/buffer_a.c	2017-10-11 14:37:13.308095192 +0200
+++ b/lib/timidity/timiditybuffer_a.c	2017-10-11 14:37:13.328095148 +0200
@@ -87,7 +87,7 @@
 				return -1; // just stop adding
 
 			output_song->stored_buffer = newbuf;
-			memcpy( (char*(output_song->stored_buffer) + output_song->stored_size, buf + copylength, stored_size );
+			memcpy( (char*)(output_song->stored_buffer) + output_song->stored_size, buf + copylength, stored_size );
 			output_song->stored_size = output_song->stored_size + stored_size;
 		}
 		else
