diff -urp fuseiso-20070708.orig/src/isofs.c fuseiso-20070708/src/isofs.c
--- fuseiso-20070708.orig/src/isofs.c	2007-07-08 20:22:59.000000000 +0800
+++ fuseiso-20070708/src/isofs.c	2008-06-03 23:15:11.000000000 +0800
@@ -18,8 +18,6 @@
  *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
  ***************************************************************************/
 
-// for struct tm->tm_gmtoff
-#define _BSD_SOURCE
 
 #include <stdio.h>
 #include <fcntl.h>
@@ -503,7 +501,6 @@ static time_t isofs_date(char *stamp, in
         tm.tm_min = stamp[4];
         tm.tm_sec = stamp[5];
         tm.tm_isdst = -1;
-        tm.tm_gmtoff = stamp[6] * 15 * 60;
     } else if(stamp_len == 17) { // ISO9660:8.4.26.1
         fprintf(stderr, "isofs_date: 17 byte date isn`t supported for the moment, sorry\n");
         return 0;
@@ -512,9 +509,6 @@ static time_t isofs_date(char *stamp, in
         return 0;
     };
 
-    if(tm.tm_gmtoff) {
-//         fprintf(stderr, "direntry2stat: non-zero timezone offset: %d\n", tm.tm_gmtoff);
-    };
     time_t time = mktime(& tm);
     
     return time;
