diff -uNr vdr-boblight-0.0.5+r06464fb.orig/Makefile vdr-boblight-0.0.5+r06464fb/Makefile
--- vdr-boblight-0.0.5+r06464fb.orig/Makefile	2014-01-27 15:41:20.100685689 +0800
+++ vdr-boblight-0.0.5+r06464fb/Makefile	2014-01-27 15:41:32.176685184 +0800
@@ -7,7 +7,7 @@
 # This name will be used in the '-P...' option of VDR to load the plugin.
 # By default the main source file also carries this name.
 
-PLUGIN = vdrboblight
+PLUGIN = boblight
 
 ### The version number of this plugin (taken from the main source file):
 
