diff -Naur acl-2.2.47.orig/include/buildmacros acl-2.2.47/include/buildmacros
--- acl-2.2.47.orig/include/buildmacros	2009-12-13 01:00:29.702519654 +0100
+++ acl-2.2.47/include/buildmacros	2009-12-13 01:00:44.063769112 +0100
@@ -42,10 +42,10 @@
 LTOBJECTS = $(OBJECTS:.o=.lo)
 LTVERSION = $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
-LTLINK = $(LIBTOOL) --mode=link $(CC)
+LTLINK = $(LIBTOOL) --tag=CC --mode=link $(CC)
 LTEXEC = $(LIBTOOL) --mode=execute
 LTINSTALL = $(LIBTOOL) --mode=install $(INSTALL)
-LTCOMPILE = $(LIBTOOL) --mode=compile $(CCF)
+LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CCF)
 
 ifeq ($(ENABLE_SHARED),yes)
 LTLDFLAGS += -rpath $(PKG_LIB_DIR)
