--- a/opkg.py	2014-11-24 12:41:57.000000000 +0100
+++ b/opkg.py	2015-12-12 19:59:32.460073185 +0100
@@ -226,7 +226,8 @@
                     line = control.readline()
                     if not line: break
                     if line[0] != ' ': break
+                    line = line.strip()
-                    value = value + '\n' + line
+                    value = value + "\n " + line
                 if name == 'size':
                     self.size = int(value)
                 elif name == 'md5sum':
@@ -474,8 +475,7 @@
         if self.filename: out = out + "Filename: %s\n" % (self.filename)
         if self.source: out = out + "Source: %s\n" % (self.source)
         if self.description:
-            printable_description = textwrap.dedent(self.description).strip()
-            out = out + "Description: %s\n" % textwrap.fill(printable_description, width=74, initial_indent=' ', subsequent_indent=' ')
+            out = out + "Description: %s\n" % (self.description.strip())
         if self.oe: out = out + "OE: %s\n" % (self.oe)
         if self.homepage: out = out + "HomePage: %s\n" % (self.homepage)
         if self.license: out = out + "License: %s\n" % (self.license)
