--- a/opkg-buildpackage	2016-05-20 10:28:47.854446432 +0200
+++ b/opkg-buildpackage	2016-05-20 10:28:47.886446431 +0200
@@ -202,7 +202,11 @@
	file=${CONTROL}/$i

	if [ -e ${file} ] ; then
-		cp $file /tmp/${pkg}/CONTROL
+		if [ -h ${file} ] && [ $(dirname $(readlink ${file})) = "." ]; then
+			ln -sf ${pkg}.$(readlink ${file}) /tmp/${pkg}/CONTROL/$(basename ${file})
+		else
+			cp $file /tmp/${pkg}/CONTROL
+		fi
	fi
 done
