diff -Naur evas-1.0.0.beta2.orig//configure evas-1.0.0.beta2/configure
--- evas-1.0.0.beta2.orig//configure	2010-11-14 20:34:46.841802002 +0100
+++ evas-1.0.0.beta2/configure	2010-11-14 20:35:12.481802002 +0100
@@ -22235,7 +22235,7 @@
       x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
       x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext -lXrender"
    fi
-   evas_engine_gl_x11_cflags="-I/usr/include ${x_cflags}"
+   evas_engine_gl_x11_cflags="${x_cflags}"
    evas_engine_gl_x11_libs="${x_libs} -lGL $gl_pt_lib"
    evas_engine_gl_common_libs="-lGL $gl_pt_lib"
 else
