diff -Naur evas-r44003.orig/configure evas-r44003/configure
--- evas-r44003.orig/configure	2009-11-26 18:28:14.672033368 +0100
+++ evas-r44003/configure	2009-11-26 18:28:52.443280123 +0100
@@ -21064,7 +21064,7 @@
       x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
       x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11"
    fi
-   evas_engine_gl_x11_cflags="-I/usr/include ${x_cflags}"
+   evas_engine_gl_x11_cflags="${x_cflags}"
    evas_engine_gl_x11_libs="${x_libs} -lGL -lpthread"
    evas_engine_gl_common_libs="-lGL -lpthread"
 else
