diff -Naur evas-r48546.orig/src/lib/file/evas_module.c evas-r48546/src/lib/file/evas_module.c
--- evas-r48546.orig/src/lib/file/evas_module.c	2010-05-22 18:48:04.567491190 +0200
+++ evas-r48546/src/lib/file/evas_module.c	2010-05-22 18:48:28.527717879 +0200
@@ -73,7 +73,7 @@
 
    /* 4. PREFIX/evas/modules/ */
 #ifndef _MSC_VER
-   path3 = PACKAGE_LIB_DIR "/evas/modules";
+   path3 = "/usr/lib/evas/modules";
    if ((path && (strcmp(path, path3) != 0)) ||
        (path2 && (strcmp(path2, path3) != 0)) ||
        (!path && !path2))
