diff -Naur evas-0.9.9.49539/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c evas-0.9.9.49539a/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c
--- evas-0.9.9.49539/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c	2010-05-24 04:37:55.000000000 +0200
+++ evas-0.9.9.49539a/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c	2010-06-07 12:54:57.366950839 +0200
@@ -68,7 +68,7 @@
 	"	cmp		%[tmp], %[d]			\n\t"
 	"	ble		"AP"loopout			\n\t"
 	AP"quadloopint:						\n\t"
-	"	ldr		%[x],	[%[m]]			\n\t"
+	"	ldr.32		%[x],	[%[m]]			\n\t"
 	"	add 		%[m], #4			\n\t"
 	"	cmp		%[x],	#0			\n\t"
 	"	beq		"AP"fastloop			\n\t"
@@ -231,7 +231,7 @@
 
 	AP"quadloopint:					\n\t"
 		// Load the mask: 4 bytes: It has d0/d1
-	"	ldr		%[x],	[%[m]]		\n\t"
+	"	ldr.32		%[x],	[%[m]]		\n\t"
 	"	add		%[m], #4		\n\t"
 	"	cmp		%[x],	#0		\n\t"
 	"	beq		"AP"fastloop		\n\t"
