diff -Naur evas-r49141.orig/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c evas-r49141/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c
--- evas-r49141.orig/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c	2010-05-23 10:55:16.717491714 +0200
+++ evas-r49141/src/lib/engines/common/evas_op_blend/op_blend_mask_color_neon.c	2010-05-23 10:56:37.447490792 +0200
@@ -69,7 +69,7 @@
 	"	ble		"AP"loopout			\n\t"
 	AP"quadloopint:						\n\t"
 //	"	vld1.32		d0[0],	[%[m]]!			\n\t"
-	"	ldr.32		%[x],	[%[m]]			\n\t"
+	"	ldr		%[x],	[%[m]]			\n\t"
 	"	add %[m], #4					\n\t"
 	"	cmp		%[x],	#0			\n\t"
 	"	beq		"AP"fastloop			\n\t"
@@ -232,7 +232,7 @@
 
 	AP"quadloopint:					\n\t"
 		// Load the mask: 4 bytes: It has d0/d1
-	"	ldr.32		%[x],	[%[m]]		\n\t"
+	"	ldr		%[x],	[%[m]]		\n\t"
 	"	add		%[m], #4		\n\t"
 	"	cmp		%[x],	#0		\n\t"
 	"	beq		"AP"fastloop		\n\t"
