diff -Naur expedite-r48914.orig/src/bin/main.c expedite-r48914/src/bin/main.c
--- expedite-r48914.orig/src/bin/main.c	2010-06-02 20:58:18.847188117 +0200
+++ expedite-r48914/src/bin/main.c	2010-06-02 20:59:12.488546403 +0200
@@ -1095,7 +1095,7 @@
         if (!prefix)
           prefix = getenv("EXPEDITE_DATA_DIR");
         if (!prefix)
-          prefix = PACKAGE_DATA_DIR"/data/";
+          prefix = "/usr/share/expedite/data/";
 
         snprintf(path, 4096, "%s/", prefix);
 
@@ -1336,7 +1336,7 @@
    if (!prefix)
      prefix = getenv("EXPEDITE_FONTS_DIR");
    if (!prefix)
-     prefix = PACKAGE_DATA_DIR"/data";
+     prefix = "/usr/share/expedite/data";
 
    snprintf(buf, 4096, "%s", prefix);
 
