--- vdr-extrecmenu-1.2.1/mymenueditrecording.c	2012-03-28 11:54:04.279684108 +0200
+++ vdr-extrecmenu-1.2.1/mymenueditrecording.c.ok	2012-03-28 11:53:52.099684110 +0200
@@ -575,7 +575,7 @@
     {
       // Casting const away is nasty, but what the heck?
       // The Recordings thread is locked and the object is going to be deleted anyway.
-      if(((cRecordingInfo *)Recording->Info())->Read(f)&&Recording->WriteInfo())
+      if(((cRecordingInfo *)Recording->Info())->Read()&&Recording->WriteInfo())
         return true;
       esyslog("[extrecmenu] error in info string '%s'",Info);
     }
