--- a/build/common.gypi	2014-10-22 13:16:34.718289318 +0200
+++ b/build/common.gypi	2014-10-22 13:17:07.982288225 +0200
@@ -3557,11 +3557,11 @@
                   # and make sure a 32-bit-on-64-bit build picks up the
                   # right files.
                   # For android build, use NDK headers instead of host headers
-                  ['host_arch!="ia32" and OS!="android"', {
-                    'include_dirs+': [
-                      '/usr/include32',
-                    ],
-                  }],
+                  #['host_arch!="ia32" and OS!="android"', {
+                  #  'include_dirs+': [
+                  #    '/usr/include32',
+                  #  ],
+                  #}],
                 ],
               }],
             ],
