diff -Nur geexbox-generator-0.98.7/config/options geexbox-generator-0.98.7.new/config/options
--- geexbox-generator-0.98.7/config/options	2005-11-11 12:35:19.000000000 +0100
+++ geexbox-generator-0.98.7.new/config/options	2005-11-11 19:26:56.000000000 +0100
@@ -23,11 +23,11 @@
 
 # Need to point to your actual cc
 # If you have ccache installed, take care that LOCAL_CC don't point to it
-LOCAL_CC=`which gcc`
+LOCAL_CC=/usr/bin/gcc
 
 # Need to point to your actual g++
 # If you have ccache installed, take care that LOCAL_CXX don't point to it
-LOCAL_CXX=`which g++`
+LOCAL_CXX=/usr/bin/g++
 
 # Theme (omc/movies)
 THEME=omc
