diff -Nur geexbox-generator-0.98.7/config/options geexbox-generator-0.98.7.new/config/options
--- geexbox-generator-0.98.7/config/options	2005-11-11 20:06:12.000000000 +0100
+++ geexbox-generator-0.98.7.new/config/options	2005-11-11 20:11:44.000000000 +0100
@@ -112,6 +112,6 @@
 CONCURRENCY_MAKE_LEVEL=1
 
 # read personal options if available
-test -f "$HOME/.geexbox-options" && . "$HOME/.geexbox-options"
+#test -f "$HOME/.geexbox-options" && . "$HOME/.geexbox-options"
 
 . config/path
