--- vompserver-0.3.0/recplayer.c
+++ vompserver-0.3.0/recplayer.c
@@ -34,7 +34,11 @@ RecPlayer::RecPlayer(cRecording* rec)
 
   // FIXME find out max file path / name lengths
 
+#if VDRVERSNUM >= 10703
+  indexFile = new cIndexFile(recording->FileName(), false, rec->IsPesRecording());
+#else
   indexFile = new cIndexFile(recording->FileName(), false);
+#endif
   if (!indexFile) log->log("RecPlayer", Log::ERR, "Failed to create indexfile!");
 
   scan();
@@ -193,12 +197,15 @@ ULLONG RecPlayer::positionFromFrameNumbe
 {
   if (!indexFile) return 0;
 
+#if VDRVERSNUM >= 10703
+  uint16_t retFileNumber;
+  off_t retFileOffset;
+#else
   uchar retFileNumber;
   int retFileOffset;
-  uchar retPicType;
-  int retLength;
+#endif
 
-  if (!indexFile->Get((int)frameNumber, &retFileNumber, &retFileOffset, &retPicType, &retLength))
+  if (!indexFile->Get((int)frameNumber, &retFileNumber, &retFileOffset))
   {
     return 0;
   }
@@ -240,13 +247,10 @@ bool RecPlayer::getNextIFrame(ULONG fram
 
   if (!indexFile) return false;
 
-  uchar waste1;
-  int waste2;
-
   int iframeLength;
   int indexReturnFrameNumber;
 
-  indexReturnFrameNumber = (ULONG)indexFile->GetNextIFrame(frameNumber, (direction==1 ? true : false), &waste1, &waste2, &iframeLength);
+  indexReturnFrameNumber = (ULONG)indexFile->GetNextIFrame(frameNumber, (direction==1 ? true : false), NULL, NULL, &iframeLength);
   log->log("RecPlayer", Log::DEBUG, "GNIF input framenumber:%lu, direction=%lu, output:framenumber=%i, framelength=%i", frameNumber, direction, indexReturnFrameNumber, iframeLength);
 
   if (indexReturnFrameNumber == -1) return false;
