--- syslinux-5.10/Makefile.caz	2013-11-05 16:43:54.679262770 +0100
+++ syslinux-5.10/Makefile	2013-11-05 16:44:02.663262907 +0100
@@ -127,10 +127,6 @@
 	-install -m 644 -c $(INSTALL_AUX_OPT) $(INSTALLROOT)$(AUXDIR)
 	mkdir -m 755 -p $(INSTALLROOT)$(DIAGDIR)
 	install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)
-	mkdir -m 755 -p $(INSTALLROOT)$(MANDIR)/man1
-	install -m 644 -c man/*.1 $(INSTALLROOT)$(MANDIR)/man1
-	: mkdir -m 755 -p $(INSTALLROOT)$(MANDIR)/man8
-	: install -m 644 -c man/*.8 $(INSTALLROOT)$(MANDIR)/man8
 
 install: local-install
 	set -e ; for i in $(INSTALLSUBDIRS) ; do $(MAKE) -C $$i $@ ; done
