diff -Naur mythplugins-0.24/configure mythplugins-0.24a/configure
--- mythplugins-0.24/configure	2010-10-27 19:52:19.000000000 +0200
+++ mythplugins-0.24a/configure	2010-12-22 13:40:03.665953009 +0100
@@ -462,13 +462,6 @@
     check_pl_lib "SOAP::Lite"  || disable_weather "SOAP::Lite"
 fi
 
-if test "$dcraw" != "no" ; then
-    dcraw="no"
-    if test x`which dcraw` != x ; then
-        dcraw="yes"
-    fi
-fi
-
 if test x"$icc" != x ; then
     # Assume relative paths are for files in the standard icc directory.
     case "$icc" in
