diff -Naur mythplugins-0.24/configure mythplugins-0.24a//configure
--- mythplugins-0.24/configure	2010-12-29 15:24:52.213005885 +0100
+++ mythplugins-0.24a//configure	2010-12-29 15:26:59.809008265 +0100
@@ -132,6 +132,7 @@
 cc="gcc"
 qmake="qmake"
 python="python"
+[ -z "$SDL_CONFIG" ] && SDL_CONFIG=`which sdl-config`
 
 targetos=`uname -s`
 
@@ -410,7 +411,7 @@
 
 if test "$sdl" != "no" ; then
     sdl="no"
-    if test x`which sdl-config` != x ; then
+    if test -x "$SDL_CONFIG" ; then
         sdl="yes"
     fi
 fi
@@ -831,9 +832,9 @@
     if test "$sdl" = "yes" ; then
         echo "        SDL            support will be included in MythMusic"
         echo "#define SDL_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
-        echo "LIBS += `sdl-config --libs`" >> ./mythmusic/mythmusic/config.pro
-        echo "QMAKE_CXXFLAGS_RELEASE += `sdl-config --cflags`" >> ./mythmusic/mythmusic/config.pro
-        echo "QMAKE_CXXFLAGS_DEBUG += `sdl-config --cflags`" >> ./mythmusic/mythmusic/config.pro
+        echo "LIBS += `$SDL_CONFIG --libs`" >> ./mythmusic/mythmusic/config.pro
+        echo "QMAKE_CXXFLAGS_RELEASE += `$SDL_CONFIG --cflags`" >> ./mythmusic/mythmusic/config.pro
+        echo "QMAKE_CXXFLAGS_DEBUG += `$SDL_CONFIG --cflags`" >> ./mythmusic/mythmusic/config.pro
     fi
     if test "$sdl" = "no" ; then
         echo "        SDL            support will not be included in MythMusic"
