diff -Naur kaa-20061116.orig/setup.py kaa-20061116/setup.py
--- kaa-20061116.orig/setup.py	2006-11-25 20:49:37.000000000 +0100
+++ kaa-20061116/setup.py	2006-11-25 20:50:27.000000000 +0100
@@ -32,7 +32,7 @@
 import distutils.core
 
 submodules = [ 'base', 'imlib2', 'display', 'mevas', 'epg', 'record',
-               'metadata', 'evas', 'canvas', 'xine', 'cherrypy',
+               'metadata', 'cherrypy',
                'beacon', 'popcorn' ]
 
 for a in sys.argv:
