--- a/record/src/fp_udpsend.cc	2006-11-26 19:07:08.000000000 +0100
+++ b/record/src/fp_udpsend.cc	2006-12-08 15:20:44.000000000 +0100
@@ -145,7 +145,7 @@
 
   // fill struct with address and port
   struct sockaddr_in ucaddr;
-  bzero((char *) &ucaddr, sizeof(ucaddr));
+  memset((char *) &ucaddr, 0, sizeof(ucaddr));
   ucaddr.sin_family      = AF_INET;
   ucaddr.sin_addr.s_addr = INADDR_ANY;
   ucaddr.sin_port        = 0;
