diff -Naur kaa-r2231.orig/display/setup.py kaa-r2231/display/setup.py
--- kaa-r2231.orig/display/setup.py	2006-12-11 03:44:28.000000000 +0100
+++ kaa-r2231/display/setup.py	2006-12-14 22:44:39.000000000 +0100
@@ -45,13 +45,7 @@
 # config file
 config = ConfigFile('src/config.h')
 
-check_library('X11', ['<X11/Xlib.h>'], '')
 check_library('imlib2', '1.1.1')
-evas = check_library('evas', '0.9.9.010')
-check_library('directfb', '0.9.20')
-
-print 'checking for pygame', '...',
-sys.__stdout__.flush()
 
 try:
     import pygame
@@ -119,12 +113,7 @@
     fb = Extension('kaa.display._FBmodule', 
                    [ 'src/fb.c', 'src/common.c'])
     fb.add_library('imlib2')
-    if evas and evas.compile(['<Evas.h>', '<Evas_Engine_FB.h>']):
-        fb.add_library('evas')
-        config.define('ENABLE_ENGINE_FB')
-        print "+ Framebuffer (imlib2, evas)"
-    else:
-        print "+ Framebuffer (imlib2)"
+    print "+ Framebuffer (imlib2)"
     modules.append(fb)
 else:
     print "- Framebuffer"
