diff -Naur kaa-20061116.orig/display/setup.py kaa-20061116/display/setup.py
--- kaa-20061116.orig/display/setup.py	2006-11-25 20:49:37.000000000 +0100
+++ kaa-20061116/display/setup.py	2006-11-26 16:25:44.000000000 +0100
@@ -46,10 +46,7 @@
 # config file
 config = Configfile('src/config.h')
 
-check_library('X11', ['<X11/Xlib.h>'], '')
 check_library('imlib2', '1.1.1')
-check_library('evas', '0.9.9.010')
-check_library('directfb', '0.9.20')
 
 print 'checking for pygame', '...',
 sys.__stdout__.flush()
