diff -Naur kaa-r2231.orig/popcorn/src/backends/mplayer/player.py kaa-r2231/popcorn/src/backends/mplayer/player.py
--- kaa-r2231.orig/popcorn/src/backends/mplayer/player.py	2006-12-08 16:12:01.000000000 +0100
+++ kaa-r2231/popcorn/src/backends/mplayer/player.py	2006-12-29 15:11:51.000000000 +0100
@@ -507,14 +507,6 @@
         if filters:
             args.extend(("-vf", ",".join(filters)))
 
-        if isinstance(self._window, kaa.display.X11Window):
-            args.extend((
-                "-wid", hex(self._window.get_id()),
-                "-display", self._window.get_display().get_string()))
-        else:
-            # no window == no video out
-            args.extend(('-vo', 'null'))
-
         if self._config.audio.passthrough:
             args += [ '-ac', 'hwac3,hwdts,' ]
         else:
