--- a/display/setup.py	2006-12-08 16:44:11.000000000 +0100
+++ b/display/setup.py	2006-12-08 16:44:17.000000000 +0100
@@ -51,11 +51,10 @@
 sys.__stdout__.flush()

 try:
-    import pygame
+    #import pygame
     print 'ok'
     print 'checking for pygame header files', '...',
-    inc = re.sub("/(lib|lib64)/", "/include/",
-                 pygame.__path__[0]).replace("site-packages/", "")
+    inc = os.environ['PYTHONPATH'].replace("site-packages/", "")
     if not os.path.isdir(inc):
         raise ImportError
