diff -Naur kaa-r2286.orig/popcorn/src/backends/mplayer/player.py kaa-r2286/popcorn/src/backends/mplayer/player.py
--- kaa-r2286.orig/popcorn/src/backends/mplayer/player.py	2006-12-29 15:44:03.000000000 +0100
+++ kaa-r2286/popcorn/src/backends/mplayer/player.py	2006-12-29 15:47:08.000000000 +0100
@@ -369,7 +369,7 @@
         # open the stream and provide information about it. After that, the
         # caller can still change stuff before calling play. Mplayer doesn't
         # work that way so we have to run mplayer with -identify first.
-        args = "-nolirc -nojoystick -nomouseinput -identify " +\
+        args = "-nolirc -nojoystick -identify " +\
                "-vo null -ao null -frames 0"
         ident = kaa.notifier.Process(self._mp_cmd)
         ident.start(args.split(' ') + [ self._file ])
@@ -500,8 +500,8 @@
         if 'overlay' in self._mp_info['video_filters']:
             filters += ["overlay=%s" % self._osd_shmkey]
 
-        args = [ "-v", "-slave", "-osdlevel", "0", "-nolirc", "-nojoystick", \
-                 "-nomouseinput", "-nodouble", "-fixed-vo", "-identify", \
+        args = [ "-v", "-slave", "-nolirc", "-nojoystick", \
+                 "-nodouble", "-fixed-vo", "-identify", \
                  "-framedrop" ]
 
         if filters:
@@ -515,28 +515,6 @@
             # no window == no video out
             args.extend(('-vo', 'null'))
 
-        if self._config.audio.passthrough:
-            args += [ '-ac', 'hwac3,hwdts,' ]
-        else:
-            args += [ '-channels', str(self._config.audio.channels) ]
-            
-        args += [ '-ao', self._config.audio.driver ]
-        if self._config.audio.driver == 'alsa':
-            args[-1] += ":noblock"
-            n_channels = self._streaminfo.get('channels')
-            if self._streaminfo.get('acodec') in ('a52', 'hwac3', 'ffdts', 'hwdts'):
-                device = self._config.audio.device.passthrough
-            elif n_channels == 1:
-                device = self._config.audio.device.mono
-            elif n_channels <= 4:
-                device = self._config.audio.device.surround40
-            elif n_channels <= 6:
-                device = self._config.audio.device.surround51
-            else:
-                device = self._config.audio.device.stereo
-            if device != '':
-                args[-1] += ':device=' + device.replace(':', '=')
-            
         # There is no way to make MPlayer ignore keys from the X11 window.  So
         # this hack makes a temp input file that maps all keys to a dummy (and
         # non-existent) command which causes MPlayer not to react to any key
