--- kaa-r2307/beacon/src/server/videothumb.py.orig	2007-01-01 19:42:38.000000000 +0100
+++ kaa-r2307/beacon/src/server/videothumb.py	2007-01-01 19:42:44.000000000 +0100
@@ -65,8 +65,8 @@
         self.notify_client = thumbnailer.notify_client
         self.create_failed = thumbnailer.create_failed
 
-        self.child = kaa.notifier.Process(['mplayer', '-nosound', '-vo',
-                                           'png:z=2', '-frames', '10',
+        self.child = kaa.notifier.Process(['mplayer', '-nosound', '-nocache', '-osdlevel', '0',
+                                           '-vo', 'png:z=2', '-frames', '1',
                                            '-zoom', '-ss' ])
         self.child.signals['completed'].connect(self._completed)

