diff -Naur kaa-r2231.orig/display/src/sdl.py kaa-r2231/display/src/sdl.py
--- kaa-r2231.orig/display/src/sdl.py	2006-11-01 18:16:37.000000000 +0100
+++ kaa-r2231/display/src/sdl.py	2006-12-17 23:11:54.000000000 +0100
@@ -35,9 +35,6 @@
 import time
 import kaa.notifier
 
-# the display module
-import _X11
-
 class PygameDisplay(object):
     def __init__(self, size):
 
@@ -76,8 +73,6 @@
         is a list of pos, size of the areas to update.
         """
         if self._surface:
-            # we need to use our tmp surface
-            _X11.image_to_surface(image, self._surface)
             if areas == None:
                 # copy everything
                 self._screen.blit(self._surface, (0,0))
@@ -85,9 +80,6 @@
                 # copy only the needed areas
                 for pos, size in areas:
                     self._screen.blit(self._surface, pos, pos + size)
-        else:
-            # copy everything
-            _X11.image_to_surface(image, self._screen)
 
         # update the screen
         if areas:
