diff -Naur a/config.h b/config.h
--- a/config.h	2011-06-21 23:43:01.000000000 +0200
+++ b/config.h	2012-03-24 23:23:02.178391311 +0100
@@ -103,7 +103,7 @@
                 if (!isempty(s)) {
                    T *l = new T;
                    if (l->Parse(s))
-                      Add(l);
+                      this->Add(l);
                    else {
                       esyslog("ERROR: error in %s, line %d", fileName, line);
                       delete l;
