Signed-off-by: Paul Menzel <paulepanter at users.sourceforge.net>
CC: Henning Heinold <heinold at inf.fu-berlin.de>
---
 tools.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/tools.h b/tools.h
index 95c35ff..1fbe005 100644
--- a/tools.h
+++ b/tools.h
@@ -17,6 +17,7 @@
 #include <iconv.h>
 #include <math.h>
 #include <poll.h>
+#include <stdarg.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <stdio.h>
-- 
1.7.2.3
