diff -Naur vdr-1.6.0-davide/libsi/si.c vdr-1.6.0/libsi/si.c
--- vdr-1.6.0-davide/libsi/si.c	2008-03-05 18:00:55.000000000 +0100
+++ vdr-1.6.0/libsi/si.c	2008-05-11 13:45:20.000000000 +0200
@@ -382,7 +382,7 @@
      if (cd != (iconv_t)-1) {
         char *fromPtr = (char *)from;
         while (fromLength > 0 && toLength > 1) {
-           if (iconv(cd, &fromPtr, &fromLength, &to, &toLength) == size_t(-1)) {
+           if (iconv(cd, (const char **)&fromPtr, &fromLength, &to, &toLength) == size_t(-1)) {
               if (errno == EILSEQ) {
                  // A character can't be converted, so mark it with '?' and proceed:
                  fromPtr++;
diff -Naur vdr-1.6.0-davide/tools.c vdr-1.6.0/tools.c
--- vdr-1.6.0-davide/tools.c	2008-03-05 18:23:47.000000000 +0100
+++ vdr-1.6.0/tools.c	2008-05-11 13:45:00.000000000 +0200
@@ -822,7 +822,7 @@
      ToLength--; // save space for terminating 0
      char *Converted = ToPtr;
      while (FromLength > 0) {
-           if (iconv(cd, &FromPtr, &FromLength, &ToPtr, &ToLength) == size_t(-1)) {
+           if (iconv(cd, (const char**)&FromPtr, &FromLength, &ToPtr, &ToLength) == size_t(-1)) {
               if (errno == E2BIG || errno == EILSEQ && ToLength < 1) {
                  if (To)
                     break; // caller provided a fixed size buffer, but it was too small
