diff -Naur vdr-1.6.0/Makefile vdr-1.6.0-davide/Makefile
--- vdr-1.6.0/Makefile	2008-02-29 22:43:03.000000000 +0100
+++ vdr-1.6.0-davide/Makefile	2008-05-11 14:02:48.000000000 +0200
@@ -20,7 +20,7 @@
 MANDIR   = $(PREFIX)/share/man
 BINDIR   = $(PREFIX)/bin
 LOCDIR   = ./locale
-LIBS     = -ljpeg -lpthread -ldl -lrt -lfreetype
+LIBS     = -ljpeg -lpthread -lgcc -ldl -lrt -lfreetype
 INCLUDES = -I/usr/include/freetype2
 
 PLUGINDIR= ./PLUGINS
