diff -Naur vdr-1.6.0/dvbdevice.h vdr-1.6.0b/dvbdevice.h
--- vdr-1.6.0/dvbdevice.h	2009-02-05 20:36:12.000000000 +0100
+++ vdr-1.6.0b/dvbdevice.h	2009-02-05 20:36:43.000000000 +0100
@@ -15,8 +15,8 @@
 #include "device.h"
 #include "dvbspu.h"
 
-#if DVB_API_VERSION != 3
-#error VDR requires Linux DVB driver API version 3!
+#if DVB_API_VERSION < 3
+#error VDR requires Linux DVB driver API version 3 or later!
 #endif
 
 #define MAXDVBDEVICES  8
