diff -Naur vdr-1.6.0/dvbdevice.c vdr-1.6.0a/dvbdevice.c
--- vdr-1.6.0/dvbdevice.c	2009-02-15 09:23:08.000000000 +0100
+++ vdr-1.6.0a/dvbdevice.c	2009-02-15 09:22:59.000000000 +0100
@@ -7,10 +7,10 @@
  * $Id: dvbdevice.c 1.170 2008/02/09 16:11:44 kls Exp $
  */
 
-#include "dvbdevice.h"
 #include <errno.h>
 #include <limits.h>
 #include <linux/videodev.h>
+#include "dvbdevice.h"
 #include <linux/dvb/audio.h>
 #include <linux/dvb/dmx.h>
 #include <linux/dvb/frontend.h>
