diff -uNr scan-s2-rac895a3ca3788ad4105e86e671e6ca1b6f2907e2.orig/Makefile scan-s2-rac895a3ca3788ad4105e86e671e6ca1b6f2907e2/Makefile
--- scan-s2-rac895a3ca3788ad4105e86e671e6ca1b6f2907e2.orig/Makefile	2013-06-22 00:37:32.000000000 +0800
+++ scan-s2-rac895a3ca3788ad4105e86e671e6ca1b6f2907e2/Makefile	2013-07-10 19:55:50.722009020 +0800
@@ -1,10 +1,11 @@
-CC=gcc
+CC?=gcc
+CFLAGS?=-O2 -march=i586
 
 SRC=atsc_psip_section.c diseqc.c dump-vdr.c dump-zap.c lnb.c scan.c section.c 
 HED=atsc_psip_section.h diseqc.h dump-vdr.h dump-zap.h lnb.h scan.h section.h list.h
 OBJ=atsc_psip_section.o diseqc.o dump-vdr.o dump-zap.o lnb.o scan.o section.o
 
-BIND=/usr/local/bin/
+BIND?=/usr/local/bin/
 INCLUDE=-I../s2/linux/include
 
 TARGET=scan-s2
@@ -12,7 +13,7 @@
 all: $(TARGET)
 
 $(TARGET): $(OBJ)
-	$(CC) $(CFLG) $(OBJ) -o $(TARGET) $(CLIB) 
+	$(CC) $(CFLAGS) $(OBJ) -o $(TARGET) $(CLIB) 
 
 $(OBJ): $(HED)
 
@@ -26,4 +27,4 @@
 	rm -f $(OBJ) $(TARGET) *~
 
 %.o: %.c
-	$(CC) $(INCLUDE) -c $< -o $@
+	$(CC) $(CFLAGS) $(INCLUDE) -c $< -o $@
