diff -Naur sgx530-mine-4.05.00.03/eurasiacon/build/linux/omap3430_linux/makefile.shared_conf sgx530-mine-4.05.00.03_patched//eurasiacon/build/linux/omap3430_linux/makefile.shared_conf
--- sgx530-mine-4.05.00.03/eurasiacon/build/linux/omap3430_linux/makefile.shared_conf	2011-10-07 15:37:46.000000000 +0200
+++ sgx530-mine-4.05.00.03_patched//eurasiacon/build/linux/omap3430_linux/makefile.shared_conf	2011-12-21 00:46:22.418511774 +0100
@@ -69,9 +69,7 @@
 OMAP_KERNEL_SUBREL		:= $(shell grep "^SUBLEVEL = " \
 	 				$(KERNELDIR)/Makefile | cut -f3 -d' ')
 
-OMAP_KERNEL_AT_LEAST_2_6_29 := $(shell test $(OMAP_KERNEL_VER) -ge 2 -a \
-			$(OMAP_KERNEL_REL) -ge 6 -a \
-			$(OMAP_KERNEL_SUBREL) -ge 29 && echo 1 || echo 0)
+OMAP_KERNEL_AT_LEAST_2_6_29 := 1
 
 ifeq ($(OMAP_KERNEL_AT_LEAST_2_6_29),1)
 SUPPORT_OMAP3430_OMAPFB3 = 1
diff -Naur sgx530-mine-4.05.00.03/eurasiacon/build/linux/omap3630_linux/makefile.shared_conf sgx530-mine-4.05.00.03_patched//eurasiacon/build/linux/omap3630_linux/makefile.shared_conf
--- sgx530-mine-4.05.00.03/eurasiacon/build/linux/omap3630_linux/makefile.shared_conf	2011-10-07 15:37:46.000000000 +0200
+++ sgx530-mine-4.05.00.03_patched//eurasiacon/build/linux/omap3630_linux/makefile.shared_conf	2011-12-21 00:46:37.038511774 +0100
@@ -69,9 +69,7 @@
 OMAP_KERNEL_SUBREL		:= $(shell grep "^SUBLEVEL = " \
 	 				$(KERNELDIR)/Makefile | cut -f3 -d' ')
 
-OMAP_KERNEL_AT_LEAST_2_6_29 := $(shell test $(OMAP_KERNEL_VER) -ge 2 -a \
-			$(OMAP_KERNEL_REL) -ge 6 -a \
-			$(OMAP_KERNEL_SUBREL) -ge 29 && echo 1 || echo 0)
+OMAP_KERNEL_AT_LEAST_2_6_29 := 1
 
 ifeq ($(OMAP_KERNEL_AT_LEAST_2_6_29),1)
 SUPPORT_OMAP3430_OMAPFB3 = 1
