diff -Naur sgx530-OMAP34xx-drv-4.05.00.03//eurasiacon/build/linux/omap3630_linux/makefile.shared_conf sgx530-OMAP34xx-drv-4.05.00.03_patched/eurasiacon/build/linux/omap3630_linux/makefile.shared_conf
--- sgx530-OMAP34xx-drv-4.05.00.03//eurasiacon/build/linux/omap3630_linux/makefile.shared_conf	2011-12-22 15:13:31.009574980 +0100
+++ sgx530-OMAP34xx-drv-4.05.00.03_patched/eurasiacon/build/linux/omap3630_linux/makefile.shared_conf	2011-12-22 15:11:45.189574974 +0100
@@ -57,7 +57,7 @@
 # passive power management isn't enabled, the driver won't see the
 # system suspend/resume events, and so won't take appropriate action.
 ifeq ($(LDM_PLATFORM),1)
-SUPPORT_ACTIVE_POWER_MANAGEMENT ?= 0
+SUPPORT_ACTIVE_POWER_MANAGEMENT ?= 1
 else
 SUPPORT_ACTIVE_POWER_MANAGEMENT = 0
 endif
