diff -Naur sgx-OMAP35x-drv-3.01.00.06/services4/system/omap3430/sysutils_linux.c sgx-OMAP35x-drv-3.01.00.06a/services4/system/omap3430/sysutils_linux.c
--- sgx-OMAP35x-drv-3.01.00.06/services4/system/omap3430/sysutils_linux.c	2010-04-28 00:04:05.000000000 +0200
+++ sgx-OMAP35x-drv-3.01.00.06a/services4/system/omap3430/sysutils_linux.c	2010-06-02 16:54:56.122940507 +0200
@@ -447,7 +447,7 @@
 		clk_disable(psSysSpecData->psSGX_FCK);
 		return PVRSRV_ERROR_GENERIC;
 	}
-
+#if 0
 	lNewRate = clk_round_rate(psSysSpecData->psSGX_FCK, SYS_SGX_CLOCK_SPEED + ONE_MHZ);
 	if (lNewRate <= 0)
 	{
@@ -461,7 +461,7 @@
 		PVR_DPF((PVR_DBG_ERROR, "EnableSGXClocks: Couldn't set SGX function clock rate (%d)", res));
 		return PVRSRV_ERROR_GENERIC;
 	}
-
+#endif
 #if defined(DEBUG)
 	{
 		
