diff -Naur sgx-OMAP35x-drv-3.01.00.06/services4/srvkm/devices/sgx/sgxinit.c sgx-OMAP35x-drv-3.01.00.06a/services4/srvkm/devices/sgx/sgxinit.c
--- sgx-OMAP35x-drv-3.01.00.06/services4/srvkm/devices/sgx/sgxinit.c	2010-04-28 00:04:05.000000000 +0200
+++ sgx-OMAP35x-drv-3.01.00.06a/services4/srvkm/devices/sgx/sgxinit.c	2010-06-02 16:55:47.939447110 +0200
@@ -1277,7 +1277,7 @@
 				PVRVERSION_BUILD, psSGXFeatures->ui32DDKBuild));
 	}
 
-
+#if 0
 	
 	ui32BuildOptions = psSGXFeatures->ui32BuildOptions;
 	if (ui32BuildOptions != (SGX_BUILD_OPTIONS))
@@ -1303,7 +1303,7 @@
 	{
 		PVR_DPF((PVR_DBG_WARNING, "(Success) SGXInit: Driver and microkernel build options match."));
 	}
-
+#endif
 #endif
 	eError = PVRSRV_OK;
 exit:
