diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c	2011-02-12 17:36:33.184345532 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/dc_ti81xx_linux/omaplfb_linux.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/dc_ti81xx_linux/omaplfb_linux.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/dc_ti81xx_linux/omaplfb_linux.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/dc_ti81xx_linux/omaplfb_linux.c	2011-02-12 17:36:33.184345532 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/linux_framebuffer/pvrlfb_linux.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/linux_framebuffer/pvrlfb_linux.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/3rdparty/linux_framebuffer/pvrlfb_linux.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/3rdparty/linux_framebuffer/pvrlfb_linux.c	2011-02-12 17:36:33.188345269 +0100
@@ -53,7 +53,7 @@
  **************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/common/resman.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/common/resman.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/common/resman.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/common/resman.c	2011-02-12 17:36:33.147899076 +0100
@@ -29,7 +29,7 @@
 
 #ifdef __linux__
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/event.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/event.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/event.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/event.c	2011-02-12 17:36:33.172337524 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mmap.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mmap.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mmap.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mmap.c	2011-02-12 17:36:33.180343832 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mm.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mm.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mm.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mm.c	2011-02-12 17:36:33.168341532 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mm.h sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mm.h
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mm.h	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mm.h	2011-02-12 17:36:33.172337524 +0100
@@ -28,7 +28,7 @@
 #define __IMG_LINUX_MM_H__
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/module.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/module.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/module.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/module.c	2011-02-12 17:36:33.180343832 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #if !defined(SUPPORT_DRI_DRM)
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mutils.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mutils.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mutils.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mutils.c	2011-02-12 17:36:33.152345377 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 #include <linux/version.h>
 
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mutils.h sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mutils.h
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/mutils.h	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/mutils.h	2011-02-12 17:36:33.176345069 +0100
@@ -28,7 +28,7 @@
 #define __IMG_LINUX_MUTILS_H__
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/osfunc.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/osfunc.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/osfunc.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/osfunc.c	2011-02-12 17:36:33.160332463 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/proc.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/proc.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/proc.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/proc.c	2011-02-12 17:36:33.164345975 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/init.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_debug.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_debug.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_debug.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_debug.c	2011-02-12 17:36:33.160332463 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <asm/io.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_drm.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_drm.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_drm.c	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_drm.c	2011-02-12 17:36:33.152345377 +0100
@@ -27,7 +27,7 @@
 #if defined(SUPPORT_DRI_DRM)
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/init.h>
diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_uaccess.h sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_uaccess.h
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/srvkm/env/linux/pvr_uaccess.h	2011-01-27 09:51:19.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/srvkm/env/linux/pvr_uaccess.h	2011-02-12 17:36:33.147899076 +0100
@@ -28,7 +28,7 @@
 #define __PVR_UACCESS_H__
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
