diff -Naur sgx-OMAP35x-drv-3.01.00.06.orig/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c sgx-OMAP35x-drv-3.01.00.06/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c
--- sgx-OMAP35x-drv-3.01.00.06.orig/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2010-07-28 20:32:32.166488206 +0200
+++ sgx-OMAP35x-drv-3.01.00.06/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2010-07-28 20:33:21.087737283 +0200
@@ -28,6 +28,7 @@
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/fs.h>
+#include <linux/slab.h>
 #include <asm/uaccess.h>
 #include <asm/io.h>
 
