diff -Naur sgx530-OMAP34xx-drv-4.03.00.01/services4/system/omap3630/sysutils_linux.c sgx530-OMAP34xx-drv-4.03.00.01a/services4/system/omap3630/sysutils_linux.c
--- sgx530-OMAP34xx-drv-4.03.00.01/services4/system/omap3630/sysutils_linux.c	2011-01-27 09:51:20.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.03.00.01a/services4/system/omap3630/sysutils_linux.c	2011-02-12 17:44:51.387836134 +0100
@@ -33,7 +33,11 @@
 
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,26))
 #include <linux/semaphore.h>
-#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,31))
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,33))
+#include <linux/semaphore.h>
+#include <asm-generic/resource.h>
+#include <plat/omap-pm.h>
+#elif (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,31))
 #include <plat/resource.h>
 #else 
 #include <mach/resource.h>
