diff -Naur sgx-OMAP35x-drv-4.00.00.01/services4/system/omap3430/sysutils_linux.c sgx-OMAP35x-drv-4.00.00.01a/services4/system/omap3430/sysutils_linux.c
--- sgx-OMAP35x-drv-4.00.00.01/services4/system/omap3430/sysutils_linux.c	2010-09-27 12:30:22.436178930 +0200
+++ sgx-OMAP35x-drv-4.00.00.01a/services4/system/omap3430/sysutils_linux.c	2010-09-27 12:30:54.708218793 +0200
@@ -31,6 +31,12 @@
 #include <linux/spinlock.h>
 #include <asm/bug.h>
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,33))
+#include <linux/semaphore.h>
+#include <asm/resource.h>
+#include <plat/omap-pm.h>
+#else
+
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,31))
 #include <linux/semaphore.h>
 #include <plat/resource.h>
@@ -48,6 +54,7 @@
 #endif
 #endif
 #endif
+#endif
 
 #include "sgxdefs.h"
 #include "services_headers.h"
diff -Naur sgx-OMAP35x-drv-4.00.00.01/services4/system/omap3630/sysutils_linux.c sgx-OMAP35x-drv-4.00.00.01a/services4/system/omap3630/sysutils_linux.c
--- sgx-OMAP35x-drv-4.00.00.01/services4/system/omap3630/sysutils_linux.c	2010-07-30 16:15:01.000000000 +0200
+++ sgx-OMAP35x-drv-4.00.00.01a/services4/system/omap3630/sysutils_linux.c	2010-09-27 12:31:06.416468840 +0200
@@ -32,6 +32,12 @@
 #include <asm/bug.h>
 #include <linux/platform_device.h>
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,33))
+#include <linux/semaphore.h>
+#include <asm/resource.h>
+#include <plat/omap-pm.h>
+#else
+
 #if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,31))
 #include <linux/semaphore.h>
 #include <plat/resource.h>
@@ -49,6 +55,7 @@
 #endif
 #endif
 #endif
+#endif
 
 #if	(LINUX_VERSION_CODE >  KERNEL_VERSION(2,6,27)) && \
 	(LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,29))
