diff -Naur sgx-OMAP35x-drv-4.00.00.01/eurasiacon/build/linux/kbuild/Makefile.kbuild_subdir_common sgx-OMAP35x-drv-4.00.00.01a/eurasiacon/build/linux/kbuild/Makefile.kbuild_subdir_common
--- sgx-OMAP35x-drv-4.00.00.01/eurasiacon/build/linux/kbuild/Makefile.kbuild_subdir_common	2010-03-05 10:51:10.000000000 +0100
+++ sgx-OMAP35x-drv-4.00.00.01a/eurasiacon/build/linux/kbuild/Makefile.kbuild_subdir_common	2010-09-27 12:35:22.972203619 +0200
@@ -30,7 +30,7 @@
 
 EXTRA_CFLAGS = $(INCLUDES)  \
 				$(ALL_CFLAGS_kbuild)	\
-				-Wall -fno-strict-aliasing -Werror
+				-Wall -fno-strict-aliasing
 
 ifeq ($(SILENT),@)
 	KBUILD_VERBOSE ?= 0
