diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c	2010-09-08 08:06:41.000000000 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c	2011-02-15 14:36:34.869612290 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_ti8168_linux/omaplfb_linux.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_ti8168_linux/omaplfb_linux.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_ti8168_linux/omaplfb_linux.c	2010-09-08 09:55:22.000000000 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_ti8168_linux/omaplfb_linux.c	2011-02-15 14:36:34.865620365 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/common/resman.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/common/resman.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/common/resman.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/common/resman.c	2011-02-15 14:36:34.757616481 +0100
@@ -29,7 +29,7 @@
 
 #ifdef __linux__
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/event.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/event.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/event.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/event.c	2011-02-15 14:36:34.825623442 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mmap.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mmap.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mmap.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mmap.c	2011-02-15 14:36:34.845613561 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mm.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mm.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mm.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mm.c	2011-02-15 14:36:34.817613082 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mm.h sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mm.h
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mm.h	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mm.h	2011-02-15 14:36:34.833657871 +0100
@@ -28,7 +28,7 @@
 #define __IMG_LINUX_MM_H__
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/module.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/module.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/module.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/module.c	2011-02-15 14:36:34.857656825 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/init.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mutils.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mutils.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mutils.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mutils.c	2011-02-15 14:36:34.781492371 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 #include <linux/version.h>
 
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mutils.h sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mutils.h
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/mutils.h	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/mutils.h	2011-02-15 14:36:34.841612602 +0100
@@ -28,7 +28,7 @@
 #define __IMG_LINUX_MUTILS_H__
 
 #ifndef AUTOCONF_INCLUDED
-#include <linux/config.h>
+
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/osfunc.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/osfunc.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/osfunc.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/osfunc.c	2011-02-15 14:36:34.793654582 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/version.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/proc.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/proc.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/proc.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/proc.c	2011-02-15 14:36:34.809621635 +0100
@@ -25,7 +25,7 @@
  ******************************************************************************/
 
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <linux/init.h>
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/pvr_debug.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/pvr_debug.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/pvr_debug.c	2010-03-05 10:47:57.000000000 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/srvkm/env/linux/pvr_debug.c	2011-02-15 14:36:34.797623066 +0100
@@ -26,7 +26,7 @@
 
   
 #ifndef AUTOCONF_INCLUDED
- #include <linux/config.h>
+ 
 #endif
 
 #include <asm/io.h>
