diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c	2010-09-08 08:06:41.000000000 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_omap3430_linux/omaplfb_displayclass.c	2011-02-15 15:00:18.917539784 +0100
@@ -235,9 +235,17 @@
 {
 	int res;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
 	res = fb_blank(psDevInfo->psLINFBInfo, 0);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 #if !defined (CONFIG_OMAP2_DSS)
 	/* DSS2 returns error if unblank from a non-suspend state */
 	if (res != 0)
@@ -1113,7 +1121,11 @@
 	OMAP_ERROR eError = OMAP_ERROR_GENERIC;
 	unsigned long FBSize;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
 
 	if (fb_idx < 0 || fb_idx >= num_registered_fb)
 	{
@@ -1253,7 +1265,11 @@
 errModPut:
 	module_put(psLINFBOwner);
 errRelSem:
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 	return eError;
 }
 
@@ -1262,7 +1278,11 @@
 	struct fb_info *psLINFBInfo = psDevInfo->psLINFBInfo;
 	struct module *psLINFBOwner;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
 
 	psLINFBOwner = psLINFBInfo->fbops->owner;
 
@@ -1273,7 +1293,11 @@
 
 	module_put(psLINFBOwner);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 }
 
 OMAP_ERROR OMAPLFBInit(void)
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_ti8168_linux/omaplfb_displayclass.c sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_ti8168_linux/omaplfb_displayclass.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/dc_ti8168_linux/omaplfb_displayclass.c	2010-09-08 16:44:31.000000000 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01a/services4/3rdparty/dc_ti8168_linux/omaplfb_displayclass.c	2011-02-15 14:58:33.717546226 +0100
@@ -235,9 +235,18 @@
 {
 	int res;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
+
 	res = fb_blank(psDevInfo->psLINFBInfo, 0);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 #if !defined (CONFIG_OMAP2_DSS)
 	/* DSS2 returns error if unblank from a non-suspend state */
 	if (res != 0)
@@ -1113,7 +1122,11 @@
 	OMAP_ERROR eError = OMAP_ERROR_GENERIC;
 	unsigned long FBSize;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
 
 	if (fb_idx < 0 || fb_idx >= num_registered_fb)
 	{
@@ -1263,7 +1276,11 @@
 errModPut:
 	module_put(psLINFBOwner);
 errRelSem:
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 	return eError;
 }
 
@@ -1272,7 +1289,11 @@
 	struct fb_info *psLINFBInfo = psDevInfo->psLINFBInfo;
 	struct module *psLINFBOwner;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_lock();
+#else
 	acquire_console_sem();
+#endif
 
 	psLINFBOwner = psLINFBInfo->fbops->owner;
 
@@ -1283,7 +1304,11 @@
 
 	module_put(psLINFBOwner);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,38)
+	console_unlock();
+#else
 	release_console_sem();
+#endif
 }
 
 OMAP_ERROR OMAPLFBInit(void)
