diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c sgx530-OMAP34xx-drv-4.00.00.01a//services4/3rdparty/bufferclass_example/bufferclass_example_linux.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2011-02-15 15:09:47.521987777 +0100
+++ sgx530-OMAP34xx-drv-4.00.00.01a//services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2011-02-15 15:11:28.469544583 +0100
@@ -56,14 +56,14 @@
 
 MODULE_SUPPORTED_DEVICE(DEVNAME);
 
-int BC_Example_Bridge(struct inode *inode, struct file *file, unsigned int cmd, unsigned long arg);
+int BC_Example_Bridge(struct file *file, unsigned int cmd, unsigned long arg);
 int FillBuffer(unsigned int uiBufferIndex);
 int GetBufferCount(unsigned int *puiBufferCount);
 
 static int AssignedMajorNumber;
 
 static struct file_operations bufferclass_example_fops = {
-	ioctl:BC_Example_Bridge,
+	unlocked_ioctl:BC_Example_Bridge,
 };
 
 
@@ -307,7 +307,7 @@
 }
 
 
-int BC_Example_Bridge(struct inode *inode, struct file *file, unsigned int cmd, unsigned long arg)
+int BC_Example_Bridge(struct file *file, unsigned int cmd, unsigned long arg)
 {
 	int err = -EFAULT;
 	int command = _IOC_NR(cmd);
