diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/osfunc.c sgx530-OMAP34xx-drv-4.00.00.01_patched//services4/srvkm/env/linux/osfunc.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/osfunc.c	2011-06-09 19:07:33.839175387 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01_patched//services4/srvkm/env/linux/osfunc.c	2011-06-09 22:19:37.546115146 +0200
@@ -1516,7 +1516,11 @@
 
 static TIMER_CALLBACK_DATA sTimers[OS_MAX_TIMERS];
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
+static spinlock_t sTimerStructLock = __SPIN_LOCK_UNLOCKED(sTimerStructLock);
+#else
 static spinlock_t sTimerStructLock = SPIN_LOCK_UNLOCKED;
+#endif
 
 static IMG_VOID OSTimerCallbackWrapper(IMG_UINT32 ui32Data)
 {
diff -Naur sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/pvr_debug.c sgx530-OMAP34xx-drv-4.00.00.01_patched//services4/srvkm/env/linux/pvr_debug.c
--- sgx530-OMAP34xx-drv-4.00.00.01/services4/srvkm/env/linux/pvr_debug.c	2011-06-09 19:07:33.839175387 +0200
+++ sgx530-OMAP34xx-drv-4.00.00.01_patched//services4/srvkm/env/linux/pvr_debug.c	2011-06-09 22:20:39.001539437 +0200
@@ -56,7 +56,11 @@
 
 static PVRSRV_LINUX_MUTEX gsDebugMutexNonIRQ;
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,39))
+static spinlock_t gsDebugLockIRQ = __SPIN_LOCK_UNLOCKED(gsDebugLockIRQ);
+#else
 static spinlock_t gsDebugLockIRQ = SPIN_LOCK_UNLOCKED;
+#endif
 
 #define	USE_SPIN_LOCK (in_interrupt() || !preemptible())
 
