diff -Naur bftpd.orig/Makefile.in bftpd/Makefile.in
--- bftpd.orig/Makefile.in	2010-08-14 12:32:29.184780509 +0200
+++ bftpd/Makefile.in	2010-08-14 12:32:42.554781450 +0200
@@ -26,10 +26,13 @@
 $(OBJS): $(HEADERS) Makefile
 
 install: all
-	$(INSTALL) -g 0 -m 700 -o 0 bftpd $(DESTDIR)/$(prefix)/sbin
-	$(INSTALL) -g 0 -m 644 -o 0 bftpd.8 $(DESTDIR)/$(mandir)/man8
+	mkdir -p $(DESTDIR)/$(prefix)/sbin
+	$(INSTALL) -m 700 bftpd $(DESTDIR)/$(prefix)/sbin
+	mkdir -p $(DESTDIR)/$(mandir)/man8
+	$(INSTALL) -m 644 bftpd.8 $(DESTDIR)/$(mandir)/man8
+	mkdir -p $(DESTDIR)/etc
 	[ -f $(DESTDIR)/etc/bftpd.conf ] || \
-		$(INSTALL) -g 0 -m 600 -o 0 bftpd.conf $(DESTDIR)/etc
+		$(INSTALL) -m 600 bftpd.conf $(DESTDIR)/etc
 	mkdir -p $(DESTDIR)/var/log
 	touch $(DESTDIR)/var/log/bftpd.log
 	chmod 644 $(DESTDIR)/var/log/bftpd.log
