diff -pur bftpd-1.5.orig/commands.c bftpd-1.5/commands.c
--- bftpd-1.5.orig/commands.c	2006-04-25 02:42:36.000000000 +0200
+++ bftpd-1.5/commands.c	2006-08-14 18:23:53.000000000 +0200
@@ -166,17 +166,14 @@ int dataconn()
 
 void init_userinfo()
 {
-    struct passwd *temp = getpwnam(user);
-    if (temp) {
-        userinfo.pw_name = strdup(temp->pw_name);
-        userinfo.pw_passwd = strdup(temp->pw_passwd);
-        userinfo.pw_uid = temp->pw_uid;
-        userinfo.pw_gid = temp->pw_gid;
-        userinfo.pw_gecos = strdup(temp->pw_gecos);
-        userinfo.pw_dir = strdup(temp->pw_dir);
-        userinfo.pw_shell = strdup(temp->pw_shell);
-        userinfo_set = 1;
-    }
+    userinfo.pw_name = strdup("geexbox");
+    userinfo.pw_passwd = strdup("ftp");
+    userinfo.pw_uid = 0;
+    userinfo.pw_gid = 0;
+    userinfo.pw_gecos = strdup("");
+    userinfo.pw_dir = strdup("/");
+    userinfo.pw_shell = strdup("/bin/sh");
+    userinfo_set = 1;
 }
 
 void command_user(char *username)
Seulement dans bftpd-1.5/: commands.c.orig
diff -pur bftpd-1.5.orig/login.c bftpd-1.5/login.c
--- bftpd-1.5.orig/login.c	2006-07-04 00:20:52.000000000 +0200
+++ bftpd-1.5/login.c	2006-08-14 18:25:06.000000000 +0200
@@ -193,10 +193,6 @@ int bftpd_login(char *password)
 
         if (! file_auth[0] )    /* not using auth file */
         {
-	   if (!getpwnam(user)) {
-                control_printf(SL_FAILURE, "421 Login incorrect.");
-		exit(0);
-           }
         }
         /* we are using auth_file */
         else
@@ -243,22 +239,6 @@ int bftpd_login(char *password)
             exit(0);
         }
        
-        /* disbale these checks when logging in via auth file */
-        if (! file_auth[0] ) 
-        {
-	    if(checkuser() || checkshell()) {
-		control_printf(SL_FAILURE, "421 Login incorrect.");
-		exit(0);
-	    }
-        }
-
-        /* do not do this check we we are using auth_file */
-        if (! file_auth[0] )
-        {
-	    if (checkpass(password))
-		return 1;
-        }
-
 	if (strcasecmp((char *) config_getoption("RATIO"), "none")) {
 		sscanf((char *) config_getoption("RATIO"), "%i/%i",
 			   &ratio_send, &ratio_recv);
