diff -Naur bftpd-1.0.24.orig/commands.c bftpd-1.0.24/commands.c
--- bftpd-1.0.24.orig/commands.c	2003-09-17 17:39:02.000000000 +0200
+++ bftpd-1.0.24/commands.c	2003-09-17 17:40:22.000000000 +0200
@@ -164,17 +164,14 @@
 
 void init_userinfo()
 {
-    struct passwd *temp = getpwnam(user);
-    if (temp) {
-        userinfo.pw_name = strdup(temp->pw_name);
-        userinfo.pw_passwd = strdup(temp->pw_passwd);
-        userinfo.pw_uid = temp->pw_uid;
-        userinfo.pw_gid = temp->pw_gid;
-        userinfo.pw_gecos = strdup(temp->pw_gecos);
-        userinfo.pw_dir = strdup(temp->pw_dir);
-        userinfo.pw_shell = strdup(temp->pw_shell);
-        userinfo_set = 1;
-    }
+    userinfo.pw_name = strdup("geexbox");
+    userinfo.pw_passwd = strdup("ftp");
+    userinfo.pw_uid = 0;
+    userinfo.pw_gid = 0;
+    userinfo.pw_gecos = strdup("");
+    userinfo.pw_dir = strdup("/");
+    userinfo.pw_shell = strdup("/bin/sh");
+    userinfo_set = 1;
 }
 
 void command_user(char *username)
diff -Naur bftpd-1.0.24.orig/login.c bftpd-1.0.24/login.c
--- bftpd-1.0.24.orig/login.c	2003-09-17 17:39:02.000000000 +0200
+++ bftpd-1.0.24/login.c	2003-09-17 17:41:37.000000000 +0200
@@ -186,10 +186,6 @@
 	char str[256];
 	char *foo;
 	int maxusers;
-	if (!getpwnam(user)) {
-        control_printf(SL_FAILURE, "421 Login incorrect.");
-		exit(0);
-    }
 	if (strncasecmp(foo = config_getoption("DENY_LOGIN"), "no", 2)) {
 		if (foo[0] != '\0') {
 			if (strncasecmp(foo, "yes", 3))
@@ -210,12 +206,6 @@
 		control_printf(SL_FAILURE, "421 User %s is already logged in %i times.", user, maxusers);
 		exit(0);
 	}
-	if(checkuser() || checkshell()) {
-		control_printf(SL_FAILURE, "421 Login incorrect.");
-		exit(0);
-	}
-	if (checkpass(password))
-		return 1;
 	if (strcasecmp((char *) config_getoption("RATIO"), "none")) {
 		sscanf((char *) config_getoption("RATIO"), "%i/%i",
 			   &ratio_send, &ratio_recv);
