diff -Naur bftpd.orig/options.c bftpd/options.c
--- bftpd.orig/options.c	2010-04-18 23:45:28.000000000 +0200
+++ bftpd/options.c	2010-09-05 00:35:21.676059971 +0200
@@ -13,7 +13,7 @@
 
 struct global config_global;
 struct group_of_users *config_groups;
-struct user *config_users;
+struct _user *config_users;
 
 /*
 Returns NULL on error. May return
@@ -165,7 +165,7 @@
 	FILE *configfile;
 	char *str;
     struct group_of_users *grp = NULL;
-    struct user *usr = NULL;
+    struct _user *usr = NULL;
     config_global.options = NULL;
     config_global.directories = NULL;
 	if (!configpath)
@@ -185,9 +185,9 @@
                 create_options(configfile, &(config_global.options), &(config_global.directories));
             } else if (strstr(str, "user ") == str) {
                 if (usr) {
-                    usr = usr->next = malloc(sizeof(struct user));
+                    usr = usr->next = malloc(sizeof(struct _user));
                 } else {
-                    config_users = usr = malloc(sizeof(struct user));
+                    config_users = usr = malloc(sizeof(struct _user));
                 }
                 /* avoid memory errors */
                 if (! usr)
@@ -325,7 +325,7 @@
 char *getoption_user(char *name)
 {
     char *result;
-    struct user *usr;
+    struct _user *usr;
     if ((usr = config_users)) {
         do {
             if (!strcmp(user, usr->name)) {
diff -Naur bftpd.orig/options.h bftpd/options.h
--- bftpd.orig/options.h	2006-02-17 02:48:06.000000000 +0100
+++ bftpd/options.h	2010-09-05 00:35:21.676059971 +0200
@@ -48,15 +48,15 @@
     struct directory *directories;
     struct group_of_users *next;
 };
-struct user {
+struct _user {
     char *name;
     struct bftpd_option *options;
     struct directory *directories;
-    struct user *next;
+    struct _user *next;
 };
 
 extern struct group_of_users *config_groups;
-extern struct user *config_users;
+extern struct _user *config_users;
 
 void expand_groups();
 void config_init();
