From 0dd6a547aa07370b5c5157c8b93ee2614c501498 Mon Sep 17 00:00:00 2001
From: Alexander Huemer <a.huemer@bct-electronic.com>
Date: Mon, 23 Jan 2012 11:01:43 +0100
Subject: [PATCH] main should return 0 here

---
 src/make_dct4_tables.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/make_dct4_tables.c b/src/make_dct4_tables.c
index 2ccd4f1..76c54e3 100644
--- a/src/make_dct4_tables.c
+++ b/src/make_dct4_tables.c
@@ -99,6 +99,8 @@ int main(int argc, char *argv[])
 
     for (i = 0;  i <= length_log;  i++)
         set_up_one_table(dct_size << i);
+
+    return 0;
 }
 /*- End of function --------------------------------------------------------*/
 /*- End of file ------------------------------------------------------------*/
-- 
1.7.4.1

