diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
@@ -438,6 +438,7 @@
 			ARRAY_SIZE(omap3_beagle_devices));
 	omap_serial_init();
 
+        printk(KERN_ERR "Booting using BeagleBoard platform mapping ...\n");
 	omap_mux_init_gpio(170, OMAP_PIN_INPUT);
 	gpio_request(170, "DVI_nPD");
 	/* REVISIT leave DVI powered down until it's needed ... */
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
@@ -536,6 +536,7 @@
 			ARRAY_SIZE(omap3_touchbook_devices));
 	omap_serial_init();
 
+        printk(KERN_ERR "Booting using TouchBook platform mapping ...\n");
 	omap_mux_init_gpio(170, OMAP_PIN_INPUT);
 	gpio_request(176, "DVI_nPD");
 	/* REVISIT leave DVI powered down until it's needed ... */
