diff -Naur a/setup.py b/setup.py
--- a/setup.py	2011-02-20 16:20:13.715171835 +0100
+++ b/setup.py	2011-02-20 16:20:01.251693111 +0100
@@ -57,7 +57,7 @@
 
 import os.path, glob, stat
 import distutils.sysconfig
-from distutils.core import setup, Extension, Command
+from setuptools import setup, Extension, Command
 from distutils.extension import read_setup_file
 from distutils.command.install_data import install_data
 
