diff -Naur a/config_unix.py b/config_unix.py
--- a/config_unix.py	2011-02-20 16:23:54.527741242 +0100
+++ b/config_unix.py	2011-02-20 16:24:46.587171739 +0100
@@ -54,7 +54,7 @@
                 elif f[:3] == '-Wl':
                     self.cflags += '-Xlinker ' + f + ' '
             if self.name == 'SDL':
-                inc = '-I' + '/usr/X11R6/include'
+                inc = ''
                 self.cflags = inc + ' ' + self.cflags
         except:
             print ('WARNING: "%s" failed!' % command)
@@ -159,10 +158,6 @@
     else:
         incdirs = []
         libdirs = []
-    incdirs += ["/usr"+d for d in origincdirs]
-    libdirs += ["/usr"+d for d in origlibdirs]
-    incdirs += ["/usr/local"+d for d in origincdirs]
-    libdirs += ["/usr/local"+d for d in origlibdirs]
 
     for arg in DEPS[0].cflags.split():
         if arg[:2] == '-I':
