diff -Naur pygame-1.7.1release.orig/config_unix.py pygame-1.7.1release/config_unix.py
--- pygame-1.7.1release.orig/config_unix.py	2006-11-12 18:18:32.000000000 +0100
+++ pygame-1.7.1release/config_unix.py	2006-11-12 18:19:04.000000000 +0100
@@ -168,14 +168,6 @@
         d.configure(incdirs, libdirs)
 
 
-    for d in DEPS[1:]:
-        if not d.found:
-            if not confirm("""
-Warning, some of the pygame dependencies were not found. Pygame can still
-compile and install, but games that depend on those missing dependencies
-will not run. Would you like to continue the configuration?"""):
-                raise SystemExit
-            break
 
     return DEPS
 
