diff -Naur opkg-0.1.8/libopkg/pkg_hash.c opkg-0.1.8a/libopkg/pkg_hash.c
--- opkg-0.1.8/libopkg/pkg_hash.c	2009-12-21 02:46:09.000000000 +0100
+++ opkg-0.1.8a/libopkg/pkg_hash.c	2010-06-26 16:00:56.990733518 +0200
@@ -105,7 +105,7 @@
 			continue;
 		}
 
-		if (!pkg->architecture || !pkg->arch_priority) {
+		if (!conf->offline_root && (!pkg->architecture || !pkg->arch_priority)) {
 			char *version_str = pkg_version_str_alloc(pkg);
 			opkg_msg(ERROR, "Package %s version %s has no "
 					"valid architecture, ignoring.\n",
@@ -276,7 +276,7 @@
 				 maybe->arch_priority, maybe->version);
                     /* We make sure not to add the same package twice. Need to search for the reason why 
                        they show up twice sometimes. */
-		    if ((maybe->arch_priority > 0) && (! pkg_vec_contains(matching_pkgs, maybe))) {
+		    if (((maybe->arch_priority > 0) || conf->offline_root) && (! pkg_vec_contains(matching_pkgs, maybe))) {
 			 max_count++;
 			 abstract_pkg_vec_insert(matching_apkgs, maybe->parent);
 			 pkg_vec_insert(matching_pkgs, maybe);
diff -Naur opkg-0.1.8/libopkg/pkg.c opkg-0.1.8a/libopkg/pkg.c
--- opkg-0.1.8/libopkg/pkg.c	2010-01-12 05:07:43.000000000 +0100
+++ opkg-0.1.8a/libopkg/pkg.c	2010-06-26 16:16:40.598990809 +0200
@@ -1306,6 +1306,9 @@
      if (!pkg->architecture)
 	  return 1;
 
+     if (conf->offline_root)
+	  return 1;
+
      list_for_each_entry(l , &conf->arch_list.head, node) {
 	  nv_pair_t *nv = (nv_pair_t *)l->data;
 	  if (strcmp(nv->name, pkg->architecture) == 0) {
