diff -Naur opkg-0.1.8/libopkg/opkg_install.c opkg-0.1.8a//libopkg/opkg_install.c
--- opkg-0.1.8/libopkg/opkg_install.c	2010-02-01 00:16:55.000000000 +0100
+++ opkg-0.1.8a//libopkg/opkg_install.c	2011-02-03 14:05:09.231220469 +0100
@@ -321,7 +321,7 @@
 			found = 0;
 
 			for (k=0; k<count1; k++) {
-				cd1 = &pkg->depends[i];
+				cd1 = &pkg->depends[k];
 				if (cd1->type != DEPEND)
 					continue;
 				for (l=0; l<cd1->possibility_count; l++) {
