--- at-3.1.14/Makefile.in.org	2014-09-18 15:27:07.890894628 +0200
+++ at-3.1.14/Makefile.in	2014-09-18 15:27:52.950893883 +0200
@@ -3,7 +3,7 @@
 VPATH 		= $(srcdir)
 
 install_prefix	=
-prefix		= @prefix@
+prefix		=  ./.install
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
 sbindir		= @sbindir@
@@ -14,7 +14,7 @@
 man8dir		= $(mandir)/man8
 docdir		= $(prefix)/doc
 atdocdir	= $(docdir)/at
-etcdir		= @ETCDIR@
+etcdir		= ./.install/etc
 systemdsystemunitdir = @systemdsystemunitdir@
 
 DAEMON_USERNAME	= @DAEMON_USERNAME@
