diff -Naur osdpip-0.0.10a/decoder.c osdpip-0.0.10a.new/decoder.c
--- osdpip-0.0.10a/decoder.c	2010-01-01 22:24:06.000000000 +0100
+++ osdpip-0.0.10a.new/decoder.c	2010-01-02 14:11:31.914298872 +0100
@@ -82,14 +82,14 @@
     context = sws_getContext(m_Context->width - (OsdPipSetup.CropLeft + OsdPipSetup.CropRight),
                              m_Context->height - (OsdPipSetup.CropTop + OsdPipSetup.CropBottom),
                              PIX_FMT_YUV420P,
-                             m_Width, m_Height, ConvertToRGB ? PIX_FMT_RGBA32 : PIX_FMT_YUV420P,
+                             m_Width, m_Height, ConvertToRGB ? PIX_FMT_RGB32 : PIX_FMT_YUV420P,
                              SWS_LANCZOS, NULL, NULL, NULL);
     if (!context) {
         printf("Error initializing scale context.\n");
         return -1;
     }
     avpicture_fill((AVPicture *) m_PicResample, m_BufferResample,
-                   ConvertToRGB ? PIX_FMT_RGBA32 : PIX_FMT_YUV420P,
+                   ConvertToRGB ? PIX_FMT_RGB32 : PIX_FMT_YUV420P,
                    m_Width, m_Height);
     sws_scale(context, pic_crop.data, pic_crop.linesize,
               0, m_Context->height - (OsdPipSetup.CropTop + OsdPipSetup.CropBottom),
