--- vdr-fritz-1.5.3/Makefile.org	2014-01-20 16:58:58.914013104 +0100
+++ vdr-fritz-1.5.3/Makefile	2014-01-20 16:54:25.742008416 +0100
@@ -61,7 +61,7 @@
 
 STATIC_LIB_DIRS = $(dir $(LIBS))
 STATIC_LIBS     = $(LIBS:%=$(CURDIR)/%)
-CXXFLAGS       += -I$(CURDIR) -std=c++11
+CXXFLAGS       += -I$(CURDIR) -std=c++11 -D_GLIBCXX_USE_NANOSLEEP
 LDFLAGS        += -lboost_system -lboost_thread -lboost_regex -lpthread -lgcrypt
 export STATIC_LIBS CXXFLAGS LDFLAGS
 
--- vdr-fritzbox-1.5.3/Makefile.org	2014-01-20 17:41:19.418056715 +0100
+++ vdr-fritzbox-1.5.3/Makefile	2014-01-20 17:41:37.154057018 +0100
@@ -86,7 +86,7 @@
 all: $(SOFILE) i18n
 
 $(SOFILE): $(OBJS) $(LIBS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) $(LIBS) $(LDFLAGS) -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -shared $(OBJS) $(LIBS) -o $@
 
 %.a: $(STATIC_LIB_DIRS)
 	@
