diff -Naur libpbc-0.5.12_orig//misc/extend_printf.c libpbc-0.5.12/misc/extend_printf.c
--- libpbc-0.5.12_orig//misc/extend_printf.c	2011-06-02 21:11:46.000000000 +0200
+++ libpbc-0.5.12/misc/extend_printf.c	2012-05-25 18:12:54.529584806 +0200
@@ -68,7 +68,7 @@
           break;
         default:
           if (strchr("diouxXeEfFgGaAcspnm", *c)) {
-            void *ptr = va_arg(ap, void *);
+            va_list ptr = va_arg(ap, va_list);
             ch = *(c+1);
             *(c+1) = '\0';
             //status = gmp_fprintf(stream, start, ptr);
