--- /root/openbricks/build/build.i386.eglibc/serdisplib-1.97.8/Makefile.in	2008-07-30 23:05:37.000000000 +0200
+++ Makefile.in	2010-04-11 16:52:05.000000000 +0200
@@ -1,5 +1,5 @@
 # Makefile for project 'serdisplib', root-directory
-# (C) 2003-2008 by Wolfgang Astleitner
+# (C) 2003-2010 by Wolfgang Astleitner
 
 
 SHELL = /bin/sh
@@ -18,10 +18,18 @@
 
 CC = @CC@
 MAKE=@MAKE@
+INSTALL=@INSTALL@
+INSTALL_PROGRAM = @INSTALL_PROGRAM@
+INSTALL_SCRIPT = @INSTALL_SCRIPT@
+INSTALL_DATA = @INSTALL_DATA@
 
 
 subdirs = src tools
 
+ifeq "@WITH_DRIVER_REMOTE@" "true"
+  subdirs += server
+endif
+
 
 all:
 	@for dir in ${subdirs}; do \
@@ -30,12 +38,15 @@
 	done && test -z "$$fail"
 
 install:
+	$(top_srcdir)/mkinstalldirs $(libdir)/pkgconfig
+	$(INSTALL_DATA) serdisplib.pc $(libdir)/pkgconfig/
 	@for dir in ${subdirs}; do \
 	  (cd $(srcdir) && cd $$dir && $(MAKE) install) \
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
 	done && test -z "$$fail"
 
 uninstall:
+	-/bin/rm -f $(libdir)/pkgconfig/serdisplib.pc
 	@for dir in ${subdirs}; do \
 	  (cd $(srcdir) && cd $$dir && $(MAKE) uninstall) \
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
@@ -50,7 +61,7 @@
 
 distclean: clean
 	-/bin/rm -f autom4te.cache/* && /bin/rmdir autom4te.cache
-	/bin/rm -f Makefile config.h config.status config.cache config.log
+	/bin/rm -f Makefile config.h config.status config.cache config.log tags
 	@for dir in ${subdirs}; do \
 	  (cd $(srcdir) && cd $$dir && $(MAKE) distclean) \
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
@@ -59,7 +70,7 @@
 # make snapshot is meant for internal use only
 snapshot: distclean
 	cd $(top_srcdir)/.. && \
-	/bin/gtar c --exclude "*~" --exclude ".svn" --exclude "temp" --exclude "*_bkp*" $(base_abs_top_srcdir) | /bin/gzip -c > $(base_abs_top_srcdir)_$(current_timestamp).tar.gz
+	/bin/gtar c --exclude "*~" --exclude ".svn" --exclude "temp" --exclude "*_bkp*" --exclude "*.o" --exclude serdisplib.pc $(base_abs_top_srcdir) | /bin/gzip -c > $(base_abs_top_srcdir)_$(current_timestamp).tar.gz
 
 
 # automatic re-running of configure if the configure.in file has changed
