Index: alsa-lib-1.0.23/src/conf/alsa.conf
===================================================================
--- alsa-lib-1.0.23.orig/src/conf/alsa.conf	2010-10-03 15:05:47.000000000 -0400
+++ alsa-lib-1.0.23/src/conf/alsa.conf	2010-10-03 15:09:27.000000000 -0400
@@ -54,11 +54,11 @@
 #
 
 # show all name hints also for definitions without hint {} section
-defaults.namehint.showall off
+defaults.namehint.showall on
 # show just basic name hints
 defaults.namehint.basic on
 # show extended name hints
-defaults.namehint.extended off
+defaults.namehint.extended on
 #
 defaults.ctl.card 0
 defaults.pcm.card 0
