--- a/include/alsa-symbols.h	2014-06-17 14:34:19.000000000 +0200
+++ b/include/alsa-symbols.h	2017-04-11 14:55:21.348473773 +0200
@@ -26,8 +26,12 @@
 #define USE_VERSIONED_SYMBOLS
 #endif
 
+#ifdef VERSIONED_SYMBOLS
 #define INTERNAL_CONCAT2_2(Pre, Post) Pre##Post
 #define INTERNAL(Name) INTERNAL_CONCAT2_2(__, Name)
+#else
+#define INTERNAL(Name) Name
+#endif
 
 #ifdef __powerpc64__
 # define symbol_version(real, name, version) 			\
@@ -61,10 +65,12 @@
 #define use_default_symbol_version(real, name, version) \
         __asm__ (".weak " ASM_NAME(#name)); \
         __asm__ (ASM_NAME(#name) " = " ASM_NAME(#real))
-#else
+#elif defined(INTERNAL_CONCAT2_2)
 #define use_default_symbol_version(real, name, version) \
 	__asm__ (".weak " ASM_NAME(#name)); \
 	__asm__ (".set " ASM_NAME(#name) "," ASM_NAME(#real))
+#else
+#define use_default_symbol_version(real, name, version)
 #endif
 #endif
 #endif
