--- vdr-1.7.42/config.h	2013-03-24 10:58:22.403521392 +0100
+++ vdr-1.7.42/config.h.org	2013-03-24 10:58:09.631521174 +0100
@@ -28,8 +28,8 @@
 
 // The plugin API's version number:
 
-#define APIVERSION  "1.7.41"
-#define APIVERSNUM   10741  // Version * 10000 + Major * 100 + Minor
+#define APIVERSION  "1.7.42"
+#define APIVERSNUM   10742  // Version * 10000 + Major * 100 + Minor
 
 // When loading plugins, VDR searches them by their APIVERSION, which
 // may be smaller than VDRVERSION in case there have been no changes to
