--- a/CMakeLists.txt	2019-11-10 19:51:18.015644870 +0100
+++ b/CMakeLists.txt	2019-11-10 19:52:17.863212673 +0100
@@ -9,11 +9,11 @@

 find_package(Kodi REQUIRED)
 find_package(p8-platform REQUIRED)
-find_package(OpenGL REQUIRED)
+#find_package(OpenGL REQUIRED)

 find_package(ProjectM REQUIRED)

-add_definitions(-DHAS_SDL_OPENGL)
+add_definitions(-DHAS_GLES)

 set(INCLUDES ${OpenGL_INCLUDE_DIR}
              ${KODI_INCLUDE_DIR}
