--- a/src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformKeyboardEventQt.cpp	2015-05-07 16:14:47.000000000 +0200
+++ b/src/3rdparty/webkit/Source/WebCore/platform/qt/PlatformKeyboardEventQt.cpp	2024-10-10 13:50:41.000000000 +0200
@@ -587,6 +587,7 @@
             return "\t";
         break;
     case Qt::Key_Enter:
+    case Qt::Key_Return:
         if (event->text().isNull())
             return "\r";
     }
