--- lirc-0.7.1.orig/configure	2005-06-04 16:12:28.596987032 +0800
+++ lirc-0.7.1/configure	2005-06-04 16:14:16.167633808 +0800
@@ -8504,7 +8504,7 @@
 #define LIRC_DRIVER_ANY 1
 EOF
 
-  lirc_driver="none"
+  lirc_driver=""
   #include creative driver
   hw_module="${hw_module} hw_creative.o serial.o"
   
@@ -8566,18 +8566,20 @@
 		#hw_module="${hw_module} hw_irreal.o serial.o receive.o transmit.o"
 		;;
 	lirc_dev-lirc_atiusb)
+		lirc_driver="${lirc_driver} lirc_dev lirc_atiusb"
 		;;
 	lirc_dev-lirc_gpio)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
+		lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
 		;;
 	lirc_dev-lirc_i2c)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
+		lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
 		;;
 	lirc_dev-lirc_igorplugusb)
 		;;
 	lirc_dev-lirc_imon)
 		;;
 	lirc_dev-lirc_it87)
+		lirc_driver="${lirc_driver} lirc_dev lirc_it87"
 		;;
 	lirc_dev-lirc_mceusb)
 		;;
@@ -8586,8 +8588,10 @@
 	lirc_dev-lirc_sasem)
 		;;
 	lirc_dev-lirc_serial)
+		lirc_driver="${lirc_driver} lirc_serial"
 		;;
 	lirc_dev-lirc_sir)
+		lirc_driver="${lirc_driver} lirc_sir"
 		;;
 	lirc_dev-lirc_streamzap)
 		;;
@@ -8644,11 +8648,14 @@
   done
   #remove duplicates
 
-  #sed segfaults when $hw_module is too long :(
-  #hw_module=`echo ${hw_module}| sed \
-  #			-e':again' \
-  #			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
-  #			-e'tagain'`
+  hw_module=`echo ${hw_module}| sed \
+  			-e':again' \
+  			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+  			-e'tagain'`
+  lirc_driver=`echo ${lirc_driver}| sed \
+  			-e':again' \
+			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+			-e'tagain'`
 
   space_to_newline()
   { for token in $*; do echo $token; done; }
@@ -9205,7 +9212,7 @@
   { echo "configure: error: no driver specified, try ./configure --help" 1>&2; exit 1; }
 fi
 
-if echo ${possible_drivers} | grep "($lirc_driver)" >/dev/null 2>&1; then
+if true; then
   dummy=
 else
   case $lirc_driver in
--- lirc-0.7.1.orig/daemons/hw-types.c	2005-06-04 16:12:28.670975784 +0800
+++ lirc-0.7.1/daemons/hw-types.c	2005-06-04 16:13:36.232704840 +0800
@@ -100,7 +100,6 @@
 	&hw_silitek,
 	/*	&hw_slinke,*/
 	&hw_tira,
-	&hw_udp,
 	&hw_uirt2,
 	&hw_uirt2_raw,
 #else
