diff -Naur lirc-0.7.0pre8.orig/configure lirc-0.7.0pre8/configure
--- lirc-0.7.0pre8.orig/configure	2004-10-16 21:02:35.409499856 +0200
+++ lirc-0.7.0pre8/configure	2004-10-16 21:02:55.709413800 +0200
@@ -8358,7 +8358,7 @@
 #define LIRC_DRIVER_ANY 1
 EOF
 
-  lirc_driver="none"
+  lirc_driver=""
   #include creative driver
   hw_module="${hw_module} hw_creative.o serial.o"
   
@@ -8414,12 +8414,13 @@
 		#hw_module="${hw_module} hw_irreal.o serial.o receive.o transmit.o"
 		;;
 	lirc_dev-lirc_atiusb)
+		lirc_driver="${lirc_driver} lirc_dev lirc_atiusb"
 		;;
 	lirc_dev-lirc_gpio)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
+		lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
 		;;
 	lirc_dev-lirc_i2c)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
+		lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
 		;;
 	lirc_dev-lirc_igorplugusb)
 		;;
@@ -8432,8 +8433,10 @@
 	lirc_dev-lirc_sasem)
 		;;
 	lirc_dev-lirc_serial)
+		lirc_driver="${lirc_driver} lirc_serial"
 		;;
 	lirc_dev-lirc_sir)
+		lirc_driver="${lirc_driver} lirc_sir"
 		;;
 	lirc_flyvideo)
 		;;
@@ -8482,11 +8485,14 @@
   done
   #remove duplicates
 
-  #sed segfaults when $hw_module is too long :(
-  #hw_module=`echo ${hw_module}| sed \
-  #			-e':again' \
-  #			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
-  #			-e'tagain'`
+  hw_module=`echo ${hw_module}| sed \
+  			-e':again' \
+  			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+  			-e'tagain'`
+  lirc_driver=`echo ${lirc_driver}| sed \
+  			-e':again' \
+			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+			-e'tagain'`
 
   space_to_newline()
   { for token in $*; do echo $token; done; }
@@ -8981,7 +8987,7 @@
   { echo "configure: error: no driver specified, try ./configure --help" 1>&2; exit 1; }
 fi
 
-if echo ${possible_drivers} | grep "($lirc_driver)" >/dev/null 2>&1; then
+if true; then
   dummy=
 else
   case $lirc_driver in
diff -Naur lirc-0.7.0pre8.orig/daemons/hw-types.c lirc-0.7.0pre8/daemons/hw-types.c
--- lirc-0.7.0pre8.orig/daemons/hw-types.c	2004-10-16 21:02:35.414499096 +0200
+++ lirc-0.7.0pre8/daemons/hw-types.c	2004-10-16 21:02:55.710413648 +0200
@@ -86,7 +86,6 @@
 	&hw_silitek,
 	/*	&hw_slinke,*/
 	&hw_tira,
-	&hw_udp,
 	&hw_uirt2,
 	&hw_uirt2_raw,
 #else
