diff -Nur lirc-0.7.2pre3.orig/configure lirc-0.7.2pre3/configure
--- lirc-0.7.2pre3.orig/configure	Mon Aug  8 14:44:07 2005
+++ lirc-0.7.2pre3/configure	Mon Aug  8 14:45:43 2005
@@ -8569,7 +8569,7 @@
 #define LIRC_DRIVER_ANY 1
 EOF
 
-  lirc_driver="any"
+  lirc_driver=""
   
   any_possible_drivers=`echo ${possible_drivers} | sed -e's/ /-/g' \
 			  -e's/)-(/ /g'  -e's/(//g' -e's/)//g' \
@@ -8639,20 +8639,22 @@
 		#hw_module="${hw_module} hw_irreal.o serial.o receive.o transmit.o"
 		;;
 	lirc_dev-lirc_atiusb)
+		lirc_driver="${lirc_driver} lirc_dev lirc_atiusb"
 		;;
 	lirc_dev-lirc_cmdir)
 		;;
 	lirc_dev-lirc_gpio)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
+		lirc_driver="${lirc_driver} lirc_dev lirc_gpio"
 		;;
 	lirc_dev-lirc_i2c)
-		#lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
+		lirc_driver="${lirc_driver} lirc_dev lirc_i2c"
 		;;
 	lirc_dev-lirc_igorplugusb)
 		;;
 	lirc_dev-lirc_imon)
 		;;
 	lirc_dev-lirc_it87)
+		lirc_driver="${lirc_driver} lirc_dev lirc_it87"
 		;;
 	lirc_dev-lirc_mceusb)
 		;;
@@ -8663,8 +8665,10 @@
 	lirc_dev-lirc_sasem)
 		;;
 	lirc_dev-lirc_serial)
+		lirc_driver="${lirc_driver} lirc_serial"
 		;;
 	lirc_dev-lirc_sir)
+		lirc_driver="${lirc_driver} lirc_sir"
 		;;
 	lirc_dev-lirc_streamzap)
 		;;
@@ -8721,11 +8725,14 @@
   done
   #remove duplicates
 
-  #sed segfaults when $hw_module is too long :(
-  #hw_module=`echo ${hw_module}| sed \
-  #			-e':again' \
-  #			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
-  #			-e'tagain'`
+  hw_module=`echo ${hw_module}| sed \
+  			-e':again' \
+  			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+  			-e'tagain'`
+  lirc_driver=`echo ${lirc_driver}| sed \
+  			-e':again' \
+			-e's/\(.*\)\(\<.*\) \(.*\) \2\(.*\)$/\1\2 \3\4/' \
+			-e'tagain'`
 
   space_to_newline()
   { for token in $*; do echo $token; done; }
@@ -9316,7 +9323,7 @@
   { echo "configure: error: no driver specified, try ./configure --help" 1>&2; exit 1; }
 fi
 
-if echo ${possible_drivers} | grep "($lirc_driver)" >/dev/null 2>&1; then
+if true; then
   dummy=
 else
   case $lirc_driver in
diff -Nur lirc-0.7.2pre3.orig/daemons/hw-types.c lirc-0.7.2pre3/daemons/hw-types.c
--- lirc-0.7.2pre3.orig/daemons/hw-types.c	Mon Aug  1 21:34:37 2005
+++ lirc-0.7.2pre3/daemons/hw-types.c	Mon Aug  8 14:44:49 2005
@@ -107,7 +107,6 @@
 	&hw_silitek,
 	/*	&hw_slinke,*/
 	&hw_tira,
-	&hw_udp,
 	&hw_uirt2,
 	&hw_uirt2_raw,
 #else
