diff -Naur lirc-0.7.1.orig/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.7.1/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.7.1.orig/drivers/lirc_atiusb/lirc_atiusb.c	2005-06-04 00:40:55.000000000 +0200
+++ lirc-0.7.1/drivers/lirc_atiusb/lirc_atiusb.c	2005-06-04 00:41:24.000000000 +0200
@@ -395,7 +395,7 @@
 	/* strip channel code */
 	if (!unique) {
 		iep->buf[len-1] &= 0x0F;
-		iep->buf[len-3] -= (chan<<4);
+		iep->buf[len-3] &= 0x0F;
 	}
 
 	if ( !((1U<<chan) & mask) ) {
