diff -Naur lirc-0.7.0pre8.orig/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.7.0pre8/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.7.0pre8.orig/drivers/lirc_atiusb/lirc_atiusb.c	2004-10-16 21:02:35.411499552 +0200
+++ lirc-0.7.0pre8/drivers/lirc_atiusb/lirc_atiusb.c	2004-10-16 21:15:40.717114952 +0200
@@ -391,6 +391,8 @@
 		dprintk(DRIVER_NAME "[%d]: accept channel %d\n",
 			ir->devnum, chan+1);
 
+		ir->buf_in[1] &= 0x0F;
+
 		/* check for repeats */
 		if (memcmp(ir->old, ir->buf_in, len) == 0) {
 			if (ir->old_jiffies + repeat_jiffies > jiffies) {
