diff -Naur lirc-0.7.0.orig/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.7.0/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.7.0.orig/drivers/lirc_atiusb/lirc_atiusb.c	2005-04-23 18:49:06.000000000 +0200
+++ lirc-0.7.0/drivers/lirc_atiusb/lirc_atiusb.c	2005-04-23 18:55:54.000000000 +0200
@@ -484,7 +484,7 @@
 	dprintk(DRIVER_NAME ": usb probe called\n");
 
 	/* determine remote type */
-	switch (dev->descriptor.idVendor) {
+	switch (cpu_to_le16(dev->descriptor.idVendor)) {
 	case VENDOR_ATI1:
 		type = ATI1_COMPATIBLE;
 		break;
