diff -Naur lirc-0.7.1.orig/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.7.1/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.7.1.orig/drivers/lirc_atiusb/lirc_atiusb.c	2005-06-04 00:40:55.000000000 +0200
+++ lirc-0.7.1/drivers/lirc_atiusb/lirc_atiusb.c	2005-06-04 00:44:50.000000000 +0200
@@ -936,7 +936,7 @@
 	devnum = dev->devnum;
 
 	/* determine remote type */
-	switch (dev->descriptor.idVendor) {
+	switch (cpu_to_le16(dev->descriptor.idVendor)) {
 	case VENDOR_ATI1:
 		type = ATI1_COMPATIBLE;
 		break;
