diff -Naur lirc-0.7.0.orig/configure lirc-0.7.0/configure
--- lirc-0.7.0.orig/configure	2004-12-30 01:30:08.498232128 +0100
+++ lirc-0.7.0/configure	2004-12-30 01:31:49.115935912 +0100
@@ -7866,16 +7866,12 @@
   kernelext=ko
   no_kernel=yes
 
-  if test `uname` != "Linux"; then
-    kerneldir="not running Linux"
-  else
-    for dir in /usr/src/kernel-source-`uname -r` /usr/src/linux-`uname -r` /usr/src/linux ${ac_kerneldir}; do
+    for dir in ${ac_kerneldir}; do
       if test -d $dir; then
         kerneldir=`dirname $dir/Makefile`/
         no_kernel=no
       fi;
     done
-  fi
 
   if test x${no_kernel} != xyes; then
     if test -f ${kerneldir}/Makefile; then
