diff -Naur lirc-0.7.0.orig/configure lirc-0.7.0/configure
--- lirc-0.7.0.orig/configure	2004-12-30 01:36:49.918207016 +0100
+++ lirc-0.7.0/configure	2004-12-30 01:36:56.205251240 +0100
@@ -7884,14 +7884,14 @@
       echo "lirc_tell_me_what_cc_is:" >>${ac_pkss_makefile}
       echo "	echo \$(CC)" >>${ac_pkss_makefile}
 
-      kernelcc=`make -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_cc_is`
+      kernelcc=`${MAKE-make} -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_cc_is`
 
       echo "lirc_tell_me_what_version_is:" >>${ac_pkss_makefile}
       echo "	echo \$(VERSION)" >>${ac_pkss_makefile}
       echo "lirc_tell_me_what_patchlevel_is:" >>${ac_pkss_makefile}
       echo "	echo \$(PATCHLEVEL)" >>${ac_pkss_makefile}
-      version=`make -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_version_is`
-      patchlevel=`make -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_patchlevel_is`
+      version=`${MAKE-make} -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_version_is`
+      patchlevel=`${MAKE-make} -s -C ${kerneldir} -f ${ac_pkss_makefile} lirc_tell_me_what_patchlevel_is`
       if test ${version} -eq 2; then
         if test ${patchlevel} -lt 5; then
           kernelext=o
