commited in cvs (in diffrent form)

diff -Nur lirc-0.7.3pre1.orig/drivers/kcompat.h lirc-0.7.3pre1/drivers/kcompat.h
--- lirc-0.7.3pre1.orig/drivers/kcompat.h	Thu Oct 20 20:41:36 2005
+++ lirc-0.7.3pre1/drivers/kcompat.h	Fri Oct 28 12:05:36 2005
@@ -219,5 +219,8 @@
 #ifndef I2C_ALGO_BIT
 #   define I2C_ALGO_BIT 0
 #endif
+#ifndef URB_ASYNC_UNLINK
+#   define URB_ASYNC_UNLINK 0
+#endif
 
 #endif /* _KCOMPAT_H */
