* fixes linux 2.6.16 compliation

taken from upstream cvs

diff -Naur lirc-0.8.0.orig/drivers/kcompat.h lirc-0.8.0/drivers/kcompat.h
--- lirc-0.8.0.orig/drivers/kcompat.h	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/kcompat.h	2006-03-20 21:25:58.000000000 +0100
@@ -5,6 +5,12 @@
 
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16)
+#define LIRC_THIS_MODULE(x) x,
+#else /* >= 2.6.16 */
+#define LIRC_THIS_MODULE(x)
+#endif
+
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 
 #include <linux/device.h>
@@ -245,6 +251,11 @@
 #   define I2C_ALGO_BIT 0
 #endif
 
+/* removed in 2.6.16 */
+#ifndef I2C_DRIVERID_EXP3
+#  define I2C_DRIVERID_EXP3 0xf003
+#endif
+
 /*************************** USB specific *****************************/
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 4, 0)
 #include <linux/usb.h>
diff -Naur lirc-0.8.0.orig/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.8.0/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.8.0.orig/drivers/lirc_atiusb/lirc_atiusb.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_atiusb/lirc_atiusb.c	2006-03-20 21:25:58.000000000 +0100
@@ -1180,7 +1180,7 @@
 }
 
 static struct usb_driver usb_remote_driver = {
-	.owner =	THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name =		DRIVER_NAME,
 	.probe =	usb_remote_probe,
 	.disconnect =	usb_remote_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_i2c/lirc_i2c.c lirc-0.8.0/drivers/lirc_i2c/lirc_i2c.c
--- lirc-0.8.0.orig/drivers/lirc_i2c/lirc_i2c.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_i2c/lirc_i2c.c	2006-03-20 21:25:19.000000000 +0100
@@ -360,9 +360,16 @@
 static int ir_command(struct i2c_client *client, unsigned int cmd, void *arg);
 
 static struct i2c_driver driver = {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 16)
         name:           "i2c ir driver",
-        id:             I2C_DRIVERID_EXP3, /* FIXME */
         flags:          I2C_DF_NOTIFY,
+#else
+	.driver = {
+		owner:  THIS_MODULE,
+		name:   "i2c ir driver",
+	},
+#endif
+        id:             I2C_DRIVERID_EXP3, /* FIXME */
         attach_adapter: ir_probe,
         detach_client:  ir_detach,
         command:        ir_command,
diff -Naur lirc-0.8.0.orig/drivers/lirc_igorplugusb/lirc_igorplugusb.c lirc-0.8.0/drivers/lirc_igorplugusb/lirc_igorplugusb.c
--- lirc-0.8.0.orig/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2006-03-20 21:25:58.000000000 +0100
@@ -615,7 +615,7 @@
 };
 
 static struct usb_driver usb_remote_driver = {
-	.owner =	THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name =		DRIVER_NAME,
 	.probe =	usb_remote_probe,
 	.disconnect =	usb_remote_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_imon/lirc_imon.c lirc-0.8.0/drivers/lirc_imon/lirc_imon.c
--- lirc-0.8.0.orig/drivers/lirc_imon/lirc_imon.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_imon/lirc_imon.c	2006-03-20 21:25:58.000000000 +0100
@@ -201,7 +201,7 @@
 
 /* USB Device data */
 static struct usb_driver imon_driver = {
-	.owner 		= THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name 		= MOD_NAME,
 	.probe 		= imon_probe,
 	.disconnect 	= imon_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_mceusb/lirc_mceusb.c lirc-0.8.0/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-0.8.0.orig/drivers/lirc_mceusb/lirc_mceusb.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_mceusb/lirc_mceusb.c	2006-03-20 21:25:58.000000000 +0100
@@ -222,7 +222,7 @@
 
 /* usb specific object needed to register this driver with the usb subsystem */
 static struct usb_driver mceusb_driver = {
-	.owner =	THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name =		DRIVER_NAME,
 	.probe =	mceusb_probe,
 	.disconnect =	mceusb_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_mceusb2/lirc_mceusb2.c lirc-0.8.0/drivers/lirc_mceusb2/lirc_mceusb2.c
--- lirc-0.8.0.orig/drivers/lirc_mceusb2/lirc_mceusb2.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_mceusb2/lirc_mceusb2.c	2006-03-20 21:25:58.000000000 +0100
@@ -585,7 +585,7 @@
 }
 
 static struct usb_driver usb_remote_driver = {
-	.owner =	THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name =		DRIVER_NAME,
 	.probe =	usb_remote_probe,
 	.disconnect =	usb_remote_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_sasem/lirc_sasem.c lirc-0.8.0/drivers/lirc_sasem/lirc_sasem.c
--- lirc-0.8.0.orig/drivers/lirc_sasem/lirc_sasem.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_sasem/lirc_sasem.c	2006-03-20 21:25:58.000000000 +0100
@@ -189,7 +189,7 @@
 
 /* USB Device data */
 static struct usb_driver sasem_driver = {
-	.owner 		= THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name 		= MOD_NAME,
 	.probe 		= sasem_probe,
 	.disconnect 	= sasem_disconnect,
diff -Naur lirc-0.8.0.orig/drivers/lirc_streamzap/lirc_streamzap.c lirc-0.8.0/drivers/lirc_streamzap/lirc_streamzap.c
--- lirc-0.8.0.orig/drivers/lirc_streamzap/lirc_streamzap.c	2006-03-20 21:24:53.000000000 +0100
+++ lirc-0.8.0/drivers/lirc_streamzap/lirc_streamzap.c	2006-03-20 21:25:58.000000000 +0100
@@ -177,7 +177,7 @@
 /* usb specific object needed to register this driver with the usb subsystem */
 
 static struct usb_driver streamzap_driver = {
-	.owner =	THIS_MODULE,
+	LIRC_THIS_MODULE(.owner = THIS_MODULE)
 	.name =		DRIVER_NAME,
 	.probe =	streamzap_probe,
 	.disconnect =	streamzap_disconnect,
