diff -Nur lirc-0.7.3pre1.orig/drivers/lirc_it87/lirc_it87.c lirc-0.7.3pre1/drivers/lirc_it87/lirc_it87.c
--- lirc-0.7.3pre1.orig/drivers/lirc_it87/lirc_it87.c	Tue Oct  4 23:10:04 2005
+++ lirc-0.7.3pre1/drivers/lirc_it87/lirc_it87.c	Fri Nov  4 11:59:59 2005
@@ -58,7 +58,6 @@
 #include <linux/delay.h>
 #include <linux/poll.h>
 #include <asm/system.h>
-#include <asm/segment.h>
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/fcntl.h>
diff -Nur lirc-0.7.3pre1.orig/drivers/lirc_serial/lirc_serial.c lirc-0.7.3pre1/drivers/lirc_serial/lirc_serial.c
--- lirc-0.7.3pre1.orig/drivers/lirc_serial/lirc_serial.c	Thu Sep 22 23:10:37 2005
+++ lirc-0.7.3pre1/drivers/lirc_serial/lirc_serial.c	Fri Nov  4 12:00:05 2005
@@ -91,7 +91,6 @@
 #include <linux/poll.h>
 
 #include <asm/system.h>
-#include <asm/segment.h>
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/fcntl.h>
diff -Nur lirc-0.7.3pre1.orig/drivers/lirc_sir/lirc_sir.c lirc-0.7.3pre1/drivers/lirc_sir/lirc_sir.c
--- lirc-0.7.3pre1.orig/drivers/lirc_sir/lirc_sir.c	Thu Sep 22 23:10:37 2005
+++ lirc-0.7.3pre1/drivers/lirc_sir/lirc_sir.c	Fri Nov  4 12:00:08 2005
@@ -75,7 +75,6 @@
 #include <linux/delay.h>
 #include <linux/poll.h>
 #include <asm/system.h>
-#include <asm/segment.h>
 #include <asm/io.h>
 #include <asm/irq.h>
 #include <asm/fcntl.h>
