diff -Nur lirc-0.8.1pre1.orig/drivers/lirc_gpio/Makefile.in lirc-0.8.1pre1/drivers/lirc_gpio/Makefile.in
--- lirc-0.8.1pre1.orig/drivers/lirc_gpio/Makefile.in	Sun Jun  4 07:31:36 2006
+++ lirc-0.8.1pre1/drivers/lirc_gpio/Makefile.in	Fri Jun 23 07:57:02 2006
@@ -128,7 +128,7 @@
 KBUILD_VERBOSE = 1
 LIRC_EXTRA_CFLAGS = -DIRCTL_DEV_MAJOR=$(lirc_major) -DEXPORT_SYMTAB $(DEFS) \
 		$(DEFAULT_INCLUDES) -I$(LIRC_DEVDIR)/../.. \
-		-I$(KERNEL_LOCATION)/include/
+		-I$(KERNEL_LOCATION)/include/ -I$(KERNEL_LOCATION)/drivers/media/video/
 
 
 CLEANFILES = $(module_DATA) .$(module_DATA).flags $(module_DATA:.o=.mod.c) $(module_DATA:.o=.@kernelext@) *~
diff -Nur lirc-0.8.1pre1.orig/drivers/lirc_gpio/lirc_gpio.c lirc-0.8.1pre1/drivers/lirc_gpio/lirc_gpio.c
--- lirc-0.8.1pre1.orig/drivers/lirc_gpio/lirc_gpio.c	Sat May 20 10:43:18 2006
+++ lirc-0.8.1pre1/drivers/lirc_gpio/lirc_gpio.c	Fri Jun 23 07:57:17 2006
@@ -48,9 +48,12 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,4,0)
 #include "../drivers/char/bttv.h"
 #include "../drivers/char/bttvp.h"
-#else
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(2,6,17)
 #include "../drivers/media/video/bttv.h"
 #include "../drivers/media/video/bttvp.h"
+#else
+#include "bt8xx/bttv.h"
+#include "bt8xx/bttvp.h"
 #endif
 
 #if BTTV_VERSION_CODE < KERNEL_VERSION(0,7,45)
