* backported from lirc CVS

diff -Naur lirc-0.8.1pre1.orig/drivers/kcompat.h lirc-0.8.1pre1/drivers/kcompat.h
--- lirc-0.8.1pre1.orig/drivers/kcompat.h	2006-09-17 21:36:10.000000000 +0200
+++ lirc-0.8.1pre1/drivers/kcompat.h	2006-09-17 21:42:43.000000000 +0200
@@ -15,8 +15,11 @@
 
 #include <linux/device.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
 #define LIRC_HAVE_DEVFS
 #define LIRC_HAVE_DEVFS_26
+#endif
+
 #define LIRC_HAVE_SYSFS
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,13)
diff -Naur lirc-0.8.1pre1.orig/drivers/lirc_dev/lirc_dev.c lirc-0.8.1pre1/drivers/lirc_dev/lirc_dev.c
--- lirc-0.8.1pre1.orig/drivers/lirc_dev/lirc_dev.c	2006-09-17 21:36:10.000000000 +0200
+++ lirc-0.8.1pre1/drivers/lirc_dev/lirc_dev.c	2006-09-17 21:37:02.000000000 +0200
@@ -49,16 +49,18 @@
 #endif
 #define __KERNEL_SYSCALLS__
 #include <linux/unistd.h>
+
+#include "drivers/kcompat.h"
+
 /* DevFS header */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,0)
+#if defined(LIRC_HAVE_DEVFS)
 #include <linux/devfs_fs_kernel.h>
 #endif
 /* SysFS header */
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
+#if defined(LIRC_HAVE_SYSFS)
 #include <linux/device.h>
 #endif
 
-#include "drivers/kcompat.h"
 #include "drivers/lirc.h"
 #include "lirc_dev.h"
 
diff -Naur lirc-0.8.1pre1.orig/drivers/lirc_imon/lirc_imon.c lirc-0.8.1pre1/drivers/lirc_imon/lirc_imon.c
--- lirc-0.8.1pre1.orig/drivers/lirc_imon/lirc_imon.c	2006-09-17 21:36:10.000000000 +0200
+++ lirc-0.8.1pre1/drivers/lirc_imon/lirc_imon.c	2006-09-17 21:44:28.000000000 +0200
@@ -58,10 +58,13 @@
 #include <linux/slab.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
+#include "drivers/kcompat.h"
+
+#if defined(LIRC_HAVE_DEVFS)
 #include <linux/devfs_fs_kernel.h>
+#endif
 
 #include "drivers/lirc.h"
-#include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
 
diff -Naur lirc-0.8.1pre1.orig/drivers/lirc_sasem/lirc_sasem.c lirc-0.8.1pre1/drivers/lirc_sasem/lirc_sasem.c
--- lirc-0.8.1pre1.orig/drivers/lirc_sasem/lirc_sasem.c	2006-09-17 21:36:10.000000000 +0200
+++ lirc-0.8.1pre1/drivers/lirc_sasem/lirc_sasem.c	2006-09-17 21:47:08.000000000 +0200
@@ -67,10 +67,13 @@
 #include <linux/slab.h>
 #include <asm/uaccess.h>
 #include <linux/usb.h>
+
+#include "drivers/kcompat.h"
+#if defined(LIRC_HAVE_DEVFS)
 #include <linux/devfs_fs_kernel.h>
+#endif
 
 #include "drivers/lirc.h"
-#include "drivers/kcompat.h"
 #include "drivers/lirc_dev/lirc_dev.h"
 
 
