diff -Naur lirc-0.7.3pre1.orig/drivers/lirc_gpio/lirc_gpio.c lirc-0.7.3pre1/drivers/lirc_gpio/lirc_gpio.c
--- lirc-0.7.3pre1.orig/drivers/lirc_gpio/lirc_gpio.c	2006-01-03 12:50:19.000000000 +0100
+++ lirc-0.7.3pre1/drivers/lirc_gpio/lirc_gpio.c	2006-01-03 12:57:55.000000000 +0100
@@ -68,7 +68,7 @@
 static int debug = 0;
 static int card = 0;
 static int minor = -1;
-static int bttv_id = BTTV_UNKNOWN;
+static int bttv_id = BTTV_BOARD_UNKNOWN;
 static unsigned long gpio_mask = 0;
 static unsigned long gpio_enable = 0;
 static unsigned long gpio_lock_mask = 0;
@@ -94,57 +94,57 @@
 };
 
 static struct rcv_info rcv_infos[] = {
-	{BTTV_UNKNOWN,                0,          0,          0,         0,          0,   0,  1,  0},
-	{BTTV_PXELVWPLTVPAK,          0, 0x00003e00,          0, 0x0010000,          0,   0, 15, 32},
-	{BTTV_PXELVWPLTVPRO,          0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
-	{BTTV_PV_BT878P_9B,           0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
-	{BTTV_PV_BT878P_PLUS,         0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
-	{BTTV_AVERMEDIA,              0, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32},
-	{BTTV_AVPHONE98,     0x00011461, 0x003b8000, 0x00004000, 0x0800000, 0x00800000,   0, 10,  0}, /*mapped to Capture98*/
-	{BTTV_AVERMEDIA98,   0x00021461, 0x003b8000, 0x00004000, 0x0800000, 0x00800000,   0, 10,  0}, /*mapped to Capture98*/
-	{BTTV_AVPHONE98,     0x00031461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
+	{BTTV_BOARD_UNKNOWN,                0,          0,          0,         0,          0,   0,  1,  0},
+	{BTTV_BOARD_PXELVWPLTVPAK,          0, 0x00003e00,          0, 0x0010000,          0,   0, 15, 32},
+	{BTTV_BOARD_PXELVWPLTVPRO,          0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
+	{BTTV_BOARD_PV_BT878P_9B,           0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
+	{BTTV_BOARD_PV_BT878P_PLUS,         0, 0x00001f00,          0, 0x0008000,          0, 500, 12, 32},
+	{BTTV_BOARD_AVERMEDIA,              0, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32},
+	{BTTV_BOARD_AVPHONE98,     0x00011461, 0x003b8000, 0x00004000, 0x0800000, 0x00800000,   0, 10,  0}, /*mapped to Capture98*/
+	{BTTV_BOARD_AVERMEDIA98,   0x00021461, 0x003b8000, 0x00004000, 0x0800000, 0x00800000,   0, 10,  0}, /*mapped to Capture98*/
+	{BTTV_BOARD_AVPHONE98,     0x00031461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
 	/* is this one correct? */
-	{BTTV_AVERMEDIA98,   0x00041461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
+	{BTTV_BOARD_AVERMEDIA98,   0x00041461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
 	/* work-around for VDOMATE */
-	{BTTV_AVERMEDIA98,   0x03001461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
+	{BTTV_BOARD_AVERMEDIA98,   0x03001461, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
 	/* reported by Danijel Korzinek, AVerTV GOw/FM */
-	{BTTV_AVERMEDIA98,   0x00000000, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
-	{BTTV_CHRONOS_VS2,            0, 0x000000f8,          0, 0x0000100,          0,   0, 20,  0},
+	{BTTV_BOARD_AVERMEDIA98,   0x00000000, 0x00f88000,          0, 0x0010000, 0x00010000,   0, 10, 32}, /*mapped to Phone98*/
+	{BTTV_BOARD_CHRONOS_VS2,            0, 0x000000f8,          0, 0x0000100,          0,   0, 20,  0},
 	/* CPH031 and CPH033 cards (?) */
 	/* MIRO was just a work-around */
-	{BTTV_MIRO,                   0, 0x00001f00,          0, 0x0004000,          0,   0, 10, 32},
-	{BTTV_DYNALINK,               0, 0x00001f00,          0, 0x0004000,          0,   0, 10, 32},
-	{BTTV_WINVIEW_601,            0, 0x00001f00,          0, 0x0004000,          0,   0,  0, 32},
-#ifdef BTTV_KWORLD
-	{BTTV_KWORLD,                 0, 0x00007f00,          0, 0x0004000,          0,   0, 12, 32},
+	{BTTV_BOARD_MIRO,                   0, 0x00001f00,          0, 0x0004000,          0,   0, 10, 32},
+	{BTTV_BOARD_DYNALINK,               0, 0x00001f00,          0, 0x0004000,          0,   0, 10, 32},
+	{BTTV_BOARD_WINVIEW_601,            0, 0x00001f00,          0, 0x0004000,          0,   0,  0, 32},
+#ifdef BTTV_BOARD_KWORLD
+	{BTTV_BOARD_KWORLD,                 0, 0x00007f00,          0, 0x0004000,          0,   0, 12, 32},
 #endif
 	/* just a guess */
-	{BTTV_MAGICTVIEW061,          0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
- 	{BTTV_MAGICTVIEW063,          0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
- 	{BTTV_PHOEBE_TVMAS,           0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
-#ifdef BTTV_BESTBUY_EASYTV2
-        {BTTV_BESTBUY_EASYTV,         0, 0x00007F00,          0, 0x0004000,          0,   0, 10,  8},
-        {BTTV_BESTBUY_EASYTV2,        0, 0x00007F00,          0, 0x0008000,          0,   0, 10,  8},
+	{BTTV_BOARD_MAGICTVIEW061,          0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
+ 	{BTTV_BOARD_MAGICTVIEW063,          0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
+ 	{BTTV_BOARD_PHOEBE_TVMAS,           0, 0x0028e000,          0, 0x0020000,          0,   0, 20, 32},
+#ifdef BTTV_BOARD_BESTBUY_EASYTV2
+        {BTTV_BOARD_BESTBUY_EASYTV,         0, 0x00007F00,          0, 0x0004000,          0,   0, 10,  8},
+        {BTTV_BOARD_BESTBUY_EASYTV2,        0, 0x00007F00,          0, 0x0008000,          0,   0, 10,  8},
 #endif
 	/* lock_mask probably also 0x100, or maybe it is 0x0 for all others !?! */
-	{BTTV_FLYVIDEO,               0, 0x000000f8,          0,         0,          0,   0,  0, 42},
- 	{BTTV_FLYVIDEO_98,            0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
- 	{BTTV_TYPHOON_TVIEW,          0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
-#ifdef BTTV_FLYVIDEO_98FM
+	{BTTV_BOARD_FLYVIDEO,               0, 0x000000f8,          0,         0,          0,   0,  0, 42},
+ 	{BTTV_BOARD_FLYVIDEO_98,            0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
+ 	{BTTV_BOARD_TYPHOON_TVIEW,          0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
+#ifdef BTTV_BOARD_FLYVIDEO_98FM
 	/* smorar@alfonzo.smuts.uct.ac.za */
-	{BTTV_FLYVIDEO_98FM,          0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
+	{BTTV_BOARD_FLYVIDEO_98FM,          0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 42},
 #endif
 	/* The Leadtek WinFast TV 2000 XP card (id 0x6606107d) uses an
 	 * extra gpio bit compared to the original TV 2000 card (id
 	 * 0x217d6606); as the bttv-0.7.100 driver does not
 	 * distinguish between the two cards, we enable the extra bit
 	 * based on the card id: */
-	{BTTV_WINFAST2000,   0x6606107d, 0x000008f8,          0, 0x0000100,          0,   0,  0, 32},
-	{BTTV_WINFAST2000,   0x6609107d, 0x000008f8,          0, 0x0000100,          0,   0,  0, 32},
+	{BTTV_BOARD_WINFAST2000,   0x6606107d, 0x000008f8,          0, 0x0000100,          0,   0,  0, 32},
+	{BTTV_BOARD_WINFAST2000,   0x6609107d, 0x000008f8,          0, 0x0000100,          0,   0,  0, 32},
 	/* default: */
-	{BTTV_WINFAST2000,            0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 32},
-#ifdef BTTV_GVBCTV5PCI
-	{BTTV_GVBCTV5PCI,             0, 0x00f0b000,          0,         0,          0,   0, 20,  8},
+	{BTTV_BOARD_WINFAST2000,            0, 0x000000f8,          0, 0x0000100,          0,   0,  0, 32},
+#ifdef BTTV_BOARD_GVBCTV5PCI
+	{BTTV_BOARD_GVBCTV5PCI,             0, 0x00f0b000,          0,         0,          0,   0, 20,  8},
 #endif
 };
 
@@ -189,7 +189,7 @@
 	
 	switch (bttv_id)
 	{
-	case BTTV_AVERMEDIA98:
+	case BTTV_BOARD_AVERMEDIA98:
 		if (bttv_write_gpio(card, gpio_enable, gpio_enable)) {
 			dprintk(LOGHEAD "cannot write to GPIO\n", card);
 			return -EIO;
@@ -221,13 +221,13 @@
 	dprintk(LOGHEAD "code is %lx\n",card,(unsigned long) codes[0]);
 	switch (bttv_id)
 	{
-	case BTTV_AVERMEDIA:
+	case BTTV_BOARD_AVERMEDIA:
 		codes[2] = (codes[0]<<2)&0xff;
 		codes[3] = (~codes[2])&0xff;
 		codes[0] = 0x02;
 		codes[1] = 0xFD;
 		break;
-	case BTTV_AVPHONE98:
+	case BTTV_BOARD_AVPHONE98:
 		codes[2] = ((codes[0]&(~0x1))<<2)&0xff;
 		codes[3] = (~codes[2])&0xff;
 		if (codes[0]&0x1) {
@@ -238,13 +238,13 @@
 			codes[1] = 0xbf;
 		}
 		break;
-	case BTTV_AVERMEDIA98:
+	case BTTV_BOARD_AVERMEDIA98:
 		break;
-	case BTTV_FLYVIDEO:
-	case BTTV_FLYVIDEO_98:
-	case BTTV_TYPHOON_TVIEW:
-#ifdef BTTV_FLYVIDEO_98FM
-	case BTTV_FLYVIDEO_98FM:
+	case BTTV_BOARD_FLYVIDEO:
+	case BTTV_BOARD_FLYVIDEO_98:
+	case BTTV_BOARD_TYPHOON_TVIEW:
+#ifdef BTTV_BOARD_FLYVIDEO_98FM
+	case BTTV_BOARD_FLYVIDEO_98FM:
 #endif
 		codes[4]=codes[0]<<3;
 		codes[5]=((~codes[4])&0xff);
@@ -254,23 +254,23 @@
 		codes[2]=0x1F;
 		codes[3]=0x2F;
 		break;
-        case BTTV_MAGICTVIEW061:
-        case BTTV_MAGICTVIEW063:
-	case BTTV_PHOEBE_TVMAS:
+        case BTTV_BOARD_MAGICTVIEW061:
+        case BTTV_BOARD_MAGICTVIEW063:
+	case BTTV_BOARD_PHOEBE_TVMAS:
 		codes[0] = (codes[0]&0x01)
 			|((codes[0]&0x02)<<1)
 			|((codes[0]&0x04)<<2)
 			|((codes[0]&0x08)>>2)
 			|((codes[0]&0x10)>>1);
 		/* FALLTHROUGH */
-	case BTTV_MIRO:
-	case BTTV_DYNALINK:
-	case BTTV_PXELVWPLTVPAK:
-	case BTTV_PXELVWPLTVPRO:
-	case BTTV_PV_BT878P_9B:
-	case BTTV_PV_BT878P_PLUS:
-#ifdef BTTV_KWORLD
-	case BTTV_KWORLD:
+	case BTTV_BOARD_MIRO:
+	case BTTV_BOARD_DYNALINK:
+	case BTTV_BOARD_PXELVWPLTVPAK:
+	case BTTV_BOARD_PXELVWPLTVPRO:
+	case BTTV_BOARD_PV_BT878P_9B:
+	case BTTV_BOARD_PV_BT878P_PLUS:
+#ifdef BTTV_BOARD_KWORLD
+	case BTTV_BOARD_KWORLD:
 #endif
 		codes[2] = reverse(codes[0],8);
 		codes[3] = (~codes[2])&0xff;
@@ -281,8 +281,8 @@
 		/* derived from e-tech config file */
 		/* 26 + 16 bits */
 		/* won't apply it until it's confirmed with a fly98 */
- 	case BTTV_FLYVIDEO_98:
-	case BTTV_FLYVIDEO_98FM:
+ 	case BTTV_BOARD_FLYVIDEO_98:
+	case BTTV_BOARD_FLYVIDEO_98FM:
 		codes[4]=codes[0]<<3;
 		codes[5]=(~codes[4])&0xff;
 		
@@ -292,10 +292,10 @@
 		codes[3]=0x2F;
 		break;
 #endif
-	case BTTV_WINFAST2000:
+	case BTTV_BOARD_WINFAST2000:
 		/* shift extra bit */
 		codes[0] = (codes[0]&0x1f) | ((codes[0]&0x20) << 1);
-	case BTTV_WINVIEW_601:
+	case BTTV_BOARD_WINVIEW_601:
 		codes[2] = reverse(codes[0],8);
 		codes[3] = (~codes[2])&0xff;
 		codes[0] = 0xC0;
@@ -471,7 +471,7 @@
 		printk(LOGHEAD "card type 0x%x, id 0x%x\n",minor,
 		       type,cardid);
 
-		if (type == BTTV_UNKNOWN) {
+		if (type == BTTV_BOARD_UNKNOWN) {
 			printk(LOGHEAD "cannot detect TV card nr %d!\n",
 			       minor, card);
 			return -EBADRQC;
@@ -493,17 +493,17 @@
 				break;
 			}
 		}
-		if (type==BTTV_AVPHONE98 && cardid==0x00011461)	{
-			bttv_id = BTTV_AVERMEDIA98;
+		if (type==BTTV_BOARD_AVPHONE98 && cardid==0x00011461)	{
+			bttv_id = BTTV_BOARD_AVERMEDIA98;
 		}
-		if (type==BTTV_AVERMEDIA98 && cardid==0x00041461) {
-			bttv_id = BTTV_AVPHONE98;
+		if (type==BTTV_BOARD_AVERMEDIA98 && cardid==0x00041461) {
+			bttv_id = BTTV_BOARD_AVPHONE98;
 		}
-		if (type==BTTV_AVERMEDIA98 && cardid==0x03001461) {
-			bttv_id = BTTV_AVPHONE98;
+		if (type==BTTV_BOARD_AVERMEDIA98 && cardid==0x03001461) {
+			bttv_id = BTTV_BOARD_AVPHONE98;
 		}
-		if (type==BTTV_AVERMEDIA98 && cardid==0x00000000) {
-			bttv_id = BTTV_AVPHONE98;
+		if (type==BTTV_BOARD_AVERMEDIA98 && cardid==0x00000000) {
+			bttv_id = BTTV_BOARD_AVPHONE98;
 		}
 		if (card_type == sizeof(rcv_infos)/sizeof(struct rcv_info)) {
 			printk(LOGHEAD "TV card type 0x%x not supported!\n",
diff -Naur lirc-0.7.3pre1.orig/drivers/lirc_imon/lirc_imon.c lirc-0.7.3pre1/drivers/lirc_imon/lirc_imon.c
--- lirc-0.7.3pre1.orig/drivers/lirc_imon/lirc_imon.c	2006-01-03 12:50:19.000000000 +0100
+++ lirc-0.7.3pre1/drivers/lirc_imon/lirc_imon.c	2006-01-03 13:02:58.000000000 +0100
@@ -216,7 +216,6 @@
 static struct usb_class_driver imon_class = {
 	.name 		= DEVFS_NAME,
 	.fops		= &vfd_fops,
-	.mode		= DEVFS_MODE,
 	.minor_base	= VFD_MINOR_BASE,
 };
 #endif
diff -Naur lirc-0.7.3pre1.orig/drivers/lirc_sasem/lirc_sasem.c lirc-0.7.3pre1/drivers/lirc_sasem/lirc_sasem.c
--- lirc-0.7.3pre1.orig/drivers/lirc_sasem/lirc_sasem.c	2006-01-03 12:50:19.000000000 +0100
+++ lirc-0.7.3pre1/drivers/lirc_sasem/lirc_sasem.c	2006-01-03 13:04:17.000000000 +0100
@@ -204,7 +204,6 @@
 static struct usb_class_driver sasem_class = {
 	.name 		= DEVFS_NAME,
 	.fops		= &vfd_fops,
-	.mode		= DEVFS_MODE,
 	.minor_base	= VFD_MINOR_BASE,
 };
 #endif
diff -Naur lirc-0.7.3pre1.orig/drivers/lirc_dev/lirc_dev.c lirc-0.7.3pre1/drivers/lirc_dev/lirc_dev.c
--- lirc-0.7.3pre1.orig/drivers/lirc_dev/lirc_dev.c	2006-01-04 00:20:49.000000000 +0100
+++ lirc-0.7.3pre1/drivers/lirc_dev/lirc_dev.c	2006-01-04 00:21:08.000000000 +0100
@@ -382,7 +382,7 @@
 			S_IFCHR|S_IRUSR|S_IWUSR,
 			DEV_LIRC "/%u", ir->p.minor);
 #endif
-	(void) class_device_create(lirc_class, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor),
+	(void) class_device_create(lirc_class, NULL, MKDEV(IRCTL_DEV_MAJOR, ir->p.minor),
 				       NULL, "lirc%u", ir->p.minor);
 
 	if(p->sample_rate || p->get_queue) {
