diff -Naur lirc-0.8.2.orig/drivers/lirc_serial/lirc_serial.c lirc-0.8.2/drivers/lirc_serial/lirc_serial.c
--- lirc-0.8.2.orig/drivers/lirc_serial/lirc_serial.c	2007-12-14 00:10:49.000000000 +0100
+++ lirc-0.8.2/drivers/lirc_serial/lirc_serial.c	2007-12-14 00:11:53.000000000 +0100
@@ -998,7 +998,7 @@
 	do_gettimeofday(&lasttv);
 
 	result=request_irq(irq,irq_handler,
-			   SA_INTERRUPT | (share_irq ? SA_SHIRQ:0),
+			   IRQF_DISABLED | (share_irq ? IRQF_SHARED:0),
 			   LIRC_DRIVER_NAME,(void *)&hardware);
 	
 	switch(result)
diff -Naur lirc-0.8.2.orig/drivers/lirc_sir/lirc_sir.c lirc-0.8.2/drivers/lirc_sir/lirc_sir.c
--- lirc-0.8.2.orig/drivers/lirc_sir/lirc_sir.c	2007-12-14 00:10:49.000000000 +0100
+++ lirc-0.8.2/drivers/lirc_sir/lirc_sir.c	2007-12-14 00:11:31.000000000 +0100
@@ -1051,7 +1051,7 @@
 		return -EBUSY;
 	}
 #endif
-	retval = request_irq(irq, sir_interrupt, SA_INTERRUPT,
+	retval = request_irq(irq, sir_interrupt, IRQF_DISABLED,
 			     LIRC_DRIVER_NAME, NULL);
 	if (retval < 0) {
 #               ifndef LIRC_ON_SA1100
