diff -Naur lirc-0.8.6/drivers/lirc_atiusb/lirc_atiusb.c lirc-0.8.6a/drivers/lirc_atiusb/lirc_atiusb.c
--- lirc-0.8.6/drivers/lirc_atiusb/lirc_atiusb.c	2009-03-11 01:21:46.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_atiusb/lirc_atiusb.c	2010-02-20 11:32:57.813510994 +0100
@@ -43,7 +43,11 @@
 #error "*******************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #include <linux/kernel.h>
 #include <linux/errno.h>
diff -Naur lirc-0.8.6/drivers/lirc_bt829/lirc_bt829.c lirc-0.8.6a/drivers/lirc_bt829/lirc_bt829.c
--- lirc-0.8.6/drivers/lirc_bt829/lirc_bt829.c	2009-03-09 19:54:17.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_bt829/lirc_bt829.c	2010-02-20 11:32:57.813510994 +0100
@@ -22,7 +22,11 @@
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 4, 0)
 #error "This driver needs kernel version 2.4.0 or higher"
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/module.h>
 #include <linux/threads.h>
diff -Naur lirc-0.8.6/drivers/lirc_dev/lirc_dev.c lirc-0.8.6a/drivers/lirc_dev/lirc_dev.c
--- lirc-0.8.6/drivers/lirc_dev/lirc_dev.c	2009-08-31 18:57:55.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_dev/lirc_dev.c	2010-02-20 11:32:57.813510994 +0100
@@ -32,7 +32,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/sched.h>
diff -Naur lirc-0.8.6/drivers/lirc_dev/lirc_dev.h lirc-0.8.6a/drivers/lirc_dev/lirc_dev.h
--- lirc-0.8.6/drivers/lirc_dev/lirc_dev.h	2009-03-15 10:34:00.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_dev/lirc_dev.h	2010-02-20 11:32:57.813510994 +0100
@@ -30,14 +30,19 @@
 
 struct lirc_buffer {
 	wait_queue_head_t wait_poll;
-	spinlock_t lock;
+	spinlock_t fifo_lock;
 	unsigned int chunk_size;
 	unsigned int size; /* in chunks */
 	/* Using chunks instead of bytes pretends to simplify boundary checking
 	 * And should allow for some performance fine tunning later */
 #ifdef LIRC_HAVE_KFIFO
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 	struct kfifo *fifo;
 #else
+	struct kfifo fifo;
+	u8 fifo_initialized;
+#endif
+#else
 	unsigned int fill; /* in chunks */
 	int head, tail;    /* in chunks */
 	unsigned char *data;
@@ -47,12 +52,12 @@
 static inline void lirc_buffer_lock(struct lirc_buffer *buf,
 				    unsigned long *flags)
 {
-	spin_lock_irqsave(&buf->lock, *flags);
+	spin_lock_irqsave(&buf->fifo_lock, *flags);
 }
 static inline void lirc_buffer_unlock(struct lirc_buffer *buf,
 				      unsigned long *flags)
 {
-	spin_unlock_irqrestore(&buf->lock, *flags);
+	spin_unlock_irqrestore(&buf->fifo_lock, *flags);
 }
 static inline void _lirc_buffer_clear(struct lirc_buffer *buf)
 {
@@ -63,11 +68,20 @@
 #endif
 static inline void lirc_buffer_clear(struct lirc_buffer *buf)
 {
+	unsigned long flags;
+
 #ifdef LIRC_HAVE_KFIFO
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 	if (buf->fifo)
 		kfifo_reset(buf->fifo);
 #else
-	unsigned long flags;
+	if (buf->fifo_initialized) {
+		spin_lock_irqsave(&buf->fifo_lock, flags);
+		kfifo_reset(&buf->fifo);
+		spin_unlock_irqrestore(&buf->fifo_lock, flags);
+	}
+#endif
+#else
 	lirc_buffer_lock(buf, &flags);
 	_lirc_buffer_clear(buf);
 	lirc_buffer_unlock(buf, &flags);
@@ -77,31 +91,47 @@
 				    unsigned int chunk_size,
 				    unsigned int size)
 {
+	int ret = 0;
+
 	init_waitqueue_head(&buf->wait_poll);
-	spin_lock_init(&buf->lock);
+	spin_lock_init(&buf->fifo_lock);
 #ifndef LIRC_HAVE_KFIFO
 	_lirc_buffer_clear(buf);
 #endif
 	buf->chunk_size = chunk_size;
 	buf->size = size;
 #ifdef LIRC_HAVE_KFIFO
-	buf->fifo = kfifo_alloc(size*chunk_size, GFP_KERNEL, &buf->lock);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
+	buf->fifo = kfifo_alloc(size*chunk_size, GFP_KERNEL, &buf->fifo_lock);
 	if (!buf->fifo)
 		return -ENOMEM;
 #else
+	ret = kfifo_alloc(&buf->fifo, size * chunk_size, GFP_KERNEL);
+	if (ret == 0)
+		buf->fifo_initialized = 1;
+#endif
+#else
 	buf->data = kmalloc(size*chunk_size, GFP_KERNEL);
 	if (buf->data == NULL)
 		return -ENOMEM;
 	memset(buf->data, 0, size*chunk_size);
 #endif
-	return 0;
+
+	return ret;
 }
 static inline void lirc_buffer_free(struct lirc_buffer *buf)
 {
 #ifdef LIRC_HAVE_KFIFO
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
 	if (buf->fifo)
 		kfifo_free(buf->fifo);
 #else
+	if (buf->fifo_initialized) {
+		kfifo_free(&buf->fifo);
+		buf->fifo_initialized = 0;
+	}
+#endif
+#else
 	kfree(buf->data);
 	buf->data = NULL;
 	buf->head = 0;
@@ -111,6 +141,25 @@
 	buf->size = 0;
 #endif
 }
+
+#ifdef LIRC_HAVE_KFIFO
+static inline int lirc_buffer_len(struct lirc_buffer *buf)
+{
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
+	return kfifo_len(buf->fifo);
+#else
+	int len;
+	unsigned long flags;
+
+	spin_lock_irqsave(&buf->fifo_lock, flags);
+	len = kfifo_len(&buf->fifo);
+	spin_unlock_irqrestore(&buf->fifo_lock, flags);
+
+	return len;
+#endif
+}
+#endif
+
 #ifndef LIRC_HAVE_KFIFO
 static inline int  _lirc_buffer_full(struct lirc_buffer *buf)
 {
@@ -120,7 +169,7 @@
 static inline int  lirc_buffer_full(struct lirc_buffer *buf)
 {
 #ifdef LIRC_HAVE_KFIFO
-	return kfifo_len(buf->fifo) == buf->size * buf->chunk_size;
+	return lirc_buffer_len(buf) == buf->size * buf->chunk_size;
 #else
 	unsigned long flags;
 	int ret;
@@ -139,7 +188,7 @@
 static inline int  lirc_buffer_empty(struct lirc_buffer *buf)
 {
 #ifdef LIRC_HAVE_KFIFO
-	return !kfifo_len(buf->fifo);
+	return !lirc_buffer_len(buf);
 #else
 	unsigned long flags;
 	int ret;
@@ -158,7 +207,7 @@
 static inline int  lirc_buffer_available(struct lirc_buffer *buf)
 {
 #ifdef LIRC_HAVE_KFIFO
-	return buf->size - (kfifo_len(buf->fifo) / buf->chunk_size);
+	return buf->size - (lirc_buffer_len(buf) / buf->chunk_size);
 #else
 	unsigned long flags;
 	int ret;
@@ -177,21 +226,30 @@
 	buf->fill -= 1;
 }
 #endif
-static inline void lirc_buffer_read(struct lirc_buffer *buf,
-				    unsigned char *dest)
+static inline unsigned int lirc_buffer_read(struct lirc_buffer *buf,
+					    unsigned char *dest)
 {
+	unsigned int ret = 0;
+
 #ifdef LIRC_HAVE_KFIFO
-	if (kfifo_len(buf->fifo) >= buf->chunk_size)
-		kfifo_get(buf->fifo, dest, buf->chunk_size);
+	if (lirc_buffer_len(buf) >= buf->chunk_size)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
+		ret = kfifo_get(buf->fifo, dest, buf->chunk_size);
+#else
+		ret = kfifo_out_locked(&buf->fifo, dest, buf->chunk_size,
+				       &buf->fifo_lock);
+#endif
 #else
 	unsigned long flags;
 	lirc_buffer_lock(buf, &flags);
 	_lirc_buffer_read_1(buf, dest);
 	lirc_buffer_unlock(buf, &flags);
 #endif
+
+	return ret;
 }
 #ifndef LIRC_HAVE_KFIFO
-static inline void _lirc_buffer_write_1(struct lirc_buffer *buf,
+static inline  _lirc_buffer_write_1(struct lirc_buffer *buf,
 				      unsigned char *orig)
 {
 	memcpy(&buf->data[buf->tail*buf->chunk_size], orig, buf->chunk_size);
@@ -199,17 +257,26 @@
 	buf->fill++;
 }
 #endif
-static inline void lirc_buffer_write(struct lirc_buffer *buf,
-				     unsigned char *orig)
+static inline unsigned int lirc_buffer_write(struct lirc_buffer *buf,
+					     unsigned char *orig)
 {
+	unsigned int ret = 0;
+
 #ifdef LIRC_HAVE_KFIFO
-	kfifo_put(buf->fifo, orig, buf->chunk_size);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
+	ret = kfifo_put(buf->fifo, orig, buf->chunk_size);
+#else
+	ret = kfifo_in_locked(&buf->fifo, orig, buf->chunk_size,
+			      &buf->fifo_lock);
+#endif
 #else
 	unsigned long flags;
 	lirc_buffer_lock(buf, &flags);
 	_lirc_buffer_write_1(buf, orig);
 	lirc_buffer_unlock(buf, &flags);
 #endif
+
+	return ret;
 }
 #ifndef LIRC_HAVE_KFIFO
 static inline void _lirc_buffer_write_n(struct lirc_buffer *buf,
@@ -234,17 +301,26 @@
 	buf->fill += count;
 }
 #endif
-static inline void lirc_buffer_write_n(struct lirc_buffer *buf,
-				       unsigned char *orig, int count)
+static inline unsigned int lirc_buffer_write_n(struct lirc_buffer *buf,
+					       unsigned char *orig, int count)
 {
+	unsigned int ret = 0;
+
 #ifdef LIRC_HAVE_KFIFO
-	kfifo_put(buf->fifo, orig, count * buf->chunk_size);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 33)
+	ret = kfifo_put(buf->fifo, orig, count * buf->chunk_size);
+#else
+	ret = kfifo_in_locked(&buf->fifo, orig, count * buf->chunk_size,
+			      &buf->fifo_lock);
+#endif
 #else
 	unsigned long flags;
 	lirc_buffer_lock(buf, &flags);
 	_lirc_buffer_write_n(buf, orig, count);
 	lirc_buffer_unlock(buf, &flags);
 #endif
+
+	return ret;
 }
 
 struct lirc_driver {
diff -Naur lirc-0.8.6/drivers/lirc_i2c/lirc_i2c.c lirc-0.8.6a/drivers/lirc_i2c/lirc_i2c.c
--- lirc-0.8.6/drivers/lirc_i2c/lirc_i2c.c	2009-08-30 18:59:53.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_i2c/lirc_i2c.c	2010-02-20 11:32:57.813510994 +0100
@@ -399,8 +399,8 @@
 		.name	= "i2c ir driver",
 	},
 #endif
-	.id		= I2C_DRIVERID_EXP3, /* FIXME */
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 31)
+	.id		= I2C_DRIVERID_EXP3, /* FIXME */
 	.attach_adapter	= ir_probe,
 	.detach_client	= ir_remove,
 #else
@@ -492,20 +492,23 @@
 		ir->l.add_to_buf = add_to_buf_pv951;
 		break;
 	case 0x71:
-#ifdef I2C_HW_B_CX2341X
-		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_BT848) ||
-		    adap->id == (I2C_ALGO_BIT | I2C_HW_B_CX2341X)) {
-#else
-		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_BT848)) {
+
+
+
+#ifdef I2C_HW_B_CX2388x
+		/* Leadtek Winfast PVR2000 or Hauppauge HVR-1300 */
+		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_CX2388x))
+			strlcpy(ir->c.name, "Hauppauge HVR1300", I2C_NAME_SIZE);
+		else
 #endif
+		{
 			/*
 			 * The PVR150 IR receiver uses the same protocol as
 			 * other Hauppauge cards, but the data flow is
 			 * different, so we need to deal with it by its own.
 			 */
 			strlcpy(ir->c.name, "Hauppauge PVR150", I2C_NAME_SIZE);
-		} else /* I2C_HW_B_CX2388x */
-			strlcpy(ir->c.name, "Hauppauge HVR1300", I2C_NAME_SIZE);
+		}
 		ir->l.code_length = 13;
 		ir->l.add_to_buf = add_to_buf_haup_pvr150;
 		break;
@@ -516,19 +519,18 @@
 		break;
 	case 0x18:
 	case 0x1a:
-#ifdef I2C_HW_B_CX2341X
-		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_BT848) ||
-		    adap->id == (I2C_ALGO_BIT | I2C_HW_B_CX2341X)) {
+#ifdef I2C_HW_B_CX2388x
+		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_CX2388x)) {
+			strlcpy(ir->c.name, "Leadtek IR", I2C_NAME_SIZE);
+			ir->l.code_length = 8;
+			ir->l.add_to_buf = add_to_buf_pvr2000;
+		} else {
 #else
-		if (adap->id == (I2C_ALGO_BIT | I2C_HW_B_BT848)) {
+		{
 #endif
 			strlcpy(ir->c.name, "Hauppauge IR", I2C_NAME_SIZE);
 			ir->l.code_length = 13;
 			ir->l.add_to_buf = add_to_buf_haup;
-		} else { /* I2C_HW_B_CX2388x */
-			strlcpy(ir->c.name, "Leadtek IR", I2C_NAME_SIZE);
-			ir->l.code_length = 8;
-			ir->l.add_to_buf = add_to_buf_pvr2000;
 		}
 		break;
 	case 0x30:
diff -Naur lirc-0.8.6/drivers/lirc_igorplugusb/lirc_igorplugusb.c lirc-0.8.6a/drivers/lirc_igorplugusb/lirc_igorplugusb.c
--- lirc-0.8.6/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2009-08-02 11:54:10.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_igorplugusb/lirc_igorplugusb.c	2010-02-20 11:32:57.813510994 +0100
@@ -47,7 +47,11 @@
 #error "*******************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/module.h>
 #include <linux/kernel.h>
 #include <linux/kmod.h>
diff -Naur lirc-0.8.6/drivers/lirc_imon/lirc_imon.c lirc-0.8.6a/drivers/lirc_imon/lirc_imon.c
--- lirc-0.8.6/drivers/lirc_imon/lirc_imon.c	2009-09-11 06:56:18.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_imon/lirc_imon.c	2010-02-20 11:32:57.813510994 +0100
@@ -30,7 +30,11 @@
 #error "*** Sorry, this driver requires a 2.6 kernel"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #include <linux/errno.h>
 #include <linux/init.h>
diff -Naur lirc-0.8.6/drivers/lirc_it87/lirc_it87.c lirc-0.8.6a/drivers/lirc_it87/lirc_it87.c
--- lirc-0.8.6/drivers/lirc_it87/lirc_it87.c	2009-06-01 14:21:31.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_it87/lirc_it87.c	2010-02-20 11:32:57.813510994 +0100
@@ -36,7 +36,11 @@
 
 #include <linux/version.h>
 #include <linux/module.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
diff -Naur lirc-0.8.6/drivers/lirc_mceusb/lirc_mceusb.c lirc-0.8.6a/drivers/lirc_mceusb/lirc_mceusb.c
--- lirc-0.8.6/drivers/lirc_mceusb/lirc_mceusb.c	2009-09-02 16:04:02.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_mceusb/lirc_mceusb.c	2010-02-20 11:32:57.813510994 +0100
@@ -52,7 +52,11 @@
 #error "Sorry, this driver needs kernel version 2.6.5 or higher"
 #error "*******************************************************"
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
diff -Naur lirc-0.8.6/drivers/lirc_parallel/lirc_parallel.c lirc-0.8.6a/drivers/lirc_parallel/lirc_parallel.c
--- lirc-0.8.6/drivers/lirc_parallel/lirc_parallel.c	2010-02-20 11:32:42.345590859 +0100
+++ lirc-0.8.6a/drivers/lirc_parallel/lirc_parallel.c	2010-02-20 11:33:17.021577990 +0100
@@ -34,13 +34,21 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #include <linux/module.h>
 #include <linux/sched.h>
 #include <linux/errno.h>
 #include <linux/signal.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/fs.h>
 #include <linux/kernel.h>
 #include <linux/ioport.h>
diff -Naur lirc-0.8.6/drivers/lirc_sasem/lirc_sasem.c lirc-0.8.6a/drivers/lirc_sasem/lirc_sasem.c
--- lirc-0.8.6/drivers/lirc_sasem/lirc_sasem.c	2009-02-28 11:27:10.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_sasem/lirc_sasem.c	2010-02-20 11:32:57.817330867 +0100
@@ -41,7 +41,11 @@
 #error "*** Sorry, this driver requires kernel version 2.4.22 or higher"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #include <linux/errno.h>
 #include <linux/init.h>
diff -Naur lirc-0.8.6/drivers/lirc_serial/lirc_serial.c lirc-0.8.6a/drivers/lirc_serial/lirc_serial.c
--- lirc-0.8.6/drivers/lirc_serial/lirc_serial.c	2009-03-15 10:34:00.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_serial/lirc_serial.c	2010-02-20 11:32:57.817330867 +0100
@@ -60,7 +60,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #if defined(CONFIG_SERIAL) || defined(CONFIG_SERIAL_8250)
 #warning "******************************************"
diff -Naur lirc-0.8.6/drivers/lirc_sir/lirc_sir.c lirc-0.8.6a/drivers/lirc_sir/lirc_sir.c
--- lirc-0.8.6/drivers/lirc_sir/lirc_sir.c	2009-07-10 00:24:23.000000000 +0200
+++ lirc-0.8.6a/drivers/lirc_sir/lirc_sir.c	2010-02-20 11:32:57.817330867 +0100
@@ -45,7 +45,11 @@
 # include <config.h>
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #if !defined(CONFIG_SERIAL_MODULE)
 #if !defined(LIRC_ON_SA1100)
 #warning "******************************************"
diff -Naur lirc-0.8.6/drivers/lirc_streamzap/lirc_streamzap.c lirc-0.8.6a/drivers/lirc_streamzap/lirc_streamzap.c
--- lirc-0.8.6/drivers/lirc_streamzap/lirc_streamzap.c	2009-03-15 10:34:00.000000000 +0100
+++ lirc-0.8.6a/drivers/lirc_streamzap/lirc_streamzap.c	2010-02-20 11:32:57.817330867 +0100
@@ -35,7 +35,11 @@
 #error "Sorry, this driver needs kernel version 2.4.0 or higher"
 #error "*******************************************************"
 #endif
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 #include <linux/kernel.h>
 #include <linux/errno.h>
 #include <linux/init.h>
diff -Naur lirc-0.8.6/drivers/lirc_wpc8769l/lirc_wpc8769l.c lirc-0.8.6a/drivers/lirc_wpc8769l/lirc_wpc8769l.c
--- lirc-0.8.6/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2010-02-20 11:32:42.341423103 +0100
+++ lirc-0.8.6a/drivers/lirc_wpc8769l/lirc_wpc8769l.c	2010-02-20 11:32:57.817330867 +0100
@@ -37,7 +37,11 @@
 #error "**********************************************************"
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
+#include <generated/autoconf.h>
+#else
 #include <linux/autoconf.h>
+#endif
 
 #include <linux/module.h>
 #include <linux/errno.h>
