Description: Fixes problems with text2skin skin enigma.
Author: Andreas Brugger <brougs78@gmx.net>
Origin: http://vdrportal.de/board/thread.php?postid=343665#post343665

Index: vdr/osdbase.c
===================================================================
--- vdr.orig/osdbase.c	2012-03-11 18:35:36.082012638 +0100
+++ vdr/osdbase.c	2012-03-11 18:36:56.002012641 +0100
@@ -312,6 +312,7 @@
 
 void cOsdMenu::CursorUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -350,6 +351,7 @@
 
 void cOsdMenu::CursorDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -390,6 +392,7 @@
 
 void cOsdMenu::PageUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current -= displayMenuItems;
@@ -424,6 +427,7 @@
 
 void cOsdMenu::PageDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current += displayMenuItems;
