Description: Use ncursesw header file
Author: Tobias Grimm <tg@e-tobi.net>

Index: vdr-1.7.27/PLUGINS/src/skincurses/skincurses.c
===================================================================
--- vdr-1.7.27.orig/PLUGINS/src/skincurses/skincurses.c	2012-03-11 15:42:52.000000000 +0100
+++ vdr-1.7.27/PLUGINS/src/skincurses/skincurses.c	2012-03-25 14:18:05.344975038 +0200
@@ -6,7 +6,7 @@
  * $Id: skincurses.c 2.8 2012/03/11 14:42:52 kls Exp $
  */
 
-#include <ncurses.h>
+#include <ncursesw/ncurses.h>
 #include <vdr/osd.h>
 #include <vdr/plugin.h>
 #include <vdr/skins.h>
