Description: Patch for VDR
Author: Tom Joad <TomJoad@vdr-portal> 
Origin: http://www.vdr-portal.de/board16-video-disk-recorder/board8-vdr-grundlagen/p962639-bug-pause-100-cpu-auslastung/#post962639

diff -urNad vdr-1.7.22-orig/dvbplayer.c vdr-1.7.22-dvbp/dvbplayer.c
--- vdr-1.7.22-orig/dvbplayer.c	2011-12-21 18:04:16.000000000 +0100
+++ vdr-1.7.22-dvbp/dvbplayer.c	2011-12-21 18:06:15.316883986 +0100
@@ -428,9 +428,9 @@
            cPoller Poller;
            DevicePoll(Poller, 10);
            Sleep = false;
+           }
            if (playMode == pmStill || playMode==pmPause)
               cCondWait::SleepMs(3);
-           }
         {
           LOCK_THREAD;
 
