--- a/Makefile	2015-01-10 23:31:18.751993045 +0100
+++ b/Makefile	2015-01-10 23:31:18.763993041 +0100
@@ -1225,7 +1225,7 @@
 $(sort $(vmlinux-deps) $(subdir-modorder)): descend ;
 
 filechk_kernel.release = \
-	echo "$(KERNELVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))"
+	echo "$(KERNELVERSION)"
 
 # Store (new) KERNELRELEASE string in include/config/kernel.release
 include/config/kernel.release: FORCE
