--- at-3.1.8/Makefile.in.org	2012-05-17 19:31:59.402267683 +0200
+++ at-3.1.8/Makefile.in	2012-05-17 19:32:12.774271757 +0200
@@ -5,7 +5,7 @@
 VPATH 		= $(srcdir)
 
 install_prefix	=
-prefix		= @prefix@
+prefix		= ./.install
 exec_prefix	= @exec_prefix@
 bindir		= @bindir@
 sbindir		= @sbindir@
--- at-3.1.13/Makefile.in.orig	2012-09-27 23:41:00.100427543 +0200
+++ at-3.1.13/Makefile.in	2012-09-27 23:41:44.396428305 +0200
@@ -14,7 +14,7 @@
 man8dir		= $(mandir)/man8
 docdir		= $(prefix)/doc
 atdocdir	= $(docdir)/at
-etcdir		= @ETCDIR@
+etcdir		= ./.install/etc
 
 DAEMON_USERNAME	= @DAEMON_USERNAME@
 DAEMON_GROUPNAME= @DAEMON_GROUPNAME@
