--- at-3.1.13/atd.c.org	2012-05-18 18:48:04.695774001 +0200
+++ at-3.1.13/atd.c	2012-05-18 18:48:34.595774017 +0200
@@ -503,11 +503,11 @@
 
 	    chdir ("/");
 
-#if defined(SENDMAIL)
-	    execl(SENDMAIL, "sendmail", "-i", mailname, (char *) NULL);
-#else
-#error      "No mail command specified."
-#endif
+//#if defined(SENDMAIL)
+//	    execl(SENDMAIL, "sendmail", "-i", mailname, (char *) NULL);
+//#else
+//#error      "No mail command specified."
+//#endif
 	    perr("Exec failed for mail command");
 
 	PRIV_END
