diff -Naur crystalhd-3.9.0-git+20101203.orig//linux_lib/libcrystalhd/libcrystalhd_if.h crystalhd-3.9.0-git+20101203/linux_lib/libcrystalhd/libcrystalhd_if.h
--- crystalhd-3.9.0-git+20101203.orig//linux_lib/libcrystalhd/libcrystalhd_if.h	2010-12-05 11:27:20.892122000 +0100
+++ crystalhd-3.9.0-git+20101203/linux_lib/libcrystalhd/libcrystalhd_if.h	2010-12-05 20:29:46.142122019 +0100
@@ -29,7 +29,9 @@
 #ifndef _BCM_LDIL_IF_H_
 #define _BCM_LDIL_IF_H_
 
+#include <inttypes.h>
 #include "bc_dts_defs.h"
+#include "bc_dts_types.h"
 
 #define FLEA_MAX_TRICK_MODE_SPEED	6
 
