diff -Naur imx6-lib-11.11.01.orig/ipu/mxc_ipu_hl_lib.c imx6-lib-11.11.01/ipu/mxc_ipu_hl_lib.c
--- imx6-lib-11.11.01.orig/ipu/mxc_ipu_hl_lib.c	2011-12-23 14:56:33.955127520 +0100
+++ imx6-lib-11.11.01/ipu/mxc_ipu_hl_lib.c	2011-12-23 14:57:55.423129435 +0100
@@ -41,7 +41,7 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include "mxc_ipu_hl_lib.h"
 
 #define DBG_DEBUG		3
diff -Naur imx6-lib-11.11.01.orig/ipu/mxc_ipu_hl_lib_dummy.c imx6-lib-11.11.01/ipu/mxc_ipu_hl_lib_dummy.c
--- imx6-lib-11.11.01.orig/ipu/mxc_ipu_hl_lib_dummy.c	2011-12-23 14:56:33.955127520 +0100
+++ imx6-lib-11.11.01/ipu/mxc_ipu_hl_lib_dummy.c	2011-12-23 14:58:18.571129749 +0100
@@ -37,7 +37,7 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include "mxc_ipu_hl_lib.h"
 
 int mxc_ipu_lib_task_init(ipu_lib_input_param_t * input,
diff -Naur imx6-lib-11.11.01.orig/screenlayer/ScreenLayer.c imx6-lib-11.11.01/screenlayer/ScreenLayer.c
--- imx6-lib-11.11.01.orig/screenlayer/ScreenLayer.c	2011-12-23 14:56:33.951127384 +0100
+++ imx6-lib-11.11.01/screenlayer/ScreenLayer.c	2011-12-23 14:57:36.699128268 +0100
@@ -26,7 +26,7 @@
 #include <semaphore.h>
 #include <linux/ipu.h>
 #include <linux/mxcfb.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
