diff -Naur imx6-lib-11.11.01.orig/Makefile imx6-lib-11.11.01/Makefile
--- imx6-lib-11.11.01.orig/Makefile	2011-12-23 14:56:33.951127384 +0100
+++ imx6-lib-11.11.01/Makefile	2011-12-23 14:57:02.579127998 +0100
@@ -1,6 +1,6 @@
 # Get all dirs with a Makefile
 TMP_DIRS := $(foreach dir, $(wildcard *), $(wildcard $(dir)/Makefile))
-DIRS := $(patsubst %/Makefile,%,$(TMP_DIRS))
+DIRS := ipu vpu
 
 .PHONY: all install clean
 .PHONY: $(DIRS)
