diff -Naur libmad-0.15.1b.orig/configure libmad-0.15.1b/configure
--- libmad-0.15.1b.orig/configure	2010-07-23 22:24:39.942478967 +0200
+++ libmad-0.15.1b/configure	2010-07-23 22:24:52.202479084 +0200
@@ -19099,7 +19099,6 @@
     case "$optimize" in
 	-O|"-O "*)
 	    optimize="-O"
-	    optimize="$optimize -fforce-mem"
 	    optimize="$optimize -fforce-addr"
 	    : #x optimize="$optimize -finline-functions"
 	    : #- optimize="$optimize -fstrength-reduce"
