--- marvell-ipp-0.2.1+hard+soft/example/Rules.make	2012-12-28 23:14:55.305981958 +0100
+++ marvell-ipp-0.2.1+hard+soft/example/Rules.make-new	2012-12-28 23:16:50.929977123 +0100
@@ -24,11 +24,13 @@
 else
 CFLAGS= -mcpu=iwmmxt -mtune=iwmmxt
 endif
+LIB_DIR=lib
 else
 #PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi/bin
-PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi-4.4.4/bin
-TOOLCHAIN_PREFIX=arm-marvell-linux-gnueabi
-CFLAGS= -mcpu=marvell-f 
+PATH_GNU_BIN=$(MY_PATH_GNU_BIN)
+TOOLCHAIN_PREFIX=$(MY_TOOLCHAIN_PREFIX)
+CFLAGS=$(MY_CFLAGS)
+LIB_DIR=$(MY_LIBDIR)
 endif
 
 #==============================================================================
@@ -53,7 +55,7 @@
 # User root path												(user update)
 #==============================================================================
 PATH_USR_ROOT=../../../..
-PATH_USR_LIB=$(PATH_USR_ROOT)/lib
+PATH_USR_LIB=$(PATH_USR_ROOT)/$(LIB_DIR)
 PATH_USR_BUILD=$(PATH_USR_ROOT)/example/$(CODEC_NAME)/build
 PATH_USR_SRC=$(PATH_USR_ROOT)/example/$(CODEC_NAME)/src
 PATH_USR_BIN=$(PATH_USR_ROOT)/example/$(CODEC_NAME)/bin
