--- marvell-ipp-0.2.1+hard+soft/example/Rules.make	2011-03-07 18:51:40.000000000 +0100
+++ marvell-ipp-0.2.1+hard+soft/example/Rules.make-new	2012-10-14 15:40:10.774442582 +0200
@@ -26,9 +26,9 @@
 endif
 else
 #PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi/bin
-PATH_GNU_BIN=/usr/local/arm-marvell-linux-gnueabi-4.4.4/bin
-TOOLCHAIN_PREFIX=arm-marvell-linux-gnueabi
-CFLAGS= -mcpu=marvell-f 
+PATH_GNU_BIN=$(MY_PATH_GNU_BIN)
+TOOLCHAIN_PREFIX=$(MY_TOOLCHAIN_PREFIX)
+CFLAGS=$(MY_CFLAGS)
 endif
 
 #==============================================================================
