diff -Naur DirectFB-1.0.0-rc2.orig/lib/direct/ppcasm_memcpy_cachable.S DirectFB-1.0.0-rc2/lib/direct/ppcasm_memcpy_cachable.S
--- DirectFB-1.0.0-rc2.orig/lib/direct/ppcasm_memcpy_cachable.S	2006-10-29 16:49:02.000000000 +0100
+++ DirectFB-1.0.0-rc2/lib/direct/ppcasm_memcpy_cachable.S	2006-12-17 15:39:59.000000000 +0100
@@ -11,8 +11,6 @@
 
 #define __ASSEMBLY__
 
-#include <linux/config.h>
-
 #if defined(CONFIG_8xx) || defined(CONFIG_403GCX)
 #define L1_CACHE_LINE_SIZE       16
 #define LG_L1_CACHE_LINE_SIZE     4 
