diff -Naur DirectFB-1.4.3.orig/configure DirectFB-1.4.3/configure
--- DirectFB-1.4.3.orig/configure	2010-06-27 00:39:42.254825799 +0200
+++ DirectFB-1.4.3/configure	2010-06-27 00:39:49.794688254 +0200
@@ -22759,7 +22759,7 @@
 
 if test "$enable_x11" = "yes"; then
   CFLAGS_saved="$CFLAGS"
-  CFLAGS="$CFLAGS -I/usr/X11R6/include"
+  CFLAGS="$CFLAGS"
   if test "${ac_cv_header_X11_X_h+set}" = set; then
   { echo "$as_me:$LINENO: checking for X11/X.h" >&5
 echo $ECHO_N "checking for X11/X.h... $ECHO_C" >&6; }
@@ -22900,8 +22900,8 @@
 echo "$as_me: WARNING:
 *** no X11/X.h found -- building without X11 support." >&2;}
   else
-    X11_LIBS="-L/usr/X11R6/lib -lX11 -lXext"
-    X11_CFLAGS="-I/usr/X11R6/include"
+    X11_LIBS="-lX11 -lXext"
+    X11_CFLAGS=""
   fi
 fi
 
