diff -Naur DirectFB-1.0.0-rc3.orig/inputdrivers/linux_input/linux_input.c DirectFB-1.0.0-rc3/inputdrivers/linux_input/linux_input.c
--- DirectFB-1.0.0-rc3.orig/inputdrivers/linux_input/linux_input.c	2007-02-01 22:55:29.000000000 +0100
+++ DirectFB-1.0.0-rc3/inputdrivers/linux_input/linux_input.c	2007-02-01 22:56:41.000000000 +0100
@@ -50,6 +50,7 @@
 #define BTN_TOOL_TRIPLETAP	0x14e
 #endif
 
+#include <linux/types.h>
 #include <linux/input.h>
 #ifndef KEY_OK
 /* Linux kernel 2.5.42+ defines additional keys in linux/input.h */
