--- libccid-1.4.5/Makefile.org	2012-05-31 22:37:41.113993627 +0200
+++ libccid-1.4.5/Makefile	2012-05-31 22:38:07.273993641 +0200
@@ -219,7 +219,7 @@
 build_vendor = unknown
 builddir = .
 bundle = ifd-ccid.bundle
-ccidtwindir = /root/openbricks/build/build.kbb.eglibc/toolchain/x86_64-openbricks-linux-gnu/sysroot/usr/local/local/pcsc/drivers/serial
+ccidtwindir = /usr/local/local/pcsc/drivers/serial
 datadir = ${datarootdir}
 datarootdir = ${prefix}/share
 docdir = ${datarootdir}/doc/${PACKAGE_TARNAME}
@@ -254,7 +254,7 @@
 top_build_prefix = 
 top_builddir = .
 top_srcdir = .
-usbdropdir = /root/openbricks/build/build.kbb.eglibc/toolchain/x86_64-openbricks-linux-gnu/sysroot/usr/local/local/pcsc/drivers
+usbdropdir = /usr/local/local/pcsc/drivers
 ACLOCAL_AMFLAGS = -I m4
 SUBDIRS = m4 readers examples contrib src
 EXTRA_DIST = bootstrap ChangeLog SCARDGETATTRIB.txt \
--- libccid-1.4.5/src/Makefile.org	2012-05-31 22:38:14.263993645 +0200
+++ libccid-1.4.5/src/Makefile	2012-05-31 22:38:35.743993655 +0200
@@ -292,7 +292,7 @@
 build_vendor = unknown
 builddir = .
 bundle = ifd-ccid.bundle
-ccidtwindir = /root/openbricks/build/build.kbb.eglibc/toolchain/x86_64-openbricks-linux-gnu/sysroot/usr/local/local/pcsc/drivers/serial
+ccidtwindir = /usr/local/local/pcsc/drivers/serial
 datadir = ${datarootdir}
 datarootdir = ${prefix}/share
 docdir = ${datarootdir}/doc/${PACKAGE_TARNAME}
@@ -327,7 +327,7 @@
 top_build_prefix = ../
 top_builddir = ..
 top_srcdir = ..
-usbdropdir = /root/openbricks/build/build.kbb.eglibc/toolchain/x86_64-openbricks-linux-gnu/sysroot/usr/local/local/pcsc/drivers
+usbdropdir = /usr/local/local/pcsc/drivers
 CCID_BUNDLE = $(bundle)
 CCID_LIB = libccid.$(DYN_LIB_EXT)
 CCIDTWIN_LIB = libccidtwin.$(DYN_LIB_EXT)
