--- a/cmake/modules/FindSSE.cmake	2017-08-30 10:00:59.574192890 +0200
+++ b/cmake/modules/FindSSE.cmake	2017-08-30 10:00:59.594192826 +0200
@@ -3,8 +3,14 @@
 include(TestCXXAcceptsFlag)

 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-   if(CPU MATCHES "x86_64" OR CPU MATCHES "i.86")
-     exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+   if(CPU MATCHES "x86_64" OR CPU MATCHES "i.86" OR CPU MATCHES "generic")
+     if(NOT CMAKE_CROSSCOMPILING)
+       exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+     elseif(CPU MATCHES "x86_64" OR ARCH MATCHES "x86_64")
+       set(CPUINFO sse2 pni ssse3 sse4_1 sse4_2)
+     else()
+       set(CPUINFO sse2)
+     endif()

      string(REGEX REPLACE "^.*(sse).*$" "\\1" _SSE_THERE ${CPUINFO})
      string(COMPARE EQUAL "sse" "${_SSE_THERE}" _SSE_TRUE)
