--- a/cmake/modules/FindTexturePacker.cmake	2017-08-27 12:44:36.088140534 +0200
+++ b/cmake/modules/FindTexturePacker.cmake	2017-08-27 12:44:36.100140444 +0200
@@ -12,10 +12,10 @@
 #   TexturePacker::TexturePacker   - The TexturePacker executable

 if(NOT TARGET TexturePacker::TexturePacker)
-  if(KODI_DEPENDSBUILD)
+  if(KODI_DEPENDSBUILD OR CMAKE_CROSSCOMPILING)
     add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
     set_target_properties(TexturePacker::TexturePacker PROPERTIES
-                                                       IMPORTED_LOCATION "${NATIVEPREFIX}/bin/TexturePacker")
+                                                       IMPORTED_LOCATION "${CORE_SOURCE_DIR}/tools/bin/TexturePacker")
   elseif(WIN32)
     add_executable(TexturePacker::TexturePacker IMPORTED GLOBAL)
     set_target_properties(TexturePacker::TexturePacker PROPERTIES
--- a/cmake/scripts/linux/Install.cmake	2017-08-29 14:27:43.686488706 +0200
+++ b/cmake/scripts/linux/Install.cmake	2017-08-29 14:27:43.742479838 +0200
@@ -139,7 +139,7 @@
         COMPONENT kodi)

 # Install kodi-tools-texturepacker
-if(NOT WITH_TEXTUREPACKER)
+if(NOT WITH_TEXTUREPACKER AND NOT CMAKE_CROSSCOMPILING)
   install(PROGRAMS $<TARGET_FILE:TexturePacker::TexturePacker>
           DESTINATION ${bindir}
           COMPONENT kodi-tools-texturepacker)
--- a/cmake/modules/FindSSE.cmake	2017-08-30 10:00:59.574192890 +0200
+++ b/cmake/modules/FindSSE.cmake	2017-08-30 10:00:59.594192826 +0200
@@ -3,8 +3,14 @@
 include(TestCXXAcceptsFlag)

 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-   if(CPU MATCHES "x86_64" OR CPU MATCHES "i.86")
-     exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+   if(CPU MATCHES "x86_64" OR CPU MATCHES "i.86" OR CPU MATCHES "generic")
+     if(NOT CMAKE_CROSSCOMPILING)
+       exec_program(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
+     elseif(CPU MATCHES "x86_64" OR ARCH MATCHES "x86_64")
+       set(CPUINFO sse2 pni ssse3 sse4_1 sse4_2)
+     else()
+       set(CPUINFO sse2)
+     endif()

      string(REGEX REPLACE "^.*(sse).*$" "\\1" _SSE_THERE ${CPUINFO})
      string(COMPARE EQUAL "sse" "${_SSE_THERE}" _SSE_TRUE)
--- a/cmake/modules/FindJsonSchemaBuilder.cmake	2017-09-06 17:56:42.525579479 +0200
+++ b/cmake/modules/FindJsonSchemaBuilder.cmake	2017-09-06 17:58:17.180919514 +0200
@@ -12,10 +12,10 @@
 #   JsonSchemaBuilder::JsonSchemaBuilder   - The JsonSchemaBuilder executable

 if(NOT TARGET JsonSchemaBuilder::JsonSchemaBuilder)
-  if(KODI_DEPENDSBUILD)
+  if(KODI_DEPENDSBUILD OR CMAKE_CROSSCOMPILING)
     add_executable(JsonSchemaBuilder::JsonSchemaBuilder IMPORTED GLOBAL)
     set_target_properties(JsonSchemaBuilder::JsonSchemaBuilder PROPERTIES
-                                                       IMPORTED_LOCATION "${NATIVEPREFIX}/bin/JsonSchemaBuilder")
+                                                       IMPORTED_LOCATION "${CORE_SOURCE_DIR}/tools/bin/JsonSchemaBuilder")
   elseif(CORE_SYSTEM_NAME STREQUAL windows OR CORE_SYSTEM_NAME STREQUAL windowsstore)
     add_executable(JsonSchemaBuilder::JsonSchemaBuilder IMPORTED GLOBAL)
     set_target_properties(JsonSchemaBuilder::JsonSchemaBuilder PROPERTIES
