diff --git a/tools/depends/native/TexturePacker/src/Makefile.am b/tools/depends/native/TexturePacker/src/Makefile.am
index c2ac65b825b0..b6d6a67bb3f2 100644
--- a/tools/depends/native/TexturePacker/src/Makefile.am
+++ b/tools/depends/native/TexturePacker/src/Makefile.am
@@ -10,7 +10,7 @@ AM_CPPFLAGS = \
   -I@KODI_SRC_DIR@/lib \
   -I@KODI_SRC_DIR@/xbmc \
   -I@KODI_SRC_DIR@/xbmc/guilib \
-  -I@KODI_SRC_DIR@/xbmc/linux \
+  -I@KODI_SRC_DIR@/xbmc/platform/linux \
   @CPPFLAGS@
 
 AM_LDFLAGS = @LIBS@ @STATIC_FLAG@
