From 98f05a368f77e71aba65fa7323a0b4d7b75a6faf Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 23 Oct 2017 14:01:08 +0200
Subject: [PATCH 03/13] fixup! windowing/gbm: add atomic modesetting

---
 xbmc/windowing/gbm/DRMAtomic.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/xbmc/windowing/gbm/DRMAtomic.cpp b/xbmc/windowing/gbm/DRMAtomic.cpp
index 3f80bd60a9..fad616ad90 100644
--- a/xbmc/windowing/gbm/DRMAtomic.cpp
+++ b/xbmc/windowing/gbm/DRMAtomic.cpp
@@ -425,7 +425,6 @@ void CDRMAtomic::DestroyDrmAtomic()
 
 bool CDRMAtomic::SetVideoMode(RESOLUTION_INFO res)
 {
-  CDRMUtils::GetMode(res);
   m_drm->need_modeset = true;
 
   return true;
-- 
2.11.0

