From f6613c6ace0ed97afb14e1379adc0e7443c6ab90 Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sat, 11 Nov 2017 23:59:07 +0100
Subject: [PATCH 06/13] windowing/gbm: register DRM PRIME video codec and
 renderer

---
 xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp b/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
index 7088c76577..1000f5b1c5 100644
--- a/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
+++ b/xbmc/windowing/gbm/WinSystemGbmGLESContext.cpp
@@ -24,6 +24,9 @@
 #include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGLES.h"
 #endif
 
+#include "cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.h"
+#include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererDRMPRIME.h"
+
 #include "cores/RetroPlayer/process/RPProcessInfo.h"
 #include "cores/RetroPlayer/rendering/VideoRenderers/RPRendererGuiTexture.h"
 #include "cores/VideoPlayer/DVDCodecs/DVDFactoryCodec.h"
@@ -62,6 +65,9 @@ bool CWinSystemGbmGLESContext::InitWindowSystem()
   }
 #endif
 
+  CRendererDRMPRIME::Register();
+  CDVDVideoCodecDRMPRIME::Register();
+
   return true;
 }
 
-- 
2.11.0

