From d6f6155a060e0a5b94e3257f7c5d75fa1a8be321 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 24 Sep 2016 15:33:46 +0200
Subject: [PATCH 33/37] Initialize first two VPU frame buffers (DivX logo
 issue)

---
 .../DVDCodecs/Video/DVDVideoCodecIMX.cpp           | 39 +++++++++++-----------
 1 file changed, 19 insertions(+), 20 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
index f692482..e3fa72e 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
@@ -371,37 +371,36 @@ bool CIMXCodec::VpuAllocFrameBuffers()
       ptrVirt = (unsigned char*)Align(ptrVirt,nAlign);
     }
 
-    VpuFrameBuffer vpuFrameBuffer;
-    m_vpuFrameBuffers.push_back(vpuFrameBuffer);
+    VpuFrameBuffer vpuFrameBuffer = {};
 
     // fill stride info
-    m_vpuFrameBuffers[i].nStrideY           = yStride;
-    m_vpuFrameBuffers[i].nStrideC           = uvStride;
+    vpuFrameBuffer.nStrideY           = yStride;
+    vpuFrameBuffer.nStrideC           = uvStride;
 
     // fill phy addr
-    m_vpuFrameBuffers[i].pbufY              = ptr;
-    m_vpuFrameBuffers[i].pbufCb             = ptr + ySize;
+    vpuFrameBuffer.pbufY              = ptr;
+    vpuFrameBuffer.pbufCb             = ptr + ySize;
 #ifdef IMX_INPUT_FORMAT_I420
-    m_vpuFrameBuffers[i].pbufCr             = ptr + ySize + uSize;
-#else
-    m_vpuFrameBuffers[i].pbufCr             = 0;
+    vpuFrameBuffer.pbufCr             = ptr + ySize + uSize;
 #endif
-    m_vpuFrameBuffers[i].pbufMvCol          = ptr + ySize + uSize + vSize;
+    vpuFrameBuffer.pbufMvCol          = ptr + ySize + uSize + vSize;
 
     // fill virt addr
-    m_vpuFrameBuffers[i].pbufVirtY          = ptrVirt;
-    m_vpuFrameBuffers[i].pbufVirtCb         = ptrVirt + ySize;
+    vpuFrameBuffer.pbufVirtY          = ptrVirt;
+    vpuFrameBuffer.pbufVirtCb         = ptrVirt + ySize;
 #ifdef IMX_INPUT_FORMAT_I420
-    m_vpuFrameBuffers[i].pbufVirtCr         = ptrVirt + ySize + uSize;
-#else
-    m_vpuFrameBuffers[i].pbufVirtCr         = 0;
+    vpuFrameBuffer.pbufVirtCr         = ptrVirt + ySize + uSize;
 #endif
-    m_vpuFrameBuffers[i].pbufVirtMvCol      = ptrVirt + ySize + uSize + vSize;
+    vpuFrameBuffer.pbufVirtMvCol      = ptrVirt + ySize + uSize + vSize;
 
-    m_vpuFrameBuffers[i].pbufY_tilebot      = 0;
-    m_vpuFrameBuffers[i].pbufCb_tilebot     = 0;
-    m_vpuFrameBuffers[i].pbufVirtY_tilebot  = 0;
-    m_vpuFrameBuffers[i].pbufVirtCb_tilebot = 0;
+    if (i < 2)
+    {
+      memset(vpuFrameBuffer.pbufVirtY, 16, ySize);
+      memset(vpuFrameBuffer.pbufVirtCb, 128, uSize + vSize);
+      memset(vpuFrameBuffer.pbufVirtMvCol, 0, mvSize);
+    }
+
+    m_vpuFrameBuffers.push_back(vpuFrameBuffer);
   }
 
   if (VPU_DEC_RET_SUCCESS != VPU_DecRegisterFrameBuffer(m_vpuHandle, &m_vpuFrameBuffers[0], m_vpuFrameBuffers.size()))
-- 
1.9.1

