--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2018-04-29 12:28:28.441445418 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2018-04-29 13:05:48.663232412 +0200
@@ -65,7 +65,8 @@
 #endif
 
 CIMXContext   g_IMXContext;
-std::shared_ptr<CIMXCodec> g_IMXCodec;
+static CCriticalSection g_IMXMutex;
+static std::shared_ptr<CIMXCodec> g_IMXCodec;
 
 std::list<VpuFrameBuffer*> m_recycleBuffers;
 
@@ -77,22 +78,26 @@
 
 CDVDVideoCodecIMX::~CDVDVideoCodecIMX()
 {
+  CSingleLock lk(g_IMXMutex);
+
   m_IMXCodec.reset();
+  
   if (g_IMXCodec.use_count() == 1)
     g_IMXCodec.reset();
 }
 
 bool CDVDVideoCodecIMX::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 {
-  if (!g_IMXCodec)
   {
-    m_IMXCodec.reset(new CIMXCodec);
-    g_IMXCodec = m_IMXCodec;
-  }
-  else
+    CSingleLock lk(g_IMXMutex);
+
+    if (!g_IMXCodec)
+      g_IMXCodec.reset(new CIMXCodec);
+
     m_IMXCodec = g_IMXCodec;
+  }
 
-  return g_IMXCodec->Open(hints, options, m_pFormatName, &m_processInfo);
+  return m_IMXCodec->Open(hints, options, m_pFormatName, &m_processInfo);
 }
 
 unsigned CDVDVideoCodecIMX::GetAllowedReferences()
@@ -499,7 +504,7 @@
   }
 #endif
 
-  if (m_hints != hints && g_IMXCodec->IsRunning())
+  if (m_hints != hints && IsRunning())
   {
     StopThread(false);
     ProcessSignals(SIGNAL_FLUSH);
@@ -761,7 +766,7 @@
     return VC_BUFFER;
 
   int ret = 0;
-  if (!g_IMXCodec->IsRunning())
+  if (!IsRunning())
   {
     if (!m_decInput.full())
     {
@@ -785,8 +790,8 @@
       m_decOpenParam.nMapType = MAPTYPE_TILED_FRAME;
 
       ptrn.Flush();
-      g_IMXCodec->Create();
-      g_IMXCodec->WaitStartup();
+      Create();
+      WaitStartup();
     }
   }
 
