--- a/configure.in	2013-12-14 01:44:01.933705064 +0100
+++ b/configure.in	2013-12-14 08:06:25.153027168 +0100
@@ -2282,13 +2282,13 @@
 
 USE_PVR_ADDONS=0
 DISABLE_PVR_ADDON_CONFIG=1
-if [[ -f "pvr-addons/Makefile.am" ]]; then
-  final_message="$final_message\n  PVR add-ons:\t\tYes"
-  USE_PVR_ADDONS=1
-  DISABLE_PVR_ADDON_CONFIG=0
-else
+#if [[ -f "pvr-addons/Makefile.am" ]]; then
+#  final_message="$final_message\n  PVR add-ons:\t\tYes"
+#  USE_PVR_ADDONS=1
+#  DISABLE_PVR_ADDON_CONFIG=0
+#else
   final_message="$final_message\n  PVR add-ons:\t\tNo"
-fi
+#fi
 
 OUTPUT_FILES="Makefile \
     Makefile.include \
