--- a/xbmc/powermanagement/PowerManager.cpp	2014-05-11 20:47:12.304539963 +0200
+++ b/xbmc/powermanagement/PowerManager.cpp	2014-05-11 20:47:12.320539963 +0200
@@ -78,13 +78,13 @@
   m_instance = new CAndroidPowerSyscall();
 #elif defined(TARGET_POSIX)
 #if defined(HAS_DBUS)
-  if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
+  /*if (CConsoleUPowerSyscall::HasConsoleKitAndUPower())
     m_instance = new CConsoleUPowerSyscall();
   else if (CConsoleDeviceKitPowerSyscall::HasDeviceConsoleKit())
     m_instance = new CConsoleDeviceKitPowerSyscall();
   else if (CLogindUPowerSyscall::HasLogind())
     m_instance = new CLogindUPowerSyscall();
-  else if (CUPowerSyscall::HasUPower())
+  else*/ if (CUPowerSyscall::HasUPower())
     m_instance = new CUPowerSyscall();
 #if defined(HAS_HAL)
   else if(1)
