--- a/xbmc/powermanagement/PowerManager.cpp	2015-06-10 23:55:59.644280840 +0200
+++ b/xbmc/powermanagement/PowerManager.cpp	2015-06-10 23:55:59.656280840 +0200
@@ -83,12 +83,14 @@
   std::list< std::pair< std::function<bool()>,
                         std::function<IPowerSyscall*()> > > powerManagers =
   {
+#if 0    
     std::make_pair(CConsoleUPowerSyscall::HasConsoleKitAndUPower,
                    [] { return new CConsoleUPowerSyscall(); }),
     std::make_pair(CConsoleDeviceKitPowerSyscall::HasDeviceConsoleKit,
                    [] { return new CConsoleDeviceKitPowerSyscall(); }),
     std::make_pair(CLogindUPowerSyscall::HasLogind,
                    [] { return new CLogindUPowerSyscall(); }),
+#endif                   
     std::make_pair(CUPowerSyscall::HasUPower,
                    [] { return new CUPowerSyscall(); })
   };
