--- a/tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp	2015-01-25 09:10:20.616219238 +0100
+++ b/tools/depends/native/TexturePacker/src/decoder/GifHelper.cpp	2015-01-25 09:10:20.636219236 +0100
@@ -82,7 +82,12 @@
 
 GifHelper::~GifHelper()
 {
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+  int err;
+  DGifCloseFile(m_gif, &err);
+#else
   int err = DGifCloseFile(m_gif);
+#endif
   if (err == D_GIF_ERR_CLOSE_FAILED)
   {
     fprintf(stderr, "Gif::~Gif(): D_GIF_ERR_CLOSE_FAILED\n");
@@ -119,7 +124,7 @@
   if (DGifSlurp(m_gif) == GIF_ERROR)
   {
 #if GIFLIB_MAJOR >= 5
-    char* error = GifErrorString(m_gif->Error);
+    const char* error = GifErrorString(m_gif->Error);
     if (error)
       fprintf(stderr, "Gif::LoadGif(): Could not read file %s - %s\n", m_filename.c_str(), error);
 #else
@@ -190,7 +195,7 @@
   if (!m_gif)
   {
 #if GIFLIB_MAJOR >= 5
-    char* error = GifErrorString(err);
+    const char* error = GifErrorString(err);
     if (error)
       fprintf(stderr, "Gif::LoadGif(): Could not open file %s - %s\n", m_filename.c_str(), error);
 #else
@@ -248,7 +253,11 @@
     {
       if (DGifSlurp(gif) && gif->ImageCount > 1)
         m_isAnimated = 1;
+#if GIFLIB_MAJOR >= 5 && GIFLIB_MINOR >= 1
+      DGifCloseFile(gif, 0);
+#else
       DGifCloseFile(gif);
+#endif
       fclose(gifFile);
     }
   }
@@ -290,7 +299,7 @@
     GraphicsControlBlock gcb;
     if (!DGifSavedExtensionToGCB(m_gif, imgIdx, &gcb))
     {
-      char* error = GifErrorString(m_gif->Error);
+      const char* error = GifErrorString(m_gif->Error);
       if (error)
         fprintf(stderr, "Gif::ExtractFrames(): Could not read GraphicsControlBlock of frame %d - %s\n", imgIdx, error);
       else
@@ -495,7 +504,7 @@
   if (!m_gif)
   {
 #if GIFLIB_MAJOR >= 5
-    char* error = GifErrorString(err);
+    const char* error = GifErrorString(err);
     if (error)
       fprintf(stderr, "Gif::LoadImageFromMemory(): Could not open gif from memory - %s\n", error);
 #else
