--- a/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp	2015-10-19 08:31:15.000000000 +0200
+++ b/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp	2016-02-07 18:32:00.431735589 +0100
@@ -1160,14 +1160,18 @@
       if (!m_vizBuffers && m_audioCallback)
       {
         AEAudioFormat vizFormat = m_internalFormat;
-        vizFormat.m_channelLayout = AE_CH_LAYOUT_2_0;
-        vizFormat.m_dataFormat = AE_FMT_FLOAT;
+
+        // accumulate samples for the visualization addon
+        if (vizFormat.m_frames < 1536)
+          vizFormat.m_frames = 1536;
 
         // input buffers
-        m_vizBuffersInput = new CActiveAEBufferPool(m_internalFormat);
+        m_vizBuffersInput = new CActiveAEBufferPool(vizFormat);
         m_vizBuffersInput->Create(2000);
 
         // resample buffers
+        vizFormat.m_channelLayout = AE_CH_LAYOUT_2_0;
+        vizFormat.m_dataFormat = AE_FMT_FLOAT;
         m_vizBuffers = new CActiveAEBufferPoolResample(m_internalFormat, vizFormat, m_settings.resampleQuality);
         // TODO use cache of sync + water level
         m_vizBuffers->Create(2000, false, false);
@@ -1947,16 +1951,24 @@
 
             if (!m_vizBuffersInput->m_freeSamples.empty())
             {
-              // copy the samples into the viz input buffer
-              CSampleBuffer *viz = m_vizBuffersInput->GetFreeBuffer();
-              int samples = std::min(512, out->pkt->nb_samples);
+              // copy the samples into the viz input buffer	      
+              CSampleBuffer *viz = m_vizBuffersInput->m_freeSamples.front();
+              int samples = std::min(viz->pkt->max_nb_samples - viz->pkt->nb_samples, out->pkt->nb_samples);
               int bytes = samples * out->pkt->config.channels / out->pkt->planes * out->pkt->bytes_per_sample;
+              int offset = viz->pkt->nb_samples * out->pkt->config.channels / out->pkt->planes * out->pkt->bytes_per_sample;
               for(int i= 0; i < out->pkt->planes; i++)
               {
-                memcpy(viz->pkt->data[i], out->pkt->data[i], bytes);
+                memcpy(viz->pkt->data[i] + offset, out->pkt->data[i], bytes);
+              }
+              viz->pkt->nb_samples += samples;
+
+              if (viz->pkt->nb_samples >= viz->pkt->max_nb_samples)
+              {
+                m_vizBuffersInput->m_freeSamples.pop_front();
+                viz->refCount = 1;
+                viz->pkt->nb_samples = 512; // we do not need more...
+                m_vizBuffers->m_inputSamples.push_back(viz);
               }
-              viz->pkt->nb_samples = samples;
-              m_vizBuffers->m_inputSamples.push_back(viz);
             }
             else
               CLog::Log(LOGWARNING,"ActiveAE::%s - viz ran out of free buffers", __FUNCTION__);
