--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.h	2016-08-02 14:56:02.133634023 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.h	2016-08-02 14:56:02.153644023 +0200
@@ -41,7 +41,6 @@
 
   // Feature support
   virtual bool Supports(EINTERLACEMETHOD method);
-  virtual bool Supports(EDEINTERLACEMODE mode);
   virtual bool Supports(ESCALINGMETHOD method);
   
 
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp	2016-08-02 14:56:04.822978065 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererIMX.cpp	2016-08-02 14:56:04.834984067 +0200
@@ -93,16 +93,6 @@
     return false;
 }
 
-bool CRendererIMX::Supports(EDEINTERLACEMODE mode)
-{
-  if(mode == VS_DEINTERLACEMODE_AUTO
-  || mode == VS_DEINTERLACEMODE_FORCE
-  || mode == VS_DEINTERLACEMODE_OFF)
-    return true;
-
-  return false;
-}
-
 bool CRendererIMX::Supports(ESCALINGMETHOD method)
 {
   return false;
