--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOpenMax.h	2016-08-02 15:04:11.270087954 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOpenMax.h	2016-08-02 15:04:11.286095955 +0200
@@ -38,7 +38,6 @@
 
   // Feature support
   virtual bool Supports(EINTERLACEMETHOD method);
-  virtual bool Supports(EDEINTERLACEMODE mode);
   virtual bool Supports(ESCALINGMETHOD method);
   
 
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOpenMax.cpp	2016-08-02 15:04:09.701303928 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererOpenMax.cpp	2016-08-02 15:04:09.717311929 +0200
@@ -71,11 +71,6 @@
   return false;
 }
 
-bool CRendererOMX::Supports(EDEINTERLACEMODE mode)
-{
-  return false;
-}
-
 bool CRendererOMX::Supports(ESCALINGMETHOD mode)
 {
   return false;
