--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.h	2016-08-02 15:14:07.680153624 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.h	2016-08-02 15:14:07.696161625 +0200
@@ -41,7 +41,6 @@
 
   // Feature support
   virtual bool Supports(EINTERLACEMETHOD method);
-  virtual bool Supports(EDEINTERLACEMODE mode);
   virtual bool Supports(ESCALINGMETHOD method);
   virtual bool Supports(ERENDERFEATURE feature);
 
--- a/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp	2016-08-02 15:14:06.131379598 +0200
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp	2016-08-02 15:14:06.147387601 +0200
@@ -81,16 +81,6 @@
   return false;
 }
 
-bool CRendererAML::Supports(EDEINTERLACEMODE mode)
-{
-  if(mode == VS_DEINTERLACEMODE_OFF
-  || mode == VS_DEINTERLACEMODE_AUTO
-  || mode == VS_DEINTERLACEMODE_FORCE)
-    return true;
-
-  return false;
-}
-
 bool CRendererAML::Supports(ESCALINGMETHOD method)
 {
   return false;
