--- a/xbmc/video/windows/GUIWindowFullScreen.cpp	2016-10-01 14:30:58.000000000 +0200
+++ b/xbmc/video/windows/GUIWindowFullScreen.cpp	2016-10-01 20:02:57.037598834 +0200
@@ -485,14 +485,16 @@
 
 void CGUIWindowFullScreen::Process(unsigned int currentTime, CDirtyRegionList &dirtyregion)
 {
-  if (g_application.m_pPlayer->IsRenderingGuiLayer())
-    MarkDirtyRegion();
+  unsigned int oldDirty = dirtyregion.size();
 
   CGUIWindow::Process(currentTime, dirtyregion);
 
   //! @todo This isn't quite optimal - ideally we'd only be dirtying up the actual video render rect
   //!       which is probably the job of the renderer as it can more easily track resizing etc.
   m_renderRegion.SetRect(0, 0, (float)g_graphicsContext.GetWidth(), (float)g_graphicsContext.GetHeight());
+
+  if (dirtyregion.size() != oldDirty || g_application.m_pPlayer->IsRenderingGuiLayer())
+    dirtyregion.push_back(m_renderRegion);
 }
 
 void CGUIWindowFullScreen::Render()
