--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-08-15 20:49:17.498056015 +0200
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-08-15 20:49:17.518056016 +0200
@@ -29,6 +29,7 @@
 #include <sys/ioctl.h>
 #ifdef HAS_IMXVPU
 #include <linux/mxcfb.h>
+#include "cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h"
 #endif
 #include "utils/log.h"
 #include "utils/RegExp.h"
--- a/xbmc/linux/imx/IMX.cpp	2016-08-15 20:49:12.690055819 +0200
+++ b/xbmc/linux/imx/IMX.cpp	2016-08-15 20:49:12.714055819 +0200
@@ -180,7 +180,6 @@
     prev = d;
   }
 
-  unsigned int patternLength = 0;
   for (auto it = m_hgraph.begin(); it != m_hgraph.end();)
   {
     if (it->second > 1)
@@ -198,11 +197,6 @@
 
   double frameNorm = CDVDCodecUtils::NormalizeFrameduration(frameDuration, &hasMatch);
 
-  if (hasMatch && !patternLength)
-    m_patternLength = 1;
-  else
-    m_patternLength = patternLength;
-
   if (!m_hasPattern && hasMatch)
     m_frameDuration = frameNorm;
 
