--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2018-02-07 10:38:14.365820779 +0100
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2018-02-07 10:38:14.405800878 +0100
@@ -724,10 +724,15 @@
       if (bits != fmtBits)
       {
         /* if we opened in 32bit and only have 24bits, signal it accordingly */
-        if (fmtBits == 32 && bits == 24)
+        if (fmt == SND_PCM_FORMAT_S32 && bits == 24)
+        {
           i = AE_FMT_S24NE4MSB;
-        else
+        }
+        else if (fmt != SND_PCM_FORMAT_S24)
+        {
+          fmt = SND_PCM_FORMAT_UNKNOWN;
           continue;
+        }
       }
 
       /* record that the format fell back to X */
