--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-08-15 21:05:07.150094800 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-08-15 21:05:07.174094801 +0200
@@ -55,9 +55,9 @@
 // priorities to those subsystems can result in a very different user
 // experience. With that setting enabled we can build some statistics,
 // as numbers are always better than "feelings"
-#define IMX_PROFILE_BUFFERS
+//#define IMX_PROFILE_BUFFERS
 
-#define IMX_PROFILE
+//#define IMX_PROFILE
 //#define TRACE_FRAMES
 
 #define RENDER_USE_G2D 0
