--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-08-27 18:09:46.908531444 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-08-27 18:09:46.924531444 +0200
@@ -1246,7 +1246,7 @@
 
 CIMXContext::~CIMXContext()
 {
-  Stop(false);
+  Stop();
   Dispose();
   CloseDevices();
   g2dCloseDevices();
@@ -1569,7 +1569,7 @@
   int page = m_fbCurrentPage.load();
 
   CSingleLock lk(m_pageSwapLock);
-  if (!m_fbHandle || !m_bFbIsConfigured) return false;
+  if (m_bStop || !m_fbHandle || !m_bFbIsConfigured) return false;
   // Protect page swapping from screen capturing that does read the current
   // front buffer. This is actually not done very frequently so the lock
   // does not hurt.
@@ -1948,7 +1948,10 @@
   CThread::StopThread(false);
   Blank();
   if (bWait && IsRunning())
+  {
+    m_waitFlip.Set();
     CThread::StopThread(true);
+  }
 }
 
 void CIMXContext::Process()
