--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2017-02-25 12:39:39.771159746 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2017-02-25 12:38:09.000000000 +0100
@@ -758,10 +758,19 @@
     inData.pVirAddr = demuxer_content;
     if ((m_decOpenParam.CodecFormat == VPU_V_MPEG2) ||
         (m_decOpenParam.CodecFormat == VPU_V_VC1_AP)||
-        (m_decOpenParam.CodecFormat == VPU_V_XVID))
+        (m_decOpenParam.CodecFormat == VPU_V_XVID) ||
+        (m_decOpenParam.CodecFormat == VPU_V_AVC))
     {
-      inData.sCodecData.pData = (unsigned char *)m_hints.extradata;
-      inData.sCodecData.nSize = m_hints.extrasize;
+      if (m_convert_bitstream)
+      {
+        inData.sCodecData.pData = m_converter->GetExtraData();
+        inData.sCodecData.nSize = m_converter->GetExtraSize();
+      }
+      else
+      {
+        inData.sCodecData.pData = (unsigned char *)m_hints.extradata;
+        inData.sCodecData.nSize = m_hints.extrasize;
+      }
     }
     else
     {
