diff -Naur a/xbmc/linux/imx/IMX.h b/xbmc/linux/imx/IMX.h
--- a/xbmc/linux/imx/IMX.h	2016-09-04 13:45:51.000000000 +0200
+++ b/xbmc/linux/imx/IMX.h	2016-09-04 13:44:59.000000000 +0200
@@ -177,7 +177,7 @@
 public:
   uint32_t     iWidth;
   uint32_t     iHeight;
-  int          pPhysAddr;
+  uint32_t     pPhysAddr;
   uint8_t     *pVirtAddr;
   int          iFormat;
   double       m_fps;
diff -Naur a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-08-28 19:02:32.000000000 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-09-04 13:58:24.000000000 +0200
@@ -1156,7 +1156,7 @@
   iWidth      = frameInfo->pExtInfo->nFrmWidth;
   iHeight     = frameInfo->pExtInfo->nFrmHeight;
   pVirtAddr   = m_frameBuffer->pbufVirtY;
-  pPhysAddr   = (int)m_frameBuffer->pbufY;
+  pPhysAddr   = (uint32_t)m_frameBuffer->pbufY;
 
 #ifdef IMX_INPUT_FORMAT_I420
   iFormat     = _4CC('I', '4', '2', '0');
diff -Naur a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-08-28 19:02:32.000000000 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2016-09-04 13:58:24.000000000 +0200
@@ -197,7 +197,7 @@
   int                            m_fbLineLength;
   int                            m_fbPageSize;
   int                            m_fbPhysSize;
-  int                            m_fbPhysAddr;
+  uint32_t                       m_fbPhysAddr;
   uint8_t                       *m_fbVirtAddr;
   struct fb_var_screeninfo       m_fbVar;
   int                            m_ipuHandle;
