diff -Naur a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-08-28 19:02:32.000000000 +0200
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-09-04 13:58:24.000000000 +0200
@@ -1209,6 +1209,7 @@
   : CThread("iMX IPU")
   , m_fbHandle(0)
   , m_fbCurrentPage(0)
+  , m_pg(0)
   , m_fbPhysAddr(0)
   , m_fbVirtAddr(NULL)
   , m_ipuHandle(0)
@@ -1520,10 +1521,10 @@
 {
   if (page == RENDER_TASK_AUTOPAGE)
     page = m_pg;
-  else if (page < 0 && page >= m_fbPages)
+  else if (page < 0 || page >= m_fbPages)
     return;
 
-  m_pg = ++m_pg % m_fbPages;
+  m_pg = (page + 1) % m_fbPages;
 
   IPUTaskPtr ipu(new IPUTask);
   ipu->page = page;
