--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-09-21 19:06:42.214823995 +0200
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2016-09-21 19:06:42.238823996 +0200
@@ -97,7 +97,7 @@
     }
   }
 
-  ShowWindow(false);
+  //ShowWindow(false);
 
   fd = open("/dev/fb0",O_RDWR);
   if (fd < 0)
@@ -131,7 +131,7 @@
 
   m_sar = GetMonitorSAR();
   g_IMXContext.create();
-  ShowWindow(true);
+  //ShowWindow(true);
 #endif
   return;
 }
@@ -257,7 +257,7 @@
   DestroyNativeWindow();
   DestroyNativeDisplay();
 
-  ShowWindow(false);
+  //ShowWindow(false);
   SysfsUtils::SetString("/sys/class/graphics/fb0/mode", res.strId + "\n");
 
   CreateNativeDisplay();
