diff -Naur acl-2.2.49/Makefile acl-2.2.49a/Makefile
--- acl-2.2.49/Makefile	2009-06-22 20:37:31.000000000 +0200
+++ acl-2.2.49a/Makefile	2010-10-17 18:18:46.944818733 +0200
@@ -88,7 +88,7 @@
 		$(MAKE) $(AM_MAKEFLAGS) include/builddefs; \
 	fi
 
-install: default $(addsuffix -install,$(SUBDIRS))
+install: default $(addsuffix -install,$(SUBDIRS)) $(addsuffix -install-dev,$(SUBDIRS)) $(addsuffix -install-lib,$(SUBDIRS))
 	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
 	$(INSTALL) -m 644 README $(PKG_DOC_DIR)
 
