diff -Naur libSockets-2.3.9.5/Makefile libSockets-2.3.9.5_patched//Makefile
--- libSockets-2.3.9.5/Makefile	2010-02-21 09:57:49.000000000 +0100
+++ libSockets-2.3.9.5_patched//Makefile	2011-06-14 18:22:33.820431664 +0200
@@ -35,7 +35,7 @@
 #PREFIX =	/usr
 
 # include paths
-INCLUDE =	-I/usr/include/libxml2 
+INCLUDE =	`xml2-config  --cflags`
 
 # CXX, CFLAGS, LIBS, LDFLAGS, LDFLAGSSO
 include		Makefile.Defines.$(PLATFORM)
diff -Naur libSockets-2.3.9.5/Makefile.Defines.linux-x86-32 libSockets-2.3.9.5_patched//Makefile.Defines.linux-x86-32
--- libSockets-2.3.9.5/Makefile.Defines.linux-x86-32	2010-02-06 20:02:11.000000000 +0100
+++ libSockets-2.3.9.5_patched//Makefile.Defines.linux-x86-32	2011-06-14 18:22:54.573543911 +0200
@@ -1,4 +1,4 @@
-CXX =		g++
+CXX ?=		g++
 
 CFLAGS +=	-Wall -g $(INCLUDE) -MD -D_VERSION='"$(VERSION)"'
 
