diff -Naur libSockets-2.3.9.5/pkgconfig/libsockets2.pc.IN libSockets-2.3.9.5_patched//pkgconfig/libsockets2.pc.IN
--- libSockets-2.3.9.5/pkgconfig/libsockets2.pc.IN	2010-02-06 20:01:39.000000000 +0100
+++ libSockets-2.3.9.5_patched//pkgconfig/libsockets2.pc.IN	2011-06-14 20:48:46.687218605 +0200
@@ -1,7 +1,7 @@
 prefix=/usr
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+includedir=${prefix}/include/Sockets
 
 Name: libsockets2
 Description: C++ Sockets Library
diff -Naur libSockets-2.3.9.5/pkgconfig/libSockets.pc.IN libSockets-2.3.9.5_patched//pkgconfig/libSockets.pc.IN
--- libSockets-2.3.9.5/pkgconfig/libSockets.pc.IN	2010-02-06 20:01:39.000000000 +0100
+++ libSockets-2.3.9.5_patched//pkgconfig/libSockets.pc.IN	2011-06-14 20:48:32.381213276 +0200
@@ -1,7 +1,7 @@
 prefix=/usr
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
-includedir=${prefix}/include
+includedir=${prefix}/include/Sockets
 
 Name: lib%NAME%
 Description: C++ Sockets Library
