diff -Naur upstart-0.6.3/udev/Makefile.am upstart-0.6.3a/udev/Makefile.am
--- upstart-0.6.3/udev/Makefile.am	2010-01-24 18:43:53.000000000 +0100
+++ upstart-0.6.3a/udev/Makefile.am	2010-01-24 18:44:17.000000000 +0100
@@ -42,7 +42,7 @@
 	../dbus/com.ubuntu.Upstart.xml
 
 $(com_ubuntu_Upstart_OUTPUTS): $(com_ubuntu_Upstart_XML) $(top_builddir)/nih-dbus-tool/nih-dbus-tool
-	$(top_builddir)/nih-dbus-tool/nih-dbus-tool \
+	$(NIH_DBUS_TOOL) \
 		--mode=proxy --prefix=upstart --default-interface=com.ubuntu.Upstart0_6 \
 		--output=$@ $<
 
diff -Naur upstart-0.6.3/udev/Makefile.in upstart-0.6.3a/udev/Makefile.in
--- upstart-0.6.3/udev/Makefile.in	2010-01-24 18:43:53.000000000 +0100
+++ upstart-0.6.3a/udev/Makefile.in	2010-01-24 18:44:34.000000000 +0100
@@ -731,7 +731,7 @@
 
 
 $(com_ubuntu_Upstart_OUTPUTS): $(com_ubuntu_Upstart_XML) $(top_builddir)/nih-dbus-tool/nih-dbus-tool
-	$(top_builddir)/nih-dbus-tool/nih-dbus-tool \
+	$(NIH_DBUS_TOOL) \
 		--mode=proxy --prefix=upstart --default-interface=com.ubuntu.Upstart0_6 \
 		--output=$@ $<
 
