--- upstart-0.6.3/nih/logging.c.orig	2010-01-28 06:15:51.000000000 -0500
+++ upstart-0.6.3/nih/logging.c	2010-01-28 06:18:12.000000000 -0500
@@ -39,13 +39,6 @@
 
 
 /**
- * __abort_msg:
- *
- * A glibc variable that keeps the assertion message in the core dump.
- **/
-extern char *__abort_msg;
-
-/**
  * logger:
  *
  * Function used to output log messages.
@@ -144,12 +137,6 @@
 	message = NIH_MUST (nih_vsprintf (NULL, format, args));
 	va_end (args);
 
-	if (priority >= NIH_LOG_FATAL) {
-		if (__abort_msg)
-			nih_discard (__abort_msg);
-		__abort_msg = NIH_MUST (nih_strdup (NULL, message));
-	}
-
 	/* Output the message */
 	ret = logger (priority, message);
 
