--- upstart-0.6.3.orig/init/job_process.c	2010-01-25 18:48:49.000000000 +0100
+++ upstart-0.6.3/init/job_process.c	2010-01-25 18:49:15.000000000 +0100
@@ -163,6 +163,7 @@
 
 		NIH_MUST (nih_str_array_add (&argv, NULL, &argc, SHELL));
 		NIH_MUST (nih_str_array_add (&argv, NULL, &argc, "-e"));
+		NIH_MUST (nih_str_array_add (&argv, NULL, &argc, "-l"));
 
 		/* If the process wasn't originally marked to be run through
 		 * a shell, prepend exec to the script so that the shell
