Description: add support for glibc 2.15
Origin: upstream, http://sourceforge.net/mailarchive/forum.php?thread_name=20111229082456.D5FE6108870%40jail0086.vps.exonetric.net&forum_name=valgrind-developers

Modified: trunk/configure.in
===================================================================
--- trunk/configure.in	2011-12-27 18:43:32 UTC (rev 12322)
+++ trunk/configure.in	2011-12-29 08:24:55 UTC (rev 12323)
@@ -778,6 +778,13 @@
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.15)
+	AC_MSG_RESULT(2.15 family)
+	AC_DEFINE([GLIBC_2_15], 1, [Define to 1 if you're using glibc 2.15.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
 	AC_DEFINE([DARWIN_LIBC], 1, [Define to 1 if you're using Darwin])
@@ -791,7 +798,7 @@
 
      *)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
-	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.14])
+	AC_MSG_ERROR([Valgrind requires glibc version 2.2 - 2.15])
 	AC_MSG_ERROR([or Darwin libc])
 	;;
 esac
