--- valgrind-3.7.0/configure.in	2011-11-18 16:10:48.045357540 +0100
+++ valgrind-3.7.0/configure.in_	2011-11-18 16:10:17.175357536 +0100
@@ -110,6 +110,7 @@
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
+             | $SED 's/.*openbricks.*-gcc//' \
              | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`]
 
 is_clang="notclang"
