diff -Naur libSockets-2.3.9.5/TcpSocket.cpp libSockets-2.3.9.5_patched//TcpSocket.cpp
--- libSockets-2.3.9.5/TcpSocket.cpp	2011-06-14 18:35:47.290006490 +0200
+++ libSockets-2.3.9.5_patched//TcpSocket.cpp	2010-07-21 09:01:59.000000000 +0200
@@ -1297,7 +1297,7 @@
 	/* Create our context*/
 	if (m_server_contexts.find(context) == m_server_contexts.end())
 	{
-		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : const_cast<SSL_METHOD *>(SSLv3_method());
 		m_ssl_ctx = m_server_contexts[context] = SSL_CTX_new(meth);
 		SSL_CTX_set_mode(m_ssl_ctx, SSL_MODE_AUTO_RETRY|SSL_MODE_ENABLE_PARTIAL_WRITE);
 		// session id
@@ -1333,7 +1333,7 @@
 	/* Create our context*/
 	if (m_server_contexts.find(context) == m_server_contexts.end())
 	{
-		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : SSLv3_method();
+		SSL_METHOD *meth = meth_in ? const_cast<SSL_METHOD *>(meth_in) : const_cast<SSL_METHOD *>(SSLv3_method());
 		m_ssl_ctx = m_server_contexts[context] = SSL_CTX_new(meth);
 		SSL_CTX_set_mode(m_ssl_ctx, SSL_MODE_AUTO_RETRY|SSL_MODE_ENABLE_PARTIAL_WRITE);
 		// session id
