diff -Naur libSockets-2.3.9.5/Makefile libSockets-2.3.9.5_patched/Makefile
--- libSockets-2.3.9.5/Makefile	2011-06-14 19:39:43.204013905 +0200
+++ libSockets-2.3.9.5_patched/Makefile	2011-06-14 19:48:31.770462362 +0200
@@ -110,13 +110,12 @@

 install_shared:	install shared
 		@mkdir -p $(DESTDIR)/$(PREFIX)/lib/pkgconfig
-		cp $(SHAREDLIBNAME) $(DESTDIR)/$(PREFIX)/lib
+		install -m 0644 $(SHAREDLIBNAME) $(DESTDIR)/$(PREFIX)/lib
-		cp -a pkgconfig/*pc $(DESTDIR)/$(PREFIX)/lib/pkgconfig
+		install -m 0644 pkgconfig/*pc $(DESTDIR)/$(PREFIX)/lib/pkgconfig
 		rm -f $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so
 		rm -f $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so.$(MAJOR)
-		ln -s $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so.$(MAJOR).$(MINOR) $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so
+		ln -s lib$(NAME).so.$(MAJOR).$(MINOR) $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so
-		ln -s $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so.$(MAJOR).$(MINOR) $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so.$(MAJOR)
+		ln -s lib$(NAME).so.$(MAJOR).$(MINOR) $(DESTDIR)/$(PREFIX)/lib/lib$(NAME).so.$(MAJOR)
-		ldconfig
 
 # no binary files, zip will translate lf to cr lf
 FILES =		*.h *.cpp \
