diff -ruN 1/config.c 2/config.c
--- 1/config.c	2013-10-25 14:13:11.000000000 +0200
+++ 2/config.c	2013-10-25 14:15:25.000000000 +0200
@@ -402,6 +402,9 @@
   StandardCompliance = STANDARD_DVB;
   MarginStart = 2;
   MarginStop = 10;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
+  JumpSecondsRepeat = 300;
   AudioLanguages[0] = -1;
   DisplaySubtitles = 0;
   SupportTeletext = 0;
@@ -618,6 +621,9 @@
   else if (!strcasecmp(Name, "StandardCompliance"))  StandardCompliance = atoi(Value);
   else if (!strcasecmp(Name, "MarginStart"))         MarginStart        = atoi(Value);
   else if (!strcasecmp(Name, "MarginStop"))          MarginStop         = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value); 
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsRepeat"))   JumpSecondsRepeat  = atoi(Value);
   else if (!strcasecmp(Name, "AudioLanguages"))      return ParseLanguages(Value, AudioLanguages);
   else if (!strcasecmp(Name, "DisplaySubtitles"))    DisplaySubtitles   = atoi(Value);
   else if (!strcasecmp(Name, "SupportTeletext"))     SupportTeletext    = atoi(Value);
@@ -738,6 +744,9 @@
   Store("StandardCompliance", StandardCompliance);
   Store("MarginStart",        MarginStart);
   Store("MarginStop",         MarginStop);
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
+  Store("JumpSecondsRepeat",  JumpSecondsRepeat);
   StoreLanguages("AudioLanguages", AudioLanguages);
   Store("DisplaySubtitles",   DisplaySubtitles);
   Store("SupportTeletext",    SupportTeletext);
diff -ruN 1/config.h 2/config.h
--- 1/config.h	2013-10-25 14:13:11.000000000 +0200
+++ 2/config.h	2013-10-25 14:15:25.000000000 +0200
@@ -286,6 +286,7 @@
   int TimeTransponder;
   int StandardCompliance;
   int MarginStart, MarginStop;
+  int JumpSeconds, JumpSecondsSlow, JumpSecondsRepeat;
   int AudioLanguages[I18N_MAX_LANGUAGES + 1];
   int DisplaySubtitles;
   int SupportTeletext;
diff -ruN 1/menu.c 2/menu.c
--- 1/menu.c	2013-10-25 14:13:11.000000000 +0200
+++ 2/menu.c	2013-10-25 14:15:25.000000000 +0200
@@ -3607,6 +3607,9 @@
   Add(new cMenuEditIntItem(tr("Setup.Replay$Resume ID"), &data.ResumeID, 0, 99));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Jump&Play"), &data.JumpPlay));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Play&Jump"), &data.PlayJump));
+  Add(new cMenuEditIntItem( tr("Setup.Recording$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem( tr("Setup.Recording$Jump Seconds Slow"), &data.JumpSecondsSlow));
+  Add(new cMenuEditIntItem( tr("Setup.Recording$Jump Seconds (Repeat)"), &data.JumpSecondsRepeat));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause at last mark"), &data.PauseLastMark));
 }
 
@@ -5658,9 +5661,11 @@
     case kRight:   Forward(); break;
     case kRed:     TimeSearch(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+                   SkipSeconds(-(Setup.JumpSecondsRepeat)); break;
+    case kGreen:   SkipSeconds(-(Setup.JumpSeconds)); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+                   SkipSeconds(Setup.JumpSecondsRepeat); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
     case kStop:
     case kBlue:    Hide();
                    Stop();
diff -ruN 1/po/de_DE.po 2/po/de_DE.po
--- 1/po/de_DE.po	2013-10-25 14:13:11.000000000 +0200
+++ 2/po/de_DE.po	2013-10-25 14:15:25.000000000 +0200
@@ -934,6 +934,15 @@
 msgid "Button$Scan"
 msgstr "Scan"
 
+msgid "Setup.Recording$Jump Seconds"
+msgstr "Sprungweite in Sekunden"
+
+msgid "Setup.Recording$Jump Seconds Slow"
+msgstr "Sprungweite in Sekunden langsam"
+
+msgid "Setup.Recording$Jump Seconds (Repeated)"
+msgstr "Sprungweite in Sek. (wiederh.)"
+
 msgid "Setup.EPG$EPG scan timeout (h)"
 msgstr "Zeit bis zur EPG-Aktualisierung (h)"
 
