diff -ruN vdr-1/eitscan.c vdr/eitscan.c
--- vdr-1/eitscan.c	2012-04-07 16:39:28.000000000 +0200
+++ vdr/eitscan.c	2014-01-08 21:52:33.000000000 +0100
@@ -145,7 +145,7 @@
                if (Device && Device->ProvidesEIT()) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (Device->Priority() < 0) {
diff -ruN vdr-1/pat.c vdr/pat.c
--- vdr-1/pat.c	2014-01-08 21:51:00.000000000 +0100
+++ vdr/pat.c	2014-01-08 21:52:33.000000000 +0100
@@ -633,6 +633,7 @@
                  TeletextSubtitlePages[NumTPages++] = tTeletextSubtitlePage(manualPageNumber);
               }
            Channel->SetTeletextSubtitlePages(TeletextSubtitlePages, NumTPages);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
