--- vdr-2.1.6/config.h.orig	2014-08-06 19:01:23.822856982 +0200
+++ vdr-2.1.6/config.h	2014-08-06 19:03:35.072735650 +0200
@@ -343,6 +343,7 @@
   int MultiSpeedMode;
   int LeftRightSkip;
   int GreenYellowSkip;
+  int OneThreeSkip;
   int ShowReplayMode;
   int ShowRemainingTime;
   int ProgressDisplayTime;
--- vdr-2.1.6/menu.c.orig	2014-08-06 18:42:23.292623237 +0200
+++ vdr-2.1.6/menu.c	2014-08-06 18:50:19.803011136 +0200
@@ -3683,6 +3683,7 @@
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Multi speed mode"), &data.MultiSpeedMode));
   Add(new cMenuEditIntItem( tr("Setup.Replay$Left Right Skip (s)"), &data.LeftRightSkip, 0, 60 * 60));
   Add(new cMenuEditIntItem( tr("Setup.Replay$Green Yellow Skip (s)"), &data.GreenYellowSkip, 0, 60 * 60));
+  Add(new cMenuEditIntItem( tr("Setup.Replay$Key 1 and 3 Skip (s)"), &data.OneThreeSkip, 0, 60 * 60));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Show replay mode"), &data.ShowReplayMode));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Show remaining time"), &data.ShowRemainingTime));
   Add(new cMenuEditIntItem( tr("Setup.Replay$Progress display time (s)"), &data.ProgressDisplayTime, 0, 60));
@@ -5714,6 +5715,11 @@
     case kLeft:    SkipSeconds(-Setup.LeftRightSkip); break;
     case kRight|k_Repeat:
     case kRight:   SkipSeconds( Setup.LeftRightSkip); break;
+    case k1|k_Repeat:
+    case k1:       SkipSeconds(-Setup.OneThreeSkip); break;
+    case k3|k_Repeat:
+    case k3:       SkipSeconds( Setup.OneThreeSkip); break;
+
     case kStop:
     case kBlue:    Hide();
                    Stop();
--- vdr-2.1.6/config.c.orig	2014-08-05 22:29:34.311930826 +0200
+++ vdr-2.1.6/config.c	2014-08-06 18:58:00.672454070 +0200
@@ -474,6 +474,7 @@
   MultiSpeedMode = 0;
   LeftRightSkip = 0;
   GreenYellowSkip = 60;
+  OneThreeSkip = 10;
   ShowReplayMode = 0;
   ShowRemainingTime = 0;
   ProgressDisplayTime = 0;
@@ -698,6 +699,7 @@
   else if (!strcasecmp(Name, "MultiSpeedMode"))      MultiSpeedMode     = atoi(Value);
   else if (!strcasecmp(Name, "LeftRightSkip"))       LeftRightSkip      = atoi(Value);
   else if (!strcasecmp(Name, "GreenYellowSkip"))     GreenYellowSkip    = atoi(Value);
+  else if (!strcasecmp(Name, "OneThreeSkip"))        OneThreeSkip       = atoi(Value);
   else if (!strcasecmp(Name, "ShowReplayMode"))      ShowReplayMode     = atoi(Value);
   else if (!strcasecmp(Name, "ShowRemainingTime"))   ShowRemainingTime  = atoi(Value);
   else if (!strcasecmp(Name, "ProgressDisplayTime")) ProgressDisplayTime= atoi(Value);
@@ -826,6 +828,7 @@
   Store("MultiSpeedMode",     MultiSpeedMode);
   Store("LeftRightSkip",      LeftRightSkip);
   Store("GreenYellowSkip",    GreenYellowSkip);
+  Store("OneThreeSkip",       OneThreeSkip);
   Store("ShowReplayMode",     ShowReplayMode);
   Store("ShowRemainingTime",  ShowRemainingTime);
   Store("ProgressDisplayTime",ProgressDisplayTime);
--- vdr/po/de_DE.po.orig	2014-08-06 21:22:14.549830140 +0200
+++ vdr/po/de_DE.po	2014-08-06 21:29:48.947459155 +0200
@@ -1227,6 +1227,9 @@
 
 msgid "Setup.Replay$Multi speed mode"
 msgstr "Mehrstufiger Vor-/Rücklauf"
+
+msgid "Setup.Replay$Key 1 and 3 Skip (s)"
+msgstr "Sprungintervall Taste 1 und 3 (s)"
 
 msgid "Setup.Replay$Show replay mode"
 msgstr "Wiedergabestatus anzeigen"
