diff -u --recursive --new-file orig/channels.c mod/channels.c
--- orig/channels.c	2014-07-31 09:41:05.893809856 +0200
+++ mod/channels.c	2014-07-31 09:40:46.545430322 +0200
@@ -821,7 +821,7 @@
 bool cChannels::Load(const char *FileName, bool AllowComments, bool MustExist)
 {
   if (cConfig<cChannel>::Load(FileName, AllowComments, MustExist)) {
-     DeleteDuplicateChannels();
+     // DeleteDuplicateChannels();
      ReNumber();
      return true;
      }
