diff -Naur vdr-loadepg-0.2.3/loadepg.h vdr-loadepg-0.2.3a//loadepg.h
--- vdr-loadepg-0.2.3/loadepg.h	2009-07-30 19:48:25.000000000 +0200
+++ vdr-loadepg-0.2.3a//loadepg.h	2011-01-04 17:51:49.343953604 +0100
@@ -6,7 +6,7 @@
 #include <sys/ioctl.h>
 #include <linux/dvb/frontend.h>
 #include <linux/dvb/dmx.h>
-#include "../../../libsi/section.h"
+#include <libsi/section.h>
 
 #include "formats.h"
 #include "i18n.h"
