--- loadepg.orig/loadepg.c	2008-05-01 17:15:02.000000000 +0200
+++ loadepg/loadepg.c	2008-05-04 11:16:47.000000000 +0200
@@ -1423,15 +1423,19 @@
 		  {
 		    int i = p + 5;
 		    int loop = 0;
+		    int ServiceProviderNameLength;
+		    int ServiceNameLength;
 		    while( loop < DescriptorsLoopLength )
 		    {
 		      DescriptorTag = Data[i];
 		      DescriptorLength = Data[i+1];
+
 		      switch( DescriptorTag )
 		      {
 		        case 0x48:
-			  int ServiceProviderNameLength = Data[i+3];
-		          int ServiceNameLength = Data[i+4+ServiceProviderNameLength] - 1;
+			{
+			  ServiceProviderNameLength = Data[i+3];
+		          ServiceNameLength = Data[i+4+ServiceProviderNameLength] - 1;
 			  if( ( pC + ServiceNameLength + 2 ) > MAX_BUFFER_SIZE_CHANNELS )
 			  {
 			    return;
@@ -1440,8 +1444,10 @@
 			  C->pData = pC;
 			  C->lenData = ServiceNameLength;
 			  pC += ( ServiceNameLength + 1 );
+			}
 		          break;
 		        case 0xc0:
+			{
 		          ServiceNameLength = DescriptorLength - 1;
 			  if( ( pC + ServiceNameLength + 2 ) > MAX_BUFFER_SIZE_CHANNELS )
 			  {
@@ -1451,6 +1457,7 @@
 			  C->pData = pC;
 			  C->lenData = ServiceNameLength;
 			  pC += ( ServiceNameLength + 1 );
+			}
 		          break;
 		        default:
 		          break;
