diff -Naur u-boot-2009-08+1~42e3a9c-origin/tools/env/fw_env.c u-boot-2009-08+1~42e3a9c/tools/env/fw_env.c
--- u-boot-2009-08+1~42e3a9c-origin/tools/env/fw_env.c	2017-10-22 18:46:28.581966704 +0200
+++ u-boot-2009-08+1~42e3a9c/tools/env/fw_env.c	2017-10-22 18:46:37.837864887 +0200
@@ -25,6 +25,7 @@
  */
 
 #include <errno.h>
+#include <stdint.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
