diff -Naur vdr-xineliboutput-cvs20101220/configure vdr-xineliboutput-cvs20101220a/configure
--- vdr-xineliboutput-cvs20101220/configure	2010-12-30 17:34:56.000000000 +0100
+++ vdr-xineliboutput-cvs20101220a/configure	2010-12-31 10:25:26.016131280 +0100
@@ -370,7 +370,7 @@
     test_library X11  xrandr       "X11/extensions/Xrandr.h"   "-lXrandr"    "XRRGetScreenInfo(0,0)"
     test_library X11  xdpms        "X11/extensions/dpms.h"     "-lXext"      "DPMSDisable(0)"
     test_library X11  xinerama     "X11/extensions/Xinerama.h" "-lXinerama"  "XineramaQueryScreens(0,0)"
-    test_library X11  opengl       "GL/glx.h"                  "-lGL -lGLU -lglut"  "glXQueryVersion(0,0,0)"
+    test_library X11  opengl       "GL/glx.h"                  "-lGL -lGLU -lglut -lstdc++"  "glXQueryVersion(0,0,0)"
     test_library none vdpau        "vdpau/vdpau_x11.h"         "-lvdpau"     "vdp_device_create_x11(0,0,0,0)"
     test_library X11  dbus-glib-1  \
       "dbus/dbus-glib.h" \
