diff -Naur vdr-xineliboutput-cvs20100818/Makefile vdr-xineliboutput-cvs20100818a/Makefile
--- vdr-xineliboutput-cvs20100818/Makefile	2010-07-10 00:14:32.000000000 +0200
+++ vdr-xineliboutput-cvs20100818a/Makefile	2010-08-18 16:29:36.317779094 +0200
@@ -239,7 +239,7 @@
 ### Dependencies:
 ###
 
-MAKEDEP = g++ -MM -MG
+MAKEDEP = g++ $(OPTFLAGS) -MM -MG
 DEPFILE = .dependencies
 $(DEPFILE): Makefile config.mak
 	@rm -f $@
@@ -335,11 +335,11 @@
 #
 
 $(VDRPLUGIN_SXFE): $(OBJS_SXFE_SO)
-	$(CC) $(CFLAGS) $(LDFLAGS_SO) $(OBJS_SXFE_SO) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) -o $@
+	$(CXX) $(CFLAGS) $(LDFLAGS_SO) $(OBJS_SXFE_SO) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) -o $@
 	@-rm -rf $(LIBDIR)/$(VDRPLUGIN_SXFE).$(VERSION)
 	@cp $@ $(LIBDIR)/$(VDRPLUGIN_SXFE).$(VERSION)
 $(VDRSXFE): $(OBJS_SXFE)
-	$(CC) -g $(OBJS_SXFE) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_PTHREAD) -o $@
+	$(CXX) -g $(OBJS_SXFE) $(LIBS_X11) $(LIBS_XINE) $(LIBS_JPEG) $(LIBS_PTHREAD) -o $@
 
 #
 # vdr-fbfe
