diff -urN vdr-1.7.38.org/device.h vdr-1.7.38/device.h
--- vdr-1.7.38.org/device.h	2013-02-18 15:37:21.183390350 +0100
+++ vdr-1.7.38/device.h	2013-02-18 15:37:32.723390004 +0100
@@ -301,12 +301,12 @@
          ///< Direction (only the sign of Direction is evaluated, positive values
          ///< switch to higher channel numbers).
 private:
-  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
-         ///< Sets the device to the given channel (general setup).
 protected:
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
          ///< Sets the device to the given channel (actual physical setup).
 public:
+  eSetChannelResult SetChannel(const cChannel *Channel, bool LiveView);
+         ///< Sets the device to the given channel (general setup).
   static int CurrentChannel(void) { return primaryDevice ? currentChannel : 0; }
          ///< Returns the number of the current channel on the primary device.
   static void SetCurrentChannel(const cChannel *Channel) { currentChannel = Channel ? Channel->Number() : 0; }
diff -urN vdr-1.7.38.org/osd.c vdr-1.7.38/osd.c
--- vdr-1.7.38.org/osd.c	2013-02-18 15:37:21.187390350 +0100
+++ vdr-1.7.38/osd.c	2013-02-18 15:37:32.727390005 +0100
@@ -1644,6 +1644,7 @@
          }
       }
   Osds.Append(this);
+  vidWin.bpp = 0;
 }
 
 cOsd::~cOsd()
diff -urN vdr-1.7.38.org/osd.h vdr-1.7.38/osd.h
--- vdr-1.7.38.org/osd.h	2013-02-18 15:37:21.183390350 +0100
+++ vdr-1.7.38/osd.h	2013-02-18 15:38:22.419388524 +0100
@@ -927,6 +927,7 @@
        ///<        delete pm;
        ///<        }
   static bool pinValid;   // PIN PATCH
+  tArea vidWin;
   };
 
 #define MAXOSDIMAGES 64
diff -urN vdr-1.7.38.org/PLUGINS/src/dvbsddevice/dvbsdffosd.c vdr-1.7.38/PLUGINS/src/dvbsddevice/dvbsdffosd.c
--- vdr-1.7.38.org/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2013-02-18 15:37:21.171390351 +0100
+++ vdr-1.7.38/PLUGINS/src/dvbsddevice/dvbsdffosd.c	2013-02-18 15:37:32.723390004 +0100
@@ -77,6 +77,10 @@
             Cmd(OSD_SetWindow, 0, i + 1);
             Cmd(OSD_Close);
             }
+        if (vidWin.bpp != 0) {
+           Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+           Cmd(OSD_Close);
+           }
         shown = false;
         }
      }
@@ -111,6 +115,10 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_Close);
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_Close);
+        }
      shown = false;
      }
   return cOsd::SetAreas(Areas, NumAreas);
@@ -194,6 +202,11 @@
          Cmd(OSD_SetWindow, 0, i + 1);
          Cmd(OSD_MoveWindow, 0, Left() + Bitmap->X0(), Top() + Bitmap->Y0());
          }
+     if (vidWin.bpp != 0) {
+        Cmd(OSD_SetWindow, 0, MAXNUMWINDOWS);
+        Cmd(OSD_OpenRaw, vidWin.bpp, vidWin.x1, vidWin.y1,
+            vidWin.x2, vidWin.y2, (void *)0);
+        }
      shown = true;
      }
 }
