--- ./config.h	2013/03/10 15:52:11	2.72
+++ ./config.h	2013/03/11 10:32:02
@@ -260,7 +260,7 @@
   int MenuScrollWrap;
   int MenuKeyCloses;
   int MarkInstantRecord;
-  char NameInstantRecord[NAME_MAX];
+  char NameInstantRecord[NAME_MAX + 1];
   int InstantRecordTime;
   int LnbSLOF;
   int LnbFrequLo;
--- ./recording.c	2013/03/04 14:02:40	2.90
+++ ./recording.c	2013/03/11 10:34:41
@@ -66,7 +66,7 @@
 
 #define MAX_LINK_LEVEL  6
 
-int DirectoryPathMax = PATH_MAX;
+int DirectoryPathMax = PATH_MAX - 1;
 int DirectoryNameMax = NAME_MAX;
 bool DirectoryEncoding = false;
 int InstanceId = 0;
--- ./timers.h	2013/02/05 11:23:24	2.6
+++ ./timers.h	2013/03/11 10:35:53
@@ -39,7 +39,7 @@
   int stop;
   int priority;
   int lifetime;
-  mutable char file[NAME_MAX * 2]; // *2 to be able to hold 'title' and 'episode', which can each be up to 255 characters long
+  mutable char file[NAME_MAX * 2 + 1]; // *2 to be able to hold 'title' and 'episode', which can each be up to 255 characters long
   char *aux;
   const cEvent *event;
 public:
--- ./vdr.c	2013/03/03 10:25:11	2.50
+++ ./vdr.c	2013/03/11 10:31:24
@@ -281,7 +281,7 @@
           case 'd' | 0x100: {
                     char *s = optarg;
                     int n = strtol(s, &s, 10);
-                    if (n <= 0 || n >= PATH_MAX) {
+                    if (n <= 0 || n >= PATH_MAX) { // PATH_MAX includes the terminating 0
                        fprintf(stderr, "vdr: invalid directory path length: %s\n", optarg);
                        return 2;
                        }
@@ -293,7 +293,7 @@
                        return 2;
                        }
                     n = strtol(s, &s, 10);
-                    if (n <= 0 || n >= NAME_MAX) {
+                    if (n <= 0 || n > NAME_MAX) { // NAME_MAX excludes the terminating 0
                        fprintf(stderr, "vdr: invalid directory name length: %s\n", optarg);
                        return 2;
                        }
@@ -530,7 +530,7 @@
                "\n",
                DEFAULTCACHEDIR,
                DEFAULTCONFDIR,
-               PATH_MAX,
+               PATH_MAX - 1,
                NAME_MAX,
                DEFAULTEPGDATAFILENAME,
                MAXVIDEOFILESIZEDEFAULT,
