--- a/src/main.cpp	2017-05-14 22:02:13.515896315 +0200
+++ b/src/main.cpp	2017-05-14 22:02:24.847799093 +0200
@@ -1533,7 +1533,7 @@
   struct INPUTSTREAM_INFO GetStream(int streamid)
   {
     static struct INPUTSTREAM_INFO dummy_info = {
-      INPUTSTREAM_INFO::TYPE_NONE, "", "", 0, 0, 0, "",
+      INPUTSTREAM_INFO::TYPE_NONE, "", "", 0, 0, 0, 0, "",
       0, 0, 0, 0, 0.0f,
       0, 0, 0, 0, 0 };
 
