diff -Naur libcap-2.19/Make.Rules libcap-2.19_patched//Make.Rules
--- libcap-2.19/Make.Rules	2010-01-14 03:04:58.000000000 +0100
+++ libcap-2.19_patched//Make.Rules	2011-11-21 12:18:11.665373218 +0100
@@ -59,11 +59,12 @@
 LD=$(CC) -Wl,-x -shared
 LDFLAGS := #-g
 
-SYSTEM_HEADERS = /usr/include
+#SYSTEM_HEADERS = /usr/include
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
 CFLAGS += -Dlinux $(WARNINGS) $(DEBUG) $(IPATH)
-PAM_CAP := $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo yes ; else echo no ; fi)
+#PAM_CAP := $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo yes ; else echo no ; fi)
+PAM_CAP := no
 INDENT := $(shell if [ -n "$(which indent 2>/dev/null)" ]; then echo "| indent -kr" ; fi)
 DYNAMIC := $(shell if [ ! -d "$(topdir)/.git" ]; then echo yes; fi)
 LIBATTR := yes
