--- a/src/libcec/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp	2015-05-22 07:13:16.461476211 +0200
+++ b/src/libcec/adapter/TDA995x/TDA995xCECAdapterCommunication.cpp	2015-05-22 07:13:16.481484779 +0200
@@ -178,7 +178,7 @@
   
   m_messageMutex.Lock();
   uint32_t msgKey = ++m_iNextMessage;
-  m_messages.insert(make_pair(msgKey, entry));
+  m_messages.insert(std::make_pair(msgKey, entry));
  
   if (m_dev->Write((char *)&frame, sizeof(frame)) == sizeof(frame))
   {
@@ -353,7 +353,7 @@
         opcode = ( frame.size > 4 ) ? frame.data[1] : (uint32_t)CEC_OPCODE_NONE;
 
         m_messageMutex.Lock();
-        for (map<uint32_t, CAdapterMessageQueueEntry *>::iterator it = m_messages.begin(); 
+        for (std::map<uint32_t, CAdapterMessageQueueEntry *>::iterator it = m_messages.begin(); 
              !bHandled && it != m_messages.end(); it++)
         {
           bHandled = it->second->CheckMatch(opcode, initiator, destination, status);
