--- a/src/libcec/cmake/CheckPlatformSupport.cmake	2016-04-23 10:28:27.310704724 +0200
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake	2016-04-23 10:28:27.334704725 +0200
@@ -112,8 +112,8 @@
   endif()
 
   # TDA995x
-  check_include_files("tda998x_ioctl.h;comps/tmdlHdmiCEC/inc/tmdlHdmiCEC_Types.h" HAVE_TDA995X_API)
-  if (HAVE_TDA995X_API)
+  check_include_files("tda998x_ioctl.h;comps/tmdlHdmiCEC/inc/tmdlHdmiCEC_Types.h" HAVE_TDA995X_API_INC)
+  if (HAVE_TDA995X_API_INC)
     set(HAVE_TDA995X_API 1)
     set(LIB_INFO "${LIB_INFO}, TDA995x")
     set(CEC_SOURCES_ADAPTER_TDA995x adapter/TDA995x/TDA995xCECAdapterDetection.cpp
