--- a/src/libcec/CECClient.cpp	2017-05-25 09:45:17.116788693 +0200
+++ b/src/libcec/CECClient.cpp	2017-05-25 09:45:17.140788478 +0200
@@ -133,7 +133,7 @@
       (*it)->SetOSDName(m_configuration.strDeviceName);
 
     // set the default menu language for devices we control
-    (*it)->SetMenuLanguage(m_configuration.strDeviceLanguage);
+    (*it)->SetMenuLanguage(std::string(m_configuration.strDeviceLanguage, 3));
   }
 
   // set the physical address
