--- a/configure.in	2011-01-04 12:55:39.000000000 +0100
+++ b/configure.in	2012-02-24 10:42:05.564338544 +0100
@@ -44,7 +44,6 @@
 AC_CONFIG_AUX_DIR(scripts)
 AC_CONFIG_HEADERS(config.h)
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION)
-AM_ACLOCAL_INCLUDE(scripts)
 
 AC_CANONICAL_HOST                                                               
 
@@ -125,7 +124,6 @@
     dnl FSM_LIB="-lpthread"
   else
     SIP_FSM_FLAGS="-DOSIP_MT"
-    ACX_PTHREAD()
   fi
 fi
 
