--- a/src/mrvl_offscreen_memory.c	2016-06-16 23:09:26.842560213 +0200
+++ b/src/mrvl_offscreen_memory.c	2016-06-16 23:09:26.866560214 +0200
@@ -2,7 +2,11 @@
 //(C) Copyright 2009 Marvell International Ltd.
 //All Rights Reserved
 ******************************************************************************/
-#include "xf86.h"
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include <unistd.h>
 #include <X11/Xmd.h>
 #include "fb.h"
