diff -Naur strace-4.5.19.orig/configure strace-4.5.19/configure
--- strace-4.5.19.orig/configure	2010-02-18 23:48:15.570757707 +0100
+++ strace-4.5.19/configure	2010-02-18 23:48:27.610758123 +0100
@@ -5880,6 +5880,7 @@
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_compile "$LINENO" "$ac_header" "$as_ac_Header" "#include <stddef.h>
+#include <sys/socket.h>
 #include <linux/socket.h>
 "
 eval as_val=\$$as_ac_Header
