diff -Naur xf86-video-nvidia/usr/src/nv/nv-i2c.c xf86-video-nvidia2/usr/src/nv/nv-i2c.c
--- xf86-video-nvidia/usr/src/nv/nv-i2c.c	2009-04-17 04:40:38.000000000 +0200
+++ xf86-video-nvidia2/usr/src/nv/nv-i2c.c	2009-07-26 15:41:50.000000000 +0200
@@ -218,8 +218,6 @@
 #if defined(NV_I2C_ADAPTER_HAS_DEC_USE)
     .dec_use           = nv_i2c_adapter_dec_use,
 #endif
-    .client_register   = nv_i2c_adapter_register_client,
-    .client_unregister = nv_i2c_adapter_unregister_client,
     .data              = NULL,
 };
 
@@ -241,8 +239,6 @@
     .owner             = THIS_MODULE,
     .algo              = &nv_i2c_algo,
     .algo_data         = NULL,
-    .client_register   = nv_i2c_adapter_register_client,
-    .client_unregister = nv_i2c_adapter_unregister_client,
 };
 
 #endif // defined(KERNEL_2_4)
diff -Naur xf86-video-nvidia/usr/src/nv/os-agp.c xf86-video-nvidia2/usr/src/nv/os-agp.c
--- xf86-video-nvidia/usr/src/nv/os-agp.c	2009-04-17 04:40:38.000000000 +0200
+++ xf86-video-nvidia2/usr/src/nv/os-agp.c	2009-07-26 15:41:50.000000000 +0200
@@ -293,7 +293,7 @@
     {
          nv_pte_t *page_ptr = at->page_table[i];
 
-         page_ptr->phys_addr = (ptr->memory[i] & PAGE_MASK);
+         page_ptr->phys_addr = (page_to_phys(ptr->pages[i]) & PAGE_MASK);
          page_ptr->virt_addr = (unsigned long) __va(page_ptr->phys_addr);
          page_ptr->dma_addr  = page_ptr->phys_addr;
     }
