diff -Naur vdr-dvd-0.3.6+cvs20070812/Makefile vdr-dvd-0.3.6+cvs20070812a/Makefile
--- vdr-dvd-0.3.6+cvs20070812/Makefile	2006-05-02 19:52:46.000000000 +0200
+++ vdr-dvd-0.3.6+cvs20070812a/Makefile	2008-06-22 18:11:25.000000000 +0200
@@ -43,7 +43,7 @@
 
 INCLUDES += -I$(VDRDIR)/include -I$(NAVDIR)
 
-DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
+DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -D__STDC_LIMIT_MACROS
 
 # to use xine videoout:
 ifdef POLLTIMEOUTS_BEFORE_DEVICECLEAR
diff -Naur vdr-dvd-0.3.6+cvs20070812/player-dvd.h vdr-dvd-0.3.6+cvs20070812a/player-dvd.h
--- vdr-dvd-0.3.6+cvs20070812/player-dvd.h	2007-09-17 21:04:43.000000000 +0200
+++ vdr-dvd-0.3.6+cvs20070812a/player-dvd.h	2008-06-22 18:11:47.000000000 +0200
@@ -11,6 +11,8 @@
 #ifndef __PLAYER_DVD_H
 #define __PLAYER_DVD_H
 
+#include <inttypes.h>
+
 #include <dvdnav/dvdnav.h>
 #include <dvdnav/dvdnav_events.h>
 
