diff -Naur libdvdcss-1.2.9.orig/Makefile.am libdvdcss-1.2.9/Makefile.am
--- libdvdcss-1.2.9.orig/Makefile.am	2008-08-04 01:07:39.000000000 +0200
+++ libdvdcss-1.2.9/Makefile.am	2008-08-04 01:08:06.000000000 +0200
@@ -1,12 +1,6 @@
-SUBDIRS = src test doc
+SUBDIRS = src test
 DIST_SUBDIRS = $(SUBDIRS) debian
 
 EXTRA_DIST = libdvdcss.spec bootstrap
 
 AUTOMAKE_OPTIONS = foreign dist-bzip2 subdir-objects
-
-doc-dummy:
-
-doc: doc-dummy
-	$(MAKE) -C doc doc
-
diff -Naur libdvdcss-1.2.9.orig/Makefile.in libdvdcss-1.2.9/Makefile.in
--- libdvdcss-1.2.9.orig/Makefile.in	2008-08-04 01:07:39.000000000 +0200
+++ libdvdcss-1.2.9/Makefile.in	2008-08-04 01:08:17.000000000 +0200
@@ -191,7 +191,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = src test doc
+SUBDIRS = src test
 DIST_SUBDIRS = $(SUBDIRS) debian
 EXTRA_DIST = libdvdcss.spec bootstrap
 AUTOMAKE_OPTIONS = foreign dist-bzip2 subdir-objects
@@ -631,10 +631,6 @@
 	tags tags-recursive uninstall uninstall-am uninstall-info-am
 
 
-doc-dummy:
-
-doc: doc-dummy
-	$(MAKE) -C doc doc
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
