diff -Nur sysfsutils-1.3.0.orig/Makefile.in sysfsutils-1.3.0/Makefile.in
--- sysfsutils-1.3.0.orig/Makefile.in	Wed Jun  1 19:17:49 2005
+++ sysfsutils-1.3.0/Makefile.in	Fri Dec  9 00:14:07 2005
@@ -31,7 +31,7 @@
 libdir = @libdir@
 infodir = @infodir@
 mandir = @mandir@
-includedir = /usr/include/sysfs
+includedir = @includedir@/sysfs
 oldincludedir = /usr/include
 pkgdatadir = $(datadir)/@PACKAGE@
 pkglibdir = $(libdir)/@PACKAGE@
