--- vdr-dbus2vdr-12.0/Makefile.org	2013-06-03 08:08:05.108476638 +0200
+++ vdr-dbus2vdr-12.0/Makefile	2013-06-03 08:07:27.740475996 +0200
@@ -28,7 +28,7 @@
 ### The compiler options:
 
 export CFLAGS   = $(call PKGCFG,cflags)
-export CXXFLAGS = $(call PKGCFG,cxxflags) $(shell pkg-config --cflags dbus-1 glib-2.0 gio-2.0) $(shell libpng-config --cflags)
+export CXXFLAGS = -lintl $(call PKGCFG,cxxflags) $(shell pkg-config --cflags dbus-1 glib-2.0 gio-2.0) $(shell libpng-config --cflags)
 export LDADD    += $(shell pkg-config --libs dbus-1 glib-2.0 gio-2.0) $(shell libpng-config --ldflags)
 
 
