--- vdr-softhddevice-0.5.2+746746d5b/Makefile.orig	2013-02-27 00:25:53.358077458 +0100
+++ vdr-softhddevice-0.5.2+746746d5b/Makefile	2013-02-27 00:34:53.890086737 +0100
@@ -30,6 +30,7 @@
 #CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
 #CONFIG += -DNO_TS_AUDIO		# disable ts audio parser
 #CONFIG += -DUSE_TS_VIDEO		# build new ts video parser
+CONFIG += -DUSE_YAEPG
 
 ifeq ($(ALSA),1)
 CONFIG += -DUSE_ALSA
