diff -ruN vdr-1/menu.h vdr-2/menu.h
--- vdr-1/menu.h	2012-09-10 22:37:18.000000000 +0200
+++ vdr-2/menu.h	2012-09-10 22:38:14.000000000 +0200
@@ -350,6 +350,7 @@
   virtual void Hide(void);
   bool Visible(void) { return visible; }
   static void SetRecording(const char *FileName);
+  static void SetRecording(const char *FileName, const char *) { SetRecording(FileName); }
   static const char *NowReplaying(void);
   static const char *LastReplayed(void);
   static void ClearLastReplayed(const char *FileName);
