diff -Naur util-linux-2.13-pre7.orig/fdisk/cfdisk.c util-linux-2.13-pre7/fdisk/cfdisk.c
--- util-linux-2.13-pre7.orig/fdisk/cfdisk.c	2006-09-18 23:44:23.000000000 +0200
+++ util-linux-2.13-pre7/fdisk/cfdisk.c	2006-09-18 23:44:51.000000000 +0200
@@ -79,6 +79,7 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <linux/types.h>
+#include <locale.h>
 
 #include "nls.h"
 #include "xstrncpy.h"
diff -Naur util-linux-2.13-pre7.orig/fdisk/sfdisk.c util-linux-2.13-pre7/fdisk/sfdisk.c
--- util-linux-2.13-pre7.orig/fdisk/sfdisk.c	2006-09-18 23:44:23.000000000 +0200
+++ util-linux-2.13-pre7/fdisk/sfdisk.c	2006-09-18 23:44:42.000000000 +0200
@@ -47,6 +47,7 @@
 #include <sys/stat.h>
 #include <sys/utsname.h>
 #include <linux/unistd.h>	/* _syscall */
+#include <locale.h>
 #include "nls.h"
 #include "common.h"
 
diff -Naur util-linux-2.13-pre7.orig/mount/mount.c util-linux-2.13-pre7/mount/mount.c
--- util-linux-2.13-pre7.orig/mount/mount.c	2008-04-05 20:02:39.000000000 +0200
+++ util-linux-2.13-pre7/mount/mount.c	2008-04-05 20:05:00.000000000 +0200
@@ -20,6 +20,7 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <sys/mount.h>
+#include <locale.h>
 
 #include "mount_blkid.h"
 #include "mount_constants.h"
