diff -Naur vdr-extrecmenu-1.1/mymenurecordings.c vdr-extrecmenu-1.1a/mymenurecordings.c
--- vdr-extrecmenu-1.1/mymenurecordings.c	2007-10-24 19:29:46.000000000 +0200
+++ vdr-extrecmenu-1.1a/mymenurecordings.c	2009-04-17 10:55:26.000000000 +0200
@@ -800,7 +800,7 @@
  if(item&&!item->IsDirectory())
  {
   cDevice::PrimaryDevice()->StopReplay();
-  cResumeFile ResumeFile(item->FileName());
+  cResumeFile ResumeFile(item->FileName(), false);
   ResumeFile.Delete();
   return Play();
  }
diff -Naur vdr-extrecmenu-1.1/tools.c vdr-extrecmenu-1.1a/tools.c
--- vdr-extrecmenu-1.1/tools.c	2007-08-28 18:40:36.000000000 +0200
+++ vdr-extrecmenu-1.1a/tools.c	2009-04-17 10:55:20.000000000 +0200
@@ -12,6 +12,12 @@
 #include "tools.h"
 #include "mymenusetup.h"
 
+// Picture types:
+#define NO_PICTURE 0
+#define I_FRAME    1
+#define P_FRAME    2
+#define B_FRAME    3
+
 using namespace std;
 extern bool VfatFileSytem;
 
@@ -374,8 +380,10 @@
   cMarks frommarks,tomarks;
   cMark *mark;
   const char *error=NULL;
-  uchar filenumber,picturetype,buffer[MAXFRAMESIZE];
-  int fileoffset,length,index,currentfilenumber=0,filesize=0,lastiframe=0;
+  uint16_t filenumber;
+  off_t fileoffset;
+  uchar picturetype,buffer[MAXFRAMESIZE];
+  int length,index,currentfilenumber=0,filesize=0,lastiframe=0;
   bool lastmark=false,cutin=true;
 
   if(frommarks.Load(From.c_str()) && frommarks.Count())
@@ -409,7 +417,7 @@
   }
 
   isyslog("[extecmenu] editing %s",From.c_str());
-  while(fromindex->Get(index++,&filenumber,&fileoffset,&picturetype,&length) && Running() && !cancelcut)
+  while(fromindex->Get(index++,&filenumber,&fileoffset,NULL,&length) && Running() && !cancelcut)
   {
     AssertFreeDiskSpace(-1);
     
