--- iw-3.6/version.sh.orig	2012-10-24 17:48:11.000000000 +0200
+++ iw-3.6/version.sh	2012-10-24 17:48:56.000000000 +0200
@@ -3,21 +3,5 @@
 VERSION="3.6"
 OUT="$1"
 
-if head=`git rev-parse --verify HEAD 2>/dev/null`; then
-	git update-index --refresh --unmerged > /dev/null
-	descr=$(git describe)
-
-	# on git builds check that the version number above
-	# is correct...
-	[ "${descr%%-*}" = "v$VERSION" ] || exit 2
-
-	v="${descr#v}"
-	if git diff-index --name-only HEAD | read dummy ; then
-		v="$v"-dirty
-	fi
-else
-	v="$VERSION"
-fi
-
 echo '#include "iw.h"' > "$OUT"
-echo "const char iw_version[] = \"$v\";" >> "$OUT"
+echo "const char iw_version[] = \"$VERSION\";" >> "$OUT"
