--- a/inc/Module/Install/PRIVATE/Net/SSLeay.pm.orig	2008-09-28 15:43:59.000000000 +0400
+++ a/inc/Module/Install/PRIVATE/Net/SSLeay.pm	2008-09-28 15:46:52.000000000 +0400
@@ -45,15 +45,8 @@
 
     $self->makemaker_args( CCCDLFLAGS => $opts->{cccdlflags} );
 
-    if ( $self->prompt(
-            "Do you want to run external tests?\n".
-            "These tests *will* *fail* if you do not have network connectivity.",
-            'n',
-    ) =~ /^y/i ) {
-        $self->tests('t/*/*.t t/*/*/*.t');
-    } else {
-        $self->tests('t/local/*.t t/handle/local/*.t');
-    }
+    # Skip tests that need network connectivity.
+    $self->tests('t/local/*.t t/handle/local/*.t');
 }
 
 sub ssleay_get_build_opts {
