diff -Naur mythtv-0.24/configure mythtv-0.24a/configure
--- mythtv-0.24/configure	2010-11-08 18:40:43.000000000 +0100
+++ mythtv-0.24a/configure	2010-12-21 14:47:23.349255951 +0100
@@ -4558,8 +4558,8 @@
 fi
 
 if enabled freetype2; then
-    echo "FREETYPE_CFLAGS=`freetype-config --cflags`" >> $MYTH_CONFIG_MAK
-    echo "FREETYPE_LIBS=`freetype-config --libs`" >> $MYTH_CONFIG_MAK
+    echo "FREETYPE_CFLAGS=`$FREETYPE_CONFIG --cflags`" >> $MYTH_CONFIG_MAK
+    echo "FREETYPE_LIBS=`$FREETYPE_CONFIG --libs`" >> $MYTH_CONFIG_MAK
 fi
 
 if test $target_os = darwin; then
