From fc9de9e0a14332e6206f31170e931c303cd93e97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Sat, 17 Jul 2010 19:57:00 +0200
Subject: [PATCH 3/8] videoscale: Fix caps fixating if the height is fixed but the width isn't

---
 gst/videoscale/gstvideoscale.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/gst/videoscale/gstvideoscale.c b/gst/videoscale/gstvideoscale.c
index 46170b8..d7656a5 100644
--- a/gst/videoscale/gstvideoscale.c
+++ b/gst/videoscale/gstvideoscale.c
@@ -475,10 +475,12 @@ gst_video_scale_fixate_caps (GstBaseTransform * base, GstPadDirection direction,
     gst_structure_get_int (ins, "width", &from_w);
     gst_structure_get_int (ins, "height", &from_h);
 
+    gst_structure_get_int (outs, "width", &w);
+    gst_structure_get_int (outs, "height", &h);
+
     /* if both width and height are already fixed, we can't do anything
      * about it anymore */
-    if (gst_structure_get_int (outs, "width", &w)
-        && gst_structure_get_int (outs, "height", &h)) {
+    if (w && h) {
       guint n, d;
 
       GST_DEBUG_OBJECT (base, "dimensions already set to %dx%d, not fixating",
-- 
1.7.1

