From 6fb91f771130dbfcb62d89ce840863384a5cf911 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Mon, 19 Jul 2010 15:44:40 +0200
Subject: [PATCH 7/8] videoscale: Rename borders property to add-borders

---
 gst/videoscale/gstvideoscale.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/videoscale/gstvideoscale.c b/gst/videoscale/gstvideoscale.c
index 0a7971f..cef6457 100644
--- a/gst/videoscale/gstvideoscale.c
+++ b/gst/videoscale/gstvideoscale.c
@@ -243,7 +243,7 @@ gst_video_scale_class_init (GstVideoScaleClass * klass)
           G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
   g_object_class_install_property (gobject_class, PROP_ADD_BORDERS,
-      g_param_spec_boolean ("borders", "Add Borders",
+      g_param_spec_boolean ("add-borders", "Add Borders",
           "Add black borders if necessary to keep the display aspect ratio",
           DEFAULT_PROP_ADD_BORDERS,
           G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
-- 
1.7.1

