diff -Naur ./tests/Makefile.am ../gst-vaapi-0.2.5_patched/tests/Makefile.am
--- ./tests/Makefile.am	2011-05-03 14:44:18.612879614 +0200
+++ ../gst-vaapi-0.2.5_patched/tests/Makefile.am	2011-05-03 14:45:16.804756501 +0200
@@ -55,8 +55,8 @@
 test_windows_LDADD	= $(TEST_LIBS) $(TEST_X11_LIBS)
 
 test_textures_SOURCES	= test-textures.c image.c
-test_textures_CFLAGS	= $(TEST_CFLAGS) $(TEST_GLX_CFLAGS)
-test_textures_LDADD	= $(TEST_LIBS) $(TEST_GLX_LIBS)
+test_textures_CFLAGS	= $(TEST_CFLAGS) $(TEST_MIX_CFLAGS)
+test_textures_LDADD	= $(TEST_LIBS) $(TEST_MIX_LIBS)
 
 EXTRA_DIST = \
 	image.h			\
