--- vdr-2.0.1/config.h.org	2013-04-13 18:31:16.003935517 +0200
+++ vdr-2.0.1/config.h	2013-04-13 18:31:39.703934811 +0200
@@ -28,8 +28,8 @@
 
 // The plugin API's version number:
 
-#define APIVERSION  "2.0.0"
-#define APIVERSNUM   20000  // Version * 10000 + Major * 100 + Minor
+#define APIVERSION  "2.0.2"
+#define APIVERSNUM   20002  // Version * 10000 + Major * 100 + Minor
 
 // When loading plugins, VDR searches them by their APIVERSION, which
 // may be smaller than VDRVERSION in case there have been no changes to
