--- a/drivers/gpu/drm/imx/dw_hdmi-imx.c	2017-10-15 13:07:59.841507314 +0200
+++ b/drivers/gpu/drm/imx/dw_hdmi-imx.c	2017-10-15 13:07:59.869507117 +0200
@@ -151,6 +151,10 @@
 imx6q_hdmi_mode_valid(struct drm_connector *con,
 		      const struct drm_display_mode *mode)
 {
+	/* Rudi: temporary disable interlaced modes */
+	if (mode->flags & DRM_MODE_FLAG_INTERLACE)
+		return MODE_NO_INTERLACE;
+
 	if (mode->clock < 13500)
 		return MODE_CLOCK_LOW;
 	/* FIXME: Hardware is capable of 266MHz, but setup data is missing. */
@@ -164,6 +168,10 @@
 imx6dl_hdmi_mode_valid(struct drm_connector *con,
 		       const struct drm_display_mode *mode)
 {
+	/* Rudi: temporary disable interlaced modes */
+	if (mode->flags & DRM_MODE_FLAG_INTERLACE)
+		return MODE_NO_INTERLACE;
+
 	if (mode->clock < 13500)
 		return MODE_CLOCK_LOW;
 	/* FIXME: Hardware is capable of 270MHz, but setup data is missing. */
