diff --git a/arch/arm/mach-mx6/board-mx6q_microsom.c b/arch/arm/mach-mx6/board-mx6q_microsom.c
index e8d84df..07b71f3 100644
--- a/arch/arm/mach-mx6/board-mx6q_microsom.c
+++ b/arch/arm/mach-mx6/board-mx6q_microsom.c
@@ -507,6 +512,13 @@ void __init fixup_usom_board(struct machine_desc *desc, struct tag *tags,
 				str += 7;
 				imx6q_gpu_pdata.reserved_mem_size = memparse(str, &str);
 			}
+			/* VPU reserved memory */
+			str = t->u.cmdline.cmdline;
+			str = strstr(str, "vpumem=");
+			if (str != NULL) {
+				str += 7;
+				vout_mem.res_msize = memparse(str, &str);
+			}
 			break;
 		}
 	}
