--- a/sound/soc/soc-io.c	2017-10-22 18:35:24.079607839 +0200
+++ b/sound/soc/soc-io.c	2017-10-22 18:35:24.099607705 +0200
@@ -30,6 +30,9 @@
 {
 	int ret;
 
+	if (!component)
+		return -EIO;
+
 	if (component->regmap)
 		ret = regmap_read(component->regmap, reg, val);
 	else if (component->read)
@@ -52,6 +55,9 @@
 int snd_soc_component_write(struct snd_soc_component *component,
 	unsigned int reg, unsigned int val)
 {
+	if (!component)
+		return -EIO;
+
 	if (component->regmap)
 		return regmap_write(component->regmap, reg, val);
 	else if (component->write)
@@ -104,6 +110,9 @@
 	bool change;
 	int ret;
 
+	if (!component)
+		return -EIO;
+
 	if (component->regmap)
 		ret = regmap_update_bits_check(component->regmap, reg, mask,
 			val, &change);
@@ -140,6 +149,9 @@
 	bool change;
 	int ret;
 
+	if (!component)
+		return -EIO;
+
 	if (component->regmap)
 		ret = regmap_update_bits_check_async(component->regmap, reg,
 			mask, val, &change);
