--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c	2017-10-09 05:53:29.000000000 +0200
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c	2017-10-24 16:30:10.948577111 +0200
@@ -87,6 +87,12 @@
 	{ 0x6756, 0x78ab, 0x2000, 0x0200 }
 };
 
+static const u16 csc_coeff_rgb_out_limited[3][4] = {
+	{ 0x1b80, 0x0000, 0x0000, 0x0020 },
+	{ 0x0000, 0x1b80, 0x0000, 0x0020 },
+	{ 0x0000, 0x0000, 0x1b80, 0x0020 }
+};
+
 struct hdmi_vmode {
 	bool mdataenablepolarity;
 
@@ -103,6 +109,8 @@
 	unsigned int pix_repet_factor;
 	unsigned int hdcp_enable;
 	struct hdmi_vmode video_mode;
+	bool rgb_quantization_range_selectable;
+	enum hdmi_quantization_range rgb_quantization_range;
 };
 
 struct dw_hdmi_i2c {
@@ -755,7 +763,9 @@
 
 static int is_color_space_conversion(struct dw_hdmi *hdmi)
 {
-	return hdmi->hdmi_data.enc_in_bus_format != hdmi->hdmi_data.enc_out_bus_format;
+	return hdmi->hdmi_data.enc_in_bus_format != hdmi->hdmi_data.enc_out_bus_format ||
+		(hdmi_bus_fmt_is_rgb(hdmi->hdmi_data.enc_out_bus_format) && 
+		 hdmi->hdmi_data.rgb_quantization_range == HDMI_QUANTIZATION_RANGE_LIMITED);
 }
 
 static int is_color_space_decimation(struct dw_hdmi *hdmi)
@@ -790,7 +800,10 @@
 
 	if (is_color_space_conversion(hdmi)) {
 		if (hdmi_bus_fmt_is_rgb(hdmi->hdmi_data.enc_out_bus_format)) {
-			if (hdmi->hdmi_data.enc_out_encoding ==
+			if (hdmi_bus_fmt_is_rgb(
+					hdmi->hdmi_data.enc_in_bus_format))
+				csc_coeff = &csc_coeff_rgb_out_limited;
+			else if (hdmi->hdmi_data.enc_out_encoding ==
 						V4L2_YCBCR_ENC_601)
 				csc_coeff = &csc_coeff_rgb_out_eitu601;
 			else
@@ -1330,6 +1343,12 @@
 	/* Initialise info frame from DRM mode */
 	drm_hdmi_avi_infoframe_from_display_mode(&frame, mode, false);
 
+	/* Set up RGB quantization range */
+	drm_hdmi_avi_infoframe_quant_range(
+		&frame, mode, 
+		hdmi->hdmi_data.rgb_quantization_range, 
+		hdmi->hdmi_data.rgb_quantization_range_selectable);
+
 	if (hdmi_bus_fmt_is_yuv444(hdmi->hdmi_data.enc_out_bus_format))
 		frame.colorspace = HDMI_COLORSPACE_YUV444;
 	else if (hdmi_bus_fmt_is_yuv422(hdmi->hdmi_data.enc_out_bus_format))
@@ -1709,6 +1728,18 @@
 	/* TOFIX: Default to RGB888 output format */
 	hdmi->hdmi_data.enc_out_bus_format = MEDIA_BUS_FMT_RGB888_1X24;
 
+	/* Perpare rgb quantization range selection */
+	hdmi->hdmi_data.rgb_quantization_range_selectable = false;
+	hdmi->hdmi_data.rgb_quantization_range = HDMI_QUANTIZATION_RANGE_FULL;
+
+	if (hdmi->sink_is_hdmi) {
+		if (drm_rgb_quant_range_selectable((struct edid *)hdmi->edid))
+			hdmi->hdmi_data.rgb_quantization_range_selectable = true;
+		else
+			hdmi->hdmi_data.rgb_quantization_range = 
+				drm_default_rgb_quant_range(mode);
+	}
+
 	hdmi->hdmi_data.pix_repet_factor = 0;
 	hdmi->hdmi_data.hdcp_enable = 0;
 	hdmi->hdmi_data.video_mode.mdataenablepolarity = true;
