--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-ahb-audio.c	2017-11-04 09:59:15.000000000 +0100
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-ahb-audio.c	2017-11-04 09:59:41.876176564 +0100
@@ -120,6 +120,36 @@
 	{ 0xff, 0x13 },	/* FL,FR,LFE,FC,RL,RR,[FR]RC,[FR]LC */
 };
 
+static struct snd_pcm_chmap_elem dw_hdmi_chmaps[] = {
+	{ .channels = 2,	/* 2.0 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR } },
+	{ .channels = 3,	/* 3.0 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR, 
+	           SNDRV_CHMAP_FC } },
+	{ .channels = 4,	/* 4.0 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR,
+		   SNDRV_CHMAP_RL, SNDRV_CHMAP_RR } },
+	{ .channels = 5,	/* 4.1 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR,
+		   SNDRV_CHMAP_LFE,
+		   SNDRV_CHMAP_RL, SNDRV_CHMAP_RR } },
+	{ .channels = 6,	/* 5.1 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR,
+		   SNDRV_CHMAP_LFE, SNDRV_CHMAP_FC,
+		   SNDRV_CHMAP_RL, SNDRV_CHMAP_RR } },
+	{ .channels = 7,	/* 6.1 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR,
+		   SNDRV_CHMAP_LFE, SNDRV_CHMAP_FC,
+		   SNDRV_CHMAP_RL, SNDRV_CHMAP_RR, 
+		   SNDRV_CHMAP_RC } },
+	{ .channels = 8,	/* 7.1 */
+	  .map = { SNDRV_CHMAP_FL, SNDRV_CHMAP_FR,
+		   SNDRV_CHMAP_LFE, SNDRV_CHMAP_FC,
+		   SNDRV_CHMAP_RL, SNDRV_CHMAP_RR,
+		   SNDRV_CHMAP_SL, SNDRV_CHMAP_SR } },
+	{ }
+};
+
 struct snd_dw_hdmi {
 	struct snd_card *card;
 	struct snd_pcm *pcm;
@@ -574,6 +604,9 @@
 	strlcpy(pcm->name, DRIVER_NAME, sizeof(pcm->name));
 	snd_pcm_set_ops(pcm, SNDRV_PCM_STREAM_PLAYBACK, &snd_dw_hdmi_ops);
 
+	snd_pcm_add_chmap_ctls(
+		pcm, SNDRV_PCM_STREAM_PLAYBACK, dw_hdmi_chmaps, 8, 0, NULL);
+
 	/*
 	 * To support 8-channel 96kHz audio reliably, we need 512k
 	 * to satisfy alsa with our restricted period (ERR004323).
