--- a/drivers/watchdog/imx2_wdt.c	2014-12-17 15:56:17.000000000 +0100
+++ b/drivers/watchdog/imx2_wdt.c	2015-06-01 17:16:51.583884780 +0200
@@ -164,7 +164,7 @@
 {
 	if (test_bit(IMX2_WDT_EXPECT_CLOSE, &imx2_wdt.status) && !nowayout)
 		imx2_wdt_stop();
-	else {
+	else if (!timer_pending(&imx2_wdt.timer)) {
 		dev_crit(imx2_wdt_miscdev.parent,
 			"Unexpected close: Expect reboot!\n");
 		imx2_wdt_ping();
@@ -213,6 +213,15 @@
 	case WDIOC_GETTIMEOUT:
 		return put_user(imx2_wdt.timeout, p);
 
+	case WDIOC_SETOPTIONS:
+		if (get_user(new_value, p))
+			return -EFAULT;
+		if (new_value & WDIOS_DISABLECARD)
+			imx2_wdt_stop();
+		if (new_value & WDIOS_ENABLECARD)
+			imx2_wdt_start();
+		return 0;
+
 	default:
 		return -ENOTTY;
 	}
