diff --git a/drivers/video/mxc/mxc_edid.c b/drivers/video/mxc/mxc_edid.c
index 88b5268..7be0f45 100644
--- a/drivers/video/mxc/mxc_edid.c
+++ b/drivers/video/mxc/mxc_edid.c
@@ -299,6 +299,10 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 	}
 
 	if (revision >= 3) {
+		cfg->max_channels = 0;
+		cfg->sample_rates = 0;
+		cfg->sample_sizes = 0;
+
 		/* short desc */
 		DPRINTK("CEA Short desc timmings\n");
 		index++;
@@ -503,10 +507,6 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 					u8 audio_format, max_ch, byte1, byte2, byte3;
 
 					i = 0;
-					cfg->max_channels = 0;
-					cfg->sample_rates = 0;
-					cfg->sample_sizes = 0;
-
 					while (i < blklen) {
 						byte1 = edid[index + 1];
 						byte2 = edid[index + 2];
