--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-ahb-audio.c	2017-11-05 11:58:04.230320156 +0100
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-ahb-audio.c	2017-11-05 11:58:04.254320055 +0100
@@ -252,7 +252,7 @@
 static void dw_hdmi_create_cs(struct snd_dw_hdmi *dw,
 	struct snd_pcm_runtime *runtime)
 {
-	u8 cs[4];
+	u8 cs[5];
 	unsigned ch, i, j;
 
 	snd_pcm_create_iec958_consumer(runtime, cs, sizeof(cs));
