From b739880578a9d49225c8264a3b01209fd6df7d96 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sun, 20 Apr 2014 16:45:22 +0200
Subject: [PATCH 2/3] Indicate that the picture on HDMI is always underscanned

---
 drivers/video/mxc/mxc_hdmi.c |    9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 0115dee..a5802f89 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -1304,7 +1304,6 @@ static void hdmi_config_AVI(struct mxc_hdmi *hdmi)
 {
 	u8 val;
 	u8 pix_fmt;
-	u8 under_scan;
 	u8 act_ratio, coded_ratio, colorimetry, ext_colorimetry;
 	struct fb_videomode mode;
 	const struct fb_videomode *edid_mode;
@@ -1333,16 +1332,12 @@ static void hdmi_config_AVI(struct mxc_hdmi *hdmi)
 	else
 		pix_fmt = HDMI_FC_AVICONF0_PIX_FMT_RGB;
 
-	if (hdmi->edid_cfg.cea_underscan)
-		under_scan = HDMI_FC_AVICONF0_SCAN_INFO_UNDERSCAN;
-	else
-		under_scan =  HDMI_FC_AVICONF0_SCAN_INFO_NODATA;
-
 	/*
 	 * Active format identification data is present in the AVI InfoFrame.
 	 * Under scan info, no bar data
 	 */
-	val = pix_fmt | under_scan |
+	val = pix_fmt | 
+		HDMI_FC_AVICONF0_SCAN_INFO_UNDERSCAN |
 		HDMI_FC_AVICONF0_ACTIVE_FMT_INFO_PRESENT |
 		HDMI_FC_AVICONF0_BAR_DATA_NO_DATA;
 
-- 
1.7.9.5

