diff --git a/arch/arm/mach-mx6/crm_regs.h b/arch/arm/mach-mx6/crm_regs.h
index c4e923e..af932fb 100644
--- a/arch/arm/mach-mx6/crm_regs.h
+++ b/arch/arm/mach-mx6/crm_regs.h
@@ -55,7 +55,7 @@
 #define IOMUXC_GPR4_VDOA_CACHE_EN	(0xf << 28)
 #define IOMUXC_GPR4_VPU_CACHE_EN	(0xcc)
 #define IOMUXC_GPR4_IPU_CACHE_EN	(0x3)
-#define IOMUXC_GPR6_IPU1_QOS		(0x007f007f)
+#define IOMUXC_GPR6_IPU1_QOS		(0xffffffff)
 #define IOMUXC_GPR7_IPU2_QOS		(0x007f007f)
 
 /* MMDC */
