--- a/arch/arm/mach-mx6/board-mx6q_c1.c	2013-12-15 15:19:05.000000000 +0100
+++ b/arch/arm/mach-mx6/board-mx6q_c1.c	2013-12-15 15:43:43.438611223 +0100
@@ -87,7 +87,8 @@
 #define USOM_REG_ON		IMX_GPIO_NR(3, 19)
 #define USOM_XTAL_ON		IMX_GPIO_NR(5, 13)
 #define USOM_SD2_CD		IMX_GPIO_NR(1, 4)
-#define GPIO_IR_IN		IMX_GPIO_NR(1, 2) /* IMX_GPIO_NR(3, 9) for CuBox-i */
+#define GPIO_IR_IN_C1		IMX_GPIO_NR(1, 2) /* IMX_GPIO_NR(3, 9) for CuBox-i */
+#define GPIO_IR_IN_CUBOX	IMX_GPIO_NR(3, 9) /* IMX_GPIO_NR(1, 2) for Carrier-1 */
 #define USOM_USB_OTG_PWR	IMX_GPIO_NR(3, 22)
 #define USOM_USB_H1_PWR		IMX_GPIO_NR(1, 0)
 #define USOM_ENET_RST		IMX_GPIO_NR(4, 15)
@@ -539,7 +540,12 @@
 
 #ifdef CONFIG_IR_GPIO_CIR
 static struct gpio_ir_recv_platform_data c1_ir_data = {
-	.gpio_nr = GPIO_IR_IN,
+	.gpio_nr = GPIO_IR_IN_C1,
+	.active_low = 1,
+};
+
+static struct gpio_ir_recv_platform_data cubox_ir_data = {
+	.gpio_nr = GPIO_IR_IN_CUBOX,
 	.active_low = 1,
 };
 
@@ -550,6 +556,17 @@
 		.platform_data  = &c1_ir_data,
 	}
 };
+
+static int __init boardtype_setup(char *str)
+{
+	if (strcmp(str, "mx6-c1") != 0)
+		c1_ir.dev.platform_data = &cubox_ir_data;
+
+	return 1;
+}
+
+__setup("board_type=", boardtype_setup);
+
 #endif
 
 #if 0
