diff --git a/arch/arm/mach-mx6/board-mx6q_microsom.c b/arch/arm/mach-mx6/board-mx6q_microsom.c
index e8d84df..8094501 100644
--- a/arch/arm/mach-mx6/board-mx6q_microsom.c
+++ b/arch/arm/mach-mx6/board-mx6q_microsom.c
@@ -639,6 +639,7 @@ void imx6q_usom_init_fec(void) {
 void __init mx6_usom_board_init(void)
 {
 	int i;
+	struct platform_device *voutdev;
 	struct clk *clko, *clko2;
 	struct clk *new_parent;
 	int rate;
@@ -680,7 +681,15 @@ void __init mx6_usom_board_init(void)
 
 	imx6q_add_vdoa();
 	imx6q_add_ldb(&ldb_data);
-	imx6q_add_v4l2_output(0);
+	voutdev = imx6q_add_v4l2_output(0);
+	if (vout_mem.res_msize && voutdev) {
+		dma_declare_coherent_memory(&voutdev->dev,
+					    vout_mem.res_mbase,
+					    vout_mem.res_mbase,
+					    vout_mem.res_msize,
+					    (DMA_MEMORY_MAP |
+					     DMA_MEMORY_EXCLUSIVE));
+	}
 	imx6q_add_v4l2_capture(0, &capture_data[0]);
 	imx6q_add_v4l2_capture(1, &capture_data[1]);
 	imx6q_add_imx_snvs_rtc();
