--- a/arch/arm/mach-mx6/board-mx6q_microsom.c	2013-12-29 16:41:39.937117033 +0100
+++ b/arch/arm/mach-mx6/board-mx6q_microsom.c	2013-12-29 16:40:42.869118716 +0100
@@ -601,7 +601,7 @@
 	imx6q_add_sdhci_usdhc_imx(0, &mx6q_usom_sd1_data);
 }
 
-void imx6q_usom_init_fec(void) {
+void __init imx6q_usom_init_fec(void) {
 	struct clk *enet;
 	/* Set GPR1, bit 21 to 1 */
 	mxc_iomux_set_gpr_register(1, 21, 1, 1);
