diff --git a/arch/arm/boot/dts/imx6q-cm-fx6.dts b/arch/arm/boot/dts/imx6q-cm-fx6.dts
index bc7587c383f68..65a42bc5c2d00 100644
--- a/arch/arm/boot/dts/imx6q-cm-fx6.dts
+++ b/arch/arm/boot/dts/imx6q-cm-fx6.dts
@@ -413,12 +413,5 @@
 	pinctrl-0 = <&pinctrl_usdhc1>;
 	mmc-pwrseq = <&awnh387_pwrseq>;
 	non-removable;
-	/*
-	 * If the OS probes the Bluetooth AMP function advertised on this bus
-	 * but the firmware in place does not support it, the WiFi/BT module
-	 * gets unresponsive.
-	 * Users who configured their OS properly can enable this node to gain
-	 * WiFi and/or plain Bluetooth support.
-	 */
-	status = "disabled";
+	status = "okay";
 };
