From 553f4d161767cefa39920d95dfb7ec33e14fff46 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 6 Sep 2014 10:40:15 +0200
Subject: [PATCH 1101/1101] video: mxc_hdmi: Use symbolic constant instead of
 0.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 drivers/video/mxc/mxc_hdmi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 2799df1..9a21417 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -702,7 +702,7 @@ static void hdmi_video_csc(struct mxc_hdmi *hdmi)
 {
 	int color_depth = 0;
 	int interpolation = HDMI_CSC_CFG_INTMODE_DISABLE;
-	int decimation = 0;
+	int decimation = HDMI_CSC_CFG_DECMODE_DISABLE;
 	u8 val;
 
 	/* YCC422 interpolation to 444 mode */
-- 
1.7.9.5

