diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index 6380c4e..e07ae43 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -1949,6 +1949,9 @@ static void mxc_hdmi_edid_rebuild_modelist(struct mxc_hdmi *hdmi)
 		 */
 		mode = &hdmi->fbi->monspecs.modedb[i];
 
+		if (mode->vmode & FB_VMODE_INTERLACED)
+			continue;
+
 		vic = mxc_edid_mode_to_vic(mode);
 		if (hdmi->edid_cfg.hdmi_cap &&
 		    (vic == 0))
@@ -2005,7 +2008,7 @@ static void  mxc_hdmi_default_modelist(struct mxc_hdmi *hdmi)
 	/*Add all no interlaced CEA mode to default modelist */
 	for (i = 0; i < ARRAY_SIZE(mxc_cea_mode); i++) {
 		mode = mxc_cea_mode[i];
-		if (mode.xres != 0) {
+		if (!(mode.vmode & FB_VMODE_INTERLACED) && (mode.xres != 0)) {
 			if (ignore_edid)
 				mode.flag |= FB_MODE_IS_STANDARD;
 			fb_add_videomode(&mode, &hdmi->fbi->modelist);
@@ -2820,7 +2823,7 @@ static int mxc_hdmi_disp_init(struct mxc_dispdrv_handle *disp,
 	/*Add all no interlaced CEA mode to default modelist */
 	for (i = 0; i < ARRAY_SIZE(mxc_cea_mode); i++) {
 		mode = &mxc_cea_mode[i];
-		if (mode->xres != 0) {
+		if (!(mode->vmode & FB_VMODE_INTERLACED) && (mode->xres != 0)) {
 			struct fb_videomode m = *mode;
 			m.flag |= FB_MODE_IS_STANDARD;
 			fb_add_videomode(&m, &hdmi->fbi->modelist);
