diff --git a/sound/soc/fsl/imx-hdmi-dma.c b/sound/soc/fsl/imx-hdmi-dma.c
index 4155122..f811799 100644
--- a/sound/soc/fsl/imx-hdmi-dma.c
+++ b/sound/soc/fsl/imx-hdmi-dma.c
@@ -496,6 +496,20 @@ static int hdmi_dma_set_thrsld_incrtype(struct device *dev, int channels)
 	return 0;
 }
 
+static void hdmi_dma_set_hbr(struct device *dev, int channels)
+{
+	u8 value = hdmi_readb(HDMI_AHB_DMA_CONF0) & (~HDMI_AHB_DMA_CONF0_HBR);
+
+	/* non audio AES and 8 channels means we have to enable HBR */
+	if ((channels == 8) &&
+	    (iec_header.B.linear_pcm == 1)) {
+		value |= HDMI_AHB_DMA_CONF0_HBR;
+		dev_info(dev, "Enabling HBR");
+	}
+
+	hdmi_writeb(value, HDMI_AHB_DMA_CONF0);
+}
+
 static int hdmi_dma_configure_dma(struct device *dev, int channels)
 {
 	u8 i, val = 0;
@@ -517,6 +531,9 @@ static int hdmi_dma_configure_dma(struct device *dev, int channels)
 
 	hdmi_writeb(val, HDMI_AHB_DMA_CONF1);
 
+	/* Handle HBR */
+	hdmi_dma_set_hbr(dev, channels);
+
 	return 0;
 }
 
