diff --git a/sound/soc/fsl/imx-hdmi-dma.c b/sound/soc/fsl/imx-hdmi-dma.c
index f811799..cdd6006 100644
--- a/sound/soc/fsl/imx-hdmi-dma.c
+++ b/sound/soc/fsl/imx-hdmi-dma.c
@@ -122,8 +122,8 @@ EXPORT_SYMBOL(iec_header);
  *    is necessary for 6 ch.
  */
 #define HDMI_DMA_PERIOD_BYTES		(6144)
-#define HDMI_DMA_BUF_SIZE		(64 * 1024)
-#define HDMI_PCM_BUF_SIZE		(64 * 1024)
+#define HDMI_DMA_BUF_SIZE		(16 * 64 * 1024)
+#define HDMI_PCM_BUF_SIZE		(16 * 64 * 1024)
 
 #define hdmi_audio_debug(dev, reg) \
 	dev_dbg(dev, #reg ": 0x%02x\n", hdmi_readb(reg))
@@ -903,7 +903,7 @@ static struct snd_pcm_hardware snd_imx_hardware = {
 	.period_bytes_min = HDMI_DMA_PERIOD_BYTES / 2,
 	.period_bytes_max = HDMI_DMA_PERIOD_BYTES / 2,
 	.periods_min = 8,
-	.periods_max = 8,
+	.periods_max = 8*16,
 	.fifo_size = 0,
 };
 
