reverted:
--- b/drivers/video/mxc/mxc_ipuv3_fb.c
+++ a/drivers/video/mxc/mxc_ipuv3_fb.c
@@ -2345,7 +2345,6 @@
 	struct fb_info *fbi;
 	struct mxcfb_info *mxcfbi;
 	struct resource *res;
-	char *fix = kmalloc(11, GFP_KERNEL);
 	int ret = 0;
 
 	dev_dbg(&pdev->dev, "%s enter\n", __func__);
@@ -2355,11 +2354,6 @@
 		return pdev->id;
 	}
 
-	strlcpy(fix, pdev->name, 11);
-	pr_info("FB name %s\n", pdev->name);
-	pr_info("FB name %s\n", fix);
-	pdev->name = fix;
-
 	plat_data = devm_kzalloc(&pdev->dev, sizeof(struct
 					ipuv3_fb_platform_data), GFP_KERNEL);
 	if (!plat_data)
