--- a/sound/soc/fsl/fsl_spdif.h	2014-03-01 15:33:50.106848860 +0100
+++ b/sound/soc/fsl/fsl_spdif.h	2014-03-01 15:34:03.118848433 +0100
@@ -184,10 +184,12 @@
 					 SNDRV_PCM_RATE_64000 | \
 					 SNDRV_PCM_RATE_96000)
 
-#define FSL_SPDIF_FORMATS_PLAYBACK	(SNDRV_PCM_FMTBIT_S16_LE | \
+// !!! temporary workaround for broken 24bit mode !!!
+#define FSL_SPDIF_FORMATS_PLAYBACK	(SNDRV_PCM_FMTBIT_S16_LE)
+/*#define FSL_SPDIF_FORMATS_PLAYBACK	(SNDRV_PCM_FMTBIT_S16_LE | \
 					 SNDRV_PCM_FMTBIT_S20_3LE | \
 					 SNDRV_PCM_FMTBIT_S24_LE)
-
+*/
 #define FSL_SPDIF_FORMATS_CAPTURE	(SNDRV_PCM_FMTBIT_S24_LE)
 
 #endif /* _FSL_SPDIF_DAI_H */
