--- a/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-04-22 09:36:27.000000000 +0200
+++ b/arch/arm/boot/dts/imx6qdl-cubox-i.dtsi	2014-07-19 09:18:57.744595248 +0200
@@ -248,6 +248,16 @@
 };
 
 &spdif {
+	clocks = <&clks 197>, <&clks 0>,
+		  <&clks 197>, <&clks 0>,
+		  <&clks 0>,   <&clks 0>,
+		  <&clks 0>,  <&clks 0>,
+		  <&clks 0>, <&clks 156>;
+	clock-names = "core",  "rxtx0",
+		      "rxtx1", "rxtx2",
+		      "rxtx3", "rxtx4",
+		      "rxtx5", "rxtx6",
+		      "rxtx7", "dma";
 	status = "okay";
 };
 
--- a/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-04-22 09:36:27.000000000 +0200
+++ b/arch/arm/boot/dts/imx6dl-hummingboard.dts	2014-07-19 10:50:25.768433095 +0200
@@ -233,6 +219,16 @@
 };
 
 &spdif {
+	clocks = <&clks 197>, <&clks 0>,
+		  <&clks 197>, <&clks 0>,
+		  <&clks 0>,   <&clks 0>,
+		  <&clks 0>,  <&clks 0>,
+		  <&clks 0>, <&clks 156>;
+	clock-names = "core",  "rxtx0",
+		      "rxtx1", "rxtx2",
+		      "rxtx3", "rxtx4",
+		      "rxtx5", "rxtx6",
+		      "rxtx7", "dma";
 	status = "okay";
 };
 
