--- a/Makefile	2015-09-29 15:33:02.699030664 +0200
+++ b/Makefile	2015-09-29 15:33:02.719030642 +0200
@@ -125,6 +125,7 @@
 EXTRA_CFLAGS       += -I$(src)/src/wl/sys -I$(src)/src/wl/phy -I$(src)/src/wl/ppr/include
 EXTRA_CFLAGS       += -I$(src)/src/shared/bcmwifi/include
 #EXTRA_CFLAGS       += -DBCMDBG_ASSERT -DBCMDBG_ERR
+EXTRA_CFLAGS       += -Wno-date-time
 
 EXTRA_LDFLAGS      := $(src)/lib/wlc_hybrid.o_shipped
 
