diff -Naur clutter-gst-1.3.14/clutter-gst/Makefile.in clutter-gst-1.3.14a/clutter-gst/Makefile.in
--- clutter-gst-1.3.14/clutter-gst/Makefile.in	2011-08-31 17:51:05.000000000 +0200
+++ clutter-gst-1.3.14a/clutter-gst/Makefile.in	2011-09-20 15:38:47.909037507 +0200
@@ -438,7 +438,6 @@
 
 AM_CPPFLAGS = \
 	-DCLUTTER_GST_COMPILATION	\
-	-DG_DISABLE_DEPRECATED		\
 	-DG_DISABLE_SINGLE_INCLUDES	\
 	-DCOGL_DISABLE_DEPRECATED	\
 	-DCLUTTER_DISABLE_DEPRECATED	\
