--- a/drivers/net/ethernet/sun4i/sun4i_wemac.c	2013-05-14 16:07:23.000000000 +0200
+++ b/drivers/net/ethernet/sun4i/sun4i_wemac.c	2013-06-23 19:50:05.635660934 +0200
@@ -1502,7 +1502,7 @@
 
 	if (netif_msg_ifup(db))
 		dev_dbg(db->dev, "enabling %s\n", dev->name);
-
+#if 0
 	/* If there is no IRQ type specified, default to something that
 	 * may work, and tell the user that this is a problem */
 
@@ -1510,7 +1510,7 @@
 		dev_warn(db->dev, "WARNING: no IRQ resource flags set.\n");
 
 	irqflags |= IRQF_SHARED;
-
+#endif
 	if (request_irq(dev->irq, &wemac_interrupt, irqflags, dev->name, dev))
 		return -EAGAIN;
 
