diff -Naur tntnet-1.6.3/sdk/tools/ecppc/tnt/ecppc/scopevar.h tntnet-1.6.3a/sdk/tools/ecppc/tnt/ecppc/scopevar.h
--- tntnet-1.6.3/sdk/tools/ecppc/tnt/ecppc/scopevar.h	2008-02-16 10:10:27.000000000 +0100
+++ tntnet-1.6.3a/sdk/tools/ecppc/tnt/ecppc/scopevar.h	2009-01-05 10:37:58.000000000 +0100
@@ -22,6 +22,7 @@
 #define TNT_ECPPC_SCOPEVAR_H
 
 #include <string>
+#include <algorithm>
 #include <iosfwd>
 #include <tnt/ecpp/scopetypes.h>
 
