diff -Naur a/configure b/configure
--- a/configure	2011-07-08 06:54:01.000000000 +0200
+++ b/configure	2011-07-08 07:23:12.000000000 +0200
@@ -23141,7 +23141,7 @@
 else
 	pkg_failed=untried
 fi
-
+pkg_failed=untried
 
 
 if test $pkg_failed = yes; then
@@ -23204,7 +23204,7 @@
 else
 	pkg_failed=untried
 fi
-
+pkg_failed=untried
 
 
 if test $pkg_failed = yes; then
@@ -23279,7 +23279,7 @@
 else
 	pkg_failed=untried
 fi
-
+pkg_failed=untried
 
 
 if test $pkg_failed = yes; then
