--- a/lib/libv4lconvert/jpeg_memsrcdest.c	2017-03-05 15:38:00.136453654 +0100
+++ b/lib/libv4lconvert/jpeg_memsrcdest.c	2017-03-05 15:38:00.148453280 +0100
@@ -24,7 +24,7 @@
 
 /* libjpeg8 and later come with their own (API compatible) memory source
    and dest */
-#if JPEG_LIB_VERSION < 80
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
 
 /* Expanded data source object for memory input */
 
--- a/lib/libv4lconvert/jpeg_memsrcdest.h	2017-03-05 15:38:03.840304727 +0100
+++ b/lib/libv4lconvert/jpeg_memsrcdest.h	2017-03-05 15:38:03.856303652 +0100
@@ -1,5 +1,7 @@
 #include <jpeglib.h>
 
+#if JPEG_LIB_VERSION < 80 && !defined(MEM_SRCDST_SUPPORTED)
+
 void
 jpeg_mem_src (j_decompress_ptr cinfo, unsigned char * buffer,
 	unsigned long bufsize);
@@ -7,3 +9,5 @@
 void
 jpeg_mem_dest (j_compress_ptr cinfo, unsigned char ** outbuffer,
 	unsigned long * outsize);
+
+#endif
