From dd3e1e53a0c083c4dc16f3db4083ad5e4b7a866e Mon Sep 17 00:00:00 2001
From: Nicolas Chauvet <kwizart@gmail.com>
Date: Fri, 11 Jan 2013 21:04:40 +0100
Subject: [PATCH] Fix libva-vdpau-driver with GL_GLEXT_VERSION >= 85

---
 src/utils_glx.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/utils_glx.h b/src/utils_glx.h
index 19d03e4..42ae06f 100644
--- a/src/utils_glx.h
+++ b/src/utils_glx.h
@@ -67,6 +67,13 @@ typedef void (*PFNGLXRELEASETEXIMAGEEXTPROC)(Display *, GLXDrawable, int);
 #define GL_PROGRAM_UNDER_NATIVE_LIMITS GL_PROGRAM_UNDER_NATIVE_LIMITS_ARB
 #endif
 
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+ * 	block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+ * 		Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+
 const char *
 gl_get_error_string(GLenum error)
     attribute_hidden;
-- 
1.7.11.7

