--- OpenBM-20120103-git3b65875de5/gateway/nbproject/Makefile-Release.mk	2012-01-03 17:48:32.000000000 +0100
+++ OpenBM-20120103-git3b65875de5/gateway/nbproject/Makefile-Release.mk.new	2012-01-04 08:16:50.595790575 +0100
@@ -56,19 +56,19 @@
 ASFLAGS=
 
 # Link Libraries and Options
-LDLIBSOPTIONS=/usr/lib/libboost_date_time.a /usr/lib/libboost_system-mt.a /usr/lib/libboost_thread-mt.a /usr/lib/libboost_program_options-mt.a -ldl
+LDLIBSOPTIONS=${LIB_P}/lib/libboost_date_time-mt.a ${LIB_P}/lib/libboost_system-mt.a ${LIB_P}/lib/libboost_thread-mt.a ${LIB_P}/lib/libboost_program_options-mt.a -ldl
 
 # Build Targets
 .build-conf: ${BUILD_SUBPROJECTS}
 	"${MAKE}"  -f nbproject/Makefile-${CND_CONF}.mk ${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway
 
-${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_date_time.a
+#${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_date_time.a
 
-${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_system-mt.a
+#${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_system-mt.a
 
-${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_thread-mt.a
+#${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_thread-mt.a
 
-${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_program_options-mt.a
+#${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: /usr/lib/libboost_program_options-mt.a
 
 ${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}/gateway: ${OBJECTFILES}
 	${MKDIR} -p ${CND_DISTDIR}/${CND_CONF}/${CND_PLATFORM}
