From 28ebdd17dc780293ece2f373323cf68051e416fb Mon Sep 17 00:00:00 2001
From: Ole Ernst <olebowle@gmx.com>
Date: Mon, 21 Sep 2015 15:25:53 +0200
Subject: [PATCH] - Removed unused longopt

---
 src/main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 3c94663..f6c50fa 100644
--- a/src/main.c
+++ b/src/main.c
@@ -54,7 +54,6 @@ int main(int argc,char **argv)
         {"evmap",required_argument,NULL,'e'},
         {"socket",required_argument,NULL,'s'},
         {"mode",required_argument,NULL,'m'},
-        {"process",required_argument,NULL,'p'},
         {"repeat-filter",no_argument,NULL,'R'},
         {"release",required_argument,NULL,'r'},
         {0, 0, 0, 0}
