diff -Naur xf86-video-ati-X11R7.2-6.6.3.orig/src/aticonfig.c xf86-video-ati-X11R7.2-6.6.3/src/aticonfig.c
--- xf86-video-ati-X11R7.2-6.6.3.orig/src/aticonfig.c	2007-04-21 11:43:33.000000000 +0200
+++ xf86-video-ati-X11R7.2-6.6.3/src/aticonfig.c	2007-04-21 11:48:26.000000000 +0200
@@ -362,8 +362,8 @@
             "Using %s acceleration architecture\n",
             pATI->useEXA ? "EXA" : "XAA");
 
-        pATI->RenderAccelEnabled = FALSE;
 #if defined(USE_EXA)
+        pATI->RenderAccelEnabled = FALSE;
         if (pATI->useEXA && RenderAccel)
             pATI->RenderAccelEnabled = TRUE;
 #endif
