Index: sgx-kernel-module/sgx/eurasiacon/build/linux2/config/core.mk
===================================================================
--- sgx-kernel-module.orig/sgx/eurasiacon/build/linux2/config/core.mk	2012-07-04 10:52:08.000000000 +0200
+++ sgx-kernel-module/sgx/eurasiacon/build/linux2/config/core.mk	2012-07-04 14:29:03.000000000 +0200
@@ -471,9 +471,6 @@
 # Build-type dependent options
 #
 $(eval $(call BothConfigMake,BUILD,$(BUILD)))
-$(eval $(call KernelConfigC,DEBUG_LINUX_MMAP_AREAS,))
-$(eval $(call KernelConfigC,DEBUG_LINUX_MEM_AREAS,))
-$(eval $(call KernelConfigC,DEBUG_LINUX_MEMORY_ALLOCATIONS,))
 $(eval $(call KernelConfigC,PVRSRV_DISABLE_UM_SYNCOBJ_MAPPINGS,1))
 
 ifeq ($(BUILD),debug)
