--- xf86-video-intel-2.13.0/src/intel_driver.c.old	2010-11-07 15:42:33.092212509 +0100
+++ xf86-video-intel-2.13.0/src/intel_driver.c	2010-11-07 15:42:55.528207662 +0100
@@ -890,6 +890,8 @@
 		   "VideoRam: %d KB\n", scrn->videoRam);
 
 #ifdef DRI2
+	if (IS_GEN6(intel))
+		intel->directRenderingType = DRI_DISABLED;
 	if (intel->directRenderingType == DRI_NONE
 	    && I830DRI2ScreenInit(screen))
 		intel->directRenderingType = DRI_DRI2;
