diff -Nurp xserver-xorg-video-intel-2.13.901//uxa/uxa-accel.c xserver-xorg-video-intel-2.13.901-working//uxa/uxa-accel.c
--- xserver-xorg-video-intel-2.13.901//uxa/uxa-accel.c	2010-09-28 20:43:10.000000000 -0700
+++ xserver-xorg-video-intel-2.13.901-working//uxa/uxa-accel.c	2011-01-21 19:15:03.421218050 -0800
@@ -120,6 +120,10 @@ uxa_fill_spans(DrawablePtr pDrawable, GC
 
 		solid = uxa_acquire_solid(screen, src->pSourcePict);
 		FreePicture(src, 0);
+		if (!solid) {
+			FreePicture(dst, 0);
+			goto solid;
+		}
 
 		src = solid;
 		src_pixmap = uxa_get_offscreen_pixmap(src->pDrawable,
@@ -1110,6 +1114,10 @@ try_solid:
 
 			solid = uxa_acquire_solid(screen, src->pSourcePict);
 			FreePicture(src, 0);
+			if (!solid) {
+				FreePicture(dst, 0);
+				goto err;
+			}
 
 			src = solid;
 			src_pixmap = uxa_get_offscreen_pixmap(src->pDrawable,
diff -Nurp xserver-xorg-video-intel-2.13.901//uxa/uxa-render.c xserver-xorg-video-intel-2.13.901-working//uxa/uxa-render.c
--- xserver-xorg-video-intel-2.13.901//uxa/uxa-render.c	2010-09-28 20:43:10.000000000 -0700
+++ xserver-xorg-video-intel-2.13.901-working//uxa/uxa-render.c	2011-01-21 19:07:49.463951184 -0800
@@ -1064,6 +1064,8 @@ try_solid:
 			int src_off_x, src_off_y;
 
 			solid = uxa_acquire_solid(screen, src->pSourcePict);
+			if (! solid)
+				goto err_src;
 			FreePicture(src, 0);
 
 			src = solid;
