diff --git a/src/i915_render.c b/src/i915_render.c
index 6d844e5..9739a72 100644
--- a/src/i915_render.c
+++ b/src/i915_render.c
@@ -546,7 +546,7 @@ i915_emit_composite_primitive(PixmapPtr dest,
 {
 	ScrnInfoPtr scrn = xf86Screens[dest->drawable.pScreen->myNum];
 	intel_screen_private *intel = intel_get_screen_private(scrn);
-	Bool is_affine_src, is_affine_mask = TRUE;
+	Bool is_affine_src = TRUE, is_affine_mask = TRUE;
 	int per_vertex, num_floats;
 	int tex_unit = 0;
 	int src_unit = -1, mask_unit = -1;
diff --git a/src/xvmc/intel_batchbuffer.h b/src/xvmc/intel_batchbuffer.h
index 6fa1c05..888abeb 100644
--- a/src/xvmc/intel_batchbuffer.h
+++ b/src/xvmc/intel_batchbuffer.h
@@ -24,7 +24,7 @@ extern int VERBOSE;
 
 #define OUT_RELOC(bo,read_domains,write_domains,delta)  \
     do { \
-        *(unsigned int *)batch_ptr = delta + bo->offset; \
+        *(unsigned int *)batch_ptr = (delta) + bo->offset; \
         intel_batch_emit_reloc(bo, read_domains, write_domains, delta, batch_ptr); \
         batch_ptr += 4;                                                 \
     } while (0)
diff --git a/src/xvmc/intel_xvmc.c b/src/xvmc/intel_xvmc.c
index 6cbec80..6bfd9d1 100644
--- a/src/xvmc/intel_xvmc.c
+++ b/src/xvmc/intel_xvmc.c
@@ -1071,7 +1071,7 @@ _X_EXPORT Status XvMCLoadQMatrix(Display * display, XvMCContext * context,
 _X_EXPORT Status XvMCPutSlice(Display * display, XvMCContext * context,
 			      char *slice, int nbytes)
 {
-	if (xvmc_driver->put_slice(display, context, slice, nbytes)) {
+	if (xvmc_driver->put_slice(display, context, (unsigned char *) slice, nbytes)) {
 		XVMC_ERR("PutSlice fail\n");
 		return BadValue;
 	}
@@ -1082,7 +1082,7 @@ _X_EXPORT Status XvMCPutSlice2(Display * display, XvMCContext * context,
 			       char *slice, int nbytes, int slice_code)
 {
 	if (xvmc_driver->put_slice2
-	    (display, context, slice, nbytes, slice_code)) {
+	    (display, context, (unsigned char *) slice, nbytes, slice_code)) {
 		XVMC_ERR("PutSlice2 fail\n");
 		return BadValue;
 	}
diff --git a/src/xvmc/intel_xvmc_dump.c b/src/xvmc/intel_xvmc_dump.c
index 974bcbb..d22d311 100644
--- a/src/xvmc/intel_xvmc_dump.c
+++ b/src/xvmc/intel_xvmc_dump.c
@@ -38,7 +38,7 @@ void intel_xvmc_dump_open(void)
 	if (xvmc_dump)
 		return;
 
-	if (d = getenv("INTEL_XVMC_DUMP"))
+	if ((d = getenv("INTEL_XVMC_DUMP")))
 		xvmc_dump = 1;
 
 	if (xvmc_dump) {
