From ae8877e30707cae586565b4deee4c7fd9ac7d3ed Mon Sep 17 00:00:00 2001
From: Cyril Brulebois <kibi@debian.org>
Date: Mon, 31 Jan 2011 05:22:35 +0000
Subject: xvmc: Get rid of unused variables.

Signed-off-by: Cyril Brulebois <kibi@debian.org>
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/xvmc/i915_xvmc.c b/src/xvmc/i915_xvmc.c
index 21a1149..fbd4555 100644
--- a/src/xvmc/i915_xvmc.c
+++ b/src/xvmc/i915_xvmc.c
@@ -391,7 +391,6 @@ static void i915_mc_static_indirect_state_set(XvMCContext * context,
 	i915XvMCContext *pI915XvMC = (i915XvMCContext *) context->privData;
 	struct intel_xvmc_surface *intel_surf = dest->privData;
 	struct i915_mc_static_indirect_state_buffer *buffer_info;
-	unsigned int w = dest->width;
 
 	drm_intel_gem_bo_map_gtt(pI915XvMC->sis_bo);
 	buffer_info = pI915XvMC->sis_bo->virtual;
@@ -688,7 +687,7 @@ static void i915_mc_load_indirect_render_emit(XvMCContext * context)
 {
 	i915XvMCContext *pI915XvMC = (i915XvMCContext *) context->privData;
 	int mem_select;
-	uint32_t load_indirect, buffer_address;
+	uint32_t load_indirect;
 	BATCH_LOCALS;
 
 	BEGIN_BATCH(5);
diff --git a/src/xvmc/intel_batchbuffer.c b/src/xvmc/intel_batchbuffer.c
index 337b2ce..2b5526c 100644
--- a/src/xvmc/intel_batchbuffer.c
+++ b/src/xvmc/intel_batchbuffer.c
@@ -64,8 +64,6 @@ static void i965_end_batch(void)
 
 Bool intelInitBatchBuffer(void)
 {
-	int i;
-
 	if ((xvmc_driver->batch.buf =
 	     drm_intel_bo_alloc(xvmc_driver->bufmgr,
 				"batch buffer", BATCH_SIZE, 0x1000)) == NULL) {
diff --git a/src/xvmc/intel_xvmc.c b/src/xvmc/intel_xvmc.c
index 6cbec80..339c133 100644
--- a/src/xvmc/intel_xvmc.c
+++ b/src/xvmc/intel_xvmc.c
@@ -105,8 +105,6 @@ unsigned int mb_bytes_420[] = {
 
 void LOCK_HARDWARE(drm_context_t ctx)
 {
-	char __ret = 0;
-
 	PPTHREAD_MUTEX_LOCK();
 	assert(!xvmc_driver->locked);
 
@@ -218,11 +216,6 @@ _X_EXPORT Status XvMCCreateContext(Display * display, XvPortID port,
 	int error_base;
 	int event_base;
 	int priv_count;
-	int isCapable;
-	int screen = DefaultScreen(display);
-	intel_xvmc_context_ptr intel_ctx;
-	int fd;
-	char *driverName = NULL, *deviceName = NULL;
 
 	/* Verify Obvious things first */
 	if (!display || !context)
@@ -482,7 +475,6 @@ _X_EXPORT Status XvMCCreateBlocks(Display * display, XvMCContext * context,
 				  unsigned int num_blocks,
 				  XvMCBlockArray * block)
 {
-	Status ret;
 	if (!display || !context || !num_blocks || !block)
 		return BadValue;
 
@@ -505,7 +497,6 @@ _X_EXPORT Status XvMCCreateBlocks(Display * display, XvMCContext * context,
  */
 _X_EXPORT Status XvMCDestroyBlocks(Display * display, XvMCBlockArray * block)
 {
-	Status ret;
 	if (!display || !block)
 		return BadValue;
 
@@ -679,7 +670,6 @@ _X_EXPORT Status XvMCPutSurface(Display * display, XvMCSurface * surface,
 _X_EXPORT Status XvMCSyncSurface(Display * display, XvMCSurface * surface)
 {
 	Status ret;
-	int stat = 0;
 
 	if (!display || !surface)
 		return XvMCBadSurface;
@@ -739,9 +729,6 @@ _X_EXPORT Status XvMCGetSurfaceStatus(Display * display, XvMCSurface * surface,
  */
 _X_EXPORT Status XvMCHideSurface(Display * display, XvMCSurface * surface)
 {
-	int stat = 0;
-	Status ret;
-
 	if (!display || !surface)
 		return XvMCBadSurface;
 
diff --git a/src/xvmc/xvmc_vld.c b/src/xvmc/xvmc_vld.c
index addc7c9..d72e105 100644
--- a/src/xvmc/xvmc_vld.c
+++ b/src/xvmc/xvmc_vld.c
@@ -785,7 +785,6 @@ static Status begin_surface(Display * display, XvMCContext * context,
 			    XvMCSurface * future,
 			    const XvMCMpegControl * control)
 {
-	struct i965_xvmc_contex *i965_ctx;
 	struct intel_xvmc_surface *priv_target, *priv_past, *priv_future;
 	intel_xvmc_context_ptr intel_ctx = context->privData;
 	Status ret;
--
cgit v0.8.3-6-g21f6
