diff -Naur xf86-video-intel-2.0.0.orig/src/i830_memory.c xf86-video-intel-2.0.0/src/i830_memory.c
--- xf86-video-intel-2.0.0.orig/src/i830_memory.c	2007-05-19 15:35:36.000000000 +0200
+++ xf86-video-intel-2.0.0/src/i830_memory.c	2007-05-19 15:36:41.000000000 +0200
@@ -231,11 +231,13 @@
     pI830->exa_965_state = NULL;
     pI830->overlay_regs = NULL;
     pI830->logical_context = NULL;
+#ifdef XF86DRI
     pI830->back_buffer = NULL;
     pI830->third_buffer = NULL;
     pI830->depth_buffer = NULL;
     pI830->textures = NULL;
     pI830->memory_manager = NULL;
+#endif
     pI830->LpRing->mem = NULL;
 
     /* Reset the fence register allocation. */
@@ -246,6 +248,7 @@
 void
 i830_free_3d_memory(ScrnInfoPtr pScrn)
 {
+#ifdef XF86DRI
     I830Ptr pI830 = I830PTR(pScrn);
 
     i830_free_memory(pScrn, pI830->back_buffer);
@@ -258,6 +261,7 @@
     pI830->textures = NULL;
     i830_free_memory(pScrn, pI830->memory_manager);
     pI830->memory_manager = NULL;
+#endif
 }
 
 /**
@@ -598,6 +602,7 @@
 		   "%s0x%08lx:            end of aperture\n",
 		   prefix, pI830->FbMapSize);
 
+#ifdef XF86DRI
     if (pI830->front_buffer != NULL) {
 	i830_describe_tiling(pScrn, verbosity, prefix, pI830->front_buffer,
 			     pI830->front_tiled);
@@ -614,6 +619,7 @@
 	i830_describe_tiling(pScrn, verbosity, prefix, pI830->depth_buffer,
 			     pI830->depth_tiled);
     }
+#endif
 }
 
 static Bool
@@ -808,13 +814,17 @@
 	front_buffer = i830_allocate_memory_tiled(pScrn, name, size,
 						  pitch, align,
 						  0, TILING_XMAJOR);
+#ifdef XF86DRI
 	pI830->front_tiled = FENCE_XMAJOR;
+#endif
     }
 
     /* If not, attempt it linear */
     if (front_buffer == NULL) {
 	front_buffer = i830_allocate_memory(pScrn, name, size, KB(64), flags);
+#ifdef XF86DRI
 	pI830->front_tiled = FENCE_LINEAR;
+#endif
     }
 
     if (front_buffer == NULL) {
