diff -Naur xf86-video-intel-2.4.97.0/aclocal.m4.org xf86-video-intel-2.4.97.0/aclocal.m4
--- xf86-video-intel-2.4.97.0/aclocal.m4.org    2008-08-26 16:21:13.000000000 +0200
+++ xf86-video-intel-2.4.97.0/aclocal.m4        2008-08-26 16:22:05.000000000 +0200
@@ -6936,9 +6936,7 @@
 # with the AM_CONDITIONAL "BUILD_LINUXDOC"
 AC_DEFUN([XORG_CHECK_LINUXDOC],[
 XORG_SGML_PATH=$prefix/share/sgml
-HAVE_DEFS_ENT=
-
-AC_CHECK_FILE([$XORG_SGML_PATH/X11/defs.ent], [HAVE_DEFS_ENT=yes])
+HAVE_DEFS_ENT=no

 AC_PATH_PROG(LINUXDOC, linuxdoc)
 AC_PATH_PROG(PS2PDF, ps2pdf)
diff -Naur xf86-video-intel-2.1.0.orig/configure xf86-video-intel-2.1.0/configure
--- xf86-video-intel-2.1.0.orig/configure	2007-07-15 17:00:59.000000000 +0200
+++ xf86-video-intel-2.1.0/configure	2007-07-15 17:01:29.000000000 +0200
@@ -21733,7 +21733,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
