diff -Naur xf86-video-intel-2.2.0.orig/configure xf86-video-intel-2.2.0/configure
--- xf86-video-intel-2.2.0.orig/configure	2008-01-14 20:29:52.000000000 +0100
+++ xf86-video-intel-2.2.0/configure	2008-01-14 20:31:03.000000000 +0100
@@ -21173,7 +21173,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
@@ -21198,7 +21198,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
