diff -Naur xf86-video-intel-2.4.0.orig/src/common.h xf86-video-intel-2.4.0/src/common.h
--- xf86-video-intel-2.4.0.orig/src/common.h	2008-07-23 22:21:26.000000000 +0200
+++ xf86-video-intel-2.4.0/src/common.h	2008-07-23 22:22:07.000000000 +0200
@@ -375,6 +375,8 @@
 #define HWS_NEED_NONSTOLEN(pI810) (IS_GM45(pI810) || IS_G4X(pI810))
 #define SUPPORTS_INTEGRATED_HDMI(pI810) (IS_GM45(pI810) || IS_G4X(pI810))
 
+#define HW_HAS_OVERLAY(pI810) (1)
+
 #define GTT_PAGE_SIZE			KB(4)
 #define ROUND_TO(x, y)			(((x) + (y) - 1) / (y) * (y))
 #define ROUND_DOWN_TO(x, y)		((x) / (y) * (y))
diff -Naur xf86-video-intel-2.4.0.orig/src/i830_driver.c xf86-video-intel-2.4.0/src/i830_driver.c
--- xf86-video-intel-2.4.0.orig/src/i830_driver.c	2008-07-23 22:21:26.000000000 +0200
+++ xf86-video-intel-2.4.0/src/i830_driver.c	2008-07-23 22:22:07.000000000 +0200
@@ -317,6 +317,7 @@
 #ifdef INTEL_XVMC
    OPTION_XVMC,
 #endif
+   OPTION_TEXTURED_XV,
 } I830Opts;
 
 static OptionInfoRec I830Options[] = {
@@ -345,6 +346,7 @@
 #ifdef INTEL_XVMC
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
+   {OPTION_TEXTURED_XV,	"TexturedVideo",OPTV_BOOLEAN,	{0},	FALSE},
    {-1,			NULL,		OPTV_NONE,	{0},	FALSE}
 };
 /* *INDENT-ON* */
@@ -1680,6 +1682,9 @@
    pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
+   pI830->TexturedXvEnabled =
+     xf86ReturnOptValBool(pI830->Options, OPTION_TEXTURED_XV, FALSE);
+   
 #ifdef I830_XV
    if (xf86GetOptValInteger(pI830->Options, OPTION_VIDEO_KEY,
 			    &(pI830->colorKey))) {
diff -Naur xf86-video-intel-2.4.0.orig/src/i830.h xf86-video-intel-2.4.0/src/i830.h
--- xf86-video-intel-2.4.0.orig/src/i830.h	2008-07-23 22:21:26.000000000 +0200
+++ xf86-video-intel-2.4.0/src/i830.h	2008-07-23 22:22:07.000000000 +0200
@@ -515,7 +515,8 @@
 
    Bool XvDisabled;			/* Xv disabled in PreInit. */
    Bool XvEnabled;			/* Xv enabled for this generation. */
-
+   Bool TexturedXvEnabled;              /* Use Textured video for Xv ? */
+  
 #ifdef I830_XV
    int colorKey;
    XF86VideoAdaptorPtr adaptor;
diff -Naur xf86-video-intel-2.4.0.orig/src/i830_video.c xf86-video-intel-2.4.0/src/i830_video.c
--- xf86-video-intel-2.4.0.orig/src/i830_video.c	2008-07-23 22:21:26.000000000 +0200
+++ xf86-video-intel-2.4.0/src/i830_video.c	2008-07-23 22:24:33.000000000 +0200
@@ -606,7 +606,7 @@
     /* Set up textured video if we can do it at this depth and we are on
      * supported hardware.
      */
-    if (pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
+    if (pI830->TexturedXvEnabled && pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
 	!(!IS_I965G(pI830) && pScrn->displayWidth > 2048))
     {
 	texturedAdaptor = I830SetupImageVideoTextured(pScreen);
@@ -620,7 +620,7 @@
     }
 
     /* Set up overlay video if we can do it at this depth. */
-    if (!OVERLAY_NOEXIST(pI830) && pScrn->bitsPerPixel != 8 &&
+    if (!OVERLAY_NOEXIST(pI830) && HW_HAS_OVERLAY(pI830) && pScrn->bitsPerPixel != 8 &&
 	    pI830->overlay_regs != NULL)
     {
 	overlayAdaptor = I830SetupImageVideoOverlay(pScreen);
@@ -2616,7 +2616,7 @@
     I830PortPrivPtr pPriv;
 
     /* no overlay */
-    if (pI830->adaptor == NULL)
+    if (!HW_HAS_OVERLAY(pI830) || pI830->adaptor == NULL)
         return;
 
     pPriv = GET_PORT_PRIVATE(pScrn);
@@ -2896,7 +2896,7 @@
     I830PortPrivPtr pPriv;
 
     /* no overlay */
-    if (pI830->adaptor == NULL)
+    if (!HW_HAS_OVERLAY(pI830) || pI830->adaptor == NULL)
 	return;
 
     pPriv = GET_PORT_PRIVATE(pScrn);
