diff -Naur xf86-video-intel-2.5.0/src/i830_driver.c xf86-video-intel-2.5.0.edit/src/i830_driver.c
--- xf86-video-intel-2.5.0/src/i830_driver.c	2008-10-20 23:02:38.000000000 +0200
+++ xf86-video-intel-2.5.0.edit/src/i830_driver.c	2008-10-25 17:53:33.000000000 +0200
@@ -310,6 +310,7 @@
    OPTION_LVDSFIXEDMODE,
    OPTION_TRIPLEBUFFER,
    OPTION_FORCEENABLEPIPEA,
+   OPTION_TEXTURED_VIDEO,
 #ifdef INTEL_XVMC
    OPTION_XVMC,
 #endif
@@ -337,6 +338,7 @@
    {OPTION_LVDSFIXEDMODE, "LVDSFixedMode", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_TRIPLEBUFFER, "TripleBuffer", OPTV_BOOLEAN,	{0},	FALSE},
    {OPTION_FORCEENABLEPIPEA, "ForceEnablePipeA", OPTV_BOOLEAN,	{0},	FALSE},
+   {OPTION_TEXTURED_VIDEO, "TexturedVideo", OPTV_BOOLEAN, {0}, FALSE},
 #ifdef INTEL_XVMC
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
@@ -1689,6 +1691,8 @@
     pI830->XvDisabled =
 	!xf86ReturnOptValBool(pI830->Options, OPTION_XVIDEO, TRUE);
 
+    pI830->TexturedXvEnabled = OVERLAY_NOEXIST(pI830) || xf86ReturnOptValBool(pI830->Options, OPTION_TEXTURED_VIDEO, FALSE);
+    
 #ifdef I830_XV
     if (xf86GetOptValInteger(pI830->Options, OPTION_VIDEO_KEY,
 			     &(pI830->colorKey))) {
diff -Naur xf86-video-intel-2.5.0/src/i830.h xf86-video-intel-2.5.0.edit/src/i830.h
--- xf86-video-intel-2.5.0/src/i830.h	2008-10-25 17:54:04.000000000 +0200
+++ xf86-video-intel-2.5.0.edit/src/i830.h	2008-10-25 17:53:33.000000000 +0200
@@ -556,7 +556,8 @@
 
    Bool XvDisabled;			/* Xv disabled in PreInit. */
    Bool XvEnabled;			/* Xv enabled for this generation. */
-
+   Bool TexturedXvEnabled;		/* Textured video enabled/disabled by user*/
+  
 #ifdef I830_XV
    int colorKey;
    XF86VideoAdaptorPtr adaptor;
diff -Naur xf86-video-intel-2.5.0/src/i830_video.c xf86-video-intel-2.5.0.edit/src/i830_video.c
--- xf86-video-intel-2.5.0/src/i830_video.c	2008-10-19 01:13:06.000000000 +0200
+++ xf86-video-intel-2.5.0.edit/src/i830_video.c	2008-10-25 17:53:33.000000000 +0200
@@ -606,7 +606,7 @@
     /* Set up textured video if we can do it at this depth and we are on
      * supported hardware.
      */
-    if (pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
+    if (pI830->TexturedXvEnabled && pScrn->bitsPerPixel >= 16 && (IS_I9XX(pI830) || IS_I965G(pI830)) &&
 	!(!IS_I965G(pI830) && pScrn->displayWidth > 2048))
     {
 	texturedAdaptor = I830SetupImageVideoTextured(pScreen);
