diff -Naur xf86-video-intel-2.5.1/src/i830_driver.c.org xf86-video-intel-2.5.1/src/i830_driver.c > /export/60_overlay.diff
--- xf86-video-intel-2.5.1/src/i830_driver.c.org	2008-11-14 20:16:27.000000000 +0100
+++ xf86-video-intel-2.5.1/src/i830_driver.c	2008-11-14 20:16:59.000000000 +0100
@@ -344,7 +344,7 @@
    {OPTION_XVMC,	"XvMC",		OPTV_BOOLEAN,	{0},	TRUE},
 #endif
    {OPTION_FORCE_SDVO_DETECT, "ForceSDVODetect", OPTV_BOOLEAN,  {0},	FALSE},
-   {OPTION_PREFER_OVERLAY, "XvPreferOverlay", OPTV_BOOLEAN, {0}, FALSE},
+   {OPTION_PREFER_OVERLAY, "XvPreferOverlay", OPTV_BOOLEAN, {0}, TRUE},
    {-1,			NULL,		OPTV_NONE,	{0},	FALSE}
 };
 /* *INDENT-ON* */
