--- xf86-video-intel-2.7.99.901/src/i830_driver.c.old	2009-06-11 21:31:18.000000000 +0200
+++ xf86-video-intel-2.7.99.901/src/i830_driver.c	2009-06-11 21:33:57.000000000 +0200
@@ -1353,13 +1353,9 @@
     MessageType from = X_PROBED;
     int i, num_pipe;
 
-    if (xf86ReturnOptValBool(pI830->Options, OPTION_NOACCEL, FALSE)) {
-	pI830->accel = ACCEL_NONE;
-    } else {
-       pI830->accel = ACCEL_UXA;
-	xf86DrvMsg(pScrn->scrnIndex, from, "Using %s for acceleration\n",
+    pI830->accel = ACCEL_NONE;
+    xf86DrvMsg(pScrn->scrnIndex, from, "Using %s for acceleration\n",
 		   accel_name[pI830->accel]);
-    }
 
     pI830->directRenderingType = DRI_NONE;
     if (!xf86ReturnOptValBool(pI830->Options, OPTION_DRI, TRUE))
