--- xf86-video-intel-2.6.0/src/i830_driver.c.org	2009-01-15 21:27:53.000000000 +0100
+++ xf86-video-intel-2.6.0/src/i830_driver.c	2009-01-15 21:30:03.000000000 +0100
@@ -1705,10 +1705,8 @@
     xf86DrvMsg(pScrn->scrnIndex, from, "Will%s try to enable page flipping\n",
 	       pI830->allowPageFlip ? "" : " not");
 
-    pI830->TripleBuffer = FALSE;
-    from =  (pI830->directRenderingType != DRI_DISABLED &&
-	     xf86GetOptValBool(pI830->Options, OPTION_TRIPLEBUFFER,
-			       &pI830->TripleBuffer)) ? X_CONFIG : X_DEFAULT;
+    pI830->TripleBuffer = TRUE;
+    from =  X_CONFIG;
 
     xf86DrvMsg(pScrn->scrnIndex, from, "Triple buffering %sabled\n",
 	       pI830->TripleBuffer ? "en" : "dis");
