--- xf86-video-intel-2.5.1/src/i830_driver.c.org	2008-11-15 17:56:04.000000000 +0100
+++ xf86-video-intel-2.5.1/src/i830_driver.c	2008-11-15 17:59:29.000000000 +0100
@@ -1740,10 +1740,8 @@
     xf86DrvMsg(pScrn->scrnIndex, from, "Will%s try to enable page flipping\n",
 	       pI830->allowPageFlip ? "" : " not");
 
-    pI830->TripleBuffer = FALSE;
-    from =  (!pI830->directRenderingDisabled &&
-	     xf86GetOptValBool(pI830->Options, OPTION_TRIPLEBUFFER,
-			       &pI830->TripleBuffer)) ? X_CONFIG : X_DEFAULT;
+    pI830->TripleBuffer = TRUE;
+    from =  (!pI830->directRenderingDisabled) ? X_CONFIG : X_DEFAULT;
 
     xf86DrvMsg(pScrn->scrnIndex, from, "Triple buffering %sabled\n",
 	       pI830->TripleBuffer ? "en" : "dis");
