diff -Naur xf86-video-intel-2.9.1/uxa/uxa-render.c xf86-video-intel-2.9.1a/uxa/uxa-render.c
--- xf86-video-intel-2.9.1/uxa/uxa-render.c	2009-10-26 12:48:05.000000000 +0100
+++ xf86-video-intel-2.9.1a/uxa/uxa-render.c	2009-11-28 15:05:18.000000000 +0100
@@ -438,11 +438,7 @@
     return pPicture;
 }
 
-#if XORG_VERSION_CURRENT < XORG_VERSION_NUMERIC (1,6,99,1,0)
 #define IMAGE_FROM_PICT(P) image_from_pict(P, 0)
-#else
-#define IMAGE_FROM_PICT(P) image_from_pict(P, 0, 0)
-#endif
 
 static PicturePtr
 uxa_acquire_pattern (ScreenPtr pScreen,
