diff -Naur xf86-input-mouse-1.2.1.orig/configure xf86-input-mouse-1.2.1/configure
--- xf86-input-mouse-1.2.1.orig/configure	2007-01-30 20:33:47.000000000 +0100
+++ xf86-input-mouse-1.2.1/configure	2007-01-30 20:33:55.000000000 +0100
@@ -21327,7 +21327,7 @@
 if { as_var=$as_ac_File; eval "test \"\${$as_var+set}\" = set"; }; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  test "$cross_compiling" = yes &&
+  test "$cross_compiling" = dummy &&
   { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
 echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
    { (exit 1); exit 1; }; }
