diff -Naur bst-kconfig-2.6.34+git20100608/Makefile bst-kconfig-2.6.34+git20100608a//Makefile
--- bst-kconfig-2.6.34+git20100608/Makefile	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a//Makefile	2010-06-29 12:07:48.954317006 +0200
@@ -37,7 +37,7 @@
 
 export HOSTCC HOSTCXX HOSTCFLAGS HOSTCXXFLAGS
 
-KBUILD_KCONFIG = $(CURDIR)/fm/main.fm
+KBUILD_KCONFIG = $(CURDIR)/../../../config/Kconfig.main
 export KBUILD_KCONFIG
 
 srctree		:= $(CURDIR)
