diff -Naur bst-kconfig-2.6.34+git20100608/scripts/kconfig/gconf.c bst-kconfig-2.6.34+git20100608a/scripts/kconfig/gconf.c
--- bst-kconfig-2.6.34+git20100608/scripts/kconfig/gconf.c	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a/scripts/kconfig/gconf.c	2010-06-29 12:11:33.402307051 +0200
@@ -266,7 +266,7 @@
 					  /*"style", PANGO_STYLE_OBLIQUE, */
 					  NULL);
 
-	sprintf(title, _("Linux Kernel v%s Configuration"),
+	sprintf(title, _("GeeXboX v%s Configuration"),
 		getenv("KERNELVERSION"));
 	gtk_window_set_title(GTK_WINDOW(main_wnd), title);
 
@@ -718,8 +718,8 @@
 {
 	GtkWidget *dialog;
 	const gchar *intro_text = _(
-	    "Welcome to gkc, the GTK+ graphical kernel configuration tool\n"
-	    "for Linux.\n"
+	    "Welcome to gkc, the GTK+ graphical configuration tool\n"
+	    "for GeeXboX.\n"
 	    "For each option, a blank box indicates the feature is disabled, a\n"
 	    "check indicates it is enabled, and a dot indicates that it is to\n"
 	    "be compiled as a module.  Clicking on the box will cycle through the three states.\n"
diff -Naur bst-kconfig-2.6.34+git20100608/scripts/kconfig/mconf.c bst-kconfig-2.6.34+git20100608a/scripts/kconfig/mconf.c
--- bst-kconfig-2.6.34+git20100608/scripts/kconfig/mconf.c	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a/scripts/kconfig/mconf.c	2010-06-29 12:12:42.167308241 +0200
@@ -25,10 +25,10 @@
 static const char mconf_readme[] = N_(
 "Overview\n"
 "--------\n"
-"Some kernel features may be built directly into the kernel.\n"
-"Some may be made into loadable runtime modules.  Some features\n"
+"Some GeeXboX features may be built directly into the rootfs.\n"
+"Some may be made into installable opk packages.  Some features\n"
 "may be completely removed altogether.  There are also certain\n"
-"kernel parameters which are not really features, but must be\n"
+"GeeXboX parameters which are not really features, but must be\n"
 "entered in as decimal or hexadecimal numbers or possibly text.\n"
 "\n"
 "Menu items beginning with following braces represent features that\n"
@@ -37,10 +37,10 @@
 "  { } can be built in or modularized (selected by other feature)\n"
 "  - - are selected by other feature,\n"
 "while *, M or whitespace inside braces means to build in, build as\n"
-"a module or to exclude the feature respectively.\n"
+"a package or to exclude the feature respectively.\n"
 "\n"
 "To change any of these features, highlight it with the cursor\n"
-"keys and press <Y> to build it in, <M> to make it a module or\n"
+"keys and press <Y> to build it in, <M> to make it a package or\n"
 "<N> to removed it.  You may also press the <Space Bar> to cycle\n"
 "through the available options (ie. Y->N->M->Y).\n"
 "\n"
@@ -305,7 +305,7 @@
 	sym = sym_lookup("KERNELVERSION", 0);
 	sym_calc_value(sym);
 	size = snprintf(menu_backtitle, sizeof(menu_backtitle),
-	                _("%s - Linux Kernel v%s Configuration"),
+	                _("%s - GeeXboX v%s Configuration"),
 		        config_filename, sym_get_string_value(sym));
 	if (size >= sizeof(menu_backtitle))
 		menu_backtitle[sizeof(menu_backtitle)-1] = '\0';
@@ -851,7 +851,7 @@
 		}
 	case -1:
 		printf(_("\n\n"
-			"*** End of Linux kernel configuration.\n"
+			"*** End of GeeXboX configuration.\n"
 			"*** Execute 'make' to build the kernel or try 'make help'."
 			"\n\n"));
 		break;
diff -Naur bst-kconfig-2.6.34+git20100608/scripts/kconfig/qconf.cc bst-kconfig-2.6.34+git20100608a/scripts/kconfig/qconf.cc
--- bst-kconfig-2.6.34+git20100608/scripts/kconfig/qconf.cc	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a/scripts/kconfig/qconf.cc	2010-06-29 12:13:26.611310155 +0200
@@ -1275,7 +1275,7 @@
 	char title[256];
 
 	QDesktopWidget *d = configApp->desktop();
-	snprintf(title, sizeof(title), _("Linux Kernel v%s Configuration"),
+	snprintf(title, sizeof(title), _("GeeXboX v%s Configuration"),
 		getenv("KERNELVERSION"));
 	setCaption(title);
 
@@ -1631,10 +1631,10 @@
 
 void ConfigMainWindow::showIntro(void)
 {
-	static const QString str = _("Welcome to the qconf graphical kernel configuration tool for Linux.\n\n"
+	static const QString str = _("Welcome to the qconf graphical kernel configuration tool for GeeXboX.\n\n"
 		"For each option, a blank box indicates the feature is disabled, a check\n"
 		"indicates it is enabled, and a dot indicates that it is to be compiled\n"
-		"as a module.  Clicking on the box will cycle through the three states.\n\n"
+		"as a package.  Clicking on the box will cycle through the three states.\n\n"
 		"If you do not see an option (e.g., a device driver) that you believe\n"
 		"should be present, try turning on Show All Options under the Options menu.\n"
 		"Although there is no cross reference yet to help you figure out what other\n"
diff -Naur bst-kconfig-2.6.34+git20100608/scripts/kconfig/zconf.tab.c_shipped bst-kconfig-2.6.34+git20100608a/scripts/kconfig/zconf.tab.c_shipped
--- bst-kconfig-2.6.34+git20100608/scripts/kconfig/zconf.tab.c_shipped	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a/scripts/kconfig/zconf.tab.c_shipped	2010-06-29 12:09:38.002307937 +0200
@@ -2224,7 +2224,7 @@
 	modules_sym = sym_lookup(NULL, 0);
 	modules_sym->type = S_BOOLEAN;
 	modules_sym->flags |= SYMBOL_AUTO;
-	rootmenu.prompt = menu_add_prompt(P_MENU, "Linux Kernel Configuration", NULL);
+	rootmenu.prompt = menu_add_prompt(P_MENU, "GeeXboX Configuration", NULL);
 
 #if YYDEBUG
 	if (getenv("ZCONF_DEBUG"))
diff -Naur bst-kconfig-2.6.34+git20100608/scripts/kconfig/zconf.y bst-kconfig-2.6.34+git20100608a/scripts/kconfig/zconf.y
--- bst-kconfig-2.6.34+git20100608/scripts/kconfig/zconf.y	2010-06-22 12:52:53.000000000 +0200
+++ bst-kconfig-2.6.34+git20100608a/scripts/kconfig/zconf.y	2010-06-29 12:09:17.538306523 +0200
@@ -479,7 +479,7 @@
 	modules_sym = sym_lookup(NULL, 0);
 	modules_sym->type = S_BOOLEAN;
 	modules_sym->flags |= SYMBOL_AUTO;
-	rootmenu.prompt = menu_add_prompt(P_MENU, "Linux Kernel Configuration", NULL);
+	rootmenu.prompt = menu_add_prompt(P_MENU, "GeeXboX Configuration", NULL);
 
 #if YYDEBUG
 	if (getenv("ZCONF_DEBUG"))
