diff -NrU5 cxxtools-1.4.7.orig/src/tcpstream.cpp cxxtools-1.4.7/src/tcpstream.cpp
--- cxxtools-1.4.7.orig/src/tcpstream.cpp	2008-03-09 17:26:59.000000000 +0100
+++ cxxtools-1.4.7/src/tcpstream.cpp	2008-04-28 02:10:52.000000000 +0200
@@ -23,10 +23,11 @@
 #include <sys/poll.h>
 #include <unistd.h>
 #include <netdb.h>
 #include <errno.h>
 #include <string.h>
+#include <cstring>
 #include "cxxtools/log.h"
 
 log_define("cxxtools.net.tcp")
 
 namespace cxxtools
