--- png++-0.2.5/Makefile.org	2014-10-31 16:21:57.924434848 +0100
+++ png++-0.2.5/Makefile	2014-10-31 16:22:20.760434845 +0100
@@ -110,7 +110,7 @@
 test-compile-headers: *.hpp
 	for i in *.hpp; do \
 		echo '#include "'$$i'"' >$$i.cpp \
-		&& g++ -c $$i.cpp $(make_cflags) `$(LIBPNG_CONFIG) --cflags`; \
+		&& $(CXX) -c $$i.cpp $(make_cflags) `$(LIBPNG_CONFIG) --cflags`; \
 	done
 	rm -f *.hpp.o *.hpp.cpp
 
--- png++-0.2.5/example/Makefile.org	2014-10-31 16:22:27.816434844 +0100
+++ png++-0.2.5/example/Makefile	2014-10-31 16:23:30.584434837 +0100
@@ -66,14 +66,14 @@
 .PHONY: all dist-copy-files clean clean-deps
 
 %$(bin_suffix): %.o
-	g++ -o $@ $< $(make_ldflags) `$(LIBPNG_CONFIG) --ldflags`
+	$(CXX) -o $@ $< $(make_ldflags) `$(LIBPNG_CONFIG) --ldflags`
 
 %.o: %.cpp
-	g++ -c -o $@ $< $(make_cflags) `$(LIBPNG_CONFIG) --cflags`
+	$(CXX) -c -o $@ $< $(make_cflags) `$(LIBPNG_CONFIG) --cflags`
 
 
 %.dep: %.cpp
-	g++ -M $(CPPFLAGS) $(make_cflags) $< -o- | \
+	$(CXX) -M $(CPPFLAGS) $(make_cflags) $< -o- | \
 	  sed 's/\($*\)\.o[ :]*/\1.o $@ : /g' > $@
 
 clean-deps:
--- a/test/Makefile	2010-10-26 09:00:28.000000000 +0200
+++ b/test/Makefile	2014-11-08 20:13:16.883299962 +0100
@@ -85,13 +85,13 @@
   clean-deps
 
 %$(bin_suffix): %.o
-	g++ -o $@ $< $(make_ldflags) `$(LIBPNG_CONFIG) --ldflags`
+	$(CXX) -o $@ $< $(make_ldflags) `$(LIBPNG_CONFIG) --ldflags`
 
 %.o: %.cpp
-	g++ -c -o $@ $< $(make_cflags) `$(LIBPNG_CONFIG) --cflags`
+	$(CXX) -c -o $@ $< $(make_cflags) `$(LIBPNG_CONFIG) --cflags`
 
 %.dep: %.cpp
-	g++ -M $(CPPFLAGS) $(make_cflags) $< -o- | \
+	$(CXX) -M $(CPPFLAGS) $(make_cflags) $< -o- | \
 	  sed 's/\($*\)\.o[ :]*/\1.o $@ : /g' > $@
 
 clean-deps:
