diff --git a/video.c b/video.c
index 40b6343..35c57f8 100644
--- a/video.c
+++ b/video.c
@@ -493,6 +493,9 @@ static void VideoUpdateOutput(AVRational input_aspect_ratio, int input_width,
     // FIXME: store different positions for the ratios
     tmp_ratio.num = 4;
     tmp_ratio.den = 3;
+    fprintf(stderr, "ratio: %d:%d %d:%d\n",
+	input_aspect_ratio.num, input_aspect_ratio.den,
+	display_aspect_ratio.num, display_aspect_ratio.den);
     if (!av_cmp_q(input_aspect_ratio, tmp_ratio)) {
 	switch (Video4to3ZoomMode) {
 	    case VideoNormal:
@@ -4626,12 +4629,12 @@ static void VaapiSyncDecoder(VaapiDecoder * decoder)
 	    ++decoder->FramesDuped;
 	    decoder->SyncCounter = 1;
 	    goto out;
-	} else if (video_clock > audio_clock + VideoAudioDelay + 45 * 90) {
+	} else if (video_clock > audio_clock + VideoAudioDelay + 65 * 90) {
 	    err = VaapiMessage(3, "video: slow down video, duping frame\n");
 	    ++decoder->FramesDuped;
 	    decoder->SyncCounter = 1;
 	    goto out;
-	} else if (audio_clock + VideoAudioDelay > video_clock + 15 * 90
+	} else if (audio_clock + VideoAudioDelay > video_clock + 35 * 90
 	    && filled > 1 + 2 * decoder->Interlaced) {
 	    err = VaapiMessage(3, "video: speed up video, droping frame\n");
 	    ++decoder->FramesDropped;
@@ -8885,7 +8888,7 @@ void VideoSetOsdSize(int width, int height)
 ///
 ///	Set the 3d OSD mode.
 ///
-///	@pram mode	OSD mode (0=off, 1=SBS, 2=Top Bottom)
+///	@param mode	OSD mode (0=off, 1=SBS, 2=Top Bottom)
 ///
 void VideoSetOsd3DMode(int mode)
 {
