diff -Naur softhddevice-9999_orig/softhddevice.cpp softhddevice-9999/softhddevice.cpp
--- softhddevice-9999_orig/softhddevice.cpp	2012-11-26 23:46:46.000000000 +0100
+++ softhddevice-9999/softhddevice.cpp	2012-11-26 23:49:23.422687612 +0100
@@ -316,13 +316,15 @@
 #ifdef USE_YAEPG
     // support yaepghd, video window
     if (vidWin.bpp) {			// restore fullsized video
-	int width;
-	int height;
-	double video_aspect;
+	// on destroying the OSD, we should _correctly_ scale to available window, at this time with the help of external calculation
+#ifdef OSD_DEBUG
+	dsyslog("[softhddev]%s: %dx%d+%d+%d\n", __FUNCTION__, vidWin.Width(),
+	    vidWin.Height(), vidWin.x1, vidWin.y2);
+#endif
 
-	::GetOsdSize(&width, &height, &video_aspect);
-	// works osd relative
-	VideoSetOutputPosition(0, 0, width, height);
+	// FIXME: vidWin is OSD relative not video window.
+	VideoSetOutputPosition(Left() + vidWin.x1, Top() + vidWin.y1,
+	    vidWin.Width(), vidWin.Height());
     }
 #endif
 }
