diff -aur --unidirectional-new-file a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am	2012-12-26 17:07:50.000000000 +0100
+++ b/src/Makefile.am	2013-06-03 19:20:29.039005371 +0200
@@ -6,14 +6,13 @@
 # copies and copies may only be made to the extent permitted
 # by a licensing agreement from ARM Limited.
 
-MALI_DDK="/work/trunk"
+MALI_DDK="/work/trunk"
 
 mali_drv_la_LTLIBRARIES = mali_drv.la
 mali_drv_la_LDFLAGS = -module -avoid-version -L$(MALI_DDK)/lib -lpixman-1 -lUMP -lpthread
 mali_drv_ladir = @moduledir@/drivers
 
 AM_CFLAGS = @XORG_CFLAGS@ \
-	-I/usr/include/libdrm \
 	-I$(MALI_DDK)/include \
 	-I$(MALI_DDK)/internal/include/khronos \
 	-I$(MALI_DDK)/src/ump/include \
