From 64a1b4b96389ffa546980df34fa320a20648a569 Mon Sep 17 00:00:00 2001
From: Rene van Dorst <opensource@vdorst.com>
Date: Sun, 13 Jan 2013 12:22:12 +0000
Subject: [PATCH] MTD: devices: Add support for serial flash Micron N25Q032

---
 drivers/mtd/devices/m25p80.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index 5d0d68c..652831b 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -659,6 +659,9 @@ static const struct spi_device_id m25p_ids[] = {
 	{ "mx25l25635e", INFO(0xc22019, 0, 64 * 1024, 512, 0) },
 	{ "mx25l25655e", INFO(0xc22619, 0, 64 * 1024, 512, 0) },
 
+	/* Micron */
+	{ "n25q032", INFO(0x20ba16,  0,  64 * 1024,  64, 0) },
+
 	/* Spansion -- single (large) sector size only, at least
 	 * for the chips listed here (without boot sectors).
 	 */
-- 
1.7.9.5

