diff -Naur linux-2.6.35.orig/Makefile linux-2.6.35/Makefile
--- linux-2.6.35.orig/Makefile	2010-08-05 20:02:32.838987735 +0200
+++ linux-2.6.35/Makefile	2010-08-06 20:31:18.066487421 +0200
@@ -886,7 +886,7 @@
 # Store (new) KERNELRELASE string in include/config/kernel.release
 include/config/kernel.release: include/config/auto.conf FORCE
 	$(Q)rm -f $@
-	$(Q)echo "$(KERNELVERSION)$$($(CONFIG_SHELL) $(srctree)/scripts/setlocalversion $(srctree))" > $@
+	$(Q)echo "$(KERNELVERSION)" > $@
 
 
 # Things we need to do before we recursively start building the kernel
