From e9e34d9ffa3bc2b823d782ac38b094f090c0829f Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Tue, 6 Aug 2013 18:27:15 +0200
Subject: [PATCH] HDMITX: Fix color space conversion bug

---
 .../comps/tmbslTDA9989/src/tmbslTDA9989_InOut.c    |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/dovefb/nxp_hdmi/comps/tmbslTDA9989/src/tmbslTDA9989_InOut.c b/drivers/video/dovefb/nxp_hdmi/comps/tmbslTDA9989/src/tmbslTDA9989_InOut.c
index 6b2334c..e334ed3 100755
--- a/drivers/video/dovefb/nxp_hdmi/comps/tmbslTDA9989/src/tmbslTDA9989_InOut.c
+++ b/drivers/video/dovefb/nxp_hdmi/comps/tmbslTDA9989/src/tmbslTDA9989_InOut.c
@@ -3391,7 +3391,7 @@ tmbslTDA9989MatrixSetConversion
     
             if(vinMode == HDMITX_VINMODE_RGB444)    /* RGB */
             {
-                cspace_in = HDMITX_CS_RGB_LIMITED;
+                cspace_in = HDMITX_CS_RGB_FULL;
             }
             else                                    /* CCIR656, YUV444, YU422 */
             {
@@ -3401,7 +3401,7 @@ tmbslTDA9989MatrixSetConversion
         default:    /* Now all the SD modes */
             if(vinMode == HDMITX_VINMODE_RGB444)    /* we're RGB */
             {
-                cspace_in = HDMITX_CS_RGB_LIMITED;
+                cspace_in = HDMITX_CS_RGB_FULL;
             }
             else                                    /* CCIR656, YUV444, YU422 */
             {
-- 
1.7.9.5

