--- a/drivers/video/dovefb/dovefb_ovly.c	2012-08-25 12:42:19.000000000 +0200
+++ b/drivers/video/dovefb/dovefb_ovly.c	2012-12-08 21:47:49.864077905 +0100
@@ -284,7 +284,9 @@
 	    (dfli->surface.viewPortInfo.srcWidth != new_info->srcWidth ||
 	    dfli->surface.viewPortInfo.srcHeight != new_info->srcHeight ||
 	    dfli->surface.viewPortInfo.zoomXSize != new_info->zoomXSize ||
-	    dfli->surface.viewPortInfo.zoomYSize != new_info->zoomYSize)) {
+	    dfli->surface.viewPortInfo.zoomYSize != new_info->zoomYSize ||
+	    dfli->surface.viewPortInfo.ycPitch   != new_info->ycPitch   ||
+	    dfli->surface.viewPortInfo.uvPitch   != new_info->uvPitch)) {
 		var->xres = new_info->srcWidth;
 		var->yres = new_info->srcHeight;
 		var->xres_virtual = new_info->srcWidth;
@@ -1250,6 +1252,8 @@
 	 * Fill in sane defaults.
 	 */
 	dovefb_set_mode(dfli, &fi->var, dmi->modes, dmi->pix_fmt, 0);
+	dfli->surface.viewPortInfo.ycPitch = (fi->var.xres * fi->var.bits_per_pixel) / 8;
+	dfli->surface.viewPortInfo.uvPitch = dfli->surface.viewPortInfo.ycPitch / 2;
 	dovefb_ovly_set_par(fi);
 
 	/*
