diff --git a/drivers/video/dovefb/dovefb_gfx.c b/drivers/video/dovefb/dovefb_gfx.c
index 30bb811..04c645f 100644
--- a/drivers/video/dovefb/dovefb_gfx.c
+++ b/drivers/video/dovefb/dovefb_gfx.c
@@ -147,7 +147,7 @@ static void set_clock_divider(struct dovefb_layer_info *dfli,
 	 * 1. Set up reference clock speed.
 	 */
 
-	printk("%s : setup reference clk to %u\n", __FUNCTION__, needed_pixclk);
+	printk(KERN_DEBUG "%s : setup reference clk to %u\n", __func__, needed_pixclk);
 
 	clk_set_rate(info->clk, needed_pixclk);
 	/*
@@ -628,9 +628,9 @@ static int dovefb_gfx_set_par(struct fb_info *fi)
 	if ((x & 0x1) == 0)
 		writel(x | 1, dfli->reg_base + LCD_SPU_DUMB_CTRL);
 #ifdef CONFIG_TDA19988
-	printk(KERN_INFO "Setting HDMI TX resolution to %dx%d%c @ %d\n",
-		m->xres, m->yres, (m->vmode & FB_VMODE_INTERLACED) ? 'i' : 'p',
-		m->refresh);
+	printk(KERN_DEBUG "%s: Setting HDMI TX resolution to %dx%d%c @ %d\n",
+	       __func__,m->xres, m->yres, 
+	       (m->vmode & FB_VMODE_INTERLACED) ? 'i' : 'p', m->refresh);
 	if (!tda19988_configure_tx_inout(m->xres, m->yres,
 		(m->vmode & FB_VMODE_INTERLACED) ? 1 : 0, m->refresh))
 		printk(KERN_ERR "Setting HDMI TX mode failed\n");
