diff --git a/sound/soc/kirkwood/kirkwood-dma.c b/sound/soc/kirkwood/kirkwood-dma.c
index 690260b..06f7cd3 100644
--- a/sound/soc/kirkwood/kirkwood-dma.c
+++ b/sound/soc/kirkwood/kirkwood-dma.c
@@ -53,7 +53,7 @@ static struct snd_pcm_hardware kirkwood_dma_snd_hw = {
 	.rate_max		= 384000,
 	.channels_min		= 1,
 	.channels_max		= 8,
-	.buffer_bytes_max	= KIRKWOOD_SND_MAX_PERIOD_BYTES * KIRKWOOD_SND_MAX_PERIODS,
+	.buffer_bytes_max	= KIRKWOOD_SND_MAX_BUFFER_BYTES,
 	.period_bytes_min	= KIRKWOOD_SND_MIN_PERIOD_BYTES,
 	.period_bytes_max	= KIRKWOOD_SND_MAX_PERIOD_BYTES,
 	.periods_min		= KIRKWOOD_SND_MIN_PERIODS,
diff --git a/sound/soc/kirkwood/kirkwood.h b/sound/soc/kirkwood/kirkwood.h
index c30899b..f7c4b6d 100644
--- a/sound/soc/kirkwood/kirkwood.h
+++ b/sound/soc/kirkwood/kirkwood.h
@@ -151,10 +151,11 @@
 
 /* Theses values come from the marvell alsa driver */
 /* need to find where they come from               */
+#define KIRKWOOD_SND_MAX_BUFFER_BYTES		0x40000		// 262144
+#define KIRKWOOD_SND_MAX_PERIOD_BYTES		0x8000		// 32768
+#define KIRKWOOD_SND_MIN_PERIOD_BYTES		0x800		// 512
+#define KIRKWOOD_SND_MAX_PERIODS		(KIRKWOOD_SND_MAX_BUFFER_BYTES/KIRKWOOD_SND_MIN_PERIOD_BYTES)
 #define KIRKWOOD_SND_MIN_PERIODS		8
-#define KIRKWOOD_SND_MAX_PERIODS		16
-#define KIRKWOOD_SND_MIN_PERIOD_BYTES		0x4000
-#define KIRKWOOD_SND_MAX_PERIOD_BYTES		0x4000
 
 struct kirkwood_dma_data {
 	struct resource *mem;
