diff --git a/sound/soc/kirkwood/kirkwood.h b/sound/soc/kirkwood/kirkwood.h
index f7c58d0..10167b9 100644
--- a/sound/soc/kirkwood/kirkwood.h
+++ b/sound/soc/kirkwood/kirkwood.h
@@ -151,8 +151,8 @@
 
 /* Theses values come from the marvell alsa driver */
 /* need to find where they come from               */
-#define KIRKWOOD_SND_MAX_BUFFER_BYTES		0x40000		// 262144
-#define KIRKWOOD_SND_MAX_PERIOD_BYTES		0x8000		// 32768
+#define KIRKWOOD_SND_MAX_BUFFER_BYTES		0x100000	// 1024K
+#define KIRKWOOD_SND_MAX_PERIOD_BYTES		0x20000		// 128K
 #define KIRKWOOD_SND_MIN_PERIOD_BYTES		0x800		// 512
 #define KIRKWOOD_SND_MAX_PERIODS		(KIRKWOOD_SND_MAX_BUFFER_BYTES/KIRKWOOD_SND_MIN_PERIOD_BYTES)
 #define KIRKWOOD_SND_MIN_PERIODS		8
