diff --git a/drivers/video/dovefb/nxp_hdmi/tda998x_i2c.h b/drivers/video/dovefb/nxp_hdmi/tda998x_i2c.h
index 2f5adcd..35b32ef 100644
--- a/drivers/video/dovefb/nxp_hdmi/tda998x_i2c.h
+++ b/drivers/video/dovefb/nxp_hdmi/tda998x_i2c.h
@@ -19,7 +19,7 @@
 #ifndef __i2c_h__
 #define __i2c_h__
 
-#define CEC_I2C_DEBUG
+//#define CEC_I2C_DEBUG
 
 struct i2c_client *txGetThisI2cClient(void);
 struct i2c_client *cecGetThisI2cClient(void);
