diff --git a/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c b/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
index 7da34b9..2ac079d 100644
--- a/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
+++ b/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
@@ -40,6 +40,7 @@ extern void tda19988_set_audio_rate(unsigned rate);
 	 	 	 SNDRV_PCM_RATE_192000)
 
 #define STUB_FORMATS	(SNDRV_PCM_FMTBIT_S16_LE | \
+			 SNDRV_PCM_FMTBIT_S24_LE | \
 			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_LE | \
 			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_BE)
 
