diff --git a/drivers/video/dovefb/nxp_hdmi/tda998x.c b/drivers/video/dovefb/nxp_hdmi/tda998x.c
index 3416c8e..e5a231b 100755
--- a/drivers/video/dovefb/nxp_hdmi/tda998x.c
+++ b/drivers/video/dovefb/nxp_hdmi/tda998x.c
@@ -2092,7 +2092,7 @@ void tda19988_set_audio_rate(unsigned rate)
       }
    }
 
-#if 1
+#if 0
    printk(KERN_INFO "HDMI TX - setting audio rate to %d (%d -> %d)\n", 
           rate, this->tda.setio.audio_in.rate, rateCode);
 #endif
@@ -2101,24 +2101,12 @@ void tda19988_set_audio_rate(unsigned rate)
 
    if (is_ready_for_output(this) && this->tda.setio.audio_in.rate != rateCode) {
 
-      // Hack: temporary switch to I2S to avoid audio lockup
       TRY(tmdlHdmiTxSetAudioMute(this->tda.instance, true));
-      msleep(5);
-   
-      this->tda.setio.audio_in.format = TMDL_HDMITX_AFMT_I2S;
-      TRY(tmdlHdmiTxSetAudioInput(this->tda.instance, \
-                                  this->tda.setio.audio_in, \
-                                  this->tda.setio.sink));
-      msleep(30);
 
-      // set new audio sampling rate
       this->tda.setio.audio_in.rate = rateCode;
-      this->tda.setio.audio_in.format = TMDL_HDMITX_AFMT_SPDIF;
       TRY(tmdlHdmiTxSetAudioInput(this->tda.instance, \
                                  this->tda.setio.audio_in, \
                                  this->tda.setio.sink));
-      msleep(5);
-
 TRY_DONE:
       tmdlHdmiTxSetAudioMute(this->tda.instance, this->tda.audio_mute);
    } else {
