diff --git a/drivers/video/dovefb/nxp_hdmi/tda998x.c b/drivers/video/dovefb/nxp_hdmi/tda998x.c
index e5a231b..8e648ec 100755
--- a/drivers/video/dovefb/nxp_hdmi/tda998x.c
+++ b/drivers/video/dovefb/nxp_hdmi/tda998x.c
@@ -1081,19 +1081,22 @@ static int hdmi_tx_init(tda_instance *this)
    this->tda.setio.audio_in.channelStatus.wordLength = TMDL_HDMITX_CSWORD_DEFAULT;
    this->tda.setio.audio_in.channelStatus.origSampleFreq = TMDL_HDMITX_CSOFREQ_44_1k;
 #else
+   this->tda.setio.audio_in.rate = TMDL_HDMITX_AFS_44K;
    this->tda.setio.audio_in.format = TMDL_HDMITX_AFMT_SPDIF;
-   this->tda.setio.audio_in.dstRate = TMDL_HDMITX_DSTRATE_DOUBLE;//TMDL_HDMITX_DSTRATE_SINGLE; /* not relevant here */
+
+   this->tda.setio.audio_in.dstRate = TMDL_HDMITX_DSTRATE_SINGLE; /* not relevant here */
    this->tda.setio.audio_in.channelAllocation = 0; /* audio channel allocation (Ref to CEA-861D p85) */
-   this->tda.setio.audio_in.rate = TMDL_HDMITX_AFS_44K;
+   
+// Rudi - defaults for I2S mode, probaly not needed for SPDIF
+   this->tda.setio.audio_in.i2sQualifier = TMDL_HDMITX_I2SQ_32BITS;
    this->tda.setio.audio_in.channelStatus.PcmIdentification = TMDL_HDMITX_AUDIO_DATA_PCM;
    this->tda.setio.audio_in.channelStatus.CopyrightInfo = TMDL_HDMITX_CSCOPYRIGHT_UNPROTECTED;
    this->tda.setio.audio_in.channelStatus.FormatInfo = TMDL_HDMITX_CSFI_PCM_2CHAN_NO_PRE;
-   this->tda.setio.audio_in.channelStatus.categoryCode = 0;
+   this->tda.setio.audio_in.channelStatus.categoryCode = 0x82;
    this->tda.setio.audio_in.channelStatus.clockAccuracy = TMDL_HDMITX_CSCLK_LEVEL_II;
    this->tda.setio.audio_in.channelStatus.maxWordLength = TMDL_HDMITX_CSMAX_LENGTH_24;
    this->tda.setio.audio_in.channelStatus.wordLength = TMDL_HDMITX_CSWORD_DEFAULT;
    this->tda.setio.audio_in.channelStatus.origSampleFreq = TMDL_HDMITX_CSOFREQ_44_1k;
-   this->tda.setio.audio_in.i2sQualifier = TMDL_HDMITX_I2SQ_32BITS;
 #endif
 
 // Rabeeh - D2 uses below?? --> This is original   this->tda.setio.sink = TMDL_HDMITX_SINK_HDMI; /* skip edid reading */
diff --git a/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c b/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
index 2ac079d..f41b25e 100644
--- a/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
+++ b/drivers/video/dovefb/nxp_hdmi/tda998x_dai.c
@@ -41,16 +41,15 @@ extern void tda19988_set_audio_rate(unsigned rate);
 
 #define STUB_FORMATS	(SNDRV_PCM_FMTBIT_S16_LE | \
 			 SNDRV_PCM_FMTBIT_S24_LE | \
-			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_LE | \
-			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_BE)
+			 SNDRV_PCM_FMTBIT_IEC958_SUBFRAME_LE)
+
 
-			 
 static int hdmi_dit_hw_params(struct snd_pcm_substream *substream,
 			       struct snd_pcm_hw_params *params, 
 			       struct snd_soc_dai *dai)
 {
 	DPRINTK("substream = %p, params = %p\n", substream, params);
-	DPRINTK("rate = %d\n", params_rate(params));
+	DPRINTK("rate = %d, channels = %d\n", params_rate(params), params_channels(params));
 	DPRINTK("dai = %s\n", dai->name);
 	
 	tda19988_set_audio_rate(params_rate(params));
