diff --git a/drivers/uio/uio.c b/drivers/uio/uio.c
index 2e1c6f8..19aa72e 100644
--- a/drivers/uio/uio.c
+++ b/drivers/uio/uio.c
@@ -507,13 +507,18 @@ static unsigned int uio_poll(struct file *filep, poll_table *wait)
 {
 	struct uio_listener *listener = filep->private_data;
 	struct uio_device *idev = listener->dev;
+	s32 event_count;
 
 	if (!idev->info->irq)
 		return -EIO;
 
 	poll_wait(filep, &idev->wait, wait);
-	if (listener->event_count != atomic_read(&idev->event))
+
+	event_count = atomic_read(&idev->event);
+	if (listener->event_count != event_count){
+		listener->event_count = event_count;
 		return POLLIN | POLLRDNORM;
+	}
 	return 0;
 }
 
