From 34a667c52ffef500f3478b30a3654342e7d1e1fa Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Fri, 29 Mar 2013 15:53:30 +0300
Subject: [PATCH 1/4] Protect BMM_DUMP with the global mutex

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/misc/bmm_drv.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/misc/bmm_drv.c b/drivers/misc/bmm_drv.c
index ab16e0e..0ce2bd6 100644
--- a/drivers/misc/bmm_drv.c
+++ b/drivers/misc/bmm_drv.c
@@ -159,10 +159,12 @@ static void bmm_dump_list(struct list_head *head)
 
 static void bmm_dump_all(void)
 {
+	mutex_lock(&bmm_mutex);
 	pr_info("free block list:\n");
 	bmm_dump_list(&(bmm_free_block.list));
 	pr_info("used block list:\n");
 	bmm_dump_list(&(bmm_used_block.list));
+	mutex_unlock(&bmm_mutex);
 }
 
 static struct bmm_block_t *bmm_search_vaddr(struct list_head *head,
-- 
1.7.9.5

