diff --git a/drivers/misc/bmm_drv.c b/drivers/misc/bmm_drv.c
index 0ce2bd6..05e4934 100644
--- a/drivers/misc/bmm_drv.c
+++ b/drivers/misc/bmm_drv.c
@@ -116,24 +116,22 @@ static unsigned long uva_to_pa(struct mm_struct *mm, unsigned long addr)
 unsigned long va_to_pa(unsigned long user_addr, unsigned int size)
 {
 	unsigned long  paddr, paddr_tmp;
-	unsigned long  size_tmp = 0;
-	int page_num = PAGE_ALIGN(size) / PAGE_SIZE;
-	unsigned int vaddr = PAGE_ALIGN(user_addr);
-	int i = 0;
+	int extra_page_count = ((user_addr & ~PAGE_MASK) + size - 1) / PAGE_SIZE;
 	struct mm_struct *mm = current->mm;
 
-	if (vaddr == 0)
+	if (user_addr == 0)
 		return 0;
 
-	paddr = uva_to_pa(mm, vaddr);
+	paddr = uva_to_pa(mm, user_addr);
+	paddr_tmp = paddr;
 
-	for (i = 0; i < page_num; i++) {
-		paddr_tmp = uva_to_pa(mm, vaddr);
-		if ((paddr_tmp - paddr) != size_tmp)
+	while (extra_page_count-- > 0) {
+		user_addr += PAGE_SIZE;
+		paddr_tmp += PAGE_SIZE;
+		if (uva_to_pa(mm, user_addr) != paddr_tmp)
 			return 0;
-		vaddr += PAGE_SIZE;
-		size_tmp += PAGE_SIZE;
 	}
+
 	return paddr;
 }
 
