From 4efbd634b86c0cb51fa316da96436d8c8de63dbf Mon Sep 17 00:00:00 2001
From: Re van Dorst <opensource@vdorst.com>
Date: Thu, 4 Apr 2013 19:01:42 +0200
Subject: [PATCH 3/3] [LEDS] Removed not needed include file.

---
 drivers/media/rc/rc-main.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index ba0a9d2..dfe2e1d 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -23,7 +23,6 @@
 
 #if defined(CONFIG_LEDS_TRIGGER_REMOTE_CONTROL)
 /* LED Trigger Remote Control. */
-#include <linux/leds.h>
 extern void ledtrig_rc_activity(void);
 #endif
 
-- 
1.7.9.5

