--- a/host_applications/linux/apps/hello_pi/hello_fft/makefile	2015-01-10 23:20:37.092012002 +0100
+++ b/host_applications/linux/apps/hello_pi/hello_fft/makefile	2015-01-10 23:20:37.104012000 +0100
@@ -24,10 +24,10 @@
 F = -lrt -lm
 
 hello_fft.bin:	$(S) $(C1D) $(H1D)
-	gcc -o hello_fft.bin $(F) $(C1D)
+	gcc -o hello_fft.bin $(C1D) $(F)
 
 hello_fft_2d.bin:	$(S) hex/shader_trans.hex $(C2D) $(H2D)
-	gcc -o hello_fft_2d.bin $(F) $(C2D)
+	gcc -o hello_fft_2d.bin $(C2D) $(F)
 
 clean:
 	rm -f *.bin
