--- a/host_applications/linux/apps/hello_pi/hello_fft/makefile	2015-02-15 09:40:04.481057194 +0100
+++ b/host_applications/linux/apps/hello_pi/hello_fft/makefile	2015-02-15 09:40:04.493057195 +0100
@@ -27,10 +27,10 @@
 all:	hello_fft.bin hello_fft_2d.bin
 
 hello_fft.bin:	$(S) $(C1D) $(H1D)
-	gcc -o hello_fft.bin $(F) $(C1D)
+	$(CC) -o hello_fft.bin $(F) $(C1D)
 
 hello_fft_2d.bin:	$(S) hex/shader_trans.hex $(C2D) $(H2D)
-	gcc -o hello_fft_2d.bin $(F) $(C2D)
+	$(CC) -o hello_fft_2d.bin $(F) $(C2D)
 
 clean:
 	rm -f *.bin
