diff -Naur systemd-13/Makefile.in systemd-13a/Makefile.in
--- systemd-13/Makefile.in	2010-11-18 23:56:52.114745754 +0100
+++ systemd-13a/Makefile.in	2010-11-21 10:59:28.125905338 +0100
@@ -966,21 +966,6 @@
 dist_udevrules_DATA = \
 	src/99-systemd.rules
 
-dbusinterface_DATA = \
-	org.freedesktop.systemd1.Manager.xml \
-	org.freedesktop.systemd1.Job.xml \
-	org.freedesktop.systemd1.Unit.xml \
-	org.freedesktop.systemd1.Service.xml \
-	org.freedesktop.systemd1.Socket.xml \
-	org.freedesktop.systemd1.Timer.xml \
-	org.freedesktop.systemd1.Target.xml \
-	org.freedesktop.systemd1.Device.xml \
-	org.freedesktop.systemd1.Mount.xml \
-	org.freedesktop.systemd1.Automount.xml \
-	org.freedesktop.systemd1.Snapshot.xml \
-	org.freedesktop.systemd1.Swap.xml \
-	org.freedesktop.systemd1.Path.xml
-
 dist_bashcompletion_DATA = \
         src/systemctl-bash-completion.sh
 
@@ -5459,9 +5444,6 @@
 @HAVE_XSLTPROC_TRUE@man/%.html.in: man/%.xml.in
 @HAVE_XSLTPROC_TRUE@	$(XSLTPROC_PROCESS_HTML_IN)
 
-org.freedesktop.systemd1.%.xml: systemd
-	$(AM_V_GEN)SYSTEMD_SKIP_API_MOUNTS=1 ./systemd --introspect=${@:.xml=} > $@
-
 install-data-hook:
 	$(MKDIR_P) -m 0755 \
 		$(DESTDIR)$(systemunitdir) \
