--- systemd-28.orig/src/ask-password-api.c	2011-06-21 17:16:36.000000000 +0200
+++ systemd-28/src/ask-password-api.c	2011-06-21 17:16:50.000000000 +0200
@@ -319,7 +319,7 @@
 
         mkdir_p("/run/systemd/ask-password", 0755);
 
-        if ((fd = mkostemp(temp, O_CLOEXEC|O_CREAT|O_WRONLY)) < 0) {
+        if ((fd = mkstemp(temp)) < 0) {
                 log_error("Failed to create password file: %m");
                 r = -errno;
                 goto finish;
--- systemd-28.orig/src/manager.c	2011-06-21 17:16:59.000000000 +0200
+++ systemd-28/src/manager.c	2011-06-21 17:17:13.000000000 +0200
@@ -2623,7 +2623,7 @@
                 return -ENOMEM;
 
         saved_umask = umask(0077);
-        fd = mkostemp(path, O_RDWR|O_CLOEXEC);
+        fd = mkstemp(path);
         umask(saved_umask);
 
         if (fd < 0) {
