diff -Naur systemd-19/src/99-systemd.rules systemd-19a/src/99-systemd.rules
--- systemd-19/src/99-systemd.rules	2011-02-28 22:00:08.000000000 +0100
+++ systemd-19a/src/99-systemd.rules	2011-03-04 12:33:37.898606756 +0100
@@ -9,6 +9,7 @@
 
 SUBSYSTEM=="tty", KERNEL=="tty[0-9]|tty1[0-2]", TAG+="systemd"
 SUBSYSTEM=="tty", KERNEL=="ttyS*", TAG+="systemd"
+SUBSYSTEM=="tty", KERNEL=="ttyO*", TAG+="systemd"
 SUBSYSTEM=="tty", KERNEL=="hvc*", TAG+="systemd"
 SUBSYSTEM=="tty", KERNEL=="ttyUSB*", TAG+="systemd"
 
