diff -Naur systemd-13/Makefile.in systemd-13a/Makefile.in
--- systemd-13/Makefile.in	2010-11-21 11:32:39.213913074 +0100
+++ systemd-13a/Makefile.in	2010-11-21 11:33:20.953985818 +0100
@@ -5466,7 +5466,8 @@
 		$(DESTDIR)$(pkgsysconfdir)/system/multi-user.target.wants \
 		$(DESTDIR)$(pkgsysconfdir)/system/getty.target.wants \
 		$(DESTDIR)$(pkgsysconfdir)/user \
-		$(DESTDIR)$(sysconfdir)/xdg/systemd
+		$(DESTDIR)$(sysconfdir)/xdg/systemd \
+		$(DESTDIR)$(dbussessionservicedir)
 	( cd $(DESTDIR)$(sysconfdir)/xdg/systemd/ && \
 		rm -f user && \
 		$(LN_S) $(pkgsysconfdir)/user user )
