--- systemd-44/src/journal/journald.c.orig	2012-04-17 14:09:06.459560709 +0300
+++ systemd-44/src/journal/journald.c	2012-04-17 14:09:49.451773397 +0300
@@ -2144,7 +2144,7 @@
                                 uint8_t buf[CMSG_SPACE(sizeof(struct ucred)) +
                                             CMSG_SPACE(sizeof(struct timeval)) +
                                             CMSG_SPACE(sizeof(int)) + /* fd */
-                                            CMSG_SPACE(NAME_MAX)]; /* selinux label */
+                                            CMSG_SPACE(getpagesize())]; /* selinux label */
                         } control;
                         ssize_t n;
                         int v;
