--- vdr-softhddevice-0.5.2+9954b939e/Makefile.org	2013-03-02 23:57:30.194871661 +0100
+++ vdr-softhddevice-0.5.2+9954b939e/Makefile	2013-03-02 23:57:33.250871715 +0100
@@ -27,6 +27,7 @@
 CONFIG := # -DDEBUG #-DOSD_DEBUG
 CONFIG += -DAV_INFO -DAV_INFO_TIME=3000	# info/debug a/v sync
 CONFIG += -DUSE_PIP			# PIP support
+CONFIG += -DUSE_YAEPG			# Yaepg HD
 #CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
 #CONFIG += -DNO_TS_AUDIO		# disable ts audio parser
 #CONFIG += -DUSE_TS_VIDEO		# build new ts video parser
