--- softhddevice.old/softhddevice.cpp	2013-04-07 08:43:55.442441002 +0200
+++ softhddevice.new/softhddevice.cpp	2013-04-07 08:45:53.958441000 +0200
@@ -22,12 +22,14 @@
 
 #define __STDC_CONSTANT_MACROS		///< needed for ffmpeg UINT64_C
 
+#pragma GCC diagnostic ignored "-Wunused-parameter"
 #include <vdr/interface.h>
 #include <vdr/plugin.h>
 #include <vdr/player.h>
 #include <vdr/osd.h>
 #include <vdr/dvbspu.h>
 #include <vdr/shutdown.h>
+#pragma GCC diagnostic warning "-Wunused-parameter"
 
 #ifdef HAVE_CONFIG
 #include "config.h"
@@ -1341,7 +1343,9 @@
 //	cReceiver
 //////////////////////////////////////////////////////////////////////////////
 
+#pragma GCC diagnostic ignored "-Wunused-parameter"
 #include <vdr/receiver.h>
+#pragma GCC diagnostic warning "-Wunused-parameter"
 
 /**
 **	Receiver class for PIP mode.
