--- vdr-softhddevice-0.6.1/softhddevice.cpp.orig	2014-01-28 13:59:58.000000000 +0100
+++ vdr-softhddevice-0.6.1/softhddevice.cpp	2014-02-01 18:15:07.438409883 +0100
@@ -2212,6 +2212,10 @@
 	    break;
 	case pmAudioOnly:
 	case pmAudioOnlyBlack:
+            dsyslog("[softhddev] play mode external audio\n");
+            Suspend(0, 1, 0);
+            SuspendMode = SUSPEND_EXTERNAL;
+            return true;
 	    break;
 	case pmVideoOnly:
 	    break;
