--- serdisplib-1.98.0-r341/Makefile.in.org	2010-04-11 16:52:05.000000000 +0200
+++ serdisplib-1.98.0-r341/Makefile.in	2013-03-06 10:37:51.949654121 +0100
@@ -12,6 +12,7 @@
 bindir = @bindir@
 libdir = @libdir@
 mandir = @mandir@
+destdir = .install
 abs_top_srcdir=@abs_top_srcdir@
 base_abs_top_srcdir=$(shell /bin/basename $(abs_top_srcdir))
 current_timestamp=$(shell /bin/date '+%Y%m%d-%H%M')
@@ -38,8 +39,8 @@
 	done && test -z "$$fail"
 
 install:
-	$(top_srcdir)/mkinstalldirs $(libdir)/pkgconfig
-	$(INSTALL_DATA) serdisplib.pc $(libdir)/pkgconfig/
+	$(top_srcdir)/mkinstalldirs $(destdir)/$(libdir)/pkgconfig
+	$(INSTALL_DATA) serdisplib.pc $(destdir)/$(libdir)/pkgconfig/
 	@for dir in ${subdirs}; do \
 	  (cd $(srcdir) && cd $$dir && $(MAKE) install) \
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
--- serdisplib-1.98.0-r341/server/Makefile.in.org	2010-10-02 21:14:22.000000000 +0200
+++ serdisplib-1.98.0-r341/server/Makefile.in	2013-03-06 10:38:02.181654296 +0100
@@ -20,6 +20,7 @@
 mandir = @mandir@
 includedir = @includedir@
 sysconfdir = @sysconfdir@
+destdir = .install
 
 CC=@CC@
 AR=@AR@ -r
@@ -111,22 +112,22 @@
 
 
 install: $(PROGRAMS)
-	$(top_srcdir)/mkinstalldirs $(bindir)
-	$(top_srcdir)/mkinstalldirs $(sbindir)
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(bindir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(sbindir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(sysconfdir)
 	list='$(PROGRAMS_SBIN)'; \
 	for prog in $$list; do \
-	  $(INSTALL_PROGRAM) $$prog $(sbindir)/ ; \
+	  $(INSTALL_PROGRAM) $$prog $(top_srcdir)/$(destdir)/$(sbindir)/ ; \
 	done
 	list='$(PROGRAMS_BIN)'; \
 	for prog in $$list; do \
-	  $(INSTALL_PROGRAM) $$prog $(bindir)/ ; \
+	  $(INSTALL_PROGRAM) $$prog $(top_srcdir)/$(destdir)/$(bindir)/ ; \
 	done
-	$(INSTALL_DATA) examples/serdispd.cfg $(sysconfdir)/
+	$(INSTALL_DATA) examples/serdispd.cfg $(top_srcdir)/$(destdir)/$(sysconfdir)/
 
 
 uninstall:
 	list='$(PROGRAMS)'; \
 	for prog in $$list; do \
-	  /bin/rm -f $(sbindir)/$$prog ; \
+	  /bin/rm -f $(top_srcdir)/$(destdir)/$(sbindir)/$$prog ; \
 	done
--- serdisplib-1.98.0-r341/src/Makefile.in.org	2011-03-20 15:21:52.000000000 +0100
+++ serdisplib-1.98.0-r341/src/Makefile.in	2013-03-06 10:38:11.021654447 +0100
@@ -31,6 +31,7 @@
 libdir = @libdir@
 mandir = @mandir@
 includedir = @includedir@
+destdir = .install
 
 CC=@CC@
 AR=@AR@ -r
@@ -196,30 +197,30 @@
 	/bin/rm -f Makefile
 
 install: $(LIB_DIR)/$(LIB_SHARED) $(LIB_DIR)/$(LIB_STATIC)
-	$(top_srcdir)/mkinstalldirs $(bindir)
-	$(top_srcdir)/mkinstalldirs $(libdir)
-	$(top_srcdir)/mkinstalldirs $(includedir)
-	$(top_srcdir)/mkinstalldirs $(includedir)/serdisplib
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(bindir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(libdir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(includedir)
+	$(top_srcdir)/mkinstalldirs $(top_srcdir)/$(destdir)/$(includedir)/serdisplib
 	
-	$(INSTALL_PROGRAM) $(PROGRAMS) $(bindir)/
-	$(INSTALL_PROGRAM) $(LIB_DIR)/$(LIB_SHARED) $(libdir)/
-	$(INSTALL_DATA) $(LIB_DIR)/$(LIB_STATIC) $(libdir)/
+	$(INSTALL_PROGRAM) $(PROGRAMS) $(top_srcdir)/$(destdir)/$(bindir)/
+	$(INSTALL_PROGRAM) $(LIB_DIR)/$(LIB_SHARED) $(top_srcdir)/$(destdir)/$(libdir)/
+	$(INSTALL_DATA) $(LIB_DIR)/$(LIB_STATIC) $(top_srcdir)/$(destdir)/$(libdir)/
 	list='$(LIB_HEADERFILES)'; \
 	for headerfile in $$list; do \
-	  $(INSTALL_DATA) ../include/serdisplib/$$headerfile $(includedir)/serdisplib/ ; \
+	  $(INSTALL_DATA) ../include/serdisplib/$$headerfile $(top_srcdir)/$(destdir)/$(includedir)/serdisplib/ ; \
 	done
 	if [ -n "$(VERSION_SUBMINOR)" ]  ; then \
-	  cd $(libdir) && $(LN_S) -f $(LIB_SHARED) lib$(LIB_NAME).so.$(VERSION_MAJOR).$(VERSION_MINOR) ; \
+	  cd $(top_srcdir)/$(destdir)/$(libdir) && $(LN_S) -f $(LIB_SHARED) lib$(LIB_NAME).so.$(VERSION_MAJOR).$(VERSION_MINOR) ; \
 	fi
-	cd $(libdir) && $(LN_S) -f $(LIB_SHARED) lib$(LIB_NAME).so.$(VERSION_MAJOR)
-	cd $(libdir) && $(LN_S) -f lib$(LIB_NAME).so.$(VERSION_MAJOR) lib$(LIB_NAME).so
+	cd $(top_srcdir)/$(destdir)/$(libdir) && $(LN_S) -f $(LIB_SHARED) lib$(LIB_NAME).so.$(VERSION_MAJOR)
+	cd $(top_srcdir)/$(destdir)/$(libdir) && $(LN_S) -f lib$(LIB_NAME).so.$(VERSION_MAJOR) lib$(LIB_NAME).so
 
 uninstall:
-	-/bin/rm -f $(libdir)/libserdisp*
-	-/bin/rm -f $(includedir)/serdisplib/*.h
-	-/bin/rmdir $(includedir)/serdisplib
+	-/bin/rm -f $(top_srcdir)/$(destdir)/$(libdir)/libserdisp*
+	-/bin/rm -f $(top_srcdir)/$(destdir)/$(includedir)/serdisplib/*.h
+	-/bin/rmdir $(top_srcdir)/$(destdir)/$(includedir)/serdisplib
 	list='$(PROGRAMS)'; \
 	for prog in $$list; do \
-	  /bin/rm -f $(bindir)/$$prog ; \
+	  /bin/rm -f $(top_srcdir)/$(destdir)/$(bindir)/$$prog ; \
 	done
 
--- serdisplib-1.98.0-r341/tools/Makefile.in.org	2010-10-02 21:14:22.000000000 +0200
+++ serdisplib-1.98.0-r341/tools/Makefile.in	2013-03-06 10:38:16.013654533 +0100
@@ -14,6 +14,7 @@
 libdir = @libdir@
 mandir = @mandir@
 includedir = @includedir@
+destdir = .install
 
 CC=@CC@
 AR=@AR@ -r
@@ -131,13 +132,14 @@
 	/bin/rm -f Makefile
 
 install: $(PROGRAMS)
+	mkdir -p $(top_srcdir)/$(destdir)/$(bindir); \
 	list='$(PROGRAMS)'; \
 	for prog in $$list; do \
-	  $(INSTALL_PROGRAM) $$prog $(bindir)/ ; \
+	  $(INSTALL_PROGRAM) $$prog $(top_srcdir)/$(destdir)/$(bindir)/ ; \
 	done
 
 uninstall: $(PROGRAMS)
 	list='$(PROGRAMS)'; \
 	for prog in $$list; do \
-	  /bin/rm -f $(bindir)/$$prog ; \
+	  /bin/rm -f $(top_srcdir)/$(destdir)/$(bindir)/$$prog ; \
 	done
