--- vdr-setup-0.3.1/Makefile.org	2013-03-14 10:50:15.104681846 +0100
+++ vdr-setup-0.3.1/Makefile	2013-03-14 10:50:37.904682237 +0100
@@ -99,9 +99,9 @@
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
 	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
 
-setupsystemmenu.c:  ../../../menu.c
+setupsystemmenu.c:  $(VDRDIR)/menu.c
 	@echo "Extracting class definitions from VDR's menu.c for standard Setup"
-	@./getVdrSetupMenu.pl ../../../menu.c
+	@./getVdrSetupMenu.pl $(VDRDIR)/menu.c
 
 	
 dist: clean
