diff -ruN setup-0.3.1//Makefile setup-mod//Makefile
--- setup-0.3.1//Makefile	2012-04-01 18:27:48.000000000 +0200
+++ setup-mod//Makefile	2012-04-03 23:50:15.422438439 +0200
@@ -16,7 +16,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -g -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -g -O2 -Wall -Werror=overloaded-virtual
 
 ### The directory environment:
 
@@ -48,7 +48,7 @@
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -DAPIVERSION="\"$(APIVERSION)\"" 
 
 ### The object files (add further files here):
-OBJS     = $(PLUGIN).o  i18n.o menus.o setupmenu.o setupsetup.o config.o plugins.o sysconfig.o util.o setupsystemmenu.o
+OBJS     = $(PLUGIN).o  menus.o setupmenu.o setupsetup.o config.o plugins.o sysconfig.o util.o setupsystemmenu.o
 
 ### The main target:
 
diff -ruN setup-0.3.1//setup.c setup-mod//setup.c
--- setup-0.3.1//setup.c	2012-04-01 18:27:48.000000000 +0200
+++ setup-mod//setup.c	2012-04-03 22:56:33.000000000 +0200
@@ -12,7 +12,7 @@
 #include <vdr/plugin.h>
 #include "setupmenu.h"
 #include "setupsetup.h"
-#include "i18n.h"
+//#include "i18n.h"
 
 static const char *VERSION        = "0.3.1-zulu-edition";
 static const char *DESCRIPTION    = trNOOP("System Setup");
@@ -68,9 +68,9 @@
 bool cPluginSetup::Initialize(void)
 {
   // Initialize any background activities the plugin shall perform.
-#if VDRVERSNUM < 10507
-  RegisterI18n(Phrases);
-#endif
+//#if VDRVERSNUM < 10507
+//  RegisterI18n(Phrases);
+//#endif
   return true;
 }
 
