--- libpcd-1.0.1.orig/GNUmakefile	2003-07-22 12:52:46.000000000 +0300
+++ libpcd-1.0.1/GNUmakefile	2004-12-03 12:17:32.000000000 +0200
@@ -6,6 +6,8 @@
 libdir	:= $(DESTDIR)$(prefix)/lib
 
 CC	:= gcc
+AR	:= ar
+RANLIB	:= ranlib
 WARN	:= -Wall -Wmissing-prototypes -Wstrict-prototypes -Wpointer-arith
 CFLAGS	?= -g -O2
 CFLAGS	+= $(WARN)
@@ -30,8 +32,8 @@
 
 libpcd.a: $(STATIC)
 	rm -f $@
-	ar rc $@ $(STATIC)
-	ranlib $@
+	$(AR) rc $@ $(STATIC)
+	$(RANLIB) $@
 
 libpcd.so.$(MINOR): $(SHARED)
 	rm -f $@
