diff -Naur a/setup.py b/setup.py
--- a/setup.py	2011-02-20 16:41:48.199175609 +0100
+++ b/setup.py	2011-02-20 16:46:43.787424954 +0100
@@ -147,14 +147,17 @@
             add_directory(library_dirs, "/opt/local/lib")
             add_directory(include_dirs, "/opt/local/include")
 
-        add_directory(library_dirs, "/usr/local/lib")
-        # FIXME: check /opt/stuff directories here?
-
-        prefix = sysconfig.get_config_var("prefix")
-        if prefix:
+        xcprefix = os.environ.get('PYTHONXCPREFIX')
+        if xcprefix:
+            add_directory(library_dirs, os.path.join(xcprefix, "lib"))
+            add_directory(include_dirs, os.path.join(xcprefix, "include"))
+        else:
+            add_directory(library_dirs, "/usr/local/lib")
             add_directory(library_dirs, os.path.join(prefix, "lib"))
             add_directory(include_dirs, os.path.join(prefix, "include"))
 
+        # FIXME: check /opt/stuff directories here?
+
         #
         # locate tkinter libraries
 
@@ -208,11 +211,11 @@
                 add_directory(include_dirs, tcl_dir)
 
         # standard locations
-        add_directory(library_dirs, "/usr/local/lib")
-        add_directory(include_dirs, "/usr/local/include")
-
-        add_directory(library_dirs, "/usr/lib")
-        add_directory(include_dirs, "/usr/include")
+        if not xcprefix:
+            add_directory(library_dirs, "/usr/local/lib")
+            add_directory(include_dirs, "/usr/local/include")
+            add_directory(library_dirs, "/usr/lib")
+            add_directory(include_dirs, "/usr/include")
 
         #
         # insert new dirs *before* default libs, to avoid conflicts
