diff -aur --unidirectional-new-file a/hybris/egl/Makefile.am b/hybris/egl/Makefile.am
--- a/hybris/egl/Makefile.am	2013-05-13 01:01:57.000000000 +0200
+++ b/hybris/egl/Makefile.am	2013-07-21 09:25:59.000000000 +0200
@@ -7,7 +7,10 @@
 	egl.c \
 	ws.c
 
-libEGL_la_CFLAGS = -I$(top_srcdir)/include -DPKGLIBDIR="\"$(pkglibdir)/\""
+#target_libdir=$(pkglibdir)
+target_libdir=/usr/local/lib/libhybris
+
+libEGL_la_CFLAGS = -I$(top_srcdir)/include -DPKGLIBDIR="\"$(target_libdir)/\""
 if WANT_MESA
 libEGL_la_CFLAGS += -DLIBHYBRIS_WANTS_MESA_X11_HEADERS
 endif
@@ -18,7 +21,7 @@
 libEGL_la_CFLAGS += -ggdb -O0
 endif
 
-libEGL_la_CXXFLAGS = -I$(top_srcdir)/include -DPKGLIBDIR="\"$(pkglibdir)/\""
+libEGL_la_CXXFLAGS = -I$(top_srcdir)/include -DPKGLIBDIR="\"$(target_libdir)/\""
 if WANT_MESA
 libEGL_la_CXXFLAGS += -DLIBHYBRIS_WANTS_MESA_X11_HEADERS
 endif
diff -aur --unidirectional-new-file a/hybris/egl/ws.c b/hybris/egl/ws.c
--- a/hybris/egl/ws.c	2013-05-13 01:01:57.000000000 +0200
+++ b/hybris/egl/ws.c	2013-07-21 09:26:27.000000000 +0200
@@ -20,7 +20,8 @@
 			egl_platform = "null";
 	
 		snprintf(ws_name, 2048, PKGLIBDIR "eglplatform_%s.so", egl_platform);	
-			
+
+		printf("####%s####\n", ws_name);
 		void *wsmod = (void *) dlopen(ws_name, RTLD_LAZY);
 		assert(wsmod != NULL);
 		ws = dlsym(wsmod, "ws_module_info");
