From: Loic Minier <lool@dooz.org>
Subject: Remove check that the compiler produces executables we can run.

Index: alsa-lib-1.0.23/configure
===================================================================
--- alsa-lib-1.0.23.orig/configure	2010-05-30 20:56:55.000000000 +0200
+++ alsa-lib-1.0.23/configure	2010-05-30 20:56:55.000000000 +0200
@@ -3598,7 +3598,7 @@
 # the compiler is broken, or we cross compile.
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether we are cross compiling" >&5
 $as_echo_n "checking whether we are cross compiling... " >&6; }
-if test "$cross_compiling" != yes; then
+if false && test "$cross_compiling" != yes; then
   { { ac_try="$ac_link"
 case "(($ac_try" in
   *\"* | *\`* | *\\*) ac_try_echo=\$ac_try;;
