--- libcec-2.1.4-2/src/testclient/main.cpp.org	2014-03-16 16:32:54.943773159 +0100
+++ libcec-2.1.4-2/src/testclient/main.cpp	2014-03-16 16:33:10.283773420 +0100
@@ -1204,7 +1204,7 @@
 
   g_config.Clear();
   g_callbacks.Clear();
-  snprintf(g_config.strDeviceName, 13, "CECTester");
+  snprintf(g_config.strDeviceName, 13, "SifBoX");
   g_config.clientVersion       = CEC_CONFIG_VERSION;
   g_config.bActivateSource     = 0;
   g_callbacks.CBCecLogMessage  = &CecLogMessage;
