diff -Naur opkg-utils-r5966//opkg-build opkg-utils-r5966a/opkg-build
--- opkg-utils-r5966//opkg-build	2010-05-25 14:18:38.000000000 +0200
+++ opkg-utils-r5966a/opkg-build	2010-05-25 15:06:22.597756432 +0200
@@ -239,7 +239,7 @@
 
 echo "2.0" > $tmp_dir/debian-binary
 
-if (( $opkext == 1)); then 
+if [ "$opkext" = 1 ]; then 
 	pkg_file=$dest_dir/${pkg}_${version}_${arch}.opk
 else
 	pkg_file=$dest_dir/${pkg}_${version}_${arch}.ipk
