diff -Naur opkg-utils-r5966.orig/opkg-build opkg-utils-r5966/opkg-build
--- opkg-utils-r5966.orig/opkg-build	2010-06-27 13:58:49.884688275 +0200
+++ opkg-utils-r5966/opkg-build	2010-06-27 13:59:17.684688051 +0200
@@ -118,8 +118,8 @@
 	disallowed_filename=`disallowed_field Filename`
 	[ "$?" -ne 0 ] && PKG_ERROR=1
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "*** Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])" >&2
+	if echo $pkg | grep '[^a-zA-Z0-9.+-]'; then
+		echo "*** Error: Package name $name contains illegal characters, (other than [a-zA-Z0-9.+-])" >&2
 		PKG_ERROR=1;
 	fi
 
