diff -Naur rt2400-1.2.1.orig/Module/rtmp.h rt2400-1.2.1/Module/rtmp.h
--- rt2400-1.2.1.orig/Module/rtmp.h	2004-08-23 17:10:51.000000000 +0200
+++ rt2400-1.2.1/Module/rtmp.h	2005-01-14 18:29:39.000000000 +0100
@@ -858,10 +858,10 @@
 // Enable & Disable NIC interrupt via writing interrupt mask register
 // Since it use ADAPTER structure, it have to be put after structure definition.
 //
-inline	VOID    NICDisableInterrupt(
+VOID    NICDisableInterrupt(
     IN  PRTMP_ADAPTER   pAd);
 
-inline	VOID    NICEnableInterrupt(
+VOID    NICEnableInterrupt(
     IN  PRTMP_ADAPTER   pAd);
 
 VOID	RTMPFreeDMAMemory(
