--- xbmc-13.alpha-imx-r22727b2de2e7a5221f19b3bc20464c4e956de49f/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2013-10-16 09:05:06.248044294 +0200
+++ xbmc-13.alpha-imx-r22727b2de2e7a5221f19b3bc20464c4e956de49f/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp-new	2013-10-16 09:07:14.312040086 +0200
@@ -1134,7 +1134,7 @@
     if (retSatus == 0)
       /* No Picture ready and Not enough VPU buffers.
        * Lets sleep a litlle to wait for VPU buffers to be freed by IPU... */
-      usleep(5000);
+      usleep(500);
 
 
 #ifdef IMX_PROFILE
