--- vdr-scraper2vdr-0.1.5/lib/Makefile.org	2014-11-17 11:37:30.085767280 +0100
+++ vdr-scraper2vdr-0.1.5/lib/Makefile	2014-11-17 11:37:38.217767419 +0100
@@ -12,7 +12,7 @@
 
 LIBOBJS = common.o curl.o imgtools.o config.o db.o tabledef.o dbdict.o
 
-CC        = g++
+CC        = $(CXX)
 doCompile = $(CC) -c $(CFLAGS) $(DEFINES)
 doLink    = $(CC) $(LFLAGS)
 doLib     = ar -rs
--- vdr-scraper2vdr-0.1.6/lib/db.c.org	2014-11-17 23:05:14.274475609 +0100
+++ vdr-scraper2vdr-0.1.6/lib/db.c	2014-11-17 23:04:41.602475048 +0100
@@ -7,7 +7,7 @@
 
 #include <stdio.h>
 
-#include <errmsg.h>
+#include <mysql/errmsg.h>
 
 #include "db.h"
 
