diff -Naur linux-2.6.35-903.19/drivers/media/IR/Kconfig linux-2.6.35-903.19a/drivers/media/IR/Kconfig
--- linux-2.6.35-903.19/drivers/media/IR/Kconfig	2010-11-21 10:53:25.000000000 +0100
+++ linux-2.6.35-903.19a/drivers/media/IR/Kconfig	2010-11-27 14:09:36.798696859 +0100
@@ -10,7 +10,7 @@
 
 config LIRC
 	tristate
-	default y
+	default n
 
 	---help---
 	   Enable this option to build the Linux Infrared Remote
