--- linux-3.4.0-1487.6/arch/arm/mach-omap2/pm.c	2012-09-10 18:50:19.000000000 +0200
+++ linux-3.4.0-1487.6/arch/arm/mach-omap2/pm.c-new	2012-10-21 11:32:25.429694124 +0200
@@ -474,7 +474,7 @@
 		omap2_set_init_voltage("mpu", "virt_dpll_mpu_ck", "mpu");
 	else
 		omap2_set_init_voltage("mpu", "dpll_mpu_ck", "mpu");
-	omap2_set_init_voltage("core", "virt_l3_ck", "l3_main_1");
+	omap2_set_init_voltage("core", "l3_div_ck", "l3_main_1");
 	omap2_set_init_voltage("iva", "virt_iva_ck", "iva");
 }
 
