diff -Naur linux-ubuntu-omap4-3.0.0-1206.11.orig//include/linux/Kbuild linux-ubuntu-omap4-3.0.0-1206.11/include/linux/Kbuild
--- linux-ubuntu-omap4-3.0.0-1206.11.orig//include/linux/Kbuild	2011-10-31 15:16:45.582318101 +0100
+++ linux-ubuntu-omap4-3.0.0-1206.11/include/linux/Kbuild	2011-10-31 15:17:07.018318610 +0100
@@ -280,7 +280,6 @@
 header-y += nvram.h
 header-y += omap3isp.h
 header-y += omapfb.h
-header-y += omap_gpu.h
 header-y += oom.h
 header-y += param.h
 header-y += parport.h
