diff --git a/include/configs/mx6solo_c1.h b/include/configs/mx6solo_c1.h
index 39b38c8..afb7609 100644
--- a/include/configs/mx6solo_c1.h
+++ b/include/configs/mx6solo_c1.h
@@ -107,7 +107,7 @@
 
 #define CONFIG_CMD_IMX_DOWNLOAD_MODE
 
-#define CONFIG_BOOTDELAY 3
+#define CONFIG_BOOTDELAY 1
 
 #define CONFIG_PRIME	"FEC0"
 
@@ -125,12 +125,13 @@
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	\
 	"bootcmd_net=run bootargs_base bootargs_nfs; "			\
 		"tftpboot ${loadaddr} ${kernel}; bootm\0"		\
-	"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "		\
-		"root=/dev/mmcblk0p1 rootwait\0"			\
-	"bootcmd_mmc=run bootargs_base bootargs_mmc; "			\
-	"mmc dev 3; "							\
-	"mmc read ${loadaddr} 0x800 0x2000; bootm\0"			\
-	"bootcmd=run bootcmd_net\0"					\
+	"bootargs_mmc=setenv bootargs consoleblank=0 console=ttymxc0,115200 "		\
+		"video=mxcfb0:dev=hdmi,1920x1080M@60,if=RGB24,bpp=16 "	\
+		"root=/dev/mmcblk0p1 rootwait\0"			\
+	"bootcmd_mmc=run bootargs_mmc; "				\
+		"ext2load mmc 0:1 0x10800000 /boot/uImage; "		\
+		"bootm 0x10800000 \0"					\
+	"bootcmd=run bootcmd_mmc\0"					\
 
 
 #define CONFIG_ARP_TIMEOUT		200UL
