--- a/scheduler/dirsvc.c
+++ b/scheduler/dirsvc.c
@@ -457,14 +457,26 @@
 #  endif /* __APPLE__ */
 #  ifdef HAVE_AVAHI
   {
-    cupsdSetString(&DNSSDComputerName, avahi_client_get_host_name(DNSSDClient));
-    cupsdSetString(&DNSSDHostName,
-                   avahi_client_get_host_name_fqdn(DNSSDClient));
+    const char	*host_name = avahi_client_get_host_name(DNSSDClient);
+    const char	*host_fqdn = avahi_client_get_host_name_fqdn(DNSSDClient);
+
+    cupsdSetString(&DNSSDComputerName, host_name ? host_name : ServerName);
+
+    if (host_fqdn)
+      cupsdSetString(&DNSSDHostName, host_fqdn);
+    else if (strchr(ServerName, '.'))
+      cupsdSetString(&DNSSDHostName, ServerName);
+    else
+      cupsdSetStringf(&DNSSDHostName, "%s.local", ServerName);
   }
 #  else /* HAVE_DNSSD */
   {
     cupsdSetString(&DNSSDComputerName, ServerName);
-    cupsdSetString(&DNSSDHostName, ServerName);
+
+    if (strchr(ServerName, '.'))
+      cupsdSetString(&DNSSDHostName, ServerName);
+    else
+      cupsdSetStringf(&DNSSDHostName, "%s.local", ServerName);
   }
 #  endif /* HAVE_AVAHI */
 
