diff -Naur cryptopy-1.2.5/setup.py cryptopy-1.2.5a/setup.py
--- cryptopy-1.2.5/setup.py	2003-03-02 20:31:42.000000000 -0800
+++ cryptopy-1.2.5a/setup.py	2012-05-21 21:18:59.049469077 -0700
@@ -12,11 +12,11 @@
     See http://software-carpentry.codesourcery.com/entries/build/Distutils/Distutils.html
     for more information about this file
 
-    Copyright © (c) 2002 by Paul A. Lambert
+    Copyright  (c) 2002 by Paul A. Lambert
     Read LICENSE.txt for license information.
 """
 
-from distutils.core import setup
+from setuptools import setup
 #import sys
 #assert sys.version >= '2', "Error -> Please install Python 2.0 or greater."
 
