diff -uNr u-boot-v2009.08-imx_3.0.35_4.1.0.orig/board/freescale/mx6q_sabresd/flash_header.S u-boot-v2009.08-imx_3.0.35_4.1.0/board/freescale/mx6q_sabresd/flash_header.S
--- u-boot-v2009.08-imx_3.0.35_4.1.0.orig/board/freescale/mx6q_sabresd/flash_header.S	2013-11-15 21:19:41.000000000 +0800
+++ u-boot-v2009.08-imx_3.0.35_4.1.0/board/freescale/mx6q_sabresd/flash_header.S	2013-12-04 18:10:08.687946295 +0800
@@ -363,9 +363,14 @@
 MXC_DCD_ITEM(68, MMDC_P0_BASE_ADDR + 0x01c, 0x00008000)
 MXC_DCD_ITEM(69, MMDC_P0_BASE_ADDR + 0x02c, 0x000026D2)
 MXC_DCD_ITEM(70, MMDC_P0_BASE_ADDR + 0x030, 0x005A1023)
-MXC_DCD_ITEM(71, MMDC_P0_BASE_ADDR + 0x040, 0x00000027)
 
+/*
+MXC_DCD_ITEM(71, MMDC_P0_BASE_ADDR + 0x040, 0x00000027)
 MXC_DCD_ITEM(72, MMDC_P0_BASE_ADDR + 0x000, 0x831A0000)
+*/
+//CS0END 1g=0x27 liuy
+MXC_DCD_ITEM(71, MMDC_P0_BASE_ADDR + 0x040, 0x00000047)
+MXC_DCD_ITEM(72, MMDC_P0_BASE_ADDR + 0x000, 0x841A0000)
 
 MXC_DCD_ITEM(73, MMDC_P0_BASE_ADDR + 0x01c, 0x04088032)
 MXC_DCD_ITEM(74, MMDC_P0_BASE_ADDR + 0x01c, 0x00008033)
diff -uNr u-boot-v2009.08-imx_3.0.35_4.1.0.orig/include/asm-arm/arch-mx6/mx6.h u-boot-v2009.08-imx_3.0.35_4.1.0/include/asm-arm/arch-mx6/mx6.h
--- u-boot-v2009.08-imx_3.0.35_4.1.0.orig/include/asm-arm/arch-mx6/mx6.h	2013-11-15 21:19:41.000000000 +0800
+++ u-boot-v2009.08-imx_3.0.35_4.1.0/include/asm-arm/arch-mx6/mx6.h	2013-12-04 18:13:03.235949908 +0800
@@ -129,7 +129,8 @@
 #define MMDC1_ARB_END_ADDR              0xFFFFFFFF
 #else
 #define MMDC0_ARB_BASE_ADDR             0x10000000
-#define MMDC0_ARB_END_ADDR              0x7FFFFFFF
+/*#define MMDC0_ARB_END_ADDR              0x7FFFFFFF liuy change to 8FFFFFFF*/
+#define MMDC0_ARB_END_ADDR              0x8FFFFFFF
 #define MMDC1_ARB_BASE_ADDR             0x80000000
 #define MMDC1_ARB_END_ADDR              0xFFFFFFFF
 #endif
diff -uNr u-boot-v2009.08-imx_3.0.35_4.1.0.orig/include/configs/mx6q_sabresd.h u-boot-v2009.08-imx_3.0.35_4.1.0/include/configs/mx6q_sabresd.h
--- u-boot-v2009.08-imx_3.0.35_4.1.0.orig/include/configs/mx6q_sabresd.h	2013-11-15 21:19:41.000000000 +0800
+++ u-boot-v2009.08-imx_3.0.35_4.1.0/include/configs/mx6q_sabresd.h	2013-12-04 18:14:57.043952263 +0800
@@ -121,17 +121,17 @@
 		"uboot=u-boot.bin\0"			\
 		"kernel=uImage\0"				\
 		"nfsroot=/opt/eldk/arm\0"				\
-		"bootargs_base=setenv bootargs console=ttymxc0,115200\0"\
+		"bootargs_base=setenv bootargs console=ttymxc0,115200 di0_primary console=tty1\0"\
 		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs "\
 			"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"\
 		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
 			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
-		"bootargs_mmc=setenv bootargs ${bootargs} ip=dhcp "     \
-			"root=/dev/mmcblk0p1 rootwait\0"                \
+		"bootargs_mmc=setenv bootargs ${bootargs} "     \
+			"root=/dev/mmcblk0p1 rootwait video=mxcfb1:dev=ldb,LDB-XGA,if=RGB666 video=mxcfb0:dev=hdmi,1920x1080M@60 video=mxcfb1:off video=mxcfb2:off fbmem=28M consoleblank=0 quiet\0"                \
 		"bootcmd_mmc=run bootargs_base bootargs_mmc; "   \
 		"mmc dev 3; "	\
-		"mmc read ${loadaddr} 0x800 0x2000; bootm\0"	\
-		"bootcmd=run bootcmd_net\0"                             \
+		"mmc read ${loadaddr} 0x800 0x4000; bootm\0"	\
+		"bootcmd=run bootcmd_mmc\0"                             \
 
 
 #define CONFIG_ARP_TIMEOUT	200UL
@@ -289,7 +289,8 @@
  */
 #define CONFIG_NR_DRAM_BANKS	1
 #define PHYS_SDRAM_1		CSD0_DDR_BASE_ADDR
-#define PHYS_SDRAM_1_SIZE	(1u * 1024 * 1024 * 1024)
+/*#define PHYS_SDRAM_1_SIZE	(1u * 1024 * 1024 * 1024) liuy change to 2u */
+#define PHYS_SDRAM_1_SIZE	(2u * 1024 * 1024 * 1024)
 #define iomem_valid_addr(addr, size) \
 	(addr >= PHYS_SDRAM_1 && addr <= (PHYS_SDRAM_1 + PHYS_SDRAM_1_SIZE))
 
