From c1a633c1ef44e72ea6305c3adb9217b118814e3f Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Wed, 6 Jul 2016 22:09:58 -0500
Subject: [PATCH 07/21] Version bumped to 0.5.101

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 1246932..c359321 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,7 +23,7 @@
 # Initialize Autoconf
 AC_PREREQ(2.57)
 AC_INIT([xf86-video-openchrome],
-        [0.5.100],
+        [0.5.101],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg&component=Driver/openchrome],
         [xf86-video-openchrome])
 
-- 
1.9.1

