From 9696d5de3fea53387faf162e0dec3cca1ad0956a Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Sun, 10 Jul 2016 13:13:30 -0500
Subject: [PATCH 16/21] Turning off interlace mode for IGA2

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/via_display.c b/src/via_display.c
index 4e28172..fd0ac6d 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -2175,6 +2175,12 @@ viaIGA2SetDisplayRegister(ScrnInfoPtr pScrn, DisplayModePtr mode)
         ViaCrtcMask(hwp, 0x62, 0x00, 0x01);
     }
 
+    /* Keep interlace mode off. */
+    /* 3X5.67[5] - Second Display Interlace Mode
+     *             0: Off
+     *             1: On */
+    ViaCrtcMask(hwp, 0x67, 0x00, 0x20);
+
 
     /* Set IGA2 horizontal total pixels.*/
     /* Horizontal Total Period: 4096 - 1 (max) */
-- 
1.9.1

