From 3ff683897d3ee33a2db935eee4d5e2ab95a5f2ac Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Sun, 10 Jul 2016 13:23:54 -0500
Subject: [PATCH 17/21] Version bumped to 0.5.105

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6f28921..3f5e2b1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,7 +23,7 @@
 # Initialize Autoconf
 AC_PREREQ(2.57)
 AC_INIT([xf86-video-openchrome],
-        [0.5.104],
+        [0.5.105],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg&component=Driver/openchrome],
         [xf86-video-openchrome])
 
-- 
1.9.1

