From bb30d2938c0574ff3a987422a15abf383d557efa Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Sun, 10 Jul 2016 13:46:04 -0500
Subject: [PATCH 18/21] Turning off spread spectrum and turning on ROC ECK

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/via_display.c b/src/via_display.c
index fd0ac6d..779028f 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -466,7 +466,7 @@ viaIGAInitCommon(ScrnInfoPtr pScrn)
      * 3C5.1E[0]   - On/Off ROC ECK
      *               0: Off
      *               1: On */
-    ViaSeqMask(hwp, 0x1E, 0xF4, 0xF4);
+    ViaSeqMask(hwp, 0x1E, 0xF5, 0xFD);
 
     /* Set DVP0 data drive strength to 0b11 (highest). */
     /* Set DVP0 clock drive strength to 0b11 (highest). */
-- 
1.9.1

