From 799e5d6ec4489c529b2afbfd4075533d3751c4c8 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Sun, 10 Jul 2016 13:47:10 -0500
Subject: [PATCH 19/21] Version bumped to 0.5.106

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 3f5e2b1..1ac17e8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -23,7 +23,7 @@
 # Initialize Autoconf
 AC_PREREQ(2.57)
 AC_INIT([xf86-video-openchrome],
-        [0.5.105],
+        [0.5.106],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg&component=Driver/openchrome],
         [xf86-video-openchrome])
 
-- 
1.9.1

