diff -Naur a/setup.py b/setup.py
--- a/setup.py	2011-02-17 21:49:16.000000000 +0100
+++ b/setup.py	2011-08-31 11:50:10.361237594 +0200
@@ -12,7 +12,7 @@
 # Freevo distutils stuff
 sys.path.append('./src')
 import version
-from distutils import core
+import setuptools
 from distutils.command import install
 from util.distribution import setup, Extension, check_libs, docbook_finder
 #try:
@@ -45,8 +45,6 @@
 if sys.hexversion < 0x2050000:
     libs_to_check.append(('elementtree', 'http://effbot.org/zone/elementtree.htm'))
 
-check_libs(libs_to_check)
-
 class CustomInstall(install.install):
     def run(self):
         for filename in files_to_remove:
@@ -56,7 +54,7 @@
         install.install.run(self)
 
 
-class Runtime(core.Command):
+class Runtime(setuptools.Command):
 
     description     = "download and install runtime"
     user_options    = []
@@ -158,10 +156,6 @@
         except Exception, why:
             traceback.print_exc()
 
-    if (not os.path.isdir('./Docs/installation/html')):
-        print 'Docs/howto not found. Please run ./autogen.sh'
-        sys.exit(0)
-
 import revision
 # only add files not in share and src
 
