diff -Nur fmio-2.0.8.orig/src/Makefile fmio-2.0.8/src/Makefile
--- fmio-2.0.8.orig/src/Makefile	Fri Apr 15 23:34:32 2005
+++ fmio-2.0.8/src/Makefile	Fri Apr 15 23:35:12 2005
@@ -6,6 +6,8 @@
 # Makefile for fmio -- small utility to manipulate FM radio card
 #
 CC?=cc
+AR?=ar
+RANLIB?=ranlib
 CFLAGS?=-O2 -Wall
 
 # don't use mixer stuff
@@ -97,8 +99,8 @@
 
 libradio.a: $(ALLHDRS) $(OBJS) $(DRVS)
 	rm -f $@
-	ar cru $@ $(OBJS) $(DRVS)
-	ranlib $@
+	$(AR) cru $@ $(OBJS) $(DRVS)
+	$(RANLIB) $@
 
 fmio.0: $(MANPAGE)
 	@echo "groff -Tascii -mandoc $(MANPAGE) > $@"
