--- vdr-rpihddevice-0.0.5/Makefile.org	2013-11-18 11:44:54.000000000 +0100
+++ vdr-rpihddevice-0.0.5/Makefile	2013-11-27 12:15:53.769050455 +0100
@@ -54,9 +54,9 @@
 CXXFLAGS += -D__STDC_CONSTANT_MACROS
 
 ILCDIR   =ilclient
-SDKSTAGE ?=/opt
-VCINCDIR =$(SDKSTAGE)/vc/include
-VCLIBDIR =$(SDKSTAGE)/vc/lib
+SDKSTAGE = /root/openrasp/build/build.raspberrypi.eglibc/toolchain/armv6-openbricks-linux-gnueabihf/sysroot/usr/local
+VCINCDIR =$(SDKSTAGE)/include
+VCLIBDIR =$(SDKSTAGE)/lib
 
 INCLUDES += -I$(ILCDIR) -I$(VCINCDIR) -I$(VCINCDIR)/interface/vcos/pthreads -I$(VCINCDIR)/interface/vmcs_host/linux
 
--- vdr-rpihddevice-0.0.5/ilclient/Makefile.org	2013-11-18 11:44:54.000000000 +0100
+++ vdr-rpihddevice-0.0.5/ilclient/Makefile	2013-11-27 12:16:01.425050305 +0100
@@ -1,10 +1,11 @@
 OBJS=ilclient.o ilcore.o
 LIB=libilclient.a
+SDKSTAGE = /root/openrasp/build/build.raspberrypi.eglibc/toolchain/armv6-openbricks-linux-gnueabihf/sysroot/usr/local
+
 
 CFLAGS+=-DSTANDALONE -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -DTARGET_POSIX -D_LINUX -fPIC -DPIC -D_REENTRANT -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -U_FORTIFY_SOURCE -Wall -g -DHAVE_LIBOPENMAX=2 -DOMX -DOMX_SKIP64BIT -ftree-vectorize -pipe -DUSE_EXTERNAL_OMX -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi
 
-SDKSTAGE ?= /opt
-INCLUDES+=-I$(SDKSTAGE)/vc/include/ -I$(SDKSTAGE)/vc/include/interface/vcos/pthreads -I$(SDKSTAGE)/vc/include/interface/vmcs_host/linux
+INCLUDES+=-I$(SDKSTAGE)/include/ -I$(SDKSTAGE)/include/interface/vcos/pthreads -I$(SDKSTAGE)/include/interface/vmcs_host/linux
 
 all: $(LIB)
 
