--- a/Makefile	2017-09-02 09:53:01.714737796 +0200
+++ b/Makefile	2017-09-02 09:53:01.726731887 +0200
@@ -1,5 +1,5 @@
-CC=gcc
-CXX=g++
+#CC=gcc
+#CXX=g++
 CFLAGS+=-D_FILE_OFFSET_BITS=64
 #CXXFLAGS+=-Wall -D_FILE_OFFSET_BITS=64 -D USE_UTF16
 CXXFLAGS+=-Wall -D_FILE_OFFSET_BITS=64
@@ -18,7 +18,7 @@
 #	$(CXX) $(LIB_OBJS) gdisk.o gpttext.o $(LDFLAGS) -licuio -licuuc -luuid -o gdisk
 
 cgdisk: $(LIB_OBJS) cgdisk.o gptcurses.o
-	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) -luuid -lncursesw -o cgdisk
+	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) -luuid -lncurses -o cgdisk
 #	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) -licuio -licuuc -luuid -lncurses -o cgdisk
 
 sgdisk: $(LIB_OBJS) sgdisk.o gptcl.o
