+* fix gcc4 compliation - "error: invalid lvalue in increment"

diff -Nur iconv-extra-2.0.orig/ces/unicode-1-1-utf-7.c iconv-extra-2.0/ces/unicode-1-1-utf-7.c
--- iconv-extra-2.0.orig/ces/unicode-1-1-utf-7.c	Tue Nov 21 00:41:22 2000
+++ iconv-extra-2.0/ces/unicode-1-1-utf-7.c	Fri Mar  3 20:46:49 2006
@@ -230,7 +230,7 @@
             if (*inbytesleft < 2)
                 return UCS_CHAR_NONE;
             needbytes = 1;
-            ch = char_type(*(++((unsigned char *)*inbuf)));
+            ch = char_type(*(++(*inbuf)));
             (*inbytesleft) --;
         case utf7_printable:
             utf7_state[0] = 0;
@@ -246,7 +246,7 @@
             (*inbytesleft) += needbytes;
             return UCS_CHAR_NONE;
         }
-        switch (char_type(*(++(unsigned char *)*inbuf))) {
+        switch (char_type(*(++(*inbuf)))) {
         case utf7_shift_out:
             (*inbuf) ++;
             (*inbytesleft) -= 2;
@@ -264,7 +264,7 @@
         return UCS_CHAR_INVALID;
     }
     (*inbytesleft) --;
-    return *((unsigned char *)*inbuf) ++;
+    return *((*inbuf)++);
 #undef utf7_state
 }
 
