--- ./src/mmstools/mmsfiletransfer.cpp	2011-05-04 17:29:25.614224158 +0200
+++ ../disko-1.7.0/src/mmstools/mmsfiletransfer2.cpp	2011-05-04 17:30:20.886506998 +0200
@@ -92,7 +92,7 @@
 
 
 int progress_callback(void *pclient, double dltotal, double dlnow, double ultotal, double ulnow) {
-	((MMSFiletransfer::MMSFiletransfer*) pclient)->progress.emit(dltotal != 0 ? (int) (.5 + 100* dlnow / dltotal) : (int) (.5 + 100* ulnow / ultotal));
+	((MMSFiletransfer*) pclient)->progress.emit(dltotal != 0 ? (int) (.5 + 100* dlnow / dltotal) : (int) (.5 + 100* ulnow / ultotal));
 	return 0;
 }
 
