--- a/ipu/mxc_ipu_hl_lib.c	2011-04-19 09:27:39.000000000 +0200
+++ b/ipu/mxc_ipu_hl_lib.c	2012-03-11 11:36:20.888697046 +0100
@@ -41,7 +41,7 @@
 #include <sys/mman.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include "mxc_ipu_hl_lib.h"
 
 #define DBG_DEBUG		3
--- a/screenlayer/ScreenLayer.c	2011-04-19 09:27:39.000000000 +0200
+++ b/screenlayer/ScreenLayer.c	2012-03-11 11:35:50.988653193 +0100
@@ -26,7 +26,7 @@
 #include <semaphore.h>
 #include <linux/ipu.h>
 #include <linux/mxcfb.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <sys/stat.h>
