diff -Naur boxee-sources-0.9.23.15885.orig//guilib/Makefile boxee-sources-0.9.23.15885/guilib/Makefile
--- boxee-sources-0.9.23.15885.orig//guilib/Makefile	2011-02-24 19:26:43.051170999 +0100
+++ boxee-sources-0.9.23.15885/guilib/Makefile	2011-02-24 19:54:34.701171001 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -I/usr/include/freetype2 -I/usr/include/SDL -ItinyXML -I../xbmc/lib/libsquish
+INCLUDES=-I. -I../ -Icommon -I../xbmc -I../xbmc/cores -I../xbmc/linux -I../xbmc/utils -ItinyXML -I../xbmc/lib/libsquish
 
 SRCS=AnimatedGif.cpp \
      AudioContext.cpp \
diff -Naur boxee-sources-0.9.23.15885.orig//xbmc/FileSystem/Makefile boxee-sources-0.9.23.15885/xbmc/FileSystem/Makefile
--- boxee-sources-0.9.23.15885.orig//xbmc/FileSystem/Makefile	2011-02-24 19:26:35.221171002 +0100
+++ boxee-sources-0.9.23.15885/xbmc/FileSystem/Makefile	2011-02-24 19:55:25.071171004 +0100
@@ -1,4 +1,4 @@
-INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include
+INCLUDES=-I. -I../ -I../cores -I../linux -I../../guilib -I../lib/UnrarXLib -I../utils
 INCLUDES+=-I../lib/libcdio/libcdio/include
 INCLUDES+=-I../lib/libBoxee
 
diff -Naur boxee-sources-0.9.23.15885.orig//xbmc/Makefile boxee-sources-0.9.23.15885/xbmc/Makefile
--- boxee-sources-0.9.23.15885.orig//xbmc/Makefile	2011-02-24 19:26:43.021170998 +0100
+++ boxee-sources-0.9.23.15885/xbmc/Makefile	2011-02-24 19:54:49.841170997 +0100
@@ -1,4 +1,4 @@
-INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -I/usr/include/freetype2 -Icores/dvdplayer -Icores/ffmpeg
+INCLUDES+=-I. -Ilinux -Icores -I../guilib -Iutils -IFileSystem -Icores/dvdplayer -Icores/ffmpeg
 
 INCLUDES+=-Ilib/libUPnP/Platinum/Source/Core \
           -Ilib/libUPnP/Platinum/Source/Platinum \
