diff -Naur boxee-sources-0.9.23.15885.orig//Makefile.in boxee-sources-0.9.23.15885/Makefile.in
--- boxee-sources-0.9.23.15885.orig//Makefile.in	2011-02-24 19:26:35.101171003 +0100
+++ boxee-sources-0.9.23.15885/Makefile.in	2011-02-24 19:58:55.371170994 +0100
@@ -95,13 +95,12 @@
 
 LIB_DIRS+= xbmc/lib/libBoxee xbmc/lib/libBoxee/tinyxpath 
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
+
 ifneq (@BUILD_GOOM@,1)
 VIS_DIRS+=xbmc/visualizations/Goom
 endif
@@ -292,7 +291,7 @@
 endif
 endif
 screensavers: exports
-	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
+
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
