diff -Naur a/configure b/configure
--- a/configure	2011-02-25 15:41:13.206956215 +0100
+++ b/configure	2011-02-25 15:41:24.423279521 +0100
@@ -6595,8 +6595,6 @@
 	INCLUDES+=" $VORBISENC_CFLAGS"; LIBS+=" $VORBISENC_LIBS"
 fi
 else
-  LIBS+="/usr/lib/libmysqlclient.so.16 "
-  LIBS+="/usr/lib/libcdio.a "
   ac_fn_c_check_header_mongrel "$LINENO" "FLAC/stream_decoder.h" "ac_cv_header_FLAC_stream_decoder_h" "$ac_includes_default"
 if test "x$ac_cv_header_FLAC_stream_decoder_h" = x""yes; then :
 
