diff -Nur samba-3.0.20.orig/source/include/debug.h samba-3.0.20/source/include/debug.h
--- samba-3.0.20.orig/source/include/debug.h	2005-07-28 16:19:49.000000000 +0300
+++ samba-3.0.20/source/include/debug.h	2005-08-23 13:48:45.000000000 +0300
@@ -155,50 +155,16 @@
  * will remove the extra conditional test.
  */
 
-#define DEBUGLVL( level ) \
-  ( ((level) <= MAX_DEBUG_LEVEL) && \
-     ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-     (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-      DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) ) )
-
-
-#define DEBUGLVLC( dbgc_class, level ) \
-  ( ((level) <= MAX_DEBUG_LEVEL) && \
-     ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-     (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-      DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-   && dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) ) )
-
-
-#define DEBUG( level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
-
-#define DEBUGC( dbgc_class, level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-	    DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbghdr( level, __FILE__, FUNCTION_MACRO, (__LINE__) )) \
-       && (dbgtext body) )
-
-#define DEBUGADD( level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-           ((DEBUGLEVEL_CLASS[ DBGC_CLASS ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ DBGC_CLASS ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
-
-#define DEBUGADDC( dbgc_class, level, body ) \
-  (void)( ((level) <= MAX_DEBUG_LEVEL) && \
-          ((DEBUGLEVEL_CLASS[ dbgc_class ] >= (level))||  \
-           (!DEBUGLEVEL_CLASS_ISSET[ dbgc_class ] && \
-            DEBUGLEVEL_CLASS[ DBGC_ALL   ] >= (level))  ) \
-       && (dbgtext body) )
+#define DEBUGLVL( level )  (0)
+
+#define DEBUGLVLC( dbgc_class, level )  (0)
+
+#define DEBUG( level, body )  ((void)0)
+
+#define DEBUGC( dbgc_class, level, body )  ((void)0)
+
+#define DEBUGADD( level, body )  ((void)0)
+
+#define DEBUGADDC( dbgc_class, level, body )  ((void)0)
 
 #endif
diff -Nur samba-3.0.20.orig/source/libsmb/clierror.c samba-3.0.20/source/libsmb/clierror.c
--- samba-3.0.20.orig/source/libsmb/clierror.c	2005-07-28 16:19:46.000000000 +0300
+++ samba-3.0.20/source/libsmb/clierror.c	2005-08-23 13:48:45.000000000 +0300
@@ -134,7 +134,7 @@
 	if (flgs2 & FLAGS2_32_BIT_ERROR_CODES) {
 		NTSTATUS status = NT_STATUS(IVAL(cli->inbuf,smb_rcls));
 
-		return nt_errstr(status);
+		return "NT_STATUS_UNSUCCESSFUL";
         }
 
 	cli_dos_error(cli, &errclass, &errnum);
diff -Nur samba-3.0.23.orig/source/Makefile.in samba-3.0.23/source/Makefile.in
--- samba-3.0.23.orig/source/Makefile.in	2006-07-10 19:27:55.000000000 +0300
+++ samba-3.0.23/source/Makefile.in	2006-07-28 11:26:10.000000000 +0300
@@ -477,12 +477,7 @@
 	$(SECRETS_OBJ) $(LIBSAMBA_OBJ) $(RPC_PARSE_OBJ1) $(DOSERR_OBJ) \
 	$(PRINTBASE_OBJ) $(ERRORMAP_OBJ)
 
-SMBTREE_OBJ = utils/smbtree.o $(PARAM_OBJ) \
-             $(PROFILE_OBJ) $(LIB_NONSMBD_OBJ) $(LIBSMB_OBJ) \
-	     $(KRBCLIENT_OBJ) $(POPT_LIB_OBJ) $(SECRETS_OBJ) \
-             rpc_client/cli_srvsvc.o rpc_parse/parse_srv.o  \
-             rpc_client/cli_pipe.o rpc_parse/parse_rpc.o \
-             rpc_client/cli_netlogon.o rpc_parse/parse_net.o
+SMBTREE_OBJ = utils/smbtree.o $(POPT_LIB_OBJ) $(SECRETS_OBJ)
 
 TESTPARM_OBJ = utils/testparm.o \
                $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) \
@@ -590,8 +585,7 @@
 CUPS_OBJ = client/smbspool.o $(PARAM_OBJ) $(LIBSMB_OBJ) \
 	  $(LIB_NONSMBD_OBJ) $(KRBCLIENT_OBJ) $(SECRETS_OBJ)
 
-MOUNT_OBJ = client/smbmount.o \
-             $(PARAM_OBJ) $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) $(LIB_NONSMBD_OBJ) $(SECRETS_OBJ)
+MOUNT_OBJ = client/smbmount.o
 
 MNT_OBJ = client/smbmnt.o lib/replace.o lib/timegm.o $(VERSION_OBJ) $(SNPRINTF_OBJ) @SOCKWRAP@
 
@@ -947,9 +941,9 @@
 	@echo Linking $@
 	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(CUPS_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) $(KRB5LIBS) $(LDAP_LIBS)
 
-bin/smbmount@EXEEXT@: $(MOUNT_OBJ) bin/.dummy
+bin/smbmount@EXEEXT@: $(MOUNT_OBJ) bin/libsmbclient.so bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) $(KRB5LIBS) $(LDAP_LIBS)
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) $(KRB5LIBS) $(LDAP_LIBS) -Lbin -lsmbclient
 
 bin/smbmnt@EXEEXT@: $(MNT_OBJ) bin/.dummy
 	@echo Linking $@
@@ -982,9 +976,9 @@
 	$(SMBCONTROL_OBJ) $(DYNEXP) $(LDFLAGS) \
 	$(LIBS) @LIBUNWIND_PTRACE@ @POPTLIBS@
 
-bin/smbtree@EXEEXT@: $(SMBTREE_OBJ) @BUILD_POPT@ bin/.dummy
+bin/smbtree@EXEEXT@: $(SMBTREE_OBJ) bin/libsmbclient.so @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(SMBTREE_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) @POPTLIBS@ $(KRB5LIBS) $(LDAP_LIBS)
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(SMBTREE_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) @POPTLIBS@ $(KRB5LIBS) $(LDAP_LIBS) -Lbin -lsmbclient
 
 bin/smbpasswd@EXEEXT@: $(SMBPASSWD_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
@@ -1093,7 +1087,7 @@
 
 bin/libsmbclient.a: $(LIBSMBCLIENT_PICOBJS)
 	@echo Linking libsmbclient non-shared library $@
-	@-$(AR) -rc $@ $(LIBSMBCLIENT_PICOBJS)
+	@-$(AR) -rc $@ $(LIBSMBCLIENT_PICOBJS) $(KRB5LIBS) $(LDAPLIBS)
 
 bin/libsmbsharemodes.@SHLIBEXT@: $(LIBSMBSHAREMODES_PICOBJS)
 	@echo Linking libsmbsharemodes shared library $@
diff -Nur samba-3.0.23a.orig/source/utils/smbtree.c samba-3.0.23a/source/utils/smbtree.c
--- samba-3.0.23a.orig/source/utils/smbtree.c	2006-05-23 21:54:35.000000000 +0300
+++ samba-3.0.23a/source/utils/smbtree.c	2006-07-28 15:17:48.000000000 +0300
@@ -127,6 +127,7 @@
         return True;
 }
 
+#if 0
 static BOOL get_rpc_shares(struct cli_state *cli, 
 			   void (*fn)(const char *, uint32, const char *, void *),
 			   void *state)
@@ -179,6 +180,7 @@
 	cli_rpc_pipe_close(pipe_hnd);
 	return True;
 }
+#endif
 
 
 static BOOL get_shares(char *server_name, struct user_auth_info *user_info)
@@ -188,9 +190,6 @@
         if (!(cli = get_ipc_connect(server_name, NULL, user_info)))
                 return False;
 
-	if (get_rpc_shares(cli, add_name, &shares))
-		return True;
-	
         if (!cli_RNetShareEnum(cli, add_name, &shares))
                 return False;
 
