diff -Naur samba-3.0.25.orig/source/Makefile.in samba-3.0.25/source/Makefile.in
--- samba-3.0.25.orig/source/Makefile.in	2007-05-17 12:28:16.000000000 +0200
+++ samba-3.0.25/source/Makefile.in	2007-05-17 12:37:56.000000000 +0200
@@ -530,12 +530,7 @@
 	$(SECRETS_OBJ) $(LIBSAMBA_OBJ) $(RPC_PARSE_OBJ1) $(DOSERR_OBJ) \
 	$(PRINTBASE_OBJ) $(ERRORMAP_OBJ)
 
-SMBTREE_OBJ = utils/smbtree.o $(PARAM_OBJ) \
-             $(PROFILE_OBJ) $(LIB_NONSMBD_OBJ) $(LIBSMB_OBJ) \
-	     $(KRBCLIENT_OBJ) $(POPT_LIB_OBJ) $(SECRETS_OBJ) \
-             rpc_client/cli_pipe.o $(RPC_PARSE_OBJ2) \
-             $(RPC_CLIENT_OBJ1) \
-	     $(LIBMSRPC_GEN_OBJ)
+SMBTREE_OBJ = utils/smbtree.o $(POPT_LIB_OBJ) $(SECRETS_OBJ)
 
 TESTPARM_OBJ = utils/testparm.o \
                $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) \
@@ -1037,9 +1031,9 @@
 	$(SMBCONTROL_OBJ) $(DYNEXP) $(LDFLAGS) \
 	$(LIBS) $(LDAP_LIBS) @LIBUNWIND_PTRACE@ @POPTLIBS@
 
-bin/smbtree@EXEEXT@: proto_exists $(SMBTREE_OBJ) @BUILD_POPT@ bin/.dummy
+bin/smbtree@EXEEXT@: proto_exists $(SMBTREE_OBJ) bin/libsmbclient.so @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) -o $@ $(SMBTREE_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) @POPTLIBS@ $(KRB5LIBS) $(LDAP_LIBS)
+	@$(CC) $(FLAGS) -o $@ $(SMBTREE_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) @POPTLIBS@ $(KRB5LIBS) $(LDAP_LIBS) -Lbin -lsmbclient
 
 bin/smbpasswd@EXEEXT@: proto_exists $(SMBPASSWD_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
diff -Naur samba-3.0.25.orig/source/utils/smbtree.c samba-3.0.25/source/utils/smbtree.c
--- samba-3.0.25.orig/source/utils/smbtree.c	2007-05-17 12:28:16.000000000 +0200
+++ samba-3.0.25/source/utils/smbtree.c	2007-05-17 12:38:37.000000000 +0200
@@ -127,6 +127,7 @@
         return True;
 }
 
+#if 0
 static BOOL get_rpc_shares(struct cli_state *cli, 
 			   void (*fn)(const char *, uint32, const char *, void *),
 			   void *state)
@@ -179,7 +180,7 @@
 	cli_rpc_pipe_close(pipe_hnd);
 	return True;
 }
-
+#endif
 
 static BOOL get_shares(char *server_name, struct user_auth_info *user_info)
 {
@@ -188,9 +189,6 @@
         if (!(cli = get_ipc_connect(server_name, NULL, user_info)))
                 return False;
 
-	if (get_rpc_shares(cli, add_name, &shares))
-		return True;
-	
         if (!cli_RNetShareEnum(cli, add_name, &shares))
                 return False;
 
