diff -Naur samba-3.0.10.orig/source/configure samba-3.0.10/source/configure
--- samba-3.0.10.orig/source/configure	2004-12-30 02:33:11.254165816 +0100
+++ samba-3.0.10/source/configure	2004-12-30 02:35:04.375968688 +0100
@@ -17313,11 +17313,7 @@
     *linux*)
        # glibc <= 2.3.2 has a broken getgrouplist
        if test "$cross_compiling" = yes; then
-  { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&5
-echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+           linux_getgrouplist_ok=no
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
