--- samba-3.0.8.orig/source/utils/smbtree.c	Fri Dec 24 14:28:46 2004
+++ samba-3.0.8/source/utils/smbtree.c	Fri Dec 24 14:31:00 2004
@@ -151,8 +151,6 @@
 
         for (wg = workgroups; wg; wg = wg->next) {
 
-                printf("%s\n", wg->name);
-
                 /* List servers */
 
                 free_name_list(servers);
@@ -164,9 +162,6 @@
 
                 for (sv = servers; sv; sv = sv->next) {
 
-                        printf("\t\\\\%-15s\t\t%s\n", 
-			       sv->name, sv->comment);
-
                         /* List shares */
 
                         free_name_list(shares);
@@ -177,8 +172,9 @@
                                 continue;
 
                         for (sh = shares; sh; sh = sh->next) {
-                                printf("\t\t\\\\%s\\%-15s\t%s\n", 
-				       sv->name, sh->name, sh->comment);
+			    if (sh->server_type == STYPE_DISKTREE && sh->name[strlen(sh->name)-1] != '$')
+                                printf("%s/%s\n",
+				       sv->name, sh->name);
                         }
                 }
         }
