--- vdr-plugin-xvdr-0.9.9+r00bb4449/src/config/config.c.org	2013-11-08 15:59:15.619666136 +0100
+++ vdr-plugin-xvdr-0.9.9+r00bb4449/src/config/config.c	2013-11-08 16:00:39.631667579 +0100
@@ -45,7 +45,11 @@
 }
 
 void cXVDRServerConfig::Load() {
+#if VDRVERSNUM >= 20102
+  cLiveQueue::SetTimeShiftDir(cVideoDirectory::Name());
+#else
   cLiveQueue::SetTimeShiftDir(VideoDirectory);
+#endif
   cRecordingsCache::GetInstance().LoadResumeData();
 
   if(!cConfig<cSetupLine>::Load(AddDirectory(ConfigDirectory, GENERAL_CONFIG_FILE), true, false))
--- vdr-plugin-xvdr-0.9.9+r00bb4449/src/xvdr/xvdrclient.c.org	2013-11-08 15:59:23.315666269 +0100
+++ vdr-plugin-xvdr-0.9.9+r00bb4449/src/xvdr/xvdrclient.c	2013-11-08 16:01:24.687668353 +0100
@@ -1380,7 +1380,11 @@
 bool cXVDRClient::processRECORDINGS_GetDiskSpace() /* OPCODE 100 */
 {
   int FreeMB;
+#if VDRVERSNUM >= 20102
+  int Percent = cVideoDirectory::VideoDiskSpace(&FreeMB);
+#else
   int Percent = VideoDiskSpace(&FreeMB);
+#endif
   int Total   = (FreeMB / (100 - Percent)) * 100;
 
   m_resp->put_U32(Total);
