diff -Naur ./tests/Makefile.in ../gst-vaapi-0.2.5_patched/tests/Makefile.in
--- ./tests/Makefile.in	2011-05-03 14:54:50.672650473 +0200
+++ ../gst-vaapi-0.2.5_patched/tests/Makefile.in	2011-05-03 14:56:42.365059615 +0200
@@ -302,8 +302,8 @@
 test_windows_CFLAGS = $(TEST_CFLAGS) $(TEST_X11_CFLAGS)
 test_windows_LDADD = $(TEST_LIBS) $(TEST_X11_LIBS)
 test_textures_SOURCES = test-textures.c image.c
-test_textures_CFLAGS = $(TEST_CFLAGS) $(TEST_GLX_CFLAGS)
-test_textures_LDADD = $(TEST_LIBS) $(TEST_GLX_LIBS)
+test_textures_CFLAGS = $(TEST_CFLAGS) $(TEST_MIX_CFLAGS)
+test_textures_LDADD = $(TEST_LIBS) $(TEST_MIX_LIBS)
 EXTRA_DIST = \
 	image.h			\
 	test-decode.h 		\
