--- syslinux/mtools/syslinux.c.orig	2010-03-31 12:24:25.000000000 -0400
+++ syslinux/mtools/syslinux.c	2010-04-03 06:09:04.000000000 -0400
@@ -229,24 +229,6 @@
     fclose(mtc);
 
     /*
-     * Run mtools to create the LDLINUX.SYS file
-     */
-    if (setenv("MTOOLSRC", mtools_conf, 1)) {
-	perror(program);
-	exit(1);
-    }
-
-    /* This command may fail legitimately */
-    system("mattrib -h -r -s s:/ldlinux.sys 2>/dev/null");
-
-    mtp = popen("mcopy -D o -D O -o - s:/ldlinux.sys", "w");
-    if (!mtp || (fwrite(syslinux_ldlinux, 1, syslinux_ldlinux_len, mtp)
-		 != syslinux_ldlinux_len) ||
-	(status = pclose(mtp), !WIFEXITED(status) || WEXITSTATUS(status))) {
-	die("failed to create ldlinux.sys");
-    }
-
-    /*
      * Now, use libfat to create a block map
      */
     fs = libfat_open(libfat_xpread, dev_fd);
