diff -Naur syslinux-2.11.orig/syslinux.c syslinux-2.11/syslinux.c
--- syslinux-2.11.orig/syslinux.c	2004-08-15 21:40:45.000000000 +0200
+++ syslinux-2.11/syslinux.c	2004-10-28 16:29:01.000000000 +0200
@@ -114,13 +114,9 @@
   static unsigned char sectbuf[512];
   int dev_fd;
   struct stat st;
-  int status;
   char **argp, *opt;
   int force = 0;		/* -f (force) option */
   off_t offset = 0;		/* -o (offset) option */
-  char mtools_conf[] = "/tmp/syslinux-mtools-XXXXXX";
-  int mtc_fd;
-  FILE *mtc, *mtp;
 
   mypid = getpid();
   program = argv[0];
@@ -183,55 +179,6 @@
   if(!syslinux_check_bootsect(sectbuf,device)) {
     exit(1);
   }
-
-  /*
-   * Create an mtools configuration file
-   */
-  mtc_fd = mkstemp(mtools_conf);
-  if ( mtc_fd < 0 || !(mtc = fdopen(mtc_fd, "w")) ) {
-    perror(program);
-    exit(1);
-  }
-  fprintf(mtc,
-	  "MTOOLS_NO_VFAT=1\n"
-	  "MTOOLS_SKIP_CHECK=1\n" /* Needed for some flash memories */
-	  "drive s:\n"
-	  "  file=\"/proc/%lu/fd/%d\"\n"
-	  "  offset=%lld\n",
-	  (unsigned long)mypid,
-	  dev_fd,
-	  (unsigned long long)offset);
-  fclose(mtc);
-  
-  /*
-   * Run mtools to create the LDLINUX.SYS file
-   */
-  if ( setenv("MTOOLSRC", mtools_conf, 1) ) {
-    perror(program);
-    exit(1);
-  }
-
-  /* This command may fail legitimately */
-  system("mattrib -h -r -s s:ldlinux.sys 2>/dev/null");
-
-  mtp = popen("mcopy -D o -D O -o - s:ldlinux.sys", "w");
-  if ( !mtp ||
-       (fwrite(syslinux_ldlinux, 1, syslinux_ldlinux_len, mtp) 
-	!= syslinux_ldlinux_len) ||
-       (status = pclose(mtp), !WIFEXITED(status) || WEXITSTATUS(status)) ) {
-    fprintf(stderr, "%s: failed to create ldlinux.sys\n", program);
-    exit(1);
-  }
-
-  status = system("mattrib +r s:ldlinux.sys");
-
-  if ( !WIFEXITED(status) || WEXITSTATUS(status) ) {
-    fprintf(stderr,
-	    "%s: warning: failed to set readonly bit on ldlinux.sys\n",
-	    program);
-  }
-
-  unlink(mtools_conf);
 
   /*
    * To finish up, write the boot sector
