diff -Naur dbus-1.3.0.orig/configure dbus-1.3.0/configure
--- dbus-1.3.0.orig/configure	2009-10-19 19:19:52.000000000 +0200
+++ dbus-1.3.0/configure	2009-10-19 19:19:59.000000000 +0200
@@ -15960,7 +15960,7 @@
     ;;
   esac
 
-  sys_lib_dlsearch_path_spec="/lib${libsuff} /usr/lib${libsuff}"
+  sys_lib_dlsearch_path_spec=""
 
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
