diff -Naur gcc-4.6.1.orig//configure gcc-4.6.1/configure
--- gcc-4.6.1.orig//configure	2011-03-16 19:27:36.000000000 +0100
+++ gcc-4.6.1/configure	2011-07-08 19:07:53.008474073 +0200
@@ -5770,7 +5770,7 @@
     LDFLAGS="$saved_LDFLAGS"
   fi
 
-  ppllibs="$ppllibs -lppl_c -lppl $pwllib -lgmpxx"
+  ppllibs="$ppllibs -lppl_c -lppl -lpwl $pwllib -lgmpxx"
 
   if test "$enable_ppl_version_check" != no; then
     saved_CFLAGS="$CFLAGS"
