diff -Naur a/freetds.conf b/freetds.conf
--- a/freetds.conf	2024-08-06 10:07:00.799906813 +0200
+++ b/freetds.conf	2024-08-06 10:07:54.787644635 +0200
@@ -11,6 +11,9 @@
         # TDS protocol version
 	tds version = auto
 
+        # enable legacy protocol version
+	enable tls v1 = yes
+
 	# Whether to write a TDSDUMP file for diagnostic purposes
 	# (setting this to /tmp is insecure on a multi-user system)
 ;	dump file = /tmp/freetds.log
