--- a/Makefile	2014-01-27 13:26:26.000000000 +0100
+++ b/Makefile	2014-08-26 22:39:00.214668468 +0200
@@ -19 +19 @@ VPATH=src $(wildcard src/cpu/*)
-MD = @mkdir
+MD = mkdir
@@ -26,0 +27,2 @@ STRIP = strip
+SDL_INSTALL_ROOT=/usr
+
@@ -34,4 +36,6 @@ CFLAGS = -fsigned-char $(DEVLIBS) \
-	-I/usr/include/SDL \
-	-I$(SDKSTAGE)/opt/vc/include -I$(SDKSTAGE)/opt/vc/include/interface/vcos/pthreads \
-	-I$(SDKSTAGE)/opt/vc/include/interface/vmcs_host/linux \
-	-I/usr/include/glib-2.0 -I/usr/lib/arm-linux-gnueabihf/glib-2.0/include \
+	-I$(SDL_INSTALL_ROOT)/include/SDL \
+	-I$(SDKSTAGE)/include/interface/vcos \
+	-I$(SDKSTAGE)/include/interface/vcos/pthreads \
+	-I$(SDKSTAGE)/include/interface/vmcs_host/linux \
+	-I$(SDKSTAGE)/include/glib-2.0 \
+	-I$(SDKSTAGE)/lib/glib-2.0/include \
