--- vdr-1.7.36.orig/ci.c	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/ci.c	2013-01-26 15:56:02.000000000 +0100
@@ -1914,6 +1914,10 @@
          AddPid(Channel->Sid(), *Dpid, STREAM_TYPE_PRIVATE);
      for (const int *Spid = Channel->Spids(); *Spid; Spid++)
          AddPid(Channel->Sid(), *Spid, STREAM_TYPE_PRIVATE);
+#ifdef USE_TTXTSUBS
+     if (Channel->Tpid() && Setup.SupportTeletext)
+        AddPid(Channel->Sid(), Channel->Tpid(), STREAM_TYPE_PRIVATE);
+#endif // USE_TTXTSUBS
      }
 }
 
@@ -1937,6 +1941,11 @@
          CaPmt.AddPid(*Dpid, STREAM_TYPE_PRIVATE);
      for (const int *Spid = Channel->Spids(); *Spid; Spid++)
          CaPmt.AddPid(*Spid, STREAM_TYPE_PRIVATE); 
+#ifdef USE_TTXTSUBS
+     if (Channel->Tpid() && Setup.SupportTeletext) {
+        CaPmt.AddPid(Channel->Tpid(), STREAM_TYPE_PRIVATE);
+        }
+#endif // USE_TTXTSUBS
      cas->SendPMT(&CaPmt);
      cTimeMs Timeout(QUERY_REPLY_TIMEOUT);
      do {
