--- vdr-1.7.36.orig/config.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/config.h	2013-01-26 15:56:02.000000000 +0100
@@ -30,12 +30,34 @@
 #define APIVERSION  "1.7.36"
 #define APIVERSNUM   10736  // Version * 10000 + Major * 100 + Minor
 
+#ifdef USE_YAEPG
+#define YAEPGHDVERSNUM 1
+#endif /* YAEPG */
+
 // When loading plugins, VDR searches them by their APIVERSION, which
 // may be smaller than VDRVERSION in case there have been no changes to
 // VDR header files since the last APIVERSION. This allows compiled
 // plugins to work with newer versions of the core VDR as long as no
 // VDR header files have changed.
 
+#ifdef USE_CHANNELBIND
+#define CHANNELBINDINGVERSNUM 2
+#endif /* CHANNELBIND */
+
+#ifdef USE_MAINMENUHOOKS
+// The MainMenuHook Patch's version number:
+#define MAINMENUHOOKSVERSION "1.0.1"
+#define MAINMENUHOOKSVERSNUM 10001  // Version * 10000 + Major * 100 + Minor
+#endif /* MAINMENUHOOKS */
+
+#ifdef USE_JUMPPLAY
+#define JUMPPLAYVERSNUM 100
+#endif /* JUMPPLAY */
+
+//#ifdef USE_VASARAJANAULOJA
+//#define LIEMIKUUTIO  134
+//#endif /* VASARAJANAULOJA */
+
 #define MAXPRIORITY       99
 #define MINPRIORITY       (-MAXPRIORITY)
 #define LIVEPRIORITY      0                  // priority used when selecting a device for live viewing
@@ -44,6 +66,10 @@
 #define MAXLIFETIME       99
 #define DEFINSTRECTIME    180 // default instant recording time (minutes)
 
+#ifdef USE_DVLVIDPREFER
+#define DVLVIDPREFER_MAX 12
+#endif /* DVLVIDPREFER */
+
 #define MINOSDWIDTH   480
 #define MAXOSDWIDTH  1920
 #define MINOSDHEIGHT  324
@@ -250,6 +276,9 @@
   char OSDLanguage[I18N_MAX_LOCALE_LEN];
   char OSDSkin[MaxSkinName];
   char OSDTheme[MaxThemeName];
+#ifdef USE_WAREAGLEICON
+  int WarEagleIcons;
+#endif /* WAREAGLEICON */
   int PrimaryDVB;
   int ShowInfoOnChSwitch;
   int TimeoutRequChInfo;
@@ -270,10 +299,19 @@
   int MarginStart, MarginStop;
   int AudioLanguages[I18N_MAX_LANGUAGES + 1];
   int DisplaySubtitles;
+#ifdef USE_TTXTSUBS
+  int SupportTeletext;
+#endif // USE_TTXTSUBS
   int SubtitleLanguages[I18N_MAX_LANGUAGES + 1];
   int SubtitleOffset;
   int SubtitleFgTransparency, SubtitleBgTransparency;
   int EPGLanguages[I18N_MAX_LANGUAGES + 1];
+#ifdef USE_DDEPGENTRY
+  int DoubleEpgTimeDelta;
+  int DoubleEpgAction;
+  int MixEpgAction;
+  int DisableVPS;
+#endif /* DDEPPGENTRY */
   int EPGScanTimeout;
   int EPGBugfixLevel;
   int EPGLinger;
@@ -293,6 +331,9 @@
   int VideoDisplayFormat;
   int VideoFormat;
   int UpdateChannels;
+#ifdef USE_CHANNELBIND
+  int ChannelBindingByRid;
+#endif /* CHANNELBIND */
   int UseDolbyDigital;
   int ChannelInfoPos;
   int ChannelInfoTime;
@@ -314,6 +355,9 @@
   int MaxVideoFileSize;
   int SplitEditedFiles;
   int DelTimeshiftRec;
+#ifdef USE_NALUDUMP
+  int DumpNaluFill;
+#endif // USE_NALUDUMP
   int MinEventTimeout, MinUserInactivity;
   time_t NextWakeupTime;
   int MultiSpeedMode;
@@ -323,12 +367,38 @@
   int PauseOnMarkSet;
   int ResumeID;
   int CurrentChannel;
+#ifdef USE_JUMPPLAY
+  int JumpPlay;
+  int PlayJump;
+  int PauseLastMark;
+#endif /* JUMPPLAY */
   int CurrentVolume;
   int CurrentDolby;
   int InitialVolume;
+#ifdef USE_VOLCTRL
+  int LRVolumeControl;
+  int LRChannelGroups;
+  int LRForwardRewind;
+#endif // USE_VOLCTRL
   int ChannelsWrap;
   int ShowChannelNamesWithSource;
   int EmergencyExit;
+#ifdef USE_JUMPINGSECONDS
+  int JumpSeconds;
+  int JumpSecondsSlow;
+  int JumpSecondsRepeat;
+#endif // USE_JUMPINGSECONDS
+#ifdef USE_LIRCSETTINGS
+  int LircRepeatDelay;
+  int LircRepeatFreq;
+  int LircRepeatTimeout;
+#endif /* LIRCSETTINGS */
+#ifdef USE_DVLVIDPREFER
+  int UseVidPrefer;  // 0 = VDR's default, 1 = use
+  int nVidPrefer;
+  int VidPreferPrio[DVLVIDPREFER_MAX];
+  int VidPreferSize[DVLVIDPREFER_MAX];
+#endif /* DVLVIDPREFER */
   int __EndData__;
   cString InitialChannel;
   cString DeviceBondings;
