--- vdr-1.7.36.orig/cutter.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/cutter.h	2013-01-26 15:56:02.000000000 +0100
@@ -24,7 +24,11 @@
   static bool error;
   static bool ended;
 public:
+#ifdef USE_VASARAJANAULOJA
+  static bool Start(const char *FileName, const char *TargetFileName = NULL, bool Overwrite = true);
+#else
   static bool Start(const char *FileName);
+#endif /* VASARAJANAULOJA */
   static void Stop(void);
   static bool Active(const char *FileName = NULL);
          ///< Returns true if the cutter is currently active.
