--- vdr-1.7.36.orig/dvbdevice.h	2013-01-26 15:55:50.000000000 +0100
+++ vdr-1.7.36/dvbdevice.h	2013-01-26 15:56:02.000000000 +0100
@@ -193,6 +193,9 @@
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
 public:
   virtual bool HasLock(int TimeoutMs = 0);
+#ifdef USE_ROTOR
+  virtual bool SendDiseqcCmd(dvb_diseqc_master_cmd cmd);
+#endif /* ROTOR */
 
 // PID handle facilities
 
