--- vdr-1.7.36.orig/epg.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/epg.h	2013-01-26 15:56:02.000000000 +0100
@@ -164,6 +164,9 @@
   void DropOutdated(time_t SegmentStart, time_t SegmentEnd, uchar TableID, uchar Version);
   void Cleanup(time_t Time);
   void Cleanup(void);
+#ifdef USE_DDEPGENTRY
+  const cEvent *GetPreviousEvent(cEvent *Event) const; //:EW
+#endif /* DDEPGENTRY */
   cEvent *AddEvent(cEvent *Event);
   void DelEvent(cEvent *Event);
   void HashEvent(cEvent *Event);
