--- vdr-1.7.36.orig/MANUAL	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/MANUAL	2013-01-26 15:56:02.000000000 +0100
@@ -756,6 +756,9 @@
                          background transparency. By default the values as broadcast
                          are used.
 
+  Record Teletext Subtitles = no
+                         If set to 'yes', teletext subtitles will be recorded.
+
   LNB:
 
   Use DiSEqC = no        Generally turns DiSEqC support on or off.
