--- vdr-1.7.36.orig/osd.c	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/osd.c	2013-01-26 15:56:02.000000000 +0100
@@ -1623,6 +1623,10 @@
 int cOsd::osdHeight = 0;
 cVector<cOsd *> cOsd::Osds;
 cMutex cOsd::mutex;
+#ifdef USE_PINPLUGIN
+bool cOsd::pinValid = false;
+#endif /* PINPLUGIN */
+
 
 cOsd::cOsd(int Left, int Top, uint Level)
 {
@@ -1637,6 +1641,9 @@
   width = height = 0;
   level = Level;
   active = false;
+#ifdef USE_YAEPG
+  vidWin.bpp = 0;
+#endif /* YAEPG */
   for (int i = 0; i < Osds.Size(); i++) {
       if (Osds[i]->level > level) {
          Osds.Insert(this, i);
