--- vdr-1.7.36.orig/osd.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/osd.h	2013-01-26 15:56:02.000000000 +0100
@@ -730,6 +730,10 @@
   int left, top, width, height;
   uint level;
   bool active;
+#ifdef USE_YAEPG
+public:
+  tArea vidWin;
+#endif /* YAEPG */
 protected:
   cOsd(int Left, int Top, uint Level);
        ///< Initializes the OSD with the given coordinates.
@@ -931,6 +935,9 @@
        ///<        MyOsdDrawPixmap(Left() + pm->ViewPort().X(), Top() + pm->ViewPort().Y(), pm->Data(), w, h, h * d);
        ///<        delete pm;
        ///<        }
+#ifdef USE_PINPLUGIN
+  static bool pinValid;
+#endif /* PINPLUGIN */
   };
 
 #define MAXOSDIMAGES 64
