--- vdr-1.7.36.orig/plugin.h	2013-01-26 15:55:50.000000000 +0100
+++ vdr-1.7.36/plugin.h	2013-01-26 15:56:02.000000000 +0100
@@ -47,6 +47,9 @@
 
   virtual const char *MainMenuEntry(void);
   virtual cOsdObject *MainMenuAction(void);
+#ifdef USE_MCLI
+  virtual cOsdObject *AltMenuAction(void) { return NULL; };
+#endif /* MCLI */
 
   virtual cMenuSetupPage *SetupMenu(void);
   virtual bool SetupParse(const char *Name, const char *Value);
