--- vdr-1.7.36.orig/receiver.c	2013-01-26 15:55:50.000000000 +0100
+++ vdr-1.7.36/receiver.c	2013-01-26 15:56:02.000000000 +0100
@@ -67,7 +67,12 @@
             (Channel->Ppid() == Channel->Vpid() || AddPid(Channel->Ppid())) &&
             AddPids(Channel->Apids()) &&
             AddPids(Channel->Dpids()) &&
+#ifdef USE_TTXTSUBS
+            AddPids(Channel->Spids()) &&
+            (!Setup.SupportTeletext || AddPid(Channel->Tpid()));
+#else
             AddPids(Channel->Spids());
+#endif // USE_TTXTSUBS
      }
   return true;
 }
