--- vdr-1.7.36.orig/recorder.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/recorder.h	2013-01-26 15:56:02.000000000 +0100
@@ -21,6 +21,9 @@
   cRingBufferLinear *ringBuffer;
   cFrameDetector *frameDetector;
   cPatPmtGenerator patPmtGenerator;
+#ifdef USE_NALUDUMP
+  cNaluStreamProcessor *naluStreamProcessor;
+#endif // USE_NALUDUMP
   cFileName *fileName;
   cIndexFile *index;
   cUnbufferedFile *recordFile;
