--- vdr-1.7.36.orig/recording.c	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/recording.c	2013-01-26 15:56:02.000000000 +0100
@@ -8,6 +8,9 @@
  */
 
 #include "recording.h"
+#ifdef USE_WAREAGLEICON
+#include "iconpatch.h"
+#endif /* WAREAGLEICON */
 #include <ctype.h>
 #include <dirent.h>
 #include <errno.h>
@@ -881,7 +884,11 @@
 
 const char *cRecording::Title(char Delimiter, bool NewIndicator, int Level) const
 {
+#ifdef USE_WAREAGLEICON
+  const char *New = NewIndicator && IsNew() ? Setup.WarEagleIcons ? IsLangUtf8() ? ICON_NEW_UTF8 : ICON_NEW : "*" : " ";
+#else
   char New = NewIndicator && IsNew() ? '*' : ' ';
+#endif /* WAREAGLEICON */
   free(titleBuffer);
   titleBuffer = NULL;
   if (Level < 0 || Level == HierarchyLevels()) {
@@ -901,7 +908,11 @@
                    Minutes % 60
                    );
         }
+#ifdef USE_WAREAGLEICON
+     titleBuffer = strdup(cString::sprintf("%02d.%02d.%02d%c%02d:%02d%s%s%c%s",
+#else
      titleBuffer = strdup(cString::sprintf("%02d.%02d.%02d%c%02d:%02d%s%c%c%s",
+#endif /* WAREAGLEICON */
                             t->tm_mday,
                             t->tm_mon + 1,
                             t->tm_year % 100,
@@ -951,6 +962,18 @@
   return NULL;
 }
 
+#ifdef USE_VASARAJANAULOJA
+const char *cRecording::UpdateFileName(const char *FileName)
+{
+	if (FileName && *FileName) {
+		free(fileName);
+		fileName = strdup(FileName);
+		return fileName;
+	}
+	return NULL;
+}
+#endif /* VASARAJANAULOJA */
+
 int cRecording::HierarchyLevels(void) const
 {
   const char *s = name;
@@ -1243,7 +1266,11 @@
      }
 }
 
+#ifdef USE_VASARAJANAULOJA
+void cRecordings::DelByName(const char *FileName, bool RemoveRecording)
+#else
 void cRecordings::DelByName(const char *FileName)
+#endif /* VASARAJANAULOJA */
 {
   LOCK_THREAD;
   cRecording *recording = GetByName(FileName);
@@ -1251,7 +1278,11 @@
      cThreadLock DeletedRecordingsLock(&DeletedRecordings);
      Del(recording, false);
      char *ext = strrchr(recording->fileName, '.');
+#ifdef USE_VASARAJANAULOJA
+	 if (ext && RemoveRecording) {
+#else
      if (ext) {
+#endif /* VASARAJANAULOJA */
         strncpy(ext, DELEXT, strlen(ext));
         if (access(recording->FileName(), F_OK) == 0) {
            recording->deleted = time(NULL);
