--- vdr-1.7.36.orig/shutdown.c	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/shutdown.c	2013-01-26 15:56:02.000000000 +0100
@@ -17,6 +17,9 @@
 #include "channels.h"
 #include "config.h"
 #include "cutter.h"
+#ifdef USE_VASARAJANAULOJA
+#include "filetransfer.h"
+#endif /* VASARAJANAULOJA */
 #include "i18n.h"
 #include "interface.h"
 #include "menu.h"
@@ -166,6 +169,12 @@
      if (!Interactive || !Interface->Confirm(tr("Editing - shut down anyway?")))
         return false;
      }
+#ifdef USE_VASARAJANAULOJA
+  if (cFileTransfer::Active()) {
+	  if (!Interactive || !Interface->Confirm(tr("Transfering file - shut down anyway?")))
+		  return false;
+  }
+#endif /* VASARAJANAULOJA */
 
   cTimer *timer = Timers.GetNextActiveTimer();
   time_t Next = timer ? timer->StartTime() : 0;
@@ -209,6 +218,12 @@
      if (!Interactive || !Interface->Confirm(tr("Editing - restart anyway?")))
         return false;
      }
+#ifdef USE_VASARAJANAULOJA
+  if (cFileTransfer::Active()) {
+	  if (!Interactive || !Interface->Confirm(tr("Transfering file - restart anyway?")))
+		  return false;
+  }
+#endif /* VASARAJANAULOJA */
 
   cTimer *timer = Timers.GetNextActiveTimer();
   time_t Next  = timer ? timer->StartTime() : 0;
