--- vdr-1.7.36.orig/svdrp.h	2013-01-26 15:55:49.000000000 +0100
+++ vdr-1.7.36/svdrp.h	2013-01-26 15:56:02.000000000 +0100
@@ -56,6 +56,9 @@
   void PrintHelpTopics(const char **hp);
   void CmdCHAN(const char *Option);
   void CmdCLRE(const char *Option);
+#ifdef USE_VASARAJANAULOJA
+  void CmdCPYR(const char *Option);
+#endif /* VASARAJANAULOJA */
   void CmdDELC(const char *Option);
   void CmdDELR(const char *Option);
   void CmdDELT(const char *Option);
@@ -71,6 +74,9 @@
   void CmdMODC(const char *Option);
   void CmdMODT(const char *Option);
   void CmdMOVC(const char *Option);
+#ifdef USE_VASARAJANAULOJA
+  void CmdMOVR(const char *Option);
+#endif /* VASARAJANAULOJA */
   void CmdNEWC(const char *Option);
   void CmdNEWT(const char *Option);
   void CmdNEXT(const char *Option);
