--- vdr-1.7.36.orig/videodir.h	2013-01-26 15:55:50.000000000 +0100
+++ vdr-1.7.36/videodir.h	2013-01-26 15:56:02.000000000 +0100
@@ -23,6 +23,9 @@
 bool VideoFileSpaceAvailable(int SizeMB);
 int VideoDiskSpace(int *FreeMB = NULL, int *UsedMB = NULL); // returns the used disk space in percent
 cString PrefixVideoFileName(const char *FileName, char Prefix);
+#ifdef USE_VASARAJANAULOJA
+cString NewVideoFileName(const char *FileName, const char *NewDirName);
+#endif /* VASARAJANAULOJA */
 void RemoveEmptyVideoDirectories(const char *IgnoreFiles[] = NULL);
 bool IsOnVideoDirectoryFileSystem(const char *FileName);
 
