--- a/include/VG/openvg.h	2018-12-29 20:12:07.618938044 +0100
+++ b/include/VG/openvg.h	2018-12-29 20:12:07.670937601 +0100
@@ -706,9 +706,9 @@
                                          VGboolean allowAutoHinting) VG_API_EXIT;
 VG_API_CALL void VG_API_ENTRY vgDrawGlyphs(VGFont font,
                                           VGint glyphCount,
-                                          VGuint *glyphIndices,
-                                          VGfloat *adjustments_x,
-                                          VGfloat *adjustments_y,
+                                          const VGuint *glyphIndices,
+                                          const VGfloat *adjustments_x,
+                                          const VGfloat *adjustments_y,
                                           VGbitfield paintModes,
                                           VGboolean allowAutoHinting) VG_API_EXIT;

