--- vdr-upnp-1.0.0+aa99a1555aed/plugins/profiler/vdrDVBProfiler/Makefile.org	2013-02-01 16:07:09.682950854 +0100
+++ vdr-upnp-1.0.0+aa99a1555aed/plugins/profiler/vdrDVBProfiler/Makefile	2013-02-01 16:05:43.274949371 +0100
@@ -13,6 +13,7 @@
 
 ROOTBUILDDIR = ../../..
 -include ../../../Make.config
+INCLUDES += -I../$(VDRDIR)/include -I$(DVBDIR)/include -I../$(VDRDIR) -I./src -I.
 
 all:
 	@$(MAKE) -f ../../../Makefile.plugins -C "$(shell pwd)" _all
