--- vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/fileProvider/Makefile.org	2013-02-01 16:10:08.966953931 +0100
+++ vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/fileProvider/Makefile	2013-02-01 16:10:27.094954242 +0100
@@ -13,6 +13,7 @@
 
 ROOTBUILDDIR = ../../..
 -include ../../../Make.config
+INCLUDES += -I../$(VDRDIR)/include -I$(DVBDIR)/include -I../$(VDRDIR) -I./src -I.
 
 all:
 	@$(MAKE) -f ../../../Makefile.plugins -C "$(shell pwd)" _all
