--- vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/recProvider/Makefile.org	2013-02-01 16:11:43.266955550 +0100
+++ vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/recProvider/Makefile	2013-02-01 16:12:02.202955876 +0100
@@ -13,7 +13,7 @@
 
 ROOTBUILDDIR = ../../..
 -include ../../../Make.config
-
+INCLUDES += -I../$(VDRDIR)/include -I$(DVBDIR)/include -I../$(VDRDIR) -I./src -I.
 all:
 	@$(MAKE) -f ../../../Makefile.plugins -C "$(shell pwd)" _all
 
