--- vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/vdrProvider/Makefile.org	2013-02-01 16:13:12.462957082 +0100
+++ vdr-upnp-1.0.0+aa99a1555aed/plugins/provider/vdrProvider/Makefile	2013-02-01 16:13:20.542957221 +0100
@@ -13,6 +13,7 @@
 
 ROOTBUILDDIR = ../../..
 -include ../../../Make.config
+INCLUDES += -I../$(VDRDIR)/include -I$(DVBDIR)/include -I../$(VDRDIR) -I./src -I.
 
 all:
 	@$(MAKE) -f ../../../Makefile.plugins -C "$(shell pwd)" _all
