--- a/os_dep/linux/os_intfs.c	2018-10-14 09:12:28.954763802 +0200
+++ b/os_dep/linux/os_intfs.c	2018-10-14 09:16:08.317612220 +0200
@@ -2746,7 +2746,7 @@
 #ifdef CONFIG_IOCTL_CFG80211
 	rtw_scan_abort(padapter);
 	wdev_to_priv(padapter->rtw_wdev)->bandroid_scan = _FALSE;
-	padapter->rtw_wdev->iftype = NL80211_IFTYPE_MONITOR; //set this at the end
+	//padapter->rtw_wdev->iftype = NL80211_IFTYPE_MONITOR; //set this at the end
 #endif //CONFIG_IOCTL_CFG80211

 	RT_TRACE(_module_os_intfs_c_,_drv_info_,("-871x_drv - drv_close\n"));
