--- a/os_dep/linux/os_intfs.c	2018-10-14 09:12:28.954763802 +0200
+++ b/os_dep/linux/os_intfs.c	2018-10-14 09:16:08.317612220 +0200
@@ -1146,16 +1146,16 @@
 		if (padapter == NULL)
 			continue;
 
+		#ifdef CONFIG_IOCTL_CFG80211
+		rtw_wdev_unregister(padapter->rtw_wdev);
+		#endif
+
 		pnetdev = padapter->pnetdev;
 
 		if((padapter->DriverState != DRIVER_DISAPPEAR) && pnetdev) {
 			unregister_netdev(pnetdev); //will call netdev_close()
 			rtw_proc_remove_one(pnetdev);
 		}
-
-		#ifdef CONFIG_IOCTL_CFG80211
-		rtw_wdev_unregister(padapter->rtw_wdev);
-		#endif
 	}
 
 }
