--- goom2k4-0.orig/src/xmmx.c	2008-02-15 14:44:04.000000000 +0100
+++ goom2k4-0/src/xmmx.c	2008-02-15 14:44:11.000000000 +0100
@@ -239,7 +239,7 @@
 
 		++loop;
 	}
-	__asm__ __volatile__ ("femms\n");
+	__asm__ __volatile__ ("emms\n");
 }
 
 #define DRAWMETHOD_PLUS_XMMX(_out,_backbuf,_col) \
@@ -387,7 +387,7 @@
 		}
 	}
 end_of_line:
-	__asm__ __volatile__ ("femms\n"); 
+	__asm__ __volatile__ ("emms\n"); 
 }
 
 #endif
