diff --git a/locale/C-translit.h.in b/locale/C-translit.h.in
index ab0b130..5286802 100644
--- a/locale/C-translit.h.in
+++ b/locale/C-translit.h.in
@@ -103,6 +103,7 @@
 "\x2061"	""	/* <U2061> FUNCTION APPLICATION */
 "\x2062"	""	/* <U2062> INVISIBLE TIMES */
 "\x2063"	""	/* <U2063> INVISIBLE SEPARATOR */
+"\x20a1"	"CRC"	/* <U20A1> COLON SIGN */
 "\x20a8"	"Rs"	/* <U20A8> RUPEE SIGN */
 "\x20ac"	"EUR"	/* <U20AC> EURO SIGN */
 "\x2100"	"a/c"	/* <U2100> ACCOUNT OF */
