diff --git a/localedata/locales/translit_neutral b/localedata/locales/translit_neutral
index 5883d28..5712361 100644
--- a/localedata/locales/translit_neutral
+++ b/localedata/locales/translit_neutral
@@ -128,6 +128,8 @@ include "translit_wide";""
 <U2063> ""
 % DONG SIGN
 <U20AB> "<U0110><U1ED3><U006E><U0067>"
+% COLON SIGN
+<U20A1> "<U0043><U0052><U0043>"
 % EURO SIGN
 <U20AC> "<U0045><U0055><U0052>"
 % TRADE MARK SIGN
