diff -Naur glib-2.12.4.orig/configure glib-2.12.4/configure
--- glib-2.12.4.orig/configure	2006-11-23 23:20:52.000000000 +0100
+++ glib-2.12.4/configure	2006-11-23 23:21:15.000000000 +0100
@@ -4824,20 +4824,6 @@
 echo "${ECHO_T}no" >&6
 fi
 
-if test x$PKG_CONFIG = xno ; then
-  { { echo "$as_me:$LINENO: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&5
-echo "$as_me: error: *** pkg-config not found. See http://www.freedesktop.org/software/pkgconfig/" >&2;}
-   { (exit 1); exit 1; }; }
-fi
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.14 ; then
-  :
-else
-  { { echo "$as_me:$LINENO: error: *** pkg-config too old; version 0.14 or better required." >&5
-echo "$as_me: error: *** pkg-config too old; version 0.14 or better required." >&2;}
-   { (exit 1); exit 1; }; }
-fi
-
 if test "x$enable_debug" = "xyes"; then
   if test x$cflags_set != xset ; then
       case " $CFLAGS " in
