--- GLib-2.32.3/autogen.sh.orig	2012-07-05 11:40:17.582579288 +0300
+++ GLib-2.32.3/autogen.sh	2012-07-05 11:40:33.262658200 +0300
@@ -7,12 +7,6 @@
 olddir=`pwd`
 cd "$srcdir"
 
-GTKDOCIZE=`which gtkdocize`
-if test -z $GTKDOCIZE; then
-        echo "*** No GTK-Doc found, please install it ***"
-        exit 1
-fi
-
 AUTORECONF=`which autoreconf`
 if test -z $AUTORECONF; then
         echo "*** No autoreconf found, please install it ***"
@@ -24,7 +18,6 @@
 # regenerated from their corresponding *.in files by ./configure anyway.
 touch README INSTALL
 
-gtkdocize || exit $?
 autoreconf --force --install --verbose || exit $?
 
 cd "$olddir"
