diff -Naur glib-2.12.4.orig/configure glib-2.12.4/configure
--- glib-2.12.4.orig/configure	2006-11-23 23:20:52.000000000 +0100
+++ glib-2.12.4/configure	2006-11-23 23:21:15.000000000 +0100
@@ -27609,7 +27595,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-        if test "$cross_compiling" = yes; then
+        if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -30441,7 +30427,7 @@
 if test "${glib_cv_monotonic_clock+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -33223,7 +33209,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-		if test "$cross_compiling" = yes; then
+		if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -34382,7 +34368,7 @@
 if test "${ac_cv_func_posix_getpwuid_r+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  if test "$cross_compiling" = yes; then
+  if test "$cross_compiling" = dummy; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
