--- glib-2.12.4.orig/glib/gnulib/vasnprintf.c	2006-11-29 17:52:03.000000000 +0100
+++ glib-2.12.4/glib/gnulib/vasnprintf.c	2006-11-29 17:52:07.000000000 +0100
@@ -50,6 +50,8 @@
 # define freea(p) free (p) 
 #endif
 
+#undef HAVE_WCHAR_T
+
 #ifndef HAVE_LONG_LONG_FORMAT
 static int
 print_long_long (char *buf, 

--- glib-2.12.4.orig/glib/gtimer.c	2006-11-29 17:53:11.000000000 +0100
+++ glib-2.12.4/glib/gtimer.c	2006-11-29 17:53:17.000000000 +0100
@@ -56,6 +56,9 @@
 #define USE_CLOCK_GETTIME 1
 #endif
 
+#define USE_CLOCK_GETTIME 0
+#define HAVE_CLOCK_GETTIME 0
+
 struct _GTimer
 {
 #ifdef G_OS_WIN32
