From 1ad25ad5365d99b1523d90dec2884e463dba2a81 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Wed, 2 Oct 2013 08:59:30 +0200
Subject: [PATCH] suppress 'mangling has changed...' warning

---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 9b2ac34..772ca7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -342,7 +342,7 @@ fi
 LIBS_LIBCEC="$LIBS"
 LIBS="$libs_client"
 
-CXXFLAGS="$CXXFLAGS -fPIC -Wall -Wextra -Wno-missing-field-initializers"
+CXXFLAGS="$CXXFLAGS -fPIC -Wall -Wextra -Wno-missing-field-initializers -Wno-psabi"
 
 if test "x$use_debug" = "xyes"; then
   CXXFLAGS="$CXXFLAGS -g"
-- 
1.7.9.5

