--- a/src/lib/adapter/AdapterFactory.cpp	2014-02-17 20:59:29.106696155 +0100
+++ b/src/lib/adapter/AdapterFactory.cpp	2014-02-17 20:59:29.114696156 +0100
@@ -149,7 +149,8 @@
 #endif
 
 #if defined(HAVE_IMX_API)
-  return new CIMXCECAdapterCommunication(m_lib->m_cec);
+  if (!strcmp(strPort, CEC_IMX_VIRTUAL_COM))
+    return new CIMXCECAdapterCommunication(m_lib->m_cec);
 #endif
 
 #if defined(HAVE_P8_USB)
