diff -Naur freevo-r8795.orig/ui/src/application/mplayer.py freevo-r8795/ui/src/application/mplayer.py
--- freevo-r8795.orig/ui/src/application/mplayer.py	2006-12-29 16:44:27.000000000 +0100
+++ freevo-r8795/ui/src/application/mplayer.py	2006-12-29 16:44:53.000000000 +0100
@@ -96,8 +96,6 @@
                 next_is_vf=True
             else:
                 ret.append(arg)
-        if vf:
-            return ret + [ '-vf-add', vf[1:] ]
         return ret
 
 
