diff -Naur a/setup.py b/setup.py
--- a/setup.py	2011-02-20 17:34:49.783527922 +0100
+++ b/setup.py	2011-02-20 17:34:52.715722084 +0100
@@ -12,7 +12,7 @@
 # Freevo distutils stuff
 sys.path.append('./src')
 import version
-from distutils import core
+import setuptools
 from util.distribution import setup, Extension, check_libs, docbook_finder
 #try:
 #    from setuptools import setup
@@ -20,27 +20,8 @@
 #    from util.distribution import setup
 
 
-libs_to_check = [
-    ('kaa', '\"svn co svn://svn.freevo.org/kaa/trunk/ kaa\"' ),
-    ('kaa.metadata', '\"svn co svn://svn.freevo.org/kaa/trunk/ kaa\"' ),
-    ('kaa.imlib2', '\"svn co svn://svn.freevo.org/kaa/trunk/ kaa\"' ),
-    ('BeautifulSoup', 'http://www.crummy.com/software/BeautifulSoup/' ),
-    ('pygame', 'http://www.pygame.org'),
-    ('Image', 'http://www.pythonware.com/products/pil/'),
-    ('twisted', 'http://www.twistedmatrix.com/'),
-    ('zope.interface', 'http://www.zope.org/Products/ZopeInterface'),
-    ('twisted.web.microdom', 'http://www.twistedmatrix.com/'),
-    ('imdb', 'http://imdbpy.sourceforge.net/'),
-    #('Numeric', 'http://numeric.scipy.org/'),
-]
 
-if sys.hexversion < 0x2050000:
-    libs_to_check.append(('elementtree', 'http://effbot.org/zone/elementtree.htm'))
-
-check_libs(libs_to_check)
-
-
-class Runtime(core.Command):
+class Runtime(setuptools.Command):
 
     description     = "download and install runtime"
     user_options    = []
@@ -142,10 +123,6 @@
         except Exception, why:
             traceback.print_exc()
 
-    if (not os.path.isdir('./Docs/installation/html')):
-        print 'Docs/howto not found. Please run ./autogen.sh'
-        sys.exit(0)
-
 import revision
 # only add files not in share and src
 
