diff -urN vcd-0.9.org/i18n.h vcd-0.9/i18n.h
--- vcd-0.9.org/i18n.h	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/i18n.h	1970-01-01 01:00:00.000000000 +0100
@@ -1,30 +0,0 @@
-/*
- * i18n.h: Internationalization
- *
- * VCD Player plugin for VDR (the Video Disk Recorder)
- * Copyright (C) 2002  Thomas Heiligenmann  <thomas@heiligenmann.de>
- *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program; if not, write to the Free Software Foundation,
- * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301 USA
- *
- */
-
-#ifndef __VCD_I18N_H
-#define __VCD_I18N_H
-
-#include <vdr/i18n.h>
-
-extern const tI18nPhrase Phrases[];
-
-#endif //__VCD_I18N_H
diff -urN vcd-0.9.org/Makefile vcd-0.9/Makefile
--- vcd-0.9.org/Makefile	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/Makefile	2014-03-20 18:23:26.124149647 +0100
@@ -53,7 +53,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o functions.o i18n.o menu.o player.o viewer.o setup.o menucontrol.o psd.o psdcontrol.o
+OBJS = $(PLUGIN).o functions.o menu.o player.o viewer.o setup.o menucontrol.o psd.o psdcontrol.o
 
 ### Implicit rules:
 
diff -urN vcd-0.9.org/menu.c vcd-0.9/menu.c
--- vcd-0.9.org/menu.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/menu.c	2014-03-20 18:23:26.124149647 +0100
@@ -26,7 +26,6 @@
 
 #include <vdr/interface.h>
 #include <vdr/status.h>
-#include "i18n.h"
 #include "menu.h"
 #include "functions.h"
 #include "setup.h"
diff -urN vcd-0.9.org/menucontrol.c vcd-0.9/menucontrol.c
--- vcd-0.9.org/menucontrol.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/menucontrol.c	2014-03-20 18:23:26.124149647 +0100
@@ -26,7 +26,6 @@
 
 #include <vdr/interface.h>
 #include <vdr/status.h>
-#include "i18n.h"
 #include "menu.h"
 #include "functions.h"
 #include "setup.h"
diff -urN vcd-0.9.org/psd.c vcd-0.9/psd.c
--- vcd-0.9.org/psd.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/psd.c	2014-03-20 18:23:26.124149647 +0100
@@ -21,7 +21,6 @@
  */
 
 
-#include "i18n.h"
 #include "psd.h"
 #include "psdcontrol.h"
 
diff -urN vcd-0.9.org/psdcontrol.c vcd-0.9/psdcontrol.c
--- vcd-0.9.org/psdcontrol.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/psdcontrol.c	2014-03-20 18:23:26.124149647 +0100
@@ -27,7 +27,6 @@
 
 #include <vdr/interface.h>
 #include <vdr/status.h>
-#include "i18n.h"
 #include "functions.h"
 #include "setup.h"
 #include "psdcontrol.h"
diff -urN vcd-0.9.org/setup.c vcd-0.9/setup.c
--- vcd-0.9.org/setup.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/setup.c	2014-03-20 18:23:26.128149647 +0100
@@ -22,7 +22,6 @@
 
 #include <vdr/menuitems.h>
 #include "setup.h"
-#include "i18n.h"
 
 cVcdSetupData VcdSetupData;
 
diff -urN vcd-0.9.org/vcd.c vcd-0.9/vcd.c
--- vcd-0.9.org/vcd.c	2014-03-20 18:22:47.276150287 +0100
+++ vcd-0.9/vcd.c	2014-03-20 18:23:26.124149647 +0100
@@ -25,7 +25,6 @@
 #include <vdr/interface.h>
 #include <vdr/plugin.h>
 #include "functions.h"
-#include "i18n.h"
 #include "menu.h"
 #include "menucontrol.h"
 #include "psdcontrol.h"
@@ -100,7 +99,6 @@
 
 bool cPluginVcd::Start(void)
 {
-  RegisterI18n(Phrases);
   if (option_vcd)
      vcd = new cVcd(option_vcd);
   else
