diff -Naur rtaudio-4.0.11_orig//Makefile.in rtaudio-4.0.11/Makefile.in
--- rtaudio-4.0.11_orig//Makefile.in	2012-05-31 22:21:04.000000000 +0200
+++ rtaudio-4.0.11/Makefile.in	2012-07-10 16:11:49.625274863 +0200
@@ -29,8 +29,7 @@
 	ranlib $(STATIC)
 	$(CC) -fPIC @libflags@ $(OBJECTS) @LIBS@
 	$(LN) -s @sharedname@ $(SHARED)
-
-#	$(CC) -shared $(OBJECTS) -o $(SHARED) @LIBS@
+	$(LN) -s @sharedname@ $(SHARED).$(MAJOR)
 
 %.o : %.cpp
 	$(CC) $(CFLAGS) $(DEFS) -c $(<) -o $@
