diff -Naur libdce-L24.9-git+20101214.orig//dce.c libdce-L24.9-git+20101214/dce.c
--- libdce-L24.9-git+20101214.orig//dce.c	2010-12-19 14:35:31.372262000 +0100
+++ libdce-L24.9-git+20101214/dce.c	2010-12-19 14:35:46.482261999 +0100
@@ -526,9 +526,6 @@
     VIDDEC3_process__args *args;
     RcmClient_Message *msg = NULL;
 
-    DEBUG(">> codec=%p, inBufs=%p, outBufs=%p, inArgs=%p, outArgs=%p",
-            codec, inBufs, outBufs, inArgs, outArgs);
-
     err = RcmClient_alloc(handle, sizeof(VIDDEC3_process__args), &msg);
     if (err < 0) {
         ERROR("fail: %08x", err);
@@ -552,8 +549,6 @@
     args = (VIDDEC3_process__args *)&(msg->data);
     ret = args->out.ret;
 
-    DEBUG("<< ret=%d", ret);
-
 out:
     if (msg) {
         RcmClient_free (handle, msg);
