--- uClibc/extra/config/confdata.c.orig	2007-10-28 00:19:06.000000000 +0200
+++ uClibc/extra/config/confdata.c	2007-10-28 00:19:24.000000000 +0200
@@ -169,10 +169,6 @@
 				if (sym->type == S_UNKNOWN)
 					sym->type = S_BOOLEAN;
 			}
-			if (sym->flags & def_flags) {
-				conf_warning("trying to reassign symbol %s", sym->name);
-				break;
-			}
 			switch (sym->type) {
 			case S_BOOLEAN:
 			case S_TRISTATE:
@@ -205,10 +201,6 @@
 				if (sym->type == S_UNKNOWN)
 					sym->type = S_OTHER;
 			}
-			if (sym->flags & def_flags) {
-				conf_warning("trying to reassign symbol %s", sym->name);
-				break;
-			}
 			switch (sym->type) {
 			case S_TRISTATE:
 				if (p[0] == 'm') {
