--- uClibc-20040825.orig/Makefile	Sun Aug 22 00:19:19 2004
+++ uClibc-20040825/Makefile	Wed Nov 17 19:44:18 2004
@@ -68,7 +68,7 @@
 	$(CC) $(CFLAGS) $(ARCH_CFLAGS) -c -o $@ $(ROOTDIR)/lib/libc/main.c
 
 bogus $(SHARED_TARGET): lib/libc.a lib/main.o Makefile
-	make -C $(ROOTDIR) relink
+	$(MAKE) -C $(ROOTDIR) relink
 	$(CC) -nostartfiles -o $(SHARED_TARGET) $(ARCH_CFLAGS) -Wl,-elf2flt -nostdlib		\
 		-Wl,-shared-lib-id,${LIBID}				\
 		lib/main.o \
@@ -114,7 +114,7 @@
 
 include/bits/uClibc_config.h: .config
 	@if [ ! -x ./extra/config/conf ] ; then \
-	    make -C extra/config conf; \
+	    $(MAKE) -C extra/config conf; \
 	fi;
 	$(RM) -r include/bits
 	$(INSTALL) -d include/bits
@@ -171,7 +171,7 @@
 endif
 ifeq ($(strip $(UCLIBC_PREGENERATED_LOCALE_DATA)),y)
 	(cd extra/locale; zcat $(LOCALE_DATA_FILENAME) | tar -xvf -)
-	make -C extra/locale pregen
+	$(MAKE) -C extra/locale pregen
 endif
 
 
@@ -315,10 +315,10 @@
 # configuration
 # ---------------------------------------------------------------------------
 extra/config/conf:
-	make -C extra/config conf
+	$(MAKE) -C extra/config conf
 
 extra/config/mconf:
-	make -C extra/config ncurses mconf
+	$(MAKE) -C extra/config ncurses mconf
 
 menuconfig: extra/config/mconf
 	$(RM) -r include/bits
