diff -Naur uClibc.orig/Makefile.in uClibc/Makefile.in
--- uClibc.orig/Makefile.in	2006-03-18 09:35:49.000000000 +0200
+++ uClibc/Makefile.in	2006-03-18 09:36:03.000000000 +0200
@@ -253,7 +253,7 @@
 	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)lib
 	$(INSTALL) -m 644 lib/lib*-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
 		$(PREFIX)$(RUNTIME_PREFIX)lib
-	cp -dRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
+	cp -PRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
 	@if [ -x lib/$(UCLIBC_LDSO_NAME)-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so ] ; then \
 	    set -e; \
 		$(SHELL_SET_X); \
