--- uClibc-20040825.orig/Makefile	Wed Nov 17 22:52:39 2004
+++ uClibc-20040825/Makefile	Wed Nov 17 23:34:58 2004
@@ -101,7 +101,7 @@
 	$(INSTALL) -d $(ROMFSDIR)/lib
 	$(INSTALL) -m 644 lib/lib*-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
 		$(ROMFSDIR)/lib
-	cp -dRf lib/*.so.* $(ROMFSDIR)/lib/.
+	cp -PRf lib/*.so.* $(ROMFSDIR)/lib/.
 	@if [ -x lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so ] ; then \
 	    set -x -e; \
 	    $(INSTALL) -m 755 lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
@@ -281,7 +281,7 @@
 	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)lib
 	$(INSTALL) -m 644 lib/lib*-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
 		$(PREFIX)$(RUNTIME_PREFIX)lib
-	cp -dRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
+	cp -PRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
 	@if [ -x lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so ] ; then \
 	    set -x -e; \
 	    $(INSTALL) -m 755 lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
@@ -396,7 +396,7 @@
 release: distclean
 	cd ..;					\
 	$(RM) -r uClibc-$(VERSION);		\
-	cp -dRf uClibc uClibc-$(VERSION);	\
+	cp -PRf uClibc uClibc-$(VERSION);	\
 	find uClibc-$(VERSION)/ -type f		\
 	    -name .\#* -exec $(RM) -r {} \; ;	\
 	find uClibc-$(VERSION)/ -type d		\
