diff -Naur uClibc-0.9.27.orig/Makefile uClibc-0.9.27/Makefile
--- uClibc-0.9.27.orig/Makefile	2005-01-12 12:59:03.839842992 +0100
+++ uClibc-0.9.27/Makefile	2005-01-12 13:01:30.403561920 +0100
@@ -232,7 +232,7 @@
 	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)lib
 	$(INSTALL) -m 644 lib/lib*-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
 		$(PREFIX)$(RUNTIME_PREFIX)lib
-	cp -dRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
+	cp -PRf lib/*.so.* $(PREFIX)$(RUNTIME_PREFIX)lib
 	@if [ -x lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so ] ; then \
 	    set -x -e; \
 	    $(INSTALL) -m 755 lib/ld-uClibc-$(MAJOR_VERSION).$(MINOR_VERSION).$(SUBLEVEL).so \
@@ -347,7 +347,7 @@
 release: distclean
 	cd ..;					\
 	$(RM) -r uClibc-$(VERSION);		\
-	cp -dRf uClibc uClibc-$(VERSION);	\
+	cp -PRf uClibc uClibc-$(VERSION);	\
 	find uClibc-$(VERSION)/ -type f		\
 	    -name .\#* -exec $(RM) -r {} \; ;	\
 	find uClibc-$(VERSION)/ -type d		\
