--- uClibc-20040825.orig/Makefile	Sun Jan  2 22:58:33 2005
+++ uClibc-20040825/Makefile	Sun Jan  2 22:57:53 2005
@@ -170,7 +170,10 @@
 	fi );
 endif
 ifeq ($(strip $(UCLIBC_PREGENERATED_LOCALE_DATA)),y)
-	(cd extra/locale; zcat $(LOCALE_DATA_FILENAME) | tar -xvf -)
+	(cd extra/locale; \
+	if [ -f $(LOCALE_DATA_FILENAME) ] ; then \
+	zcat $(LOCALE_DATA_FILENAME) | tar -xvf -; \
+	fi );
 	$(MAKE) -C extra/locale pregen
 endif
 
