--- uClibc/libc/sysdeps/linux/common/bits/uClibc_stdio.h	2007-01-06 01:57:50.000000000 +0100
+++ uClibc/libc/sysdeps/linux/common/bits/uClibc_stdio.h	2007-01-06 01:58:19.000000000 +0100
@@ -341,7 +341,7 @@
 /**********************************************************************
  * PROTOTYPES OF INTERNAL FUNCTIONS
  **********************************************************************/
-#ifdef _LIBC
+#if defined _LIBC && (defined IS_IN_libc || defined NOT_IN_libc)
 
 extern void _stdio_init(void) attribute_hidden;
 extern void _stdio_term(void) attribute_hidden;
