mplayer loader uses modify_ldt() syscall.

diff -Nur uClibc.orig/libc/sysdeps/linux/common/modify_ldt.c uClibc/libc/sysdeps/linux/common/modify_ldt.c
--- uClibc.orig/libc/sysdeps/linux/common/modify_ldt.c	Thu Jun 22 10:12:06 2006
+++ uClibc/libc/sysdeps/linux/common/modify_ldt.c	Fri Jun 23 08:10:49 2006
@@ -8,7 +8,7 @@
  */
 
 /* Nobody really uses this ... */
-#if 0
+#if 1
 
 #include "syscalls.h"
 
