diff -Naur uClibc.orig/libc/sysdeps/linux/powerpc/crti.S uClibc/libc/sysdeps/linux/powerpc/crti.S
--- uClibc.orig/libc/sysdeps/linux/powerpc/crti.S	2007-03-01 21:09:06.000000000 +0100
+++ uClibc/libc/sysdeps/linux/powerpc/crti.S	2007-03-01 21:09:43.000000000 +0100
@@ -1,8 +1,6 @@
 	.file	"initfini.c"
-#APP
 	
 	.section .init
-#NO_APP
 	.align 2
 	.globl _init
 	.type	_init, @function
@@ -10,13 +8,11 @@
 	stwu 1,-32(1)
 	mflr 0
 	stw 0,36(1)
-#APP
 	
 	.align 2
 	
 	
 	.section .fini
-#NO_APP
 	.align 2
 	.globl _fini
 	.type	_fini, @function
@@ -24,7 +20,6 @@
 	stwu 1,-32(1)
 	mflr 0
 	stw 0,36(1)
-#APP
 	.align 2
 	
 	
diff -Naur uClibc.orig/libc/sysdeps/linux/powerpc/crtn.S uClibc/libc/sysdeps/linux/powerpc/crtn.S
--- uClibc.orig/libc/sysdeps/linux/powerpc/crtn.S	2007-03-01 21:09:06.000000000 +0100
+++ uClibc/libc/sysdeps/linux/powerpc/crtn.S	2007-03-01 21:09:43.000000000 +0100
@@ -1,30 +1,23 @@
 	.file	"initfini.c"
-#APP
 	
 	.section .init
-#NO_APP
 	.align 2
 	.globl _init
 	.type	_init, @function
-#NO_APP
 	lwz 0,36(1)
 	addi 1,1,32
 	mtlr 0
 	blr
 	.size	_init, .-_init
-#APP
 	
 	.section .fini
-#NO_APP
 	.align 2
 	.globl _fini
 	.type	_fini, @function
-#NO_APP
 	lwz 0,36(1)
 	addi 1,1,32
 	mtlr 0
 	blr
 	.size	_fini, .-_fini
-#APP
 	
 	.ident	"GCC: (GNU) 3.3.1 20030626 (Debian prerelease)"
