diff -Naur uClibc-0.9.28.orig/libc/sysdeps/linux/powerpc/bits/kernel_types.h uClibc-0.9.28/libc/sysdeps/linux/powerpc/bits/kernel_types.h
--- uClibc-0.9.28.orig/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2006-01-03 14:49:26.000000000 +0100
+++ uClibc-0.9.28/libc/sysdeps/linux/powerpc/bits/kernel_types.h	2006-01-03 14:50:59.000000000 +0100
@@ -4,9 +4,10 @@
  * our private content, and not the kernel header, will win.
  *  -Erik
  */
-#if ! defined _PPC_POSIX_TYPES_H && ! defined _PPC64_POSIX_TYPES_H
+#if ! defined _PPC_POSIX_TYPES_H && ! defined _PPC64_POSIX_TYPES_H && ! defined _ASM_POWERPC_POSIX_TYPES_H
 #define _PPC_POSIX_TYPES_H
 #define _PPC64_POSIX_TYPES_H
+#define _ASM_POWERPC_POSIX_TYPES_H
 
 # if __WORDSIZE == 64
 typedef unsigned int	__kernel_dev_t;
