--- squeezelite-1.5/Makefile.org	2014-02-01 16:05:04.634275941 +0100
+++ squeezelite-1.5/Makefile	2014-02-01 16:05:13.750276098 +0100
@@ -1,6 +1,6 @@
 # Cross compile support - create a Makefile which defines these three variables and then includes this Makefile...
 CFLAGS  ?= -Wall -fPIC -O2 $(OPTS)
-LDFLAGS ?= -lasound -lpthread -lm -lrt
+LDFLAGS += -lasound -lpthread -lm -lrt
 EXECUTABLE ?= squeezelite
 
 # passing one or more of these in $(OPTS) enables optional feature inclusion
