diff -Nur iconv-base-2.0.orig/lib/GNUmakefile iconv-base-2.0/lib/GNUmakefile
--- iconv-base-2.0.orig/lib/GNUmakefile	Sat Mar 12 13:59:11 2005
+++ iconv-base-2.0/lib/GNUmakefile	Sat Mar 12 13:59:23 2005
@@ -4,7 +4,7 @@
 SHLIB_MAJOR =	2
 SHLIB_MINOR =	0
 
-SRCS =		aliases.c ccs.c ces.c ces_euc.c ces_iso2022.c ces_table.c \
+SRCS =		aliases.c ccs.c ces.c ces_euc.c ces_table.c \
 		converter.c iconv.c utils.c
 
 BUILTIN_CCS = $(shell cd ../ccs/ && $(MAKE) lib >/dev/null && ${AR} t libccs.a)
