--- a/qtbase/mkspecs/features/toolchain.prf	2019-10-20 10:20:11.647965358 +0200
+++ b/qtbase/mkspecs/features/toolchain.prf	2019-10-20 10:20:11.687965070 +0200
@@ -235,6 +235,7 @@
                         paths = $$split(line, $$QMAKE_DIRLIST_SEP)
                     for (path, paths): \
                         QMAKE_DEFAULT_LIBDIRS += $$clean_path($$path)
+                    QMAKE_DEFAULT_LIBDIRS += $$clean_path($$(LIBRARY_PATH))
                 } else: contains(line, "Library search paths:") {
                     add_libraries = true
                 } else: contains(line, "$${LITERAL_HASH}include <.*") {  # #include <...> search starts here:
