--- a/qtbase/mkspecs/devices/linux-generic-g++/qmake.conf	2019-01-01 11:16:45.495801861 +0100
+++ b/qtbase/mkspecs/devices/linux-generic-g++/qmake.conf	2019-01-01 11:16:45.503801802 +0100
@@ -5,5 +5,8 @@
 # ./configure -device linux-generic-g++ -device-option CROSS_COMPILE=mipsel-linux-gnu-

 include(../common/linux_device_pre.conf)
+
+EGLFS_DEVICE_INTEGRATION = $$EGLFS_INTEGRATION
+
 include(../common/linux_device_post.conf)
 load(qt_config)
