diff -Nur nvram-wakeup.orig/set_timer nvram-wakeup/set_timer
--- nvram-wakeup.orig/set_timer	2003-09-18 21:45:54.000000000 +0200
+++ nvram-wakeup/set_timer	2004-07-09 00:17:54.000000000 +0200
@@ -30,10 +30,10 @@
 
 # programs to use:
 HWCLOCK=/sbin/hwclock
-TIME=/usr/local/sbin/time
+TIME=/usr/sbin/time
 
 # files to use:
-TIME_DIFF=/video/time_diff
+TIME_DIFF=/var/tmp/time_diff
 
 
 if [ ! -x $HWCLOCK ] ; then 
