diff -Naur hal-0.5.14/hald/linux/probing/probe-video4linux.c hal-0.5.14a//hald/linux/probing/probe-video4linux.c
--- hal-0.5.14/hald/linux/probing/probe-video4linux.c	2009-08-24 14:42:30.000000000 +0200
+++ hal-0.5.14a//hald/linux/probing/probe-video4linux.c	2011-02-25 15:34:39.959208719 +0100
@@ -30,7 +30,6 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
-#include <linux/videodev.h>
 #include <linux/videodev2.h>
 #include <errno.h>
 #include <fcntl.h>
@@ -50,7 +49,6 @@
 	int ret = -1;
 	char *udi;
 	char *device_file;
-	struct video_capability v1cap;
 	struct v4l2_capability v2cap;
 	LibHalContext *ctx = NULL;
 	LibHalChangeSet *cset;
@@ -108,30 +106,6 @@
 		}
 	} else {
 		HAL_DEBUG (("ioctl VIDIOC_QUERYCAP failed"));
-
-		if (ioctl (fd, VIDIOCGCAP, &v1cap) == 0) {
-			libhal_changeset_set_property_string (cset,
-			                                      "video4linux.version", "1");
-
-			libhal_changeset_set_property_string (cset,
-			                                      "info.product", v1cap.name);
-
-			if ((v1cap.type & VID_TYPE_CAPTURE) > 0) {
-				LIBHAL_FREE_DBUS_ERROR (&error);
-				libhal_device_add_capability (ctx, udi, "video4linux.video_capture", &error);
-			} if ((v1cap.type & VID_TYPE_OVERLAY) > 0) {
-				LIBHAL_FREE_DBUS_ERROR (&error);
-				libhal_device_add_capability (ctx, udi, "video4linux.video_overlay", &error);
-			} if (v1cap.audios > 0) {
-				LIBHAL_FREE_DBUS_ERROR (&error);
-				libhal_device_add_capability (ctx, udi, "video4linux.audio", &error);
-			} if ((v1cap.type & VID_TYPE_TUNER) > 0) {
-				LIBHAL_FREE_DBUS_ERROR (&error);
-				libhal_device_add_capability (ctx, udi, "video4linux.tuner", &error);
-			}
-		} else {
-			HAL_DEBUG (("ioctl VIDIOCGCAP failed"));
-		}
 	}
 
 	LIBHAL_FREE_DBUS_ERROR (&error);
