--- hal-0.5.14/hald/linux/addons/addon-generic-backlight.c.orig	2012-10-23 23:08:32.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-generic-backlight.c	2012-10-23 23:08:56.000000000 +0200
@@ -35,7 +35,7 @@
 #include <sys/stat.h>
 #include <unistd.h> 
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hal-0.5.14/hald/linux/addons/addon-leds.c.orig	2012-10-23 23:12:17.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-leds.c	2012-10-23 23:12:31.000000000 +0200
@@ -36,8 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
-#include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>

--- hal-0.5.14/hald/linux/addons/addon-rfkill-killswitch.c.orig	2012-10-23 23:14:53.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-rfkill-killswitch.c	2012-10-23 23:15:03.000000000 +0200
@@ -36,8 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
-#include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hal-0.5.14/hald/linux/addons/addon-storage.c.orig	2012-10-23 23:16:56.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-storage.c	2012-10-23 23:17:14.000000000 +0200
@@ -39,7 +39,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <unistd.h>
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hal-0.5.14/hald/linux/addons/addon-input.c.orig	2012-10-23 23:18:49.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-input.c	2012-10-23 23:19:06.000000000 +0200
@@ -41,8 +41,7 @@
 #else
   #include <linux/input.h>
 #endif
-#include <glib/gmain.h>
-#include <glib/gprintf.h>
+#include <glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
 #include "libhal/libhal.h"
--- hal-0.5.14/hald/linux/addons/addon-ipw-killswitch.c.orig	2012-10-23 23:20:51.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-ipw-killswitch.c	2012-10-23 23:21:00.000000000 +0200
@@ -36,8 +36,6 @@
 #include <unistd.h> 
 
 #include <glib.h>
-#include <glib/gmain.h>
-#include <glib/gstdio.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 
--- hal-0.5.14/hald/linux/addons/addon-omap-backlight.c.orig	2012-10-23 23:22:52.000000000 +0200
+++ hal-0.5.14/hald/linux/addons/addon-omap-backlight.c	2012-10-23 23:23:06.000000000 +0200
@@ -41,7 +41,7 @@
 #include <fcntl.h>
 #include <unistd.h> 
 
-#include <glib/gmain.h>
+#include <glib.h>
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-lowlevel.h>
 

