diff -Naur python-pycurl-7.19.0-orig/setup.py python-pycurl-7.19.0/setup.py
--- python-pycurl-7.19.0-orig/setup.py	2008-09-09 19:40:34.000000000 +0200
+++ python-pycurl-7.19.0/setup.py	2012-11-21 13:57:51.134719057 +0100
@@ -11,8 +11,7 @@
 
 import glob, os, re, sys, string
 import distutils
-from distutils.core import setup
-from distutils.extension import Extension
+from setuptools import Extension, setup
 from distutils.util import split_quoted
 from distutils.version import LooseVersion
 
@@ -97,8 +96,7 @@
         else:
             extra_compile_args.append(e)
     libs = split_quoted(
-        os.popen("'%s' --libs" % CURL_CONFIG).read()+\
-        os.popen("'%s' --static-libs" % CURL_CONFIG).read())
+        os.popen("'%s' --libs" % CURL_CONFIG).read())
     for e in libs:
         if e[:2] == "-l":
             libraries.append(e[2:])
