diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 17eccbe..7ae3ba4 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -2317,6 +2317,15 @@ void CApplication::Render()
           singleFrameTime = 200;  // 5 fps, <=200 ms latency to wake up
       }
 
+#ifdef ALLWINNERA10
+      if (IsInScreenSaver())
+      {
+        //so the mk802 does not burn down the house when i'm sleeping :)
+        limitFrames = true;
+        singleFrameTime = 100;
+      }
+#endif
+
       decrement = true;
     }
   }

