diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index 1da026e..7ed2cdf 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -785,6 +785,15 @@ void CAESinkALSA::EnumerateDevicesEx(AEDeviceInfoList &list, bool force)
       {
         EnumerateDevice(list, name, desc ? desc : name, config);
       }
+      else
+      {
+        /* sun4i whilelist */
+        std::string strDevice = std::string(name);
+        size_t found = strDevice.find("sunxi");
+        
+        if(found != std::string::npos)
+          EnumerateDevice(list, name, desc ? desc : name, config);
+      }
     }
     free(io);
     free(name);
@@ -886,6 +895,10 @@ AEDeviceType CAESinkALSA::AEDeviceTypeFromName(const std::string &name)
   else if (name.substr(0, 6) == "iec958" || name.substr(0, 5) == "spdif")
     return AE_DEVTYPE_IEC958;
 
+  /* sun4i hdmi device */
+  if(name.find("sunxisndhdmi") != std::string::npos)
+    return AE_DEVTYPE_HDMI;
+
   return AE_DEVTYPE_PCM;
 }
 

