--- a/xbmc/cores/VideoRenderers/LinuxRendererA10.h	2013-05-25 11:10:15.000000000 +0200
+++ b/xbmc/cores/VideoRenderers/LinuxRendererA10.h	2013-06-22 16:29:00.000000000 +0200
@@ -40,6 +40,15 @@
 #ifndef CEDARV_FRAME_HAS_PHY_ADDR
 #include <os_adapter.h>
 #endif
+
+#ifndef SUNXI_DISP_VERSION
+#define SUNXI_DISP_VERSION_MAJOR 1
+#define SUNXI_DISP_VERSION_MINOR 0
+#define SUNXI_DISP_VERSION ((SUNXI_DISP_VERSION_MAJOR << 16) | SUNXI_DISP_VERSION_MINOR)
+#define SUNXI_DISP_VERSION_MAJOR_GET(x) (((x) >> 16) & 0x7FFF)
+#define SUNXI_DISP_VERSION_MINOR_GET(x) ((x) & 0xFFFF)
+#define DISP_CMD_VERSION DISP_CMD_RESERVE0
+#endif
 }
 
 class CRenderCapture;
--- a/xbmc/cores/VideoRenderers/LinuxRendererA10.cpp	2013-05-25 11:10:15.000000000 +0200
+++ b/xbmc/cores/VideoRenderers/LinuxRendererA10.cpp	2013-06-22 16:34:15.000000000 +0200
@@ -1495,6 +1495,16 @@
     return false;
   }
 
+  // tell /dev/disp the API version we are using
+  args[0] = SUNXI_DISP_VERSION;
+  args[1] = 0;
+  args[2] = 0;
+  args[3] = 0;
+  i = ioctl(g_hdisp, DISP_CMD_VERSION, args);
+  CLog::Log(LOGNOTICE, "A10: display API version is: %d.%d\n", 
+            SUNXI_DISP_VERSION_MAJOR_GET(i),
+            SUNXI_DISP_VERSION_MINOR_GET(i));
+
   args[0] = g_screenid;
   args[1] = 0;
   args[2] = 0;
