From a0338fcc008ae4753e7eb97c99b5fbfde1f34b65 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Fri, 31 May 2013 21:04:28 +0200
Subject: [PATCH 2/7] Enable all codecs. Needs testing!

---
 .../dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
index 3c9ff46..c7d2daf 100755
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
@@ -27,7 +27,7 @@ static long g_cedaropen = 0;
 #define A10ENABLE_MPEG1         CODEC_OK
 #define A10ENABLE_MPEG2         CODEC_OK
 #define A10ENABLE_H264          CODEC_OK
-#define A10ENABLE_H263          CODEC_BAD //fails completely
+#define A10ENABLE_H263          CODEC_OK //fails completely
 #define A10ENABLE_VC1_WVC1      CODEC_OK
 #define A10ENABLE_VP6           CODEC_OK
 #define A10ENABLE_VP8           CODEC_OK
@@ -36,8 +36,8 @@ static long g_cedaropen = 0;
 #define A10ENABLE_WMV1          CODEC_OK
 #define A10ENABLE_WMV2          CODEC_OK
 #define A10ENABLE_WMV3          CODEC_OK
-#define A10ENABLE_MPEG4V1       CODEC_BAD //??
-#define A10ENABLE_MPEG4V2       CODEC_BAD //fails completely
+#define A10ENABLE_MPEG4V1       CODEC_OK //??
+#define A10ENABLE_MPEG4V2       CODEC_OK //fails completely
 #define A10ENABLE_MPEG4V3       CODEC_OK
 #define A10ENABLE_DIVX4         CODEC_OK
 #define A10ENABLE_DIVX5         CODEC_OK
-- 
1.7.9.5

