From 6aeacf0b0656961f902affdab32846fa84c6cf07 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Fri, 31 May 2013 22:28:26 +0200
Subject: [PATCH 3/7] Use ffmpeg for thumb extraction

---
 xbmc/cores/dvdplayer/DVDFileInfo.cpp |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDFileInfo.cpp b/xbmc/cores/dvdplayer/DVDFileInfo.cpp
index 30f59f5..3607617 100644
--- a/xbmc/cores/dvdplayer/DVDFileInfo.cpp
+++ b/xbmc/cores/dvdplayer/DVDFileInfo.cpp
@@ -162,6 +162,11 @@ bool CDVDFileInfo::ExtractThumb(const CStdString &strPath, CTextureDetails &deta
     CDVDStreamInfo hint(*pDemuxer->GetStream(nVideoStream), true);
     hint.software = true;
 
+#ifdef ALLWINNERA10
+      // always use ffmpeg for thumb extraction instead of A10 HW
+      CDVDCodecOptions dvdOptions;
+      pVideoCodec = CDVDFactoryCodec::OpenCodec(new CDVDVideoCodecFFmpeg(), hint, dvdOptions);
+#else
     if (hint.codec == CODEC_ID_MPEG2VIDEO || hint.codec == CODEC_ID_MPEG1VIDEO)
     {
       // libmpeg2 is not thread safe so use ffmepg for mpeg2/mpeg1 thumb extraction
@@ -172,6 +177,7 @@ bool CDVDFileInfo::ExtractThumb(const CStdString &strPath, CTextureDetails &deta
     {
       pVideoCodec = CDVDFactoryCodec::CreateVideoCodec( hint );
     }
+#endif
 
     if (pVideoCodec)
     {
-- 
1.7.9.5

