From c957b60515d04e2bc143512c4324373a17266a36 Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Sun, 23 Jun 2013 12:46:32 +0200
Subject: [PATCH 4/7] Improve Makefile a bit

---
 tools/a10/depends/xbmc/Makefile |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/tools/a10/depends/xbmc/Makefile b/tools/a10/depends/xbmc/Makefile
index 4e6558a..84b9b9f 100644
--- a/tools/a10/depends/xbmc/Makefile
+++ b/tools/a10/depends/xbmc/Makefile
@@ -15,14 +15,24 @@ CONFIGURE=./configure \
 	--enable-neon \
 	--enable-gles \
 	--disable-mysql \
+	--disable-optical-drive \
+	--disable-crystalhd \
+	--disable-vtbdecoder \
+	--disable-pulse \
+	--disable-projectm \
+	--disable-libcec \
+	--disable-optimizations \
 
 SRC=../../../../
 
 all: dep
 	(cd $(SRC); make -j$(JOBS))
 
+install:
+	(cd $(SRC); make install)
+
 install-only:
-	(cd $(SRC); make -j$(JOBS) install)
+	(cd $(SRC); make install)
 
 bin-only:
 	(cd $(SRC); make -j$(JOBS))
-- 
1.7.9.5

