From e485214c51e36222e2903ebdb91ea067e83adbff Mon Sep 17 00:00:00 2001
From: Andreas Baierl <ichgeh@imkreisrum.de>
Date: Sun, 23 Jun 2013 12:47:02 +0200
Subject: [PATCH 5/7] Add some logging.

---
 .../dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp |   21 ++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
index c7d2daf..6e247a8 100755
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecA10.cpp
@@ -311,6 +311,10 @@ bool CDVDVideoCodecA10::DoOpen()
     CLog::Log(LOGERROR, "A10: cedar already in use");
     return false;
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "A10: cedar not in use yet");
+  }
 
   m_hcedarv = libcedarv_init(&ret);
   if (ret < 0)
@@ -318,6 +322,10 @@ bool CDVDVideoCodecA10::DoOpen()
     CLog::Log(LOGERROR, "A10: libcedarv_init failed. (%d)\n", ret);
     goto Error;
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "A10: cedar initialized");
+  }
 
   ret = m_hcedarv->set_vstream_info(m_hcedarv, &m_info);
   if (ret < 0)
@@ -325,6 +333,10 @@ bool CDVDVideoCodecA10::DoOpen()
     CLog::Log(LOGERROR, "A10: set_vstream_info failed. (%d)\n", ret);
     goto Error;
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "A10: vstream_info set");
+  }
 
   ret = m_hcedarv->open(m_hcedarv);
   if (ret < 0)
@@ -332,6 +344,11 @@ bool CDVDVideoCodecA10::DoOpen()
     CLog::Log(LOGERROR, "A10: open failed. (%d)\n", ret);
     goto Error;
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "A10: cedarv opened");
+  }
+
 
   m_hcedarv->ioctrl(m_hcedarv, CEDARV_COMMAND_RESET, 0);
 
@@ -341,6 +358,10 @@ bool CDVDVideoCodecA10::DoOpen()
     CLog::Log(LOGERROR, "A10: CEDARV_COMMAND_PLAY failed. (%d)\n", ret);
     goto Error;
   }
+  else
+  {
+    CLog::Log(LOGDEBUG, "A10: cedarv playing");
+  }
 
   m_prebuffer = true;
   m_nframes   = 0;
-- 
1.7.9.5

