diff --git a/addons/weather.wunderground/default.py b/addons/weather.wunderground/default.py
index c86590b..3419f8a 100644
--- a/addons/weather.wunderground/default.py
+++ b/addons/weather.wunderground/default.py
@@ -513,7 +513,7 @@ def properties(data,loc):
     set_property('Alerts.IsFetched', 'true')
     if str(data['alerts']) != '[]':
         rss = ''
-        alerts = ''
+        alerts = u''
         for count, item in enumerate(data['alerts']):
             set_property('Alerts.%i.Description'     % (count+1), item['description'])
             set_property('Alerts.%i.Message'         % (count+1), item['message'])
@@ -523,7 +523,7 @@ def properties(data,loc):
             rss    = rss + item['description'] + ' - '
             alerts = alerts + item['message']
         set_property('Alerts.RSS'   , rss.rstrip(' - '))
-        set_property('Alerts'       , alerts)
+        set_property('Alerts'       , alerts.encode('iso-8859-1'))
         set_property('Alerts.Count' , str(count+1))
     else:
         set_property('Alerts.RSS'   , '')

