diff --git a/txtrecv.c b/txtrecv.c
index 15bd88f..ea76f23 100644
--- a/txtrecv.c
+++ b/txtrecv.c
@@ -483,8 +483,12 @@ cTxtStatus::~cTxtStatus()
     delete receiver;
 }
 
-void cTxtStatus::ChannelSwitch(const cDevice *Device, int ChannelNumber)
-{
+#if VDRVERSNUM >= 10726
+void cTxtStatus::ChannelSwitch(const cDevice * Device, int ChannelNumber, bool LiveView) {
+#else
+void cTxtStatus::ChannelSwitch(const cDevice * Device, int ChannelNumber) {
+    bool LiveView = Device && Device->IsPrimaryDevice();
+#endif
    // ignore if channel is 0
    if (ChannelNumber == 0) return;
 
@@ -494,7 +498,7 @@ void cTxtStatus::ChannelSwitch(const cDevice *Device, int ChannelNumber)
    if (newLiveChannel == NULL) return;
 
    // ignore non-live-channel-switching
-   if (!Device->IsPrimaryDevice() ||
+   if (!LiveView ||
       ChannelNumber != cDevice::CurrentChannel()) return;
 
    // live channel was changed
diff --git a/txtrecv.h b/txtrecv.h
index 8e53daa..436f5a3 100644
--- a/txtrecv.h
+++ b/txtrecv.h
@@ -176,7 +176,11 @@ private:
    bool storeTopText;
    Storage* storage;
 protected:
-   virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
+#if VDRVERSNUM >= 10726
+    virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber, bool LiveView);
+#else
+    virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
+#endif
 public:
    cTxtStatus(bool storeTopText, Storage* storage);
    ~cTxtStatus();
