--- atitvout.orig/Makefile	Mon Sep 16 22:33:35 2002
+++ atitvout/Makefile	Wed Nov 17 20:24:45 2004
@@ -8,7 +8,7 @@
 all: $(BINARY)
 
 $(BINARY): $(OBJS)
-	make -C lrmi-0.6
+	$(MAKE) -C lrmi-0.6
 	$(CC) $(OBJS) lrmi-0.6/liblrmi.a -o $(BINARY)
 
 strip: $(BINARY)
@@ -18,7 +18,7 @@
 
 clean:
 	rm -f *.o $(BINARY) *~ README index.html *.tar.gz test.out
-	make -C lrmi-0.6 clean
+	$(MAKE) -C lrmi-0.6 clean
 
 install: strip
 	install -g root -o root -m 555 $(BINARY) /usr/local/sbin
