taken from Fedora atitvout source rpm

--- atitvout.orig/atitvout.c	2002-09-16 22:37:08.000000000 +0300
+++ atitvout/atitvout.c	2004-07-23 21:43:27.426417887 +0300
@@ -118,8 +118,20 @@
     } else {
         if (strncmp(oem, "ATI MACH64", 10) == 0)
             radeon = 0;
+        /* From http://amd.colony.ee/pmwiki.php?pagename=Main.Atitvout */
+        else if (strncmp(oem, "ATI MOBILE M3", 13) == 0)
+            radeon = 0;
+        /* From Ville S */
+        else if (strcmp(oem, "ATI MOBILITY RADEON") == 0)
+            radeon = 0;
+        /* From NetBSD package */
+        else if (strcmp(oem, "ATI MOBILITY RADEON 9000") == 0)
+            radeon = 0;
         else if (strncmp(oem, "ATI RAGE128", 11) == 0)
             radeon = 1;
+        /* From Ville S */
+        else if (strcmp(oem, "ATI RV200") == 0)
+            radeon = 1;
         /* Note: Also it's possible to have here 'RG6' for first radeons */
         else if (strstr(oem, "RADEON") != 0)
             radeon = 1;
