diff -uN 13.4/common/lib/modules/fglrx/build_mod/firegl_public.c 13.4/common/lib/modules/fglrx/build_mod/firegl_public.c
--- 13.4/common/lib/modules/fglrx/build_mod/firegl_public.c	2013-04-16 23:29:55.000000000 +0200
+++ 13.4/common/lib/modules/fglrx/build_mod/firegl_public.c	2013-05-19 12:05:33.842076474 +0200
@@ -3088,7 +3156,11 @@
 int ATI_API_CALL KCL_EFI_IS_ENABLED(void)
 {
 #ifdef CONFIG_EFI
+#ifdef EFI_BOOT
+    return efi_enabled(EFI_BOOT);
+#else
     return efi_enabled;
+#endif
 #else
     return 0;
 #endif
