--- work.orig/common/lib/modules/fglrx/build_mod/firegl_public.c	2010-08-20 21:51:48.000000000 +0200
+++ work/common/lib/modules/fglrx/build_mod/firegl_public.c	2010-08-20 23:46:37.000000000 +0200
@@ -320,7 +320,11 @@
     return firegl_release((KCL_IO_FILE_Handle)filp);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
+int ip_firegl_ioctl(struct file* filp, unsigned int cmd, unsigned long arg)
+#else
 int ip_firegl_ioctl(struct inode* inode, struct file* filp, unsigned int cmd, unsigned long arg)
+#endif
 {
     return firegl_ioctl((KCL_IO_FILE_Handle)filp, cmd, arg);
 }
@@ -407,7 +411,11 @@
 #endif
     open:    ip_firegl_open,
     release: ip_firegl_release,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,36)
+    unlocked_ioctl:   ip_firegl_ioctl,
+#else
     ioctl:   ip_firegl_ioctl,
+#endif
     mmap:    ip_firegl_mmap,
 
     write:   ip_firegl_write,
