--- ati.orig/common/lib/modules/fglrx/build_mod/firegl_public.c	2010-11-18 18:08:34.831997690 +0100
+++ ati/common/lib/modules/fglrx/build_mod/firegl_public.c	2010-11-18 18:10:28.711997706 +0100
@@ -5094,7 +5094,11 @@
 unsigned int ATI_API_CALL KAS_Mutex_Initialize(void* hMutex)
 {
     kasMutex_t* mutex_obj = (kasMutex_t*)hMutex;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,37)
+    sema_init(&(mutex_obj->mutex),1);
+#else
     init_MUTEX(&(mutex_obj->mutex));
+#endif
     return 1;
 }
 
