diff -Naur a/common/lib/modules/fglrx/build_mod/kcl_osconfig.h b/common/lib/modules/fglrx/build_mod/kcl_osconfig.h
--- a/common/lib/modules/fglrx/build_mod/kcl_osconfig.h	2010-05-04 15:07:20.134758660 +0200
+++ b/common/lib/modules/fglrx/build_mod/kcl_osconfig.h	2010-05-04 15:29:13.617745041 +0200
@@ -27,8 +27,8 @@
 #endif
 
 // Choose modern way to call 32-on-64 IOCTLs if configured in the kernel
-#if defined(CONFIG_COMPAT) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+//#if defined(CONFIG_COMPAT) && LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
 #define KCL_OSCONFIG_IOCTL_COMPAT
-#endif
+//#endif
 
 #endif
