--- a/common/lib/modules/fglrx/build_mod/kcl_ioctl.c	2010-12-02 22:28:43.000000000 +0100
+++ b/common/lib/modules/fglrx/build_mod/kcl_ioctl.c.new	2011-01-09 17:40:22.000000000 +0100
@@ -193,7 +193,7 @@
  */
 void* ATI_API_CALL KCL_IOCTL_AllocUserSpace32(long size)
 {
-    void __user *ret = COMPAT_ALLOC_USER_SPACE(size);
+    void __user *ret = arch_compat_alloc_user_space(size);

     /* prevent stack overflow */
     if (!access_ok(VERIFY_WRITE, ret, size))

