--- a/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-06-10 09:50:54.260781327 +0200
+++ b/common/lib/modules/fglrx/build_mod/firegl_public.c	2012-06-10 11:35:18.000898839 +0200
@@ -4154,7 +4154,7 @@
 {
     unsigned int p;
     KCL_DEBUG5(FN_FIREGL_KAS, "%d\n", level_init);
-    for_each_cpu_mask(p, cpu_possible_map)
+    for_each_possible_cpu(p)
     {
         KCL_DEBUG1(FN_FIREGL_KAS,"Setting initial execution level for CPU # %d\n", p);
         preempt_disable();
--- ./common/lib/modules/fglrx/build_mod/kcl_ioctl.c	2012-06-10 09:50:54.268781326 +0200
+++ kcl_ioctl.c	2012-06-10 11:34:40.184898130 +0200
@@ -217,6 +217,9 @@
  *  \param size [in] Number of bytes to allocate
  *  \return Pointer to allocated memory
  */
+
+DEFINE_PER_CPU(unsigned long, old_rsp);
+
 void* ATI_API_CALL KCL_IOCTL_AllocUserSpace32(long size)
 {
     void __user *ret = COMPAT_ALLOC_USER_SPACE(size);
