diff -ruN vdr-1.7.38_iptv/menu.h vdr-1.7.38/menu.h
--- vdr-1.7.38_iptv/menu.h	2013-02-25 02:30:57.000000000 +0100
+++ vdr-1.7.38/menu.h	2013-02-25 02:35:33.000000000 +0100
@@ -336,6 +336,7 @@
   virtual void Hide(void);
   bool Visible(void) { return visible; }
   static void SetRecording(const char *FileName);
+  static void SetRecording(const char *FileName, const char *) { SetRecording(FileName); }
   static const char *NowReplaying(void);
   static const char *LastReplayed(void);
   static void ClearLastReplayed(const char *FileName);
