From 5260df1deef2a488afa690ff2c3544e7501e3320 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 23 Aug 2012 16:27:01 +0200
Subject: [PATCH 1/4] Add configure options for CuBox target

---
 configure.ac |   26 ++++++++++++++++++++++++++
 1 files changed, 26 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index ae1b5a5..d02493a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -50,6 +50,20 @@ AC_ARG_ENABLE([rpi],
   [use_rpi=$enableval],
   [use_rpi=auto])
 
+## CuBox support
+AC_ARG_ENABLE([cubox],
+  [AS_HELP_STRING([--enable-cubox],
+  [enable support for the CuBox (default is no)])],
+  [use_nxp=$enableval],
+  [use_nxp=no])
+
+## Optional path to the NXP dev toolkit
+AC_ARG_WITH([nxp-toolkit-path],
+  [AS_HELP_STRING([--with-nxp-toolkit-path],
+    [location of the NXP HDMI toolkit (default is ./nxp_hdmi)])],
+  [NXP_CFLAGS="-I$withval/inc"],
+  [NXP_CFLAGS="-I\$(abs_top_srcdir)/nxp_hdmi/inc"])
+
 ## Optional path to the RPi's dev headers
 AC_ARG_WITH([rpi-include-path],
   [AS_HELP_STRING([--with-rpi-include-path],
@@ -235,6 +249,18 @@ else
   features="$features\n  Raspberry Pi support :\t\tno"
 fi
 
+## mark CuBox support as available
+if test "x$use_nxp" != "xno"; then
+  AC_DEFINE([HAVE_NXP_API],[1],[Define to 1 to include CuBox support])
+  AM_CONDITIONAL(USE_NXP_API, true)
+  features="$features\n  CuBox support :\t\t\tyes"
+  LIB_INFO="$LIB_INFO 'CuBox'"
+  CPPFLAGS="$CPPFLAGS $NXP_CFLAGS"
+else
+  AM_CONDITIONAL(USE_NXP_API, false)
+  features="$features\n  CuBox support :\t\t\tno"
+fi
+
 ## check if our build system is complete
 AC_CHECK_HEADER(algorithm,,AC_MSG_ERROR($msg_required_header_missing))
 AC_CHECK_HEADER(ctype.h,,AC_MSG_ERROR($msg_required_header_missing))
-- 
1.7.4.1

