From 31172d06e57e5e924d8d841b2eeb335ddd155b3c Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 23 Aug 2012 18:11:28 +0200
Subject: [PATCH 2/4] Added CuBox section to README

---
 README |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/README b/README
index 5c7364b..de118ea 100644
--- a/README
+++ b/README
@@ -100,6 +100,23 @@ for 'configure':
 --with-rpi-lib-path="/path/to/libbcm_host.so"
 
 ===============================================================================
+                           === CuBox ===
+===============================================================================
+
+Solid-Run's CuBox uses an HDMI tranceiver by NXP. The device driver for this,
+which is based on an SDK by the chip vendor, is built into the kernel. The
+following options for 'configure' have been introduced:
+
+To enable support for the CuBox:
+--enable-cubox
+
+To specify the path for the NXP SDK:
+--with-nxp-toolkit-path='path/to/linux/drivers/video/dovefb/nxp_hdmi'
+
+If the toolkit path is not specified, it is assumed that a directory named
+'nxp_hdmi' (or a link to it) exists in the top level of the source tree.
+
+===============================================================================
                             === Debugging / Testing ===
 ===============================================================================
 
-- 
1.7.4.1

