From 580e5f0c21609225ecbf444b1ab3f70ec90d3d04 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Sun, 2 Sep 2012 22:28:56 +0200
Subject: [PATCH 4/4] Fixed vendor id shown as zero on big endian systems

---
 src/testclient/main.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/testclient/main.cpp b/src/testclient/main.cpp
index e594eaf..7b94d01 100644
--- a/src/testclient/main.cpp
+++ b/src/testclient/main.cpp
@@ -649,7 +649,7 @@ bool ProcessCommandVEN(ICECAdapter *parser, const string &command, string &argum
       if (iDev >= 0 && iDev < 15)
       {
         uint64_t iVendor = parser->GetDeviceVendorId((cec_logical_address) iDev);
-        PrintToStdOut("vendor id: %06x", iVendor);
+        PrintToStdOut("vendor id: %06llx", iVendor);
         return true;
       }
     }
-- 
1.7.4.1

