diff -Naur xbmc-10.1/configure xbmc-10.1a/configure
--- xbmc-10.1/configure	2011-04-25 17:28:12.370863302 +0200
+++ xbmc-10.1a/configure	2011-04-25 17:30:52.354114240 +0200
@@ -10111,7 +10111,7 @@
   xyes) :
 
     INCLUDES="$INCLUDES $LIBBLURAY_CFLAGS";
-    if test "$host_vendor" != "apple"; then
+    if test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
 
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libbluray soname" >&5
@@ -11578,7 +11578,7 @@
   HOST_NOT_APPLE=0
 fi
 
-if test "$host_vendor" != "apple"; then
+if test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
 
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libcurl soname" >&5
@@ -11804,7 +11804,6 @@
 
   fi
 
-fi
 
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libogg soname" >&5
@@ -11985,6 +11984,7 @@
 
   fi
 
+fi
 
 # WebServer
 if test "$host_vendor" != "apple" ; then
@@ -12513,7 +12513,7 @@
 
 done
 
-  if test "$use_librtmp" != "no" && test "$host_vendor" != "apple"; then
+  if test "$use_librtmp" != "no" && test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
 
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for librtmp soname" >&5
@@ -13296,7 +13296,7 @@
 fi
 
 
-    if test "$host_vendor" != "apple"; then
+    if test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
 
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libcrystalhd soname" >&5
diff -Naur xbmc-10.1/configure.in xbmc-10.1a/configure.in
--- xbmc-10.1/configure.in	2011-03-05 04:33:37.000000000 +0100
+++ xbmc-10.1a/configure.in	2011-04-25 17:29:40.418246838 +0200
@@ -689,17 +689,18 @@
   HOST_NOT_APPLE=0
 fi
 
-if test "$host_vendor" != "apple"; then
+if test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
   XB_FIND_SONAME([CURL],        [curl])
   XB_FIND_SONAME([FLAC],        [FLAC])
   XB_FIND_SONAME([MODPLUG],     [modplug])
   XB_FIND_SONAME([FAAD],        [faad])
   XB_FIND_SONAME([MAD],         [mad])
-fi
+
 XB_FIND_SONAME([OGG],         [ogg])
 XB_FIND_SONAME([VORBIS],      [vorbis])
 XB_FIND_SONAME([VORBISENC],   [vorbisenc])
 XB_FIND_SONAME([VORBISFILE],  [vorbisfile])
+fi
 
 # WebServer
 if test "$host_vendor" != "apple" ; then
@@ -822,7 +823,7 @@
       use_librtmp="no"
     fi
    ])
-  if test "$use_librtmp" != "no" && test "$host_vendor" != "apple"; then
+  if test "$use_librtmp" != "no" && test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
     XB_FIND_SONAME([RTMP], [rtmp], [use_librtmp])
   fi
   if test "$use_librtmp" != "no"; then
@@ -1007,7 +1008,7 @@
         AC_MSG_RESULT($crystalhd_not_found)
       fi],
     [#define __LINUX_USER__])
-    if test "$host_vendor" != "apple"; then
+    if test "$host_vendor" != "apple" -a "$cross_compiling" != yes; then
       XB_FIND_SONAME([CRYSTALHD], [crystalhd], [use_crystalhd])
     fi
     if test "x$use_crystalhd" != "xno"; then
