diff --git a/configure.in b/configure.in
index c8bf4f1..859bd90 100755
--- a/configure.in
+++ b/configure.in
@@ -2186,6 +2186,7 @@ XB_CONFIG_MODULE([lib/libid3tag/libid3tag],[
   set -x
   rm -f CMakeCache.txt &&                              \
   CC="$ORIGCC" CXX="$ORIGCXX" LDFLAGS="$LDFLAGS" cmake \
+    -DCMAKE_FIND_ROOT_PATH=${LIB_P}                    \
     -DCMAKE_BUILD_TYPE=None -DUSE_FTGL:BOOL=OFF        \
     -DCMAKE_C_FLAGS:STRING="${CPPFLAGS} ${CFLAGS}"     \
     -DCMAKE_CXX_FLAGS:STRING="${CPPFLAGS} ${CXXFLAGS}" \
@@ -2218,6 +2219,7 @@ XB_CONFIG_MODULE([xbmc/visualizations/Goom/goom2k4-0],[
   else
     TEMPCFLAGS="$CFLAGS";
   fi
+  export jm_cv_func_gettimeofday_clobber=no
   ./configure \
     CC="$CC" \
     CXX="$CXX" \
diff --git a/lib/libdvd/libdvdcss/src/libdvdcss.pc.in b/lib/libdvd/libdvdcss/src/libdvdcss.pc.in
index 9d25167..e61ef3a 100644
--- a/lib/libdvd/libdvdcss/src/libdvdcss.pc.in
+++ b/lib/libdvd/libdvdcss/src/libdvdcss.pc.in
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: libdvdcss
 Description: DVD access and decryption library.
 Version: @VERSION@
-Libs: -L${libdir} -ldvdcss
-Cflags: -I{includedir} -I${includedir}/@PACKAGE@
+Libs: -ldvdcss
+Cflags: -I.
diff --git a/lib/libdvd/libdvdnav/misc/dvdnav-config.in b/lib/libdvd/libdvdnav/misc/dvdnav-config.in
index 4e65cc1..88eb971 100644
--- a/lib/libdvd/libdvdnav/misc/dvdnav-config.in
+++ b/lib/libdvd/libdvdnav/misc/dvdnav-config.in
@@ -76,13 +76,13 @@ if test "$echo_exec_prefix" = "yes"; then
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@ -I@includedir@/libdvdread @THREAD_CFLAGS@
+      echo @THREAD_CFLAGS@
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdnav -ldvdread @THREAD_LIBS@
+      echo -ldvdnav -ldvdread @THREAD_LIBS@
 fi
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L@libdir@ -ldvdnavmini @THREAD_LIBS@
+      echo -ldvdnavmini @THREAD_LIBS@
 fi      
diff --git a/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh b/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh
index c8cdd05..227b393 100644
--- a/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh
+++ b/lib/libdvd/libdvdnav/misc/dvdnav-config2.sh
@@ -56,17 +56,17 @@ if test "$echo_prefix" = "yes"; then
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $dvdread_cflags $extracflags $threadcflags
+      echo $dvdread_cflags $extracflags $threadcflags
 fi
 
 if test "$echo_minicflags" = "yes"; then
-      echo -I$prefix/include -I$prefix/include/dvdnav $extracflags $threadcflags
+      echo $extracflags $threadcflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir -ldvdnav $dvdread_libs $threadlib
+      echo -ldvdnav $dvdread_libs $threadlib
 fi      
 
 if test "$echo_minilibs" = "yes"; then
-      echo -L$libdir -ldvdnavmini $threadlib
+      echo -ldvdnavmini $threadlib
 fi
diff --git a/lib/libdvd/libdvdnav/misc/dvdnav.pc.in b/lib/libdvd/libdvdnav/misc/dvdnav.pc.in
index 7ed56d3..dfe72f2 100644
--- a/lib/libdvd/libdvdnav/misc/dvdnav.pc.in
+++ b/lib/libdvd/libdvdnav/misc/dvdnav.pc.in
@@ -8,5 +8,5 @@ Description: DVD Navigation library
 Version: @VERSION@
 
 Requires.private: dvdread >= 4.1.2
-Cflags: -I${includedir} @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff --git a/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in b/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in
index 52945b3..21dc5b4 100644
--- a/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in
+++ b/lib/libdvd/libdvdnav/misc/dvdnavmini.pc.in
@@ -7,5 +7,5 @@ Name: libdvdnavmini
 Description: DVD Navigation mini library
 Version: @VERSION@
 
-Cflags: -I${includedir} @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
-Libs: -L${libdir} -ldvdnav @THREAD_LIBS@
+Cflags: @DVDREAD_CFLAGS@ @THREAD_CFLAGS@
+Libs: -ldvdnav @THREAD_LIBS@
diff --git a/lib/libdvd/libdvdread/misc/dvdread-config.in b/lib/libdvd/libdvdread/misc/dvdread-config.in
index cf38462..0a0974d 100644
--- a/lib/libdvd/libdvdread/misc/dvdread-config.in
+++ b/lib/libdvd/libdvdread/misc/dvdread-config.in
@@ -72,9 +72,9 @@ if test "$echo_exec_prefix" = "yes"; then
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I@includedir@
+      echo ""
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L@libdir@ -ldvdread
+      echo -ldvdread
 fi
diff --git a/lib/libdvd/libdvdread/misc/dvdread-config.sh b/lib/libdvd/libdvdread/misc/dvdread-config.sh
index e170c7e..84df135 100644
--- a/lib/libdvd/libdvdread/misc/dvdread-config.sh
+++ b/lib/libdvd/libdvdread/misc/dvdread-config.sh
@@ -48,9 +48,9 @@ if test "$echo_prefix" = "yes"; then
 fi
 
 if test "$echo_cflags" = "yes"; then
-      echo -I$prefix/include $extracflags
+      echo $extracflags
 fi
 
 if test "$echo_libs" = "yes"; then
-      echo -L$libdir $dvdreadlib
+      echo $dvdreadlib
 fi      
diff --git a/lib/libdvd/libdvdread/misc/dvdread.pc.in b/lib/libdvd/libdvdread/misc/dvdread.pc.in
index 917d0e6..43a4c52 100644
--- a/lib/libdvd/libdvdread/misc/dvdread.pc.in
+++ b/lib/libdvd/libdvdread/misc/dvdread.pc.in
@@ -7,5 +7,5 @@ Name: libdvdread
 Description: Low level DVD access library
 Version: @VERSION@
 
-Cflags: -I${includedir}
-Libs: -L${libdir} -ldvdread
+Cflags: -I.
+Libs: -ldvdread
diff --git a/system/Lircmap.xml b/system/Lircmap.xml
index b009eab..2d0cd06 100644
--- a/system/Lircmap.xml
+++ b/system/Lircmap.xml
@@ -409,7 +409,6 @@
 
 	<remote device="linux-input-layer">
 	<altname>cx23885_remote</altname>
-	<altname>devinput</altname>
 		<left>KEY_LEFT</left>
 		<right>KEY_RIGHT</right>
 		<up>KEY_UP</up>
@@ -509,4 +508,59 @@
 		<mypictures>yellow</mypictures>
 		<myvideo>blue</myvideo>
 	</remote>
+
+	<remote device="devinput">
+		<left>KEY_LEFT</left>
+		<right>KEY_RIGHT</right>
+		<up>KEY_UP</up>
+		<down>KEY_DOWN</down>
+		<select>KEY_OK</select>
+		<enter>KEY_ENTER</enter>
+		<clear>KEY_DELETE</clear>
+		<start>KEY_PROG1</start>
+		<back>KEY_EXIT</back>
+		<record>KEY_RECORD</record>
+		<play>KEY_PLAY</play>
+		<pause>KEY_PAUSE</pause>
+		<stop>KEY_STOP</stop>
+		<forward>KEY_FASTFORWARD</forward>
+		<reverse>KEY_REWIND</reverse>
+		<volumeplus>KEY_VOLUMEUP</volumeplus>
+		<volumeminus>KEY_VOLUMEDOWN</volumeminus>
+		<channelplus>KEY_CHANNELUP</channelplus>
+		<channelminus>KEY_CHANNELDOWN</channelminus>
+		<skipplus>KEY_NEXT</skipplus>
+		<skipminus>KEY_PREVIOUS</skipminus>
+		<title>KEY_EPG</title>
+		<subtitle>KEY_SUBTITLE</subtitle>
+		<language>KEY_LANGUAGE</language>
+		<info>KEY_INFO</info>
+		<display>KEY_ZOOM</display>
+		<mute>KEY_MUTE</mute>
+		<power>KEY_POWER</power>
+		<eject>KEY_EJECT</eject>
+		<menu>KEY_DVD</menu>
+		<menu>KEY_MENU</menu>
+		<myvideo>KEY_VIDEO</myvideo>
+		<mymusic>KEY_AUDIO</mymusic>
+		<mypictures>KEY_CAMERA</mypictures>
+		<mytv>KEY_TUNER</mytv>
+		<teletext>KEY_TEXT</teletext>
+		<one>KEY_NUMERIC_1</one>
+		<two>KEY_NUMERIC_2</two>
+		<three>KEY_NUMERIC_3</three>
+		<four>KEY_NUMERIC_4</four>
+		<five>KEY_NUMERIC_5</five>
+		<six>KEY_NUMERIC_6</six>
+		<seven>KEY_NUMERIC_7</seven>
+		<eight>KEY_NUMERIC_8</eight>
+		<nine>KEY_NUMERIC_9</nine>
+		<zero>KEY_NUMERIC_0</zero>
+		<star>KEY_NUMERIC_STAR</star>
+		<hash>KEY_NUMERIC_POUND</hash>
+		<red>KEY_RED</red>
+		<green>KEY_GREEN</green>
+		<yellow>KEY_YELLOW</yellow>
+		<blue>KEY_BLUE</blue>
+	</remote>
 </lircmap>
diff --git a/system/keymaps/keyboard.xml b/system/keymaps/keyboard.xml
index f35c79c..0345437 100644
--- a/system/keymaps/keyboard.xml
+++ b/system/keymaps/keyboard.xml
@@ -91,7 +91,7 @@
       <numpadseven>Number7</numpadseven>
       <numpadeight>Number8</numpadeight>
       <numpadnine>Number9</numpadnine>
-      <backslash>ToggleFullScreen</backslash>
+<!--      <backslash>ToggleFullScreen</backslash>   -->
       <home>FirstPage</home>
       <end>LastPage</end>
       <!-- PVR windows -->
diff --git a/userdata/RssFeeds.xml b/userdata/RssFeeds.xml
index c74d461..e8f1674 100644
--- a/userdata/RssFeeds.xml
+++ b/userdata/RssFeeds.xml
@@ -3,6 +3,7 @@
   <!-- RSS feeds. To have multiple feeds, just add a feed to the set. You can also have multiple sets. 	!-->
   <!-- To use different sets in your skin, each must be called from skin with a unique id.             	!-->
   <set id="1">
+    <feed updateinterval="30">http://www.geexbox.org/feed/</feed>
     <feed updateinterval="30">http://feeds.xbmc.org/xbmc</feed>
   </set>
 </rssfeeds>

