From f7b52fd8a3904bfdc469e880a5c003fb9e1c1fb2 Mon Sep 17 00:00:00 2001
From: Tobias Arrskog <topfs2@xbmc.org>
Date: Wed, 9 Feb 2011 21:38:00 -0500
Subject: [PATCH 2/4] Do not DEFINE types, typedef them

---
 xbmc/linux/PlatformDefs.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xbmc/linux/PlatformDefs.h b/xbmc/linux/PlatformDefs.h
index 76c929b..28737d5 100644
--- a/xbmc/linux/PlatformDefs.h
+++ b/xbmc/linux/PlatformDefs.h
@@ -161,7 +161,7 @@
 #define ZeroMemory(dst,size) memset(dst, 0, size)
 
 #define VOID    void
-#define byte    unsigned char
+typedef unsigned char byte;
 #define __int8    char
 #define __int16   short
 #define __int32   int
-- 
1.7.2.5

