diff -Naur xbmc-10.0.orig//configure xbmc-10.0/configure
--- xbmc-10.0.orig//configure	2010-12-20 00:01:10.785073994 +0100
+++ xbmc-10.0/configure	2010-12-20 00:03:19.975073995 +0100
@@ -14483,12 +14483,6 @@
   final_message="$final_message\n  MID Support:\tNo"
 fi
 
-if test -d "/usr/include/lzo"; then
-  INCLUDES="$INCLUDES -I/usr/include/lzo"
-else
-  INCLUDES="$INCLUDES"
-fi
-
 if test "x$use_ccache" != "xno"; then
   # Extract the first word of "ccache", so it can be a program name with args.
 set dummy ccache; ac_word=$2
diff -Naur xbmc-10.0.orig//tools/TexturePacker/XBMCTex.cpp xbmc-10.0/tools/TexturePacker/XBMCTex.cpp
--- xbmc-10.0.orig//tools/TexturePacker/XBMCTex.cpp	2010-12-20 00:00:52.515073995 +0100
+++ xbmc-10.0/tools/TexturePacker/XBMCTex.cpp	2010-12-20 00:03:38.075073998 +0100
@@ -40,7 +40,7 @@
 #endif
 
 #ifdef _LINUX
-#include <lzo1x.h>
+#include <lzo/lzo1x.h>
 #else
 #include "../../xbmc/lib/liblzo/LZO1X.H"
 #endif
