diff -Naur xbmc-10.1.orig//xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in xbmc-10.1/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in
--- xbmc-10.1.orig//xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in	2011-06-25 23:14:32.600777228 +0200
+++ xbmc-10.1/xbmc/cores/dvdplayer/Codecs/libdvd/Makefile.in	2011-07-09 15:32:49.512887309 +0200
@@ -49,12 +49,12 @@
 else
 
 $(SYSDIR)/libdvdcss-$(ARCH).so: $(WRAPPER) libdvdcss/src/.libs/libdvdcss.a
-	$(CC) -o $@ $(LDFLAGS) --soname,$@ \
+	$(CC) -o $@ $(LDFLAGS) -Wl,-soname,$@ \
 		libdvdcss/src/*.o \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
 $(SYSDIR)/libdvdnav-$(ARCH).so: $(WRAPPER) $(DVDCSS_A) libdvdread/obj/libdvdread.a libdvdnav/obj/libdvdnav.a
-	$(CC) -o $@ $(LDFLAGS) --soname,$@ $(DVDCSS_O) libdvdread/obj/*.o libdvdnav/obj/*.o \
+	$(CC) -o $@ $(LDFLAGS) -Wl,-soname,$@ $(DVDCSS_O) libdvdread/obj/*.o libdvdnav/obj/*.o \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
 endif
