--- a/configure	2011-10-31 08:17:10.954582157 +0100
+++ b/configure	2011-10-31 08:16:09.204491584 +0100
@@ -28891,6 +28891,7 @@
 
   set -x
   rm -f CMakeCache.txt && CC="$ORIGCC" CXX="$ORIGCXX" cmake    \
+    -DCMAKE_FIND_ROOT_PATH=${LIB_P} \
     -DCMAKE_BUILD_TYPE=None -DUSE_FTGL:BOOL=OFF        \
     -DCMAKE_C_FLAGS:STRING="${CPPFLAGS} ${CFLAGS}"     \
     -DCMAKE_CXX_FLAGS:STRING="${CPPFLAGS} ${CXXFLAGS}" \
