--- xbmc-pvr-20120102-git957bc3f/configure.in	2012-01-02 16:57:51.292973894 +0100
+++ xbmc-pvr-20120102-git957bc3f/configure.in.new	2012-01-02 17:01:31.613297037 +0100
@@ -2269,6 +2269,7 @@
 XB_CONFIG_MODULE([xbmc/visualizations/XBMCProjectM/libprojectM],[
   set -x
   rm -f CMakeCache.txt && CC="$ORIGCC" CXX="$ORIGCXX" cmake    \
+    -DCMAKE_FIND_ROOT_PATH=${LIB_P} \
     -DCMAKE_BUILD_TYPE=None -DUSE_FTGL:BOOL=OFF        \
     -DCMAKE_C_FLAGS:STRING="${CPPFLAGS} ${CFLAGS}"     \
     -DCMAKE_CXX_FLAGS:STRING="${CPPFLAGS} ${CXXFLAGS}" \
