diff -Naur xbmc-10.1/configure xbmc-10.1a/configure
--- xbmc-10.1/configure	2011-08-01 10:16:58.841596503 +0200
+++ xbmc-10.1a/configure	2011-08-01 10:17:12.113063194 +0200
@@ -13412,9 +13412,6 @@
 fi
 
 
-    if test "$use_openmax" = "yes"; then
-      INCLUDES="$INCLUDES -I/usr/include/OpenMAX/IL"
-    fi
   else
     { $as_echo "$as_me:${as_lineno-$LINENO}: $openmax_disabled" >&5
 $as_echo "$as_me: $openmax_disabled" >&6;}
diff -Naur xbmc-10.1/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp xbmc-10.1a/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp
--- xbmc-10.1/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp	2011-08-01 10:16:58.833552017 +0200
+++ xbmc-10.1a/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp	2011-08-01 10:17:12.254555201 +0200
@@ -32,10 +32,10 @@
 #include "DVDStreamInfo.h"
 #include "utils/log.h"
 
-#include <OMX_Core.h>
-#include <OMX_Component.h>
-#include <OMX_Index.h>
-#include <OMX_Image.h>
+#include <OpenMAX/IL/OMX_Core.h>
+#include <OpenMAX/IL/OMX_Component.h>
+#include <OpenMAX/IL/OMX_Index.h>
+#include <OpenMAX/IL/OMX_Image.h>
 
 #define CLASSNAME "COpenMax"
 // EGL extension functions
diff -Naur xbmc-10.1/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.h xbmc-10.1a/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.h
--- xbmc-10.1/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.h	2011-03-05 04:33:37.000000000 +0100
+++ xbmc-10.1a/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.h	2011-08-01 10:17:49.437053792 +0200
@@ -31,7 +31,7 @@
 #include <semaphore.h>
 #include <EGL/egl.h>
 #include <EGL/eglext.h>
-#include <OMX_Core.h>
+#include <OpenMAX/IL/OMX_Core.h>
 
 typedef struct omx_codec_capability {
     // level is OMX_VIDEO_AVCPROFILETYPE, OMX_VIDEO_H263PROFILETYPE, 
