diff -Naur xbmc-10.1.orig//Makefile.in xbmc-10.1/Makefile.in
--- xbmc-10.1.orig//Makefile.in	2011-06-25 23:14:42.210777232 +0200
+++ xbmc-10.1/Makefile.in	2011-06-25 23:15:38.900777252 +0200
@@ -104,13 +104,11 @@
 LIB_DIRS+=lib/libmicrohttpd
 endif
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
 ifeq ($(findstring osx,$(ARCH)), osx)
 VIS_DIRS+=xbmc/visualizations/iTunes
 endif
@@ -308,7 +306,6 @@
 ifneq (arm, $(ARCH))
 	$(MAKE) -C xbmc/visualizations/OpenGLSpectrum
 	$(MAKE) -C xbmc/visualizations/WaveForm
-	$(MAKE) -C xbmc/visualizations/XBMCProjectM
 ifeq ($(findstring osx,$(ARCH)), osx)
 	$(MAKE) -C xbmc/visualizations/iTunes
 endif
@@ -319,9 +316,7 @@
 endif
 endif
 screensavers: exports
-ifneq (arm, $(ARCH))
-	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
-endif
+
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
