diff -Naur xbmc-10.0.orig//Makefile.in xbmc-10.0/Makefile.in
--- xbmc-10.0.orig//Makefile.in	2010-12-20 00:00:49.635073993 +0100
+++ xbmc-10.0/Makefile.in	2010-12-23 00:09:36.919329001 +0100
@@ -104,13 +104,11 @@
 LIB_DIRS+=lib/libmicrohttpd
 endif
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
 
 VIS_DIRS=\
 	xbmc/visualizations/OpenGLSpectrum \
-	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM
+	xbmc/visualizations/WaveForm
 ifeq ($(findstring osx,$(ARCH)), osx)
 VIS_DIRS+=xbmc/visualizations/iTunes
 endif
@@ -319,9 +317,7 @@
 endif
 endif
 screensavers: exports
-ifneq (arm, $(ARCH))
-	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
-endif
+
 libpython: dllloader
 	$(MAKE) -C xbmc/lib/libPython
 	$(MAKE) -C xbmc/lib/libPython/xbmcmodule
