diff --git a/xbmc/utils/CPUInfo.cpp b/xbmc/utils/CPUInfo.cpp
index 76ddb76..c9be950 100644
--- a/xbmc/utils/CPUInfo.cpp
+++ b/xbmc/utils/CPUInfo.cpp
@@ -241,6 +241,9 @@
     m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/temp1_input", "r");
   if (m_fProcTemperature == NULL)   
     m_fProcTemperature = fopen("/sys/class/thermal/thermal_zone0/temp", "r");  // On Raspberry PIs
+  // some systems put this information here ...
+  if (m_fProcTemperature == NULL)   
+    m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/device/temp1_input", "r");
 
   m_fCPUFreq = fopen ("/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq", "r");
 
diff --git a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
index 6cad1fd..6826f51 100644
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -90,7 +90,7 @@
 CStdString CSysInfoJob::GetCPUFreqInfo()
 {
   double CPUFreq = GetCPUFrequency();
-  return StringUtils::Format("%4.2fMHz", CPUFreq);;
+  return (CPUFreq > 0) ?  StringUtils::Format("%4.2fMHz", CPUFreq) : CStdString();
 }
 
 CSysData::INTERNET_STATE CSysInfoJob::GetInternetState()
