diff -Naur xbmc-10.0.orig//xbmc/RenderSystemGLES.cpp xbmc-10.0/xbmc/RenderSystemGLES.cpp
--- xbmc-10.0.orig//xbmc/RenderSystemGLES.cpp	2010-12-20 00:01:10.745073996 +0100
+++ xbmc-10.0/xbmc/RenderSystemGLES.cpp	2010-12-20 00:28:34.995073994 +0100
@@ -206,7 +206,7 @@
   }
 }
 
-static int64_t abs(int64_t a)
+static int64_t _abs(int64_t a)
 {
   if(a < 0)
     return -a;
@@ -250,7 +250,7 @@
     diff = curr - m_iSwapStamp;
     m_iSwapStamp = curr;
 
-    if (abs(diff - m_iSwapRate) < abs(diff))
+    if (_abs(diff - m_iSwapRate) < _abs(diff))
       CLog::Log(LOGDEBUG, "%s - missed requested swap",__FUNCTION__);
   }
   
