diff -Naur a/xbmc/windowing/egl/WinSystemGLES.cpp b/xbmc/windowing/egl/WinSystemGLES.cpp
--- a/xbmc/windowing/egl/WinSystemGLES.cpp	2012-06-02 09:41:05.000000000 +0200
+++ b/xbmc/windowing/egl/WinSystemGLES.cpp	2012-06-02 10:26:45.430295461 +0200
@@ -20,7 +20,7 @@
  */
 #include "system.h"
 
-#ifdef HAS_EGL
+#ifdef HAS_EGLGLES
 
 #include "WinSystemGLES.h"
 #include "filesystem/SpecialProtocol.h"
