diff -Naur xbmc-10.0.orig//xbmc/cores/VideoRenderers/OverlayRendererGL.cpp xbmc-10.0/xbmc/cores/VideoRenderers/OverlayRendererGL.cpp
--- xbmc-10.0.orig//xbmc/cores/VideoRenderers/OverlayRendererGL.cpp	2010-12-20 00:01:02.795073995 +0100
+++ xbmc-10.0/xbmc/cores/VideoRenderers/OverlayRendererGL.cpp	2010-12-20 00:39:08.225073997 +0100
@@ -125,7 +125,11 @@
             , o->width * 4
             , &m_u, &m_v
             , GL_RGBA
+#ifndef HAS_GLES
             , GL_BGRA
+#else
+            , GL_RGBA
+#endif
             , rgba);
   free(rgba);
 
@@ -199,7 +203,11 @@
             , o->width * 4
             , &m_u, &m_v
             , GL_RGBA
+#ifndef HAS_GLES
             , GL_BGRA
+#else
+            , GL_RGBA
+#endif
             , rgba + min_x + min_y * o->width);
 
   free(rgba);
