--- a/xbmc/powermanagement/PowerManager.h	2013-10-23 20:12:12.728657068 +0200
+++ b/xbmc/powermanagement/PowerManager.h	2013-10-23 20:12:55.068655818 +0200
@@ -32,8 +32,8 @@
   virtual bool Reboot()       { return false; }
 
   virtual bool CanPowerdown() { return true; }
-  virtual bool CanSuspend()   { return true; }
-  virtual bool CanHibernate() { return true; }
+  virtual bool CanSuspend()   { return false; }
+  virtual bool CanHibernate() { return false; }
   virtual bool CanReboot()    { return true; }
 
   virtual int  BatteryLevel() { return 0; }
