diff -Naur xbmc-10.0.orig//Makefile.in xbmc-10.0/Makefile.in
--- xbmc-10.0.orig//Makefile.in	2010-12-20 00:00:49.635073993 +0100
+++ xbmc-10.0/Makefile.in	2010-12-20 00:48:31.815073998 +0100
@@ -467,8 +467,8 @@
 	xbmc/cores/DllLoader/exports/exports.a
 
 ifeq (arm, $(ARCH))
-# Force external python2.5 for now!
-LIBS +=-lpython2.5
+# Force external python2.7 for now!
+LIBS +=-lpython2.7
 endif
 
 xbmc.bin: $(OBJSXBMC) $(DYNOBJSXBMC)
