--- a/xbmc/cores/AudioEngine/Utils/AEConvert.cpp	2014-04-17 08:25:30.494948270 +0200
+++ b/xbmc/cores/AudioEngine/Utils/AEConvert.cpp	2014-04-17 08:29:24.078938939 +0200
@@ -247,7 +247,7 @@
 {
   for (unsigned int i = 0; i < samples; ++i, data += 3)
   {
-    int s = (data[1] << 24) | (data[2] << 16) | (data[3] << 8);
+    int s = (data[0] << 24) | (data[1] << 16) | (data[2] << 8);
     *dest++ = (float)s * INT32_SCALE;
   }
   return samples;
@@ -886,7 +886,7 @@
   {
     const uint32_t odd = samples - even;
     if (odd == 1)
-      dst[0] = safeRound(data[0] * ((float)INT24_MAX+.5f)) & 0xFFFFFF;
+      dst[0] = (safeRound(data[0] * ((float)INT24_MAX+.5f)) & 0xFFFFFF) << leftShift;
     else
     {
       __m128 in;
