diff -Naur xbmc-10.0.orig//xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp xbmc-10.0/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp
--- xbmc-10.0.orig//xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp	2010-12-20 00:00:59.125073997 +0100
+++ xbmc-10.0/xbmc/cores/dvdplayer/DVDCodecs/Video/OpenMax.cpp	2010-12-21 00:28:27.030785986 +0100
@@ -112,6 +112,20 @@
 #define FACTORFORMAT422 2
 #define FACTORFORMAT420 1.5
 
+// check for potentially undefined OpenMAX version numbers
+#ifndef OMX_VERSION_MAJOR
+#define OMX_VERSION_MAJOR 1
+#endif
+#ifndef OMX_VERSION_MINOR
+#define OMX_VERSION_MINOR 1
+#endif
+#ifndef OMX_VERSION_REVISION
+#define OMX_VERSION_REVISION 2
+#endif
+#ifndef OMX_VERSION_STEP
+#define OMX_VERSION_STEP 0
+#endif
+
 #define OMX_INIT_STRUCTURE(a) \
   memset(&(a), 0, sizeof(a)); \
   (a).nSize = sizeof(a); \
