diff -Naur xbmc-10.0/config.h.in xbmc-10.0a/config.h.in
--- xbmc-10.0/config.h.in	2010-12-19 15:08:46.000000000 +0100
+++ xbmc-10.0a/config.h.in	2011-02-20 10:21:08.003170977 +0100
@@ -243,6 +243,9 @@
 /* Define to 1 if you have the 'python2.6' library. */
 #undef HAVE_LIBPYTHON2_6
 
+/* Define to 1 if you have the 'python2.7' library. */
+#undef HAVE_LIBPYTHON2_7
+
 /* Define to 1 if you have the `resolv' library (-lresolv). */
 #undef HAVE_LIBRESOLV
 
diff -Naur xbmc-10.0/configure xbmc-10.0a/configure
--- xbmc-10.0/configure	2011-02-20 10:20:14.607168326 +0100
+++ xbmc-10.0a/configure	2011-02-20 10:21:08.003170977 +0100
@@ -613,6 +613,7 @@
 USE_PYTHON2_4
 USE_PYTHON2_5
 USE_PYTHON2_6
+USE_PYTHON2_7
 USE_EXTERNAL_PYTHON
 USE_EXTERNAL_LIBWAVPACK
 USE_EXTERNAL_LIBMPEG2
@@ -12454,6 +12455,42 @@
 
 # External Python
 if test "$use_external_python" = "yes"; then
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lpython2.7" >&5
+$as_echo_n "checking for main in -lpython2.7... " >&6; }
+if test "${ac_cv_lib_python2_7_main+set}" = set; then :
+  $as_echo_n "(cached) " >&6
+else
+  ac_check_lib_save_LIBS=$LIBS
+LIBS="-lpython2.7  $LIBS"
+cat confdefs.h - <<_ACEOF >conftest.$ac_ext
+/* end confdefs.h.  */
+
+
+int
+main ()
+{
+return main ();
+  ;
+  return 0;
+}
+_ACEOF
+if ac_fn_c_try_link "$LINENO"; then :
+  ac_cv_lib_python2_7_main=yes
+else
+  ac_cv_lib_python2_7_main=no
+fi
+rm -f core conftest.err conftest.$ac_objext \
+    conftest$ac_exeext conftest.$ac_ext
+LIBS=$ac_check_lib_save_LIBS
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_python2_7_main" >&5
+$as_echo "$ac_cv_lib_python2_7_main" >&6; }
+if test "x$ac_cv_lib_python2_7_main" = x""yes; then :
+
+$as_echo "#define HAVE_LIBPYTHON2_7 1" >>confdefs.h
+
+    USE_PYTHON2_7=1
+else
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lpython2.6" >&5
 $as_echo_n "checking for main in -lpython2.6... " >&6; }
 if test "${ac_cv_lib_python2_6_main+set}" = set; then :
@@ -12464,7 +12501,6 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
-
 int
 main ()
 {
@@ -12569,6 +12605,7 @@
 
 fi
 
+fi
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: $external_python_enabled" >&5
 $as_echo "$as_me: $external_python_enabled" >&6;}
diff -Naur xbmc-10.0/xbmc/lib/libPython/linux/Makefile.in xbmc-10.0a/xbmc/lib/libPython/linux/Makefile.in
--- xbmc-10.0/xbmc/lib/libPython/linux/Makefile.in	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/linux/Makefile.in	2011-02-20 10:21:08.007266415 +0100
@@ -5,7 +5,10 @@
 SHELL=/bin/bash
 SYSDIR=../../../../system/python
 
-ifeq (@USE_PYTHON2_6@,1)
+ifeq (@USE_PYTHON2_7@,1)
+    PYVERSION=python2.7
+    SO=python27-$(ARCH).so
+else ifeq (@USE_PYTHON2_6@,1)
     PYVERSION=python2.6
     SO=python26-$(ARCH).so
 else ifeq (@USE_PYTHON2_5@,1)
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/action.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/action.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/action.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/action.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlbutton.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlbutton.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlbutton.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlbutton.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlcheckmark.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlfadelabel.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlgroup.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlgroup.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlgroup.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlgroup.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/control.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/control.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/control.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/control.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlimage.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlimage.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlimage.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlimage.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controllabel.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controllabel.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controllabel.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controllabel.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controllist.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controllist.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controllist.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controllist.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlprogress.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlprogress.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlprogress.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlprogress.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlradiobutton.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlslider.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlslider.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlslider.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlslider.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlspin.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlspin.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controlspin.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controlspin.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controltextbox.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controltextbox.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/controltextbox.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/controltextbox.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/dialog.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/dialog.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/dialog.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/dialog.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -24,7 +24,9 @@
 #endif
 #include "dialog.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/dialog.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/dialog.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/dialog.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/dialog.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/GUIPythonWindow.h	2011-02-20 10:21:08.007266415 +0100
@@ -26,7 +26,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/infotagmusic.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/infotagmusic.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/infotagmusic.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/infotagmusic.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/infotagvideo.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/infotagvideo.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/infotagvideo.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/infotagvideo.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/keyboard.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/keyboard.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/keyboard.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/keyboard.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/listitem.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/listitem.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/listitem.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/listitem.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/listitem.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/listitem.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/listitem.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/listitem.h	2011-02-20 10:21:08.007266415 +0100
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/player.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/player.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/player.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/player.h	2011-02-20 10:21:08.007266415 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+   #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyplaylist.cpp	2011-02-20 10:21:08.007266415 +0100
@@ -26,7 +26,9 @@
 #include "Util.h"
 #include "pyplaylist.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyplaylist.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyplaylist.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyplaylist.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyplaylist.h	2011-02-20 10:21:08.007266415 +0100
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/PythonAddon.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/PythonAddon.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/PythonAddon.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/PythonAddon.h	2011-02-20 10:21:08.007266415 +0100
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/PythonPlayer.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/PythonPlayer.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/PythonPlayer.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/PythonPlayer.h	2011-02-20 10:21:08.011704017 +0100
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyutil.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyutil.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/pyutil.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/pyutil.h	2011-02-20 10:21:08.011704017 +0100
@@ -25,7 +25,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/window.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/window.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/window.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/window.h	2011-02-20 10:21:08.011704017 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxml.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxml.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxml.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxml.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -24,7 +24,9 @@
 #endif
 #include "winxml.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxmldialog.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -24,7 +24,9 @@
 #endif
 #include "winxml.h"
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxml.h xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxml.h
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/winxml.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/winxml.h	2011-02-20 10:21:08.011704017 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcaddonmodule.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcaddonmodule.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcaddonmodule.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcaddonmodule.cpp	2011-02-20 10:21:53.811227108 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcguimodule.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
     #include <python2.6/structmember.h>
   #elif (defined HAVE_LIBPYTHON2_5)
diff -Naur xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp
--- xbmc-10.0/xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/xbmcmodule/xbmcmodule.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/XBPython.cpp xbmc-10.0a/xbmc/lib/libPython/XBPython.cpp
--- xbmc-10.0/xbmc/lib/libPython/XBPython.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/XBPython.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -24,7 +24,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
@@ -61,7 +63,9 @@
 #define PYTHON_DLL "special://xbmcbin/system/python/python24-x86-osx.so"
 #endif
 #elif defined(__x86_64__)
-#if (defined HAVE_LIBPYTHON2_6)
+#if (defined HAVE_LIBPYTHON2_7)
+#define PYTHON_DLL "special://xbmcbin/system/python/python27-x86_64-linux.so"
+#elif (defined HAVE_LIBPYTHON2_6)
 #define PYTHON_DLL "special://xbmcbin/system/python/python26-x86_64-linux.so"
 #elif (defined HAVE_LIBPYTHON2_5)
 #define PYTHON_DLL "special://xbmcbin/system/python/python25-x86_64-linux.so"
@@ -69,7 +73,9 @@
 #define PYTHON_DLL "special://xbmcbin/system/python/python24-x86_64-linux.so"
 #endif
 #elif defined(_POWERPC)
-#if (defined HAVE_LIBPYTHON2_6)
+#if (defined HAVE_LIBPYTHON2_7)
+#define PYTHON_DLL "special://xbmcbin/system/python/python27-powerpc-linux.so"
+#elif (defined HAVE_LIBPYTHON2_6)
 #define PYTHON_DLL "special://xbmcbin/system/python/python26-powerpc-linux.so"
 #elif (defined HAVE_LIBPYTHON2_5)
 #define PYTHON_DLL "special://xbmcbin/system/python/python25-powerpc-linux.so"
@@ -77,7 +83,9 @@
 #define PYTHON_DLL "special://xbmcbin/system/python/python24-powerpc-linux.so"
 #endif
 #elif defined(_POWERPC64)
-#if (defined HAVE_LIBPYTHON2_6)
+#if (defined HAVE_LIBPYTHON2_7)
+#define PYTHON_DLL "special://xbmcbin/system/python/python27-powerpc64-linux.so"
+#elif (defined HAVE_LIBPYTHON2_6)
 #define PYTHON_DLL "special://xbmcbin/system/python/python26-powerpc64-linux.so"
 #elif (defined HAVE_LIBPYTHON2_5)
 #define PYTHON_DLL "special://xbmcbin/system/python/python25-powerpc64-linux.so"
@@ -85,7 +93,9 @@
 #define PYTHON_DLL "special://xbmcbin/system/python/python24-powerpc64-linux.so"
 #endif
 #elif defined(_ARMEL)
-#if (defined HAVE_LIBPYTHON2_6)
+#if (defined HAVE_LIBPYTHON2_7)
+#define PYTHON_DLL "special://xbmc/system/python/python27-arm.so"
+#elif (defined HAVE_LIBPYTHON2_6)
 #define PYTHON_DLL "special://xbmc/system/python/python26-arm.so"
 #elif (defined HAVE_LIBPYTHON2_5)
 #define PYTHON_DLL "special://xbmc/system/python/python25-arm.so"
@@ -93,7 +103,9 @@
 #define PYTHON_DLL "special://xbmc/system/python/python24-arm.so"
 #endif
 #else /* !__x86_64__ && !__powerpc__ */
-#if (defined HAVE_LIBPYTHON2_6)
+#if (defined HAVE_LIBPYTHON2_7)
+#define PYTHON_DLL "special://xbmcbin/system/python/python27-i486-linux.so"
+#elif (defined HAVE_LIBPYTHON2_6)
 #define PYTHON_DLL "special://xbmcbin/system/python/python26-i486-linux.so"
 #elif (defined HAVE_LIBPYTHON2_5)
 #define PYTHON_DLL "special://xbmcbin/system/python/python25-i486-linux.so"
diff -Naur xbmc-10.0/xbmc/lib/libPython/XBPythonDll.cpp xbmc-10.0a/xbmc/lib/libPython/XBPythonDll.cpp
--- xbmc-10.0/xbmc/lib/libPython/XBPythonDll.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/XBPythonDll.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -23,7 +23,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/pyconfig.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/pyconfig.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/XBPythonDllFuncs.S xbmc-10.0a/xbmc/lib/libPython/XBPythonDllFuncs.S
--- xbmc-10.0/xbmc/lib/libPython/XBPythonDllFuncs.S	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/XBPythonDllFuncs.S	2011-02-20 10:21:08.011704017 +0100
@@ -2,7 +2,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/pyconfig.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/pyconfig.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/pyconfig.h>
diff -Naur xbmc-10.0/xbmc/lib/libPython/XBPyThread.cpp xbmc-10.0a/xbmc/lib/libPython/XBPyThread.cpp
--- xbmc-10.0/xbmc/lib/libPython/XBPyThread.cpp	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/XBPyThread.cpp	2011-02-20 10:21:08.011704017 +0100
@@ -24,7 +24,10 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+    #include <python2.7/osdefs.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
     #include <python2.6/osdefs.h>
   #elif (defined HAVE_LIBPYTHON2_5)
diff -Naur xbmc-10.0/xbmc/lib/libPython/XBPyThread.h xbmc-10.0a/xbmc/lib/libPython/XBPyThread.h
--- xbmc-10.0/xbmc/lib/libPython/XBPyThread.h	2010-12-17 07:17:41.000000000 +0100
+++ xbmc-10.0a/xbmc/lib/libPython/XBPyThread.h	2011-02-20 10:21:08.011704017 +0100
@@ -26,7 +26,9 @@
   #include "config.h"
 #endif
 #if (defined USE_EXTERNAL_PYTHON)
-  #if (defined HAVE_LIBPYTHON2_6)
+  #if (defined HAVE_LIBPYTHON2_7)
+    #include <python2.7/Python.h>
+  #elif (defined HAVE_LIBPYTHON2_6)
     #include <python2.6/Python.h>
   #elif (defined HAVE_LIBPYTHON2_5)
     #include <python2.5/Python.h>
