diff -Nur acx-20060521.orig/common.c acx-20060521/common.c
--- acx-20060521.orig/common.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/common.c	2006-12-08 16:21:12.000000000 +0100
@@ -30,7 +30,6 @@
 ** ---------------------------------------------------------------------
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/module.h>
 #include <linux/kernel.h>
diff -Nur acx-20060521.orig/conv.c acx-20060521/conv.c
--- acx-20060521.orig/conv.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/conv.c	2006-12-08 16:21:21.000000000 +0100
@@ -30,7 +30,6 @@
 ** ---------------------------------------------------------------------
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/skbuff.h>
 #include <linux/if_arp.h>
diff -Nur acx-20060521.orig/ioctl.c acx-20060521/ioctl.c
--- acx-20060521.orig/ioctl.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/ioctl.c	2006-12-08 16:21:29.000000000 +0100
@@ -30,7 +30,6 @@
 ** ---------------------------------------------------------------------
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/types.h>
diff -Nur acx-20060521.orig/pci.c acx-20060521/pci.c
--- acx-20060521.orig/pci.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/pci.c	2006-12-08 16:21:37.000000000 +0100
@@ -31,7 +31,6 @@
 */
 #define ACX_PCI 1
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/compiler.h> /* required for Lx 2.6.8 ?? */
 #include <linux/kernel.h>
@@ -51,6 +51,7 @@
 #include <linux/pm.h>
 #include <linux/vmalloc.h>
 #include <linux/dma-mapping.h>
+#include <linux/utsrelease.h>

 #include "acx.h"

diff -Nur acx-20060521.orig/usb.c acx-20060521/usb.c
--- acx-20060521.orig/usb.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/usb.c	2006-12-08 16:21:52.000000000 +0100
@@ -45,7 +45,6 @@
 #define ACX_USB 1
 
 #include <linux/version.h>
-#include <linux/config.h>
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/moduleparam.h>
diff -Nur acx-20060521.orig/wlan.c acx-20060521/wlan.c
--- acx-20060521.orig/wlan.c	2006-12-08 16:20:30.000000000 +0100
+++ acx-20060521/wlan.c	2006-12-08 16:20:55.000000000 +0100
@@ -37,7 +37,6 @@
 ** http://www.linux-wlan.com
 */
 
-#include <linux/config.h>
 #include <linux/version.h>
 #include <linux/types.h>
 #include <linux/if_arp.h>
