* from the gentoo package, cf. http://bugs.gentoo.org/185845

--- acx-20070101/pci.c.orig	2007-07-24 03:57:41.000000000 -0700
+++ acx-20070101/pci.c	2007-07-24 03:59:20.000000000 -0700
@@ -2104,7 +2104,11 @@
 /* TODO: pci_set_power_state(pdev, PCI_D0); ? */
 
 	/* request shared IRQ handler */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,21)	
+	if (request_irq(ndev->irq, acxpci_i_interrupt, IRQF_SHARED, ndev->name, ndev)) {
+#else
 	if (request_irq(ndev->irq, acxpci_i_interrupt, SA_SHIRQ, ndev->name, ndev)) {
+#endif		
 		printk("%s: request_irq FAILED\n", ndev->name);
 		result = -EAGAIN;
 		goto done;
--- acx-20070101/common.c.orig	2007-07-24 03:59:40.000000000 -0700
+++ acx-20070101/common.c	2007-07-24 04:01:23.000000000 -0700
@@ -2753,8 +2753,11 @@
 
 	skb->dev = adev->ndev;
 	skb->dev->last_rx = jiffies;
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22)
+	skb_reset_mac_header(skb);
+#else
 	skb->mac.raw = skb->data;
+#endif
 	skb->ip_summed = CHECKSUM_NONE;
 	skb->pkt_type = PACKET_OTHERHOST;
 	skb->protocol = htons(ETH_P_80211_RAW);
