--- acx-20071003.orig/pci.c	2007-07-24 03:57:41.000000000 -0700
+++ acx-20071003/pci.c	2007-07-24 03:59:20.000000000 -0700
@@ -2116,7 +2104,11 @@
 /* TODO: pci_set_power_state(pdev, PCI_D0); ? */
 
 	/* request shared IRQ handler */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,21)	
+	if (request_irq(ndev->irq, acxpci_i_interrupt, IRQF_SHARED, ndev->name, ndev)) {
+#else
 	if (request_irq(ndev->irq, acxpci_i_interrupt, SA_SHIRQ, ndev->name, ndev)) {
+#endif		
 		printk("%s: request_irq FAILED\n", ndev->name);
 		result = -EAGAIN;
 		goto done;

--- acx-20071003.orig/usb.c	2007-12-13 17:03:21.000000000 +0100
+++ acx-20071003/usb.c	2007-12-13 17:03:24.000000000 +0100
@@ -859,7 +859,6 @@
 	ndev->watchdog_timeo = 4 * HZ;
 #endif
 	ndev->change_mtu = &acx_e_change_mtu;
-	SET_MODULE_OWNER(ndev);
 
 	/* Setup private driver context */
 
