--- acx-20060521/ioctl.c~	2006-12-09 22:52:26.000000000 +0100
+++ acx-20060521/ioctl.c	2006-12-09 22:55:26.000000000 +0100
@@ -675,8 +675,8 @@
 
 		if (len > sizeof(adev->essid))
 			len = sizeof(adev->essid);
-		memcpy(adev->essid, extra, len-1);
-		adev->essid[len-1] = '\0';
+		memcpy(adev->essid, extra, len);
+		adev->essid[len] = '\0';
 		/* Paranoia: just in case there is a '\0'... */
 		adev->essid_len = strlen(adev->essid);
 		adev->essid_active = 1;
