diff -Naur acx-20080831.orig/acx_struct.h acx-20080831/acx_struct.h
--- acx-20080831.orig/acx_struct.h	2010-07-31 15:25:28.826487669 +0200
+++ acx-20080831/acx_struct.h	2010-07-31 15:26:05.297737917 +0200
@@ -83,6 +83,7 @@
 /* Use worker_queues for 2.5/2.6 kernels and queue tasks for 2.4 kernels
    (used for the 'bottom half' of the interrupt routine) */
 
+#include <linux/semaphore.h>
 #include <linux/workqueue.h>
 #define USE_WORKER_TASKS
 #define WORK_STRUCT struct work_struct
