--- ./SConstruct	2010-03-22 14:16:04.000000000 +0100
+++ ../disko-demos-1.7.0_patched/SConstruct	2011-05-09 20:29:45.668146088 +0200
@@ -16,8 +16,6 @@
 	env['CXX'] = [os.environ['CXX'].split()] 
 if os.environ.has_key('CXXFLAGS'):
 	env['CCFLAGS'] = [os.environ['CXXFLAGS'].split()]
-if os.environ.has_key('LD'):
-	env['LINK'] = [os.environ['LD'].split()]
 if os.environ.has_key('LDFLAGS'):
 	env['LINKFLAGS'] = [os.environ['LDFLAGS'].split()]
 
