--- libmusicbrainz-5.0.1-git0749dd0a35/CMakeLists.txt	2012-09-21 09:23:26.000000000 +0200
+++ libmusicbrainz-5.0.1-git0749dd0a35/CMakeLists.txt-new	2012-09-28 19:32:37.605790805 +0200
@@ -36,8 +36,6 @@
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libmusicbrainz5.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
 
 ADD_SUBDIRECTORY(src)
-ADD_SUBDIRECTORY(tests)
-ADD_SUBDIRECTORY(examples)
 
 ADD_CUSTOM_TARGET(docs
 	doxygen
