diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/NVOMX_IndexExtensions.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/NVOMX_IndexExtensions.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/NVOMX_IndexExtensions.h	2010-10-24 01:27:04.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/NVOMX_IndexExtensions.h	2011-08-01 10:26:29.829052406 +0200
@@ -26,8 +26,8 @@
 #ifndef _NVOMX_IndexExtensions_h_
 #define _NVOMX_IndexExtensions_h_
 
-#include <OMX_Core.h>
-#include <OMX_Component.h>
+#include <OpenMAX/IL/OMX_Core.h>
+#include <OpenMAX/IL/OMX_Component.h>
 
 #include "OMX_Core.h"
 
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Audio.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Audio.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Audio.h	2010-10-24 01:27:04.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Audio.h	2011-08-01 10:26:29.837052293 +0200
@@ -39,7 +39,7 @@
  *  for this header file to compile successfully 
  */
 
-#include <OMX_Core.h>
+#include <OpenMAX/IL/OMX_Core.h>
 
 /** @defgroup midi MIDI
  * @ingroup audio
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Component.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Component.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Component.h	2010-10-24 01:27:04.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Component.h	2011-08-01 10:26:29.837052293 +0200
@@ -41,10 +41,10 @@
  *  for this header file to compile successfully 
  */
 
-#include <OMX_Audio.h>
-#include <OMX_Video.h>
-#include <OMX_Image.h>
-#include <OMX_Other.h>
+#include <OpenMAX/IL/OMX_Audio.h>
+#include <OpenMAX/IL/OMX_Video.h>
+#include <OpenMAX/IL/OMX_Image.h>
+#include <OpenMAX/IL/OMX_Other.h>
 
 /** @ingroup comp */
 typedef enum OMX_PORTDOMAINTYPE { 
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Core.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Core.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Core.h	2010-10-24 01:27:04.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Core.h	2011-08-01 10:26:29.845051512 +0200
@@ -39,7 +39,7 @@
  *  for this header file to compile successfully 
  */
 
-#include <OMX_Index.h>
+#include <OpenMAX/IL/OMX_Index.h>
 
 
 /** The OMX_COMMANDTYPE enumeration is used to specify the action in the
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Image.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Image.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Image.h	2010-10-24 01:27:05.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Image.h	2011-08-01 10:26:29.845051512 +0200
@@ -39,7 +39,7 @@
  * for this header file to compile successfully 
  */
 
-#include <OMX_IVCommon.h>
+#include <OpenMAX/IL/OMX_IVCommon.h>
 
 /** @defgroup imaging OpenMAX IL Imaging Domain
  * @ingroup iv
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Index.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Index.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Index.h	2010-10-24 01:27:05.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Index.h	2011-08-01 10:26:29.849053452 +0200
@@ -39,7 +39,7 @@
  *  header to compile without errors.  The includes below are required
  *  for this header file to compile successfully 
  */
-#include <OMX_Types.h>
+#include <OpenMAX/IL/OMX_Types.h>
 
 
 /** The OMX_INDEXTYPE enumeration is used to select a structure when either
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_IVCommon.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_IVCommon.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_IVCommon.h	2010-10-24 01:27:05.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_IVCommon.h	2011-08-01 10:26:29.853051898 +0200
@@ -39,7 +39,7 @@
  * file to compile successfully 
  */
 
-#include <OMX_Core.h>
+#include <OpenMAX/IL/OMX_Core.h>
 
 /** @defgroup iv OpenMAX IL Imaging and Video Domain
  * Common structures for OpenMAX IL Imaging and Video domains
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Other.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Other.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Other.h	2010-10-24 01:27:05.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Other.h	2011-08-01 10:26:29.853051898 +0200
@@ -39,7 +39,7 @@
  *  for this header file to compile successfully 
  */
 
-#include <OMX_Core.h>
+#include <OpenMAX/IL/OMX_Core.h>
 
 
 /** 
diff -Naur tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Video.h tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Video.h
--- tegra-openmax-includes-10.9.3_20101029/usr/include/OpenMAX/IL/OMX_Video.h	2010-10-24 01:27:05.000000000 +0200
+++ tegra-openmax-includes-10.9.3_20101029a//usr/include/OpenMAX/IL/OMX_Video.h	2011-08-01 10:26:29.857052422 +0200
@@ -46,7 +46,7 @@
  * for this header file to compile successfully 
  */
 
-#include <OMX_IVCommon.h>
+#include <OpenMAX/IL/OMX_IVCommon.h>
 
 
 /**
