diff -Naur djmount-0.30.orig/djmount/device_list.c djmount-0.30/djmount/device_list.c
--- djmount-0.30.orig/djmount/device_list.c	2005-09-06 22:46:36.000000000 +0200
+++ djmount-0.30/djmount/device_list.c	2005-10-09 16:51:23.000000000 +0200
@@ -35,7 +35,7 @@
 #include <upnp/ithread.h>
 #include <upnp/upnptools.h>
 #include <upnp/LinkedList.h>
-
+#include <unistd.h>
 
 static UpnpClient_Handle g_ctrlpt_handle = -1;
 
@@ -456,6 +456,7 @@
 	AddDevice (d_event->DeviceId, d_event->Location, d_event->Expires);
 	Log_Print (LOG_DEBUG, "Discovery: DeviceList after AddDevice =");
 	DeviceList_PrintStatus (LOG_DEBUG);
+        symlink ("/tmp/UPnP", "/mnt/UPnP");
       }
       
       break;
