diff -Nur djmount-0.70.orig/djmount/device_list.c djmount-0.70/djmount/device_list.c
--- djmount-0.70.orig/djmount/device_list.c	Tue Jul 25 22:33:12 2006
+++ djmount-0.70/djmount/device_list.c	Sat Aug 26 11:36:43 2006
@@ -35,6 +35,7 @@
 #include "service.h"
 #include "talloc_util.h"
 
+#include <unistd.h>
 #include <stdbool.h>
 #include <upnp/upnp.h>
 #include <upnp/ithread.h>
@@ -478,6 +479,7 @@
 			Log_Printf (LOG_DEBUG, "Discovery: "
 				    "DeviceList after AddDevice = \n%s",
 				    DeviceList_GetStatusString (tmp_ctx));
+			symlink ("/tmp/UPnP", "/mnt/UPnP");
 		}
 		
 		break;
