diff -Naur djmount-0.50.orig/djmount/device_list.c djmount-0.50/djmount/device_list.c
--- djmount-0.50.orig/djmount/device_list.c	2005-12-06 23:05:02.000000000 +0100
+++ djmount-0.50/djmount/device_list.c	2005-12-10 12:01:09.000000000 +0100
@@ -35,6 +35,7 @@
 #include "service.h"
 #include "talloc_util.h"
 
+#include <unistd.h>
 #include <stdbool.h>
 #include <upnp/upnp.h>
 #include <upnp/ithread.h>
@@ -462,6 +463,7 @@
 	AddDevice (d_event->DeviceId, d_event->Location, d_event->Expires);
 	Log_Print (LOG_DEBUG, "Discovery: DeviceList after AddDevice =");
 	DeviceList_PrintStatus (LOG_DEBUG);
+	symlink ("/tmp/UPnP", "/mnt/UPnP");
       }
       
       break;
