diff -Naur djmount-0.31.orig/djmount/djfs.c djmount-0.31/djmount/djfs.c
--- djmount-0.31.orig/djmount/djfs.c	2005-10-16 10:37:08.000000000 +0200
+++ djmount-0.31/djmount/djfs.c	2005-10-28 18:14:43.000000000 +0200
@@ -374,11 +374,6 @@
       for (i = 0; i < names->nb; i++) {
 	const char* const devName = names->str[i];
 	DIR_BEGIN(devName) {
-	  FILE_BEGIN("status") {
-	    talloc_string = DeviceList_GetDeviceStatusString (talloc_context,
-							      devName, true);
-	  } FILE_END;
-	  DIR_BEGIN("browse") {
 	    size_t nb_matched = 0;
 	    const ContentDir_BrowseResult* const res = 
 	      _DJFS_BrowseCDS (tmp_ctx, devName, ptr, &nb_matched);
@@ -418,7 +413,6 @@
 		}
 	      } DIR_END;
 	    }
-	  } DIR_END; // "browse"
 	} DIR_END; // devName
       }
     }
