diff -Naur djmount-0.30.orig/djmount/djfs.c djmount-0.30/djmount/djfs.c
--- djmount-0.30.orig/djmount/djfs.c	2005-09-04 19:49:43.000000000 +0200
+++ djmount-0.30/djmount/djfs.c	2005-10-09 22:40:53.000000000 +0200
@@ -373,11 +373,6 @@
       for (i = 0; i < names->nb; i++) {
 	const char* const devName = names->str[i];
 	DIR_BEGIN(devName) {
-	  FILE_BEGIN("status") {
-	    talloc_string = DeviceList_GetDeviceStatusString (talloc_context,
-							      devName);
-	  } FILE_END;
-	  DIR_BEGIN("browse") {
 	    size_t nb_matched = 0;
 	    const ContentDir_BrowseResult* const res = 
 	      _DJFS_BrowseCDS (tmp_ctx, devName, ptr, &nb_matched);
@@ -417,7 +412,6 @@
 		}
 	      } DIR_END;
 	    }
-	  } DIR_END; // "browse"
 	} DIR_END; // devName
       }
     }
