diff -Naur djmount-0.71.orig/configure djmount-0.71/configure
--- djmount-0.71.orig/configure	2007-12-15 18:30:05.000000000 +0100
+++ djmount-0.71/configure	2007-12-15 18:31:05.000000000 +0100
@@ -26450,6 +26450,7 @@
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
+export rt_package_find_fuse_link_ok=yes
 echo "$as_me:$LINENO: result: $rt_package_find_fuse_link_ok" >&5
 echo "${ECHO_T}$rt_package_find_fuse_link_ok" >&6
 if test x"$rt_package_find_fuse_link_ok" = xyes ; then
