diff -Nur djmount-0.70.orig/djmount/djfs.c djmount-0.70/djmount/djfs.c
--- djmount-0.70.orig/djmount/djfs.c	Wed Jul 26 23:43:16 2006
+++ djmount-0.70/djmount/djfs.c	Sat Aug 26 11:40:42 2006
@@ -291,21 +291,6 @@
 	  }
 	}
       } PTR_ARRAY_FOR_EACH_PTR_END;
-
-      if ( (self->flags & DJFS_SHOW_METADATA) && 
-	   !PtrArray_IsEmpty(children->objects) ) {
-	DIR_BEGIN (".metadata") {
-	  PTR_ARRAY_FOR_EACH_PTR (children->objects, o) {
-	    char* const name = MediaFile_GetName (tmp_ctx, o, "xml");
-	    FILE_BEGIN (name) {
-	      const char* const str = talloc_asprintf
-		(tmp_ctx, "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n%s",
-		 DIDLObject_GetElementString (o, tmp_ctx));
-	      FILE_SET_STRING (str, FILE_BUFFER_STRING_STEAL);
-	    } FILE_END;
-	  } PTR_ARRAY_FOR_EACH_PTR_END;
-	} DIR_END;
-      }
     } // if children
     
     if (searchable && parent->searchable) {
