diff -Naur djmount-0.30.orig/djmount/djfs.c djmount-0.30/djmount/djfs.c
--- djmount-0.30.orig/djmount/djfs.c	2005-10-09 22:41:11.000000000 +0200
+++ djmount-0.30/djmount/djfs.c	2005-10-09 22:42:25.000000000 +0200
@@ -398,15 +398,6 @@
 							GET_CONTENT);
 		      } FILE_END;
 		    }
-		    char* name = talloc_asprintf (tmp_ctx, "%s.xml", o->title);
-		    FILE_BEGIN (name) {
-		      talloc_string = 
-			talloc_asprintf 
-			(talloc_context, 
-			 "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n%s",
-			 XMLUtil_GetNodeString (tmp_ctx, 
-						(IXML_Node*) o->element));
-		    } FILE_END;
 		  }
 		  o = o->next;
 		}
