diff -Naur djmount-0.50.orig/djmount/djfs.c djmount-0.50/djmount/djfs.c
--- djmount-0.50.orig/djmount/djfs.c	2005-12-08 18:30:53.000000000 +0100
+++ djmount-0.50/djmount/djfs.c	2005-12-10 12:05:42.000000000 +0100
@@ -356,15 +356,6 @@
 			} FILE_END;
 		      }
 		    }
-		    char* const name = MediaFile_GetName (tmp_ctx, o, "xml");
-		    FILE_BEGIN (name) {
-		      const char* const str = talloc_asprintf 
-			(tmp_ctx, 
-			 "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n%s",
-			 DIDLObject_GetElementString (o, tmp_ctx));
-		      FILE_SET_STRING (str);
-		      FILE_SET_SIZE (str ? strlen (str) : 0);
-		    } FILE_END;
 		  }
 		} PTR_LIST_FOR_EACH_PTR_END;
 	      } DIR_END;
