diff -Nur djmount-0.40.orig/djmount/Makefile.in djmount-0.40/djmount/Makefile.in
--- djmount-0.40.orig/djmount/Makefile.in	Sat Nov 12 16:12:28 2005
+++ djmount-0.40/djmount/Makefile.in	Sat Nov 12 18:27:47 2005
@@ -62,7 +62,6 @@
 @INTERNAL_LIBUPNP_TRUE@am__append_1 = -I$(UPNP_TMPINC)
 bin_PROGRAMS = djmount$(EXEEXT)
 check_PROGRAMS = test_charset$(EXEEXT) test_upnp$(EXEEXT)
-@WANT_ICONV_FALSE@am__append_2 = charset_internal.c
 subdir = djmount
 DIST_COMMON = $(noinst_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in
@@ -89,11 +88,10 @@
 am__djmount_SOURCES_DIST = log.c object.c service.c device.c \
 	device_list.c content_dir.c djfs.c upnp_util.c string_util.c \
 	xml_util.c charset.c charset_internal.c fuse_main.c
-@WANT_ICONV_FALSE@am__objects_1 = charset_internal.$(OBJEXT)
 am__objects_2 = log.$(OBJEXT) object.$(OBJEXT) service.$(OBJEXT) \
 	device.$(OBJEXT) device_list.$(OBJEXT) content_dir.$(OBJEXT) \
 	djfs.$(OBJEXT) upnp_util.$(OBJEXT) string_util.$(OBJEXT) \
-	xml_util.$(OBJEXT) charset.$(OBJEXT) $(am__objects_1)
+	xml_util.$(OBJEXT) $(am__objects_1)
 am_djmount_OBJECTS = $(am__objects_2) fuse_main.$(OBJEXT)
 djmount_OBJECTS = $(am_djmount_OBJECTS)
 djmount_LDADD = $(LDADD)
@@ -263,13 +261,13 @@
 		  	  $(LIBUPNP_CFLAGS)
 
 LDADD = $(TALLOC_LIBS) $(FUSE_LIBS) \
-			  $(LIBUPNP_LIBS) $(LIBICONV)
+			  $(LIBUPNP_LIBS)
 
 # TBD auto run test ?
 # TESTS			= test_charset
 COMMON_SRCS = log.c object.c service.c device.c device_list.c \
 	content_dir.c djfs.c upnp_util.c string_util.c xml_util.c \
-	charset.c $(am__append_2)
+	$(am__append_2)
 noinst_HEADERS = \
 			log.h object.h object_p.h service.h service_p.h \
 		  	device.h device_list.h \
@@ -387,8 +385,6 @@
 distclean-compile:
 	-rm -f *.tab.c
 
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/charset.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/charset_internal.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/content_dir.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/device.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/device_list.Po@am__quote@
diff -Nur djmount-0.40.orig/djmount/fuse_main.c djmount-0.40/djmount/fuse_main.c
--- djmount-0.40.orig/djmount/fuse_main.c	Sun Nov  6 21:59:10 2005
+++ djmount-0.40/djmount/fuse_main.c	Sat Nov 12 18:03:52 2005
@@ -48,7 +48,6 @@
 #include "string_util.h"
 #include "djfs.h"
 #include "content_dir.h"
-#include "charset.h"
 #include "minmax.h"
 
 
@@ -83,52 +82,14 @@
  * Charset conversions (display <-> UTF-8) for filesystem
  *****************************************************************************/
 
-typedef struct {
-	fuse_dirh_t    h;
-	fuse_dirfil_t  filler;
-} my_dir_handle;
-
-static int filler_from_utf8 (fuse_dirh_t h, const char *name, 
-			     int type, ino_t ino)
-{
-	// Convert filename to display charset
-	char buffer [NAME_MAX + 1];
-	char* display_name = Charset_ConvertString (CHARSET_FROM_UTF8, 
-						    name, 
-						    buffer, sizeof (buffer),
-						    NULL);
-	my_dir_handle* const my_h = (my_dir_handle*) h;
-	int rc = my_h->filler (my_h->h, display_name, type, ino);
-	if (display_name != buffer && display_name != name)
-		talloc_free (display_name);
-	return rc;
-}
-
 static int
 Browse (const char* path, 
 	/* for STAT => */	struct stat* stbuf, 
 	/* for GETDIR => */	fuse_dirh_t h, fuse_dirfil_t filler, 
 	/* for READ => */	void* talloc_context, char** file_content)
 {
-	int rc = -EIO;
-	if (! Charset_IsConverting()) {
-		rc = DJFS_Browse (path, stbuf, h, filler, 
+	return DJFS_Browse (path, stbuf, h, filler, 
 				  talloc_context, file_content);
-	} else {
-		// Convert filename from display charset 
-		char buffer [PATH_MAX];
-		char* const utf_path = Charset_ConvertString 
-			(CHARSET_TO_UTF8, path, buffer, sizeof (buffer), NULL);
-		my_dir_handle my_h = { .h = h, .filler = filler };
-		
-		rc = DJFS_Browse (utf_path, stbuf, 
-				  (filler ? (void*)&my_h : h), 
-				  (filler ? filler_from_utf8 : NULL),
-				  talloc_context, file_content);
-		if (utf_path != buffer && utf_path != path)
-			talloc_free (utf_path);
-	}
-	return rc;
 }
 
 /*****************************************************************************
@@ -570,7 +531,7 @@
 	}
 	
 	// Convert message to display charset, and print
-	Charset_PrintString (CHARSET_FROM_UTF8, msg, stdout);
+	fputs (msg, stdout);
 	Log_EndColor (level, stdout);
 	printf ("\n");
 }
@@ -601,7 +562,6 @@
      "    -f                     foreground operation (default: daemonized)\n"
      "\n"
      "Mount options (one or more comma separated options) :\n"
-     "    iocharset=<charset>    filenames encoding (default: environment)\n"
      "\n"
      "Debug levels are one or more comma separated words :\n"
 #ifdef DEBUG
@@ -671,7 +631,6 @@
 	/*
 	 * Handle options
 	 */
-	char* charset = NULL;
 	
 	char* fuse_argv[32] = { argv[0] };
 	int fuse_argc = 1;
@@ -706,15 +665,9 @@
 			for (s = strtok_r (options_copy, ",", &tokptr); 
 			     s != NULL; 
 			     s = strtok_r (NULL, ",", &tokptr)) {
-				if (strncmp (s, "iocharset=", 10) == 0) {
-					charset = talloc_strdup 
-						(talloc_autofree_context(), 
-						 s+10);
-				} else {
 					bad_usage (argv[0], 
 						   "unknown mount option '%s'",
 						   s); // ---------->
-				}
 			}
 			free (options_copy);
 			Log_Printf (LOG_INFO, "  Mount options = %s", options);
@@ -795,16 +748,6 @@
 	
 
 	/*
-	 * Set charset encoding
-	 */
-	rc = Charset_Initialize (charset);
-	if (rc) {
-		Log_Printf (LOG_ERROR, "Error initialising charset='%s'",
-			    NN(charset));
-	}
-
-
-	/*
 	 * Initialise UPnP Control point and starts FUSE file system
 	 */
 	
@@ -824,7 +767,6 @@
 	Log_Printf (LOG_DEBUG, "Shutting down ...");
 	DeviceList_Stop();
 	
-	Charset_Finish();
 	Log_Finish();
 
 	return rc; 
