diff -Naur djmount-0.51.orig/djmount/media_file.c djmount-0.51/djmount/media_file.c
--- djmount-0.51.orig/djmount/media_file.c	2006-01-10 22:41:22.000000000 +0100
+++ djmount-0.51/djmount/media_file.c	2006-02-12 18:41:38.000000000 +0100
@@ -73,6 +73,9 @@
 	{ "video/x-motion-jpeg",	   	"m3u",	"mjpg"	},
 	{ "video/quicktime",		   	"m3u",	"mov" 	},
 	{ "video/x-matroska",			"m3u",	"mkv"	},
+	{ "video/x-ms-wmv",			"m3u",	"wmv"	},
+	{ "video/mpeg",  			"m3u",	"mpg"	},
+	{ "video/mp2p",  			"m3u",	"vob"	},
 	{ "video/",			   	"m3u",	NULL  	},
 	{ "application/x-matroska",		"m3u",	"mkv"	},
 	{ "application/ogg",			"m3u",	"ogg"	},
@@ -87,7 +90,26 @@
 	{ "image/x-xpixmap",			NULL,	"xpm" 	},
 	{ "image/x-xbitmap",			NULL,	"xbm" 	},
 	{ "image/x-photo-cd",			NULL,	"pcd" 	},
+	{ "image/bmp",  			NULL,	"bmp" 	},
+	{ "image/gif",  			NULL,	"gif" 	},
+	{ "image/png",  			NULL,	"png" 	},
+	{ "image/x-quicktime", 			NULL,	"qti" 	},
+	{ "image/tiff", 			NULL,	"tif" 	},
 	{ "image/",				NULL,	NULL  	},
+	{ "text/srt", 	        		NULL,	"srt" 	},
+	{ "text/ssa", 	        		NULL,	"ssa" 	},
+	{ "text/psb", 	        		NULL,	"psb" 	},
+	{ "text/pjs", 	        		NULL,	"pjs" 	},
+	{ "text/sub", 	        		NULL,	"sub" 	},
+	{ "text/idx", 	        		NULL,	"idx" 	},
+	{ "text/dks", 	        		NULL,	"dks" 	},
+	{ "text/tts", 	        		NULL,	"tts" 	},
+	{ "text/vsf", 	        		NULL,	"vsf" 	},
+	{ "text/zeg", 	        		NULL,	"zeg" 	},
+	{ "text/mpl", 	        		NULL,	"mpl" 	},
+	{ "text/bup", 	        		NULL,	"bup" 	},
+	{ "text/ifo", 	        		NULL,	"ifo" 	},
+	{ "text/",				NULL,	NULL  	},
 	{ NULL,					NULL,	NULL  	}
 };
 
