diff -Nur djmount-0.71.orig/djmount/media_file.c djmount-0.71/djmount/media_file.c
--- djmount-0.71.orig/djmount/media_file.c	Sun Aug 27 23:12:20 2006
+++ djmount-0.71/djmount/media_file.c	Fri Nov 10 14:53:25 2006
@@ -90,6 +90,7 @@
 	{ "video/x-motion-jpeg",		"m3u",	"mjpg"	},
 	{ "video/quicktime",			"m3u",	"mov"	},
 	{ "video/x-matroska",			"m3u",	"mkv"	},
+	{ "video/x-ms-wmv",			"m3u",	"wmv"	},
 	{ "video/mpeg",				"m3u",	"mpg"	},
 	{ "video/mp2p",				"m3u",	"vob"	},
 	// Default for all other video files : asf, mpeg2, x-ms-wmv, ...
@@ -109,6 +110,9 @@
 	{ "image/x-xpixmap",			NULL,	"xpm"	},
 	{ "image/x-xbitmap",			NULL,	"xbm"	},
 	{ "image/x-photo-cd",			NULL,	"pcd"	},
+	{ "image/bmp",  			NULL,	"bmp" 	},
+	{ "image/gif",  			NULL,	"gif" 	},
+	{ "image/png",  			NULL,	"png" 	},
 	{ "image/x-quicktime",			NULL,	"qti"	}, 
 	{ "image/x-icon",			NULL,	"ico"	}, 
 	{ "image/tiff",				NULL,	"tif"	},
@@ -127,6 +131,19 @@
 	 */
 	{ "text/plain",				NULL,	"txt"	},
 	// Default for all text files : sub, idx, ssa, ifo, ...
+	{ "text/srt", 	        		NULL,	"srt" 	},
+	{ "text/ssa", 	        		NULL,	"ssa" 	},
+	{ "text/psb", 	        		NULL,	"psb" 	},
+	{ "text/pjs", 	        		NULL,	"pjs" 	},
+	{ "text/sub", 	        		NULL,	"sub" 	},
+	{ "text/idx", 	        		NULL,	"idx" 	},
+	{ "text/dks", 	        		NULL,	"dks" 	},
+	{ "text/tts", 	        		NULL,	"tts" 	},
+	{ "text/vsf", 	        		NULL,	"vsf" 	},
+	{ "text/zeg", 	        		NULL,	"zeg" 	},
+	{ "text/mpl", 	        		NULL,	"mpl" 	},
+	{ "text/bup", 	        		NULL,	"bup" 	},
+	{ "text/ifo", 	        		NULL,	"ifo" 	},
 	{ "text/",				NULL,	NULL	},
 
 	{ NULL,					NULL,	NULL	}
