diff -Naur djmount-0.71.orig/djmount/media_file.c djmount-0.71/djmount/media_file.c
--- djmount-0.71.orig/djmount/media_file.c	2007-12-08 17:28:09.000000000 +0100
+++ djmount-0.71/djmount/media_file.c	2007-12-08 17:29:29.000000000 +0100
@@ -87,6 +87,7 @@
 	 */
 	{ "video/vnd.rn-realvideo",		"ram",	"ram"	},
 	{ "video/x-msvideo",			"m3u",	"avi"	},
+	{ "video/avi",   			"m3u",	"avi"	},
 	{ "video/x-motion-jpeg",		"m3u",	"mjpg"	},
 	{ "video/quicktime",			"m3u",	"mov"	},
 	{ "video/x-matroska",			"m3u",	"mkv"	},
