diff -Naur djmount-0.71.orig/configure.ac djmount-0.71/configure.ac
--- djmount-0.71.orig/configure.ac	2007-12-15 18:30:05.000000000 +0100
+++ djmount-0.71/configure.ac	2007-12-28 12:44:15.000000000 +0100
@@ -213,6 +213,12 @@
 CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 CC="$PTHREAD_CC"
 
+#
+# Force _GNU_SOURCE for libupnp releases >= 1.6.3
+# -----------------------------------------------
+#
+CFLAGS="$CFLAGS -D_GNU_SOURCE"
+
 
 #
 # FUSE 
