diff -Nur djmount-0.31.orig/djmount/Makefile djmount-0.31/djmount/Makefile
--- djmount-0.31.orig/djmount/Makefile	Sun Sep  4 18:22:14 2005
+++ djmount-0.31/djmount/Makefile	Wed Nov  2 18:59:36 2005
@@ -49,20 +49,7 @@
 endif
 
 
-ifneq ($(wildcard /usr/include/readline/readline.h),)
-CPPFLAGS += -DHAVE_READLINE=1
-test_upnp: LIBS +=  -lreadline -lhistory -ltermcap
-endif
-
-
-ifneq ($(wildcard /usr/include/iconv.h),)
-HAVE_ICONV = 1
-endif
-ifeq ($(HAVE_ICONV),1)
-CPPFLAGS += -DHAVE_ICONV=1
-else
 OBJS += charset_internal.o
-endif
 
 
 CFLAGS += -Wall $(OPT) $(CPPFLAGS) $(FUSEFLAGS) -pthread
