diff -Nur djmount-0.31.orig/djmount/Makefile djmount-0.31/djmount/Makefile
--- djmount-0.31.orig/djmount/Makefile	Wed Nov  2 19:23:14 2005
+++ djmount-0.31/djmount/Makefile	Wed Nov  2 19:23:25 2005
@@ -33,8 +33,8 @@
 
 LIBUPNP=../libupnp/upnp
 
-CPPFLAGS += -I../talloc -I../sglib -I.
-LIBS += -L../talloc -ltalloc
+CPPFLAGS += -I../sglib -I.
+LIBS += -ltalloc
 ifeq ($(DEBUG),1)
 OPT = -g -O2 
 CPPFLAGS += -DDEBUG=1
