--- vdr-pin-0.1.14/pin.h.org	2014-03-21 16:03:12.134178382 +0100
+++ vdr-pin-0.1.14/pin.h	2014-03-21 15:57:44.086172751 +0100
@@ -26,8 +26,8 @@
 //***************************************************************************
 
 static const char *VERSION        = "0.1.14";
-static const char *DESCRIPTION    = "Childlock plugin";
-static const char *MAINMENUENTRY  = tr("Childlock");
+static const char *DESCRIPTION    = "Filtro Famiglia";
+static const char *MAINMENUENTRY  = tr("Filtro Famiglia");
 
 #define PROTECTION_FILE "protection.fsk"
 // #define __DEBUG__
