diff -Naur pin-0.1.14.orig/pin.c pin-0.1.14/pin.c
--- pin-0.1.14.orig/pin.c	2013-05-22 11:27:30.000000000 +0200
+++ pin-0.1.14/pin.c	2013-10-29 00:56:40.229242000 +0100
@@ -574,7 +574,11 @@
 
    if (isDirectory)
       res = asprintf(&path, "%s%s%s/%s/%s", 
+#if APIVERSNUM > 20101
+                     cVideoDirectory::Name(),
+#else
                      VideoDirectory, 
+#endif
                      base ? "/" : "",
                      base ? base : "",
                      name ? name : "",
