Index: udisks/data/org.freedesktop.UDisks.service.in
===================================================================
--- udisks.orig/data/org.freedesktop.UDisks.service.in	2011-06-29 18:16:14.000000000 +0200
+++ udisks/data/org.freedesktop.UDisks.service.in	2012-02-11 16:39:57.804348611 +0100
@@ -2,4 +2,4 @@
 Name=org.freedesktop.UDisks
 Exec=@libexecdir@/udisks-daemon
 User=root
-
+SystemdService=udisks.service
