--- vdr-dvbapi-1.02+2df8c881/Makefile	2012-06-28 23:57:48.000000000 +0200
+++ vdr-dvbapi-1.02+2df8c881.patch/Makefile	2012-08-14 19:32:48.506674050 +0200
@@ -18,7 +18,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -march=athlon64 -fPIC -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -fPIC -O2 -Wall -Woverloaded-virtual
 
 ### The directory environment:
 
@@ -54,9 +54,8 @@
 OBJS = CAPMT.o DeCSA.o DeCsaTSBuffer.o dll.o DVBAPI.o DVBAPISetup.o SCDeviceProbe.o simplelist.o device.o deviceplugin.o UDPSocket.o SCCIAdapter.o Frame.o SCCAMSlot.o
 
 # FFdeCSA
-CPUOPT     ?= athlon64
-PARALLEL   ?= PARALLEL_128_SSE
-CSAFLAGS   ?= -fPIC -O3 -fexpensive-optimizations -funroll-loops -mmmx -msse -msse2 -msse3
+PARALLEL   ?= PARALLEL_32_INT
+CSAFLAGS   ?= -fPIC -O3 -fexpensive-optimizations -funroll-loops
 FFDECSADIR  = FFdecsa
 FFDECSA     = $(FFDECSADIR)/FFdecsa.o
 FFDECSATEST = $(FFDECSADIR)/FFdecsa_test.done
@@ -108,7 +107,7 @@
 	@cp -p $< $@
 
 $(FFDECSA): $(FFDECSADIR)/*.c $(FFDECSADIR)/*.h
-	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CSAFLAGS) -march=$(CPUOPT)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
+	@$(MAKE) COMPILER="$(CXX)" FLAGS="$(CSAFLAGS)" PARALLEL_MODE=$(PARALLEL) -C $(FFDECSADIR) all
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
