--- vdr-reelsoftcam-1.0/Makefile.org	2013-05-06 09:17:37.000000000 +0200
+++ vdr-reelsoftcam-1.0/Makefile	2013-05-06 10:34:35.289658451 +0200
@@ -140,12 +140,11 @@
 
 # FFdeCSA
 ### VDR-SC Makefile FFdeCSA OPTS ###
-CPUOPT     ?= native
-PARALLEL   ?= PARALLEL_128_2MMX
-CSAFLAGS   ?= -O2 -fexpensive-optimizations -fomit-frame-pointer -funroll-loops
+PARALLEL   ?= PARALLEL_32_INT
+CSAFLAGS   ?= -Wall -fPIC -g -O3 -fomit-frame-pointer -fexpensive-optimizations -funroll-loops
 
 ### GENERIC FFdeCSA make OPTS ###
-FLAGS="-O2 -march=native -fexpensive-optimizations -fomit-frame-pointer -funroll-loops" PARALLEL_MODE=PARALLEL_128_2MMX
+#FLAGS="-O2 -march=native -fexpensive-optimizations -fomit-frame-pointer -funroll-loops" PARALLEL_MODE=PARALLEL_128_2MMX
 FFDECSADIR  = FFdecsa
 FFDECSA     = $(FFDECSADIR)/FFdecsa.o
 FFDECSATEST = $(FFDECSADIR)/FFdecsa_test.done
