--- mysql-python-1.2.5/setup_posix.py	2015-03-25 21:23:32.353684738 +0100
+++ mysql-python-1.2.5/setup_posix.py-new	2015-03-25 21:24:36.169682641 +0100
@@ -69,6 +69,8 @@
         include_dirs = [ dequote(i[2:])
                          for i in mysql_config('cflags')
                          if i.startswith(compiler_flag('I')) ]
+    include_dirs.append("LIB_P/include")
+    include_dirs.append("LIB_P/include/mysql")
 
     if static:
         extra_objects.append(os.path.join(library_dirs[0],'lib%s.a' % client))
