--- vdr-xineliboutput-1.0.9/Makefile.org	2012-10-24 11:33:19.816513503 +0200
+++ vdr-xineliboutput-1.0.9/Makefile	2012-10-24 11:34:34.948511264 +0200
@@ -26,8 +26,8 @@
 # Override configuration here or in ../../../Make.config
 #
 
-#NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal.mpv
-#STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
+NOSIGNAL_IMAGE_FILE=/usr/share/vdr/xineliboutput/nosignal_sifbox_720x576.mpg
+STARTUP_IMAGE_FILE=/usr/share/vdr/xineliboutput/logodisplay.mpv
 XINELIBOUTPUT_CONFIGURE_OPTS =
 
 
