From 9da4aceadbf7d5b3ff8c90dcc3107cb8bede4f99 Mon Sep 17 00:00:00 2001
From: rofafor
Date: Fri, 25 Oct 2013 17:22:38 +0000
Subject: Added support for vdr-2.1.2.

---
diff --git a/config.c b/config.c
index f7384e8..5f3e132 100644
--- a/config.c
+++ b/config.c
@@ -4,7 +4,7 @@
  * See the main source file 'xineliboutput.c' for copyright information and
  * how to reach the author.
  *
- * $Id: config.c,v 1.115 2012-03-19 11:38:43 phintuka Exp $
+ * $Id: config.c,v 1.116 2013-10-25 17:22:37 rofafor Exp $
  *
  */
 
@@ -703,9 +703,15 @@ config_t::config_t() {
   ff_h264_skip_loop_filter     = FF_H264_SKIP_LOOPFILTER_auto;
 
   strn0cpy(media_root_dir,    "/",            sizeof(media_root_dir));
+#if defined(APIVERSNUM) && (APIVERSNUM < 20102)
   strn0cpy(browse_files_dir,  VideoDirectory, sizeof(browse_files_dir));
   strn0cpy(browse_music_dir,  VideoDirectory, sizeof(browse_music_dir));
   strn0cpy(browse_images_dir, VideoDirectory, sizeof(browse_images_dir));
+#else
+  strn0cpy(browse_files_dir,  cVideoDirectory::Name(), sizeof(browse_files_dir));
+  strn0cpy(browse_music_dir,  cVideoDirectory::Name(), sizeof(browse_music_dir));
+  strn0cpy(browse_images_dir, cVideoDirectory::Name(), sizeof(browse_images_dir));
+#endif
   show_hidden_files = 0;
   cache_implicit_playlists = 1;
   enable_id3_scanner = 1;
diff --git a/menu.c b/menu.c
index c0a152f..8ee89ed 100644
--- a/menu.c
+++ b/menu.c
@@ -4,7 +4,7 @@
  * See the main source file 'xineliboutput.c' for copyright information and
  * how to reach the author.
  *
- * $Id: menu.c,v 1.96 2012-03-19 11:29:14 phintuka Exp $
+ * $Id: menu.c,v 1.97 2013-10-25 17:22:38 rofafor Exp $
  *
  */
 
@@ -220,7 +220,11 @@ void cMenuBrowseFiles::Set(void)
   }
 
   if (m_CurrentDir[0] != '/') {
+#if defined(APIVERSNUM) && (APIVERSNUM < 20102)
     m_CurrentDir = VideoDirectory;
+#else
+    m_CurrentDir = cVideoDirectory::Name();
+#endif
   }
 
   // find deepest accessible directory from path
diff --git a/tools/udp_pes_scheduler.c b/tools/udp_pes_scheduler.c
index 615b43b..505637b 100644
--- a/tools/udp_pes_scheduler.c
+++ b/tools/udp_pes_scheduler.c
@@ -4,7 +4,7 @@
  * See the main source file 'xineliboutput.c' for copyright information and
  * how to reach the author.
  *
- * $Id: udp_pes_scheduler.c,v 1.56 2011-02-28 13:15:08 phintuka Exp $
+ * $Id: udp_pes_scheduler.c,v 1.57 2013-10-25 17:22:38 rofafor Exp $
  *
  */
 
@@ -635,7 +635,12 @@ void cUdpScheduler::Send_SAP(bool Announce)
 #if 1
   /* store copy of SDP data */
   if(m_fd_sap < 0) {
-    cString fname = AddDirectory(VideoDirectory,
+    cString fname = AddDirectory(
+#if defined(APIVERSNUM) && (APIVERSNUM < 20102)
+				 VideoDirectory,
+#else
+				 cVideoDirectory::Name(),
+#endif
 				 cString::sprintf("xineliboutput@%s.sdp",
 						  ip));
     FILE *fp = fopen(fname, "w");
--
cgit v0.9.0.2-40-g7c79
