--- a/src/webui/webui.c	2018-04-28 15:09:59.603646266 +0200
+++ b/src/webui/webui.c	2018-04-28 15:09:59.647645957 +0200
@@ -112,7 +112,7 @@
 {
   if (hc->hc_access != NULL &&
       hc->hc_access->aa_username != NULL &&
-      hc->hc_access->aa_username != '\0') {
+      *hc->hc_access->aa_username != '\0') {
     http_redirect(hc, "/", &hc->hc_req_args, 0);
     return 0;
   } else {
@@ -125,7 +125,7 @@
 {
   if (hc->hc_access == NULL ||
       hc->hc_access->aa_username == NULL ||
-      hc->hc_access->aa_username == '\0') {
+      *hc->hc_access->aa_username == '\0') {
 redirect:
     http_redirect(hc, "/", &hc->hc_req_args, 0);
     return 0;
--- a/src/dvr/dvr_rec.c	2018-04-28 15:13:16.722247524 +0200
+++ b/src/dvr/dvr_rec.c	2018-04-28 15:13:16.746247355 +0200
@@ -206,7 +206,7 @@
     return -1;
 
   cfg = de->de_config;
-  if (cfg->dvr_storage == NULL || cfg->dvr_storage == '\0')
+  if (cfg->dvr_storage == NULL || *cfg->dvr_storage == '\0')
     return -1;
   strncpy(path, cfg->dvr_storage, sizeof(path));
   path[sizeof(path)-1] = '\0';
--- a/src/input/mpegts/scanfile.c	2018-04-28 15:18:55.659842541 +0200
+++ b/src/input/mpegts/scanfile.c	2018-04-28 15:18:55.715842144 +0200
@@ -374,7 +374,7 @@
 
   /* validity check for [text] */
   s = str_trim(line);
-  if (s == '\0' || s[strlen(s) - 1] != ']')
+  if (s == NULL || s[strlen(s) - 1] != ']')
     return 1;
 
   l = htsmsg_create_map();
