--- a/src/input/mpegts/tvhdhomerun/tvhdhomerun.c	2015-02-18 19:55:17.000000000 +0100
+++ b/src/input/mpegts/tvhdhomerun/tvhdhomerun.c	2016-03-17 21:53:35.000000000 +0100
@@ -343,7 +343,7 @@
   if (!tvheadend_running)
     return;
 
-  int numDevices = hdhomerun_discover_find_devices_custom(0,
+  int numDevices = hdhomerun_discover_find_devices_custom_v2(0,
                                                           HDHOMERUN_DEVICE_TYPE_TUNER,
                                                           HDHOMERUN_DEVICE_ID_WILDCARD,
                                                           result_list,
