Index: btfm/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ btfm/Makefile	2010-09-28 19:10:55.000000000 +0200
@@ -0,0 +1,27 @@
+# This top-level makefile was created to ease Debian packaging.
+# It must provide targets: 'clean', 'build', 'install' and 'check'.
+# The 'install' target must install files under $(DESTDIR).
+# Making 'build' the default target is nice to have.
+
+.PHONY: clean build install check
+.DEFAULT: build
+
+DIR_BT := uim
+DIR_FM := kfmapp
+BINDIR := $(DESTDIR)/usr/bin
+CROSS_COMPILE ?=
+ARCH ?= arm
+
+clean:
+	$(MAKE) -C $(DIR_BT) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) clean
+	$(MAKE) -C $(DIR_FM) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) clean
+
+build:
+	$(MAKE) -C $(DIR_BT) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH)
+	$(MAKE) -C $(DIR_FM) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH)
+
+install:
+	@install -d "$(BINDIR)"; \
+	install -m 755 "uim/uim" "$(BINDIR)"; \
+	install -m 755 "kfmapp/kfmapp" "$(BINDIR)"; \
+
