Index: btfm/kfmapp/makefile
===================================================================
--- btfm.orig/kfmapp/makefile	2010-08-31 12:09:05.000000000 +0200
+++ btfm/kfmapp/makefile	2010-08-31 12:09:36.000000000 +0200
@@ -10,7 +10,7 @@
 all: kfmapp
 
 kfmapp: kfmapp.c
-	$(CC) $^ -o $@ $(LIB_DIR) -lpthread -lm
+	$(CC) $(CFLAGS) $^ -o $@ $(LIB_DIR) -lpthread -lm
 
 .PHONY: clean
 clean:
Index: btfm/uim/makefile
===================================================================
--- btfm.orig/uim/makefile	2010-08-31 12:09:05.000000000 +0200
+++ btfm/uim/makefile	2010-08-31 12:09:27.000000000 +0200
@@ -8,7 +8,7 @@
 all: uim
 
 uim: uim.c
-	$(CC) $^ -o $@
+	$(CC) $(CFLAGS) $^ -o $@
 
 .PHONY: clean
 clean:
