--- a/libcedarv/linux-armhf/Makefile	2013-06-16 12:59:42.459407526 +0200
+++ b/libcedarv/linux-armhf/Makefile	2013-06-16 12:59:42.467407529 +0200
@@ -27,7 +27,7 @@
 all: $(LIB)
 
 $(LIB): $(OBJS)
-	$(AR) cq $(LIB) $(OBJS)
+	$(AR) cr $(LIB) $(OBJS)
 
 clean:
 	rm -f $(LIB) $(OBJS)
--- a/libcedarv/linux-armel/Makefile	2013-06-16 13:00:44.643408496 +0200
+++ b/libcedarv/linux-armel/Makefile	2013-06-16 13:00:02.603407842 +0200
@@ -26,7 +26,7 @@
 all: $(LIB)
 
 $(LIB): $(OBJS)
-	$(AR) cq $(LIB) $(OBJS)
+	$(AR) cr $(LIB) $(OBJS)
 
 clean:
 	rm -f $(LIB) $(OBJS)
--- a/libcedarv/android/Makefile	2013-06-16 13:16:21.579423081 +0200
+++ a/libcedarv/android/Makefile	2013-06-16 13:16:21.591423080 +0200
@@ -26,7 +26,7 @@
 all: $(LIB)
 
 $(LIB): $(OBJS)
-	$(AR) cq $(LIB) $(OBJS)
+	$(AR) cr $(LIB) $(OBJS)
 
 clean:
 	rm -f $(LIB) $(OBJS)
