From d08d39a7acc7532940b46ed016f3a81c3ce68457 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Tue, 17 May 2011 22:38:14 +0200
Subject: [PATCH 07/11] baseparse: set correct buffer size

---
 libs/gst/base/gstbaseparse.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libs/gst/base/gstbaseparse.c b/libs/gst/base/gstbaseparse.c
index e80c8f4..357788d 100644
--- a/libs/gst/base/gstbaseparse.c
+++ b/libs/gst/base/gstbaseparse.c
@@ -2224,7 +2224,7 @@ gst_base_parse_chain (GstPad * pad, GstBuffer * buffer)
       /* always pass all available data */
       data = gst_adapter_peek (parse->priv->adapter, av);
       GST_BUFFER_DATA (tmpbuf) = (guint8 *) data;
-      GST_BUFFER_SIZE (tmpbuf) = min_size;
+      GST_BUFFER_SIZE (tmpbuf) = av;
       GST_BUFFER_OFFSET (tmpbuf) = parse->priv->offset;
       GST_BUFFER_FLAG_SET (tmpbuf, GST_MINI_OBJECT_FLAG_READONLY);
 
-- 
1.7.1

