diff --git a/gst/gstelementfactory.c b/gst/gstelementfactory.c
index 0c3d8aa..1e3c304 100644
--- a/gst/gstelementfactory.c
+++ b/gst/gstelementfactory.c
@@ -395,7 +395,7 @@ gst_element_factory_create (GstElementFactory * factory, con
    */
   oclass = GST_ELEMENT_GET_CLASS (element);
   if (!g_atomic_pointer_compare_and_exchange (
-          (gpointer) & oclass->elementfactory, NULL, factory))
+        (volatile gpointer *)&oclass->elementfactory, NULL, factory))
     gst_object_unref (factory);

   GST_DEBUG ("created element \"%s\"", GST_PLUGIN_FEATURE_NAME (factory));
