From 2f7820d91d7b0c1818bd2caab51d4df0f24a6951 Mon Sep 17 00:00:00 2001
From: Alexander Huemer <a.huemer@bct-electronic.com>
Date: Mon, 23 Jan 2012 10:42:15 +0100
Subject: [PATCH] don't compile build utility for target

---
 src/Makefile.am |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index ec21765..aff0468 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -78,8 +78,8 @@ noinst_HEADERS = basop32.h \
                  tables.h \
                  utilities.h
 
-noinst_PROGRAMS = make_dct4_tables \
-                  make_tables
+#noinst_PROGRAMS = make_dct4_tables \
+#                  make_tables
 
 dct4.$(OBJEXT): dct4.h
 
-- 
1.7.4.1

