From 59e6beed0a9677a92690e94804beb3707ac18621 Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Tue, 20 Mar 2012 19:04:45 +0100
Subject: [PATCH 06/17] [dove] workaround for eglCreateContext

---
 xbmc/windowing/X11/WinSystemX11GLES.cpp |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/xbmc/windowing/X11/WinSystemX11GLES.cpp b/xbmc/windowing/X11/WinSystemX11GLES.cpp
index a94906d..36a9f97 100644
--- a/xbmc/windowing/X11/WinSystemX11GLES.cpp
+++ b/xbmc/windowing/X11/WinSystemX11GLES.cpp
@@ -34,8 +34,12 @@
 using namespace std;
 
 // Comment out one of the following defines to select the colourspace to use
-//#define RGBA8888
+#ifdef HAS_MARVELL_DOVE
+/* For dove we use 32 bit per pixel on graphics overlay */
+#define RGBA8888
+#else
 #define RGB565
+#endif
 
 #if defined(RGBA8888)
 #define RSIZE	8
@@ -385,11 +389,13 @@ bool CWinSystemX11GLES::RefreshEGLContext()
   if (m_eglContext)
     eglDestroyContext(m_eglDisplay, m_eglContext);
 
+#ifndef HAS_MARVELL_DOVE /* Dove GL engine doesn't like the following. Probably EGL_NO_CONTEXT flag */
   if ((m_eglContext = eglCreateContext(m_eglDisplay, eglConfig, EGL_NO_CONTEXT, contextAttributes)) == EGL_NO_CONTEXT)
   {
     CLog::Log(LOGERROR, "EGL Error: Could not create context");
     return false;
   }
+#endif
 
   if ((m_eglContext = eglCreateContext(m_eglDisplay, eglConfig, m_eglContext, contextAttributes)) == EGL_NO_CONTEXT)
   {
-- 
1.7.4.1

