diff -aur --unidirectional-new-file a/xbmc/utils/MathUtils.h b/xbmc/utils/MathUtils.h
--- a/xbmc/utils/MathUtils.h	2013-01-28 03:07:32.000000000 +0100
+++ b/xbmc/utils/MathUtils.h	2013-02-24 10:57:20.000000000 +0100
@@ -34,6 +34,7 @@
 
 #if defined(__ppc__) || \
     defined(__powerpc__) || \
+    defined(TARGET_MARVELL_DOVE) || \
    (defined(__APPLE__) && defined(__arm__) && defined(__llvm__)) || \
    (defined(__ANDROID__) && defined(__arm__)) || \
     defined(TARGET_RASPBERRY_PI)
@@ -42,6 +43,7 @@
 
 #if defined(__ppc__) || \
     defined(__powerpc__) || \
+    defined(TARGET_MARVELL_DOVE) || \
    (defined(__APPLE__) && defined(__llvm__)) || \
    (defined(__ANDROID__) && defined(__arm__)) || \
     defined(TARGET_RASPBERRY_PI)
