From 35bb457fe891955d7cf1ee4a91eb3ba30065d7c5 Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Tue, 20 Mar 2012 19:06:27 +0100
Subject: [PATCH 08/17] [dove] missing typdef leads in XKBlib.h include errors

---
 xbmc/windowing/WinEventsSDL.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/xbmc/windowing/WinEventsSDL.cpp b/xbmc/windowing/WinEventsSDL.cpp
index f8b9806..882e097 100644
--- a/xbmc/windowing/WinEventsSDL.cpp
+++ b/xbmc/windowing/WinEventsSDL.cpp
@@ -36,6 +36,7 @@
 #endif
 
 #if defined(_LINUX) && !defined(__APPLE__)
+typedef int Status; /* somehow Status typedef is missing, leads error in XKBlib.h */
 #include <X11/Xlib.h>
 #include <X11/XKBlib.h>
 #include "input/XBMC_keysym.h"
-- 
1.7.4.1

