From 53928bd32a07ebbb75574773272dc1e6c5be1f75 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 17 May 2014 12:02:26 +0200
Subject: [PATCH 09/26] Dove: add settings constants

---
 xbmc/guilib/Resolution.h |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/xbmc/guilib/Resolution.h b/xbmc/guilib/Resolution.h
index 08be14e..007f8d2 100644
--- a/xbmc/guilib/Resolution.h
+++ b/xbmc/guilib/Resolution.h
@@ -63,6 +63,20 @@ enum VSYNC {
   VSYNC_DRIVER       =  3
 };
 
+#ifdef TARGET_MARVELL_DOVE
+enum VMETA_CLK {
+  VMETA_CLK_500      = 500,
+  VMETA_CLK_667      = 667
+};
+
+enum GRAPHICS_SCALING {
+  GR_SCALE_100       = 100,
+  GR_SCALE_150       = 150,
+  GR_SCALE_200       = 200,
+  GR_SCALE_250       = 250
+};
+#endif
+
 struct OVERSCAN
 {
   int left;
-- 
1.7.9.5

