--- a/configure.in	2013-06-12 19:17:02.000000000 +0200
+++ b/configure.in	2013-06-12 19:35:04.499894778 +0200
@@ -675,6 +675,32 @@
      CXXFLAGS="$CXXFLAGS"
      ffmpeg_target_os=linux
      ;;
+
+  marvell-dove)
+     target_platform=target_marvell_dove
+     use_neon=no
+     use_texturepacker=yes
+     use_texturepacker_native=yes
+     use_arch="arm"
+     use_cpu=cortex-a9
+     use_armv5te="yes"
+     use_armv6t2="no"
+     use_armv6="yes"
+     use_armvfp="yes"
+     use_hardcoded_tables="yes"
+     use_alsa="yes"
+     USE_TEXTUREPACKER_NATIVE_ROOT="$TEXTUREPACKER_NATIVE_ROOT"
+     ARCH="arm"
+     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_MARVELL_DOVE")
+     AC_DEFINE(HAS_MARVELL_DOVE, [1], [Define if supporting Marvell Armada 510 (Dove)])
+     CFLAGS="$CFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb -marm -O3 -ftree-vectorize"
+     CXXFLAGS="$CXXFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -mthumb -marm -O3 -ftree-vectorize"
+     FFMPEG_EXTRACFLAGS="$FFMPEG_EXTRACFLAGS -mtune=cortex-a9 -mfpu=vfpv3-d16 -O3 -ftree-vectorize -mthumb -marm"
+     USE_MARVELL_DOVE=1
+     ;;
+
+  *)
+     use_alsa="yes"
 esac
 
 if test "$build_shared_lib" = "yes"; then
@@ -756,6 +782,10 @@
   CFLAGS="$CFLAGS -mno-apcs-stack-check"
   CXXFLAGS="$CXXFLAGS -mno-apcs-stack-check"
   FFMPEG_EXTRACFLAGS=""
+elif test "$target_platform" = "target_marvell_dove"; then
+  ARCH="arm"
+  use_arch="arm"
+else
   if test "$use_tegra" = "yes"; then
     # Compile for ARMv7a architecture, need to test gcc for vfpv3-d16 support 
     SAVE_CFLAGS="$CFLAGS"
@@ -2415,6 +2445,7 @@
 AC_SUBST(USE_AIRPLAY)
 AC_SUBST(USE_VDA)
 AC_SUBST(USE_OPENMAX)
+AC_SUBST(USE_MARVELL_DOVE)
 AC_SUBST(USE_PULSE)
 AC_SUBST(USE_XRANDR)
 AC_SUBST(USE_ALSA)
@@ -2569,6 +2600,10 @@
       `if test "$use_arch" != "no"; then echo --arch=$use_arch; fi`\
       `if test "$use_cpu" != "no"; then echo --cpu=$use_cpu; fi`\
       `if test "$use_neon" = "yes"; then echo --enable-neon; else echo --disable-neon; fi`\
+      `if test "$use_armv5te" = "yes"; then echo --enable-armv5te; else echo --disable-armv5te; fi`\
+      `if test "$use_armv6t2" = "yes"; then echo --enable-armv6t2; else echo --disable-armv6t2; fi`\
+      `if test "$use_armv6" = "yes"; then echo --enable-armv6; else echo --disable-armv6; fi`\
+      `if test "$use_armvfp" = "yes"; then echo --enable-armvfp; else echo --disable-armvfp; fi`\
       --target-os=$ffmpeg_target_os \
       --disable-muxers \
       --enable-muxer=spdif \
