From a69f24f45e21c5d6982394ec7420e32de7784ab7 Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Tue, 20 Mar 2012 20:06:10 +0100
Subject: [PATCH 11/17] [dove] changed vblank waiting is only needed for DOVEFB_IOCTL_NEXT_FRAME_PRESENT.

---
 xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp b/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp
index 82502d5..9091651 100644
--- a/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp
+++ b/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp
@@ -331,8 +331,12 @@ void CDoveOverlayRenderer::FlipPage(int source)
 
   }
 
+  /*
+   * Is only needed for DOVEFB_IOCTL_NEXT_FRAME_PRESENT
+   *
   if (ioctl(m_overlayfd, DOVEFB_IOCTL_WAIT_VSYNC, 0) != 0)
     CLog::Log(LOGERROR, "%s::%s - Error waiting for vsync\n", CLASSNAME, __func__);
+  */
 
   if( source >= 0 && source < NUM_BUFFERS )
     m_currentBuffer = source;
-- 
1.7.4.1

