--- ./configure.in.orig	2012-05-21 15:40:42.830481291 -0600
+++ ./configure.in	2012-05-22 16:24:02.230644140 -0600
@@ -572,7 +572,8 @@ case $use_platform in
      use_armv6="yes"
      use_armvfp="yes"
      use_hardcoded_tables="yes"
-     use_alsa="yes"
+     use_alsa="no"
+     use_pulse="yes"
      USE_TEXTUREPACKER_NATIVE_ROOT="$TEXTUREPACKER_NATIVE_ROOT"
      ARCH="arm"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_MARVELL_DOVE")
