From ed4d0717e799f43bc75892eb4a56bf940ff700e5 Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Thu, 22 Mar 2012 21:58:53 +0100
Subject: [PATCH 13/17] [dove] changed use bigger alsa buffer and ask the buffersize after we commited our setup to the driver

---
 xbmc/cores/AudioRenderers/ALSADirectSound.cpp |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/cores/AudioRenderers/ALSADirectSound.cpp b/xbmc/cores/AudioRenderers/ALSADirectSound.cpp
index 2ebacb3..3e17d13 100644
--- a/xbmc/cores/AudioRenderers/ALSADirectSound.cpp
+++ b/xbmc/cores/AudioRenderers/ALSADirectSound.cpp
@@ -112,7 +112,7 @@ bool CALSADirectSound::Initialize(IAudioCallback* pCallback, const CStdString& d
   if (!m_bPassthrough)
      m_amp.SetVolume(m_nCurrentVolume);
 
-  m_dwFrameCount = 512;
+  m_dwFrameCount = 1024;
   m_dwNumPackets = 16;
   m_uiBufferSize = 0;
 
@@ -253,13 +253,13 @@ bool CALSADirectSound::Initialize(IAudioCallback* pCallback, const CStdString& d
   nErr = snd_pcm_hw_params_set_periods_near(m_pPlayHandle, hw_params, &m_dwNumPackets, NULL);
   CHECK_ALSA_RETURN(LOGERROR,"hw_params_set_periods",nErr);
 
-  nErr = snd_pcm_hw_params_get_buffer_size(hw_params, &m_uiBufferSize);
-  CHECK_ALSA_RETURN(LOGERROR,"hw_params_get_buffer_size",nErr);
-
   /* Assign them to the playback handle and free the parameters structure */
   nErr = snd_pcm_hw_params(m_pPlayHandle, hw_params);
   CHECK_ALSA_RETURN(LOGERROR,"snd_pcm_hw_params",nErr);
 
+  nErr = snd_pcm_hw_params_get_buffer_size(hw_params, &m_uiBufferSize);
+  CHECK_ALSA_RETURN(LOGERROR,"hw_params_get_buffer_size",nErr);
+
   nErr = snd_pcm_sw_params_current(m_pPlayHandle, sw_params);
   CHECK_ALSA_RETURN(LOGERROR,"sw_params_current",nErr);
 
-- 
1.7.4.1

