From c1e618d4104749b97738f88b5f0773db33dfa9df Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Fri, 23 Mar 2012 22:23:39 +0100
Subject: [PATCH 14/17] [dove] workaround for marvell dove. EGL_NO_CONTEXT causes problems.

---
 xbmc/windowing/X11/WinSystemX11GLES.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xbmc/windowing/X11/WinSystemX11GLES.cpp b/xbmc/windowing/X11/WinSystemX11GLES.cpp
index 36a9f97..ea7735f 100644
--- a/xbmc/windowing/X11/WinSystemX11GLES.cpp
+++ b/xbmc/windowing/X11/WinSystemX11GLES.cpp
@@ -369,7 +369,9 @@ bool CWinSystemX11GLES::RefreshEGLContext()
   if ((m_eglWindow == info.info.x11.window) && m_eglSurface && m_eglContext)
   {
     CLog::Log(LOGWARNING, "EGL: Same window as before, refreshing context");
+#ifndef HAS_MARVELL_DOVE /* Dove GL engine doesn't like the following. Probably EGL_NO_CONTEXT flag */
     eglMakeCurrent(m_eglDisplay, EGL_NO_SURFACE, EGL_NO_SURFACE, EGL_NO_CONTEXT);
+#endif
     eglMakeCurrent(m_eglDisplay, m_eglSurface,   m_eglSurface,   m_eglContext);
     return true;
   }
-- 
1.7.4.1

