From 0355175cc33314c792fcce935b7de444aa17afaa Mon Sep 17 00:00:00 2001
From: huceke <gimli@i7.(none)>
Date: Tue, 3 Apr 2012 22:55:18 +0200
Subject: [PATCH 17/17] [dove] fixed exit hang. this is a workaround for a bug in the Vivante gpu driver.

---
 xbmc/xbmc.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/xbmc/xbmc.cpp b/xbmc/xbmc.cpp
index b6d3bac..c91b11d 100644
--- a/xbmc/xbmc.cpp
+++ b/xbmc/xbmc.cpp
@@ -98,5 +98,10 @@ int main(int argc, char* argv[])
     status = -1;
   }
 
+  // workaround for a bug in the Vivante gpu driver
+#if defined(TARGET_MARVELL_DOVE)
+  _exit(status);
+#else
   return status;
+#endif
 }
-- 
1.7.4.1

