From 6b036938c849da120410aec567e24938df3eba8e Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 17 May 2014 14:59:27 +0200
Subject: [PATCH 18/26] Dove: support graphics scaling #2

---
 xbmc/cores/VideoRenderers/OverlayRenderer.cpp |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/xbmc/cores/VideoRenderers/OverlayRenderer.cpp b/xbmc/cores/VideoRenderers/OverlayRenderer.cpp
index e6a2d81..8e73056 100644
--- a/xbmc/cores/VideoRenderers/OverlayRenderer.cpp
+++ b/xbmc/cores/VideoRenderers/OverlayRenderer.cpp
@@ -277,6 +277,15 @@ void CRenderer::Render(COverlay* o)
 
   state.x += GetStereoscopicDepth();
 
+#ifdef HAS_MARVELL_DOVE
+  GRAPHICS_SCALING scale = g_graphicsContext.getGraphicsScale();
+
+  state.x = (state.x * 100) / scale;
+  state.y = (state.y * 100) / scale;
+  state.width = (state.width * 100) /scale;
+  state.height = (state.height *100) /scale;
+#endif
+
   o->Render(state);
 }
 
-- 
1.7.9.5

