diff -Naur a/xbmc/cores/dvdplayer//DVDFileInfo.cpp b/xbmc/cores/dvdplayer//DVDFileInfo.cpp
--- a/xbmc/cores/dvdplayer//DVDFileInfo.cpp	2012-07-11 21:59:17.000000000 +0200
+++ b/xbmc/cores/dvdplayer//DVDFileInfo.cpp	2012-08-12 13:55:20.035629732 +0200
@@ -143,6 +143,13 @@
     CDVDStreamInfo hint(*pDemuxer->GetStream(nVideoStream), true);
     hint.software = true;
 
+#if defined(TARGET_MARVELL_DOVE)
+	// VMETA Codec doesn't work for thumbs so use FFmpeg
+    {
+      CDVDCodecOptions dvdOptions;
+      pVideoCodec = CDVDFactoryCodec::OpenCodec(new CDVDVideoCodecFFmpeg(), hint, dvdOptions);
+    }
+#else
     if (hint.codec == CODEC_ID_MPEG2VIDEO || hint.codec == CODEC_ID_MPEG1VIDEO)
     {
       // libmpeg2 is not thread safe so use ffmepg for mpeg2/mpeg1 thumb extraction
@@ -153,6 +160,7 @@
     {
       pVideoCodec = CDVDFactoryCodec::CreateVideoCodec( hint );
     }
+#endif
 
     if (pVideoCodec)
     {
