From 70c6284be579c0c5a6a07895f2c03b307c2e46d8 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sun, 18 May 2014 15:40:12 +0200
Subject: [PATCH 22/26] Dove: disable HD audio formats

---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index c608231..acbf104 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -986,6 +986,10 @@ void CAESinkALSA::EnumerateDevice(AEDeviceInfoList &list, const std::string &dev
             {
               if (!AE_IS_RAW(i))
                 continue;
+#ifdef TARGET_MARVELL_DOVE
+              if (AE_IS_RAW_HD(i) && i != AE_FMT_EAC3)
+                continue;
+#endif
               it = find(info.m_dataFormats.begin(), info.m_dataFormats.end(), i);
               if (it == info.m_dataFormats.end())
               {
-- 
1.7.9.5

