diff --git a/xbmc/windowing/X11/WinSystemX11GLES.h b/xbmc/windowing/X11/WinSystemX11GLES.h
index b5bd2cc..559a96c 100644
--- a/xbmc/windowing/X11/WinSystemX11GLES.h
+++ b/xbmc/windowing/X11/WinSystemX11GLES.h
@@ -23,6 +23,7 @@
  *
  */
 #include "windowing/WinSystem.h"
+#include <SDL/SDL.h>
 #include <EGL/egl.h>
 #include <X11/Xlib.h>
 #include "rendering/gles/RenderSystemGLES.h"
diff --git a/xbmc/windowing/X11/WinSystemX11GLES.cpp b/xbmc/windowing/X11/WinSystemX11GLES.cpp
index 06ae8ef..743ff6a 100644
--- a/xbmc/windowing/X11/WinSystemX11GLES.cpp
+++ b/xbmc/windowing/X11/WinSystemX11GLES.cpp
@@ -162,11 +162,12 @@ bool CWinSystemX11GLES::CreateNewWindow(const CStdString& name, bool fullScreen,
   if(!SetFullScreen(fullScreen, res, false))
 	return false;
 
-  CTexture iconTexture;
-  iconTexture.LoadFromFile("special://xbmc/media/icon256x256.png");
+  CBaseTexture* iconTexture = CTexture::LoadFromFile("special://xbmc/media/icon256x256.png");
 
-  SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture.GetPixels(), 256, 256, BPP, 256, 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
+  if (iconTexture)
+    SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture->GetPixels(), iconTexture->GetWidth(), iconTexture->GetHeight(), BPP, iconTexture->GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
   SDL_WM_SetCaption("XBMC Media Center", NULL);
+  delete iconTexture;
 
   m_bWindowCreated = true;
 
