--- ./xbmc/windowing/X11/WinSystemX11GLES.cpp.orig	2012-08-17 12:17:41.641969581 -0600
+++ ./xbmc/windowing/X11/WinSystemX11GLES.cpp	2012-08-17 12:17:41.994966010 -0600
@@ -162,7 +162,7 @@ bool CWinSystemX11GLES::CreateNewWindow(
   CTexture iconTexture;
   iconTexture.LoadFromFile("special://xbmc/media/icon.png");
 
-  SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture.GetPixels(), iconTexture.GetWidth(), iconTexture.GetHeight(), BPP, iconTexture.GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000L), NULL);
+  SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture.GetPixels(), iconTexture.GetWidth(), iconTexture.GetHeight(), BPP, iconTexture.GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
   SDL_WM_SetCaption("XBMC Media Center", NULL);
 
   m_bWindowCreated = true;
--- ./xbmc/windowing/X11/WinSystemX11.cpp.orig	2012-08-17 12:13:36.000000000 -0600
+++ ./xbmc/windowing/X11/WinSystemX11.cpp	2012-08-17 12:19:49.970671789 -0600
@@ -135,7 +135,7 @@ bool CWinSystemX11::CreateNewWindow(cons
   CBaseTexture* iconTexture = CTexture::LoadFromFile("special://xbmc/media/icon.png");
 
   if (iconTexture)
-    SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture->GetPixels(), iconTexture->GetWidth(), iconTexture->GetHeight(), 32, iconTexture->GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000L), NULL);
+    SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture->GetPixels(), iconTexture->GetWidth(), iconTexture->GetHeight(), 32, iconTexture->GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
   SDL_WM_SetCaption("XBMC Media Center", NULL);
   delete iconTexture;
 
