--- xbmc-frodo-rcbfd2846/xbmc/windowing/X11/WinSystemX11GLES.cpp	2012-11-11 14:38:57.488051524 +0100
+++ xbmc-frodo-rcbfd2846/xbmc/windowing/X11/WinSystemX11GLES.cpp-new	2012-11-11 14:45:52.698660500 +0100
@@ -161,7 +161,7 @@
   CTexture iconTexture;
   iconTexture.LoadFromFile("special://xbmc/media/icon.png");
 
-  SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture.GetPixels(), iconTexture.GetWidth(), iconTexture.GetHeight(), BPP, iconTexture.GetPitch(), 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
+  SDL_WM_SetIcon(SDL_CreateRGBSurfaceFrom(iconTexture.GetPixels(), 256, 256, BPP, 256, 0xff0000, 0x00ff00, 0x0000ff, 0xff000000), NULL);
   SDL_WM_SetCaption("XBMC Media Center", NULL);
 
   m_bWindowCreated = true;
