From 5df2820e1f837f3deaf85e3be4c0e1de983a593b Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 25 Nov 2012 22:03:57 +0200
Subject: [PATCH] Make vmeta clock default to 667MHz

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 xbmc/settings/GUISettings.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/settings/GUISettings.cpp b/xbmc/settings/GUISettings.cpp
index 63cee70..04a89cd 100644
--- a/xbmc/settings/GUISettings.cpp
+++ b/xbmc/settings/GUISettings.cpp
@@ -439,7 +439,7 @@ void CGUISettings::Initialize()
   map<int,int> vmeta_clk;
   vmeta_clk.insert(make_pair(37001,VMETA_CLK_500));
   vmeta_clk.insert(make_pair(37002,VMETA_CLK_667));
-  AddInt(vs, "videoscreen.vmeta_clk", 37000, VMETA_CLK_500, vmeta_clk, SPIN_CONTROL_TEXT);
+  AddInt(vs, "videoscreen.vmeta_clk", 37000, VMETA_CLK_667, vmeta_clk, SPIN_CONTROL_TEXT);
 #endif
   map<int,int> vsync;
 #if defined(_LINUX) && !defined(TARGET_DARWIN)
-- 
1.7.9.5

