diff -aur --unidirectional-new-file a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-01-05 22:10:04.169198347 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-01-05 22:10:18.313197756 +0100
@@ -744,8 +744,10 @@
     pDvdVideoPicture->iLineSize[2]    = 0;
 
     pDvdVideoPicture->iFlags          = DVP_FLAG_ALLOCATED;
-    pDvdVideoPicture->iFlags         |= m_drop_state ? DVP_FLAG_DROPPED : 0;
-
+    
+    if (m_drop_state || (unsigned)pPicture->pUsrData1 < 2)
+      pDvdVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+    
 #ifdef ENABLE_PTS
     if (!m_pts_queue.isEmpty())
       m_pts_queue.getHead(pDvdVideoPicture->pts);
