diff -aur --unidirectional-new-file a/xbmc/settings/GUISettings.cpp b/xbmc/settings/GUISettings.cpp
--- a/xbmc/settings/GUISettings.cpp	2013-02-05 20:11:11.590350783 +0100
+++ b/xbmc/settings/GUISettings.cpp	2013-02-05 20:19:19.022330397 +0100
@@ -469,13 +469,17 @@
   CSettingsCategory* ao = AddCategory(SETTINGS_SYSTEM, "audiooutput", 772);
 
   map<int,int> audiomode;
+#if !defined(TARGET_MARVELL_DOVE)
   audiomode.insert(make_pair(338,AUDIO_ANALOG));
+#endif
 #if !defined(TARGET_RASPBERRY_PI)
   audiomode.insert(make_pair(339,AUDIO_IEC958));
 #endif
   audiomode.insert(make_pair(420,AUDIO_HDMI  ));
 #if defined(TARGET_RASPBERRY_PI)
   AddInt(ao, "audiooutput.mode", 337, AUDIO_HDMI, audiomode, SPIN_CONTROL_TEXT);
+#elif defined(TARGET_MARVELL_DOVE)  
+  AddInt(ao, "audiooutput.mode", 337, AUDIO_IEC958, audiomode, SPIN_CONTROL_TEXT);
 #else
   AddInt(ao, "audiooutput.mode", 337, AUDIO_ANALOG, audiomode, SPIN_CONTROL_TEXT);
 #endif
@@ -497,10 +501,10 @@
   AddBool(aocat, "audiooutput.dtspassthrough"   , 254, true);
 
 
-#if !defined(TARGET_DARWIN) && !defined(TARGET_RASPBERRY_PI)
+#if !defined(TARGET_DARWIN) && !defined(TARGET_RASPBERRY_PI) && !defined(TARGET_MARVELL_DOVE)
   AddBool(aocat, "audiooutput.passthroughaac"   , 299, false);
 #endif
-#if !defined(TARGET_DARWIN_IOS) && !defined(TARGET_RASPBERRY_PI)
+#if !defined(TARGET_DARWIN_IOS) && !defined(TARGET_RASPBERRY_PI) && !defined(TARGET_MARVELL_DOVE)
   AddBool(aocat, "audiooutput.multichannellpcm" , 348, true );
 #endif
 #if !defined(TARGET_DARWIN) && !defined(TARGET_RASPBERRY_PI)
