diff -aur --unidirectional-new-file a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-02-17 08:57:29.476103283 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-02-17 08:51:00.932119533 +0100
@@ -45,7 +45,7 @@
 #include "utils/BitstreamConverter.h"
 
 
-//#define ENABLE_MPEG1            // use vMeta for MPEG1 decoding
+#define ENABLE_MPEG1            // use vMeta for MPEG1 decoding
 //#define ENABLE_PTS              // honour presentation time stamps
 
 
@@ -733,6 +733,13 @@
   pDvdVideoPicture->iWidth          = m_picture_width;
   pDvdVideoPicture->iHeight         = m_picture_height;
 
+#ifdef ENABLE_MPEG1
+  // Workaround for bug in vMeta that causes a flickering 
+  // line on the bottom of the screen
+  if (m_VDecParSet.strm_fmt == IPP_VIDEO_STRM_FMT_MPG1)
+    pDvdVideoPicture->iHeight--;
+#endif
+
   if (m_output_ready.getHead(pPicture))
   {
     // clone the video picture buffer settings.
