diff -aur --unidirectional-new-file a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-04-18 00:32:41.162552662 +0200
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2013-04-18 00:30:54.000000000 +0200
@@ -44,6 +44,7 @@
 
 #include "utils/BitstreamConverter.h"
 #include "settings/GUISettings.h"
+#include "DVDCodecs/DVDCodecs.h"
 
 
 #define ENABLE_MPEG1            // use vMeta for MPEG1 decoding
@@ -208,7 +209,17 @@
       m_VDecParSet.strm_fmt = IPP_VIDEO_STRM_FMT_MPG4;
       m_video_codec_name = "vmeta-mpeg4";
       bSendCodecConfig = true;
-      m_codec_species = 2; /* TODO - Better detect xvid, and divx versions */
+      
+      if (hints.codec_tag == MKTAG('X','V','I','D'))
+        m_codec_species = 3;
+      else if (hints.codec_tag == MKTAG('D','X','5','0'))
+        m_codec_species = 2;
+      else if (hints.codec_tag == MKTAG('D','I','V','X'))
+        m_codec_species = 1;
+      else
+        m_codec_species = 0;
+      
+      //m_codec_species = 2; /* TODO - Better detect xvid, and divx versions */
       break;
 
     case CODEC_ID_MPEG2VIDEO:

