diff -u a/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp b/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp
--- a/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp	2013-12-22 23:22:11.239338612 +0100
+++ b/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp	2013-12-22 23:20:18.000000000 +0100
@@ -126,7 +126,8 @@
   m_overlaySurface.viewPortInfo.srcWidth  = m_sourceRect.x2 - m_sourceRect.x1;
   m_overlaySurface.viewPortInfo.srcHeight = m_sourceRect.y2 - m_sourceRect.y1;
   m_overlaySurface.viewPortInfo.zoomXSize = m_destRect.x2 - m_destRect.x1;
-  m_overlaySurface.viewPortInfo.zoomYSize = m_destRect.y2 - m_destRect.y1;
+  // Hack to avoid flickering line on bottom of screen with some MPEG1 videos
+  m_overlaySurface.viewPortInfo.zoomYSize = m_destRect.y2 - m_destRect.y1 - 1;
 
   m_overlaySurface.viewPortOffset.xOffset = m_destRect.x1;
   m_overlaySurface.viewPortOffset.yOffset = m_destRect.y1;
