diff -Naur a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2014-01-26 21:35:04.310933171 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp	2014-01-26 21:39:00.070926206 +0100
@@ -469,12 +469,12 @@
       // special handling: MPEG4 packed bitstream
       uint8_t *digest_ret = digest_mpeg4_inbuf(pData, iSize);
 
-      if ((uint32_t)digest_ret == 0xffffffff)
+      if (((uint32_t)digest_ret | 0x1) == 0xffffffff)
       {
         pData = 0;
-        iSize = 0;      // Skip null VOP
+        iSize = 0;      // Skip null VOP and stuffing byte
       }
-      else if ((uint32_t)digest_ret)
+      else if (digest_ret)
       {
         int iTemp = digest_ret - pData;
         iSize2nd = iSize - iTemp;
