--- ./tools/TexturePacker/Makefile.in	2012-10-24 22:53:23.000000000 +0200
+++ ./tools/TexturePacker/Makefile.in-new	2012-10-30 09:30:01.551383647 +0100
@@ -26,7 +26,7 @@
                   -I@abs_top_srcdir@/xbmc/linux
 NATIVE_LIBS    += -L$(NATIVE_ROOT_PATH)/lib
 endif
-NATIVE_LIBS    += -lSDL_image -lSDL -llzo2
+NATIVE_LIBS    += -lSDL_image -lSDL -llzo2 -lpthread -ljpeg -lpng -lz
 NATIVE_LIBS    += -L@abs_top_srcdir@/lib/libsquish -lsquish-native
 else
 LIBS    += -L@abs_top_srcdir@/lib/libsquish -lsquish
