diff -Naur c/tools/TexturePacker/Makefile.in d/tools/TexturePacker/Makefile.in
--- c/tools/TexturePacker/Makefile.in	2012-07-11 21:58:21.000000000 +0200
+++ d/tools/TexturePacker/Makefile.in	2012-10-28 13:32:40.000000000 +0100
@@ -3,26 +3,25 @@
 DEFINES += -DHOST_BIGENDIAN
 endif
 
-CXXFLAGS+= \
-  -I. \
-  -I@abs_top_srcdir@/lib \
-  -I@abs_top_srcdir@/xbmc \
-  -I@abs_top_srcdir@/xbmc/linux
-
 ifeq (@USE_TEXTUREPACKER_NATIVE@,1)
 NATIVE_ROOT_PATH=@USE_TEXTUREPACKER_NATIVE_ROOT@
 ifdef NATIVE_ROOT_PATH
+CXXFLAGS = -I$(NATIVE_ROOT_PATH)/include
+LIBS     = -L$(NATIVE_ROOT_PATH)/lib
+endif
 ifeq ($(findstring osx,@ARCH@),osx)
 CXXFLAGS+= -m32
 endif
-CXXFLAGS+= -I$(NATIVE_ROOT_PATH)/include
-LIBS    += -L$(NATIVE_ROOT_PATH)/lib
-endif
 LIBS    += -L@abs_top_srcdir@/lib/libsquish -lsquish-native
 else
 LIBS    += -L@abs_top_srcdir@/lib/libsquish -lsquish
 endif
 
+CXXFLAGS+= \
+  -I. \
+  -I@abs_top_srcdir@/lib \
+  -I@abs_top_srcdir@/xbmc \
+  -I@abs_top_srcdir@/xbmc/linux
+LIBS    += -lSDL_image -lSDL -llzo2 -lpthread -ljpeg -lpng -lz 
-LIBS    += -lSDL_image -lSDL -llzo2
 
 SRCS = \

