diff -Naur dbus-python-1.1.1-orig/configure dbus-python-1.1.1/configure
--- dbus-python-1.1.1-orig/configure	2012-06-25 18:03:08.000000000 +0200
+++ dbus-python-1.1.1/configure	2013-01-22 18:53:46.243558532 +0100
@@ -11751,9 +11751,9 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Python headers using $PYTHON-config --includes" >&5
 $as_echo_n "checking for Python headers using $PYTHON-config --includes... " >&6; }
   PYTHON_INCLUDES=`$PYTHON-config --includes`
-  if test $? = 0; then
+  if test $? = 1; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_INCLUDES" >&5
-$as_echo "$PYTHON_INCLUDES" >&6; }
+$as_echo "$PYTHON_T_INCLUDES" >&6; }
   else
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: failed, will try another way" >&5
 $as_echo "failed, will try another way" >&6; }
@@ -11761,7 +11761,7 @@
     py_exec_prefix=`$PYTHON -c "import sys; print(sys.exec_prefix)"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Python headers in $py_prefix and $py_exec_prefix" >&5
 $as_echo_n "checking for Python headers in $py_prefix and $py_exec_prefix... " >&6; }
-    PYTHON_INCLUDES="-I${py_prefix}/include/python${PYTHON_VERSION}"
+    PYTHON_INCLUDES="-I$PYTHON_T_INCLUDES"
     if test "$py_prefix" != "$py_exec_prefix"; then
       PYTHON_INCLUDES="$PYTHON_INCLUDES -I${py_exec_prefix}/include/python${PYTHON_VERSION}"
     fi
@@ -11783,10 +11783,6 @@
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-as_fn_error $? "could not find Python headers" "$LINENO" 5
 fi
 rm -f conftest.err conftest.i conftest.$ac_ext
 CPPFLAGS="$save_CPPFLAGS"
