--- sgx540-OMAP44xx-drv-1.7.10.1.20/sgx/services4/srvkm/devices/sgx/mmu.c.orig	2012-04-12 14:22:18.475066936 +0300
+++ sgx540-OMAP44xx-drv-1.7.10.1.20/sgx/services4/srvkm/devices/sgx/mmu.c	2012-04-12 14:22:36.343155536 +0300
@@ -980,7 +980,7 @@
 		if(ppsPTInfoList[i]->hPTPageOSMemHandle == IMG_NULL
 		&& ppsPTInfoList[i]->PTPageCpuVAddr == IMG_NULL)
 		{
-			IMG_DEV_PHYADDR	sDevPAddr;
+			IMG_DEV_PHYADDR	sDevPAddr = {0};
 #if defined(SUPPORT_SGX_MMU_DUMMY_PAGE)
 			IMG_UINT32 *pui32Tmp;
 			IMG_UINT32 j;
