diff -Naur vdr-skinelchi-0.2.1/videostatus.h vdr-skinelchi-0.2.1a/videostatus.h
--- vdr-skinelchi-0.2.1/videostatus.h	2008-03-21 13:22:49.000000000 +0100
+++ vdr-skinelchi-0.2.1a/videostatus.h	2009-08-02 18:13:12.000000000 +0200
@@ -10,6 +10,8 @@
 #ifndef __SKINELCHI_VIDEOSTATUS_H_
 #define __SKINELCHI_VIDEOSTATUS_H_
 
+typedef long __kernel_time_t;
+
 #include <vdr/thread.h>
 #include <vdr/plugin.h>
 #include <linux/dvb/video.h>
