--- vdr-skinelchi-0.2.1/Makefile.org	2012-04-11 15:40:50.934407009 +0200
+++ vdr-skinelchi-0.2.1/Makefile	2012-04-11 15:42:11.674407051 +0200
@@ -33,7 +33,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual
+CXXFLAGS ?= -O2 -fPIC -Wall -Werror=overloaded-virtual
 
 ### The directory environment:
 
@@ -71,7 +71,7 @@
 
 ### The object files (add further files here):
 OBJS = $(PLUGIN).o DisplayChannel.o DisplayMenu.o DisplayReplay.o DisplayVolume.o DisplayTracks.o DisplayMessage.o
-OBJS += config.o vdrstatus.o scroll.o i18n.o setup.o videostatus.o
+OBJS += config.o vdrstatus.o scroll.o setup.o videostatus.o
 
 ifdef SKINELCHI_HAVE_IMAGEMAGICK
 	LIBS += -lMagick++
