diff -ruN skinelchi-0.2.7.orig/DisplayChannel.c skinelchi-0.2.7/DisplayChannel.c
--- skinelchi-0.2.7.orig/DisplayChannel.c	2012-01-15 19:27:05.000000000 +0100
+++ skinelchi-0.2.7/DisplayChannel.c	2012-12-08 14:19:37.463479933 +0100
@@ -741,7 +741,7 @@
 
       if (Present) {
          ///isyslog("SetEvents: present, hastimer=%d, IsRunning=%d", Present->HasTimer(),Present->IsRunning());
-         int TimerMatch = tmNone;
+         eTimerMatch TimerMatch = tmNone;
          const cTimer * Timer = Timers.GetMatch(Present, &TimerMatch);
          if (Timer && Timer->Recording()) {
             ///isyslog("SetEvents: present has Timer");
