--- vdr-markad-0.1.4/plugin/Makefile.org	2012-11-28 15:14:37.432793782 +0100
+++ vdr-markad-0.1.4/plugin/Makefile	2012-11-28 15:14:58.972794176 +0100
@@ -99,11 +99,11 @@
 ### Targets:
 
 install: all 
-	cp --remove-destination libvdr-$(PLUGIN).so $(LIBDIR)/libvdr-$(PLUGIN).so.$(APIVERSION)
+#	cp --remove-destination libvdr-$(PLUGIN).so ../libvdr-$(PLUGIN).so.$(APIVERSION)
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) $(LIBS) -o $@
-	cp --remove-destination libvdr-$(PLUGIN).so $(LIBDIR)/libvdr-$(PLUGIN).so.$(APIVERSION)
+	cp --remove-destination libvdr-$(PLUGIN).so ../libvdr-$(PLUGIN).so.$(APIVERSION)
 
 clean:
-	@-rm -f $(OBJS) $(DEPFILE) *.so *.so.* *.tgz core* *~ $(PODIR)/*.mo $(PODIR)/*.pot
+#	@-rm -f $(OBJS) $(DEPFILE) *.so *.so.* *.tgz core* *~ $(PODIR)/*.mo $(PODIR)/*.pot
--- vdr-markad-0.1.4/plugin/Makefile.org	2013-03-06 21:43:13.050339394 +0100
+++ vdr-markad-0.1.4/plugin/Makefile	2013-03-06 21:42:56.058339103 +0100
@@ -74,7 +74,7 @@
 ### Internationalization (I18N):
 
 PODIR     = po
-LOCALEDIR = $(VDRDIR)/locale
+LOCALEDIR = ../.install/usr/share/locale
 I18Npo    = $(wildcard $(PODIR)/*.po)
 I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
 I18Npot   = $(PODIR)/$(PLUGIN).pot
