diff -Naur c/configure d/configure
--- c/configure	2011-07-10 07:46:57.000000000 +0200
+++ d/configure	2011-07-10 14:25:52.000000000 +0200
@@ -19570,7 +19570,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libbluray soname" >&5
 $as_echo_n "checking for libbluray soname... " >&6; }
     BLURAY_SONAME=$( $CC -print-file-name=libbluray.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20496,7 +20496,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libmad soname" >&5
 $as_echo_n "checking for libmad soname... " >&6; }
     MAD_SONAME=$( $CC -print-file-name=libmad.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20541,7 +20541,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libogg soname" >&5
 $as_echo_n "checking for libogg soname... " >&6; }
     OGG_SONAME=$( $CC -print-file-name=libogg.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20586,7 +20586,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libcurl soname" >&5
 $as_echo_n "checking for libcurl soname... " >&6; }
     CURL_SONAME=$( $CC -print-file-name=libcurl.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20631,7 +20631,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libFLAC soname" >&5
 $as_echo_n "checking for libFLAC soname... " >&6; }
     FLAC_SONAME=$( $CC -print-file-name=libFLAC.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20676,7 +20676,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libvorbis soname" >&5
 $as_echo_n "checking for libvorbis soname... " >&6; }
     VORBIS_SONAME=$( $CC -print-file-name=libvorbis.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20721,7 +20721,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libvorbisenc soname" >&5
 $as_echo_n "checking for libvorbisenc soname... " >&6; }
     VORBISENC_SONAME=$( $CC -print-file-name=libvorbisenc.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20766,7 +20766,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libvorbisfile soname" >&5
 $as_echo_n "checking for libvorbisfile soname... " >&6; }
     VORBISFILE_SONAME=$( $CC -print-file-name=libvorbisfile.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20810,7 +20810,7 @@
   if [ "$host_vendor" != "apple" ]; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libmodplug soname" >&5
 $as_echo_n "checking for libmodplug soname... " >&6; }
-    MODPLUG_SONAME=$( $CC -print-file-name=libmodplug.so | \
+    MODPLUG_SONAME=$( $CC -print-file-name=$LIB_P/lib/libmodplug.so | \
       while read output; do objdump -p $output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
@@ -20856,7 +20856,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libass soname" >&5
 $as_echo_n "checking for libass soname... " >&6; }
     ASS_SONAME=$( $CC -print-file-name=libass.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -20901,7 +20901,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libmpeg2 soname" >&5
 $as_echo_n "checking for libmpeg2 soname... " >&6; }
     MPEG2_SONAME=$( $CC -print-file-name=libmpeg2.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -21424,7 +21424,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for librtmp soname" >&5
 $as_echo_n "checking for librtmp soname... " >&6; }
     RTMP_SONAME=$( $CC -print-file-name=librtmp.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -21505,7 +21505,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libnfs soname" >&5
 $as_echo_n "checking for libnfs soname... " >&6; }
     NFS_SONAME=$( $CC -print-file-name=libnfs.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
@@ -22421,7 +22421,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libcrystalhd soname" >&5
 $as_echo_n "checking for libcrystalhd soname... " >&6; }
     CRYSTALHD_SONAME=$( $CC -print-file-name=libcrystalhd.so | \
-      while read output; do objdump -p $output | \
+      while read output; do objdump -p $LIB_P/lib/$output | \
         grep "SONAME" | \
         sed -e 's/  *SONAME  *//'; done 2> /dev/null )
   else
