diff -Naur xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620/xbmc/windowing/WindowingFactory.h xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620a/xbmc/windowing/WindowingFactory.h
--- xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620/xbmc/windowing/WindowingFactory.h	2011-10-07 07:11:39.000000000 +0200
+++ xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620a/xbmc/windowing/WindowingFactory.h	2011-10-08 17:46:02.958983438 +0200
@@ -33,10 +33,10 @@
 #elif defined(TARGET_LINUX)   && defined(HAS_GLX)
 #include "X11/WinSystemX11GL.h"
 
-#elif defined(TARGET_LINUX)   &&  defined(HAS_X11) && defined(HAS_GLES)
+#elif defined(TARGET_LINUX)   &&  defined(HAVE_X11) && defined(HAS_GLES)
 #include "X11/WinSystemX11GLES.h"
 
-#elif defined(TARGET_LINUX)   && !defined(HAS_X11) && defined(HAS_GLES)
+#elif defined(TARGET_LINUX)   && !defined(HAVE_X11) && defined(HAS_GLES)
 #include "egl/WinSystemGLES.h"
 
 #elif defined(TARGET_DARWIN_OSX)
diff -Naur xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620/xbmc/windowing/X11/WinSystemX11GLES.cpp xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620a/xbmc/windowing/X11/WinSystemX11GLES.cpp
--- xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620/xbmc/windowing/X11/WinSystemX11GLES.cpp	2011-10-07 07:11:39.000000000 +0200
+++ xbmc-pvr-657c930017dddcd781d3ba9880a2527173495620a/xbmc/windowing/X11/WinSystemX11GLES.cpp	2011-10-08 17:45:40.822980417 +0200
@@ -20,7 +20,7 @@
  */
 #include "system.h"
 
-#if defined(HAS_X11) && defined(HAS_GLES)
+#if defined(HAVE_X11) && defined(HAS_GLES)
 
 #include "WinSystemX11GLES.h"
 #include "utils/log.h"
