diff -Naur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2011-08-07 10:35:32.000000000 +0200
+++ b/Makefile.in	2011-08-07 15:09:18.270868295 +0200
@@ -146,16 +146,12 @@
 	lib/xbmc-dll-symbols
 
 SS_DIRS=
-ifeq (@USE_OPENGL@,1)
-SS_DIRS+= xbmc/screensavers/rsxs-0.9/xbmc
-endif
 
 VIS_DIRS=
 ifeq (@USE_OPENGL@,1)
 VIS_DIRS+=\
 	xbmc/visualizations/OpenGLSpectrum \
 	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM \
 
 endif
 ifeq (($(findstring osx,@ARCH@), osx),(@USE_OPENGL@,1)) 
