diff -Naur c/Makefile.in d/Makefile.in
--- c/Makefile.in	2011-07-09 15:58:43.000000000 +0200
+++ d/Makefile.in	2011-07-10 14:26:07.000000000 +0200
@@ -137,15 +137,14 @@
 	lib/cpluff \
 	lib/xbmc-dll-symbols
 
-SS_DIRS=\
-	xbmc/screensavers/rsxs-0.9/xbmc
+SS_DIRS=
 
 VIS_DIRS=
 ifeq (@USE_OPENGL@,1)
 VIS_DIRS+=\
 	xbmc/visualizations/OpenGLSpectrum \
 	xbmc/visualizations/WaveForm \
-	xbmc/visualizations/XBMCProjectM \
+
 
 endif
 ifeq (($(findstring osx,@ARCH@), osx),(@USE_OPENGL@,1)) 
@@ -429,9 +428,7 @@
 	$(MAKE) -C xbmc/pvrclients/vdr-vnsi
 	$(MAKE) -C xbmc/pvrclients/tvheadend
 screensavers: exports
-ifeq (@USE_OPENGL@,1)
-	$(MAKE) -C xbmc/screensavers/rsxs-0.9/xbmc
-endif
+
 libaddon: exports
 	$(MAKE) -C lib/addons/library.xbmc.addon
 	$(MAKE) -C lib/addons/library.xbmc.gui
