diff -Naur attr-2.4.43.orig/include/buildmacros attr-2.4.43/include/buildmacros
--- attr-2.4.43.orig/include/buildmacros	2009-12-13 00:52:53.233769000 +0100
+++ attr-2.4.43/include/buildmacros	2009-12-13 00:53:08.471269847 +0100
@@ -42,10 +42,10 @@
 LTOBJECTS = $(OBJECTS:.o=.lo)
 LTVERSION = $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 
-LTLINK = $(LIBTOOL) --mode=link $(CC)
+LTLINK = $(LIBTOOL) --tag=CC --mode=link $(CC)
 LTEXEC = $(LIBTOOL) --mode=execute
 LTINSTALL = $(LIBTOOL) --mode=install $(INSTALL)
-LTCOMPILE = $(LIBTOOL) --mode=compile $(CCF)
+LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CCF)
 
 ifeq ($(ENABLE_SHARED),yes)
 LTLDFLAGS += -rpath $(PKG_LIB_DIR)
