diff -Naur attr-2.4.44/Makefile attr-2.4.44a/Makefile
--- attr-2.4.44/Makefile	2009-07-31 11:02:14.000000000 +0200
+++ attr-2.4.44a/Makefile	2010-10-17 18:14:44.399805351 +0200
@@ -89,7 +89,7 @@
 		$(MAKE) $(AM_MAKEFLAGS) include/builddefs; \
 	fi
 
-install: default $(addsuffix -install,$(SUBDIRS))
+install: default $(addsuffix -install,$(SUBDIRS)) $(addsuffix -install-dev,$(SUBDIRS)) $(addsuffix -install-lib,$(SUBDIRS))
 	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
 	$(INSTALL) -m 644 README $(PKG_DOC_DIR)
 
