--- a/boost/exception/detail/exception_ptr_base.hpp	2012-01-04 07:17:47.690594223 +0100
+++ b/boost/exception/detail/exception_ptr_base.hpp	2012-01-04 07:18:16.990637208 +0100
@@ -22,10 +22,7 @@
 
                         protected:
 
-                        virtual
-                        ~exception_ptr_base() throw()
-                            {
-                            }
+                        ~exception_ptr_base() throw() { }
                         };
                 }
     }
