diff -Naur gcc-4.6.1.orig//gcc/c-family/c-common.h gcc-4.6.1/gcc/c-family/c-common.h
--- gcc-4.6.1.orig//gcc/c-family/c-common.h	2011-07-08 19:10:47.708474171 +0200
+++ gcc-4.6.1/gcc/c-family/c-common.h	2011-07-08 19:18:43.588474438 +0200
@@ -192,6 +192,7 @@
   RID_LAST_PQ = RID_ONEWAY,
   RID_FIRST_PATTR = RID_GETTER,
   RID_LAST_PATTR = RID_NONATOMIC
+  ,__LAST_AND_UNUSED_RID=32767 /* Force 16bit width.  */
 };
 
 #define OBJC_IS_AT_KEYWORD(rid) \
diff -Naur gcc-4.6.1.orig//gcc/rtl.h gcc-4.6.1/gcc/rtl.h
--- gcc-4.6.1.orig//gcc/rtl.h	2011-07-08 19:10:47.708474171 +0200
+++ gcc-4.6.1/gcc/rtl.h	2011-07-08 19:17:20.898474392 +0200
@@ -50,9 +50,11 @@
 #include "rtl.def"		/* rtl expressions are documented here */
 #undef DEF_RTL_EXPR
 
-  LAST_AND_UNUSED_RTX_CODE};	/* A convenient way to get a value for
+  LAST_AND_UNUSED_RTX_CODE	/* A convenient way to get a value for
 				   NUM_RTX_CODE.
 				   Assumes default enum value assignment.  */
+  ,__LAST_AND_UNUSED_RTX_CODE=32767 /* Force 16bit width.  */
+};
 
 #define NUM_RTX_CODE ((int) LAST_AND_UNUSED_RTX_CODE)
 				/* The cast here, saves many elsewhere.  */
