--- a/utils/loadndisdriver.c	2007-06-19 08:59:41.000000000 +0800
+++ b/utils/loadndisdriver.c	2007-06-19 09:04:00.000000000 +0800
@@ -50,7 +50,8 @@
 #endif

+FILE *fp; 
 #define LOG_MSG(where, fmt, ...)					\
-	syslog(LOG_KERN | where, "%s: %s(%d): " fmt "\n",		\
+	fprintf(fp, "%s: %s(%d): " fmt "\n",		\
 	       PROG_NAME, __FUNCTION__, __LINE__ , ## __VA_ARGS__)
 #define ERROR(fmt, ...) LOG_MSG(LOG_INFO, fmt, ## __VA_ARGS__)
 #define INFO(fmt, ...) LOG_MSG(LOG_INFO, fmt, ## __VA_ARGS__)
@@ -508,7 +508,10 @@
 	int i, ioctl_device, res;
 	char *cmd;
 
+    fp = fopen("/var/log/loadndisdriver.log","a");
+/*
 	openlog(PROG_NAME, LOG_PERROR | LOG_CONS, LOG_KERN | LOG_DEBUG);
+*/
 
 	DBG("argc: %d", argc);
 
@@ -587,7 +590,10 @@
 out:
 	if (ioctl_device != -1)
 		close(ioctl_device);
+    fclose(fp);
+/*
 	closelog();
+*/
 	return res;
 }
 
