diff -Naur ndiswrapper-1.55.orig/utils/loadndisdriver.c ndiswrapper-1.55/utils/loadndisdriver.c
--- ndiswrapper-1.55.orig/utils/loadndisdriver.c	2009-07-25 14:35:00.000000000 +0200
+++ ndiswrapper-1.55/utils/loadndisdriver.c	2009-07-25 14:37:05.000000000 +0200
@@ -49,8 +49,9 @@
 	directory only
 #endif
 
+static FILE *fp;
 #define LOG_MSG(where, fmt, ...)					\
-	syslog(LOG_KERN | where, "%s: %s(%d): " fmt "\n",		\
+	 fprintf(fp, "%s: %s(%d): " fmt "\n",            \
 	       PROG_NAME, __func__, __LINE__ , ## __VA_ARGS__)
 #define ERROR(fmt, ...) LOG_MSG(LOG_INFO, fmt, ## __VA_ARGS__)
 #define INFO(fmt, ...) LOG_MSG(LOG_INFO, fmt, ## __VA_ARGS__)
@@ -508,7 +509,7 @@
 	int i, ioctl_device, res;
 	char *cmd;
 
-	openlog(PROG_NAME, LOG_PERROR | LOG_CONS, LOG_KERN | LOG_DEBUG);
+	fp = fopen("/var/log/loadndisdriver.log","a");
 
 	DBG("argc: %d", argc);
 
@@ -523,7 +524,7 @@
 	}
 
 	cmd = argv[1];
-	i = -1;
+	i = 1;
 	i = atoi(argv[2]);
 	if (i < 0) {
 		ERROR("invalid debug value %d", i);
@@ -587,7 +588,7 @@
 out:
 	if (ioctl_device != -1)
 		close(ioctl_device);
-	closelog();
+	fclose(fp);
 	return res;
 }
 
