diff -Naur ndiswrapper-1.55.orig/driver/ntoskernel.h ndiswrapper-1.55/driver/ntoskernel.h
--- ndiswrapper-1.55.orig/driver/ntoskernel.h	2009-07-25 14:35:00.000000000 +0200
+++ ndiswrapper-1.55/driver/ntoskernel.h	2009-07-25 14:38:28.000000000 +0200
@@ -875,7 +875,6 @@
 #define u64_low_32(x) ((u32)x)
 #define u64_high_32(x) ((u32)(x >> 32))
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,31)
 static inline u64 cmpxchg8b(volatile u64 *ptr, u64 old, u64 new)
 {
 	u64 prev;
@@ -887,7 +886,6 @@
 		: "A" (old), "b" (u64_low_32(new)), "c" (u64_high_32(new)));
 	return prev;
 }
-#endif
 
 /* slist routines below update slist atomically - no need for
  * spinlocks */
