diff -Naur ndiswrapper-1.56/driver/wrapndis.c ndiswrapper-1.56a/driver/wrapndis.c
--- ndiswrapper-1.56/driver/wrapndis.c	2011-02-17 11:26:06.545875417 +0100
+++ ndiswrapper-1.56a/driver/wrapndis.c	2011-02-17 11:26:52.929380487 +0100
@@ -13,16 +13,16 @@
  *
  */
 
-#include "ndis.h"
-#include "iw_ndis.h"
-#include "pnp.h"
-#include "loader.h"
-#include "wrapndis.h"
 #include <linux/inetdevice.h>
 #include <linux/ip.h>
 #include <linux/tcp.h>
 #include <linux/udp.h>
 #include <linux/in.h>
+#include "ndis.h"
+#include "iw_ndis.h"
+#include "pnp.h"
+#include "loader.h"
+#include "wrapndis.h"
 #include "wrapper.h"
 
 /* Functions callable from the NDIS driver */
