dont override CFLAGS.

diff -Nur ndiswrapper-1.21.orig/utils/Makefile ndiswrapper-1.21/utils/Makefile
--- ndiswrapper-1.21.orig/utils/Makefile	Sat Jul 15 20:40:28 2006
+++ ndiswrapper-1.21/utils/Makefile	Sat Aug  5 20:23:23 2006
@@ -5,7 +5,7 @@
 usrsbindir = $(DESTDIR)/usr/sbin
 
 CC = gcc
-CFLAGS = -g -Wall -DUTILS_VERSION=\"$(UTILS_VERSION)\" 
+CFLAGS += -g -Wall -DUTILS_VERSION=\"$(UTILS_VERSION)\" 
 
 DISTFILES=Makefile ndiswrapper loadndisdriver.c ndiswrapper-buginfo
 
