fix warnings about variables not being initilized.

diff -Nur ndiswrapper-1.21.orig/utils/loadndisdriver.c ndiswrapper-1.21/utils/loadndisdriver.c
--- ndiswrapper-1.21.orig/utils/loadndisdriver.c	Sat Jul 15 20:40:28 2006
+++ ndiswrapper-1.21/utils/loadndisdriver.c	Sat Aug  5 20:30:15 2006
@@ -152,7 +152,7 @@
 	char setting_name[MAX_SETTING_NAME_LEN];
 	char setting_value[MAX_SETTING_VALUE_LEN];
 	int ret, nr_settings;
-	int i, vendor, device, subvendor, subdevice, bus_type;
+	int vendor, device, subvendor, subdevice, bus_type;
 
 	if (lstat(conf_file_name, &statbuf)) {
 		ERROR("unable to open config file: %s", strerror(errno));
@@ -167,8 +167,8 @@
 			  &bus_type) == 5) {
 		DBG("bus_type: %X", bus_type);
 	} else {
-		ERROR("unable to parse conf file name %s (%d)",
-		      conf_file_name, i);
+		ERROR("unable to parse conf file name %s",
+		      conf_file_name);
 		return -EINVAL;
 	}
 
@@ -599,7 +599,7 @@
 */
 int main(int argc, char *argv[0])
 {
-	int i, ioctl_device, res;
+	int i, ioctl_device = -1, res = 0;
 	char *cmd;
 
 	openlog(PROG_NAME, LOG_PERROR | LOG_CONS, LOG_KERN | LOG_DEBUG);
