diff -Naur vdr-xineliboutput-cvs20110103/Makefile vdr-xineliboutput-cvs20110103a/Makefile
--- vdr-xineliboutput-cvs20110103/Makefile	2010-12-30 17:34:56.000000000 +0100
+++ vdr-xineliboutput-cvs20110103a/Makefile	2011-01-10 10:07:32.712615867 +0100
@@ -269,7 +269,7 @@
 ###
 
 mpg2c: mpg2c.c
-	$(CC) $(CFLAGS) $(LDFLAGS) mpg2c.c -o $@
+	$(HOST_CC) $(HOST_CFLAGS) $(HOST_LDFLAGS) mpg2c.c -o $@
 
 # data
 black_720x576.c: mpg2c black_720x576.mpg
