Index: vdr-plugin-xineliboutput_1.0.7+cvs20111217.2347/osd.c
===================================================================
--- vdr-plugin-xineliboutput_1.0.7+cvs20111217.2347.orig/osd.c	2011-12-18 17:58:32.506780226 +0100
+++ vdr-plugin-xineliboutput_1.0.7+cvs20111217.2347/osd.c	2011-12-18 18:01:11.000000000 +0100
@@ -526,10 +526,10 @@
   if(!m_IsVisible)
     return;
 
-#ifdef YAEPGHDVERSNUM
+
   if (vidWin.bpp)
     CmdVideoWindow(vidWin.x1, vidWin.y1, vidWin.Width(), vidWin.Height());
-#endif
+
 
 #if VDRVERSNUM >= 10717
   if (IsTrueColor()) {
