Index: wlan/wlan/CUDK/configurationutility/src/cu_cmd.c
===================================================================
--- wlan.orig/wlan/CUDK/configurationutility/src/cu_cmd.c	2010-09-28 14:23:25.000000000 +0200
+++ wlan/wlan/CUDK/configurationutility/src/cu_cmd.c	2010-09-28 14:23:56.000000000 +0200
@@ -5447,7 +5447,7 @@
 #ifdef _WINDOWS
     PS8 nvsFilePath = (PS8)"/windows/nvs_map.bin";
 #else
-    PS8 nvsFilePath = (PS8)"/lib/firmware/ti-wlan/nvs_map.bin";
+    PS8 nvsFilePath = (PS8)"/lib/firmware/tiwlan-wl1271/nvs_map.bin";
 #endif /*_WINDOWS*/
 	TI_UINT8		currentNVSbuffer[1500];
 	TI_UINT16		lengthOfCurrentNVSBufer;
Index: wlan/wlan/CUDK/tiwlan_loader/tiwlan_loader.c
===================================================================
--- wlan.orig/wlan/CUDK/tiwlan_loader/tiwlan_loader.c	2010-09-28 14:21:03.000000000 +0200
+++ wlan/wlan/CUDK/tiwlan_loader/tiwlan_loader.c	2010-09-28 14:22:59.000000000 +0200
@@ -202,9 +202,9 @@
 S32 user_main(S32 argc, PPS8 argv)
 {
     S32 i;
-    PS8 eeprom_file_name = (PS8)"/lib/firmware/ti-wlan/nvs_map.bin";
-    PS8 init_file_name = (PS8)"/lib/firmware/ti-wlan/tiwlan.ini";
-    PS8 firmware_file_name = (PS8)"/lib/firmware/ti-wlan/firmware.bin";
+    PS8 eeprom_file_name = (PS8)"/lib/firmware/tiwlan-wl1271/nvs_map.bin";
+    PS8 init_file_name = (PS8)"/lib/firmware/tiwlan-wl1271/tiwlan.ini";
+    PS8 firmware_file_name = (PS8)"/lib/firmware/tiwlan-wl1271/firmware.bin";
 
     /* Parse command line parameters */
     if( argc > 1 )
