Index: wlan/wlan/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ wlan/wlan/Makefile	2010-09-09 00:09:11.000000000 +0200
@@ -0,0 +1,25 @@
+# This top-level makefile is called by DKMS.
+# It must provide targets: 'all' and 'clean'.
+# Making 'all' the default target is nice to have.
+# This makefile must take the $KVERSION variable into account.
+
+.PHONY: all clean
+
+KVERSION ?= $(shell uname -r)
+DIR := platforms/os/linux
+KERNEL_HEADERS ?= /usr/src/linux-headers-$(KVERSION)
+BINDIR := /usr/bin
+CROSS_COMPILE ?=
+ARCH ?= arm
+
+all:
+	$(MAKE) -C $(DIR) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) HOST_PLATFORM=sdc4430 \
+	KERNEL_DIR=$(KERNEL_HEADERS) LDFLAGS= BUILD_SUPPL=n \
+	../../../platforms/os/linux/tiwlan_drv.ko sdio \
+	&& cp ./platforms/os/linux/tiwlan_drv.ko .\
+	&& cp ./external_drivers/sdc4430/Linux/sdio/sdio.ko . \
+# Note: we replace the 'make install' step by our cp-install for now.
+
+clean:
+	$(MAKE) -C $(DIR) CROSS_COMPILE=$(CROSS_COMPILE) ARCH=$(ARCH) HOST_PLATFORM=sdc4430 \
+	KERNEL_DIR=$(KERNEL_HEADERS) BUILD_SUPPL=n clean
