Index: wlan/wlan/CUDK/Makefile
===================================================================
--- wlan.orig/wlan/CUDK/Makefile	2010-12-15 14:46:01.000000000 +0100
+++ wlan/wlan/CUDK/Makefile	2010-12-15 14:46:05.000000000 +0100
@@ -54,9 +54,9 @@
 ifeq ($(SUPPL),WPA)
   SUPPL_TARGET = $(OUTPUT_DIR)/wpa_supplicant
   TI_SUPP_LIB_DIR = $(WPASUPPL_DIR)
-  ifeq ($(BUILD_SUPPL),y)
-    ALL_TARGETS += $(SUPPL_TARGET)
-  endif
+#  ifeq ($(BUILD_SUPPL),y)
+#    ALL_TARGETS += $(SUPPL_TARGET)
+#  endif
 endif
 
 ifeq ($(SUPPL),DEVICESCAPE)
