Index: wlan/wlan/CUDK/Makefile
===================================================================
--- wlan.orig/wlan/CUDK/Makefile	2010-11-03 18:17:16.000000000 +0100
+++ wlan/wlan/CUDK/Makefile	2010-11-03 18:18:22.000000000 +0100
@@ -81,7 +81,7 @@
 endif
 
 ifeq ($(SUPPL),WPA)
-	$(MAKE) -e -C $(TI_SUPP_LIB_DIR) DEBUG=$(DEBUG) XCC=$(XCC) CROSS_COMPILE=$(CROSS_COMPILE) STATIC_LIB=$(STATIC_LIB) TI_WLAN_DRIVER=y CONFIG_CTRL_IFACE=unix CONFIG_EAP_WSC=y CONFIG_WPS=y
+	$(MAKE) -e -C $(TI_SUPP_LIB_DIR)/wpa_supplicant DEBUG=$(DEBUG) XCC=$(XCC) CROSS_COMPILE=$(CROSS_COMPILE) STATIC_LIB=$(STATIC_LIB) TI_WLAN_DRIVER=y CONFIG_CTRL_IFACE=unix CONFIG_EAP_WSC=y CONFIG_WPS=y
 endif
 
 ifeq ($(SUPPL),GEM)
