Index: wlan/wlan/platforms/os/linux/src/WlanDrvIf.c
===================================================================
--- wlan.orig/wlan/platforms/os/linux/src/WlanDrvIf.c	2010-11-05 10:48:22.000000000 +0100
+++ wlan/wlan/platforms/os/linux/src/WlanDrvIf.c	2010-11-05 13:54:13.000000000 +0100
@@ -80,6 +80,12 @@
 /* save driver handle just for module cleanup */
 static TWlanDrvIfObj *pDrvStaticHandle;  
 
+static struct platform_device_id wifi_id_table[] = {
+	{ "device_wifi", },
+	{ "device_wifi_1271", },
+	{ },
+};
+
 #define OS_SPECIFIC_RAM_ALLOC_LIMIT			(0xFFFFFFFF)	/* assume OS never reach that limit */
 
 
@@ -136,6 +142,7 @@
                 .driver         = {
                 .name   = "device_wifi",
             },
+	        .id_table = wifi_id_table,
 };
 
 
@@ -1058,4 +1065,5 @@
 module_init (wlanDrvIf_ModuleInit);
 module_exit (wlanDrvIf_ModuleExit);
 MODULE_ALIAS("platform:device_wifi");
+MODULE_ALIAS("platform:device_wifi_1271");
 
