diff --git a/xbmc/cores/paplayer/DVDPlayerCodec.cpp b/xbmc/cores/paplayer/DVDPlayerCodec.cpp
index c60abc8..74ed872 100644
--- a/xbmc/cores/paplayer/DVDPlayerCodec.cpp
+++ b/xbmc/cores/paplayer/DVDPlayerCodec.cpp
@@ -202,8 +202,9 @@ bool DVDPlayerCodec::Init(const CStdString &strFile, unsigned int filecache)
   }
   else
   {
-    m_pInputStream->Seek(0, SEEK_SET);
-    m_pDemuxer->Reset();
+    // do not reset demuxer when input non-seekable (pipe)
+    if (m_pInputStream->Seek(0, SEEK_SET) == 0)
+      m_pDemuxer->Reset();
     m_bCanSeek = false;
   }
 
