--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2013-12-21 17:58:36.883932924 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2013-12-21 17:59:17.263931732 +0100
@@ -810,6 +810,7 @@
   m_convert_bitstream = false;
   switch(m_hints.codec)
   {
+  case CODEC_ID_MPEG1VIDEO:
   case CODEC_ID_MPEG2VIDEO:
   case CODEC_ID_MPEG2VIDEO_XVMC:
     m_decOpenParam.CodecFormat = VPU_V_MPEG2;
@@ -1103,7 +1104,7 @@
 
     do // Decode as long as the VPU consumes data
     {
-      if (m_decOpenParam.CodecFormat == VPU_V_MPEG2)
+      if (m_decOpenParam.CodecFormat == VPU_V_MPEG2 && m_hints.codec != CODEC_ID_MPEG1VIDEO)
       {
         parse_mpeg2_aspect_ratio(inData.pVirAddr, inData.nSize);
       }
