diff -u a/DVDVideoCodecIMX.cpp b/DVDVideoCodecIMX.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2014-01-03 00:02:58.000000000 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2014-01-02 23:55:36.000000000 +0100
@@ -453,7 +453,7 @@
   return true;
 }
 
-bool CDVDVideoCodecIMX::VpuPushFrame(VpuFrameBuffer *frameBuffer, VpuFieldType field)
+bool CDVDVideoCodecIMX::VpuPushFrame(VpuFrameBuffer *frameBuffer, VpuFieldType field, VpuFrameExtInfo *extraInfo)
 {
   #ifdef IMX_PROFILE
   static unsigned long long previous_ts;
@@ -494,6 +494,8 @@
     outputFrame.v4l2_buffer = &m_v4lBuffers[i];
     outputFrame.pts = pts;
     outputFrame.field = field;
+    outputFrame.nQ16ShiftWidthDivHeightRatio = 
+      ( extraInfo ) ? extraInfo->nQ16ShiftWidthDivHeightRatio : 0x00010000;
 #ifdef IMX_PROFILE
     outputFrame.push_ts = get_time();
     CLog::Log(LOGDEBUG, "%s - Time between push %llu\n",
@@ -1181,7 +1183,7 @@
           CLog::Log(LOGERROR, "%s - VPU Cannot get output frame(%d).\n", __FUNCTION__, ret);
           goto out_error;
         }
-        if (VpuPushFrame(frameInfo.pDisplayFrameBuf, frameInfo.eFieldType))
+        if (VpuPushFrame(frameInfo.pDisplayFrameBuf, frameInfo.eFieldType, frameInfo.pExtInfo))
         {
           retSatus |= VC_PICTURE;
         }        
@@ -1349,8 +1351,9 @@
   pDvdVideoPicture->dts = DVD_NOPTS_VALUE;
   pDvdVideoPicture->iWidth = m_initInfo.PicCropRect.nRight - m_initInfo.PicCropRect.nLeft;
   pDvdVideoPicture->iHeight = m_initInfo.PicCropRect.nBottom - m_initInfo.PicCropRect.nTop;
-  pDvdVideoPicture->iDisplayWidth = m_initInfo.PicCropRect.nRight - m_initInfo.PicCropRect.nLeft;
-  pDvdVideoPicture->iDisplayHeight = m_initInfo.PicCropRect.nBottom - m_initInfo.PicCropRect.nTop;
+  pDvdVideoPicture->iDisplayWidth = ((pDvdVideoPicture->iWidth * 
+                                      outputFrame.nQ16ShiftWidthDivHeightRatio) + 32767) >> 16;
+  pDvdVideoPicture->iDisplayHeight = pDvdVideoPicture->iHeight;
   pDvdVideoPicture->format = RENDER_FMT_IMX;
   pDvdVideoPicture->imx = this;
   return true;
diff -u a/DVDVideoCodecIMX.h b/DVDVideoCodecIMX.h
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2014-01-03 00:02:58.000000000 +0100
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.h	2014-01-02 23:50:59.000000000 +0100
@@ -51,6 +51,7 @@
 #ifdef IMX_PROFILE
   unsigned long long push_ts;
 #endif
+  unsigned nQ16ShiftWidthDivHeightRatio;
 } outputFrameType;
   
 class CDVDVideoCodecIMX : public CDVDVideoCodec
@@ -92,7 +93,7 @@
   bool                m_dropState;
   /* FIXME V4L rendering stuff & Frame Buffers: To be moved in a dedicated class */
   bool VpuAllocFrameBuffers(void);
-  bool VpuPushFrame(VpuFrameBuffer *, VpuFieldType);
+  bool VpuPushFrame(VpuFrameBuffer *, VpuFieldType, VpuFrameExtInfo *);
   bool VpuDeQueueFrame(bool);
   int GetAvailableBufferNb(void);
   void InitFB(void);
