--- xbmc-frodo-r8885f2f2/configure.in	2012-10-27 11:30:20.946526020 +0200
+++ xbmc-frodo-r8885f2f2/configure.in-new	2012-10-27 11:29:47.636477144 +0200
@@ -627,7 +627,7 @@
      use_texturepacker=no
      ARCH="arm"
      use_arch="arm"
-     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
+     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -DLINUX")
      ;;
   arm*-*linux-android*)
      target_platform=target_android
