diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
index a6f3712..4e4f68f 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp
@@ -611,7 +611,7 @@ int CDVDVideoCodecIMX::Decode(BYTE *pData, int iSize, double dts, double pts)
 */
 
   if ((pData && iSize) ||
-     (m_bytesToBeConsumed))
+     ((m_bytesToBeConsumed) && (m_previousPts != DVD_NOPTS_VALUE)))
   {
     if ((m_convert_bitstream) && (iSize))
     {
