--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2015-12-20 14:08:50.650504414 +0100
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2015-12-20 14:08:50.674504415 +0100
@@ -218,7 +218,8 @@
   RESOLUTION_INFO res;
   for (size_t i = 0; i < probe_str.size(); i++)
   {
-    if(!StringUtils::StartsWith(probe_str[i], "S:"))
+    if(!StringUtils::StartsWith(probe_str[i], "S:") && !StringUtils::StartsWith(probe_str[i], "D:") &&
+       !StringUtils::StartsWith(probe_str[i], "V:"))
       continue;
     if(ModeToResolution(probe_str[i], &res))
       resolutions.push_back(res);
