--- xbmc-gotham-alpha-imx-r527c967916/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2014-02-28 19:45:26.585191676 +0100
+++ xbmc-gotham-alpha-imx-r527c967916-new/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2014-02-28 18:54:46.000000000 +0100
@@ -43,10 +43,12 @@
 
 bool CEGLNativeTypeIMX::CheckCompatibility()
 {
-  std::string strName;
+ std::string strName;
+  std::string str2 ("mxc");
   get_sysfs_str("/sys/class/graphics/fb0/device/modalias", strName);
   StringUtils::Trim(strName);
-  if (strName == "platform:mxc_sdc_fb")
+  size_t found = strName.find(str2);
+  if(found!=std::string::npos)
     return true;
   return false;
 }
