--- module-init-tools-3.0.orig/configure	Wed Nov 17 12:03:44 2004
+++ module-init-tools-3.0/configure	Wed Nov 17 12:04:01 2004
@@ -2817,13 +2817,6 @@
 LDADD="$LDADD $zlib_flags"
 
 
-case $target in
-*-*-linux*) ;;
-*) { { echo "$as_me:$LINENO: error: Linux only, dude!" >&5
-echo "$as_me: error: Linux only, dude!" >&2;}
-   { (exit 1); exit 1; }; };;
-esac
-
 # Thanks to Martin Pool
 if test x"$GCC" = xyes
 then
