diff -Naur parted-1.8.8.orig/configure parted-1.8.8/configure
--- parted-1.8.8.orig/configure	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/configure	2008-04-24 21:03:28.000000000 +0200
@@ -11895,7 +11895,6 @@
    gl_LIBOBJS="$gl_LIBOBJS calloc.$ac_objext"
 
 cat >>confdefs.h <<\_ACEOF
-#define calloc rpl_calloc
 _ACEOF
 
 fi
@@ -12369,7 +12368,6 @@
     gl_LIBOBJS="$gl_LIBOBJS free.$ac_objext"
 
 cat >>confdefs.h <<\_ACEOF
-#define free rpl_free
 _ACEOF
 
   fi
@@ -13112,7 +13110,6 @@
    gl_LIBOBJS="$gl_LIBOBJS malloc.$ac_objext"
 
 cat >>confdefs.h <<\_ACEOF
-#define malloc rpl_malloc
 _ACEOF
 
 fi
@@ -13455,7 +13452,6 @@
     if test $ac_cv_func_memcmp_working = no; then
 
 cat >>confdefs.h <<\_ACEOF
-#define memcmp rpl_memcmp
 _ACEOF
 
     :
@@ -14130,7 +14126,6 @@
    gl_LIBOBJS="$gl_LIBOBJS realloc.$ac_objext"
 
 cat >>confdefs.h <<\_ACEOF
-#define realloc rpl_realloc
 _ACEOF
 
 fi
