diff -Naur parted-1.8.8.orig/libparted/fs/Makefile.am parted-1.8.8/libparted/fs/Makefile.am
--- parted-1.8.8.orig/libparted/fs/Makefile.am	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/libparted/fs/Makefile.am	2008-05-02 16:38:54.000000000 +0200
@@ -3,7 +3,7 @@
 #
 # This file may be modified and/or distributed without restriction.
 
-SUBDIRS       = amiga ext2 ufs fat ntfs hfs linux_swap xfs jfs reiserfs # bfs
+SUBDIRS       = ext2 ufs fat ntfs linux_swap xfs jfs reiserfs # bfs
 
 partedincludedir      =	-I$(top_srcdir)/include
 noinst_LTLIBRARIES    =	libfs.la
@@ -14,12 +14,10 @@
 		    $(INTLLIBS)			\
 		    $(OS_LIBS)			\
 		    $(REISER_LIBS)		\
-		    amiga/libamigafs.la		\
 		    ext2/libext2.la		\
 		    ufs/libufs.la		\
 		    fat/libfat.la		\
 		    ntfs/libntfs.la		\
-		    hfs/libhfs.la		\
 		    linux_swap/liblinuxswap.la	\
 		    xfs/libxfs.la		\
 		    jfs/libjfs.la		\
diff -Naur parted-1.8.8.orig/libparted/fs/Makefile.in parted-1.8.8/libparted/fs/Makefile.in
--- parted-1.8.8.orig/libparted/fs/Makefile.in	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/libparted/fs/Makefile.in	2008-05-02 16:39:24.000000000 +0200
@@ -87,8 +87,8 @@
 am__DEPENDENCIES_1 =
 libfs_la_DEPENDENCIES = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	amiga/libamigafs.la ext2/libext2.la ufs/libufs.la \
-	fat/libfat.la ntfs/libntfs.la hfs/libhfs.la \
+	ext2/libext2.la ufs/libufs.la \
+	fat/libfat.la ntfs/libntfs.la \
 	linux_swap/liblinuxswap.la xfs/libxfs.la jfs/libjfs.la \
 	reiserfs/libreiserfs.la
 am_libfs_la_OBJECTS =
@@ -390,7 +390,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = amiga ext2 ufs fat ntfs hfs linux_swap xfs jfs reiserfs # bfs
+SUBDIRS = ext2 ufs fat ntfs linux_swap xfs jfs reiserfs # bfs
 partedincludedir = -I$(top_srcdir)/include
 noinst_LTLIBRARIES = libfs.la
 libfs_la_LDFLAGS = -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
@@ -400,12 +400,10 @@
 		    $(INTLLIBS)			\
 		    $(OS_LIBS)			\
 		    $(REISER_LIBS)		\
-		    amiga/libamigafs.la		\
 		    ext2/libext2.la		\
 		    ufs/libufs.la		\
 		    fat/libfat.la		\
 		    ntfs/libntfs.la		\
-		    hfs/libhfs.la		\
 		    linux_swap/liblinuxswap.la	\
 		    xfs/libxfs.la		\
 		    jfs/libjfs.la		\
diff -Naur parted-1.8.8.orig/libparted/libparted.c parted-1.8.8/libparted/libparted.c
--- parted-1.8.8.orig/libparted/libparted.c	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/libparted/libparted.c	2008-05-02 16:46:49.000000000 +0200
@@ -116,28 +116,24 @@
 }
 
 #ifdef ENABLE_FS
-extern void ped_file_system_amiga_init (void);
 extern void ped_file_system_xfs_init (void);
 extern void ped_file_system_ufs_init (void);
 extern void ped_file_system_reiserfs_init (void);
 extern void ped_file_system_ntfs_init (void);
 extern void ped_file_system_linux_swap_init (void);
 extern void ped_file_system_jfs_init (void);
-extern void ped_file_system_hfs_init (void);
 extern void ped_file_system_fat_init (void);
 extern void ped_file_system_ext2_init (void);
 
 static void
 init_file_system_types ()
 {
-	ped_file_system_amiga_init ();
 	ped_file_system_xfs_init ();
 	ped_file_system_ufs_init ();
 	ped_file_system_reiserfs_init ();
 	ped_file_system_ntfs_init ();
 	ped_file_system_linux_swap_init ();
 	ped_file_system_jfs_init ();
-	ped_file_system_hfs_init ();
 	ped_file_system_fat_init ();
 	ped_file_system_ext2_init ();
 }
@@ -207,28 +203,24 @@
 #ifdef ENABLE_FS
 extern void ped_file_system_ext2_done (void);
 extern void ped_file_system_fat_done (void);
-extern void ped_file_system_hfs_done (void);
 extern void ped_file_system_jfs_done (void);
 extern void ped_file_system_linux_swap_done (void);
 extern void ped_file_system_ntfs_done (void);
 extern void ped_file_system_reiserfs_done (void);
 extern void ped_file_system_ufs_done (void);
 extern void ped_file_system_xfs_done (void);
-extern void ped_file_system_amiga_done (void);
 
 static void
 done_file_system_types ()
 {
 	ped_file_system_ext2_done ();
 	ped_file_system_fat_done ();
-	ped_file_system_hfs_done ();
 	ped_file_system_jfs_done ();
 	ped_file_system_linux_swap_done ();
 	ped_file_system_ntfs_done ();
 	ped_file_system_reiserfs_done ();
 	ped_file_system_ufs_done ();
 	ped_file_system_xfs_done ();
-	ped_file_system_amiga_done ();
 }
 #endif /* ENABLE_FS */
 
