diff -Naur parted-1.8.8.orig/libparted/labels/Makefile.am parted-1.8.8/libparted/labels/Makefile.am
--- parted-1.8.8.orig/libparted/labels/Makefile.am	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/libparted/labels/Makefile.am	2008-05-02 16:51:25.000000000 +0200
@@ -13,18 +13,10 @@
 noinst_LTLIBRARIES    =	liblabels.la
 
 liblabels_la_SOURCES  = rdb.c		\
-			bsd.c  		\
 			$(S390_SRCS)	\
 			efi_crc32.c	\
 			dos.c  		\
-			dvh.h		\
-			dvh.c  		\
-			gpt.c  		\
-			loop.c 		\
-			mac.c  		\
-			pc98.c 		\
-			sun.c		\
-			aix.c
+			gpt.c  		
 
 liblabels_la_LIBADD = $(OS_LIBS) $(INTLLIBS)
 
diff -Naur parted-1.8.8.orig/libparted/labels/Makefile.in parted-1.8.8/libparted/labels/Makefile.in
--- parted-1.8.8.orig/libparted/labels/Makefile.in	2008-04-24 21:03:21.000000000 +0200
+++ parted-1.8.8/libparted/labels/Makefile.in	2008-05-02 16:52:41.000000000 +0200
@@ -87,12 +87,10 @@
 am__DEPENDENCIES_1 =
 liblabels_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
-am__liblabels_la_SOURCES_DIST = rdb.c bsd.c dasd.c fdasd.c vtoc.c \
-	efi_crc32.c dos.c dvh.h dvh.c gpt.c loop.c mac.c pc98.c sun.c \
-	aix.c
+am__liblabels_la_SOURCES_DIST = rdb.c efi_crc32.c dos.c gpt.c
 @COMPILE_FOR_S390_TRUE@am__objects_1 = dasd.lo fdasd.lo vtoc.lo
-am_liblabels_la_OBJECTS = rdb.lo bsd.lo $(am__objects_1) efi_crc32.lo \
-	dos.lo dvh.lo gpt.lo loop.lo mac.lo pc98.lo sun.lo aix.lo
+am_liblabels_la_OBJECTS = rdb.lo $(am__objects_1) efi_crc32.lo \
+	dos.lo gpt.lo
 liblabels_la_OBJECTS = $(am_liblabels_la_OBJECTS)
 DEFAULT_INCLUDES = -I. -I$(top_builddir)/lib@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/build-aux/depcomp
@@ -385,18 +383,10 @@
 partedincludedir = -I$(top_srcdir)/include
 noinst_LTLIBRARIES = liblabels.la
 liblabels_la_SOURCES = rdb.c		\
-			bsd.c  		\
 			$(S390_SRCS)	\
 			efi_crc32.c	\
 			dos.c  		\
-			dvh.h		\
-			dvh.c  		\
-			gpt.c  		\
-			loop.c 		\
-			mac.c  		\
-			pc98.c 		\
-			sun.c		\
-			aix.c
+			gpt.c  		
 
 liblabels_la_LIBADD = $(OS_LIBS) $(INTLLIBS)
 INCLUDES = $(partedincludedir) $(INTLINCS)
diff -Naur parted-1.8.8.orig/libparted/libparted.c parted-1.8.8/libparted/libparted.c
--- parted-1.8.8.orig/libparted/libparted.c	2008-05-02 16:54:06.000000000 +0200
+++ parted-1.8.8/libparted/libparted.c	2008-05-02 16:50:51.000000000 +0200
@@ -81,38 +81,14 @@
 	return 1;
 }
 
-extern void ped_disk_aix_init ();
-extern void ped_disk_bsd_init ();
-extern void ped_disk_dvh_init ();
 extern void ped_disk_gpt_init ();
-extern void ped_disk_loop_init ();
-extern void ped_disk_mac_init ();
 extern void ped_disk_msdos_init ();
-extern void ped_disk_pc98_init ();
-extern void ped_disk_sun_init ();
-extern void ped_disk_amiga_init ();
-extern void ped_disk_dasd_init ();
 
 static void
 init_disk_types ()
 {
-	ped_disk_loop_init ();	/* must be last in the probe list */
-
-#if defined(__s390__) || defined(__s390x__)
-	ped_disk_dasd_init();
-#endif
-
-	ped_disk_sun_init ();
-#ifdef ENABLE_PC98
-	ped_disk_pc98_init ();
-#endif
 	ped_disk_msdos_init ();
-	ped_disk_mac_init ();
 	ped_disk_gpt_init ();
-	ped_disk_dvh_init ();
-	ped_disk_bsd_init ();
-	ped_disk_amiga_init ();
-	ped_disk_aix_init ();
 }
 
 #ifdef ENABLE_FS
@@ -139,36 +115,14 @@
 }
 #endif /* ENABLE_FS */
 
-extern void ped_disk_aix_done ();
-extern void ped_disk_bsd_done ();
-extern void ped_disk_dvh_done ();
 extern void ped_disk_gpt_done ();
-extern void ped_disk_loop_done ();
-extern void ped_disk_mac_done ();
 extern void ped_disk_msdos_done ();
-extern void ped_disk_pc98_done ();
-extern void ped_disk_sun_done ();
-extern void ped_disk_amiga_done ();
-extern void ped_disk_dasd_done ();
 
 static void
 done_disk_types ()
 {
-#if defined(__s390__) || (__s390x__)
-	ped_disk_dasd_done ();
-#endif
-	ped_disk_sun_done ();
-#ifdef ENABLE_PC98
-	ped_disk_pc98_done ();
-#endif
 	ped_disk_msdos_done ();
-	ped_disk_mac_done ();
-	ped_disk_loop_done ();
 	ped_disk_gpt_done ();
-	ped_disk_dvh_done ();
-	ped_disk_bsd_done ();
-	ped_disk_amiga_done ();
-	ped_disk_aix_done ();
 }
 
 static void _init() __attribute__ ((constructor));
