--- loadepg-0.2.5/loadepg.c	2013-01-21 18:55:00.000000000 +0100
+++ loadepg-0.2.5/loadepg.c_new	2013-04-15 16:57:49.247329500 +0200
@@ -2026,7 +2026,7 @@
           {
             SatelliteTimeOffset = SatelliteTimeOffsetH * 3600;
           }
-          EpgTimeOffset = (LocalTimeOffset - SatelliteTimeOffset);
+          EpgTimeOffset =  0; //(LocalTimeOffset - SatelliteTimeOffset);
           esyslog("LoadEPG: Satellite Time Offset=[UTC]%+i", SatelliteTimeOffset / 3600);
           esyslog("LoadEPG: Epg Time Offset=%+i seconds", EpgTimeOffset);
           if(DEBUG)
