diff -Naur u-boot-omap3-git-20100429.orig/common/cmd_i2c.c u-boot-omap3-git-20100429/common/cmd_i2c.c
--- u-boot-omap3-git-20100429.orig/common/cmd_i2c.c	2010-04-29 01:08:02.268811228 +0200
+++ u-boot-omap3-git-20100429/common/cmd_i2c.c	2010-04-29 01:08:49.399546000 +0200
@@ -1243,10 +1243,7 @@
 {
 	int bus_idx, ret=0;
 
-	if (argc == 1)
-		/* querying current setting */
-		printf("Current bus is %d\n", i2c_get_bus_num());
-	else {
+	if (argc != 1) {
 		bus_idx = simple_strtoul(argv[1], NULL, 10);
 		printf("Setting bus to %d\n", bus_idx);
 		ret = i2c_set_bus_num(bus_idx);
