--- A/Makefile.orig	2013-03-18 12:51:07.152288866 +0200
+++ B/Makefile	2013-03-18 12:51:48.356493187 +0200
@@ -108,7 +108,7 @@
 DEBUG_FLAGS := $(if ${DEBUG},-g -O0,-Os)
 COMMON_FLAGS := -nostdinc -pipe \
 	-ffreestanding -fno-builtin -fno-stack-protector \
-	-Werror -Wall -Wstrict-prototypes ${DEBUG_FLAGS}
+	-Wall -Wstrict-prototypes ${DEBUG_FLAGS}
 
 # Note: FIRMWARE_ARCH is defined by the Chromium OS ebuild.
 ifeq (${FIRMWARE_ARCH}, arm)
@@ -131,7 +131,7 @@
 else
 # FIRMWARE_ARCH not defined; assuming local compile.
 CC ?= gcc
-CFLAGS += -DCHROMEOS_ENVIRONMENT -Wall -Werror # HEY: always want last two?
+CFLAGS += -DCHROMEOS_ENVIRONMENT -Wall # HEY: always want last two?
 endif
 
 ifneq (${OLDDIR},)
