--- xf86-video-nvidia-96xx-96.43.23/usr/src/nv/nv.c.org	2015-02-28 23:15:39.521431012 +0100
+++ xf86-video-nvidia-96xx-96.43.23/usr/src/nv/nv.c	2015-02-28 23:16:17.609431547 +0100
@@ -2299,7 +2299,7 @@
 
     nvl = NVL_FROM_FILEP(file);
 
-    if (NV_IS_CONTROL_DEVICE(file->f_dentry->d_inode))
+    if (NV_IS_CONTROL_DEVICE(file->f_path.dentry->d_inode))
         return nv_kern_ctl_poll(file, wait);
 
     nvfp = NV_GET_NVFP(file);
@@ -2494,7 +2494,7 @@
     unsigned long i_arg
 )
 {
-    return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
 }
 
 long nv_kern_compat_ioctl(
@@ -2503,7 +2503,7 @@
     unsigned long i_arg
 )
 {
-    return nv_kern_ioctl(file->f_dentry->d_inode, file, cmd, i_arg);
+    return nv_kern_ioctl(file->f_path.dentry->d_inode, file, cmd, i_arg);
 }
 
 /*
