--- vdr-karaoke-0.0.4/data.c.org	2013-12-29 20:02:34.769417734 +0100
+++ vdr-karaoke-0.0.4/data.c	2013-12-29 20:02:50.873417257 +0100
@@ -479,7 +479,7 @@
     {
         do
         {
-            char *s = index( Include, '/' );
+            const char *s = index( Include, '/' );
             int l = s ? s - Include : strlen( Include );
             if( l )
             {
--- vdr-karaoke-0.0.4/decoder.c.org	2013-12-29 20:03:04.601416844 +0100
+++ vdr-karaoke-0.0.4/decoder.c	2013-12-29 20:03:36.681415891 +0100
@@ -109,7 +109,7 @@
     // if no title, try to build a reasonable from the filename
     if( !Title && filename )
     {
-        char *s = rindex( filename, '/' );
+        const char *s = rindex( filename, '/' );
         if( s && *s == '/' )
         {
             s++;
@@ -125,7 +125,7 @@
             {                   // strip any extention
                 s = rindex( Title, '.' );
                 if( s && *s == '.' && strlen( s ) <= 5 )
-                    *s = 0;
+                    s = 0;
             }
         d( printf( "[karaoke] karaoke: faking title '%s' from filename '%s'\n", Title, filename ) )}
     }
