--- vdr-karaoke-0.0.4/decoder.c.org	2013-12-29 20:28:39.941371090 +0100
+++ vdr-karaoke-0.0.4/decoder.c	2013-12-29 20:31:12.553366538 +0100
@@ -644,7 +644,12 @@
 
             char *cInfoCache::CacheFile( void )
             {
+#if APIVERSNUM > 20101
+		return AddPath( cachedir ? cachedir : cVideoDirectory::Name(), CACHEFILENAME );
+#else
+
                 return AddPath( cachedir ? cachedir : VideoDirectory, CACHEFILENAME );
+#endif
             }
 
             void cInfoCache::Save( bool force )
