diff -Naur yaboot-1.3.13.orig/second/yaboot.c yaboot-1.3.13/second/yaboot.c
--- yaboot-1.3.13.orig/second/yaboot.c	2005-03-24 01:24:15.000000000 +0100
+++ yaboot-1.3.13/second/yaboot.c	2005-03-24 01:34:05.000000000 +0100
@@ -344,8 +344,10 @@
 	  strcpy(conf_path, "/etc/");
      else if (path && *path)
 	  strcpy(conf_path, path);
+     else if (strstr(device, "ethernet") || strstr(device, "enet"))
+	  strcpy(conf_path, "GEEXBOX.powerpc/boot/netboot/");
      else
-	  conf_path[0] = 0;
+	  strcpy(conf_path, "/GEEXBOX/boot/");
      strcat(conf_path, CONFIG_FILE_NAME);
 
      /* Open it */
