From a3c6b3e5f3dd926aefecc3e7fe3d0923cdcde6cd Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Sat, 6 Feb 2010 22:10:16 -0600
Subject: [PATCH 19/26] modify playbin to use stridetransform

---
 gst/playback/gstplaysink.c |   26 ++++----------------------
 1 files changed, 4 insertions(+), 22 deletions(-)

diff --git a/gst/playback/gstplaysink.c b/gst/playback/gstplaysink.c
index a8c9b33..969a785 100644
--- a/gst/playback/gstplaysink.c
+++ b/gst/playback/gstplaysink.c
@@ -1224,13 +1224,13 @@ gen_video_chain (GstPlaySink * playsink, gboolean raw, gboolean async,
   }
 
   if (raw && !(playsink->flags & GST_PLAY_FLAG_NATIVE_VIDEO)) {
-    GST_DEBUG_OBJECT (playsink, "creating ffmpegcolorspace");
-    chain->conv = gst_element_factory_make ("ffmpegcolorspace", "vconv");
+    GST_DEBUG_OBJECT (playsink, "creating stridetransform");
+    chain->conv = gst_element_factory_make ("stridetransform", "vconv");
     if (chain->conv == NULL) {
-      post_missing_element_message (playsink, "ffmpegcolorspace");
+      post_missing_element_message (playsink, "stridetransform");
       GST_ELEMENT_WARNING (playsink, CORE, MISSING_PLUGIN,
           (_("Missing element '%s' - check your GStreamer installation."),
-              "ffmpegcolorspace"), ("video rendering might fail"));
+              "stridetransform"), ("video rendering might fail"));
     } else {
       gst_bin_add (bin, chain->conv);
       if (prev) {
@@ -1241,24 +1241,6 @@ gen_video_chain (GstPlaySink * playsink, gboolean raw, gboolean async,
       }
       prev = chain->conv;
     }
-
-    GST_DEBUG_OBJECT (playsink, "creating videoscale");
-    chain->scale = gst_element_factory_make ("videoscale", "vscale");
-    if (chain->scale == NULL) {
-      post_missing_element_message (playsink, "videoscale");
-      GST_ELEMENT_WARNING (playsink, CORE, MISSING_PLUGIN,
-          (_("Missing element '%s' - check your GStreamer installation."),
-              "videoscale"), ("possibly a liboil version mismatch?"));
-    } else {
-      gst_bin_add (bin, chain->scale);
-      if (prev) {
-        if (!gst_element_link_pads (prev, "src", chain->scale, "sink"))
-          goto link_failed;
-      } else {
-        head = chain->scale;
-      }
-      prev = chain->scale;
-    }
   }
 
   if (prev) {
-- 
1.7.1

