From 35575b7c1df6162120e858732d068594344086f4 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 13 Sep 2010 19:05:56 -0500
Subject: [PATCH 23/26] video: fix endianess issue for 16bit RGB formats

---
 gst-libs/gst/video/video.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/gst-libs/gst/video/video.c b/gst-libs/gst/video/video.c
index d40efa7..312cb1a 100644
--- a/gst-libs/gst/video/video.c
+++ b/gst-libs/gst/video/video.c
@@ -621,6 +621,7 @@ gst_video_format_new_caps_simple (GstVideoFormat format, int rowstride,
     int depth;
     int bpp;
     gboolean have_alpha;
+    int endianness = G_BIG_ENDIAN;
     unsigned int mask = 0;
 
     switch (format) {
@@ -699,6 +700,7 @@ gst_video_format_new_caps_simple (GstVideoFormat format, int rowstride,
         default:
           return NULL;
       }
+      endianness = G_BYTE_ORDER;
     } else {
       return NULL;
     }
@@ -707,7 +709,7 @@ gst_video_format_new_caps_simple (GstVideoFormat format, int rowstride,
         "video/x-raw-rgb-strided" : "video/x-raw-rgb",
         "bpp", G_TYPE_INT, bpp,
         "depth", G_TYPE_INT, depth,
-        "endianness", G_TYPE_INT, G_BIG_ENDIAN,
+        "endianness", G_TYPE_INT, endianness,
         "red_mask", G_TYPE_INT, red_mask,
         "green_mask", G_TYPE_INT, green_mask,
         "blue_mask", G_TYPE_INT, blue_mask,
-- 
1.7.1

