From 10c1010afad545008508d43470d91158528f3667 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Wed, 6 Jul 2011 19:02:21 +0200
Subject: [PATCH 29/32] encodebin: set GST_ENC_FLAG_NATIVE_VIDEO on omap

---
 gst/encoding/gstencodebin.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst/encoding/gstencodebin.c b/gst/encoding/gstencodebin.c
index 8d9e90f..ee5cced 100644
--- a/gst/encoding/gstencodebin.c
+++ b/gst/encoding/gstencodebin.c
@@ -225,7 +225,7 @@ struct _StreamGroup
 #define DEFAULT_QUEUE_TIME_MAX     GST_SECOND
 #define DEFAULT_AUDIO_JITTER_TOLERANCE 20 * GST_MSECOND
 #define DEFAULT_AVOID_REENCODING   FALSE
-#define DEFAULT_FLAGS              0
+#define DEFAULT_FLAGS              GST_ENC_FLAG_NATIVE_VIDEO
 
 #define DEFAULT_RAW_CAPS			\
   "video/x-raw-yuv; "				\
@@ -1223,7 +1223,7 @@ _create_stream_group (GstEncodeBin * ebin, GstEncodingProfile * sprof,
   /* 3.2. restriction elements */
   /* FIXME : Once we have properties for specific converters, use those */
   if (GST_IS_ENCODING_VIDEO_PROFILE (sprof)) {
-    const gboolean native_video = ! !(ebin->flags & GST_ENC_FLAG_NATIVE_VIDEO);
+    const gboolean native_video = !!(ebin->flags & GST_ENC_FLAG_NATIVE_VIDEO);
     GstElement *cspace = NULL, *scale, *vrate, *cspace2 = NULL;
 
     GST_LOG ("Adding conversion elements for video stream");
-- 
1.7.1

