diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -751,7 +751,6 @@
   check_lib xine.h xine_init -lxine
   if [ "$?" = 0 ]; then
     add_extracflags -DHAVE_XINE
-    add_pkgconfig_requires libxine
     wrapper_xine="yes"
   fi
 fi
@@ -804,7 +802,6 @@
   check_lib gst/gst.h gst_init -lgstreamer-0.10
   if [ "$?" = 0 ]; then
     add_extracflags -DHAVE_GSTREAMER
-    add_pkgconfig_requires gstreamer-0.10
     wrapper_gstreamer="yes"
   fi
 fi
@@ -823,7 +820,6 @@
     check_lib X11/X.h XInitThreads -lX11
     if [ "$?" = 0 ]; then
       add_extracflags -DUSE_X11
-      add_pkgconfig_requires x11
       x11="yes"
     fi
   fi
