diff -Naur libplayer-1.0.0/src/wrapper_mplayer.c libplayer-1.0.0a/src/wrapper_mplayer.c
--- libplayer-1.0.0/src/wrapper_mplayer.c	2009-12-06 15:27:55.000000000 +0100
+++ libplayer-1.0.0a/src/wrapper_mplayer.c	2010-02-02 10:05:18.000000000 +0100
@@ -2869,7 +2869,7 @@
       params[pp++] = "-vo";
       params[pp++] = "vdpau,xv,x11";
       params[pp++] = "-vc";
-      params[pp++] = "ffh264vdpau,ffmpeg12vdpau,ffvc1vdpau,ffwmv3vdpau,";
+      params[pp++] = "ffh264vdpau,ffodivxvdpau,ffmpeg12vdpau,ffvc1vdpau,ffwmv3vdpau,";
       break;
 
     case PLAYER_VO_AUTO:
