diff -Naur libvpx-0.9.0+git20100616.orig/build/make/configure.sh libvpx-0.9.0+git20100616/build/make/configure.sh
--- libvpx-0.9.0+git20100616.orig/build/make/configure.sh	2010-06-16 22:10:53.788439276 +0200
+++ libvpx-0.9.0+git20100616/build/make/configure.sh	2010-06-16 22:13:25.944687636 +0200
@@ -799,8 +799,8 @@
                 add_ldflags -i-static
                 ;;
             gcc*)
-                add_cflags  -m${bits}
-                add_ldflags -m${bits}
+                add_cflags
+                add_ldflags
                 link_with_cc=gcc
             setup_gnu_toolchain
                 ;;
