diff -Naur libvpx-0.9.0+git20100616.orig/vp8/decoder/dboolhuff.c libvpx-0.9.0+git20100616/vp8/decoder/dboolhuff.c
--- libvpx-0.9.0+git20100616.orig/vp8/decoder/dboolhuff.c	2010-06-16 22:10:53.768439270 +0200
+++ libvpx-0.9.0+git20100616/vp8/decoder/dboolhuff.c	2010-06-16 22:47:25.234690691 +0200
@@ -108,7 +108,6 @@
     bc->decode_buffer = 0;
 }
 
-#if 0
 /*
  * Until optimized versions of these functions are available, we
  * keep the implementation in the header to allow inlining.
@@ -172,4 +171,3 @@
     }
     return z;
 }
-#endif
