Description: Fixes problems with text2skin skin enigma.
Author: Andreas Brugger <brougs78@gmx.net>
Origin: http://vdrportal.de/board/thread.php?postid=343665#post343665

--- a/osdbase.c
+++ b/osdbase.c
@@ -325,6 +325,7 @@
 
 void cOsdMenu::CursorUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -363,6 +364,7 @@
 
 void cOsdMenu::CursorDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -403,6 +405,7 @@
 
 void cOsdMenu::PageUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current -= displayMenuItems;
@@ -437,6 +440,7 @@
 
 void cOsdMenu::PageDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current += displayMenuItems;
