Description: Use ncursesw header file
Author: Tobias Grimm <tg@e-tobi.net>

Index: vdr-1.7.28/PLUGINS/src/skincurses/skincurses.c
===================================================================
--- vdr-1.7.28.orig/PLUGINS/src/skincurses/skincurses.c	2012-04-23 10:53:13.000000000 +0200
+++ vdr-1.7.28/PLUGINS/src/skincurses/skincurses.c	2012-06-03 17:10:38.519576571 +0200
@@ -6,7 +6,7 @@
  * $Id: skincurses.c 2.9 2012/04/23 08:53:13 kls Exp $
  */
 
-#include <ncurses.h>
+#include <ncursesw/ncurses.h>
 #include <vdr/osd.h>
 #include <vdr/plugin.h>
 #include <vdr/skins.h>
