--- a/config.c
+++ b/config.c
@@ -475,6 +475,8 @@
   ChannelsWrap = 0;
   ShowChannelNamesWithSource = 0;
   EmergencyExit = 1;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
 }
 
 cSetup& cSetup::operator= (const cSetup &s)
@@ -682,6 +684,8 @@
   else if (!strcasecmp(Name, "ChannelsWrap"))        ChannelsWrap       = atoi(Value);
   else if (!strcasecmp(Name, "ShowChannelNamesWithSource")) ShowChannelNamesWithSource = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
   else
      return false;
   return true;
@@ -792,6 +796,8 @@
   Store("ChannelsWrap",       ChannelsWrap);
   Store("ShowChannelNamesWithSource", ShowChannelNamesWithSource);
   Store("EmergencyExit",      EmergencyExit);
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
 
   Sort();
 
--- a/config.h
+++ b/config.h
@@ -342,6 +342,8 @@
   int ChannelsWrap;
   int ShowChannelNamesWithSource;
   int EmergencyExit;
+  int JumpSeconds;
+  int JumpSecondsSlow;
   int __EndData__;
   cString InitialChannel;
   cString DeviceBondings;
--- a/menu.c
+++ b/menu.c
@@ -3523,6 +3523,8 @@
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Jump&Play"), &data.JumpPlay));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Play&Jump"), &data.PlayJump));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause at last mark"), &data.PauseLastMark));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds Slow"), &data.JumpSecondsSlow));
 }
 
 void cMenuSetupReplay::Store(void)
@@ -5485,9 +5487,9 @@
     case kRight:   Forward(); break;
     case kRed:     TimeSearch(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+    case kGreen:   SkipSeconds(-Setup.JumpSeconds); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
     case kPrev|k_Repeat:
     case kPrev:    if (lastSkipTimeout.TimedOut()) {
                       lastSkipSeconds = REPLAYCONTROLSKIPSECONDS;
@@ -5534,7 +5536,7 @@
                                   marks.Save();
                                   }
                                else
-                                  SkipSeconds(-20);
+                                  SkipSeconds(-Setup.JumpSecondsSlow);
                                break;
         case k3|k_Repeat:
         case k3:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
@@ -5547,7 +5549,7 @@
                                   marks.Save();
                                   }
                                else
-                                  SkipSeconds(20);
+                                  SkipSeconds(Setup.JumpSecondsSlow);
                                break;
         case kMarkToggle:      if (GetReplayMode(play, forward, speed) && play) {
                                   displayFrames = true;
