diff -Naur u-boot-2009.11-tegra-0.9.108.0/board/tegra2/harmony/config.mk u-boot-2009.11-tegra-0.9.108.0a/board/tegra2/harmony/config.mk
--- u-boot-2009.11-tegra-0.9.108.0/board/tegra2/harmony/config.mk	2010-11-01 13:49:25.000000000 +0100
+++ u-boot-2009.11-tegra-0.9.108.0a/board/tegra2/harmony/config.mk	2010-11-06 11:24:40.716990850 +0100
@@ -21,4 +21,4 @@
 #  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
 #  MA 02111-1307 USA
 # 
-TEXT_BASE = 0x00e08000
+TEXT_BASE =  0x00108000
