diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2012-06-03 20:17:01.899662209 -0700
+++ b/configure.ac	2012-06-03 20:17:26.479596382 -0700
@@ -104,7 +104,7 @@
 ])
 
 MAYBE_WARN="-Wall -Wextra \
--Wsign-compare -Werror-implicit-function-declaration \
+-Wsign-compare \
 -Wpointer-arith -Wwrite-strings -Wstrict-prototypes \
 -Wnested-externs \
 -Wpacked -Wswitch-enum -Wmissing-format-attribute \
