diff -Naur a/util/demux.c b/util/demux.c
--- a/util/demux.c	2012-06-05 00:31:35.000000000 -0700
+++ b/util/demux.c	2012-06-19 22:01:05.161041859 -0700
@@ -50,7 +50,7 @@
 open_file(const char *filename)
 {
 	AVFormatContext *afc;
-	int err = av_open_input_file(&afc, filename, NULL, 0, NULL);
+	int err = avformat_open_input(&afc, filename, NULL, NULL);
 
 	if (!err)
 		err = av_find_stream_info(afc);
@@ -60,7 +60,7 @@
 		exit(1);
 	}
 
-	dump_format(afc, 0, filename, 0);
+	av_dump_format(afc, 0, filename, 0);
 
 	return afc;
 }
