--- vdr-2.0.6/dvbdevice.c.orig	2014-03-22 15:44:41.795144172 +0100
+++ vdr-2.0.6/dvbdevice.c	2014-03-22 15:42:42.339147069 +0100
@@ -565,6 +560,8 @@
     case 0x13C21019: // TT-budget S2-3200 (DVB-S/DVB-S2)
     case 0x1AE40001: // TechniSat SkyStar HD2 (DVB-S/DVB-S2)
                      MaxSignal = 670; break;
+    case 0x13D02103: // TechniSat SkyStar 2 DVB-S rev 2.3P
+                     MaxSignal = 0x5500; break;
     }
   int s = int(Signal) * 100 / MaxSignal;
   if (s > 100)

