--- xine-lib-1.2-r11592/src/vdr/Makefile.in	2010-12-23 17:53:01.000000000 +0100
+++ xine-lib-1.2-r11592a/src/vdr/Makefile.in	2010-12-23 18:09:56.978834564 +0100
@@ -440,7 +440,7 @@
 AM_LDFLAGS = $(xineplug_ldflags)
 @ENABLE_VDR_TRUE@xineplug_LTLIBRARIES = xineplug_vdr.la
 xineplug_vdr_la_SOURCES = combined_vdr.c combined_vdr.h input_vdr.c post_vdr_video.c post_vdr_audio.c
-xineplug_vdr_la_LIBADD = $(XINE_LIB) $(PTHREAD_LIBS)
+xineplug_vdr_la_LIBADD = $(XINE_LIB) $(PTHREAD_LIBS) -ljpeg
 all: all-am
 
 .SUFFIXES:
