diff -Naur xine-lib-1.2-r11592/src/video_out/video_out_vdpau.c xine-lib-1.2-r11592a/src/video_out/video_out_vdpau.c
--- xine-lib-1.2-r11592/src/video_out/video_out_vdpau.c	2011-01-04 11:51:59.360396514 +0100
+++ xine-lib-1.2-r11592a/src/video_out/video_out_vdpau.c	2011-01-04 11:52:19.175904408 +0100
@@ -61,7 +61,6 @@
 #define NUM_FRAMES_BACK 1
 
 #define LOCKDISPLAY /*define this if you have a buggy libX11/xcb*/
-#undef LOCKDISPLAY 
 
 
 #define DEINT_BOB                    1
