--- vdr-trayopenng-0.5.2/trayopenng.cc.org	2012-10-03 23:59:06.953413396 +0200
+++ vdr-trayopenng-0.5.2/trayopenng.cc	2012-10-03 23:59:50.577414145 +0200
@@ -16,8 +16,8 @@
 
 cPluginTrayopenng::cPluginTrayopenng(void)
 {
-    strcpy(mDevice, "/dev/dvd");
-    strcpy(mCommand, "eject -T");
+    strcpy(mDevice, "/dev/sr0");
+    strcpy(mCommand, "eject -s");
 }
 
 const char *cPluginTrayopenng::CommandLineHelp(void)
