--- loadepg-0.2.3/loadepg.h	2011-11-15 20:45:34.609894174 +0100
+++ loadepg-0.2.3/loadepg.h_equiv	2011-11-15 20:49:49.662232800 +0100
@@ -69,16 +69,16 @@
 
 typedef struct
 {
-  unsigned short int OriginalSourceId;
-  unsigned short int OriginalNid;
-  unsigned short int OriginalTid;
-  unsigned short int OriginalSid;
-  unsigned short int OriginalRid;
-  unsigned short int EquivSourceId;
-  unsigned short int EquivNid;
-  unsigned short int EquivTid;
-  unsigned short int EquivSid;
-  unsigned short int EquivRid;
+  unsigned int OriginalSourceId;
+  unsigned int OriginalNid;
+  unsigned int OriginalTid;
+  unsigned int OriginalSid;
+  unsigned int OriginalRid;
+  unsigned int EquivSourceId;
+  unsigned int EquivNid;
+  unsigned int EquivTid;
+  unsigned int EquivSid;
+  unsigned int EquivRid;
 } sEquivChannel;
 
 typedef struct

