--- a/tools/Makefile.am	2010-01-06 15:57:57.000000000 +0100
+++ b/tools/Makefile.am	2012-02-24 11:48:32.736265704 +0100
@@ -3,9 +3,10 @@
 bin_PROGRAMS = sip_reg
 endif
 
-AM_CFLAGS = @CFLAGS@ @PTHREAD_CFLAGS@ @EXOSIP_FLAGS@
+AM_CFLAGS = @CFLAGS@ `pkg-config --cflags pthread-stubs` @EXOSIP_FLAGS@
 
 sip_reg_SOURCES = sip_reg.c
-sip_reg_LDADD = $(top_builddir)/src/libeXosip2.la @TOOLS_LIBS@ $(OSIP_LIBS) $(EXOSIP_LIB) $(PTHREAD_LIBS)
+sip_reg_LDFLAGS = `pkg-config --libs pthread-stubs`
+sip_reg_LDADD = $(top_builddir)/src/libeXosip2.la @TOOLS_LIBS@ $(OSIP_LIBS) $(EXOSIP_LIB)
 
 INCLUDES = -I$(top_srcdir) -I$(top_srcdir)/include $(OSIP_CFLAGS)
