* proposed upstream

--- ivtv-0.9.1.orig/driver/ivtv-driver.c	2007-01-24 11:43:36.000000000 +0100
+++ ivtv-0.9.1/driver/ivtv-driver.c	2007-01-24 11:45:34.000000000 +0100
@@ -696,7 +696,11 @@
 		IVTV_ERR("Could not create VBI workqueue\n");
 		return -1;
 	}
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,20)
+	INIT_WORK(&itv->vbi_work_queue, vbi_work_handler);
+#else
 	INIT_WORK(&itv->vbi_work_queue, vbi_work_handler, itv);
+#endif
 #else
 	INIT_TQUEUE(&itv->vbi_work_queue, vbi_work_handler, itv);
 #endif
