--- kconfig-frontends-3.3.0-0/frontends/gconf/Makefile.in.orig	2012-03-20 23:34:58.408091001 +0200
+++ kconfig-frontends-3.3.0-0/frontends/gconf/Makefile.in	2012-03-20 23:36:56.116091003 +0200
@@ -251,7 +251,7 @@
                  $(GETTEXT)             \
                  -I../../libs/parser    \
                  -I../../libs/images    \
-                 -DGUI_PATH='"$(pkgdatadir)/gconf.glade"'
+                 -DGUI_PATH='"$(abs_srcdir)/gconf.glade"'
 
 gconf_CFLAGS = $(AM_CFLAGS) \
                $(gtk_CFLAGS)
