diff -Naur vdr-eepg-0.0.3/eepg.c vdr-eepg-0.0.3a/eepg.c
--- vdr-eepg-0.0.3/eepg.c	2010-09-03 17:01:36.676447228 +0200
+++ vdr-eepg-0.0.3a/eepg.c	2010-09-03 17:02:33.156176480 +0200
@@ -2684,7 +2684,7 @@
   class cEIT2:public SI::EIT
   {
   public:
-    cEIT2 (cSchedules::cSchedules * Schedules, int Source, u_char Tid, const u_char * Data,
+    cEIT2 (cSchedules * Schedules, int Source, u_char Tid, const u_char * Data,
 	   bool OnlyRunningStatus = false);
 
 // originally from libdtv, Copyright Rolf Hakenes <hakenes@hippomi.de>
@@ -2773,7 +2773,7 @@
   }
 #endif /* NOEPG */
 
-  cEIT2::cEIT2 (cSchedules::cSchedules * Schedules, int Source, u_char Tid, const u_char * Data, bool OnlyRunningStatus)
+  cEIT2::cEIT2 (cSchedules * Schedules, int Source, u_char Tid, const u_char * Data, bool OnlyRunningStatus)
 :  SI::EIT (Data, false) {
     if (!CheckCRCAndParse ())
       return;
