diff --git linux-2.6.33.5/scripts/gen_initramfs_list.sh linux-2.6.33.5/scripts/gen_initramfs_list.sh
--- linux-2.6.33.5/scripts/gen_initramfs_list.sh
+++ linux-2.6.33.5/scripts/gen_initramfs_list.sh
@@ -105,9 +105,7 @@
 # for links, devices etc the format differs. See gen_init_cpio for details
 parse() {
 	local location="$1"
-	local name="${location/${srcdir}//}"
-	# change '//' into '/'
-	name="${name//\/\///}"
+	local name="$(echo "$location" | sed -e 's%$srcdir%%' -e 's%//*%/%g')"
 	local mode="$2"
 	local uid="$3"
 	local gid="$4"
