diff --git linux-2.6.33.5/arch/x86/boot/tty.c linux-2.6.33.5/arch/x86/boot/tty.c
--- linux-2.6.33.5/arch/x86/boot/tty.c
+++ linux-2.6.33.5/arch/x86/boot/tty.c
@@ -94,7 +94,7 @@
 
 int getchar_timeout(void)
 {
-	int cnt = 30;
+	int cnt = 3;
 	int t0, t1;
 
 	t0 = gettime();
diff --git linux-2.6.33.5/arch/x86/boot/video.c linux-2.6.33.5/arch/x86/boot/video.c
--- linux-2.6.33.5/arch/x86/boot/video.c
+++ linux-2.6.33.5/arch/x86/boot/video.c
@@ -201,7 +201,7 @@
 	unsigned int sel;
 
 	puts("Press <ENTER> to see video modes available, "
-	     "<SPACE> to continue, or wait 30 sec\n");
+	     "<SPACE> to continue, or wait 3 sec\n");
 
 	kbd_flush();
 	while (1) {
