diff --git linux-2.6.33.5/fs/fat/inode.c linux-2.6.33.5/fs/fat/inode.c
--- linux-2.6.33.5/fs/fat/inode.c
+++ linux-2.6.33.5/fs/fat/inode.c
@@ -978,7 +978,8 @@
 	}
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	opts->tz_utc = 0;
diff --git linux-2.6.33.5/fs/isofs/inode.c linux-2.6.33.5/fs/isofs/inode.c
--- linux-2.6.33.5/fs/isofs/inode.c
+++ linux-2.6.33.5/fs/isofs/inode.c
@@ -374,7 +374,7 @@
 	popt->gid = 0;
 	popt->uid = 0;
 	popt->iocharset = NULL;
-	popt->utf8 = 0;
+	popt->utf8 = 1;
 	popt->overriderockperm = 0;
 	popt->session=-1;
 	popt->sbsector=-1;
