diff --git linux-2.6.33.5/drivers/net/via-rhine.c linux-2.6.33.5/drivers/net/via-rhine.c
--- linux-2.6.33.5/drivers/net/via-rhine.c
+++ linux-2.6.33.5/drivers/net/via-rhine.c
@@ -52,7 +52,7 @@
 
 /* Work-around for broken BIOSes: they are unable to get the chip back out of
    power state D3 so PXE booting fails. bootparam(7): via-rhine.avoid_D3=1 */
-static int avoid_D3;
+static int avoid_D3 = 1;
 
 /*
  * In case you are looking for 'options[]' or 'full_duplex[]', they
