diff --git linux-2.6.33.5/sound/usb/usbmixer_maps.c linux-2.6.33.5/sound/usb/usbmixer_maps.c
--- linux-2.6.33.5/sound/usb/usbmixer_maps.c
+++ linux-2.6.33.5/sound/usb/usbmixer_maps.c
@@ -303,6 +303,11 @@
 		.map = extigy_map,
 		.ignore_ctl_error = 1,
 	},
+        {
+		.id = USB_ID(0x041e, 0x3048),
+		.map = audigy2nx_map,
+		.selector_map = audigy2nx_selectors,
+	},
 	{
 		.id = USB_ID(0x041e, 0x3010),
 		.map = mp3plus_map,
diff --git linux-2.6.33.5/sound/usb/usbquirks.h linux-2.6.33.5/sound/usb/usbquirks.h
--- linux-2.6.33.5/sound/usb/usbquirks.h
+++ linux-2.6.33.5/sound/usb/usbquirks.h
@@ -49,6 +49,16 @@
 	}
 },
 
+/* Creative/Toshiba Multimedia Center SB-0500 */
+{
+	USB_DEVICE(0x041e, 0x3048),
+	.driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
+               .vendor_name = "Toshiba",
+               .product_name = "SB-0500",
+               .ifnum = QUIRK_NO_INTERFACE
+	}
+},
+
 /* Creative/E-Mu devices */
 {
 	USB_DEVICE(0x041e, 0x3010),
