diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
@@ -183,7 +183,7 @@
 	return 0;
 }
 
-static struct twl4030_hsmmc_info mmc[] __initdata = {
+static struct omap2_hsmmc_info mmc[] __initdata = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -202,7 +202,7 @@
 
 	platform_add_devices(sdp2430_devices, ARRAY_SIZE(sdp2430_devices));
 	omap_serial_init();
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 	usb_musb_init();
 	board_smc91x_init();
 
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
@@ -326,7 +326,7 @@
 	.tblsize		= ARRAY_SIZE(sdp3430_batt_table),
 };
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		/* 8 bits (default) requires S6.3 == ON,
@@ -363,7 +363,7 @@
 	 */
 	mmc[0].gpio_cd = gpio + 0;
 	mmc[1].gpio_cd = gpio + 1;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters ... we "know" the
 	 * regulators will be set up only *after* we return.
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
@@ -364,7 +364,7 @@
 	.rep		= 1,
 };
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -413,7 +413,7 @@
 
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters */
 	cm_t35_vmmc1_supply.dev = mmc[0].dev;
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
@@ -121,7 +121,7 @@
 	.consumer_supplies      = &igep2_vmmc1_supply,
 };
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -142,7 +142,7 @@
 {
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters ... we "know" the
 	 * regulators will be set up only *after* we return.
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
@@ -359,7 +359,7 @@
 	return 0;
 }
 
-static struct twl4030_hsmmc_info mmc[] __initdata = {
+static struct omap2_hsmmc_info mmc[] __initdata = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -396,7 +396,7 @@
 	omap_serial_init();
 	usb_musb_init();
 
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 	/* link regulators to MMC adapters */
 	ldp_vmmc1_supply.dev = mmc[0].dev;
 }
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
@@ -208,7 +208,7 @@
 
 #include "sdram-micron-mt46h32m32lf-6.h"
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 8,
@@ -238,7 +238,7 @@
 	}
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters */
 	beagle_vmmc1_supply.dev = mmc[0].dev;
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
@@ -185,7 +185,7 @@
 	.consumer_supplies	= &omap3evm_vsim_supply,
 };
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -225,7 +225,7 @@
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	omap_mux_init_gpio(63, OMAP_PIN_INPUT);
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters */
 	omap3evm_vmmc1_supply.dev = mmc[0].dev;
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
@@ -192,7 +192,7 @@
 	.rep		= 1,
 };
 
-static struct twl4030_hsmmc_info omap3pandora_mmc[] = {
+static struct omap2_hsmmc_info omap3pandora_mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -231,7 +231,7 @@
 	/* gpio + {0,1} is "mmc{0,1}_cd" (input/IRQ) */
 	omap3pandora_mmc[0].gpio_cd = gpio + 0;
 	omap3pandora_mmc[1].gpio_cd = gpio + 1;
-	twl4030_mmc_init(omap3pandora_mmc);
+	omap2_hsmmc_init(omap3pandora_mmc);
 
 	/* link regulators to MMC adapters */
 	pandora_vmmc1_supply.dev = omap3pandora_mmc[0].dev;
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
@@ -220,7 +220,7 @@
 
 #include "sdram-micron-mt46h32m32lf-6.h"
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 8,
@@ -250,7 +250,7 @@
 	}
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters */
 	touchbook_vmmc1_supply.dev = mmc[0].dev;
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
@@ -272,7 +272,7 @@
 	}
 }
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] = {
 	{
 		.mmc		= 1,
 		.wires		= 4,
@@ -297,7 +297,7 @@
 static int overo_twl_gpio_setup(struct device *dev,
 		unsigned gpio, unsigned ngpio)
 {
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	overo_vmmc1_supply.dev = mmc[0].dev;
 
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-rx51-peripherals.c linux-2.6.33.5/arch/arm/mach-omap2/board-rx51-peripherals.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-rx51-peripherals.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-rx51-peripherals.c
@@ -209,7 +209,7 @@
 	.irq_line		= 1,
 };
 
-static struct twl4030_hsmmc_info mmc[] = {
+static struct omap2_hsmmc_info mmc[] __initdata = {
 	{
 		.name		= "external",
 		.mmc		= 1,
@@ -232,15 +232,18 @@
 };
 
 static struct regulator_consumer_supply rx51_vmmc1_supply = {
-	.supply			= "vmmc",
+	.supply   = "vmmc",
+	.dev_name = "mmci-omap-hs.0",
 };
 
 static struct regulator_consumer_supply rx51_vmmc2_supply = {
-	.supply			= "vmmc",
+	.supply   = "vmmc",
+	.dev_name = "mmci-omap-hs.1",
 };
 
 static struct regulator_consumer_supply rx51_vsim_supply = {
-	.supply			= "vmmc_aux",
+	.supply   = "vmmc_aux",
+	.dev_name = "mmci-omap-hs.1",
 };
 
 static struct regulator_init_data rx51_vaux1 = {
@@ -375,12 +378,6 @@
 	gpio_request(gpio + 7, "speaker_en");
 	gpio_direction_output(gpio + 7, 1);
 
-	/* set up MMC adapters, linking their regulators to them */
-	twl4030_mmc_init(mmc);
-	rx51_vmmc1_supply.dev = mmc[0].dev;
-	rx51_vmmc2_supply.dev = mmc[1].dev;
-	rx51_vsim_supply.dev = mmc[1].dev;
-
 	return 0;
 }
 
@@ -751,5 +748,6 @@
 	rx51_init_wl1251();
 	spi_register_board_info(rx51_peripherals_spi_board_info,
 				ARRAY_SIZE(rx51_peripherals_spi_board_info));
+	omap2_hsmmc_init(mmc);
 }
 
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-zoom-peripherals.c linux-2.6.33.5/arch/arm/mach-omap2/board-zoom-peripherals.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-zoom-peripherals.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-zoom-peripherals.c
@@ -150,7 +150,7 @@
 	.consumer_supplies      = &zoom_vsim_supply,
 };
 
-static struct twl4030_hsmmc_info mmc[] __initdata = {
+static struct omap2_hsmmc_info mmc[] __initdata = {
 	{
 		.name		= "external",
 		.mmc		= 1,
@@ -175,7 +175,7 @@
 {
 	/* gpio + 0 is "mmc0_cd" (input/IRQ) */
 	mmc[0].gpio_cd = gpio + 0;
-	twl4030_mmc_init(mmc);
+	omap2_hsmmc_init(mmc);
 
 	/* link regulators to MMC adapters ... we "know" the
 	 * regulators will be set up only *after* we return.
