Index: linux-2.6.16/arch/i386/boot/tools/build.c
===================================================================
--- linux-2.6.16.orig/arch/i386/boot/tools/build.c	2006-03-20 18:41:36.000000000 +0100
+++ linux-2.6.16/arch/i386/boot/tools/build.c	2006-03-20 18:41:38.000000000 +0100
@@ -29,7 +29,6 @@
 #include <stdarg.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/sysmacros.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <asm/boot.h>
@@ -41,6 +40,11 @@
 #define DEFAULT_MAJOR_ROOT 0
 #define DEFAULT_MINOR_ROOT 0
 
+#undef major
+#define major(dev) ((int)(((dev) >> 8) & 0xff))
+#undef minor
+#define minor(dev) ((int)((dev) & 0xff))
+
 /* Minimal number of setup sectors (see also bootsect.S) */
 #define SETUP_SECTS 4
 
