diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-2430sdp.c
@@ -194,6 +194,12 @@
 	{}	/* Terminator */
 };
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap_2430sdp_init(void)
 {
 	int ret;
@@ -203,7 +209,7 @@
 	platform_add_devices(sdp2430_devices, ARRAY_SIZE(sdp2430_devices));
 	omap_serial_init();
 	omap2_hsmmc_init(mmc);
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	board_smc91x_init();
 
 	/* Turn off secondary LCD backlight */
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
@@ -650,6 +650,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap_3430sdp_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
@@ -664,7 +670,7 @@
 				ARRAY_SIZE(sdp3430_spi_board_info));
 	ads7846_dev_init();
 	omap_serial_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	board_smc91x_init();
 	sdp3430_display_init();
 	enable_board_wakeup_source();
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
@@ -575,6 +575,12 @@
 	{ .reg_offset = OMAP_MUX_TERMINATOR },
 };
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init cm_t35_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CUS);
@@ -585,7 +591,7 @@
 	cm_t35_init_ethernet();
 	cm_t35_init_led();
 
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 }
 
 MACHINE_START(CM_T35, "Compulab CM-T35")
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
@@ -436,6 +436,12 @@
 	return 0;
 }
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
@@ -461,7 +467,7 @@
 	igep2_i2c_init();
 	platform_add_devices(igep2_devices, ARRAY_SIZE(igep2_devices));
 	omap_serial_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
 
 	igep2_flash_init();
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-ldp.c
@@ -383,6 +383,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap_ldp_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
@@ -394,7 +400,7 @@
 				ARRAY_SIZE(ldp_spi_board_info));
 	ads7846_dev_init();
 	omap_serial_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 
 	omap2_hsmmc_init(mmc);
 	/* link regulators to MMC adapters */
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
@@ -505,6 +505,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap3_beagle_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
@@ -519,7 +525,7 @@
 	/* REVISIT leave DVI powered down until it's needed ... */
 	gpio_direction_output(170, true);
 
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
 	omap3beagle_flash_init();
 
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
@@ -430,6 +430,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap3_evm_init(void)
 {
 	omap3_evm_get_revision();
@@ -469,7 +475,7 @@
 		omap_mux_init_gpio(135, OMAP_PIN_OUTPUT);
 		ehci_pdata.reset_gpio_port[1] = 135;
 	}
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
 	ads7846_dev_init();
 	omap3evm_init_smsc911x();
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
@@ -401,6 +401,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap3pandora_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
@@ -413,7 +419,7 @@
 	omap3pandora_ads7846_init();
 	usb_ehci_init(&ehci_pdata);
 	pandora_keys_gpio_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 
 	/* Ensure SDRC pins are mux'd for self-refresh */
 	omap_mux_init_signal("sdrc_cke0", OMAP_PIN_OUTPUT);
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
@@ -600,6 +600,12 @@
 }
 __early_param("tbr=", early_touchbook_revision);
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init omap3_touchbook_init(void)
 {
 	pm_power_off = omap3_touchbook_poweroff;
@@ -619,7 +625,7 @@
 	spi_register_board_info(omap3_ads7846_spi_board_info,
 				ARRAY_SIZE(omap3_ads7846_spi_board_info));
 	omap3_ads7846_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
 	omap3touchbook_flash_init();
 
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
@@ -413,6 +413,12 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_OTG,
+	.power			= 100,
+};
+
 static void __init overo_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
@@ -420,7 +426,7 @@
 	platform_add_devices(overo_devices, ARRAY_SIZE(overo_devices));
 	omap_serial_init();
 	overo_flash_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	usb_ehci_init(&ehci_pdata);
 	overo_ads7846_init();
 	overo_init_smsc911x();
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-rx51.c linux-2.6.33.5/arch/arm/mach-omap2/board-rx51.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-rx51.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-rx51.c
@@ -78,11 +78,17 @@
 #define board_mux	NULL
 #endif
 
+static struct omap_musb_board_data musb_board_data = {
+	.interface_type		= MUSB_INTERFACE_ULPI,
+	.mode			= MUSB_PERIPHERAL,
+	.power			= 0,
+};
+
 static void __init rx51_init(void)
 {
 	omap3_mux_init(board_mux, OMAP_PACKAGE_CBB);
 	omap_serial_init();
-	usb_musb_init();
+	usb_musb_init(&musb_board_data);
 	rx51_peripherals_init();
 
 	/* Ensure SDRC pins are mux'd for self-refresh */
diff --git linux-2.6.33.5/arch/arm/mach-omap2/usb-musb.c linux-2.6.33.5/arch/arm/mach-omap2/usb-musb.c
--- linux-2.6.33.5/arch/arm/mach-omap2/usb-musb.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/usb-musb.c
@@ -146,7 +146,7 @@
 	.resource	= musb_resources,
 };
 
-void __init usb_musb_init(void)
+void __init usb_musb_init(struct omap_musb_board_data *board_data)
 {
 	if (cpu_is_omap243x())
 		musb_resources[0].start = OMAP243X_HS_BASE;
@@ -159,6 +159,9 @@
 	 * musb_core.c have been converted to use use clkdev.
 	 */
 	musb_plat.clock = "ick";
+	musb_plat.board_data = board_data;
+	musb_plat.power = board_data->power >> 1;
+	musb_plat.mode = board_data->mode;
 
 	if (platform_device_register(&musb_device) < 0) {
 		printk(KERN_ERR "Unable to register HS-USB (MUSB) device\n");
@@ -167,7 +170,7 @@
 }
 
 #else
-void __init usb_musb_init(void)
+void __init usb_musb_init(struct omap_musb_board_data *board_data)
 {
 }
 #endif /* CONFIG_USB_MUSB_SOC */
diff --git linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
--- linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
+++ linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
@@ -3,6 +3,7 @@
 #ifndef	__ASM_ARCH_OMAP_USB_H
 #define	__ASM_ARCH_OMAP_USB_H
 
+#include <linux/usb/musb.h>
 #include <plat/board.h>
 
 #define OMAP3_HS_USB_PORTS	3
@@ -42,7 +43,15 @@
 #define UDC_BASE			OMAP2_UDC_BASE
 #define OMAP_OHCI_BASE			OMAP2_OHCI_BASE
 
-extern void usb_musb_init(void);
+struct omap_musb_board_data {
+	u8	interface_type;
+	u8	mode;
+	u8	power;
+};
+
+enum musb_interface    {MUSB_INTERFACE_ULPI, MUSB_INTERFACE_UTMI};
+
+extern void usb_musb_init(struct omap_musb_board_data *board_data);
 
 extern void usb_ehci_init(struct ehci_hcd_omap_platform_data *pdata);
 
diff --git linux-2.6.33.5/include/linux/usb/musb.h linux-2.6.33.5/include/linux/usb/musb.h
--- linux-2.6.33.5/include/linux/usb/musb.h
+++ linux-2.6.33.5/include/linux/usb/musb.h
@@ -84,6 +84,9 @@
 
 	/* MUSB configuration-specific details */
 	struct musb_hdrc_config	*config;
+
+	/* Architecture specific board data	*/
+	void		*board_data;
 };
 
 
