diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-3430sdp.c
@@ -630,7 +630,7 @@
 		OMAP_WAKEUP_EN | OMAP_PIN_INPUT_PULLUP);
 }
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-3630sdp.c linux-2.6.33.5/arch/arm/mach-omap2/board-3630sdp.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-3630sdp.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-3630sdp.c
@@ -54,7 +54,7 @@
 		OMAP_WAKEUP_EN | OMAP_PIN_INPUT_PULLUP);
 }
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-am3517evm.c linux-2.6.33.5/arch/arm/mach-omap2/board-am3517evm.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-am3517evm.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-am3517evm.c
@@ -51,7 +51,7 @@
 	omap_gpio_init();
 }
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initdata = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[2] = EHCI_HCD_OMAP_MODE_UNKNOWN,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-cm-t35.c
@@ -383,7 +383,7 @@
 	{}	/* Terminator */
 };
 
-static struct ehci_hcd_omap_platform_data ehci_pdata = {
+static struct ehci_hcd_omap_platform_data ehci_pdata __initdata = {
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[2] = EHCI_HCD_OMAP_MODE_UNKNOWN,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-igep0020.c
@@ -442,7 +442,7 @@
 	.power			= 100,
 };
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[2] = EHCI_HCD_OMAP_MODE_UNKNOWN,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3beagle.c
@@ -485,7 +485,7 @@
 	}
 }
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3evm.c
@@ -409,7 +409,7 @@
 	&omap3_evm_lcd_device,
 };
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static struct ehci_hcd_omap_platform_data ehci_pdata __initdata = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3pandora.c
@@ -381,7 +381,7 @@
 	&pandora_keys_gpio,
 };
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_UNKNOWN,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-omap3touchbook.c
@@ -566,7 +566,7 @@
 	}
 }
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
--- linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/board-overo.c
@@ -394,7 +394,7 @@
 	&overo_lcd_device,
 };
 
-static struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
+static const struct ehci_hcd_omap_platform_data ehci_pdata __initconst = {
 	.port_mode[0] = EHCI_HCD_OMAP_MODE_UNKNOWN,
 	.port_mode[1] = EHCI_HCD_OMAP_MODE_PHY,
 	.port_mode[2] = EHCI_HCD_OMAP_MODE_UNKNOWN,
diff --git linux-2.6.33.5/arch/arm/mach-omap2/usb-ehci.c linux-2.6.33.5/arch/arm/mach-omap2/usb-ehci.c
--- linux-2.6.33.5/arch/arm/mach-omap2/usb-ehci.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/usb-ehci.c
@@ -70,7 +70,7 @@
 /*
  * setup_ehci_io_mux - initialize IO pad mux for USBHOST
  */
-static void setup_ehci_io_mux(enum ehci_hcd_omap_mode *port_mode)
+static void setup_ehci_io_mux(const enum ehci_hcd_omap_mode *port_mode)
 {
 	switch (port_mode[0]) {
 	case EHCI_HCD_OMAP_MODE_PHY:
@@ -213,7 +213,7 @@
 	return;
 }
 
-void __init usb_ehci_init(struct ehci_hcd_omap_platform_data *pdata)
+void __init usb_ehci_init(const struct ehci_hcd_omap_platform_data *pdata)
 {
 	platform_device_add_data(&ehci_device, pdata, sizeof(*pdata));
 
@@ -229,7 +229,7 @@
 
 #else
 
-void __init usb_ehci_init(struct ehci_hcd_omap_platform_data *pdata)
+void __init usb_ehci_init(const struct ehci_hcd_omap_platform_data *pdata)
 
 {
 }
diff --git linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
--- linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
+++ linux-2.6.33.5/arch/arm/plat-omap/include/plat/usb.h
@@ -53,7 +53,7 @@
 
 extern void usb_musb_init(struct omap_musb_board_data *board_data);
 
-extern void usb_ehci_init(struct ehci_hcd_omap_platform_data *pdata);
+extern void usb_ehci_init(const struct ehci_hcd_omap_platform_data *pdata);
 
 #endif
 
