define PAGE_MASK for !KERNEL

diff -Nur linux-2.6.18.orig/include/asm-powerpc/page.h linux-2.6.18/include/asm-powerpc/page.h
--- linux-2.6.18.orig/include/asm-powerpc/page.h	2006-09-17 16:28:08.000000000 +0200
+++ linux-2.6.18/include/asm-powerpc/page.h	2006-09-17 16:28:01.000000000 +0200
@@ -10,7 +10,6 @@
  * 2 of the License, or (at your option) any later version.
  */
 
-#ifdef __KERNEL__
 #include <asm/asm-compat.h>
 #include <asm/kdump.h>
 
@@ -37,6 +36,7 @@
  */
 #define PAGE_MASK      (~((1 << PAGE_SHIFT) - 1))
 
+#ifdef __KERNEL__
 /*
  * KERNELBASE is the virtual address of the start of the kernel, it's often
  * the same as PAGE_OFFSET, but _might not be_.
