diff -Naur linux-2.6.38.8.orig//drivers/input/misc/ati_remote.c linux-2.6.38.8/drivers/input/misc/ati_remote.c
--- linux-2.6.38.8.orig//drivers/input/misc/ati_remote.c	2011-07-18 14:26:46.902341081 +0200
+++ linux-2.6.38.8/drivers/input/misc/ati_remote.c	2011-07-18 14:28:39.302341142 +0200
@@ -248,13 +248,13 @@
 	{KIND_FILTERED, 0xc6, 0x01, EV_KEY, KEY_B, 1},
 	{KIND_FILTERED, 0xde, 0x19, EV_KEY, KEY_C, 1},
 	{KIND_FILTERED, 0xe0, 0x1b, EV_KEY, KEY_D, 1},
-	{KIND_FILTERED, 0xe6, 0x21, EV_KEY, KEY_E, 1},
+	{KIND_FILTERED, 0xe6, 0x21, EV_KEY, KEY_ESC, 1},
 	{KIND_FILTERED, 0xe8, 0x23, EV_KEY, KEY_F, 1},
 
 	/* "special" keys */
 	{KIND_FILTERED, 0xdd, 0x18, EV_KEY, KEY_KPENTER, 1},    /* "check" */
 	{KIND_FILTERED, 0xdb, 0x16, EV_KEY, KEY_MENU, 1},       /* "menu" */
-	{KIND_FILTERED, 0xc7, 0x02, EV_KEY, KEY_POWER, 1},      /* Power */
+	{KIND_FILTERED, 0xc7, 0x02, EV_KEY, KEY_S, 1},      /* Power */
 	{KIND_FILTERED, 0xc8, 0x03, EV_KEY, KEY_TV, 1},         /* TV */
 	{KIND_FILTERED, 0xc9, 0x04, EV_KEY, KEY_DVD, 1},        /* DVD */
 	{KIND_FILTERED, 0xca, 0x05, EV_KEY, KEY_WWW, 1},        /* WEB */
@@ -266,7 +266,7 @@
 	{KIND_FILTERED, 0xe4, 0x1f, EV_KEY, KEY_RIGHT, 1},      /* right */
 	{KIND_FILTERED, 0xe7, 0x22, EV_KEY, KEY_DOWN, 1},       /* down */
 	{KIND_FILTERED, 0xdf, 0x1a, EV_KEY, KEY_UP, 1},         /* up */
-	{KIND_FILTERED, 0xe3, 0x1e, EV_KEY, KEY_OK, 1},         /* "OK" */
+	{KIND_FILTERED, 0xe3, 0x1e, EV_KEY, KEY_ENTER, 1},         /* "OK" */
 	{KIND_FILTERED, 0xce, 0x09, EV_KEY, KEY_VOLUMEDOWN, 1}, /* VOL + */
 	{KIND_FILTERED, 0xcd, 0x08, EV_KEY, KEY_VOLUMEUP, 1},   /* VOL - */
 	{KIND_FILTERED, 0xcf, 0x0a, EV_KEY, KEY_MUTE, 1},       /* MUTE  */
