diff --git linux-2.6.33.5/arch/arm/mach-omap2/Makefile linux-2.6.33.5/arch/arm/mach-omap2/Makefile
--- linux-2.6.33.5/arch/arm/mach-omap2/Makefile
+++ linux-2.6.33.5/arch/arm/mach-omap2/Makefile
@@ -67,6 +67,8 @@
 i2c-omap-$(CONFIG_I2C_OMAP)		:= i2c.o
 obj-y					+= $(i2c-omap-m) $(i2c-omap-y)
 
+obj-$(CONFIG_MPU_BRIDGE)		+= dspbridge.o
+
 # Specific board support
 obj-$(CONFIG_MACH_OMAP_GENERIC)		+= board-generic.o
 obj-$(CONFIG_MACH_OMAP_H4)		+= board-h4.o
diff --git linux-2.6.33.5/arch/arm/mach-omap2/dspbridge.c linux-2.6.33.5/arch/arm/mach-omap2/dspbridge.c
new file mode 100644
--- /dev/null
+++ linux-2.6.33.5/arch/arm/mach-omap2/dspbridge.c
@@ -0,0 +1,76 @@
+/*
+ * TI's dspbridge platform device registration
+ *
+ * Copyright (C) 2005-2006 Texas Instruments, Inc.
+ * Copyright (C) 2009 Nokia Corporation
+ *
+ * Written by Hiroshi DOYU <Hiroshi.DOYU@nokia.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include <linux/platform_device.h>
+
+#ifdef CONFIG_BRIDGE_DVFS
+#include <mach/omap-pm.h>
+#endif
+
+#include <dspbridge/host_os.h>
+
+static struct platform_device *dspbridge_pdev;
+
+static struct dspbridge_platform_data dspbridge_pdata __initdata = {
+#ifdef CONFIG_BRIDGE_DVFS
+	.dsp_set_min_opp = omap_pm_dsp_set_min_opp,
+	.dsp_get_opp	 = omap_pm_dsp_get_opp,
+	.cpu_set_freq	 = omap_pm_cpu_set_freq,
+	.cpu_get_freq	 = omap_pm_cpu_get_freq,
+#endif
+};
+
+static int __init dspbridge_init(void)
+{
+	struct platform_device *pdev;
+	int err = -ENOMEM;
+	struct dspbridge_platform_data *pdata = &dspbridge_pdata;
+
+	pdata->phys_mempool_base = dspbridge_get_mempool_base();
+
+	if (pdata->phys_mempool_base) {
+		pdata->phys_mempool_size = CONFIG_BRIDGE_MEMPOOL_SIZE;
+		pr_info("%s: %x bytes @ %x\n", __func__,
+			pdata->phys_mempool_size, pdata->phys_mempool_base);
+	}
+
+	pdev = platform_device_alloc("C6410", -1);
+	if (!pdev)
+		goto err_out;
+
+	err = platform_device_add_data(pdev, pdata, sizeof(*pdata));
+	if (err)
+		goto err_out;
+
+	err = platform_device_add(pdev);
+	if (err)
+		goto err_out;
+
+	dspbridge_pdev = pdev;
+	return 0;
+
+err_out:
+	platform_device_put(pdev);
+	return err;
+}
+module_init(dspbridge_init);
+
+static void __exit dspbridge_exit(void)
+{
+	platform_device_unregister(dspbridge_pdev);
+}
+module_exit(dspbridge_exit);
+
+MODULE_AUTHOR("Hiroshi DOYU");
+MODULE_DESCRIPTION("TI's dspbridge platform device registration");
+MODULE_LICENSE("GPL v2");
diff --git linux-2.6.33.5/arch/arm/mach-omap2/io.c linux-2.6.33.5/arch/arm/mach-omap2/io.c
--- linux-2.6.33.5/arch/arm/mach-omap2/io.c
+++ linux-2.6.33.5/arch/arm/mach-omap2/io.c
@@ -49,6 +49,8 @@
 #include "omap_hwmod_2430.h"
 #include "omap_hwmod_34xx.h"
 
+#include <dspbridge/host_os.h>
+
 /*
  * The machine specific code may provide the extra mapping besides the
  * default mapping provided here.
@@ -248,6 +250,7 @@
 	omap_sram_init();
 	omapfb_reserve_sdram();
 	omap_vram_reserve_sdram();
+	dspbridge_reserve_sdram();
 }
 
 #ifdef CONFIG_ARCH_OMAP2420
diff --git linux-2.6.33.5/arch/arm/plat-omap/devices.c linux-2.6.33.5/arch/arm/plat-omap/devices.c
--- linux-2.6.33.5/arch/arm/plat-omap/devices.c
+++ linux-2.6.33.5/arch/arm/plat-omap/devices.c
@@ -14,6 +14,8 @@
 #include <linux/init.h>
 #include <linux/platform_device.h>
 #include <linux/io.h>
+#include <linux/bootmem.h>
+#include <plat/dsp_common.h>
 
 #include <mach/hardware.h>
 #include <asm/mach-types.h>
@@ -90,6 +92,33 @@
 static inline void omap_init_dsp(void) { }
 #endif	/* CONFIG_OMAP_DSP */
 
+#if defined(CONFIG_MPU_BRIDGE) ||  defined(CONFIG_MPU_BRIDGE_MODULE)
+
+static unsigned long dspbridge_phys_mempool_base;
+
+void dspbridge_reserve_sdram(void)
+{
+	void *va;
+	unsigned long size = CONFIG_BRIDGE_MEMPOOL_SIZE;
+
+	if (!size)
+		return;
+
+	va = __alloc_bootmem_nopanic(size, SZ_1M, 0);
+	if (!va) {
+		pr_err("%s: Failed to bootmem allocation(%lu bytes)\n",
+		       __func__, size);
+		return;
+	}
+	dspbridge_phys_mempool_base = virt_to_phys(va);
+}
+
+unsigned long dspbridge_get_mempool_base(void)
+{
+	return dspbridge_phys_mempool_base;
+}
+EXPORT_SYMBOL(dspbridge_get_mempool_base);
+#endif
 /*-------------------------------------------------------------------------*/
 #if	defined(CONFIG_KEYBOARD_OMAP) || defined(CONFIG_KEYBOARD_OMAP_MODULE)
 
diff --git linux-2.6.33.5/drivers/Kconfig linux-2.6.33.5/drivers/Kconfig
--- linux-2.6.33.5/drivers/Kconfig
+++ linux-2.6.33.5/drivers/Kconfig
@@ -113,4 +113,6 @@
 source "drivers/staging/Kconfig"
 
 source "drivers/platform/Kconfig"
+
+source "drivers/dsp/bridge/Kconfig"
 endmenu
diff --git linux-2.6.33.5/drivers/Makefile linux-2.6.33.5/drivers/Makefile
--- linux-2.6.33.5/drivers/Makefile
+++ linux-2.6.33.5/drivers/Makefile
@@ -36,6 +36,8 @@
 obj-$(CONFIG_FB_I810)           += video/i810/
 obj-$(CONFIG_FB_INTEL)          += video/intelfb/
 
+obj-$(CONFIG_MPU_BRIDGE)	+= dsp/bridge/
+
 obj-y				+= serial/
 obj-$(CONFIG_PARPORT)		+= parport/
 obj-y				+= base/ block/ misc/ mfd/
diff --git linux-2.6.33.5/mm/memory.c linux-2.6.33.5/mm/memory.c
--- linux-2.6.33.5/mm/memory.c
+++ linux-2.6.33.5/mm/memory.c
@@ -1243,6 +1243,7 @@
 		return ERR_PTR(-EFAULT);
 	return page;
 }
+EXPORT_SYMBOL_GPL(follow_page);
 
 int __get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
 		     unsigned long start, int nr_pages, unsigned int gup_flags,
