diff -Naur linux-2.6.35.orig/arch/arm/mach-omap2/control.c linux-2.6.35/arch/arm/mach-omap2/control.c
--- linux-2.6.35.orig/arch/arm/mach-omap2/control.c	2010-08-05 20:02:54.407472213 +0200
+++ linux-2.6.35/arch/arm/mach-omap2/control.c	2010-08-05 20:04:45.498988811 +0200
@@ -166,6 +166,7 @@
 {
 	return __raw_readl(OMAP_CTRL_REGADDR(offset));
 }
+EXPORT_SYMBOL(omap_ctrl_readl);
 
 void omap_ctrl_writeb(u8 val, u16 offset)
 {
@@ -181,6 +182,7 @@
 {
 	__raw_writel(val, OMAP_CTRL_REGADDR(offset));
 }
+EXPORT_SYMBOL(omap_ctrl_writel);
 
 #if defined(CONFIG_ARCH_OMAP3) && defined(CONFIG_PM)
 /*
