--- linux-2.6.9.orig/include/linux/fs.h	Sun Dec 26 09:47:03 2004
+++ linux-2.6.9/include/linux/fs.h	Sun Dec 26 09:46:29 2004
@@ -16,7 +16,6 @@
 #include <linux/dcache.h>
 #include <linux/stat.h>
 #include <linux/cache.h>
-#include <linux/prio_tree.h>
 #include <linux/kobject.h>
 #include <asm/atomic.h>
 
@@ -224,6 +223,7 @@
 #ifdef __KERNEL__
 
 #include <linux/list.h>
+#include <linux/prio_tree.h>
 #include <linux/radix-tree.h>
 #include <linux/audit.h>
 #include <linux/init.h>
