diff -Naur linux-2.6.31-rc4.orig/scripts/Makefile.lib linux-2.6.31-rc4/scripts/Makefile.lib
--- linux-2.6.31-rc4.orig/scripts/Makefile.lib	2009-07-25 12:47:45.000000000 +0200
+++ linux-2.6.31-rc4/scripts/Makefile.lib	2009-07-25 12:49:12.000000000 +0200
@@ -228,5 +228,5 @@
 
 quiet_cmd_lzma = LZMA    $@
 cmd_lzma = (cat $(filter-out FORCE,$^) | \
-	lzma -9 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
+	lzma e -a1 -so && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
 	(rm -f $@ ; false)
