diff -Naur linux-2.6.31-rc4.orig/drivers/gpu/drm/i915/i915_dma.c linux-2.6.31-rc4/drivers/gpu/drm/i915/i915_dma.c
--- linux-2.6.31-rc4.orig/drivers/gpu/drm/i915/i915_dma.c	2009-07-25 12:47:32.000000000 +0200
+++ linux-2.6.31-rc4/drivers/gpu/drm/i915/i915_dma.c	2009-07-25 15:39:48.000000000 +0200
@@ -1186,8 +1186,8 @@
 	if (ret)
 		goto out_iomapfree;
 
-	/* enable GEM by default */
-	dev_priv->has_gem = 1;
+	/* disable GEM by default */
+	dev_priv->has_gem = 0;
 
 	if (prealloc_size > agp_size * 3 / 4) {
 		DRM_ERROR("Detected broken video BIOS with %d/%dkB of video "
