diff -Naur linux-2.6.11.6.orig/init/main.c linux-2.6.11.6/init/main.c
--- linux-2.6.11.6.orig/init/main.c	2005-06-27 21:51:41.000000000 +0200
+++ linux-2.6.11.6/init/main.c	2005-06-27 21:59:23.000000000 +0200
@@ -635,6 +635,8 @@
 
 static int init(void * unused)
 {
+	char *console = "/dev_console";
+
 	lock_kernel();
 	/*
 	 * init can run on any cpu.
@@ -688,8 +690,12 @@
 	system_state = SYSTEM_RUNNING;
 	numa_default_policy();
 
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0) {
+		sys_mknod(console, S_IFCHR|0600, (TTYAUX_MAJOR<<8)|1);
+		if (sys_open(console, O_RDWR, 0) < 0)
+			printk(KERN_WARNING "Warning: unable to open an initial console.\n");
+		sys_unlink(console);
+	}
 
 	(void) sys_dup(0);
 	(void) sys_dup(0);
