--- linux-2.6.8.1.orig/drivers/video/bootsplash/bootsplash.c	2004-09-20 20:45:33.953903864 +0200
+++ linux-2.6.8.1/drivers/video/bootsplash/bootsplash.c	2004-09-20 23:47:01.810698040 +0200
@@ -471,7 +471,7 @@
 	    silentsize = 0;
 	}
 	sboxcnt = splash_gets(ndata, SPLASH_OFF_SBOXCNT);
-	if (silentsize) {
+	if (silentsize > 12 * sboxcnt) {
 	    unsigned char *simage = ndata + len + splash_size + 12 * sboxcnt;
 	    if (!jpeg_check_size(simage, width, height) ||
 		splash_check_jpeg(simage, width, height, info->var.bits_per_pixel)) {
@@ -488,12 +488,12 @@
 	sd->splash_sboxes = 0;
 	sd->splash_sboxcount = 0;
 	if (silentsize) {
-	    sd->splash_silentjpeg = vmalloc(silentsize);
-	    if (sd->splash_silentjpeg) {
-		memcpy(sd->splash_silentjpeg, ndata + len + splash_size, silentsize);
-		sd->splash_sboxes = vc->vc_splash_data->splash_silentjpeg;
-		sd->splash_silentjpeg += 12 * sboxcnt;
+	    sd->splash_sboxes = vmalloc(silentsize);
+	    if (sd->splash_sboxes) {
+		memcpy(sd->splash_sboxes, ndata + len + splash_size, silentsize);
 		sd->splash_sboxcount = sboxcnt;
+		if (silentsize > 12 * sboxcnt)
+		    sd->splash_silentjpeg = sd->splash_sboxes + 12 * sboxcnt;
 	    }
 	}
 	sd->splash_state = splash_getb(ndata, SPLASH_OFF_STATE);
@@ -542,6 +542,8 @@
 	sd->splash_palette = sd->splash_boxes + boxcnt * 12;
 	sd->splash_jpeg = sd->splash_palette + palcnt;
 	sd->splash_palcnt = palcnt / 3;
+	if (splash_usesilent && sd->splash_silentjpeg == 0)
+	    sd->splash_silentjpeg = sd->splash_jpeg;
 	sd->splash_dosilent = sd->splash_silentjpeg != 0;
 	return unit;
     }
