* allow bootsplash to work on non-vesa fb drivers

diff -Naur linux-2.6.11.orig/drivers/video/bootsplash/bootsplash.c linux-2.6.11/drivers/video/bootsplash/bootsplash.c
--- linux-2.6.11.orig/drivers/video/bootsplash/bootsplash.c	2005-03-30 19:32:11.000000000 +0200
+++ linux-2.6.11/drivers/video/bootsplash/bootsplash.c	2005-03-30 19:32:52.000000000 +0200
@@ -28,7 +28,6 @@
 #include "bootsplash.h"
 #include "decode-jpg.h"
 
-extern struct fb_ops vesafb_ops;
 extern signed char con2fb_map[MAX_NR_CONSOLES];
 
 #define SPLASH_VERSION "3.1.7.4-geexbox"
@@ -545,7 +544,7 @@
 	    printk(" found, but has oversized text area!\n");
 	    return -1;
 	}
-	if (!vc_cons[unit].d || info->fbops != &vesafb_ops) {
+	if (!vc_cons[unit].d) {
 	    splash_free(vc, info);
 	    printk(" found, but framebuffer can't handle it!\n");
 	    return -1;
