diff -Naur linux-2.6.19.orig/drivers/video/bootsplash/bootsplash.c linux-2.6.19/drivers/video/bootsplash/bootsplash.c
--- linux-2.6.19.orig/drivers/video/bootsplash/bootsplash.c	2006-11-30 20:31:43.000000000 +0100
+++ linux-2.6.19/drivers/video/bootsplash/bootsplash.c	2006-11-30 20:32:45.000000000 +0100
@@ -12,7 +12,6 @@
  *  For more information on this code check http://www.bootsplash.org/
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/types.h>
 #include <linux/fb.h>
diff -Naur linux-2.6.19.orig/drivers/video/bootsplash/decode-jpg.c linux-2.6.19/drivers/video/bootsplash/decode-jpg.c
--- linux-2.6.19.orig/drivers/video/bootsplash/decode-jpg.c	2006-11-30 20:31:43.000000000 +0100
+++ linux-2.6.19/drivers/video/bootsplash/decode-jpg.c	2006-11-30 20:40:08.000000000 +0100
@@ -5,7 +5,6 @@
  *                  
  */
 
-#include <linux/config.h>
 #include <linux/string.h>
 #include <asm/byteorder.h>
 
diff -Naur linux-2.6.19.orig/drivers/video/bootsplash/render.c linux-2.6.19/drivers/video/bootsplash/render.c
--- linux-2.6.19.orig/drivers/video/bootsplash/render.c	2006-11-30 20:31:43.000000000 +0100
+++ linux-2.6.19/drivers/video/bootsplash/render.c	2006-11-30 20:40:19.000000000 +0100
@@ -2,7 +2,6 @@
  *    linux/drivers/video/bootsplash/render.c - splash screen render functions.
  */
 
-#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/types.h>
 #include <linux/fb.h>
