Index: linux-2.6.16/arch/i386/boot/video.S
===================================================================
--- linux-2.6.16.orig/arch/i386/boot/video.S	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6.16/arch/i386/boot/video.S	2006-03-20 18:47:03.000000000 +0100
@@ -1853,10 +1853,10 @@
 	int	$0x16
 	ret
 
-# Read a key with a timeout of 30 seconds.
+# Read a key with a timeout of 3 seconds.
 # The hardware clock is used to get the time.
 getkt:	call	gettime
-	addb	$30, %al			# Wait 30 seconds
+	addb	$3, %al			# Wait 3 seconds
 	cmpb	$60, %al
 	jl	lminute
 
@@ -1968,7 +1968,7 @@
 
 # Status messages
 keymsg:		.ascii	"Press <RETURN> to see video modes available, "
-		.ascii	"<SPACE> to continue or wait 30 secs"
+		.ascii	"<SPACE> to continue or wait 3 secs"
 		.byte	0x0d, 0x0a, 0
 
 listhdr:	.byte	0x0d, 0x0a
