diff -Nur linux-2.6.12.orig/scripts/kconfig/lkc.h linux-2.6.12/scripts/kconfig/lkc.h
--- linux-2.6.12.orig/scripts/kconfig/lkc.h	Fri Jun 17 22:48:29 2005
+++ linux-2.6.12/scripts/kconfig/lkc.h	Sat Jun 18 09:20:30 2005
@@ -8,8 +8,6 @@
 
 #include "expr.h"
 
-#include <libintl.h>
-
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -28,7 +26,7 @@
 #define PACKAGE "linux"
 #define LOCALEDIR "/usr/share/locale"
 
-#define _(text) gettext(text)
+#define _(text) (text)
 #define N_(text) (text)
 
 int zconfparse(void);
diff -Nur linux-2.6.12.orig/scripts/kconfig/mconf.c linux-2.6.12/scripts/kconfig/mconf.c
--- linux-2.6.12.orig/scripts/kconfig/mconf.c	Fri Jun 17 22:48:29 2005
+++ linux-2.6.12/scripts/kconfig/mconf.c	Sat Jun 18 09:21:55 2005
@@ -1042,10 +1042,6 @@
 	char *mode;
 	int stat;
 
-	setlocale(LC_ALL, "");
-	bindtextdomain(PACKAGE, LOCALEDIR);
-	textdomain(PACKAGE);
-
 	conf_parse(av[1]);
 	conf_read(NULL);
 
