diff -Nur linux-2.6.12.orig/fs/fat/inode.c linux-2.6.12/fs/fat/inode.c
--- linux-2.6.12.orig/fs/fat/inode.c	Fri Jul 22 16:04:30 2005
+++ linux-2.6.12/fs/fat/inode.c	Fri Jul 22 16:04:37 2005
@@ -847,7 +847,8 @@
 		opts->shortname = 0;
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->nocase = 0;
 	*debug = 0;
diff -Nur linux-2.6.12.orig/fs/isofs/inode.c linux-2.6.12/fs/isofs/inode.c
--- linux-2.6.12.orig/fs/isofs/inode.c	Fri Jul 22 16:10:17 2005
+++ linux-2.6.12/fs/isofs/inode.c	Fri Jul 22 16:10:19 2005
@@ -376,7 +376,7 @@
 	popt->gid = 0;
 	popt->uid = 0;
 	popt->iocharset = NULL;
-	popt->utf8 = 0;
+	popt->utf8 = 1;
 	popt->session=-1;
 	popt->sbsector=-1;
 	if (!options)
