* dirty hack to get sound on a Mac Mini

diff -Naur linux-2.6.11.orig/sound/ppc/pmac.c linux-2.6.11/sound/ppc/pmac.c
--- linux-2.6.11.orig/sound/ppc/pmac.c	2005-03-22 16:46:58.000000000 +0100
+++ linux-2.6.11/sound/ppc/pmac.c	2005-03-22 16:49:02.000000000 +0100
@@ -964,7 +964,7 @@
 	if (device_is_compatible(sound, "AOAKeylargo")) {
 		/* Seems to support the stock AWACS frequencies, but has
 		   a snapper mixer */
-		chip->model = PMAC_SNAPPER;
+		chip->model = PMAC_AWACS;
 		// chip->can_byte_swap = 0; /* FIXME: check this */
 		chip->control_mask = MASK_IEPC | 0x11; /* disable IEE */
 	}
