diff -Naur linux-2.6.14.4-old/drivers/net/via-rhine.c linux-2.6.14.4-new/drivers/net/via-rhine.c
--- linux-2.6.14.4-old/drivers/net/via-rhine.c	2005-12-14 15:50:41.000000000 -0800
+++ linux-2.6.14.4-new/drivers/net/via-rhine.c	2005-12-15 05:34:30.000000000 -0800
@@ -1934,9 +1934,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }

