Index: linux-2.6.16/drivers/net/via-rhine.c
===================================================================
--- linux-2.6.16.orig/drivers/net/via-rhine.c	2006-03-20 06:53:29.000000000 +0100
+++ linux-2.6.16/drivers/net/via-rhine.c	2006-03-20 18:47:13.000000000 +0100
@@ -1964,9 +1964,6 @@
 		iowrite8(ioread8(ioaddr + StickyHW) | 0x04, ioaddr + StickyHW);
 	}
 
-	/* Hit power state D3 (sleep) */
-	iowrite8(ioread8(ioaddr + StickyHW) | 0x03, ioaddr + StickyHW);
-
 	/* TODO: Check use of pci_enable_wake() */
 
 }
