diff -Naur linux-2.6.13.orig/include/linux/joystick.h linux-2.6.13/include/linux/joystick.h
--- linux-2.6.13.orig/include/linux/joystick.h	2005-08-29 01:41:01.000000000 +0200
+++ linux-2.6.13/include/linux/joystick.h	2005-08-29 23:53:32.000000000 +0200
@@ -134,6 +134,7 @@
 	struct JS_DATA_TYPE JS_CORR;
 };
 
+#ifdef __KERNEL__
 #if BITS_PER_LONG == 64
 #define JS_DATA_SAVE_TYPE JS_DATA_SAVE_TYPE_64
 #elif BITS_PER_LONG == 32
@@ -141,5 +142,6 @@
 #else
 #error Unexpected BITS_PER_LONG
 #endif
+#endif
 
 #endif /* _LINUX_JOYSTICK_H */
