diff -Naur linux-2.6.16.orig/drivers/scsi/libata-core.c linux-2.6.16/drivers/scsi/libata-core.c
--- linux-2.6.16.orig/drivers/scsi/libata-core.c	2006-03-29 22:05:34.000000000 +0200
+++ linux-2.6.16/drivers/scsi/libata-core.c	2006-03-29 22:05:53.000000000 +0200
@@ -78,7 +78,7 @@
 static unsigned int ata_unique_id = 1;
 static struct workqueue_struct *ata_wq;
 
-int atapi_enabled = 0;
+int atapi_enabled = 1;
 module_param(atapi_enabled, int, 0444);
 MODULE_PARM_DESC(atapi_enabled, "Enable discovery of ATAPI devices (0=off, 1=on)");
 
