diff -Naur xorg-server-1.7.0.orig/configure xorg-server-1.7.0/configure
--- xorg-server-1.7.0.orig/configure	2009-10-12 19:27:02.000000000 +0200
+++ xorg-server-1.7.0/configure	2009-10-12 19:27:24.000000000 +0200
@@ -30510,7 +30510,7 @@
   exec_prefix_NONE=
   test "x$prefix" = xNONE && prefix_NONE=yes && prefix=$ac_default_prefix
   test "x$exec_prefix" = xNONE && exec_prefix_NONE=yes && exec_prefix=$prefix
-  eval ac_define_dir="\"$prefix\""
+  eval ac_define_dir="\"$exec_prefix\""
   eval ac_define_dir="\"$ac_define_dir\""
   PROJECTROOT="$ac_define_dir"
 
