Index: xorg-server/hw/dmx/dmxinit.c
===================================================================
--- xorg-server.orig/hw/dmx/dmxinit.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/dmx/dmxinit.c	2010-07-14 14:02:26.161453002 +1000
@@ -830,7 +830,7 @@
 /** This function is called in Xserver/os/utils.c from \a AbortServer().
  * We must ensure that backend and console state is restored in the
  * event the server shutdown wasn't clean. */
-void AbortDDX(void)
+void SigAbortDDX(int signo)
 {
     int i;
 
@@ -848,6 +848,11 @@
 }
 #endif
 
+void AbortDDX(void)
+{
+    SigAbortDDX(0);
+}
+
 /** This function is called in Xserver/dix/main.c from \a main() when
  * dispatchException & DE_TERMINATE (which is the only way to exit the
  * main loop without an interruption. */
Index: xorg-server/hw/kdrive/src/kdrive.c
===================================================================
--- xorg-server.orig/hw/kdrive/src/kdrive.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/kdrive/src/kdrive.c	2010-07-14 14:02:26.161453002 +1000
@@ -321,7 +321,7 @@
 }
 
 void
-AbortDDX(void)
+SigAbortDDX(int signo)
 {
     KdDisableScreens ();
     if (kdOsFuncs)
@@ -338,6 +338,12 @@
 }
 
 void
+AbortDDX(void)
+{
+    SigAbortDDX(0);
+}
+
+void
 ddxGiveUp (void)
 {
     AbortDDX ();
Index: xorg-server/hw/vfb/InitOutput.c
===================================================================
--- xorg-server.orig/hw/vfb/InitOutput.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/vfb/InitOutput.c	2010-07-14 14:02:26.161453002 +1000
@@ -201,11 +201,17 @@
 }
 
 void
-AbortDDX(void)
+SigAbortDDX(int signo)
 {
     ddxGiveUp();
 }
 
+void
+AbortDDX(void)
+{
+    SigAbortDDX(0);
+}
+
 #ifdef __APPLE__
 void
 DarwinHandleGUI(int argc, char *argv[])
Index: xorg-server/hw/xfree86/common/xf86Events.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86Events.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/xfree86/common/xf86Events.c	2010-07-14 14:02:26.171453002 +1000
@@ -348,6 +348,8 @@
 int
 xf86SigWrapper(int signo)
 {
+  static Bool beenhere = FALSE;
+
   if ((signo == SIGILL) && xf86SigIllHandler) {
     (*xf86SigIllHandler)();
     return 0; /* continue */
Index: xorg-server/hw/xfree86/common/xf86Init.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86Init.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/xfree86/common/xf86Init.c	2010-07-14 14:02:26.171453002 +1000
@@ -899,14 +899,17 @@
 }
 
 /*
- * ddxGiveUp --
+ * ddxSigGiveUp --
  *      Device dependent cleanup. Called by by dix before normal server death.
  *      For SYSV386 we must switch the terminal back to normal mode. No error-
  *      checking here, since there should be restored as much as possible.
+ *
+ *      If a non-zero signo is passed, re-raise that signal rather than
+ *      calling abort().
  */
 
 void
-ddxGiveUp(void)
+ddxSigGiveUp(int signo)
 {
     int i;
 
@@ -933,24 +936,45 @@
     if (xorgHWOpenConsole)
 	xf86CloseConsole();
 
+    ErrorF (" ddxSigGiveUp: Closing log\n");
     xf86CloseLog();
 
     /* If an unexpected signal was caught, dump a core for debugging */
-    if (xf86Info.caughtSignal)
-	OsAbort();
+    if (xf86Info.caughtSignal) {
+        if (signo != 0) {
+            raise(signo);
+        } else {
+            OsAbort();
+        }
+    }
 }
 
+/*
+ * ddxGiveUp --
+ *      Device dependent cleanup. Called by by dix before normal server death.
+ *      For SYSV386 we must switch the terminal back to normal mode. No error-
+ *      checking here, since there should be restored as much as possible.
+ */
+
+void
+ddxGiveUp()
+{
+    ddxSigGiveUp(0);
+}
 
 
 /*
- * AbortDDX --
+ * SigAbortDDX --
  *      DDX - specific abort routine.  Called by AbortServer(). The attempt is
  *      made to restore all original setting of the displays. Also all devices
  *      are closed.
+ *
+ *      If a non-zero signo is passed, re-raise that signal rather than calling
+ *      abort()
  */
 
 void
-AbortDDX(void)
+SigAbortDDX(int signo)
 {
   int i;
 
@@ -983,7 +1007,20 @@
    * This is needed for an abnormal server exit, since the normal exit stuff
    * MUST also be performed (i.e. the vt must be left in a defined state)
    */
-  ddxGiveUp();
+  ddxSigGiveUp(signo);
+}
+
+/*
+ * AbortDDX --
+ *      DDX - specific abort routine.  The attempt is made to restore
+ *      all original setting of the displays. Also all devices are
+ *      closed.
+ */
+
+void
+AbortDDX()
+{
+    SigAbortDDX(0);
 }
 
 void
Index: xorg-server/hw/xnest/Init.c
===================================================================
--- xorg-server.orig/hw/xnest/Init.c	2010-07-05 15:56:19.000000000 +1000
+++ xorg-server/hw/xnest/Init.c	2010-07-14 14:02:26.171453002 +1000
@@ -117,12 +117,17 @@
 /*
  * DDX - specific abort routine.  Called by AbortServer().
  */
-void AbortDDX(void)
+void SigAbortDDX(int signo)
 {
   xnestDoFullGeneration = True;
   xnestCloseDisplay();
 }
 
+void AbortDDX(void)
+{
+    SigAbortDDX(0);
+}
+
 /* Called by GiveUp(). */
 void ddxGiveUp(void)
 {
Index: xorg-server/hw/xquartz/darwin.c
===================================================================
--- xorg-server.orig/hw/xquartz/darwin.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/xquartz/darwin.c	2010-07-14 14:02:26.171453002 +1000
@@ -765,17 +765,22 @@
 
 
 /*
- * AbortDDX --
+ * [Sig]AbortDDX --
  *      DDX - specific abort routine.  Called by AbortServer(). The attempt is
  *      made to restore all original setting of the displays. Also all devices
  *      are closed.
  */
-void AbortDDX( void )
+void SigAbortDDX( void )
 {
     ErrorF( "   AbortDDX\n" );
     OsAbort();
 }
 
+void AbortDDX( void )
+{
+    SigAbortDDX(0);
+}
+
 #include "mivalidate.h" // for union _Validate used by windowstr.h
 #include "windowstr.h"  // for struct _Window
 #include "scrnintstr.h" // for struct _Screen
Index: xorg-server/hw/xwin/InitOutput.c
===================================================================
--- xorg-server.orig/hw/xwin/InitOutput.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/hw/xwin/InitOutput.c	2010-07-14 14:02:26.171453002 +1000
@@ -288,7 +288,7 @@
 
 /* See Porting Layer Definition - p. 57 */
 void
-AbortDDX (void)
+SigAbortDDX (int signo)
 {
 #if CYGDEBUG
   winDebug ("AbortDDX\n");
@@ -296,6 +296,12 @@
   ddxGiveUp ();
 }
 
+void
+AbortDDX (void)
+{
+    SigAbortDDX(0);
+}
+
 #ifdef __CYGWIN__
 /* hasmntopt is currently not implemented for cygwin */
 static const char *winCheckMntOpt(const struct mntent *mnt, const char *opt)
Index: xorg-server/include/os.h
===================================================================
--- xorg-server.orig/include/os.h	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/include/os.h	2010-07-14 14:02:26.171453002 +1000
@@ -477,7 +477,9 @@
 /* stuff for FlushCallback */
 extern _X_EXPORT CallbackListPtr FlushCallback;
 
+extern _X_EXPORT void SigAbortDDX(int signo);
 extern _X_EXPORT void AbortDDX(void);
+extern _X_EXPORT void ddxSigGiveUp(int signo);
 extern _X_EXPORT void ddxGiveUp(void);
 extern _X_EXPORT int TimeSinceLastInputEvent(void);
 
Index: xorg-server/os/log.c
===================================================================
--- xorg-server.orig/os/log.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/os/log.c	2010-07-14 14:06:06.531453001 +1000
@@ -411,9 +411,11 @@
 
 void
 AbortServer(void) _X_NORETURN;
+void
+SigAbortServer(int signo) _X_NORETURN;
 
 void
-AbortServer(void)
+SigAbortServer(int signo)
 {
 #ifdef XF86BIGFONT
     XF86BigfontCleanup();
@@ -421,11 +423,21 @@
     CloseWellKnownConnections();
     OsCleanup(TRUE);
     CloseDownDevices();
-    AbortDDX();
+    SigAbortDDX(signo);
     fflush(stderr);
-    if (CoreDump)
-	OsAbort();
-    exit (1);
+    if (CoreDump) {
+        if (signo != 0)
+            raise(signo);
+        else
+            OsAbort();
+    } else
+       exit (1);
+}
+
+void
+AbortServer(void)
+{
+    SigAbortServer(0);
 }
 
 #define AUDIT_PREFIX "AUDIT: %s: %ld: "
@@ -528,6 +540,27 @@
 }
 
 void
+FatalSignal(int signo)
+{
+    static Bool beenhere = FALSE;
+
+    if (beenhere)
+	ErrorF("\nFatalSignal re-entered, aborting\n");
+    else
+	ErrorF("\nCaught signal %d (%s). Server aborting\n",
+               signo, strsignal(signo));
+
+    if (!beenhere)
+	OsVendorFatalError();
+    if (!beenhere) {
+	beenhere = TRUE;
+	SigAbortServer(signo);
+    } else
+	abort();
+    /*NOTREACHED*/
+}
+
+void
 FatalError(const char *f, ...)
 {
     va_list args;
Index: xorg-server/os/osinit.c
===================================================================
--- xorg-server.orig/os/osinit.c	2010-07-14 13:21:02.000000000 +1000
+++ xorg-server/os/osinit.c	2010-07-14 14:02:26.181453002 +1000
@@ -148,13 +148,13 @@
           case SIGBUS:
           case SIGILL:
           case SIGFPE:
+	      signal(signo,SIG_DFL);
 	      ErrorF("%s at address %p\n", strsignal(signo), sip->si_addr);
       }
   }
 #endif
 
-  FatalError("Caught signal %d (%s). Server aborting\n",
-	     signo, strsignal(signo));
+  FatalSignal(signo);
 }
 
 void
