diff -Naur xorg-server-1.9.0.orig//hw/xfree86/common/xf86pciBus.c xorg-server-1.9.0/hw/xfree86/common/xf86pciBus.c
--- xorg-server-1.9.0.orig//hw/xfree86/common/xf86pciBus.c	2010-10-04 23:26:49.623916999 +0200
+++ xorg-server-1.9.0/hw/xfree86/common/xf86pciBus.c	2010-10-04 23:29:56.113917003 +0200
@@ -1102,8 +1102,16 @@
 	case 0x1142:		    driverList[0] = "apm"; break;
 	case 0xedd8:		    driverList[0] = "ark"; break;
 	case 0x1a03:		    driverList[0] = "ast"; break;
-	case 0x1002:		    driverList[0] = "ati"; break;
-	case 0x102c:		    driverList[0] = "chips"; break;
+        case 0x1002:
+            /* Technical fglrx works only on 0x791e and newer, but
+               if the user has it installed, -ati and -radeonhd
+               probably won't work anyway so we may as well try it
+               in case it randomly does work.  */
+            driverList[0] = "fglrx"; /* Supports R600 and newer only */
+            driverList[1] = "radeonhd";
+            driverList[2] = "ati";
+            break;
+        case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
 	case 0x3d3d:		    driverList[0] = "glint"; break;
 	case 0x105d:		    driverList[0] = "i128"; break;
