Index: xorg-server/hw/xfree86/common/xf86VidMode.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86VidMode.c	2011-08-24 12:52:45.565647144 +0300
+++ xorg-server/hw/xfree86/common/xf86VidMode.c	2011-08-24 12:56:49.605650623 +0300
@@ -222,6 +222,9 @@
 
     pScrn = xf86Screens[scrnIndex];
     pVidMode = VMPTR(pScrn->pScreen);
+    if (pScrn->modes == NULL)
+        return FALSE;
+
     pVidMode->First = pScrn->modes;
     pVidMode->Next =  pVidMode->First->next;
 
