Index: xorg-server/hw/xfree86/common/xf86VidMode.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xf86VidMode.c	2009-12-02 23:06:16.000000000 +0200
+++ xorg-server/hw/xfree86/common/xf86VidMode.c	2009-12-02 23:06:21.000000000 +0200
@@ -220,6 +220,9 @@
 
     pScrn = xf86Screens[scrnIndex];
     pVidMode = VMPTR(pScrn->pScreen);
+    if (pScrn->modes == NULL)
+        return FALSE;
+
     pVidMode->First = pScrn->modes;
     pVidMode->Next =  pVidMode->First->next;
 
