diff -Naur xorg-server-1.7.3.orig/config/Makefile.am xorg-server-1.7.3/config/Makefile.am
--- xorg-server-1.7.3.orig/config/Makefile.am	2009-12-18 18:28:04.501655881 +0100
+++ xorg-server-1.7.3/config/Makefile.am	2009-12-18 18:29:31.632519049 +0100
@@ -1,15 +1,7 @@
 AM_CFLAGS = @DIX_CFLAGS@
 
 noinst_LTLIBRARIES = libconfig.la
-libconfig_la_SOURCES = config.c config-backends.h
-
-if CONFIG_UDEV
-
-AM_CFLAGS += @UDEV_CFLAGS@
-libconfig_la_SOURCES += udev.c
-libconfig_la_LIBADD = @UDEV_LIBS@
-
-else
+libconfig_la_SOURCES = config.c udev.c config-backends.h
 
 if CONFIG_NEED_DBUS
 AM_CFLAGS += @DBUS_CFLAGS@
@@ -31,6 +23,4 @@
 
 endif # CONFIG_NEED_DBUS
 
-endif # !CONFIG_UDEV
-
 EXTRA_DIST = xorg-server.conf x11-input.fdi
diff -Naur xorg-server-1.7.3.orig/config/Makefile.in xorg-server-1.7.3/config/Makefile.in
--- xorg-server-1.7.3.orig/config/Makefile.in	2009-12-12 15:25:53.271272011 +0100
+++ xorg-server-1.7.3/config/Makefile.in	2009-12-18 18:34:41.692551463 +0100
@@ -67,7 +67,7 @@
 @CONFIG_NEED_DBUS_TRUE@am__objects_1 = dbus-core.lo
 @CONFIG_DBUS_API_TRUE@am__objects_2 = dbus.lo
 @CONFIG_HAL_TRUE@am__objects_3 = hal.lo
-am_libconfig_la_OBJECTS = config.lo $(am__objects_1) $(am__objects_2) \
+am_libconfig_la_OBJECTS = config.lo udev.lo $(am__objects_1) $(am__objects_2) \
 	$(am__objects_3)
 libconfig_la_OBJECTS = $(am_libconfig_la_OBJECTS)
 AM_V_lt = $(am__v_lt_$(V))
@@ -418,7 +418,7 @@
 top_srcdir = @top_srcdir@
 AM_CFLAGS = @DIX_CFLAGS@ $(am__append_1)
 noinst_LTLIBRARIES = libconfig.la
-libconfig_la_SOURCES = config.c config-backends.h $(am__append_2) \
+libconfig_la_SOURCES = config.c udev.c config-backends.h $(am__append_2) \
 	$(am__append_3) $(am__append_4)
 @CONFIG_DBUS_API_TRUE@dbusconfigdir = $(sysconfdir)/dbus-1/system.d
 @CONFIG_DBUS_API_TRUE@dbusconfig_DATA = xorg-server.conf
