Index: xorg-server/fb/fbpict.c
===================================================================
--- xorg-server.orig/fb/fbpict.c	2010-07-05 16:12:34.118394001 +1000
+++ xorg-server/fb/fbpict.c	2010-07-05 17:20:51.548394001 +1000
@@ -35,7 +35,7 @@
 #include "mipict.h"
 #include "fbpict.h"
 
-#define mod(a,b) ((b) == 1 ? 0 : (a) >= 0 ? (a) % (b) : (b) - (-a) % (b))
+#define mod(a,b) ((b) == 1 ? 0 : (a) >= 0 ? (a) % (b) : (b) - (-(a)) % (b))
 
 void
 fbWalkCompositeRegion (CARD8 op,
