Index: xorg-server/hw/xfree86/modes/xf86RandR12.c
===================================================================
--- xorg-server.orig/hw/xfree86/modes/xf86RandR12.c	2011-04-09 10:36:57.000000000 +0300
+++ xorg-server/hw/xfree86/modes/xf86RandR12.c	2011-04-09 10:38:10.000000000 +0300
@@ -930,7 +930,8 @@
     for (c = 0; c < config->num_crtc; c++) {
 	xf86CrtcPtr    crtc = config->crtc[c];
 
-	RRCrtcSetRotations (crtc->randr_crtc, rotations);
+        if (crtc != NULL)
+            RRCrtcSetRotations (crtc->randr_crtc, rotations);
     }
 #endif
     randrp->supported_rotations = rotations;
