Index: xorg-server/hw/xfree86/common/xisb.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xisb.c	2011-08-24 12:52:45.315647139 +0300
+++ xorg-server/hw/xfree86/common/xisb.c	2011-08-24 12:56:49.655650624 +0300
@@ -98,6 +98,9 @@
 {
 	int ret;
 
+        if (b == NULL)
+            return -2;
+
 	if (b->current >= b->end)
 	{
 		if (b->block_duration >= 0)
