Index: xorg-server/hw/xfree86/common/xisb.c
===================================================================
--- xorg-server.orig/hw/xfree86/common/xisb.c	2011-04-09 10:36:57.000000000 +0300
+++ xorg-server/hw/xfree86/common/xisb.c	2011-04-09 10:38:10.000000000 +0300
@@ -98,6 +98,9 @@
 {
 	int ret;
 
+        if (b == NULL)
+            return -2;
+
 	if (b->current >= b->end)
 	{
 		if (b->block_duration >= 0)
