=== modified file 'hw/vfb/InitOutput.c'
Index: xorg-server/hw/vfb/InitOutput.c
===================================================================
--- xorg-server.orig/hw/vfb/InitOutput.c	2010-07-14 15:06:03.511453001 +1000
+++ xorg-server/hw/vfb/InitOutput.c	2010-07-14 15:07:00.101453002 +1000
@@ -66,6 +66,7 @@
 #endif /* HAS_SHM */
 #include "dix.h"
 #include "miline.h"
+#include "randrstr.h"
 
 #define VFB_DEFAULT_WIDTH      1280
 #define VFB_DEFAULT_HEIGHT     1024
@@ -881,6 +882,12 @@
 
     if (!ret) return FALSE;
 
+    if (!miRandRInit(pScreen))
+        /* Not sure how to emit warnings in xorg. Should warn here though:
+	 * ("Could not initialise RANDR\n");
+	 */
+        return FALSE;
+
     miInitializeBackingStore(pScreen);
 
     /*
