=== modified file 'randr/randr.c'
Index: xorg-server/randr/randr.c
===================================================================
--- xorg-server.orig/randr/randr.c	2011-08-24 12:52:44.745647133 +0300
+++ xorg-server/randr/randr.c	2011-08-24 12:57:30.195651199 +0300
@@ -235,7 +235,7 @@
     /*
      * Calling function best set these function vectors
      */
-    pScrPriv->rrGetInfo = 0;
+    pScrPriv->rrGetInfo = NULL;
     pScrPriv->maxWidth = pScrPriv->minWidth = pScreen->width;
     pScrPriv->maxHeight = pScrPriv->minHeight = pScreen->height;
     
