=== modified file 'randr/randr.c'
--- a/randr/randr.c	2009-12-19 01:44:12 +0000
+++ b/randr/randr.c	2010-02-02 16:28:26 +0000
@@ -234,7 +234,7 @@
     /*
      * Calling function best set these function vectors
      */
-    pScrPriv->rrGetInfo = 0;
+    pScrPriv->rrGetInfo = NULL;
     pScrPriv->maxWidth = pScrPriv->minWidth = pScreen->width;
     pScrPriv->maxHeight = pScrPriv->minHeight = pScreen->height;
     

