=== modified file 'randr/randr.c'
Index: xorg-server/randr/randr.c
===================================================================
--- xorg-server.orig/randr/randr.c	2011-04-09 10:36:56.000000000 +0300
+++ xorg-server/randr/randr.c	2011-04-09 10:38:10.000000000 +0300
@@ -235,7 +235,7 @@
     /*
      * Calling function best set these function vectors
      */
-    pScrPriv->rrGetInfo = 0;
+    pScrPriv->rrGetInfo = NULL;
     pScrPriv->maxWidth = pScrPriv->minWidth = pScreen->width;
     pScrPriv->maxHeight = pScrPriv->minHeight = pScreen->height;
     
