Index: xorg-server/os/utils.c
===================================================================
--- xorg-server.orig/os/utils.c	2011-01-10 17:10:56.190132573 -0600
+++ xorg-server/os/utils.c	2011-01-10 17:14:10.000000000 -0600
@@ -499,6 +499,7 @@
     ErrorF("-nolisten string       don't listen on protocol\n");
     ErrorF("-noreset               don't reset after last client exists\n");
     ErrorF("-background [none]     create root window with no background\n");
+    ErrorF("-nr                    (Ubuntu-specific) Synonym for -background none\n");
     ErrorF("-reset                 reset after last client exists\n");
     ErrorF("-p #                   screen-saver pattern duration (minutes)\n");
     ErrorF("-pn                    accept failure to listen on all ports\n");
@@ -837,6 +838,8 @@
                     UseMsg();
             }
         }
+	else if ( strcmp( argv[i], "-nr") == 0)
+	    bgNoneRoot = TRUE;
         else if ( strcmp( argv[i], "-maxbigreqsize") == 0) {
              if(++i < argc) {
                  long reqSizeArg = atol(argv[i]);
