diff -Naur xorg-server-1.4.99.905.orig/hw/xfree86/common/extramodes xorg-server-1.4.99.905/hw/xfree86/common/extramodes
--- xorg-server-1.4.99.905.orig/hw/xfree86/common/extramodes	2008-07-06 13:44:52.000000000 +0200
+++ xorg-server-1.4.99.905/hw/xfree86/common/extramodes	2008-07-06 13:48:14.000000000 +0200
@@ -7,12 +7,45 @@
 # 832x624 @ 75Hz (74.55Hz) (fix if the official/Apple spec is different) hsync: 49.725kHz
 ModeLine "832x624" 57.284 832  864  928 1152  624  625  628  667 -Hsync -Vsync
 
+# 1152x768 @ 54.8Hz (Titanium PowerBook) hsync: 44.2kHz
+ModeLine "1152x768" 64.995 1152 1178 1314 1472  768  771 777 806 +hsync +vsync
+
+# 1152x864 @ 85Hz (Red Hat custom modeline)
+ModeLine "1152x864" 121.5 1152 1216 1344 1568 864  865  868  911 +hsync -vsync
+
+# 1280x768 @ 60Hz (LCD native res) hsync: 47.70kHz
+ModeLine "1280x768" 80.14 1280 1344 1480 1680  768 769 772 795
+
+# 1280x800 @ 60Hz (LCD native res) hsync: 49.68kHz
+ModeLine "1280x800" 83.46 1280 1344 1480 1680  800 801 804 828
+
 # 1400x1050 @ 60Hz (VESA GTF) hsync: 65.5kHz
 ModeLine "1400x1050" 122.0 1400 1488 1640 1880   1050 1052 1064 1082 +hsync +vsync
 
+# 1400x1050 @ 70Hz (Red Hat custom modeline) hsync: 76.8kHz
+ModeLine "1400x1050" 151.0 1400 1464 1656 1960 1050 1051 1054 1100 +hsync +vsync
+
 # 1400x1050 @ 75Hz (VESA GTF) hsync: 82.2kHz
 ModeLine "1400x1050" 155.8 1400 1464 1784 1912   1050 1052 1064 1090 +hsync +vsync
 
+# 1400x1050 @ 85Hz (Red Hat custom modeline) hsync: 93.2 kHz
+ModeLine "1400x1050" 184.0 1400 1464 1656 1960   1050 1051 1054 1100 +hsync +vsync
+
+# 1440x900 @ 100Hz (17" iMac LCD) hsync: 67 kHz
+ModeLine "1440x900" 108.84 1440 1472 1880 1912   900 918 927 946 +hsync +vsync
+
+# 1600x1024 @ 60Hz (SGI 1600SW) hsync: 64.0kHz
+ModeLine "1600x1024" 106.910 1600 1620 1640 1670 1024 1027 1030 1067 -hsync -vsync
+
+# 1680x1050 @ 60Hz (LCD native res) hsync: 64.22kHz
+ModeLine "1680x1050" 147.14 1680 1784 1968 2256  1050 1051 1054 1087
+
+# 1920x1200 @ 60Hz (LCD native res) hsync: 74.52kHz
+ModeLine "1920x1200" 193.16 1920 2048 2256 2592  1200 1201 1204 1242
+
+# 1920x1200 @ 70Hz (Red Hat custom modeline - bugzilla #53364) hsync: 87.50 KHz
+ModeLine "1920x1200" 230.0 1920 1936 2096 2528  1200 1201 1204 1250 -HSync -VSync
+
 # 1920x1440 @ 85Hz (VESA GTF) hsync: 128.5kHz
 Modeline "1920x1440" 341.35  1920 2072 2288 2656  1440 1441 1444 1512 -hsync +vsync
 
