diff -Naur xorg-server-X11R7.1-1.1.0.orig/Xext/fontcache.c xorg-server-X11R7.1-1.1.0/Xext/fontcache.c
--- xorg-server-X11R7.1-1.1.0.orig/Xext/fontcache.c	2007-01-27 23:00:55.000000000 +0100
+++ xorg-server-X11R7.1-1.1.0/Xext/fontcache.c	2007-01-28 00:47:52.000000000 +0100
@@ -46,8 +46,8 @@
 #include "inputstr.h"
 #include "servermd.h"
 #define _FONTCACHE_SERVER_
-#include "fontcacheP.h"
-#include "fontcachstr.h"
+#include <X11/extensions/fontcacheP.h>
+#include <X11/extensions/fontcachstr.h>
 #include <X11/Xfuncproto.h>
 
 #include "swaprep.h"
