From: Ian Pilcher <arequipeno@gmail.com>
Subject: Backout commit 6dd775f57d2f94f0ddaee324aeec33b9b66ed5bc

Index: xorg-server/composite/compalloc.c
===================================================================
--- xorg-server.orig/composite/compalloc.c	2011-04-06 18:25:23.000000000 +0300
+++ xorg-server/composite/compalloc.c	2011-04-09 10:39:42.000000000 +0300
@@ -508,17 +508,6 @@
     return Success;
 }
 
-static int
-bgNoneVisitWindow(WindowPtr pWin, void *null)
-{
-    if (pWin->backgroundState != BackgroundPixmap)
-	return WT_WALKCHILDREN;
-    if (pWin->background.pixmap != None)
-	return WT_WALKCHILDREN;
-
-    return WT_STOPWALKING;
-}
-
 static PixmapPtr
 compNewPixmap (WindowPtr pWin, int x, int y, int w, int h, Bool map)
 {
@@ -540,17 +529,6 @@
 	return pPixmap;
 
     /*
-     * If there's no bg=None in the tree, we're done.
-     *
-     * We could optimize this more by collection the regions of all the
-     * bg=None subwindows and feeding that in as the clip for the
-     * CopyArea below, but since window trees are shallow these days it
-     * might not be worth the effort.
-     */
-    if (TraverseTree(pWin, bgNoneVisitWindow, NULL) == WT_NOMATCH)
-	return pPixmap;
-
-    /*
      * Copy bits from the parent into the new pixmap so that it will
      * have "reasonable" contents in case for background None areas.
      */
