diff -Naur xorg-server-1.6.3.901/configure.org xorg-server-1.6.3.901/configure
--- xorg-server-1.6.3.901/configure.org       2009-08-26 10:28:12.000000000 +0200
+++ xorg-server-1.6.3.901/configure   2009-08-26 10:30:04.000000000 +0200
@@ -16311,12 +16311,12 @@
         pkg_cv_GL_CFLAGS="$GL_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9 gl >= 7.1.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9 gl >= 7.1.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GL_CFLAGS=`$PKG_CONFIG --cflags "glproto >= 1.4.9 gl >= 7.1.0" 2>/dev/null`
+  pkg_cv_GL_CFLAGS=`$PKG_CONFIG --cflags "glproto >= 1.4.9" 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -16329,12 +16329,12 @@
         pkg_cv_GL_LIBS="$GL_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9 gl >= 7.1.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9 gl >= 7.1.0") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"glproto >= 1.4.9\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "glproto >= 1.4.9") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GL_LIBS=`$PKG_CONFIG --libs "glproto >= 1.4.9 gl >= 7.1.0" 2>/dev/null`
+  pkg_cv_GL_LIBS=`$PKG_CONFIG --libs "glproto >= 1.4.9" 2>/dev/null`
 else
   pkg_failed=yes
 fi
