--- a/Xi/exevents.c
+++ b/Xi/exevents.c
@@ -1647,7 +1647,8 @@ ProcessTouchOwnership(DeviceIntPtr dev,
                     if (ev == ti->history + ti->history_size)
                         ev = ti->history;
                 }
-            } else if (tc->type == TOUCH_SELECT_UNOWNED &&
+            } else if ((tc->type == TOUCH_SELECT_UNOWNED ||
+                        tc->type == TOUCH_GRAB) &&
                        !ti->emulate_pointer) {
                 DeliverTouchOwnershipEvent(tc, ti);
             }
