diff -Naur xorg-server-1.6.3.orig/hw/xfree86/common/xf86AutoConfig.c xorg-server-1.6.3/hw/xfree86/common/xf86AutoConfig.c
--- xorg-server-1.6.3.orig/hw/xfree86/common/xf86AutoConfig.c	2009-08-09 11:40:49.000000000 +0200
+++ xorg-server-1.6.3/hw/xfree86/common/xf86AutoConfig.c	2009-08-09 11:42:53.000000000 +0200
@@ -179,7 +179,7 @@
 	case 0x102b:		    driverList[0] = "mga";	break;
 	case 0x10c8:		    driverList[0] = "neomagic"; break;
 	case 0x105d:		    driverList[0] = "i128";	break;
-	case 0x10de: case 0x12d2:   driverList[0] = "nv";	break;
+	case 0x10de: case 0x12d2:   driverList[0] = "nvidia"; driverList[1] = "nv";	break;
 	case 0x1163:		    driverList[0] = "rendition"; break;
 	case 0x5333:
 	    switch (dev->device_id)
