diff -Naur xorg-server-1.7.5/hw/xfree86/common/xf86AutoConfig.c xorg-server-1.7.5a/hw/xfree86/common/xf86AutoConfig.c
--- xorg-server-1.7.5/hw/xfree86/common/xf86AutoConfig.c	2010-07-29 18:48:34.454383797 +0200
+++ xorg-server-1.7.5a/hw/xfree86/common/xf86AutoConfig.c	2010-07-29 18:53:09.553869239 +0200
@@ -176,7 +176,14 @@
 	case 0x1142:		    driverList[0] = "apm"; break;
 	case 0xedd8:		    driverList[0] = "ark"; break;
 	case 0x1a03:		    driverList[0] = "ast"; break;
-	case 0x1002:		    driverList[0] = "ati"; break;
+	case 0x1002:
+		if (dev->device_id < 0x791e)
+			driverList[0] = "ati";
+		else {
+			driverList[0] = "fglrx"; /* Supports R600 and newer only */
+			driverList[1] = "ati";
+		}
+		break;
 	case 0x102c:		    driverList[0] = "chips"; break;
 	case 0x1013:		    driverList[0] = "cirrus"; break;
 	case 0x3d3d:		    driverList[0] = "glint"; break;
