diff -Naur RT2500-Linux-STA-1.4.3.0.orig/STA/Module/rtmp.h RT2500-Linux-STA-1.4.3.0/STA/Module/rtmp.h
--- RT2500-Linux-STA-1.4.3.0.orig/STA/Module/rtmp.h	2004-09-15 05:46:42.000000000 +0200
+++ RT2500-Linux-STA-1.4.3.0/STA/Module/rtmp.h	2005-01-14 18:32:28.000000000 +0100
@@ -1064,10 +1064,10 @@
 // Enable & Disable NIC interrupt via writing interrupt mask register
 // Since it use ADAPTER structure, it have to be put after structure definition.
 //
-inline  VOID    NICDisableInterrupt(
+VOID    NICDisableInterrupt(
     IN  PRTMP_ADAPTER   pAdapter);
 
-inline  VOID    NICEnableInterrupt(
+VOID    NICEnableInterrupt(
     IN  PRTMP_ADAPTER   pAdapter);
 
 BOOLEAN NICCheckForHang(
@@ -1319,7 +1319,7 @@
     IN  ULONG           Length,
     IN  UCHAR           TxRate);
 
-inline NDIS_STATUS RTMPCheckRxDescriptor(
+NDIS_STATUS RTMPCheckRxDescriptor(
     IN  PRXD_STRUC      pRxD);
 
 NDIS_STATUS RTMPApplyPacketFilter(
