diff -Naur vdr-dvd-0.3.6+cvs20090118/Makefile vdr-dvd-0.3.6+cvs20090118a/Makefile
--- vdr-dvd-0.3.6+cvs20090118/Makefile	2009-04-17 10:07:09.000000000 +0200
+++ vdr-dvd-0.3.6+cvs20090118a/Makefile	2009-04-17 10:09:56.000000000 +0200
@@ -69,10 +69,6 @@
 OBJS = $(PLUGIN).o dvddev.o player-dvd.o control-dvd.o dvdspu.o     \
 	           ca52.o i18n.o setup-dvd.o 
 
-### The main target:
-
-all: libvdr-$(PLUGIN).so i18n
-
 ### Implicit rules:
 
 %.o: %.c
@@ -87,33 +83,10 @@
 
 -include $(DEPFILE)
 
-### Internationalization (I18N):
-
-PODIR     = po
-LOCALEDIR = $(VDRDIR)/locale
-I18Npo    = $(wildcard $(PODIR)/*.po)
-I18Nmsgs  = $(addprefix $(LOCALEDIR)/, $(addsuffix /LC_MESSAGES/vdr-$(PLUGIN).mo, $(notdir $(foreach file, $(I18Npo), $(basename $(file))))))
-I18Npot   = $(PODIR)/$(PLUGIN).pot
-
-%.mo: %.po
-	msgfmt -c -o $@ $<
-
-$(I18Npot): $(wildcard *.c)
-	xgettext -C -cTRANSLATORS --no-wrap --no-location -k -ktr -ktrNOOP --msgid-bugs-address='<marco@lordzodiac.de>' -o $@ $^
-
-%.po: $(I18Npot)
-	msgmerge -U --no-wrap --no-location --backup=none -q $@ $<
-	@touch $@
-
-$(I18Nmsgs): $(LOCALEDIR)/%/LC_MESSAGES/vdr-$(PLUGIN).mo: $(PODIR)/%.mo
-	@mkdir -p $(dir $@)
-	cp $< $@
-
-.PHONY: i18n
-i18n: $(I18Nmsgs) $(I18Npot)
-
 ### Targets:
 
+all: libvdr-$(PLUGIN).so
+
 libvdr-$(PLUGIN).so: $(OBJS) retain-sym
 	$(CXX) $(LDFLAGS) -shared $(OBJS) $(LIBS) -o $@
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
@@ -129,4 +102,4 @@
 	echo "VDRPluginCreator" > retain-sym
 
 clean:
-	@-rm -f $(OBJS) $(DEPFILE) *.so *.tgz core* retain-sym *~ $(PODIR)/*.mo $(PODIR)/*.pot
+	@-rm -f $(OBJS) $(DEPFILE) *.so *.tgz core* retain-sym *~
