--- vdr-dvd-0.3.6+cvs20090118/player-dvd.c.org	2014-05-14 06:37:27.256261423 +0200
+++ vdr-dvd-0.3.6+cvs20090118/player-dvd.c	2014-05-14 06:41:00.512265083 +0200
@@ -375,7 +375,13 @@
 	        if (playMode == pmSlow)
                 sp=2;
         }
-        DeviceTrickSpeed(sp);
+#if VDRVERSNUM >= 20102
+	bool forward;
+	if (playDir == pdBackward) forward=false; else forward=true;
+	DeviceTrickSpeed(sp,forward);
+#else
+	DeviceTrickSpeed(sp);
+#endif
 
     } else if ( nts>0 && nts-NORMAL_SPEED <= MAX_MAX_SPEEDS ) {
         fastWindFactor  = 1;
