--- vdr-2.0.0/device.c.orig	2013-04-05 11:08:09.924421454 +0200
+++ vdr-2.0.0/device.c	2013-04-05 16:11:59.991878156 +0200
@@ -108,11 +108,11 @@
 
   player = NULL;
   isPlayingVideo = false;
+  keepTracks = false; // used in ClrAvailableTracks()!
   ClrAvailableTracks();
   currentAudioTrack = ttNone;
   currentAudioTrackMissingCount = 0;
   currentSubtitleTrack = ttNone;
-  keepTracks = false;
   liveSubtitle = NULL;
   dvbSubtitleConverter = NULL;
   autoSelectPreferredSubtitleLanguage = true;
