--- vdr-dvbhddevice-1.0/hdffosd.c.orf	2015-02-21 12:04:49.945916236 +0100
+++ vdr-dvbhddevice-1.0/hdffosd.c	2015-02-21 12:05:07.181915898 +0100
@@ -681,7 +681,7 @@
                 return;
         }
         LOCK_PIXMAPS;
-        while (cPixmapMemory *pm = RenderPixmaps())
+        while (cPixmapMemory *pm = dynamic_cast<cPixmapMemory*>(RenderPixmaps()))
         {
             int w = pm->ViewPort().Width();
             int h = pm->ViewPort().Height();
