diff -ruN vdr-1.7.26/Makefile vdr-1.7.26.mod/Makefile
--- vdr-1.7.26/Makefile	2012-03-11 13:38:52.086314317 +0100
+++ vdr-1.7.26.mod/Makefile	2012-03-11 13:41:12.471010585 +0100
@@ -179,7 +179,7 @@
 	@failed="";\
 	noapiv="";\
 	for i in `ls $(PLUGINDIR)/src | grep -v '[^a-z0-9]'`; do\
-	    echo "Plugin $$i:";\
+	    true echo "Plugin $$i:";\
 	    if ! grep -q "\$$(LIBDIR)/.*\$$(APIVERSION)" "$(PLUGINDIR)/src/$$i/Makefile" ; then\
 	       echo "ERROR: plugin $$i doesn't honor APIVERSION - not compiled!";\
 	       noapiv="$$noapiv $$i";\
diff -ruN vdr-1.7.26/menu.c vdr-1.7.26.mod/menu.c
--- vdr-1.7.26/menu.c	2012-03-11 13:38:52.090314318 +0100
+++ vdr-1.7.26.mod/menu.c	2012-03-11 13:41:12.475010544 +0100
@@ -3824,7 +3824,7 @@
 {
   Clear();
   char buffer[64];
-  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("Setup"), VDRVERSION);
+  snprintf(buffer, sizeof(buffer), "%s - VDR %s", tr("VDR Setup"), VDRVERSION);
   SetTitle(buffer);
   SetHasHotkeys();
   Add(new cOsdItem(hk(tr("OSD")),           osUser1));
@@ -3903,9 +3903,9 @@
 {
 #ifdef USE_SETUP
   // Load Menu Configuration
-  cString menuXML = cString::sprintf("%s/setup/vdr-menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
+  cString menuXML = cString::sprintf("%s/../menu.%s.xml", cPlugin::ConfigDirectory(), Setup.OSDLanguage);
   if (access(menuXML, 04) == -1)
-     menuXML = cString::sprintf("%s/setup/vdr-menu.xml", cPlugin::ConfigDirectory());
+     menuXML = cString::sprintf("%s/../menu.xml", cPlugin::ConfigDirectory());
   subMenu.LoadXml(menuXML);
   nrDynamicMenuEntries = 0;
 #endif /* SETUP */
@@ -4055,7 +4055,7 @@
          else if (strcmp(item, "Recordings") == 0)
             Add(new cOsdItem(hk(tr("Recordings")), osRecordings));
          else if (strcmp(item, "Setup") == 0) {
-            cString itemSetup = cString::sprintf("%s%s", tr("Setup"), *subMenu.GetMenuSuffix());
+            cString itemSetup = cString::sprintf("%s%s", tr("VDR Setup"), *subMenu.GetMenuSuffix());
             Add(new cOsdItem(hk(itemSetup), osSetup));
             }
          else if (strcmp(item, "Commands") == 0 && Commands.Count() > 0) {
@@ -4106,9 +4106,9 @@
   // More basic menu items:
 
 #ifdef USE_PINPLUGIN
-  if (!cStatus::MsgMenuItemProtected("Setup", true)) Add(new cOsdItem(hk(tr("Setup")), osSetup));
+  if (!cStatus::MsgMenuItemProtected("Setup", true)) Add(new cOsdItem(hk(tr("VDR Setup")), osSetup));
 #else
-  Add(new cOsdItem(hk(tr("Setup")),      osSetup));
+  Add(new cOsdItem(hk(tr("VDR Setup")),      osSetup));
 #endif /* PINPLUGIN */
   if (Commands.Count())
 #ifdef USE_PINPLUGIN
diff -ruN vdr-1.7.26/menuitems.c vdr-1.7.26.mod/menuitems.c
--- vdr-1.7.26/menuitems.c	2012-03-11 13:37:10.505810662 +0100
+++ vdr-1.7.26.mod/menuitems.c	2012-03-11 13:41:12.475010544 +0100
@@ -1129,7 +1129,7 @@
 
 void cMenuSetupPage::SetSection(const char *Section)
 {
-  SetTitle(cString::sprintf("%s - %s", tr("Setup"), Section));
+  SetTitle(cString::sprintf("%s - %s", tr("VDR Setup"), Section));
 }
 
 eOSState cMenuSetupPage::ProcessKey(eKeys Key)
diff -ruN vdr-1.7.26/submenu.c vdr-1.7.26.mod/submenu.c
--- vdr-1.7.26/submenu.c	2012-03-11 13:37:10.513810701 +0100
+++ vdr-1.7.26.mod/submenu.c	2012-03-11 13:41:12.479010487 +0100
@@ -446,10 +446,6 @@
   if ((ok = xmlDoc.LoadFile())) {
      if ((root = xmlDoc.FirstChildElement("menus")) != NULL) {
         cString tmp = root->Attribute("suffix");
-#ifdef USE_WAREAGLEICON
-        if      (strcmp(tmp, "ICON_FOLDER") == 0)      tmp = cString::sprintf(" %s", IsLangUtf8() ? ICON_FOLDER_UTF8 : ICON_FOLDER);
-        else if (strcmp(tmp, "ICON_MOVE_FOLDER") == 0) tmp = cString::sprintf(" %s", IsLangUtf8() ? ICON_MOVE_FOLDER_UTF8 : ICON_MOVE_FOLDER);
-#endif /* WAREAGLEICON */
         if (*tmp)
            _menuSuffix = tmp;
         else
