diff -ruN vdr-1.7.26/config.c vdr-1.7.26.mod/config.c
--- vdr-1.7.26/config.c	2012-03-11 13:54:10.822870224 +0100
+++ vdr-1.7.26.mod/config.c	2012-03-11 13:54:21.634923768 +0100
@@ -480,6 +480,8 @@
   InitialVolume = -1;
   ChannelsWrap = 0;
   EmergencyExit = 1;
+  JumpSeconds = 60;
+  JumpSecondsSlow = 10;
 #ifdef USE_LIVEBUFFER
   LiveBufferSize = 15;
   LiveBufferMaxFileSize = 100;
@@ -700,6 +702,8 @@
   else if (!strcasecmp(Name, "DeviceBondings"))      DeviceBondings     = Value;
   else if (!strcasecmp(Name, "ChannelsWrap"))        ChannelsWrap       = atoi(Value);
   else if (!strcasecmp(Name, "EmergencyExit"))       EmergencyExit      = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSeconds"))         JumpSeconds        = atoi(Value);
+  else if (!strcasecmp(Name, "JumpSecondsSlow"))     JumpSecondsSlow    = atoi(Value);
 #ifdef USE_LIRCSETTINGS
   else if (!strcasecmp(Name, "LircRepeatDelay"))     LircRepeatDelay    = atoi(Value);
   else if (!strcasecmp(Name, "LircRepeatFreq"))      LircRepeatFreq     = atoi(Value);
@@ -846,6 +850,8 @@
   Store("DeviceBondings",     DeviceBondings);
   Store("ChannelsWrap",       ChannelsWrap);
   Store("EmergencyExit",      EmergencyExit);
+  Store("JumpSeconds",        JumpSeconds);
+  Store("JumpSecondsSlow",    JumpSecondsSlow);
 #ifdef USE_LIVEBUFFER
   Store("LiveBufferSize",     LiveBufferSize);
 #endif  /* LIVEBUFFER */
diff -ruN vdr-1.7.26/config.h vdr-1.7.26.mod/config.h
--- vdr-1.7.26/config.h	2012-03-11 13:54:10.822870224 +0100
+++ vdr-1.7.26.mod/config.h	2012-03-11 13:54:21.638923775 +0100
@@ -359,6 +359,8 @@
   int InitialVolume;
   int ChannelsWrap;
   int EmergencyExit;
+  int JumpSeconds;
+  int JumpSecondsSlow;
 #ifdef USE_LIVEBUFFER
   int LiveBufferSize;
   int LiveBufferMaxFileSize;
diff -ruN vdr-1.7.26/menu.c vdr-1.7.26.mod/menu.c
--- vdr-1.7.26/menu.c	2012-03-11 13:54:10.830870034 +0100
+++ vdr-1.7.26.mod/menu.c	2012-03-11 13:54:21.642923789 +0100
@@ -3703,6 +3703,8 @@
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Play&Jump"), &data.PlayJump));
   Add(new cMenuEditBoolItem(tr("Setup.Replay$Pause at last mark"), &data.PauseLastMark));
 #endif /* JUMPPLAY */
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds"), &data.JumpSeconds));
+  Add(new cMenuEditIntItem(tr("Setup.Replay$Jump Seconds Slow"), &data.JumpSecondsSlow));
 }
 
 void cMenuSetupReplay::Store(void)
@@ -6061,9 +6063,13 @@
     case kFastFwd:
     case kRight:   Forward(); break;
     case kGreen|k_Repeat:
-    case kGreen:   SkipSeconds(-60); break;
+    case kGreen:   SkipSeconds(-Setup.JumpSeconds); break;
+    case k1|k_Repeat:
+    case k1:       SkipSeconds(-Setup.JumpSecondsSlow); break;
     case kYellow|k_Repeat:
-    case kYellow:  SkipSeconds( 60); break;
+    case kYellow:  SkipSeconds(Setup.JumpSeconds); break;
+    case k3|k_Repeat:
+    case k3:       SkipSeconds(Setup.JumpSecondsSlow); break;
 #ifdef USE_LIVEBUFFER
     case kRed:     if(cRecordControls::GetLiveIndex(fileName)) {
 
