diff -ruN vdr-1.7.26/eitscan.c vdr-1.7.26.mod/eitscan.c
--- vdr-1.7.26/eitscan.c	2012-03-07 14:54:34.000000000 +0100
+++ vdr-1.7.26.mod/eitscan.c	2012-03-11 13:57:41.335914027 +0100
@@ -145,7 +145,7 @@
                if (Device && Device->ProvidesEIT()) {
                   for (cScanData *ScanData = scanList->First(); ScanData; ScanData = scanList->Next(ScanData)) {
                       const cChannel *Channel = ScanData->GetChannel();
-                      if (Channel) {
+                      if (Channel && !cSource::IsType(Channel->Source(), 'I')) {
                          if (!Channel->Ca() || Channel->Ca() == Device->DeviceNumber() + 1 || Channel->Ca() >= CA_ENCRYPTED_MIN) {
                             if (Device->ProvidesTransponder(Channel)) {
                                if (!Device->Receiving()) {
diff -ruN vdr-1.7.26/pat.c vdr-1.7.26.mod/pat.c
--- vdr-1.7.26/pat.c	2012-03-02 11:56:45.000000000 +0100
+++ vdr-1.7.26.mod/pat.c	2012-03-11 13:57:41.335914027 +0100
@@ -537,6 +537,7 @@
             }
         if (Setup.UpdateChannels >= 2) {
            Channel->SetPids(Vpid, Ppid, Vtype, Apids, Atypes, ALangs, Dpids, Dtypes, DLangs, Spids, SLangs, Tpid);
+           if (!cSource::IsType(Channel->Source(), 'I'))
            Channel->SetCaIds(CaDescriptors->CaIds());
            Channel->SetSubtitlingDescriptors(SubtitlingTypes, CompositionPageIds, AncillaryPageIds);
            }
