--- a/configure	2014-09-14 19:22:19.480440387 +0200
+++ b/configure	2014-09-14 19:31:25.716424247 +0200
@@ -3294,16 +3294,17 @@
             subarch=$(echo $cpu | sed 's/[^a-z0-9]//g')
         ;;
         *)
-            cpuflags="-mcpu=$cpu"
             case $cpu in
-                cortex-a*)                               subarch=armv7a  ;;
-                cortex-r*)                               subarch=armv7r  ;;
-                cortex-m*)                 enable thumb; subarch=armv7m  ;;
+                cortex-a*)                               subarch=armv7-a ;;
+                cortex-r*)                               subarch=armv7-r ;;
+                cortex-m*)                 enable thumb; subarch=armv7-m ;;
                 arm11*)                                  subarch=armv6   ;;
                 arm[79]*e*|arm9[24]6*|arm96*|arm102[26]) subarch=armv5te ;;
                 armv4*|arm7*|arm9[24]*)                  subarch=armv4   ;;
                 *)                             subarch=$(probe_arm_arch) ;;
             esac
+            cpuflags="-march=$subarch -mtune=$cpu"
+            subarch=$(echo $subarch | sed 's/-//g')
         ;;
     esac
 
