diff -Naur ffmpeg-r23630.orig/libavcodec/arm/mpegvideo_arm.c ffmpeg-r23630/libavcodec/arm/mpegvideo_arm.c
--- ffmpeg-r23630.orig/libavcodec/arm/mpegvideo_arm.c	2010-06-16 23:05:21.968439314 +0200
+++ ffmpeg-r23630/libavcodec/arm/mpegvideo_arm.c	2010-06-16 23:05:48.097389703 +0200
@@ -24,6 +24,7 @@
 #include "mpegvideo_arm.h"
 #include "asm-offsets.h"
 
+#if 0
 #if HAVE_NEON
 CHK_OFFS(MpegEncContext, y_dc_scale,       Y_DC_SCALE);
 CHK_OFFS(MpegEncContext, c_dc_scale,       C_DC_SCALE);
@@ -32,6 +33,7 @@
 CHK_OFFS(MpegEncContext, inter_scantable.raster_end, INTER_SCANTAB_RASTER_END);
 CHK_OFFS(MpegEncContext, h263_aic,         H263_AIC);
 #endif
+#endif
 
 void ff_dct_unquantize_h263_inter_neon(MpegEncContext *s, DCTELEM *block,
                                        int n, int qscale);
