diff -Naur pastebincl-1.0-orig/src/data_devcode.h pastebincl-1.0/src/data_devcode.h
--- pastebincl-1.0-orig/src/data_devcode.h	2012-01-08 15:52:55.000000000 +0100
+++ pastebincl-1.0/src/data_devcode.h	2012-11-20 09:31:23.404386294 +0100
@@ -46,7 +46,7 @@
 #ifndef DEF_DATADEVCODE
 #define DEF_DATADEVCODE
 
-#define DEVELOPPER_CODE "" // TODO: Enter your Pastebin Developper Key here
+#define DEVELOPPER_CODE "751f0a2ed34877816b8aa860deb5bb4c" // TODO: Enter your Pastebin Developper Key here
 
 #endif//DEF_DATADEVCODE
 
diff -Naur pastebincl-1.0-orig/src/Makefile pastebincl-1.0/src/Makefile
--- pastebincl-1.0-orig/src/Makefile	2012-01-08 16:01:37.000000000 +0100
+++ pastebincl-1.0/src/Makefile	2012-11-20 09:32:44.614505409 +0100
@@ -8,7 +8,7 @@
 TARGET=pastebincl
 OBJS=CurlPost.o func.o main.o ParameterRead.o ConfigRead.o
 
-all: inputdevcode checkdevcode $(TARGET)
+all:  $(TARGET)
 	cp $(TARGET) ../
 
 inputdevcode:
