--- a/src/shared/missing.h	2015-10-08 14:33:54.357678366 +0200
+++ b/src/shared/missing.h	2015-10-08 14:33:54.365678367 +0200
@@ -25,6 +25,7 @@

 #include <sys/resource.h>
 #include <sys/syscall.h>
+#include <sys/socket.h>
 #include <fcntl.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -589,3 +589,7 @@
 #ifndef NET_NAME_RENAMED
 #  define NET_NAME_RENAMED 4
 #endif
+
+#ifndef BPF_XOR
+#  define BPF_XOR 0xa0
+#endif
