diff -uNr vdr-dvbapi_tt-1.0.2.orig/Makefile vdr-dvbapi_tt-1.0.2/Makefile
--- vdr-dvbapi_tt-1.0.2.orig/Makefile	2013-09-07 11:00:30.491315447 +0800
+++ vdr-dvbapi_tt-1.0.2/Makefile	2013-09-07 11:01:37.439312647 +0800
@@ -9,7 +9,7 @@
 # IMPORTANT: the presence of this macro is important for the Make.config
 # file. So it must be defined, even if it is not used here!
 #
-PLUGIN = dvbapi
+PLUGIN = dvbapi_ttcam
 
 ### The version number of this plugin (taken from the main source file):
 
