--- netceiver-dkms_0.9.4/mcast/dvbloop/Makefile.org	2013-04-12 07:41:19.887672612 +0200
+++ netceiver-dkms_0.9.4/mcast/dvbloop/Makefile	2013-04-12 07:41:44.923671866 +0200
@@ -44,12 +44,12 @@
 
 ifneq ($(SRCDIR),)
 EXTRA_CFLAGS = \
-        -I$(SRCDIR)/drivers/media/dvb \
-        -Idrivers/media/dvb -I$(@D)/../linux
+        -I$(SRCDIR)/drivers/media \
+        -Idrivers/media -I$(@D)/../linux
 else
 EXTRA_CFLAGS = \
-	-I/usr/src/linux/drivers/media/dvb \
-	-Idrivers/media/dvb -I$(@D)/../linux
+	-I/usr/src/linux/drivers/media \
+	-Idrivers/media -I$(@D)/../linux
 endif
 
 default: build
