--- vdr-svdrpservice-0.0.4/svdrpservice.c.org	2013-02-22 15:38:37.555839255 +0100
+++ vdr-svdrpservice-0.0.4/svdrpservice.c	2013-02-22 15:38:58.723839617 +0100
@@ -61,8 +61,8 @@
 
 const char *cPluginSvdrpService::CommandLineHelp(void)
 {
-	return "  IP[:PORT]  Default server IP and optional port (e.g. 192.168.0.1:2001).\n"
-	       "             If no port is given, the default SVDRP port 2001 is asumed.\n";
+	return "  IP[:PORT]  Default server IP and optional port (e.g. 192.168.0.1:6419).\n"
+	       "             If no port is given, the default SVDRP port 6419 is asumed.\n";
 }
 
 bool cPluginSvdrpService::ProcessArgs(int argc, char *argv[])
--- vdr-svdrpservice-0.0.4/setup.c.org	2013-02-22 15:39:12.443839854 +0100
+++ vdr-svdrpservice-0.0.4/setup.c	2013-02-22 15:39:27.339840111 +0100
@@ -17,7 +17,7 @@
 
 cSvdrpServiceSetup::cSvdrpServiceSetup() {
 	serverIp[0] = 0;
-	serverPort = 2001;
+	serverPort = 6419;
 	connectTimeout = 2;
 	readTimeout = 5;
 }
@@ -34,7 +34,7 @@
 	if (!strcasecmp(Name, "ServerIp"))
 		strn0cpy(serverIp, opt_serverIp ? opt_serverIp : Value, sizeof(serverIp));
 	else if (!strcasecmp(Name, "ServerPort"))
-		serverPort = opt_serverIp ? (opt_serverPort ? atoi(opt_serverPort) : 2001) :  atoi(Value);
+		serverPort = opt_serverIp ? (opt_serverPort ? atoi(opt_serverPort) : 6419) :  atoi(Value);
 	else if (!strcasecmp(Name, "ConnectTimeout"))
 		connectTimeout = atoi(Value);
 	else if (!strcasecmp(Name, "ReadTimeout"))
