--- vdr-svdrpservice-0.0.4/setup.c.orig	2013-03-04 18:27:38.033498152 +0100
+++ vdr-svdrpservice-0.0.4/setup.c	2013-03-04 18:22:06.245492456 +0100
@@ -54,7 +54,7 @@
 
 cSvdrpServiceMenuSetup::cSvdrpServiceMenuSetup() {
 	setupTmp = SvdrpServiceSetup;
-	Add(new cMenuEditStrItem(tr("Default server IP"), setupTmp.serverIp, 15, ".1234567890"));
+	Add(new cMenuEditStrItem(tr("Default server IP"), setupTmp.serverIp, 16, ".1234567890"));
 	Add(new cMenuEditIntItem(tr("Default server port"), &setupTmp.serverPort, 1, 65535));
 	if (cSvdrpServiceSetup::opt_serverIp)
 	{
