reverted:
--- b/libkmod/libkmod-elf.c
+++ a/libkmod/libkmod-elf.c
@@ -280,10 +280,10 @@
 	size_t hdr_size, shdr_size, min_size;
 	int class;
 
+	assert(sizeof(uint16_t) == sizeof(Elf32_Half));
+	assert(sizeof(uint16_t) == sizeof(Elf64_Half));
+	assert(sizeof(uint32_t) == sizeof(Elf32_Word));
+	assert(sizeof(uint32_t) == sizeof(Elf64_Word));
-	assert_cc(sizeof(uint16_t) == sizeof(Elf32_Half));
-	assert_cc(sizeof(uint16_t) == sizeof(Elf64_Half));
-	assert_cc(sizeof(uint32_t) == sizeof(Elf32_Word));
-	assert_cc(sizeof(uint32_t) == sizeof(Elf64_Word));
 
 	class = elf_identify(memory, size);
 	if (class < 0) {
@@ -513,7 +513,7 @@
 	int i, count, err;
 #define MODVERSION_SEC_SIZE (sizeof(struct kmod_modversion64))
 
+	assert(sizeof(struct kmod_modversion64) ==
-	assert_cc(sizeof(struct kmod_modversion64) ==
 					sizeof(struct kmod_modversion32));
 
 	if (elf->class == KMOD_ELF_32)
