--- gst-plugins-gl-1.0+4497be0c97c/gst-libs/gst/gl/gstglwindow.h	2012-10-10 15:24:51.000000000 +0200
+++ gst-plugins-gl-1.0+4497be0c97c/gst-libs/gst/gl/gstglwindow.h-new	2012-10-16 09:03:21.222965447 +0200
@@ -79,12 +79,12 @@
 GQuark gst_gl_window_error_quark (void);
 GType gst_gl_window_get_type (void);
 
-GstGLWindow * gst_gl_window_new (guintptr external_gl_context);
+GstGLWindow * gst_gl_window_new (gulong external_gl_context);
 
-guintptr gst_gl_window_get_internal_gl_context (GstGLWindow *window);
+gulong gst_gl_window_get_internal_gl_context (GstGLWindow *window);
 void gst_gl_window_activate_gl_context (GstGLWindow *window, gboolean activate);
 
-void gst_gl_window_set_external_window_id (GstGLWindow *window, guintptr id);
+void gst_gl_window_set_external_window_id (GstGLWindow *window, gulong id);
 void gst_gl_window_set_draw_callback (GstGLWindow *window, GstGLWindowCB callback, gpointer data);
 void gst_gl_window_set_resize_callback (GstGLWindow *window, GstGLWindowCB2 callback, gpointer data);
 void gst_gl_window_set_close_callback (GstGLWindow *window, GstGLWindowCB callback, gpointer data);
