diff -Naur ncurses-5.7.orig//ncurses/tinfo/comp_hash.c ncurses-5.7/ncurses/tinfo/comp_hash.c
--- ncurses-5.7.orig//ncurses/tinfo/comp_hash.c	2011-01-08 17:28:07.569843002 +0100
+++ ncurses-5.7/ncurses/tinfo/comp_hash.c	2011-01-08 17:29:03.809843001 +0100
@@ -39,6 +39,7 @@
  */
 
 #define USE_TERMLIB 1
+#include <wchar.h>
 #include <curses.priv.h>
 
 #include <tic.h>
diff -Naur ncurses-5.7.orig//ncurses/tinfo/make_keys.c ncurses-5.7/ncurses/tinfo/make_keys.c
--- ncurses-5.7.orig//ncurses/tinfo/make_keys.c	2011-01-08 17:28:07.579843002 +0100
+++ ncurses-5.7/ncurses/tinfo/make_keys.c	2011-01-08 17:29:18.249843004 +0100
@@ -37,6 +37,7 @@
  */
 
 #define USE_TERMLIB 1
+#include <wchar.h>
 #include <build.priv.h>
 
 MODULE_ID("$Id: make_keys.c,v 1.19 2010/06/05 22:08:00 tom Exp $")
