diff -Naur nbench-2.2.3_orig//Makefile nbench-2.2.3/Makefile
--- nbench-2.2.3_orig//Makefile	2004-12-30 03:23:30.000000000 +0100
+++ nbench-2.2.3/Makefile	2012-07-13 14:48:51.724630492 +0200
@@ -19,10 +19,10 @@
 # You should leave -static in the CFLAGS so that your sysinfo can be
 # compiled into the executable.
 
-CC = gcc
+CC ?= gcc
 
 # generic options for gcc
-CFLAGS = -s -static -Wall -O3
+CFLAGS += -static
 
 # if your gcc lets you do it, then try this one
 #CFLAGS = -s -static -Wall -O3 -fomit-frame-pointer -funroll-loops
