--- a/configure.ac	2011-12-08 18:16:24.817503893 +0100
+++ b/configure.ac	2011-12-08 18:19:09.927746052 +0100
@@ -291,12 +291,6 @@
 	nouveau/libdrm_nouveau.pc
 	glamo/Makefile
 	glamo/libdrm_glamo.pc
-	tests/Makefile
-	tests/modeprint/Makefile
-	tests/modetest/Makefile
-	tests/kmstest/Makefile
-	tests/radeon/Makefile
-	tests/vbltest/Makefile
 	include/Makefile
 	include/drm/Makefile
 	libdrm.pc])
--- a/Makefile.am	2011-12-08 18:16:24.817503893 +0100
+++ b/Makefile.am	2011-12-08 18:18:24.167678937 +0100
@@ -45,7 +45,7 @@
 GLAMO_SUBDIR = glamo
 endif
 
-SUBDIRS = . $(LIBKMS_SUBDIR) $(INTEL_SUBDIR) $(NOUVEAU_SUBDIR) $(RADEON_SUBDIR) $(GLAMO_SUBDIR) tests include
+SUBDIRS = . $(LIBKMS_SUBDIR) $(INTEL_SUBDIR) $(NOUVEAU_SUBDIR) $(RADEON_SUBDIR) $(GLAMO_SUBDIR) include
 
 libdrm_la_LTLIBRARIES = libdrm.la
 libdrm_ladir = $(libdir)
