--- a/CMakeLists.txt	2016-03-12 18:34:05.205622814 +0100
+++ b/CMakeLists.txt	2016-03-12 18:34:05.221622815 +0100
@@ -17,19 +17,19 @@
   endif()
 
   add_options(CXX ALL_BUILDS -std=c++11)
-  find_package(OpenGL)
-  if(OPENGL_FOUND)
-    find_package(GLEW REQUIRED)
-    set(INCLUDES ${OpenGL_INCLUDE_DIR}
-                 ${GLEW_INCLUDE_DIR})
-    add_definitions(-DHAS_OPENGL)
-    set(DEPLIBS ${DEPLIBS} ${OPENGL_LIBRARIES} ${GLEW_LIBRARIES})
+  find_package(OpenGLES2)
+  if(OPENGLES2_FOUND)
+    set(INCLUDES ${OPENGLES2_INCLUDE_DIR})
+    add_definitions(-DHAS_GLES2 -DHAS_GLES)
+    set(DEPLIBS ${DEPLIBS} ${OPENGLES2_LIBRARIES})
   else()
-    find_package(OpenGLES2)
-    if(OPENGLES2_FOUND)
-      set(INCLUDES ${OPENGLES2_INCLUDE_DIR})
-      add_definitions(-DHAS_GLES2 -DHAS_GLES)
-      set(DEPLIBS ${DEPLIBS} ${OPENGLES2_LIBRARIES})
+    find_package(OpenGL)
+    if(OPENGL_FOUND)
+      find_package(GLEW REQUIRED)
+      set(INCLUDES ${OpenGL_INCLUDE_DIR}
+                   ${GLEW_INCLUDE_DIR})
+      add_definitions(-DHAS_OPENGL)
+      set(DEPLIBS ${DEPLIBS} ${OPENGL_LIBRARIES} ${GLEW_LIBRARIES})
     else()
       message(FATAL_ERROR "No rendering API found. Bailing")
     endif()
@@ -45,8 +45,6 @@
   set(SHADERTOY_SOURCES src/lodepng.cpp
                         src/main.cpp)
 
-  set(DEPLIBS ${DEPLIBS} ${OPENGL_LIBRARIES} ${GLEW_LIBRARIES})
-
 elseif(WIN32)
   include_directories(${KODI_INCLUDE_DIR}
                       ${platform_INCLUDE_DIRS}
