--- a/src/include/scsi_ioctls.h	2017-04-10 17:22:28.407697706 +0200
+++ b/src/include/scsi_ioctls.h	2017-04-10 17:22:28.419697556 +0200
@@ -34,8 +34,8 @@
 } Scsi_Idlun;
 
 
-inline int get_scsi_idlun(int fd, Scsi_Idlun *idlun);
-inline int get_scsi_pci(int fd, char *slot_name);
+int get_scsi_idlun(int fd, Scsi_Idlun *idlun);
+int get_scsi_pci(int fd, char *slot_name);
 int idlun_to_components (Scsi_Idlun *idlun,
 			 unsigned char *host,
 			 unsigned char *channel,
