diff -Naur webvideo-0.5.0_old/src/vdr-plugin/webvideo.c webvideo-0.5.0/src/vdr-plugin/webvideo.c
--- webvideo-0.5.0_old/src/vdr-plugin/webvideo.c	2013-04-14 18:38:02.000000000 +0200
+++ webvideo-0.5.0/src/vdr-plugin/webvideo.c	2013-10-28 00:16:20.235572000 +0100
@@ -150,7 +150,11 @@
 
   // default values if not given on the command line
   if ((const char *)destdir == NULL)
+#if APIVERSNUM > 20101
+    webvideoConfig->SetDownloadPath(cString(cVideoDirectory::Name()));
+#else
     webvideoConfig->SetDownloadPath(cString(VideoDirectory));
+#endif
   if ((const char *)conffile == NULL)
     conffile = AddDirectory(ConfigDirectory(Name()), "webvi.plugin.conf");
 
