--- old/y4mscaler.C	2005-12-11 21:27:03.000000000 +0100
+++ y4mscaler.C	2008-04-23 20:46:29.318734652 +0200
@@ -194,7 +194,7 @@
   if (source.read_stream_header(fd_in) != Y4M_OK)
     mjpeg_error_exit1("Failed to read YUV4MPEG2 header!");
   mjpeg_info("Input Stream Header:");
-  source.stream().log_info(LOG_INFO, "<<< ");
+  source.stream().log_info(mjpeg_loglev_t("info"), "<<< ");
 
   /* set target stream defaults from source stream */
   target.init_stream(source);
@@ -216,7 +216,7 @@
   /* set up target stream */
   target.stream().write_stream_header(fd_out);
   mjpeg_info("Output Stream Header:");
-  target.stream().log_info(LOG_INFO, ">>> ");
+  target.stream().log_info(mjpeg_loglev_t("info"), ">>> ");
 
   /* do some scaling */
   scaling.create_scalers(source, target);
