--- vdr-pin-0.1.10/pin.c.oirg	2013-03-04 15:35:52.233321246 +0100
+++ vdr-pin-0.1.10/pin.c	2013-02-20 21:56:38.000000000 +0100
@@ -512,7 +512,7 @@
 // Channel Switch
 //***************************************************************************
 
-void cPinStatusMonitor::ChannelSwitch(const cDevice *Device, int ChannelNumber)
+void cPinStatusMonitor::ChannelSwitch(const cDevice *Device, int ChannelNumber, bool LiveView)
 {
    // cChannel* channel = Channels.GetByNumber(ChannelNumber);
 }
--- vdr-pin-0.1.10/pin.h.org	2013-03-04 15:38:32.881324003 +0100
+++ vdr-pin-0.1.10/pin.h	2013-02-20 21:56:38.000000000 +0100
@@ -12,7 +12,9 @@
  * build  version  date       comment
  *
 
- * #36    0.1.11   24.04.11,  bugfix: i18n, stopp support vdr<1.6.0
+ * #38    0.1.13   10.12.12,  added:  patch for vdr 1.7.33
+ * #37    0.1.12   07.12.12,  added:  patch for vdr 1.7.32
+ * #36    0.1.11   31.03.11,  added:  patch for vdr 1.7.17
  * #35    0.1.10   20.07.08,  added:  italian translations (thanks to Gringo)
  * #34    0.1.9    19.05.07,  added:  setup option for log-level
                                       More frensh translations from pat, thanks!
@@ -137,24 +139,20 @@
 
 #include <vdr/status.h>
 #include <vdr/plugin.h>
-//#include <vdr/i18n.h>
+
 #include "locks.h"
 #include "def.h"
+//#include "i18n.h"
 #include "talk.h"
 
-//extern const tI18nPhrase Phrases[];
-
-#if defined(APIVERSNUM) && APIVERSNUM < 10600
-#error "VDR-1.6.0 API version or greater is required!"
-#endif
 
 //***************************************************************************
 // Constants
 //***************************************************************************
 
-static const char *VERSION        = "0.1.11";
-static const char *DESCRIPTION    = trNOOP("Childlock plugin");
-static const char *MAINMENUENTRY  = trNOOP("Childlock");
+static const char *VERSION        = "0.1.13";
+static const char *DESCRIPTION    = "Childlock plugin";
+static const char *MAINMENUENTRY  = tr("Childlock");
 
 #define PROTECTION_FILE "protection.fsk"
 // #define __DEBUG__
@@ -235,7 +233,7 @@
 
       // interface
 
-      virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
+      virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber, bool LiveView);
       virtual bool ChannelProtected(const cDevice *Device, const cChannel* Channel);
       virtual bool ReplayProtected(const cRecording* Recording, const char* Name, 
                                    const char* Base, bool isDirectory, int menuView = false);
