--- wpa-supplicant-2.3/src/drivers/drivers.mak.org      2014-10-20 09:38:29.366686961 +0200
+++ wpa-supplicant-2.3/src/drivers/drivers.mak  2014-10-20 09:39:00.406687486 +0200
@@ -35,7 +35,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 ifdef CONFIG_LIBNL3_ROUTE
   DRV_LIBS += -lnl-route-3
   DRV_CFLAGS += -DCONFIG_LIBNL3_ROUTE
--- wpa-supplicant-2.3/src/drivers/drivers.mk.org       2014-10-20 09:39:07.854687612 +0200
+++ wpa-supplicant-2.3/src/drivers/drivers.mk   2014-10-20 09:39:32.690688031 +0200
@@ -30,7 +30,7 @@
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 ifdef CONFIG_LIBNL3_ROUTE
   DRV_LIBS += -lnl-route-3
   DRV_CFLAGS += -DCONFIG_LIBNL3_ROUTE
