diff -Naur MPlayer-1.0pre8.orig/libvo/gtf.c MPlayer-1.0pre8/libvo/gtf.c
--- MPlayer-1.0pre8.orig/libvo/gtf.c	2006-11-12 15:39:38.000000000 +0100
+++ MPlayer-1.0pre8/libvo/gtf.c	2006-11-12 15:39:58.000000000 +0100
@@ -26,7 +26,7 @@
 static GTF_constants GTF_given_constants = { 3.0,550.0,1,8,1.8,8,40,20,128,600 };
 
 #ifndef HAVE_ROUND
-static double round(double v) 
+double round(double v) 
 { 
         return floor(v + 0.5); 
 } 
