* fix #include <linux/pci.h> for ppc target (linux 2.6).

diff -r 73fec1c7390a libdha/sysdep/pci_linux.c
--- a/libdha/sysdep/pci_linux.c	Sun Apr 30 16:54:30 2006 +0200
+++ b/libdha/sysdep/pci_linux.c	Sun Apr 30 16:54:30 2006 +0200
@@ -131,6 +131,7 @@ static int pci_config_type( void ) { ret
    mplayer-dev-eng@mplayerhq.hu, colin@colino.net*/
 #include <fcntl.h>
 //#include <sys/io.h>
+#define LINUX_MOD_DEVICETABLE_H
 #include <linux/pci.h>
 #include "../../bswap.h"
 
