diff -Naur MPlayer-r27054.ori/configure MPlayer-r27054/configure
--- MPlayer-r27054.ori/configure	2008-08-04 21:55:06.000000000 +0200
+++ MPlayer-r27054/configure	2008-08-04 21:55:44.000000000 +0200
@@ -140,7 +140,7 @@
 
 ppc() {
   case "$host_arch" in
-    ppc|powerpc) return 0;;
+    ppc|ppc64|powerpc|powerpc64) return 0;;
     *) return 1;;
   esac
 }
@@ -2005,7 +2009,7 @@
     _optimizing=''
     ;;
 
-  ppc|powerpc)
+  ppc|ppc64|powerpc|powerpc64)
     _arch='POWERPC PPC'
     _def_dcbzl='#undef HAVE_DCBZL'
     _target_arch='ARCH_POWERPC = yes'
