* backported from upstream

diff -Nur MPlayer-1.0rc1.orig/configure MPlayer-1.0rc1/configure
--- MPlayer-1.0rc1.orig/configure	2006-12-06 10:18:28.000000000 +0100
+++ MPlayer-1.0rc1/configure	2006-12-06 10:18:50.000000000 +0100
@@ -3808,8 +3808,6 @@
       break
     fi
   done
-  #FIXME: This is ugly as it can duplicate a -I parameter..
-  _inc_extra="$_inc_extra $_inc_x11"
   if test -z "$_inc_x11" ; then
     _x11=no
     _x11_headers="no"
@@ -3835,6 +3833,8 @@
   done
 fi
 if test "$_x11" = yes ; then
+  #FIXME: This is ugly as it can duplicate a -I parameter..
+  _inc_extra="$_inc_extra $_inc_x11"
   _def_x11='#define HAVE_X11 1'
   _vosrc="$_vosrc x11_common.c vo_x11.c vo_xover.c"
   _vomodules="x11 xover $_vomodules"
