* applied upstream

--- MPlayer/libvo/vo_vesa.c	2007-03-01 19:28:20.000000000 +0900
+++ MPlayer/libvo/vo_vesa.c	2007-03-01 19:33:34.000000000 +0900
@@ -621,7 +621,8 @@
 static int
 config(uint32_t width, uint32_t height, uint32_t d_width, uint32_t d_height, uint32_t flags, char *title, uint32_t format)
 {
-  struct VbeInfoBlock vib;  
+  static struct VbeInfoBlock vib;
+  static int vib_set = 0;
   struct VesaModeInfoBlock vmib;
   struct VesaCRTCInfoBlock crtc_pass;
   size_t i,num_modes;
@@ -650,12 +651,14 @@
 	} 
 	if((err=vbeInit()) != VBE_OK) { PRINT_VBE_ERR("vbeInit",err); return -1; }
 	memcpy(vib.VESASignature,"VBE2",4);
-	if((err=vbeGetControllerInfo(&vib)) != VBE_OK)
+	if(!vib_set && ((err=vbeGetControllerInfo(&vib)) != VBE_OK))
 	{
 	  PRINT_VBE_ERR("vbeGetControllerInfo",err);
 	  mp_msg(MSGT_VO,MSGL_ERR, MSGTR_LIBVO_VESA_PossibleReasonNoVbe2BiosFound);
 	  return -1;
 	}
+   else
+     vib_set = 1;
 	/* Print general info here */
 	mp_msg(MSGT_VO,MSGL_INFO, MSGTR_LIBVO_VESA_FoundVesaVbeBiosVersion,
 		(int)(vib.VESAVersion >> 8) & 0xff,
