/* already applied upstream */
diff -Naur MPlayer-1.0rc1.orig/libdha/oth/pci.db MPlayer-1.0rc1/libdha/oth/pci.db
--- MPlayer-1.0rc1.orig/libdha/oth/pci.db	2007-01-27 15:18:12.000000000 +0100
+++ MPlayer-1.0rc1/libdha/oth/pci.db	2007-01-27 15:19:22.000000000 +0100
@@ -279,7 +279,9 @@
 d	10024148	Radeon R350 AH [Radeon 9800]	0
 d	10024149	Radeon R350 AI [Radeon 9800]	0
 d	10024150	Radeon RV350 AP [Radeon 9600 Pro]	0
+d	10024151	Radeon RV350 AQ [Radeon 9600SE]	0
 d	10024170	Radeon RV350 AP [Radeon 9600 Pro] (Secondary)	0
+d	10024171	Radeon RV350 AP [Radeon 9600SE] (Secondary)	0
 d	10024152	Radeon RV350 AR [Radeon 9600XT]	0
 d	10024153	Radeon RV350 BK [Radeon 9550]	0
 d	10024168	Radeon R350 [Radeon 9800] (Secondary)	0
diff -Naur MPlayer-1.0rc1.orig/vidix/drivers/radeon_vid.c MPlayer-1.0rc1/vidix/drivers/radeon_vid.c
--- MPlayer-1.0rc1.orig/vidix/drivers/radeon_vid.c	2007-01-27 15:18:12.000000000 +0100
+++ MPlayer-1.0rc1/vidix/drivers/radeon_vid.c	2007-01-27 15:20:06.000000000 +0100
@@ -904,6 +904,7 @@
  DEVICE_ATI_RADEON_R300_AE,
  DEVICE_ATI_RADEON_R300_AF,
  DEVICE_ATI_RADEON_RV350_AP,
+ DEVICE_ATI_RADEON_RV350_AQ,
  DEVICE_ATI_RADEON_RV350_AR,
  DEVICE_ATI_RADEON_RV350_BK,
  DEVICE_ATI_RADEON_R350_AH,
@@ -1094,6 +1095,7 @@
             /* Radeon 9600/9800 */
             case DEVICE_ATI_RV370_5B60_RADEON:
             case DEVICE_ATI_RADEON_RV350_AP:
+            case DEVICE_ATI_RADEON_RV350_AQ:
             case DEVICE_ATI_RADEON_RV350_AR:
             case DEVICE_ATI_RADEON_RV350_BK:
             case DEVICE_ATI_RADEON_R350_NH:
