/* disable yuv4mpeg vo driver */
diff -r 974208590746 Makefile
--- a/Makefile	Sun Apr 30 16:54:29 2006 +0200
+++ b/Makefile	Sun Apr 30 16:54:29 2006 +0200
@@ -506,7 +506,6 @@ SRCS=aclib.c \
                libvo/video_out.c \
                libvo/vo_mpegpes.c \
                libvo/vo_null.c \
-               libvo/vo_yuv4mpeg.c \
                $(addprefix libvo/,$(VO_SRCS)) \
 
 SRCS_MPLAYER-$(APPLE_IR)     += input/appleir.c
diff -r 974208590746 libvo/video_out.c
--- a/libvo/video_out.c	Sun Apr 30 16:54:29 2006 +0200
+++ b/libvo/video_out.c	Sun Apr 30 16:54:29 2006 +0200
@@ -221,7 +221,6 @@ vo_functions_t* video_out_drivers[] =
         &video_out_xvmc,
 #endif
 	&video_out_mpegpes,
-	&video_out_yuv4mpeg,
 #ifdef HAVE_PNG
 	&video_out_png,
 #endif	
