diff -Naur MPlayer-1.0rc1.orig/configure MPlayer-1.0rc1/configure
--- MPlayer-1.0rc1.orig/configure	2006-10-23 00:32:31.000000000 +0200
+++ MPlayer-1.0rc1/configure	2006-12-17 16:02:00.000000000 +0100
@@ -3807,7 +3807,7 @@
 if test "$_directfb" = auto ; then
   _directfb=no
   cat > $TMPC <<EOF
-#include <directfb.h>
+#include <directfb/directfb.h>
 int main(void) { IDirectFB *foo; DirectFBInit(0,0); return 0; }
 EOF
   for _inc_tmp in "" -I/usr/local/include/directfb \
@@ -3823,7 +3823,7 @@
 
 if test "$_directfb" = yes; then
   cat > $TMPC << EOF
-#include <directfb_version.h>
+#include <directfb/directfb_version.h>
 int
 dfb_ver = DIRECTFB_MAJOR_VERSION.DIRECTFB_MINOR_VERSION.DIRECTFB_MICRO_VERSION
 ;
diff -Naur MPlayer-1.0rc1.orig/libvo/vo_dfbmga.c MPlayer-1.0rc1/libvo/vo_dfbmga.c
--- MPlayer-1.0rc1.orig/libvo/vo_dfbmga.c	2006-10-23 00:32:26.000000000 +0200
+++ MPlayer-1.0rc1/libvo/vo_dfbmga.c	2006-12-17 16:05:39.000000000 +0100
@@ -23,7 +23,7 @@
 */
 
 /* directfb includes */
-#include <directfb.h>
+#include <directfb/directfb.h>
 
 #define DFB_VERSION(a,b,c) (((a)<<16)|((b)<<8)|(c))
 
diff -Naur MPlayer-1.0rc1.orig/libvo/vo_directfb2.c MPlayer-1.0rc1/libvo/vo_directfb2.c
--- MPlayer-1.0rc1.orig/libvo/vo_directfb2.c	2006-10-23 00:32:26.000000000 +0200
+++ MPlayer-1.0rc1/libvo/vo_directfb2.c	2006-12-17 16:05:55.000000000 +0100
@@ -23,7 +23,7 @@
 
 // directfb includes
 
-#include <directfb.h>
+#include <directfb/directfb.h>
 
 #define DFB_VERSION(a,b,c) (((a)<<16)|((b)<<8)|(c))
 
