--- MPlayer/vidix/drivers/sis_vid.c	2006-10-29 10:00:16.000000000 +0800
+++ MPlayer/vidix/drivers/sis_vid.c	2006-10-29 10:03:39.000000000 +0800
@@ -297,6 +297,7 @@
           case DEVICE_SIS_650_VGA:
             /* M650 & 651 have 2 overlays */
             /* JCP: I think this works, but not really tested yet */
+            if (enable_app_io() == 0 )
             {
               unsigned char CR5F;
               unsigned char tempreg1, tempreg2;
@@ -318,6 +321,7 @@
               if (sis_has_two_overlays) 
                 if (sis_verbose > 0)
                   printf ("[SiS] detected M650/651 with 2 overlays\n");
+              disable_app_io();
             }
             sis_vga_engine = SIS_315_VGA;
             break;
@@ -348,6 +352,12 @@
     return (EINTR);
   }
 
+  if (enable_app_io() != 0)
+  {
+    printf("[SiS] can't enable register I/O\n");
+    return(EINTR);
+  }
+
   /* JCP: this is WRONG.  Need to coordinate w/ sisfb to use correct mem */
   /* map 16MB scary hack for now. */
   sis_mem_base = map_phys_mem(pci_info.base0, 0x1000000);
@@ -396,6 +406,7 @@
   /* unmap_phys_mem(sis_reg_base, 0x20000); */
   /* JCP: see above, hence also a hack. */
   unmap_phys_mem(sis_mem_base, 0x1000000);
+  disable_app_io();
 }
 
 int vixGetCapability(vidix_capability_t * to)
