--- MPlayer/vidix/drivers/sis_bridge.c	2007-02-09 11:54:59.000000000 +0900
+++ MPlayer/vidix/drivers/sis_bridge.c	2007-02-09 11:54:59.000000000 +0900
@@ -41,240 +41,159 @@
 }
 
 
-static int sis_do_sense(int tempbl, int tempbh, int tempcl, int tempch)
+static int sis_do_sense(int type, int test)
 {
-  int temp;
+  int temp, mytest, result, i, j;
 
-  outSISIDXREG(SISPART4, 0x11, tempbl);
-  temp = tempbh | tempcl;
-  setSISIDXREG(SISPART4, 0x10, 0xe0, temp);
-  sis_ddc2_delay(0x1000);
-  tempch &= 0x7f;
-  inSISIDXREG(SISPART4, 0x03, temp);
-  temp ^= 0x0e;
-  temp &= tempch;
-  return (temp == tempch);
+  for (j = 0; j < 10; j++)
+  {
+    result = 0;
+    for (i = 0; i < 3; i++)
+    {
+      mytest = test;
+      outSISIDXREG(SISPART4, 0x11, (type & 0x00ff));
+      temp = (type >> 8) | (mytest & 0x00ff);
+      setSISIDXREG(SISPART4, 0x10, 0xe0, temp);
+      sis_ddc2_delay(0x1500);
+      mytest >>= 8;
+      mytest &= 0x7f;
+      inSISIDXREG(SISPART4, 0x03, temp);
+      temp ^= 0x0e;
+      temp &= mytest;
+      if (temp == mytest)
+        result++;
+      outSISIDXREG(SISPART4, 0x11, 0x00);
+      andSISIDXREG(SISPART4, 0x10, 0xe0);
+      sis_ddc2_delay(0x1000);
+    }
+    if ((result == 0) || (result >= 2))
+      break;
+  }
+  return result;
 }
 

 /* sense connected devices on 30x bridge */
 static void sis_sense_30x(void)
 {
-  unsigned char backupP4_0d, backupP2_00, biosflag;
-  unsigned char testsvhs_tempbl, testsvhs_tempbh;
-  unsigned char testsvhs_tempcl, testsvhs_tempch;
-  unsigned char testcvbs_tempbl, testcvbs_tempbh;
-  unsigned char testcvbs_tempcl, testcvbs_tempch;
-  unsigned char testvga2_tempbl, testvga2_tempbh;
-  unsigned char testvga2_tempcl, testvga2_tempch;
-  int myflag, result = 0, i, j, haveresult;
-  unsigned short temp;
-
-  inSISIDXREG(SISPART4, 0x0d, backupP4_0d);
-  outSISIDXREG(SISPART4, 0x0d, (backupP4_0d | 0x04));
-
-  inSISIDXREG(SISPART2, 0x00, backupP2_00);
-  outSISIDXREG(SISPART2, 0x00, (backupP2_00 | 0x1c));
+  unsigned char backupP4_0d, backupP2_00, backupP2_4d, backupSR_1e;
+  unsigned char biosflag = 0;
+  unsigned short svhs = 0, svhs_c = 0;
+  unsigned short cvbs = 0, cvbs_c = 0;
+  unsigned short vga2 = 0, vga2_c = 0;
+  int myflag, result;
 
-  sis_do_sense(0, 0, 0, 0);
+  if (!(sis_vbflags & VB_SISBRIDGE))
+    return;
 
-  if (sis_vga_engine == SIS_300_VGA) 
+  if (sis_vbflags & VB_301)
   {
-    testvga2_tempbh = 0x00;
-    testvga2_tempbl = 0xd1;
-    testsvhs_tempbh = 0x00;
-    testsvhs_tempbl = 0xb9;
-    testcvbs_tempbh = 0x00;
-    testcvbs_tempbl = 0xb3;
-    biosflag = 0;
-
-    if (sis_vbflags & (VB_301B | VB_302B | VB_301LV | VB_302LV)) {
-      testvga2_tempbh = 0x01;
-      testvga2_tempbl = 0x90;
-      testsvhs_tempbh = 0x01;
-      testsvhs_tempbl = 0x6b;
-      testcvbs_tempbh = 0x01;
-      testcvbs_tempbl = 0x74;
-    }
-
+    svhs = 0x00b9; cvbs = 0x00b3; vga2 = 0x00d1;
     inSISIDXREG(SISPART4, 0x01, myflag);
-    if (myflag & 0x04) {
-      testvga2_tempbh = 0x00;
-      testvga2_tempbl = 0xfd;
-      testsvhs_tempbh = 0x00;
-      testsvhs_tempbl = 0xdd;
-      testcvbs_tempbh = 0x00;
-      testcvbs_tempbl = 0xee;
-    }
-    testvga2_tempch = 0x0e;
-    testvga2_tempcl = 0x08;
-    testsvhs_tempch = 0x06;
-    testsvhs_tempcl = 0x04;
-    testcvbs_tempch = 0x08;
-    testcvbs_tempcl = 0x04;
-
-    if (sis_device_id == DEVICE_SIS_300) {
-      inSISIDXREG(SISSR, 0x3b, myflag);
-      if (!(myflag & 0x01)) {
-        testvga2_tempbh = 0x00;
-        testvga2_tempbl = 0x00;
-        testvga2_tempch = 0x00;
-        testvga2_tempcl = 0x00;
-      }
+    if (myflag & 0x04)
+    {
+      svhs = 0x00dd; cvbs = 0x00ee; vga2 = 0x00fd;
     }
   }
+  else if (sis_vbflags & (VB_301B | VB_302B))
+  {
+    svhs = 0x016b; cvbs = 0x0174; vga2 = 0x0190;
+  }
+  else if (sis_vbflags & (VB_301LV | VB_302LV))
+  {
+    svhs = 0x0200; cvbs = 0x0100;
+  }
   else
+    return;
+
+  vga2_c = 0x0e08; svhs_c = 0x0404; cvbs_c = 0x0804;
+  if (sis_vbflags & (VB_301LV | VB_302LV))
   {
-    testvga2_tempbh = 0x00;
-    testvga2_tempbl = 0xd1;
-    testsvhs_tempbh = 0x00;
-    testsvhs_tempbl = 0xb9;
-    testcvbs_tempbh = 0x00;
-    testcvbs_tempbl = 0xb3;
-    biosflag = 0;
-  }
-
-  if (sis_vbflags & (VB_301B | VB_302B | VB_301LV | VB_302LV))
-  {
-    if (sis_vbflags & (VB_301B | VB_302B)) {
-      testvga2_tempbh = 0x01;
-      testvga2_tempbl = 0x90;
-      testsvhs_tempbh = 0x01;
-      testsvhs_tempbl = 0x6b;
-      testcvbs_tempbh = 0x01;
-      testcvbs_tempbl = 0x74;
-    } 
-    else {
-      testvga2_tempbh = 0x00;
-      testvga2_tempbl = 0x00;
-      testsvhs_tempbh = 0x02;
-      testsvhs_tempbl = 0x00;
-      testcvbs_tempbh = 0x01;
-      testcvbs_tempbl = 0x00;
-    }        
+    svhs_c = 0x0408; cvbs_c = 0x0808;
   }
+  biosflag = 2;
 
-  if (sis_vbflags & (VB_301 | VB_301B | VB_302B))
+  if (sis_vga_engine == SIS_300_VGA)
   {
-    inSISIDXREG(SISPART4, 0x01, myflag);
-    if (myflag & 0x04) {
-      testvga2_tempbh = 0x00;
-      testvga2_tempbl = 0xfd;
-      testsvhs_tempbh = 0x00;
-      testsvhs_tempbl = 0xdd;
-      testcvbs_tempbh = 0x00;
-      testcvbs_tempbl = 0xee;
-    }
-  }
-  if (sis_vbflags & (VB_301LV | VB_302LV)) 
-  {
-    /* TW: No VGA2 or SCART on LV bridges */
-    testvga2_tempbh = 0x00;
-    testvga2_tempbl = 0x00;
-    testvga2_tempch = 0x00;
-    testvga2_tempcl = 0x00;
-    testsvhs_tempch = 0x04;
-    testsvhs_tempcl = 0x08;
-    testcvbs_tempch = 0x08;
-    testcvbs_tempcl = 0x08;
-  } 
-  else 
+    inSISIDXREG(SISSR, 0x3b, myflag);
+    if (!(myflag & 0x01))
+      vga2 = vga2_c = 0;
+  }
+
+  if (!(sis_vbflags & VB_SISBRIDGE))
+    vga2 = vga2_c = 0;
+
+  inSISIDXREG(SISSR, 0x1e, backupSR_1e);
+  orSISIDXREG(SISSR, 0x1e, 0x20);
+  inSISIDXREG(SISPART4, 0x0d, backupP4_0d);
+
+  if (sis_vbflags & VB_301)
+    setSISIDXREG(SISPART4, 0x0d, ~0x07, 0x05);
+  else
+    orSISIDXREG(SISPART4, 0x0d, 0x04);
+
+  sis_ddc2_delay(0x2000);
+
+  inSISIDXREG(SISPART2, 0x00, backupP2_00);
+  outSISIDXREG(SISPART2, 0x00, ((backupP2_00 | 0x3c) & 0xfc));
+  inSISIDXREG(SISPART2, 0x4d, backupP2_4d);
+
+  sis_do_sense(0,0);
+
+  andSISIDXREG(SISCR, 0x32, ~0x14);
+
+  if (vga2_c || vga2)
   {
-    testvga2_tempch = 0x0e;
-    testvga2_tempcl = 0x08;
-    testsvhs_tempch = 0x06;
-    testsvhs_tempcl = 0x04;
-    testcvbs_tempch = 0x08;
-    testcvbs_tempcl = 0x04;
-  }
-
-  /* scan for VGA2/SCART */
-  if (testvga2_tempch || testvga2_tempcl ||
-    testvga2_tempbh || testvga2_tempbl)
-  {
-    haveresult = 0;
-    for (j = 0; j < 10; j++)
-    {
-      result = 0;
-      for (i = 0; i < 3; i++)
-        if (sis_do_sense(testvga2_tempbl, testvga2_tempbh,
-            testvga2_tempcl, testvga2_tempch))
-          result++;
-      if ((result == 0) || (result >= 2))
-        break;
-    }
-    if (result)
+    if (sis_do_sense(vga2, vga2_c)) 
     {
       if (biosflag & 0x01) 
       {
         if (sis_verbose > 1)
           printf("[SiS] SiS30x: Detected TV connected to SCART output\n");
         sis_vbflags |= TV_SCART;
         orSISIDXREG(SISCR, 0x32, 0x04);
       }
       else
       {
         if (sis_verbose > 1)
           printf("[SiS] SiS30x: Detected secondary VGA connection\n");
         sis_vbflags |= VGA2_CONNECTED;
         orSISIDXREG(SISCR, 0x32, 0x10);
       }
     }
   }
+  andSISIDXREG(SISCR, 0x32, 0x3f);
+  andSISIDXREG(SISCR, 0x32, ~0x03);
 
   /* scanning for TV */
-
-  result = sis_do_sense(testsvhs_tempbl, testsvhs_tempbh,
-              testsvhs_tempcl, testsvhs_tempch);
-
-  haveresult = 0;
-  for (j = 0; j < 10; j++)
-  {
-    result = 0;
-    for (i = 0; i < 3; i++)
-      if (sis_do_sense(testsvhs_tempbl, testsvhs_tempbh,
-          testsvhs_tempcl, testsvhs_tempch))
-        result++;
-    if ((result == 0) || (result >= 2))
-      break;
-  }
-  if (result)
+  if ((result = sis_do_sense(svhs, svhs_c)))
   {
     if (sis_verbose > 1)
       printf("[SiS] SiS30x: Detected TV connected to SVIDEO output\n");
     /* TW: So we can be sure that there IS a SVIDEO output */
     sis_vbflags |= TV_SVIDEO;
     orSISIDXREG(SISCR, 0x32, 0x02);
   }
 
-  if ((biosflag & 0x02) || (!(result)))
+  if ((biosflag & 0x02) || (!result))
   {
-    haveresult = 0;
-    for (j = 0; j < 10; j++)
-    {
-      result = 0;
-      for (i = 0; i < 3; i++)
-        if (sis_do_sense(testcvbs_tempbl, testcvbs_tempbh,
-            testcvbs_tempcl, testcvbs_tempch))
-          result++;
-      if ((result == 0) || (result >= 2))
-        break;
-    }
-    if (result)
+    if (sis_do_sense(cvbs, cvbs_c))
     {
       if (sis_verbose > 1)
         printf("[SiS] SiS30x: Detected TV connected to COMPOSITE output\n");
       sis_vbflags |= TV_AVIDEO;
       orSISIDXREG(SISCR, 0x32, 0x01);
     }
   }
 
-  sis_do_sense(0, 0, 0, 0);
+  sis_do_sense(0, 0);
+
   outSISIDXREG(SISPART2, 0x00, backupP2_00);
   outSISIDXREG(SISPART4, 0x0d, backupP4_0d);
+  outSISIDXREG(SISSR, 0x1e, backupSR_1e);
+  outSISIDXREG(SISPART2, 0x00, backupP2_00);
 }
 
-
 static void sis_detect_crt1(void)
 {
   unsigned char CR32;
