--- MPlayer-1.0rc1/vidix/drivers/sis_bridge.c.old	2007-02-10 01:21:02.000000000 +0900
+++ MPlayer-1.0rc1/vidix/drivers/sis_bridge.c	2007-02-10 01:42:36.000000000 +0900
@@ -406,23 +406,27 @@
   if (temp == 1)
   {
     inSISIDXREG(SISPART4, 0x01, temp1);
-    temp1 &= 0xff;
-    if (temp1 >= 0xE0)
+    temp1 &= 0xf0;
+    if (temp1 == 0xE0)
     {
+      inSISIDXREG(SISPART4, 0x39, temp2);
+      if (temp2 == 0xff)
+      {
       sis_vbflags |= VB_302LV;
       if (sis_verbose > 1)
         printf("[SiS] Detected SiS302LV video bridge (ID 1; Revision 0x%x)\n",
              temp1);
 
+      }
     } 
-    else if (temp1 >= 0xD0)
+    else if (temp1 == 0xD0)
     {
       sis_vbflags |= VB_301LV;
       if (sis_verbose > 1)
         printf("[SiS] Detected SiS301LV video bridge (ID 1; Revision 0x%x)\n",
              temp1);
     }
-    else if (temp1 >= 0xB0)
+    else if (temp1 == 0xB0)
     {
       sis_vbflags |= VB_301B;
       inSISIDXREG(SISPART4, 0x23, temp2);
@@ -432,7 +435,7 @@
         printf("[SiS] Detected SiS301B%s video bridge (Revision 0x%x)\n",
              (temp2 & 0x02) ? "" : " (DH)", temp1);
     }
-    else
+    else if (temp1 <= 0xA0)
     {
       sis_vbflags |= VB_301;
       if (sis_verbose > 1)
@@ -445,8 +448,8 @@
   else if (temp == 2)
   {
     inSISIDXREG(SISPART4, 0x01, temp1);
-    temp1 &= 0xff;
-    if (temp1 >= 0xE0)
+    temp1 &= 0xf0;
+    if (temp1 == 0xE0)
     {
       sis_vbflags |= VB_302LV;
       if (sis_verbose > 1)
@@ -454,14 +457,14 @@
              temp1);
       
     } 
-    else if (temp1 >= 0xD0)
+    else if (temp1 == 0xD0)
     {
       sis_vbflags |= VB_301LV;
       if (sis_verbose > 1)
         printf("[SiS] Detected SiS301LV video bridge (ID 2; Revision 0x%x)\n",
              temp1);
     }
-    else
+    else if (temp1 <= 0xC0)
     {
       sis_vbflags |= VB_302B;
       inSISIDXREG(SISPART4, 0x23, temp2);
