diff -Naur MPlayer-1.0rc1.orig/libvo/vo_ivtv.c MPlayer-1.0rc1/libvo/vo_ivtv.c
--- MPlayer-1.0rc1.orig/libvo/vo_ivtv.c	2007-02-26 20:01:08.000000000 +0100
+++ MPlayer-1.0rc1/libvo/vo_ivtv.c	2007-02-26 20:01:21.000000000 +0100
@@ -78,9 +78,11 @@
 {
   struct ivtv_cfg_stop_decode sd;
   struct ivtv_cfg_start_decode sd1;
+  int flags = 0;
 
-  sd.hide_last = blank_screen;
-  sd.pts_stop = 0;
+  if (blank_screen)
+    flags |= IVTV_STOP_FL_HIDE_FRAME;
+  sd.flags = flags;
  
   if (ioctl (ivtv_fd, IVTV_IOC_STOP_DECODE, &sd) < 0)
   {
