--- MPlayer-r25980.orig/configure       2008-03-20 16:41:45.000000000 +0100
+++ MPlayer-r25980/configure    2008-03-20 16:42:36.000000000 +0100
@@ -3875,6 +3875,7 @@
       break
     fi
   done
+ if test $_cross_compile = no; then
   for I in /usr/X11/include /usr/X11R6/include /usr/include/X11R6 /usr/openwin/include ; do
     if test -f "$I/X11/Xlib.h" ; then
       _inc_extra="$_inc_extra -I$I"
@@ -3883,6 +3884,7 @@
       break
     fi
   done
+ fi
 echores "$_x11_headers"
 
 

