diff -Naur MPlayer-1.0pre6.orig/configure MPlayer-1.0pre6/configure
--- MPlayer-1.0pre6.orig/configure	2004-12-30 19:31:43.348159480 +0100
+++ MPlayer-1.0pre6/configure	2004-12-30 19:37:18.203253768 +0100
@@ -4549,7 +4549,7 @@
 #include <alsa/asoundlib.h>
 int main(void) { return (!(SND_LIB_MAJOR==1 && SND_LIB_MINOR==0)); }
 EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='1.0.x-alsa'
+  cc_check -lasound $_ld_dl $_ld_pthread && _alsaver='1.0.x-alsa'
 fi
 _def_alsa5='#undef HAVE_ALSA5'
 _def_alsa9='#undef HAVE_ALSA9'
