--- old/mplayer.c	2006-05-28 12:34:59.000000000 +0800
+++ new/mplayer.c	2006-05-28 12:48:05.000000000 +0800
@@ -1622,13 +1624,16 @@
     case M_PROPERTY_SET:
         if(edl_muted) return M_PROPERTY_DISABLED;
         if(!arg) return 0;
-        if((!!*(int*)arg) != mixer.muted)
+        if((!!*(int*)arg) != mixer.muted) {
             mixer_mute(&mixer);
+            user_muted = !user_muted;
+        }
         return 1;
     case M_PROPERTY_STEP_UP:
     case M_PROPERTY_STEP_DOWN:
         if(edl_muted) return M_PROPERTY_DISABLED;
         mixer_mute(&mixer);
+        user_muted = !user_muted;
         return 1;
     case M_PROPERTY_PRINT:
         if(!arg) return 0;

