--- old/mplayer.c	2006-05-28 12:34:59.000000000 +0800
+++ new/mplayer.c	2006-05-28 12:48:05.000000000 +0800
@@ -437,6 +437,11 @@
 #define INITED_VCODEC  2048
 #define INITED_ALL 0xFFFF
 
+#include "mixer.h"
+mixer_t mixer;
+/// step size of mixer changes
+int volstep = 3;
+
 static void uninit_player(unsigned int mask){
   mask=inited_flags&mask;
 
@@ -520,6 +525,8 @@
   if(mask&INITED_AO){
     inited_flags&=~INITED_AO;
     current_module="uninit_ao";
+    if (user_muted | edl_muted)
+      mixer_mute(&mixer);
     audio_out->uninit(eof?0:1); audio_out=NULL;
   }
 
@@ -664,11 +671,6 @@
 
 extern void mp_input_register_options(m_config_t* cfg);
 
-#include "mixer.h"
-mixer_t mixer;
-/// step size of mixer changes
-int volstep = 3;
-
 #include "cfg-mplayer.h"
 
 void parse_cfgfiles( m_config_t* conf )
