Index: MPlayer-20051213/libao2/ao_alsa.c
===================================================================
--- MPlayer-20051213.orig/libao2/ao_alsa.c	2006-02-22 15:56:25.000000000 +0100
+++ MPlayer-20051213/libao2/ao_alsa.c	2006-02-22 16:01:39.000000000 +0100
@@ -558,10 +558,14 @@
 	if ((err = snd_pcm_hw_params_set_period_size_near(alsa_handler, alsa_hwparams, 
 							  &chunk_size, NULL)) < 0)
 	  {
-	    mp_msg(MSGT_AO,MSGL_ERR,"alsa-init: unable to set periodsize(%ld): %s\n",
-			    chunk_size, snd_strerror(err));
+	    chunk_size = 512 + 1024 - chunk_size;
+	    if ((err = snd_pcm_hw_params_set_period_size(alsa_handler, alsa_hwparams, 
+	                                                 chunk_size, 0)) < 0) {
+	      mp_msg(MSGT_AO,MSGL_ERR,"alsa-init: unable to set periodsize(%ld): %s\n",
+	      chunk_size, snd_strerror(err));
+	    }
 	  }
-	else {
+	if (err >= 0) {
 	  mp_msg(MSGT_AO,MSGL_V,"alsa-init: chunksize set to %li\n", chunk_size);
 	}
 	if ((err = snd_pcm_hw_params_set_periods_near(alsa_handler, alsa_hwparams,
