diff -Naur MPlayer-1.0pre6.orig/libao2/ao_alsa.c MPlayer-1.0pre6/libao2/ao_alsa.c
--- MPlayer-1.0pre6.orig/libao2/ao_alsa.c	2004-12-30 19:38:06.742874624 +0100
+++ MPlayer-1.0pre6/libao2/ao_alsa.c	2004-12-30 19:38:19.805888744 +0100
@@ -652,10 +652,14 @@
 	if ((err = snd_pcm_hw_params_set_period_size_near(alsa_handler, alsa_hwparams, 
 							  &chunk_size, &dir)) < 0)
 	  {
-	    mp_msg(MSGT_AO,MSGL_ERR,"alsa-init: unable to set periodsize(%d): %s\n",
-			    chunk_size, snd_strerror(err));
+            chunk_size = 512 + 1024 - chunk_size;
+            if ((err = snd_pcm_hw_params_set_period_size(alsa_handler, alsa_hwparams, 
+                                                         chunk_size, 0)) < 0) {
+              mp_msg(MSGT_AO,MSGL_ERR,"alsa-init: unable to set periodsize(%d): %s\n",
+                     chunk_size, snd_strerror(err));
+            }
 	  }
-	else {
+	if (err >= 0) {
 	  mp_msg(MSGT_AO,MSGL_V,"alsa-init: chunksize set to %i\n", chunk_size);
 	}
 	if ((err = snd_pcm_hw_params_set_periods_near(alsa_handler, alsa_hwparams,
