diff -r d4838c53105a parser-cfg.c
--- a/parser-cfg.c	Sun Apr 30 16:54:28 2006 +0200
+++ b/parser-cfg.c	Sun Apr 30 16:54:28 2006 +0200
@@ -39,7 +39,7 @@ int m_config_parse_config_file(m_config_
 #define PRINT_LINENUM	mp_msg(MSGT_CFGPARSER,MSGL_V,"%s(%d): ", conffile, line_num)
 #define MAX_LINE_LEN	10000
 #define MAX_OPT_LEN	1000
-#define MAX_PARAM_LEN	1000
+#define MAX_PARAM_LEN	1500
 	FILE *fp;
 	char *line;
 	char opt[MAX_OPT_LEN + 1];
