diff -r 9fdb997b3020 configure
--- a/configure	Sun Apr 30 16:54:28 2006 +0200
+++ b/configure	Sun Apr 30 16:54:29 2006 +0200
@@ -1660,6 +1660,7 @@ _inet6=auto
 _inet6=auto
 _gethostbyname2=auto
 _ftp=yes
+_additional_filters=yes
 _musepack=auto
 _vstream=auto
 _pthreads=yes
@@ -2057,6 +2058,8 @@ for ac_option do
   --disable-win32) _win32=no _dshow=no ;;
   --enable-dshow) _win32=yes _dshow=yes ;;
   --disable-dshow) _dshow=no ;;
+  --enable-additional-filters) _additional_filters=yes ;;
+  --disable-additional-filters) _additional_filters=no ;;
 
   --with-x11incdir=*)
     _inc_x11=-I`echo $ac_option | cut -d '=' -f 2 | sed 's,:, -I,g'`
@@ -6853,6 +6856,12 @@ fi
 fi
 echores "$_gethostbyname2"
 
+if test "$_additional_filters" = yes ; then
+  _def_additional_filters='#define USE_ADDITIONAL_FILTERS 1'
+else
+  _def_additional_filters='#undef USE_ADDITIONAL_FILTERS'
+fi
+
 # --------------- GUI specific tests begin -------------------
 echocheck "GUI"
 echo "$_gui"
@@ -7353,6 +7362,7 @@ CONFIG_LIBAVFORMAT_SO = $_libavformat_so
 CONFIG_LIBAVFORMAT_SO = $_libavformat_so
 CONFIG_LIBPOSTPROC = $_libpostproc
 CONFIG_LIBPOSTPROC_SO = $_libpostproc_so
+ADDITIONAL_FILTERS = $_additional_filters
 ZORAN = $_zr
 FAME = $_fame
 FAME_LIB = $_ld_fame
@@ -7737,6 +7747,9 @@ int setenv(const char *name, const char 
 /* DirectShow support */
 $_def_dshow
 
+/* Additionnal filters support */
+$_def_additional_filters
+
 /* Mac OS X specific features */
 $_def_macosx
 $_def_macosx_finder_support
diff -r 9fdb997b3020 libmpcodecs/Makefile
--- a/libmpcodecs/Makefile	Sun Apr 30 16:54:28 2006 +0200
+++ b/libmpcodecs/Makefile	Sun Apr 30 16:54:29 2006 +0200
@@ -86,12 +86,18 @@ VIDEO_SRCS=dec_video.c \
            $(VIDEO_SRCS_OPT) \
 
 VFILTER_SRCS=vf.c \
-             pullup.c \
+             vf_crop.c \
+             vf_expand.c \
+             vf_format.c \
+             vf_scale.c \
+             vf_vo.c \
+
+ifeq ($(ADDITIONAL_FILTERS),yes)
+VFILTER_SRCS+=pullup.c \
              vf_1bpp.c \
              vf_2xsai.c \
              vf_bmovl.c \
              vf_boxblur.c \
-             vf_crop.c \
              vf_cropdetect.c \
              vf_decimate.c \
              vf_delogo.c \
@@ -104,12 +110,10 @@ VFILTER_SRCS=vf.c \
              vf_dvbscale.c \
              vf_eq2.c \
              vf_eq.c \
-             vf_expand.c \
              vf_field.c \
              vf_fil.c \
              vf_filmdint.c \
              vf_flip.c \
-             vf_format.c \
              vf_framestep.c \
              vf_halfpack.c \
              vf_harddup.c \
@@ -133,7 +137,6 @@ VFILTER_SRCS=vf.c \
              vf_rgbtest.c \
              vf_rotate.c \
              vf_sab.c \
-             vf_scale.c \
              vf_smartblur.c \
              vf_softpulldown.c \
              vf_softskip.c \
@@ -144,11 +147,12 @@ VFILTER_SRCS=vf.c \
              vf_tile.c \
              vf_tinterlace.c \
              vf_unsharp.c \
-             vf_vo.c \
              vf_yuvcsp.c \
              vf_yuy2.c \
              vf_yvu9.c \
              vf_screenshot.c \
+
+endif
 
 VFILTER_LAVC_SRCS += vf_lavc.c \
                      vf_lavcdeint.c \
diff -r 9fdb997b3020 libmpcodecs/vf.c
--- a/libmpcodecs/vf.c	Sun Apr 30 16:54:28 2006 +0200
+++ b/libmpcodecs/vf.c	Sun Apr 30 16:54:29 2006 +0200
@@ -103,22 +103,26 @@ extern vf_info_t vf_info_screenshot;
 
 // list of available filters:
 static vf_info_t* filter_list[]={
-    &vf_info_rectangle,
-#ifndef HAVE_NO_POSIX_SELECT
-    &vf_info_bmovl,
-#endif
     &vf_info_crop,
     &vf_info_expand,
 #if defined(USE_LIBPOSTPROC) || defined(USE_LIBPOSTPROC_SO)
     &vf_info_pp,
 #endif
     &vf_info_scale,
+#ifdef USE_LIBAVCODEC
+    &vf_info_lavc,
+#endif
 //    &vf_info_osd,
     &vf_info_vo,
+    &vf_info_format,
+#ifdef USE_ADDITIONAL_FILTERS
+    &vf_info_rectangle,
+#ifndef HAVE_NO_POSIX_SELECT
+    &vf_info_bmovl,
+#endif
 #ifdef USE_LIBFAME
     &vf_info_fame,
 #endif
-    &vf_info_format,
     &vf_info_noformat,
     &vf_info_yuy2,
     &vf_info_flip,
@@ -127,7 +131,6 @@ static vf_info_t* filter_list[]={
     &vf_info_mirror,
     &vf_info_palette,
 #ifdef USE_LIBAVCODEC
-    &vf_info_lavc,
     &vf_info_lavcdeint,
 #endif
 #ifdef HAVE_ZR
@@ -190,6 +193,7 @@ static vf_info_t* filter_list[]={
     &vf_info_softskip,
 #ifdef HAVE_PNG
     &vf_info_screenshot,
+#endif
 #endif
     NULL
 };
