* commited on mplayerhq.hu (20050415)

diff -Naur MPlayer-1.0pre6.orig/Makefile MPlayer-1.0pre6/Makefile
--- MPlayer-1.0pre6.orig/Makefile	2005-02-04 02:28:04.000000000 +0100
+++ MPlayer-1.0pre6/Makefile	2005-02-04 02:28:30.000000000 +0100
@@ -267,7 +267,7 @@
 $(MENCODER_DEP): version.h help_mp.h
 
 $(PRG_CFG): version.h codec-cfg.c codec-cfg.h
-	$(HOST_CC) $(CFLAGS) -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(I18NLIBS)
+	$(HOST_CC) $(HOST_CFLAGS) -I. -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(I18NLIBS)
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
