diff -Naur MPlayer-1.0pre7.orig/parser-cfg.c MPlayer-1.0pre7/parser-cfg.c
--- MPlayer-1.0pre7.orig/parser-cfg.c	2004-11-15 17:09:27.000000000 +0800
+++ MPlayer-1.0pre7/parser-cfg.c	2005-11-23 21:54:00.000000000 +0800
@@ -23,7 +23,7 @@
 #define PRINT_LINENUM	mp_msg(MSGT_CFGPARSER,MSGL_V,"%s(%d): ", conffile, line_num)
 #define MAX_LINE_LEN	10000
 #define MAX_OPT_LEN	1000
-#define MAX_PARAM_LEN	1000
+#define MAX_PARAM_LEN	1500
 	FILE *fp;
 	char *line;
 	char opt[MAX_OPT_LEN + 1];
