diff -Naur MPlayer-1.0pre6.orig/configure MPlayer-1.0pre6/configure
--- MPlayer-1.0pre6.orig/configure	2005-02-02 00:25:29.709304472 +0100
+++ MPlayer-1.0pre6/configure	2005-02-02 00:26:04.723981432 +0100
@@ -5264,7 +5264,10 @@
     _faad_external=no
     cat > $TMPC << EOF
 #include <faad.h>
-int main(void) { faacDecHandle testhand; faacDecFrameInfo testinfo; testhand = faacDecOpen(); return 0; }
+#ifndef FAAD_MIN_STREAMSIZE
+#error Too old version
+#endif
+int main(void) { faacDecHandle testhand; faacDecFrameInfo testinfo; testhand = faacDecOpen(); faacDecInit(NULL, NULL, 0, NULL, NULL); return 0; }
 EOF
     cc_check $_inc_faad $_ld_faad $_ld_lm && _faad_external=yes
   fi
@@ -5273,34 +5276,6 @@
   echores "no"
 fi
 
-if test "$_faad_internal" = yes -o "$_faad_external" = yes; then
-echocheck "FAAD2 version"
-cat > $TMPC <<EOF
-#include <faad.h>
-#ifndef FAAD_MIN_STREAMSIZE
-#error Too old version
-#endif
-int main(void) {
-#ifdef FAAD2_VERSION
-  printf("%s\n",FAAD2_VERSION);
-#else
-  printf("1.0\n");
-#endif
-  return 0;
-}
-EOF
-  if cc_check -I- $_inc_faad $_ld_faad $_ld_lm && "$TMPO" >> "$TMPLOG" ; then
-    _faad_version=`"$TMPO"`
-    _faad_tempversion=`"$TMPO" | sed -e 's/^\([0-9]\{1,\}\)\.\([0-9]\{1,\}\).*/\1\2/'`
-    _def_faad_version="#define FAADVERSION $_faad_tempversion"
-    echores "$_faad_version"
-  else
-    _faad_external=no
-    _faad_internal=no
-    echores "failed to get version"
-  fi
-fi
-
 if test "$_faad_external" = yes; then
   _def_faad='#define HAVE_FAAD 1'
   _codecmodules="faad2(external) $_codecmodules"
@@ -7209,7 +7184,6 @@
 /* enable FAAD (AAC) support */
 $_def_faad
 $_def_faad_internal
-$_def_faad_version
 
 /* enable LADSPA plugin support */
 $_def_ladspa
diff -Naur MPlayer-1.0pre6.orig/libmpcodecs/ad_faad.c MPlayer-1.0pre6/libmpcodecs/ad_faad.c
--- MPlayer-1.0pre6.orig/libmpcodecs/ad_faad.c	2005-02-02 00:25:31.282065376 +0100
+++ MPlayer-1.0pre6/libmpcodecs/ad_faad.c	2005-02-02 00:26:04.835964408 +0100
@@ -94,13 +94,8 @@
     sh->a_in_buffer_len = demux_read_data(sh->ds, sh->a_in_buffer, sh->a_in_buffer_size);
 
     /* init the codec */
-#if (FAADVERSION <= 11)
-    faac_init = faacDecInit(faac_hdec, sh->a_in_buffer,
-       &faac_samplerate, &faac_channels);
-#else
     faac_init = faacDecInit(faac_hdec, sh->a_in_buffer,
        sh->a_in_buffer_len, &faac_samplerate, &faac_channels);
-#endif
 
     sh->a_in_buffer_len -= (faac_init > 0)?faac_init:0; // how many bytes init consumed
     // XXX FIXME: shouldn't we memcpy() here in a_in_buffer ?? --A'rpi
@@ -179,11 +174,7 @@
   if(!sh->codecdata_len){
    // raw aac stream:
    do {
-#if (FAADVERSION <= 11)
-    faac_sample_buffer = faacDecDecode(faac_hdec, &faac_finfo, sh->a_in_buffer+j);
-#else
     faac_sample_buffer = faacDecDecode(faac_hdec, &faac_finfo, sh->a_in_buffer+j, sh->a_in_buffer_len);
-#endif
 	
     /* update buffer index after faacDecDecode */
     if(faac_finfo.bytesconsumed >= sh->a_in_buffer_len) {
@@ -205,11 +196,7 @@
     unsigned char* bufptr=NULL;
     int buflen=ds_get_packet(sh->ds, &bufptr);
     if(buflen<=0) break;
-#if (FAADVERSION <= 11)
-    faac_sample_buffer = faacDecDecode(faac_hdec, &faac_finfo, bufptr);
-#else
     faac_sample_buffer = faacDecDecode(faac_hdec, &faac_finfo, bufptr, buflen);
-#endif
 //    printf("FAAC decoded %d of %d  (err: %d)  \n",faac_finfo.bytesconsumed,buflen,faac_finfo.error);
   }
   
