diff -Naur MPlayer-20060519.orig/libmpdemux/realrtsp/rtsp.c MPlayer-20060519/libmpdemux/realrtsp/rtsp.c
--- MPlayer-20060519.orig/libmpdemux/realrtsp/rtsp.c	2006-06-08 20:32:18.000000000 +0200
+++ MPlayer-20060519/libmpdemux/realrtsp/rtsp.c	2006-06-08 20:33:24.000000000 +0200
@@ -906,7 +906,8 @@
 
 static int open_s(stream_t *stream,int mode, void* opts, int* file_format) {
   URL_t *url;
-
+  extern int index_mode;
+  
   mp_msg(MSGT_OPEN, MSGL_INFO, "STREAM_RTSP, URL: %s\n", stream->url);
   stream->streaming_ctrl = streaming_ctrl_new();
   if( stream->streaming_ctrl==NULL )
@@ -918,6 +919,7 @@
   //url_free(url);
 
   stream->fd = -1;
+  index_mode = -1;
   if(realrtsp_streaming_start( stream ) < 0) {
     streaming_ctrl_free(stream->streaming_ctrl);
     stream->streaming_ctrl = NULL;
