diff -Naur MPlayer-20050915.orig/configure MPlayer-20050915/configure
--- MPlayer-20050915.orig/configure	2005-09-15 23:24:55.000000000 +0200
+++ MPlayer-20050915/configure	2005-09-15 23:25:05.000000000 +0200
@@ -1487,6 +1487,7 @@
 _inet6=auto
 _gethostbyname2=auto
 _ftp=yes
+_additional_filters=yes
 _musepack=auto
 _vstream=auto
 _pthreads=yes
@@ -1876,6 +1877,8 @@
   --disable-win32) _win32=no _dshow=no ;;
   --enable-dshow) _win32=yes _dshow=yes ;;
   --disable-dshow) _dshow=no ;;
+  --enable-additional-filters) _additional_filters=yes ;;
+  --disable-additional-filters) _additional_filters=no ;;
 
   --with-x11incdir=*)
     _inc_x11=-I`echo $ac_option | cut -d '=' -f 2 | sed 's,:, -I,g'`
@@ -6557,6 +6560,12 @@
 fi
 echores "$_gethostbyname2"
 
+if test "$_additional_filters" = yes ; then
+  _def_additional_filters='#define USE_ADDITIONAL_FILTERS 1'
+else
+  _def_additional_filters='#undef USE_ADDITIONAL_FILTERS'
+fi
+
 # --------------- GUI specific tests begin -------------------
 echocheck "GUI"
 echo "$_gui"
@@ -7029,6 +7038,7 @@
 CONFIG_LIBAVCODEC = $_libavcodec
 CONFIG_LIBAVCODECSO = $_libavcodecso
 CONFIG_LIBAVFORMAT = $_libavformat
+ADDITIONAL_FILTERS = $_additional_filters
 ZORAN = $_zr
 FAME = $_fame
 FAME_LIB = $_ld_fame
@@ -7389,6 +7399,9 @@
 /* DirectShow support */
 $_def_dshow
 
+/* Additionnal filters support */
+$_def_additional_filters
+
 /* Mac OS X specific features */
 $_def_macosx
 $_def_macosx_finder_support
diff -Naur MPlayer-20050915.orig/libmpcodecs/Makefile MPlayer-20050915/libmpcodecs/Makefile
--- MPlayer-20050915.orig/libmpcodecs/Makefile	2005-09-15 23:24:55.000000000 +0200
+++ MPlayer-20050915/libmpcodecs/Makefile	2005-09-15 23:27:48.000000000 +0200
@@ -61,79 +61,83 @@
 VIDEO_SRCS=dec_video.c vd.c $(VIDEO_SRCS_NAT) $(VIDEO_SRCS_LIB) $(VIDEO_SRCS_OPT)
 
 VFILTER_SRCS=vf.c \
-             pullup.c \
-             vf_1bpp.c \
-             vf_2xsai.c \
-             vf_bmovl.c \
-             vf_boxblur.c \
              vf_crop.c \
-             vf_cropdetect.c \
-             vf_decimate.c \
-             vf_delogo.c \
-             vf_denoise3d.c \
-             vf_detc.c \
-             vf_dint.c \
-             vf_divtc.c \
-             vf_down3dright.c \
-             vf_dsize.c \
-             vf_dvbscale.c \
-             vf_eq2.c \
-             vf_eq.c \
              vf_expand.c \
-             vf_field.c \
-             vf_fil.c \
-             vf_filmdint.c \
-             vf_flip.c \
              vf_format.c \
-             vf_framestep.c \
-             vf_fspp.c \
-             vf_halfpack.c \
-             vf_harddup.c \
-             vf_hqdn3d.c \
-             vf_hue.c \
-             vf_il.c \
-             vf_ilpack.c \
-             vf_ivtc.c \
-             vf_kerndeint.c \
              vf_lavc.c \
-             vf_lavcdeint.c \
-             vf_mirror.c \
-             vf_noformat.c \
-             vf_noise.c \
-             vf_palette.c \
-             vf_perspective.c \
-             vf_phase.c \
-             vf_pp7.c \
-             vf_pullup.c \
-             vf_qp.c \
-             vf_rectangle.c \
-             vf_remove_logo.c \
-             vf_rgb2bgr.c \
-             vf_rgbtest.c \
-             vf_rotate.c \
-             vf_sab.c \
              vf_scale.c \
-             vf_smartblur.c \
-             vf_softpulldown.c \
-             vf_softskip.c \
-             vf_spp.c \
-             vf_swapuv.c \
-             vf_telecine.c \
-             vf_test.c \
-             vf_tfields.c \
-             vf_tile.c \
-             vf_tinterlace.c \
-             vf_unsharp.c \
              vf_vo.c \
-             vf_yuvcsp.c \
-             vf_yuy2.c \
-             vf_yvu9.c \
-             vf_screenshot.c \
 
 ifeq ($(HAVE_FFPOSTPROCESS),yes)
 VFILTER_SRCS += vf_pp.c
 endif
 
+ifeq ($(ADDITIONAL_FILTERS),yes)
+VFILTER_SRCS+=pullup.c \
+              vf_1bpp.c \
+              vf_2xsai.c \
+              vf_bmovl.c \
+              vf_boxblur.c \
+              vf_cropdetect.c \
+              vf_decimate.c \
+              vf_delogo.c \
+              vf_denoise3d.c \
+              vf_detc.c \
+              vf_dint.c \
+              vf_divtc.c \
+              vf_down3dright.c \
+              vf_dsize.c \
+              vf_dvbscale.c \
+              vf_eq2.c \
+              vf_eq.c \
+              vf_field.c \
+              vf_fil.c \
+              vf_filmdint.c \
+              vf_flip.c \
+              vf_framestep.c \
+              vf_fspp.c \
+              vf_halfpack.c \
+              vf_harddup.c \
+              vf_hqdn3d.c \
+              vf_hue.c \
+              vf_il.c \
+              vf_ilpack.c \
+              vf_ivtc.c \
+              vf_kerndeint.c \
+              vf_lavcdeint.c \
+              vf_mirror.c \
+              vf_noformat.c \
+              vf_noise.c \
+              vf_palette.c \
+              vf_perspective.c \
+              vf_phase.c \
+              vf_pp7.c \
+              vf_pullup.c \
+              vf_qp.c \
+              vf_rectangle.c \
+              vf_remove_logo.c \
+              vf_rgb2bgr.c \
+              vf_rgbtest.c \
+              vf_rotate.c \
+              vf_sab.c \
+              vf_smartblur.c \
+              vf_softpulldown.c \
+              vf_softskip.c \
+              vf_spp.c \
+              vf_swapuv.c \
+              vf_telecine.c \
+              vf_test.c \
+              vf_tfields.c \
+              vf_tile.c \
+              vf_tinterlace.c \
+              vf_unsharp.c \
+              vf_yuvcsp.c \
+              vf_yuy2.c \
+              vf_yvu9.c \
+              vf_screenshot.c \
+
+endif
+
 ENCODER_SRCS=ae.c \
              ae_pcm.c \
              ve.c \
diff -Naur MPlayer-20050915.orig/libmpcodecs/vf.c MPlayer-20050915/libmpcodecs/vf.c
--- MPlayer-20050915.orig/libmpcodecs/vf.c	2005-09-15 23:24:55.000000000 +0200
+++ MPlayer-20050915/libmpcodecs/vf.c	2005-09-15 23:28:15.000000000 +0200
@@ -24,16 +24,17 @@
 #include "../libvo/fastmemcpy.h"
 
 extern vf_info_t vf_info_vo;
-extern vf_info_t vf_info_rectangle;
-#ifndef HAVE_NO_POSIX_SELECT
-extern vf_info_t vf_info_bmovl;
-#endif
 extern vf_info_t vf_info_crop;
 extern vf_info_t vf_info_expand;
 #ifdef FF_POSTPROCESS
 extern vf_info_t vf_info_pp;
 #endif
 extern vf_info_t vf_info_scale;
+extern vf_info_t vf_info_lavc;
+extern vf_info_t vf_info_rectangle;
+#ifndef HAVE_NO_POSIX_SELECT
+extern vf_info_t vf_info_bmovl;
+#endif
 #ifdef USE_LIBFAME
 extern vf_info_t vf_info_fame;
 #endif
@@ -45,7 +46,6 @@
 extern vf_info_t vf_info_rotate;
 extern vf_info_t vf_info_mirror;
 extern vf_info_t vf_info_palette;
-extern vf_info_t vf_info_lavc;
 extern vf_info_t vf_info_zrmjpeg;
 extern vf_info_t vf_info_dvbscale;
 extern vf_info_t vf_info_cropdetect;
@@ -102,22 +102,26 @@
 
 // list of available filters:
 static vf_info_t* filter_list[]={
-    &vf_info_rectangle,
-#ifndef HAVE_NO_POSIX_SELECT
-    &vf_info_bmovl,
-#endif
     &vf_info_crop,
     &vf_info_expand,
 #ifdef FF_POSTPROCESS
     &vf_info_pp,
 #endif
     &vf_info_scale,
+#ifdef USE_LIBAVCODEC
+    &vf_info_lavc,
+#endif
 //    &vf_info_osd,
     &vf_info_vo,
+    &vf_info_format,
+#ifdef USE_ADDITIONAL_FILTERS
+    &vf_info_rectangle,
+#ifndef HAVE_NO_POSIX_SELECT
+    &vf_info_bmovl,
+#endif
 #ifdef USE_LIBFAME
     &vf_info_fame,
 #endif
-    &vf_info_format,
     &vf_info_noformat,
     &vf_info_yuy2,
     &vf_info_flip,
@@ -126,7 +130,6 @@
     &vf_info_mirror,
     &vf_info_palette,
 #ifdef USE_LIBAVCODEC
-    &vf_info_lavc,
     &vf_info_lavcdeint,
 #endif
 #ifdef HAVE_ZR
@@ -189,6 +192,7 @@
 #ifdef HAVE_PNG
     &vf_info_screenshot,
 #endif
+#endif
     NULL
 };
 
