Index: MPlayer-20051213/configure
===================================================================
--- MPlayer-20051213.orig/configure	2006-02-22 15:56:20.000000000 +0100
+++ MPlayer-20051213/configure	2006-02-22 16:04:47.000000000 +0100
@@ -1607,6 +1607,7 @@
 _inet6=auto
 _gethostbyname2=auto
 _ftp=yes
+_additional_filters=yes
 _musepack=auto
 _vstream=auto
 _pthreads=yes
@@ -2012,6 +2013,8 @@
   --disable-win32) _win32=no _dshow=no ;;
   --enable-dshow) _win32=yes _dshow=yes ;;
   --disable-dshow) _dshow=no ;;
+  --enable-additional-filters) _additional_filters=yes ;;
+  --disable-additional-filters) _additional_filters=no ;;
 
   --with-x11incdir=*)
     _inc_x11=-I`echo $ac_option | cut -d '=' -f 2 | sed 's,:, -I,g'`
@@ -6830,6 +6833,12 @@
 fi
 echores "$_gethostbyname2"
 
+if test "$_additional_filters" = yes ; then
+  _def_additional_filters='#define USE_ADDITIONAL_FILTERS 1'
+else
+  _def_additional_filters='#undef USE_ADDITIONAL_FILTERS'
+fi
+
 # --------------- GUI specific tests begin -------------------
 echocheck "GUI"
 echo "$_gui"
@@ -7310,6 +7319,7 @@
 CONFIG_LIBAVFORMAT_SO = $_libavformat_so
 CONFIG_LIBPOSTPROC = $_libpostproc
 CONFIG_LIBPOSTPROC_SO = $_libpostproc_so
+ADDITIONAL_FILTERS = $_additional_filters
 ZORAN = $_zr
 FAME = $_fame
 FAME_LIB = $_ld_fame
@@ -7689,6 +7699,9 @@
 /* DirectShow support */
 $_def_dshow
 
+/* Additionnal filters support */
+$_def_additional_filters
+
 /* Mac OS X specific features */
 $_def_macosx
 $_def_macosx_finder_support
Index: MPlayer-20051213/libmpcodecs/Makefile
===================================================================
--- MPlayer-20051213.orig/libmpcodecs/Makefile	2006-02-22 15:56:26.000000000 +0100
+++ MPlayer-20051213/libmpcodecs/Makefile	2006-02-22 16:09:19.000000000 +0100
@@ -86,12 +86,18 @@
            $(VIDEO_SRCS_OPT) \
 
 VFILTER_SRCS=vf.c \
-             pullup.c \
+             vf_crop.c \
+             vf_expand.c \
+             vf_format.c \
+             vf_scale.c \
+             vf_vo.c \
+
+ifeq ($(ADDITIONAL_FILTERS),yes)
+VFILTER_SRCS+=pullup.c \
              vf_1bpp.c \
              vf_2xsai.c \
              vf_bmovl.c \
              vf_boxblur.c \
-             vf_crop.c \
              vf_cropdetect.c \
              vf_decimate.c \
              vf_delogo.c \
@@ -104,12 +110,10 @@
              vf_dvbscale.c \
              vf_eq2.c \
              vf_eq.c \
-             vf_expand.c \
              vf_field.c \
              vf_fil.c \
              vf_filmdint.c \
              vf_flip.c \
-             vf_format.c \
              vf_framestep.c \
              vf_halfpack.c \
              vf_harddup.c \
@@ -133,7 +137,6 @@
              vf_rgbtest.c \
              vf_rotate.c \
              vf_sab.c \
-             vf_scale.c \
              vf_smartblur.c \
              vf_softpulldown.c \
              vf_softskip.c \
@@ -144,11 +147,12 @@
              vf_tile.c \
              vf_tinterlace.c \
              vf_unsharp.c \
-             vf_vo.c \
              vf_yuvcsp.c \
              vf_yuy2.c \
              vf_yvu9.c \
              vf_screenshot.c \
+
+endif
 
 VFILTER_LAVC_SRCS += vf_uspp.c \
                      vf_fspp.c \
Index: MPlayer-20051213/libmpcodecs/vf.c
===================================================================
--- MPlayer-20051213.orig/libmpcodecs/vf.c	2006-02-22 15:56:26.000000000 +0100
+++ MPlayer-20051213/libmpcodecs/vf.c	2006-02-22 16:17:18.000000000 +0100
@@ -103,22 +103,26 @@
 
 // list of available filters:
 static vf_info_t* filter_list[]={
-    &vf_info_rectangle,
-#ifndef HAVE_NO_POSIX_SELECT
-    &vf_info_bmovl,
-#endif
     &vf_info_crop,
     &vf_info_expand,
 #if defined(USE_LIBPOSTPROC) || defined(USE_LIBPOSTPROC_SO)
     &vf_info_pp,
 #endif
     &vf_info_scale,
+#ifdef USE_LIBAVCODEC
+    &vf_info_lavc,
+#endif
 //    &vf_info_osd,
     &vf_info_vo,
+    &vf_info_format,
+#ifdef USE_ADDITIONAL_FILTERS
+    &vf_info_rectangle,
+#ifndef HAVE_NO_POSIX_SELECT
+    &vf_info_bmovl,
+#endif
 #ifdef USE_LIBFAME
     &vf_info_fame,
 #endif
-    &vf_info_format,
     &vf_info_noformat,
     &vf_info_yuy2,
     &vf_info_flip,
@@ -127,7 +131,6 @@
     &vf_info_mirror,
     &vf_info_palette,
 #ifdef USE_LIBAVCODEC
-    &vf_info_lavc,
     &vf_info_lavcdeint,
 #endif
 #ifdef HAVE_ZR
@@ -191,6 +194,7 @@
 #ifdef HAVE_PNG
     &vf_info_screenshot,
 #endif
+#endif
     NULL
 };
 
