--- MPlayer-1.0rc4/libmpdemux/demux_lavf.c.orig	2011-11-28 13:39:16.831448863 +0200
+++ MPlayer-1.0rc4/libmpdemux/demux_lavf.c	2011-11-28 13:39:38.475453370 +0200
@@ -140,7 +140,7 @@
 static void list_formats(void) {
     AVInputFormat *fmt;
     mp_msg(MSGT_DEMUX, MSGL_INFO, "Available lavf input formats:\n");
-    for (fmt = first_iformat; fmt; fmt = fmt->next)
+    for (fmt = av_iformat_next(NULL); fmt; av_iformat_next(fmt))
         mp_msg(MSGT_DEMUX, MSGL_INFO, "%15s : %s\n", fmt->name, fmt->long_name);
 }
 
