diff -r 09daeb8bedd2 input/joystick.c
--- a/input/joystick.c	Sun Apr 30 16:54:32 2006 +0200
+++ b/input/joystick.c	Sun Apr 30 16:54:32 2006 +0200
@@ -71,7 +71,6 @@ int mp_input_joystick_init(char* dev) {
 	mp_msg(MSGT_INPUT,MSGL_WARN,MSGTR_INPUT_JOYSTICK_LoosingBytes,l);	  
       break;
     }
-    ev.type &= ~JS_EVENT_INIT;
     if(ev.type == JS_EVENT_BUTTON)
       btns |= (ev.value << ev.number);
     if(ev.type == JS_EVENT_AXIS)
