diff -Naur MPlayer-20050806.orig/libvo/Makefile MPlayer-20050806/libvo/Makefile
--- MPlayer-20050806.orig/libvo/Makefile	2005-09-14 22:58:02.000000000 +0200
+++ MPlayer-20050806/libvo/Makefile	2005-09-14 23:24:04.000000000 +0200
@@ -15,7 +15,6 @@
      video_out.c \
      vo_mpegpes.c \
      vo_null.c \
-     vo_yuv4mpeg.c \
      $(OPTIONAL_SRCS) \
 
 OBJS_TEMP=$(basename $(SRCS))
diff -Naur MPlayer-20050806.orig/libvo/video_out.c MPlayer-20050806/libvo/video_out.c
--- MPlayer-20050806.orig/libvo/video_out.c	2005-09-14 22:58:02.000000000 +0200
+++ MPlayer-20050806/libvo/video_out.c	2005-09-14 23:23:14.000000000 +0200
@@ -235,7 +235,6 @@
         &video_out_null,
 	// should not be auto-selected
 	&video_out_mpegpes,
-	&video_out_yuv4mpeg,
 #ifdef HAVE_PNG
 	&video_out_png,
 #endif	
