diff -Naur MPlayer-1.0pre6.orig/configure MPlayer-1.0pre6/configure
--- MPlayer-1.0pre6.orig/configure	2004-12-30 19:39:19.516811304 +0100
+++ MPlayer-1.0pre6/configure	2004-12-30 19:39:28.784402416 +0100
@@ -6448,12 +6448,7 @@
 # TODO cleanup the VIDIX stuff here
 _def_vidix='#define CONFIG_VIDIX 1'
 test "$_vidix" = no && _def_vidix='#undef CONFIG_VIDIX'
-if test "$_vidix" = yes; then
-  _vosrc="$_vosrc vo_cvidix.c"
-  _vomodules="cvidix $_vomodules"
-else
   _novomodules="cvidix $_novomodules"
-fi
 if test "$_vidix" = yes && (win32); then
   _vosrc="$_vosrc vo_winvidix.c"
   _vomodules="winvidix $_vomodules"
diff -Naur MPlayer-1.0pre6.orig/libvo/Makefile MPlayer-1.0pre6/libvo/Makefile
--- MPlayer-1.0pre6.orig/libvo/Makefile	2004-12-30 19:38:06.781868696 +0100
+++ MPlayer-1.0pre6/libvo/Makefile	2004-12-30 19:39:28.784402416 +0100
@@ -3,7 +3,7 @@
 
 LIBNAME = libvo.a
 
-SRCS=geometry.c aspect.c aclib.c osd.c font_load.c gtf.c spuenc.c video_out.c vo_null.c vo_mpegpes.c vo_yuv4mpeg.c $(OPTIONAL_SRCS) sub.c font_load_ft.c
+SRCS=geometry.c aspect.c aclib.c osd.c font_load.c gtf.c spuenc.c video_out.c vo_null.c vo_mpegpes.c $(OPTIONAL_SRCS) sub.c font_load_ft.c
 OBJS=$(SRCS:.c=.o)
 
 ifeq ($(VIDIX),yes)
diff -Naur MPlayer-1.0pre6.orig/libvo/video_out.c MPlayer-1.0pre6/libvo/video_out.c
--- MPlayer-1.0pre6.orig/libvo/video_out.c	2004-12-30 19:38:06.783868392 +0100
+++ MPlayer-1.0pre6/libvo/video_out.c	2004-12-30 19:39:28.785402264 +0100
@@ -224,12 +224,10 @@
 #ifdef WIN32
     &video_out_winvidix,
 #endif
-    &video_out_cvidix,
 #endif
         &video_out_null,
 	// should not be auto-selected
 	&video_out_mpegpes,
-	&video_out_yuv4mpeg,
 #ifdef HAVE_PNG
 	&video_out_png,
 #endif	
