Index: MPlayer-20051213/libvo/Makefile
===================================================================
--- MPlayer-20051213.orig/libvo/Makefile	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/libvo/Makefile	2006-02-23 10:49:45.000000000 +0100
@@ -14,7 +14,6 @@
      video_out.c \
      vo_mpegpes.c \
      vo_null.c \
-     vo_yuv4mpeg.c \
      $(OPTIONAL_SRCS) \
 
 LIBAV_INC =
Index: MPlayer-20051213/libvo/video_out.c
===================================================================
--- MPlayer-20051213.orig/libvo/video_out.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/libvo/video_out.c	2006-02-23 10:49:45.000000000 +0100
@@ -234,7 +234,6 @@
         &video_out_xvmc,
 #endif
 	&video_out_mpegpes,
-	&video_out_yuv4mpeg,
 #ifdef HAVE_PNG
 	&video_out_png,
 #endif	
