diff -r 3f43ce128995 cfg-mplayer.h
--- a/cfg-mplayer.h	Sun Apr 30 16:54:32 2006 +0200
+++ b/cfg-mplayer.h	Sun Apr 30 17:12:13 2006 +0200
@@ -344,6 +344,8 @@ m_option_t mplayer_opts[]={
 	{"guiwid", &guiWinID, CONF_TYPE_INT, 0, 0, 0, NULL},
 #endif
 
+	{"bgvideo", &bg_video, CONF_TYPE_STRING, 0, 0, 0, NULL},
+
 	{"noloop", &loop_times, CONF_TYPE_FLAG, 0, 0, -1, NULL},
 	{"loop", &loop_times, CONF_TYPE_INT, CONF_RANGE, -1, 10000, NULL},
 	{"playlist", NULL, CONF_TYPE_STRING, 0, 0, 0, NULL},
diff -r 3f43ce128995 mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:32 2006 +0200
+++ b/mplayer.c	Sun Apr 30 17:12:13 2006 +0200
@@ -281,6 +281,9 @@ static char* spudec_ifo=NULL;
 char* filename=NULL; //"MI2-Trailer.avi";
 int forced_subs_only=0;
 int file_filter=1;
+
+static char* bg_video = NULL;
+demuxer_t *bg_demuxer = NULL;
 
 // cache2:
        int stream_cache_size=-1;
@@ -447,6 +450,13 @@ static void uninit_player(unsigned int m
 	free_demuxer(demuxer);
     }
     demuxer=NULL;
+    current_module="free_bg_demuxer";
+    if(bg_demuxer) {
+      stream_t* bg_s = bg_demuxer->stream;
+      free_demuxer(bg_demuxer);
+      bg_demuxer = NULL;
+      free_stream(bg_s);
+    }
   }
 
   // kill the cache process:
@@ -1247,7 +1257,7 @@ void set_osd_bar(int type,char* name,dou
  * 
  */
 
-static void update_osd_msg(void) {
+static void update_osd_msg(demux_stream_t *d_audio) {
     mp_osd_msg_t *msg;
     static char osd_text[64] = "";
     static char osd_text_timer[64];
@@ -1278,6 +1288,7 @@ static void update_osd_msg(void) {
             int percentage = -1;
             char percentage_text[10];
             int pts = sh_video->pts;
+            if (bg_demuxer) pts=d_audio->pts+(ds_tell_pts(d_audio)-sh_audio->a_in_buffer_len)/sh_audio->i_bps;
             
             if (osd_show_percentage)
                 percentage = demuxer_get_percent_pos(demuxer);
@@ -3163,6 +3174,34 @@ sh_audio=d_audio->sh;
 sh_audio=d_audio->sh;
 sh_video=d_video->sh;
 
+while(sh_audio && !sh_video && bg_video) {
+  int bg_file_format = 0;
+  stream_t* bg_s;
+  mp_msg(MSGT_CPLAYER,MSGL_INFO, "Open bgvideo: %s\n",bg_video);
+  bg_s = open_stream(bg_video,NULL,&bg_file_format);
+  if(!bg_s) {
+    mp_msg(MSGT_CPLAYER,MSGL_ERR, "Couldn't open the stream for back ground video: %s\n",bg_video);
+    break;
+  }
+  bg_demuxer = demux_open(bg_s,bg_file_format,-2,video_id,dvdsub_id,bg_video);
+  if(!bg_demuxer) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"Failed to open background video\n");
+    free_stream(bg_s);
+    break;
+  }
+  if(!bg_demuxer->video && !bg_demuxer->video->sh) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"The background video don't have a video stream\n");
+    free_demuxer(bg_demuxer);
+    bg_demuxer = NULL;
+    free_stream(bg_s);
+    break;
+  }
+  d_video = bg_demuxer->video;
+  sh_video = d_video->sh;
+  mp_msg(MSGT_DEMUXER,MSGL_INFO,"Background video should work ;)\n");
+  break;
+}
+
 if(sh_video){
 
   current_module="video_read_properties";
@@ -3501,7 +3540,7 @@ int was_paused=0;
 // Make sure old OSD does not stay around,
 // e.g. with -fixed-vo and same-resolution files
 clear_osd_msgs();
-update_osd_msg();
+update_osd_msg(d_audio);
 
 //================ SETUP AUDIO ==========================
 
@@ -3675,6 +3714,8 @@ while(sh_audio){
   break;
 } // while(sh_audio)
 
+if(bg_demuxer && d_audio->eof) eof=1;
+
 if(!sh_video) {
   // handle audio-only case:
   if(!quiet) {
@@ -3701,7 +3742,16 @@ if(!sh_video) {
 	// get it!
 	current_module="video_read_frame";
         in_size=video_read_frame(sh_video,&next_frame_time,&start,force_fps);
-	if(in_size<0){ eof=1; break; }
+	if(in_size<0){
+	  if(bg_demuxer) {
+	    if(!demux_seek(bg_demuxer,0,0,1))
+	      eof = 1;
+	    else
+	      continue;
+	  } else
+	    eof=1;
+	  break;
+	}
 	if(in_size>max_framesize) max_framesize=in_size; // stats
 	sh_video->timer+=frame_time;
 	if(sh_audio) sh_audio->delay-=frame_time;
@@ -3901,6 +3951,15 @@ if(time_frame>0.001 && !(vo_flags&256)){
 	   if(vo_config_count) video_out->flip_page();
 	   if (play_n_frames >= 0) {
 	     --play_n_frames;
+	     if(play_n_frames<0) {
+		if(bg_demuxer) {
+		   if(!demux_seek(bg_demuxer,0,0,1))
+			eof = PT_NEXT_ENTRY;
+		   else
+			play_n_frames = play_n_frames_mf;
+		} else
+		   eof = PT_NEXT_ENTRY;
+	     }
 	     if (play_n_frames <= 0) eof = PT_NEXT_ENTRY;
 	   }
 
@@ -3922,6 +3981,13 @@ if(time_frame>0.001 && !(vo_flags&256)){
   current_module="av_sync";
 
   if(sh_audio){
+    if(bg_demuxer) {
+      if(!quiet) mp_msg(MSGT_AVSYNC,MSGL_STATUS,"A:%6.1f %4.1f%% %d%%   \r"
+ 		       ,sh_audio->delay-audio_out->get_delay()
+ 		       ,(sh_audio->delay>0.5)?100.0*audio_time_usage/(double)sh_audio->delay:0
+ 		       ,cache_fill_status
+ 		       );
+    } else {
     double a_pts, v_pts;
 
     if (autosync)
@@ -3986,7 +4052,7 @@ if(time_frame>0.001 && !(vo_flags&256)){
         if(!quiet)
           print_status(a_pts - audio_delay - delay, AV_delay, c_total);
       }
-    
+   }
   } else {
     // No audio:
     
@@ -4046,7 +4112,7 @@ if(auto_quality>0){
 	  int mlen = strlen(msg);
 	  msg[mlen-1] = '\0';
 	  set_osd_msg(OSD_MSG_PAUSE,1,0,"%s",msg+1);
-	  update_osd_msg();
+	  update_osd_msg(d_audio);
 	} else
 	  mp_msg(MSGT_CPLAYER,MSGL_STATUS,MSGTR_Paused);
         mp_msg(MSGT_IDENTIFY, MSGL_INFO, "ID_PAUSED\n");
@@ -4948,7 +5014,7 @@ if ((user_muted | edl_muted) != mixer.mu
 
 //================= Update OSD ====================
   
-  update_osd_msg();
+  update_osd_msg(d_audio);
     
 #ifdef USE_SUB
   // find sub
