diff -r 3f43ce128995 cfg-mplayer.h
--- a/cfg-mplayer.h	Sun Apr 30 16:54:32 2006 +0200
+++ b/cfg-mplayer.h	Sun Apr 30 17:12:13 2006 +0200
@@ -327,6 +327,8 @@ m_option_t mplayer_opts[]={
 	{"guiwid", &guiWinID, CONF_TYPE_INT, 0, 0, 0, NULL},
 #endif
 
+	{"bgvideo", &bg_video, CONF_TYPE_STRING, 0, 0, 0, NULL},
+
 	{"noloop", &mpctx_s.loop_times, CONF_TYPE_FLAG, 0, 0, -1, NULL},
 	{"loop", &mpctx_s.loop_times, CONF_TYPE_INT, CONF_RANGE, -1, 10000, NULL},
 	{"playlist", NULL, CONF_TYPE_STRING, 0, 0, 0, NULL},
diff -r 3f43ce128995 mp_core.h
--- a/mp_core.h	2007-05-01 02:23:10.000000000 +0200
+++ b/mp_core.h	2007-05-01 02:23:12.000000000 +0200
@@ -55,6 +55,7 @@
 
     stream_t *stream;
     demuxer_t *demuxer;
+    demuxer_t *bg_demuxer;
     sh_audio_t *sh_audio;
     sh_video_t *sh_video;
     demux_stream_t *d_audio;
diff -r 3f43ce128995 mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:32 2006 +0200
+++ b/mplayer.c	Sun Apr 30 17:12:13 2006 +0200
@@ -277,6 +277,8 @@ static char* spudec_ifo=NULL;
 char* filename=NULL; //"MI2-Trailer.avi";
 int forced_subs_only=0;
 int file_filter=1;
+
+static char* bg_video = NULL;
 
 // cache2:
        int stream_cache_size=-1;
@@ -410,7 +412,8 @@
   case META_VIDEO_BITRATE:
   case META_VIDEO_RESOLUTION:
   {
-    if (!mpctx->sh_video)
+    if (!mpctx->sh_video || (mpctx->bg_demuxer && mpctx->bg_demuxer->video && mpctx->bg_demuxer->video->sh
+        && mpctx->sh_video == mpctx->bg_demuxer->video->sh))
       return 0;
     break;
   }
@@ -599,6 +603,13 @@ static void uninit_player(unsigned int m
 	free_demuxer(mpctx->demuxer);
     }
     mpctx->demuxer=NULL;
+    current_module="free_bg_demuxer";
+    if(mpctx->bg_demuxer) {
+      stream_t* bg_s = mpctx->bg_demuxer->stream;
+      free_demuxer(mpctx->bg_demuxer);
+      mpctx->bg_demuxer = NULL;
+      free_stream(bg_s);
+    }
   }
 
   // kill the cache process:
@@ -1432,6 +1432,8 @@ static void update_osd_msg(void) {
             int percentage = -1;
             char percentage_text[10];
             int pts = demuxer_get_current_time(mpctx->demuxer);
+            if (mpctx->bg_demuxer)
+                pts = playing_audio_pts(mpctx->sh_audio, mpctx->d_audio, mpctx->audio_out);
             
             if (mpctx->osd_show_percentage)
                 percentage = demuxer_get_percent_pos(mpctx->demuxer);
@@ -1618,10 +1618,17 @@ if(!sh_video) {
 	current_module = "video_read_frame";
 	in_size = ds_get_packet_pts(d_video, &start, &pts);
 	if (in_size < 0) {
+	  if(mpctx->bg_demuxer) {
+	    if(!demux_seek(mpctx->bg_demuxer,0,0,1))
+	      hit_eof = 1;
+	    else
+	      continue;
+	  } else {
 	    // try to extract last frames in case of decoder lag
 	    in_size = 0;
 	    pts = 1e300;
 	    hit_eof = 1;
+	  }
 	}
 	if (in_size > max_framesize)
 	    max_framesize = in_size;
@@ -1684,6 +1684,13 @@ if(time_frame>0.001 && !(vo_flags&256)){
     current_module="av_sync";
 
     if(mpctx->sh_audio){
+     if(mpctx->bg_demuxer) {
+      if(!quiet) mp_msg(MSGT_AVSYNC,MSGL_STATUS,"A:%6.1f %4.1f%% %d%%   \r"
+ 		       ,mpctx->delay - mpctx->audio_out->get_delay()
+ 		       ,(mpctx->delay>0.5)?100.0*audio_time_usage/(double)mpctx->delay:0
+ 		       ,cache_fill_status
+ 		       );
+     } else {
 	double a_pts, v_pts;
 
 	if (autosync)
@@ -1735,6 +1735,7 @@ if(time_frame>0.001 && !(vo_flags&256)){
 		print_status(a_pts - audio_delay, AV_delay, c_total);
 	}
     
+     }
     } else {
 	// No audio:
     
@@ -3069,6 +3069,34 @@ sh_audio=d_audio->sh;
 mpctx->sh_audio=mpctx->d_audio->sh;
 mpctx->sh_video=mpctx->d_video->sh;
 
+while(mpctx->sh_audio && !mpctx->sh_video && bg_video) {
+  int bg_file_format = 0;
+  stream_t* bg_s;
+  mp_msg(MSGT_CPLAYER,MSGL_INFO, "Open bgvideo: %s\n",bg_video);
+  bg_s = open_stream(bg_video,NULL,&bg_file_format);
+  if(!bg_s) {
+    mp_msg(MSGT_CPLAYER,MSGL_ERR, "Couldn't open the stream for back ground video: %s\n",bg_video);
+    break;
+  }
+  mpctx->bg_demuxer = demux_open(bg_s,bg_file_format,-2,video_id,dvdsub_id,bg_video);
+  if(!mpctx->bg_demuxer) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"Failed to open background video\n");
+    free_stream(bg_s);
+    break;
+  }
+  if(!mpctx->bg_demuxer->video && !mpctx->bg_demuxer->video->sh) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"The background video don't have a video stream\n");
+    free_demuxer(mpctx->bg_demuxer);
+    mpctx->bg_demuxer = NULL;
+    free_stream(bg_s);
+    break;
+  }
+  mpctx->d_video = mpctx->bg_demuxer->video;
+  mpctx->sh_video = mpctx->d_video->sh;
+  mp_msg(MSGT_DEMUXER,MSGL_INFO,"Background video should work ;)\n");
+  break;
+}
+
 if(mpctx->sh_video){
 
   current_module="video_read_properties";
@@ -3368,6 +3368,7 @@ while(sh_audio){
 	if (!mpctx->sh_video)
 	    mpctx->eof = PT_NEXT_ENTRY;
 
+if(mpctx->bg_demuxer && mpctx->d_audio->eof) mpctx->eof=1;
 
 if(!mpctx->sh_video) {
   // handle audio-only case:
@@ -3397,6 +3397,11 @@
 	  mpctx->eof = 1; goto goto_next_file;
       }
       if (frame_time < 0)
+        if(mpctx->bg_demuxer) {
+          if(!demux_seek(mpctx->bg_demuxer,0,0,1))
+            mpctx->eof = PT_NEXT_ENTRY;
+        }
+        else
 	  mpctx->eof = 1;
       else {
 	  // might return with !eof && !blit_frame if !correct_pts
