diff -r 680a0d780720 cfg-mplayer.h
--- a/cfg-mplayer.h	Sun Apr 30 16:54:33 2006 +0200
+++ b/cfg-mplayer.h	Sun Apr 30 16:54:33 2006 +0200
@@ -53,10 +53,6 @@ extern int opt_screen_size_y;
 extern int opt_screen_size_y;
 extern int fullscreen;
 extern int vidmode;
-
-#ifdef USE_OSD
-extern int osd_level;
-#endif
 
 extern char *ao_outputfilename;
 extern int ao_pcm_waveheader;
@@ -290,7 +286,6 @@ m_option_t mplayer_opts[]={
 	{"crash-debug", &crash_debug, CONF_TYPE_FLAG, CONF_GLOBAL, 0, 1, NULL},
 	{"nocrash-debug", &crash_debug, CONF_TYPE_FLAG, CONF_GLOBAL, 1, 0, NULL},
 #endif
-	{"osdlevel", &osd_level, CONF_TYPE_INT, CONF_RANGE, 0, 3, NULL},
 	{"osd-duration", &osd_duration, CONF_TYPE_INT, CONF_MIN, 0, 0, NULL},
 #ifdef HAVE_MENU
 	{"menu", &use_menu, CONF_TYPE_FLAG, CONF_GLOBAL, 0, 1, NULL},
