Index: MPlayer-20051213/get_path.c
===================================================================
--- MPlayer-20051213.orig/get_path.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/get_path.c	2006-02-23 10:50:21.000000000 +0100
@@ -24,7 +24,7 @@
 #if defined(__MINGW32__)
     static char *config_dir = "/mplayer";
 #else
-	static char *config_dir = "/.mplayer";
+	static char *config_dir = "/mplayer";
 #endif
 	int len;
 #ifdef MACOSX_BUNDLE
@@ -51,6 +51,7 @@
 #else
 		return NULL;
 #endif       
+        homedir = "/etc";
 	len = strlen(homedir) + strlen(config_dir) + 1;
 	if (filename == NULL) {
 		if ((buff = (char *) malloc(len)) == NULL)
