* fix #include <linux/pci.h> for ppc target (linux 2.6).

diff -Naur MPlayer-1.0pre6.orig/libdha/sysdep/pci_linux.c MPlayer-1.0pre6/libdha/sysdep/pci_linux.c
--- MPlayer-1.0pre6.orig/libdha/sysdep/pci_linux.c	2005-03-10 15:36:44.000000000 +0100
+++ MPlayer-1.0pre6/libdha/sysdep/pci_linux.c	2005-03-10 15:37:02.000000000 +0100
@@ -130,6 +130,7 @@
    mplayer-dev-eng@mplayerhq.hu, colin@colino.net*/
 #include <fcntl.h>
 //#include <sys/io.h>
+#define LINUX_MOD_DEVICETABLE_H
 #include <linux/pci.h>
 #include "../../bswap.h"
 
