* fix #include <linux/pci.h> for ppc target (linux 2.6).

Index: MPlayer-20051213/libdha/sysdep/pci_linux.c
===================================================================
--- MPlayer-20051213.orig/libdha/sysdep/pci_linux.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/libdha/sysdep/pci_linux.c	2006-02-23 10:50:33.000000000 +0100
@@ -131,6 +131,7 @@
    mplayer-dev-eng@mplayerhq.hu, colin@colino.net*/
 #include <fcntl.h>
 //#include <sys/io.h>
+#define LINUX_MOD_DEVICETABLE_H
 #include <linux/pci.h>
 #include "../../bswap.h"
 
