diff -Naur MPlayer-r27630.orig/libvo/x11_common.c MPlayer-r27630/libvo/x11_common.c
--- MPlayer-r27630.orig/libvo/x11_common.c	2008-09-27 19:00:32.000000000 +0200
+++ MPlayer-r27630/libvo/x11_common.c	2008-09-27 19:01:11.000000000 +0200
@@ -1250,7 +1250,7 @@
                                    unsigned int width, unsigned int height,
                                    int depth, Colormap col_map)
 {
-    unsigned long xswamask = CWBackingStore | CWBorderPixel;
+    unsigned long xswamask = CWBorderPixel;
     XSetWindowAttributes xswa;
     Window ret_win;
 
@@ -1261,7 +1261,7 @@
     }
     xswa.background_pixel = 0;
     xswa.border_pixel = 0;
-    xswa.backing_store = Always;
+    xswa.backing_store = NotUseful;
     xswa.bit_gravity = StaticGravity;
 
     ret_win =
