diff -Naur MPlayer-1.0pre6.orig/playtree.c MPlayer-1.0pre6/playtree.c
--- MPlayer-1.0pre6.orig/playtree.c	2004-12-30 19:38:06.694881920 +0100
+++ MPlayer-1.0pre6/playtree.c	2004-12-30 19:42:45.717464040 +0100
@@ -707,8 +707,9 @@
   iter->tree = iter->tree->parent;
 
   // Pop subtree params
-  if(iter->config) {
+  if(iter->config && iter->entry_pushed > 0) {
     m_config_pop(iter->config);
+    iter->entry_pushed = 0;
     if(iter->mode == PLAY_TREE_ITER_RND)
       iter->tree->flags |= PLAY_TREE_RND_PLAYED;
   }
