diff -r 8eab14a155a8 stream/stream.c
--- a/stream/stream.c	Sun Apr 30 16:54:35 2006 +0200
+++ b/stream/stream.c	Sun Apr 30 16:54:35 2006 +0200
@@ -216,6 +216,21 @@ stream_t* open_stream_full(char* filenam
   stream_info_t* sinfo;
   stream_t* s;
   char *redirected_url = NULL;
+  char *ext = NULL;
+
+  ext = strrchr (filename, '.');
+  if (ext && !strcasecmp (ext + 1, "ifo"))
+  {
+    extern char *dvd_device;
+    struct stat st;
+    dvd_device = dirname (strdup (filename));
+    dvd_title = -1; /* play longest track */
+
+    if (stat ("/var/dvdnav", &st) == 0)
+      filename = "dvdnav://";
+    else
+      filename = "dvd://";
+  }
 
   for(i = 0 ; auto_open_streams[i] ; i++) {
     sinfo = auto_open_streams[i];
