diff -Naur MPlayer-r23418.orig/cfg-common.h MPlayer-r23418/cfg-common.h
--- MPlayer-r23418.orig/cfg-common.h	2007-05-30 19:30:33.000000000 +0200
+++ MPlayer-r23418/cfg-common.h	2007-05-30 19:31:23.000000000 +0200
@@ -26,7 +26,8 @@
 #endif
 	{"vcd", "-vcd N has been removed, use vcd://N instead.\n", CONF_TYPE_PRINT, CONF_NOCFG ,0,0, NULL},
 	{"cuefile", "-cuefile has been removed, use cue://filename:N where N is the track number.\n", CONF_TYPE_PRINT, 0, 0, 0, NULL},
-	{"cdrom-device", &cdrom_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+        {"vcd-device", &vcd_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+        {"cdda-device", &cdda_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
 #ifdef USE_DVDREAD
 	{"dvd-device", &dvd_device,  CONF_TYPE_STRING, 0, 0, 0, NULL}, 
 	{"dvd-speed", &dvd_speed, CONF_TYPE_INT, 0, 0, 0, NULL},
@@ -554,7 +554,7 @@
 extern int vivo_param_width;
 extern int vivo_param_height;
 extern int vivo_param_vformat;
-extern char *dvd_device, *cdrom_device;
+extern char *dvd_device, *vcd_device, *cdda_device;
 
 m_option_t vivoopts_conf[]={
 	{"version", &vivo_param_version, CONF_TYPE_INT, 0, 0, 0, NULL},
diff -Naur MPlayer-r23418.orig/command.c MPlayer-r23418/command.c
--- MPlayer-r23418.orig/command.c	2007-05-30 19:30:33.000000000 +0200
+++ MPlayer-r23418/command.c	2007-05-30 19:33:22.000000000 +0200
@@ -2171,8 +2171,9 @@
           char buf[11];
           play_tree_t *cdda, *tracks;
           int nb_tracks, i; 
-          
-          nb_tracks = get_cdda_tracks_num (cdrom_device ? cdrom_device : DEFAULT_CDROM_DEVICE);
+
+          extern char *cdda_device;
+          nb_tracks = get_cdda_tracks_num (cdda_device ? cdda_device : DEFAULT_CDROM_DEVICE);
           if (nb_tracks <= 0)
             break;
           
diff -Naur MPlayer-r23418.orig/stream/stream_cdda.c MPlayer-r23418/stream/stream_cdda.c
--- MPlayer-r23418.orig/stream/stream_cdda.c	2007-05-30 19:30:33.000000000 +0200
+++ MPlayer-r23418/stream/stream_cdda.c	2007-05-30 19:32:07.000000000 +0200
@@ -20,7 +20,7 @@
 #endif
 
 
-extern char *cdrom_device;
+char *cdda_device;
 
 static struct cdda_params {
   int speed;
@@ -118,8 +118,8 @@
   }
 
   if(!p->device) {
-    if (cdrom_device)
-      p->device = strdup(cdrom_device);
+    if (cdda_device)
+      p->device = strdup(cdda_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
diff -Naur MPlayer-r23418.orig/stream/stream_vcd.c MPlayer-r23418/stream/stream_vcd.c
--- MPlayer-r23418.orig/stream/stream_vcd.c	2007-05-30 19:30:33.000000000 +0200
+++ MPlayer-r23418/stream/stream_vcd.c	2007-05-30 19:32:32.000000000 +0200
@@ -31,7 +31,7 @@
 
 #include "libmpdemux/demuxer.h"
 
-extern char *cdrom_device;
+char *vcd_device;
 
 static struct stream_priv_s {
   int track;
@@ -96,8 +96,8 @@
   }
 
   if (!p->device) {
-    if(cdrom_device)
-      p->device = strdup(cdrom_device);
+    if(vcd_device)
+      p->device = strdup(vcd_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
