diff -r 1b903304e52c cfg-common.h
--- a/cfg-common.h	Sun Apr 30 16:54:35 2006 +0200
+++ b/cfg-common.h	Sun Apr 30 16:54:35 2006 +0200
@@ -26,7 +26,8 @@
 #endif
 	{"vcd", "-vcd N is deprecated, use vcd://N instead.\n", CONF_TYPE_PRINT, CONF_NOCFG ,0,0, NULL},
 	{"cuefile", "-cuefile is deprecated, use cue://filename:N where N is the track number.\n", CONF_TYPE_PRINT, 0, 0, 0, NULL},
-	{"cdrom-device", &cdrom_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"vcd-device", &vcd_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"cdda-device", &cdda_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
 #if defined(USE_DVDREAD) || defined(USE_DVDNAV)
 	{"dvd-device", &dvd_device,  CONF_TYPE_STRING, 0, 0, 0, NULL}, 
 #else
@@ -452,7 +453,7 @@ extern int vivo_param_width;
 extern int vivo_param_width;
 extern int vivo_param_height;
 extern int vivo_param_vformat;
-extern char *dvd_device, *cdrom_device;
+extern char *dvd_device, *vcd_device, *cdda_device;
 
 m_option_t vivoopts_conf[]={
 	{"version", &vivo_param_version, CONF_TYPE_INT, 0, 0, 0, NULL},
diff -r 1b903304e52c libmpdemux/cdda.c
--- a/stream/stream_cdda.c	Sun Apr 30 16:54:35 2006 +0200
+++ b/stream/stream_cdda.c	Sun Apr 30 16:54:35 2006 +0200
@@ -21,7 +21,7 @@
 #endif
 
 
-extern char *cdrom_device;
+char *cdda_device;
 
 static struct cdda_params {
   int speed;
@@ -119,8 +119,8 @@ static int open_cdda(stream_t *st,int m,
   }
 
   if(!p->device) {
-    if (cdrom_device)
-      p->device = strdup(cdrom_device);
+    if (cdda_device)
+      p->device = strdup(cdda_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
diff -r 1b903304e52c libmpdemux/stream_vcd.c
--- a/stream/stream_vcd.c	Sun Apr 30 16:54:35 2006 +0200
+++ b/stream/stream_vcd.c	Sun Apr 30 16:54:35 2006 +0200
@@ -25,7 +25,7 @@
 #include "vcd_read.h"
 #endif
 
-extern char *cdrom_device;
+char *vcd_device;
 
 static struct stream_priv_s {
   int track;
@@ -82,8 +82,8 @@ static int open_s(stream_t *stream,int m
   }
 
   if (!p->device) {
-    if(cdrom_device)
-      p->device = strdup(cdrom_device);
+    if(vcd_device)
+      p->device = strdup(vcd_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
diff -r 1b903304e52c mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:35 2006 +0200
+++ b/mplayer.c	Sun Apr 30 16:54:35 2006 +0200
@@ -4393,7 +4393,7 @@ if (stream->type==STREAMTYPE_DVDNAV && d
       play_tree_t *cdda, *tracks;
       int nb_tracks, i; 
 
-      nb_tracks = get_cdda_tracks_num (cdrom_device ? cdrom_device : DEFAULT_CDROM_DEVICE);
+      nb_tracks = get_cdda_tracks_num (cdda_device ? cdda_device : DEFAULT_CDROM_DEVICE);
       if (nb_tracks <= 0)
         break;
 
