diff -Naur MPlayer-1.0pre6.orig/libavcodec/Makefile MPlayer-1.0pre6/libavcodec/Makefile
--- MPlayer-1.0pre6.orig/libavcodec/Makefile	2004-12-30 19:38:06.802865504 +0100
+++ MPlayer-1.0pre6/libavcodec/Makefile	2004-12-30 19:42:57.252710416 +0100
@@ -7,7 +7,7 @@
 VPATH=$(SRC_PATH)/libavcodec
 
 # NOTE: -I.. is needed to include config.h
-CFLAGS=$(OPTFLAGS) -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
+CFLAGS=$(OPTFLAGS) -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE $(EXTRA_INC)
 
 OBJS= common.o utils.o mem.o allcodecs.o \
       mpegvideo.o jrevdct.o jfdctfst.o jfdctint.o\
diff -Naur MPlayer-1.0pre6.orig/libavcodec/fastmemcpy.h MPlayer-1.0pre6/libavcodec/fastmemcpy.h
--- MPlayer-1.0pre6.orig/libavcodec/fastmemcpy.h	1970-01-01 01:00:00.000000000 +0100
+++ MPlayer-1.0pre6/libavcodec/fastmemcpy.h	2004-12-30 19:42:57.252710416 +0100
@@ -0,0 +1 @@
+#include "../libvo/fastmemcpy.h"
diff -Naur MPlayer-1.0pre6.orig/libavcodec/libpostproc/fastmemcpy.h MPlayer-1.0pre6/libavcodec/libpostproc/fastmemcpy.h
--- MPlayer-1.0pre6.orig/libavcodec/libpostproc/fastmemcpy.h	1970-01-01 01:00:00.000000000 +0100
+++ MPlayer-1.0pre6/libavcodec/libpostproc/fastmemcpy.h	2004-12-30 19:42:57.252710416 +0100
@@ -0,0 +1 @@
+#include "../../libvo/fastmemcpy.h"
