diff -Naur MPlayer-1.0pre6.orig/input/input.c MPlayer-1.0pre6/input/input.c
--- MPlayer-1.0pre6.orig/input/input.c	2004-12-30 19:38:06.737875384 +0100
+++ MPlayer-1.0pre6/input/input.c	2004-12-30 19:43:27.889052984 +0100
@@ -634,6 +634,7 @@
 	ptr = NULL;
 	break;
       } else if(!e) e = ptr+strlen(ptr);
+      ptr = e;
       l = e-start;
       ptr2 = start;
        for(e = strchr(ptr2,'\\') ; e ; e = strchr(ptr2,'\\')) {
