Index: MPlayer-20051213/input/input.c
===================================================================
--- MPlayer-20051213.orig/input/input.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/input/input.c	2006-02-23 10:50:42.000000000 +0100
@@ -1141,8 +1141,6 @@
 	  continue;
 	mp_msg(MSGT_INPUT,MSGL_ERR,MSGTR_INPUT_INPUT_ErrSelect,strerror(errno));
       }
-      if(!got_cmd)
-	return NULL;
     }
     break;
   }
