diff -Naur MPlayer-1.0pre7.orig/input/joystick.c MPlayer-1.0pre7/input/joystick.c
--- MPlayer-1.0pre7.orig/input/joystick.c	2003-07-09 03:30:22.000000000 +0200
+++ MPlayer-1.0pre7/input/joystick.c	2005-07-26 19:14:52.000000000 +0200
@@ -65,7 +65,6 @@
 	printf("Joystick : we loose %d bytes of data\n",l);
       break;
     }
-    ev.type &= ~JS_EVENT_INIT;
     if(ev.type == JS_EVENT_BUTTON)
       btns |= (ev.value << ev.number);
     if(ev.type == JS_EVENT_AXIS)
