Index: MPlayer-20051213/cfg-mplayer.h
===================================================================
--- MPlayer-20051213.orig/cfg-mplayer.h	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/cfg-mplayer.h	2006-02-23 10:50:53.000000000 +0100
@@ -363,6 +363,8 @@
 	{"guiwid", &guiWinID, CONF_TYPE_INT, 0, 0, 0, NULL},
 #endif
 
+	{"bgvideo", &bg_video, CONF_TYPE_STRING, 0, 0, 0, NULL},
+
 	{"noloop", &loop_times, CONF_TYPE_FLAG, 0, 0, -1, NULL},
 	{"loop", &loop_times, CONF_TYPE_INT, CONF_RANGE, -1, 10000, NULL},
 	{"playlist", NULL, CONF_TYPE_STRING, 0, 0, 0, NULL},
Index: MPlayer-20051213/mplayer.c
===================================================================
--- MPlayer-20051213.orig/mplayer.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/mplayer.c	2006-02-23 10:50:53.000000000 +0100
@@ -274,6 +274,9 @@
 char* filename=NULL; //"MI2-Trailer.avi";
 int forced_subs_only=0;
 
+static char* bg_video = NULL;
+static demuxer_t *bg_demuxer = NULL;
+
 // cache2:
        int stream_cache_size=-1;
 #ifdef USE_STREAM_CACHE
@@ -439,6 +442,13 @@
 	free_demuxer(demuxer);
     }
     demuxer=NULL;
+    current_module="free_bg_demuxer";
+    if(bg_demuxer) {
+      stream_t* bg_s = bg_demuxer->stream;
+      free_demuxer(bg_demuxer);
+      bg_demuxer = NULL;
+      free_stream(bg_s);
+    }
   }
 
   // kill the cache process:
@@ -1228,7 +1238,7 @@
  * 
  */
 
-static void update_osd_msg(void) {
+static void update_osd_msg(demux_stream_t *d_audio) {
     mp_osd_msg_t *msg;
     static char osd_text[64] = "";
     static char osd_text_timer[64];
@@ -1259,6 +1269,7 @@
             char percentage_text[10];
             static int last_pts = -303;
             int pts = sh_video->pts;
+            if (bg_demuxer) pts=d_audio->pts+(ds_tell_pts(d_audio)-sh_audio->a_in_buffer_len)/sh_audio->i_bps;
             if(pts==last_pts-1) ++pts; else last_pts=pts;
             
             if (osd_show_percentage)
@@ -2123,6 +2134,34 @@
 sh_audio=d_audio->sh;
 sh_video=d_video->sh;
 
+while(sh_audio && !sh_video && bg_video) {
+  int bg_file_format = 0;
+  stream_t* bg_s;
+  mp_msg(MSGT_CPLAYER,MSGL_INFO, "Open bgvideo: %s\n",bg_video);
+  bg_s = open_stream(bg_video,NULL,&bg_file_format);
+  if(!bg_s) {
+    mp_msg(MSGT_CPLAYER,MSGL_ERR, "Couldn't open the stream for back ground video: %s\n",bg_video);
+    break;
+  }
+  bg_demuxer = demux_open(bg_s,bg_file_format,-2,video_id,dvdsub_id,bg_video);
+  if(!bg_demuxer) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"Failed to open background video\n");
+    free_stream(bg_s);
+    break;
+  }
+  if(!bg_demuxer->video && !bg_demuxer->video->sh) {
+    mp_msg(MSGT_DEMUXER,MSGL_ERR,"The background video don't have a video stream\n");
+    free_demuxer(bg_demuxer);
+    bg_demuxer = NULL;
+    free_stream(bg_s);
+    break;
+  }
+  d_video = bg_demuxer->video;
+  sh_video = d_video->sh;
+  mp_msg(MSGT_DEMUXER,MSGL_INFO,"Background video should work ;)\n");
+  break;
+}
+
 if(sh_video){
 
   current_module="video_read_properties";
@@ -2637,6 +2676,8 @@
   break;
 } // while(sh_audio)
 
+if(bg_demuxer && d_audio->eof) eof=1;
+
 if(!sh_video) {
   // handle audio-only case:
   if(!quiet) {
@@ -2663,7 +2704,16 @@
 	// get it!
 	current_module="video_read_frame";
         in_size=video_read_frame(sh_video,&next_frame_time,&start,force_fps);
-	if(in_size<0){ eof=1; break; }
+	if(in_size<0){
+	  if(bg_demuxer) {
+	    if(!demux_seek(bg_demuxer,0,0,1))
+	      eof = 1;
+	    else
+	      continue;
+	  } else
+	    eof=1;
+	  break;
+	}
 	if(in_size>max_framesize) max_framesize=in_size; // stats
 	sh_video->timer+=frame_time;
 	if(sh_audio) sh_audio->delay-=frame_time;
@@ -2867,6 +2917,15 @@
 	   if(vo_config_count) video_out->flip_page();
 	   if (play_n_frames >= 0) {
 	     --play_n_frames;
+	     if(play_n_frames<0) {
+		if(bg_demuxer) {
+		   if(!demux_seek(bg_demuxer,0,0,1))
+			eof = PT_NEXT_ENTRY;
+		   else
+			play_n_frames = play_n_frames_mf;
+		} else
+		   eof = PT_NEXT_ENTRY;
+	     }
 	     if (play_n_frames <= 0) eof = PT_NEXT_ENTRY;
 	   }
 
@@ -2888,6 +2947,13 @@
   current_module="av_sync";
 
   if(sh_audio){
+   if(bg_demuxer) {
+     if(!quiet) mp_msg(MSGT_AVSYNC,MSGL_STATUS,"A:%6.1f %4.1f%% %d%%   \r"
+		       ,sh_audio->delay-audio_out->get_delay()
+		       ,(sh_audio->delay>0.5)?100.0*audio_time_usage/(double)sh_audio->delay:0
+		       ,cache_fill_status
+		       );
+   } else {
     float a_pts=0;
     float v_pts=0;
 
@@ -2959,7 +3025,7 @@
         if(!quiet)
           print_status(a_pts - audio_delay - delay, AV_delay, c_total);
       }
-    
+   }
   } else {
     // No audio:
     
@@ -3016,7 +3082,7 @@
 	  int mlen = strlen(msg);
 	  msg[mlen-1] = '\0';
 	  set_osd_msg(OSD_MSG_PAUSE,1,0,"%s",msg+1);
-	  update_osd_msg();
+	  update_osd_msg(d_audio);
 	} else
 	  mp_msg(MSGT_CPLAYER,MSGL_STATUS,MSGTR_Paused);
         if (identify)
@@ -4380,7 +4446,7 @@
 //  vo_osd_text=osd_text_buffer;
 #endif /* USE_OSD */
   
-  update_osd_msg();
+  update_osd_msg(d_audio);
     
 #ifdef USE_SUB
   // find sub
