diff -Naur MPlayer-1.0pre6.orig/cfg-mplayer.h MPlayer-1.0pre6/cfg-mplayer.h
--- MPlayer-1.0pre6.orig/cfg-mplayer.h	2004-12-30 19:44:26.399158096 +0100
+++ MPlayer-1.0pre6/cfg-mplayer.h	2004-12-30 19:45:19.558076712 +0100
@@ -62,10 +62,6 @@
 extern int fullscreen;
 extern int vidmode;
 
-#ifdef USE_OSD
-extern int osd_level;
-#endif
-
 extern char *ao_outputfilename;
 extern int ao_pcm_waveheader;
 
@@ -316,7 +312,6 @@
 	{"crash-debug", &crash_debug, CONF_TYPE_FLAG, CONF_GLOBAL, 0, 1, NULL},
 	{"nocrash-debug", &crash_debug, CONF_TYPE_FLAG, CONF_GLOBAL, 1, 0, NULL},
 #endif
-	{"osdlevel", &osd_level, CONF_TYPE_INT, CONF_RANGE, 0, 3, NULL},
 #ifdef HAVE_MENU
 	{"menu", &use_menu, CONF_TYPE_FLAG, CONF_GLOBAL, 0, 1, NULL},
 	{"nomenu", &use_menu, CONF_TYPE_FLAG, CONF_GLOBAL, 1, 0, NULL},
