diff -Naur MPlayer-20060519.orig/mplayer.c MPlayer-20060519/mplayer.c
--- MPlayer-20060519.orig/mplayer.c	2006-05-25 12:55:35.000000000 +0200
+++ MPlayer-20060519/mplayer.c	2006-05-25 12:56:06.000000000 +0200
@@ -4927,7 +4927,7 @@
       }
       break;
     case MP_CMD_SWITCH_RATIO : {
-      if (cmd->nargs == 0)
+      if (cmd->nargs == 0 || cmd->args[0].v.f == -1)
 	movie_aspect = (float) sh_video->disp_w / sh_video->disp_h;
       else
 	movie_aspect = cmd->args[0].v.f;
