diff -Naur MPlayer-20060429.orig/libmenu/menu_list.h MPlayer-20060429/libmenu/menu_list.h
--- MPlayer-20060429.orig/libmenu/menu_list.h	2006-05-01 12:19:04.000000000 +0200
+++ MPlayer-20060429/libmenu/menu_list.h	2006-05-01 12:19:33.000000000 +0200
@@ -56,9 +56,9 @@
   0,0, \
   5, 3, \
   NULL, \
-  0xE0, 0x20, \
-  0x80, 0x50, \
-  0xC0, 0x50 \
+  0xFF, 0xFF, \
+  0xFF, 0xFF, \
+  0xA4, 0x50 \
 }
   
 
