diff -r d805860eb5fa libmenu/menu.c
--- a/libmenu/menu.c	Sun Apr 30 16:54:41 2006 +0200
+++ b/libmenu/menu.c	Sun Apr 30 16:54:41 2006 +0200
@@ -301,6 +301,7 @@ menu_t* menu_open(char *name) {
     return NULL;
   }
   m = calloc(1,sizeof(menu_t));
+  strncpy(m->name,name,sizeof(m->name));
   m->priv_st = &(menu_list[i].type->priv_st);
   m->priv = m_struct_copy(m->priv_st,menu_list[i].cfg);
   m->ctx = menu_ctx;
diff -r d805860eb5fa libmenu/menu.h
--- a/libmenu/menu.h	Sun Apr 30 16:54:41 2006 +0200
+++ b/libmenu/menu.h	Sun Apr 30 16:54:41 2006 +0200
@@ -16,6 +16,7 @@ struct  menu_s {
   int cl; // Close request (user sent a close cmd or
   menu_t* parent;
   menu_def_t *type;
+  char name[20];
 };
 
 typedef struct menu_info_s {
diff -r d805860eb5fa libmenu/vf_menu.c
--- a/libmenu/vf_menu.c	Sun Apr 30 16:54:41 2006 +0200
+++ b/libmenu/vf_menu.c	Sun Apr 30 16:54:41 2006 +0200
@@ -41,6 +41,18 @@ static mp_image_t* alloc_mpi(int w, int 
   menu_t* current;
   int passthrough;
 };
+
+void menu_stream_opened(int onlyhidden, const char *menu_name) {
+  char cmd[30];
+  if ( st_priv && ( (onlyhidden && !st_priv->current->show) ||
+         !strncmp(menu_name,st_priv->current->name,20) ) ) {
+    sprintf(cmd, "set_menu %s", menu_name);
+    if (!st_priv->current->show &&
+         strncmp(menu_name,st_priv->current->name,20))
+      mp_input_queue_cmd(mp_input_parse_cmd("menu toggle"));
+    mp_input_queue_cmd(mp_input_parse_cmd(cmd));
+  }
+}
 
 static int put_image(struct vf_instance_s* vf, mp_image_t *mpi, double pts);
 
@@ -96,6 +96,11 @@ static int cmd_filter(mp_cmd_t* cmd, int
       mp_msg(MSGT_GLOBAL,MSGL_WARN,MSGTR_LIBMENU_FailedToOpenMenu,menu);
       priv->current = l;
       priv->current->show = 0;
+    } else if (!strcmp("properties",menu)
+               && !strcmp(l->name,menu)) {
+      priv->current->show = l->show;
+      priv->current->parent = l->parent;
+      menu_close(l);
     } else {
       priv->current->show = 1;
       priv->current->parent = l;
diff -r d805860eb5fa mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:41 2006 +0200
+++ b/mplayer.c	Sun Apr 30 16:54:41 2006 +0200
@@ -348,6 +348,7 @@ extern int vo_gamma_hue;
 #include "m_struct.h"
 #include "libmenu/menu.h"
 extern void vf_menu_pause_update(struct vf_instance_s* vf);
+extern void menu_stream_opened(int onlyhidden, const char *menu_name);
 extern vf_info_t vf_info_menu;
 static vf_info_t* libmenu_vfs[] = {
   &vf_info_menu,
@@ -3142,6 +3142,9 @@ while(sh_audio && !sh_video && bg_video)
   mpctx->d_video = mpctx->bg_demuxer->video;
   mpctx->sh_video = mpctx->d_video->sh;
   mp_msg(MSGT_DEMUXER,MSGL_INFO,"Background video should work ;)\n");
+#ifdef HAVE_MENU
+  menu_stream_opened(1, "properties");
+#endif
   break;
 }
 
@@ -3326,6 +3326,10 @@ current_module="main";
 
 // Disable the term OSD in verbose mode
 if(verbose) term_osd = 0;
+
+#ifdef HAVE_MENU
+menu_stream_opened(0, "properties");
+#endif
 
 {
 //int frame_corr_num=0;   //
