diff -Naur MPlayer-1.0pre6.orig/libmpdemux/stream_vcd.c MPlayer-1.0pre6/libmpdemux/stream_vcd.c
--- MPlayer-1.0pre6.orig/libmpdemux/stream_vcd.c	2004-12-30 19:38:06.774869760 +0100
+++ MPlayer-1.0pre6/libmpdemux/stream_vcd.c	2004-12-30 19:46:06.124997464 +0100
@@ -93,6 +93,9 @@
     m_struct_free(&stream_opts,opts);
     return STREAM_ERROR;
   }
+#ifdef __linux__
+  ioctl(f, CDROM_LOCKDOOR, 0);
+#endif
 
   vcd = vcd_read_toc(f);
   if(!vcd) {
diff -Naur MPlayer-1.0pre6.orig/libmpdvdkit2/device.c MPlayer-1.0pre6/libmpdvdkit2/device.c
--- MPlayer-1.0pre6.orig/libmpdvdkit2/device.c	2004-12-30 19:38:06.777869304 +0100
+++ MPlayer-1.0pre6/libmpdvdkit2/device.c	2004-12-30 19:46:06.124997464 +0100
@@ -43,6 +43,11 @@
 #   include <sys/uio.h>                                      /* struct iovec */
 #endif
 
+#ifdef __linux__
+#include <linux/cdrom.h>
+#include <sys/ioctl.h>
+#endif
+
 #include "dvdcss.h"
 
 #include "common.h"
@@ -250,6 +255,9 @@
         _dvdcss_error( dvdcss, "failed opening device" );
         return -1;
     }
+#ifdef __linux__
+    ioctl(dvdcss->i_fd, CDROM_LOCKDOOR, 0);
+#endif
 
     dvdcss->i_pos = 0;
 
