Index: MPlayer-20051213/libmpdemux/stream_vcd.c
===================================================================
--- MPlayer-20051213.orig/libmpdemux/stream_vcd.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/libmpdemux/stream_vcd.c	2006-02-23 10:51:12.000000000 +0100
@@ -94,6 +94,9 @@
     m_struct_free(&stream_opts,opts);
     return STREAM_ERROR;
   }
+#ifdef __linux__
+  ioctl(f, CDROM_LOCKDOOR, 0);
+#endif
 
   vcd = vcd_read_toc(f);
   if(!vcd) {
Index: MPlayer-20051213/libmpdvdkit2/device.c
===================================================================
--- MPlayer-20051213.orig/libmpdvdkit2/device.c	2006-02-23 10:49:35.000000000 +0100
+++ MPlayer-20051213/libmpdvdkit2/device.c	2006-02-23 10:51:12.000000000 +0100
@@ -58,6 +58,11 @@
 #   include <sys/uio.h>                                      /* struct iovec */
 #endif
 
+#ifdef __linux__
+#include <linux/cdrom.h>
+#include <sys/ioctl.h>
+#endif
+
 #include "dvdcss.h"
 
 #include "common.h"
@@ -266,6 +271,9 @@
         print_error( dvdcss, "failed to open device" );
         return -1;
     }
+#ifdef __linux__
+    ioctl(dvdcss->i_fd, CDROM_LOCKDOOR, 0);
+#endif
 
     dvdcss->i_pos = 0;
 
