diff -r 7d7fcdab76a3 libmpdemux/stream_vcd.c
--- a/libmpdemux/stream_vcd.c	Sun Apr 30 16:54:34 2006 +0200
+++ b/libmpdemux/stream_vcd.c	Sun Apr 30 16:54:34 2006 +0200
@@ -94,6 +94,9 @@ static int open_s(stream_t *stream,int m
     m_struct_free(&stream_opts,opts);
     return STREAM_ERROR;
   }
+#ifdef __linux__
+  ioctl(f, CDROM_LOCKDOOR, 0);
+#endif
 
   vcd = vcd_read_toc(f);
   if(!vcd) {
diff -r 7d7fcdab76a3 libmpdvdkit2/device.c
--- a/libmpdvdkit2/device.c	Sun Apr 30 16:54:34 2006 +0200
+++ b/libmpdvdkit2/device.c	Sun Apr 30 16:54:34 2006 +0200
@@ -58,6 +58,11 @@
 #   include <sys/uio.h>                                      /* struct iovec */
 #endif
 
+#ifdef __linux__
+#include <linux/cdrom.h>
+#include <sys/ioctl.h>
+#endif
+
 #include "dvdcss.h"
 
 #include "common.h"
@@ -266,6 +271,9 @@ static int libc_open ( dvdcss_t dvdcss, 
         print_error( dvdcss, "failed to open device" );
         return -1;
     }
+#ifdef __linux__
+    ioctl(dvdcss->i_fd, CDROM_LOCKDOOR, 0);
+#endif
 
     dvdcss->i_pos = 0;
 
