diff -Naur MPlayer-20060429.orig/libmenu/menu_filesel.c MPlayer-20060429/libmenu/menu_filesel.c
--- MPlayer-20060429.orig/libmenu/menu_filesel.c	2006-05-06 15:36:36.000000000 +0200
+++ MPlayer-20060429/libmenu/menu_filesel.c	2006-05-06 15:37:25.000000000 +0200
@@ -208,6 +208,8 @@
   while ((dp = readdir(dirp)) != NULL) {
     if(dp->d_name[0] == '.' && strcmp(dp->d_name,"..") != 0)
       continue;
+    if (!strcmp (dp->d_name,"..") && !strcmp (mpriv->dir, "/mnt/"))
+      continue;
     mylstat(args,dp->d_name,&st);
     if (file_filter && extensions && !S_ISDIR(st.st_mode)) {
       if((ext = strrchr(dp->d_name,'.')) == NULL)
