diff -Naur MPlayer-r23358.orig/libmenu/menu_filesel.c MPlayer-r23358/libmenu/menu_filesel.c
--- MPlayer-r23358.orig/libmenu/menu_filesel.c	2007-05-25 17:04:17.000000000 +0200
+++ MPlayer-r23358/libmenu/menu_filesel.c	2007-05-25 17:05:15.000000000 +0200
@@ -298,8 +298,6 @@
 static void read_cmd(menu_t* menu,int cmd) {
   mp_cmd_t* c = NULL;
   switch(cmd) {
-  case MENU_CMD_LEFT:
-    mpriv->p.current = mpriv->p.menu; // Hack : we consider that the first entry is ../
   case MENU_CMD_RIGHT:
   case MENU_CMD_OK: {
     // Directory
