diff -Naur MPlayer-20060429.orig/libmenu/menu.c MPlayer-20060429/libmenu/menu.c
--- MPlayer-20060429.orig/libmenu/menu.c	2006-05-05 16:36:10.000000000 +0200
+++ MPlayer-20060429/libmenu/menu.c	2006-05-05 16:37:24.000000000 +0200
@@ -200,8 +200,6 @@
     break;
   case KEY_LEFT:
   case JOY_AXIS0_MINUS:
-    menu->read_cmd(menu,MENU_CMD_LEFT);
-    break;
   case KEY_ESC:
   case JOY_BTN1:
     menu->read_cmd(menu,MENU_CMD_CANCEL);
