* applied upstream

diff -Naur MPlayer-r23358.orig/libmenu/menu_txt.c MPlayer-r23358/libmenu/menu_txt.c
--- MPlayer-r23358.orig/libmenu/menu_txt.c	2007-05-25 18:27:13.000000000 +0200
+++ MPlayer-r23358/libmenu/menu_txt.c	2007-05-25 18:27:32.000000000 +0200
@@ -62,6 +62,7 @@
       mpriv->cur_line = mpriv->num_lines - 1;
     break;
   case MENU_CMD_CANCEL:
+  case MENU_CMD_LEFT:
     menu->show = 0;
     menu->cl = 1;
     break;
