diff -Naur MPlayer-20051213.orig/libmpdemux/stream.c MPlayer-20051213/libmpdemux/stream.c
--- MPlayer-20051213.orig/libmpdemux/stream.c	2005-12-13 00:15:49.000000000 +0100
+++ MPlayer-20051213/libmpdemux/stream.c	2006-02-22 16:02:25.000000000 +0100
@@ -179,7 +179,17 @@
   int i,j,l,r;
   stream_info_t* sinfo;
   stream_t* s;
+  char *ext = NULL;
 
+  ext = strrchr (filename, '.');
+  if (ext && !strcasecmp (ext + 1, "ifo"))
+  {
+    extern char *dvd_device;
+    dvd_device = dirname (strdup (filename));
+    dvd_title = -1; /* play longest track */
+    filename = "dvd://";
+  }
+  
   for(i = 0 ; auto_open_streams[i] ; i++) {
     sinfo = auto_open_streams[i];
     if(!sinfo->protocols) {
