diff -Naur MPlayer-20050918.orig/libmpdemux/stream.c MPlayer-20050918/libmpdemux/stream.c
--- MPlayer-20050918.orig/libmpdemux/stream.c	2005-09-19 01:16:34.000000000 +0200
+++ MPlayer-20050918/libmpdemux/stream.c	2005-09-19 01:32:10.000000000 +0200
@@ -179,6 +179,13 @@
   int i,j,l,r;
   stream_info_t* sinfo;
   stream_t* s;
+  char *ext;
+  int ifo = 0;
+  extern char *dvd_device;
+
+  ext = strrchr(filename, '.');
+  if (ext && !strcasecmp(ext+1, "ifo"))
+    ifo = 1;
 
   for(i = 0 ; auto_open_streams[i] ; i++) {
     sinfo = auto_open_streams[i];
@@ -190,7 +197,12 @@
       l = strlen(sinfo->protocols[j]);
       // l == 0 => Don't do protocol matching (ie network and filenames)
       if((l == 0) || ((strncmp(sinfo->protocols[j],filename,l) == 0) &&
-		      (strncmp("://",filename+l,3) == 0))) {
+		      (strncmp("://",filename+l,3) == 0))
+                  || (ifo && (strcmp(sinfo->protocols[j],"dvd")))) {
+        if(ifo) {
+          dvd_device = strdup (filename);
+          dvd_title = 1;
+        }
 	*file_format = DEMUXER_TYPE_UNKNOWN;
 	s = open_stream_plugin(sinfo,filename,mode,options,file_format,&r);
 	if(s) return s;
