diff -Naur MPlayer-1.0rc1.orig/libmenu/menu_param.c MPlayer-1.0rc1/libmenu/menu_param.c
--- MPlayer-1.0rc1.orig/libmenu/menu_param.c	2007-05-05 20:40:48.000000000 +0200
+++ MPlayer-1.0rc1/libmenu/menu_param.c	2007-05-05 20:41:08.000000000 +0200
@@ -72,6 +72,7 @@
 #define OPT_VERSION "Version"
 #define OPT_IP "IP"
 #define OPT_MAC "MAC"
+#define OPT_SCREEN_SIZE "ScreenSize"
 
 #define OPT_NAME "name"
 #define OPT_VCODEC "vcodec"
@@ -313,6 +314,12 @@
           if_freenameindex (ifs);
           close (fd);
         }
+      }
+      else if (!strcmp (name, OPT_SCREEN_SIZE)) {
+        char tmp[32];
+        extern int vo_screenwidth, vo_screenheight;
+        sprintf (tmp, "%dx%d", vo_screenwidth, vo_screenheight);
+        val = strdup (tmp);
       } else if (!strcmp (name, OPT_NAME))
         val = get_metadata (META_NAME);
       else if (!strcmp (name, OPT_VCODEC))
