diff -Naur MPlayer-r23418.orig/libmenu/menu_param.c MPlayer-r23418/libmenu/menu_param.c
--- MPlayer-r23418.orig/libmenu/menu_param.c	2007-06-02 16:51:54.000000000 +0200
+++ MPlayer-r23418/libmenu/menu_param.c	2007-06-02 16:52:58.000000000 +0200
@@ -76,6 +76,7 @@
 #define OPT_MPREV "MPlayerRev"
 #define OPT_IP "IP"
 #define OPT_MAC "MAC"
+#define OPT_SCREEN_SIZE "ScreenSize"
 
 static void entry_set_text(menu_t* menu, list_entry_t* e) {
   char* val = e->txt ? property_expand_string(menu->ctx, e->txt) :
@@ -252,6 +253,12 @@
         val = strdup(VERSION);
       else if (is_part(name) || is_cdrom(name))
         val = strdup(part_size(name));
+      else if (!strcmp (name, OPT_SCREEN_SIZE)) {
+        char tmp[32];
+        extern int vo_screenwidth, vo_screenheight;
+        sprintf (tmp, "%dx%d", vo_screenwidth, vo_screenheight);
+        val = strdup (tmp);
+      }
     }
     if (val) {
       int l = strlen(name) + 2 + strlen(val) + 1;
