diff -r 90db519a2f3f mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:43 2006 +0200
+++ b/mplayer.c	Sun Apr 30 16:54:44 2006 +0200
@@ -287,6 +287,7 @@
 
 static char* bg_video = NULL;
 demuxer_t *bg_demuxer = NULL;
+char* info_fifo=NULL;
 
 // cache2:
        int stream_cache_size=-1;
@@ -921,6 +921,66 @@ static int libmpdemux_was_interrupted(in
 }
 
 #define mp_basename(s) (strrchr(s,'\\')==NULL?(mp_basename2(s)):(strrchr(s,'\\')+1))
+
+void dump_stream_info()
+{
+  int fd;
+  FILE * log_file;
+
+  if(!info_fifo) return;
+
+  mkfifo(info_fifo, 0644);
+  fd = open(info_fifo, O_RDWR | O_NONBLOCK);
+  if(fd < 0) return;
+  log_file = fdopen(fd, "w");
+  if(!log_file) return;
+
+  fprintf(log_file, "ID_FILENAME=%s\n", mp_basename(filename));
+  fprintf(log_file, "ID_DEMUXER=%s\n", demuxer->desc->name);
+  fprintf(log_file, "ID_LENGTH=%.2lf\n", demuxer_get_time_length(demuxer));
+  fprintf(log_file, "ID_VIDEO_PERCENT=%d\n", demuxer_get_percent_pos(demuxer));
+
+  if (sh_video) {
+    /* Assume FOURCC if all bytes >= 0x20 (' ') */
+    if (sh_video->format >= 0x20202020)
+        fprintf(log_file, "ID_VIDEO_FORMAT=%.4s\n", (char *)&sh_video->format);
+    else
+        fprintf(log_file, "ID_VIDEO_FORMAT=0x%08X\n", sh_video->format);
+    fprintf(log_file, "ID_VIDEO_BITRATE=%d\n", sh_video->i_bps*8);
+    fprintf(log_file, "ID_VIDEO_WIDTH=%d\n", sh_video->disp_w);
+    fprintf(log_file, "ID_VIDEO_HEIGHT=%d\n", sh_video->disp_h);
+    fprintf(log_file, "ID_VIDEO_FPS=%5.3f\n", sh_video->fps);
+    fprintf(log_file, "ID_VIDEO_ASPECT=%1.4f\n", sh_video->aspect);
+  }
+  if (sh_audio) {
+    char *info;
+    if ((info = demux_info_get(demuxer, "Title")) != NULL)
+      fprintf(log_file, "ID_AUDIO_TITLE=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Artist")) != NULL)
+      fprintf(log_file, "ID_AUDIO_ARTIST=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Album")) != NULL)
+      fprintf(log_file, "ID_AUDIO_ALBUM=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Year")) != NULL)
+      fprintf(log_file, "ID_AUDIO_YEAR=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Comment")) != NULL)
+      fprintf(log_file, "ID_AUDIO_COMMENT=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Track")) != NULL)
+      fprintf(log_file, "ID_AUDIO_TRACK=%s\n", info );
+    if ((info = demux_info_get(demuxer, "Genre")) != NULL)
+      fprintf(log_file, "ID_AUDIO_GENRE=%s\n", info );
+    if (sh_audio->codec)
+      fprintf(log_file, "ID_AUDIO_CODEC=%s\n", sh_audio->codec->name);
+    /* Assume FOURCC if all bytes >= 0x20 (' ') */
+    if (sh_audio->format >= 0x20202020)
+      fprintf(log_file, "ID_AUDIO_FORMAT=%.4s\n", (char *)&sh_audio->format);
+    else
+      fprintf(log_file, "ID_AUDIO_FORMAT=%d\n", sh_audio->format);
+    fprintf(log_file, "ID_AUDIO_BITRATE=%d\n", sh_audio->i_bps*8);
+    fprintf(log_file, "ID_AUDIO_RATE=%d\n", sh_audio->samplerate);
+    fprintf(log_file, "ID_AUDIO_NCH=%d\n", sh_audio->channels);
+  }
+  fclose(log_file);
+}
 
 int playtree_add_playlist(play_tree_t* entry)
 {
@@ -5517,6 +5593,8 @@ if (stream->type==STREAMTYPE_DVDNAV && d
     loop_seek = 1;
   }
 
+  dump_stream_info();
+
 if(rel_seek_secs || abs_seek_pos){
   current_module="seek";
   if(demux_seek(demuxer,rel_seek_secs,audio_delay,abs_seek_pos)){
+++ a/cfg-mplayer.h	2007-04-28 14:08:07.000000000 +0200
--- a/cfg-mplayer.h	2007-04-28 14:08:07.000000000 +0200
@@ -342,6 +342,7 @@
 #endif
 
 	{"bgvideo", &bg_video, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"info-fifo", &info_fifo, CONF_TYPE_STRING, 0, 0, 0, NULL},

 	{"noloop", &loop_times, CONF_TYPE_FLAG, 0, 0, -1, NULL},
 	{"loop", &loop_times, CONF_TYPE_INT, CONF_RANGE, -1, 10000, NULL},
