diff -ur MPlayer-1.0pre7.orig/cfg-common.h MPlayer-1.0pre7/cfg-common.h
--- MPlayer-1.0pre7.orig/cfg-common.h	Tue Jul 26 10:47:42 2005
+++ MPlayer-1.0pre7/cfg-common.h	Tue Jul 26 10:49:38 2005
@@ -22,7 +22,8 @@
 #endif
 	{"vcd", "-vcd N is deprecated, use vcd://N instead.\n", CONF_TYPE_PRINT, CONF_NOCFG ,0,0, NULL},
 	{"cuefile", "-cuefile is deprecated, use cue://filename:N where N is the track number.\n", CONF_TYPE_PRINT, 0, 0, 0, NULL},
-	{"cdrom-device", &cdrom_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"vcd-device", &vcd_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"cdda-device", &cdda_device, CONF_TYPE_STRING, 0, 0, 0, NULL},
 #ifdef USE_DVDNAV
 	{"dvdnav", "-dvdnav is deprecated, use dvdnav:// instead.\n", CONF_TYPE_PRINT, 0, 0, 1, NULL},
 	{"skipopening", &dvd_nav_skip_opening, CONF_TYPE_FLAG, 0, 0, 1, NULL},
@@ -432,7 +433,7 @@
 extern int vivo_param_width;
 extern int vivo_param_height;
 extern int vivo_param_vformat;
-extern char *dvd_device, *cdrom_device;
+extern char *dvd_device, *vcd_device, *cdda_device;
 
 m_option_t vivoopts_conf[]={
 	{"version", &vivo_param_version, CONF_TYPE_INT, 0, 0, 0, NULL},
diff -ur MPlayer-1.0pre7.orig/libmpdemux/cdda.c MPlayer-1.0pre7/libmpdemux/cdda.c
--- MPlayer-1.0pre7.orig/libmpdemux/cdda.c	Tue Jul 26 10:44:52 2005
+++ MPlayer-1.0pre7/libmpdemux/cdda.c	Tue Jul 26 10:50:45 2005
@@ -13,7 +13,7 @@
 
 #include "cdd.h"
 
-extern char *cdrom_device;
+extern char *cdda_device;
 
 static struct cdda_params {
   int speed;
@@ -105,8 +105,8 @@
   }
 
   if(!p->device) {
-    if (cdrom_device)
-      p->device = strdup(cdrom_device);
+    if (cdda_device)
+      p->device = strdup(cdda_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
diff -ur MPlayer-1.0pre7.orig/libmpdemux/open.c MPlayer-1.0pre7/libmpdemux/open.c
--- MPlayer-1.0pre7.orig/libmpdemux/open.c	Tue Jul 26 10:44:52 2005
+++ MPlayer-1.0pre7/libmpdemux/open.c	Tue Jul 26 10:50:14 2005
@@ -42,7 +42,8 @@
 int dvd_chapter_pos[99];
 int dvd_angle=1;
 char* dvd_device=NULL;
-char* cdrom_device=NULL;
+char* vcd_device=NULL;
+char* cdda_device=NULL;
 
 #ifdef USE_DVDNAV
 #include "dvdnav_stream.h"
diff -ur MPlayer-1.0pre7.orig/libmpdemux/stream_vcd.c MPlayer-1.0pre7/libmpdemux/stream_vcd.c
--- MPlayer-1.0pre7.orig/libmpdemux/stream_vcd.c	Tue Jul 26 10:44:52 2005
+++ MPlayer-1.0pre7/libmpdemux/stream_vcd.c	Tue Jul 26 10:50:31 2005
@@ -25,7 +25,7 @@
 #include "vcd_read.h"
 #endif
 
-extern char *cdrom_device;
+extern char *vcd_device;
 
 static struct stream_priv_s {
   int track;
@@ -80,8 +80,8 @@
   }
 
   if (!p->device) {
-    if(cdrom_device)
-      p->device = strdup(cdrom_device);
+    if(vcd_device)
+      p->device = strdup(vcd_device);
     else
       p->device = strdup(DEFAULT_CDROM_DEVICE);
   }
diff -ur MPlayer-1.0pre7.orig/mplayer.c MPlayer-1.0pre7/mplayer.c
--- MPlayer-1.0pre7.orig/mplayer.c	Tue Jul 26 10:44:52 2005
+++ MPlayer-1.0pre7/mplayer.c	Tue Jul 26 10:48:11 2005
@@ -2959,7 +2959,7 @@
       play_tree_t *cdda, *tracks;
       int nb_tracks, i; 
 
-      nb_tracks = get_cdda_tracks_num (cdrom_device ? cdrom_device : DEFAULT_CDROM_DEVICE);
+      nb_tracks = get_cdda_tracks_num (cdda_device ? cdda_device : DEFAULT_CDROM_DEVICE);
       if (nb_tracks <= 0)
         break;
 
