Index: mplayer/libmenu/menu_filesel.c
===================================================================
--- mplayer/libmenu/menu_filesel.c	(revision 25316)
+++ mplayer/libmenu/menu_filesel.c	(working copy)
@@ -180,7 +180,7 @@
   }
 }
 
-static int open_dir(menu_t* menu,char* args) {
+static int _open_dir(menu_t* menu,char* args) {
   char **namelist, **tp;
   struct dirent *dp;
   struct stat st;
@@ -291,7 +291,13 @@
 
   return 1;
 }
-    
+static int open_dir(menu_t* menu,char* args) {
+  if (_open_dir(menu, args))
+    return 1;
+  if (menu_chroot && _open_dir(menu, menu_chroot))
+    return 1;
+  return 0;
+}
 
 static char *action;
 
