diff -r 92ef6acb17e7 libmenu/menu_filesel.c
--- a/libmenu/menu_filesel.c	Sun Apr 30 16:54:46 2006 +0200
+++ b/libmenu/menu_filesel.c	Sun Apr 30 16:54:46 2006 +0200
@@ -192,7 +192,7 @@ free_extensions (char **extensions)
   }
 }
 
-static int open_dir(menu_t* menu,char* args) {
+static int _open_dir(menu_t* menu,char* args) {
   char **namelist, **tp;
   struct dirent *dp;
   struct stat st;
@@ -304,6 +304,9 @@ bailout:
   return 1;
 }
     
+static int open_dir(menu_t* menu,char* args) {
+  return (_open_dir(menu, args) || _open_dir(menu, "/mnt/"));
+}
 
 static char *action;
 
