diff -Naur MPlayer-1.0pre6.orig/cfg-common.h MPlayer-1.0pre6/cfg-common.h
--- MPlayer-1.0pre6.orig/cfg-common.h	2004-12-30 19:38:06.690882528 +0100
+++ MPlayer-1.0pre6/cfg-common.h	2004-12-30 19:46:18.780073600 +0100
@@ -254,6 +254,7 @@
 #endif
 #ifdef USE_OSD
 	{"font", &font_name, CONF_TYPE_STRING, 0, 0, 0, NULL},
+	{"subfont", &sub_font_name, CONF_TYPE_STRING, 0, 0, 0, NULL},
 	{"ffactor", &font_factor, CONF_TYPE_FLOAT, CONF_RANGE, 0.0, 10.0, NULL},
  	{"subpos", &sub_pos, CONF_TYPE_INT, CONF_RANGE, 0, 100, NULL},
 	{"subalign", &sub_alignment, CONF_TYPE_INT, CONF_RANGE, 0, 2, NULL},
diff -Naur MPlayer-1.0pre6.orig/libvo/font_load.h MPlayer-1.0pre6/libvo/font_load.h
--- MPlayer-1.0pre6.orig/libvo/font_load.h	2004-12-30 19:38:06.781868696 +0100
+++ MPlayer-1.0pre6/libvo/font_load.h	2004-12-30 19:46:18.780073600 +0100
@@ -62,6 +62,7 @@
 } font_desc_t;
 
 extern font_desc_t* vo_font;
+extern font_desc_t* sub_font;
 
 #ifdef HAVE_FREETYPE
 
diff -Naur MPlayer-1.0pre6.orig/libvo/font_load_ft.c MPlayer-1.0pre6/libvo/font_load_ft.c
--- MPlayer-1.0pre6.orig/libvo/font_load_ft.c	2004-12-30 19:38:06.781868696 +0100
+++ MPlayer-1.0pre6/libvo/font_load_ft.c	2004-12-30 19:46:18.781073448 +0100
@@ -938,7 +938,7 @@
 {
     FT_Vector kern;
     
-    if (!vo_font->dynamic) return 0;
+    if (!desc->dynamic) return 0;
     if (prevc < 0 || c < 0) return 0;
     if (desc->font[prevc] != desc->font[c]) return 0;
     if (desc->font[prevc] == -1 || desc->font[c] == -1) return 0;
diff -Naur MPlayer-1.0pre6.orig/libvo/sub.c MPlayer-1.0pre6/libvo/sub.c
--- MPlayer-1.0pre6.orig/libvo/sub.c	2004-12-30 19:38:06.783868392 +0100
+++ MPlayer-1.0pre6/libvo/sub.c	2004-12-30 19:46:18.782073296 +0100
@@ -55,6 +55,7 @@
 
 //static int vo_font_loaded=-1;
 font_desc_t* vo_font=NULL;
+font_desc_t* sub_font=NULL;
 
 unsigned char* vo_osd_text=NULL;
 int sub_unicode=0;
@@ -342,7 +343,7 @@
    
    obj->flags|=OSDFLAG_CHANGED|OSDFLAG_VISIBLE;
 
-   if(!vo_sub || !vo_font || !sub_visibility || (vo_font->font[40]<0)){
+   if(!vo_sub || !sub_font || !sub_visibility || (sub_font->font[40]<0)){
        obj->flags&=~OSDFLAG_VISIBLE;
        return;
    }
@@ -352,7 +353,7 @@
 
       // too long lines divide into a smaller ones
       i=k=lasth=0;
-      h=vo_font->height;
+      h=sub_font->height;
       lastStripPosition=-1;
       l=vo_sub->lines;
 
@@ -363,7 +364,7 @@
 	int *char_seq, char_position, xlimit = dxs * sub_width_p / 100, counter;
 
       while (l) {
-	    xsize = -vo_font->charspace;
+	    xsize = -sub_font->charspace;
 	  l--;
 	  t=vo_sub->text[i++];	  
 	  len=strlen(t)-1;
@@ -394,7 +395,7 @@
 		 mp_msg(MSGT_OSD,MSGL_WARN,"\nMAX_UCS exceeded!\n");
 	      }
 	      if (!c) c++; // avoid UCS 0
-	      render_one_glyph(vo_font, c);
+	      render_one_glyph(sub_font, c);
 
 		if (c == ' ') {
 		    struct osd_text_t *tmp_ott = (struct osd_text_t *) calloc(1, sizeof(struct osd_text_t));
@@ -405,7 +406,7 @@
 			tmp_ott->prev = cp_ott;
 			cp_ott->next = tmp_ott;
 			tmp_ott->osd_kerning =
-			    vo_font->charspace + vo_font->width[' '];
+			    sub_font->charspace + sub_font->width[' '];
 			cp_ott = tmp_ott;
 		    }
 		    tmp_ott->osd_length = xsize;
@@ -417,16 +418,16 @@
 		    xsize = 0;
 		    prevc = c;
 		} else {
-		    int delta_xsize = vo_font->width[c] + vo_font->charspace + kerning(vo_font, prevc, c);
+		    int delta_xsize = sub_font->width[c] + sub_font->charspace + kerning(sub_font, prevc, c);
 		    
 		    if (xsize + delta_xsize <= dxs) {
 			if (!x) x = 1;
 			prevc = c;
 			char_seq[char_position++] = c;
 			xsize += delta_xsize;
-			if ((!suboverlap_enabled) && ((font = vo_font->font[c]) >= 0)) {
-			    if (vo_font->pic_a[font]->h > h) {
-				h = vo_font->pic_a[font]->h;
+			if ((!suboverlap_enabled) && ((font = sub_font->font[c]) >= 0)) {
+			    if (sub_font->pic_a[font]->h > h) {
+				h = sub_font->pic_a[font]->h;
 			    }
 			}
 		    } else {
@@ -448,7 +449,7 @@
 		    tmp_ott->prev = cp_ott;
 		    cp_ott->next = tmp_ott;
 		    tmp_ott->osd_kerning =
-			vo_font->charspace + vo_font->width[' '];
+			sub_font->charspace + sub_font->width[' '];
 		    cp_ott = tmp_ott;
 		}
 		tmp_ott->osd_length = xsize;
@@ -457,7 +458,7 @@
 		for (counter = 0; counter < char_position; ++counter)
 		    tmp_ott->text[counter] = char_seq[counter];
 		char_position = 0;
-		xsize = -vo_font->charspace;
+		xsize = -sub_font->charspace;
 	    }
 
 	    if (osl != NULL) {
@@ -479,7 +480,7 @@
 			tmp->prev = tmp_otp;
 			tmp_otp = tmp;
 			tmp_otp->ott = tmp_ott;
-			value = -2 * vo_font->charspace - vo_font->width[' '];
+			value = -2 * sub_font->charspace - sub_font->width[' '];
 		    } else {
 			tmp_otp->value = value;
 			exit = 1;
@@ -580,7 +581,7 @@
 		break;
 
 	    if (h > obj->y) {	// out of the screen so end parsing
-		obj->y -= lasth - vo_font->height;	// correct the y position
+		obj->y -= lasth - sub_font->height;	// correct the y position
 		break;
 	    }
 	    xsize = tmp_otp->value;
@@ -597,17 +598,17 @@
 			break;
 		    }
 		    c = tmp_ott->text[counter];
-		    render_one_glyph(vo_font, c);
+		    render_one_glyph(sub_font, c);
 		    obj->params.subtitle.utbl[utblc++] = c;
 		    k++;
 		}
 		obj->params.subtitle.utbl[utblc++] = ' ';
 	    }
 	    obj->params.subtitle.utbl[utblc - 1] = 0;
-	    obj->y -= vo_font->height;
+	    obj->y -= sub_font->height;
 	}
 	if(obj->params.subtitle.lines)
-	    obj->y = dys - ((obj->params.subtitle.lines - 1) * vo_font->height + vo_font->pic_a[vo_font->font[40]]->h);
+	    obj->y = dys - ((obj->params.subtitle.lines - 1) * sub_font->height + sub_font->pic_a[sub_font->font[40]]->h);
 	
 	// free memory
 	if (otp_sub != NULL) {
@@ -646,7 +647,7 @@
     obj->bbox.x1=xmin;
     obj->bbox.x2=xmax;
     obj->bbox.y1=obj->y;
-//    obj->bbox.y2=obj->y+obj->params.subtitle.lines*vo_font->height;
+//    obj->bbox.y2=obj->y+obj->params.subtitle.lines*sub_font->height;
     obj->flags|=OSDFLAG_BBOX;
 
     alloc_buf(obj);
@@ -692,18 +693,18 @@
 	    }
 	 prevc = -1;
 	 while ((c=obj->params.subtitle.utbl[j++])){
-	       x += kerning(vo_font,prevc,c);
-	       if ((font=vo_font->font[c])>=0)
+	       x += kerning(sub_font,prevc,c);
+	       if ((font=sub_font->font[c])>=0)
 		  draw_alpha_buf(obj,x,y,
-			     vo_font->width[c],
-			     vo_font->pic_a[font]->h+y<obj->dys ? vo_font->pic_a[font]->h : obj->dys-y,
-			     vo_font->pic_b[font]->bmp+vo_font->start[c],
-			     vo_font->pic_a[font]->bmp+vo_font->start[c],
-			     vo_font->pic_a[font]->w);
-	       x+=vo_font->width[c]+vo_font->charspace;
+			     sub_font->width[c],
+			     sub_font->pic_a[font]->h+y<obj->dys ? sub_font->pic_a[font]->h : obj->dys-y,
+			     sub_font->pic_b[font]->bmp+sub_font->start[c],
+			     sub_font->pic_a[font]->bmp+sub_font->start[c],
+			     sub_font->pic_a[font]->w);
+	       x+=sub_font->width[c]+sub_font->charspace;
                prevc = c;
 	    }
-         y+=vo_font->height;
+         y+=sub_font->height;
 	}
     }
     
diff -Naur MPlayer-1.0pre6.orig/mencoder.c MPlayer-1.0pre6/mencoder.c
--- MPlayer-1.0pre6.orig/mencoder.c	2004-12-30 19:38:06.692882224 +0100
+++ MPlayer-1.0pre6/mencoder.c	2004-12-30 19:46:18.783073144 +0100
@@ -188,6 +188,7 @@
 
 // sub:
 char *font_name=NULL;
+char *sub_font_name=NULL;
 #ifdef HAVE_FONTCONFIG
 extern int font_fontconfig;
 #endif
diff -Naur MPlayer-1.0pre6.orig/mplayer.c MPlayer-1.0pre6/mplayer.c
--- MPlayer-1.0pre6.orig/mplayer.c	2004-12-30 19:45:55.765572336 +0100
+++ MPlayer-1.0pre6/mplayer.c	2004-12-30 19:46:18.785072840 +0100
@@ -295,6 +295,7 @@
 
 // sub:
 char *font_name=NULL;
+char *sub_font_name=NULL;
 #ifdef HAVE_FONTCONFIG
 extern int font_fontconfig;
 #endif
@@ -1280,6 +1281,10 @@
        if(!vo_font)
        vo_font=read_font_desc(MPLAYER_DATADIR "/font/font.desc",font_factor,verbose>1);
   }
+  if (sub_font_name)
+    sub_font = read_font_desc(sub_font_name, font_factor, verbose>1);
+  else
+    sub_font = vo_font;
 #ifdef HAVE_FONTCONFIG
   }
 #endif
diff -Naur MPlayer-1.0pre6.orig/mplayer.h MPlayer-1.0pre6/mplayer.h
--- MPlayer-1.0pre6.orig/mplayer.h	2004-12-30 19:38:06.693882072 +0100
+++ MPlayer-1.0pre6/mplayer.h	2004-12-30 19:46:18.785072840 +0100
@@ -23,6 +23,7 @@
 extern int osd_visible;
 
 extern char * font_name;
+extern char * sub_font_name;
 extern float  font_factor;
 extern float movie_aspect;
 extern float force_fps;
