* commited upstream on 20050622
diff -Naur MPlayer-1.0pre7.orig/configure MPlayer-1.0pre7/configure
--- MPlayer-1.0pre7.orig/configure	Tue Jun 21 00:16:14 2005
+++ MPlayer-1.0pre7/configure	Tue Jun 21 00:19:52 2005
@@ -5012,9 +5012,11 @@
 	cat > $TMPC << EOF
 #include <stdio.h>
+/* workaround for fribidi 0.10.4 and below */
+#define FRIBIDI_CHARSET_UTF8 FRIBIDI_CHAR_SET_UTF8
 #include <fribidi/fribidi.h>
 int main()
 {
-    if(fribidi_parse_charset("UTF-8") != FRIBIDI_CHARSET_UTF8) {
+    if(fribidi_parse_charset("UTF-8") != FRIBIDI_CHAR_SET_UTF8) {
        printf("Fribidi headers are not consistents with the library!\n");
        exit(1);
     }
