diff -Naur MPlayer-20060223.orig/cfg-common.h MPlayer-20060223/cfg-common.h
--- MPlayer-20060223.orig/cfg-common.h	2006-03-04 15:22:00.000000000 +0100
+++ MPlayer-20060223/cfg-common.h	2006-03-04 15:24:29.000000000 +0100
@@ -356,6 +356,9 @@
 #endif
 
 #ifdef USE_TV
+#ifdef FOR_MENCODER
+extern m_option_t tvopts_conf[];
+#else
 m_option_t tvopts_conf[]={
 	{"on", "-tv on is deprecated, use tv:// instead.\n", CONF_TYPE_PRINT, 0, 0, 0, NULL},
 	{"immediatemode", &tv_param_immediate, CONF_TYPE_INT, CONF_RANGE, 0, 1, NULL},
@@ -401,6 +404,7 @@
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
 #endif
+#endif
 
 #ifdef HAS_DVBIN_SUPPORT
 #include "libmpdemux/dvbin.h"
@@ -422,6 +426,9 @@
 extern float sws_chr_sharpen;
 extern float sws_lum_sharpen;
 
+#ifdef FOR_MENCODER
+extern m_option_t scaler_filter_conf[];
+#else
 m_option_t scaler_filter_conf[]={
 	{"lgb", &sws_lum_gblur, CONF_TYPE_FLOAT, 0, 0, 100.0, NULL},
 	{"cgb", &sws_chr_gblur, CONF_TYPE_FLOAT, 0, 0, 100.0, NULL},
@@ -431,6 +438,7 @@
 	{"cs", &sws_chr_sharpen, CONF_TYPE_FLOAT, 0, -100.0, 100.0, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
 /* VIVO demuxer options: */
 extern int vivo_param_version;
@@ -443,6 +451,9 @@
 extern int vivo_param_vformat;
 extern char *dvd_device, *vcd_device, *cdda_device;
 
+#ifdef FOR_MENCODER
+extern m_option_t vivoopts_conf[];
+#else
 m_option_t vivoopts_conf[]={
 	{"version", &vivo_param_version, CONF_TYPE_INT, 0, 0, 0, NULL},
 	/* audio options */
@@ -456,6 +467,7 @@
 	{"vformat", &vivo_param_vformat, CONF_TYPE_INT, 0, 0, 0, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
 extern int    mf_w;
 extern int    mf_h;
@@ -464,6 +476,9 @@
 extern m_obj_settings_t* vf_settings;
 extern m_obj_list_t vf_obj_list;
 
+#ifdef FOR_MENCODER
+extern m_option_t mfopts_conf[];
+#else
 m_option_t mfopts_conf[]={
         {"on", "-mf on is deprecated, use mf:// instead.\n", CONF_TYPE_PRINT, 0, 0, 1, NULL},
         {"w", &mf_w, CONF_TYPE_INT, 0, 0, 0, NULL},
@@ -472,17 +487,27 @@
         {"type", &mf_type, CONF_TYPE_STRING, 0, 0, 0, NULL},
         {NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 						
 extern m_obj_settings_t* vo_plugin_args;
 
 #include "libaf/af.h"
 extern af_cfg_t af_cfg; // Audio filter configuration, defined in libmpcodecs/dec_audio.c
+
+#ifdef FOR_MENCODER
+extern m_option_t audio_filter_conf[];
+#else
 m_option_t audio_filter_conf[]={       
 	{"list", &af_cfg.list, CONF_TYPE_STRING_LIST, 0, 0, 0, NULL},
         {"force", &af_cfg.force, CONF_TYPE_INT, CONF_RANGE, 0, 7, NULL},
 	{NULL, NULL, 0, 0, 0, 0, NULL}
 };
+#endif
 
+
+#ifdef FOR_MENCODER
+extern m_option_t msgl_config[];
+#else
 m_option_t msgl_config[]={
 	{ "all", &mp_msg_level_all, CONF_TYPE_INT, CONF_RANGE, -1, 9, NULL},
 
@@ -569,6 +594,7 @@
         "   muxer      - muxer layer\n"
         "\n", CONF_TYPE_PRINT, CONF_NOCFG, 0, 0, NULL},
 };
+#endif
 
 #ifdef WIN32
 
diff -Naur MPlayer-20060223.orig/cfg-mencoder.h MPlayer-20060223/cfg-mencoder.h
--- MPlayer-20060223.orig/cfg-mencoder.h	2006-03-04 15:22:00.000000000 +0100
+++ MPlayer-20060223/cfg-mencoder.h	2006-03-04 15:24:29.000000000 +0100
@@ -3,7 +3,9 @@
  * config for cfgparser
  */
 
+#define FOR_MENCODER 1
 #include "cfg-common.h"
+#undef FOR_MENCODER
 
 #ifdef USE_FAKE_MONO
 extern int fakemono; // defined in dec_audio.c
diff -Naur MPlayer-20060223.orig/Makefile MPlayer-20060223/Makefile
--- MPlayer-20060223.orig/Makefile	2006-03-04 15:22:00.000000000 +0100
+++ MPlayer-20060223/Makefile	2006-03-04 15:32:25.000000000 +0100
@@ -27,47 +27,45 @@
 DO_MAKE = @ for i in $(SUBDIRS); do $(MAKE) -C $$i $@; done
 endif
 
-SRCS_COMMON = asxparser.c \
-              codec-cfg.c \
-              cpudetect.c \
-              edl.c \
-              find_sub.c \
-              m_config.c \
-              m_option.c \
-              m_struct.c \
-              parser-cfg.c \
-              playtree.c \
-              playtreeparser.c \
-              spudec.c \
-              sub_cc.c \
-              subreader.c \
-              vobsub.c \
-
-SRCS_MENCODER = mencoder.c \
-                mp_msg-mencoder.c \
-                $(SRCS_COMMON) \
-                divx4_vbr.c \
-                libvo/aclib.c \
-                libvo/font_load.c \
-                libvo/font_load_ft.c \
-                libvo/osd.c \
-                libvo/sub.c \
-                parser-mecmd.c \
-                xvid_vbr.c \
-
-SRCS_MPLAYER = mplayer.c \
-               mp_msg.c \
-               $(SRCS_COMMON) \
-               mixer.c \
-               parser-mpcmd.c \
-               subopt-helper.c \
+SRCS =  mplayer.c \
+        mp_msg.c \
+        asxparser.c \
+        codec-cfg.c \
+        cpudetect.c \
+        edl.c \
+        find_sub.c \
+        m_config.c \
+        m_option.c \
+        m_struct.c \
+        parser-cfg.c \
+        playtree.c \
+        playtreeparser.c \
+        spudec.c \
+        sub_cc.c \
+        subreader.c \
+        vobsub.c \
+        mixer.c \
+        parser-mpcmd.c \
+        subopt-helper.c \
+
+ifeq ($(MENCODER),yes)
+SRCS += mencoder.c \
+        divx4_vbr.c \
+        libvo/aclib.c \
+        libvo/font_load.c \
+        libvo/font_load_ft.c \
+        libvo/osd.c \
+        libvo/sub.c \
+        parser-mecmd.c \
+        xvid_vbr.c \
+
+endif
 
 ifeq ($(UNRARLIB),yes)
-SRCS_COMMON += unrarlib.c
+SRCS += unrarlib.c
 endif
 
-OBJS_MENCODER = $(SRCS_MENCODER:.c=.o)
-OBJS_MPLAYER = $(SRCS_MPLAYER:.c=.o)
+OBJS = $(SRCS:.c=.o)
 
 VO_LIBS = $(AA_LIB) \
           $(X_LIB) \
@@ -356,17 +354,19 @@
 libmenu/libmenu.a:
 	$(MAKE) -C libmenu
 
-MPLAYER_DEP = $(OBJS_MPLAYER) $(COMMON_DEPS)
+DEP = $(OBJS) $(COMMON_DEPS)
 
 ifeq ($(LIBMENU),yes)
-MPLAYER_DEP += libmenu/libmenu.a
+DEP += libmenu/libmenu.a
 MENU_LIBS = libmenu/libmenu.a
 PARTS += libmenu
 else
 MENU_LIBS =
 endif
 
-MENCODER_DEP = $(OBJS_MENCODER) $(COMMON_DEPS) libmpcodecs/libmpencoders.a
+ifeq ($(MENCODER),yes)
+DEP += libmpcodecs/libmpencoders.a
+endif
 
 ifeq ($(VIDIX),yes)
 VIDIX_LIBS = vidix/libvidix.a
@@ -375,10 +375,17 @@
 endif
 
 ifeq ($(TARGET_WIN32),yes)
-OBJS_MPLAYER += osdep/mplayer-rc.o
+OBJS += osdep/mplayer-rc.o
 endif
 
-LIBS_MPLAYER = libvo/libvo.a \
+ifeq ($(MENCODER),yes)
+LIBS += libmpcodecs/libmpencoders.a \
+	$(ENCORE_LIB) \
+	$(MLIB_LIB) \
+ 
+endif
+
+LIBS        += libvo/libvo.a \
                libao2/libao2.a \
                $(MENU_LIBS) \
                $(VIDIX_LIBS) \
@@ -396,11 +403,11 @@
                $(MATH_LIB) \
                $(LIBC_LIB) \
 
-$(PRG):	$(MPLAYER_DEP)
+$(PRG):	$(DEP)
     ifeq ($(TARGET_WIN32),yes)
 	windres -o osdep/mplayer-rc.o osdep/mplayer.rc
     endif
-	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) $(LIBS_MPLAYER)
+	$(CC) $(CFLAGS) -o $(PRG) $(OBJS) $(LIBS)
 
 mplayer.exe.spec.c: libmpcodecs/libmpcodecs.a
 	winebuild -fPIC -o mplayer.exe.spec.c -exe mplayer.exe -mcui \
@@ -408,26 +415,14 @@
 	-L/usr/local/lib/wine -lkernel32
 
 mplayer.exe.so:	$(MPLAYER_DEP) mplayer.exe.spec.c
-	$(CC) $(CFLAGS) -Wall -shared  -Wl,-rpath,/usr/local/lib -Wl,-Bsymbolic  -o mplayer.exe.so $(OBJS_MPLAYER) mplayer.exe.spec.c libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) -lwine $(MATH_LIB) 
+	$(CC) $(CFLAGS) -Wall -shared  -Wl,-rpath,/usr/local/lib -Wl,-Bsymbolic  -o mplayer.exe.so $(OBJS) mplayer.exe.spec.c libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) -lwine $(MATH_LIB) 
 
 mplayer_wine.so:	$(MPLAYER_DEP)
-	$(CC) $(CFLAGS) -shared -Wl,-Bsymbolic -o mplayer_wine.so mplayer_wine.spec.c $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) -lwine $(ARCH_LIB) $(MATH_LIB)
+	$(CC) $(CFLAGS) -shared -Wl,-Bsymbolic -o mplayer_wine.so mplayer_wine.spec.c $(OBJS) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) -lwine $(ARCH_LIB) $(MATH_LIB)
 
 ifeq ($(MENCODER),yes)
-LIBS_MENCODER = libmpcodecs/libmpencoders.a \
-                $(ENCORE_LIB) \
-                $(COMMON_LIBS) \
-                $(EXTRA_LIB) \
-                $(MLIB_LIB) \
-                $(LIRC_LIB) \
-                $(LIRCC_LIB) \
-                $(ARCH_LIB) \
-                $(I18NLIBS) \
-                $(MATH_LIB) \
-                $(LIBC_LIB) \
-
-$(PRG_MENCODER): $(MENCODER_DEP)
-	$(CC) $(CFLAGS) -o $(PRG_MENCODER) $(OBJS_MENCODER) $(LIBS_MENCODER)
+$(PRG_MENCODER): $(PRG)
+	-ln -sf $(PRG) $(PRG_MENCODER)
 endif
 
 codecs.conf.h: $(PRG_CFG) etc/codecs.conf
@@ -441,8 +436,7 @@
 # in a recursive "make distclean" and we must wait for this "make distclean" to
 # finish before we can start building new object files.
 # help_mp.h is also required by a lot of files, so force generating it early.
-$(MPLAYER_DEP): version.h help_mp.h
-$(MENCODER_DEP): version.h help_mp.h
+$(DEP): version.h help_mp.h
 
 $(PRG_CFG): version.h codec-cfg.c codec-cfg.h
 	$(HOST_CC) $(HOST_CFLAGS) -I. codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(EXTRA_INC) $(I18NLIBS)
@@ -466,7 +460,7 @@
 		fi ; \
 	done
 ifeq ($(MENCODER),yes)
-	$(INSTALL) -m 755 $(INSTALLSTRIP) $(PRG_MENCODER) $(BINDIR)/$(PRG_MENCODER)
+	-ln -sf $(PRG) $(BINDIR)/$(PRG_MENCODER)
 	for i in $(MAN_LANG); do \
 		if test "$$i" = en ; then \
 			ln -sf mplayer.1 $(MANDIR)/man1/mencoder.1 ; \
@@ -537,7 +531,7 @@
 
 depend: help_mp.h
 	./version.sh `$(CC) -dumpversion`
-	$(CC) -MM $(CFLAGS) -DCODECS2HTML mplayer.c mencoder.c $(SRCS_MPLAYER) $(SRCS_MENCODER) 1>.depend
+	$(CC) -MM $(CFLAGS) -DCODECS2HTML mplayer.c mencoder.c $(SRCS) 1>.depend
 	@for a in $(PARTS); do $(MAKE) -C $$a dep; done
 
 # ./configure must be run if it changed in CVS
diff -Naur MPlayer-20060223.orig/mencoder.c MPlayer-20060223/mencoder.c
--- MPlayer-20060223.orig/mencoder.c	2006-03-04 15:22:00.000000000 +0100
+++ MPlayer-20060223/mencoder.c	2006-03-04 15:35:13.000000000 +0100
@@ -74,7 +74,7 @@
 
 #include "osdep/timer.h"
 
-#include "get_path.c"
+extern char *get_path(char *filename);
 
 #ifdef USE_LIBAVCODEC
 #ifdef USE_LIBAVCODEC_SO
@@ -85,30 +85,27 @@
 #endif
 
 #include "libmpcodecs/ae.h"
-int vo_doublebuffering=0;
-int vo_directrendering=0;
-int vo_config_count=0;
-int forced_subs_only=0;
+extern forced_subs_only;
 
 //--------------------------
 
 // cache2:
-int stream_cache_size=-1;
+extern int stream_cache_size;
 #ifdef USE_STREAM_CACHE
 extern int cache_fill_status;
 
-float stream_cache_min_percent=20.0;
-float stream_cache_seek_min_percent=50.0;
+extern float stream_cache_min_percent;
+extern float stream_cache_seek_min_percent;
 #else
 #define cache_fill_status 0
 #endif
 
-int audio_id=-1;
-int video_id=-1;
-int dvdsub_id=-2;
-int vobsub_id=-1;
-char* audio_lang=NULL;
-char* dvdsub_lang=NULL;
+extern int audio_id;
+extern int video_id;
+extern int dvdsub_id;
+extern int vobsub_id;
+extern char* audio_lang;
+extern char* dvdsub_lang;
 static char* spudec_ifo=NULL;
 
 static char** audio_codec_list=NULL;  // override audio codec
@@ -130,15 +127,15 @@
 //void resync_audio_stream(sh_audio_t *sh_audio){}
 
 extern int verbose; // must be global!
-int identify=0;
-int quiet=0;
-double video_time_usage=0;
-double vout_time_usage=0;
+extern int identify;
+extern int quiet;
+extern double video_time_usage;
+extern double vout_time_usage;
 double max_video_time_usage=0;
 double max_vout_time_usage=0;
 double cur_video_time_usage=0;
 double cur_vout_time_usage=0;
-int benchmark=0;
+extern int benchmark;
 
 extern int mp_msg_levels[MSGT_MAX];
 extern int mp_msg_level_all;
@@ -159,10 +156,10 @@
 static int ignore_start=0;
 static int audio_density=2;
 
-float force_fps=0;
+extern float force_fps;
 static float force_ofps=0; // set to 24 for inverse telecine
 static int skip_limit=-1;
-float playback_speed=1.0;
+extern float playback_speed;
 
 static int force_srate=0;
 static int audio_output_format=0;
@@ -185,22 +182,22 @@
 #include "libvo/sub.h"
 
 // sub:
-char *font_name=NULL;
-char *sub_font_name=NULL;
+extern char *font_name;
+extern char *sub_font_name;
 #ifdef HAVE_FONTCONFIG
 extern int font_fontconfig;
 #endif
-float font_factor=0.75;
-char **sub_name=NULL;
-float sub_delay=0;
-float sub_fps=0;
-int   sub_auto = 0;
-int   subcc_enabled=0;
-int   suboverlap_enabled = 1;
+extern float font_factor;
+extern char **sub_name;
+extern float sub_delay;
+extern float sub_fps;
+extern int   sub_auto;
+extern int   subcc_enabled;
+extern int   suboverlap_enabled;
 
 #ifdef USE_SUB
 static sub_data* subdata=NULL;
-float sub_last_pts = -303;
+extern float sub_last_pts;
 #endif
 
 int auto_expand=1;
@@ -298,7 +295,7 @@
     exit(level);
 }
 
-void parse_cfgfiles( m_config_t* conf )
+void me_parse_cfgfiles( m_config_t* conf )
 {
   char *conffile;
   if ((conffile = get_path("mencoder")) == NULL) {
@@ -359,7 +356,7 @@
 
 extern void print_wave_header(WAVEFORMATEX *h);
 
-int main(int argc,char* argv[]){
+int me_main(int argc,char* argv[]){
 
 stream_t* stream=NULL;
 demuxer_t* demuxer=NULL;
@@ -404,6 +401,7 @@
 
 audio_encoding_params_t aparams;
 audio_encoder_t *aencoder = NULL;
+ sub_auto = 0;
 
   mp_msg_init();
   mp_msg(MSGT_CPLAYER,MSGL_INFO, "MEncoder " VERSION " (C) 2000-2006 MPlayer Team\n");
@@ -462,7 +460,7 @@
 
  mconfig = m_config_new();
  m_config_register_options(mconfig,mencoder_opts);
- parse_cfgfiles(mconfig);
+ me_parse_cfgfiles(mconfig);
  filelist = m_config_parse_me_command_line(mconfig, argc, argv);
  if(!filelist) mencoder_exit(1, MSGTR_ErrorParsingCommandLine);
 
diff -Naur MPlayer-20060223.orig/mplayer.c MPlayer-20060223/mplayer.c
--- MPlayer-20060223.orig/mplayer.c	2006-03-04 15:22:00.000000000 +0100
+++ MPlayer-20060223/mplayer.c	2006-03-04 15:24:29.000000000 +0100
@@ -1314,7 +1314,7 @@
 }
 
 
-int main(int argc,char* argv[]){
+int mp_main(int argc,char* argv[]){
 
 
 char * mem_ptr;
@@ -4748,3 +4748,14 @@
 
 return 1;
 }
+
+int
+main (int argc, char **argv)
+{
+#ifdef HAVE_MENCODER
+  if (strstr (argv[0], "mencoder"))
+    return me_main (argc, argv);
+  else
+#endif
+    return mp_main (argc, argv);
+}
