diff -r 37c3bd5363cf input/input.c
--- a/input/input.c	Sun Apr 30 16:54:36 2006 +0200
+++ b/input/input.c	Sun Apr 30 16:54:37 2006 +0200
@@ -117,6 +117,7 @@ static mp_cmd_t mp_cmds[] = {
   { MP_CMD_VO_ONTOP, "vo_ontop", 0, { {MP_CMD_ARG_INT,{-1}}, {-1,{0}} } },
   { MP_CMD_VO_ROOTWIN, "vo_rootwin", 0, { {MP_CMD_ARG_INT,{-1}}, {-1,{0}} } },
   { MP_CMD_VO_BORDER, "vo_border", 0, { {MP_CMD_ARG_INT,{-1}}, {-1,{0}} } },
+  { MP_CMD_RECORD, "record", 0, { {-1,{0}} } },
   { MP_CMD_SCREENSHOT, "screenshot", 0, { {-1,{0}} } },
   { MP_CMD_PANSCAN, "panscan",1,  { {MP_CMD_ARG_FLOAT,{0}}, {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
   { MP_CMD_SWITCH_VSYNC, "switch_vsync", 0, { {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
diff -r 37c3bd5363cf input/input.h
--- a/input/input.h	Sun Apr 30 16:54:36 2006 +0200
+++ b/input/input.h	Sun Apr 30 16:54:37 2006 +0200
@@ -75,6 +75,8 @@
 #define MP_CMD_DVD_PLAY 81
 #define MP_CMD_CDDA_PLAY 82
 
+#define MP_CMD_RECORD 95
+
 #define MP_CMD_GUI_EVENTS       5000
 #define MP_CMD_GUI_LOADFILE     5001
 #define MP_CMD_GUI_LOADSUBTITLE 5002
diff -r 37c3bd5363cf mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:36 2006 +0200
+++ b/mplayer.c	Sun Apr 30 16:54:37 2006 +0200
@@ -91,6 +91,8 @@ int player_idle_mode=0;
 int player_idle_mode=0;
 extern int verbose;
 int quiet=0;
+static int record=0;
+static int old_index_mode=0;
 
 #ifdef WIN32
 char * proc_priority=NULL;
@@ -4732,6 +4734,27 @@ if (stream->type==STREAMTYPE_DVDNAV && d
 	video_out->control(VOCTRL_SCREENSHOT, NULL);
       }
       break;
+    case MP_CMD_RECORD:
+    {
+      extern int index_mode;
+      char cmd[1024];
+      
+      if (record == 0)
+      {
+        record = 1;
+        old_index_mode = index_mode;
+        index_mode = 0; /* do _NOT_ try to rebuild the index */
+        
+        sprintf (cmd, "/usr/bin/mp_recorder \"%s\"&", filename);
+      }
+      else
+      {
+        record = 0;
+        sprintf (cmd, "/usr/bin/mp_recorder &");
+        index_mode = old_index_mode; /* restore previous index mode */
+      }
+      system (cmd);
+    } break;
     case MP_CMD_VF_CHANGE_RECTANGLE:
 	set_rectangle(sh_video, cmd->args[0].v.i, cmd->args[1].v.i);
 	break;
