diff -Naur MPlayer-20050915.orig/configure MPlayer-20050915/configure
--- MPlayer-20050915.orig/configure	2005-09-15 23:24:55.000000000 +0200
+++ MPlayer-20050915/configure	2005-09-15 23:30:24.000000000 +0200
@@ -5141,7 +5141,7 @@
 }
 EOF
 	_fribidi=no
-	cc_check `$_fribidiconfig --cflags` `$_fribidiconfig --libs` && $TMPO >> "$TMPLOG" && _fribidi=yes
+	cc_check `$_fribidiconfig --cflags` `$_fribidiconfig --libs` && _fribidi=yes
     else
 	_fribidi=no
     fi
