diff -Naur MPlayer-1.0pre6.orig/configure MPlayer-1.0pre6/configure
--- MPlayer-1.0pre6.orig/configure	2004-12-30 19:42:53.548273576 +0100
+++ MPlayer-1.0pre6/configure	2004-12-30 19:46:30.719258568 +0100
@@ -4945,7 +4945,7 @@
 }
 EOF
 	_fribidi=no
-	cc_check `$_fribidiconfig --cflags` `$_fribidiconfig --libs` && ( $TMPO >> "$TMPLOG" ) && _fribidi=yes
+	cc_check `$_fribidiconfig --cflags` `$_fribidiconfig --libs` && _fribidi=yes
     else
 	_fribidi=no
     fi
