diff -Naur MPlayer-1.0pre6.orig/configure MPlayer-1.0pre6/configure
--- MPlayer-1.0pre6.orig/configure	Fri Mar  4 11:44:31 2005
+++ MPlayer-1.0pre6/configure	Fri Mar  4 11:44:33 2005
@@ -2223,6 +2223,9 @@
   elif cc_check $_ld_lm -liconv ; then
     _iconv=yes
     _ld_iconv='-liconv'
+  elif cc_check $_ld_lm -liconv -ldl ; then
+    _iconv=yes
+    _ld_iconv='-liconv -ldl'
   fi
 fi
 if test "$_iconv" = yes ; then
