Index: mencoder.c
===================================================================
RCS file: /cvsroot/mplayer/main/mencoder.c,v
retrieving revision 1.342
diff -u -r1.342 mencoder.c
--- MPlayer-20051213.orig/mencoder.c	24 Feb 2006 15:43:20 -0000	1.342
+++ MPlayer-20051213/mencoder.c	13 Mar 2006 16:17:04 -0000
@@ -300,7 +300,7 @@
 void parse_cfgfiles( m_config_t* conf )
 {
   char *conffile;
-  if ((conffile = get_path("mencoder")) == NULL) {
+  if ((conffile = get_path("mencoder.conf")) == NULL) {
     mp_msg(MSGT_CPLAYER,MSGL_ERR,MSGTR_GetpathProblem);
   } else {
     if (m_config_parse_config_file(conf, conffile) < 0)

