diff -r 6cd0bd7e2d49 libmpdemux/tv.c
--- a/stream/tv.c	Sun Apr 30 16:54:43 2006 +0200
+++ b/stream/tv.c	Sun Apr 30 16:54:43 2006 +0200
@@ -466,7 +466,11 @@ static int open_tv(tvi_handle_t *tvh)
 	int channel = 0;
 	if (tvh->tv_param->channel)
 	 {
-	   if (isdigit(*tvh->tv_param->channel))
+	   if (*tvh->tv_param->channel == '-' && isdigit(tvh->tv_param->channel[1])) {
+		tvh->tv_param->input = atoi(tvh->tv_param->channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tvh->tv_param->input);
+	   }
+	   else if (isdigit(*tvh->tv_param->channel))
 		/* if tvh->tv_param->channel begins with a digit interpret it as a number */
 		channel = atoi(tvh->tv_param->channel);
 	   else
@@ -514,6 +514,11 @@ static int open_tv(tvi_handle_t *tvh)
 	    if (tvh->tv_param->channel) {
 	struct CHANLIST cl;
 
+	   if (*tvh->tv_param->channel == '-' && isdigit(tvh->tv_param->channel[1])) {
+		tvh->tv_param->input = atoi(tvh->tv_param->channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tvh->tv_param->input);
+	   }
+	   else {
 	mp_msg(MSGT_TV, MSGL_V, MSGTR_TV_RequestedChannel, tvh->tv_param->channel);
 	for (i = 0; i < chanlists[tvh->chanlist].count; i++)
 	{
@@ -529,6 +529,7 @@ static int open_tv(tvi_handle_t *tvh)
 		tv_set_freq(tvh, (unsigned long)(((float)cl.freq/1000)*16));
 		break;
 	    }
+	}
 	}
     }
     }
