diff -r 5f88e25ccf52 mplayer.c
--- a/mplayer.c	Sun Apr 30 16:54:44 2006 +0200
+++ b/mplayer.c	Sun Apr 30 16:54:44 2006 +0200
@@ -2004,6 +2004,7 @@ if(!(video_out=init_best_video_out(video
       mp_msg(MSGT_CPLAYER,MSGL_FATAL,MSGTR_ErrorInitializingVODevice);
       goto err_out;
     }
+    system("[ -x /usr/bin/tvkeepon ] && /usr/bin/tvkeepon");
     sh_video->video_out=mpctx->video_out;
     inited_flags|=INITED_VO;
   }
