* already applied in trunk.

diff -Naur MPlayer-1.0pre8.orig/libmpdemux/tv.c MPlayer-1.0pre8/libmpdemux/tv.c
--- MPlayer-1.0pre8.orig/libmpdemux/tv.c	2006-06-20 21:56:14.000000000 +0200
+++ MPlayer-1.0pre8/libmpdemux/tv.c	2006-06-20 21:58:07.000000000 +0200
@@ -331,7 +331,8 @@
 
 		if (!sep) continue; // Wrong syntax, but mplayer should not crash
 
-		strcpy(tv_channel_current->name, sep + 1);
+		strlcpy(tv_channel_current->name, sep + 1,
+		        sizeof(tv_channel_current->name));
 		sep[0] = '\0';
 		strncpy(tv_channel_current->number, tmp, 5);
 
