* already applied upstream

diff -Naur MPlayer-1.0pre8.orig/libmpdemux/dvbin.c MPlayer-1.0pre8/libmpdemux/dvbin.c
--- MPlayer-1.0pre8.orig/libmpdemux/dvbin.c	2006-07-15 14:41:16.000000000 +0200
+++ MPlayer-1.0pre8/libmpdemux/dvbin.c	2006-07-15 14:46:58.000000000 +0200
@@ -69,7 +69,7 @@
 }
 stream_defaults =
 {
-	"", 1, "", 0, 0, 3, NULL
+	"", 1, "", 0, 0, 30, NULL
 };
 
 #define ST_OFF(f) M_ST_OFF(struct stream_priv_s, f)
@@ -772,6 +772,8 @@
 	stream->fill_buffer = dvb_streaming_read;
 	stream->close = dvbin_close;
 	m_struct_free(&stream_opts, opts);
+	
+	*file_format = DEMUXER_TYPE_MPEG_TS;
 
     return STREAM_OK;
 }
