diff -Naur MPlayer-1.0pre8.orig/libmpdemux/dvbin.c MPlayer-1.0pre8/libmpdemux/dvbin.c
--- MPlayer-1.0pre8.orig/libmpdemux/dvbin.c	2006-12-03 17:54:10.000000000 +0100
+++ MPlayer-1.0pre8/libmpdemux/dvbin.c	2006-12-03 17:54:46.000000000 +0100
@@ -168,7 +168,7 @@
 		if((line[0] == '#') || (strlen(line) == 0))
 			continue;
 
-		colon = index(line, ':');
+		colon = strchr(line, ':');
 		if(colon)
 		{
 			k = colon - line;
