diff -Nur MPlayer-1.0pre7.orig/input/input.c MPlayer-1.0pre7/input/input.c
--- MPlayer-1.0pre7.orig/input/input.c	Fri Apr 15 17:52:33 2005
+++ MPlayer-1.0pre7/input/input.c	Sat Nov  5 13:20:25 2005
@@ -107,8 +107,8 @@
   { MP_CMD_SCREENSHOT, "screenshot", 0, { {-1,{0}} } },
   { MP_CMD_PANSCAN, "panscan",1,  { {MP_CMD_ARG_FLOAT,{0}}, {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
   { MP_CMD_SWITCH_VSYNC, "switch_vsync", 0, { {MP_CMD_ARG_INT,{0}}, {-1,{0}} } },
-  { MP_CMD_LOADFILE, "loadfile", 1, { {MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
-  { MP_CMD_LOADLIST, "loadlist", 1, { {MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
+  { MP_CMD_LOADFILE, "loadfile", 1, { {MP_CMD_ARG_STRING, {0}}, {MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
+  { MP_CMD_LOADLIST, "loadlist", 1, { {MP_CMD_ARG_STRING, {0}}, {MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
   { MP_CMD_RUN, "run", 1, { {MP_CMD_ARG_STRING,{0}}, {-1,{0}} } },
   { MP_CMD_VF_CHANGE_RECTANGLE, "change_rectangle", 2, { {MP_CMD_ARG_INT,{0}}, {MP_CMD_ARG_INT,{0}}, {-1,{0}}}},
 
diff -Nur MPlayer-1.0pre7.orig/mplayer.c MPlayer-1.0pre7/mplayer.c
--- MPlayer-1.0pre7.orig/mplayer.c	Fri Apr 15 15:14:24 2005
+++ MPlayer-1.0pre7/mplayer.c	Sat Nov  5 13:25:43 2005
@@ -2950,6 +2950,8 @@
       play_tree_t* e = play_tree_new();
       play_tree_add_file(e,cmd->args[0].v.s);
 
+      if (cmd->nargs==2 && !strcmp(cmd->args[1].v.s, "hide_menu"))
+        mp_input_queue_cmd(mp_input_parse_cmd("menu hide"));
       // Go back to the start point
       while(play_tree_iter_up_step(playtree_iter,0,1) != PLAY_TREE_ITER_END)
 	/* NOP */;
@@ -2964,6 +2966,8 @@
       if(!e)
 	mp_msg(MSGT_CPLAYER,MSGL_ERR,MSGTR_PlaylistLoadUnable,cmd->args[0].v.s);
       else {
+	if (cmd->nargs==2 && !strcmp(cmd->args[1].v.s, "hide_menu"))
+	  mp_input_queue_cmd(mp_input_parse_cmd("menu hide"));
 	// Go back to the start point
 	while(play_tree_iter_up_step(playtree_iter,0,1) != PLAY_TREE_ITER_END)
 	  /* NOP */;
