--- MPlayer-r25315/vidix/radeon_vid.c.orig	2007-12-09 18:16:02.000000000 +0100
+++ MPlayer-r25315/vidix/radeon_vid.c	2007-12-09 18:16:38.000000000 +0100
@@ -3400,6 +3400,7 @@
 	besr.graphics_key_msk=0;
 	besr.graphics_key_clr=0;
 	besr.ckey_cntl = VIDEO_KEY_FN_TRUE|GRAPHIC_KEY_FN_TRUE|CMP_MIX_AND;
+	besr.merge_cntl |= 0x00000100;  /* DISP_RGB_OFFSET_EN */
     }
     radeon_fifo_wait(3);
     OUTREG(OV0_GRAPHICS_KEY_MSK, besr.graphics_key_msk);
