Fix

diff -Nur MPlayer-20060519.orig/libmpdemux/demux_rtp.cpp MPlayer-20060519/libmpdemux/demux_rtp.cpp
--- MPlayer-20060519.orig/libmpdemux/demux_rtp.cpp	2006-06-02 20:12:02.000000000 +0300
+++ MPlayer-20060519/libmpdemux/demux_rtp.cpp	2006-06-02 20:12:48.000000000 +0300
@@ -241,7 +241,7 @@
   if (demux_is_multiplexed_rtp_stream(demuxer)) {
     stream_t* s = new_ds_stream(demuxer->video);
     demuxer_t* od = demux_open(s, DEMUXER_TYPE_UNKNOWN,
-			       audio_id, video_id, dvdsub_id, NULL, 0, 0, 0);
+			       audio_id, video_id, dvdsub_id, NULL, 0);
     demuxer = new_demuxers_demuxer(od, od, od);
   }
 
