* backported from upstream (r21380 && r21459)

diff -Nur MPlayer-1.0rc1.orig/libvo/sub.c MPlayer-1.0rc1/libvo/sub.c
--- MPlayer-1.0rc1.orig/libvo/sub.c	2006-12-04 00:35:42.000000000 +0100
+++ MPlayer-1.0rc1/libvo/sub.c	2006-12-04 00:38:13.000000000 +0100
@@ -238,7 +238,9 @@
 //printf("dvdnav box image: %x alpha: %x \n",dvdnav_image,dvdnav_alpha);
     memset(obj->bitmap_buffer,dvdnav_image,len);
     memset(obj->alpha_buffer,dvdnav_alpha,len);
-    obj->flags|=OSDFLAG_BBOX|OSDFLAG_VISIBLE|OSDFLAG_CHANGED;
+    obj->flags|=OSDFLAG_BBOX|OSDFLAG_CHANGED;
+    if (obj->bbox.y2 > obj->bbox.y1 && obj->bbox.x2 > obj->bbox.x1)
+      obj->flags |= OSDFLAG_VISIBLE;
     }
     else
     obj->flags&=~OSDFLAG_VISIBLE;
@@ -1024,7 +1026,9 @@
     while(obj){
 	if(obj->flags&OSDFLAG_VISIBLE){
 	    if(	(obj->bbox.x1<=x2 && obj->bbox.x2>=x1) &&
-		(obj->bbox.y1<=y2 && obj->bbox.y2>=y1) ) return 1;
+		(obj->bbox.y1<=y2 && obj->bbox.y2>=y1) &&
+		obj->bbox.y2 > obj->bbox.y1 && obj->bbox.x2 > obj->bbox.x1
+		) return 1;
 	}
 	obj=obj->next;
     }
