--- old/mplayer.c	2006-05-28 12:34:59.000000000 +0800
+++ new/mplayer.c	2006-05-28 12:48:05.000000000 +0800
@@ -520,7 +525,7 @@
   if(mask&INITED_AO){
     inited_flags&=~INITED_AO;
     current_module="uninit_ao";
-    if (user_muted | edl_muted)
+    if (user_muted | edl_muted | sw_muted)
       mixer_mute(&mixer);
     audio_out->uninit(eof?0:1); audio_out=NULL;
   }
@@ -5815,7 +5820,7 @@
 #ifdef USE_EDL
 if (sh_audio && !user_muted && !edl_muted && !sw_muted && mixer.muted) mixer_mute(&mixer);
 #else
-if (sh_audio && sw_muted != mixer.muted) mixer_mute(&mixer);
+if (sh_audio && !user_muted && sw_muted != mixer.muted) mixer_mute(&mixer);
 #endif
 
 #ifdef HAVE_NEW_GUI
