diff -Naur MPlayer-20060124.orig/libmpdemux/aviheader.c MPlayer-20060124/libmpdemux/aviheader.c
--- MPlayer-20060124.orig/libmpdemux/aviheader.c	2006-01-12 21:04:34.000000000 +0100
+++ MPlayer-20060124/libmpdemux/aviheader.c	2006-01-24 11:09:46.000000000 +0100
@@ -432,7 +432,7 @@
     priv->isodml = 1;
 }
 
-if (priv->isodml && (index_mode==-1 || index_mode==0)) {
+if (priv->isodml && (index_mode==-1 || index_mode==0 || index_mode==1)) {
     int i, j, k;
     int safety=1000;
 
