diff -Naur MPlayer-r26334.orig/mplayer.c MPlayer-r26334/mplayer.c
--- MPlayer-r26334.orig/mplayer.c	2008-05-02 18:56:33.000000000 +0200
+++ MPlayer-r26334/mplayer.c	2008-05-02 18:56:45.000000000 +0200
@@ -3767,6 +3767,7 @@
 mp_input_set_section(NULL);
 //TODO: add desired (stream-based) sections here
 if (mpctx->stream->type==STREAMTYPE_TV) mp_input_set_section("tv");
+if (mpctx->stream->type==STREAMTYPE_DVDNAV) mp_input_set_section("dvdnav");
 
 //==================== START PLAYING =======================
 
