diff -Naur MPlayer-1.0pre6.orig/libvo/vo_vesa.c MPlayer-1.0pre6/libvo/vo_vesa.c
--- MPlayer-1.0pre6.orig/libvo/vo_vesa.c	2004-12-30 19:38:06.786867936 +0100
+++ MPlayer-1.0pre6/libvo/vo_vesa.c	2004-12-30 19:50:15.167137304 +0100
@@ -468,6 +468,8 @@
 #ifdef CONFIG_VIDIX
     if(vidix_name)return(vidix_query_fourcc(format));
 #endif
+    if (format == IMGFMT_MPEGPES)
+      return 0;
     return 1 | VFCAP_OSD | VFCAP_SWSCALE | VFCAP_ACCEPT_STRIDE; /* due new SwScale code */
 }
 
