--- MPlayer-20051213/libmenu/vf_menu.c	2005-12-23 23:18:18.000000000 +0800
+++ MPlayer-20051213.new/libmenu/vf_menu.c	2005-12-24 00:02:35.000000000 +0800
@@ -102,7 +102,7 @@
   case MP_CMD_MENU : {  // Convert txt cmd from the users into libmenu stuff
     char* arg = cmd->args[0].v.s;
     
-    if(!priv->current->show)
+    if(!priv->current->show && !(strcmp(arg,"hide") == 0) )
       priv->current->show = 1;
     else if(strcmp(arg,"up") == 0)
       menu_read_cmd(priv->current,MENU_CMD_UP);
@@ -114,6 +114,6 @@
       menu_read_cmd(priv->current,MENU_CMD_OK);
     else if(strcmp(arg,"cancel") == 0)
       menu_read_cmd(priv->current,MENU_CMD_CANCEL);
-    else if(strcmp(arg,"hide") == 0)
+    else if(strcmp(arg,"hide") == 0 || strcmp(arg,"toggle") == 0)
       priv->current->show = 0;
     else
       printf("Unknown menu command: '%s'\n",arg);
