* commited on mplayerhq.hu on 20050410

diff -Naur MPlayer-1.0pre6.orig/vidix/drivers/radeon_vid.c MPlayer-1.0pre6/vidix/drivers/radeon_vid.c
--- MPlayer-1.0pre6.orig/vidix/drivers/radeon_vid.c	2004-12-29 16:11:12.217446952 +0100
+++ MPlayer-1.0pre6/vidix/drivers/radeon_vid.c	2004-12-29 16:41:13.033681272 +0100
@@ -1153,6 +1153,7 @@
   /* Rage Mobility (rage128) also has memsize bug */
   if (radeon_ram_size == 0 &&
       (def_cap.device_id == DEVICE_ATI_RAGE_MOBILITY_M3 ||
+       def_cap.device_id == DEVICE_ATI_RAGE_128_RL_VR ||
        def_cap.device_id == DEVICE_ATI_RAGE_MOBILITY_M32))
   {
       printf(RADEON_MSG" Workarounding buggy Rage Mobility M3 (0 vs. 8MB ram)\n");
