--- orig/libmenu/menu_cmdlist.c	2006-04-03 16:16:31.000000000 +0800
+++ orig/libmenu/menu_cmdlist.c	2006-04-03 16:18:26.000000000 +0800
@@ -35,15 +35,20 @@
 
 struct menu_priv_s {
   menu_list_priv_t p;
+  int auto_close;
 };
 
+#define ST_OFF(m) M_ST_OFF(struct menu_priv_s, m)
+
 static struct menu_priv_s cfg_dflt = {
-  MENU_LIST_PRIV_DFLT
+  MENU_LIST_PRIV_DFLT,
+  0,
 };
 
 static m_option_t cfg_fields[] = {
   MENU_LIST_PRIV_FIELDS,
   { "title",M_ST_OFF(struct menu_priv_s,p.title), CONF_TYPE_STRING, 0, 0, 0, NULL },
+  { "auto-close", ST_OFF(auto_close), CONF_TYPE_FLAG, 0, 0, 1, NULL },
   { NULL, NULL, NULL, 0,0,0,NULL }
 };
 
@@ -55,7 +60,11 @@
     if(mpriv->p.current->ok) {
       mp_cmd_t* c = mp_input_parse_cmd(mpriv->p.current->ok);
       if(c)
+        {
+          if (mpriv->auto_close)
+              mp_input_queue_cmd (mp_input_parse_cmd ("menu hide"));
 	mp_input_queue_cmd(c);
+        }
     }
    } break;
   case MENU_CMD_CANCEL:
