diff -Naur MPlayer-1.0pre6.orig/libdha/oth/pci.db MPlayer-1.0pre6/libdha/oth/pci.db
--- MPlayer-1.0pre6.orig/libdha/oth/pci.db	2005-03-10 15:36:44.000000000 +0100
+++ MPlayer-1.0pre6/libdha/oth/pci.db	2005-03-25 17:37:31.000000000 +0100
@@ -438,6 +438,7 @@
 d	10025961	RV280 [Radeon 9200]	0	
 s	1002596117af200c	Excalibur Radeon 9200	0	
 s	1002596118bc0051	GeXcube GC-R9200-C3	0	     
+d	10025962	RV280 [Radeon 9200]	0	
 d	10025964	RV280 [Radeon 9200 SE]	0	
 s	10025964148c2073	CN-AG92E	0	
 d	10025c61	RV250 5c61 [Radeon Mobility 9200 M9+]	0	
diff -Naur MPlayer-1.0pre6.orig/vidix/drivers/radeon_vid.c MPlayer-1.0pre6/vidix/drivers/radeon_vid.c
--- MPlayer-1.0pre6.orig/vidix/drivers/radeon_vid.c	2005-03-10 15:36:45.000000000 +0100
+++ MPlayer-1.0pre6/vidix/drivers/radeon_vid.c	2005-03-25 17:47:27.000000000 +0100
@@ -895,6 +895,7 @@
  DEVICE_ATI_RV280_RADEON_92003,
  DEVICE_ATI_RV280_RADEON_92004,
  DEVICE_ATI_RV280_RADEON_92005,
+ DEVICE_ATI_RV280_RADEON_92006,
  DEVICE_ATI_RADEON_R300_ND,
  DEVICE_ATI_RADEON_R300_NE,
  DEVICE_ATI_RADEON_R300_NF,
@@ -1074,6 +1075,7 @@
             case DEVICE_ATI_RV280_RADEON_92003:
             case DEVICE_ATI_RV280_RADEON_92004:
             case DEVICE_ATI_RV280_RADEON_92005:
+            case DEVICE_ATI_RV280_RADEON_92006:
               RadeonFamily = 280;
               break;
 
