diff -Naur MPlayer-1.0pre7/libmpdemux/tv.c MPlayerMod/libmpdemux/tv.c
--- MPlayer-1.0pre7/libmpdemux/tv.c	2005-01-29 20:24:43.000000000 +0800
+++ MPlayerMod/libmpdemux/tv.c	2005-09-06 23:05:33.000000000 +0800
@@ -736,22 +736,25 @@
 int tv_step_channel(tvi_handle_t *tvh, int direction) {
 	if (tv_channel_list) {
 		if (direction == TV_CHANNEL_HIGHER) {
-			if (tv_channel_current->next) {
-				tv_channel_last = tv_channel_current;
+			tv_channel_last = tv_channel_current;
+			if (tv_channel_current->next)
 				tv_channel_current = tv_channel_current->next;
+			else
+				tv_channel_current = tv_channel_list;
 				tv_set_freq(tvh, (unsigned long)(((float)tv_channel_current->freq/1000)*16));
 				mp_msg(MSGT_TV, MSGL_INFO, "Selected channel: %s - %s (freq: %.3f)\n",
 			tv_channel_current->number, tv_channel_current->name, (float)tv_channel_current->freq/1000);
-			}
 		}
 		if (direction == TV_CHANNEL_LOWER) {
-			if (tv_channel_current->prev) {
-				tv_channel_last = tv_channel_current;
+			tv_channel_last = tv_channel_current;
+			if (tv_channel_current->prev)
 				tv_channel_current = tv_channel_current->prev;
+			else
+				while (tv_channel_current->next)
+					tv_channel_current = tv_channel_current->next;
 				tv_set_freq(tvh, (unsigned long)(((float)tv_channel_current->freq/1000)*16));
 				mp_msg(MSGT_TV, MSGL_INFO, "Selected channel: %s - %s (freq: %.3f)\n",
 			tv_channel_current->number, tv_channel_current->name, (float)tv_channel_current->freq/1000);
-			}
 		}
 	} else tv_step_channel_real(tvh, direction);
 	return(1);
