diff -Naur MPlayer-1.0pre7.orig/libmenu/menu_dvbin.c MPlayer-1.0pre7/libmenu/menu_dvbin.c
--- MPlayer-1.0pre7.orig/libmenu/menu_dvbin.c	2004-04-26 23:20:37.000000000 +0200
+++ MPlayer-1.0pre7/libmenu/menu_dvbin.c	2005-09-04 16:31:10.000000000 +0200
@@ -39,6 +39,7 @@
   char* file;
   int card;
   int level;
+  int auto_close;
   dvb_config_t *config;
 };
 
@@ -49,6 +50,7 @@
 static m_option_t cfg_fields[] = {
   MENU_LIST_PRIV_FIELDS,
   { "title", ST_OFF(title), CONF_TYPE_STRING, 0, 0, 0, NULL },
+  { "auto-close", ST_OFF(auto_close), CONF_TYPE_FLAG, 0, 0, 1, NULL },
   { NULL, NULL, NULL, 0,0,0,NULL },
 };
 
@@ -59,6 +61,7 @@
   "channels.conf",
   0,
   0,
+  1,
   NULL,
 };
 
@@ -210,7 +213,11 @@
 		
 		c = mp_input_parse_cmd(cmd_name);
     	if(c)
-		  mp_input_queue_cmd(c);
+          {
+            if (mpriv->auto_close)
+              mp_input_queue_cmd (mp_input_parse_cmd ("menu hide"));
+            mp_input_queue_cmd(c);
+          }
   	}
   	}
   	break;
