diff -Naur MPlayer-1.0rc1.orig/configure MPlayer-1.0rc1/configure
--- MPlayer-1.0rc1.orig/configure	2007-04-13 19:18:26.000000000 +0200
+++ MPlayer-1.0rc1/configure	2007-04-13 19:18:36.000000000 +0200
@@ -1281,7 +1281,10 @@
     mingw32msvc) system_name=MINGW32 ;;
   esac
   # We need to convert underscores so that values like k6-2 and pentium-mmx can be passed
-  host_arch=`echo $_target | cut -d '-' -f 1 | tr '_' '-'`
+  host_arch=`echo $_target | cut -d '-' -f 1`
+  if test `echo $host_arch` != "x86_64" ; then
+    host_arch=`echo $host_arch | tr '_' '-'`
+  fi
 fi
 
 echo "Detected operating system: $system_name"
