* already applied upstream

Modified: trunk/configure
==============================================================================
--- trunk/configure	(original)
+++ trunk/configure	Tue Jun  5 20:35:34 2007
@@ -7233,7 +7233,7 @@ if test "$_vidix" = yes ; then
   _def_vidix='#define CONFIG_VIDIX 1'
   _vosrc="$_vosrc vo_cvidix.c"
   _vomodules="cvidix $_vomodules"
-  test $_vidix_drivers || _vidix_drivers="cyberblade ivtv mach64 mga mga_crtc2 nvidia pm2 pm3 radeon rage128 savage sis unichrome"
+  test "$_vidix_drivers" || _vidix_drivers="cyberblade ivtv mach64 mga mga_crtc2 nvidia pm2 pm3 radeon rage128 savage sis unichrome"
   test $_ivtv = "yes" || _vidix_drivers=`echo $_vidix_drivers | sed s/ivtv//`
 
   for driver in $_vidix_drivers ; do
