diff -Naur MPlayer-r23418.orig/vidix/drivers.c MPlayer-r23418/vidix/drivers.c
--- MPlayer-r23418.orig/vidix/drivers.c	2007-07-01 18:59:30.000000000 +0200
+++ MPlayer-r23418/vidix/drivers.c	2007-07-01 19:00:24.000000000 +0200
@@ -65,9 +65,6 @@
 #ifdef CONFIG_VIDIX_DRV_IVTV
   vidix_register_driver (&ivtv_drv);
 #endif
-#ifdef CONFIG_VIDIX_DRV_MACH64
-  vidix_register_driver (&mach64_drv);
-#endif
 #ifdef CONFIG_VIDIX_DRV_MGA
   vidix_register_driver (&mga_drv);
 #endif
@@ -89,6 +86,9 @@
 #ifdef CONFIG_VIDIX_DRV_RAGE128
   vidix_register_driver (&rage128_drv);
 #endif
+#ifdef CONFIG_VIDIX_DRV_MACH64
+  vidix_register_driver (&mach64_drv);
+#endif
 #ifdef CONFIG_VIDIX_DRV_SAVAGE
   vidix_register_driver (&savage_drv);
 #endif
