diff -Naur MPlayer-20051213.orig/input/input.c MPlayer-20051213/input/input.c
--- MPlayer-20051213.orig/input/input.c	2006-01-06 00:05:50.000000000 +0100
+++ MPlayer-20051213/input/input.c	2006-01-06 00:19:04.000000000 +0100
@@ -592,6 +592,7 @@
   int pausing = 0;
   char *ptr,*e;
   mp_cmd_t *cmd, *cmd_def;
+  mp_cmd_t *cmd_ret = NULL;
 
 #ifdef MP_DEBUG
   assert(str != NULL);
@@ -602,6 +603,8 @@
     str = &str[8];
   }
 
+ do {
+
   for(ptr = str ; ptr[0] != '\0'  && ptr[0] != '\t' && ptr[0] != ' ' ; ptr++)
     /* NOTHING */;
   if(ptr[0] != '\0')
@@ -633,6 +636,11 @@
     ptr = strchr(ptr,' ');
     if(!ptr) break;
     while(ptr[0] == ' ' || ptr[0] == '\t') ptr++;
+    if (ptr[0] == ';')
+      {
+        str = ptr + 1;
+        break;
+      }
     if(ptr[0] == '\0') break;
     cmd->args[i].type = cmd_def->args[i].type;
     switch(cmd_def->args[i].type) {
@@ -711,7 +719,14 @@
   if(i < MP_CMD_MAX_ARGS)
     cmd->args[i].type = -1;
 
-  return cmd;
+  if(!cmd_ret)
+    cmd_ret = cmd;
+  else
+    mp_input_queue_cmd(cmd);
+
+ } while (ptr && ptr[0] == ';');
+
+  return cmd_ret;
 }
 
 static int
