--- MPlayer-20051213/input/input.c	2005-12-24 14:47:18.000000000 +0800
+++ MPlayer-20051213.new/input/input.c	2005-12-24 14:45:41.000000000 +0800
@@ -601,13 +600,17 @@
 mp_input_parse_cmd(char* str) {
   int i,l;
   int pausing = 0;
-  char *ptr,*e;
+  char *ptr,*e,*next_cmd=NULL;
   mp_cmd_t *cmd, *cmd_def;
 
 #ifdef MP_DEBUG
   assert(str != NULL);
 #endif
 
+  next_cmd=strchr(str, ';');
+  if (next_cmd && strlen(next_cmd) > 1)
+    mp_input_queue_cmd(mp_input_parse_cmd(next_cmd+1));
+
   if (strncmp(str, "pausing ", 8) == 0) {
     pausing = 1;
     str = &str[8];
--- MPlayer-20051213/subopt-helper.c	2005-10-17 06:31:18.000000000 +0800
+++ MPlayer-20051213.new/subopt-helper.c	2005-12-24 14:27:21.000000000 +0800
@@ -37,7 +37,7 @@
 /**
  * \brief Try to parse all options in str and fail if it was not possible.
  *
- * \param str Pointer to the zero terminated string to be parsed.
+ * \param str Pointer to the zero or ';' terminated string to be parsed.
  * \param opts Pointer to a options array. The array must be terminated
  *             with an element having set name to NULL in its opt_t structure.
  *
@@ -59,7 +59,7 @@
 
   if ( str )
   {
-    while ( str[parse_pos] && !parse_err )
+    while ( str[parse_pos] && !parse_err && strcmp(str[parse_pos],';'))
     {
       int next = 0;
 
