diff -Naur MPlayer-1.0pre6.orig/libmenu/menu_list.c MPlayer-1.0pre6/libmenu/menu_list.c
--- MPlayer-1.0pre6.orig/libmenu/menu_list.c	2004-12-30 19:38:06.757872344 +0100
+++ MPlayer-1.0pre6/libmenu/menu_list.c	2004-12-30 19:54:39.984878888 +0100
@@ -85,6 +85,8 @@
   } else
     m = mpriv->menu;
 
+  dy += 0.1 * h;
+
   if(th > 0) {
     menu_draw_text_full(mpi,mpriv->title,
 			x < 0 ? mpi->w / 2 : x,
