diff -Naur MPlayer-20060429.orig/libmenu/menu_filesel.c MPlayer-20060429/libmenu/menu_filesel.c
--- MPlayer-20060429.orig/libmenu/menu_filesel.c	2006-05-01 13:59:13.000000000 +0200
+++ MPlayer-20060429/libmenu/menu_filesel.c	2006-05-01 13:59:52.000000000 +0200
@@ -338,6 +338,11 @@
 	  p = malloc(l + strlen(mpriv->p.current->p.txt) + 1);
 	  sprintf(p,"%s%s",mpriv->dir,mpriv->p.current->p.txt);
 	}
+        if (!strcmp (p, "/")) /* chroot user to /mnt */
+        {
+          free (p);
+          break;
+        }
 	menu_list_uninit(menu,free_entry);
 	if(!open_dir(menu,p)) {
 	  mp_msg(MSGT_GLOBAL,MSGL_ERR,MSGTR_LIBMENU_CantOpenDirectory,p);
