diff -Naur MPlayer-1.0pre6.orig/libmenu/menu_filesel.c MPlayer-1.0pre6/libmenu/menu_filesel.c
--- MPlayer-1.0pre6.orig/libmenu/menu_filesel.c	2003-08-15 20:45:35.000000000 +0200
+++ MPlayer-1.0pre6/libmenu/menu_filesel.c	2005-02-24 15:47:02.000000000 +0100
@@ -6,6 +6,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <fcntl.h>
 #include <ctype.h>
 #include <unistd.h>
 #include <limits.h>
@@ -126,6 +127,7 @@
   struct dirent *dp;
   struct stat st;
   int n;
+  int path_fp;
   char* p = NULL;
   list_entry_t* e;
   DIR* dirp;
@@ -146,6 +148,13 @@
     return 0;
   }
 
+  path_fp = open ("/var/mp_current_path", O_CREAT | O_WRONLY | O_TRUNC);
+  if (path_fp)
+    {
+      write (path_fp, mpriv->dir, strlen (mpriv->dir));
+      close (path_fp);
+    }
+
   namelist = (char **) malloc(sizeof(char *));
 
   n=0;
