diff -Naur MPlayer-1.0pre6.orig/libmpdemux/tv.c MPlayer-1.0pre6/libmpdemux/tv.c
--- MPlayer-1.0pre6.orig/libmpdemux/tv.c	2004-12-30 19:38:06.774869760 +0100
+++ MPlayer-1.0pre6/libmpdemux/tv.c	2004-12-30 19:57:40.016509920 +0100
@@ -359,7 +359,11 @@
 	int channel = 0;
 	if (tv_param_channel)
 	 {
-	   if (isdigit(*tv_param_channel))
+	   if (*tv_param_channel == '-' && isdigit(tv_param_channel[1])) {
+		tv_param_input = atoi(tv_param_channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tv_param_input);
+	   }
+	   else if (isdigit(*tv_param_channel))
 		/* if tv_param_channel begins with a digit interpret it as a number */
 		channel = atoi(tv_param_channel);
 	   else
@@ -407,6 +411,11 @@
 	    if (tv_param_channel) {
 	struct CHANLIST cl;
 
+	   if (*tv_param_channel == '-' && isdigit(tv_param_channel[1])) {
+		tv_param_input = atoi(tv_param_channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tv_param_input);
+	   }
+	   else {
 	mp_msg(MSGT_TV, MSGL_V, "Requested channel: %s\n", tv_param_channel);
 	for (i = 0; i < chanlists[tvh->chanlist].count; i++)
 	{
@@ -423,6 +432,7 @@
 		break;
 	    }
 	}
+	}
     }
     }
     
