Index: MPlayer-20051213/libmpdemux/tv.c
===================================================================
--- MPlayer-20051213.orig/libmpdemux/tv.c	2006-02-23 10:49:34.000000000 +0100
+++ MPlayer-20051213/libmpdemux/tv.c	2006-02-23 10:52:28.000000000 +0100
@@ -363,7 +363,11 @@
 	int channel = 0;
 	if (tv_param_channel)
 	 {
-	   if (isdigit(*tv_param_channel))
+	   if (*tv_param_channel == '-' && isdigit(tv_param_channel[1])) {
+		tv_param_input = atoi(tv_param_channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tv_param_input);
+	   }
+	   else if (isdigit(*tv_param_channel))
 		/* if tv_param_channel begins with a digit interpret it as a number */
 		channel = atoi(tv_param_channel);
 	   else
@@ -411,6 +415,11 @@
 	    if (tv_param_channel) {
 	struct CHANLIST cl;
 
+	   if (*tv_param_channel == '-' && isdigit(tv_param_channel[1])) {
+		tv_param_input = atoi(tv_param_channel+1);
+		funcs->control(tvh->priv, TVI_CONTROL_SPC_SET_INPUT, &tv_param_input);
+	   }
+	   else {
 	mp_msg(MSGT_TV, MSGL_V, "Requested channel: %s\n", tv_param_channel);
 	for (i = 0; i < chanlists[tvh->chanlist].count; i++)
 	{
@@ -427,6 +436,7 @@
 		break;
 	    }
 	}
+	}
     }
     }
     
