* Already fixed upstream.

diff -Naur MPlayer-1.0pre7.orig/libmpdemux/tvi_v4l2.c MPlayer-1.0pre7/libmpdemux/tvi_v4l2.c
--- MPlayer-1.0pre7.orig/libmpdemux/tvi_v4l2.c	2005-07-24 22:47:25.000000000 +0200
+++ MPlayer-1.0pre7/libmpdemux/tvi_v4l2.c	2005-07-24 22:47:40.000000000 +0200
@@ -776,7 +776,7 @@
     }
     PRIV->video_fd = -1;
 
-    PRIV->video_dev = strdup(video_dev? video_dev: "/dev/video");
+    PRIV->video_dev = strdup(video_dev? video_dev: "/dev/video0");
     if (!PRIV->video_dev) {
 	free_handle(tvi_handle);
 	return NULL;
