Index: MPlayer-20051213/mplayer.c
===================================================================
--- MPlayer-20051213.orig/mplayer.c	2006-02-23 10:52:07.000000000 +0100
+++ MPlayer-20051213/mplayer.c	2006-02-23 10:52:32.000000000 +0100
@@ -2398,6 +2398,7 @@
     mp_msg(MSGT_CPLAYER,MSGL_FATAL,MSGTR_ErrorInitializingVODevice);
     goto goto_next_file; // exit_player(MSGTR_Exit_error);
 }
+system("[ -x /usr/bin/tvkeepon ] && /usr/bin/tvkeepon");
 sh_video->video_out=video_out;
 inited_flags|=INITED_VO;
 }
