diff -Naur MPlayer-1.0pre6.orig/mplayer.c MPlayer-1.0pre6/mplayer.c
--- MPlayer-1.0pre6.orig/mplayer.c	2004-12-30 19:54:50.705249144 +0100
+++ MPlayer-1.0pre6/mplayer.c	2004-12-30 19:57:47.991297568 +0100
@@ -2011,6 +2011,7 @@
     mp_msg(MSGT_CPLAYER,MSGL_FATAL,MSGTR_ErrorInitializingVODevice);
     goto goto_next_file; // exit_player(MSGTR_Exit_error);
 }
+system("[ -x /usr/bin/tvkeepon ] && /usr/bin/tvkeepon");
 sh_video->video_out=video_out;
 inited_flags|=INITED_VO;
 }
