diff -Naur MPlayer-1.0pre7.orig/libmpdemux/demux_ts.c MPlayer-1.0pre7/libmpdemux/demux_ts.c
--- MPlayer-1.0pre7.orig/libmpdemux/demux_ts.c	2005-04-09 13:30:59.000000000 +0200
+++ MPlayer-1.0pre7/libmpdemux/demux_ts.c	2005-08-20 16:26:33.000000000 +0200
@@ -893,12 +893,6 @@
 	else
 		memset(params.slang, 0, 4);
 
-	if(audio_lang != NULL)
-	{
-		strncpy(params.alang, audio_lang, 3);
-		params.alang[3] = 0;
-	}
-	else
 		memset(params.alang, 0, 4);
 
 	start_pos = ts_detect_streams(demuxer, &params);
