diff -Naur MPlayer-1.0pre7.orig/libmpdemux/demux_ts.c MPlayer-1.0pre7/libmpdemux/demux_ts.c
--- MPlayer-1.0pre7.orig/libmpdemux/demux_ts.c	2005-04-09 13:30:59.000000000 +0200
+++ MPlayer-1.0pre7/libmpdemux/demux_ts.c	2005-08-28 16:18:45.000000000 +0200
@@ -885,7 +885,7 @@
 	params.prog = ts_prog;
 	params.probe = ts_probe;
 
-	if(dvdsub_lang != NULL)
+	if(dvdsub_lang != NULL && demuxer->stream->type != STREAMTYPE_DVB)
 	{
 		strncpy(params.slang, dvdsub_lang, 3);
 		params.slang[3] = 0;
@@ -893,7 +893,7 @@
 	else
 		memset(params.slang, 0, 4);
 
-	if(audio_lang != NULL)
+	if(audio_lang != NULL && demuxer->stream->type != STREAMTYPE_DVB)
 	{
 		strncpy(params.alang, audio_lang, 3);
 		params.alang[3] = 0;
