Index: MPlayer-20051213/libmpdemux/demux_ts.c
===================================================================
--- MPlayer-20051213.orig/libmpdemux/demux_ts.c	2006-02-23 10:49:34.000000000 +0100
+++ MPlayer-20051213/libmpdemux/demux_ts.c	2006-02-23 10:52:34.000000000 +0100
@@ -896,20 +896,8 @@
 	params.prog = ts_prog;
 	params.probe = ts_probe;
 
-	if(dvdsub_lang != NULL)
-	{
-		strncpy(params.slang, dvdsub_lang, 3);
-		params.slang[3] = 0;
-	}
-	else
 		memset(params.slang, 0, 4);
 
-	if(audio_lang != NULL)
-	{
-		strncpy(params.alang, audio_lang, 3);
-		params.alang[3] = 0;
-	}
-	else
 		memset(params.alang, 0, 4);
 
 	start_pos = ts_detect_streams(demuxer, &params);
