diff -Nur MPlayer-1.0pre7.orig/input/input.c MPlayer-1.0pre7/input/input.c
--- MPlayer-1.0pre7.orig/input/input.c	Tue Jul 26 14:37:42 2005
+++ MPlayer-1.0pre7/input/input.c	Tue Jul 26 14:39:30 2005
@@ -83,6 +83,7 @@
   { MP_CMD_GET_PERCENT_POS, "get_percent_pos", 0, { {-1,{0}} } },
   { MP_CMD_GET_TIME_LENGTH, "get_time_length", 0, { {-1,{0}} } },
   { MP_CMD_SWITCH_AUDIO, "switch_audio", 0, { {-1,{0}} } },
+  { MP_CMD_SET_OPTION, "set_option", 2, { { MP_CMD_ARG_STRING, {0}}, { MP_CMD_ARG_STRING, {0}}, { MP_CMD_ARG_STRING, {0}}, {-1,{0}} } },
 #ifdef USE_TV
   { MP_CMD_TV_STEP_CHANNEL, "tv_step_channel", 1,  { { MP_CMD_ARG_INT ,{0}}, {-1,{0}} }},
   { MP_CMD_TV_STEP_NORM, "tv_step_norm",0, { {-1,{0}} }  },
diff -Nur MPlayer-1.0pre7.orig/input/input.h MPlayer-1.0pre7/input/input.h
--- MPlayer-1.0pre7.orig/input/input.h	Tue Jul 26 14:37:42 2005
+++ MPlayer-1.0pre7/input/input.h	Tue Jul 26 14:39:53 2005
@@ -70,6 +70,8 @@
 #define MP_CMD_DVD_PLAY 71
 #define MP_CMD_CDDA_PLAY 72
 
+#define MP_CMD_SET_OPTION 90
+
 #define MP_CMD_GUI_EVENTS       5000
 #define MP_CMD_GUI_LOADFILE     5001
 #define MP_CMD_GUI_LOADSUBTITLE 5002
diff -Nur MPlayer-1.0pre7.orig/mplayer.c MPlayer-1.0pre7/mplayer.c
--- MPlayer-1.0pre7.orig/mplayer.c	Tue Jul 26 14:37:42 2005
+++ MPlayer-1.0pre7/mplayer.c	Tue Jul 26 14:41:48 2005
@@ -2985,6 +2985,21 @@
     } break;
 #endif
 
+    case MP_CMD_SET_OPTION: {
+      char *option = cmd->args[0].v.s, *value = cmd->args[1].v.s;
+      m_config_set_option(mconfig, option, value);
+      if (cmd->nargs==3 && !strcmp(cmd->args[2].v.s, "save")) {
+        char *buf;
+        size_t len;
+        len = strlen(option) + strlen(value) + sizeof("mp_set_option '' ''");
+        if ((buf = (char*)malloc(len))) {
+          sprintf(buf, "mp_set_option '%s' '%s'", option, value);
+          system(buf);
+          free(buf);
+        }
+      }
+    } break;
+
     case MP_CMD_GRAB_FRAMES : {
       grab_frames=2;
     } break;
