Index: MPlayer-20051213/libmenu/menu_filesel.c
===================================================================
--- MPlayer-20051213.orig/libmenu/menu_filesel.c	2006-02-23 10:52:25.000000000 +0100
+++ MPlayer-20051213/libmenu/menu_filesel.c	2006-02-23 10:52:49.000000000 +0100
@@ -192,7 +192,7 @@
     }
 }
 
-static int open_dir(menu_t* menu,char* args) {
+static int _open_dir(menu_t* menu,char* args) {
   char **namelist, **tp;
   struct dirent *dp;
   struct stat st;
@@ -298,6 +298,9 @@
   return 1;
 }
     
+static int open_dir(menu_t* menu,char* args) {
+  return (_open_dir(menu, args) || _open_dir(menu, "/mnt/"));
+}
 
 static char *action;
 
