--- a/CMakeLists.txt	2017-06-10 09:37:59.768207518 +0200
+++ b/CMakeLists.txt	2017-06-10 09:37:59.800206924 +0200
@@ -53,6 +53,7 @@
 externalproject_add(rsxs SOURCE_DIR ${PROJECT_SOURCE_DIR}/${rsxs_dir}
                     CONFIGURE_COMMAND ${configure_start}
                                       ac_cv_type__Bool=yes
+                                      ${RSXS_CROSSCOMPILE}
                                       --prefix=<INSTALL_DIR>
                                       --without-xscreensaver
                                       --disable-cyclone
