--- fuseiso-20070708.orig/src/fuseiso.c	2007-05-07 01:01:17.000000000 +0800
+++ fuseiso-20070708/src/fuseiso.c	2009-05-23 22:32:27.000000000 +0800
@@ -309,7 +309,7 @@ int main(int argc, char *argv[])
     maintain_mtab = 1;
     iocharset = NULL;
     
-    char c;
+    int c;
     while((c = getopt(argc, argv, "+npc:h")) > 0) {
         switch(c) {
             case 'n':
--- fuseiso-20070708.orig/src/isofs.h	2006-10-17 09:50:39.000000000 +0800
+++ fuseiso-20070708/src/isofs.h	2009-05-24 15:40:24.000000000 +0800
@@ -109,7 +109,7 @@ static inline int isonum_711(unsigned ch
 // }
 static inline unsigned int isonum_721(char *p)
 {
-#if defined(WORDS_BIGENDIAN)
+#if defined(_BIG_ENDIAN)
     return *(unsigned short *)p;
 #else
     return bswap_16(*(unsigned short *)p);
@@ -122,7 +122,7 @@ static inline unsigned int isonum_721(ch
 static inline unsigned int isonum_723(char *p)
 {
     /* Ignore bigendian datum due to broken mastering programs */
-#if defined(WORDS_BIGENDIAN)
+#if defined(_BIG_ENDIAN)
     return bswap_16(*(unsigned short *)p);
 #else
     return *(unsigned short *)p;
@@ -130,7 +130,7 @@ static inline unsigned int isonum_723(ch
 }
 static inline unsigned int isonum_731(char *p)
 {
-#if defined(WORDS_BIGENDIAN)
+#if defined(_BIG_ENDIAN)
     return bswap_32(*(unsigned int *)p);
 #else
     return *(unsigned int *)p;
@@ -143,7 +143,7 @@ static inline unsigned int isonum_731(ch
 static inline unsigned int isonum_733(char *p)
 {
     /* Ignore bigendian datum due to broken mastering programs */
-#if defined(WORDS_BIGENDIAN)
+#if defined(_BIG_ENDIAN)
     return bswap_32(*(unsigned int *)p);
 #else
     return *(unsigned int *)p;
