--- a/src/Makefile	2014-02-16 10:39:11.681684917 +0100
+++ b/src/Makefile	2014-02-16 10:39:11.697684918 +0100
@@ -40,7 +40,7 @@
 ARCREATE=cr
 
 # base of installation location
-BASE=/usr/local
+BASE=$(PREFIX)/usr/local
 O= ../bin/unknown
 D= ../doc
 TRUE=/bin/true
@@ -139,10 +139,10 @@
 	@env CFLAGS=-O MAKE="$(MAKE)"  MAKEFLAGS="$(MAKEFLAGS)" CC="${CC}" OS="${OS}" ../scripts/build install-target
 
 install-target:
-	if [ ! -d $(BASE) ]; then mkdir $(BASE); fi
-	if [ ! -d $(BASE)/bin ]; then mkdir $(BASE)/bin; fi
-	if [ ! -d $(BASE)/include ]; then mkdir $(BASE)/include; fi
-	if [ ! -d $(BASE)/lib ]; then mkdir $(BASE)/lib; fi
+	if [ ! -d $(BASE) ]; then mkdir -p $(BASE); fi
+	if [ ! -d $(BASE)/bin ]; then mkdir -p $(BASE)/bin; fi
+	if [ ! -d $(BASE)/include ]; then mkdir -p $(BASE)/include; fi
+	if [ ! -d $(BASE)/lib ]; then mkdir -p $(BASE)/lib; fi
 	cp $(EXES) $(BASE)/bin
 	cp $(INCS) $(BASE)/include
 	cp $O/lmbench.a $(BASE)/lib/libmbench.a
