diff -Naur lcd4linux-0.10.1-RC1.orig/drv_generic_text.c lcd4linux-0.10.1-RC1/drv_generic_text.c
--- lcd4linux-0.10.1-RC1.orig/drv_generic_text.c	2007-03-23 11:51:24.000000000 +0100
+++ lcd4linux-0.10.1-RC1/drv_generic_text.c	2007-03-23 11:51:57.000000000 +0100
@@ -320,16 +320,16 @@
     int i;
     int flag = 0;
 
-    char *line1[] = { "* LCD4Linux " VERSION " *",
-	"LCD4Linux " VERSION,
-	"* LCD4Linux *",
-	"LCD4Linux",
-	"L4Linux",
+    char *line1[] = { "GeeXboX OMC Welcome",
+	"GeeXboX Welcome",
+	"GeeXboX OMC",
+	"GeeXboX",
+	"GB",
 	NULL
     };
 
-    char *line2[] = { "http://lcd4linux.bulix.org",
-	"lcd4linux.bulix.org",
+    char *line2[] = { "http://www.openbricks.org",
+	"www.openbricks.org",
 	NULL
     };
 
