diff -Naur libX11-1.1.4.orig/src/util/Makefile.am libX11-1.1.4/src/util/Makefile.am
--- libX11-1.1.4.orig/src/util/Makefile.am	2008-04-12 14:56:49.000000000 +0200
+++ libX11-1.1.4/src/util/Makefile.am	2008-04-12 14:57:36.000000000 +0200
@@ -5,7 +5,7 @@
 makekeys_CFLAGS=$(X11_CFLAGS)
 
 #override CC = gcc
-LINK = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(HOST_CC) $(AM_CFLAGS) $(HOST_CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 
 EXTRA_DIST = mkks.sh
 
diff -Naur libX11-1.1.4.orig/src/util/Makefile.in libX11-1.1.4/src/util/Makefile.in
--- libX11-1.1.4.orig/src/util/Makefile.in	2008-04-12 14:56:49.000000000 +0200
+++ libX11-1.1.4/src/util/Makefile.in	2008-04-12 15:03:50.000000000 +0200
@@ -51,17 +51,17 @@
 makekeys_OBJECTS = makekeys-makekeys.$(OBJEXT)
 makekeys_LDADD = $(LDADD)
 makekeys_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=link $(CCLD) $(makekeys_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) \
-	$(LDFLAGS) -o $@
+	--mode=link host-gcc $(makekeys_CFLAGS) $(HOST_CFLAGS) $(AM_LDFLAGS) \
+	$(HOST_LDFLAGS) -o $@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/src -I$(top_builddir)/include/X11
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
-COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+COMPILE = $(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(HOST_CPPFLAGS) $(AM_CFLAGS) $(HOST_CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
-	--mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+	--mode=compile host-gcc $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
+	$(AM_CPPFLAGS) $(HOST_CPPFLAGS) $(AM_CFLAGS) $(HOST_CFLAGS)
+CCLD = $(HOST_CC)
 SOURCES = makekeys.c
 DIST_SOURCES = makekeys.c
 ETAGS = etags
@@ -80,9 +80,9 @@
 AWK = @AWK@
 BIGFONT_CFLAGS = @BIGFONT_CFLAGS@
 BIGFONT_LIBS = @BIGFONT_LIBS@
-CC = @CC@
+CC = host-gcc
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = $(HOST_CFLAGS)
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
