diff -Naur libX11-X11R7.1-1.0.1.orig/src/util/Makefile.am libX11-X11R7.1-1.0.1/src/util/Makefile.am
--- libX11-X11R7.1-1.0.1.orig/src/util/Makefile.am	2007-01-27 22:00:41.000000000 +0100
+++ libX11-X11R7.1-1.0.1/src/util/Makefile.am	2007-01-27 22:02:14.000000000 +0100
@@ -5,6 +5,6 @@
 makekeys_CFLAGS=$(X11_CFLAGS)
 
 #override CC = gcc
-LINK = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(HOST_CC) $(AM_CFLAGS) $(HOST_CFLAGS) $(AM_LDFLAGS) $(HOST_LDFLAGS) -o $@
 
 EXTRA_DIST = mkks.sh
diff -Naur libX11-X11R7.1-1.0.1.orig/src/util/Makefile.in libX11-X11R7.1-1.0.1/src/util/Makefile.in
--- libX11-X11R7.1-1.0.1.orig/src/util/Makefile.in	2007-01-27 22:00:41.000000000 +0100
+++ libX11-X11R7.1-1.0.1/src/util/Makefile.in	2007-01-27 22:02:17.000000000 +0100
@@ -57,12 +57,13 @@
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)/src -I$(top_builddir)/include/X11
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
-COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
-LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
-	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
-CCLD = $(CC)
+COMPILE = $(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) 
+$(AM_CPPFLAGS) \
+	$(CPPFLAGS) $(AM_CFLAGS) $(HOST_CFLAGS)
+LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(HOST_CC) $(DEFS) \
+	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(HOST_CPPFLAGS) \
+	$(AM_CFLAGS) $(HOST_CFLAGS)
+CCLD = $(HOST_CC)
 SOURCES = makekeys.c
 DIST_SOURCES = makekeys.c
 ETAGS = etags
@@ -83,7 +84,7 @@
 AWK = @AWK@
 BIGFONT_CFLAGS = @BIGFONT_CFLAGS@
 BIGFONT_LIBS = @BIGFONT_LIBS@
-CC = @CC@
+CC = host-gcc
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
 CPP = @CPP@
