diff -Naur a/configure b/configure
--- a/configure	2009-10-02 15:46:26.000000000 +0200
+++ b/configure	2009-10-02 16:00:33.000000000 +0200
@@ -13552,73 +13552,7 @@
 	X11_EXTRA_DEPS="xau"
 
 pkg_failed=no
-{ $as_echo "$as_me:$LINENO: checking for XDMCP" >&5
-$as_echo_n "checking for XDMCP... " >&6; }
 
-if test -n "$XDMCP_CFLAGS"; then
-    pkg_cv_XDMCP_CFLAGS="$XDMCP_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"xdmcp\"") >&5
-  ($PKG_CONFIG --exists --print-errors "xdmcp") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_XDMCP_CFLAGS=`$PKG_CONFIG --cflags "xdmcp" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$XDMCP_LIBS"; then
-    pkg_cv_XDMCP_LIBS="$XDMCP_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { ($as_echo "$as_me:$LINENO: \$PKG_CONFIG --exists --print-errors \"xdmcp\"") >&5
-  ($PKG_CONFIG --exists --print-errors "xdmcp") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-  pkg_cv_XDMCP_LIBS=`$PKG_CONFIG --libs "xdmcp" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        XDMCP_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "xdmcp" 2>&1`
-        else
-	        XDMCP_PKG_ERRORS=`$PKG_CONFIG --print-errors "xdmcp" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$XDMCP_PKG_ERRORS" >&5
-
-	{ $as_echo "$as_me:$LINENO: result: no" >&5
-$as_echo "no" >&6; }
-                { $as_echo "$as_me:$LINENO: result: no" >&5
-$as_echo "no" >&6; }
-elif test $pkg_failed = untried; then
-	{ $as_echo "$as_me:$LINENO: result: no" >&5
-$as_echo "no" >&6; }
-else
-	XDMCP_CFLAGS=$pkg_cv_XDMCP_CFLAGS
-	XDMCP_LIBS=$pkg_cv_XDMCP_LIBS
-        { $as_echo "$as_me:$LINENO: result: yes" >&5
-$as_echo "yes" >&6; }
-	{ $as_echo "$as_me:$LINENO: checking for XdmcpWrap in -lXdmcp" >&5
-$as_echo_n "checking for XdmcpWrap in -lXdmcp... " >&6; }
 if test "${ac_cv_lib_Xdmcp_XdmcpWrap+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
@@ -13764,8 +13698,6 @@
 
 fi
 
-fi
-
 cat >>confdefs.h <<\_ACEOF
 #define USE_XCB 0
 _ACEOF
