diff -Naur libX11-1.1.99.2/configure libX11-1.1.99.2b/configure
--- libX11-1.1.99.2/configure	2008-12-22 16:48:17.000000000 +0100
+++ libX11-1.1.99.2b/configure	2008-12-22 16:50:59.000000000 +0100
@@ -20123,7 +20123,7 @@
     case "$arg" in
         --mode=compile) modeok=true ;;
         --tag=CC|--tag=CXX) tagok=true ;;
-        *) args+=("$arg")
+        *) args[${#args[@]}]="$arg"
     esac
 done
 if $modeok && $tagok ; then
