diff -Naur zsnes_1_51.orig/src/Makefile.in zsnes_1_51/src/Makefile.in
--- zsnes_1_51.orig/src/Makefile.in	2007-10-06 20:21:50.000000000 +0200
+++ zsnes_1_51/src/Makefile.in	2007-10-06 20:23:59.000000000 +0200
@@ -104,7 +104,7 @@
 	rm -f version.o
 
 $(PSR): parsegen.cpp
-	@CXX@ @CXXFLAGS@ -o $@ $< -lz
+	$(HOST_CXX) @CXXFLAGS@ -o $@ $< -lz
 
 TOOLSEXE=$(TOOL_D)/archopt $(TOOL_D)/cutrtype $(TOOL_D)/extraext\
 	$(TOOL_D)/macroll $(TOOL_D)/minwhite $(TOOL_D)/nreplace\
@@ -115,7 +115,7 @@
 $(TOOL_D)/cutrtype: $(TOOL_D)/cutrtype.cpp $(TOOL_O)
 	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/depbuild: $(TOOL_D)/depbuild.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/extraext: $(TOOL_D)/extraext.cpp $(TOOL_O)
 	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/macroll: $(TOOL_D)/macroll.cpp $(TOOL_O)
