--- zsnes-1.5-RC1-svn/src/Makefile.in.orig	2006-12-07 10:16:09.000000000 +0100
+++ zsnes-1.5-RC1-svn/src/Makefile.in	2006-12-08 10:12:49.000000000 +0100
@@ -103,7 +103,9 @@
 	@ZC@ -o @ZSNESEXE@ $(Z_OBJS) @ZCFLAGS@ @LDFLAGS@
 	rm -f version.o
 
-$(PSR): parsegen.cpp
+
+$(PSR):
+	$(HOST_CXX) @CXXFLAGS@ -o $(PSR) $(PSR).cpp
 
 TOOLSEXE=$(TOOL_D)/archopt $(TOOL_D)/cutrtype $(TOOL_D)/extraext\
 	$(TOOL_D)/macroll $(TOOL_D)/minwhite $(TOOL_D)/nreplace\
@@ -114,7 +116,7 @@
 $(TOOL_D)/cutrtype: $(TOOL_D)/cutrtype.cpp $(TOOL_O)
 	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/depbuild: $(TOOL_D)/depbuild.cpp $(TOOL_O)
-	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
+	$(HOST_CXX) @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/extraext: $(TOOL_D)/extraext.cpp $(TOOL_O)
 	@CXX@ @CXXFLAGS@ -o $@ $@.cpp $(TOOL_O)
 $(TOOL_D)/macroll: $(TOOL_D)/macroll.cpp $(TOOL_O)
