From b1023e676cf47bcaca2b624a0fefbbfaf4044f1d Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 16 Mar 2017 09:30:40 +0100
Subject: [PATCH] Dove: Hook new platform into Kodi build system

---
 configure.ac    |   26 ++++++++++++++++++++++++--
 m4/xbmc_arch.m4 |    4 ++++
 2 files changed, 28 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5d47a4a..0fe3e0c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -709,6 +709,25 @@ case $use_platform in
      CFLAGS="$CFLAGS"
      CXXFLAGS="$CXXFLAGS"
      ;;
+
+  marvell-dove)
+     target_platform=target_marvell_dove
+     use_x11=yes
+     use_sdl=yes
+     use_neon=no
+     use_arch="arm"
+     use_cpu=cortex-a9
+     use_hardcoded_tables="yes"
+     use_alsa="yes"
+     ARCH="arm"
+     AC_DEFINE(HAS_MARVELL_DOVE, [1], [Define if supporting Marvell Armada 510 (Dove)])
+     CFLAGS="$CFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -marm -O3 -ftree-vectorize"
+     CXXFLAGS="$CXXFLAGS -Wno-psabi -Wa,-march=armv7a -mtune=cortex-a9 -mfpu=vfpv3-d16 -marm -O3 -ftree-vectorize"
+     USE_MARVELL_DOVE=1
+     ;;
+
+  *)
+     use_alsa="yes"
 esac
 
 if test "$host_vendor" = "apple"; then
@@ -804,7 +823,8 @@ if test "$host_vendor" = "apple" ; then
     #arm/ios
     LIBS="$LIBS -framework CoreFoundation"
   fi
-elif test "$target_platform" = "target_raspberry_pi"; then
+elif test "$target_platform" = "target_raspberry_pi" || \
+     test "$target_platform" = "target_marvell_dove"; then
   ARCH="arm"
   use_arch="arm"
 elif test "$use_arch" = "arm"; then
@@ -1122,7 +1142,8 @@ int main() {
   )])
 AC_LANG_POP([C++])
 
-if test "${target_platform}" = "target_linux" || test "${target_platform}" = "target_raspberry_pi" || test "${target_platform}" = "target_android"; then
+if test "${target_platform}" = "target_linux" || test "${target_platform}" = "target_raspberry_pi" || \
+   test "${target_platform}" = "target_android" || test "${target_platform}" = "target_marvell_dove"; then
   PKG_CHECK_MODULES([UUID], [uuid],, AC_MSG_ERROR(libuuid not found))
 fi
 
@@ -2512,6 +2533,7 @@ AC_SUBST(USE_LIBNFS)
 AC_SUBST(USE_LIBGIF)
 AC_SUBST(USE_AIRPLAY)
 AC_SUBST(USE_OPENMAX)
+AC_SUBST(USE_MARVELL_DOVE)
 AC_SUBST(USE_PULSE)
 AC_SUBST(HAVE_LIBPULSE)
 AC_SUBST(USE_ALSA)
diff --git a/m4/xbmc_arch.m4 b/m4/xbmc_arch.m4
index 0b66a82..0f23e7b 100644
--- a/m4/xbmc_arch.m4
+++ b/m4/xbmc_arch.m4
@@ -81,5 +81,9 @@ case $use_platform in
   raspberry-pi)
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_RASPBERRY_PI")
      ;;
+     
+  marvell-dove)
+     AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_ARMEL -DTARGET_MARVELL_DOVE")
+     ;;
 esac
 ])
-- 
1.7.9.5

