From 7707795ac6dd2f7a139f30bf5d295173c8b9debd Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 16 Mar 2017 09:30:44 +0100
Subject: [PATCH] Dove: Disable HD audio formats

---
 xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index e22db7a..a66244d 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -1545,6 +1545,10 @@ void CAESinkALSA::EnumerateDevice(AEDeviceInfoList &list, const std::string &dev
     {
       if (!AE_IS_RAW(i))
         continue;
+      
+      if (AE_IS_RAW_HD(i) && i != AE_FMT_EAC3)
+        continue;
+
       it = find(info.m_dataFormats.begin(), info.m_dataFormats.end(), i);
       if (it == info.m_dataFormats.end())
       {
-- 
1.7.9.5

