From 4b02e90bb09cf79e92e20cdbc3dc9cb3a6c9d931 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 16 Mar 2017 09:30:44 +0100
Subject: [PATCH] Dove: Enable CPU temperature retrieval

---
 xbmc/utils/CPUInfo.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/utils/CPUInfo.cpp b/xbmc/utils/CPUInfo.cpp
index f1d07dd..e9100de 100644
--- a/xbmc/utils/CPUInfo.cpp
+++ b/xbmc/utils/CPUInfo.cpp
@@ -266,6 +266,8 @@ CCPUInfo::CCPUInfo(void)
   if (m_fProcTemperature == NULL)   
     m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/temp1_input", "r");
   if (m_fProcTemperature == NULL)   
+    m_fProcTemperature = fopen("/sys/class/hwmon/hwmon0/device/temp1_input", "r");
+  if (m_fProcTemperature == NULL)   
     m_fProcTemperature = fopen("/sys/class/thermal/thermal_zone0/temp", "r");  // On Raspberry PIs
 
   m_fCPUFreq = fopen ("/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq", "r");
-- 
1.7.9.5

