From 2e0201fb7c145ab03a4d581c3641c8f00e17c275 Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Sat, 17 May 2014 15:20:31 +0200
Subject: [PATCH 20/26] Dove: workaround for hang on exit

---
 xbmc/xbmc.cpp |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/xbmc/xbmc.cpp b/xbmc/xbmc.cpp
index 4d1047e..7f49fff 100644
--- a/xbmc/xbmc.cpp
+++ b/xbmc/xbmc.cpp
@@ -136,5 +136,10 @@ extern "C" int XBMC_Run(bool renderGUI)
   g_RBP.Deinitialize();
 #endif
 
+  // workaround for a bug in the Vivante gpu driver
+#if defined(TARGET_MARVELL_DOVE)
+  _exit(status);
+#else
   return status;
+#endif
 }
-- 
1.7.9.5

