From 2e6b9573dffac4292100eca4a01f76e1a03d665e Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 16 Mar 2017 09:30:45 +0100
Subject: [PATCH] Dove: Fix HW decoder startup failure on some H264 videos

---
 .../DVDCodecs/Video/DVDVideoCodecVMETA.cpp         |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
index 7fe43ec..928c8bc 100644
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecVMETA.cpp
@@ -124,7 +124,6 @@ CDVDVideoCodecVMETA::~CDVDVideoCodecVMETA()
 bool CDVDVideoCodecVMETA::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 {
   IppCodecStatus ret;
-  bool bSendCodecConfig = true;
 
   m_picture_width  = m_display_width  = hints.width;
   m_picture_height = m_display_height = hints.height;
@@ -188,7 +187,6 @@ bool CDVDVideoCodecVMETA::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
       }
 
       m_VDecParSet.strm_fmt = IPP_VIDEO_STRM_FMT_H264;
-      bSendCodecConfig = false;
     }
     break;
 
@@ -290,8 +288,7 @@ bool CDVDVideoCodecVMETA::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
   m_frame_no = 0;
   m_is_open = true;
 
-  if(bSendCodecConfig)
-    SendCodecConfig();
+  SendCodecConfig();
 
   CLog::Log(LOGDEBUG, "%s::%s - VMETA Decoder opened with codec : %s [%dx%d]",
             CLASSNAME, __func__, m_video_codec_name.c_str(), m_display_width, m_display_height);
-- 
1.7.9.5

