--- a/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp	2017-01-17 20:44:57.607277505 +0100
+++ b/xbmc/cores/VideoRenderers/DoveOverlayRenderer.cpp	2017-01-17 20:48:31.065378012 +0100
@@ -78,25 +78,20 @@
 
 void CDoveOverlayRenderer::ManageDisplay(bool first)
 {
-  CRect view;
   int interpolation = m_interpolation;
   struct _sOvlySurface overlaySurface = m_overlaySurface;
   const CDisplaySettings& dispInfo = CDisplaySettings::GetInstance();
-  const RESOLUTION_INFO& resInfo = dispInfo.GetCurrentResolutionInfo();
   const CVideoSettings& vidInfo = CMediaSettings::GetInstance().GetCurrentVideoSettings();
 
-  view.x1 = (float)resInfo.Overscan.left;
-  view.y1 = (float)resInfo.Overscan.top;
-  view.x2 = (float)resInfo.Overscan.right;
-  view.y2 = (float)resInfo.Overscan.bottom;
+  m_viewRect = g_graphicsContext.GetViewWindow();
 
   m_sourceRect.x1 = 0.0;
   m_sourceRect.y1 = 0.0;
   m_sourceRect.x2 = (float)m_sourceWidth;
   m_sourceRect.y2 = (float)m_sourceHeight;
 
-  CalcNormalDisplayRect(view.x1, view.y1,
-                        view.Width(), view.Height(),
+  CalcNormalDisplayRect(m_viewRect.x1, m_viewRect.y1,
+                        m_viewRect.Width(), m_viewRect.Height(),
                         GetAspectRatio() * dispInfo.GetPixelRatio(),
                         dispInfo.GetZoomAmount(), dispInfo.GetVerticalShift());
 
