From 791b894ec1f8d683578f0478bd9634f0992e661d Mon Sep 17 00:00:00 2001
From: Pascal Patry <ppatry@spectrumdt.com>
Date: Wed, 21 Sep 2016 09:36:01 -0400
Subject: [PATCH] Ticket #16906 - Fix iEndOffset property when parsing cue
 files

---
 xbmc/CueDocument.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/CueDocument.cpp b/xbmc/CueDocument.cpp
index bb430040..c3d00d6 100644
--- a/xbmc/CueDocument.cpp
+++ b/xbmc/CueDocument.cpp
@@ -324,7 +324,7 @@ bool CCueDocument::Parse(CueReader& reader, const std::string& strFile)
         CLog::Log(LOGERROR, "Mangled Time in INDEX 0x tag in CUE file!");
         return false;
       }
-      if (totalTracks > 0) // Set the end time of the last track
+      if (totalTracks > 0 && m_tracks[totalTracks - 1].strFile == strCurrentFile) // Set the end time of the last track
         m_tracks[totalTracks - 1].iEndTime = time;
 
       if (totalTracks >= 0) // start time of the next track
