--- a/xbmc/interfaces/python/XBPython.cpp	2015-03-15 11:35:48.855803590 +0100
+++ b/xbmc/interfaces/python/XBPython.cpp	2015-03-15 11:35:48.867803590 +0100
@@ -564,8 +564,7 @@


     // Darwin packs .pyo files, we need PYTHONOPTIMIZE on in order to load them.
-    // linux built with unified builds only packages the pyo files so need it
-#if defined(TARGET_DARWIN) || defined(TARGET_LINUX)
+#if defined(TARGET_DARWIN)
     setenv("PYTHONOPTIMIZE", "1", 1);
 #endif
     // Info about interesting python envvars available
