--- a/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2015-12-20 13:57:24.770476402 +0100
+++ b/xbmc/windowing/egl/EGLNativeTypeIMX.cpp	2015-12-20 13:57:24.798476403 +0100
@@ -302,7 +302,7 @@
   RESOLUTION_INFO res;
   for (size_t i = 0; i < probe_str.size(); i++)
   {
-    if(!StringUtils::StartsWith(probe_str[i], "S:") && !StringUtils::StartsWith(probe_str[i], "U:") &&
+    if(!StringUtils::StartsWith(probe_str[i], "S:") && !StringUtils::StartsWith(probe_str[i], "D:") &&
        !StringUtils::StartsWith(probe_str[i], "V:"))
       continue;

