--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-09-18 17:53:01.250676755 +0200
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp	2016-09-18 17:53:01.274676756 +0200
@@ -1499,13 +1499,14 @@
 bool CIMXContext::Blank()
 {
   if (!m_fbHandle) return false;
-  return ioctl(m_fbHandle, FBIOBLANK, 1) == 0;
+  return ioctl(m_fbHandle, FBIOBLANK, FB_BLANK_NORMAL) == 0;
 }
 
 bool CIMXContext::Unblank()
 {
   if (!m_fbHandle) return false;
-  return ioctl(m_fbHandle, FBIOBLANK, FB_BLANK_UNBLANK) == 0;
+  return ioctl(m_fbHandle, FBIOBLANK, FB_BLANK_UNBLANK) == 0 &&
+         ioctl(m_fbHandle, MXCFB_WAIT_FOR_VSYNC, NULL);
 }
 
 bool CIMXContext::SetVSync(bool enable)
