--- bootsplash-3.1.orig/Utilities/splash.c	2004-09-20 20:54:42.832461640 +0200
+++ bootsplash-3.1/Utilities/splash.c	2004-09-20 23:49:27.094611528 +0200
@@ -517,13 +517,13 @@
 				exit(1);
 			}
 			fclose(sfp);
-			if (sboxcount)
-				memmove(pic + 47 + l + boxcount * 12,
-					sboxes, sboxcount * 12);
-			silentl += sboxcount * 12;
-			l += silentl;
 		}
 		fclose(fp);
+		if (sboxcount)
+			memmove(pic + 47 + l + boxcount * 12,
+				sboxes, sboxcount * 12);
+		silentl += sboxcount * 12;
+		l += silentl;
 	} else {
 		l = 0;
 		pic = calloc(47, 1);
