diff -Naur vdr-xine-0.9.3/xineDevice.c vdr-xine-0.9.3a/xineDevice.c
--- vdr-xine-0.9.3/xineDevice.c	2009-06-17 20:06:35.000000000 +0200
+++ vdr-xine-0.9.3a/xineDevice.c	2010-02-16 12:10:01.520203908 +0100
@@ -1539,7 +1539,13 @@
   {
 #if APIVERSNUM >= 10701
     if (Length >= TS_SIZE)
-      m_tsVideoPid = TsPid(Data);   
+    {
+#if APIVERSNUM >= 10712
+      m_tsVideoPid = PatPmtParser()->Vpid();
+#else
+      m_tsVideoPid = TsPid(Data);
+#endif
+    }
 #endif
     return Length;
   }
@@ -4119,6 +4125,8 @@
   
   void cXineDevice::MakePrimaryDevice(bool On)
   {
+    cDevice::MakePrimaryDevice(On);
+
     xfprintf(stderr, "-------------------------\n");
     xfprintf(stderr, "MakePrimaryDevice: %d\n", On);
     xfprintf(stderr, "=========================\n");
