diff -Naur a/ipu/Makefile b/ipu/Makefile
--- a/ipu/Makefile	2013-06-21 15:22:18.000000000 +0200
+++ b/ipu/Makefile	2013-12-22 16:44:49.376043061 +0100
@@ -31,7 +31,7 @@
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ -o $@ $^ -lpthread
 
 $(LIBNAME).so: $(LIBNAME).so.$(SONAMEVERSION)
-	ln -s $< $@
+	ln -sf $< $@
 
 %.o: %.c
 ifeq "$(PLATFORM)" ""
diff -Naur a/vpu/Makefile b/vpu/Makefile
--- a/vpu/Makefile	2013-06-21 15:22:18.000000000 +0200
+++ b/vpu/Makefile	2013-12-22 16:45:04.896042602 +0100
@@ -39,7 +39,7 @@
 	$(CC) -shared -nostartfiles -Wl,-soname,$@ $^ -o $@ -lpthread
 
 $(LIBNAME).so: $(LIBNAME).so.$(SONAMEVERSION)
-	ln -s $< $@
+	ln -sf $< $@
 
 $(LIBNAME).a: $(OBJ)
 	$(AR) -rc $@  $^
