diff -ruN enna-0.5.0+c39ea212.orig/data/theme/default/Makefile.am enna-0.5.0+c39ea212/data/theme/default/Makefile.am
--- enna-0.5.0+c39ea212.orig/data/theme/default/Makefile.am	2013-04-04 00:12:13.203775552 +0300
+++ enna-0.5.0+c39ea212/data/theme/default/Makefile.am	2013-04-04 00:13:17.784095490 +0300
@@ -425,7 +425,7 @@
   $(EDC)
 
 default.edj: Makefile $(EXTRA_DIST)
-	$(AM_V_GEN)$(EDJE_CC) $(EDJE_VERB) $(EDJE_FLAGS) \
+	$(EDJE_CC) $(EDJE_VERB) $(EDJE_FLAGS) \
 	$(top_srcdir)/data/theme/default/default.edc \
 	$(top_builddir)/data/theme/default/default.edj
 
diff -ruN enna-0.5.0+c39ea212.orig/data/theme/stb/Makefile.am enna-0.5.0+c39ea212/data/theme/stb/Makefile.am
--- enna-0.5.0+c39ea212.orig/data/theme/stb/Makefile.am	2013-04-04 00:12:13.195775555 +0300
+++ enna-0.5.0+c39ea212/data/theme/stb/Makefile.am	2013-04-04 00:12:52.239968090 +0300
@@ -234,7 +234,7 @@
 	$(IMG_WEATHER_ICONS)
 
 stb.edj: Makefile $(EXTRA_DIST)
-	$(AM_V_GEN)$(EDJE_CC) $(EDJE_FLAGS) \
+	$(EDJE_CC) $(EDJE_FLAGS) \
 	$(top_srcdir)/data/theme/stb/stb.edc \
 	$(top_builddir)/data/theme/stb/stb.edj
 
