diff -Naur enna-r2479/configure enna-r2479a/configure
--- enna-r2479/configure	2009-12-25 12:03:09.000000000 +0100
+++ enna-r2479a/configure	2009-12-25 13:53:08.000000000 +0100
@@ -14087,7 +14087,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14104,7 +14103,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14124,7 +14122,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14153,7 +14150,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14170,7 +14166,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14190,7 +14185,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14225,7 +14219,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 "`
         else
@@ -14243,7 +14236,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 "`
         fi
@@ -14264,7 +14256,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0 libplayer < 1.1.0
    libvalhalla >= 1.0.0 libvalhalla < 2.0.0
-   dbus-1 >= 1.2.0
 
 ) were not met:
 
