--- a/configure	2009-12-18 22:46:28.000000000 +0100
+++ b/configure	2009-12-18 22:46:28.000000000 +0100
@@ -14053,7 +14053,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14070,7 +14069,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14090,7 +14088,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14119,7 +14116,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 \""; } >&5
   ($PKG_CONFIG --exists --print-errors "
@@ -14136,7 +14132,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 ") 2>&5
   ac_status=$?
@@ -14156,7 +14151,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 " 2>/dev/null`
 else
@@ -14191,7 +14185,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 "`
         else
@@ -14209,7 +14202,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 "`
         fi
@@ -14230,7 +14222,6 @@
    elementary >= 0.6.0.063
    libplayer >= 1.0.0
    libvalhalla >= 1.0.0
-   dbus-1 >= 1.2.0
 
 ) were not met:
 
