diff -r f2c716d0954e src/bin/enna.c
--- a/src/bin/enna.c	Sat Jan 09 11:15:54 2010 +0100
+++ b/src/bin/enna.c	Sun Jan 10 15:50:20 2010 +0100
@@ -32,6 +32,7 @@
 #include <Ecore_Str.h>
 #ifdef BUILD_ECORE_X
 #include <Ecore_X.h>
+#include <X11/Xlib.h>
 #endif
 #include <Elementary.h>
 
@@ -516,6 +517,7 @@
         return EXIT_SUCCESS;
 
     url_global_init();
+    XInitThreads();
     eina_init();
 
     /* Must be called first */
