# See enna r3136
--- enna-r3117/src/bin/utils.c	2010-05-02 14:17:17.000000000 +0200
+++ enna-r3117/src/bin/utils.c	2010-06-02 09:40:58.000000000 +0200
@@ -31,7 +31,6 @@
 #include <Edje.h>
 
 #include "enna.h"
-#include "logs.h"
 #include "enna_config.h"
 #include "utils.h"
 #include "vfs.h"
@@ -56,12 +55,7 @@
     if (!home)
         return getenv("CWD")?strdup(getenv("CWD")):NULL;
     if (!home)
-    {
-        enna_log(ENNA_MSG_WARNING, NULL,
-                 "Neither HOME/CWD has been set, using / instead");
-
         return strdup("/");
-    }
     return home;
 }
 
