diff -Naur enna-r1786.orig/po/POTFILES.in enna-r1786/po/POTFILES.in
--- enna-r1786.orig/po/POTFILES.in	2009-09-15 21:51:41.000000000 +0200
+++ enna-r1786/po/POTFILES.in	2009-09-15 21:51:50.000000000 +0200
@@ -16,7 +16,6 @@
 src/modules/activity/weather/weather.c
 src/modules/browser/cdda/cdda.c
 src/modules/browser/dvd/dvd.c
-src/modules/browser/lms/lms.c
 src/modules/browser/localfiles/localfiles.c
 src/modules/browser/netstreams/netstreams.c
 src/modules/browser/podcasts/podcasts.c
