diff -Naur enna-0.4.0.orig/src/modules/browser/localfiles/localfiles.c enna-0.4.0/src/modules/browser/localfiles/localfiles.c
--- enna-0.4.0.orig/src/modules/browser/localfiles/localfiles.c	2010-01-09 12:03:35.945902871 +0100
+++ enna-0.4.0/src/modules/browser/localfiles/localfiles.c	2010-01-09 12:04:15.586771780 +0100
@@ -405,16 +405,6 @@
             data->config->root_directories, root);
     }
 
-    // add home directory entry
-    root = ENNA_NEW(Root_Directories, 1);
-    snprintf(buf, sizeof(buf), "file://%s", enna_util_user_home_get());
-    root->uri = strdup(buf);
-    root->label = strdup("Home");
-    root->icon = strdup("icon/favorite");
-
-    data->config->root_directories = eina_list_append(
-        data->config->root_directories, root);
-
     /* add localfiles to the list of volumes listener */
     data->vl = enna_volumes_listener_add("localfiles", _add_volumes_cb, _remove_volumes_cb, data);
 }
