diff -Naur enna-r3117/configure enna-r3117a/configure
--- enna-r3117/configure	2010-05-02 14:17:50.000000000 +0200
+++ enna-r3117a/configure	2010-06-07 14:15:11.650200181 +0200
@@ -14105,7 +14105,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14120,7 +14120,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14138,7 +14138,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14165,7 +14165,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14180,7 +14180,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14198,7 +14198,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14231,7 +14231,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14247,7 +14247,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
@@ -14266,7 +14266,7 @@
    ecore-evas >= 0.9.9.063
    ecore-file >= 0.9.9.063
    ecore-ipc >= 0.9.9.063
-   edje >= 0.9.93.063
+   edje >= 0.9.9.063
    elementary >= 0.6.0.063
    libvalhalla >= 2.0.0 libvalhalla < 3.0.0
    libxml-2.0
