diff -Naur zvbi-0.2.33.orig/src/misc.h zvbi-0.2.33/src/misc.h
--- zvbi-0.2.33.orig/src/misc.h	2010-09-05 19:45:16.768937097 +0200
+++ zvbi-0.2.33/src/misc.h	2010-09-05 19:45:53.385187878 +0200
@@ -35,6 +35,10 @@
 #include <limits.h>		/* (S)SIZE_MAX */
 #include <assert.h>
 
+#ifndef S_ISCHR
+#define S_ISCHR(m)      (((m) & S_IFMT) == S_IFCHR)
+#endif
+
 #include "macros.h"
 #include "version.h"
 
