diff -urN lzma-4.22-orig/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp lzma-4.22/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp
--- lzma-4.22-orig/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp	2005-06-08 15:44:38.000000000 +0200
+++ lzma-4.22/SRC/7zip/Compress/LZMA_Alone/LzmaAlone.cpp	2005-07-09 21:38:19.000000000 +0200
@@ -27,11 +27,9 @@
 #include "../LZMA/LZMAEncoder.h"
 
 #include "LzmaBench.h"
-#include "LzmaRam.h"
 
 extern "C"
 {
-#include "LzmaRamDecode.h"
 }
 
 using namespace NCommandLineParser;
@@ -51,7 +49,6 @@
   kEOS,
   kStdIn,
   kStdOut,
-  kFilter86
 };
 }
 
@@ -69,7 +66,6 @@
   { L"EOS", NSwitchType::kSimple, false },
   { L"SI",  NSwitchType::kSimple, false },
   { L"SO",  NSwitchType::kSimple, false },
-  { L"F86",  NSwitchType::kSimple, false }
 };
 
 static const int kNumSwitches = sizeof(kSwitchForms) / sizeof(kSwitchForms[0]);
@@ -250,6 +246,7 @@
     }
   }
 
+#if 0
   if (parser[NKey::kFilter86].ThereIs)
   {
     // -f86 switch is for x86 filtered mode: BCJ + LZMA.
@@ -308,6 +305,7 @@
     MyFree(inBuffer);
     return 0;
   }
+#endif
 
 
   UInt64 fileSize;
diff -urN lzma-4.22-orig/SRC/7zip/Compress/LZMA_Alone/makefile lzma-4.22/SRC/7zip/Compress/LZMA_Alone/makefile
--- lzma-4.22-orig/SRC/7zip/Compress/LZMA_Alone/makefile	2005-04-05 19:20:54.000000000 +0200
+++ lzma-4.22/SRC/7zip/Compress/LZMA_Alone/makefile	2005-07-09 21:38:43.000000000 +0200
@@ -8,10 +8,7 @@
 OBJS = \
   LzmaAlone.o \
   LzmaBench.o \
-  LzmaRam.o \
-  LzmaRamDecode.o \
   LzmaDecode.o \
-  BranchX86.o \
   LZMADecoder.o \
   LZMAEncoder.o \
   LZInWindow.o \
