diff -uNr enigma2pc-20130727.orig/data/xine.conf enigma2pc-20130727/data/xine.conf
--- enigma2pc-20130727.orig/data/xine.conf	2014-01-11 22:24:09.618032189 +0800
+++ enigma2pc-20130727/data/xine.conf	2014-01-12 17:04:29.012756151 +0800
@@ -75,7 +75,7 @@
 
 # vdpau: HD deinterlace method
 # { bob  half temporal  half temporal_spatial  temporal  temporal_spatial }, default: 3
-video.output.vdpau_hd_deinterlace_method:temporal
+video.output.vdpau_hd_deinterlace_method:bob
 
 # vdpau: disable deinterlacing when progressive_frame flag is set
 # bool, default: 0
@@ -83,7 +83,7 @@
 
 # vdpau: SD deinterlace method
 # { bob  half temporal  half temporal_spatial  temporal  temporal_spatial }, default: 3
-video.output.vdpau_sd_deinterlace_method:temporal_spatial
+video.output.vdpau_sd_deinterlace_method:bob
 
 # vdpau: restrict enabling video properties for SD video only
 # { none  noise  sharpness  noise+sharpness }, default: 0
