diff -Nur i810tvout-0.9.orig/ldvo.c i810tvout-0.9/ldvo.c
--- i810tvout-0.9.orig/ldvo.c	Wed Dec 22 17:10:12 2004
+++ i810tvout-0.9/ldvo.c	Thu Nov  3 14:29:12 2005
@@ -134,7 +134,7 @@
   //pci_scan_bus(access);
 
   //Look for i810
-  dev = pci_get_dev(access,0,1,0);
+  dev = pci_get_dev(access,0,0,1,0);
   if(dev) {
     pci_fill_info(dev,PCI_FILL_IDENT|PCI_FILL_BASES);
     LDVO_DEBUG("DEVICE 0:1:0 == %x %x\n",dev->vendor_id, dev->device_id);
@@ -155,7 +155,7 @@
     if(dev) {
       pci_free_dev(dev);
     }
-    dev = pci_get_dev(access,0,2,0);
+    dev = pci_get_dev(access,0,0,2,0);
     if(dev) {
       pci_fill_info(dev,PCI_FILL_IDENT|PCI_FILL_BASES);
       LDVO_DEBUG("DEVICE 0:2:0 == %x %x\n",dev->vendor_id, dev->device_id);    
