--- a/mono/Makefile.in	2018-01-10 09:11:41.282345328 +0100
+++ b/mono/Makefile.in	2018-01-10 09:11:41.322339914 +0100
@@ -391,10 +391,10 @@
 top_srcdir = @top_srcdir@
 @SUPPORT_SGEN_TRUE@sgen_dirs = sgen
 @BTLS_TRUE@btls_dirs = btls
-@CROSS_COMPILING_FALSE@@INSTALL_MONOTOUCH_FALSE@SUBDIRS = $(btls_dirs) arch utils io-layer cil metadata $(sgen_dirs) mini dis tests unit-tests benchmark profiler
+@CROSS_COMPILING_FALSE@@INSTALL_MONOTOUCH_FALSE@SUBDIRS = $(btls_dirs) arch utils io-layer cil metadata $(sgen_dirs) mini dis profiler
 @CROSS_COMPILING_FALSE@@INSTALL_MONOTOUCH_TRUE@SUBDIRS = $(btls_dirs) arch utils io-layer metadata $(sgen_dirs) mini profiler
 @CROSS_COMPILING_TRUE@SUBDIRS = $(btls_dirs) arch utils io-layer cil metadata $(sgen_dirs) mini dis profiler
-DIST_SUBDIRS = btls arch utils io-layer cil metadata $(sgen_dirs) mini dis tests unit-tests benchmark profiler
+DIST_SUBDIRS = btls arch utils io-layer cil metadata $(sgen_dirs) mini dis profiler
 all: all-recursive
 
 .SUFFIXES:
--- a/mono/Makefile.am	2018-01-10 09:22:52.296258358 +0100
+++ a/mono/Makefile.am	2018-01-10 09:22:52.336253008 +0100
@@ -34,7 +34,7 @@
 	  (cd $$subdir && $(MAKE) $(AM_MAKEFLAGS) $$target); \
     done;
 else
-SUBDIRS = $(btls_dirs) arch utils io-layer cil metadata $(sgen_dirs) mini dis tests unit-tests benchmark profiler
+SUBDIRS = $(btls_dirs) arch utils io-layer cil metadata $(sgen_dirs) mini dis profiler
 endif
 endif
-DIST_SUBDIRS = btls arch utils io-layer cil metadata $(sgen_dirs) mini dis tests unit-tests benchmark profiler
+DIST_SUBDIRS = btls arch utils io-layer cil metadata $(sgen_dirs) mini dis profiler
