--- weston-0.85.0/clients/Makefile.in.orig	2012-02-09 20:01:06.000000000 +0200
+++ weston-0.85.0/clients/Makefile.in	2012-07-26 20:40:38.000000000 +0300
@@ -69,7 +69,7 @@
 @BUILD_CLIENTS_TRUE@am__EXEEXT_3 = weston-tablet-shell$(EXEEXT)
 @BUILD_FULL_GL_CLIENTS_TRUE@am__EXEEXT_4 = gears$(EXEEXT) \
 @BUILD_FULL_GL_CLIENTS_TRUE@	wscreensaver$(EXEEXT)
-@BUILD_CLIENTS_TRUE@am__EXEEXT_5 = flower$(EXEEXT) screenshot$(EXEEXT) \
+@BUILD_CLIENTS_TRUE@am__EXEEXT_5 = flower$(EXEEXT) \
 @BUILD_CLIENTS_TRUE@	image$(EXEEXT) dnd$(EXEEXT) smoke$(EXEEXT) \
 @BUILD_CLIENTS_TRUE@	resizor$(EXEEXT) eventdemo$(EXEEXT) \
 @BUILD_CLIENTS_TRUE@	$(am__EXEEXT_4)
@@ -439,8 +439,6 @@
 @BUILD_CLIENTS_TRUE@	../shared/libconfig-parser.la
 
 @BUILD_CLIENTS_TRUE@BUILT_SOURCES = \
-@BUILD_CLIENTS_TRUE@	screenshooter-client-protocol.h		\
-@BUILD_CLIENTS_TRUE@	screenshooter-protocol.c		\
 @BUILD_CLIENTS_TRUE@	desktop-shell-client-protocol.h		\
 @BUILD_CLIENTS_TRUE@	desktop-shell-protocol.c		\
 @BUILD_CLIENTS_TRUE@	tablet-shell-client-protocol.h		\
