--- weston-1.0.0/clients/Makefile.am.orig	2012-10-23 12:34:34.511921827 +0000
+++ weston-1.0.0/clients/Makefile.am	2012-10-23 12:35:18.518351992 +0000
@@ -11,7 +11,6 @@
 libexec_PROGRAMS =				\
 	$(desktop_shell)			\
 	$(tablet_shell)				\
-	$(screenshooter)			\
 	$(screensaver)
 
 AM_CFLAGS = $(GCC_CFLAGS)
@@ -69,7 +68,6 @@
 tablet_shell = weston-tablet-shell
 endif
 
-screenshooter = weston-screenshooter
 
 noinst_LIBRARIES = libtoytoolkit.a
 
@@ -89,8 +87,6 @@
 flower_SOURCES = flower.c
 flower_LDADD = $(toolkit_libs)
 
-weston_screenshooter_SOURCES = screenshot.c screenshooter-protocol.c
-weston_screenshooter_LDADD = $(toolkit_libs)
 
 weston_terminal_SOURCES = terminal.c
 weston_terminal_LDADD = $(toolkit_libs) -lutil
@@ -150,8 +146,6 @@
 weston_tablet_shell_LDADD = $(toolkit_libs)
 
 BUILT_SOURCES =					\
-	screenshooter-client-protocol.h		\
-	screenshooter-protocol.c		\
 	text-cursor-position-client-protocol.h	\
 	text-cursor-position-protocol.c		\
 	text-protocol.c				\
