--- performous-0.7.0/game/ffmpeg.cc.org	2013-12-28 20:32:22.647939373 +0100
+++ performous-0.7.0/game/ffmpeg.cc	2013-12-28 20:33:08.003938022 +0100
@@ -16,6 +16,10 @@
 #	define AV_SAMPLE_FMT_S16 SAMPLE_FMT_S16
 #endif
 
+#ifndef AVCODEC_MAX_AUDIO_FRAME_SIZE
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000 // 1 second of 48khz 32bit audio
+#endif
+
 #define AUDIO_CHANNELS 2
 
 /*static*/ boost::mutex FFmpeg::s_avcodec_mutex;
