--- a/lib/libdvd/Makefile.in	2015-09-22 21:35:38.479997968 +0200
+++ b/lib/libdvd/Makefile.in	2015-09-22 21:35:38.499997969 +0200
@@ -55,12 +55,12 @@
 else
 
 $(SYSDIR)/libdvdcss-$(ARCH).so: $(WRAPPER) libdvdcss/.libs/libdvdcss.a
-	$(CC) -o $@ $(LDFLAGS) -Wl,--soname,$@ \
+	$(CC) -o $@ $(LDFLAGS) -Wl,--soname,$(@F) \
 		libdvdcss/src/*.o -Wl,--unresolved-symbols=ignore-all -lm \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
 $(SYSDIR)/libdvdnav-$(ARCH).so: $(WRAPPER) $(DVDCSS_A) libdvdread/obj/libdvdread.a libdvdnav/obj/libdvdnav.a
-	$(CC) -o $@ $(LDFLAGS) -Wl,--soname,$@ $(DVDCSS_O) libdvdread/obj/*.o libdvdnav/obj/*.o -lm \
+	$(CC) -o $@ $(LDFLAGS) -Wl,--soname,$(@F) $(DVDCSS_O) libdvdread/obj/*.o libdvdnav/obj/*.o -lm \
 		-Wl,--unresolved-symbols=ignore-all \
 		`cat $(WRAPPER:.o=.def)` $(WRAPPER)
 
