diff -Naur util-linux-ng-2.16.orig/login-utils/Makefile.am util-linux-ng-2.16/login-utils/Makefile.am
--- util-linux-ng-2.16.orig/login-utils/Makefile.am	2010-07-10 18:21:44.134722129 +0200
+++ util-linux-ng-2.16/login-utils/Makefile.am	2010-07-10 18:22:12.355698610 +0200
@@ -108,7 +108,7 @@
 if USE_TTY_GROUP
 if MAKEINSTALL_DO_CHOWN
 install-exec-hook::
-	chgrp tty $(DESTDIR)$(usrbin_execdir)/wall
+	true tty $(DESTDIR)$(usrbin_execdir)/wall
 	chmod g+s $(DESTDIR)$(usrbin_execdir)/wall
 endif
 endif
diff -Naur util-linux-ng-2.16.orig/login-utils/Makefile.in util-linux-ng-2.16/login-utils/Makefile.in
--- util-linux-ng-2.16.orig/login-utils/Makefile.in	2010-07-10 18:21:44.134722129 +0200
+++ util-linux-ng-2.16/login-utils/Makefile.in	2010-07-10 18:22:20.084718825 +0200
@@ -1282,7 +1282,7 @@
 @BUILD_LOGIN_UTILS_TRUE@	cd $(DESTDIR)$(usrsbin_execdir) && ln -sf vipw vigr
 
 @BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@install-exec-hook::
-@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chgrp tty $(DESTDIR)$(usrbin_execdir)/wall
+@BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	true tty $(DESTDIR)$(usrbin_execdir)/wall
 @BUILD_WALL_TRUE@@MAKEINSTALL_DO_CHOWN_TRUE@@USE_TTY_GROUP_TRUE@	chmod g+s $(DESTDIR)$(usrbin_execdir)/wall
 
 @BUILD_INIT_TRUE@install-exec-hook::
