diff -ruN ffnetdev-2012.01.07//ffnetdev.c ffnetdev-mod//ffnetdev.c
--- ffnetdev-2012.01.07//ffnetdev.c	2012-01-07 16:51:41.000000000 +0100
+++ ffnetdev-mod//ffnetdev.c	2012-04-02 20:30:05.000000000 +0200
@@ -10,7 +10,6 @@
 
 #include <vdr/tools.h>
 
-#include "i18n.h"
 #include "tsworker.h"
 #include "netosd.h"
 #include "ffnetdev.h"
@@ -163,8 +162,7 @@
 bool cPluginFFNetDev::Start(void)
 {
   // Start any background activities the plugin shall perform.
-  RegisterI18n(Phrases);
-  
+   
   	  
   cOSDWorker::Init(OSDPort, this);
   cTSWorker::Init(m_StreamDevice, TSPort, this);
diff -ruN ffnetdev-2012.01.07//Makefile ffnetdev-mod//Makefile
--- ffnetdev-2012.01.07//Makefile	2012-01-07 16:51:41.000000000 +0100
+++ ffnetdev-mod//Makefile	2012-04-02 20:29:43.000000000 +0200
@@ -19,7 +19,7 @@
 ### The C++ compiler and options:
 
 CXX      ?= g++
-CXXFLAGS ?= -W -Woverloaded-virtual
+CXXFLAGS ?= -W -Werror=overloaded-virtual
 
 ### The directory environment:
 
@@ -50,7 +50,7 @@
 
 ### The object files (add further files here):
 
-COMMONOBJS = i18n.o \
+COMMONOBJS = \
 	\
 	tools/source.o tools/select.o tools/socket.o tools/tools.o 
 
