diff -Naur sgx530-mine-4.05.00.03/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c sgx530-mine-4.05.00.03_patched//services4/3rdparty/bufferclass_example/bufferclass_example_linux.c
--- sgx530-mine-4.05.00.03/services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2011-12-21 00:42:14.888511758 +0100
+++ sgx530-mine-4.05.00.03_patched//services4/3rdparty/bufferclass_example/bufferclass_example_linux.c	2011-12-21 00:43:14.788511762 +0100
@@ -67,7 +67,7 @@
 static int AssignedMajorNumber;
 
 static struct file_operations bufferclass_example_fops = {
-	ioctl:BC_Example_Bridge,
+	unlocked_ioctl:BC_Example_Bridge,
 };
 
 
