--- a/eurasiacon/build/linux/kbuild/Makefile.kbuild	2012-10-21 21:08:20.000000000 +0200
+++ b/eurasiacon/build/linux/kbuild/Makefile.kbuild	2012-10-21 21:08:28.000000000 +0200
@@ -63,7 +63,7 @@
 	$(EXTRA_SUBDIRS)
 
 ifeq ($(SILENT),)
-all %:
+all:
 	set -x; \
 	for dir in $(KBUILD_SUBDIRS); \
 	do \
@@ -75,7 +75,7 @@
 	fi \
 	done
 else
-all %:
+all:
 	$(SILENT)for dir in $(KBUILD_SUBDIRS); \
 	do \
 	if test -e $$dir;\
--- a/eurasiacon/build/linux/makefile.shared_conf	2012-10-21 21:07:49.000000000 +0200
+++ b/eurasiacon/build/linux/makefile.shared_conf	2012-10-21 21:08:13.000000000 +0200
@@ -161,7 +161,7 @@
 
 KERNELVERSION	= $(KERNEL_VER).$(KERNEL_REL).$(KERNEL_SUBREL)
 
-KERNEL_ID		?= $(shell grep -h '\#define UTS_RELEASE' $(KERNELDIR)/include/linux/* | \
+KERNEL_ID		?= $(shell grep -h '\#define UTS_RELEASE' $(KERNELDIR)/include/generated/* | \
 				$(CUT) -f3 -d' ' | \
 				$(SED) s/\"//g)
 
