--- a/configure.ac	2014-07-16 11:22:26.000000000 +0200
+++ b/configure.ac	2014-12-29 11:36:28.090191583 +0100
@@ -124,7 +124,7 @@
 esac
 
 ### External libraries checks
-AC_CHECK_LIB([GL],[main],have_gl=yes, have_gl=no)  
+AC_CHECK_LIB([GL],[main],[AC_CHECK_HEADER([GL/gl.h],have_gl=yes, have_gl=no)], have_gl=no)
 if test "$have_gl" = "yes"; then
   AC_DEFINE([HAVE_GL],[1],["Define to 1 if we have gl"])
 fi
