diff -Naur nfs-utils-1.2.5.orig/support/include/sockaddr.h nfs-utils-1.2.5/support/include/sockaddr.h
--- nfs-utils-1.2.5.orig/support/include/sockaddr.h	2011-10-27 14:43:25.000000000 -0400
+++ nfs-utils-1.2.5/support/include/sockaddr.h	2011-10-27 14:43:59.000000000 -0400
@@ -20,7 +20,7 @@
 #ifndef NFS_UTILS_SOCKADDR_H
 #define NFS_UTILS_SOCKADDR_H
 
-#include <libio.h>
+#include <stdio.h>
 #include <stdbool.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
