--- nfs-utils-1.2.5/utils/blkmapd/device-process.c	2011-10-28 23:03:53.046615017 -0400
+++ nfs-utils-1.2.5/utils/blkmapd/device-process.c	2011-10-28 23:03:20.973407095 -0400
@@ -296,7 +296,7 @@
 		off_t stripe_unit = vol->param.bv_stripe_unit;
 		/* Check limitations imposed by device-mapper */
 		if ((stripe_unit & (stripe_unit - 1)) != 0
-		    || stripe_unit < (off_t) (PAGE_SIZE >> 9))
+		    || stripe_unit < (off_t) (getpagesize() >> 9))
 			return -EIO;
 		BLK_READBUF(p, end, 4);
 		READ32(vol->bv_vol_n);
