enable only: yesno, msgbox, menu, inputbox

diff -Nur dialog-1.0-20060126.orig/dialog.c dialog-1.0-20060126/dialog.c
--- dialog-1.0-20060126.orig/dialog.c	Thu Jan 19 00:51:37 2006
+++ dialog-1.0-20060126/dialog.c	Sat Feb 11 13:51:58 2006
@@ -895,14 +895,8 @@
 {
     {o_yesno,           4, 4, call_yesno},
     {o_msgbox,          4, 4, call_msgbox},
-    {o_infobox,         4, 4, call_infobox},
-    {o_textbox,         4, 4, call_textbox},
     {o_menu,            7, 0, call_menu},
-    {o_inputmenu,       7, 0, call_inputmenu},
-    {o_checklist,       8, 0, call_checklist},
-    {o_radiolist,       8, 0, call_radiolist},
     {o_inputbox,        4, 5, call_inputbox},
-    {o_passwordbox,     4, 5, call_passwordbox},
 #ifdef HAVE_XDIALOG
     {o_calendar,        4, 7, call_calendar},
     {o_fselect,         4, 5, call_fselect},
