enable only: yesno, msgbox, menu, inputbox

diff -Naur dialog-1.0-20051030.orig/dialog.c dialog-1.0-20051030/dialog.c
--- dialog-1.0-20051030.orig/dialog.c	2005-10-30 22:16:52.000000000 +0200
+++ dialog-1.0-20051030/dialog.c	2005-10-31 17:50:12.000000000 +0200
@@ -853,15 +853,21 @@
 {
     {o_yesno,       4, 4, call_yesno},
     {o_msgbox,      4, 4, call_msgbox},
+#if 0
     {o_infobox,     4, 4, call_infobox},
     {o_textbox,     4, 4, call_textbox},
+#endif
     {o_menu,        7, 0, call_menu},
+#if 0
     {o_inputmenu,   7, 0, call_inputmenu},
     {o_checklist,   8, 0, call_checklist},
     {o_radiolist,   8, 0, call_radiolist},
+#endif
     {o_inputbox,    4, 5, call_inputbox},
+#if 0
     {o_passwordbox, 4, 5, call_passwordbox},
     {o_pause,       5, 5, call_pause},
+#endif
 #ifdef HAVE_XDIALOG
     {o_calendar,    4, 7, call_calendar},
     {o_fselect,     4, 5, call_fselect},
