diff -Naur dialog-1.0-20041118.orig/dialog.c dialog-1.0-20041118/dialog.c
--- dialog-1.0-20041118.orig/dialog.c	2004-11-18 23:23:32.000000000 +0100
+++ dialog-1.0-20041118/dialog.c	2004-12-30 02:12:11.000000000 +0100
@@ -621,6 +621,7 @@
 			 numeric_arg(av, 3), 1);
 }
 
+#if 0
 static int
 j_infobox(JUMPARGS)
 {
@@ -640,6 +641,7 @@
 			  numeric_arg(av, 2),
 			  numeric_arg(av, 3));
 }
+#endif
 
 static int
 j_menu(JUMPARGS)
@@ -655,6 +657,7 @@
 		       tags, av + 5);
 }
 
+#if 0
 static int
 j_inputmenu(JUMPARGS)
 {
@@ -704,6 +707,7 @@
 			    numeric_arg(av, 4),
 			    tags, av + 5, FLAG_RADIO);
 }
+#endif
 
 static int
 j_inputbox(JUMPARGS)
@@ -716,6 +720,7 @@
 			   optional_str(av, 4, 0), 0);
 }
 
+#if 0
 static int
 j_passwordbox(JUMPARGS)
 {
@@ -726,6 +731,7 @@
 			   numeric_arg(av, 3),
 			   optional_str(av, 4, 0), 1);
 }
+#endif
 
 #ifdef HAVE_XDIALOG
 static int
@@ -822,14 +828,20 @@
 {
     {o_yesno,       4, 4, j_yesno},
     {o_msgbox,      4, 4, j_msgbox},
+#if 0
     {o_infobox,     4, 4, j_infobox},
     {o_textbox,     4, 4, j_textbox},
+#endif
     {o_menu,        7, 0, j_menu},
+#if 0
     {o_inputmenu,   7, 0, j_inputmenu},
     {o_checklist,   8, 0, j_checklist},
     {o_radiolist,   8, 0, j_radiolist},
+#endif
     {o_inputbox,    4, 5, j_inputbox},
+#if 0
     {o_passwordbox, 4, 5, j_passwordbox},
+#endif
 #ifdef HAVE_XDIALOG
     {o_calendar,    4, 7, j_calendar},
     {o_fselect,     4, 5, j_fselect},
