diff -Naur dialog-1.1-20070227.orig/dialog.c dialog-1.1-20070227/dialog.c
--- dialog-1.1-20070227.orig/dialog.c	2007-08-27 15:01:37.000000000 +0200
+++ dialog-1.1-20070227/dialog.c	2008-04-06 14:57:01.000000000 +0200
@@ -978,14 +978,9 @@
 {
     {o_yesno,           4, 4, call_yesno},
     {o_msgbox,          4, 4, call_msgbox},
-    {o_infobox,         4, 4, call_infobox},
     {o_textbox,         4, 4, call_textbox},
     {o_menu,            7, 0, call_menu},
-    {o_inputmenu,       7, 0, call_inputmenu},
-    {o_checklist,       8, 0, call_checklist},
-    {o_radiolist,       8, 0, call_radiolist},
     {o_inputbox,        4, 5, call_inputbox},
-    {o_passwordbox,     4, 5, call_passwordbox},
 #ifdef HAVE_DLG_GAUGE
     {o_gauge,           4, 5, call_gauge},
     {o_pause,           5, 5, call_pause},
