diff -Naur dialog-1.1-20070227.orig/configure dialog-1.1-20070227/configure
--- dialog-1.1-20070227.orig/configure	2007-08-27 15:01:37.000000000 +0200
+++ dialog-1.1-20070227/configure	2007-08-27 15:01:57.000000000 +0200
@@ -7445,9 +7445,6 @@
 
 cf_cv_ncurses_header=curses.h
 
-CPPFLAGS="`$NCURSES_CONFIG --cflags` $CPPFLAGS"
-LIBS="`$NCURSES_CONFIG --libs` $LIBS"
-
 cat >>confdefs.h <<\EOF
 #define NCURSES 1
 EOF
