--- a/addons/pvr.wmc/addon/addon.xml.in	2015-06-16 20:27:19.207196105 +0200
+++ b/addons/pvr.wmc/addon/addon.xml.in	2015-06-16 20:27:19.219196102 +0200
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon
-  id="pvr.wmc-@OS@-@ARCHITECTURE@"
+  id="pvr.wmc"
   version="0.2.99"
-  name="Windows Media Center Client - @OS@ @ARCHITECTURE@ Edition"
+  name="Windows Media Center Client"
   provider-name="KrustyReturns">
   <requires>
     <c-pluff version="0.1"/>
