--- a/addons/Makefile.include.am	2019-11-19 20:44:44.624166259 +0100
+++ b/addons/Makefile.include.am	2019-11-19 20:44:44.660166047 +0100
@@ -3,7 +3,7 @@
 #

 INCLUDES        = -I. -I$(abs_top_srcdir)/xbmc -I$(abs_top_srcdir)/lib @HOST_INCLUDES@
-WARNINGS        = -Wall -Wextra -Wno-missing-field-initializers -Woverloaded-virtual -Wno-parentheses
+WARNINGS        = -Wall -Wextra -Wno-missing-field-initializers -Woverloaded-virtual -Wno-parentheses -Wno-narrowing
 DEFINES         = @ARCH_DEFINES@ -DUSE_DEMUX -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS
 AM_CXXFLAGS     = -g -O2 -fPIC $(WARNINGS) $(DEFINES) @HOST_CXXFLAGS@
 LIB             = @abs_top_srcdir@/addons/$(ADDONNAME)/addon/@BINPREFIX@$(ADDONBINNAME)@BIN_EXT@
--- a/addons/pvr.mythtv.cmyth/src/pvrclient-mythtv.cpp  2019-11-19 20:51:27.457826785 +0100
+++ b/addons/pvr.mythtv.cmyth/src/pvrclient-mythtv.cpp  2019-11-19 20:51:27.509826487 +0100
@@ -1194,7 +1194,7 @@
     PVR->TransferTimerEntry(handle, &tag);
     // Add it to memorandom: cf UpdateTimer()
     boost::shared_ptr<PVR_TIMER> pTag = boost::shared_ptr<PVR_TIMER>(new PVR_TIMER(tag));
-    m_PVRtimerMemorandum.insert(std::make_pair(tag.iClientIndex, pTag));
+    m_PVRtimerMemorandum.insert(std::make_pair((unsigned)tag.iClientIndex, pTag));
   }

   if (g_bExtraDebug)
