diff -Naur xf86-video-nv-2.1.16.orig/src/nv_driver.c xf86-video-nv-2.1.16/src/nv_driver.c
--- xf86-video-nv-2.1.16.orig/src/nv_driver.c	2009-12-18 17:22:12.192519522 +0100
+++ xf86-video-nv-2.1.16/src/nv_driver.c	2009-12-18 17:23:33.221270155 +0100
@@ -300,6 +300,8 @@
   { 0x10DE0163, "GeForce 6200 LE" },
   { 0x10DE0164, "GeForce Go 6200" },
   { 0x10DE0165, "Quadro NVS 285" },
+  { 0x10DE053B, "GeForce 7050" },
+  { 0x10DE053E, "GeForce 7025" },
   { 0x10DE0166, "GeForce Go 6400" },
   { 0x10DE0167, "GeForce Go 6200" },
   { 0x10DE0168, "GeForce Go 6400" },
@@ -1717,6 +1719,7 @@
     case 0x0290:   /* GeForce 7900 */
     case 0x0390:   /* GeForce 7600 */
     case 0x0240:   /* GeForce 6100 */
+    case 0x0530:   /* GeForce 7050, 7025 */
     case 0x03D0:
          pNv->Architecture =  NV_ARCH_40;
          break;
diff -Naur xf86-video-nv-2.1.16.orig/src/nv_hw.c xf86-video-nv-2.1.16/src/nv_hw.c
--- xf86-video-nv-2.1.16.orig/src/nv_hw.c	2009-12-18 17:22:12.192519522 +0100
+++ xf86-video-nv-2.1.16/src/nv_hw.c	2009-12-18 17:28:16.131554283 +0100
@@ -890,7 +890,8 @@
         case NV_ARCH_30:
         default:
             if(((pNv->Chipset & 0xfff0) == 0x0240) ||
-               ((pNv->Chipset & 0xfff0) == 0x03D0))
+               ((pNv->Chipset & 0xfff0) == 0x03D0) ||
+               ((pNv->Chipset & 0xfff0) == 0x0530))
             {
                 state->arbitration0 = 128; 
                 state->arbitration1 = 0x0480; 
@@ -1226,6 +1227,7 @@
               case 0x01D0:
               case 0x0240:
               case 0x03D0:
+              case 0x0530:
                  pNv->PMC[0x1700/4] = pNv->PFB[0x020C/4];
                  pNv->PMC[0x1704/4] = 0;
                  pNv->PMC[0x1708/4] = 0;
@@ -1305,7 +1307,8 @@
                  ((pNv->Chipset & 0xfff0) == 0x01D0) ||
                  ((pNv->Chipset & 0xfff0) == 0x0290) ||
                  ((pNv->Chipset & 0xfff0) == 0x0390) ||
-                 ((pNv->Chipset & 0xfff0) == 0x03D0))
+                 ((pNv->Chipset & 0xfff0) != 0x0240) &&
+                 ((pNv->Chipset & 0xfff0) != 0x0530))
               {
                  for(i = 0; i < 60; i++) {
                    pNv->PGRAPH[(0x0D00/4) + i] = pNv->PFB[(0x0600/4) + i];
