diff -Naur xf86-video-nv-2.1.15.orig/src/nv_driver.c xf86-video-nv-2.1.15/src/nv_driver.c
--- xf86-video-nv-2.1.15.orig/src/nv_driver.c	2009-09-26 13:30:48.000000000 +0200
+++ xf86-video-nv-2.1.15/src/nv_driver.c	2009-09-26 13:34:14.000000000 +0200
@@ -420,6 +420,8 @@
   { 0x10DE05E1, "GeForce GTX 280" },
   { 0x10DE05E2, "GeForce GTX 260" },
   { 0x10DE05E3, "GeForce GTX 285" },
+  { 0x10DE053B, "GeForce 7050" },
+  { 0x10DE053E, "GeForce 7025" },
   { 0x10DE05F9, "Quadro CX" },
   { 0x10DE05FD, "Quadro FX 5800" },
   { 0x10DE05FE, "Quadro FX 4800" },
@@ -1688,6 +1690,7 @@
     case 0x0290:   /* GeForce 7900 */
     case 0x0390:   /* GeForce 7600 */
     case 0x0240:   /* GeForce 6100 */
+    case 0x0530:   /* GeForce 7050, 7025 */
     case 0x03D0:
          pNv->Architecture =  NV_ARCH_40;
          break;
diff -Naur xf86-video-nv-2.1.15.orig/src/nv_hw.c xf86-video-nv-2.1.15/src/nv_hw.c
--- xf86-video-nv-2.1.15.orig/src/nv_hw.c	2009-09-26 13:30:48.000000000 +0200
+++ xf86-video-nv-2.1.15/src/nv_hw.c	2009-09-26 13:33:25.000000000 +0200
@@ -890,7 +890,8 @@
         case NV_ARCH_30:
         default:
             if(((pNv->Chipset & 0xfff0) == 0x0240) ||
-               ((pNv->Chipset & 0xfff0) == 0x03D0))
+               ((pNv->Chipset & 0xfff0) == 0x03D0) ||
+               ((pNv->Chipset & 0xfff0) == 0x0530))
             {
                 state->arbitration0 = 128; 
                 state->arbitration1 = 0x0480; 
@@ -1226,6 +1227,7 @@
               case 0x01D0:
               case 0x0240:
               case 0x03D0:
+              case 0x0530:
                  pNv->PMC[0x1700/4] = pNv->PFB[0x020C/4];
                  pNv->PMC[0x1704/4] = 0;
                  pNv->PMC[0x1708/4] = 0;
@@ -1316,7 +1318,8 @@
                    pNv->PGRAPH[(0x0900/4) + i] = pNv->PFB[(0x0600/4) + i];
                    if(((pNv->Chipset & 0xfff0) != 0x0160) &&
                       ((pNv->Chipset & 0xfff0) != 0x0220) &&
-                      ((pNv->Chipset & 0xfff0) != 0x0240))
+                      ((pNv->Chipset & 0xfff0) != 0x0240) &&
+                      ((pNv->Chipset & 0xfff0) != 0x0530))
                    {
                       pNv->PGRAPH[(0x6900/4) + i] = pNv->PFB[(0x0600/4) + i];
                    }
