Index: xserver-xorg-video-nv-2.1.14-2ubuntu1/src/nv_setup.c
===================================================================
--- xserver-xorg-video-nv-2.1.14-2ubuntu1.orig/src/nv_setup.c	2009-07-29 12:52:14.000000000 +0300
+++ xserver-xorg-video-nv-2.1.14-2ubuntu1/src/nv_setup.c	2009-07-29 12:57:15.000000000 +0300
@@ -664,27 +664,14 @@
        }
      
        if(monitorA) {
-           if((monitorA->features.input_type && pNv->FlatPanel) ||
-              (!monitorA->features.input_type && !pNv->FlatPanel))
-           {
-               if(monitorB) { 
-                  xfree(monitorB);
-                  monitorB = NULL;
-               }
-           } else {
-              xfree(monitorA);
-              monitorA = NULL;
-           }
+	   if(monitorB) { 
+	       xfree(monitorB);
+	       monitorB = NULL;
+	   }
        }
 
        if(monitorB) {
-           if((monitorB->features.input_type && !pNv->FlatPanel) ||
-              (!monitorB->features.input_type && pNv->FlatPanel)) 
-           {
-              xfree(monitorB);
-           } else {
-              monitorA = monitorB;
-           }
+           monitorA = monitorB;
            monitorB = NULL;
        }
 
