--- vdr-1.7.18.orig/svdrphosts.conf	2006-04-17 14:19:27.000000000 +0200
+++ vdr-1.7.18/svdrphosts.conf	2011-08-11 19:03:35.865943003 +0200
@@ -7,7 +7,7 @@
 # IP-Address[/Netmask]
 #
 
-127.0.0.1             # always accept localhost
+#127.0.0.1             # always accept localhost
 #192.168.100.0/24     # any host on the local net
 #204.152.189.113      # a specific host
-#0.0.0.0/0            # any host on any net (USE THIS WITH CARE!)
+0.0.0.0/0            # any host on any net (USE THIS WITH CARE!)
