--- a/dvbdevice.c	2018-04-28 13:19:32.315210156 +0200
+++ b/dvbdevice.c	2018-04-28 13:19:32.335210015 +0200
@@ -880,7 +880,7 @@
         CHECK(ioctl(fd_frontend, FE_SET_VOLTAGE, volt));
         CHECK(ioctl(fd_frontend, FE_SET_TONE, tone));
         }
-     frequency = abs(frequency); // Allow for C-band, where the frequency is less than the LOF
+     frequency = abs((int)frequency); // Allow for C-band, where the frequency is less than the LOF
 
      // DVB-S/DVB-S2 (common parts)
      SETCMD(DTV_FREQUENCY, frequency * 1000UL);
--- a/osdbase.c	2018-04-28 13:24:43.461020300 +0200
+++ b/osdbase.c	2018-04-28 13:24:43.493020075 +0200
@@ -512,7 +512,7 @@
       const char *s = item->Text();
       i = 0;
       item_nr = 0;
-      if (s && (s = skipspace(s)) != '\0' && '0' <= s[i] && s[i] <= '9') {
+      if (s && (s = skipspace(s)) != NULL && '0' <= s[i] && s[i] <= '9') {
          do {
             item_nr = item_nr * 10 + (s[i] - '0');
             }
--- a/remux.c	2018-04-28 13:29:23.935049632 +0200
+++ b/remux.c	2018-04-28 13:29:23.959049464 +0200
@@ -1641,11 +1641,11 @@
                        uint32_t Delta = ptsValues[0] / Div;
                        // determine frame info:
                        if (isVideo) {
-                          if (abs(Delta - 3600) <= 1)
+                          if (abs((int)(Delta - 3600)) <= 1)
                              framesPerSecond = 25.0;
                           else if (Delta % 3003 == 0)
                              framesPerSecond = 30.0 / 1.001;
-                          else if (abs(Delta - 1800) <= 1)
+                          else if (abs((int)(Delta - 1800)) <= 1)
                              framesPerSecond = 50.0;
                           else if (Delta == 1501)
                              framesPerSecond = 60.0 / 1.001;
