--- vdr-1.7.29/dvbsubtitle.c.org	2012-09-03 17:04:59.578348051 +0200
+++ vdr-1.7.29/dvbsubtitle.c	2012-09-03 17:05:23.122348458 +0200
@@ -1043,6 +1043,7 @@
             page->SetVersion(pageVersion);
             page->SetTimeout(pageTimeout);
             page->SetState(bs.GetBits(2));
+	    page->regions.Clear();
             bs.SkipBits(2); // reserved
             dbgpages("Update page id %d version %d pts %"PRId64" timeout %d state %d\n", pageId, page->Version(), page->Pts(), page->Timeout(), page->State());
             while (!bs.IsEOF()) {
