--- a/src/core/shutdown.c	2014-04-29 19:32:34.163841249 +0200
+++ b/src/core/shutdown.c	2014-04-29 19:32:44.823840898 +0200
@@ -27,6 +27,7 @@
 #include <sys/stat.h>
 #include <sys/mount.h>
 #include <sys/syscall.h>
+#include <sys/socket.h>
 #include <fcntl.h>
 #include <dirent.h>
 #include <errno.h>
--- a/src/test/test-udev.c	2014-02-13 03:42:34.733681325 +0100
+++ b/src/test/test-udev.c	2014-04-29 19:39:33.299827477 +0200
@@ -31,6 +31,7 @@
 #include <sched.h>
 #include <sys/mount.h>
 #include <sys/signalfd.h>
+#include <sys/socket.h>
 
 #include "missing.h"
 #include "udev.h"
--- a/src/readahead/readahead-collect.c	2014-02-13 03:42:33.819685017 +0100
+++ b/src/readahead/readahead-collect.c	2014-04-29 20:31:49.235724440 +0200
@@ -42,6 +42,7 @@
 #include <sys/vfs.h>
 #include <getopt.h>
 #include <sys/inotify.h>
+#include <sys/socket.h>
 #include <math.h>
 
 #ifdef HAVE_LINUX_BTRFS_H
--- a/src/readahead/readahead-replay.c	2014-02-13 03:42:33.793685122 +0100
+++ b/src/readahead/readahead-replay.c	2014-04-29 20:49:32.571689502 +0200
@@ -34,6 +34,7 @@
 #include <unistd.h>
 #include <getopt.h>
 #include <sys/inotify.h>
+#include <sys/socket.h>
 
 #include <systemd/sd-daemon.h>
 
