From 6149b974ab71e45649fedc83aa6cab04ee532c27 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 12 May 2011 11:38:17 +0200
Subject: [PATCH 007/119] omxcamerabinsrc: forward the zoom property to
 omx_camera

---
 gst/camerabin2/gstomxcamerabinsrc.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/gst/camerabin2/gstomxcamerabinsrc.c b/gst/camerabin2/gstomxcamerabinsrc.c
index 1fb3038..d90ee1c 100644
--- a/gst/camerabin2/gstomxcamerabinsrc.c
+++ b/gst/camerabin2/gstomxcamerabinsrc.c
@@ -505,10 +505,15 @@ static void
 gst_omx_camera_bin_src_set_zoom (GstBaseCameraBinSrc * bcamsrc, gfloat zoom)
 {
   GstOmxCameraBinSrc *self = GST_OMX_CAMERA_BIN_SRC (bcamsrc);
+  gint omx_zoom;
 
   GST_INFO_OBJECT (self, "setting zoom %f", zoom);
 
-//  g_object_set (G_OBJECT (self->video_source), "zoom", (gint) (zoom * 100), NULL);
+  omx_zoom = zoom * 80;
+  if (omx_zoom < 100)
+    omx_zoom = 100;
+
+  g_object_set (G_OBJECT (self->video_source), "zoom", omx_zoom, NULL);
 }
 
 static GstCaps *
-- 
1.7.5.4

