From a7abcb59e109ee744c6ced7df201fef6b79ace00 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20D=C3=ADaz?= <yosoy@danieldiaz.org>
Date: Thu, 17 Dec 2009 17:24:37 -0600
Subject: [PATCH 010/119] freeze: Add timeout property.

---
 gst/freeze/gstfreeze.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/gst/freeze/gstfreeze.c b/gst/freeze/gstfreeze.c
index 90e3ea3..1885d8c 100644
--- a/gst/freeze/gstfreeze.c
+++ b/gst/freeze/gstfreeze.c
@@ -122,6 +122,12 @@ gst_freeze_class_init (GstFreezeClass * klass)
           "timeout",
           "Timeout before closing stream", 0, G_MAXINT, 1, G_PARAM_READWRITE));
 
+  g_object_class_install_property (object_class,
+      ARG_TIMEOUT,
+      g_param_spec_int ("timeout",
+          "timeout",
+          "Timeout before closing stream", 0, G_MAXINT, 1, G_PARAM_READWRITE));
+
   object_class->dispose = gst_freeze_dispose;
 
 }
-- 
1.7.5.4

