From 4c64527c787570d6ca2989c7641ff45b859f940c Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Thu, 12 May 2011 11:56:52 +0200
Subject: [PATCH 014/119] h264parse: fixup boilerplate and debug code style

---
 gst/videoparsers/gsth264parse.c |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index 8b996fe..4d8e1e1 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -68,7 +68,8 @@ static GstStaticPadTemplate srctemplate = GST_STATIC_PAD_TEMPLATE ("src",
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS ("video/x-h264, parsed = (boolean) true"));
 
-GST_BOILERPLATE (GstH264Parse, gst_h264_parse, GstElement, GST_TYPE_BASE_PARSE);
+GST_BOILERPLATE (GstH264Parse, gst_h264_parse, GstBaseParse,
+    GST_TYPE_BASE_PARSE);
 
 static void gst_h264_parse_finalize (GObject * object);
 
@@ -203,7 +204,7 @@ gst_h264_parse_start (GstBaseParse * parse)
 {
   GstH264Parse *h264parse = GST_H264_PARSE (parse);
 
-  GST_DEBUG ("Start");
+  GST_DEBUG_OBJECT (parse, "start");
   gst_h264_parse_reset (h264parse);
 
   gst_h264_params_create (&h264parse->params, GST_ELEMENT (h264parse));
@@ -218,7 +219,7 @@ gst_h264_parse_stop (GstBaseParse * parse)
 {
   GstH264Parse *h264parse = GST_H264_PARSE (parse);
 
-  GST_DEBUG ("Stop");
+  GST_DEBUG_OBJECT (parse, "stop");
   gst_h264_parse_reset (h264parse);
 
   gst_h264_params_free (h264parse->params);
-- 
1.7.5.4

