From 93f4377f4ae243fac93c9c783626702e4131b7c3 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Thu, 12 May 2011 11:57:19 +0200
Subject: [PATCH 015/119] h264parse: fix some buffer leaks

Fixes #650323.
---
 gst/videoparsers/gsth264parse.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index 4d8e1e1..5e1d0c4 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -746,6 +746,7 @@ gst_h264_parse_parse_frame (GstBaseParse * parse, GstBaseParseFrame * frame)
     buf = gst_adapter_take_buffer (h264parse->frame_out, av);
     gst_buffer_copy_metadata (buf, buffer, GST_BUFFER_COPY_ALL);
     gst_buffer_replace (&frame->buffer, buf);
+    gst_buffer_unref (buf);
   }
 
   return GST_FLOW_OK;
@@ -869,6 +870,7 @@ gst_h264_parse_pre_push_frame (GstBaseParse * parse, GstBaseParseFrame * frame)
           new_buf = gst_byte_writer_reset_and_get_buffer (&bw);
           gst_buffer_copy_metadata (new_buf, buffer, GST_BUFFER_COPY_ALL);
           gst_buffer_replace (&frame->buffer, new_buf);
+          gst_buffer_unref (new_buf);
         }
       }
       /* we pushed whatever we had */
-- 
1.7.5.4

