From b49f36cc0e3057e7ddacd4b93c6e5435ffda5f17 Mon Sep 17 00:00:00 2001
From: David Schleef <ds@schleef.org>
Date: Wed, 18 May 2011 13:35:34 -0700
Subject: [PATCH 016/119] h264parse: max pps is 255

Fixes #650484.
---
 gst/h264parse/gsth264parse.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/h264parse/gsth264parse.h b/gst/h264parse/gsth264parse.h
index e245ea0..e2fae84 100644
--- a/gst/h264parse/gsth264parse.h
+++ b/gst/h264parse/gsth264parse.h
@@ -48,7 +48,7 @@ typedef struct _GstH264Sps GstH264Sps;
 typedef struct _GstH264Pps GstH264Pps;
 
 #define MAX_SPS_COUNT	32
-#define MAX_PPS_COUNT   32
+#define MAX_PPS_COUNT   256
 
 #define CLOCK_BASE 9LL
 #define CLOCK_FREQ (CLOCK_BASE * 10000)
-- 
1.7.5.4

