From 0f52d8130b9e4312ec65d4741b34460be3af03b0 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Tue, 17 May 2011 22:42:29 +0200
Subject: [PATCH 018/119] h264parse: simplify minimum frame size handling

---
 gst/videoparsers/gsth264parse.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index bc615c7..f9b6187 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -209,7 +209,7 @@ gst_h264_parse_start (GstBaseParse * parse)
 
   gst_h264_params_create (&h264parse->params, GST_ELEMENT (h264parse));
 
-  gst_base_parse_set_min_frame_size (parse, 512);
+  gst_base_parse_set_min_frame_size (parse, 6);
 
   return TRUE;
 }
@@ -565,8 +565,8 @@ gst_h264_parse_check_valid_frame (GstBaseParse * parse,
   return TRUE;
 
 more:
-  /* Ask for 1024 bytes more - this is an arbitrary choice */
-  gst_base_parse_set_min_frame_size (parse, GST_BUFFER_SIZE (buffer) + 1024);
+  /* ask for best next available */
+  *framesize = G_MAXUINT;
 
   /* skip up to initial startcode */
   *skipsize = sc_pos;
-- 
1.7.5.4

