From 26141adb7933db5685790089082798856b90f32b Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Tue, 21 Jun 2011 12:41:47 +0200
Subject: [PATCH 026/119] h264parse: avoid bogus frame parsing state

---
 gst/videoparsers/gsth264parse.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index d5b539e..f8b403e 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -507,6 +507,8 @@ gst_h264_parse_check_valid_frame (GstBaseParse * parse,
     if (sc_pos == -1) {
       /* SC not found, need more data */
       sc_pos = GST_BUFFER_SIZE (buffer) - 3;
+      /* avoid going < 0 later on */
+      nal_pos = next_sc_pos = sc_pos;
       goto more;
     }
 
-- 
1.7.5.4

