From 908a2b596144d87d3b7c8266bff170c78c1a34c0 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Wed, 6 Jul 2011 08:14:55 +0200
Subject: [PATCH 028/119] h264parse: bump rank to PRIMARY + 1

---
 gst/videoparsers/plugin.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/videoparsers/plugin.c b/gst/videoparsers/plugin.c
index cfd8a48..eec5013 100644
--- a/gst/videoparsers/plugin.c
+++ b/gst/videoparsers/plugin.c
@@ -34,7 +34,7 @@ plugin_init (GstPlugin * plugin)
   ret = gst_element_register (plugin, "h263parse",
       GST_RANK_NONE, GST_TYPE_H263_PARSE);
   ret = gst_element_register (plugin, "h264parse",
-      GST_RANK_NONE, GST_TYPE_H264_PARSE);
+      GST_RANK_PRIMARY + 1, GST_TYPE_H264_PARSE);
   ret = gst_element_register (plugin, "diracparse",
       GST_RANK_NONE, GST_TYPE_DIRAC_PARSE);
 
-- 
1.7.5.4

