From a4f4328d0e2b8c999b8d7e0c8d31d7afc5f13024 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Tue, 12 Jul 2011 19:04:27 +0200
Subject: [PATCH 032/119] h264parse: remove a check made useless by the last
 commit

---
 gst/videoparsers/gsth264parse.c |   10 ----------
 1 files changed, 0 insertions(+), 10 deletions(-)

diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index b374b72..def1ce5 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -1040,16 +1040,6 @@ gst_h264_parse_set_caps (GstBaseParse * parse, GstCaps * caps)
   /* negotiate with downstream, sets ->format and ->align */
   gst_h264_parse_negotiate (h264parse);
 
-  /* if upstream sets codec_data without setting stream-format and alignment, we
-   * assume stream-format=avc,alignment=au */
-  if (format == GST_H264_PARSE_FORMAT_NONE) {
-    if (codec_data == NULL)
-      goto unknown_input_format;
-
-    format = GST_H264_PARSE_FORMAT_AVC;
-    align = GST_H264_PARSE_ALIGN_AU;
-  }
-
   if (format == h264parse->format && align == h264parse->align) {
     gst_base_parse_set_passthrough (parse, TRUE);
 
-- 
1.7.5.4

