From 78072bd58cda4d169835836eaeb8bcd8b6fca037 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Mon, 23 May 2011 12:02:00 +0200
Subject: [PATCH 036/119] mpegvideoparse: rename to legacympegvideoparse

---
 gst/mpegvideoparse/mpegvideoparse.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst/mpegvideoparse/mpegvideoparse.c b/gst/mpegvideoparse/mpegvideoparse.c
index 70fdc89..92e38a1 100644
--- a/gst/mpegvideoparse/mpegvideoparse.c
+++ b/gst/mpegvideoparse/mpegvideoparse.c
@@ -1022,10 +1022,10 @@ gst_mpegvideoparse_change_state (GstElement * element,
 static gboolean
 plugin_init (GstPlugin * plugin)
 {
-  GST_DEBUG_CATEGORY_INIT (mpv_parse_debug, "mpegvideoparse", 0,
+  GST_DEBUG_CATEGORY_INIT (mpv_parse_debug, "legacympegvideoparse", 0,
       "MPEG Video Parser");
 
-  return gst_element_register (plugin, "mpegvideoparse",
+  return gst_element_register (plugin, "legacympegvideoparse",
       GST_RANK_PRIMARY, GST_TYPE_MPEGVIDEOPARSE);
 }
 
-- 
1.7.5.4

