From ef59c5985483d0d299af8a9151b51ddcc8780e6c Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 4 Aug 2011 13:46:59 +0200
Subject: [PATCH 043/119] mpeg4videoparse: bump rank to PRIMARY + 1

---
 gst/mpeg4videoparse/mpeg4videoparse.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/mpeg4videoparse/mpeg4videoparse.c b/gst/mpeg4videoparse/mpeg4videoparse.c
index 3040b61..a63b03f 100644
--- a/gst/mpeg4videoparse/mpeg4videoparse.c
+++ b/gst/mpeg4videoparse/mpeg4videoparse.c
@@ -611,7 +611,7 @@ plugin_init (GstPlugin * plugin)
   GST_DEBUG_CATEGORY_INIT (mpeg4v_parse_debug, "mpeg4videoparse", 0,
       "MPEG-4 video parser");
 
-  if (!gst_element_register (plugin, "mpeg4videoparse", GST_RANK_SECONDARY,
+  if (!gst_element_register (plugin, "mpeg4videoparse", GST_RANK_PRIMARY + 1,
           gst_mpeg4vparse_get_type ()))
     return FALSE;
 
-- 
1.7.5.4

