From 27d44cf0ef5906e526c1bb097e10519e336e420e Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 4 Aug 2011 16:35:06 +0200
Subject: [PATCH 044/119] omxcamerabinsrc: make recordings start at
 running_time=0

---
 gst/camerabin2/gstomxcamerabinsrc.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/gst/camerabin2/gstomxcamerabinsrc.c b/gst/camerabin2/gstomxcamerabinsrc.c
index 930ce57..f572771 100644
--- a/gst/camerabin2/gstomxcamerabinsrc.c
+++ b/gst/camerabin2/gstomxcamerabinsrc.c
@@ -214,9 +214,17 @@ gst_omx_camera_bin_src_vidsrc_probe (GstPad * pad, GstBuffer * buffer,
   if (self->video_rec_status == GST_VIDEO_RECORDING_STATUS_DONE) {
     /* NOP */
   } else if (self->video_rec_status == GST_VIDEO_RECORDING_STATUS_STARTING) {
+    GstClockTime timestamp;
+
     GST_DEBUG_OBJECT (self, "Starting video recording");
     self->video_rec_status = GST_VIDEO_RECORDING_STATUS_RUNNING;
 
+    timestamp = GST_BUFFER_TIMESTAMP (buffer);
+    if (!GST_CLOCK_TIME_IS_VALID (timestamp))
+      timestamp = 0;
+    gst_pad_push_event (pad, gst_event_new_new_segment (FALSE, 1.0,
+            GST_FORMAT_TIME, timestamp, -1, 0));
+
     /* post preview */
     GST_DEBUG_OBJECT (self, "Posting preview for video");
     gst_base_camera_src_post_preview (camerasrc, buffer);
-- 
1.7.5.4

