From eab81f72b716d228ec60f5fac99acd07de54c102 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Tue, 13 Sep 2011 00:38:04 -0300
Subject: [PATCH 060/119] pvr: Swap drawable after the blit

---
 sys/pvr2d/gstpvrvideosink.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 688fb21..df6c5c7 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -315,6 +315,7 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
       return;
       break;
   }
+  dcontext->wsegl_table->pfnWSEGL_SwapDrawable (dcontext->drawable_handle, 1);
   GST_DEBUG_OBJECT (pvrvideosink, "end");
 }
 
-- 
1.7.5.4

