From d7fb852aa7157ab0c158fc2901c342256d5f7e00 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Tue, 13 Sep 2011 00:38:34 -0300
Subject: [PATCH 061/119] pvr: bufferpool workaround

---
 sys/pvr2d/gstpvrvideosink.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index df6c5c7..550a233 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -650,7 +650,7 @@ gst_pvrvideosink_buffer_alloc (GstBaseSink * bsink, guint64 offset, guint size,
 
     if (!(pvrvideosink->buffer_pool =
             gst_pvr_bufferpool_new (GST_ELEMENT (pvrvideosink),
-                caps, pvrvideosink->num_buffers, size))) {
+                caps, 8, size))) {
       return GST_FLOW_ERROR;
     }
     /* Notify application to set xwindow id now */
-- 
1.7.5.4

