From c77e969356c28ebe22dd3f8f40c57f631a7fa523 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Tue, 13 Sep 2011 16:25:05 +0200
Subject: [PATCH 065/119] pvrsink: gstducati.h => gstpvr.h in Makefile.am

---
 sys/pvr2d/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/Makefile.am b/sys/pvr2d/Makefile.am
index 9823282..2927870 100644
--- a/sys/pvr2d/Makefile.am
+++ b/sys/pvr2d/Makefile.am
@@ -17,6 +17,6 @@ libgstpvr_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
 libgstpvr_la_LIBTOOLFLAGS = --tag=disable-static
 
 noinst_HEADERS = \
-	gstducati.h \
+	gstpvr.h \
 	gstpvrbufferpool.h \
 	gstpvrvideosink.h
-- 
1.7.5.4

