From 61d245c2fcbcf7ae5c64b173f661ebcbd5377b2f Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Tue, 13 Sep 2011 17:09:11 +0200
Subject: [PATCH 066/119] pvrvideosink: don't crash if PVR2DEnumerateDevices
 fails

---
 sys/pvr2d/gstpvrvideosink.c |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index e3ec4dd..86b49d9 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -95,14 +95,15 @@ gst_pvrvideosink_get_dcontext (GstPVRVideoSink * pvrvideosink)
 
   dcontext->x_display = XOpenDisplay (NULL);
 
-  num_devices = PVR2DEnumerateDevices (NULL);
+  pvr_error = num_devices = PVR2DEnumerateDevices (NULL);
+  if (num_devices < 0)
+    goto enumerate_devices_error;
+
   p_dev_info = g_new0 (PVR2DDEVICEINFO, num_devices);
   pvr_error = PVR2DEnumerateDevices (p_dev_info);
-  if (pvr_error != PVR2D_OK) {
-    GST_ERROR_OBJECT (pvrvideosink, "Failed to enumerate devices "
-        "returned %d", pvr_error);
-    return NULL;
-  }
+  if (pvr_error != PVR2D_OK)
+    goto enumerate_devices_error;
+
   pvr_error = PVR2DCreateDeviceContext (p_dev_info[0].ulDevID,
       &(dcontext->pvr_context), 0);
   switch (pvr_error) {
@@ -136,6 +137,11 @@ gst_pvrvideosink_get_dcontext (GstPVRVideoSink * pvrvideosink)
   dcontext->p_blt_info->DstStride = dcontext->stride;
   dcontext->p_blt_info->DstFormat = dcontext->display_format;
   return dcontext;
+
+enumerate_devices_error:
+  GST_ERROR_OBJECT (pvrvideosink, "Failed to enumerate devices "
+      "returned %d", pvr_error);
+  return NULL;
 }
 
 static GstXWindow *
-- 
1.7.5.4

