From 868e3c539bf14c17091262308fd058d142bd4186 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Wed, 14 Sep 2011 12:31:10 +0200
Subject: [PATCH 067/119] pvrvideosink: reset GST_BUFFER_SIZE on revived
 buffers

---
 sys/pvr2d/gstpvrbufferpool.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrbufferpool.c b/sys/pvr2d/gstpvrbufferpool.c
index 9dad41b..35ebf9e 100644
--- a/sys/pvr2d/gstpvrbufferpool.c
+++ b/sys/pvr2d/gstpvrbufferpool.c
@@ -223,6 +223,7 @@ gst_pvr_bufferpool_get (GstPvrBufferPool * self, GstBuffer * orig)
     GST_BUFFER_TIMESTAMP (buf) = GST_BUFFER_TIMESTAMP (orig);
     GST_BUFFER_DURATION (buf) = GST_BUFFER_DURATION (orig);
   }
+  GST_BUFFER_SIZE (buf) = self->size;
 
   return buf;
 }
-- 
1.7.5.4

