From 92386199b5c1166947cbc0d55abdbdf3011425cb Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Wed, 14 Sep 2011 18:11:10 +0200
Subject: [PATCH 070/119] pvrvideosink: make the pool log under the
 pvrvideosink category

---
 sys/pvr2d/gstpvrbufferpool.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrbufferpool.c b/sys/pvr2d/gstpvrbufferpool.c
index 35ebf9e..af996d5 100644
--- a/sys/pvr2d/gstpvrbufferpool.c
+++ b/sys/pvr2d/gstpvrbufferpool.c
@@ -19,6 +19,9 @@
 
 #include "gstpvrbufferpool.h"
 
+GST_DEBUG_CATEGORY_EXTERN (gst_debug_pvrvideosink);
+#define GST_CAT_DEFAULT gst_debug_pvrvideosink
+
 /*
  * GstDucatiBuffer
  */
-- 
1.7.5.4

