From 5ca9579f77186ffe14cdd083f8659d1b47c1d6d0 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Wed, 14 Sep 2011 11:49:44 -0300
Subject: [PATCH 072/119] pvr: Space changes

---
 sys/pvr2d/gstpvrvideosink.c |   36 ++++++++++++++++++------------------
 1 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 44bc0d1..b073aa5 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -40,23 +40,23 @@ GST_DEBUG_CATEGORY_EXTERN (gst_debug_pvrvideosink);
 
 typedef struct DRI2WS_Display_TAG
 {
-	unsigned int			ui32RefCount;
-	
-	Display				*display;
-	int				screen;
-	unsigned int			ui32Flags;
-
-	unsigned int			ui32Width;
-	unsigned int			ui32Height;
-	unsigned int			ui32StrideInBytes;
-	unsigned int			ui32BytesPerPixel;
-	WSEGLPixelFormat		ePixelFormat;
-
-	PVR2DFORMAT				ePVR2DPixelFormat;
-	PVR2DCONTEXTHANDLE		hContext;
-	PVR2DMEMINFO			*psMemInfo;
-
-	int				iDRMfd;
+  unsigned int ui32RefCount;
+
+  Display *display;
+  int screen;
+  unsigned int ui32Flags;
+
+  unsigned int ui32Width;
+  unsigned int ui32Height;
+  unsigned int ui32StrideInBytes;
+  unsigned int ui32BytesPerPixel;
+  WSEGLPixelFormat ePixelFormat;
+
+  PVR2DFORMAT ePVR2DPixelFormat;
+  PVR2DCONTEXTHANDLE hContext;
+  PVR2DMEMINFO *psMemInfo;
+
+  int iDRMfd;
 } DRI2WSDisplay;
 
 static void gst_pvrvideosink_reset (GstPVRVideoSink * pvrvideosink);
@@ -68,7 +68,7 @@ GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS (GST_VIDEO_CAPS_BGRx)
-);
+    );
 
 enum
 {
-- 
1.7.5.4

