From 46023bd536e7804cece31fe262f1f5d3f369cfa0 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Tue, 13 Sep 2011 14:02:34 -0300
Subject: [PATCH 073/119] pvr: Unwrap memory also if there is an error

---
 sys/pvr2d/gstpvrvideosink.c |   18 ++++++++++--------
 1 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index b073aa5..6d9875c 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -249,7 +249,7 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
   ret &= gst_structure_get_int (structure, "height", &video_height);
   if (!ret) {
     GST_ERROR_OBJECT (pvrvideosink, "Failed to get dimensions of the buffer");
-    return;
+    goto done;
   }
 
   pvr_error = PVR2DMemWrap (dcontext->pvr_context, data, 0, size, NULL,
@@ -257,7 +257,7 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
   if (pvr_error != PVR2D_OK) {
     GST_ERROR_OBJECT (pvrvideosink, "Failed to Wrap buffer memory"
         "returned %d", pvr_error);
-    return;
+    goto done;
   }
 
   if (video_width > dcontext->display_width)
@@ -293,33 +293,35 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
       break;
     case PVR2DERROR_DEVICE_UNAVAILABLE:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, device unavailable");
-      return;
+      goto done;
       break;
     case PVR2DERROR_INVALID_CONTEXT:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, invalid context");
-      return;
+      goto done;
       break;
     case PVR2DERROR_INVALID_PARAMETER:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, invalid parameter");
-      return;
+      goto done;
       break;
     case PVR2DERROR_HW_FEATURE_NOT_SUPPORTED:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, "
           "hardware feature not supported");
-      return;
+      goto done;
       break;
     case PVR2DERROR_GENERIC_ERROR:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, generic error");
-      return;
+      goto done;
       break;
     default:
       GST_ERROR_OBJECT (pvrvideosink, "Failed to blit, "
           "undefined error %d", pvr_error);
-      return;
+      goto done;
       break;
   }
   dcontext->wsegl_table->pfnWSEGL_SwapDrawable (dcontext->drawable_handle, 1);
   PVR2DQueryBlitsComplete (dcontext->pvr_context, dcontext->dst_mem, 1);
+
+done:
   pvr_error = PVR2DMemFree (dcontext->pvr_context, dcontext->src_mem);
   if (pvr_error != PVR2D_OK) {
     GST_ERROR_OBJECT (pvrvideosink, "Failed to Unwrap buffer memory"
-- 
1.7.5.4

