From 40e52a2544da2f9ed0e9777f3306b09bcb8c524c Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <luciana.fujii@collabora.co.uk>
Date: Tue, 27 Sep 2011 17:00:32 -0300
Subject: [PATCH 097/119] pvrvideosink: Remove unused variables

---
 sys/pvr2d/gstpvrvideosink.c |    9 ---------
 1 files changed, 0 insertions(+), 9 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 368ed34..b99569b 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -1080,7 +1080,6 @@ gst_pvrvideosink_buffer_alloc (GstBaseSink * bsink, guint64 offset, guint size,
   GstStructure *structure = NULL;
   GstFlowReturn ret = GST_FLOW_OK;
   gint width, height;
-  GstVideoRectangle src;
 
   pvrvideosink = GST_PVRVIDEOSINK (bsink);
 
@@ -1108,9 +1107,6 @@ gst_pvrvideosink_buffer_alloc (GstBaseSink * bsink, guint64 offset, guint size,
     goto beach;
   }
 
-  src.w = width;
-  src.h = height;
-
   /* initialize the buffer pool if not initialized yet (first buffer): */
   if (G_UNLIKELY (!pvrvideosink->buffer_pool)) {
     /* workaround for bug in omap_vout driver, when we ask for more
@@ -1138,9 +1134,6 @@ gst_pvrvideosink_buffer_alloc (GstBaseSink * bsink, guint64 offset, guint size,
       goto beach;
     }
 
-    src.w = width;
-    src.h = height;
-
     GST_LOG_OBJECT (pvrvideosink, "Creating a buffer pool with %d buffers",
         pvrvideosink->num_buffers);
 
@@ -1518,12 +1511,10 @@ gst_pvrvideosink_class_init (GstPVRVideoSinkClass * klass)
   GObjectClass *gobject_class;
   GstElementClass *gstelement_class;
   GstBaseSinkClass *gstbasesink_class;
-  GstVideoSinkClass *videosink_class;
 
   gobject_class = (GObjectClass *) klass;
   gstelement_class = (GstElementClass *) klass;
   gstbasesink_class = (GstBaseSinkClass *) klass;
-  videosink_class = (GstVideoSinkClass *) klass;
 
   parent_class = g_type_class_peek_parent (klass);
 
-- 
1.7.5.4

