From 565813abf82d16f2f96c0ef905dcccce79c0fa70 Mon Sep 17 00:00:00 2001
From: Luciana Fujii Pontello <lucianafujii@collabora.co.uk>
Date: Wed, 19 Oct 2011 19:39:14 -0400
Subject: [PATCH 106/119] pvrvideosink: Remove unused variables

---
 sys/pvr2d/gstpvrvideosink.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 18a1d55..bca1e3b 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -481,8 +481,6 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
 {
   PVR2DERROR pvr_error;
   GstDrawContext *dcontext = pvrvideosink->dcontext;
-  void *data;
-  guint size;
   GstCaps *caps;
   GstStructure *structure;
   gint video_width;
@@ -506,8 +504,6 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
     return;
   }
 
-  data = GST_BUFFER_DATA (buffer);
-  size = GST_BUFFER_SIZE (buffer);
   caps = GST_BUFFER_CAPS (buffer);
   src_mem = gst_ducati_buffer_get_meminfo ((GstDucatiBuffer *) buffer);
   p_blt_3d = dcontext->p_blt_info;
-- 
1.7.5.4

