From cc066f061ce5949f18f3d8ccd0073c4ee45e88a7 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 27 Oct 2011 13:13:43 +0200
Subject: [PATCH 114/119] pvrvideosink: link explicitly to pvr2d

---
 sys/pvr2d/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sys/pvr2d/Makefile.am b/sys/pvr2d/Makefile.am
index 2927870..b450710 100644
--- a/sys/pvr2d/Makefile.am
+++ b/sys/pvr2d/Makefile.am
@@ -9,7 +9,7 @@ libgstpvr_la_CFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_CFLAGS) $(PVR_CFLAGS)
 
 libgstpvr_la_LIBADD = $(GST_LIBS) $(GST_BASE_LIBS) \
 	$(GST_PLUGINS_BASE_LIBS) $(X11_LIBS) -lgstvideo-$(GST_MAJORMINOR) \
-	-lgstinterfaces-$(GST_MAJORMINOR) -lpvrPVR2D_DRIWSEGL\
+	-lgstinterfaces-$(GST_MAJORMINOR) -lpvr2d -lpvrPVR2D_DRIWSEGL\
 	$(PVR_LIBS) \
 	$(LIBM)
 
-- 
1.7.5.4

