From f00064efd79dd2895a77aaf1a2729aa7921d79b1 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 26 Dec 2011 21:57:48 -0600
Subject: [PATCH 140/154] videoparsers: set interlaced field in caps for mpeg2

---
 gst/videoparsers/gstmpegvideoparse.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/gst/videoparsers/gstmpegvideoparse.c b/gst/videoparsers/gstmpegvideoparse.c
index 458f699..eefbeab 100644
--- a/gst/videoparsers/gstmpegvideoparse.c
+++ b/gst/videoparsers/gstmpegvideoparse.c
@@ -598,6 +598,9 @@ gst_mpegv_parse_update_src_caps (GstMpegvParse * mpvparse)
       gst_caps_set_simple (caps, "level", G_TYPE_STRING, level, NULL);
     else
       GST_DEBUG_OBJECT (mpvparse, "Invalid level - %u", level_c);
+
+    gst_caps_set_simple (caps, "interlaced",
+        G_TYPE_BOOLEAN, !mpvparse->params.progressive, NULL);
   }
 
   gst_pad_set_caps (GST_BASE_PARSE_SRC_PAD (mpvparse), caps);
-- 
1.7.5.4

