From f5e84e6dd525e3af09a1d41107ba826e28946d5a Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Sat, 31 Dec 2011 15:13:11 -0600
Subject: [PATCH 148/154] dvdspu: use refcnting instead of explicit copy

---
 gst/dvdspu/gstdvdspu.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst/dvdspu/gstdvdspu.c b/gst/dvdspu/gstdvdspu.c
index c1b97f3..fa68b24 100644
--- a/gst/dvdspu/gstdvdspu.c
+++ b/gst/dvdspu/gstdvdspu.c
@@ -563,7 +563,7 @@ dvdspu_handle_vid_buffer (GstDVDSpu * dvdspu, GstBuffer * buf)
       goto no_ref_frame;
     }
 
-    buf = gst_buffer_copy (dvdspu->ref_frame);
+    buf = gst_buffer_ref (dvdspu->ref_frame);
 
 #if 0
     g_print ("Duping frame %" GST_TIME_FORMAT " with new TS %" GST_TIME_FORMAT
@@ -601,7 +601,7 @@ dvdspu_handle_vid_buffer (GstDVDSpu * dvdspu, GstBuffer * buf)
 
       /* Take a copy in case we hit a still frame and need the pristine 
        * frame around */
-      copy = gst_buffer_copy (buf);
+      copy = gst_buffer_ref (buf);
       gst_buffer_replace (&dvdspu->ref_frame, copy);
       gst_buffer_unref (copy);
     }
-- 
1.7.5.4

