From 1f3e439e981adc6cb340c215bcdd4dfb46cf4ee4 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 2 Jan 2012 11:12:24 -0600
Subject: [PATCH 152/154] pvrvideosink: fix NPE when closing window

---
 sys/pvr2d/gstpvrvideosink.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 440944a..bf2926a 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -720,6 +720,10 @@ gst_pvrvideosink_blit (GstPVRVideoSink * pvrvideosink, GstBuffer * buffer)
     draw_border = TRUE;
   }
 
+  if (!pvrvideosink->xwindow) {
+    goto done;
+  }
+
   /* Sometimes the application hasn't really given us valid dimensions
    * when we want to render the first frame, which throws pvr into a
    * tizzy, so let's just detect it and bail early:
-- 
1.7.5.4

