--- a/Makefile	2015-11-07 20:17:20.141292298 +0100
+++ b/Makefile	2015-11-07 20:17:20.149292298 +0100
@@ -1,6 +1,6 @@
 
 ply-image: ply-image.c ply-frame-buffer.c Makefile
-	gcc $(CFLAGS) `pkg-config --cflags libpng`  ply-image.c ply-frame-buffer.c -o ply-image  -lm `pkg-config --libs libpng`
+	$(CC) $(CFLAGS) `pkg-config --cflags libpng`  ply-image.c ply-frame-buffer.c -o ply-image  -lm `pkg-config --libs libpng`
 	
 clean:
 	rm -f ply-image *~ gmon.out
