--- a/ply-frame-buffer.c	2015-11-07 20:22:47.513305198 +0100
+++ b/ply-frame-buffer.c	2015-11-07 20:22:47.537305199 +0100
@@ -145,7 +145,6 @@
   x2 = x1 + buffer->area_to_flush.width;
   y2 = y1 + buffer->area_to_flush.height;
 
-  bytes_per_row = buffer->area_to_flush.width * buffer->bytes_per_pixel;
   row_buffer = malloc (buffer->row_stride * buffer->bytes_per_pixel);
   for (row = y1; row < y2; row++)
     {
