diff -Naur a/ply-image.c b/ply-image.c
--- a/ply-image.c	2011-07-17 14:19:54.000000000 +0200
+++ b/ply-image.c	2011-07-17 14:21:35.737894946 +0200
@@ -221,7 +221,7 @@
     png_set_palette_to_rgb (png);
 
   if ((color_type == PNG_COLOR_TYPE_GRAY) && (bits_per_pixel < 8))
-    png_set_gray_1_2_4_to_8 (png);
+    png_set_expand_gray_1_2_4_to_8 (png);
 
   if (png_get_valid (png, info, PNG_INFO_tRNS))
     png_set_tRNS_to_alpha (png);
