--- a/driver/os_dep/linux/rtw_android.c	2014-12-30 08:04:23.000000000 +0100
+++ b/driver/os_dep/linux/rtw_android.c	2015-05-25 09:07:58.803013247 +0200
@@ -81,6 +81,10 @@
 #define PNO_TLV_FREQ_REPEAT		'R'
 #define PNO_TLV_FREQ_EXPO_MAX		'M'
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,0,0)
+#define strncasecmp strnicmp
+#endif
+
 typedef struct cmd_tlv {
 	char prefix;
 	char version;
@@ -224,7 +228,7 @@
 {
 	int cmd_num;
 	for(cmd_num=0 ; cmd_num<ANDROID_WIFI_CMD_MAX; cmd_num++)
-		if(0 == strnicmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
+		if(0 == strncasecmp(cmdstr , android_wifi_cmd_str[cmd_num], strlen(android_wifi_cmd_str[cmd_num])) )
 			break;
 		
 	return cmd_num;
