diff -Naur libmpcdec-1.2/include/mpcdec/internal.h libmpcdec-1.2.new/include/mpcdec/internal.h
--- libmpcdec-1.2/include/mpcdec/internal.h	2005-05-05 23:43:34.000000000 +0200
+++ libmpcdec-1.2.new/include/mpcdec/internal.h	2005-07-03 18:19:14.000000000 +0200
@@ -46,10 +46,8 @@
 /// Big/little endian 32 bit byte swapping routine.
 static __inline
 mpc_uint32_t swap32(mpc_uint32_t val) {
-    const unsigned char* src = (const unsigned char*)&val;
-    return 
-        (mpc_uint32_t)src[0] | 
-        ((mpc_uint32_t)src[1] << 8) | ((mpc_uint32_t)src[2] << 16) | ((mpc_uint32_t)src[3] << 24);
+  return (((val & 0xff000000) >> 24) | ((val & 0x00ff0000) >>  8) |
+          ((val & 0x0000ff00) <<  8) | ((val & 0x000000ff) << 24));
 }
 
 /// Searches for a ID3v2-tag and reads the length (in bytes) of it.
diff -Naur libmpcdec-1.2/include/mpcdec/mpcdec.h libmpcdec-1.2.new/include/mpcdec/mpcdec.h
--- libmpcdec-1.2/include/mpcdec/mpcdec.h	2005-05-05 23:24:17.000000000 +0200
+++ libmpcdec-1.2.new/include/mpcdec/mpcdec.h	2005-07-03 18:21:13.000000000 +0200
@@ -100,6 +100,8 @@
 /// \return TRUE if decoder was initalized successfully, FALSE otherwise    
 mpc_bool_t mpc_decoder_initialize(mpc_decoder *d, mpc_streaminfo *si);
 
+void mpc_decoder_set_streaminfo(mpc_decoder *d, mpc_streaminfo *si);
+
 /// Sets decoder sample scaling factor.  All decoded samples will be multiplied
 /// by this factor.
 /// \param scale_factor multiplicative scaling factor
@@ -119,6 +121,12 @@
     mpc_uint32_t *vbr_update_acc, 
     mpc_uint32_t *vbr_update_bits);
 
+mpc_uint32_t mpc_decoder_decode_frame(
+    mpc_decoder *d,
+    mpc_uint32_t *in_buffer,
+    mpc_uint32_t in_len,
+    MPC_SAMPLE_FORMAT *out_buffer);
+
 /// Seeks to the specified sample in the source stream.
 mpc_bool_t mpc_decoder_seek_sample(mpc_decoder *d, mpc_int64_t destsample);
 
diff -Naur libmpcdec-1.2/src/mpc_decoder.c libmpcdec-1.2.new/src/mpc_decoder.c
--- libmpcdec-1.2/src/mpc_decoder.c	2005-05-05 23:24:24.000000000 +0200
+++ libmpcdec-1.2.new/src/mpc_decoder.c	2005-07-03 18:22:38.000000000 +0200
@@ -313,6 +313,37 @@
     memset(d->MS_Flag         , 0, sizeof d->MS_Flag          );
 }
 
+mpc_uint32_t
+mpc_decoder_decode_frame(mpc_decoder *d, mpc_uint32_t *in_buffer,
+                         mpc_uint32_t in_len, MPC_SAMPLE_FORMAT *out_buffer)
+{
+  int i;
+  mpc_decoder_reset_bitstream_decode(d);
+  if (in_len > sizeof(d->Speicher)) in_len = sizeof(d->Speicher);
+  memcpy(d->Speicher, in_buffer, in_len);
+#ifdef MPC_LITTLE_ENDIAN
+  for (i = 0; i < (in_len + 3) / 4; i++)
+    d->Speicher[i] = swap32(d->Speicher[i]);
+#endif
+  d->dword = d->Speicher[0];
+  switch (d->StreamVersion) {
+    case 0x04:
+    case 0x05:
+    case 0x06:
+        mpc_decoder_read_bitstream_sv6(d);
+        break;
+    case 0x07:
+    case 0x17:
+        mpc_decoder_read_bitstream_sv7(d);
+        break;
+    default:
+        return (mpc_uint32_t)(-1);
+  }
+  mpc_decoder_requantisierung(d, d->Max_Band);
+  mpc_decoder_synthese_filter_float(d, out_buffer);
+  return mpc_decoder_bits_read(d);
+}
+
 static mpc_uint32_t
 mpc_decoder_decode_internal(mpc_decoder *d, MPC_SAMPLE_FORMAT *buffer) 
 {
@@ -1174,7 +1205,7 @@
   mpc_decoder_init_huffman_sv7(d);
 }
 
-static void mpc_decoder_set_streaminfo(mpc_decoder *d, mpc_streaminfo *si)
+void mpc_decoder_set_streaminfo(mpc_decoder *d, mpc_streaminfo *si)
 {
     mpc_decoder_reset_synthesis(d);
     mpc_decoder_reset_globals(d);
