diff -Naur gnu-fdisk-1.0.orig/configure gnu-fdisk-1.0/configure
--- gnu-fdisk-1.0.orig/configure	2008-04-24 21:13:44.000000000 +0200
+++ gnu-fdisk-1.0/configure	2008-04-24 21:14:01.000000000 +0200
@@ -8161,7 +8161,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <ncurses.h>
+#include <ncurses/ncurses.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -8201,7 +8201,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <ncurses.h>
+#include <ncurses/ncurses.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
diff -Naur gnu-fdisk-1.0.orig/src/cfdisk.c gnu-fdisk-1.0/src/cfdisk.c
--- gnu-fdisk-1.0.orig/src/cfdisk.c	2008-04-24 21:13:44.000000000 +0200
+++ gnu-fdisk-1.0/src/cfdisk.c	2008-04-24 21:14:11.000000000 +0200
@@ -43,7 +43,7 @@
 
 
 #if HAVE_NCURSES_H
-  #include <ncurses.h>
+  #include <ncurses/ncurses.h>
 #else
   #include <curses.h>
 #endif
