diff -Naur gnu-fdisk-1.0.orig/src/cfdisk.c gnu-fdisk-1.0/src/cfdisk.c
--- gnu-fdisk-1.0.orig/src/cfdisk.c	2008-04-24 21:13:44.000000000 +0200
+++ gnu-fdisk-1.0/src/cfdisk.c	2008-05-02 17:09:23.000000000 +0200
@@ -1389,33 +1389,6 @@
 }
 
 static int
-do_mkfs (Context *c, PedPartition **part) {
-	char buf[SMALLBUF];
-	const PedFileSystemType *type = NULL;
-	int go;
-	if ((*part)->fs_type) {
-		/* If we have an fs_type, we hope we can create it */
-		if ((*part)->fs_type->ops->create != NULL) {
-			go = 1;
-			snprintf(buf,SMALLBUF,_("The partition has %s set as a filesystem. Use it?"),
-			         (*part)->fs_type->name);
-			getbool(buf,&go);
-			if (go) type = (*part)->fs_type;
-		}
-		else {
-			snprintf(buf,SMALLBUF, N_("Can't create filesystem %s, you'll have to choose another"), (*part)->fs_type->name);
-			warning_waitkey(buf);
-		}
-	}
-	if (!perform_mkfs (c->disk, *part, type, UI_WARN_COMMIT)) {
-		warning_waitkey(N_("The filesystem was not created successfully"));
-		return 0;
-	}
-	notice_waitkey(N_("The filesystem was created successfully."));
-	return 1;
-}
-
-static int
 do_check (Context *c, PedPartition **part) {
 	if (!perform_check(c->disk,*part)) {
 		warning_waitkey(N_("The consistency check failed."));
@@ -2086,7 +2059,6 @@
 	{ 'n', N_("New"), N_("Create new partition from free space") },
 	{ 'e', N_("Edit"), N_("Edit this BSD disklabel") },
 	{ 's', N_("Rescue"), N_("Look for deleted and corrupted partitions in the free space") },
-	{ 'f', N_("Make FS"), N_("Creates a filesystem on the partition") },
 	{ 'c', N_("Check"), N_("Check partition for consistency") },
 	{ 'm', N_("Rename"), N_("Change partition name") },
 	{ 'y', N_("Copy"), N_("Write another partition over this one (requires commit)") },
@@ -2151,11 +2123,6 @@
 			//redraw = 1;
 			key = 0;
 			break;
-		case 'f':
-			do_mkfs(c,part);
-			//redraw = 1;
-			key = 0;
-			break;
 		case 'c':
 			do_check(c,part);
 			//redraw = 1;
