diff -Naur gnu-fdisk-1.0.orig/src/cfdisk.c gnu-fdisk-1.0/src/cfdisk.c
--- gnu-fdisk-1.0.orig/src/cfdisk.c	2008-12-30 23:46:25.000000000 +0100
+++ gnu-fdisk-1.0/src/cfdisk.c	2008-12-30 23:46:36.000000000 +0100
@@ -1097,7 +1097,6 @@
 	static MenuItem part_type[] = {
 		{ 'p', N_("Primary"), N_("Create primary partition") },
 		{ 'e', N_("Extended"), N_("Create extended partition") },
-		{ 'b', N_("BSD disklabel"), N_("Create a new BSD disklabel") },
 		{ ESC, N_("Cancel"), N_("Do not create a partition") },
 		{ 0, NULL, NULL }
 	};
@@ -1113,7 +1112,6 @@
 	char type_keys[5];
 	type_keys[i++] = ESC;
 	type_keys[i++] = 'p'; /* We can always create a regular partition */
-	type_keys[i++] = 'b'; /* The BSD disklabel can be counted as a regular partition */
 	PedPartitionType type = 0;
 	PartPos pos;
 	int key,success = 1,done;
@@ -1139,15 +1137,6 @@
 			if (key == ESC)	return 1;
 			else if (key == 'p') type = PED_PARTITION_NORMAL;
 			else if (key == 'e') type = PED_PARTITION_EXTENDED;
-		else if (key == 'b') {
-			type = PED_PARTITION_NORMAL;
-			key = do_menu(bsd_type, 8, MENU_BUTTON | MENU_TITLE, bsd_keys, NULL);
-			if (key == ESC) return 1;
-			else if (key == 'f') bsd_disklabel = 0xa5;
-			else if (key == 'o') bsd_disklabel = 0xa6;
-			else if (key == 'n') bsd_disklabel = 0xa9;
-		}
-
 	}
 
 	pos.start.sector = (*part)->geom.start;
