diff -Naur gnu-fdisk-1.0.orig/src/cfdisk.c gnu-fdisk-1.0/src/cfdisk.c
--- gnu-fdisk-1.0.orig/src/cfdisk.c	2008-12-30 23:46:25.000000000 +0100
+++ gnu-fdisk-1.0/src/cfdisk.c	2008-12-30 23:46:36.000000000 +0100
@@ -2369,11 +2358,6 @@
 		{ 'q', N_("Quit"), N_("Quit program") },
 		{ 0, NULL, NULL }
 	};
-	static MenuItem custom_label[] = {
-		{ 'c', N_("Create"), N_("Create new partition table") },
-		{ 'q', N_("Quit"), N_("Quit program") },
-		{ 0, NULL, NULL }
-	};
 	/* If we can't find the device automatically. FIXME: This is not tested. */
 	if (!dev) {
 		menu_title(_("Cannot find a device automatically"));
@@ -2403,12 +2387,8 @@
 		if (!disk) {
 			ped_exception_catch();
 			ped_exception_leave_all();
-			menu_title(_("Cannot find a partition table on the disk"));
-			if (type || 'c' == do_menu(custom_label, 8, MENU_BUTTON | MENU_TITLE, "cq", 0)) {
-				if (!perform_mklabel (dev, &disk, type))
-					do_quit(1,_("Creation of partition table failed"));
-			}
-			else do_quit (0,NULL);
+                        if (!perform_mklabel (dev, &disk, type))
+                          do_quit(1,_("Creation of partition table failed"));
 		} else {
 			ped_exception_leave_all();
 		}
