diff -Naur gnu-fdisk-1.0.orig/src/common.c gnu-fdisk-1.0/src/common.c
--- gnu-fdisk-1.0.orig/src/common.c	2008-12-30 23:46:25.000000000 +0100
+++ gnu-fdisk-1.0/src/common.c	2008-12-30 23:46:41.000000000 +0100
@@ -1245,10 +1249,8 @@
                 }
                 ped_disk_destroy (*disk);
         }
-	if (!type) {
-        	if (!get_disk_type (_("New disk label type"), &type))
-                	goto error;
-	}
+	if (!type)
+		type = ped_disk_type_get ("msdos");
 
         *disk = ped_disk_new_fresh (dev, type);
         if (!*disk)
