--- a/src/pcm/FormatConverter.cxx	2017-04-13 10:47:46.814830355 +0200
+++ b/src/pcm/FormatConverter.cxx	2017-04-13 10:47:46.830830154 +0200
@@ -44,7 +44,7 @@
 			     "PCM conversion from %s to %s is not implemented",
 			     sample_format_to_string(_src_format),
 			     sample_format_to_string(_dest_format));
-		return nullptr;
+		return false;
 
 	case SampleFormat::S16:
 	case SampleFormat::S24_P32:
--- a/src/storage/plugins/NfsStorage.cxx	2017-04-13 11:02:50.775466034 +0200
+++ b/src/storage/plugins/NfsStorage.cxx	2017-04-13 11:02:50.795465780 +0200
@@ -287,7 +287,7 @@
 		return false;
 
 	if (!WaitConnected(error))
-		return nullptr;
+		return false;
 
 	NfsGetInfoOperation operation(*connection, path.c_str(), info);
 	return operation.Run(error);
--- a/src/db/plugins/simple/SimpleDatabasePlugin.cxx	2017-04-13 11:05:28.329485310 +0200
+++ b/src/db/plugins/simple/SimpleDatabasePlugin.cxx	2017-04-13 11:05:28.349485057 +0200
@@ -445,13 +445,13 @@
 	if (r.uri == nullptr) {
 		error.Format(db_domain, DB_CONFLICT,
 			     "Already exists: %s", uri);
-		return nullptr;
+		return false;
 	}
 
 	if (strchr(r.uri, '/') != nullptr) {
 		error.Format(db_domain, DB_NOT_FOUND,
 			     "Parent not found: %s", uri);
-		return nullptr;
+		return false;
 	}
 
 	Directory *mnt = r.directory->CreateChild(r.uri);
@@ -478,7 +478,7 @@
 	if (cache_path.IsNull()) {
 		error.Format(db_domain, DB_NOT_FOUND,
 			     "No 'cache_directory' configured");
-		return nullptr;
+		return false;
 	}
 
 	std::string name(storage_uri);
--- a/src/db/plugins/ProxyDatabasePlugin.cxx	2017-04-21 16:14:30.180294958 +0200
+++ b/src/db/plugins/ProxyDatabasePlugin.cxx	2017-04-21 16:14:30.208294608 +0200
@@ -731,7 +731,7 @@
 {
 	// TODO: eliminate the const_cast
 	if (!const_cast<ProxyDatabase *>(this)->EnsureConnected(error))
-		return nullptr;
+		return false;
 
 	if (!visit_directory && !visit_playlist && selection.recursive &&
 	    (ServerSupportsSearchBase(connection)
@@ -757,7 +757,7 @@
 {
 	// TODO: eliminate the const_cast
 	if (!const_cast<ProxyDatabase *>(this)->EnsureConnected(error))
-		return nullptr;
+		return false;
 
 	enum mpd_tag_type tag_type2 = Convert(tag_type);
 	if (tag_type2 == MPD_TAG_COUNT) {
@@ -810,7 +810,7 @@
 
 	// TODO: eliminate the const_cast
 	if (!const_cast<ProxyDatabase *>(this)->EnsureConnected(error))
-		return nullptr;
+		return false;
 
 	struct mpd_stats *stats2 =
 		mpd_run_stats(connection);
