--- a/CMakeLists.txt	2017-06-15 20:16:22.938356239 +0200
+++ b/CMakeLists.txt	2017-06-15 20:16:22.910356510 +0200
@@ -18,7 +18,7 @@
 endif()
 externalproject_add(libgoom SOURCE_DIR ${PROJECT_SOURCE_DIR}/lib/goom2k4-0
                     "${update_command}"
-                    CONFIGURE_COMMAND ${PROJECT_SOURCE_DIR}/lib/goom2k4-0/configure --disable-shared --enable-static --with-pic --prefix=<INSTALL_DIR>
+                    CONFIGURE_COMMAND ${PROJECT_SOURCE_DIR}/lib/goom2k4-0/configure ${GOOM_CROSSCOMPILE} --disable-shared --enable-static --with-pic --prefix=<INSTALL_DIR>
                     BUILD_COMMAND ${MAKE}
                     INSTALL_COMMAND "")
 
