--- vdr-noad-0.8.5/audiotools.cpp.org	2013-11-25 16:12:23.613180633 +0100
+++ vdr-noad-0.8.5/audiotools.cpp	2013-11-25 16:13:28.145181743 +0100
@@ -1,4 +1,6 @@
 #include "audiotools.h"
+#define MAX_AUDIO_FRAME_SIZE 192000
+#define AVCODEC_MAX_AUDIO_FRAME_SIZE 192000
 
 bool havesilence = false;
 #ifdef HAVE_LIBAVCODEC
@@ -102,12 +104,12 @@
 #endif
 
   /* open it */
-  if (avcodec_open(codecContext, codec) < 0)
+  if (avcodec_open2(codecContext, codec, NULL) < 0)
   {
     fprintf(stderr, "could not open codec\n");
   }
 
-  outbuf = (uint8_t *)malloc(AVCODEC_MAX_AUDIO_FRAME_SIZE);
+  outbuf = (uint8_t *)malloc(MAX_AUDIO_FRAME_SIZE);
 
   av_codec_initialised = true;
         
--- vdr-noad-0.8.5/ffmpeg_decoder.cpp.org	2013-11-25 16:13:36.953181893 +0100
+++ vdr-noad-0.8.5/ffmpeg_decoder.cpp	2013-11-25 16:14:05.181182378 +0100
@@ -223,7 +223,7 @@
 		pCodecCtx->flags|=CODEC_FLAG_TRUNCATED;
 
 	// Open codec
-	if(avcodec_open(pCodecCtx, pCodec)<0)
+        if(avcodec_open2(pCodecCtx, pCodec, NULL)<0)
 		return -1; // Could not open codec
 
 
