diff -Naur before/configure.ac after/configure.ac
--- before/configure.ac	2010-11-12 08:53:36.000000000 -0500
+++ after/configure.ac	2012-10-18 15:48:46.000000000 -0400
@@ -45,7 +45,7 @@
 PKG_CHECK_MODULES([SYSLINK], [syslink])
 
 dnl Check for tiler memmgr
-PKG_CHECK_MODULES([MEMMGR], [libmemmgr])
+PKG_CHECK_MODULES([MEMMGR], [libtimemmgr])
 
 dnl Check for d2cmap
 PKG_CHECK_MODULES([D2CMAP], [libd2cmap])
