diff -urN vdr-graphlcd-0.1.9.orf//state.c vdr-graphlcd-0.1.9/state.c
--- vdr-graphlcd-0.1.9.orf//state.c	2012-03-18 10:00:24.265919180 +0100
+++ vdr-graphlcd-0.1.9/state.c	2012-03-18 10:05:41.025919130 +0100
@@ -74,12 +74,16 @@
 {
 }
 
+#if VDRVERSNUM >= 10726
+void cGraphLCDState::ChannelSwitch(const cDevice * Device, int ChannelNumber, bool LiveView) {
+#else
 void cGraphLCDState::ChannelSwitch(const cDevice * Device, int ChannelNumber)
-{
     //printf("graphlcd plugin: cGraphLCDState::ChannelSwitch %d %d\n", Device->CardIndex(), ChannelNumber);
+    bool LiveView = Device && Device->IsPrimaryDevice() && !EITScanner.UsesDevice(Device)
+#endif
     if (GraphLCDSetup.PluginActive)
     {
-        if (ChannelNumber > 0 && Device->IsPrimaryDevice() && !EITScanner.UsesDevice(Device))
+        if (ChannelNumber > 0 && LiveView)
         {
             if (ChannelNumber == cDevice::CurrentChannel())
             {
diff -urN vdr-graphlcd-0.1.9.orf//state.h vdr-graphlcd-0.1.9/state.h
--- vdr-graphlcd-0.1.9.orf//state.h	2012-03-18 10:00:24.145919180 +0100
+++ vdr-graphlcd-0.1.9/state.h	2012-03-18 10:07:20.635919114 +0100
@@ -100,7 +100,11 @@
     void SetChannel(int ChannelNumber);
     void GetProgramme();
 protected:
+#if VDRVERSNUM >= 10726
+    virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber, bool LiveView);
+#else
     virtual void ChannelSwitch(const cDevice *Device, int ChannelNumber);
+#endif
 #if VDRVERSNUM < 10338
     virtual void Recording(const cDevice *Device, const char *Name);
     virtual void Replaying(const cControl *Control, const char *Name);
