--- s2-liplianin-v39+r2c75558/linux/drivers/media/i2c/Makefile.old	2014-10-31 12:14:11.076436622 +0100
+++ s2-liplianin-v39+r2c75558/linux/drivers/media/i2c/Makefile	2014-10-31 12:14:21.444436620 +0100
@@ -64,7 +64,6 @@
 obj-$(CONFIG_VIDEO_NOON010PC30)	+= noon010pc30.o
 obj-$(CONFIG_VIDEO_S5K6AA)	+= s5k6aa.o
 obj-$(CONFIG_VIDEO_S5K4ECGX)	+= s5k4ecgx.o
-obj-$(CONFIG_VIDEO_S5C73M3)	+= s5c73m3/
 obj-$(CONFIG_VIDEO_ADP1653)	+= adp1653.o
 obj-$(CONFIG_VIDEO_AS3645A)	+= as3645a.o
 obj-$(CONFIG_VIDEO_SMIAPP_PLL)	+= smiapp-pll.o
