diff -uNr s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/bt8xx/bttv-i2c.c s2-liplianin-v39+r2c75558/linux/drivers/media/pci/bt8xx/bttv-i2c.c
--- s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/bt8xx/bttv-i2c.c	2013-07-02 02:13:31.000000000 +0800
+++ s2-liplianin-v39+r2c75558/linux/drivers/media/pci/bt8xx/bttv-i2c.c	2013-10-09 22:10:45.320562923 +0800
@@ -400,5 +401,5 @@
 	if (0 != btv->i2c_rc)
 		return 0;
 
-	return i2c_del_adapter(&btv->c.i2c_adap);
+	i2c_del_adapter(&btv->c.i2c_adap);
 }
diff -uNr s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/ttpci/av7110_ir.c s2-liplianin-v39+r2c75558/linux/drivers/media/pci/ttpci/av7110_ir.c
--- s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/ttpci/av7110_ir.c	2013-07-02 02:13:31.000000000 +0800
+++ s2-liplianin-v39+r2c75558/linux/drivers/media/pci/ttpci/av7110_ir.c	2013-10-09 22:11:04.368563418 +0800
@@ -375,7 +375,8 @@
 	if (av_cnt == 1) {
 		e = proc_create("av7110_ir", S_IWUSR, NULL, &av7110_ir_proc_fops);
 		if (e)
-			e->size = 4 + 256 * sizeof(u16);
+//			e->size = 4 + 256 * sizeof(u16);
+			proc_set_size(e, 4 + 256 * sizeof(u16));
 	}
 
 	tasklet_init(&av7110->ir.ir_tasklet, av7110_emit_key, (unsigned long) &av7110->ir);
diff -uNr s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/zoran/zoran_procfs.c s2-liplianin-v39+r2c75558/linux/drivers/media/pci/zoran/zoran_procfs.c
--- s2-liplianin-v39+r2c75558.orig/linux/drivers/media/pci/zoran/zoran_procfs.c	2013-07-02 02:13:31.000000000 +0800
+++ s2-liplianin-v39+r2c75558/linux/drivers/media/pci/zoran/zoran_procfs.c	2013-10-09 22:11:23.352563910 +0800
@@ -130,14 +130,16 @@
 
 static int zoran_open(struct inode *inode, struct file *file)
 {
-	struct zoran *data = PDE(inode)->data;
+//	struct zoran *data = PDE(inode)->data;
+	struct zoran *data = PDE_DATA(inode);
 	return single_open(file, zoran_show, data);
 }
 
 static ssize_t zoran_write(struct file *file, const char __user *buffer,
 			size_t count, loff_t *ppos)
 {
-	struct zoran *zr = PDE(file_inode(file))->data;
+//	struct zoran *zr = PDE(file_inode(file))->data;
+	struct zoran *zr = PDE_DATA(file_inode(file));
 	char *string, *sp;
 	char *line, *ldelim, *varname, *svar, *tdelim;
 
@@ -201,7 +203,8 @@
 		dprintk(2,
 			KERN_INFO
 			"%s: procfs entry /proc/%s allocated. data=%p\n",
-			ZR_DEVNAME(zr), name, zr->zoran_proc->data);
+//			ZR_DEVNAME(zr), name, zr->zoran_proc->data);
+			ZR_DEVNAME(zr), name, zr);
 	} else {
 		dprintk(1, KERN_ERR "%s: Unable to initialise /proc/%s\n",
 			ZR_DEVNAME(zr), name);
