From adbaee209dd5c106465a3e87695096313d6d80fb Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Thu, 24 Jul 2014 00:58:07 +0200
Subject: [PATCH] Modify eglwayland versions for Vivante GPUs

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 ozone/wayland/wayland.gyp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/ozone/wayland/wayland.gyp b/ozone/wayland/wayland.gyp
index 062573e..8474546 100644
--- a/ozone/wayland/wayland.gyp
+++ b/ozone/wayland/wayland.gyp
@@ -26,12 +26,13 @@
       'type': 'static_library',
       'variables': {
         'WAYLAND_VERSION': '1.4.0',
-        'MESA_VERSION': '9.1.3',
+        'WAYLAND_EGL_VERSION': '1.0.0',
+        'VIVEGL_VERSION': '8.0',
         'wayland_packages': [
-          'egl >= <(MESA_VERSION)',
+          'egl >= <(VIVEGL_VERSION)',
           'wayland-client >= <(WAYLAND_VERSION)',
           'wayland-cursor >= <(WAYLAND_VERSION)',
-          'wayland-egl >= <(MESA_VERSION)',
+          'wayland-egl >= <(WAYLAND_EGL_VERSION)',
           'xkbcommon',
         ],
       },
-- 
1.8.3.2

