diff -Naur eina-r48546.orig/src/lib/eina_mempool.c eina-r48546/src/lib/eina_mempool.c
--- eina-r48546.orig/src/lib/eina_mempool.c	2010-05-22 18:38:09.877491458 +0200
+++ eina-r48546/src/lib/eina_mempool.c	2010-05-22 18:38:34.187490683 +0200
@@ -170,7 +170,7 @@
    _backends = eina_hash_string_superfast_new(NULL);
 
    /* dynamic backends */
-   _modules = eina_module_arch_list_get(NULL, PACKAGE_LIB_DIR "/eina/modules/mp", MODULE_ARCH);
+   _modules = eina_module_arch_list_get(NULL, "/usr/lib/eina/modules/mp", MODULE_ARCH);
 
    path = eina_module_environment_path_get("HOME", "/.eina/mp/modules/mp");
    _modules = eina_module_arch_list_get(_modules, path, MODULE_ARCH);
