--- glibc/elf/dl-load.c	2013-01-23 19:29:03.412797023 -0700
+++ glibc/elf/dl-load.c	2013-01-23 19:30:02.556794931 -0700
@@ -1999,6 +2000,15 @@
 	      *found_other_class = true;
 	      goto close_and_out;
 	    }
+#ifdef __arm__
+          else if (!VALID_FLOAT_ABI (ehdr->e_flags))
+	    {
+	      /* This is not a fatal error.  On architectures where
+		 soft-float and hard-float binaries can be run this
+		 might happen.  */
+	      goto close_and_out;
+	    }
+#endif
 	  else if (ehdr->e_ident[EI_DATA] != byteorder)
 	    {
 	      if (BYTE_ORDER == BIG_ENDIAN)
